From 2c1ba636c623e579444230c4f5e8f23974464126 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 30 Jun 2015 20:04:08 -0600
Subject: [PATCH] fix includes

Signed-off-by: Kevin Mihelich <kevin@archlinuxarm.org>
---
 bluetooth/brcm_patchram_plus.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/bluetooth/brcm_patchram_plus.c b/bluetooth/brcm_patchram_plus.c
index b4fa5a8..f31d8ef 100644
--- a/bluetooth/brcm_patchram_plus.c
+++ b/bluetooth/brcm_patchram_plus.c
@@ -112,6 +112,7 @@
 #include <fcntl.h>
 
 #include <stdlib.h>
+#include <unistd.h>
 
 #ifdef ANDROID
 #include <termios.h>
-- 
2.4.4

