--- configure.in	2002-07-13 01:31:52.000000000 +0000
+++ configure.in.patched	2003-09-10 14:37:31.000000000 +0000
@@ -319,13 +319,13 @@
 [AC_TRY_RUN([#include <stdio.h>
 int test() {
 #if defined(__GNUC__)
-  __asm__ __volatile__ ("
-  .align 4
-mfx_a1:
-  .byte 0
-  .align 4
-mfx_a2:
-  ");
+  __asm__ __volatile__ (
+  ".align 4\n"
+"mfx_a1:\n"
+  ".byte 0\n"
+  ".align 4\n"
+"mfx_a2:\n"
+  );
 #endif
   return 0;
 }
@@ -363,11 +363,11 @@
 #endif
   int mfx_asm_func();
   static void test() {
-    __asm__ __volatile__("
-    .globl mfx_asm_func
-    mfx_asm_func:
-      .byte 0
-    ");
+    __asm__ __volatile__(
+    ".globl mfx_asm_func\n"
+    "mfx_asm_func:\n"
+      ".byte 0\n"
+    );
   }
 ], [
   return mfx_asm_func();
