From 72f50a2d29648248e03eb871f896f0c940e44c78 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 5 Jul 2016 18:47:56 -0600
Subject: [PATCH] optionally enable OpenGL

Signed-off-by: Kevin Mihelich <kevin@archlinuxarm.org>
---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 47f6b10..7615b7f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -104,6 +104,7 @@ option(ENABLE_KFILEMETADATASUPPORT  "Build digiKam with KDE files indexer suppor
 option(ENABLE_AKONADICONTACTSUPPORT "Build digiKam with KDE Mail Contacts support (default=OFF)"                       OFF)
 option(ENABLE_MEDIAPLAYER           "Build digiKam with QtMultimedia support (default=OFF)"                            OFF)
 option(ENABLE_DBUS                  "Build digiKam with DBUS support (default=ON)"                                     ON)
+option(ENABLE_OPENGL                "Build digiKam with OpenGL support (default=ON)"                                   ON)
 
 # Mysql support options (experimental):
 option(ENABLE_MYSQLSUPPORT          "Build digiKam with MySQL dabatase support (default=OFF)"                          OFF)
@@ -284,7 +285,7 @@ else()
 endif()
 
 # decide if Presentation tool can be built with OpenGL
-if(OPENGL_FOUND AND OPENGL_GLU_FOUND AND Qt5OpenGL_FOUND)
+if(OPENGL_FOUND AND OPENGL_GLU_FOUND AND Qt5OpenGL_FOUND AND ENABLE_OPENGL)
     set(HAVE_OPENGL TRUE)
 elseif()
     set(HAVE_OPENGL FALSE)
-- 
2.8.3

