--- digikam-2.5.0/core/digikam/album/albummanager.h	2012-01-03 02:32:41.000000000 +0200
+++ digikam-2.5.0/core/digikam/album/albummanager.h	2012-01-05 03:18:16.291639995 +0200
@@ -831,7 +831,7 @@
     {
         foreach(T* t, list)
         {
-            append(AlbumPointer<T>(t));
+            this->append(AlbumPointer<T>(t));
         }
         return *this;
     }
--- digikam-2.5.0/core/libs/database/imagehistory/imagehistorygraph_boost.h	2012-01-03 02:32:26.000000000 +0200
+++ digikam-2.5.0/core/libs/database/imagehistory/imagehistorygraph_boost.h	2012-01-05 03:01:56.965609602 +0200
@@ -1362,7 +1362,7 @@
             template <typename VertexType, typename GraphType>
             void discover_vertex(VertexType u, const GraphType&) const
             {
-                record(u);
+                this->record(u);
             }
         };
 
@@ -1373,7 +1373,7 @@
             template <typename VertexType, typename GraphType>
             void discover_vertex(VertexType u, const GraphType&) const
             {
-                record(u);
+                this->record(u);
             }
         };
 
--- digikam-2.5.0/core/utilities/cameragui/devices/gpcamera.cpp	2012-01-03 02:32:04.000000000 +0200
+++ digikam-2.5.0/core/utilities/cameragui/devices/gpcamera.cpp	2012-01-05 01:22:34.009352997 +0200
@@ -30,6 +30,7 @@
 extern "C"
 {
 #include <utime.h>
+#include <unistd.h>
 }
 
 // C++ includes
--- digikam-2.5.0/extra/kipi-plugins/htmlexport/wizard.cpp	2012-01-03 02:31:42.000000000 +0200
+++ digikam-2.5.0/extra/kipi-plugins/htmlexport/wizard.cpp	2012-01-05 02:35:01.061054821 +0200
@@ -78,7 +78,7 @@
 public:
     WizardPage(KAssistantDialog* dialog, const QString& title)
     : QWidget(dialog) {
-        setupUi(this);
+        this->setupUi(this);
         layout()->setMargin(0);
         mPage = dialog->addPage(this, title);
     }
--- digikam-2.5.0/extra/kipi-plugins/printimages/wizard/wizard.cpp	2012-01-03 02:31:34.000000000 +0200
+++ digikam-2.5.0/extra/kipi-plugins/printimages/wizard/wizard.cpp	2012-01-05 02:17:15.249914790 +0200
@@ -88,7 +88,7 @@
     WizardPage ( KAssistantDialog* dialog, const QString& title )
         : QWidget ( dialog )
     {
-        setupUi ( this );
+        this->setupUi ( this );
         layout()->setMargin ( 0 );
         mPage = dialog->addPage ( this, title );
     }
