--- digikam-5.1.0/core/libs/rawengine/libraw/libraw/libraw_types.h.orig	2016-08-09 15:05:10.449224741 +0000
+++ digikam-5.1.0/core/libs/rawengine/libraw/libraw/libraw_types.h	2016-08-09 15:05:50.725789944 +0000
@@ -618,10 +618,10 @@
 #elif defined(_M_X64) || defined(__amd64__)
 #define LibRawBigEndian 0
 
-#elif defined(__LITTLE_ENDIAN__)
+#elif __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
 #define LibRawBigEndian 0
 
-#elif defined(__BIG_ENDIAN__)
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
 #define LibRawBigEndian 1
 #elif defined(_ARM_)
 #define LibRawBigEndian 0
