--- digikam-5.4.0/core/cmake/modules/FindQtAV.cmake.0	2017-01-09 22:56:43.303991660 +0000
+++ digikam-5.4.0/core/cmake/modules/FindQtAV.cmake	2017-01-09 22:57:24.843934106 +0000
@@ -24,15 +24,17 @@
               HINTS ${_qt5_install_prefix}                                           # For MXE
                     ${_qt5_install_prefix}/../qt5/include                            # For Mageia
                     ${_qt5_install_prefix}/../../include/qt5                         # For Suse
+                    ${_qt5_install_prefix}/../../include/qt
                     ${_qt5_install_prefix}/../../../include/x86_64-linux-gnu/qt5     # For Debian
               PATH_SUFFIXES QtAV
     )
 
     find_path(QTAV_WIDGETS_INCLUDE_DIR
               NAMES QtAVWidgets.h
               HINTS ${_qt5_install_prefix}                                           # For MXE
                     ${_qt5_install_prefix}/../qt5/include                            # For Mageia
                     ${_qt5_install_prefix}/../../include/qt5                         # For Suse
+                    ${_qt5_install_prefix}/../../include/qt
                     ${_qt5_install_prefix}/../../../include/x86_64-linux-gnu/qt5     # For Debian
               PATH_SUFFIXES QtAVWidgets
     )
 
