diff -wbBur mysql-connector-odbc-5.1.5r1144/driver/myutil.h mysql-connector-odbc-5.1.5r1144.my/driver/myutil.h
--- mysql-connector-odbc-5.1.5r1144/driver/myutil.h	2008-04-15 05:03:13.000000000 +0400
+++ mysql-connector-odbc-5.1.5r1144.my/driver/myutil.h	2009-09-23 19:52:45.000000000 +0400
@@ -204,7 +204,7 @@
 				      SQLLEN *pcbValue);
 SQLRETURN SQL_API my_SQLExtendedFetch(SQLHSTMT hstmt, SQLUSMALLINT fFetchType,
 				      SQLROWOFFSET irow, SQLULEN *pcrow,
-				      SQLUSMALLINT FAR *rgfRowStatus, bool upd_status);
+				      SQLUSMALLINT FAR *rgfRowStatus, my_bool upd_status);
 SQLRETURN copy_stmt_error(STMT FAR *src, STMT FAR *dst);
 int unireg_to_c_datatype(MYSQL_FIELD *field);
 int default_c_type(int sql_data_type);
diff -wbBur mysql-connector-odbc-5.1.5r1144/driver/results.c mysql-connector-odbc-5.1.5r1144.my/driver/results.c
--- mysql-connector-odbc-5.1.5r1144/driver/results.c	2008-06-06 23:37:04.000000000 +0400
+++ mysql-connector-odbc-5.1.5r1144.my/driver/results.c	2009-09-23 19:52:39.000000000 +0400
@@ -1184,7 +1184,7 @@
                                        SQLROWOFFSET         irow,
                                        SQLULEN             *pcrow,
                                        SQLUSMALLINT FAR    *rgfRowStatus,
-                                       bool                 upd_status )
+                                       my_bool                 upd_status )
 {
     ulong rows_to_fetch;
     long cur_row, max_row;
