--- a/src/prefsdlg.cpp	2007-09-12 23:37:46.000000000 -0500
+++ b/src/prefsdlg.cpp	2007-09-12 23:38:54.000000000 -0500
@@ -726,7 +726,7 @@
 	GtkWidget *vbox1 = gtk_vbox_new(false, 6);
 	gtk_box_pack_start(GTK_BOX(vbox),vbox1,false,false, 0);
 
-	GtkWidget *check_button;
+	GtkWidget *check_button, *label;
 	check_button = gtk_check_button_new_with_mnemonic(_("_Enable sound event."));
 	bool enable=
 		conf->get_bool_at("dictionary/enable_sound_event");
@@ -748,7 +748,7 @@
 	gtk_box_pack_start(GTK_BOX(vbox1), hbox2, FALSE, FALSE, 0);
 #endif
 
-	GtkWidget *label = gtk_label_new(_("RealPeopleTTS search path:"));
+	label = gtk_label_new(_("RealPeopleTTS search path:"));
 	gtk_misc_set_alignment(GTK_MISC(label), 0, .5);
 	gtk_box_pack_start(GTK_BOX(vbox1),label,false,false,0);
 	tts_textview = gtk_text_view_new();
--- a/configure.in	2007-09-12 23:52:01.000000000 -0500
+++ b/configure.in	2007-09-12 23:52:33.000000000 -0500
@@ -192,7 +192,7 @@
 elif test "x${enable_maemo_support}" = "xyes" ; then
   DEP_MODULES="gtk+-2.0 >= 2.6 gconf-2.0 >= 2.6 hildon-libs >= 0.12"
 elif test "x${enable_gnome_support}" = "xno" ; then
-  DEP_MODULES="gtk+-2.0 >= 2.6"
+  DEP_MODULES="gtk+-2.0 >= 2.6 gthread-2.0"
 else
   DEP_MODULES="libgnomeui-2.0 >= 2.2.0"
 fi
