diff -Naur stardict-3.0.1.orig/src/gconf_file.cpp stardict-3.0.1/src/gconf_file.cpp
--- stardict-3.0.1.orig/src/gconf_file.cpp	2007-07-10 08:16:04.000000000 +0100
+++ stardict-3.0.1/src/gconf_file.cpp	2008-04-11 18:18:43.000000000 +0100
@@ -22,6 +22,8 @@
 #  include "config.h"
 #endif
 
+#include <memory>
+
 #include <glib/gi18n.h>
 
 #include "gconf_file.hpp"
diff -Naur stardict-3.0.1.orig/src/inifile.cpp stardict-3.0.1/src/inifile.cpp
--- stardict-3.0.1.orig/src/inifile.cpp	2007-08-01 10:43:00.000000000 +0100
+++ stardict-3.0.1/src/inifile.cpp	2008-04-11 18:18:43.000000000 +0100
@@ -25,6 +25,7 @@
 #include <vector>
 #include <cerrno>
 #include <cstring>
+#include <cstdlib>
 #include <cstdio>
 #include <glib/gi18n.h>
 #include <glib/gstdio.h>
diff -Naur stardict-3.0.1.orig/src/pangoview.cpp stardict-3.0.1/src/pangoview.cpp
--- stardict-3.0.1.orig/src/pangoview.cpp	2007-09-25 08:11:48.000000000 +0100
+++ stardict-3.0.1/src/pangoview.cpp	2008-04-11 18:18:43.000000000 +0100
@@ -22,6 +22,8 @@
 #  include "config.h"
 #endif
 
+#include <cstring>
+
 #include "gtktextviewpango.h"
 #include "utils.h"
 #include "skin.h"//for SkinCursor definition
diff -Naur stardict-3.0.1.orig/src/prefsdlg.cpp stardict-3.0.1/src/prefsdlg.cpp
--- stardict-3.0.1.orig/src/prefsdlg.cpp	2007-10-30 08:14:07.000000000 +0000
+++ stardict-3.0.1/src/prefsdlg.cpp	2008-04-11 18:18:43.000000000 +0100
@@ -21,6 +21,8 @@
 #  include "config.h"
 #endif
 
+#include <cstdlib>
+
 #include <glib/gi18n.h>
 #include <glib/gstdio.h>
 
diff -Naur stardict-3.0.1.orig/src/readword.cpp stardict-3.0.1/src/readword.cpp
--- stardict-3.0.1.orig/src/readword.cpp	2007-07-10 08:16:04.000000000 +0100
+++ stardict-3.0.1/src/readword.cpp	2008-04-11 17:53:43.000000000 +0100
@@ -3,6 +3,7 @@
 #endif
 
 #include <cstring>
+#include <cstdlib>
 #include <string>
 
 #include <glib/gi18n.h>
diff -Naur stardict-3.0.1.orig/src/sigc++/signal.h stardict-3.0.1/src/sigc++/signal.h
--- stardict-3.0.1.orig/src/sigc++/signal.h	2007-07-10 08:16:01.000000000 +0100
+++ stardict-3.0.1/src/sigc++/signal.h	2008-04-11 17:50:11.000000000 +0100
@@ -18,7 +18,7 @@
   //Compilers, such as older versions of SUN Forte C++, that do not allow this also often
   //do not allow a typedef to have the same name as a class in the typedef's definition.
   //For Sun Forte CC 5.7 (SUN Workshop 10), comment this out to fix the build.
-  #define SIGC_TYPEDEF_REDEFINE_ALLOWED 1
+//  #define SIGC_TYPEDEF_REDEFINE_ALLOWED 1
 #endif
 
 namespace sigc {
diff -Naur stardict-3.0.1.orig/src/utils.cpp stardict-3.0.1/src/utils.cpp
--- stardict-3.0.1.orig/src/utils.cpp	2007-10-22 03:25:02.000000000 +0100
+++ stardict-3.0.1/src/utils.cpp	2008-04-11 18:18:43.000000000 +0100
@@ -22,6 +22,8 @@
 #  include "config.h"
 #endif
 
+#include <cstring>
+
 #include <glib.h>
 #include <glib/gi18n.h>
 #include <cstdlib>
diff -Naur stardict-3.0.1.orig/src/x11_iskeyspressed.hpp stardict-3.0.1/src/x11_iskeyspressed.hpp
--- stardict-3.0.1.orig/src/x11_iskeyspressed.hpp	2007-07-10 08:16:04.000000000 +0100
+++ stardict-3.0.1/src/x11_iskeyspressed.hpp	2008-04-11 17:57:08.000000000 +0100
@@ -1,6 +1,8 @@
 #ifndef _X11_ISKEYSPRESSED_HPP_
 #define _X11_ISKEYSPRESSED_HPP_
 
+#include <memory>
+
 #include <gdk/gdkx.h>
 #include <X11/keysym.h>
 #include <gtk/gtk.h>

 #ifdef _WIN32
diff -Naur stardict-3.0.1.orig/stardict-plugins/stardict-espeak-tts-plugin/stardict_espeak.cpp stardict-3.0.1/stardict-plugins/stardict-espeak-tts-plugin/stardict_espeak.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-espeak-tts-plugin/stardict_espeak.cpp	2007-09-19 09:27:18.000000000 +0100
+++ stardict-3.0.1/stardict-plugins/stardict-espeak-tts-plugin/stardict_espeak.cpp	2008-04-11 18:02:55.000000000 +0100
@@ -1,4 +1,5 @@
 #include "stardict_espeak.h"
+#include <cstring>
 #include <espeak/speak_lib.h>
 #include <glib/gi18n.h>
 
diff -Naur stardict-3.0.1.orig/stardict-plugins/stardict-html-parsedata-plugin/stardict_html_parsedata.cpp stardict-3.0.1/stardict-plugins/stardict-html-parsedata-plugin/stardict_html_parsedata.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-html-parsedata-plugin/stardict_html_parsedata.cpp	2007-09-13 08:51:55.000000000 +0100
+++ stardict-3.0.1/stardict-plugins/stardict-html-parsedata-plugin/stardict_html_parsedata.cpp	2008-04-11 17:59:50.000000000 +0100
@@ -1,4 +1,6 @@
 #include "stardict_html_parsedata.h"
+#include <cstring>
+#include <cstdlib>
 #include <glib/gi18n.h>
 
 #ifdef _WIN32

diff -Naur stardict-3.0.1.orig/stardict-plugins/stardict-spell-plugin/stardict_spell.cpp stardict-3.0.1/stardict-plugins/stardict-spell-plugin/stardict_spell.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-spell-plugin/stardict_spell.cpp	2007-09-19 09:29:21.000000000 +0100
+++ stardict-3.0.1/stardict-plugins/stardict-spell-plugin/stardict_spell.cpp	2008-04-11 18:02:38.000000000 +0100
@@ -1,4 +1,5 @@
 #include "stardict_spell.h"
+#include <cstring>
 #include <glib.h>
 #include <glib/gi18n.h>
 #include <enchant.h>

diff -Naur stardict-3.0.1.orig/stardict-plugins/stardict-wordnet-plugin/court_widget.cpp stardict-3.0.1/stardict-plugins/stardict-wordnet-plugin/court_widget.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-wordnet-plugin/court_widget.cpp	2007-10-18 02:36:22.000000000 +0100
+++ stardict-3.0.1/stardict-plugins/stardict-wordnet-plugin/court_widget.cpp	2008-04-11 18:01:10.000000000 +0100
@@ -1,4 +1,5 @@
 #include "court_widget.h"
+#include <cstring>
 #include <math.h>
 #include <list>
 
diff -Naur stardict-3.0.1.orig/stardict-plugins/stardict-wordnet-plugin/stardict_wordnet.cpp stardict-3.0.1/stardict-plugins/stardict-wordnet-plugin/stardict_wordnet.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-wordnet-plugin/stardict_wordnet.cpp	2007-10-15 04:32:04.000000000 +0100
+++ stardict-3.0.1/stardict-plugins/stardict-wordnet-plugin/stardict_wordnet.cpp	2008-04-11 18:00:35.000000000 +0100
@@ -1,5 +1,6 @@
 #include "stardict_wordnet.h"
 #include "court_widget.h"
+#include <cstring>
 #include <glib/gi18n.h>
 
 #ifdef _WIN32

diff -Naur stardict-3.0.1.orig/stardict-plugins/stardict-xdxf-parsedata-plugin/stardict_xdxf_parsedata.cpp stardict-3.0.1/stardict-plugins/stardict-xdxf-parsedata-plugin/stardict_xdxf_parsedata.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-xdxf-parsedata-plugin/stardict_xdxf_parsedata.cpp	2007-08-31 07:41:54.000000000 +0100
+++ stardict-3.0.1/stardict-plugins/stardict-xdxf-parsedata-plugin/stardict_xdxf_parsedata.cpp	2008-04-11 18:00:12.000000000 +0100
@@ -1,4 +1,5 @@
 #include "stardict_xdxf_parsedata.h"
+#include <cstring>
 #include <glib/gi18n.h>
 
 #ifdef _WIN32
diff -Naur stardict-3.0.1.orig/tests/t_config_file.cpp stardict-3.0.1/tests/t_config_file.cpp
--- stardict-3.0.1.orig/tests/t_config_file.cpp	2007-07-10 08:16:04.000000000 +0100
+++ stardict-3.0.1/tests/t_config_file.cpp	2008-04-11 18:04:13.000000000 +0100
@@ -8,6 +8,7 @@
 #include <cstring>
 #include <iterator>
 #include <iostream>
+#include <memory>
 #include <gtk/gtk.h>
 
 #include "config_file.hpp"
diff -Naur stardict-3.0.1.orig/tests/t_xml.cpp stardict-3.0.1/tests/t_xml.cpp
--- stardict-3.0.1.orig/tests/t_xml.cpp	2007-07-10 08:16:04.000000000 +0100
+++ stardict-3.0.1/tests/t_xml.cpp	2008-04-11 18:04:58.000000000 +0100
@@ -5,6 +5,7 @@
 #include <glib.h>
 #include <cstdlib>
 #include <string>
+#include <cstring>
 
 
 static void xml_decode(const char *str, std::string& decoded)
