--- plugins/mp4/mp4.cpp~	2009-05-09 16:00:27.000000000 +0200
+++ plugins/mp4/mp4.cpp	2009-05-09 16:00:27.000000000 +0200
@@ -232,7 +232,7 @@
   
   strcpy(mdata->fileFormat, fileName + strlen(fileName) - 3); 
   
-  if (!MP4Close(mp4file))
+  MP4Close(mp4file);
     return 0;
   
   return 1;
@@ -316,7 +316,7 @@
   sprintf(temp, "%d", mdata->nonAlbum);  
   MP4SetMetadataFreeForm(mp4file, "MusicBrainz Non-Album", (u_int8_t *)temp, strlen(temp) + 1);
   
-  if (!MP4Close(mp4file))
+  MP4Close(mp4file);
     return 0;
 
 #ifndef WIN32
