diff -up lib/metadata.cpp.gcc43 lib/metadata.cpp
--- lib/metadata.cpp.gcc43 2006-11-18 04:52:33.000000000 -0600
+++ lib/metadata.cpp 2008-02-21 20:39:10.000000000 -0600
@@ -26,6 +26,7 @@
 ----------------------------------------------------------------------------*/
 #include <math.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include "metadata.h"
 #include "../config.h"
 using namespace std;
diff -up lib/protocol.cpp.gcc43 lib/protocol.cpp
--- lib/protocol.cpp.gcc43 2006-11-28 14:25:04.000000000 -0600
+++ lib/protocol.cpp 2008-02-21 20:39:10.000000000 -0600
@@ -8,6 +8,7 @@
 -------------------------------------------------------------------*/
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <string>
 #include <map>
 #include <expat.h>
diff -up lib/utf8/utf8util.cpp.gcc43 lib/utf8/utf8util.cpp
--- lib/utf8/utf8util.cpp.gcc43 2006-11-18 04:52:26.000000000 -0600
+++ lib/utf8/utf8util.cpp 2008-02-21 20:39:10.000000000 -0600
@@ -26,6 +26,7 @@
 ----------------------------------------------------------------------------*/
 
 #include <stdio.h>
+#include <stdlib.h>
 #include "utf8util.h"
 #include "utf8.h"
 #ifdef WIN32
diff -up lib/tunepimp.cpp.gcc43 lib/tunepimp.cpp
--- lib/tunepimp.cpp.gcc43 2006-11-18 04:52:33.000000000 -0600
+++ lib/tunepimp.cpp 2008-02-21 20:39:10.000000000 -0600
@@ -30,7 +30,9 @@
 #endif
 #endif
 
+#include <string.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <map>
 using namespace std;
 
diff -up lib/c_wrapper.cpp.gcc43 lib/c_wrapper.cpp
--- lib/c_wrapper.cpp.gcc43 2006-11-18 04:52:33.000000000 -0600
+++ lib/c_wrapper.cpp 2008-02-21 20:39:10.000000000 -0600
@@ -24,6 +24,7 @@
    $Id: tunepimp-0.5.3-gcc43.patch,v 1.1 2008/04/07 05:49:36 philantrop Exp $
 
 ----------------------------------------------------------------------------*/
+#include <stdlib.h>
 #include "tunepimp.h"
 #include "mutex.h"
 #include "tp_c.h"
diff -up lib/readmeta.cpp.gcc43 lib/readmeta.cpp
--- lib/readmeta.cpp.gcc43 2006-11-18 04:52:33.000000000 -0600
+++ lib/readmeta.cpp 2008-02-21 20:39:10.000000000 -0600
@@ -31,6 +31,7 @@
 #endif
 
 #include <stdio.h>
+#include <stdlib.h>
 #ifndef WIN32
 #include <unistd.h>
 #endif
diff -up lib/fileio.cpp.gcc43 lib/fileio.cpp
--- lib/fileio.cpp.gcc43 2006-11-18 04:52:33.000000000 -0600
+++ lib/fileio.cpp 2008-02-21 20:39:10.000000000 -0600
@@ -27,6 +27,8 @@
 
 #include <assert.h>
 #include <errno.h>
+#include <stdlib.h>
+#include <string.h>
 #include <string>
 #ifndef WIN32
 #include <unistd.h>
diff -up plugins/mp4/mp4.cpp.gcc43 plugins/mp4/mp4.cpp
--- plugins/mp4/mp4.cpp.gcc43 2006-11-18 04:51:08.000000000 -0600
+++ plugins/mp4/mp4.cpp 2008-02-21 20:39:10.000000000 -0600
@@ -27,6 +27,7 @@
 
 #include <string.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <mp4.h>
 #include "metadata.h"
 #include "plugin.h"
diff -up plugins/mp3/id3_meta.cpp.gcc43 plugins/mp3/id3_meta.cpp
--- plugins/mp3/id3_meta.cpp.gcc43 2006-11-18 04:51:07.000000000 -0600
+++ plugins/mp3/id3_meta.cpp 2008-02-21 20:41:35.000000000 -0600
@@ -26,6 +26,7 @@
 ----------------------------------------------------------------------------*/
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <time.h>
 #include <ctype.h>
 #include <musicbrainz/mb_c.h>
diff -up plugins/mp3/id3_2_3_meta.cpp.gcc43 plugins/mp3/id3_2_3_meta.cpp
--- plugins/mp3/id3_2_3_meta.cpp.gcc43 2006-11-18 04:51:07.000000000 -0600
+++ plugins/mp3/id3_2_3_meta.cpp 2008-02-21 20:41:59.000000000 -0600
@@ -26,6 +26,7 @@
 ----------------------------------------------------------------------------*/
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <assert.h>
 #include <ctype.h>
 #include <musicbrainz/mb_c.h>
diff -up plugins/wma/wma.cpp.gcc43 plugins/wma/wma.cpp
--- plugins/wma/wma.cpp.gcc43 2006-11-18 04:51:28.000000000 -0600
+++ plugins/wma/wma.cpp 2008-02-21 20:39:10.000000000 -0600
@@ -27,6 +27,7 @@
 
 #include <string.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <wmafile.h>
 #include "metadata.h"
 #include "plugin.h"
diff -up plugins/wma/wmafile.cpp.gcc43 plugins/wma/wmafile.cpp
--- plugins/wma/wmafile.cpp.gcc43 2006-11-18 04:51:28.000000000 -0600
+++ plugins/wma/wmafile.cpp 2008-02-21 20:39:10.000000000 -0600
@@ -19,6 +19,7 @@
  *   USA                                                                   *
  ***************************************************************************/
 
+#include <string.h>
 #include <tstring.h>
 #include <wmafile.h>
 #include <wmatag.h>
diff -up include/tunepimp-0.5/metadata.h.gcc43 include/tunepimp-0.5/metadata.h
--- include/tunepimp-0.5/metadata.h.gcc43 2006-11-18 04:52:08.000000000 -0600
+++ include/tunepimp-0.5/metadata.h 2008-02-21 20:39:10.000000000 -0600
@@ -27,6 +27,7 @@
 #ifndef __METADATA_H__
 #define __METADATA_H__
 
+#include <string.h>
 #include <string>
 #include <cstdio> 