--- sw/source/core/doc/docbm.cxx	2009-05-01 08:09:45.000000000 -0500
+++ sw/source/core/doc/docbm.cxx	2009-05-01 13:31:49.000000000 -0500
@@ -33,6 +33,7 @@
 #include "precompiled_sw.hxx"
 
 
+#include <stdio.h>
 #include <tools/pstm.hxx>
 #include <svx/linkmgr.hxx>
 #include <fmtanchr.hxx>
--- sw/source/filter/ww8/WW8TableInfo.cxx	2009-04-21 04:51:46.000000000 -0500
+++ sw/source/filter/ww8/WW8TableInfo.cxx	2009-05-01 15:20:53.000000000 -0500
@@ -31,6 +31,7 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_sw.hxx"
 
+#include <stdio.h>
 #include <iostream> 
 #include "WW8TableInfo.hxx"
 #include "swtable.hxx"

