--- officecfg/registry/data/org/openoffice/Office/Common.xcu
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu
@@ -576,8 +576,8 @@
     </node>
   </node>
   <node oor:name="Misc">
-    <prop oor:name="UseSystemFileDialog"  install:module="UseOOoFileDialogs">
-      <value>false</value>
+    <prop oor:name="UseSystemFileDialog"  install:module="UseOOoFileDialogs">
+      <value>true</value>
     </prop>
     <prop oor:name="UseSystemPrintDialog"  install:module="macosx">
       <value>true</value>
