--- desktop/scripts/soffice.sh
+++ desktop/scripts/soffice.sh
@@ -207,6 +207,24 @@
 OPENOFFICE_MOZILLA_FIVE_HOME="$sd_inst/program"
 export OPENOFFICE_MOZILLA_FIVE_HOME
 
+if [ -z "$MOZILLA_CERTIFICATE_FOLDER" ]; then
+	db=""
+	dir=""
+	for d in $HOME/.mozilla/firefox $HOME/.firefox $HOME/.thunderbird $HOME/.gnome2/epiphany/mozilla $HOME/.mozilla/default $HOME/.mozilla; do
+		if [ -z "$dir" -a -d "$d" ]; then
+			dir=$d;
+		fi;
+	done
+	if [ -n "$dir" ]; then
+		cert8=`find $dir -name "cert8.db" | head -n 1`
+		if [ -n "$cert8" ]; then
+			profiledir=`dirname "$cert8"`
+			MOZILLA_CERTIFICATE_FOLDER=$profiledir
+			export MOZILLA_CERTIFICATE_FOLDER
+		fi
+	fi
+fi
+
 unset XENVIRONMENT
 
 # uncomment line below to disable anti aliasing of fonts
