Index: /trunk/Source/JavaScriptCore/runtime/JSArrayBufferView.h
===================================================================
--- trunk/Source/JavaScriptCore/runtime/JSArrayBufferView.h	(revision 211433)
+++ trunk/Source/JavaScriptCore/runtime/JSArrayBufferView.h	(revision 211434)
@@ -168,5 +168,5 @@
     void neuter();
     
-    void* vector()
+    void* vector() const
     {
         return m_vector.getPredicated(
             this,
