--- webkit-qtwebkit-tp5/Source/JavaScriptCore/inspector/InspectorBackendDispatcher.h.orig	2017-06-01 06:14:55.587195469 +0000
+++ webkit-qtwebkit-tp5/Source/JavaScriptCore/inspector/InspectorBackendDispatcher.h	2017-06-01 06:15:23.040491577 +0000
@@ -32,6 +32,7 @@
 #include <wtf/Optional.h>
 #include <wtf/RefCounted.h>
 #include <wtf/text/WTFString.h>
+#include <functional>
 
 namespace Inspector {
 
--- webkit-qtwebkit-tp5/Source/WebCore/css/CSSValue.h.orig	2017-06-01 07:48:18.099623726 +0000
+++ webkit-qtwebkit-tp5/Source/WebCore/css/CSSValue.h	2017-06-01 07:48:41.542925091 +0000
@@ -27,6 +27,7 @@
 #include <wtf/RefCounted.h>
 #include <wtf/RefPtr.h>
 #include <wtf/TypeCasts.h>
+#include <functional>
 
 namespace WebCore {
 
--- webkit-qtwebkit-tp5/Source/WebCore/css/StyleSheetContents.h.orig	2017-06-01 08:33:31.792577087 +0000
+++ webkit-qtwebkit-tp5/Source/WebCore/css/StyleSheetContents.h	2017-06-01 08:33:44.715892348 +0000
@@ -30,6 +30,7 @@
 #include <wtf/Vector.h>
 #include <wtf/text/AtomicStringHash.h>
 #include <wtf/text/TextPosition.h>
+#include <functional>
 
 namespace WebCore {
 
--- webkit-qtwebkit-tp5/Source/WebCore/loader/ResourceLoader.h.orig	2017-06-01 09:47:50.463011889 +0000
+++ webkit-qtwebkit-tp5/Source/WebCore/loader/ResourceLoader.h	2017-06-01 09:48:21.976300680 +0000
@@ -35,6 +35,7 @@
 #include "ResourceRequest.h"
 #include "ResourceResponse.h"
 #include <wtf/Forward.h>
+#include <functional>
 
 namespace WTF {
 class SchedulePair;
