From 058bf8ae8686bd441188a62e950227e50214056b Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Mon, 14 Jan 2019 21:34:02 -0700
Subject: [PATCH] Use only unsigned long for rb_serial_t

---
 internal.h | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/internal.h b/internal.h
index 28a2f74..8c16c5a 100644
--- a/internal.h
+++ b/internal.h
@@ -941,19 +941,9 @@ struct rb_subclass_entry {
     rb_subclass_entry_t *next;
 };
 
-#if defined(HAVE_LONG_LONG)
-typedef unsigned LONG_LONG rb_serial_t;
-#define SERIALT2NUM ULL2NUM
-#define PRI_SERIALT_PREFIX PRI_LL_PREFIX
-#elif defined(HAVE_UINT64_T)
-typedef uint64_t rb_serial_t;
-#define SERIALT2NUM SIZET2NUM
-#define PRI_SERIALT_PREFIX PRI_64_PREFIX
-#else
 typedef unsigned long rb_serial_t;
 #define SERIALT2NUM ULONG2NUM
 #define PRI_SERIALT_PREFIX PRI_LONG_PREFIX
-#endif
 
 struct rb_classext_struct {
     struct st_table *iv_index_tbl;
-- 
2.20.1

