From 53b9e3e7da9da9a797d0b9c0b07bf869dcc7d1bf Mon Sep 17 00:00:00 2001
From: Johannes Dewender <github@JonnyJD.net>
Date: Tue, 26 May 2015 22:41:54 +0200
Subject: [PATCH] fix a syntax error in setup.py

Signed-off-by: Johannes Dewender <github@JonnyJD.net>
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 442ac06..fcbae40 100755
--- a/setup.py
+++ b/setup.py
@@ -28,7 +28,7 @@ setup (
     url = 'http://github.com/agrover/rtslib-fb',
     packages = ['rtslib_fb', 'rtslib'],
     scripts = ['scripts/targetctl'],
-    "classifiers": [
+    classifiers = [
         "Programming Language :: Python",
         "Programming Language :: Python :: 3",
         "License :: OSI Approved :: Apache Software License",
-- 
2.1.3

