From 26ded571425d3f5b651bdd956013b049bacd718d Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 29 Jan 2017 11:07:56 -0700
Subject: [PATCH] AArch64 fixes

---
 libco/sjlj.c        | 1 +
 nall/intrinsics.hpp | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/libco/sjlj.c b/libco/sjlj.c
index 1d0cb59a..885b1d36 100644
--- a/libco/sjlj.c
+++ b/libco/sjlj.c
@@ -4,6 +4,7 @@
 */
 
 #define LIBCO_C
+#define _XOPEN_SOURCE 700
 #include "libco.h"
 #include "settings.h"
 
diff --git a/nall/intrinsics.hpp b/nall/intrinsics.hpp
index e5dcaf64..20ed3460 100644
--- a/nall/intrinsics.hpp
+++ b/nall/intrinsics.hpp
@@ -122,7 +122,7 @@ namespace nall {
 #elif defined(__amd64__) || defined(_M_AMD64)
   #define PROCESSOR_AMD64
   auto Intrinsics::processor() -> Processor { return Processor::amd64; }
-#elif defined(__arm__)
+#elif defined(__arm__) || defined(__aarch64__)
   #define PROCESSOR_ARM
   auto Intrinsics::processor() -> Processor { return Processor::ARM; }
 #elif defined(__ppc64__) || defined(_ARCH_PPC64)
-- 
2.13.3

