--- interface/gtk_i.c	2004/12/21 16:59:33	1.1
+++ interface/gtk_i.c	2004/12/21 16:59:41
@@ -365,6 +365,7 @@ void
 Launch_Gtk_Process(int pipe_number)
 {
     int	argc = 0;
+    gchar **argv = NULL;
     GtkWidget *button, *mbar, *swin;
     GtkWidget *table, *align, *handlebox;
     GtkWidget *vbox, *hbox, *vbox2, *scrolled_win;
@@ -373,7 +374,7 @@ Launch_Gtk_Process(int pipe_number)
     /* enable locale */
     gtk_set_locale ();
 
-    gtk_init (&argc, NULL);
+    gtk_init (&argc, &argv);
 
     ttip = create_yellow_tooltips();
     window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
