--- a/doc/doc.pri
+++ b/doc/doc.pri
@@ -1,6 +1,6 @@
 unix {
-    QDOC = SRCDIR=$$PWD OUTDIR=$$OUT_PWD/doc/html $$(QTDIR)/tools/qdoc3/qdoc3
-    HELPGENERATOR = $$(QTDIR)/bin/qhelpgenerator
+    QDOC = SRCDIR=$$PWD OUTDIR=$$OUT_PWD/doc/html /usr/bin/qdoc3
+    HELPGENERATOR = /usr/bin/qhelpgenerator
 } else {
     QDOC = set SRCDIR=$$PWD&& set OUTDIR=$$OUT_PWD/doc/html&& $$(QTDIR)\tools\qdoc3\release\qdoc3.exe
     HELPGENERATOR = $$(QTDIR)\bin\qhelpgenerator.exe
