diff -rupN qt-creator-1.1.0-src/src/app/main.cpp orig/src/app/main.cpp
--- qt-creator-1.1.0-src/src/app/main.cpp	2009-04-21 16:13:13.000000000 +0200
+++ qt-creator-1.1.0-src/src/app/main.cpp	2009-05-02 16:53:57.985960034 +0200
@@ -179,11 +179,7 @@ static inline QStringList getPluginPaths
     // 1) "plugins" (Win/Linux)
     QString pluginPath = rootDirPath;
     pluginPath += QDir::separator();
-#if defined(QT_ARCH_X86_64) && defined(Q_OS_UNIX) && !defined(Q_OS_MAC)
-    pluginPath += QLatin1String("lib64");
-#else
     pluginPath += QLatin1String("lib");
-#endif
     pluginPath += QDir::separator();
     pluginPath += QLatin1String("qtcreator");
     pluginPath += QDir::separator();
diff -rupN qt-creator-1.1.0-src/src/qworkbench.pri orig/src/qworkbench.pri
--- qt-creator-1.1.0-src/src/qworkbench.pri	2009-04-21 16:13:13.000000000 +0200
+++ qt-creator-1.1.0-src/src/qworkbench.pri	2009-05-02 16:53:59.801793855 +0200
@@ -6,11 +6,7 @@ isEmpty(TEST):CONFIG(debug, debug|releas
     }
 }
 
-linux-*-64 {
-    IDE_LIBRARY_BASENAME = lib64
-} else {
-    IDE_LIBRARY_BASENAME = lib
-}
+IDE_LIBRARY_BASENAME = lib
 
 equals(TEST, 1) {
     QT +=testlib
