Index: motion-3.2.10.1/configure.in
===================================================================
--- motion-3.2.10.1.orig/configure.in
+++ motion-3.2.10.1/configure.in
@@ -415,12 +415,18 @@ if test "${FFMPEG_OK}" = "found"; then
 	elif test -f ${FFMPEG}/avformat.h; then
 		AC_MSG_RESULT(found ${FFMPEG}/avformat.h)
 		FFMPEG_CFLAGS="-I${FFMPEG}"
+	elif test -f ${FFMPEG}/libavformat/avformat.h; then
+		AC_MSG_RESULT(found ${FFMPEG}/libavformat/avformat.h)
+		FFMPEG_CFLAGS="-I${FFMPEG}/libavformat -I${FFMPEG}/libavutil"
 	elif test -f `AS_DIRNAME([${FFMPEG}])`/include/avformat.h; then
 		AC_MSG_RESULT(found ${FFMPEG}/include/avformat.h)
 		FFMPEG_CFLAGS="-I`AS_DIRNAME([${FFMPEG}])`/include"
 	elif test -f `AS_DIRNAME([${FFMPEG}])`/include/ffmpeg/avformat.h; then
 		AC_MSG_RESULT(found ${FFMPEG}/include/ffmpeg/avformat.h)
 		FFMPEG_CFLAGS="-I`AS_DIRNAME([${FFMPEG}])`/include/ffmpeg"
+	elif test -f `AS_DIRNAME([${FFMPEG}])`/include/libavformat/avformat.h; then
+		AC_MSG_RESULT(found ${FFMPEG}/include/libavformat/avformat.h)
+		FFMPEG_CFLAGS="-I`AS_DIRNAME([${FFMPEG}])`/include/libavformat -I`AS_DIRNAME([${FFMPEG}])`/include/libavutil"
 	else
 		AC_MSG_RESULT(not found)
 		FFMPEG_OK="no_found"
