#Source: http://launchpadlibrarian.net/18515541/thinkfinger-uinput-hack.patch (26-12-2008)
Index: pam/pam_thinkfinger-uinput.c
===================================================================
--- pam/pam_thinkfinger-uinput.c	(revision 118)
+++ pam/pam_thinkfinger-uinput.c	(working copy)
@@ -95,6 +95,7 @@
 	/* our single key keyboard */
 	i  = ioctl (*fd, UI_SET_EVBIT, EV_KEY) < 0;
 	i |= ioctl (*fd, UI_SET_KEYBIT, KEY_ENTER) < 0;
+	i |= ioctl (*fd, UI_SET_KEYBIT, KEY_A) < 0;
 
 	if (write (*fd, &device, device_size) != device_size) {
 		retval = errno;
