--- comm-1.9.1/mozilla/modules/libpref/src/init/all.js	2009-10-11 02:12:44.000000000 +0200
+++ comm-1.9.1/mozilla/modules/libpref/src/init/all.js	2009-09-16 02:56:45.000000000 +0200
@@ -890,7 +890,7 @@
 pref("intl.charset.detector",               "chrome://global/locale/intl.properties");
 pref("intl.charset.default",                "chrome://global-platform/locale/intl.properties");
 pref("intl.ellipsis",                       "chrome://global-platform/locale/intl.properties");
-pref("intl.locale.matchOS",                 false);
+pref("intl.locale.matchOS",                 true);
 // fallback charset list for Unicode conversion (converting from Unicode)
 // currently used for mail send only to handle symbol characters (e.g Euro, trademark, smartquotes)
 // for ISO-8859-1
