diff -up xulrunner-27.0/mozilla-release/gfx/ycbcr/moz.build.arm xulrunner-27.0/mozilla-release/gfx/ycbcr/moz.build
--- firefox-27.0/mozilla-release/gfx/ycbcr/moz.build.arm	2014-01-28 05:03:42.000000000 +0100
+++ firefox-27.0/mozilla-release/gfx/ycbcr/moz.build	2014-02-03 11:14:34.062219509 +0100
@@ -61,7 +61,7 @@ else:
         'yuv_row_other.cpp',
     ]
 
-if CONFIG['CPU_ARCH'] == 'arm' and CONFIG['HAVE_ARM_NEON']:
+if CONFIG['HAVE_ARM_NEON']:
     SOURCES += [
         'yuv_row_arm.s',
     ]
