diff -u -r --exclude=.deps --exclude=Makefile mysql-gui-tools-5.0r12/common/source/linux/MGFileBrowserList.cc mysql-gui-tools-5.0r12.good/common/source/linux/MGFileBrowserList.cc
--- mysql-gui-tools-5.0r12/common/source/linux/MGFileBrowserList.cc	2006-07-25 22:03:52.000000000 +0200
+++ mysql-gui-tools-5.0r12.good/common/source/linux/MGFileBrowserList.cc	2008-04-05 23:53:43.000000000 +0200
@@ -21,7 +21,7 @@
 #include <sys/stat.h>
 #include <errno.h>
 #include <stdio.h> /* rename() in FreeBSD */
-
+#include <cstring>
 
 
 MGFileBrowserList::MGFileBrowserList(const Glib::ustring &title,
diff -u -r --exclude=.deps --exclude=Makefile mysql-gui-tools-5.0r12/common/source/linux/MGHMeterGraphPlotter.cc mysql-gui-tools-5.0r12.good/common/source/linux/MGHMeterGraphPlotter.cc
--- mysql-gui-tools-5.0r12/common/source/linux/MGHMeterGraphPlotter.cc	2005-03-13 14:16:59.000000000 +0100
+++ mysql-gui-tools-5.0r12.good/common/source/linux/MGHMeterGraphPlotter.cc	2008-04-05 23:54:39.000000000 +0200
@@ -16,7 +16,7 @@
 
 
 #include "MGHMeterGraphPlotter.h"
-
+#include <cstring>
 
 
 MGHMeterGraphPlotter::MGHMeterGraphPlotter(GtkDrawingArea *darea)
diff -u -r --exclude=.deps --exclude=Makefile mysql-gui-tools-5.0r12/query-browser/source/linux/MGCodeEditor.cc mysql-gui-tools-5.0r12.good/query-browser/source/linux/MGCodeEditor.cc
--- mysql-gui-tools-5.0r12/query-browser/source/linux/MGCodeEditor.cc	2007-01-29 19:20:57.000000000 +0100
+++ mysql-gui-tools-5.0r12.good/query-browser/source/linux/MGCodeEditor.cc	2008-04-06 00:07:31.000000000 +0200
@@ -17,7 +17,7 @@
 #include "MGCodeEditor.h"
 
 #include <gtksourceview/gtksourcelanguagesmanager.h>
-
+#include <cstring>
 
 
 
