diff -Naurp mysql-gui-tools-5.0r12/administrator/source/linux/MABackupPanel.cc mysql-gui-tools-5.0r12.oden/administrator/source/linux/MABackupPanel.cc
--- mysql-gui-tools-5.0r12/administrator/source/linux/MABackupPanel.cc	2007-02-28 14:51:50.000000000 +0100
+++ mysql-gui-tools-5.0r12.oden/administrator/source/linux/MABackupPanel.cc	2007-11-08 16:21:44.000000000 +0100
@@ -1282,7 +1282,7 @@ void MABackupPanel::perform_backup()
 
   if (_inst->check_connection())
   {
-    SigC::Connection con= Glib::signal_timeout().connect(sigc::bind<void*>
+    sigc::connection con= Glib::signal_timeout().connect(sigc::bind<void*>
                                                          (sigc::mem_fun(*this, &MABackupPanel::update_backup_progress),&arg.pdata),
                                                          50);
     dlg->show();
diff -Naurp mysql-gui-tools-5.0r12/administrator/source/linux/MACatalogsPanel.cc mysql-gui-tools-5.0r12.oden/administrator/source/linux/MACatalogsPanel.cc
--- mysql-gui-tools-5.0r12/administrator/source/linux/MACatalogsPanel.cc	2007-02-21 01:48:34.000000000 +0100
+++ mysql-gui-tools-5.0r12.oden/administrator/source/linux/MACatalogsPanel.cc	2007-11-08 16:21:23.000000000 +0100
@@ -1488,7 +1488,7 @@ void MACatalogsPanel::maint_next_page()
 
       _maint_dlg_xml->get_button("next_button")->set_sensitive(false);
       _maint_dlg_xml->get_note("note")->set_current_page(4);
-      SigC::Connection conn = Glib::signal_timeout().connect(sigc::mem_fun(*this, &MACatalogsPanel::maint_pulse_progressbar), 200);
+      sigc::connection conn = Glib::signal_timeout().connect(sigc::mem_fun(*this, &MACatalogsPanel::maint_pulse_progressbar), 200);
 
       status= (MYX_TABLE_COMMAND_STATUSES*)
         _data->get_instance()->perform_data_fetch3((MInstanceInfo::DataFetcher3)myx_optimize_table,
@@ -1528,7 +1528,7 @@ void MACatalogsPanel::maint_next_page()
 
       _maint_dlg_xml->get_button("next_button")->set_sensitive(false);
       _maint_dlg_xml->get_note("note")->set_current_page(4);      
-      SigC::Connection conn = Glib::signal_timeout().connect(sigc::mem_fun(*this, &MACatalogsPanel::maint_pulse_progressbar), 200);
+      sigc::connection conn = Glib::signal_timeout().connect(sigc::mem_fun(*this, &MACatalogsPanel::maint_pulse_progressbar), 200);
 
       status= (MYX_TABLE_COMMAND_STATUSES*)
         _data->get_instance()->perform_data_fetch3((MInstanceInfo::DataFetcher3)myx_check_table,
@@ -1568,7 +1568,7 @@ void MACatalogsPanel::maint_next_page()
 
       _maint_dlg_xml->get_button("next_button")->set_sensitive(false);
       _maint_dlg_xml->get_note("note")->set_current_page(4);
-      SigC::Connection conn = Glib::signal_timeout().connect(sigc::mem_fun(*this, &MACatalogsPanel::maint_pulse_progressbar), 200);
+      sigc::connection conn = Glib::signal_timeout().connect(sigc::mem_fun(*this, &MACatalogsPanel::maint_pulse_progressbar), 200);
 
       status= (MYX_TABLE_COMMAND_STATUSES*)
         _data->get_instance()->perform_data_fetch3((MInstanceInfo::DataFetcher3)myx_repair_table,
diff -Naurp mysql-gui-tools-5.0r12/administrator/source/linux/MAdministrator.h mysql-gui-tools-5.0r12.oden/administrator/source/linux/MAdministrator.h
--- mysql-gui-tools-5.0r12/administrator/source/linux/MAdministrator.h	2007-03-27 15:09:56.000000000 +0200
+++ mysql-gui-tools-5.0r12.oden/administrator/source/linux/MAdministrator.h	2007-11-08 16:21:37.000000000 +0100
@@ -66,7 +66,7 @@ class MAdministrator : public Glib::Obje
 
     sigc::signal0<void> _signal_prefs_changed;
     
-    SigC::Connection _pulse_conn;
+    sigc::connection _pulse_conn;
     bool pulse_progress();
     
     void setup_sidebar();
diff -Naurp mysql-gui-tools-5.0r12/administrator/source/linux/MARestorePanel.cc mysql-gui-tools-5.0r12.oden/administrator/source/linux/MARestorePanel.cc
--- mysql-gui-tools-5.0r12/administrator/source/linux/MARestorePanel.cc	2006-09-26 20:12:56.000000000 +0200
+++ mysql-gui-tools-5.0r12.oden/administrator/source/linux/MARestorePanel.cc	2007-11-08 16:21:16.000000000 +0100
@@ -728,7 +728,7 @@ MYX_BACKUP_CONTENT *MARestorePanel::perf
   op_aborted= false;
 
   {
-    SigC::Connection con= Glib::signal_timeout().connect(sigc::bind<void*>
+    sigc::connection con= Glib::signal_timeout().connect(sigc::bind<void*>
                                                          (sigc::mem_fun(*this, &MARestorePanel::update_status),&arg.pdata),
                                                          50);
 
@@ -854,7 +854,7 @@ void MARestorePanel::perform_restore()
   //{
     MYX_BACKUP_ERROR err;
 
-    SigC::Connection con= Glib::signal_timeout().connect(sigc::bind<void*>
+    sigc::connection con= Glib::signal_timeout().connect(sigc::bind<void*>
                                                          (sigc::mem_fun(*this, &MARestorePanel::update_status),&arg.pdata),
                                                          50);
 
diff -Naurp mysql-gui-tools-5.0r12/administrator/source/linux/MAServerConnectionsPanel.h mysql-gui-tools-5.0r12.oden/administrator/source/linux/MAServerConnectionsPanel.h
--- mysql-gui-tools-5.0r12/administrator/source/linux/MAServerConnectionsPanel.h	2005-04-22 05:21:41.000000000 +0200
+++ mysql-gui-tools-5.0r12.oden/administrator/source/linux/MAServerConnectionsPanel.h	2007-11-08 16:21:32.000000000 +0100
@@ -87,7 +87,7 @@ class MAServerConnectionsPanel : public 
 
     Glib::RefPtr<Gdk::Pixbuf> _thread_icon;
 
-    SigC::Connection _timer;
+    sigc::connection _timer;
     
     int _current_page;
 
diff -Naurp mysql-gui-tools-5.0r12/common/source/linux/MGConnectDialog.cc mysql-gui-tools-5.0r12.oden/common/source/linux/MGConnectDialog.cc
--- mysql-gui-tools-5.0r12/common/source/linux/MGConnectDialog.cc	2006-12-05 23:44:08.000000000 +0100
+++ mysql-gui-tools-5.0r12.oden/common/source/linux/MGConnectDialog.cc	2007-11-08 16:21:54.000000000 +0100
@@ -659,9 +659,9 @@ void MGConnectDialog::open_connection_ed
   // select back the original item
   ((Gtk::OptionMenu*)_xml->get_widget("connection_list"))->set_history(_current_selected_item);
 
-  SigC::Connection c1= MGPreferencesEditor::instance()->signal_closed().connect(sigc::mem_fun(*this,
+  sigc::connection c1= MGPreferencesEditor::instance()->signal_closed().connect(sigc::mem_fun(*this,
                                                                                            &MGConnectDialog::preferences_closed));
-  SigC::Connection c2= MGPreferencesEditor::instance()->signal_changed().connect(sigc::mem_fun(*this,
+  sigc::connection c2= MGPreferencesEditor::instance()->signal_changed().connect(sigc::mem_fun(*this,
                                                                                             &MGConnectDialog::preferences_changed));
   MGPreferencesEditor::instance()->show(true);
   MGPreferencesEditor::instance()->set_modal(true);
@@ -676,7 +676,7 @@ void MGConnectDialog::show_connection_fa
 {
   Gtk::Button *btn;
   pid_t ping_pid= 0;
-  SigC::Connection input_handler;
+  sigc::connection input_handler;
   int rc;
   bool pinging= false;
   int myerror= 0;
diff -Naurp mysql-gui-tools-5.0r12/common/source/linux/MGTreeTooltip.h mysql-gui-tools-5.0r12.oden/common/source/linux/MGTreeTooltip.h
--- mysql-gui-tools-5.0r12/common/source/linux/MGTreeTooltip.h	2005-10-07 13:31:36.000000000 +0200
+++ mysql-gui-tools-5.0r12.oden/common/source/linux/MGTreeTooltip.h	2007-11-08 16:21:49.000000000 +0100
@@ -39,7 +39,7 @@ class MGTreeTooltip : public Glib::Objec
     
     WillShowSignal _show_signal;
     
-    SigC::Connection _timeout;
+    sigc::connection _timeout;
 
     void expose_event(GdkEventExpose *event);
     void leave_event(GdkEventCrossing *event);
diff -Naurp mysql-gui-tools-5.0r12/query-browser/source/linux/MQResultTab.h mysql-gui-tools-5.0r12.oden/query-browser/source/linux/MQResultTab.h
--- mysql-gui-tools-5.0r12/query-browser/source/linux/MQResultTab.h	2006-12-06 16:50:09.000000000 +0100
+++ mysql-gui-tools-5.0r12.oden/query-browser/source/linux/MQResultTab.h	2007-11-08 16:21:09.000000000 +0100
@@ -45,7 +45,7 @@ class MQResultTab : public MQBaseTab {
 
       Gtk::Paned *paned;
 
-      SigC::Connection scroll_con;
+      sigc::connection scroll_con;
     };
     
     enum CompareAction {
@@ -71,7 +71,7 @@ class MQResultTab : public MQBaseTab {
 
     bool _vertical;
 
-    SigC::Connection _sync_con1, _sync_con2;
+    sigc::connection _sync_con1, _sync_con2;
 
     void scrolled(MQResultSetView *sender);
     void activated(MQResultSetView *sender);
