Index: query-browser/source/linux/gtksourceview/gtksourceview/gtksourcelanguage.h
===================================================================
--- query-browser/source/linux/gtksourceview/gtksourceview/gtksourcelanguage.h	(revision 488)
+++ query-browser/source/linux/gtksourceview/gtksourceview/gtksourcelanguage.h	(working copy)
@@ -30,10 +30,10 @@
 G_BEGIN_DECLS
 
 #define GTK_TYPE_SOURCE_LANGUAGE		(gtk_source_language_get_type ())
-#define GTK_SOURCE_LANGUAGE(obj)		(GTK_CHECK_CAST ((obj), GTK_TYPE_SOURCE_LANGUAGE, GtkSourceLanguage))
-#define GTK_SOURCE_LANGUAGE_CLASS(klass)	(GTK_CHECK_CLASS_CAST ((klass), GTK_TYPE_SOURCE_LANGUAGE, GtkSourceLanguageClass))
-#define GTK_IS_SOURCE_LANGUAGE(obj)		(GTK_CHECK_TYPE ((obj), GTK_TYPE_SOURCE_LANGUAGE))
-#define GTK_IS_SOURCE_LANGUAGE_CLASS(klass)	(GTK_CHECK_CLASS_TYPE ((klass), GTK_TYPE_SOURCE_LANGUAGE))
+#define GTK_SOURCE_LANGUAGE(obj)		(G_TYPE_CHECK_INSTANCE_CAST ((obj), GTK_TYPE_SOURCE_LANGUAGE, GtkSourceLanguage))
+#define GTK_SOURCE_LANGUAGE_CLASS(klass)	(G_TYPE_CHECK_CLASS_CAST ((klass), GTK_TYPE_SOURCE_LANGUAGE, GtkSourceLanguageClass))
+#define GTK_IS_SOURCE_LANGUAGE(obj)		(G_TYPE_CHECK_INSTANCE_TYPE ((obj), GTK_TYPE_SOURCE_LANGUAGE))
+#define GTK_IS_SOURCE_LANGUAGE_CLASS(klass)	(G_TYPE_CHECK_CLASS_TYPE ((klass), GTK_TYPE_SOURCE_LANGUAGE))
 #define GTK_SOURCE_LANGUAGE_GET_CLASS(obj)      (G_TYPE_INSTANCE_GET_CLASS ((obj), GTK_TYPE_SOURCE_LANGUAGE, GtkSourceLanguageClass))
 
 
Index: query-browser/source/linux/gtksourceview/gtksourceview/gtksourcelanguagesmanager.h
===================================================================
--- query-browser/source/linux/gtksourceview/gtksourceview/gtksourcelanguagesmanager.h	(revision 488)
+++ query-browser/source/linux/gtksourceview/gtksourceview/gtksourcelanguagesmanager.h	(working copy)
@@ -27,10 +27,10 @@
 G_BEGIN_DECLS
 
 #define GTK_TYPE_SOURCE_LANGUAGES_MANAGER		(gtk_source_languages_manager_get_type ())
-#define GTK_SOURCE_LANGUAGES_MANAGER(obj)		(GTK_CHECK_CAST ((obj), GTK_TYPE_SOURCE_LANGUAGES_MANAGER, GtkSourceLanguagesManager))
-#define GTK_SOURCE_LANGUAGES_MANAGER_CLASS(klass)	(GTK_CHECK_CLASS_CAST ((klass), GTK_TYPE_SOURCE_LANGUAGES_MANAGER, GtkSourceLanguagesManagerClass))
-#define GTK_IS_SOURCE_LANGUAGES_MANAGER(obj)		(GTK_CHECK_TYPE ((obj), GTK_TYPE_SOURCE_LANGUAGES_MANAGER))
-#define GTK_IS_SOURCE_LANGUAGES_MANAGER_CLASS(klass)	(GTK_CHECK_CLASS_TYPE ((klass), GTK_TYPE_SOURCE_LANGUAGES_MANAGER))
+#define GTK_SOURCE_LANGUAGES_MANAGER(obj)		(G_TYPE_CHECK_INSTANCE_CAST ((obj), GTK_TYPE_SOURCE_LANGUAGES_MANAGER, GtkSourceLanguagesManager))
+#define GTK_SOURCE_LANGUAGES_MANAGER_CLASS(klass)	(G_TYPE_CHECK_CLASS_CAST ((klass), GTK_TYPE_SOURCE_LANGUAGES_MANAGER, GtkSourceLanguagesManagerClass))
+#define GTK_IS_SOURCE_LANGUAGES_MANAGER(obj)		(G_TYPE_CHECK_INSTANCE_TYPE ((obj), GTK_TYPE_SOURCE_LANGUAGES_MANAGER))
+#define GTK_IS_SOURCE_LANGUAGES_MANAGER_CLASS(klass)	(G_TYPE_CHECK_CLASS_TYPE ((klass), GTK_TYPE_SOURCE_LANGUAGES_MANAGER))
 
 
 typedef struct _GtkSourceLanguagesManager		GtkSourceLanguagesManager;
Index: query-browser/source/linux/gtksourceview/gtksourceview/gtksourceundomanager.h
===================================================================
--- query-browser/source/linux/gtksourceview/gtksourceview/gtksourceundomanager.h	(revision 488)
+++ query-browser/source/linux/gtksourceview/gtksourceview/gtksourceundomanager.h	(working copy)
@@ -29,11 +29,11 @@
 #include <gtk/gtktextbuffer.h>
 
 #define GTK_SOURCE_TYPE_UNDO_MANAGER             	(gtk_source_undo_manager_get_type ())
-#define GTK_SOURCE_UNDO_MANAGER(obj)			(GTK_CHECK_CAST ((obj), GTK_SOURCE_TYPE_UNDO_MANAGER, GtkSourceUndoManager))
-#define GTK_SOURCE_UNDO_MANAGER_CLASS(klass)		(GTK_CHECK_CLASS_CAST ((klass), GTK_SOURCE_TYPE_UNDO_MANAGER, GtkSourceUndoManagerClass))
-#define GTK_SOURCE_IS_UNDO_MANAGER(obj)			(GTK_CHECK_TYPE ((obj), GTK_SOURCE_TYPE_UNDO_MANAGER))
-#define GTK_SOURCE_IS_UNDO_MANAGER_CLASS(klass)  	(GTK_CHECK_CLASS_TYPE ((klass), GTK_SOURCE_TYPE_UNDO_MANAGER))
-#define GTK_SOURCE_UNDO_MANAGER_GET_CLASS(obj)  	(GTK_CHECK_GET_CLASS ((obj), GTK_SOURCE_TYPE_UNDO_MANAGER, GtkSourceUndoManagerClass))
+#define GTK_SOURCE_UNDO_MANAGER(obj)			(G_TYPE_CHECK_INSTANCE_CAST ((obj), GTK_SOURCE_TYPE_UNDO_MANAGER, GtkSourceUndoManager))
+#define GTK_SOURCE_UNDO_MANAGER_CLASS(klass)		(G_TYPE_CHECK_CLASS_CAST ((klass), GTK_SOURCE_TYPE_UNDO_MANAGER, GtkSourceUndoManagerClass))
+#define GTK_SOURCE_IS_UNDO_MANAGER(obj)			(G_TYPE_CHECK_INSTANCE_TYPE ((obj), GTK_SOURCE_TYPE_UNDO_MANAGER))
+#define GTK_SOURCE_IS_UNDO_MANAGER_CLASS(klass)  	(G_TYPE_CHECK_CLASS_TYPE ((klass), GTK_SOURCE_TYPE_UNDO_MANAGER))
+#define GTK_SOURCE_UNDO_MANAGER_GET_CLASS(obj)  	(G_TYPE_INSTANCE_GET_CLASS ((obj), GTK_SOURCE_TYPE_UNDO_MANAGER, GtkSourceUndoManagerClass))
 
 
 typedef struct _GtkSourceUndoManager        	GtkSourceUndoManager;
