diff -Naur xbmc-pvr-20120528-git8e5fa796/configure.in xbmc-pvr-20120528-git8e5fa796-new/configure.in
--- xbmc-pvr-20120528-git8e5fa796/configure.in	2012-07-12 23:41:00.463812887 +0200
+++ xbmc-pvr-20120528-git8e5fa796-new/configure.in	2012-07-12 23:42:59.163986997 +0200
@@ -684,8 +684,9 @@
       AC_MSG_RESULT(== WARNING: OpenGLES support is assumed.)
       LIBS="$LIBS -lEGL -lGLESv2 -lbcm_host -lvcos -lvchiq_arm"
     else
-      AC_CHECK_LIB([EGL],   [main],, AC_MSG_ERROR($missing_library))
-      AC_CHECK_LIB([GLESv2],[main],, AC_MSG_ERROR($missing_library))
+    AC_CHECK_LIB([EGL],   [main],, AC_MSG_ERROR($missing_library),[-lVIVANTE])
+    AC_CHECK_LIB([GLESv2],[main],, AC_MSG_ERROR($missing_library),[-lVIVANTE])
+    LIBS="$LIBS -lVIVANTE"
     fi
   fi
 else
