diff -Naur c/xbmc/utils/CPUInfo.cpp d/xbmc/utils/CPUInfo.cpp
--- c/xbmc/utils/CPUInfo.cpp	2012-10-24 20:28:26.274392014 +0200
+++ d/xbmc/utils/CPUInfo.cpp	2012-10-24 20:08:08.000000000 +0200
@@ -154,7 +154,10 @@
   // read from the new location of the temperature data on new kernels, 2.6.39, 3.0 etc
   if (m_fProcTemperature == NULL)   
     m_fProcTemperature = fopen("/sys/class/hwmon/hwmon0/temp1_input", "r");
-  
+  // some systems put this information here ...
+  if (m_fProcTemperature == NULL)   
+    m_fProcTemperature = fopen("/sys/class/hwmon/hwmon0/device/temp1_input", "r");
+
   m_fCPUInfo = fopen("/proc/cpuinfo", "r");
   m_cpuCount = 0;
   if (m_fCPUInfo)
@@ -164,7 +167,17 @@
     int nCurrId = 0;
     while (fgets(buffer, sizeof(buffer), m_fCPUInfo))
     {
-      if (strncmp(buffer, "processor", strlen("processor"))==0)
+      if (strncmp(buffer, "Processor", strlen("Processor"))==0)
+      {
+        char *needle = strstr(buffer, ":");
+        if (needle)
+        {
+          CStdString s(needle+2);
+          s.Trim();
+          m_cpuModel = s;
+        }
+      }
+      else if (strncmp(buffer, "processor", strlen("processor"))==0)
       {
         char *needle = strstr(buffer, ":");
         if (needle)
@@ -383,10 +370,16 @@
       }
     }
   }
-  else
+
+  if (m_cpuModel.empty())
+    m_cpuModel = "Unknown";
+
+  if (m_cpuCount == 0)
   {
+    CoreInfo core;
+    core.m_strModel = m_cpuModel;
+    m_cores[0] = core;
     m_cpuCount = 1;
-    m_cpuModel = "Unknown";
   }
 
 #endif
diff -Naur c/xbmc/utils/SystemInfo.cpp d/xbmc/utils/SystemInfo.cpp
--- c/xbmc/utils/SystemInfo.cpp	2012-07-11 21:58:22.000000000 +0200
+++ d/xbmc/utils/SystemInfo.cpp	2012-10-25 20:48:37.000000000 +0200
@@ -74,7 +74,8 @@
 {
   CStdString strCPUFreq;
   double CPUFreq = GetCPUFrequency();
-  strCPUFreq.Format("%4.2fMHz", CPUFreq);
+  if (CPUFreq > 0)
+    strCPUFreq.Format("%4.2fMHz", CPUFreq);
   return strCPUFreq;
 }
 
