diff -ruN BitchX-orig/include/module.h BitchX/include/module.h
--- BitchX-orig/include/module.h	2003-04-11 03:09:07.000000000 +0200
+++ BitchX/include/module.h	2008-10-04 19:28:13.000000000 +0200
@@ -15,7 +15,7 @@
 
 #include "struct.h"
 
-typedef int (*Function_ptr)();
+typedef long (*Function_ptr)();
 
 #ifdef WANT_DLL
 #ifdef HPUX
diff -ruN BitchX-orig/include/modval.h BitchX/include/modval.h
--- BitchX-orig/include/modval.h	2003-04-11 03:09:07.000000000 +0200
+++ BitchX/include/modval.h	2008-10-04 19:28:13.000000000 +0200
@@ -318,7 +318,7 @@
 #define show_window(x) ((void) (global[SHOW_WINDOW]((Window *)x)))
 #define get_status_by_refnum(x, y) ((char *) (global[GET_STATUS_BY_REFNUM]((unsigned int)x, (unsigned int)y)))
 #define get_visible_by_refnum(x) ((int) (global[GET_VISIBLE_BY_REFNUM]((char *)x)))
-#define get_window_by_desc(x) ((Window *) (global[GET_WINDOW_BY_DESC]((unsigned int)x)))
+#define get_window_by_desc(x) ((Window *) (global[GET_WINDOW_BY_DESC]((unsigned long)x)))
 #define get_window_by_refnum(x) ((Window *) (global[GET_WINDOW_BY_REFNUM]((unsigned int)x)))
 #define get_window_by_name(x) ((Window *) (global[GET_WINDOW_BY_NAME]((char *)x)))
 #define next_window(x, y) ((void) (global[FUNC_NEXT_WINDOW]((char)x, (char *)y)))
