diff -ruN BitchX-orig/include/irc_std.h BitchX/include/irc_std.h
--- BitchX-orig/include/irc_std.h	2003-04-11 03:09:07.000000000 +0200
+++ BitchX/include/irc_std.h	2008-10-04 19:27:29.000000000 +0200
@@ -98,7 +98,7 @@
 # define __N
 # define __inline__
 #else
-# if (__GNUC__ >= 2) && (__GNUC_MINOR__ >= 7)
+# if (__GNUC__ > 2) || ((__GNUC__ == 2) && (__GNUC_MINOR__ >= 7))
 #  define __A(x) __attribute__ ((format (printf, x, x + 1)))
 #  define __N    __attribute__ ((noreturn))
 # else
