diff -ruN BitchX-orig/dll/aim/toc/server.c BitchX/dll/aim/toc/server.c
--- BitchX-orig/dll/aim/toc/server.c	2003-04-11 03:09:06.000000000 +0200
+++ BitchX/dll/aim/toc/server.c	2008-10-04 19:17:25.000000000 +0200
@@ -7,14 +7,14 @@
 #include <unistd.h>
 #include "toc.h"
 
-static time_t lastsent = 0;
+time_t lastsent = 0;
 time_t login_time = 0;
 int my_evil;
 int is_idle = 0;
 int lag_ms = 0;
 int time_to_idle = 600;
 int is_away = 0;
-static struct timeval lag_tv;
+struct timeval lag_tv;
 
 void serv_add_buddy(char *name)
 {
diff -ruN BitchX-orig/dll/aim/toc/toc.c BitchX/dll/aim/toc/toc.c
--- BitchX-orig/dll/aim/toc/toc.c	2003-04-11 03:09:06.000000000 +0200
+++ BitchX/dll/aim/toc/toc.c	2008-10-04 19:17:25.000000000 +0200
@@ -39,9 +39,9 @@
 #include "toc.h"
 
 /* descriptor for talking to TOC */
-static int toc_fd;
-static int seqno;
-static unsigned int peer_ver=0;
+int toc_fd;
+int seqno;
+unsigned int peer_ver=0;
 int state;
 /* static int inpa=-1; */
 int permdeny = PERMIT_PERMITALL;
diff -ruN BitchX-orig/include/modval.h BitchX/include/modval.h
--- BitchX-orig/include/modval.h	2003-04-11 03:09:07.000000000 +0200
+++ BitchX/include/modval.h	2008-10-04 19:17:25.000000000 +0200
@@ -650,7 +650,7 @@
 #define start_time ((time_t) *((time_t *)global[START_TIME]))
 #define idle_time() ((time_t) *((time_t *)global[IDLE_TIME]()))
 
-#define loading_global ((int) *((int *)global[LOADING_GLOBAL]))
+#define loading_global (*((int *)global[LOADING_GLOBAL]))
 #define target_window (*((Window **)global[TARGET_WINDOW]))
 #define current_window (*((Window **)global[CURRENT_WINDOW]))
 #define invisible_list (*((Window **)global[INVISIBLE_LIST]))
@@ -665,12 +665,12 @@
 #define dll_variable (*((IrcVariableDll **)global[DLL_VARIABLE]))
 #define dll_ctcp (*((CtcpEntryDll **)global[DLL_CTCP]))
 #define dll_window (*((WindowDll **)global[DLL_WINDOW]))
-#define window_display ((int) *((int *)global[WINDOW_DISPLAY]))
-#define status_update_flag ((int) *((int *)global[STATUS_UPDATE_FLAG]))
+#define window_display (*((int *)global[WINDOW_DISPLAY]))
+#define status_update_flag (*((int *)global[STATUS_UPDATE_FLAG]))
 #define tabkey_array (*((NickTab **)global[TABKEY_ARRAY]))
 #define autoreply_array (*((NickTab *)global[AUTOREPLY_ARRAY]))
-#define identd ((int) *((int *)global[IDENTD_SOCKET]))
-#define doing_notice ((int) *((int *)global[DOING_NOTICE]))
+#define identd (*((int *)global[IDENTD_SOCKET]))
+#define doing_notice (*((int *)global[DOING_NOTICE]))
 #define last_sent_msg_body (*((char **)global[LAST_SENT_MSG_BODY]))
 #define sent_nick (*((char **)global[SENT_NICK]))
 
@@ -698,9 +698,9 @@
 #ifdef GUI
 #ifndef MAIN_SOURCE
 #define lastclicklinedata ((char *) *global[LASTCLICKLINEDATA])
-#define contextx ((int) *((int *)global[CONTEXTX]))
-#define contexty ((int) *((int *)global[CONTEXTY]))
-#define guiipc ((int) *((int *)global[GUIIPC]))
+#define contextx (*((int *)global[CONTEXTX]))
+#define contexty (*((int *)global[CONTEXTY]))
+#define guiipc (*((int *)global[GUIIPC]))
 #endif
 #define gui_mutex_lock() ((void (*)(void)) global[GUI_MUTEX_LOCK])()
 #define gui_mutex_unlock() ((void (*)(void))global[GUI_MUTEX_UNLOCK])()
diff -ruN BitchX-orig/source/ctcp.c BitchX/source/ctcp.c
--- BitchX-orig/source/ctcp.c	2003-05-27 09:00:22.000000000 +0200
+++ BitchX/source/ctcp.c	2008-10-04 19:17:25.000000000 +0200
@@ -176,7 +176,7 @@
 
 /* CDE do ops and unban logging */
 
-static char	*ctcp_type[] =
+char	*ctcp_type[] =
 {
 	"PRIVMSG",
 	"NOTICE"
diff -ruN BitchX-orig/source/term.c BitchX/source/term.c
--- BitchX-orig/source/term.c	2003-04-11 03:09:07.000000000 +0200
+++ BitchX/source/term.c	2008-10-04 19:17:25.000000000 +0200
@@ -91,8 +91,10 @@
 #define Tgetflag(x) 	tgetflag(x.tname)
 #endif
 
-extern  char    *getenv();
-extern	char	*tparm();
+extern  char    *getenv(const char *name);
+#ifndef TPARM_DECLARED
+extern	char	*tparm(char *str, ...);
+#endif
 
 /*
  * The old code assumed termcap. termcap is almost always present, but on
diff -ruN BitchX-orig/source/timer.c BitchX/source/timer.c
--- BitchX-orig/source/timer.c	2003-04-11 03:09:07.000000000 +0200
+++ BitchX/source/timer.c	2008-10-04 19:17:25.000000000 +0200
@@ -148,7 +148,7 @@
  * This is put here on purpose -- we dont want any of the above functions
  * to have any knowledge of this struct.
  */
-static TimerList *PendingTimers;
+TimerList *PendingTimers;
 static char *schedule_timer (TimerList *ntimer);
 
 static char *current_exec_timer = empty_string;
