diff -ruN BitchX-orig/include/config.h BitchX/include/config.h
--- BitchX-orig/include/config.h	2003-07-26 09:00:32.000000000 +0200
+++ BitchX/include/config.h	2008-10-04 19:46:29.000000000 +0200
@@ -330,7 +330,7 @@
  *
  * Is here there any solution to use both fonts nethertheless ?
  */
-#undef LATIN1
+#define LATIN1
 
 /*
  * If you use LINUX and non ISO8859-1 fonts with custom screen mapping,
