diff -ruN BitchX-orig/include/config.h BitchX/include/config.h
--- BitchX-orig/include/config.h	2003-07-26 09:00:32.000000000 +0200
+++ BitchX/include/config.h	2008-10-04 19:25:45.000000000 +0200
@@ -37,7 +37,26 @@
  *
  * List last updated: 03/24/01.  -dialtone.
  */
-#define DEFAULT_SERVER  "[efnet] "\
+#define DEFAULT_SERVER "[freenode] "\
+				"irc.freenode.net "\
+				"vinge.freenode.net "\
+				"truong.freenode.net "\
+				"stephenson.freenode.net "\
+				"orwell.freenode.net "\
+				"leguin.freenode.net "\
+				"kornbluth.freenode.net "\
+			        "jordan.freenode.net "\
+				"herbert.freenode.net "\
+				"goethe.freenode.net "\
+				"gibson.freenode.net "\
+				"gerrold.freenode.net "\
+				"carter.freenode.net "\
+				"carneiro.freenode.net "\
+				"capek.freenode.net "\
+				"calkins.freenode.net "\
+				"brin.freenode.net "\
+				"adams.freenode.net "\
+			"[efnet] "\
 				"irc.foxlink.net "\
 				"irc.weblook2k.com "\
 				"irc.wh.verio.net "\
