diff -ruN csound-6.11.0-a/CMakeLists.txt csound-6.11.0-b/CMakeLists.txt
--- csound-6.11.0-a/CMakeLists.txt	2018-05-10 10:31:20.000000000 +0200
+++ csound-6.11.0-b/CMakeLists.txt	2018-05-17 20:40:50.659122644 +0200
@@ -15,8 +15,8 @@
 if(NOT MSVC AND NOT WASM)
 set(CMAKE_CXX_FLAGS_RELEASE "-O3 ")
 set(CMAKE_C_FLAGS_RELEASE "-O3 ")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}  -Wall -Werror -Wno-missing-field-initializers")
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror -Wno-missing-field-initializers")
+set(CMAKE_CXX_FLAGS " -Wall -Werror -Wno-missing-field-initializers ${CMAKE_CXX_FLAGS}")
+set(CMAKE_C_FLAGS " -Wall -Werror -Wno-missing-field-initializers ${CMAKE_C_FLAGS}")
 endif()
 
 # C++11 needed
