diff -ruN a/kernel/printk.c b/kernel/printk.c
--- a/kernel/printk/printk.c	2013-05-07 21:58:03.000000000 -0600
+++ b/kernel/printk/printk.c	2013-05-08 23:24:39.000000000 -0600
@@ -61,7 +61,7 @@
 
 /* We show everything that is MORE important than this.. */
 #define MINIMUM_CONSOLE_LOGLEVEL 1 /* Minimum loglevel we let people use */
-#define DEFAULT_CONSOLE_LOGLEVEL 7 /* anything MORE serious than KERN_DEBUG */
+#define DEFAULT_CONSOLE_LOGLEVEL 4 /* anything MORE serious than KERN_DEBUG */
 
 int console_printk[4] = {
 	DEFAULT_CONSOLE_LOGLEVEL,	/* console_loglevel */
