diff -ruN a/arch/arm/boot/dts/kirkwood-dns320.dts b/arch/arm/boot/dts/kirkwood-dns320.dts
--- a/arch/arm/boot/dts/kirkwood-dns320.dts	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-dns320.dts	2012-12-24 11:35:26.000000000 -0500
@@ -12,7 +12,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	gpio-leds {
diff -ruN a/arch/arm/boot/dts/kirkwood-dns325.dts b/arch/arm/boot/dts/kirkwood-dns325.dts
--- a/arch/arm/boot/dts/kirkwood-dns325.dts	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-dns325.dts	2012-12-24 11:35:43.000000000 -0500
@@ -12,7 +12,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	gpio-leds {
diff -ruN a/arch/arm/boot/dts/kirkwood-dockstar.dts b/arch/arm/boot/dts/kirkwood-dockstar.dts
--- a/arch/arm/boot/dts/kirkwood-dockstar.dts	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-dockstar.dts	2012-12-24 11:36:38.000000000 -0500
@@ -13,7 +13,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk root=/dev/sda1 rootdelay=10";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-dreamplug.dts b/arch/arm/boot/dts/kirkwood-dreamplug.dts
--- a/arch/arm/boot/dts/kirkwood-dreamplug.dts	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-dreamplug.dts	2012-12-24 11:36:58.000000000 -0500
@@ -13,7 +13,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-goflexnet.dts b/arch/arm/boot/dts/kirkwood-goflexnet.dts
--- a/arch/arm/boot/dts/kirkwood-goflexnet.dts	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-goflexnet.dts	2012-12-24 11:37:18.000000000 -0500
@@ -13,7 +13,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk root=/dev/sda1 rootdelay=10";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-ib62x0.dts b/arch/arm/boot/dts/kirkwood-ib62x0.dts
--- a/arch/arm/boot/dts/kirkwood-ib62x0.dts	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-ib62x0.dts	2012-12-24 11:37:36.000000000 -0500
@@ -13,7 +13,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-iconnect.dts b/arch/arm/boot/dts/kirkwood-iconnect.dts
--- a/arch/arm/boot/dts/kirkwood-iconnect.dts	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-iconnect.dts	2012-12-24 11:38:01.000000000 -0500
@@ -13,9 +13,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
-		linux,initrd-start = <0x4500040>;
-		linux,initrd-end   = <0x4800000>;
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-iomega_ix2_200.dts b/arch/arm/boot/dts/kirkwood-iomega_ix2_200.dts
--- a/arch/arm/boot/dts/kirkwood-iomega_ix2_200.dts	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-iomega_ix2_200.dts	2012-12-24 11:38:17.000000000 -0500
@@ -13,7 +13,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-km_kirkwood.dts b/arch/arm/boot/dts/kirkwood-km_kirkwood.dts
--- a/arch/arm/boot/dts/kirkwood-km_kirkwood.dts	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-km_kirkwood.dts	2012-12-24 11:38:58.000000000 -0500
@@ -13,7 +13,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-lsxl.dtsi b/arch/arm/boot/dts/kirkwood-lsxl.dtsi
--- a/arch/arm/boot/dts/kirkwood-lsxl.dtsi	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-lsxl.dtsi	2012-12-24 11:39:36.000000000 -0500
@@ -3,7 +3,7 @@
 
 / {
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-mplcec4.dts b/arch/arm/boot/dts/kirkwood-mplcec4.dts
--- a/arch/arm/boot/dts/kirkwood-mplcec4.dts	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-mplcec4.dts	2012-12-24 11:39:56.000000000 -0500
@@ -13,7 +13,7 @@
         };
 
         chosen {
-                bootargs = "console=ttyS0,115200n8 earlyprintk";
+                bootargs = "";
         };
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-ns2-common.dtsi b/arch/arm/boot/dts/kirkwood-ns2-common.dtsi
--- a/arch/arm/boot/dts/kirkwood-ns2-common.dtsi	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-ns2-common.dtsi	2012-12-24 11:40:11.000000000 -0500
@@ -2,7 +2,7 @@
 
 / {
 	chosen {
-		bootargs = "console=ttyS0,115200n8";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-nsa310.dts b/arch/arm/boot/dts/kirkwood-nsa310.dts
--- a/arch/arm/boot/dts/kirkwood-nsa310.dts	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-nsa310.dts	2012-12-24 11:41:02.000000000 -0500
@@ -12,7 +12,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-openblocks_a6.dts b/arch/arm/boot/dts/kirkwood-openblocks_a6.dts
--- a/arch/arm/boot/dts/kirkwood-openblocks_a6.dts	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-openblocks_a6.dts	2012-12-24 11:41:20.000000000 -0500
@@ -13,7 +13,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-topkick.dts b/arch/arm/boot/dts/kirkwood-topkick.dts
--- a/arch/arm/boot/dts/kirkwood-topkick.dts	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-topkick.dts	2012-12-24 11:41:40.000000000 -0500
@@ -12,7 +12,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-ts219.dtsi b/arch/arm/boot/dts/kirkwood-ts219.dtsi
--- a/arch/arm/boot/dts/kirkwood-ts219.dtsi	2012-12-21 20:19:00.000000000 -0500
+++ b/arch/arm/boot/dts/kirkwood-ts219.dtsi	2012-12-24 11:42:09.000000000 -0500
@@ -10,7 +10,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
