diff -ruN a/arch/arm/boot/dts/kirkwood-dns320.dts b/arch/arm/boot/dts/kirkwood-dns320.dts
--- a/arch/arm/boot/dts/kirkwood-dns320.dts	2012-10-11 10:40:49.000000000 -0400
+++ b/arch/arm/boot/dts/kirkwood-dns320.dts	2012-10-11 10:44:31.000000000 -0400
@@ -12,7 +12,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	gpio-leds {
diff -ruN a/arch/arm/boot/dts/kirkwood-dns325.dts b/arch/arm/boot/dts/kirkwood-dns325.dts
--- a/arch/arm/boot/dts/kirkwood-dns325.dts	2012-10-11 10:40:49.000000000 -0400
+++ b/arch/arm/boot/dts/kirkwood-dns325.dts	2012-10-11 10:44:59.000000000 -0400
@@ -12,7 +12,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	gpio-leds {
diff -ruN a/arch/arm/boot/dts/kirkwood-dockstar.dts b/arch/arm/boot/dts/kirkwood-dockstar.dts
--- a/arch/arm/boot/dts/kirkwood-dockstar.dts	2012-10-11 10:40:49.000000000 -0400
+++ b/arch/arm/boot/dts/kirkwood-dockstar.dts	2012-10-11 10:46:05.000000000 -0400
@@ -12,7 +12,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk root=/dev/sda1 rootdelay=10";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-dreamplug.dts b/arch/arm/boot/dts/kirkwood-dreamplug.dts
--- a/arch/arm/boot/dts/kirkwood-dreamplug.dts	2012-10-11 10:40:49.000000000 -0400
+++ b/arch/arm/boot/dts/kirkwood-dreamplug.dts	2012-10-11 10:46:21.000000000 -0400
@@ -12,7 +12,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-goflexnet.dts b/arch/arm/boot/dts/kirkwood-goflexnet.dts
--- a/arch/arm/boot/dts/kirkwood-goflexnet.dts	2012-10-11 10:40:49.000000000 -0400
+++ b/arch/arm/boot/dts/kirkwood-goflexnet.dts	2012-10-11 10:46:44.000000000 -0400
@@ -12,7 +12,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk root=/dev/sda1 rootdelay=10";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-ib62x0.dts b/arch/arm/boot/dts/kirkwood-ib62x0.dts
--- a/arch/arm/boot/dts/kirkwood-ib62x0.dts	2012-10-11 10:40:49.000000000 -0400
+++ b/arch/arm/boot/dts/kirkwood-ib62x0.dts	2012-10-11 10:46:58.000000000 -0400
@@ -12,7 +12,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-iconnect.dts b/arch/arm/boot/dts/kirkwood-iconnect.dts
--- a/arch/arm/boot/dts/kirkwood-iconnect.dts	2012-10-11 10:40:49.000000000 -0400
+++ b/arch/arm/boot/dts/kirkwood-iconnect.dts	2012-10-11 10:47:18.000000000 -0400
@@ -12,7 +12,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 		linux,initrd-start = <0x4500040>;
 		linux,initrd-end   = <0x4800000>;
 	};
diff -ruN a/arch/arm/boot/dts/kirkwood-iomega_ix2_200.dts b/arch/arm/boot/dts/kirkwood-iomega_ix2_200.dts
--- a/arch/arm/boot/dts/kirkwood-iomega_ix2_200.dts	2012-10-11 10:40:49.000000000 -0400
+++ b/arch/arm/boot/dts/kirkwood-iomega_ix2_200.dts	2012-10-11 10:47:33.000000000 -0400
@@ -12,7 +12,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-km_kirkwood.dts b/arch/arm/boot/dts/kirkwood-km_kirkwood.dts
--- a/arch/arm/boot/dts/kirkwood-km_kirkwood.dts	2012-10-11 10:40:49.000000000 -0400
+++ b/arch/arm/boot/dts/kirkwood-km_kirkwood.dts	2012-10-11 10:47:49.000000000 -0400
@@ -12,7 +12,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-lsxl.dtsi b/arch/arm/boot/dts/kirkwood-lsxl.dtsi
--- a/arch/arm/boot/dts/kirkwood-lsxl.dtsi	2012-10-11 10:40:49.000000000 -0400
+++ b/arch/arm/boot/dts/kirkwood-lsxl.dtsi	2012-10-11 10:48:08.000000000 -0400
@@ -2,7 +2,7 @@
 
 / {
 	chosen {
-		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
diff -ruN a/arch/arm/boot/dts/kirkwood-ts219.dtsi b/arch/arm/boot/dts/kirkwood-ts219.dtsi
--- a/arch/arm/boot/dts/kirkwood-ts219.dtsi	2012-10-11 10:40:49.000000000 -0400
+++ b/arch/arm/boot/dts/kirkwood-ts219.dtsi	2012-10-11 10:48:27.000000000 -0400
@@ -10,7 +10,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n8";
+		bootargs = "";
 	};
 
 	ocp@f1000000 {
