diff -ruN a/mm/filemap.c b/mm/filemap.c
--- a/mm/filemap.c	2017-04-21 01:32:55.000000000 -0600
+++ b/mm/filemap.c	2017-04-23 14:10:10.140023733 -0600
@@ -1776,10 +1776,6 @@
 	unsigned int prev_offset;
 	int error = 0;
 
-	if (unlikely(*ppos >= inode->i_sb->s_maxbytes))
-		return 0;
-	iov_iter_truncate(iter, inode->i_sb->s_maxbytes);
-
 	index = *ppos >> PAGE_SHIFT;
 	prev_index = ra->prev_pos >> PAGE_SHIFT;
 	prev_offset = ra->prev_pos & (PAGE_SIZE-1);
