--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -114,6 +114,7 @@
 	kirkwood-openrd-client.dtb \
 	kirkwood-openrd-ultimate.dtb \
 	kirkwood-pogo_e02.dtb \
+	kirkwood-pogoplugv4.dtb \
 	kirkwood-rd88f6192.dtb \
 	kirkwood-rd88f6281-z0.dtb \
 	kirkwood-rd88f6281-a.dtb \

--- /dev/null
+++ b/arch/arm/boot/dts/kirkwood-pogoplugv4.dts
@@ -0,0 +1,162 @@
+/dts-v1/;
+
+#include "kirkwood.dtsi"
+#include "kirkwood-6281.dtsi"
+
+/ {
+    model = "CloudEngines Pogoplug V4";
+    compatible = "cloudengines,pogoplugv4", "marvell,kirkwood-88f6192", "marvell,kirkwood";
+
+    memory {
+        device_type = "memory";
+        reg = <0x00000000 0x8000000>;
+    };
+
+    chosen {
+        bootargs = "console=ttyS0,115200 earlyprintk root=/dev/mmcblk0p1 rootdelay=10";
+        stdout-path = &uart0;
+    };
+
+    mbus {
+        pcie-controller {
+            status = "okay";
+
+            pcie@1,0 {
+                status = "okay";
+            };
+        };
+    };
+
+    ocp@f1000000 {
+        pinctrl: pin-controller@10000 {
+            pmx_led_green: pmx-led-green {
+                marvell,pins = "mpp22";
+                marvell,function = "gpio";
+            };
+            pmx_led_red: pmx-led-red {
+                marvell,pins = "mpp24";
+                marvell,function = "gpio";
+            };
+            pmx_sdio_cd: pmx-sdio-cd {
+                marvell,pins = "mpp27";
+                marvell,function = "gpio";
+            };
+            pmx_sdio_wp: pmx-sdio-wp {
+                marvell,pins = "mpp28";
+                marvell,function = "gpio";
+            };
+            pmx_button_eject: pmx-button-eject {
+                marvell,pins = "mpp29";
+                marvell,function = "gpio";
+            };
+        };
+
+        spi@10600 {
+            status = "okay";
+
+            m25p05@0 {
+                compatible = "m25p05-nonjedec";
+                reg = <0>;
+                spi-max-frequency = <20000000>;
+                mode = <0>;
+            };
+        };
+
+        serial@12000 {
+            status = "ok";
+        };
+
+        sata@80000 {
+            status = "okay";
+            nr-ports = <1>;
+        };
+
+        i2c0: i2c@11000 {
+            status = "ok";
+        };
+
+        mvsdio@90000 {
+            pinctrl-0 = <&pmx_sdio &pmx_sdio_cd &pmx_sdio_wp>;
+            pinctrl-names = "default";
+            status = "okay";
+            cd-gpios = <&gpio0 27 1>;
+            wp-gpios = <&gpio0 28 1>;
+        };
+    };
+
+    gpio-leds {
+        compatible = "gpio-leds";
+        pinctrl-0 = <&pmx_led_green &pmx_led_red>;
+        pinctrl-names = "default";
+        health {
+            label = "status:green:health";
+            gpios = <&gpio0 22 1>;
+            linux,default-trigger = "default-on";
+        };
+        fault {
+            label = "status:red:fault";
+            gpios = <&gpio0 24 1>;
+            linux,default-trigger = "none";
+        };
+    };
+
+    gpio_keys {
+        compatible = "gpio-keys";
+        #address-cells = <1>;
+        #size-cells = <0>;
+        pinctrl-0 = <&pmx_button_eject>;
+        pinctrl-names = "default";
+
+        button@1 {
+            label = "Eject Button";
+            linux,code = <KEY_EJECTCD>;
+            gpios = <&gpio0 29 1>;
+        };
+    };
+};
+
+&mdio {
+    status = "okay";
+
+    ethphy0: ethernet-phy@0 {
+        device_type = "ethernet-phy";
+        reg = <0>;
+    };
+};
+
+&eth0 {
+    status = "okay";
+    ethernet0-port@0 {
+        phy-handle = <&ethphy0>;
+    };
+};
+
+&nand {
+    status = "okay";
+
+    partition@0 {
+        label = "u-boot";
+        reg = <0x0000000 0x200000>;
+        read-only;
+    };
+
+    partition@200000 {
+        label = "uImage";
+        reg = <0x200000 0x300000>;
+    };
+
+    partition@500000 {
+        label = "uImage2";
+        reg = <0x500000 0x300000>;
+    };
+
+    partition@800000 {
+        label = "failsafe";
+        reg = <0x800000 0x800000>;
+    };
+
+    partition@2500000 {
+        label = "root";
+        reg = <0x1000000 0x7000000>;
+    };
+};

