diff -urN a/clamd/fan-syscalllib.h b/clamd/fan-syscalllib.h
--- a/clamd/fan-syscalllib.h	2013-09-16 13:28:14.000000000 -0600
+++ b/clamd/fan-syscalllib.h	2013-09-19 21:29:47.314868930 -0600
@@ -10,6 +10,9 @@
 #elif defined(__i386__)
 # define __NR_fanotify_init	338
 # define __NR_fanotify_mark	339
+#elif defined(__ARM_EABI__)
+# define __NR_fanotify_init	367
+# define __NR_fanotify_mark	368
 #else
 # error "System call numbers not defined for this architecture"
 #endif
