diff -arU 0 old/etc/clamd.conf.sample new/etc/clamd.conf.sample
--- old/etc/clamd.conf.sample	2010-12-02 19:31:21.773357389 +0100
+++ new/etc/clamd.conf.sample	2010-12-02 19:31:45.873357389 +0100
@@ -14 +14 @@
-#LogFile /tmp/clamd.log
+LogFile /var/log/clamav/clamd.log
@@ -34 +34 @@
-#LogTime yes
+LogTime yes
@@ -61 +61 @@
-#PidFile /var/run/clamd.pid
+PidFile /run/clamav/clamd.pid
@@ -65 +65 @@
-#TemporaryDirectory /var/tmp
+TemporaryDirectory /tmp
@@ -80 +80 @@
-#LocalSocket /tmp/clamd.socket
+LocalSocket /var/lib/clamav/clamd.sock
@@ -190 +190 @@
-#User clamav
+User clamav
diff -arU 0 old/etc/freshclam.conf.sample new/etc/freshclam.conf.sample
--- old/etc/freshclam.conf.sample	2010-12-02 19:31:21.773357389 +0100
+++ new/etc/freshclam.conf.sample	2010-12-02 19:31:51.190024057 +0100
@@ -17 +17 @@
-#UpdateLogFile /var/log/freshclam.log
+UpdateLogFile /var/log/clamav/freshclam.log
@@ -121 +121 @@
-#NotifyClamd /path/to/clamd.conf
+NotifyClamd /etc/clamav/clamd.conf
