--- gcc/config/alpha/linux-elf.h.orig	2010-12-09 23:27:07.000000000 +1000
+++ gcc/config/alpha/linux-elf.h	2011-03-11 10:01:47.770000457 +1000
@@ -41,7 +41,7 @@
 
 #define ELF_DYNAMIC_LINKER	LINUX_DYNAMIC_LINKER
 
-#define LINK_SPEC "-m elf64alpha %{G*} %{relax:-relax}		\
+#define LINK_SPEC "-m elf64alpha --hash-style=both %{G*} %{relax:-relax}	\
   %{O*:-O3} %{!O*:-O1}						\
   %{shared:-shared}						\
   %{!shared:							\
--- gcc/config/i386/linux64.h.orig	2011-03-03 08:35:36.000000000 +1000
+++ gcc/config/i386/linux64.h	2011-03-11 10:01:47.770000457 +1000
@@ -78,7 +78,7 @@
  %{!mno-sse2avx:%{mavx:-msse2avx}} %{msse2avx:%{!mavx:-msse2avx}}"
 
 #undef	LINK_SPEC
-#define LINK_SPEC "%{" SPEC_64 ":-m elf_x86_64} %{" SPEC_32 ":-m elf_i386} \
+#define LINK_SPEC "%{" SPEC_64 ":-m elf_x86_64} %{" SPEC_32 ":-m elf_i386} --hash-style=both \
   %{shared:-shared} \
   %{!shared: \
     %{!static: \
--- gcc/config/i386/linux.h.orig	2011-01-15 04:45:06.000000000 +1000
+++ gcc/config/i386/linux.h	2011-03-11 10:01:47.770000457 +1000
@@ -104,7 +104,7 @@
   { "dynamic_linker", LINUX_DYNAMIC_LINKER }
 
 #undef	LINK_SPEC
-#define LINK_SPEC "-m %(link_emulation) %{shared:-shared} \
+#define LINK_SPEC "-m %(link_emulation) --hash-style=both %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
--- gcc/config/ia64/linux.h.orig	2010-12-09 23:27:07.000000000 +1000
+++ gcc/config/ia64/linux.h	2011-03-11 10:01:47.770000457 +1000
@@ -64,7 +64,7 @@
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-ia64.so.2"
 
 #undef LINK_SPEC
-#define LINK_SPEC "\
+#define LINK_SPEC "--hash-style=both \
   %{shared:-shared} \
   %{!shared: \
     %{!static: \
--- gcc/config/rs6000/linux64.h.orig	2011-02-11 03:30:10.000000000 +1000
+++ gcc/config/rs6000/linux64.h	2011-03-11 10:03:34.280000457 +1000
@@ -389,11 +389,11 @@
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKER64, UCLIBC_DYNAMIC_LINKER64)
 
 
-#define LINK_OS_LINUX_SPEC32 "-m elf32ppclinux %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC32 "-m elf32ppclinux --hash-style=both %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
   -dynamic-linker " LINUX_DYNAMIC_LINKER32 "}}"
 
-#define LINK_OS_LINUX_SPEC64 "-m elf64ppc %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC64 "-m elf64ppc --hash-style=both %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
   -dynamic-linker " LINUX_DYNAMIC_LINKER64 "}}"
 
--- gcc/config/rs6000/sysv4.h.orig	2011-01-28 04:36:03.000000000 +1000
+++ gcc/config/rs6000/sysv4.h	2011-03-11 10:01:47.773333792 +1000
@@ -830,7 +830,7 @@
 #define LINUX_DYNAMIC_LINKER \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKER, UCLIBC_DYNAMIC_LINKER)
 
-#define LINK_OS_LINUX_SPEC "-m elf32ppclinux %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC "-m elf32ppclinux --hash-style=both %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
   -dynamic-linker " LINUX_DYNAMIC_LINKER "}}"
 
--- gcc/config/s390/linux.h.orig	2010-12-09 23:27:07.000000000 +1000
+++ gcc/config/s390/linux.h	2011-03-11 10:01:47.770000457 +1000
@@ -77,7 +77,7 @@
 
 #undef  LINK_SPEC
 #define LINK_SPEC \
-  "%{m31:-m elf_s390}%{m64:-m elf64_s390} \
+  "%{m31:-m elf_s390}%{m64:-m elf64_s390} --hash-style=both \
    %{shared:-shared} \
    %{!shared: \
       %{static:-static} \
--- gcc/config/sparc/linux64.h.orig	2011-02-17 23:57:21.000000000 +1000
+++ gcc/config/sparc/linux64.h	2011-03-11 10:01:47.770000457 +1000
@@ -113,7 +113,7 @@
   { "link_arch_default", LINK_ARCH_DEFAULT_SPEC },	  \
   { "link_arch",	 LINK_ARCH_SPEC },
 
-#define LINK_ARCH32_SPEC "-m elf32_sparc -Y P,%R/usr/lib %{shared:-shared} \
+#define LINK_ARCH32_SPEC "-m elf32_sparc --hash-style=both -Y P,%R/usr/lib %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
@@ -121,7 +121,7 @@
       %{static:-static}} \
 "
 
-#define LINK_ARCH64_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_ARCH64_SPEC "-m elf64_sparc --hash-style=both -Y P,%R/usr/lib64 %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
@@ -193,7 +193,7 @@
 #else /* !SPARC_BI_ARCH */
 
 #undef LINK_SPEC
-#define LINK_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_SPEC "-m elf64_sparc --hash-style=both -Y P,%R/usr/lib64 %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
--- gcc/config/sparc/linux.h.orig	2011-01-27 06:30:12.000000000 +1000
+++ gcc/config/sparc/linux.h	2011-03-11 10:01:47.770000457 +1000
@@ -74,7 +74,7 @@
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
 
 #undef  LINK_SPEC
-#define LINK_SPEC "-m elf32_sparc -Y P,/usr/lib %{shared:-shared} \
+#define LINK_SPEC "-m elf32_sparc --hash-style=both -Y P,/usr/lib %{shared:-shared} \
   %{!mno-relax:%{!r:-relax}} \
   %{!shared: \
     %{!static: \
