diff -Naur a/gcc/d/d-incpath.cc b/gcc/d/d-incpath.cc
--- a/gcc/d/d-incpath.cc	2019-01-01 13:31:55.000000000 +0100
+++ b/gcc/d/d-incpath.cc	2019-06-28 08:32:00.326241502 +0200
@@ -140,7 +140,7 @@
 	    path = xstrdup (p->fname);
 
 	  /* Add D-specific suffix.  */
-	  path = concat (path, "/d", NULL);
+	  path = concat (path, "/dlang/gdc", NULL);
 
 	  /* Ignore duplicate entries.  */
 	  bool found = false;

 
