diff -Naur gtk+-2.16.4-orig/gdk-pixbuf/io-jpeg.c gtk+-2.16.4/gdk-pixbuf/io-jpeg.c
--- gtk+-2.16.4-orig/gdk-pixbuf/io-jpeg.c	2009-06-19 12:59:18.000000000 +1000
+++ gtk+-2.16.4/gdk-pixbuf/io-jpeg.c	2009-07-18 17:35:05.000000000 +1000
@@ -922,7 +922,7 @@
 				}
 			}
 			
-			for (cinfo->scale_denom = 2; cinfo->scale_denom <= 8; cinfo->scale_denom *= 2) {
+			for (cinfo->scale_denom = 2; cinfo->scale_denom <= 16; cinfo->scale_denom *= 2) {
 				jpeg_calc_output_dimensions (cinfo);
 				if (cinfo->output_width < width || cinfo->output_height < height) {
 					cinfo->scale_denom /= 2;
