--- doc/depmod.conf.sgml.orig	2011-06-25 10:07:15.897805024 -0400
+++ doc/depmod.conf.sgml	2011-06-25 10:07:48.810805020 -0400
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.1.2//EN"
+  "http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd" [
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
--- doc/depmod.sgml.orig	2011-06-25 10:07:15.898805024 -0400
+++ doc/depmod.sgml	2011-06-25 10:07:56.636805021 -0400
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.1.2//EN"
+  "http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd" [
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
@@ -52,7 +52,7 @@
       <arg><option>-n</option></arg>
       <arg><option>-v</option></arg>
       <arg><option>-P <replaceable>prefix</replaceable></option></arg>
-      <arg><option>-w</option><arg>
+      <arg><option>-w</option></arg>
       <arg><option><replaceable>version</replaceable></option></arg>
       <arg rep='repeat'><option><replaceable>filename</replaceable></option></arg>
     </cmdsynopsis>
--- doc/insmod.sgml.orig	2011-06-25 10:07:15.899805024 -0400
+++ doc/insmod.sgml	2011-06-25 10:07:59.228805019 -0400
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.1.2//EN"
+  "http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd" [
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
--- doc/lsmod.sgml.orig	2011-06-25 10:07:15.899805024 -0400
+++ doc/lsmod.sgml	2011-06-25 10:08:04.804805020 -0400
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.1.2//EN"
+  "http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd" [
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
--- doc/modinfo.sgml.orig	2011-06-25 10:07:15.900805024 -0400
+++ doc/modinfo.sgml	2011-06-25 10:08:07.630805019 -0400
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.1.2//EN"
+  "http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd" [
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
--- doc/modprobe.conf.sgml.orig	2011-06-25 10:07:15.901805024 -0400
+++ doc/modprobe.conf.sgml	2011-06-25 10:08:10.173805018 -0400
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.1.2//EN"
+  "http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd" [
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
--- doc/modprobe.sgml.orig	2011-06-25 10:07:15.902805024 -0400
+++ doc/modprobe.sgml	2011-06-25 10:08:13.623805019 -0400
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.1.2//EN"
+  "http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd" [
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
--- doc/modules.dep.sgml.orig	2011-06-25 10:07:15.903805024 -0400
+++ doc/modules.dep.sgml	2011-06-25 10:08:19.323805019 -0400
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.1.2//EN"
+  "http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd" [
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
--- doc/rmmod.sgml.orig	2011-06-25 10:07:15.904805024 -0400
+++ doc/rmmod.sgml	2011-06-25 10:08:27.691805017 -0400
@@ -1,4 +1,5 @@
-<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.1.2//EN"
+  "http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd" [
   <!ENTITY debian  "<productname>Debian GNU/Linux</productname>">
   <!ENTITY docbook "<productname>DocBook</productname>">
   <!ENTITY sgml    "<abbrev>SGML</abbrev>">
