--- Makefile~	2008-05-28 16:39:20.000000000 +0200
+++ Makefile	2008-05-28 16:38:27.000000000 +0200
@@ -23,16 +23,12 @@
 default:	$(GLIBNAME) $(OLIBNAME)
 
 install:	default
-	install -d $(AUTOLOAD_DIR)
-	cp $(GLIBNAME) $(AUTOLOAD_DIR)
-	cp $(OLIBNAME) $(AUTOLOAD_DIR)
-	test -d ${GRASSTABLES_DIR} || mkdir ${GRASSTABLES_DIR}
-	test -d ${GRASSTABLES_DIR}/etc || mkdir ${GRASSTABLES_DIR}/etc
-	cp /opt/grass/etc/ellipse.table ${GRASSTABLES_DIR}/etc
-	cp /opt/grass/etc/datum.table /opt/grass/etc/datumtransform.table ${GRASSTABLES_DIR}/etc
-	test -d ${GRASSTABLES_DIR}/driver || mkdir ${GRASSTABLES_DIR}/driver
-	test -d ${GRASSTABLES_DIR}/driver/db || mkdir ${GRASSTABLES_DIR}/driver/db
-	cp -r /opt/grass/driver/db/* ${GRASSTABLES_DIR}/driver/db/
+	install -d $(DESTDIR)/$(AUTOLOAD_DIR)
+	cp $(GLIBNAME) $(DESTDIR)/$(AUTOLOAD_DIR)
+	cp $(OLIBNAME) $(DESTDIR)/$(AUTOLOAD_DIR)
+	cp /opt/grass/etc/ellipse.table $(DESTDIR)/${GRASSTABLES_DIR}/etc
+	cp /opt/grass/etc/datum.table $(DESTDIR)/${GRASSTABLES_DIR}/etc
+	cp /opt/grass/driver/db/* $(DESTDIR)/${GRASSTABLES_DIR}/driver/db/
 
 clean:
 	rm -f $(OLIBNAME) $(GLIBNAME) *.o 
