--- a/src/greeterbackground.c
+++ b/src/greeterbackground.c
@@ -1603,6 +1603,8 @@
     return surface;
 }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-align"
 /* Sets the "ESETROOT_PMAP_ID" property to later be used to free the pixmap */
 static void
 set_root_pixmap_id(GdkScreen* screen,
@@ -1669,6 +1671,7 @@
     XChangeProperty (display, xroot, atoms[1], XA_PIXMAP, 32,
                      PropModeReplace, (unsigned char *) &xpixmap, 1);
 }
+#pragma GCC diagnostic pop
 
 /**
 * set_surface_as_root:

