--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -2560,6 +2560,46 @@ static const struct panel_desc arm_rtsm = {
 	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
 };

+static const struct drm_display_mode clockwork_cpi3_lcd_mode = {
+	.clock = 5800,
+	.hdisplay = 320,
+	.hsync_start = 320 + 6,
+	.hsync_end = 320 + 6 + 2,
+	.htotal = 320 + 6 + 2 + 60,
+	.vdisplay = 240,
+	.vsync_start = 240 + 2,
+	.vsync_end = 240 + 2 + 2,
+	.vtotal = 240 + 2 + 2 + 6,
+	.vrefresh = 60,
+	.flags = DRM_MODE_FLAG_PVSYNC | DRM_MODE_FLAG_PHSYNC,
+};
+
+static const struct panel_desc clockwork_cpi3_lcd = {
+	.modes = &clockwork_cpi3_lcd_mode,
+	.num_modes = 1,
+	.bpc = 8,
+};
+
+static const struct drm_display_mode clockwork_cpi3_hdmi_mode = {
+	.clock = 74250,
+	.hdisplay = 1280,
+	.hsync_start = 1280 + 110,
+	.hsync_end = 1280 + 110 + 40,
+	.htotal = 1280 + 110 + 40 + 220,
+	.vdisplay = 720,
+	.vsync_start = 720 + 5,
+	.vsync_end = 720 + 5 + 5,
+	.vtotal = 720 + 5 + 5 + 20,
+	.vrefresh = 60,
+	.flags = DRM_MODE_FLAG_PVSYNC | DRM_MODE_FLAG_PHSYNC,
+};
+
+static const struct panel_desc clockwork_cpi3_hdmi = {
+	.modes = &clockwork_cpi3_hdmi_mode,
+	.num_modes = 1,
+	.bpc = 8,
+};
+
 static const struct of_device_id platform_of_match[] = {
 	{
 		.compatible = "ampire,am-480272h3tmqw-t01h",
@@ -2837,6 +2877,12 @@ static const struct of_device_id platform_of_match[] = {
 	}, {
 		.compatible = "winstar,wf35ltiacd",
 		.data = &winstar_wf35ltiacd,
+	}, {
+		.compatible = "clockwork,cpi3-lcd",
+		.data = &clockwork_cpi3_lcd,
+	}, {
+		.compatible = "clockwork,cpi3-hdmi",
+		.data = &clockwork_cpi3_hdmi,
 	}, {
 		/* sentinel */
 	}
