--- gresolver-0.0.5/gresolver.desktop	2005-07-04 08:24:33.000000000 -0700
+++ gresolver-0.0.5-new/gresolver.desktop	2006-01-30 17:00:03.000000000 -0800
@@ -1,15 +1,12 @@
 [Desktop Entry]
 Encoding=UTF-8
-Version=1.0
 Type=Application
 Exec=gresolver
-TryExec=
-X-GNOME-DocPath=
 Terminal=false
 StartupNotify=true
 Name=DNS Query Tool
 Name[en_GB]=DNS Query Tool
 Comment=Perform advanced DNS queries
 Comment[en_GB]=Perform advanced DNS queries
-Icon=gresolver
-Categories=Categories=Application;Network;
+Icon=gresolver.png
+Categories=Application;Network;
