diff -rup portmidi.orig/pm_java/CMakeLists.txt portmidi/pm_java/CMakeLists.txt
--- portmidi.orig/pm_java/CMakeLists.txt	2009-11-04 16:20:44.000000000 +0100
+++ portmidi/pm_java/CMakeLists.txt	2012-01-24 21:13:08.513182169 +0100
@@ -15,13 +15,11 @@ if(UNIX)
     add_custom_command(OUTPUT pmdefaults/PmDefaultsFrame.class
         COMMAND javac -classpath . pmdefaults/PmDefaultsFrame.java
 	MAIN_DEPENDENCY pmdefaults/PmDefaultsFrame.java
-	DEPENDS pmdefaults/PmDefaults.java
-	WORKING_DIRECTORY pm_java)
+	DEPENDS pmdefaults/PmDefaults.java)
     add_custom_command(OUTPUT pmdefaults/PmDefaults.class
         COMMAND javac -classpath . pmdefaults/PmDefaults.java
 	MAIN_DEPENDENCY pmdefaults/PmDefaults.java
-	DEPENDS pmdefaults/PmDefaultsFrame.java
-	WORKING_DIRECTORY pm_java)
+	DEPENDS pmdefaults/PmDefaultsFrame.java)
     add_custom_command(OUTPUT ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/pmdefaults.jar
         COMMAND	cp pmdefaults/portmusic_logo.png .
         COMMAND	jar cmf pmdefaults/manifest.txt pmdefaults.jar
@@ -31,8 +29,7 @@ if(UNIX)
 	COMMAND mv pmdefaults.jar ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
 	COMMAND rm portmusic_logo.png
 	MAIN_DEPENDENCY pmdefaults/PmDefaults.class
-	DEPENDS ${PMDEFAULTS_ALL_CLASSES}
-	WORKING_DIRECTORY pm_java)
+	DEPENDS ${PMDEFAULTS_ALL_CLASSES})
     add_custom_target(pmdefaults_target ALL 
         DEPENDS ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/pmdefaults.jar)
     # message(STATUS "add_custom_target: pmdefaults.jar")
