--- a/SConstruct	2013-03-16 02:59:55.000000000 +0800
+++ b/SConstruct	2013-03-20 10:55:20.009645869 +0800
@@ -692,7 +692,7 @@
                          "-Winvalid-pch"] )
     # env.Append( " -Wconversion" ) TODO: this doesn't really work yet
     if linux:
-        env.Append( CCFLAGS=["-Werror", "-pipe"] )
+        env.Append( CCFLAGS=["-pipe"] )
         if not has_option('clang'):
             env.Append( CCFLAGS=["-fno-builtin-memcmp"] ) # glibc's memcmp is faster than gcc's
 
--- a/src/third_party/v8/SConscript	2013-04-17 03:21:23.000000000 +0800
+++ b/src/third_party/v8/SConscript	2013-04-18 17:41:29.878618892 +0800
@@ -47,7 +47,6 @@
   'gcc': {
     'all': {
       'CCFLAGS':      ['-Wall',
-                       '-Werror',
                        '-W',
                        '-Wno-unused-parameter',
                        '-Woverloaded-virtual',
