diff --git a/src/SConscript.client b/src/SConscript.client
index 7a6bdc9..3fb55e5 100644
--- a/src/SConscript.client
+++ b/src/SConscript.client
@@ -134,7 +134,7 @@ env.Install(
 # install
 prefix = GetOption("prefix")
 
-env.Install(prefix + "/lib", '${LIBPREFIX}mongoclient${LIBSUFFIX}')
+env.InstallAs(prefix + "/lib", '${LIBPREFIX}mongoclient${LIBSUFFIX}')
 
 for x in clientHeaderDirectories:
     env.Install(prefix + "/include/mongo/" + x,
