diff -uprw gst-plugins-bad.orig/ext/modplug/gstmodplug.cc gst-plugins-bad/ext/modplug/gstmodplug.cc
--- gst-plugins-bad.orig/ext/modplug/gstmodplug.cc	2015-04-21 03:15:25.929456272 +0300
+++ gst-plugins-bad/ext/modplug/gstmodplug.cc	2015-04-21 03:18:54.624976723 +0300
@@ -50,7 +50,7 @@
 #define WORDS_BIGENDIAN 0
 #endif
 
-#include <stdafx.h>
+#include <libmodplug/stdafx.h>
 #include <libmodplug/sndfile.h>
 
 #include "gstmodplug.h"
