diff -Naur scim-hangul-0.3.2-orig/src/scim_hangul_imengine.cpp scim-hangul-0.3.2/src/scim_hangul_imengine.cpp
--- scim-hangul-0.3.2-orig/src/scim_hangul_imengine.cpp	2008-03-26 09:33:43.000000000 +0100
+++ scim-hangul-0.3.2/src/scim_hangul_imengine.cpp	2008-03-26 09:34:41.000000000 +0100
@@ -33,6 +33,7 @@
   #include <config.h>
 #endif
 
+#include <cstring>
 #include <scim.h>
 #include "scim_hangul_imengine.h"
 
diff -Naur scim-hangul-0.3.2-orig/src/scim_hangul_imengine_setup.cpp scim-hangul-0.3.2/src/scim_hangul_imengine_setup.cpp
--- scim-hangul-0.3.2-orig/src/scim_hangul_imengine_setup.cpp	2008-03-26 09:33:43.000000000 +0100
+++ scim-hangul-0.3.2/src/scim_hangul_imengine_setup.cpp	2008-03-26 09:34:53.000000000 +0100
@@ -34,6 +34,7 @@
   #include <config.h>
 #endif
 
+#include <cstring>
 #include <scim.h>
 #include <gtk/scimkeyselection.h>
 
