diff -aur codeblocks-8.02.orig/src/src/prefix.cpp codeblocks-8.02.new/src/src/prefix.cpp
--- codeblocks-8.02.orig/src/src/prefix.cpp	2009-07-27 22:53:45.000000000 +0200
+++ codeblocks-8.02.new/src/src/prefix.cpp	2009-07-27 22:56:24.000000000 +0200
@@ -389,7 +389,8 @@
 char *
 br_extract_dir (const char *path)
 {
-	char *end, *result;
+  const char *end;
+  char *result;
 
 	br_return_val_if_fail (path != (char *) NULL, (char *) NULL);
 
@@ -424,7 +425,8 @@
 char *
 br_extract_prefix (const char *path)
 {
-	char *end, *tmp, *result;
+  const char *end;
+  char *tmp, *result;
 
 	br_return_val_if_fail (path != (char *) NULL, (char *) NULL);
 
