http://bugs.gentoo.org/209036

fix building with gcc-4.3

--- src/recodext.h	2008-04-03 18:35:42 +0000
+++ src/recodext.h	2008-04-03 18:36:32 +0000
@@ -215,10 +215,10 @@
     struct recode_single *unsurfacer;
 
     /* Non zero if this is an acceptable charset (not only a surface).  */
-    enum recode_symbol_type type : 3;
+    enum recode_symbol_type type : 2;
 
     /* Non zero if this one should be ignored.  */
-    bool ignore : 2;
+    bool ignore : 1;
   };
 
 struct recode_surface_list

