--- lftp-3.7.14/configure.ac	2009-05-15 08:22:34.000000000 +0200
+++ lftp.configure.ac	2009-06-13 12:58:39.000000000 +0200
@@ -266,9 +266,10 @@
 			with_gnutls=no;;
 esac
 if test x$with_gnutls = xyes; then
-   AM_PATH_LIBGNUTLS([1.0.0], [
+   PKG_PROG_PKG_CONFIG
+   PKG_CHECK_MODULES([LIBGNUTLS], [gnutls >= 1.0.0], [
       AC_DEFINE([USE_GNUTLS], 1, [Define to 1 when using GNU TLS library])
-      gnutls_version_code=`echo $libgnutls_config_version | $AWK -F. '{ printf "0x%02X%02X%02X\n",$1,$2,$3 }'`
+      gnutls_version_code=`$PKG_CONFIG --modversion gnutls | $AWK -F. '{ printf "0x%02X%02X%02X\n",$1,$2,$3 }'`
       AC_DEFINE_UNQUOTED([LFTP_LIBGNUTLS_VERSION_CODE], $gnutls_version_code, [Define to libgnutls version, e.g. 0x010203 for 1.2.3])
    ])
 fi

