diff -Naur blassic-0.10.0-old/blassic.cpp blassic-0.10.0/blassic.cpp
--- blassic-0.10.0-old/blassic.cpp	2005-02-08 01:22:06.000000000 +1000
+++ blassic-0.10.0/blassic.cpp	2008-08-03 12:20:03.000000000 +1000
@@ -17,6 +17,7 @@
 #include "error.h"
 
 #include <string>
+#include <cstring>
 
 #include <iostream>
 #include <cctype>
diff -Naur blassic-0.10.0-old/cursor.cpp blassic-0.10.0/cursor.cpp
--- blassic-0.10.0-old/cursor.cpp	2005-02-08 02:58:37.000000000 +1000
+++ blassic-0.10.0/cursor.cpp	2008-08-03 12:22:15.000000000 +1000
@@ -19,7 +19,7 @@
 #include <map>
 #include <queue>
 #include <sstream>
-
+#include <cstring>
 
 #ifdef BLASSIC_USE_WINDOWS
 
diff -Naur blassic-0.10.0-old/graphics.cpp blassic-0.10.0/graphics.cpp
--- blassic-0.10.0-old/graphics.cpp	2005-02-08 01:24:07.000000000 +1000
+++ blassic-0.10.0/graphics.cpp	2008-08-03 12:25:27.000000000 +1000
@@ -19,6 +19,7 @@
 #include "trace.h"
 
 #include <string>
+#include <cstring>
 #include <vector>
 #include <algorithm>
 #include <sstream>
diff -Naur blassic-0.10.0-old/memory.cpp blassic-0.10.0/memory.cpp
--- blassic-0.10.0-old/memory.cpp	2005-02-02 01:45:04.000000000 +1000
+++ blassic-0.10.0/memory.cpp	2008-08-03 12:27:02.000000000 +1000
@@ -7,6 +7,8 @@
 #include "error.h"
 #include "trace.h"
 
+#include <cstring>
+
 //#ifdef HAVE_SYS_MMAN_H
 #ifdef HAVE_MMAP
 
diff -Naur blassic-0.10.0-old/program.cpp blassic-0.10.0/program.cpp
--- blassic-0.10.0-old/program.cpp	2005-02-09 01:00:54.000000000 +1000
+++ blassic-0.10.0/program.cpp	2008-08-03 12:28:00.000000000 +1000
@@ -11,6 +11,7 @@
 #include "trace.h"
 
 #include <string>
+#include <cstring>
 #include <iostream>
 #include <fstream>
 #include <iomanip>
diff -Naur blassic-0.10.0-old/runnerline_instructions.cpp blassic-0.10.0/runnerline_instructions.cpp
--- blassic-0.10.0-old/runnerline_instructions.cpp	2005-02-08 01:40:33.000000000 +1000
+++ blassic-0.10.0/runnerline_instructions.cpp	2008-08-03 12:30:26.000000000 +1000
@@ -16,6 +16,8 @@
 
 #include "trace.h"
 
+#include <cstring>
+
 #include <memory>
 using std::auto_ptr;
 
diff -Naur blassic-0.10.0-old/socket.cpp blassic-0.10.0/socket.cpp
--- blassic-0.10.0-old/socket.cpp	2005-02-10 04:59:12.000000000 +1000
+++ blassic-0.10.0/socket.cpp	2008-08-03 12:31:45.000000000 +1000
@@ -3,6 +3,8 @@
 
 #include "socket.h"
 
+#include <cstring>
+
 //------------------------------------------------
 // Changed this: now do not use winsock in Cygwin.
 //------------------------------------------------
diff -Naur blassic-0.10.0-old/util.h blassic-0.10.0/util.h
--- blassic-0.10.0-old/util.h	2005-02-03 02:51:14.000000000 +1000
+++ blassic-0.10.0/util.h	2008-08-03 12:24:10.000000000 +1000
@@ -8,6 +8,7 @@
 #include <sstream>
 #include <stdexcept>
 #include <iostream>
+#include <cstdlib>
 
 #ifdef __BORLANDC__
 #pragma warn -8027
