diff -x .cvsignore -x .gitignore -ruN xf86-input-tek4957-4a39872189c42c4fe99aecce2e7b710ce18afb61/README xf86-input-tek4957-ea5d265dc0e73796ad600aa96c5e0ea970a57bb2/README
--- xf86-input-tek4957-4a39872189c42c4fe99aecce2e7b710ce18afb61/README	1970-01-01 00:00:00.000000000 +0000
+++ xf86-input-tek4957-ea5d265dc0e73796ad600aa96c5e0ea970a57bb2/README	2009-01-31 04:29:42.000000000 +0000
@@ -0,0 +1,20 @@
+xf86-input-tek4957 - Tektronix 4957 input driver for the Xorg X server
+
+Please submit bugs & patches to the Xorg bugzilla:
+
+        https://bugs.freedesktop.org/enter_bug.cgi?product=xorg
+
+All questions regarding this software should be directed at the
+Xorg mailing list:
+
+        http://lists.freedesktop.org/mailman/listinfo/xorg
+
+The master development code repository can be found at:
+
+        git://anongit.freedesktop.org/git/xorg/driver/xf86-input-tek4957
+
+        http://cgit.freedesktop.org/xorg/driver/xf86-input-tek4957
+
+For more information on the git code manager, see:
+
+        http://wiki.x.org/wiki/GitPage
diff -x .cvsignore -x .gitignore -ruN xf86-input-tek4957-4a39872189c42c4fe99aecce2e7b710ce18afb61/man/tek4957.man xf86-input-tek4957-ea5d265dc0e73796ad600aa96c5e0ea970a57bb2/man/tek4957.man
--- xf86-input-tek4957-4a39872189c42c4fe99aecce2e7b710ce18afb61/man/tek4957.man	2008-03-20 20:48:45.000000000 +0000
+++ xf86-input-tek4957-ea5d265dc0e73796ad600aa96c5e0ea970a57bb2/man/tek4957.man	2009-01-31 04:29:42.000000000 +0000
@@ -80,6 +80,6 @@
 .SH "BUGS / LIMITATIONS"
 Currently, only "Absolute" mode is supported ( Sorry )
 .SH "SEE ALSO"
-__xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), xorgconfig(__appmansuffix__), Xserver(__appmansuffix__), X(__miscmansuffix__).
+__xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), Xserver(__appmansuffix__), X(__miscmansuffix__).
 .SH AUTHORS
 Olivier DANET <odanet@caramail.com>
diff -x .cvsignore -x .gitignore -ruN xf86-input-tek4957-4a39872189c42c4fe99aecce2e7b710ce18afb61/src/xf86Tek4957.c xf86-input-tek4957-ea5d265dc0e73796ad600aa96c5e0ea970a57bb2/src/xf86Tek4957.c
--- xf86-input-tek4957-4a39872189c42c4fe99aecce2e7b710ce18afb61/src/xf86Tek4957.c	2008-03-20 20:48:45.000000000 +0000
+++ xf86-input-tek4957-ea5d265dc0e73796ad600aa96c5e0ea970a57bb2/src/xf86Tek4957.c	2009-01-31 04:29:42.000000000 +0000
@@ -152,7 +152,11 @@
 
     /* Gets current screen size. It can change dynamically in the case of
     a multi-head configuration with different screen sizes ... */
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) == 0
     SP=miPointerCurrentScreen();
+#else
+    SP=miPointerGetScreen(local->dev);
+#endif
     W=SP->width;
     H=SP->height;
 
@@ -446,7 +450,10 @@
 		return !Success;
 	    }
 
-	    if (InitValuatorClassDeviceStruct(pDev,2,xf86GetMotionEvents,
+	    if (InitValuatorClassDeviceStruct(pDev,2,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
+			xf86GetMotionEvents,
+#endif
 		   local->history_size,Absolute)== FALSE) {
 		xf86Msg(X_ERROR,"Tek4957:Unable to allocate Valuator class device\n");
 		return !Success;
