diff --git a/src/swig/python/build b/src/swig/python/build
index 9b86703c..1c51399f 100755
--- a/src/swig/python/build
+++ b/src/swig/python/build
@@ -25,7 +25,7 @@ then
 	${CXX} -fPIC -D_GNU_SOURCE ${CXXFLAGS} -c -I../.. $PYTHON_INCLUDE mlt_wrap.cxx || exit $?
 
 	# Create the module
-	${CXX} ${CXXFLAGS} -shared mlt_wrap.o -L../../mlt++ -lmlt++ -L../../framework -lmlt $("${PYTHON}-config" --ldflags) -o _mlt.so || exit $?
+	${CXX} ${CXXFLAGS} -shared mlt_wrap.o -L../../../../build/src/mlt++ -lmlt++ -L../../../../build/src/framework -lmlt ${LDFLAGS} $("${PYTHON}-config" --ldflags) -o _mlt.so || exit $?
 else
 	echo Python not installed.
 	exit 1
