diff --git a/src/melt/CMakeLists.txt b/src/melt/CMakeLists.txt
index b79afaf8..e8402276 100644
--- a/src/melt/CMakeLists.txt
+++ b/src/melt/CMakeLists.txt
@@ -1,11 +1,7 @@
 set(melt_lib mlt Threads::Threads)
 find_package(SDL2 QUIET)
 if(SDL2_FOUND)
-    add_library(sdl2 SHARED IMPORTED GLOBAL)
-    set_target_properties(sdl2 PROPERTIES
-        INTERFACE_INCLUDE_DIRECTORIES ${SDL2_INCLUDE_DIRS}
-        IMPORTED_LOCATION ${libdir}/libSDL2.so
-    )
+    add_library(sdl2 ALIAS SDL2::SDL2)
 else()
     pkg_check_modules(sdl2 IMPORTED_TARGET GLOBAL sdl2)
     if(TARGET PkgConfig::sdl2)
