From 9c326a34e088971a3c11b2e5289cd1a8e4d439df Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tapani=20P=C3=A4lli?= <tapani.palli@intel.com>
Date: Wed, 30 Apr 2014 14:05:11 +0300
Subject: [PATCH] drawtex: resolve glDrawTexfOES extension function
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=78101
Signed-off-by: Tapani Pälli <tapani.palli@intel.com>
Reviewed-by: Brian Paul <brianp@vmware.com>
Reviewed-by: Ian Romanick <ian.d.romanick@intel.com>
---
 src/egl/opengles1/drawtex.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/egl/opengles1/drawtex.c b/src/egl/opengles1/drawtex.c
index 524f931..1ba8ac7 100644
--- a/src/egl/opengles1/drawtex.c
+++ b/src/egl/opengles1/drawtex.c
@@ -25,13 +25,14 @@ static GLfloat width = 200, height = 200;
 static GLboolean animate = GL_FALSE;
 static int win;
 
+static PFNGLDRAWTEXFOESPROC glDrawTexfOES_func = NULL;
 
 static void
 draw(void)
 {
    glClear(GL_COLOR_BUFFER_BIT);
 
-   glDrawTexfOES(view_posx, view_posy, 0.0, width, height);
+   glDrawTexfOES_func(view_posx, view_posy, 0.0, width, height);
 }
 
 
@@ -128,6 +129,13 @@ init(void)
       exit(1);
    }
 
+   glDrawTexfOES_func = eglGetProcAddress("glDrawTexfOES");
+
+   if (!glDrawTexfOES_func) {
+      fprintf(stderr, "Sorry, failed to resolve glDrawTexfOES function\n");
+      exit(1);
+   }
+
    glClearColor(0.4, 0.4, 0.4, 0.0);
 
    make_smile_texture();
-- 
1.9.2

From 6ad8820c457845300806a2f9469599460ca96af9 Mon Sep 17 00:00:00 2001
From: Stefan Dirsch <sndirsch@suse.de>
Date: Thu, 8 May 2014 11:48:44 +0200
Subject: [PATCH] drawtex: fixed compiler warnings related to use of
 eglGetProcAddress
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add proper <EGL/egl.h> include and cast in order to fix compiler
warnings related to the use of eglGetProcAddress.

Reviewed-By: Tapani Pälli <tapani.palli@intel.com>
---
 src/egl/opengles1/drawtex.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/egl/opengles1/drawtex.c b/src/egl/opengles1/drawtex.c
index 1ba8ac7..dc1de13 100644
--- a/src/egl/opengles1/drawtex.c
+++ b/src/egl/opengles1/drawtex.c
@@ -16,6 +16,7 @@
 #include <string.h>
 #include <GLES/gl.h>
 #include <GLES/glext.h>
+#include <EGL/egl.h>
 
 #include "eglut.h"
 
@@ -129,7 +130,7 @@ init(void)
       exit(1);
    }
 
-   glDrawTexfOES_func = eglGetProcAddress("glDrawTexfOES");
+   glDrawTexfOES_func = (PFNGLDRAWTEXFOESPROC) eglGetProcAddress("glDrawTexfOES");
 
    if (!glDrawTexfOES_func) {
       fprintf(stderr, "Sorry, failed to resolve glDrawTexfOES function\n");
-- 
1.9.2

