diff -ru lensfun-0.2.3.orig/build/mak/compiler/gcc.mak lensfun-0.2.3/build/mak/compiler/gcc.mak
--- lensfun-0.2.3.orig/build/mak/compiler/gcc.mak	2009-03-15 02:42:40.000000000 +0100
+++ lensfun-0.2.3/build/mak/compiler/gcc.mak	2009-03-15 02:43:37.000000000 +0100
@@ -71,7 +71,7 @@
 endef
 
 LINK.GCC.AR = $(GCC.AR) $(GCC.ARFLAGS) $@ $^
-LINK.GCC.EXEC = $(GCC.LD) -o $@ $(GCC.LDFLAGS) $(LDFLAGS) $1 $^ $(GCC.LDFLAGS.LIBS) $(LDFLAGS.LIBS) $2
+LINK.GCC.EXEC = $(GCC.LD) $(LDFLAGS) $^ $2 $(GCC.LDFLAGS.LIBS) $(LDFLAGS.LIBS) $(GCC.LDFLAGS) $1 -o $@
 define LINK.GCC.SO
 	$(GCC.LD) -o $@.$(SHARED.$3) -Wl,"-soname=$(notdir $@).$(basename $(basename $(SHARED.$3)))" $(GCC.LDFLAGS.SHARED) $(GCC.LDFLAGS) $(LDFLAGS) $1 $^ $(GCC.LDFLAGS.LIBS) $(LDFLAGS.LIBS) $2
 	ln -fs $(notdir $@.$(SHARED.$3)) $@.$(basename $(basename $(SHARED.$3)))
Only in lensfun-0.2.3/out/posix/release: deps
