From 36203d048e32f6f32937581e4d3676d23836c0a6 Mon Sep 17 00:00:00 2001
From: Alexander Holler <holler@ahsoftware.de>
Date: Thu, 4 Feb 2010 05:42:09 +0100
Subject: [PATCH 01/14] DockStar: Change RAM definitions to one bank (128 MB).

---
 board/Marvell/sheevaplug/kwbimage.cfg |   10 ++++------
 1 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/board/Marvell/sheevaplug/kwbimage.cfg b/board/Marvell/sheevaplug/kwbimage.cfg
index 3b9c53f..c8c7efe 100644
--- a/board/Marvell/sheevaplug/kwbimage.cfg
+++ b/board/Marvell/sheevaplug/kwbimage.cfg
@@ -74,11 +74,11 @@ DATA 0xFFD0140C 0x00000a33	#  DDR Timing (High)
 # bit12-11: TW2W
 # bit31-13: zero required
 
-DATA 0xFFD01410 0x000000cc	#  DDR Address Control
+DATA 0xFFD01410 0x0000000d	#  DDR Address Control
 # bit1-0:   00, Cs0width=x8
 # bit3-2:   11, Cs0size=1Gb
-# bit5-4:   00, Cs1width=x8
-# bit7-6:   11, Cs1size=1Gb
+# bit5-4:   00, Cs1width=nonexistent
+# bit7-6:   00, Cs1size =nonexistent
 # bit9-8:   00, Cs2width=nonexistent
 # bit11-10: 00, Cs2size =nonexistent
 # bit13-12: 00, Cs3width=nonexistent
@@ -142,9 +142,7 @@ DATA 0xFFD01504 0x0FFFFFF1	# CS[0]n Size
 # bit23-4: ones, required
 # bit31-24: 0x0F, Size (i.e. 256MB)
 
-DATA 0xFFD01508 0x10000000	# CS[1]n Base address to 256Mb
-DATA 0xFFD0150C 0x0FFFFFF5	# CS[1]n Size 256Mb Window enabled for CS1
-
+DATA 0xFFD0150C 0x00000000	# CS[1]n Size, window disabled
 DATA 0xFFD01514 0x00000000	# CS[2]n Size, window disabled
 DATA 0xFFD0151C 0x00000000	# CS[3]n Size, window disabled
 
-- 
1.6.4.4

