From ee152f0e659b69f46e52339c65996d5a0addfa4b Mon Sep 17 00:00:00 2001
From: Alexander Holler <holler@ahsoftware.de>
Date: Thu, 4 Feb 2010 06:29:43 +0100
Subject: [PATCH 03/14] DockStar: MTD partitions.

---
 include/configs/sheevaplug.h |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/configs/sheevaplug.h b/include/configs/sheevaplug.h
index e020cb0..cccd142 100644
--- a/include/configs/sheevaplug.h
+++ b/include/configs/sheevaplug.h
@@ -133,8 +133,8 @@
 	"setenv bootargs ${x_bootargs} ${x_bootargs_root}; "	\
 	"${x_bootcmd_usb}; bootm 0x6400000;"
 
-#define CONFIG_MTDPARTS		"orion_nand:512k(uboot),"	\
-	"3m@1m(kernel),1m@4m(psm),13m@5m(rootfs) rw\0"
+#define CONFIG_MTDPARTS		"orion_nand:1m(u-boot),"	\
+	"4m@1m(uImage),32m@5m(root),219m@37m(data) rw\0"
 
 #define CONFIG_EXTRA_ENV_SETTINGS	"x_bootargs=console"	\
 	"=ttyS0,115200 mtdparts="CONFIG_MTDPARTS	\
-- 
1.6.4.4

