From 854ad4382cfcb24a5f9225f19f924085305a71ce Mon Sep 17 00:00:00 2001
From: Alexander Holler <holler@ahsoftware.de>
Date: Thu, 4 Feb 2010 06:16:03 +0100
Subject: [PATCH 05/14] DockStar: Disable environment in NAND. We don't want to change that.

---
 include/configs/sheevaplug.h |    5 -----
 1 files changed, 0 insertions(+), 5 deletions(-)

diff --git a/include/configs/sheevaplug.h b/include/configs/sheevaplug.h
index ecb794c..98c86b7 100644
--- a/include/configs/sheevaplug.h
+++ b/include/configs/sheevaplug.h
@@ -112,12 +112,7 @@
 /*
  *  Environment variables configurations
  */
-#ifdef CONFIG_CMD_NAND
-#define CONFIG_ENV_IS_IN_NAND		1
-#define CONFIG_ENV_SECT_SIZE		0x20000	/* 128K */
-#else
 #define CONFIG_ENV_IS_NOWHERE		1	/* if env in SDRAM */
-#endif
 /*
  * max 4k env size is enough, but in case of nand
  * it has to be rounded to sector size
-- 
1.6.4.4

