From 21b28944090f61500787e6087e65d4d8c4a291ee Mon Sep 17 00:00:00 2001
From: Alexander Holler <holler@ahsoftware.de>
Date: Tue, 16 Feb 2010 21:11:57 +0100
Subject: [PATCH 07/14] DockStar: Include long help messages.

---
 include/configs/sheevaplug.h |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/include/configs/sheevaplug.h b/include/configs/sheevaplug.h
index 6b56652..4861f3a 100644
--- a/include/configs/sheevaplug.h
+++ b/include/configs/sheevaplug.h
@@ -98,6 +98,7 @@
 #define CONFIG_CMD_NAND
 #define CONFIG_CMD_PING
 #define CONFIG_CMD_USB
+#define CONFIG_SYS_LONGHELP
 
 /*
  * NAND configuration
-- 
1.6.4.4

