From bca0f450beeecc38fce940a1207d7b6ec7602dda Mon Sep 17 00:00:00 2001
From: Alexander Holler <holler@ahsoftware.de>
Date: Wed, 17 Feb 2010 13:40:14 +0100
Subject: [PATCH 08/14] DockStar: Change TEXT_BASE from 0x600000 to 0xc00000 for chainloading U-Boot.

---
 board/Marvell/sheevaplug/config.mk |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/board/Marvell/sheevaplug/config.mk b/board/Marvell/sheevaplug/config.mk
index 2bd9f79..786dc17 100644
--- a/board/Marvell/sheevaplug/config.mk
+++ b/board/Marvell/sheevaplug/config.mk
@@ -22,7 +22,7 @@
 # MA 02110-1301 USA
 #
 
-TEXT_BASE = 0x00600000
+TEXT_BASE = 0x00c00000
 
 # Kirkwood Boot Image configuration file
 KWD_CONFIG = $(SRCTREE)/board/$(BOARDDIR)/kwbimage.cfg
-- 
1.6.4.4

