From 5c009c8ebd07699b95edce0f8e73cfbf88b19b20 Mon Sep 17 00:00:00 2001
From: Alexander Holler <holler@ahsoftware.de>
Date: Tue, 16 Feb 2010 22:03:57 +0100
Subject: [PATCH 11/14] DockStar: CONFIG_PREBOOT (kernel and netconsole) CHANGE THIS TO YOUR IPS!

---
 include/configs/sheevaplug.h |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/include/configs/sheevaplug.h b/include/configs/sheevaplug.h
index 80d26d4..127302d 100644
--- a/include/configs/sheevaplug.h
+++ b/include/configs/sheevaplug.h
@@ -99,6 +99,7 @@
 #define CONFIG_CMD_PING
 #define CONFIG_CMD_USB
 #define CONFIG_SYS_LONGHELP
+#define CONFIG_PREBOOT "setenv arcNumber 2097;setenv mainlineLinux yes"

 
 /*
-- 
1.6.4.4

