--- imwheel-1.0.0pre12/Makefile.in	2004-08-29 06:36:33.000000000 -0400
+++ imwheel-1.0.0pre12-fixed/Makefile.in	2005-01-04 14:46:37.000000000 -0500
@@ -781,10 +781,10 @@
 @SUID_TRUE@	progname=$(DESTDIR)$(bindir)/`echo imwheel|sed 's/$(EXEEXT)$$//'|sed '$(transform)'|sed 's/$$/$(EXEEXT)/'` ; \
 @SUID_TRUE@	chown @suid@ $$progname ; \
 @SUID_TRUE@	chmod u+s $$progname
-	$(mkinstalldirs) $(ETCDIR)
-	-[ -f /etc/imwheelrc ] && mv /etc/imwheelrc $(ETCDIR)/imwheelrc || true
-	-[ -f /etc/X11/imwheelrc ] && mv /etc/X11/imwheelrc $(ETCDIR)/imwheelrc || true
-	-[ -f $(ETCDIR)/imwheelrc ] && install -m 644 imwheelrc $(ETCDIR)/imwheelrc.new || install -m 644 imwheelrc $(ETCDIR)/imwheelrc
+	$(mkinstalldirs) $(DESTDIR)$(ETCDIR)
+	-[ -f /etc/imwheelrc ] && mv /etc/imwheelrc $(DESTDIR)$(ETCDIR)/imwheelrc || true
+	-[ -f /etc/X11/imwheelrc ] && mv /etc/X11/imwheelrc $(DESTDIR)$(ETCDIR)/imwheelrc || true
+	-[ -f $(ETCDIR)/imwheelrc ] && install -m 644 imwheelrc $(DESTDIR)$(ETCDIR)/imwheelrc.new || install -m 644 imwheelrc $(DESTDIR)$(ETCDIR)/imwheelrc
 
 extras: $(EXTRA_PROGRAMS)
 
