diff -urNad 3ddesktop-0.2.9/event.hpp /tmp/dpep.QstFBy/3ddesktop-0.2.9/event.hpp
--- 3ddesktop-0.2.9/event.hpp   2004-12-06 03:57:32.000000000 +0100
+++ /tmp/dpep.QstFBy/3ddesktop-0.2.9/event.hpp  2005-07-06 21:50:00.000000000 +0200
@@ -74,7 +74,7 @@
         list<Event *>::iterator k;
         for (k = events.begin(); k != events.end(); ++k) {
             Event *e = *k;
-            printf(":: Event %d  0x%x\n", e->type, (unsigned int)(e->function));
+            printf(":: Event %d  %p\n", e->type, e->function);
         }
     }


