diff -ru polymake-4.12.orig/bundled/lrs/apps/polytope/src/lrs_interface.cc polymake-4.12/bundled/lrs/apps/polytope/src/lrs_interface.cc
--- polymake-4.12.orig/bundled/lrs/apps/polytope/src/lrs_interface.cc	2024-06-21 19:35:09.144851916 +0200
+++ polymake-4.12/bundled/lrs/apps/polytope/src/lrs_interface.cc	2024-06-21 19:35:25.521671387 +0200
@@ -554,7 +554,7 @@
 
    Bitset V(Points.rows());
    for (Int index = D.Q->lastdv+1, end = D.P->m_A+D.P->d; index <= end; ++index)
-      if ( !checkindex(D.P,D.Q,index) )
+      if ( !checkindex(D.P,D.Q,index,0) )
          V += D.Q->inequality[index - D.Q->lastdv]-1;
 
    return std::pair< Bitset, Matrix<Rational> >(V,AH);
diff -ru polymake-4.12.orig/bundled/sympol/external/sympol/sympol/raycomputationlrs.cpp polymake-4.12/bundled/sympol/external/sympol/sympol/raycomputationlrs.cpp
--- polymake-4.12.orig/bundled/sympol/external/sympol/sympol/raycomputationlrs.cpp	2024-06-21 19:35:09.154852010 +0200
+++ polymake-4.12/bundled/sympol/external/sympol/sympol/raycomputationlrs.cpp	2024-06-21 19:35:41.175150886 +0200
@@ -245,7 +245,7 @@
     for (ulong index = lastdv + 1; index <= m + d; index++) {
         ineq = Q->inequality[index - lastdv]; /* the input inequality number corr. to this index */
 
-        redineq[ineq] = checkindex (P, Q, index);
+        redineq[ineq] = checkindex (P, Q, index, 0);
     }  /* end for index ..... */
     
     std::list<ulong> redundancies;
