diff --git a/build/make/configure.sh b/build/make/configure.sh
index 25c9f80..1675044 100644
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -623,10 +623,14 @@ process_common_toolchain() {
 
     # detect tgt_isa
     case "$gcctarget" in
+      armv6*-gnueabihf)
+        tgt_isa=armv6
+        float_abi=hard
+        ;;
       armv6*)
         tgt_isa=armv6
         ;;
-      armv7*-hardfloat*)
+      armv7*-gnueabihf)
         tgt_isa=armv7
         float_abi=hard
         ;;
@@ -634,6 +638,10 @@ process_common_toolchain() {
         tgt_isa=armv7
         float_abi=softfp
         ;;
+      armv5te*)
+        tgt_isa=armv5te
+        float_abi=soft
+        ;;
       *x86_64*|*amd64*)
         tgt_isa=x86_64
         ;;
@@ -811,7 +819,7 @@ process_common_toolchain() {
           soft_enable neon
           ;;
         armv7|armv7s)
-          soft_enable neon
+          #soft_enable neon
           # Only enable neon_asm when neon is also enabled.
           enabled neon && soft_enable neon_asm
           # If someone tries to force it through, die.
@@ -825,13 +833,16 @@ process_common_toolchain() {
           soft_enable media
           soft_enable fast_unaligned
           ;;
+        armv5)
+          soft_enable media
+          soft_enable fast_unaligned
+          ;;
       esac
 
       asm_conversion_cmd="cat"
 
       case ${tgt_cc} in
         gcc)
-          CROSS=${CROSS:-arm-none-linux-gnueabi-}
           link_with_cc=gcc
           setup_gnu_toolchain
           arch_int=${tgt_isa##armv}
@@ -839,13 +850,6 @@ process_common_toolchain() {
           check_add_asflags --defsym ARCHITECTURE=${arch_int}
           tune_cflags="-mtune="
           if [ ${tgt_isa} = "armv7" ] || [ ${tgt_isa} = "armv7s" ]; then
-            if [ -z "${float_abi}" ]; then
-              check_cpp <<EOF && float_abi=hard || float_abi=softfp
-#ifndef __ARM_PCS_VFP
-#error "not hardfp"
-#endif
-EOF
-            fi
             check_add_cflags  -march=armv7-a -mfloat-abi=${float_abi}
             check_add_asflags -march=armv7-a -mfloat-abi=${float_abi}
 
@@ -854,8 +858,8 @@ EOF
               check_add_asflags -mfpu=neon
             fi
           else
-            check_add_cflags -march=${tgt_isa}
-            check_add_asflags -march=${tgt_isa}
+            check_add_cflags -march=${tgt_isa} -mfloat-abi=${float_abi}
+            check_add_asflags -march=${tgt_isa} -mfloat-abi=${float_abi}
           fi
 
           enabled debug && add_asflags -g
diff --git a/configure b/configure
index f3708f9..7fb89ca 100755
--- a/configure
+++ b/configure
@@ -97,6 +97,7 @@ EOF
 
 # all_platforms is a list of all supported target platforms. Maintain
 # alphabetically by architecture, generic-gnu last.
+all_platforms="${all_platforms} armv5te-linux-gcc"
 all_platforms="${all_platforms} armv6-darwin-gcc"
 all_platforms="${all_platforms} armv6-linux-rvct"
 all_platforms="${all_platforms} armv6-linux-gcc"
