diff -ur gtklp-1.2.6.orig/configure.in gtklp-1.2.6/configure.in
--- gtklp-1.2.6.orig/configure.in	2007-11-15 18:56:57.000000000 +0100
+++ gtklp-1.2.6/configure.in	2008-12-16 12:38:39.000000000 +0100
@@ -43,20 +43,10 @@
 
 # gtk #######################################################################
 GTKVER=""
-if test x$enable_gtk1 = xyes; then
-	AM_PATH_GTK([1.2.8],,[AC_ERROR([GTK+ 1.2.8 or newer is needed for GtkLP!])])
-	GTKVER="-DHAVE_GTK1"
-	OLDDIALOG="-DOLDDIALOG"
-	AC_SUBST(OLDDIALOG)
-else
-	AC_MSG_RESULT([ ******************************************************** ])
-	AC_MSG_RESULT([ * If checking for Gtk2 fails, try to use --enable-gtk1 * ])
-	AC_MSG_RESULT([ ******************************************************** ])
-	PKG_CHECK_MODULES(GTK2, gtk+-2.0)
-	AC_SUBST(GTK2_CFLAGS)
-	AC_SUBST(GTK2_LIBS)
-	GTKVER="-DHAVE_GTK2"
-fi
+PKG_CHECK_MODULES(GTK2, gtk+-2.0)
+AC_SUBST(GTK2_CFLAGS)
+AC_SUBST(GTK2_LIBS)
+GTKVER="-DHAVE_GTK2"
 
 AC_SUBST(GTKVER)
 
