--- tcp_wrappers_7.6/options.c
+++ tcp_wrappers_7.6/options.c
@@ -34,6 +34,8 @@
 
 /* System libraries. */
 
+#include <unistd.h>
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/param.h>
 #include <sys/socket.h>
--- tcp_wrappers_7.6/safe_finger.c
+++ tcp_wrappers_7.6/safe_finger.c
@@ -20,6 +20,10 @@
 
 /* System libraries */
 
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <sys/wait.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <signal.h>
@@ -27,7 +31,7 @@
 #include <ctype.h>
 #include <pwd.h>
 
-extern void exit();
+int pipe_stdin(char **argv);
 
 /* Local stuff */
 
--- tcp_wrappers_7.6/scaffold.c
+++ tcp_wrappers_7.6/scaffold.c
@@ -10,6 +10,7 @@
 
 /* System libraries. */
 
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/socket.h>
--- tcp_wrappers_7.6/shell_cmd.c
+++ tcp_wrappers_7.6/shell_cmd.c
@@ -14,6 +14,10 @@
 
 /* System libraries. */
 
+#include <unistd.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <sys/wait.h>
 #include <sys/types.h>
 #include <sys/param.h>
 #include <signal.h>
@@ -25,8 +25,6 @@
 #include <syslog.h>
 #include <string.h>
 
-extern void exit();
-
 /* Local stuff. */
 
 #include "tcpd.h"
--- tcp_wrappers_7.6/tcpdchk.c
+++ tcp_wrappers_7.6/tcpdchk.c
@@ -20,6 +20,8 @@
 
 /* System libraries. */
 
+#include <unistd.h>
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #ifdef INET6
@@ -35,11 +36,6 @@
 #include <netdb.h>
 #include <string.h>
 
-extern int errno;
-extern void exit();
-extern int optind;
-extern char *optarg;
-
 #ifndef INADDR_NONE
 #define INADDR_NONE     (-1)		/* XXX should be 0xffffffff */
 #endif
--- tcp_wrappers_7.6/clean_exit.c
+++ tcp_wrappers_7.6/clean_exit.c
@@ -13,8 +13,8 @@
 #endif
 
 #include <stdio.h>
-
-extern void exit();
+#include <unistd.h>
+#include <stdlib.h>
 
 #include "tcpd.h"
 
--- tcp_wrappers_7.6/hosts_access.c
+++ tcp_wrappers_7.6/hosts_access.c
@@ -23,6 +23,7 @@
 
 /* System libraries. */
 
+#include <stdlib.h>
 #include <sys/types.h>
 #ifdef INT32_T
     typedef uint32_t u_int32_t;
@@ -43,8 +44,7 @@
 #include <netdb.h>
 #endif
 
-extern char *fgets();
-extern int errno;
+int match_pattern_ylo(const char *s, const char *pattern);
 
 #ifndef	INADDR_NONE
 #define	INADDR_NONE	(-1)		/* XXX should be 0xffffffff */
--- tcp_wrappers_7.6/inetcf.c
+++ tcp_wrappers_7.6/inetcf.c
@@ -9,15 +9,14 @@
 static char sccsid[] = "@(#) inetcf.c 1.7 97/02/12 02:13:23";
 #endif
 
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <stdio.h>
 #include <errno.h>
 #include <string.h>
 
-extern int errno;
-extern void exit();
-
+#include "scaffold.h"
 #include "tcpd.h"
 #include "inetcf.h"
 
--- tcp_wrappers_7.6/percent_x.c
+++ tcp_wrappers_7.6/percent_x.c
@@ -16,12 +16,12 @@
 
 /* System libraries. */
 
+#include <unistd.h>
+#include <stdlib.h>
 #include <stdio.h>
 #include <syslog.h>
 #include <string.h>
 
-extern void exit();
-
 /* Local stuff. */
 
 #include "tcpd.h"
--- tcp_wrappers_7.6/rfc931.c
+++ tcp_wrappers_7.6/rfc931.c
@@ -15,6 +15,7 @@
 
 /* System libraries. */
 
+#include <unistd.h>
 #include <stdio.h>
 #include <syslog.h>
 #include <sys/types.h>
--- tcp_wrappers_7.6/tcpd.c
+++ tcp_wrappers_7.6/tcpd.c
@@ -16,6 +16,7 @@
 
 /* System libraries. */
 
+#include <unistd.h>
 #include <sys/types.h>
 #include <sys/param.h>
 #include <sys/stat.h>
--- tcp_wrappers_7.6/tcpdmatch.c
+++ tcp_wrappers_7.6/tcpdmatch.c
@@ -19,6 +19,8 @@
 
 /* System libraries. */
 
+#include <unistd.h>
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/socket.h>
@@ -30,9 +32,6 @@
 #include <setjmp.h>
 #include <string.h>
 
-extern void exit();
-extern int optind;
-extern char *optarg;
 
 #ifndef	INADDR_NONE
 #define	INADDR_NONE	(-1)		/* XXX should be 0xffffffff */
--- tcp_wrappers_7.6/update.c
+++ tcp_wrappers_7.6/update.c
@@ -19,6 +19,7 @@
 
 /* System libraries */
 
+#include <unistd.h>
 #include <stdio.h>
 #include <syslog.h>
 #include <string.h>
