--- Makefile.old	2009-10-28 10:37:40.138328073 +0100
+++ Makefile	2009-10-28 10:37:57.014326831 +0100
@@ -154,7 +154,7 @@
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS=$(MYLIB) RANLIB=ranlib ARFLAGS=rv AUX_OBJ=weak_symbols.o \
 	NETGROUP=-DNETGROUP TLI= VSYSLOG= BUGS= all \
-	EXTRA_CFLAGS="-DSYS_ERRLIST_DEFINED -DHAVE_WEAKSYMS -D_REENTRANT"
+	EXTRA_CFLAGS="-DINET6 -DSYS_ERRLIST_DEFINED -DHAVE_WEAKSYMS -D_REENTRANT"
 
 gnu:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
--- tcpd.h.old	2009-10-28 10:48:19.285326834 +0100
+++ tcpd.h	2009-10-28 10:47:32.951325793 +0100
@@ -91,7 +91,11 @@
 
 extern void shell_cmd __P((char *));	/* execute shell command */
 extern char *percent_x __P((char *, int, char *, struct request_info *)); /* do %<char> expansion */
+#ifdef INET6
+extern void rfc931 __P((struct sockaddr *, struct sockaddr *, char *)); /* client name from RFC 931 daemon */
+#else
 extern void rfc931 __P((struct sockaddr_in *, struct sockaddr_in *, char *)); /* client name from RFC 931 daemon */
+#endif
 extern void clean_exit __P((struct request_info *)); /* clean up and exit */
 extern void refuse __P((struct request_info *));	/* clean up and exit */
 extern char *xgets __P((char *, int, FILE *));	/* fgets() on steroids */
--- scaffold.c.old	2009-10-28 10:50:13.584449353 +0100
+++ scaffold.c	2009-10-28 10:52:53.765443023 +0100
@@ -182,8 +237,13 @@
 /* ARGSUSED */
 
 void    rfc931(rmt_sin, our_sin, dest)
+#ifdef INET6
+struct sockaddr *rmt_sin;
+struct sockaddr *our_sin;
+#else
 struct sockaddr_in *rmt_sin;
 struct sockaddr_in *our_sin;
+#endif
 char   *dest;
 {
     strcpy(dest, unknown); 
