diff --git i/libgda-ui/data-entries/plugins/libmain.c w/libgda-ui/data-entries/plugins/libmain.c
index a30c2db2c3258111..1356488c8b0940f9 100644
--- i/libgda-ui/data-entries/plugins/libmain.c
+++ w/libgda-ui/data-entries/plugins/libmain.c
@@ -39,15 +39,7 @@
 #endif
 
 #ifdef HAVE_GTKSOURCEVIEW
-  #ifdef GTK_DISABLE_SINGLE_INCLUDES
-    #undef GTK_DISABLE_SINGLE_INCLUDES
-  #endif
-
-  #include <gtksourceview/gtksourceview.h>
-  #include <gtksourceview/gtksourcelanguagemanager.h>
-  #include <gtksourceview/gtksourcebuffer.h>
-  #include <gtksourceview/gtksourcestyleschememanager.h>
-  #include <gtksourceview/gtksourcestylescheme.h>
+#include <gtksourceview/gtksource.h>
 #endif
 
 static GdauiDataEntry *plugin_entry_filesel_create_func (GdaDataHandler *handler, GType type, const gchar *options);
