From de6b51ae1dc02ecb48cb9642af31dd2d27d3ae04 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Fri, 26 Jun 2015 21:31:12 -0600
Subject: [PATCH] drmmode_display: include sys/stat.h

Signed-off-by: Kevin Mihelich <kevin@archlinuxarm.org>
---
 src/drmmode_display.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/drmmode_display.c b/src/drmmode_display.c
index 58c1a84..cf64424 100644
--- a/src/drmmode_display.c
+++ b/src/drmmode_display.c
@@ -51,6 +51,8 @@
 #include <libudev.h>
 #include "drmmode_driver.h"
 
+#include <sys/stat.h>
+
 struct drmmode_cursor_rec {
 	/* hardware cursor: */
 	struct armsoc_bo *bo;
-- 
2.4.4

