diff --git a/src/armsoc_driver.c b/src/armsoc_driver.c
index d8d82a1..7914d1c 100644
--- a/src/armsoc_driver.c
+++ b/src/armsoc_driver.c
@@ -1125,6 +1125,9 @@ ARMSOCScreenInit(SCREEN_INIT_ARGS_DECL)
 	 */
 	pScrn->vtSema = TRUE;
 
+	/* Do some XRandR initialization. Return value is not useful */
+	(void)xf86CrtcScreenInit(pScreen);
+
 	/* Take over the virtual terminal from the console, set the
 	 * desired mode, etc.:
 	 */
@@ -1133,9 +1136,6 @@ ARMSOCScreenInit(SCREEN_INIT_ARGS_DECL)
 		goto fail6;
 	}
 
-	/* Do some XRandR initialization. Return value is not useful */
-	(void)xf86CrtcScreenInit(pScreen);
-
 	if (!miCreateDefColormap(pScreen)) {
 		ERROR_MSG("Cannot create colormap!");
 		goto fail7;
