diff -Naur bmpx-0.40.13-orig/json/json_reader.cpp bmpx-0.40.13/json/json_reader.cpp
--- bmpx-0.40.13-orig/json/json_reader.cpp	2007-10-18 13:17:09.000000000 -0400
+++ bmpx-0.40.13/json/json_reader.cpp	2008-03-17 21:34:10.000000000 -0400
@@ -5,6 +5,7 @@
 #include <assert.h>
 #include <iostream>
 #include <stdexcept>
+#include <cstring>
 
 #if _MSC_VER >= 1400 // VC++ 8.0
 #pragma warning( disable : 4996 )   // disable warning about strdup being deprecated.
diff -Naur bmpx-0.40.13-orig/json/json_value.cpp bmpx-0.40.13/json/json_value.cpp
--- bmpx-0.40.13-orig/json/json_value.cpp	2007-10-18 13:17:09.000000000 -0400
+++ bmpx-0.40.13/json/json_value.cpp	2008-03-17 21:35:23.000000000 -0400
@@ -3,6 +3,7 @@
 #include <json/writer.h>
 #include <utility>
 #include <stdexcept>
+#include <cstring>
 #include "assert.h"
 #ifdef JSON_USE_CPPTL
 # include <cpptl/conststring.h>
diff -Naur bmpx-0.40.13-orig/json/json_writer.cpp bmpx-0.40.13/json/json_writer.cpp
--- bmpx-0.40.13-orig/json/json_writer.cpp	2007-10-18 13:17:09.000000000 -0400
+++ bmpx-0.40.13/json/json_writer.cpp	2008-03-17 21:35:44.000000000 -0400
@@ -3,6 +3,7 @@
 #include <assert.h>
 #include <stdio.h>
 #include <iostream>
+#include <cstring>
 
 #if _MSC_VER >= 1400 // VC++ 8.0
 #pragma warning( disable : 4996 )   // disable warning about strdup being deprecated.
diff -Naur bmpx-0.40.13-orig/libhal++/util.cc bmpx-0.40.13/libhal++/util.cc
--- bmpx-0.40.13-orig/libhal++/util.cc	2007-10-18 13:17:21.000000000 -0400
+++ bmpx-0.40.13/libhal++/util.cc	2008-03-17 21:37:01.000000000 -0400
@@ -1,6 +1,6 @@
 #include <string>
 #include <cstring>
-
+#include <cstdlib>
 #include "types.hh"
 
 namespace Hal
diff -Naur bmpx-0.40.13-orig/plugins/vfs/container/libcontainer_folder.cc bmpx-0.40.13/plugins/vfs/container/libcontainer_folder.cc
--- bmpx-0.40.13-orig/plugins/vfs/container/libcontainer_folder.cc	2007-10-18 13:17:14.000000000 -0400
+++ bmpx-0.40.13/plugins/vfs/container/libcontainer_folder.cc	2008-03-17 21:38:50.000000000 -0400
@@ -23,6 +23,7 @@
 //  BMPx is covered by.
 
 #include <glibmm.h>
+#include <cstring>
 
 // BMP Audio
 #include "src/audio/audio.hh"
diff -Naur bmpx-0.40.13-orig/plugins/vfs/container/libcontainer_m3u.cc bmpx-0.40.13/plugins/vfs/container/libcontainer_m3u.cc
--- bmpx-0.40.13-orig/plugins/vfs/container/libcontainer_m3u.cc	2007-10-18 13:17:14.000000000 -0400
+++ bmpx-0.40.13/plugins/vfs/container/libcontainer_m3u.cc	2008-03-17 21:39:21.000000000 -0400
@@ -23,6 +23,7 @@
 
 #include <glibmm.h>
 #include <glibmm/i18n.h>
+#include <cstring>
 
 #include "src/database.hh"
 #include "src/uri.hh"
diff -Naur bmpx-0.40.13-orig/plugins/vfs/container/libcontainer_xspf.cc bmpx-0.40.13/plugins/vfs/container/libcontainer_xspf.cc
--- bmpx-0.40.13-orig/plugins/vfs/container/libcontainer_xspf.cc	2007-10-18 13:17:14.000000000 -0400
+++ bmpx-0.40.13/plugins/vfs/container/libcontainer_xspf.cc	2008-03-17 21:41:39.000000000 -0400
@@ -27,6 +27,7 @@
 
 #include <glibmm.h>
 #include <glibmm/i18n.h>
+#include <cstring>
 
 #include <boost/optional.hpp>
 #include <boost/format.hpp>
diff -Naur bmpx-0.40.13-orig/src/hal.hh bmpx-0.40.13/src/hal.hh
--- bmpx-0.40.13-orig/src/hal.hh	2007-10-18 13:17:08.000000000 -0400
+++ bmpx-0.40.13/src/hal.hh	2008-03-18 00:11:05.000000000 -0400
@@ -32,6 +32,7 @@
 #include <string>
 #include <map>
 #include <vector>
+#include <ctime>
 
 #include <glib/gtypes.h>
 #include <glibmm/ustring.h>
@@ -169,7 +170,7 @@
 
         void  device_condition            (std::string const& udi,
                                            std::string const& cond_name,
-                                           std::string const& cond_name );
+                                           std::string const& cond_details );
         void  device_added                (std::string const& udi );
         void  device_removed              (std::string const& udi );
         void  device_property             (std::string const& udi,
diff -Naur bmpx-0.40.13-orig/src/minisoup.cc bmpx-0.40.13/src/minisoup.cc
--- bmpx-0.40.13-orig/src/minisoup.cc	2007-10-18 13:17:08.000000000 -0400
+++ bmpx-0.40.13/src/minisoup.cc	2008-03-18 01:18:11.000000000 -0400
@@ -28,6 +28,7 @@
 
 #include <glib/gstdio.h>
 #include <glibmm/i18n.h>
+#include <cstring>
 #include <boost/format.hpp>
 #include "debug.hh"
 #include "minisoup.hh"
diff -Naur bmpx-0.40.13-orig/src/parser/libxml2-sax-base.cc bmpx-0.40.13/src/parser/libxml2-sax-base.cc
--- bmpx-0.40.13-orig/src/parser/libxml2-sax-base.cc	2007-10-18 13:17:06.000000000 -0400
+++ bmpx-0.40.13/src/parser/libxml2-sax-base.cc	2008-03-17 23:36:34.000000000 -0400
@@ -28,6 +28,7 @@
 
 #include <libxml/parser.h>
 #include <glibmm.h>
+#include <cstring>
 
 using namespace Glib;
 using namespace std;
diff -Naur bmpx-0.40.13-orig/src/sanity.cc bmpx-0.40.13/src/sanity.cc
--- bmpx-0.40.13-orig/src/sanity.cc	2007-10-18 13:17:08.000000000 -0400
+++ bmpx-0.40.13/src/sanity.cc	2008-03-17 23:44:47.000000000 -0400
@@ -69,7 +69,7 @@
 }
 
 C_FUNCTION
-gchar *
+const gchar *
 sanity_check_gtk (void)
 {
   return gtk_check_version (GTK_REQUIRED_MAJOR_VER,
diff -Naur bmpx-0.40.13-orig/src/streams-shoutcast.cc bmpx-0.40.13/src/streams-shoutcast.cc
--- bmpx-0.40.13-orig/src/streams-shoutcast.cc	2007-10-18 13:17:08.000000000 -0400
+++ bmpx-0.40.13/src/streams-shoutcast.cc	2008-03-18 00:03:38.000000000 -0400
@@ -27,6 +27,7 @@
 
 #include <iostream>
 #include <sstream>
+#include <cstring>
 
 #include <glibmm.h>
 #include <glibmm/markup.h>
diff -Naur bmpx-0.40.13-orig/src/vfs.hh bmpx-0.40.13/src/vfs.hh
--- bmpx-0.40.13-orig/src/vfs.hh	2007-10-18 13:17:08.000000000 -0400
+++ bmpx-0.40.13/src/vfs.hh	2008-03-17 23:34:57.000000000 -0400
@@ -25,6 +25,7 @@
 #define BMP_VFS_HH
 
 #include <iostream>
+#include <cstring>
 #include <sstream>
 #include <glibmm.h>
 #include <boost/shared_ptr.hpp>
diff -Naur bmpx-0.40.13-orig/src/video-widget.cc bmpx-0.40.13/src/video-widget.cc
--- bmpx-0.40.13-orig/src/video-widget.cc	2007-10-18 13:17:08.000000000 -0400
+++ bmpx-0.40.13/src/video-widget.cc	2008-03-18 00:04:38.000000000 -0400
@@ -23,6 +23,7 @@
 //  BMPx is covered by.
 
 #include <gtkmm.h>
+#include <cstring>
 #include "video-widget.hh"
 #include "audio/play.hh"
 #include <gst/video/video.h>
diff -Naur bmpx-0.40.13-orig/src/xds.cc bmpx-0.40.13/src/xds.cc
--- bmpx-0.40.13-orig/src/xds.cc	2007-10-18 13:17:08.000000000 -0400
+++ bmpx-0.40.13/src/xds.cc	2008-03-17 23:59:04.000000000 -0400
@@ -23,6 +23,7 @@
 //  BMPx is covered by.
 
 #include <gtkmm.h>
+#include <cstring>
 #include "xds.hh"
 #define MAX_URI_LEN 4096
 
