--- src/shell.cpp.orig	2006-08-27 16:03:26.000000000 -0400
+++ src/shell.cpp	2006-08-27 16:04:15.000000000 -0400
@@ -1558,11 +1558,6 @@
       try {
         notify = new Bmp::Notify (tray_icon);
       } catch (Bmp::NotifyException& cxe) {
-        bmp_system_control_message_dispatch (bmp_system_control,
-                                             msg_domain_id,
-                                             GTK_MESSAGE_ERROR,
-                                             _("BMP was unable to start and/or connect to notification-daemon."
-                                               "Notification popups will not be available."));
         notify = 0;
       }
 #endif
