diff -up qtdeclarative-opensource-src-5.6.0/src/qml/qml.pro.1135 qtdeclarative-opensource-src-5.6.0/src/qml/qml.pro
--- qtdeclarative-opensource-src-5.6.0/src/qml/qml.pro.1135	2016-06-02 08:43:24.509068141 -0500
+++ qtdeclarative-opensource-src-5.6.0/src/qml/qml.pro	2016-06-02 10:25:28.813766581 -0500
@@ -21,7 +21,7 @@ exists("qqml_enable_gcov") {
 
 greaterThan(QT_GCC_MAJOR_VERSION, 5) {
     # Our code is bad. Temporary workaround.
-    QMAKE_CXXFLAGS += -fno-delete-null-pointer-checks
+    QMAKE_CXXFLAGS += -fno-delete-null-pointer-checks -fno-lifetime-dse
 }
 
 QMAKE_DOCS = $$PWD/doc/qtqml.qdocconf
