diff -aur lightdm-gtk-greeter-1.5.0.src/src/lightdm-gtk-greeter.c lightdm-gtk-greeter-1.5.0.patch/src/lightdm-gtk-greeter.c
--- lightdm-gtk-greeter-1.5.0.src/src/lightdm-gtk-greeter.c	2013-01-25 01:31:00.000000000 +1100
+++ lightdm-gtk-greeter-1.5.0.patch/src/lightdm-gtk-greeter.c	2013-01-30 19:42:27.612325811 +1100
@@ -872,6 +872,7 @@
            g_warning ("Failed to load background: %s", error->message);
         g_clear_error (&error);
         g_free (path);
+        default_background_pixbuf=gdk_pixbuf_add_alpha (default_background_pixbuf,FALSE,255,255,255);
     }
     else
     {
