diff -upNr beagle-0.3.9.orign/beagled/BeagleDaemon.cs beagle-0.3.9/beagled/BeagleDaemon.cs
--- beagle-0.3.9.orign/beagled/BeagleDaemon.cs	2008-06-11 20:42:20.000000000 +0200
+++ beagle-0.3.9/beagled/BeagleDaemon.cs	2009-01-31 12:44:23.000000000 +0100
@@ -456,7 +456,7 @@ namespace Beagle.Daemon {
 			// debugging this code, after all...
 			// arg_debug ? LogLevel.Debug : LogLevel.Warn
 			
-			Log.Initialize (PathFinder.LogDir, "Beagle", LogLevel.Debug, arg_fg);
+			Log.Initialize (PathFinder.LogDir, "Beagle", arg_debug ? LogLevel.Debug : LogLevel.Warn, arg_fg);
 			Log.Always ("Starting Beagle Daemon (version {0})", ExternalStringsHack.Version);
 			Log.Always ("Running on {0}", SystemInformation.MonoRuntimeVersion);
 			Log.Always ("Command Line: {0}",
