From 81a351257b66b2c2e06e6b372b88fa06f60ce128 Mon Sep 17 00:00:00 2001
From: Ivan Shapovalov <intelfx@intelfx.name>
Date: Sun, 28 Jul 2024 20:43:08 +0200
Subject: [PATCH] Fix libbpf build (workaround for netdata/libbpf#4)

---
 packaging/cmake/Modules/NetdataLibBPF.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/packaging/cmake/Modules/NetdataLibBPF.cmake b/packaging/cmake/Modules/NetdataLibBPF.cmake
index 17a26a10f..823a45b4a 100644
--- a/packaging/cmake/Modules/NetdataLibBPF.cmake
+++ b/packaging/cmake/Modules/NetdataLibBPF.cmake
@@ -72,11 +72,11 @@ function(netdata_bundle_libbpf)
         GIT_REPOSITORY https://github.com/netdata/libbpf.git
         GIT_TAG ${_libbpf_tag}
         SOURCE_DIR "${libbpf_SOURCE_DIR}"
-        CONFIGURE_COMMAND ""
-        BUILD_COMMAND ${MAKE_COMMAND} -C src CC=${CMAKE_C_COMPILER} BUILD_STATIC_ONLY=1 OBJDIR=build/ DESTDIR=../ install
+        CONFIGURE_COMMAND mkdir -pv src/build/
+        BUILD_COMMAND ${MAKE_COMMAND} -C src CC=${CMAKE_C_COMPILER} BUILD_STATIC_ONLY=1 OBJDIR=build/ all
         BUILD_IN_SOURCE 1
         BUILD_BYPRODUCTS "${_libbpf_library}"
-        INSTALL_COMMAND ""
+        INSTALL_COMMAND ${MAKE_COMMAND} -C src CC=${CMAKE_C_COMPILER} BUILD_STATIC_ONLY=1 OBJDIR=build/ DESTDIR=../ install
         EXCLUDE_FROM_ALL 1
     )
 
-- 
2.45.2.5.g411eac4b43

