diff --git a/aclk/schema-wrappers/alarm_stream.cc b/aclk/schema-wrappers/alarm_stream.cc
index 11b9284f5..9e34b78a2 100644
--- a/aclk/schema-wrappers/alarm_stream.cc
+++ b/aclk/schema-wrappers/alarm_stream.cc
@@ -1,11 +1,11 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 
-#include "alarm_stream.h"
-
 #include "proto/alarm/v1/stream.pb.h"
 
 #include "libnetdata/libnetdata.h"
 
+#include "alarm_stream.h"
+
 #include "schema_wrapper_utils.h"
 
 using namespace alarms::v1;
diff --git a/aclk/schema-wrappers/schema_wrapper_utils.h b/aclk/schema-wrappers/schema_wrapper_utils.h
index 2815d0f20..751caa324 100644
--- a/aclk/schema-wrappers/schema_wrapper_utils.h
+++ b/aclk/schema-wrappers/schema_wrapper_utils.h
@@ -3,11 +3,12 @@
 #ifndef SCHEMA_WRAPPER_UTILS_H
 #define SCHEMA_WRAPPER_UTILS_H
 
+#include <google/protobuf/timestamp.pb.h>
+#include <google/protobuf/map.h>
+
 #include "database/rrd.h"
 
 #include <sys/time.h>
-#include <google/protobuf/timestamp.pb.h>
-#include <google/protobuf/map.h>
 
 #if GOOGLE_PROTOBUF_VERSION < 3001000
 #define PROTO_COMPAT_MSG_SIZE(msg) (size_t)msg.ByteSize();
