diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0f02a2d23..63f0459a8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3943,7 +3943,7 @@ endif()
 #
 
 install(CODE "
-        execute_process(COMMAND gzip -c \"${CMAKE_BINARY_DIR}/CMakeCache.txt\"
+        execute_process(COMMAND gzip -nc \"${CMAKE_BINARY_DIR}/CMakeCache.txt\"
                         OUTPUT_FILE \"${BUILD_INFO_CMAKE_CACHE_ARCHIVE_NAME}\"
                         WORKING_DIRECTORY \"${CMAKE_BINARY_DIR}\"
                         RESULT_VARIABLE result)
