From b3b18976a7b68572ea831018d8bdb4b857d62938 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:25:58 -0700
Subject: [PATCH 1/8] makechrootpkg: cache dir

---
 src/makechrootpkg.in | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/makechrootpkg.in b/src/makechrootpkg.in
index e7cbc9c..4458e3a 100644
--- a/src/makechrootpkg.in
+++ b/src/makechrootpkg.in
@@ -84,6 +84,7 @@ usage() {
 	echo '-T                Build in a temporary directory'
 	echo '-U                Run makepkg as a specified user'
 	echo '-x <when>         Inspect chroot after build (never, always, failure)'
+	echo '-X <dir>          Set pacman cache to pass to arch-nspawn'
 	exit 1
 }
 
@@ -293,7 +294,7 @@ move_products() {
 }
 # }}}
 
-while getopts 'hcur:I:l:nCTD:d:U:x:t:' arg; do
+while getopts 'hcuX:r:I:l:nCTD:d:U:x:t:' arg; do
 	case "$arg" in
 		c) clean_first=1 ;;
 		D) bindmounts_ro+=("--bind-ro=$OPTARG") ;;
@@ -308,6 +309,7 @@ while getopts 'hcur:I:l:nCTD:d:U:x:t:' arg; do
 		T) temp_chroot=1; copy+="-$$" ;;
 		U) makepkg_user="$OPTARG" ;;
 		x) inspect="$OPTARG" ;;
+		X) cache_dir="$OPTARG" ;;
 		h|*) usage ;;
 	esac
 done
@@ -323,6 +325,10 @@ chrootdir=$(readlink -e "$passeddir")
 [[ ! -d $chrootdir ]] && die "No chroot dir defined, or invalid path '%s'" "$passeddir"
 [[ ! -d $chrootdir/root ]] && die "Missing chroot dir root directory. Try using: mkarchroot %s/root base-devel" "$chrootdir"
 
+if [ -n "$cache_dir" ]; then
+	cache_dir="-c $cache_dir"
+fi
+
 if [[ ${copy:0:1} = / ]]; then
 	copydir=$copy
 else
@@ -396,7 +402,7 @@ nspawn_build_args=(
 	"${bindmounts_tmpfs[@]}"
 )
 
-if arch-nspawn "$copydir" \
+if arch-nspawn $cache_dir "$copydir" \
 	"${nspawn_build_args[@]}" \
 	/chrootbuild "${makepkg_args[@]}"
 then
-- 
2.51.2

