From 849d82cd9eed1d4ad08aee9cb1c47ee0cf82c186 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:27:34 -0700
Subject: [PATCH 2/9] arch-nspawn: keep mirrorlist

---
 arch-nspawn.in | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch-nspawn.in b/arch-nspawn.in
index 275cff7..2c61371 100644
--- a/arch-nspawn.in
+++ b/arch-nspawn.in
@@ -86,8 +86,7 @@ done
 copy_hostconf () {
 	unshare --fork --pid gpg --homedir "$working_dir"/etc/pacman.d/gnupg/ --no-permission-warning --quiet --batch --import --import-options import-local-sigs "$(pacman-conf GpgDir)"/pubring.gpg >/dev/null 2>&1
 	pacman-key --gpgdir "$working_dir"/etc/pacman.d/gnupg/ --import-trustdb "$(pacman-conf GpgDir)" >/dev/null 2>&1
-
-	printf 'Server = %s\n' "${host_mirrors[@]}" >"$working_dir/etc/pacman.d/mirrorlist"
+	#printf 'Server = %s\n' "${host_mirrors[@]}" >"$working_dir/etc/pacman.d/mirrorlist"
 
 	[[ -n $pac_conf ]] && cp "$pac_conf" "$working_dir/etc/pacman.conf"
 	[[ -n $makepkg_conf ]] && cp "$makepkg_conf" "$working_dir/etc/makepkg.conf"
-- 
2.36.1

