diff -urN a/arch-nspawn.in b/arch-nspawn.in
--- a/arch-nspawn.in	2013-11-07 06:57:03.000000000 -0700
+++ b/arch-nspawn.in	2014-01-21 18:33:11.817113085 -0700
@@ -70,8 +70,8 @@
 }
 
 copy_hostconf () {
-	cp -a /etc/pacman.d/gnupg "$working_dir/etc/pacman.d"
-	echo "Server = $host_mirror" >"$working_dir/etc/pacman.d/mirrorlist"
+	#cp -a /etc/pacman.d/gnupg "$working_dir/etc/pacman.d"
+	#echo "Server = $host_mirror" >"$working_dir/etc/pacman.d/mirrorlist"
 
 	[[ -n $pac_conf ]] && cp $pac_conf "$working_dir/etc/pacman.conf"
 	[[ -n $makepkg_conf ]] && cp $makepkg_conf "$working_dir/etc/makepkg.conf"
