From 381a5954863cab1a49fba9568fc351ad5dc62177 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:27:34 -0700
Subject: [PATCH 2/8] arch-nspawn: keep mirrorlist

---
 src/arch-nspawn.in | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/arch-nspawn.in b/src/arch-nspawn.in
index fdafe9c..d7636b0 100644
--- a/src/arch-nspawn.in
+++ b/src/arch-nspawn.in
@@ -106,8 +106,7 @@ done
 copy_hostconf () {
 	unshare --fork --pid gpg --homedir "$working_dir"/etc/pacman.d/gnupg/ --no-permission-warning --quiet --batch --import --import-options import-local-sigs "$(pacman-conf GpgDir)"/pubring.gpg >/dev/null 2>&1
 	pacman-key --gpgdir "$working_dir"/etc/pacman.d/gnupg/ --import-trustdb "$(pacman-conf GpgDir)" >/dev/null 2>&1
-
-	printf 'Server = %s\n' "${host_mirrors[@]}" >"$working_dir/etc/pacman.d/mirrorlist"
+	#printf 'Server = %s\n' "${host_mirrors[@]}" >"$working_dir/etc/pacman.d/mirrorlist"
 
 	[[ -n $host_cachemirrors ]] && printf 'CacheServer = %s\n' "${host_cachemirrors[@]}" >>"$working_dir/etc/pacman.d/mirrorlist"
 
-- 
2.51.2

