diff -urN a/mkarchroot.in b/mkarchroot.in
--- a/mkarchroot.in	2013-04-08 10:09:19.000000000 -0600
+++ b/mkarchroot.in	2013-05-31 10:53:46.465229553 -0600
@@ -142,7 +142,7 @@
 
 	chroot_lock
 	build_mount_args
-	copy_hostconf
+	#copy_hostconf
 
 	chroot_run "${working_dir}" ${RUN}
 
@@ -178,7 +178,7 @@
 	chroot_run "${working_dir}" locale-gen
 	echo 'LANG=C' > "${working_dir}/etc/locale.conf"
 
-	copy_hostconf
+	#copy_hostconf
 
 	echo "${CHROOT_VERSION}" > "${working_dir}/.arch-chroot"
 	# }}}
