From eb112c0d69e14ee6eb02e59c0c35089ed0cc6ea1 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:27:34 -0700
Subject: [PATCH 2/6] arch-nspawn: keep mirrorlist

---
 arch-nspawn.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch-nspawn.in b/arch-nspawn.in
index 19f1410..20726e8 100644
--- a/arch-nspawn.in
+++ b/arch-nspawn.in
@@ -73,7 +73,7 @@ build_mount_args() {
 
 copy_hostconf () {
 	cp -a /etc/pacman.d/gnupg "$working_dir/etc/pacman.d"
-	echo "Server = $host_mirror" >"$working_dir/etc/pacman.d/mirrorlist"
+	#echo "Server = $host_mirror" >"$working_dir/etc/pacman.d/mirrorlist"
 
 	[[ -n $pac_conf ]] && cp $pac_conf "$working_dir/etc/pacman.conf"
 	[[ -n $makepkg_conf ]] && cp $makepkg_conf "$working_dir/etc/makepkg.conf"
-- 
2.3.2

