From 375082ca4e5c03e0ffa0d81ee3739cd131d58ca4 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:32:15 -0700
Subject: [PATCH 3/6] makechrootpkg: distcc

---
 makechrootpkg.in | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/makechrootpkg.in b/makechrootpkg.in
index c0a15c0..bdc9938 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -347,7 +347,10 @@ _chrootbuild() {
 		exit 1
 	fi
 
-	sudo -u builduser makepkg "$@"
+	mkdir /build/.distcc
+	chown builduser /build/.distcc
+	
+	sudo -u builduser DISTCC_IO_TIMEOUT=0 DISTCC_DIR='/build/.distcc' makepkg "$@"
 }
 
 move_products() {
-- 
2.2.1

