diff -urN a/makechrootpkg.in b/makechrootpkg.in
--- a/makechrootpkg.in	2014-01-21 18:14:12.156219500 -0700
+++ b/makechrootpkg.in	2014-01-21 18:16:41.715633742 -0700
@@ -324,7 +324,9 @@
 		exit 1
 	fi
 
-	sudo -u nobody makepkg $makepkg_args || exit 1
+	mkdir /build/.distcc
+	chown nobody /build/.distcc
+	sudo -u nobody DISTCC_IO_TIMEOUT=0 DISTCC_DIR='/build/.distcc' makepkg $makepkg_args || exit 1
 
 	if $run_namcap; then
 		pacman -S --needed --noconfirm namcap
