From a94a490e14cadde174ca0a31fe7ae01323bb6116 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:32:15 -0700
Subject: [PATCH 3/7] makechrootpkg: distcc

---
 makechrootpkg.in | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/makechrootpkg.in b/makechrootpkg.in
index 2c4a926..4b86084 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -202,7 +202,10 @@ _chrootbuild() {
 	. /etc/profile
 	export HOME=/build
 	cd /startdir
-	sudo -u builduser makepkg "$@"
+	mkdir /build/.distcc
+	chown builduser /build/.distcc
+	
+	sudo -u builduser DISTCC_IO_TIMEOUT=0 DISTCC_DIR='/build/.distcc' makepkg "$@"
 }
 
 _chrootnamcap() {
-- 
2.9.0

