diff -urN a/makechrootpkg.in b/makechrootpkg.in
--- a/makechrootpkg.in	2013-05-31 10:49:38.866569102 -0600
+++ b/makechrootpkg.in	2013-05-31 10:57:39.183970506 -0600
@@ -280,7 +280,9 @@
 export HOME=/build
 
 cd /build
-sudo -u nobody makepkg $makepkg_args || exit 1
+mkdir .distcc
+chown nobody .distcc
+sudo -u nobody DISTCC_IO_TIMEOUT=0 DISTCC_DIR='/build/.distcc' makepkg $makepkg_args || exit 1
 
 if $run_namcap; then
 	pacman -S --needed --noconfirm namcap
