From 27859bef1f1a87cf2d2d1163ee3dc8e906b90abd Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:32:15 -0700
Subject: [PATCH 3/8] makechrootpkg: distcc

---
 makechrootpkg.in | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/makechrootpkg.in b/makechrootpkg.in
index 71ee9c0..af3cc28 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -214,7 +214,9 @@ _chrootbuild() {
 	# use "$" in arguments to commands with "sudo -i".  ${foo} or
 	# ${1} is OK, but $foo or $1 isn't.
 	# https://bugzilla.sudo.ws/show_bug.cgi?id=765
-	sudo --preserve-env=SOURCE_DATE_EPOCH -iu builduser bash -c 'cd /startdir; makepkg "$@"' -bash "$@"
+	mkdir /build/.distcc
+	chown builduser /build/.distcc
+	sudo --preserve-env=SOURCE_DATE_EPOCH -iu builduser DISTCC_IO_TIMEOUT=1200 DISTCC_DIR='/build/.distcc' bash -c 'cd /startdir; makepkg "$@"' -bash "$@"
 	ret=$?
 	case $ret in
 		0|14)
-- 
2.30.0

