From 3ffcd8940c52a9039c84cc5eb2693fa5189d10c7 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:32:15 -0700
Subject: [PATCH 3/6] makechrootpkg: distcc

---
 makechrootpkg.in | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/makechrootpkg.in b/makechrootpkg.in
index 53c0a07..1e73785 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -289,7 +289,10 @@ _chrootbuild() {
 		exit 1
 	fi
 
-	sudo -u builduser makepkg "$@"
+	mkdir /build/.distcc
+	chown builduser /build/.distcc
+	
+	sudo -u builduser DISTCC_IO_TIMEOUT=0 DISTCC_DIR='/build/.distcc' makepkg "$@"
 }
 
 move_products() {
-- 
2.3.2

