From 20af170ed679522bb78ffec04f9f4234209dab00 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:32:15 -0700
Subject: [PATCH 3/9] makechrootpkg: distcc

---
 makechrootpkg.in | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/makechrootpkg.in b/makechrootpkg.in
index aef2702..0929df1 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -208,10 +208,12 @@ _chrootbuild() {
 	# use "$" in arguments to commands with "sudo -i".  ${foo} or
 	# ${1} is OK, but $foo or $1 isn't.
 	# https://bugzilla.sudo.ws/show_bug.cgi?id=765
+	mkdir /build/.distcc
+	chown builduser /build/.distcc
 	sudo --preserve-env=SOURCE_DATE_EPOCH \
 		--preserve-env=BUILDTOOL \
 		--preserve-env=BUILDTOOLVER \
-		-iu builduser bash -c 'cd /startdir; makepkg "$@"' -bash "$@"
+		-iu builduser DISTCC_IO_TIMEOUT=1200 DISTCC_DIR='/build/.distcc' bash -c 'cd /startdir; makepkg "$@"' -bash "$@"
 	ret=$?
 	case $ret in
 		0|14)
-- 
2.36.1

