From 448f11996fc2c167084d96954746ab0153d26e9e Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:33:00 -0700
Subject: [PATCH 4/6] arch-nspawn: arm fix

---
 arch-nspawn.in   | 2 +-
 makechrootpkg.in | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch-nspawn.in b/arch-nspawn.in
index 20726e8..3f3d563 100644
--- a/arch-nspawn.in
+++ b/arch-nspawn.in
@@ -96,7 +96,7 @@ copy_hostconf
 
 eval $(grep '^CARCH=' "$working_dir/etc/makepkg.conf")
 
-exec ${CARCH:+setarch "$CARCH"} systemd-nspawn 2>/dev/null \
+exec systemd-nspawn 2>/dev/null \
 	-D "$working_dir" \
 	--register=no \
 	"${mount_args[@]}" \
diff --git a/makechrootpkg.in b/makechrootpkg.in
index bdc9938..46aff89 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -297,10 +297,10 @@ download_sources() {
 	# Ensure sources are downloaded
 	if [[ -n $SUDO_USER ]]; then
 		sudo -u $SUDO_USER env SRCDEST="$SRCDEST" BUILDDIR="$builddir" \
-			makepkg --config="$copydir/etc/makepkg.conf" --verifysource -o
+			makepkg -A --config="$copydir/etc/makepkg.conf" --verifysource -o
 	else
 		( export SRCDEST BUILDDIR="$builddir"
-			makepkg --asroot --config="$copydir/etc/makepkg.conf" --verifysource -o
+			makepkg -A --config="$copydir/etc/makepkg.conf" --verifysource -o
 		)
 	fi
 	(( $? != 0 )) && die "Could not download sources."
-- 
2.2.1

