From 0e72ba89cc863b7545577c3c463a30725cfa7340 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:33:00 -0700
Subject: [PATCH 4/9] arch-nspawn: arm fix

---
 arch-nspawn.in   | 2 +-
 makechrootpkg.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch-nspawn.in b/arch-nspawn.in
index 2c61371..42f19cc 100644
--- a/arch-nspawn.in
+++ b/arch-nspawn.in
@@ -121,7 +121,7 @@ else
 	set_arch="${CARCH}"
 fi
 
-exec ${CARCH:+setarch "$set_arch"} systemd-nspawn -q \
+exec systemd-nspawn -q \
 	-D "$working_dir" \
 	-E "PATH=/usr/local/sbin:/usr/local/bin:/usr/bin" \
 	--register=no --keep-unit --as-pid2 \
diff --git a/makechrootpkg.in b/makechrootpkg.in
index 0929df1..367f030 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -238,7 +238,7 @@ download_sources() {
 	# Ensure sources are downloaded
 	sudo -u "$makepkg_user" --preserve-env=GNUPGHOME,SSH_AUTH_SOCK \
 		env SRCDEST="$SRCDEST" BUILDDIR="$WORKDIR" \
-		makepkg --config="$copydir/etc/makepkg.conf" --verifysource -o "${verifysource_args[@]}" ||
+		makepkg -A --config="$copydir/etc/makepkg.conf" --verifysource -o "${verifysource_args[@]}" ||
 		die "Could not download sources."
 }
 
-- 
2.36.1

