From d9d880448cf065cf5a9ad8649d2ce53f9055e289 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:33:00 -0700
Subject: [PATCH 4/8] arch-nspawn: arm fix

---
 arch-nspawn.in   | 2 +-
 makechrootpkg.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch-nspawn.in b/arch-nspawn.in
index 18febda..be9f49a 100644
--- a/arch-nspawn.in
+++ b/arch-nspawn.in
@@ -126,7 +126,7 @@ eval "$(grep -a '^CARCH=' "$working_dir/etc/makepkg.conf")"
 
 [[ -z $nosetarch ]] || unset CARCH
 
-exec ${CARCH:+setarch "$CARCH"} systemd-nspawn -q \
+exec systemd-nspawn -q \
 	-D "$working_dir" \
 	-E "PATH=/usr/local/sbin:/usr/local/bin:/usr/bin" \
 	--register=no --keep-unit --as-pid2 \
diff --git a/makechrootpkg.in b/makechrootpkg.in
index af3cc28..aeb6d4a 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -241,7 +241,7 @@ download_sources() {
 	# Ensure sources are downloaded
 	sudo -u "$makepkg_user" --preserve-env=GNUPGHOME \
 		env SRCDEST="$SRCDEST" BUILDDIR="$WORKDIR" \
-		makepkg --config="$copydir/etc/makepkg.conf" --verifysource -o "${verifysource_args[@]}" ||
+		makepkg -A --config="$copydir/etc/makepkg.conf" --verifysource -o "${verifysource_args[@]}" ||
 		die "Could not download sources."
 }
 
-- 
2.30.0

