diff -urN a/arch-nspawn.in b/arch-nspawn.in
--- a/arch-nspawn.in	2014-01-21 18:16:31.735672829 -0700
+++ b/arch-nspawn.in	2014-01-21 18:17:31.245439755 -0700
@@ -97,7 +97,7 @@
 machine_name="${working_dir//[![:alnum:]_-]/-}"
 machine_name="${machine_name#-}"
 
-exec ${CARCH:+setarch "$CARCH"} systemd-nspawn 2>/dev/null \
+exec systemd-nspawn 2>/dev/null \
 	-D "$working_dir" \
 	--machine "$machine_name" \
 	"${mount_args[@]}" \
diff -urN a/makechrootpkg.in b/makechrootpkg.in
--- a/makechrootpkg.in	2014-01-21 18:17:21.820476669 -0700
+++ b/makechrootpkg.in	2014-01-21 18:17:31.245439755 -0700
@@ -272,10 +272,10 @@
 	# Ensure sources are downloaded
 	if [[ -n $SUDO_USER ]]; then
 		sudo -u $SUDO_USER env SRCDEST="$SRCDEST" BUILDDIR="$builddir" \
-			makepkg --config="$copydir/etc/makepkg.conf" --verifysource -o
+			makepkg -A --config="$copydir/etc/makepkg.conf" --verifysource -o
 	else
 		( export SRCDEST BUILDDIR="$builddir"
-			makepkg --asroot --config="$copydir/etc/makepkg.conf" --verifysource -o
+			makepkg -A --asroot --config="$copydir/etc/makepkg.conf" --verifysource -o
 		)
 	fi
 	(( $? != 0 )) && die "Could not download sources."
