From 44a2b161185fc8ea89bde621badfcc3c2fc82c64 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:33:41 -0700
Subject: [PATCH 5/7] makechrootpkg: no default logging

---
 makechrootpkg.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makechrootpkg.in b/makechrootpkg.in
index bb561ea..3ff3673 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -307,7 +307,7 @@ move_products() {
 # }}}
 
 main() {
-	default_makepkg_args=(--syncdeps --noconfirm --log --holdver --skipinteg)
+	default_makepkg_args=(--syncdeps --noconfirm --holdver --skipinteg)
 	makepkg_args=("${default_makepkg_args[@]}")
 	keepbuilddir=false
 	update_first=false
-- 
2.16.2

