From 3111a45460053111c3f736ccbee88c2f30a403ce Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:33:41 -0700
Subject: [PATCH 5/8] makechrootpkg: no default logging

---
 makechrootpkg.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makechrootpkg.in b/makechrootpkg.in
index e29d42c..ba27249 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -305,7 +305,7 @@ move_products() {
 # }}}
 
 main() {
-	default_makepkg_args=(--syncdeps --noconfirm --log --holdver --skipinteg)
+	default_makepkg_args=(--syncdeps --noconfirm --holdver --skipinteg)
 	makepkg_args=("${default_makepkg_args[@]}")
 	keepbuilddir=false
 	update_first=false
-- 
2.14.1

