From 1f6b9d855cfad03203364e286d48a8aaeb984abf Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:33:41 -0700
Subject: [PATCH 5/7] makechrootpkg: no default logging

---
 makechrootpkg.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makechrootpkg.in b/makechrootpkg.in
index c342d18..086a0a6 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -15,7 +15,7 @@ m4_include(lib/archroot.sh)
 
 shopt -s nullglob
 
-default_makepkg_args=(--syncdeps --noconfirm --log --holdver --skipinteg)
+default_makepkg_args=(--syncdeps --noconfirm --holdver --skipinteg)
 makepkg_args=("${default_makepkg_args[@]}")
 verifysource_args=()
 chrootdir=
-- 
2.22.0

