From b186821bb6ac5e7aee3c3b63074bae2a2ebbaabd Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:33:41 -0700
Subject: [PATCH 5/6] makechrootpkg: no default logging

---
 makechrootpkg.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makechrootpkg.in b/makechrootpkg.in
index 46aff89..e1da896 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -12,7 +12,7 @@ m4_include(lib/common.sh)
 
 shopt -s nullglob
 
-makepkg_args=(-s --noconfirm -L --holdver)
+makepkg_args=(-s --noconfirm --holdver)
 repack=false
 update_first=false
 clean_first=false
-- 
2.2.1

