From be3d9440b826c00a3f94b728d3680cc3bf7e5470 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:33:41 -0700
Subject: [PATCH 5/8] makechrootpkg: no default logging

---
 src/makechrootpkg.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/makechrootpkg.in b/src/makechrootpkg.in
index 7f9031b..a3a2c0a 100644
--- a/src/makechrootpkg.in
+++ b/src/makechrootpkg.in
@@ -17,7 +17,7 @@ source /usr/share/makepkg/util/util.sh
 
 shopt -s nullglob
 
-default_makepkg_args=(--syncdeps --noconfirm --log --holdver --skipinteg)
+default_makepkg_args=(--syncdeps --noconfirm --holdver --skipinteg)
 makepkg_args=("${default_makepkg_args[@]}")
 verifysource_args=()
 chrootdir=
-- 
2.51.2

