From bc5a5e2554771a6310bbb35fd6c5e9afffc6d3a0 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:34:04 -0700
Subject: [PATCH 6/8] archbuild: no setarch

---
 archbuild.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/archbuild.in b/archbuild.in
index 5b98976..b62794f 100644
--- a/archbuild.in
+++ b/archbuild.in
@@ -69,7 +69,7 @@ if ${clean_first} || [[ ! -d "${chroots}/${repo}-${arch}" ]]; then
 
 	rm -rf --one-file-system "${chroots}/${repo}-${arch}"
 	(umask 0022; mkdir -p "${chroots}/${repo}-${arch}")
-	setarch "${arch}" mkarchroot \
+	mkarchroot \
 		-C "${pacman_config}" \
 		-M "${makepkg_config}" \
 		"${chroots}/${repo}-${arch}/root" \
-- 
2.30.0

