From 8243eccb65f5756506e1ef9c1ac3f80176d3c314 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:34:04 -0700
Subject: [PATCH 6/6] archbuild: no setarch

---
 archbuild.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/archbuild.in b/archbuild.in
index 9c5d706..a4b5bed 100644
--- a/archbuild.in
+++ b/archbuild.in
@@ -62,7 +62,7 @@ if ${clean_first} || [[ ! -d "${chroots}/${repo}-${arch}" ]]; then
 
 	rm -rf --one-file-system "${chroots}/${repo}-${arch}"
 	mkdir -p "${chroots}/${repo}-${arch}"
-	setarch "${arch}" mkarchroot \
+	mkarchroot \
 		-C "@pkgdatadir@/pacman-${repo}.conf" \
 		-M "@pkgdatadir@/makepkg-${arch}.conf" \
 		"${chroots}/${repo}-${arch}/root" \
-- 
2.3.2

