--- a/archbuild.in	2014-02-16 07:57:21.777365491 -0500
+++ b/archbuild.in	2014-02-16 07:57:54.280700229 -0500
@@ -62,7 +62,7 @@
 
 	rm -rf --one-file-system "${chroots}/${repo}-${arch}"
 	mkdir -p "${chroots}/${repo}-${arch}"
-	setarch "${arch}" mkarchroot \
+	mkarchroot \
 		-C "@pkgdatadir@/pacman-${repo}.conf" \
 		-M "@pkgdatadir@/makepkg-${arch}.conf" \
 		"${chroots}/${repo}-${arch}/root" \
