From a1f1443c13ada00068162ccc87171664d83e5078 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 28 Dec 2014 22:34:04 -0700
Subject: [PATCH 6/8] archbuild: no setarch

---
 src/archbuild.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/archbuild.in b/src/archbuild.in
index 1f77e58..7dc7169 100644
--- a/src/archbuild.in
+++ b/src/archbuild.in
@@ -102,7 +102,7 @@ if ${clean_first} || [[ ! -d "${chroots}/${repo}-${arch}" ]]; then
 
 	rm -rf --one-file-system "${chroots}/${repo}-${arch}"
 	(umask 0022; mkdir -p "${chroots}/${repo}-${arch}")
-	setarch "${set_arch}" mkarchroot \
+	mkarchroot \
 		-C "${pacman_config}" \
 		-M "${makepkg_config}" \
 		"${chroots}/${repo}-${arch}/root" \
-- 
2.51.2

