From e4864e010881eeedb8efa8aac36fa43543dcdc94 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 3 Jul 2016 21:32:15 -0600
Subject: [PATCH 7/8] makechrootpkg: don't delete MAKEFLAGS and PACKAGER

---
 src/makechrootpkg.in | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/makechrootpkg.in b/src/makechrootpkg.in
index a3a2c0a..d2bf13c 100644
--- a/src/makechrootpkg.in
+++ b/src/makechrootpkg.in
@@ -178,9 +178,7 @@ prepare_chroot() {
 
 	$install -d "$copydir"/{build,startdir,{pkg,srcpkg,src,log}dest}
 
-	sed -e '/^MAKEFLAGS=/d' -e '/^PACKAGER=/d' -i "$copydir/etc/makepkg.conf"
-	for x in BUILDDIR=/build PKGDEST=/pkgdest SRCPKGDEST=/srcpkgdest SRCDEST=/srcdest LOGDEST=/logdest \
-		"MAKEFLAGS='${MAKEFLAGS:-}'" "PACKAGER='${PACKAGER:-}'"
+	for x in BUILDDIR=/build PKGDEST=/pkgdest SRCPKGDEST=/srcpkgdest SRCDEST=/srcdest LOGDEST=/logdest
 	do
 		grep -q "^$x" "$copydir/etc/makepkg.conf" && continue
 		echo "$x" >>"$copydir/etc/makepkg.conf"
-- 
2.51.2

