From 50a063470c8881025539379ee9af9f252eb5e457 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 3 Jul 2016 21:32:15 -0600
Subject: [PATCH 7/8] makechrootpkg: don't delete MAKEFLAGS and PACKAGER

---
 src/makechrootpkg.in | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/makechrootpkg.in b/src/makechrootpkg.in
index 951deec..598dc40 100644
--- a/src/makechrootpkg.in
+++ b/src/makechrootpkg.in
@@ -178,10 +178,7 @@ prepare_chroot() {
 
 	$install -d "$copydir"/{build,startdir,{pkg,srcpkg,src,log}dest}
 
-	sed -e '/^NPROC=/d' -e '/^MAKEFLAGS=/d' -e '/^PACKAGER=/d' -i "$copydir/etc/makepkg.conf"
-	local nproc="${NPROC:-}" makeflags="${MAKEFLAGS:-}" packager="${PACKAGER:-}"
-	for x in BUILDDIR=/build PKGDEST=/pkgdest SRCPKGDEST=/srcpkgdest SRCDEST=/srcdest LOGDEST=/logdest \
-		"NPROC=${nproc@Q}" "MAKEFLAGS=${makeflags@Q}" "PACKAGER=${packager@Q}"
+	for x in BUILDDIR=/build PKGDEST=/pkgdest SRCPKGDEST=/srcpkgdest SRCDEST=/srcdest LOGDEST=/logdest
 	do
 		grep -q "^$x" "$copydir/etc/makepkg.conf" && continue
 		echo "$x" >>"$copydir/etc/makepkg.conf"
-- 
2.54.0

