From 9e66d2ff9e974918683b0286b6d543547faa1dc1 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 3 Jul 2016 21:32:15 -0600
Subject: [PATCH 7/7] makechrootpkg: don't delete MAKEFLAGS and PACKAGER

---
 makechrootpkg.in | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/makechrootpkg.in b/makechrootpkg.in
index 9a69dda..a018130 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -165,9 +165,7 @@ prepare_chroot() {
 		$install -m 644 "$USER_HOME/$x" "$copydir/build/$x"
 	done
 
-	sed -e '/^MAKEFLAGS=/d' -e '/^PACKAGER=/d' -i "$copydir/etc/makepkg.conf"
-	for x in BUILDDIR=/build PKGDEST=/pkgdest SRCPKGDEST=/srcpkgdest SRCDEST=/srcdest LOGDEST=/logdest \
-		"MAKEFLAGS='$MAKEFLAGS'" "PACKAGER='$PACKAGER'"
+	for x in BUILDDIR=/build PKGDEST=/pkgdest SRCPKGDEST=/srcpkgdest SRCDEST=/srcdest LOGDEST=/logdest
 	do
 		grep -q "^$x" "$copydir/etc/makepkg.conf" && continue
 		echo "$x" >>"$copydir/etc/makepkg.conf"
-- 
2.9.0

