From 46b1986e26b017b3d38eaf238e3d8c8947243fb7 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 6 Oct 2015 17:39:00 -0600
Subject: [PATCH 7/7] arch-nspawn: keep stderr

---
 arch-nspawn.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch-nspawn.in b/arch-nspawn.in
index 3f3d563..13eba35 100644
--- a/arch-nspawn.in
+++ b/arch-nspawn.in
@@ -96,7 +96,7 @@ copy_hostconf
 
 eval $(grep '^CARCH=' "$working_dir/etc/makepkg.conf")
 
-exec systemd-nspawn 2>/dev/null \
+exec systemd-nspawn -q \
 	-D "$working_dir" \
 	--register=no \
 	"${mount_args[@]}" \
-- 
2.5.2

