From ad93f852bc62ef7a10b60ec7cf56ee1174e959da Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 3 Jul 2016 21:32:15 -0600
Subject: [PATCH 7/7] makechrootpkg: don't delete MAKEFLAGS and PACKAGER

---
 makechrootpkg.in | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/makechrootpkg.in b/makechrootpkg.in
index c13890b..6035971 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -184,9 +184,7 @@ prepare_chroot() {
 
 	$install -d "$copydir"/{build,startdir,{pkg,srcpkg,src,log}dest}
 
-	sed -e '/^MAKEFLAGS=/d' -e '/^PACKAGER=/d' -i "$copydir/etc/makepkg.conf"
-	for x in BUILDDIR=/build PKGDEST=/pkgdest SRCPKGDEST=/srcpkgdest SRCDEST=/srcdest LOGDEST=/logdest \
-		"MAKEFLAGS='${MAKEFLAGS:-}'" "PACKAGER='${PACKAGER:-}'"
+	for x in BUILDDIR=/build PKGDEST=/pkgdest SRCPKGDEST=/srcpkgdest SRCDEST=/srcdest LOGDEST=/logdest
 	do
 		grep -q "^$x" "$copydir/etc/makepkg.conf" && continue
 		echo "$x" >>"$copydir/etc/makepkg.conf"
-- 
2.21.0

