From 93b9e4890fe1dacad876f51d4a1cbe9abfab9333 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Thu, 22 Jul 2021 18:22:43 -0600
Subject: [PATCH 8/8] makechrootpkg: gotmpdir

---
 src/makechrootpkg.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/makechrootpkg.in b/src/makechrootpkg.in
index d2bf13c..aaff2d2 100644
--- a/src/makechrootpkg.in
+++ b/src/makechrootpkg.in
@@ -230,7 +230,7 @@ _chrootbuild() {
 	sudo --preserve-env=SOURCE_DATE_EPOCH \
 		--preserve-env=BUILDTOOL \
 		--preserve-env=BUILDTOOLVER \
-		-iu builduser DISTCC_IO_TIMEOUT=1200 DISTCC_DIR='/build/.distcc' bash -c 'cd /startdir; makepkg "$@"' -bash "$@"
+		-iu builduser DISTCC_IO_TIMEOUT=1200 DISTCC_DIR='/build/.distcc' GOTMPDIR='/build' bash -c 'cd /startdir; makepkg "$@"' -bash "$@"
 	ret=$?
 	case $ret in
 		0|14)
-- 
2.51.2

