From 7ce31cb2855ca525dba932e566646f2e337930ed Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Thu, 22 Jul 2021 18:22:43 -0600
Subject: [PATCH 8/9] makechrootpkg: gotmpdir

---
 makechrootpkg.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makechrootpkg.in b/makechrootpkg.in
index f0378b3..179e8c6 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -211,7 +211,7 @@ _chrootbuild() {
 	sudo --preserve-env=SOURCE_DATE_EPOCH \
 		--preserve-env=BUILDTOOL \
 		--preserve-env=BUILDTOOLVER \
-		-iu builduser DISTCC_IO_TIMEOUT=1200 DISTCC_DIR='/build/.distcc' bash -c 'cd /startdir; makepkg "$@"' -bash "$@"
+		-iu builduser DISTCC_IO_TIMEOUT=1200 DISTCC_DIR='/build/.distcc' GOTMPDIR='/build' bash -c 'cd /startdir; makepkg "$@"' -bash "$@"
 	ret=$?
 	case $ret in
 		0|14)
-- 
2.36.1

