From 44d15a4567a148d185af82f1b60bba7c48c17496 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 14 Nov 2017 22:33:27 -0700
Subject: [PATCH 8/8] fix namcap running by default

---
 makechrootpkg.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makechrootpkg.in b/makechrootpkg.in
index 6b53337..45ef1b9 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -417,7 +417,7 @@ main() {
 
 	download_sources "$copydir" "$makepkg_user"
 
-	prepare_chroot "$copydir" "$USER_HOME" "$keepbuilddir"
+	prepare_chroot "$copydir" "$USER_HOME" "$keepbuilddir" $run_namcap
 
 	if arch-nspawn $cache_dir "$copydir" \
 		--bind="$PWD:/startdir" \
-- 
2.14.1

