--- a/Makefile
+++ b/Makefile
@@ -28,6 +28,8 @@ BINPROGS = \
 	$(IN_PROGS)
 
 CONFIGFILES = \
+	makepkg-aarch64.conf \
+	makepkg-armv7h.conf \
 	makepkg-x86_64.conf \
 	makepkg-x86_64_v3.conf \
 	pacman-extra.conf \
@@ -59,10 +61,16 @@ COMMITPKG_LINKS = \
 	gnome-unstablepkg
 
 ARCHBUILD_LINKS = \
+	extra-aarch64-build \
+	extra-armv7h-build \
 	extra-x86_64-build \
 	extra-x86_64_v3-build \
+	testing-aarch64-build \
+	testing-armv7h-build \
 	testing-x86_64-build \
 	testing-x86_64_v3-build \
+	staging-aarch64-build \
+	staging-armv7h-build \
 	staging-x86_64-build \
 	staging-x86_64_v3-build \
 	multilib-build \

