From a7ded663b5a89110a899e130d05364d86940790b Mon Sep 17 00:00:00 2001
From: Aldrik Dunbar <aldrik@gmail.com>
Date: Wed, 21 Nov 2012 17:47:46 +0100
Subject: [PATCH] cubox : fix overlay issue with 480p videos (patch by Rabeeh
 Khoury)

---
 xbmc/cores/VideoRenderers/LinuxRendererDoveOverlay.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/xbmc/cores/VideoRenderers/LinuxRendererDoveOverlay.cpp b/xbmc/cores/VideoRenderers/LinuxRendererDoveOverlay.cpp
index c1fe0cf..ab96003 100644
--- a/xbmc/cores/VideoRenderers/LinuxRendererDoveOverlay.cpp
+++ b/xbmc/cores/VideoRenderers/LinuxRendererDoveOverlay.cpp
@@ -104,8 +104,14 @@ void CLinuxRendererDoveOverlay::ManageDisplayDove()
 
   if (m_format == RENDER_FMT_UYVY422)
   {
+    int delta;
     m_overlaySurface.videoMode = DOVEFB_VMODE_YUV422PACKED_SWAPYUorV;
     m_overlaySurface.viewPortInfo.ycPitch = (m_sourceRect.x2 - m_sourceRect.x1) * 2;
+    /* Align ycPitch to 16 pixels (32 bytes) since vmeta outputs in 16byte alignment per stride */
+    /* TODO check of decoded via vmeta or ffmpeg and change accordingly */
+    delta = m_overlaySurface.viewPortInfo.ycPitch % 32;
+    if (delta)
+      m_overlaySurface.viewPortInfo.ycPitch = m_overlaySurface.viewPortInfo.ycPitch - delta + 32;
     m_overlaySurface.viewPortInfo.uvPitch = 0;
   }
   else if (m_format == RENDER_FMT_YUV420P)
-- 
1.8.0

