diff -Nru openswan-2.4.14.orig/lib/liblwres/Makefile openswan-2.4.14/lib/liblwres/Makefile
--- openswan-2.4.14.orig/lib/liblwres/Makefile	2007-10-22 14:33:11.000000000 +0000
+++ openswan-2.4.14/lib/liblwres/Makefile	2009-04-08 23:07:06.000000000 +0000
@@ -17,8 +17,8 @@
 
 CINCLUDES =	-I${srcdir}/unix/include \
 		-I. -I./include -I${srcdir}/include 
-CDEFINES = -g
-CWARNINGS = -Werror
+CDEFINES =
+CWARNINGS =
 
 CFLAGS+=${CINCLUDES} ${CDEFINES} ${CWARNINGS}
 
diff -Nru openswan-2.4.14.orig/programs/Makefile.program openswan-2.4.14/programs/Makefile.program
--- openswan-2.4.14.orig/programs/Makefile.program	2007-06-19 14:49:19.000000000 +0000
+++ openswan-2.4.14/programs/Makefile.program	2009-04-08 23:08:31.000000000 +0000
@@ -34,7 +34,6 @@
 WERROR:= -Werror
 endif
 
-CFLAGS+= ${WERROR}
 
 ifneq ($(LD_LIBRARY_PATH),)
 LDFLAGS=-L$(LD_LIBRARY_PATH)
diff -Nru openswan-2.4.14.orig/testing/utils/ikeping/Makefile openswan-2.4.14/testing/utils/ikeping/Makefile
--- openswan-2.4.14.orig/testing/utils/ikeping/Makefile	2004-04-03 19:44:52.000000000 +0000
+++ openswan-2.4.14/testing/utils/ikeping/Makefile	2009-04-08 23:09:18.000000000 +0000
@@ -27,10 +27,9 @@
 FREESWANINCLS= -I$(FREESWANLIBDIR) -I${OPENSWANSRCDIR}
 FREESWANLIB=$(FREESWANLIBDIR)/libfreeswan.a
 
-CFLAGS = -g -Wall -W -Wmissing-prototypes -Wpointer-arith -Wbad-function-cast \
+CFLAGS = -Wall -W -Wmissing-prototypes -Wpointer-arith -Wbad-function-cast \
 	-Wcast-qual -Wmissing-declarations -Wwrite-strings 
 CFLAGS+= -Wstrict-prototypes 
-CFLAGS+= -Werror
 #CFLAGS+= -Wundef
 
 HDRDIRS =  $(FREESWANINCLS)
