diff -wbBur ktechlab-0.3/microbe/pic14.h ktechlab-0.3.my/microbe/pic14.h
--- ktechlab-0.3/microbe/pic14.h	2005-12-30 11:22:04.000000000 +0000
+++ ktechlab-0.3.my/microbe/pic14.h	2009-07-01 11:00:11.000000000 +0000
@@ -96,7 +96,7 @@
 			Delay_50S	= 4,
 		};
 		
-		PIC14::PIC14( Microbe * master, Type type );
+		PIC14( Microbe * master, Type type );
 		~PIC14();
 		
 		/**
diff -wbBur ktechlab-0.3/src/cells.cpp ktechlab-0.3.my/src/cells.cpp
--- ktechlab-0.3/src/cells.cpp	2005-12-30 11:22:12.000000000 +0000
+++ ktechlab-0.3.my/src/cells.cpp	2009-07-01 11:00:11.000000000 +0000
@@ -8,10 +8,9 @@
  *   (at your option) any later version.                                   *
  ***************************************************************************/
 
+#include <qt.h>
 #include "cells.h"
 
-
-
 #if 0
 class CellSmall
 {
diff -wbBur ktechlab-0.3/src/cnitem.cpp ktechlab-0.3.my/src/cnitem.cpp
--- ktechlab-0.3/src/cnitem.cpp	2005-12-30 11:22:12.000000000 +0000
+++ ktechlab-0.3.my/src/cnitem.cpp	2009-07-01 11:00:11.000000000 +0000
@@ -22,6 +22,14 @@
 
 #include <cmath>
 
+namespace std
+{
+inline static int abs(int a)
+{
+    return a >=0 ? a : -a;
+}
+}
+
 // Degrees per radian
 const double DPR = 57.29577951308232087665461840231273527024;
 
diff -wbBur ktechlab-0.3/src/conrouter.cpp ktechlab-0.3.my/src/conrouter.cpp
--- ktechlab-0.3/src/conrouter.cpp	2005-12-30 11:22:12.000000000 +0000
+++ ktechlab-0.3.my/src/conrouter.cpp	2009-07-01 11:00:11.000000000 +0000
@@ -8,6 +8,8 @@
  *   (at your option) any later version.                                   *
  ***************************************************************************/
 
+#include <qt.h>
+
 #include "conrouter.h"
 #include "icndocument.h"
 
@@ -16,6 +18,13 @@
 #include <assert.h>
 #include <cmath>
 
+namespace std
+{
+inline static int abs(int a)
+{
+    return a >=0 ? a : -a;
+}
+}
 inline static int toCanvas( int pos )
 {
 	return pos*8+4;
diff -wbBur ktechlab-0.3/src/core/main.cpp ktechlab-0.3.my/src/core/main.cpp
--- ktechlab-0.3/src/core/main.cpp	2005-12-30 11:22:10.000000000 +0000
+++ ktechlab-0.3.my/src/core/main.cpp	2009-07-01 11:22:15.000000000 +0000
@@ -17,6 +17,14 @@
 #include <kconfig.h>
 #include <klocale.h>
 
+#include <stdlib.h>
+extern void exit_cli(void);
+void exit_gpsim(int ret)
+{
+  exit_cli();
+  exit(ret);
+}
+
 static const char description[] =
     I18N_NOOP("An IDE for microcontrollers and electronics");
 
diff -wbBur ktechlab-0.3/src/electronics/simulation/matrix.cpp ktechlab-0.3.my/src/electronics/simulation/matrix.cpp
--- ktechlab-0.3/src/electronics/simulation/matrix.cpp	2005-12-30 11:22:06.000000000 +0000
+++ ktechlab-0.3.my/src/electronics/simulation/matrix.cpp	2009-07-01 11:00:11.000000000 +0000
@@ -288,7 +288,7 @@
 Map::Map( const uint size )
 {
 	m_size = size;
-	m_map = new ETMap( m_size, m_size );
+	m_map = new ETMap(m_size, std::vector<uint>(m_size));
 	reset();
 }
 
diff -wbBur ktechlab-0.3/src/itemdocument.h ktechlab-0.3.my/src/itemdocument.h
--- ktechlab-0.3/src/itemdocument.h	2005-12-30 11:22:12.000000000 +0000
+++ ktechlab-0.3.my/src/itemdocument.h	2009-07-01 11:00:11.000000000 +0000
@@ -112,7 +112,7 @@
 		 * (such as PIC/START) have restrictions, and can only have one instance of
 		 * themselves on the canvas, and adds the operation to the undo list
 		 */
-		virtual Item* addItem( const QString &id, const QPoint &p, bool newItem ) = 0l;
+		virtual Item* addItem( const QString &id, const QPoint &p, bool newItem ) = 0;
 		/**
 		 * @returns A pointer to the canvas
 		 */
@@ -215,7 +215,7 @@
 		 * Set the given QCanvasItem (which will attempt to be casted to known
 		 * items to be deleted.
 		 */
-		virtual void appendDeleteList( QCanvasItem * ) = 0l;
+		virtual void appendDeleteList( QCanvasItem * ) = 0;
 		/**
 		 * Save the current state of the document to the undo/redo history.
 		 * @param actionTicket if this is non-negative, and the last state save
diff -wbBur ktechlab-0.3/src/nodegroup.cpp ktechlab-0.3.my/src/nodegroup.cpp
--- ktechlab-0.3/src/nodegroup.cpp	2005-12-30 11:22:12.000000000 +0000
+++ ktechlab-0.3.my/src/nodegroup.cpp	2009-07-01 11:00:11.000000000 +0000
@@ -17,6 +17,14 @@
 #include <kdebug.h>
 #include <assert.h>
 
+namespace std
+{
+inline static int abs(int a)
+{
+    return a >=0 ? a : -a;
+}
+}
+
 NodeGroup::NodeGroup( ICNDocument *icnDocument, const char *name )
 	: QObject( icnDocument, name )
 {
