--- cups-2.4.0/config-scripts/cups-directories.m4	2021-11-29 16:12:17.094244942 +0100
+++ cups-2.4.0/config-scripts/cups-directories.m4.new	2021-11-29 16:15:16.958747398 +0100
@@ -268,7 +268,7 @@
 	CUPS_STATEDIR="$CUPS_SERVERROOT"
     ], [*], [
 	# All others
-	CUPS_STATEDIR="$localstatedir/run/cups"
+	CUPS_STATEDIR="/run/cups"
     ])
 ])
 AC_DEFINE_UNQUOTED([CUPS_STATEDIR], ["$CUPS_STATEDIR"], [Location of transient state files.])
