--- cups-2.3.0/scheduler/cups.service.in	2019-08-23 17:19:38.000000000 +0200
+++ cups-2.3.0/scheduler/cups.service.in.new	2019-08-25 20:20:25.805339727 +0200
@@ -1,8 +1,9 @@
 [Unit]
 Description=CUPS Scheduler
 Documentation=man:cupsd(8)
-After=network.target sssd.service ypbind.service
+After=network-online.target sssd.service ypbind.service
 Requires=cups.socket
+Wants=network-online.target

 [Service]
 ExecStart=@sbindir@/cupsd -l
 
