diff -up cups-2.0.0/cups/util.c.str4500 cups-2.0.0/cups/util.c
--- cups-2.0.0/cups/util.c.str4500	2014-10-15 12:59:27.105942488 +0100
+++ cups-2.0.0/cups/util.c	2014-10-15 13:03:38.618187112 +0100
@@ -846,10 +846,10 @@ cupsGetPPD3(http_t     *http,		/* I  - H
 
     snprintf(ppdname, sizeof(ppdname), "%s/ppd/%s.ppd", cg->cups_serverroot,
              name);
-    if (!stat(ppdname, &ppdinfo))
+    if (!stat(ppdname, &ppdinfo) && !access(ppdname, R_OK))
     {
      /*
-      * OK, the file exists, use it!
+      * OK, the file exists and is readable, use it!
       */
 
       if (buffer[0])

