Index: conf/cups-files.conf.in
===================================================================
--- conf/cups-files.conf.in	(revision 12220)
+++ conf/cups-files.conf.in	(working copy)
@@ -15,6 +15,7 @@
 #Group @CUPS_GROUP@

 # Administrator user group, used to match @SYSTEM in cupsd.conf policy rules...
+# This cannot contain the Group value for security reasons...
 SystemGroup @CUPS_SYSTEM_GROUPS@
 @CUPS_SYSTEM_AUTHKEY@

Index: scheduler/conf.c
===================================================================
--- scheduler/conf.c	(revision 12220)
+++ scheduler/conf.c	(working copy)
@@ -995,6 +995,9 @@

       cupsdLogMessage(CUPSD_LOG_NOTICE,
                       "Group and SystemGroup cannot use the same groups.");
+      if (FatalErrors & (CUPSD_FATAL_CONFIG | CUPSD_FATAL_PERMISSIONS))
+        return (0);
+
       cupsdLogMessage(CUPSD_LOG_INFO, "Resetting Group to \"nobody\"...");

       group = getgrnam("nobody");
