Index: scheduler/main.c
===================================================================
--- scheduler/main.c	(revision 12213)
+++ scheduler/main.c	(working copy)
@@ -763,6 +763,9 @@
 
     if (timeout == 86400 && OnDemand && IdleExitTimeout &&
         !cupsArrayCount(ActiveJobs) &&
+#  ifdef HAVE_SYSTEMD
+        !WebInterface &&
+#  endif /* HAVE_SYSTEMD */
 	(!Browsing || !BrowseLocalProtocols || !cupsArrayCount(Printers)))
     {
       timeout		= IdleExitTimeout;
@@ -2114,8 +2117,12 @@
   * jobs or shared printers to advertise...
   */
 
-  if (cupsArrayCount(ActiveJobs) ||
+  if (cupsArrayCount(ActiveJobs) ||	/* Active jobs */
+#  ifdef HAVE_SYSTEMD
+      WebInterface ||			/* Web interface enabled */
+#  endif /* HAVE_SYSTEMD */
       (Browsing && BrowseLocalProtocols && cupsArrayCount(Printers)))
+					/* Printers being shared */
   {
     cupsdLogMessage(CUPSD_LOG_DEBUG, "Creating keep-alive file \"" CUPS_KEEPALIVE "\".");
 
Index: scheduler/org.cups.cupsd.socket.in
===================================================================
--- scheduler/org.cups.cupsd.socket.in	(revision 12213)
+++ scheduler/org.cups.cupsd.socket.in	(working copy)
@@ -3,10 +3,6 @@
 
 [Socket]
 ListenStream=@CUPS_DEFAULT_DOMAINSOCKET@
-ListenStream=[::1]:631
-ListenStream=127.0.0.1:631
-BindIPv6Only=ipv6-only
-ReusePort=true
 
 [Install]
 WantedBy=sockets.target
