diff -up nautilus-2.25.93/nautilus.desktop.in.condrestart nautilus-2.25.93/nautilus.desktop.in
--- nautilus-2.25.93/nautilus.desktop.in.condrestart	2009-03-12 10:49:41.449753624 -0400
+++ nautilus-2.25.93/nautilus.desktop.in	2009-03-12 10:49:51.814754208 -0400
@@ -17,3 +17,4 @@ X-GNOME-Autostart-Phase=Desktop
 X-GNOME-Autostart-Notify=true
 X-GNOME-AutoRestart=true
 X-GNOME-Provides=filemanager
+AutostartCondition=GNOME /apps/nautilus/preferences/show_desktop
diff -up nautilus-2.25.93/nautilus.desktop.in.in.condrestart nautilus-2.25.93/nautilus.desktop.in.in
--- nautilus-2.25.93/nautilus.desktop.in.in.condrestart	2009-03-12 10:47:41.681753834 -0400
+++ nautilus-2.25.93/nautilus.desktop.in.in	2009-03-12 10:49:29.663754303 -0400
@@ -17,3 +17,4 @@ X-GNOME-Autostart-Phase=Desktop
 X-GNOME-Autostart-Notify=true
 X-GNOME-AutoRestart=true
 X-GNOME-Provides=filemanager
+AutostartCondition=GNOME /apps/nautilus/preferences/show_desktop
