diff -ur redland-1.0.9.orig/src/Makefile.am redland-1.0.9/src/Makefile.am
--- redland-1.0.9.orig/src/Makefile.am	2009-04-13 04:01:30.000000000 +0300
+++ redland-1.0.9/src/Makefile.am	2009-08-18 17:13:24.000000000 +0300
@@ -119,7 +119,7 @@
 plugin_LTLIBRARIES += librdf_storage_tstore.la
 librdf_storage_tstore_la_SOURCES = rdf_storage_tstore.c
 librdf_storage_tstore_la_CPPFLAGS = @TSTORE_CPPFLAGS@
-librdf_storage_tstore_la_LIBADD = @TSTORE_LIBS@
+librdf_storage_tstore_la_LIBADD = @TSTORE_LIBS@ librdf.la
 librdf_storage_tstore_la_LDFLAGS = -module -avoid-version
 endif
 
@@ -127,7 +127,7 @@
 plugin_LTLIBRARIES += librdf_storage_mysql.la
 librdf_storage_mysql_la_SOURCES = rdf_storage_mysql.c
 librdf_storage_mysql_la_CPPFLAGS = @MYSQL_CPPFLAGS@
-librdf_storage_mysql_la_LIBADD = @MYSQL_LIBS@
+librdf_storage_mysql_la_LIBADD = @MYSQL_LIBS@ librdf.la
 librdf_storage_mysql_la_LDFLAGS = -module -avoid-version
 endif
 
@@ -135,7 +135,7 @@
 plugin_LTLIBRARIES += librdf_storage_sqlite.la
 librdf_storage_sqlite_la_SOURCES = rdf_storage_sqlite.c
 librdf_storage_sqlite_la_CPPFLAGS = @SQLITE_CPPFLAGS@
-librdf_storage_sqlite_la_LIBADD = @SQLITE_LIBS@
+librdf_storage_sqlite_la_LIBADD = @SQLITE_LIBS@ librdf.la
 librdf_storage_sqlite_la_LDFLAGS = -module -avoid-version
 endif
 
@@ -143,7 +143,7 @@
 plugin_LTLIBRARIES += librdf_storage_postgresql.la
 librdf_storage_postgresql_la_SOURCES = rdf_storage_postgresql.c
 librdf_storage_postgresql_la_CPPFLAGS = @POSTGRESQL_CPPFLAGS@
-librdf_storage_postgresql_la_LIBADD = @POSTGRESQL_LIBS@
+librdf_storage_postgresql_la_LIBADD = @POSTGRESQL_LIBS@ librdf.la
 librdf_storage_postgresql_la_LDFLAGS = -module -avoid-version
 endif
 
@@ -202,7 +202,8 @@
 rdf_query_test rdf_serializer_test rdf_stream_test rdf_iterator_test \
 rdf_init_test rdf_cache_test
 
-TESTS_ENVIRONMENT=REDLAND_MODULE_PATH=
+# Set the place to find storage modules for testing
+TESTS_ENVIRONMENT=REDLAND_MODULE_PATH=$(abs_builddir)/.libs
 
 CLEANFILES=$(TESTS) $(local_tests) test test*.db test.rdf
 
diff -ur redland-1.0.9.orig/src/Makefile.in redland-1.0.9/src/Makefile.in
--- redland-1.0.9.orig/src/Makefile.in	2009-04-13 04:04:20.000000000 +0300
+++ redland-1.0.9/src/Makefile.in	2009-08-18 17:13:24.000000000 +0300
@@ -141,7 +141,8 @@
 librdf_la_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(librdf_la_LDFLAGS) $(LDFLAGS) -o $@
-librdf_storage_mysql_la_DEPENDENCIES =
+@MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@librdf_storage_mysql_la_DEPENDENCIES =  \
+@MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@	librdf.la
 am__librdf_storage_mysql_la_SOURCES_DIST = rdf_storage_mysql.c
 @MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@am_librdf_storage_mysql_la_OBJECTS = librdf_storage_mysql_la-rdf_storage_mysql.lo
 librdf_storage_mysql_la_OBJECTS =  \
@@ -151,7 +152,8 @@
 	$(librdf_storage_mysql_la_LDFLAGS) $(LDFLAGS) -o $@
 @MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@am_librdf_storage_mysql_la_rpath =  \
 @MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@	-rpath $(plugindir)
-librdf_storage_postgresql_la_DEPENDENCIES =
+@MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@librdf_storage_postgresql_la_DEPENDENCIES =  \
+@MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@	librdf.la
 am__librdf_storage_postgresql_la_SOURCES_DIST =  \
 	rdf_storage_postgresql.c
 @MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@am_librdf_storage_postgresql_la_OBJECTS = librdf_storage_postgresql_la-rdf_storage_postgresql.lo
@@ -164,7 +166,8 @@
 @MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@am_librdf_storage_postgresql_la_rpath =  \
 @MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@	-rpath \
 @MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@	$(plugindir)
-librdf_storage_sqlite_la_DEPENDENCIES =
+@MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@librdf_storage_sqlite_la_DEPENDENCIES =  \
+@MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@	librdf.la
 am__librdf_storage_sqlite_la_SOURCES_DIST = rdf_storage_sqlite.c
 @MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@am_librdf_storage_sqlite_la_OBJECTS = librdf_storage_sqlite_la-rdf_storage_sqlite.lo
 librdf_storage_sqlite_la_OBJECTS =  \
@@ -174,7 +177,8 @@
 	$(librdf_storage_sqlite_la_LDFLAGS) $(LDFLAGS) -o $@
 @MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@am_librdf_storage_sqlite_la_rpath =  \
 @MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@	-rpath $(plugindir)
-librdf_storage_tstore_la_DEPENDENCIES =
+@MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@librdf_storage_tstore_la_DEPENDENCIES =  \
+@MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@	librdf.la
 am__librdf_storage_tstore_la_SOURCES_DIST = rdf_storage_tstore.c
 @MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@am_librdf_storage_tstore_la_OBJECTS = librdf_storage_tstore_la-rdf_storage_tstore.lo
 librdf_storage_tstore_la_OBJECTS =  \
@@ -463,19 +467,19 @@
 @MODULAR_LIBRDF_TRUE@	$(am__append_6)
 @MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@librdf_storage_tstore_la_SOURCES = rdf_storage_tstore.c
 @MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@librdf_storage_tstore_la_CPPFLAGS = @TSTORE_CPPFLAGS@
-@MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@librdf_storage_tstore_la_LIBADD = @TSTORE_LIBS@
+@MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@librdf_storage_tstore_la_LIBADD = @TSTORE_LIBS@ librdf.la
 @MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@librdf_storage_tstore_la_LDFLAGS = -module -avoid-version
 @MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@librdf_storage_mysql_la_SOURCES = rdf_storage_mysql.c
 @MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@librdf_storage_mysql_la_CPPFLAGS = @MYSQL_CPPFLAGS@
-@MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@librdf_storage_mysql_la_LIBADD = @MYSQL_LIBS@
+@MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@librdf_storage_mysql_la_LIBADD = @MYSQL_LIBS@ librdf.la
 @MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@librdf_storage_mysql_la_LDFLAGS = -module -avoid-version
 @MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@librdf_storage_sqlite_la_SOURCES = rdf_storage_sqlite.c
 @MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@librdf_storage_sqlite_la_CPPFLAGS = @SQLITE_CPPFLAGS@
-@MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@librdf_storage_sqlite_la_LIBADD = @SQLITE_LIBS@
+@MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@librdf_storage_sqlite_la_LIBADD = @SQLITE_LIBS@ librdf.la
 @MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@librdf_storage_sqlite_la_LDFLAGS = -module -avoid-version
 @MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@librdf_storage_postgresql_la_SOURCES = rdf_storage_postgresql.c
 @MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@librdf_storage_postgresql_la_CPPFLAGS = @POSTGRESQL_CPPFLAGS@
-@MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@librdf_storage_postgresql_la_LIBADD = @POSTGRESQL_LIBS@
+@MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@librdf_storage_postgresql_la_LIBADD = @POSTGRESQL_LIBS@ librdf.la
 @MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@librdf_storage_postgresql_la_LDFLAGS = -module -avoid-version
 librdf_la_LIBADD = \
 @LTLIBOBJS@ @DIGEST_OBJS@ @HASH_OBJS@ \
@@ -502,7 +506,9 @@
 rdf_query_test rdf_serializer_test rdf_stream_test rdf_iterator_test \
 rdf_init_test rdf_cache_test
 
-TESTS_ENVIRONMENT = REDLAND_MODULE_PATH=
+
+# Set the place to find storage modules for testing
+TESTS_ENVIRONMENT = REDLAND_MODULE_PATH=$(abs_builddir)/.libs
 CLEANFILES = $(TESTS) $(local_tests) test test*.db test.rdf
 AM_CPPFLAGS = @LIBRDF_INTERNAL_CPPFLAGS@ @LIBRDF_CPPFLAGS@ \
 	@LIBRDF_EXTERNAL_CPPFLAGS@ -DPKGDATADIR="\"$(pkgdatadir)\""
