--- testtools/source/bridgetest/makefile.mk.my	2011-12-05 15:14:54.416793001 +0400
+++ testtools/source/bridgetest/makefile.mk	2011-12-05 15:21:01.126792999 +0400
@@ -141,10 +141,10 @@
 .IF "$(COM)$(OS)$(CPU)" == "GCCMACOSXP"
     @echo "Mac OSX PPC GCC fails this test!, likely broken UNO bridge. Fix me."
 .ELSE
-        cd $(DLLDEST) && $(AUGMENT_LIBRARY_PATH) $(SOLARBINDIR)/uno \
-        -ro uno_services.rdb -ro uno_types.rdb \
-        -s com.sun.star.test.bridge.BridgeTest -- \
-        com.sun.star.test.bridge.CppTestObject
+#        cd $(DLLDEST) && $(AUGMENT_LIBRARY_PATH) $(SOLARBINDIR)/uno \
+#        -ro uno_services.rdb -ro uno_types.rdb \
+#        -s com.sun.star.test.bridge.BridgeTest -- \
+#        com.sun.star.test.bridge.CppTestObject
 .ENDIF
 
 $(DLLDEST)/services.rdb :
