--- testtools/source/bridgetest/makefile.mk	2011-11-23 12:00:26.079719303 +0000
+++ testtools/source/bridgetest/makefile.mk	2011-11-23 12:00:52.176244593 +0000
@@ -123,29 +123,6 @@
 # --- Targets ------------------------------------------------------
 .ENDIF # L10N_framework
 
-.INCLUDE :	target.mk
-.IF "$(L10N_framework)"==""
-ALLTAR: \
-       runtest \
-        $(DLLDEST)/services.rdb \
-        $(DLLDEST)$/uno_types.rdb \
-        $(DLLDEST)$/uno_services.rdb \
-        $(DLLDEST)$/bridgetest_server$(BATCH_SUFFIX) \
-        $(DLLDEST)$/bridgetest_client$(BATCH_SUFFIX) \
-        $(JAVATARGETS)
-
-#################################################################
-
-runtest : $(DLLDEST)$/uno_types.rdb $(DLLDEST)$/uno_services.rdb makefile.mk \
-        $(SHL1TARGETN) $(SHL2TARGETN) $(SHL3TARGETN)
-.IF "$(COM)$(OS)$(CPU)" == "GCCMACOSXP"
-    @echo "Mac OSX PPC GCC fails this test!, likely broken UNO bridge. Fix me."
-.ELSE
-       cd $(DLLDEST) && $(AUGMENT_LIBRARY_PATH) $(SOLARBINDIR)/uno \
-        -ro uno_services.rdb -ro uno_types.rdb \
-        -s com.sun.star.test.bridge.BridgeTest -- \
-        com.sun.star.test.bridge.CppTestObject
-.ENDIF
 
 $(DLLDEST)/services.rdb :
     $(COPY) $(SOLARXMLDIR)/ure/services.rdb $@
@@ -226,4 +203,3 @@
         'vnd.sun.star.expand:$${{MY_CLASSDIR_URL}}testComponent.jar' -o $@ \
         $(SOLARENV)/bin/createcomponent.xslt testComponent.component
 
-.ENDIF # L10N_framework
