From 90de903b28cb1884042be0eae3f8a55458c5bc29 Mon Sep 17 00:00:00 2001
From: Caolán McNamara <caolanm@redhat.com>
Date: Tue, 12 Apr 2011 19:35:51 +0000
Subject: duplicate NULLs

---
diff --git a/ct2n/prj/build.lst b/ct2n/prj/build.lst
index 91f44b7..714d926 100644
--- a/ct2n/prj/build.lst
+++ b/ct2n/prj/build.lst
@@ -1,3 +1,3 @@
-ct2n	ct2n	:	solenv NULL NULL
+ct2n	ct2n	:	solenv NULL
 ct2n	ct2n		usr1	-	all	ct2n_mkout NULL
 ct2n	ct2n      nmake	-	all	ct2n_ct2n NULL
--
cgit v0.8.3-6-g21f6
