--- vbahelper/inc/vbahelper/vbacollectionimpl.hxx	2011-01-28 20:27:51.507604173 +0000
+++ vbahelper/inc/vbahelper/vbacollectionimpl.hxx	2011-01-28 20:28:26.230045727 +0000
@@ -238,7 +238,7 @@
 
 // including a HelperInterface implementation
 template< typename Ifc1 >
-class ScVbaCollectionBase : public InheritedHelperInterfaceImpl< Ifc1 >
+class VBAHELPER_DLLPUBLIC ScVbaCollectionBase : public InheritedHelperInterfaceImpl< Ifc1 >
 {
 typedef InheritedHelperInterfaceImpl< Ifc1 > BaseColBase;
 protected:
--- sc/Library_vbaobj.mk
+++ sc/Library_vbaobj.mk
@@ -118,7 +118,6 @@ $(eval $(call gb_Library_add_exception_objects,vbaobj,\
         sc/source/ui/vba/vbaquerytable \
         sc/source/ui/vba/vbarange \
         sc/source/ui/vba/vbasheetobject \
-        sc/source/ui/vba/vbasheetobjects \
         sc/source/ui/vba/vbastyle \
         sc/source/ui/vba/vbastyles \
         sc/source/ui/vba/vbatextboxshape \
@@ -133,6 +132,11 @@ $(eval $(call gb_Library_add_exception_objects,vbaobj,\
         sc/source/ui/vba/vbawsfunction \
 ))
 
+$(eval $(call gb_Library_add_cxxobjects,vbaobj,\
+        sc/source/ui/vba/vbasheetobjects \
+        , $(gb_COMPILERNOOPTFLAGS) $(gb_LinkTarget_EXCEPTIONFLAGS) \
+))
+
 ifneq (,$(filter LINUX DRAGONFLY OPENBSD FREEBSD NETBSD, $(OS)))
 $(eval $(call gb_Library_set_ldflags,vbaobj,\
 	$$(LDFLAGS) \
