--- a/source/blender/blenkernel/intern/writeffmpeg.org.c	2021-05-05 15:45:54.647257119 -0700
+++ a/source/blender/blenkernel/intern/writeffmpeg.c	2021-05-05 15:47:57.737338545 -0700
@@ -163,6 +163,7 @@
   frame->pts = context->audio_time / av_q2d(c->time_base);
   frame->nb_samples = context->audio_input_samples;
   frame->format = c->sample_fmt;
+  frame->channels = c->channels;
 #      ifdef FFMPEG_HAVE_FRAME_CHANNEL_LAYOUT
   frame->channel_layout = c->channel_layout;
 #      endif
