diff --git intern/cycles/util/util_optimization.h intern/cycles/util/util_optimization.h
index c951c35..48a7e0f 100644
--- intern/cycles/util/util_optimization.h
+++ intern/cycles/util/util_optimization.h
@@ -101,8 +101,10 @@
 
 #ifdef _MSC_VER
 #include <intrin.h>
-#else
+#elif defined(GNUC) && (defined(x86_64) || defined(i386))
 #include <x86intrin.h>
+#elif defined(GNUC) && defined(ARM_NEON)
+#include <arm_neon.h>
 #endif
 
 #else
