diff --git a/io_scene_gltf2/io/com/gltf2_io_draco_compression_extension.py b/io_scene_gltf2/io/com/gltf2_io_draco_compression_extension.py
index e7518ac4..c0d78564 100644
--- a/io_scene_gltf2/io/com/gltf2_io_draco_compression_extension.py
+++ b/io_scene_gltf2/io/com/gltf2_io_draco_compression_extension.py
@@ -13,6 +13,7 @@
 # limitations under the License.
 
 import os
+import site
 import sys
 from pathlib import Path
 import bpy
@@ -30,7 +31,7 @@ def dll_path() -> Path:
     python_lib = Path('{v[0]}.{v[1]}/python/lib'.format(v=bpy.app.version))
     python_version = 'python{v[0]}.{v[1]}'.format(v=sys.version_info)
 
-    path = os.environ.get('BLENDER_EXTERN_DRACO_LIBRARY_PATH')
+    path = site.getsitepackages()[0]
     if path is None:
         path = {
             'win32': blender_root / python_lib / 'site-packages',
