diff --git a/CMakeLists.txt b/CMakeLists.txt
index 75f5795db68..d50686adfb0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -754,10 +754,6 @@ endif()
 set_and_warn_dependency(WITH_PYTHON WITH_CYCLES        OFF)
 set_and_warn_dependency(WITH_PYTHON WITH_DRACO         OFF)
 
-if(WITH_DRACO AND NOT WITH_PYTHON_INSTALL)
-  message(STATUS "WITH_DRACO requires WITH_PYTHON_INSTALL to be ON, disabling WITH_DRACO for now")
-  set(WITH_DRACO OFF)
-endif()
 
 # enable boost for cycles, audaspace or i18n
 # otherwise if the user disabled
