--- a/blender-addons/io_scene_gltf2/io/imp/gltf2_io_gltf.py	2020-12-20 16:36:07.512946749 -0800
+++ b/blender-addons/io_scene_gltf2/io/imp/gltf2_io_gltf.py	2020-12-20 16:40:12.173781840 -0800
@@ -61,7 +61,7 @@
         def bad_constant(val):
             raise ImportError('Bad glTF: json contained %s' % val)
         try:
-            return json.loads(bytes(content), encoding='utf-8', parse_constant=bad_constant)
+            return json.loads(bytes(content), parse_constant=bad_constant)
         except ValueError as e:
             raise ImportError('Bad glTF: json error: %s' % e.args[0])
 
