commit 22bba02bc8109f392fce50a24a4825390bca1bda
Author: Gaia Clary <gaia.clary@machinimatrix.org>
Date:   Tue Dec 4 20:28:20 2018 +0100

    fix T58568: used wrong case in variable name

diff --git a/source/blender/collada/DocumentImporter.cpp b/source/blender/collada/DocumentImporter.cpp
index 24b7fc4317d..7f93dd4823f 100644
--- a/source/blender/collada/DocumentImporter.cpp
+++ b/source/blender/collada/DocumentImporter.cpp
@@ -1352,7 +1352,7 @@ bool DocumentImporter::writeAnimationList(const COLLADAFW::AnimationList *animat
 #if OPENCOLLADA_WITH_ANIMATION_CLIP
 // Since opencollada 1.6.68
 // called on post-process stage after writeVisualScenes
-bool DocumentImporter::writeAnimationClip(const COLLADAFW::AnimationClip *AnimationClip)
+bool DocumentImporter::writeAnimationClip(const COLLADAFW::AnimationClip *animationClip)
 {
 	if (mImportStage != General)
 		return true;
diff --git a/source/blender/collada/DocumentImporter.h b/source/blender/collada/DocumentImporter.h
index 5e9f899d42c..110389030de 100644
--- a/source/blender/collada/DocumentImporter.h
+++ b/source/blender/collada/DocumentImporter.h
@@ -108,7 +108,7 @@ public:
 
 #if OPENCOLLADA_WITH_ANIMATION_CLIP
 	// Please enable this when building with Collada 1.6.65 or newer (also in DocumentImporter.cpp)
-	bool DocumentImporter::writeAnimationClip(const COLLADAFW::AnimationClip *AnimationClip);
+	bool DocumentImporter::writeAnimationClip(const COLLADAFW::AnimationClip *animationClip);
 #endif
 
 	bool writeGeometry(const COLLADAFW::Geometry*);
