diff --git a/object_facemap_auto/auto_fmap_widgets.py b/object_facemap_auto/auto_fmap_widgets.py
index 5b26f75..646e844 100644
--- a/object_facemap_auto/auto_fmap_widgets.py
+++ b/object_facemap_auto/auto_fmap_widgets.py
@@ -324,7 +324,7 @@ class AutoFaceMapWidgetGroup(ManipulatorGroup):
 
         # foo;bar=baz;bonzo=bingo --> {"bar": baz", "bonzo": bingo}
         mpr.fmap_target_rules = dict(
-            item.partition("=")[::2] for item in fmap_rules,
+            item.partition("=")[::2] for item in fmap_rules
         )
 
         # XXX, we might want to have some way to extract a 'center' from a face-map
