diff --git a/release/scripts/addons/io_mesh_stl/blender_utils.py b/release/scripts/addons/io_mesh_stl/blender_utils.py
index 864335ab..c74853db 100644
--- a/release/scripts/addons/io_mesh_stl/blender_utils.py
+++ b/release/scripts/addons/io_mesh_stl/blender_utils.py
@@ -86,7 +86,7 @@ def faces_from_mesh(ob, global_matrix, use_mesh_modifiers=False, triangulate=Tru
     try:
         mesh = ob.to_mesh(bpy.context.scene, use_mesh_modifiers, "PREVIEW")
     except RuntimeError:
-        raise StopIteration
+        return
 
     mat = global_matrix * ob.matrix_world
     mesh.transform(mat)
