diff -ru src/dialogs/export.cpp src/dialogs/export.cpp
--- src/dialogs/export.cpp	2009-05-25 02:21:18.067302917 +0300
+++ src/dialogs/export.cpp	2009-05-25 02:21:17.804059091 +0300
@@ -1041,9 +1043,7 @@
 static gchar *
 filename_add_extension (const gchar *filename, const gchar *extension)
 {
-  gchar *dot;
-
-  dot = strrchr (filename, '.');
+  const gchar *dot = strrchr (filename, '.');
   if ( !dot )
     return g_strconcat (filename, ".", extension, NULL);
   {
diff -ru src/document.cpp src/document.cpp
--- src/document.cpp	2009-05-25 02:21:18.052299751 +0300
+++ src/document.cpp	2009-05-25 02:21:17.797058619 +0300
@@ -746,7 +746,7 @@
         }

         if ( NULL != document_language ) {
-            gchar *pos = strchr(document_language, '_');
+	    const char *pos = strchr(document_language, '_');
             if ( NULL != pos ) {
                 return Glib::ustring(document_language, pos - document_language);
             }
diff -ru src/dom/io/domstream.cpp src/dom/io/domstream.cpp
--- src/dom/io/domstream.cpp	2009-05-25 02:21:18.011048623 +0300
+++ src/dom/io/domstream.cpp	2009-05-25 02:21:17.755301212 +0300
@@ -34,6 +34,7 @@
  *
  */

+#include <cstdio>
 #include <math.h>
 #include <stdarg.h>

diff -ru src/dom/io/gzipstream.cpp src/dom/io/gzipstream.cpp
--- src/dom/io/gzipstream.cpp	2009-05-25 02:21:18.012048750 +0300
+++ src/dom/io/gzipstream.cpp	2009-05-25 02:21:17.755301212 +0300
@@ -29,6 +29,8 @@

 #include "gzipstream.h"

+#include <cstdio>
+
 #include "dom/util/ziptool.h"


diff -ru src/dom/io/socket.cpp src/dom/io/socket.cpp
--- src/dom/io/socket.cpp	2009-05-25 02:21:18.011048623 +0300
+++ src/dom/io/socket.cpp	2009-05-25 02:21:17.754336705 +0300
@@ -35,6 +35,7 @@
 #include <sys/filio.h>   // needed on Solaris 8
 #endif

+#include <cstdio>
 #include "socket.h"
 #include "dom/util/thread.h"

diff -ru src/dom/io/uristream.cpp src/dom/io/uristream.cpp
--- src/dom/io/uristream.cpp	2009-05-25 02:21:18.012048750 +0300
+++ src/dom/io/uristream.cpp	2009-05-25 02:21:17.755301212 +0300
@@ -41,6 +41,7 @@


 #include "uristream.h"
+#include <cstdio>



diff -ru src/dom/lsimpl.cpp src/dom/lsimpl.cpp
--- src/dom/lsimpl.cpp	2009-05-25 02:21:18.014048586 +0300
+++ src/dom/lsimpl.cpp	2009-05-25 02:21:17.760050211 +0300
@@ -33,6 +33,7 @@
 #include "lsimpl.h"

 #include <stdarg.h>
+#include <cstdio>

 namespace org
 {
diff -ru src/dom/svg/svgparser.cpp src/dom/svg/svgparser.cpp
--- src/dom/svg/svgparser.cpp	2009-05-25 02:21:18.014048586 +0300
+++ src/dom/svg/svgparser.cpp	2009-05-25 02:21:17.759333709 +0300
@@ -32,6 +32,7 @@
 #include "dom/cssparser.h"
 #include "dom/charclass.h"

+#include <cstdio>
 #include <stdarg.h>

 #define SVG_NAMESPACE "http://www.w3.org/2000/svg"
diff -ru src/prefix.cpp src/prefix.cpp
--- src/prefix.cpp	2009-05-25 02:21:18.037083918 +0300
+++ src/prefix.cpp	2009-05-25 02:21:17.787058605 +0300
@@ -355,7 +355,8 @@
 char *
 br_extract_dir (const char *path)
 {
-	char *end, *result;
+	const char *end;
+	char *result;

 	br_return_val_if_fail (path != (char*)NULL, (char*)NULL);

@@ -390,7 +391,8 @@
 char *
 br_extract_prefix (const char *path)
 {
-	char *end, *tmp, *result;
+	const char *end;
+	char *tmp, *result;

 	br_return_val_if_fail (path != (char*)NULL, (char*)NULL);

--- src/libavoid/vertices.h.orig	2009-07-06 20:37:23.000000000 -0400
+++ src/libavoid/vertices.h	2009-07-06 20:37:46.000000000 -0400
@@ -23,6 +23,7 @@
 #ifndef AVOID_VERTICES_H
 #define AVOID_VERTICES_H
 
+#include <cstdio>
 #include <list>
 #include <set>
 #include <map>
--- src/streams-handles.cpp~    2008-11-14 07:41:07.000000000 +0000
+++ src/streams-handles.cpp     2008-11-14 07:41:14.000000000 +0000
@@ -12,6 +12,7 @@
 #include "streams-handles.h"
 #include "uri.h"
 
+#include <cstdio>
 #include <iostream>
 
 namespace Inkscape {
--- src/streams-zlib.cpp~       2008-11-14 07:41:24.000000000 +0000
+++ src/streams-zlib.cpp        2008-11-14 07:41:30.000000000 +0000
@@ -9,6 +9,7 @@
  * Released under GNU LGPL, read the file 'COPYING.LIB' for more information
  */
 
+#include <cstdio>
 #include <cstring>
 #include <string>
 #include <string.h>
--- src/dom/domimpl.cpp~        2008-11-14 07:49:10.000000000 +0000
+++ src/dom/domimpl.cpp 2008-11-14 07:49:18.000000000 +0000
@@ -30,6 +30,8 @@
 
 #include "domimpl.h"
 
+#include <cstdio>
+
 namespace org
 {
 namespace w3c
--- src/dom/io/domstream.h~     2008-11-14 07:51:18.000000000 +0000
+++ src/dom/io/domstream.h      2008-11-14 07:51:28.000000000 +0000
@@ -33,6 +33,8 @@
 
 #include <dom/dom.h>
 
+#include <cstdio>
+
 namespace org
 {
 namespace w3c
--- src/dom/xpathtoken.h~       2008-11-14 07:52:27.000000000 +0000
+++ src/dom/xpathtoken.h        2008-11-14 07:52:36.000000000 +0000
@@ -35,6 +35,7 @@
 
 #include <math.h>
 
+#include <cstdio>
 #include <vector>
 
 namespace org
--- src/dom/svg/svgtypes.h~     2008-11-14 07:53:23.000000000 +0000
+++ src/dom/svg/svgtypes.h      2008-11-14 07:53:32.000000000 +0000
@@ -47,6 +47,7 @@
 #include "dom/smil.h"
 
 
+#include <cstdio>
 #include <math.h>
 
 
--- src/extension/internal/bitmap/imagemagick.cpp.gcc44	2009-01-28 22:12:57.000000000 +0100
+++ src/extension/internal/bitmap/imagemagick.cpp	2009-03-02 17:05:57.000000000 +0100
@@ -114,7 +114,7 @@ ImageMagickDocCache::readImage(const cha
 	char *search = (char *) g_strndup(xlink, 30);
 	if (strstr(search, "base64") != (char*)NULL) {
 		// 7 = strlen("base64") + strlen(",")
-		char* pureBase64 = strstr(xlink, "base64") + 7;		
+		const char* pureBase64 = strstr(xlink, "base64") + 7;		
 		Magick::Blob blob;
 		blob.base64(pureBase64);
 		image->read(blob);
--- src/libnr/nr-object.cpp.gcc44	2009-03-02 17:06:46.000000000 +0100
+++ src/libnr/nr-object.cpp	2009-03-02 17:06:58.000000000 +0100
@@ -198,8 +198,8 @@ NRObject *NRObject::alloc(NRType type)
     );
     memset(object, 0xf0, c->isize);
 
-    object->klass = c;
     c->cpp_ctor(object);
+    object->klass = c;
     nr_class_tree_object_invoke_init (c, object);
 
     return object;
--- src/io/inkscapestream.h~    2008-11-14 08:02:53.000000000 +0000
+++ src/io/inkscapestream.h     2008-11-14 08:03:00.000000000 +0000
@@ -12,6 +12,7 @@
  */
 
 
+#include <cstdio>
 #include <glibmm.h>
 
 namespace Inkscape
--- src/io/sys.cpp~     2008-11-14 08:04:54.000000000 +0000
+++ src/io/sys.cpp      2008-11-14 08:05:11.000000000 +0000
@@ -22,6 +22,8 @@
 #include <glibmm/ustring.h>
 #include <gtk/gtkmessagedialog.h>
 
+#include <cstdio>
+
 #include "prefs-utils.h"
 #include "sys.h"
 
--- src/2geom/svg-path-parser.h~        2008-11-14 08:06:45.000000000 +0000
+++ src/2geom/svg-path-parser.h 2008-11-14 08:06:51.000000000 +0000
@@ -32,6 +32,7 @@
 #ifndef SEEN_SVG_PATH_PARSER_H
 #define SEEN_SVG_PATH_PARSER_H
 
+#include <cstdio>
 #include <vector>
 #include <iterator>
 #include <stdexcept>
--- src/libnr/nr-matrix-fns.cpp~        2008-11-14 08:16:43.000000000 +0000
+++ src/libnr/nr-matrix-fns.cpp 2008-11-14 08:16:48.000000000 +0000
@@ -1,5 +1,7 @@
 #include <libnr/nr-matrix-fns.h>
 
+#include <cstdio>
+
 namespace NR {
 
 Matrix elliptic_quadratic_form(Matrix const &m) {
--- src/libnr/nr-matrix.cpp~    2008-11-14 08:19:20.000000000 +0000
+++ src/libnr/nr-matrix.cpp     2008-11-14 08:19:26.000000000 +0000
@@ -11,6 +11,7 @@
  * This code is in public domain
  */
 
+#include <cstdio>
 #include <cstdlib>
 #include "nr-matrix.h"
 
--- src/xml/event.cpp~  2008-11-14 08:27:20.000000000 +0000
+++ src/xml/event.cpp   2008-11-14 08:27:26.000000000 +0000
@@ -14,6 +14,7 @@
  */
 
 #include <glib.h> // g_assert()
+#include <cstdio>
 
 #include "event.h"
 #include "event-fns.h"
--- src/2geom/sbasis-math.cpp~  2008-11-14 08:29:42.000000000 +0000
+++ src/2geom/sbasis-math.cpp   2008-11-14 08:31:32.000000000 +0000
@@ -38,11 +38,11 @@
 //#define ZERO 1e-3
 
 
-namespace Geom {
+#include <math.h>
+#include <cstdio>
 
+namespace Geom {
 
-#include <stdio.h>
-#include <math.h>
 
 //-|x|-----------------------------------------------------------------------
 Piecewise<SBasis> abs(SBasis const &f){
