diff -Naur inkscape-0.46/src/Makefile.am inkscape-0.46-p/src/Makefile.am
--- inkscape-0.46/src/Makefile.am	2008-03-11 05:44:31.000000000 +0100
+++ inkscape-0.46-p/src/Makefile.am	2008-03-14 22:21:53.000000000 +0100
@@ -25,7 +25,8 @@
 	$(POPPLER_GLIB_CFLAGS)	\
 	-DPOTRACE=\"potrace\"	\
 	$(INKSCAPE_CFLAGS) \
-	-I$(top_srcdir)/cxxtest
+	-I$(top_srcdir)/cxxtest \
+	-I$(top_srcdir)/src/extension/script

 include Makefile_insert
 include application/Makefile_insert
diff -Naur inkscape-0.46/src/extension/script/inkscape_perl_wrap.cpp inkscape-0.46-p/src/extension/script/inkscape_perl_wrap.cpp
--- inkscape-0.46/src/extension/script/inkscape_perl_wrap.cpp	2008-03-11 05:20:25.000000000 +0100
+++ inkscape-0.46-p/src/extension/script/inkscape_perl_wrap.cpp	2008-03-14 22:36:24.000000000 +0100
@@ -724,7 +724,7 @@
 /* #define SWIG_croakf(x...) { SWIG_SetErrorf(x); goto fail; } */


-typedef XS(SwigPerlWrapper);
+typedef XSPROTO(SwigPerlWrapper);
 typedef SwigPerlWrapper *SwigPerlWrapperPtr;

 /* Structure for command table */
