diff -urN a/native/fdlibm/ieeefp.h b/native/fdlibm/ieeefp.h
--- a/native/fdlibm/ieeefp.h	2008-01-13 10:33:50.000000000 -0700
+++ b/native/fdlibm/ieeefp.h	2015-07-25 10:17:49.689641997 -0600
@@ -4,6 +4,14 @@
 #ifndef __IEEE_BIG_ENDIAN
 #ifndef __IEEE_LITTLE_ENDIAN
 
+#ifdef __aarch64__
+#ifdef __AARCH64EB__
+#define __IEEE_BIG_ENDIAN
+#else
+#define __IEEE_LITTLE_ENDIAN
+#endif
+#endif
+
 #ifdef __alpha__
 #define __IEEE_LITTLE_ENDIAN
 #endif
