diff -aur classpath-0.98/native/jni/gtk-peer/gnu_java_awt_peer_gtk_FreetypeGlyphVector.c classpath-0.98.new/native/jni/gtk-peer/gnu_java_awt_peer_gtk_FreetypeGlyphVector.c
--- classpath-0.98/native/jni/gtk-peer/gnu_java_awt_peer_gtk_FreetypeGlyphVector.c	2008-12-18 01:43:41.000000000 +0100
+++ classpath-0.98.new/native/jni/gtk-peer/gnu_java_awt_peer_gtk_FreetypeGlyphVector.c	2014-03-11 22:03:55.888827143 +0100
@@ -42,8 +42,9 @@
 #include <pango/pango.h>
 #include <pango/pangoft2.h>
 #include <pango/pangofc-font.h>
-#include <freetype/ftglyph.h>
-#include <freetype/ftoutln.h>
+#include <ft2build.h>
+#include FT_GLYPH_H
+#include FT_OUTLINE_H
 #include "jcl.h"
 #include "gdkfont.h"
 #include "gnu_java_awt_peer_gtk_FreetypeGlyphVector.h"
diff -aur classpath-0.98/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GdkFontPeer.c classpath-0.98.new/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GdkFontPeer.c
--- classpath-0.98/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GdkFontPeer.c	2007-04-25 16:53:04.000000000 +0200
+++ classpath-0.98.new/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GdkFontPeer.c	2014-03-11 22:03:18.788834707 +0100
@@ -39,10 +39,11 @@
 #include <pango/pango.h>
 #include <pango/pangoft2.h>
 #include <pango/pangofc-font.h>
-#include <freetype/ftglyph.h>
-#include <freetype/ftoutln.h>
-#include <freetype/fttypes.h>
-#include <freetype/tttables.h>
+#include <ft2build.h>
+#include FT_GLYPH_H
+#include FT_OUTLINE_H
+#include FT_TYPES_H
+#include FT_TRUETYPE_TABLES_H
 #include "gdkfont.h"
 #include "gtkpeer.h"
 #include "gnu_java_awt_peer_gtk_GdkFontPeer.h"
