diff -Naur diacanvas2-0.14.4.orig/python/diacanvas.override diacanvas2-0.14.4.new/python/diacanvas.override
--- diacanvas2-0.14.4.orig/python/diacanvas.override	2004-10-26 03:54:46.000000000 -0400
+++ diacanvas2-0.14.4.new/python/diacanvas.override	2007-06-27 13:13:31.000000000 -0400
@@ -182,55 +182,6 @@
 	return create_canvas_item (self, args, kwargs, DIA_TYPE_CANVAS_TEXT);
 }
 %%
-override dia_canvas_item_new kwargs
-static int
-_wrap_dia_canvas_item_new (PyGObject *self, PyObject *args, PyObject *kwargs)
-{
-	return create_canvas_item (self, args, kwargs, DIA_TYPE_CANVAS_ITEM);
-}
-%%
-override dia_canvas_group_new kwargs
-static int
-_wrap_dia_canvas_group_new (PyGObject *self, PyObject *args, PyObject *kwargs)
-{
-	return create_canvas_item (self, args, kwargs, DIA_TYPE_CANVAS_GROUP);
-}
-%%
-override dia_canvas_element_new kwargs
-static int
-_wrap_dia_canvas_element_new (PyGObject *self, PyObject *args, PyObject *kwargs)
-{
-	return create_canvas_item (self, args, kwargs, DIA_TYPE_CANVAS_ELEMENT);
-}
-%%
-override dia_canvas_line_new kwargs
-static int
-_wrap_dia_canvas_line_new (PyGObject *self, PyObject *args, PyObject *kwargs)
-{
-	return create_canvas_item (self, args, kwargs, DIA_TYPE_CANVAS_LINE);
-}
-%%
-override dia_canvas_box_new kwargs
-static int
-_wrap_dia_canvas_box_new (PyGObject *self, PyObject *args, PyObject *kwargs)
-{
-	return create_canvas_item (self, args, kwargs, DIA_TYPE_CANVAS_BOX);
-}
-%%
-override dia_canvas_image_new kwargs
-static int
-_wrap_dia_canvas_image_new (PyGObject *self, PyObject *args, PyObject *kwargs)
-{
-	return create_canvas_item (self, args, kwargs, DIA_TYPE_CANVAS_IMAGE);
-}
-%%
-override dia_canvas_text_new kwargs
-static int
-_wrap_dia_canvas_text_new (PyGObject *self, PyObject *args, PyObject *kwargs)
-{
-	return create_canvas_item (self, args, kwargs, DIA_TYPE_CANVAS_TEXT);
-}
-%%
 override dia_canvas_get_pango_layout noargs
 static PyObject *
 _wrap_dia_canvas_get_pango_layout(PyGObject *self)
