diff --git a/src/Main/MainGui.cpp b/src/Main/MainGui.cpp
index dd032c367b..89e2d2e1a7 100644
--- a/src/Main/MainGui.cpp
+++ b/src/Main/MainGui.cpp
@@ -164,7 +164,7 @@ int main( int argc, char ** argv )
     App::Application::Config()["SplashInfoColor" ] = "#8aadf4"; // light blue 
     App::Application::Config()["SplashInfoPosition" ] = "6,75";
 
-    QGuiApplication::setDesktopFileName(QStringLiteral("org.freecad.FreeCAD.desktop"));
+    QGuiApplication::setDesktopFileName(QStringLiteral("org.freecad.FreeCAD"));
 
     try {
         // Init phase ===========================================================
diff --git a/src/XDGData/CMakeLists.txt b/src/XDGData/CMakeLists.txt
index 9c70207f59..8c4cb7c74e 100644
--- a/src/XDGData/CMakeLists.txt
+++ b/src/XDGData/CMakeLists.txt
@@ -25,12 +25,12 @@ install(
 )
 
 install(
-    FILES org.freecadweb.FreeCAD.desktop
+    FILES org.freecad.FreeCAD.desktop
     DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/applications
 )
 
 install(
-    FILES org.freecadweb.FreeCAD.svg
+    FILES org.freecad.FreeCAD.svg
     DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/scalable/apps
 )
 
diff --git a/src/XDGData/org.freecadweb.FreeCAD.desktop b/src/XDGData/org.freecad.FreeCAD.desktop
similarity index 97%
rename from src/XDGData/org.freecadweb.FreeCAD.desktop
rename to src/XDGData/org.freecad.FreeCAD.desktop
index df82a16063..1326f37ca7 100644
--- a/src/XDGData/org.freecadweb.FreeCAD.desktop
+++ b/src/XDGData/org.freecad.FreeCAD.desktop
@@ -17,7 +17,7 @@ GenericName[ru]=Система автоматизированного проек
 Exec=FreeCAD %F
 Terminal=false
 Type=Application
-Icon=org.freecadweb.FreeCAD
+Icon=org.freecad.FreeCAD
 Categories=Graphics;Science;Education;Engineering;
 StartupNotify=true
 MimeType=application/x-extension-fcstd;model/obj;model/iges;image/vnd.dwg;image/vnd.dxf;model/vnd.collada+xml;application/iges;model/iges;model/step;model/step+zip;model/stl;application/vnd.shp;model/vrml;
diff --git a/src/XDGData/org.freecadweb.FreeCAD.svg b/src/XDGData/org.freecad.FreeCAD.svg
similarity index 100%
rename from src/XDGData/org.freecadweb.FreeCAD.svg
rename to src/XDGData/org.freecad.FreeCAD.svg
