diff --git a/cMake/FreeCAD_Helpers/SetupShibokenAndPyside.cmake b/cMake/FreeCAD_Helpers/SetupShibokenAndPyside.cmake
index 0569c1fde9..f5e3d0931f 100644
--- a/cMake/FreeCAD_Helpers/SetupShibokenAndPyside.cmake
+++ b/cMake/FreeCAD_Helpers/SetupShibokenAndPyside.cmake
@@ -165,6 +165,8 @@ macro(SetupShibokenAndPyside)
         message(STATUS "PySide ${PySide_VERSION} Python module found at ${PRINT_OUTPUT}.\n")
     endif()
 
+    set_target_properties(Shiboken6::libshiboken PROPERTIES INTERFACE_COMPILE_DEFINITIONS "")
+
 endmacro(SetupShibokenAndPyside)
 
 # Locate the include directory for a pip-installed package -- uses pip show to find the base pip
