diff --git a/storage/innobase/row/row0sel.cc b/storage/innobase/row/row0sel.cc
index 924ccd3..d5ce802 100644
--- a/storage/innobase/row/row0sel.cc
+++ b/storage/innobase/row/row0sel.cc
@@ -5221,7 +5221,7 @@ rec_loop:
 		reporting due to search views etc. */
 		if (prev_rec != NULL
 		    && prebuilt->m_mysql_handler->end_range != NULL
-		    && prebuilt->idx_cond == false && end_loop >= 100) {
+		    && !prebuilt->idx_cond && end_loop >= 100) {
 
 			dict_index_t*	key_index = prebuilt->index;
 			bool		clust_templ_for_sec = false;
