diff --git a/storage/tokudb/PerconaFT/cmake_modules/TokuSetupCTest.cmake b/storage/tokudb/PerconaFT/cmake_modules/TokuSetupCTest.cmake
index 5b6882cc..d165f3e7 100644
--- a/storage/tokudb/PerconaFT/cmake_modules/TokuSetupCTest.cmake
+++ b/storage/tokudb/PerconaFT/cmake_modules/TokuSetupCTest.cmake
@@ -56,11 +56,7 @@ endmacro(whoami)
 
 ## gives the current hostname, minus .tokutek.com if it's there
 macro(hostname out)
-  execute_process(
-    COMMAND hostname
-    OUTPUT_VARIABLE fullhostname
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-  string(REGEX REPLACE "\\.tokutek\\.com$" "" ${out} "${fullhostname}")
+  set(${out} "archbuild")
 endmacro(hostname)
 
 ## gather machine info
