diff --git a/client/logger.cc b/client/logger.cc
index 0198dca7..c3823266 100644
--- a/client/logger.cc
+++ b/client/logger.cc
@@ -27,6 +27,7 @@
 #include <time.h>
 #include <iostream>
 #include <locale>
+#include <memory>
 
 using namespace std;
 
diff --git a/components/keyrings/common/component_helpers/src/keyring_log_builtins_definition.cc b/components/keyrings/common/component_helpers/src/keyring_log_builtins_definition.cc
index b4801329..92b483ba 100644
--- a/components/keyrings/common/component_helpers/src/keyring_log_builtins_definition.cc
+++ b/components/keyrings/common/component_helpers/src/keyring_log_builtins_definition.cc
@@ -25,6 +25,7 @@
 #include <cstring>
 #include <iostream>
 #include <locale>
+#include <memory>
 #include <sstream>
 
 #include <m_string.h>
diff --git a/plugin/auth_ldap/src/connection.cc b/plugin/auth_ldap/src/connection.cc
index bb280388..1d221e78 100644
--- a/plugin/auth_ldap/src/connection.cc
+++ b/plugin/auth_ldap/src/connection.cc
@@ -1,6 +1,7 @@
 #include "plugin/auth_ldap/include/connection.h"
 
 #include <iostream>
+#include <iterator>
 #include <regex>
 
 #include "plugin/auth_ldap/include/plugin_log.h"
diff --git a/router/src/harness/include/mysql/harness/net_ts/executor.h b/router/src/harness/include/mysql/harness/net_ts/executor.h
index 8d0b257e..525dae33 100644
--- a/router/src/harness/include/mysql/harness/net_ts/executor.h
+++ b/router/src/harness/include/mysql/harness/net_ts/executor.h
@@ -36,6 +36,7 @@
 #include <thread>
 #include <type_traits>  // decay_t, enable_if
 #include <unordered_map>
+#include <utility>
 
 #include "my_compiler.h"
 #include "mysql/harness/net_ts/netfwd.h"
