diff --git a/plugin/tokudb-backup-plugin/Percona-TokuBackup/backup/CMakeLists.txt b/plugin/tokudb-backup-plugin/Percona-TokuBackup/backup/CMakeLists.txt
index 9d2c1dd2..cefa9190 100644
--- a/plugin/tokudb-backup-plugin/Percona-TokuBackup/backup/CMakeLists.txt
+++ b/plugin/tokudb-backup-plugin/Percona-TokuBackup/backup/CMakeLists.txt
@@ -20,7 +20,7 @@ set_property(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS
   _FILE_OFFSET_BITS=64 
   _LARGEFILE64_SOURCE)
 
-set(CMAKE_CXX_FLAGS "-Werror -W -Wall -Wshadow ${CMAKE_CXX_FLAGS}")
+set(CMAKE_CXX_FLAGS "-W -Wall -Wshadow ${CMAKE_CXX_FLAGS}")
 
 set(USE_VALGRIND OFF CACHE BOOL "whether to use valgrind headers")
 if (USE_VALGRIND)
diff --git a/storage/tokudb/PerconaFT/cmake_modules/TokuSetupCompiler.cmake b/storage/tokudb/PerconaFT/cmake_modules/TokuSetupCompiler.cmake
index 92897b88..95dad8ac 100644
--- a/storage/tokudb/PerconaFT/cmake_modules/TokuSetupCompiler.cmake
+++ b/storage/tokudb/PerconaFT/cmake_modules/TokuSetupCompiler.cmake
@@ -171,8 +171,8 @@ if (NOT CMAKE_CXX_COMPILER_ID STREQUAL Clang)
 endif ()
 
 ## always want these
-set(CMAKE_C_FLAGS "-Wall -Werror ${CMAKE_C_FLAGS}")
-set(CMAKE_CXX_FLAGS "-Wall -Werror ${CMAKE_CXX_FLAGS}")
+set(CMAKE_C_FLAGS "-Wall ${CMAKE_C_FLAGS}")
+set(CMAKE_CXX_FLAGS "-Wall ${CMAKE_CXX_FLAGS}")
 
 # pick language dialect
 set(CMAKE_C_FLAGS "-std=c99 ${CMAKE_C_FLAGS}")
