--- a/script/bootchartd	2007-10-27 17:47:04.000000000 +0200
+++ b/script/bootchartd	2007-10-27 18:45:10.000000000 +0200
@@ -114,7 +114,7 @@
 # Wait for the boot process to end.
 wait_boot()
 {
-	local runlevel=$( sed -n 's/.*:\(.*\):initdefault:.*/\1/gp' /etc/inittab )
+	local runlevel=$( sed -n -e 's/#.*//g' -e 's/.*:\(.*\):initdefault:.*/\1/gp' /etc/inittab )
 
 	# The processes we have to wait for
 	local exit_proc="gdmgreeter gdm-binary kdm_greet kdm"
