--- lame-3.96.1.orig/frontend/Makefile.am	2004-01-21 01:41:36.000000000 -0800
+++ lame-3.96.1/frontend/Makefile.am	2004-10-03 16:39:12.415901779 -0700
@@ -50,7 +50,7 @@
 endif
 
 CFLAGS = @CFLAGS@ @GTK_CFLAGS@ @FRONTEND_CFLAGS@
-LDFLAGS = @LDFLAGS@ @FRONTEND_LDFLAGS@ -static
+LDFLAGS = @LDFLAGS@ @FRONTEND_LDFLAGS@
 
 INCLUDES = -I$(top_srcdir)/libmp3lame -I$(top_srcdir)/include -I$(top_builddir)
 
diff -Naur lame-3.96.1.orig/frontend/Makefile.in lame-3.96.1/frontend/Makefile.in
--- lame-3.96.1.orig/frontend/Makefile.in	2004-01-21 05:50:29.000000000 -0800
+++ lame-3.96.1/frontend/Makefile.in	2004-10-03 16:39:19.711701553 -0700
@@ -91,7 +91,7 @@
 	$(top_builddir)/libmp3lame/libmp3lame.la \
 	@FRONTEND_LDADD@
 
-LDFLAGS = @LDFLAGS@ @FRONTEND_LDFLAGS@ -static
+LDFLAGS = @LDFLAGS@ @FRONTEND_LDFLAGS@
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LIBTOOL = @LIBTOOL@
