diff -wbBur hula.org/autogen.sh hula/autogen.sh
--- hula.org/autogen.sh	2009-02-27 10:55:05.000000000 +0000
+++ hula/autogen.sh	2009-02-27 11:09:41.000000000 +0000
@@ -7,7 +7,7 @@
 
 # default version requirements ...
 REQUIRED_AUTOCONF_VERSION=${REQUIRED_AUTOCONF_VERSION:-2.53}
-REQUIRED_AUTOMAKE_VERSION=${REQUIRED_AUTOMAKE_VERSION:-1.6}
+REQUIRED_AUTOMAKE_VERSION=${REQUIRED_AUTOMAKE_VERSION:-1.10}
 REQUIRED_LIBTOOL_VERSION=${REQUIRED_LIBTOOL_VERSION:-1.4.3}
 REQUIRED_GETTEXT_VERSION=${REQUIRED_GETTEXT_VERSION:-0.10.40}
 REQUIRED_GLIB_GETTEXT_VERSION=${REQUIRED_GLIB_GETTEXT_VERSION:-2.2.0}
@@ -276,14 +276,7 @@
     "http://ftp.gnu.org/pub/gnu/autoconf/autoconf-$REQUIRED_AUTOCONF_VERSION.tar.gz" || DIE=1
 AUTOHEADER=`echo $AUTOCONF | sed s/autoconf/autoheader/`
 
-case $REQUIRED_AUTOMAKE_VERSION in
-    1.4*) automake_progs="automake-1.4" ;;
-    1.5*) automake_progs="automake-1.5 automake-1.6 automake-1.7 automake-1.8 automake-1.9" ;;
-    1.6*) automake_progs="automake-1.6 automake-1.7 automake-1.8 automake-1.9 automake19" ;;
-    1.7*) automake_progs="automake-1.7 automake-1.8 automake-1.9" ;;
-    1.8*) automake_progs="automake-1.8 automake-1.9" ;;
-    1.9*) automake_progs="automake-1.9" ;;
-esac
+automake_progs="automake"
 version_check automake AUTOMAKE "$automake_progs" $REQUIRED_AUTOMAKE_VERSION \
     "http://ftp.gnu.org/pub/gnu/automake/automake-$REQUIRED_AUTOMAKE_VERSION.tar.gz" || DIE=1
 ACLOCAL=`echo $AUTOMAKE | sed s/automake/aclocal/`
diff -wbBur hula.org/src/libs/hula-sharp/Hula.Dav/AddressbookQueryReport.cs hula/src/libs/hula-sharp/Hula.Dav/AddressbookQueryReport.cs
--- hula.org/src/libs/hula-sharp/Hula.Dav/AddressbookQueryReport.cs	2009-02-27 10:52:29.000000000 +0000
+++ hula/src/libs/hula-sharp/Hula.Dav/AddressbookQueryReport.cs	2009-02-27 11:21:17.000000000 +0000
@@ -42,7 +42,7 @@
 		break;
 	    case "text-match" :
 		// FIXME: Don't do caseless param filters for now.
-		TermQuery q = new TermQuery (new Term("vcard." + prop + ".param", + name + "=" + child.InnerText));
+		TermQuery q = new TermQuery (new Term("vcard." + prop + ".param", name + "=" + child.InnerText));
 
 		query.Add(q, true, false);
 		break;
diff -wbBur hula.org/src/libs/hula-sharp/Hula.Dav/CalQueryReport.cs hula/src/libs/hula-sharp/Hula.Dav/CalQueryReport.cs
--- hula.org/src/libs/hula-sharp/Hula.Dav/CalQueryReport.cs	2009-02-27 10:52:29.000000000 +0000
+++ hula/src/libs/hula-sharp/Hula.Dav/CalQueryReport.cs	2009-02-27 11:21:45.000000000 +0000
@@ -59,7 +59,7 @@
 		break;
 	    case "text-match" :
 		// FIXME: Don't do caseless param filters for now.
-		TermQuery q = new TermQuery (new Term("ical." + prop + ".param", + name + "=" + child.InnerText));
+		TermQuery q = new TermQuery (new Term("ical." + prop + ".param", name + "=" + child.InnerText));
 
 		query.Add(q, true, false);
 		break;
