diff -upr php-gtk-2.0.1.orig/config.m4 php-gtk-2.0.1/config.m4
--- php-gtk-2.0.1.orig/config.m4	2008-02-13 07:05:27.000000000 +0200
+++ php-gtk-2.0.1/config.m4	2009-09-11 20:43:29.000000000 +0300
@@ -22,7 +22,7 @@ Please use the --with-php-config option 
 the location of php-config for the required version.])
         ;;
     esac
-    PHP=$PHP_PREFIX/bin/php
+    PHP="$PHP_PREFIX/bin/php -d 'open_basedir='"
   else
     AC_MSG_ERROR(Could not locate PHP executable)
   fi
