diff -upr php-gtk-2.0.1.orig/ext/extra/config.m4 php-gtk-2.0.1/ext/extra/config.m4
--- php-gtk-2.0.1.orig/ext/extra/config.m4	2007-05-29 02:53:32.000000000 +0300
+++ php-gtk-2.0.1/ext/extra/config.m4	2008-10-12 15:19:46.000000000 +0300
@@ -15,8 +15,8 @@ if test "$PHP_GTK_EXTRA" != "no"; then
     AC_DEFINE(HAVE_GTKEXTRA, 1, [gtkextra support])
     PHP_EVAL_INCLINE($GTKEXTRA_CFLAGS)
     if test "$php_gtk_ext_shared" = "yes"; then
-      PHP_EVAL_LIBLINE($GTKEXTRA_LIBS, GTKEXTRA_SHARED_LIBADD)
-      PHP_SUBST(GTKEXTRA_SHARED_LIBADD)
+      PHP_EVAL_LIBLINE($GTKEXTRA_LIBS, PHP_GTK2_EXTRA_SHARED_LIBADD)
+      PHP_SUBST(PHP_GTK2_EXTRA_SHARED_LIBADD)
     else
       PHP_EVAL_LIBLINE($GTKEXTRA_LIBS, PHP_GTK2_SHARED_LIBADD)
     fi
diff -upr php-gtk-2.0.1.orig/ext/html/config.m4 php-gtk-2.0.1/ext/html/config.m4
--- php-gtk-2.0.1.orig/ext/html/config.m4	2007-05-29 02:53:32.000000000 +0300
+++ php-gtk-2.0.1/ext/html/config.m4	2008-10-12 15:20:03.000000000 +0300
@@ -15,8 +15,8 @@ if test "$PHP_GTK_HTML" != "no"; then
     AC_DEFINE(HAVE_HTML, 1, [html support])
     PHP_EVAL_INCLINE($HTML_CFLAGS)
     if test "$php_gtk_ext_shared" = "yes"; then
-      PHP_EVAL_LIBLINE($HTML_LIBS, HTML_SHARED_LIBADD)
-      PHP_SUBST(HTML_SHARED_LIBADD)
+      PHP_EVAL_LIBLINE($HTML_LIBS, PHP_GTK2_HTML_SHARED_LIBADD)
+      PHP_SUBST(PHP_GTK2_HTML_SHARED_LIBADD)
     else
       PHP_EVAL_LIBLINE($HTML_LIBS, PHP_GTK2_SHARED_LIBADD)
     fi
diff -upr php-gtk-2.0.1.orig/ext/libglade/config.m4 php-gtk-2.0.1/ext/libglade/config.m4
--- php-gtk-2.0.1.orig/ext/libglade/config.m4	2007-05-29 02:53:33.000000000 +0300
+++ php-gtk-2.0.1/ext/libglade/config.m4	2008-10-12 15:20:17.000000000 +0300
@@ -14,8 +14,8 @@ if test "$PHP_GTK_LIBGLADE" != "no"; the
     AC_DEFINE(HAVE_LIBGLADE, 1, [libglade support])
     PHP_EVAL_INCLINE($LIBGLADE_CFLAGS)
     if test "$php_gtk_ext_shared" = "yes"; then
-      PHP_EVAL_LIBLINE($LIBGLADE_LIBS, LIBGLADE_SHARED_LIBADD)
-      PHP_SUBST(LIBGLADE_SHARED_LIBADD)
+      PHP_EVAL_LIBLINE($LIBGLADE_LIBS, PHP_GTK2_LIBGLADE_SHARED_LIBADD)
+      PHP_SUBST(PHP_GTK2_LIBGLADE_SHARED_LIBADD)
     else
       PHP_EVAL_LIBLINE($LIBGLADE_LIBS, PHP_GTK2_SHARED_LIBADD)
     fi
diff -upr php-gtk-2.0.1.orig/ext/libsexy/config.m4 php-gtk-2.0.1/ext/libsexy/config.m4
--- php-gtk-2.0.1.orig/ext/libsexy/config.m4	2007-05-29 02:53:33.000000000 +0300
+++ php-gtk-2.0.1/ext/libsexy/config.m4	2008-10-12 15:20:24.000000000 +0300
@@ -14,8 +14,8 @@ if test "$PHP_GTK_LIBSEXY" != "no"; then
     AC_DEFINE(HAVE_LIBSEXY, 1, [libsexy support])
     PHP_EVAL_INCLINE($LIBSEXY_CFLAGS)
     if test "$php_gtk_ext_shared" = "yes"; then
-      PHP_EVAL_LIBLINE($LIBSEXY_LIBS, LIBSEXY_SHARED_LIBADD)
-      PHP_SUBST(LIBSEXY_SHARED_LIBADD)
+      PHP_EVAL_LIBLINE($LIBSEXY_LIBS, PHP_GTK2_LIBSEXY_SHARED_LIBADD)
+      PHP_SUBST(PHP_GTK2_LIBSEXY_SHARED_LIBADD)
     else
       PHP_EVAL_LIBLINE($LIBSEXY_LIBS, PHP_GTK2_SHARED_LIBADD)
     fi
diff -upr php-gtk-2.0.1.orig/ext/mozembed/config.m4 php-gtk-2.0.1/ext/mozembed/config.m4
--- php-gtk-2.0.1.orig/ext/mozembed/config.m4	2007-05-29 02:53:33.000000000 +0300
+++ php-gtk-2.0.1/ext/mozembed/config.m4	2008-10-12 15:20:32.000000000 +0300
@@ -21,8 +21,8 @@ if test "$PHP_GTK_MOZEMBED" != "no"; the
     AC_DEFINE(HAVE_MOZEMBED, 1, [mozembed support])
     PHP_EVAL_INCLINE($MOZEMBED_CFLAGS)
     if test "$php_gtk_ext_shared" = "yes"; then
-      PHP_EVAL_LIBLINE($MOZEMBED_LIBS, MOZEMBED_SHARED_LIBADD)
-      PHP_SUBST(MOZEMBED_SHARED_LIBADD)
+      PHP_EVAL_LIBLINE($MOZEMBED_LIBS, PHP_GTK2_MOZEMBED_SHARED_LIBADD)
+      PHP_SUBST(PHP_GTK2_MOZEMBED_SHARED_LIBADD)
     else
       PHP_EVAL_LIBLINE($MOZEMBED_LIBS, PHP_GTK2_SHARED_LIBADD)
     fi
diff -upr php-gtk-2.0.1.orig/ext/sourceview/config.m4 php-gtk-2.0.1/ext/sourceview/config.m4
--- php-gtk-2.0.1.orig/ext/sourceview/config.m4	2007-05-29 02:53:33.000000000 +0300
+++ php-gtk-2.0.1/ext/sourceview/config.m4	2008-10-12 15:21:04.000000000 +0300
@@ -15,8 +15,8 @@ if test "$PHP_GTK_SOURCEVIEW" != "no"; t
     AC_DEFINE(HAVE_SOURCEVIEW, 1, [sourceview support])
     PHP_EVAL_INCLINE($SOURCEVIEW_CFLAGS)
     if test "$php_gtk_ext_shared" = "yes"; then
-      PHP_EVAL_LIBLINE($SOURCEVIEW_LIBS, SOURCEVIEW_SHARED_LIBADD)
-      PHP_SUBST(SOURCEVIEW_SHARED_LIBADD)
+      PHP_EVAL_LIBLINE($SOURCEVIEW_LIBS, PHP_GTK2_SOURCEVIEW_SHARED_LIBADD)
+      PHP_SUBST(PHP_GTK2_SOURCEVIEW_SHARED_LIBADD)
     else
       PHP_EVAL_LIBLINE($SOURCEVIEW_LIBS, PHP_GTK2_SHARED_LIBADD)
     fi
diff -upr php-gtk-2.0.1.orig/ext/spell/config.m4 php-gtk-2.0.1/ext/spell/config.m4
--- php-gtk-2.0.1.orig/ext/spell/config.m4	2007-05-29 02:53:33.000000000 +0300
+++ php-gtk-2.0.1/ext/spell/config.m4	2008-10-12 15:21:12.000000000 +0300
@@ -15,8 +15,8 @@ if test "$PHP_GTK_SPELL" != "no"; then
     AC_DEFINE(HAVE_SPELL, 1, [spell support])
     PHP_EVAL_INCLINE($SPELL_CFLAGS)
     if test "$php_gtk_ext_shared" = "yes"; then
-      PHP_EVAL_LIBLINE($SPELL_LIBS, SPELL_SHARED_LIBADD)
-      PHP_SUBST(SPELL_SHARED_LIBADD)
+      PHP_EVAL_LIBLINE($SPELL_LIBS, PHP_GTK2_SPELL_SHARED_LIBADD)
+      PHP_SUBST(PHP_GTK2_SPELL_SHARED_LIBADD)
     else
       PHP_EVAL_LIBLINE($SPELL_LIBS, PHP_GTK2_SHARED_LIBADD)
     fi
diff -upr php-gtk-2.0.1.orig/php_gtk.m4 php-gtk-2.0.1/php_gtk.m4
--- php-gtk-2.0.1.orig/php_gtk.m4	2008-02-23 18:21:06.000000000 +0200
+++ php-gtk-2.0.1/php_gtk.m4	2008-10-12 15:18:25.000000000 +0300
@@ -497,7 +497,7 @@ dnl ------------------------------------
     # shared module
 	ifelse($3,,,PHP_ADD_SOURCES_X($php_gtk_ext_builddir,$3,$ac_extra,[shared_objects_]translit($1,a-z-+,A-Z__),yes))
 	ifelse($4,,,PHP_ADD_SOURCES_X(/ext/$1,$4,$ac_extra,[shared_objects_]translit($1,a-z-+,A-Z__),yes))
-	PHP_GTK_SHARED_MODULE(php_gtk_$1,[shared_objects_]translit($1,a-z-+,A-Z__), $php_gtk_ext_builddir)
+	PHP_GTK_SHARED_MODULE(php_gtk2_$1,[shared_objects_]translit($1,a-z-+,A-Z__), $php_gtk_ext_builddir)
 	AC_DEFINE_UNQUOTED([PHP_GTK_COMPILE_DL_]translit($1,a-z-+,A-Z__), 1, Whether to build $1 as dynamic module)
   fi
 
