diff -ur siefs-0.5.orig/siefs/Makefile.am siefs-0.5/siefs/Makefile.am
--- siefs-0.5.orig/siefs/Makefile.am	2005-04-05 16:36:28.000000000 +0200
+++ siefs-0.5/siefs/Makefile.am	2005-04-08 16:52:04.000000000 +0200
@@ -12,5 +12,5 @@
 LDADD = $(fuseinst)/lib/libfuse.a -lpthread
 
 install-exec-hook:
-	-rm -f /sbin/mount.siefs
-	-ln -s $(DESTDIR)$(bindir)/siefs /sbin/mount.siefs
+	-rm -f $(DESTDIR)/sbin/mount.siefs
+	-ln -s $(bindir)/siefs $(DESTDIR)/sbin/mount.siefs
diff -ur siefs-0.5.orig/siefs/Makefile.in siefs-0.5/siefs/Makefile.in
--- siefs-0.5.orig/siefs/Makefile.in	2005-04-05 16:36:45.000000000 +0200
+++ siefs-0.5/siefs/Makefile.in	2005-04-08 16:52:55.000000000 +0200
@@ -345,8 +345,8 @@
 
 
 install-exec-hook:
-	-rm -f /sbin/mount.siefs
-	-ln -s $(DESTDIR)$(bindir)/siefs /sbin/mount.siefs
+	-rm -f $(DESTDIR)/sbin/mount.siefs
+	-ln -s $(bindir)/siefs $(DESTDIR)/sbin/mount.siefs
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
