diff -rN -u old-gshare/src/gshared/daemon.cs new-gshare/src/gshared/daemon.cs
--- old-gshare/src/gshared/daemon.cs	2006-09-16 20:31:47.881173352 +0100
+++ new-gshare/src/gshared/daemon.cs	2006-09-16 20:31:47.959161496 +0100
@@ -308,11 +308,11 @@
             if (reason == FTPServerShutdownReason.Error)
             {
                 Log.ErrorMessage("Shutting down FTP server because of an error");
-                Utility.InvokeLater(delegate
+                Utility.InvokeLater((GShare.Utility.BackgroundTask)(delegate
                     { 
                         Utility.ShowGenericFatalErrorDialog(); 
                         Shutdown(); 
-                    });
+                    }));
             }
         }
         
@@ -367,24 +367,24 @@
             {
                 case ClientState.Collision:
                     /*TODO figure out what to do*/
-                    Utility.InvokeLater(delegate
+                    Utility.InvokeLater((GShare.Utility.BackgroundTask)(delegate
                         { 
                             StopDaemon();
-                        });
+                        }));
                     break;
                 case ClientState.Failure:
                     Log.ErrorMessage("Zeroconf client failed to register");
-                    Utility.InvokeLater(delegate
+                    Utility.InvokeLater((GShare.Utility.BackgroundTask)(delegate
                         {
                             Utility.ShowCannotPublishServiceDialog(); 
                             Shutdown(); 
-                        });
+                        }));
                     break;
                 case ClientState.Running:
-                    Utility.InvokeLater(delegate
+                    Utility.InvokeLater((GShare.Utility.BackgroundTask)(delegate
                         {
                             PublishFTPServer(); 
-                        });
+                        }));
                     break;
             }
 
@@ -410,19 +410,19 @@
                         iAvahiServiceName = EntryGroup.GetAlternativeServiceName(iAvahiServiceName);
                     }
                     
-                    Utility.InvokeLater(delegate
+                    Utility.InvokeLater((GShare.Utility.BackgroundTask)(delegate
                         {
                             StartZeroConf(iFtpServer.ServerPort);
-                        });
+                        }));
                     
                     break;
                 case EntryGroupState.Failure:
                     Log.ErrorMessage("Failed to publish FTP service on zeroconf network");
-                    Utility.InvokeLater(delegate
+                    Utility.InvokeLater((GShare.Utility.BackgroundTask)(delegate
                         {
                             Utility.ShowCannotPublishServiceDialog();
                             Shutdown(); 
-                        });
+                        }));
                     break;
 
                 case EntryGroupState.Established:
@@ -460,11 +460,11 @@
                                     ex.Message,
                                     ex.StackTrace);
                                 
-                Utility.InvokeLater(delegate 
+                Utility.InvokeLater((GShare.Utility.BackgroundTask)(delegate 
                     {
                         Utility.ShowCannotPublishServiceDialog(); 
                         Shutdown(); 
-                    });
+                    }));
                 
             }
 

