--- configure.ac.orig	2007-11-04 04:24:32.000000000 -0300
+++ configure.ac	2009-04-30 19:03:09.393728806 -0300
@@ -67,7 +67,8 @@
 		struct user_regs_struct.rax,
 		struct user_regs_struct.rdi,
 		struct user_regs_struct.rsi,
-		struct user_regs_struct.rdx], [],[], [#include <sys/user.h>])
+		struct user_regs_struct.rdx], [],[], [AC_INCLUDES_DEFAULT
+#include <sys/user.h>])
 AC_CHECK_MEMBERS([struct pt_regs.orig_gpr3,
 		struct pt_regs.gpr], [],[], [#include <linux/ptrace.h>])
 AM_CONDITIONAL(WANT_PEEKFD_I386,
