diff --git a/libalpm/hooks/90-mkinitcpio-install.hook b/libalpm/hooks/90-mkinitcpio-install.hook
index 446b7d7..5582f12 100644
--- a/libalpm/hooks/90-mkinitcpio-install.hook
+++ b/libalpm/hooks/90-mkinitcpio-install.hook
@@ -6,6 +6,9 @@ Operation = Remove
 Target = usr/lib/initcpio/*
 Target = usr/lib/firmware/*
 Target = usr/src/*/dkms.conf
+Target = usr/lib/systemd/systemd
+Target = usr/bin/cryptsetup
+Target = usr/bin/lvm
 
 [Trigger]
 Type = Path
