From 2dd521db90f82156b04a63576db946c97a75a0f1 Mon Sep 17 00:00:00 2001
From: Janne Grunau <j@jannau.net>
Date: Sun, 30 Jan 2022 10:01:43 +0000
Subject: [PATCH 2/2] functions: handle gzip compressed kernels in kver_generic

Signed-off-by: Janne Grunau <j@jannau.net>
---
 functions | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/functions b/functions
index 2148ed8..f2b8bac 100644
--- a/functions
+++ b/functions
@@ -160,6 +160,11 @@ kver_generic() {
 
     read _ _ kver _ < <(grep -m1 -aoE 'Linux version .(\.[-[:alnum:]+]+)+' "$1")
 
+    # try if the image is gzip compressed
+    if [[ -z "$kver" ]]; then
+        read _ _ kver _ < <(gzip -c -d "$1" | grep -m1 -aoE 'Linux version .(\.[-[:alnum:]]+)+')
+    fi
+
     printf '%s' "$kver"
 }
 
-- 
2.36.1

