From 65f5014fe991afcd2572bdba235e5d90b35c9d2f Mon Sep 17 00:00:00 2001
From: Janne Grunau <j@jannau.net>
Date: Sun, 30 Jan 2022 10:01:43 +0000
Subject: [PATCH 2/3] functions: handle gzip compressed kernels in kver_generic

Signed-off-by: Janne Grunau <j@jannau.net>
---
 functions | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/functions b/functions
index 8a689ee..6eab867 100644
--- a/functions
+++ b/functions
@@ -301,6 +301,11 @@ kver_generic() {
 
     read -r _ _ kver _ < <($reader "$1" | grep -m1 -aoE 'Linux version .(\.[-[:alnum:]+]+)+')
 
+    # try if the image is gzip compressed
+    if [[ -z "$kver" ]]; then
+        read _ _ kver _ < <(gzip -c -d "$1" | grep -m1 -aoE 'Linux version .(\.[-[:alnum:]]+)+')
+    fi
+
     printf '%s' "$kver"
 }
 
-- 
2.36.1

