From d0a64f47286f04701fbe1e6e856147b372ad1c9a Mon Sep 17 00:00:00 2001
From: Morten Linderud <morten@linderud.pw>
Date: Sat, 20 Dec 2025 16:01:10 +0100
Subject: [PATCH] install/systemd: add new dlopened dependencies

Signed-off-by: Morten Linderud <morten@linderud.pw>
---
 install/sd-shutdown | 2 ++
 install/systemd     | 8 +++++++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/install/sd-shutdown b/install/sd-shutdown
index f8aad2e7..238a9464 100644
--- a/install/sd-shutdown
+++ b/install/sd-shutdown
@@ -2,6 +2,8 @@
 # SPDX-License-Identifier: GPL-2.0-only
 
 build() {
+    # TODO: should be pulled inn by .notes.dlopen in the future
+    add_binary /usr/lib/libmount.so
     add_binary /usr/lib/systemd/systemd-shutdown /shutdown
 
     if type -P kexec >/dev/null; then
diff --git a/install/systemd b/install/systemd
index a487b14a..4bf05c43 100644
--- a/install/systemd
+++ b/install/systemd
@@ -145,7 +145,13 @@ build() {
     #  qrencode -> bsod
     #  seccomp  -> systemd
     #  acl      -> systemd-tmpfiles
-    for LIB in kmod tss2-{esys,rc,mu,tcti-'*'} qrencode seccomp acl; do
+    #  Things that should be gone with the new .dlopen note section
+    #  acl     -> libsystemd-shared
+    #  blkid   -> libsystemd-shared
+    #  mount   -> libsystemd-shared
+    #  audit   -> libsystemd-shared
+    #  pam     -> libsystemd-shared
+    for LIB in kmod tss2-{esys,rc,mu,tcti-'*'} qrencode seccomp acl blkid mount audit pam; do
         LC_ALL=C.UTF-8 find /usr/lib/ -maxdepth 1 -name "lib${LIB}.so*" | while read -r FILE; do
             if [[ -L "${FILE}" ]]; then
                 add_symlink "${FILE}"
-- 
GitLab

