From 3d672171d33fc995035d4908dab3509fda47185f Mon Sep 17 00:00:00 2001
From: Christian Hesse <eworm@archlinux.org>
Date: Wed, 27 May 2026 15:25:16 +0200
Subject: [mkinitcpio] [PATCH 1/1] install/encrypt: add dlopen'ed libcryptsetup

---
 install/sd-encrypt | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/install/sd-encrypt b/install/sd-encrypt
index 856f640..684d6e3 100644
--- a/install/sd-encrypt
+++ b/install/sd-encrypt
@@ -36,12 +36,14 @@ build() {
     add_binary '/usr/lib/ossl-modules/legacy.so'
 
     # add libraries dlopen()ed by systemd-cryptsetup
-    LC_ALL=C.UTF-8 find /usr/lib/ -maxdepth 1 -name "libfido2.so*" | while read -r FILE; do
-        if [[ -L "${FILE}" ]]; then
-            add_symlink "${FILE}"
-        else
-            add_binary "${FILE}"
-        fi
+    for LIB in fido2 cryptsetup; do
+        LC_ALL=C.UTF-8 find /usr/lib/ -maxdepth 1 -name "lib${LIB}.so*" | while read -r FILE; do
+            if [[ -L "${FILE}" ]]; then
+                add_symlink "${FILE}"
+            else
+                add_binary "${FILE}"
+            fi
+        done
     done
 
     # add mkswap for creating swap space on the fly (see 'swap' in crypttab(5))
