diff --git a/install/systemd b/install/systemd
index c40f0a02807b..e85a5f6a5e74 100644
--- a/install/systemd
+++ b/install/systemd
@@ -193,9 +193,11 @@ build() {
     add_file "/usr/lib/tmpfiles.d/20-systemd-stub.conf"
 
     # add hwdb binaries to the initramfs
-    map add_file \
-        /usr/lib/udev/hwdb.bin \
-        /etc/udev/hwdb.bin
+    for f in /usr/lib/udev/hwdb.bin /etc/udev/hwdb.bin; do
+        if [[  -f "$f" ]]; then
+            add_file "$f"
+        fi
+    done
 
     # Include nvpcr files
     for nvpcr in /usr/lib/nvpcr/*.nvpcr; do
diff --git a/install/udev b/install/udev
index 33ed07be7bc9..9e1fb74e7140 100644
--- a/install/udev
+++ b/install/udev
@@ -21,9 +21,11 @@ build() {
         '80-drivers.rules'
 
     # add hwdb binaries to the initramfs
-    map add_file \
-        /usr/lib/udev/hwdb.bin \
-        /etc/udev/hwdb.bin
+    for f in /usr/lib/udev/hwdb.bin /etc/udev/hwdb.bin; do
+        if [[  -f "$f" ]]; then
+            add_file "$f"
+        fi
+    done
 
     add_runscript
 }
