Index: src/common/ssl_certificate.c
===================================================================
RCS file: //claws/src/common/ssl_certificate.c,v
retrieving revision 1.4.2.40
retrieving revision 1.4.2.41
diff -u -r1.4.2.40 -r1.4.2.41
--- src/common/ssl_certificate.c	30 Aug 2011 06:18:09 -0000	1.4.2.40
+++ src/common/ssl_certificate.c	16 May 2012 05:43:14 -0000	1.4.2.41
@@ -686,7 +686,7 @@
 	      gnutls_x509_privkey * key,
 	      gnutls_x509_crt_t * cert)
 {
-  gnutls_pkcs12_bag bag = NULL;
+  gnutls_pkcs12_bag_t bag = NULL;
   int index = 0;
   int ret;
 

