diff -Nur claws-mail-3.15.1-orig/src/plugins/vcalendar/vcalendar.c claws-mail-3.15.1/src/plugins/vcalendar/vcalendar.c
--- claws-mail-3.15.1-orig/src/plugins/vcalendar/vcalendar.c	2017-08-28 19:48:24.000000000 +0200
+++ claws-mail-3.15.1/src/plugins/vcalendar/vcalendar.c	2017-11-16 17:51:01.139709415 +0100
@@ -157,8 +157,8 @@
 			gchar *orgname = NULL;
 			gchar *summary = g_strdup(msginfo->subject ? msginfo->subject:_("no subject"));
 			gchar *description = file_read_stream_to_str(fp);
-			gchar *dtstart = g_strdup(icaltime_as_ical_string(icaltime_from_timet(t, FALSE)));
-			gchar *dtend = g_strdup(icaltime_as_ical_string(icaltime_from_timet(t2, FALSE)));
+			gchar *dtstart = g_strdup(icaltime_as_ical_string(icaltime_from_timet_with_zone(t, FALSE,NULL)));
+			gchar *dtend = g_strdup(icaltime_as_ical_string(icaltime_from_timet_with_zone(t2, FALSE,NULL)));
 			gchar *recur = NULL;
 			gchar *tzid = g_strdup("UTC");
 			gchar *url = NULL;
diff -Nur claws-mail-3.15.1-orig/src/plugins/vcalendar/vcal_manager.c claws-mail-3.15.1/src/plugins/vcalendar/vcal_manager.c
--- claws-mail-3.15.1-orig/src/plugins/vcalendar/vcal_manager.c	2017-08-28 19:48:24.000000000 +0200
+++ claws-mail-3.15.1/src/plugins/vcalendar/vcal_manager.c	2017-11-16 17:43:34.911106211 +0100
@@ -395,7 +395,7 @@
 	icalcomponent_add_property(ievent,
                 icalproperty_new_uid(event->uid));
 	icalcomponent_add_property(ievent,
-		icalproperty_vanew_dtstamp(icaltime_from_timet(time(NULL), TRUE), (void*)0));
+                               icalproperty_vanew_dtstamp(icaltime_from_timet_with_zone(time(NULL), TRUE, NULL), (void*)0));
 	icalcomponent_add_property(ievent,
 		icalproperty_vanew_dtstart((icaltime_from_string(event->dtstart)), (void*)0));
 	icalcomponent_add_property(ievent,
@@ -423,9 +423,9 @@
 	icalcomponent_add_property(ievent,
 		icalproperty_new_status(ICAL_STATUS_CONFIRMED));
 	icalcomponent_add_property(ievent,
-		icalproperty_vanew_created(icaltime_from_timet(time(NULL), TRUE), (void*)0));
+                               icalproperty_vanew_created(icaltime_from_timet_with_zone(time(NULL), TRUE, NULL), (void*)0));
 	icalcomponent_add_property(ievent,
-		icalproperty_vanew_lastmodified(icaltime_from_timet(time(NULL), TRUE), (void*)0));
+                               icalproperty_vanew_lastmodified(icaltime_from_timet_with_zone(time(NULL), TRUE, NULL), (void*)0));
 	icalcomponent_add_property(ievent,		
                 orgprop);
 
diff -Nur claws-mail-3.15.1-orig/src/plugins/vcalendar/vcal_meeting_gtk.c claws-mail-3.15.1/src/plugins/vcalendar/vcal_meeting_gtk.c
--- claws-mail-3.15.1-orig/src/plugins/vcalendar/vcal_meeting_gtk.c	2017-08-28 19:48:24.000000000 +0200
+++ claws-mail-3.15.1/src/plugins/vcalendar/vcal_meeting_gtk.c	2017-11-16 17:54:45.798495950 +0100
@@ -477,7 +477,7 @@
 	debug_print("DST change offset to apply to time %d\n", dst_offset);
 	t += dst_offset;
 	debug_print("%s\n", ctime(&t));
-	return g_strdup(icaltime_as_ical_string(icaltime_from_timet(t, FALSE)));
+	return g_strdup(icaltime_as_ical_string(icaltime_from_timet_with_zone(t, FALSE,NULL)));
 }
 
 static gchar *get_location(VCalMeeting *meet)
@@ -893,8 +893,8 @@
 	found = FALSE;
 	while (!found && offset >= -3600*6) {
 		gboolean ok = TRUE;
-		struct icaltimetype new_start = icaltime_from_timet(icaltime_as_timet(start)+offset, FALSE);
-		struct icaltimetype new_end   = icaltime_from_timet(icaltime_as_timet(end)+offset, FALSE);
+		struct icaltimetype new_start = icaltime_from_timet_with_zone(icaltime_as_timet(start)+offset, FALSE, NULL);
+		struct icaltimetype new_end   = icaltime_from_timet_with_zone(icaltime_as_timet(end)+offset, FALSE, NULL);
 		for (cur = attendees; cur; cur = cur->next) {
 			VCalAttendee *attendee = (VCalAttendee *)cur->data;
 			debug_print("trying %s - %s (offset %d)\n", 
@@ -919,8 +919,8 @@
 	offset = 1800;
 	while (!found && offset <= 3600*6) {
 		gboolean ok = TRUE;
-		struct icaltimetype new_start = icaltime_from_timet(icaltime_as_timet(start)+offset, FALSE);
-		struct icaltimetype new_end   = icaltime_from_timet(icaltime_as_timet(end)+offset, FALSE);
+		struct icaltimetype new_start = icaltime_from_timet_with_zone(icaltime_as_timet(start)+offset, FALSE, NULL);
+		struct icaltimetype new_end   = icaltime_from_timet_with_zone(icaltime_as_timet(end)+offset, FALSE, NULL);
 		for (cur = attendees; cur; cur = cur->next) {
 			VCalAttendee *attendee = (VCalAttendee *)cur->data;
 			debug_print("trying %s - %s (offset %d)\n", 
@@ -2172,8 +2172,8 @@
 
 	icalcomponent_add_component(calendar, timezone);
 
-	itt_start = icaltime_from_timet(whole_start, FALSE);
-	itt_end = icaltime_from_timet(whole_end, FALSE);
+	itt_start = icaltime_from_timet_with_zone(whole_start, FALSE, NULL);
+	itt_end = icaltime_from_timet_with_zone(whole_end, FALSE, NULL);
 	itt_start.second = itt_start.minute = itt_start.hour = 0;
 	itt_end.second = 59; itt_end.minute = 59; itt_end.hour = 23;
 
