Index: src/main.c
===================================================================
RCS file: //claws/src/main.c,v
retrieving revision 1.115.2.237
retrieving revision 1.115.2.238
diff -u -r1.115.2.237 -r1.115.2.238
--- src/main.c	10 Apr 2011 17:19:04 -0000	1.115.2.237
+++ src/main.c	30 Apr 2011 19:27:15 -0000	1.115.2.238
@@ -331,7 +331,7 @@
 		gtk_widget_show(hack);
 	}
 
-	xdisplay = gdk_display_get_default();
+	xdisplay = GDK_DISPLAY_XDISPLAY(gdk_display_get_default());
 	sn_display = sn_display_new(xdisplay,
 				sn_error_trap_push,
 				sn_error_trap_pop);

