--- claws-mail-3.15.1/src/gtk/gtkaspell.c.orig	2017-08-28 17:48:24.000000000 +0000
+++ claws-mail-3.15.1/src/gtk/gtkaspell.c	2017-11-22 15:05:59.371073969 +0000
@@ -1348,7 +1348,7 @@
 {
    	GtkAspell *gtkaspell = (GtkAspell *) data; 
 
-	enchant_dict_add_to_pwl(gtkaspell->gtkaspeller->speller, gtkaspell->theword, strlen(gtkaspell->theword));
+	enchant_dict_add(gtkaspell->gtkaspeller->speller, gtkaspell->theword, strlen(gtkaspell->theword));
 
 	gtkaspell->ctx.check_word(gtkaspell->ctx.data);
 	gtkaspell_dict_changed(gtkaspell);
