--- claws-mail-3.7.10/src/common/ssl.c	2011-12-07 14:55:08.682553870 +0100
+++ claws-mail-3.7.10/src/common/ssl.c	2011-12-07 14:55:43.808349602 +0100
@@ -270,7 +270,9 @@ gboolean ssl_init_socket_with_method(Soc
 	if (session == NULL || r != 0)
 		return FALSE;
   
+#if GNUTLS_VERSION_NUMBER < 0x020c00
 	gnutls_transport_set_lowat (session, 1); 
+#endif
 	gnutls_set_default_priority(session);
 	gnutls_protocol_set_priority (session, proto_prio);
 	gnutls_cipher_set_priority (session, cipher_prio);
