--- hugin-0.7.0/src/hugin_base/appbase/ProgressDisplayOld.cpp~	2008-09-14 21:00:41.000000000 +0000
+++ hugin-0.7.0/src/hugin_base/appbase/ProgressDisplayOld.cpp	2009-05-08 21:53:02.000000000 +0000
@@ -22,6 +22,7 @@
  */
 
 #include "ProgressDisplayOld.h"
+#include <cstdio>
 
 #ifdef WIN32
 #define snprintf _snprintf
--- hugin-0.7.0/src/hugin_base/hugin_utils/utils.cpp~	2008-09-14 21:00:41.000000000 +0000
+++ hugin-0.7.0/src/hugin_base/hugin_utils/utils.cpp	2009-05-08 21:56:25.000000000 +0000
@@ -30,7 +30,7 @@
     #include <sys/time.h>
 #endif
 #include <time.h>
-#include <stdio.h>
+#include <cstdio>
 
 
 namespace hugin_utils {
--- hugin-0.7.0/src/hugin_base/panodata/PTScriptParsing.cpp~	2008-09-14 21:00:41.000000000 +0000
+++ hugin-0.7.0/src/hugin_base/panodata/PTScriptParsing.cpp	2009-05-08 21:59:52.000000000 +0000
@@ -25,6 +25,7 @@
 #include "PTScriptParsing.h"
 
 #include <hugin_utils/utils.h>
+#include <cstdio>
 
 
 namespace HuginBase {
