diff -Nur klibc-1.5.14.orig/Kbuild klibc-1.5.14/Kbuild
--- klibc-1.5.14.orig/Kbuild	2008-07-30 23:06:03.000000000 +0200
+++ klibc-1.5.14/Kbuild	2008-09-18 13:08:42.000000000 +0200
@@ -10,9 +10,8 @@
 	$(Q)$(MAKE) $(klibc)=usr/kinit
 	$(Q)$(MAKE) $(klibc)=usr/dash
 	$(Q)$(MAKE) $(klibc)=usr/utils
-	$(Q)$(MAKE) $(klibc)=usr/gzip
 
 
 # Directories to visit during clean and install
-subdir- := scripts/basic klcc usr/klibc usr/dash usr/utils usr/gzip \
+subdir- := scripts/basic klcc usr/klibc usr/dash usr/utils \
 	   usr/kinit usr/klibc/tests
diff -Nur klibc-1.5.14.orig/usr/Kbuild klibc-1.5.14/usr/Kbuild
--- klibc-1.5.14.orig/usr/Kbuild	2008-07-30 23:06:03.000000000 +0200
+++ klibc-1.5.14/usr/Kbuild	2008-09-18 13:08:42.000000000 +0200
@@ -6,7 +6,7 @@
 
 include-subdir := include
 klibc-subdir := klibc
-usr-subdirs  := kinit utils dash gzip
+usr-subdirs  := kinit utils dash
 subdir-      := $(include-subdir) $(klibc-subdir) $(usr-subdirs)
 
 usr-subdirs  := $(addprefix _usr_,$(usr-subdirs))
diff -Nur klibc-1.5.14.orig/usr/dash/Kbuild klibc-1.5.14/usr/dash/Kbuild
--- klibc-1.5.14.orig/usr/dash/Kbuild	2008-07-30 23:06:03.000000000 +0200
+++ klibc-1.5.14/usr/dash/Kbuild	2008-09-18 13:08:42.000000000 +0200
@@ -23,11 +23,8 @@
 hostprogs-y := mkinit mksyntax mknodes mksignames
 gen-h-files := arith.h builtins.h nodes.h syntax.h token.h
 
-static-y := sh
-
 # The shared binary
-shared-y    := sh.shared
-sh.shared-y := $(sh-y)
+shared-y    := sh
 
 # For cleaning
 targets := sh sh.g sh.shared sh.shared.g $(gen-o-files)
@@ -108,4 +105,4 @@
 	$(Q):
 
 # Targets to install
-install-y := sh.shared
+install-y := sh
diff -Nur klibc-1.5.14.orig/usr/kinit/Kbuild klibc-1.5.14/usr/kinit/Kbuild
--- klibc-1.5.14.orig/usr/kinit/Kbuild	2008-07-30 23:06:03.000000000 +0200
+++ klibc-1.5.14/usr/kinit/Kbuild	2008-09-18 13:08:42.000000000 +0200
@@ -17,9 +17,7 @@
 kinit-y  += fstype/
 kinit-y  += resume/
 
-static-y := kinit
-shared-y := kinit.shared
-kinit.shared-y := $(kinit-y)
+shared-y := kinit
 
 # Additional include paths files
 KLIBCCFLAGS += -I$(srctree)/$(src)/fstype \
@@ -34,4 +32,4 @@
 
 
 # install binary
-install-y := kinit kinit.shared
+install-y := kinit
diff -Nur klibc-1.5.14.orig/usr/utils/Kbuild klibc-1.5.14/usr/utils/Kbuild
--- klibc-1.5.14.orig/usr/utils/Kbuild	2008-07-30 23:06:03.000000000 +0200
+++ klibc-1.5.14/usr/utils/Kbuild	2008-09-18 13:08:56.000000000 +0200
@@ -4,7 +4,7 @@
 
 progs := chroot dd mkdir mkfifo mknod mount pivot_root umount
 progs += true false sleep ln nuke minips cat
-progs += uname halt kill readlink cpio sync dmesg
+progs += uname halt readlink sync dmesg
 
 static-y := $(addprefix static/, $(progs))
 shared-y := $(addprefix shared/, $(progs))
