--- klibc-1.5.15.orig/usr/klibc/arch/arm/MCONFIG	2009-01-04 19:28:03.000000000 +0000
+++ klibc-1.5.15/usr/klibc/arch/arm/MCONFIG	2009-08-31 11:57:05.000000000 +0000
@@ -7,29 +7,13 @@
 # accordingly.
 #
 
-CPU_ARCH := armv4
-CPU_TUNE := strongarm
+CPU_ARCH := armv4t
 
-KLIBCOPTFLAGS += -Os -march=$(CPU_ARCH) -mtune=$(CPU_TUNE)
+KLIBCOPTFLAGS += -Os -march=$(CPU_ARCH) -msoft-float
 KLIBCBITSIZE  = 32
-KLIBCREQFLAGS += -fno-exceptions
 KLIBCSTRIPFLAGS += -R .ARM.exidx
 
-ifeq ($(CONFIG_KLIBC_THUMB),y)
-CPU_ARCH := $(CPU_ARCH)t
-KLIBCREQFLAGS += -mthumb
-KLIBCLDFLAGS  += --thumb-entry _start
-KLIBCEMAIN     = --thumb-entry _start
-KLIBCREQFLAGS += -mabi=aapcs-linux
-KLIBCSHAREDFLAGS = -Ttext 0x380200
-else
 # Extra linkflags when building the shared version of the library
 # This address needs to be reachable using normal inter-module
 # calls, and work on the memory models for this architecture
 KLIBCSHAREDFLAGS = -Ttext 0x01800200
-ifeq ($(CONFIG_AEABI),y)
-KLIBCREQFLAGS += -mabi=aapcs-linux -mno-thumb-interwork
-else
-KLIBCREQFLAGS += -mabi=apcs-gnu -mno-thumb-interwork
-endif
-endif
