From 67a4b80e80ffa30d8bca1087ce0cf5e92a85e093 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Fri, 6 Dec 2019 06:42:03 -0700
Subject: [PATCH] Remove ARM march and float specification

---
 Make.inc | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Make.inc b/Make.inc
index 7612499..52ffda2 100644
--- a/Make.inc
+++ b/Make.inc
@@ -67,8 +67,6 @@ override ARCH := aarch64
 endif
 ifeq ($(findstring arm,$(ARCH)),arm)
 override ARCH := arm
-MARCH ?= armv7-a+fp
-CFLAGS_add += -mhard-float
 endif
 ifeq ($(findstring powerpc,$(ARCH)),powerpc)
 override ARCH := powerpc
-- 
2.36.1

