From 53a4684ea8194e7766a5e5f624b28876e09710d5 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Fri, 6 Dec 2019 06:42:03 -0700
Subject: [PATCH] Remove ARM march and float specification

---
 Make.inc | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Make.inc b/Make.inc
index 34888d2..a3c43ae 100644
--- a/Make.inc
+++ b/Make.inc
@@ -57,8 +57,6 @@ endif
 # OS-specific stuff
 ifeq ($(findstring arm,$(ARCH)),arm)
 override ARCH := arm
-MARCH ?= armv7-a
-CFLAGS_add += -mhard-float
 endif
 ifeq ($(findstring powerpc,$(ARCH)),powerpc)
 override ARCH := powerpc
-- 
2.23.0

