diff -up commoncpp2-1.7.3/src/cidr.cpp.orig commoncpp2-1.7.3/src/cidr.cpp
--- commoncpp2-1.7.3/src/cidr.cpp.orig	2009-04-06 21:24:07.857718325 +0200
+++ commoncpp2-1.7.3/src/cidr.cpp	2009-04-06 21:50:26.874088776 +0200
@@ -202,7 +202,7 @@ void IPV4Cidr::set(const char *cp)
 #if defined(_MSC_VER) && _MSC_VER >= 1500
 	ep = (char *)strchr(cp, '/');
 #else
- 	ep = strchr(cp, '/');
+ 	ep = (char *)strchr(cp, '/');
 #endif
 
 	if(ep)
@@ -332,7 +332,7 @@ void IPV6Cidr::set(const char *cp)
 	memset(&netmask, 0, sizeof(netmask));
 	bitset((bit_t *)&netmask, getMask(cp));
 	setString(cbuf, sizeof(cbuf), cp);
-	ep = strchr(cp, '/');
+	ep = (char *)strchr(cp, '/');
 	if(ep)
 		*ep = 0;
 
