From c471352db4764bdcc24223338639e4c3f52b3e51 Mon Sep 17 00:00:00 2001
From: Pascal Terjan <pterjan@google.com>
Date: Wed, 4 May 2016 09:36:21 +0000
Subject: Fix test_parse with no executable backend

https://bugzilla.gnome.org/show_bug.cgi?id=765953

diff --git a/testsuite/test_parse.c b/testsuite/test_parse.c
index 65b17e9..87fa4ea 100644
--- a/testsuite/test_parse.c
+++ b/testsuite/test_parse.c
@@ -38,7 +38,7 @@ main (int argc, char *argv[])
       printf ("%s\n", programs[i]->name);
     orc_test_compare_output_full (programs[i], 0);
     cres = orc_program_compile (programs[i]);
-    if (cres != ORC_COMPILE_RESULT_OK) {
+    if (ORC_COMPILE_RESULT_IS_FATAL (cres)) {
       fprintf (stderr, "compile error: %d\n", cres);
       error = TRUE;
     }
@@ -55,7 +55,7 @@ main (int argc, char *argv[])
       printf ("%s\n", programs[i]->name);
     orc_test_compare_output_full (programs[i], 0);
     cres = orc_program_compile (programs[i]);
-    if (cres != ORC_COMPILE_RESULT_OK) {
+    if (ORC_COMPILE_RESULT_IS_FATAL (cres)) {
       fprintf (stderr, "compile error: %d\n", cres);
       error = TRUE;
     }
-- 
cgit v0.10.2

