--- mcpp-2.7.2/src/mbchar.c	2008-03-08 13:06:13.000000000 +0000
+++ mcpp-2.7.2/src/mbchar.c	2009-03-08 13:14:17.000000000 +0000
@@ -345,7 +345,7 @@
 static size_t   mb_read_utf8( int c1, char ** in_pp, char ** out_pp);
                 /* For UTF8 mbchar encoding         */
 
-#define NAMLEN          20
+#define NAMLEN          40
 #define UPPER           1               /* To upper */
 #define LOWER           0               /* To lower */
 
