--- mozilla-central-fec90cbfbaad/nsprpub/config/make-system-wrappers.pl.orig	2015-04-10 04:57:56.000000000 +0200
+++ mozilla-central-fec90cbfbaad/nsprpub/config/make-system-wrappers.pl	2016-05-07 19:51:53.832876185 +0200
@@ -19,7 +19,9 @@
     open OUT, ">$output_dir/$_";
     print OUT "#pragma GCC system_header\n";  # suppress include_next warning
     print OUT "#pragma GCC visibility push(default)\n";
+    print OUT "#define _GLIBCXX_INCLUDE_NEXT_C_HEADERS\n";
     print OUT "#include_next \<$_\>\n";
+    print OUT "#undef _GLIBCXX_INCLUDE_NEXT_C_HEADERS\n";
     print OUT "#pragma GCC visibility pop\n";
     close OUT;
 }
