diff -Naur mozilla-central-f14898695ee0.orig/ipc/chromium/src/base/message_pump_libevent.cc mozilla-central-f14898695ee0/ipc/chromium/src/base/message_pump_libevent.cc
--- mozilla-central-f14898695ee0.orig/ipc/chromium/src/base/message_pump_libevent.cc	2016-03-20 05:25:17.000000000 +0100
+++ mozilla-central-f14898695ee0/ipc/chromium/src/base/message_pump_libevent.cc	2017-06-11 19:44:51.294979917 +0200
@@ -18,11 +18,11 @@
 #include "event.h"
 #include "mozilla/UniquePtr.h"
 
-// This macro checks that the _EVENT_SIZEOF_* constants defined in
+// This macro checks that the EVENT__SIZEOF_* constants defined in
 // ipc/chromiume/src/third_party/<platform>/event2/event-config.h are correct.
 #define CHECK_EVENT_SIZEOF(TYPE, type) \
-    static_assert(_EVENT_SIZEOF_##TYPE == sizeof(type), \
-    "bad _EVENT_SIZEOF_"#TYPE);
+    static_assert(EVENT__SIZEOF_##TYPE == sizeof(type), \
+    "bad EVENT__SIZEOF_"#TYPE);
 
 CHECK_EVENT_SIZEOF(LONG,      long);
 CHECK_EVENT_SIZEOF(LONG_LONG, long long);
