diff --git a/configure.ac b/configure.ac
index d2cd5b2..686f4a2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2,7 +2,7 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.59)
-AC_INIT([kodi], [15.1.903], [http://trac.kodi.tv])
+AC_INIT([kodi], [15.2.000], [http://trac.kodi.tv])
 AC_CONFIG_HEADERS([xbmc/config.h])
 AH_TOP([#pragma once])
 m4_include([m4/ax_prog_cc_for_build.m4])
diff --git a/version.txt b/version.txt
index c427834..6bf96a6 100644
--- a/version.txt
+++ b/version.txt
@@ -3,9 +3,9 @@ COMPANY_NAME XBMC-Foundation
 WEBSITE http://kodi.tv
 VERSION_MAJOR 15
 VERSION_MINOR 2
-VERSION_TAG RC3
-VERSION_CODE 151903
-ADDON_API 15.1.903
+VERSION_TAG
+VERSION_CODE 152000
+ADDON_API 15.2.000
 
 # Notes:
 # Change AC_INIT in configure.ac
