diff -ruN poppler-0.10.0/Makefile.am poppler-0.10.0-bindings/Makefile.am
--- poppler-0.10.0/Makefile.am	2008-09-09 23:46:50.000000000 +0200
+++ poppler-0.10.0-bindings/Makefile.am	2008-11-08 00:03:17.000000000 +0100
@@ -29,7 +29,7 @@
 utils_subdir = utils
 endif
 
-SUBDIRS = goo fofi $(splash_subdir) poppler $(utils_subdir) $(glib_subdir) $(qt_subdir) test $(qt4_subdir)
+SUBDIRS = $(glib_subdir) $(qt_subdir) $(qt4_subdir)
 
 EXTRA_DIST =					\
 	README-XPDF				\
@@ -40,9 +40,6 @@
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA =				\
-	poppler.pc				\
-	$(cairo_pc_file)			\
-	$(splash_pc_file)			\
 	$(glib_pc_file)				\
 	$(qt_pc_file)                           \
 	$(qt4_pc_file)
diff -ruN poppler-0.10.0/glib/Makefile.am poppler-0.10.0-bindings/glib/Makefile.am
--- poppler-0.10.0/glib/Makefile.am	2008-09-27 17:21:45.000000000 +0200
+++ poppler-0.10.0-bindings/glib/Makefile.am	2008-11-08 00:03:17.000000000 +0100
@@ -87,7 +87,7 @@
 	poppler-private.h
 
 libpoppler_glib_la_LIBADD =				\
-	$(top_builddir)/poppler/libpoppler.la		\
+	-lpoppler					\
 	$(POPPLER_GLIB_LIBS)				\
 	$(GDK_LIBS)					\
 	$(FREETYPE_LIBS)				\
@@ -103,7 +103,7 @@
        test-poppler-glib.cc
 
 test_poppler_glib_LDADD =			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	libpoppler-glib.la			\
 	$(POPPLER_GLIB_LIBS)			\
 	$(GDK_LIBS)				\
diff -ruN poppler-0.10.0/glib/demo/Makefile.am poppler-0.10.0-bindings/glib/demo/Makefile.am
--- poppler-0.10.0/glib/demo/Makefile.am	2008-08-24 18:19:55.000000000 +0200
+++ poppler-0.10.0-bindings/glib/demo/Makefile.am	2008-11-08 00:03:17.000000000 +0100
@@ -36,5 +36,5 @@
 
 LDADD =						\
 	$(top_builddir)/glib/libpoppler-glib.la	\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler	\
 	$(GTK_TEST_LIBS)
diff -ruN poppler-0.10.0/glib/reference/Makefile.am poppler-0.10.0-bindings/glib/reference/Makefile.am
--- poppler-0.10.0/glib/reference/Makefile.am	2008-07-24 22:45:05.000000000 +0200
+++ poppler-0.10.0-bindings/glib/reference/Makefile.am	2008-11-08 00:03:17.000000000 +0100
@@ -73,7 +73,7 @@
 	$(GDK_CFLAGS)					\
 	$(FREETYPE_CFLAGS)
 
-GTKDOC_LIBS=$(top_builddir)/poppler/libpoppler.la	\
+GTKDOC_LIBS=-lpoppler	\
 	$(top_builddir)/glib/libpoppler-glib.la		\
 	$(POPPLER_GLIB_LIBS)				\
 	$(FREETYPE_LIBS)				\
diff -ruN poppler-0.10.0/qt/Makefile.am poppler-0.10.0-bindings/qt/Makefile.am
--- poppler-0.10.0/qt/Makefile.am	2008-07-24 22:45:05.000000000 +0200
+++ poppler-0.10.0-bindings/qt/Makefile.am	2008-11-08 00:03:17.000000000 +0100
@@ -24,7 +24,7 @@
 	poppler-private.h
 
 libpoppler_qt_la_LIBADD = 			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(POPPLER_QT_LIBS)			\
 	$(FREETYPE_LIBS)			\
 	$(FONTCONFIG_LIBS)
@@ -41,7 +41,7 @@
        test-poppler-qt.cpp
 
 test_poppler_qt_LDADD =				\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	libpoppler-qt.la			\
 	$(POPPLER_QT_LIBS)			\
 	$(FREETYPE_LIBS)
diff -ruN poppler-0.10.0/qt4/demos/Makefile.am poppler-0.10.0-bindings/qt4/demos/Makefile.am
--- poppler-0.10.0/qt4/demos/Makefile.am	2008-07-24 22:45:06.000000000 +0200
+++ poppler-0.10.0-bindings/qt4/demos/Makefile.am	2008-11-08 00:05:09.000000000 +0100
@@ -7,7 +7,7 @@
 	$(POPPLER_QT4_CFLAGS)
 
 LDADDS =					\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler	\
 	$(top_builddir)/qt4/src/libpoppler-qt4.la	\
 	$(FONTCONFIG_LIBS)				\
 	$(POPPLER_QT4_LIBS)
diff -ruN poppler-0.10.0/qt4/src/Makefile.am poppler-0.10.0-bindings/qt4/src/Makefile.am
--- poppler-0.10.0/qt4/src/Makefile.am	2008-09-10 00:02:49.000000000 +0200
+++ poppler-0.10.0-bindings/qt4/src/Makefile.am	2008-11-08 00:03:17.000000000 +0100
@@ -52,7 +52,7 @@
 	poppler-private.h
 
 libpoppler_qt4_la_LIBADD = 			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(FONTCONFIG_LIBS)				\
 	$(POPPLER_QT4_LIBS)
 
diff -ruN poppler-0.10.0/qt4/tests/Makefile.am poppler-0.10.0-bindings/qt4/tests/Makefile.am
--- poppler-0.10.0/qt4/tests/Makefile.am	2008-09-05 18:41:33.000000000 +0200
+++ poppler-0.10.0-bindings/qt4/tests/Makefile.am	2008-11-08 00:03:17.000000000 +0100
@@ -7,7 +7,7 @@
 	$(POPPLER_QT4_CFLAGS)
 
 LDADDS =					\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(top_builddir)/qt4/src/libpoppler-qt4.la	\
 	$(FONTCONFIG_LIBS)				\
 	$(POPPLER_QT4_LIBS)
diff -ruN poppler-0.10.0/test/Makefile.am poppler-0.10.0-bindings/test/Makefile.am
--- poppler-0.10.0/test/Makefile.am	2008-10-06 19:35:46.000000000 +0200
+++ poppler-0.10.0-bindings/test/Makefile.am	2008-11-08 00:05:42.000000000 +0100
@@ -51,14 +51,14 @@
        gtk-splash-test.cc
 
 gtk_splash_test_LDADD =				\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler	\
 	$(GTK_TEST_LIBS)
 
 gtk_cairo_test_SOURCES =			\
        gtk-cairo-test.cc
 
 gtk_cairo_test_LDADD =					\
-	$(top_builddir)/poppler/libpoppler.la		\
+	-lpoppler		\
 	$(top_builddir)/glib/libpoppler-glib.la		\
 	$(CAIRO_LIBS)					\
 	$(GTK_TEST_LIBS)				\
@@ -69,7 +69,7 @@
 
 pdf_inspector_LDADD =					\
 	$(top_builddir)/poppler/libpoppler-cairo.la	\
-	$(top_builddir)/poppler/libpoppler.la		\
+	-lpoppler		\
 	$(CAIRO_LIBS)					\
 	$(FREETYPE_LIBS)				\
 	$(GTK_TEST_LIBS)				\
@@ -80,7 +80,7 @@
        perf-test-preview-dummy.cc
 
 perf_test_LDADD =				\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler	\
 	$(FREETYPE_LIBS)					\
 	$(X_EXTRA_LIBS)
 
@@ -88,7 +88,7 @@
 	pdf-fullrewrite.cc
 
 pdf_fullrewrite_LDADD = \
-	$(top_builddir)/poppler/libpoppler.la
+	-lpoppler
 
 EXTRA_DIST =					\
 	pdf-operators.c				\
diff -ruN poppler-0.10.0/utils/Makefile.am poppler-0.10.0-bindings/utils/Makefile.am
--- poppler-0.10.0/utils/Makefile.am	2008-09-09 20:55:24.000000000 +0200
+++ poppler-0.10.0-bindings/utils/Makefile.am	2008-11-08 00:03:17.000000000 +0100
@@ -29,7 +29,7 @@
 	$(ABIWORD_CFLAGS)
 
 LDADD =						\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler	\
 	$(UTILS_LIBS)				\
 	$(FONTCONFIG_LIBS)
 
