diff -aur UMLet 9.1.old/UMLet.sh UMLet 9.1/UMLet.sh
--- UMLet 9.1.old/UMLet.sh	2009-03-08 16:37:36.000000000 -0300
+++ UMLet 9.1/UMLet.sh	2009-03-10 18:04:23.000000000 -0300
@@ -21,6 +21,7 @@
 
 # uncomment this to be able to start umlet from anywhere (UMLET HOME has to be set)
 # cd $UMLET_HOME
+cd /usr/share/umlet
 if [ $# -gt 0 ]
  then java -jar umlet.jar -filename=$currentDir/$1
  else java -jar umlet.jar
