diff -Naur wbar-1.3.3.orig/IconLoader.h wbar-1.3.3/IconLoader.h
--- wbar-1.3.3.orig/IconLoader.h	2006-10-20 02:25:01.000000000 +0300
+++ wbar-1.3.3/IconLoader.h	2008-03-29 14:39:46.000000000 +0200
@@ -4,6 +4,7 @@
 extern "C" {
 #include <sys/types.h>
 #include <regex.h>
+#include <stdlib.h>
 };
 
 #include <fstream>
diff -Naur wbar-1.3.3.orig/Main.cc wbar-1.3.3/Main.cc
--- wbar-1.3.3.orig/Main.cc	2007-08-06 15:28:11.000000000 +0300
+++ wbar-1.3.3/Main.cc	2008-03-29 14:42:25.000000000 +0200
@@ -82,7 +82,7 @@
 /*}}}*/
 
     /* window configuration *//*{{{*/
-    barwin.setName("wbar");
+    barwin.setName((char *)"wbar");
     if( optparser.isset("above-desk") ){
 	barwin.setDockWindow();
 	barwin.skipTaskNPager();
diff -Naur wbar-1.3.3.orig/SuperBar.h wbar-1.3.3/SuperBar.h
--- wbar-1.3.3.orig/SuperBar.h	2007-02-12 01:32:17.000000000 +0200
+++ wbar-1.3.3/SuperBar.h	2008-03-29 14:39:57.000000000 +0200
@@ -1,6 +1,7 @@
 #ifndef _SUPERBAR_H_
 #define _SUPERBAR_H_
 
+#include <stdlib.h>
 #include "ImgWrap.h"
 #include "Bar.h"
 
diff -Naur wbar-1.3.3.orig/XWin.cc wbar-1.3.3/XWin.cc
--- wbar-1.3.3.orig/XWin.cc	2007-01-26 21:15:59.000000000 +0200
+++ wbar-1.3.3/XWin.cc	2008-03-29 14:40:16.000000000 +0200
@@ -30,7 +30,7 @@
     /* Set WM Protocols to report window delete event */
     XSetWMProtocols(display, window, &delWindow, 1);
 
-    XClassHint ch = {"wbar", "wbar"};
+    XClassHint ch = {(char *)"wbar", (char *)"wbar"};
     XSetClassHint(display, window, &ch);
 }
 
