--- admin/cvs.sh	2003-07-11 23:00:30.000000000 +0800
+++ admin/cvs.sh	2006-08-28 14:54:23.000000000 +0800
@@ -29,7 +29,7 @@
 {
 AUTOCONF_VERSION=`$AUTOCONF --version | head -n 1`
 case $AUTOCONF_VERSION in
-  Autoconf*2.5* | autoconf*2.5* ) : ;;
+  Autoconf*2.[56]* | autoconf*2.[56]* ) : ;;
   "" )
     echo "*** AUTOCONF NOT FOUND!."
     echo "*** KDE requires autoconf 2.52, 2.53 or 2.54"
@@ -44,7 +44,7 @@
  
 AUTOHEADER_VERSION=`$AUTOHEADER --version | head -n 1`
 case $AUTOHEADER_VERSION in
-  Autoconf*2.5* | autoheader*2.5* ) : ;;
+  Autoconf*2.[56]* | autoheader*2.[56]* ) : ;;
   "" )
     echo "*** AUTOHEADER NOT FOUND!."
     echo "*** KDE requires autoheader 2.52 or 2.53 (part of autoconf)"
--- src/hdsensorslist.cpp	2004-08-09 02:30:05.000000000 +0800
+++ src/hdsensorslist.cpp	2006-08-28 15:27:02.000000000 +0800
@@ -206,13 +206,17 @@
 
 bool HDSensorsList::getDisks(QStringList &disks )
 {
-  DIR *dir= opendir("/proc/ide");
-  if(!dir) return false;
+  DIR *dir;
+
+  /* Get a listing of the hard drives looking under sysfs first before falling back to /proc/ide */
+  if((dir = opendir ("/sys/block")) == NULL)
+    if ((dir = opendir ("/proc/ide")) == NULL)
+      return false;
   QString str;
   struct dirent *ptr;
   while((ptr= readdir(dir))) {
-    if(ptr->d_name[0]=='h' && ptr->d_name[1]=='d') {
-       str.sprintf("/dev/hd%c",ptr->d_name[2]);
+    if((ptr->d_name[0] == 'h' || ptr->d_name[0] == 's') && ptr->d_name[1] == 'd') {
+       str.sprintf("/dev/%s", ptr->d_name);
        disks << str;
     }
   }
--- src/lmsensorscfgdesign.ui	2004-08-11 22:36:54.000000000 +0800
+++ src/lmsensorscfgdesign.ui	2006-08-28 14:54:23.000000000 +0800
@@ -748,29 +748,13 @@
                                 <property name="name">
                                     <cstring>unnamed</cstring>
                                 </property>
-                                <widget class="QLabel">
+                                <widget class="QButtonGroup">
                                     <property name="name">
-                                        <cstring>TextLabel1_3_2_2_2_2</cstring>
+                                        <cstring>ButtonGroup9</cstring>
                                     </property>
-                                    <property name="text">
+                                    <property name="title">
                                         <string>On reach alarm value:</string>
                                     </property>
-                                </widget>
-                                <widget class="QFrame">
-                                    <property name="name">
-                                        <cstring>Frame11_2_2_2</cstring>
-                                    </property>
-                                    <property name="frameShape">
-                                        <enum>HLine</enum>
-                                    </property>
-                                    <property name="frameShadow">
-                                        <enum>Sunken</enum>
-                                    </property>
-                                </widget>
-                                <widget class="QLayoutWidget">
-                                    <property name="name">
-                                        <cstring>layout9</cstring>
-                                    </property>
                                     <grid>
                                         <property name="name">
                                             <cstring>unnamed</cstring>
