--- tools.h	2005-07-07 00:00:00.000000000 +0200
+++ tools.h~	2006-03-13 08:27:07.000000000 +0100
@@ -69,14 +69,7 @@
 extern void lprintf  (const char *, const char *, ...);
 extern void vlprintf (const char *, const char *, va_list);
 
-/*---------------------------------------------------------------------------*\
-\*---------------------------------------------------------------------------*/
-static inline unsigned long atomic_xchg (
-	volatile atomic_t *	v, 
-	unsigned		value
-) {
-	return __xchg (value, &v->counter, sizeof (unsigned));
-} /* atomic_xchg */
+
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
