diff -aur FreeImage.orig/Makefile.fip FreeImage.new/Makefile.fip
--- FreeImage.orig/Makefile.fip	2009-05-09 17:31:33.000000000 +0200
+++ FreeImage.new/Makefile.fip	2009-05-09 18:41:47.000000000 +0200
@@ -8,13 +8,13 @@
 CXX = g++
 AR = ar
 
-INCDIR = /usr/include
-INSTALLDIR = /usr/lib
+INCDIR = $(DESTDIR)/usr/include
+INSTALLDIR = $(DESTDIR)/usr/lib
 
 # Converts cr/lf to just lf
 DOS2UNIX = dos2unix
 
-COMPILERFLAGS = -O3 -fexceptions -fvisibility=hidden
+COMPILERFLAGS = -O3 -fexceptions -fvisibility=hidden -fPIC
 LIBRARIES = -lstdc++
 
 MODULES = $(SRCS:.c=.o)
@@ -63,9 +63,6 @@
 	install -m 644 -o root -g root $(HEADERFIP) $(INCDIR)
 	install -m 644 -o root -g root $(STATICLIB) $(INSTALLDIR)
 	install -m 755 -o root -g root $(SHAREDLIB) $(INSTALLDIR)
-	ln -sf $(SHAREDLIB) $(INSTALLDIR)/$(VERLIBNAME)
-	ln -sf $(VERLIBNAME) $(INSTALLDIR)/$(LIBNAME)
-	ldconfig
 
 clean:
 	rm -f core Dist/*.* u2dtmp* $(MODULES) $(STATICLIB) $(SHAREDLIB) $(LIBNAME)
diff -aur FreeImage.orig/Makefile.gnu FreeImage.new/Makefile.gnu
--- FreeImage.orig/Makefile.gnu	2009-05-09 17:31:33.000000000 +0200
+++ FreeImage.new/Makefile.gnu	2009-05-09 17:59:54.000000000 +0200
@@ -8,8 +8,8 @@
 CXX = g++
 AR = ar
 
-INCDIR = /usr/include
-INSTALLDIR = /usr/lib
+INCDIR = $(DESTDIR)/usr/include
+INSTALLDIR = $(DESTDIR)/usr/lib
 
 # Converts cr/lf to just lf
 DOS2UNIX = dos2unix
@@ -61,9 +61,6 @@
 	install -m 644 -o root -g root $(HEADER) $(INCDIR)
 	install -m 644 -o root -g root $(STATICLIB) $(INSTALLDIR)
 	install -m 755 -o root -g root $(SHAREDLIB) $(INSTALLDIR)
-	ln -sf $(SHAREDLIB) $(INSTALLDIR)/$(VERLIBNAME)
-	ln -sf $(VERLIBNAME) $(INSTALLDIR)/$(LIBNAME)	
-	ldconfig
 
 clean:
 	rm -f core Dist/*.* u2dtmp* $(MODULES) $(STATICLIB) $(SHAREDLIB) $(LIBNAME)
