diff -wbBur fxprocessview-0.5.0/src/processlist.cpp fxprocessview-0.5.0.my/src/processlist.cpp
--- fxprocessview-0.5.0/src/processlist.cpp	2005-07-15 06:39:17.000000000 +0400
+++ fxprocessview-0.5.0.my/src/processlist.cpp	2007-03-07 19:21:34.000000000 +0300
@@ -696,7 +696,7 @@
 
 long FXProcessList::onCmdUserMode(FXObject* sender,FXSelector,void* ){
   if (userfilter.empty()){
-	userfilter=FXFile::getCurrentUserName();
+	userfilter=FXSystem::currentUserName();
 	}
   else {
  	userfilter="";
@@ -942,12 +942,17 @@
 
 
 
-  FXint num_processes = FXFile::listFiles(dirlist,"/proc","[0123456789]*",LIST_MATCH_ALL|LIST_NO_FILES|LIST_NO_PARENT);
+  FXint num_processes = FXDir::listFiles(dirlist,"/proc","[0123456789]*",FXDir::MatchAll | FXDir::NoFiles | FXDir::NoParent);
   task_total=num_processes;
   for (FXint i=0;i<num_processes;i++){
 
     /// Check the Owner
-    owner = FXFile::owner("/proc/" + dirlist[i]);
+    FXStat stat;
+    FXStat::statFile("/proc/" + dirlist[i], stat);
+    owner = FXSystem::userName(stat.user());
+
+//    owner = FXSystem::userName(FXStat::user("/proc/" + dirlist[i]));
+
     if (!userfilter.empty() && (owner!=userfilter)) continue;
 
     /// Read in Process Stat
@@ -1027,7 +1032,7 @@
       }
 
     filename = "/proc/" + dirlist[i] + "/cmdline";
-    if (FXFile::exists(filename)) {
+    if (FXStat::exists(filename)) {
       fp = fopen(filename.text(),"r");
       if (fp) {
         if (fgets(buffer,80,fp)!=NULL){
