From 933ca7c0f6f6c68d68d24373df3733eeb2efc974 Mon Sep 17 00:00:00 2001
From: Alexander Holler <holler@ahsoftware.de>
Date: Thu, 4 Feb 2010 06:14:55 +0100
Subject: [PATCH 02/14] DockStar: environment is at 0xa0000.

---
 include/configs/sheevaplug.h |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/configs/sheevaplug.h b/include/configs/sheevaplug.h
index fc401a8..e020cb0 100644
--- a/include/configs/sheevaplug.h
+++ b/include/configs/sheevaplug.h
@@ -123,8 +123,8 @@
  * it has to be rounded to sector size
  */
 #define CONFIG_ENV_SIZE			0x20000	/* 128k */
-#define CONFIG_ENV_ADDR			0x40000
-#define CONFIG_ENV_OFFSET		0x40000	/* env starts here */
+#define CONFIG_ENV_ADDR			0xA0000
+#define CONFIG_ENV_OFFSET		0xA0000	/* env starts here */
 
 /*
  * Default environment variables
-- 
1.6.4.4

