From 1ba6365d89537f2bcf389df971dc10d16cc6bc27 Mon Sep 17 00:00:00 2001
From: Alexander Holler <holler@ahsoftware.de>
Date: Thu, 4 Feb 2010 06:53:53 +0100
Subject: [PATCH 10/14] DockStar: Definition for the MAC. CHANGE THIS TO THE MAC OF YOUR DEVICE!

---
 include/configs/sheevaplug.h |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/include/configs/sheevaplug.h b/include/configs/sheevaplug.h
index 4861f3a..80d26d4 100644
--- a/include/configs/sheevaplug.h
+++ b/include/configs/sheevaplug.h
@@ -123,6 +123,10 @@
 #define CONFIG_ENV_ADDR			0xA0000
 #define CONFIG_ENV_OFFSET		0xA0000	/* env starts here */
 
+/* The MAC for the Ethernet interface */
+/* CHANGE THIS TO THE MAC FOUND ON THE BOTTOM OF YOUR DOCKSTAR! */
+#define CONFIG_ETHADDR 00:10:75:12:34:56
+
 /*
  * Default environment variables
  */
-- 
1.6.4.4

