From f0654fcc032176c18f0b45f985a79ffe53d191e2 Mon Sep 17 00:00:00 2001
From: Alexander Holler <holler@ahsoftware.de>
Date: Wed, 17 Feb 2010 00:02:04 +0100
Subject: [PATCH 12/14] DockStar: Boot from USB and netconsole. CHANGE THIS TO YOUR IPS!

---
 include/configs/sheevaplug.h |   18 ++++++++++--------
 1 files changed, 10 insertions(+), 8 deletions(-)

diff --git a/include/configs/sheevaplug.h b/include/configs/sheevaplug.h
index 127302d..42b873b 100644
--- a/include/configs/sheevaplug.h
+++ b/include/configs/sheevaplug.h
@@ -140,18 +140,20 @@
 /*
  * Default environment variables
  */
-#define CONFIG_BOOTCOMMAND		"${x_bootcmd_kernel}; "	\
-	"setenv bootargs ${x_bootargs} ${x_bootargs_root}; "	\
-	"${x_bootcmd_usb}; bootm 0x6400000;"
+#define CONFIG_BOOTCOMMAND \
+        "${x_bootcmd_usb}; "    \
+        "setenv bootargs ${x_bootargs} ${x_bootargs_root}; "    \
+        "${x_bootload_kernel}; bootm 0x800000;"
 
 #define CONFIG_MTDPARTS		"orion_nand:1m(u-boot),"	\
 	"4m@1m(uImage),32m@5m(root),219m@37m(data) rw\0"
 
-#define CONFIG_EXTRA_ENV_SETTINGS	"x_bootargs=console"	\
-	"=ttyS0,115200 mtdparts="CONFIG_MTDPARTS	\
-	"x_bootcmd_kernel=nand read 0x6400000 0x100000 0x300000\0" \
-	"x_bootcmd_usb=usb start\0" \
-	"x_bootargs_root=root=/dev/mtdblock3 rw rootfstype=jffs2\0"
+/* CHANGE THIS TO THE IPS YOU WANT TO USE! */
+#define CONFIG_EXTRA_ENV_SETTINGS \
+	"x_bootargs=console=ttyS0,115200 netconsole=@192.168.ip.dockstar/,@192.168.ip.netconsole/ mtdparts="CONFIG_MTDPARTS \
+	"x_bootargs_root=root=/dev/sda2 rootdelay=5\0" \
+	"x_bootload_kernel=ext2load usb 0:1 0x800000 /uImage\0" \
+	"x_bootcmd_usb=usb start\0"
 
 /*
  * Size of malloc() pool
-- 
1.6.4.4

