diff -ruN a/include/configs/mx6_cubox-i.h b/include/configs/mx6_cubox-i.h
--- a/include/configs/mx6_cubox-i.h	2015-10-13 11:56:13.000000000 +0000
+++ b/include/configs/mx6_cubox-i.h	2015-11-05 18:35:27.766174690 +0000
@@ -97,6 +97,7 @@
 #define CONFIG_FS_EXT4
 #define CONFIG_FS_FAT
 #define CONFIG_CMD_FS_GENERIC
+#define CONFIG_SUPPORT_RAW_INITRD
 
 /* Ethernet Configuration */
 #define CONFIG_FEC_MXC
@@ -176,11 +177,14 @@
         "bootfile=auto\0" \
         "bootenv=uEnv.txt\0" \
         "boot_prefixes=/ /boot/\0" \
+        "fdtfile_prefix=dtbs/\0" \
         "console=ttymxc0\0" \
         "splashpos=m,m\0" \
         "fdt_high=0xffffffff\0" \
         "initrd_high=0xffffffff\0" \
         "fdt_addr=0x18000000\0" \
+        "ramdisk_file=initramfs-linux.img\0" \
+        "ramdisk_addr=0x15100000\0" \
         "boot_fdt=try\0" \
         "ip_dyn=yes\0" \
         "mmcdev=" __stringify(CONFIG_SYS_MMC_ENV_DEV) "\0" \
@@ -220,7 +224,9 @@
                 "fi;\0" \
         "loadbootenv=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${file_prefix}${bootenv};\0" \
         "loadfdt=if test ${boottype} = mmc; then " \
-                     "load mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${file_prefix}${fdt_file}; " \
+                     "if load mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${file_prefix}${fdt_file}; then;  else " \
+                               "load mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${file_prefix}${fdtfile_prefix}${fdt_file}; " \
+                     "fi; " \ 
 		"else " \
                      "${get_cmd} ${fdt_addr} ${fdt_file}; " \
 		"fi;\0 " \
@@ -228,13 +234,14 @@
                      "load mmc ${mmcdev}:${mmcpart} ${ramdisk_addr} ${file_prefix}${ramdisk_file}; " \
 		"else " \
                      "${get_cmd} ${ramdisk_addr} ${ramdisk_file}; " \
-		"fi;\0 " \
+		"fi; " \
+               "setenv ramdisk_size :${filesize}; \0" \
         "loadbootfile=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${file_prefix}${bootfile};\0" \
         "importbootenv=echo Importing environment from mmc${mmcdev} ...; " \
                 "env import -t ${loadaddr} ${filesize};\0" \
         "autoboot=echo Booting ${boot_file}; " \
 		"if test ${boot_file} = zImage; then " \
-		    "bootz ${loadaddr} ${ramdisk_addr} ${fdt_addr}; " \
+		    "bootz ${loadaddr} ${ramdisk_addr}${ramdisk_size} ${fdt_addr}; " \
 		"else " \
 		    "bootm ${loadaddr} ${ramdisk_addr} ${fdt_addr}; " \
 		"fi;\0 " \
