diff -ruN a/include/configs/mx6_cubox-i.h b/include/configs/mx6_cubox-i.h
--- a/include/configs/mx6_cubox-i.h	2015-06-30 15:34:48.000000000 +0000
+++ b/include/configs/mx6_cubox-i.h	2015-08-28 05:48:43.225896471 +0000
@@ -176,6 +176,7 @@
         "bootfile=auto\0" \
         "bootenv=uEnv.txt\0" \
         "boot_prefixes=/ /boot/\0" \
+        "fdtfile_prefix=dtbs/\0" \
         "console=ttymxc0\0" \
         "splashpos=m,m\0" \
         "fdt_high=0xffffffff\0" \
@@ -220,7 +221,9 @@
                 "fi;\0" \
         "loadbootenv=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${file_prefix}${bootenv};\0" \
         "loadfdt=if test ${boottype} = mmc; then " \
-                     "load mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${file_prefix}${fdt_file}; " \
+                     "if load mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${file_prefix}${fdt_file}; then;  else " \
+                               "load mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${file_prefix}${fdtfile_prefix}${fdt_file}; " \
+                     "fi; " \ 
 		"else " \
                      "${get_cmd} ${fdt_addr} ${fdt_file}; " \
 		"fi;\0 " \
