From 38f696ece361beffcc3f843c786f8b165d7ad81f Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sat, 9 Dec 2017 10:20:06 -0700
Subject: [PATCH] no neon

---
 cmake/modules/SIMDExt.cmake | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/cmake/modules/SIMDExt.cmake b/cmake/modules/SIMDExt.cmake
index 021524e11e..2c4adc7307 100644
--- a/cmake/modules/SIMDExt.cmake
+++ b/cmake/modules/SIMDExt.cmake
@@ -68,10 +68,7 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
 
 elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "arm|ARM")
   set(HAVE_ARM 1)
-  CHECK_C_COMPILER_FLAG(-mfpu=neon HAVE_ARM_NEON)
-  if(HAVE_ARM_NEON)
-    set(SIMD_COMPILE_FLAGS "${SIMD_COMPILE_FLAGS} -mfpu=neon")
-  endif()
+  set(HAVE_ARM_NEON 0)
 
 elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "i386|i686|amd64|x86_64|AMD64")
   set(HAVE_INTEL 1)
-- 
2.17.1

