--- a/CMakeLists.txt	2017-09-26 18:27:07.000000000 +0200
+++ b/CMakeLists.txt	2017-10-06 14:10:11.032378659 +0200
@@ -63,7 +63,7 @@
 option(WITH_MANPAGE "Build man pages." ON)
 if(WITH_MANPAGE)
   find_program(SPHINX_BUILD
-    sphinx-build)
+    sphinx-build2)
   if(NOT SPHINX_BUILD)
     message(FATAL_ERROR "Can't find sphinx-build.")
   endif(NOT SPHINX_BUILD)
--- a/src/tools/setup-virtualenv.sh	2017-09-26 18:27:07.000000000 +0200
+++ b/src/tools/setup-virtualenv.sh	2017-10-06 15:02:01.736664556 +0200
@@ -18,10 +18,10 @@
 DIR=$1
 rm -fr $DIR
 mkdir -p $DIR
-virtualenv --python python2.7 $DIR
+virtualenv2 --python python2.7 $DIR
 . $DIR/bin/activate
 
-if pip --help | grep -q disable-pip-version-check; then
+if pip2 --help | grep -q disable-pip-version-check; then
     DISABLE_PIP_VERSION_CHECK=--disable-pip-version-check
 else
     DISABLE_PIP_VERSION_CHECK=
@@ -29,12 +29,12 @@
 
 # older versions of pip will not install wrap_console scripts
 # when using wheel packages
-pip $DISABLE_PIP_VERSION_CHECK --log $DIR/log.txt install --upgrade 'pip >= 6.1'
+pip2 $DISABLE_PIP_VERSION_CHECK --log $DIR/log.txt install --upgrade 'pip >= 6.1'
 
 # workaround of https://github.com/pypa/setuptools/issues/1042
-pip $DISABLE_PIP_VERSION_CHECK --log $DIR/log.txt install --upgrade "setuptools < 36"
+pip2 $DISABLE_PIP_VERSION_CHECK --log $DIR/log.txt install --upgrade "setuptools < 36"
 
-if pip --help | grep -q disable-pip-version-check; then
+if pip2 --help | grep -q disable-pip-version-check; then
     DISABLE_PIP_VERSION_CHECK=--disable-pip-version-check
 else
     DISABLE_PIP_VERSION_CHECK=
@@ -44,7 +44,7 @@
     export NO_INDEX=--no-index
 fi
 
-pip $DISABLE_PIP_VERSION_CHECK --log $DIR/log.txt install $NO_INDEX --use-wheel --find-links=file://$(pwd)/wheelhouse 'tox >=1.9'
+pip2 $DISABLE_PIP_VERSION_CHECK --log $DIR/log.txt install $NO_INDEX --use-wheel --find-links=file://$(pwd)/wheelhouse 'tox >=1.9'
 if test -f requirements.txt ; then
-    pip $DISABLE_PIP_VERSION_CHECK --log $DIR/log.txt install $NO_INDEX --use-wheel --find-links=file://$(pwd)/wheelhouse -r requirements.txt
+    pip2 $DISABLE_PIP_VERSION_CHECK --log $DIR/log.txt install $NO_INDEX --use-wheel --find-links=file://$(pwd)/wheelhouse -r requirements.txt
 fi
