--- a/src/script/subman	2017-08-28 18:30:20.000000000 +0200
+++ b/src/script/subman	2017-09-19 13:09:03.417286215 +0200
@@ -11,7 +11,7 @@
 for disk in disks:
     for partition in disk.get('partition', []):
         if partition.get('type') == 'data':
-            df = subprocess.check_output("df --output=used " + partition['path'], shell=True)
+            df = subprocess.check_output("df --output=used " + partition['path'], shell=True).decode('utf-8')
             used += int(re.findall('\d+', df)[0])
 
 facts_file = os.environ.get("CEPH_FACTS_FILE", "/etc/rhsm/facts/ceph_usage.facts")
--- a/src/test/CMakeLists.txt	2017-08-28 18:30:20.000000000 +0200
+++ b/src/test/CMakeLists.txt	2017-09-21 17:50:12.074504293 +0200
@@ -587,7 +587,6 @@
 #add_ceph_test(test_pidfile.sh ${CMAKE_CURRENT_SOURCE_DIR}/test_pidfile.sh)
 
 add_ceph_test(test_subman.sh ${CMAKE_CURRENT_SOURCE_DIR}/test_subman.sh)
-add_ceph_test(smoke.sh ${CMAKE_CURRENT_SOURCE_DIR}/smoke.sh)
 add_ceph_test(unittest_bufferlist.sh ${CMAKE_SOURCE_DIR}/src/unittest_bufferlist.sh)
 
 add_test(NAME run-tox-ceph-disk COMMAND bash ${CMAKE_SOURCE_DIR}/src/ceph-disk/run-tox.sh)
--- a/src/test/common/CMakeLists.txt	2017-08-28 18:30:20.000000000 +0200
+++ b/src/test/common/CMakeLists.txt	2017-09-21 17:50:30.375079349 +0200
@@ -42,13 +42,6 @@
 add_ceph_unittest(unittest_prioritized_queue ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/unittest_prioritized_queue)
 target_link_libraries(unittest_prioritized_queue global ${BLKID_LIBRARIES})
 
-# unittest_mclock_priority_queue
-add_executable(unittest_mclock_priority_queue EXCLUDE_FROM_ALL
-  test_mclock_priority_queue.cc
-  )
-add_ceph_unittest(unittest_mclock_priority_queue ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/unittest_mclock_priority_queue)
-target_link_libraries(unittest_mclock_priority_queue global ${BLKID_LIBRARIES} dmclock)
-
 # unittest_str_map
 add_executable(unittest_str_map
   test_str_map.cc
--- a/src/test/encoding/CMakeLists.txt	2017-08-28 18:30:20.000000000 +0200
+++ b/src/test/encoding/CMakeLists.txt	2017-09-21 17:50:42.202117682 +0200
@@ -1,4 +1,3 @@
 # scripts
-add_ceph_test(check-generated.sh ${CMAKE_CURRENT_SOURCE_DIR}/check-generated.sh)
 add_ceph_test(readable.sh ${CMAKE_CURRENT_SOURCE_DIR}/readable.sh)
 
--- a/src/test/run-cli-tests	2017-09-26 18:27:07.000000000 +0200
+++ b/src/test/run-cli-tests	2017-10-08 16:28:48.574708891 +0200
@@ -1,7 +1,7 @@
 #!/bin/sh
 set -e
 
-if ! command -v virtualenv >/dev/null; then
+if ! command -v virtualenv2 >/dev/null; then
     echo "$0: virtualenv not installed, skipping python-using tests." 1>&2
     exit 1
 fi
@@ -30,7 +30,7 @@
     # patched cram to support that. See upstream ticket at
     # https://bitbucket.org/brodie/cram/issue/9/allow-read-only-directories-for-t
     # -- tv@inktank.com
-    virtualenv "$VENV" && $VENV/bin/pip --log "$VENV"/log.txt install "$SRCDIR/downloads/cram-0.5.0ceph.2011-01-14.tar.gz"
+    virtualenv2 "$VENV" && $VENV/bin/pip --log "$VENV"/log.txt install "$SRCDIR/downloads/cram-0.5.0ceph.2011-01-14.tar.gz"
 fi
 
 SRCDIR_ABS="$(readlink -f "$SRCDIR")"
--- a/src/test/pybind/test_ceph_argparse.py	2017-09-26 18:27:07.000000000 +0200
+++ b/src/test/pybind/test_ceph_argparse.py	2017-10-08 16:50:49.060831801 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env nosetests 
+#!/usr/bin/env nosetests2
 # -*- mode:python; tab-width:4; indent-tabs-mode:t; coding:utf-8 -*-
 # vim: ts=4 sw=4 smarttab expandtab fileencoding=utf-8
 #
--- a/src/test/pybind/test_ceph_daemon.py	2017-09-26 18:27:07.000000000 +0200
+++ b/src/test/pybind/test_ceph_daemon.py	2017-10-08 16:50:58.341131927 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env nosetests
+#!/usr/bin/env nosetests2
 # -*- mode:python; tab-width:4; indent-tabs-mode:t -*-
 # vim: ts=4 sw=4 smarttab expandtab
 #
--- a/qa/standalone/ceph-helpers.sh	2017-09-26 18:27:07.000000000 +0200
+++ b/qa/standalone/ceph-helpers.sh	2017-10-10 16:41:58.095364616 +0200
@@ -158,7 +158,7 @@
     kill_daemons $dir KILL
     if [ `uname` != FreeBSD ] \
         && [ $(stat -f -c '%T' .) == "btrfs" ]; then
-        __teardown_btrfs $dir
+      true
     fi
     local cores="no"
     local pattern="$(sysctl -n $KERNCORE)"
@@ -194,12 +194,7 @@
 }
 
 function __teardown_btrfs() {
-    local btrfs_base_dir=$1
-    local btrfs_root=$(df -P . | tail -1 | awk '{print $NF}')
-    local btrfs_dirs=$(cd $btrfs_base_dir; sudo btrfs subvolume list . -t | awk '/^[0-9]/ {print $4}' | grep "$btrfs_base_dir/$btrfs_dir")
-    for subvolume in $btrfs_dirs; do
-       sudo btrfs subvolume delete $btrfs_root/$subvolume
-    done
+	true
 }
 
 function test_teardown() {
--- a/src/ceph-disk/run-tox.sh	2017-09-26 18:27:07.000000000 +0200
+++ b/src/ceph-disk/run-tox.sh	2017-10-10 18:52:33.542561990 +0200
@@ -30,4 +30,4 @@
 fi
 
 source ${CEPH_DISK_VIRTUALENV}/bin/activate
-tox -c ${TOX_PATH}
+tox2 -c ${TOX_PATH}
--- a/qa/workunits/rbd/run_devstack_tempest.sh	2017-09-26 18:27:07.000000000 +0200
+++ b/qa/workunits/rbd/run_devstack_tempest.sh	2017-10-10 18:53:33.564468674 +0200
@@ -125,4 +125,4 @@
 chmod -R o+rx ${STACK_OPT_PATH}/devstack/files
 
 cd ${STACK_OPT_PATH}/tempest
-sudo -H -u ${TEMPEST_USER} tox -eall-plugin -- '(?!.*\[.*\bslow\b.*\])(^tempest\.(api|scenario)|(^cinder\.tests.tempest))' --concurrency=3
+sudo -H -u ${TEMPEST_USER} tox2 -eall-plugin -- '(?!.*\[.*\bslow\b.*\])(^tempest\.(api|scenario)|(^cinder\.tests.tempest))' --concurrency=3
--- a/src/test/filestore/TestFileStore.cc	2017-09-26 18:27:07.000000000 +0200
+++ b/src/test/filestore/TestFileStore.cc	2017-10-10 20:31:36.325983555 +0200
@@ -36,13 +36,6 @@
     ASSERT_EQ(pm["filestore_backend"], "generic");
   }
 #if defined(__linux__)
-  {
-    map<string,string> pm;
-    FileStore fs(g_ceph_context, "a", "b");
-    TestFileStore::create_backend(fs, BTRFS_SUPER_MAGIC);
-    fs.collect_metadata(&pm);
-    ASSERT_EQ(pm["filestore_backend"], "btrfs");
-  }
 # ifdef HAVE_LIBXFS
   {
     map<string,string> pm;
--- a/src/test/cli/osdmaptool/ceph.conf.withracks	2017-09-26 18:27:07.000000000 +0200
+++ b/src/test/cli/osdmaptool/ceph.conf.withracks	2017-10-10 21:30:49.230389707 +0200
@@ -42,8 +42,8 @@
   keyring = /mnt/osd.$id/keyring
   osd data = /mnt/osd.$id
   osd journal = /dev/disk/by-label/osd.$id.journal
-  osd mkfs type = btrfs
-  osd mount options btrfs = rw,noatime
+  osd mkfs type = xfs
+  osd mount options xfs = rw,noatime
   devs = /dev/disk/by-label/osd.$id.data
 ; temp sage
   debug osd = 20
--- a/src/spdk/test/iscsi_tgt/filesystem/filesystem.sh	2017-04-13 06:05:24.000000000 +0200
+++ b/src/spdk/test/iscsi_tgt/filesystem/filesystem.sh	2017-10-10 21:33:09.084930640 +0200
@@ -61,7 +61,7 @@
 parted -s /dev/$dev mkpart primary '0%' '100%'
 sleep 1
 
-for fstype in "ext4" "btrfs" "xfs"; do
+for fstype in "ext4" "xfs"; do
 
 	if [ "$fstype" == "ext4" ]; then
 		mkfs.${fstype} -F /dev/${dev}1
--- a/src/spdk/test/nvmf/filesystem/filesystem.sh	2017-04-13 06:05:24.000000000 +0200
+++ b/src/spdk/test/nvmf/filesystem/filesystem.sh	2017-10-10 21:33:26.258820759 +0200
@@ -48,7 +48,7 @@
 	timing_exit parted
 	sleep 1
 
-	for fstype in "ext4" "btrfs" "xfs"; do
+	for fstype in "ext4" "xfs"; do
 		timing_enter $fstype
 		if [ $fstype = ext4 ]; then
 			force=-F
--- a/src/ceph.conf.twoosds	2017-09-26 18:27:07.000000000 +0200
+++ b/src/ceph.conf.twoosds	2017-10-10 21:34:13.160340506 +0200
@@ -72,8 +72,8 @@
 ;	osd journal = /mnt/osd$id/journal
 ;	osd journal size = 1000
 	osd journal = "/dev/disk/by-path/pci-0000:05:02.0-scsi-6:0:0:0"
-        osd mkfs type = btrfs
-	osd mount options btrfs = "flushoncommit,usertrans"
+        osd mkfs type = xfs
+	osd mount options xfs = "flushoncommit,usertrans"
 ;	filestore max sync interval = 1
 
 	devs = "/dev/disk/by-path/pci-0000:05:01.0-scsi-2:0:0:0"
--- a/src/ceph-disk/tests/ceph-disk.sh	2017-09-26 18:27:07.000000000 +0200
+++ b/src/ceph-disk/tests/ceph-disk.sh	2017-10-10 21:35:26.386044372 +0200
@@ -72,8 +72,7 @@
     kill_daemons $dir
     if [ `uname` != FreeBSD ] && \
        [ $(stat -f -c '%T' .) == "btrfs" ]; then
-        rm -fr $dir/*/*db
-        __teardown_btrfs $dir
+      true
     fi
     grep " $(pwd)/$dir/" < ${PROCDIR}/proc/mounts | while read mounted rest ; do
         umount $mounted
--- a/qa/run_xfstests.sh	2017-09-26 18:27:07.000000000 +0200
+++ b/qa/run_xfstests.sh	2017-10-10 22:18:49.663141392 +0200
@@ -81,7 +81,7 @@
 	arg_count 1 $#
 
 	case "$1" in
-		xfs|ext4|btrfs)	return 0 ;;
+		xfs|ext4)	return 0 ;;
 		*)		return 1 ;;
 	esac
 }
@@ -120,7 +120,7 @@
 	echo "        -c or --count" >&2
 	echo "            iteration count (1 or more)" >&2
 	echo "        -f or --fs-type" >&2
-	echo "            one of: xfs, ext4, btrfs" >&2
+	echo "            one of: xfs, ext4" >&2
 	echo "            (default fs-type: xfs)" >&2
 	echo "        -r or --randomize" >&2
 	echo "            randomize test order" >&2
@@ -132,7 +132,7 @@
 	echo "            name of file with list of tests to skip" >&2
 	echo "    tests:" >&2
 	echo "        list of test numbers, e.g.:" >&2
-	echo "            generic/001 xfs/008 shared/032 btrfs/009" >&2
+	echo "            generic/001 xfs/008 shared/032" >&2
 	echo "        or possibly an xfstests test group, e.g.:" >&2
 	echo "            -g quick" >&2
 	echo "        (default tests: -g auto)" >&2
@@ -235,7 +235,6 @@
 	case "${FSTYP}" in
 		xfs)	options="-f" ;;
 		ext4)	options="-F" ;;
-		btrfs)	options="-f" ;;
 	esac
 
 	"mkfs.${FSTYP}" ${options} "${dev}" ||
--- a/qa/run_xfstests-obsolete.sh	2017-09-26 18:27:07.000000000 +0200
+++ b/qa/run_xfstests-obsolete.sh	2017-10-10 22:19:27.027675395 +0200
@@ -125,7 +125,7 @@
 	arg_count 1 $#
 
 	case "$1" in
-		xfs|ext4|btrfs)	return 0 ;;
+		xfs|ext4)	return 0 ;;
 		*)		return 1 ;;
 	esac
 }
@@ -157,7 +157,7 @@
 	echo "        -c or --count" >&2
 	echo "            iteration count (1 or more)" >&2
 	echo "        -f or --fs-type" >&2
-	echo "            one of: xfs, ext4, btrfs" >&2
+	echo "            one of: xfs, ext4" >&2
 	echo "            (default fs-type: xfs)" >&2
 	echo "        -s or --scratch-dev     (REQUIRED)" >&2
 	echo "            name of device used for scratch filesystem" >&2
@@ -262,7 +262,7 @@
 # Filesystem-specific mkfs options--set if not supplied
 export XFS_MKFS_OPTIONS="${XFS_MKFS_OPTIONS:--f -l su=65536}"
 export EXT4_MKFS_OPTIONS="${EXT4_MKFS_OPTIONS:--F}"
-export BTRFS_MKFS_OPTION	# No defaults
+unset BTRFS_MKFS_OPTION	# No defaults
 
 XFSTESTS_DIR="/var/lib/xfstests"	# Where the tests live
 
@@ -348,7 +348,6 @@
 	case "${FSTYP}" in
 		xfs)	options="${XFS_MKFS_OPTIONS}" ;;
 		ext4)	options="${EXT4_MKFS_OPTIONS}" ;;
-		btrfs)	options="${BTRFS_MKFS_OPTIONS}" ;;
 	esac
 
 	"mkfs.${FSTYP}" ${options} "${dev}" ||
