--- a/src/common/util.cc	2017-09-26 18:27:07.000000000 +0200
+++ b/src/common/util.cc	2017-10-05 15:58:14.852647976 +0200
@@ -158,8 +158,7 @@
 {
   static const map<string, string> kvm = {
     { "distro", "ID=" },
-    { "distro_description", "PRETTY_NAME=" },
-    { "distro_version", "VERSION_ID=" }
+    { "distro_description", "PRETTY_NAME=" }
   };
 
   FILE *fp = fopen("/etc/os-release", "r");
@@ -182,7 +181,7 @@
     lderr(cct) << "distro_detect - /etc/os-release is required" << dendl;
   }
 
-  for (const char* rk: {"distro", "distro_version"}) {
+  for (const char* rk: {"distro"}) {
     if (m->find(rk) == m->end())
       lderr(cct) << "distro_detect - can't detect " << rk << dendl;
   }
