diff -ruN -x .gitignore -x .cvsignore xf86-input-elo2300-585245ac87ce421acd890f065dbcb187ed901c1a/Makefile.am xf86-input-elo2300-77d10116e972591a5f593a008ce5f680219d1e01/Makefile.am
--- xf86-input-elo2300-585245ac87ce421acd890f065dbcb187ed901c1a/Makefile.am	2008-05-21 18:54:57.000000000 +0000
+++ xf86-input-elo2300-77d10116e972591a5f593a008ce5f680219d1e01/Makefile.am	2009-02-02 21:02:43.000000000 +0000
@@ -20,11 +20,12 @@
 
 AUTOMAKE_OPTIONS = foreign
 SUBDIRS = src man
-
-CLEANFILES = ChangeLog
 EXTRA_DIST = ChangeLog
+MAINTAINERCLEANFILES = ChangeLog
 
 .PHONY: ChangeLog
 
 ChangeLog:
-	(GIT_DIR=$(top_srcdir)/.git git-log > .changelog.tmp && mv .changelog.tmp ChangeLog; rm -f .changelog.tmp) || (touch ChangeLog; echo 'git directory not found: installing possibly empty changelog.' >&2)
+	$(CHANGELOG_CMD)
+
+dist-hook: ChangeLog
diff -ruN -x .gitignore -x .cvsignore xf86-input-elo2300-585245ac87ce421acd890f065dbcb187ed901c1a/README xf86-input-elo2300-77d10116e972591a5f593a008ce5f680219d1e01/README
--- xf86-input-elo2300-585245ac87ce421acd890f065dbcb187ed901c1a/README	1970-01-01 00:00:00.000000000 +0000
+++ xf86-input-elo2300-77d10116e972591a5f593a008ce5f680219d1e01/README	2009-02-02 21:02:43.000000000 +0000
@@ -0,0 +1,20 @@
+xf86-input-elo2300 - EloGraphics 2300 input driver for the Xorg server
+
+Please submit bugs & patches to the Xorg bugzilla:
+
+        https://bugs.freedesktop.org/enter_bug.cgi?product=xorg
+
+All questions regarding this software should be directed at the
+Xorg mailing list:
+
+        http://lists.freedesktop.org/mailman/listinfo/xorg
+
+The master development code repository can be found at:
+
+        git://anongit.freedesktop.org/git/xorg/driver/xf86-input-elo2300
+
+        http://cgit.freedesktop.org/xorg/driver/xf86-input-elo2300
+
+For more information on the git code manager, see:
+
+        http://wiki.x.org/wiki/GitPage
diff -ruN -x .gitignore -x .cvsignore xf86-input-elo2300-585245ac87ce421acd890f065dbcb187ed901c1a/configure.ac xf86-input-elo2300-77d10116e972591a5f593a008ce5f680219d1e01/configure.ac
--- xf86-input-elo2300-585245ac87ce421acd890f065dbcb187ed901c1a/configure.ac	2008-05-21 18:54:57.000000000 +0000
+++ xf86-input-elo2300-77d10116e972591a5f593a008ce5f680219d1e01/configure.ac	2009-02-02 21:02:43.000000000 +0000
@@ -32,15 +32,16 @@
 
 AM_MAINTAINER_MODE
 
-DRIVER_NAME=elo2300
-AC_SUBST([DRIVER_NAME])
-
+# Require xorg-macros: XORG_CWARNFLAGS, XORG_CHANGELOG
+m4_ifndef([XORG_MACROS_VERSION], [AC_FATAL([must install xorg-macros 1.2 or later before running autoconf/autogen])])
+XORG_MACROS_VERSION(1.2)
 AM_CONFIG_HEADER([config.h])
 
 # Checks for programs.
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
 AC_PROG_CC
+XORG_CWARNFLAGS
 
 AH_TOP([#include "xorg-server.h"])
 
@@ -77,16 +78,18 @@
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XORG, [xorg-server >= 1.0.99.901] xproto $REQUIRED_MODULES)
 sdkdir=$(pkg-config --variable=sdkdir xorg-server)
-
-CFLAGS="$CFLAGS $XORG_CFLAGS "' -I$(top_srcdir)/src'
-AC_SUBST([CFLAGS])
+XORG_CFLAGS="$CWARNFLAGS $XORG_CFLAGS"
 
 # Checks for libraries.
 
 # Checks for header files.
 AC_HEADER_STDC
 
+DRIVER_NAME=elo2300
+AC_SUBST([DRIVER_NAME])
+
 XORG_MANPAGE_SECTIONS
 XORG_RELEASE_VERSION
+XORG_CHANGELOG
 
 AC_OUTPUT([Makefile src/Makefile man/Makefile])
diff -ruN -x .gitignore -x .cvsignore xf86-input-elo2300-585245ac87ce421acd890f065dbcb187ed901c1a/src/Makefile.am xf86-input-elo2300-77d10116e972591a5f593a008ce5f680219d1e01/src/Makefile.am
--- xf86-input-elo2300-585245ac87ce421acd890f065dbcb187ed901c1a/src/Makefile.am	2008-05-21 18:54:57.000000000 +0000
+++ xf86-input-elo2300-77d10116e972591a5f593a008ce5f680219d1e01/src/Makefile.am	2009-02-02 21:02:43.000000000 +0000
@@ -24,6 +24,8 @@
 # -avoid-version prevents gratuitous .0.0.0 version numbers on the end
 # _ladir passes a dummy rpath to libtool so the thing will actually link
 # TODO: -nostdlib/-Bstatic/-lgcc platform magic, not installing the .a, etc.
+AM_CFLAGS = $(XORG_CFLAGS)
+
 @DRIVER_NAME@_drv_la_LTLIBRARIES = @DRIVER_NAME@_drv.la
 @DRIVER_NAME@_drv_la_LDFLAGS = -module -avoid-version
 @DRIVER_NAME@_drv_ladir = @inputdir@
diff -ruN -x .gitignore -x .cvsignore xf86-input-elo2300-585245ac87ce421acd890f065dbcb187ed901c1a/src/elo.c xf86-input-elo2300-77d10116e972591a5f593a008ce5f680219d1e01/src/elo.c
--- xf86-input-elo2300-585245ac87ce421acd890f065dbcb187ed901c1a/src/elo.c	2008-05-21 18:54:57.000000000 +0000
+++ xf86-input-elo2300-77d10116e972591a5f593a008ce5f680219d1e01/src/elo.c	2009-02-02 21:02:43.000000000 +0000
@@ -309,7 +309,10 @@
 	 * Device reports motions on 2 axes in absolute coordinates.
 	 * Device may reports touch pressure on the 3rd axis.
 	 */
-	if (InitValuatorClassDeviceStruct (dev, priv->axes, xf86GetMotionEvents,
+	if (InitValuatorClassDeviceStruct (dev, priv->axes,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
+                    xf86GetMotionEvents,
+#endif
 									local->history_size, Absolute) == FALSE)
 	{
 		ErrorF ("Unable to allocate Elographics touchscreen ValuatorClassDeviceStruct\n");
