diff -wbBur tse3-0.3.1.org/src/examples/recording/recording.cpp tse3-0.3.1/src/examples/recording/recording.cpp
--- tse3-0.3.1.org/src/examples/recording/recording.cpp	2005-07-25 15:22:31.000000000 +0400
+++ tse3-0.3.1/src/examples/recording/recording.cpp	2008-08-08 15:26:56.000000000 +0400
@@ -26,6 +26,7 @@
      *
      **************************************************************/
 
+#include <stdlib.h>
 #include <iostream>
 
 // Used in step 1
diff -wbBur tse3-0.3.1.org/src/tse3/MidiFile.cpp tse3-0.3.1/src/tse3/MidiFile.cpp
--- tse3-0.3.1.org/src/tse3/MidiFile.cpp	2005-07-25 15:23:00.000000000 +0400
+++ tse3-0.3.1/src/tse3/MidiFile.cpp	2008-08-08 15:22:38.000000000 +0400
@@ -32,6 +32,7 @@
 #include <string>
 #include <queue>
 #include <math.h>
+#include <string.h>
 
 using namespace TSE3;
 
@@ -1199,7 +1200,7 @@
         writeVariable(out, 0);    // delta time of zero
         writeFixed(out, 0xff, 1); // meta event
         writeFixed(out, 0x01, 1); // text event
-        writeVariable(out, std::strlen(TSE3MESSAGE)+1); // len
+        writeVariable(out, strlen(TSE3MESSAGE)+1); // len
         writeString(out, TSE3MESSAGE, true);
     }
     else
diff -wbBur tse3-0.3.1.org/src/tse3/Serializable.h tse3-0.3.1/src/tse3/Serializable.h
--- tse3-0.3.1.org/src/tse3/Serializable.h	2005-07-25 15:23:00.000000000 +0400
+++ tse3-0.3.1/src/tse3/Serializable.h	2008-08-08 15:18:18.000000000 +0400
@@ -17,6 +17,7 @@
 #ifndef TSE3_SERIALIZABLE_H
 #define TSE3_SERIALIZABLE_H
 
+#include <iostream>
 #include <iosfwd>
 #include <iomanip>
 #include <cstddef>
diff -wbBur tse3-0.3.1.org/src/tse3/TSE2MDL.cpp tse3-0.3.1/src/tse3/TSE2MDL.cpp
--- tse3-0.3.1.org/src/tse3/TSE2MDL.cpp	2005-07-25 15:23:00.000000000 +0400
+++ tse3-0.3.1/src/tse3/TSE2MDL.cpp	2008-08-08 15:19:47.000000000 +0400
@@ -38,6 +38,7 @@
 #include "tse3/Progress.h"
 
 #include <fstream>
+#include <string.h>
 
 using namespace TSE3;
 
diff -wbBur tse3-0.3.1.org/src/tse3/cmd/Phrase.h tse3-0.3.1/src/tse3/cmd/Phrase.h
--- tse3-0.3.1.org/src/tse3/cmd/Phrase.h	2005-07-25 15:22:38.000000000 +0400
+++ tse3-0.3.1/src/tse3/cmd/Phrase.h	2008-08-08 15:12:35.000000000 +0400
@@ -19,6 +19,7 @@
 
 #include "tse3/cmd/Command.h"
 #include "tse3/Phrase.h"
+#include "tse3/PhraseEdit.h"
 #include "tse3/DisplayParams.h"
 
 #include <vector>
diff -wbBur tse3-0.3.1.org/src/tse3play/tse3play.cpp tse3-0.3.1/src/tse3play/tse3play.cpp
--- tse3-0.3.1.org/src/tse3play/tse3play.cpp	2005-08-23 16:58:35.000000000 +0400
+++ tse3-0.3.1/src/tse3play/tse3play.cpp	2008-08-08 15:27:26.000000000 +0400
@@ -32,6 +32,7 @@
 #include "tse3/Error.h"
 #include "tse3/Metronome.h"
 #include <fstream>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
