diff --git a/src/modules/audio_processing/transient/file_utils.h b/src/modules/audio_processing/transient/file_utils.h
index b7483377..687ebcef 100644
--- a/src/modules/audio_processing/transient/file_utils.h
+++ b/src/modules/audio_processing/transient/file_utils.h
@@ -12,6 +12,7 @@
 #define MODULES_AUDIO_PROCESSING_TRANSIENT_FILE_UTILS_H_
 
 #include <string.h>
+#include <cstdint>
 
 #include "rtc_base/system/file_wrapper.h"
 
diff --git a/src/rtc_base/third_party/base64/base64.h b/src/rtc_base/third_party/base64/base64.h
index 4190a79d..e224ffd0 100644
--- a/src/rtc_base/third_party/base64/base64.h
+++ b/src/rtc_base/third_party/base64/base64.h
@@ -14,6 +14,7 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
 
 #include "absl/strings/string_view.h"
 #include "rtc_base/system/rtc_export.h"
