diff -ru egenix-mx-base-2.0.5/mxSetup.py egenix-mx-base-2.0.5.new/mxSetup.py
--- egenix-mx-base-2.0.5/mxSetup.py     2003-08-09 16:52:16.000000000 -0400
+++ egenix-mx-base-2.0.5.new/mxSetup.py 2004-05-24 12:33:48.000000000 -0400
@@ -118,8 +118,8 @@
         so = os.environ.get('SO', so)
         compiler.set_executables(
             preprocessor='%s -E' % cc,
-            compiler='%s %s' % (cc, opt),
-            compiler_so='%s %s %s' % (cc, opt, ccshared),
+            compiler='%s %s' % (cc, "-DNDEBUG -O0 -Wall -Wstrict-prototypes"),
+            compiler_so='%s %s %s' % (cc, "-DNDEBUG -O0 -Wall -Wstrict-prototypes", ccshared),
             linker_so=ldshared,
             linker_exe=cc)
         compiler.shared_lib_extension = so
