--- a/CMakeLists.txt	2008-11-12 14:00:53.000000000 +0200
+++ b/CMakeLists.txt	2008-11-12 19:05:36.000000000 +0200
@@ -18,13 +18,8 @@
 set(RELATIVE_DATADIR "share/cuneiform")
 set(INSTALL_DATADIR "${CMAKE_INSTALL_PREFIX}/${RELATIVE_DATADIR}")
 
-include(CheckTypeSize)
-check_type_size("void*" PTRSIZE)
-if(PTRSIZE EQUAL 8 AND CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    set(LIBDIR "lib64")
-else()
-    set(LIBDIR "lib")
-endif()
+set(LIBDIR "lib/cuneiform-${CF_VERSION}")
+set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${LIBDIR}")
 
 add_subdirectory(datafiles)
 add_subdirectory(cuneiform_src)
