Source: https://www.olimex.com/forum/index.php?topic=1810

diff --git a/arch/arm/boot/dts/imx23-olinuxino.dts b/arch/arm/boot/dts/imx23-olinuxino.dts
index d107c4a..993da2e 100644
--- a/arch/arm/boot/dts/imx23-olinuxino.dts
+++ b/arch/arm/boot/dts/imx23-olinuxino.dts
@@ -64,7 +64,14 @@
 				compatible = "fsl,imx23-spi";
 				pinctrl-names = "default";
 				pinctrl-0 = <&spi2_pins_a>;
+				clock-frequency = <1000000>;
 				status = "okay";
+
+				spidev0: spidev@0 {
+					compatible = "spidev";
+					spi-max-frequency = <1000000>;
+					reg = <0>;
+				};
 			};
 		};
 
