diff -Naur dcron-orig/defs.h dcron/defs.h
--- dcron-orig/defs.h	2006-04-29 10:20:58.000000000 -0700
+++ dcron/defs.h	2006-05-04 18:40:05.000000000 -0700
@@ -28,7 +28,7 @@
 #define arysize(ary)	(sizeof(ary)/sizeof((ary)[0]))
 
 #ifndef CRONTABS
-#define CRONTABS	"/var/spool/cron/crontabs"
+#define CRONTABS	"/var/spool/cron"
 #endif
 #ifndef SCRONTABS
 #define SCRONTABS	"/etc/cron.d"
@@ -44,7 +44,7 @@
 #if defined(__DragonFly__)
 #define SENDMAIL	"/usr/sbin/sendmail"
 #else
-#define SENDMAIL	"/usr/lib/sendmail"
+#define SENDMAIL	"/usr/sbin/sendmail"
 #endif
 #endif
 
diff -Naur dcron-orig/subs.c dcron/subs.c
--- dcron-orig/subs.c	2006-04-27 10:29:56.000000000 -0700
+++ dcron/subs.c	2006-05-04 18:40:19.000000000 -0700
@@ -79,7 +79,7 @@
 
     buf[0] = 0;
     if (useDate)
-	strftime(buf, 128, "%d-%b-%y %H:%M  ", tp);
+	strftime(buf, 128, "%d-%b-%Y %H:%M  ", tp);
     vsnprintf(buf + strlen(buf), nmax, ctl, va);
     return(strlen(buf));
 }
