diff -Naur dcron-orig/job.c dcron/job.c
--- dcron-orig/job.c	2009-04-24 16:35:20.000000000 +0400
+++ dcron/job.c	2009-04-24 16:35:41.000000000 +0400
@@ -213,7 +213,7 @@
 	/*
 	 * note: 8 is a file descriptor
 	 */
-	logfd(8, "unable to exec %s %s, user %s, output to sink null", 
+	logfd(8, "unable to exec %s %s, user %s, output to sink null\n", 
 	    SENDMAIL,
 	    SENDMAIL_ARGS,
 	    file->cf_UserName
