diff -a -u -d -r pymsnt-0.11.3/src/legacy/msn/msn.py pymsnt/src/legacy/msn/msn.py
--- pymsnt-0.11.3/src/legacy/msn/msn.py	2008-02-08 16:55:07.000000000 +0300
+++ pymsnt-0.11.3/src/legacy/msn/msn.py	2008-08-05 20:40:46.000000000 +0400
@@ -110,7 +110,7 @@
 from urllib import quote, unquote
 
 
-MSN_PROTOCOL_VERSION = "MSNP11 CVR0"      # protocol version
+MSN_PROTOCOL_VERSION = "MSNP11"      # protocol version
 MSN_PORT             = 1863               # default dispatch server port
 MSN_MAX_MESSAGE      = 1664               # max message length
 MSN_CVR_STR          = "0x040c winnt 5.1 i386 MSNMSGR 7.0.0777 msmsgs"
