diff -ruN -x .gitignore -x .cvsignore xf86-input-palmax-17f1dc5d593a3056bee74d53a31e38065b7236d0/README xf86-input-palmax-5f8c3d97ac2edb3a7ee744369a9826a645b322f9/README
--- xf86-input-palmax-17f1dc5d593a3056bee74d53a31e38065b7236d0/README	1970-01-01 00:00:00.000000000 +0000
+++ xf86-input-palmax-5f8c3d97ac2edb3a7ee744369a9826a645b322f9/README	2009-01-31 04:28:25.000000000 +0000
@@ -0,0 +1,20 @@
+xf86-input-palmax - Palmax (TR88L803) touchscreen driver for the Xorg X server
+
+Please submit bugs & patches to the Xorg bugzilla:
+
+        https://bugs.freedesktop.org/enter_bug.cgi?product=xorg
+
+All questions regarding this software should be directed at the
+Xorg mailing list:
+
+        http://lists.freedesktop.org/mailman/listinfo/xorg
+
+The master development code repository can be found at:
+
+        git://anongit.freedesktop.org/git/xorg/driver/xf86-input-palmax
+
+        http://cgit.freedesktop.org/xorg/driver/xf86-input-palmax
+
+For more information on the git code manager, see:
+
+        http://wiki.x.org/wiki/GitPage
diff -ruN -x .gitignore -x .cvsignore xf86-input-palmax-17f1dc5d593a3056bee74d53a31e38065b7236d0/man/palmax.man xf86-input-palmax-5f8c3d97ac2edb3a7ee744369a9826a645b322f9/man/palmax.man
--- xf86-input-palmax-17f1dc5d593a3056bee74d53a31e38065b7236d0/man/palmax.man	2008-03-20 20:34:08.000000000 +0000
+++ xf86-input-palmax-5f8c3d97ac2edb3a7ee744369a9826a645b322f9/man/palmax.man	2009-01-31 04:28:25.000000000 +0000
@@ -82,7 +82,7 @@
 __xservername__ needs a nice calibration tool.
 .PP
 .SH "SEE ALSO"
-__xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), xorgconfig(__appmansuffix__), Xserver(__appmansuffix__), X(__miscmansuffix__).
+__xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), Xserver(__appmansuffix__), X(__miscmansuffix__).
 .SH AUTHORS
 Authors include...
  Alan Cox
diff -ruN -x .gitignore -x .cvsignore xf86-input-palmax-17f1dc5d593a3056bee74d53a31e38065b7236d0/src/xf86Palmax.c xf86-input-palmax-5f8c3d97ac2edb3a7ee744369a9826a645b322f9/src/xf86Palmax.c
--- xf86-input-palmax-17f1dc5d593a3056bee74d53a31e38065b7236d0/src/xf86Palmax.c	2008-03-20 20:34:08.000000000 +0000
+++ xf86-input-palmax-5f8c3d97ac2edb3a7ee744369a9826a645b322f9/src/xf86Palmax.c	2009-01-31 04:28:25.000000000 +0000
@@ -68,7 +68,9 @@
 	"xf86CollectInputOptions",
 	"xf86ErrorFVerb",
 	"xf86FindOptionValue",
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
 	"xf86GetMotionEvents",
+#endif
 	"xf86GetVerbosity",
 	"xf86MotionHistoryAllocate",
 	"xf86NameCmp",
@@ -467,7 +469,10 @@
 		 * screen to fit one meter.
 		 */
 	
-		if (InitValuatorClassDeviceStruct(dev, 2, xf86GetMotionEvents,
+		if (InitValuatorClassDeviceStruct(dev, 2,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
+				xf86GetMotionEvents,
+#endif
 				local->history_size, Absolute) == FALSE) {
 			ErrorF("Unable to allocate ValuatorClassDeviceStruct\n");
 			return !Success;
