--- src/terminal.c~	2009-04-17 18:30:45.000000000 +0200
+++ src/terminal.c	2009-05-29 19:47:28.071360678 +0200
@@ -234,9 +234,6 @@
   return g_string_free (name, FALSE);
 }
 
-/* Evil hack alert: this is exported from libgconf-2 but not in a public header */
-extern gboolean gconf_ping_daemon (void);
-         
 int
 main (int argc, char **argv)
 {
@@ -444,16 +441,6 @@
   g_free (argv_copy);
   g_free (factory_name);
 
-  /* If the gconf daemon isn't available (e.g. because there's no dbus
-   * session bus running), we'd crash later on. Tell the user about it
-   * now, and exit. See bug #561663.
-   */
-  if (!gconf_ping_daemon ())
-    {
-      g_printerr ("Failed to contact the GConf daemon; exiting.\n");
-      exit (1);
-    }
-
   gtk_window_set_default_icon_name (GNOME_TERMINAL_ICON_NAME);
 
   gtk_about_dialog_set_url_hook (about_url_hook, NULL, NULL);
