--- gphpedit-0.9.91/src/tab.c.orig	2008-07-08 00:21:04.000000000 +0200
+++ gphpedit-0.9.91/src/tab.c	2008-07-08 00:20:50.000000000 +0200
@@ -562,6 +562,17 @@
 	if (long_filename)
 		return long_filename;
 
+	// For ArchLinux - KISS ;)
+	long_filename = tab_help_try_filename("/usr/share/doc/php/html/function.", command, ".html");
+	if (long_filename)
+		return long_filename;
+	long_filename = tab_help_try_filename("/usr/share/doc/php/html/ref.", command, ".html");
+	if (long_filename)
+		return long_filename;
+	long_filename = tab_help_try_filename("/usr/share/doc/php/html/", command, NULL);
+	if (long_filename)
+		return long_filename;
+
 	// For Gentoo, as much as I love it - it's twatty to put docs in a version specific folder like this!
 	long_filename = tab_help_try_filename("/usr/doc/php-docs-200403/html/function.", command, ".html");
 	if (long_filename)
