--- util/bsetbg.orig	2008-03-26 21:17:04.000000000 -0700
+++ util/bsetbg	2008-03-26 21:21:57.000000000 -0700
@@ -22,7 +22,7 @@
 
 PATH=$PATH:/usr/bin:/usr/local/bin:/usr/X11R6/bin
 
-img_apps="display xli xsetbg Esetroot qiv wmsetbg xv"
+img_apps="display xli xsetbg Esetroot qiv wmsetbg xv feh"
 
 display_full_cmd="display -geometry 800x600 -window root"
 display_tile_cmd="display -window root"
@@ -59,6 +59,11 @@
 xsetbg_center_cmd="xsetbg -center"
 xsetbg_default_cmd="$xsetbg_center_cmd"
 
+feh_full_cmd="feh --bg-scale"
+feh_tile_cmd="feh --bg-tile"
+feh_center_cmd="feh --bg-center"
+feh_default_cmd="$feh_center_cmd"
+
 ##################################
 
 me=`basename $0`
