diff -Nur blackbox-0.70.1/lib/Image.cc blackbox-0.70.1-new/lib/Image.cc
--- blackbox-0.70.1/lib/Image.cc	2005-04-08 08:41:09.000000000 -0700
+++ blackbox-0.70.1-new/lib/Image.cc	2008-05-21 20:52:44.000000000 -0700
@@ -45,6 +45,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 
+#include <cstring>
+
 // #define COLORTABLE_DEBUG
 // #define MITSHM_DEBUG
 
diff -Nur blackbox-0.70.1/lib/Resource.cc blackbox-0.70.1-new/lib/Resource.cc
--- blackbox-0.70.1/lib/Resource.cc	2005-04-06 14:16:50.000000000 -0700
+++ blackbox-0.70.1-new/lib/Resource.cc	2008-05-21 20:52:44.000000000 -0700
@@ -30,6 +30,8 @@
 
 #include <stdio.h>
 
+#include <cstring>
+
 
 bt::Resource::Resource(void)
   : db(NULL)
diff -Nur blackbox-0.70.1/lib/XDG.cc blackbox-0.70.1-new/lib/XDG.cc
--- blackbox-0.70.1/lib/XDG.cc	2005-04-06 07:04:38.000000000 -0700
+++ blackbox-0.70.1-new/lib/XDG.cc	2008-05-21 20:52:44.000000000 -0700
@@ -26,7 +26,7 @@
 #include "XDG.hh"
 
 #include <stdlib.h>
-
+#include <algorithm>
 
 // make sure directory names end with a slash
 static std::string terminateDir(const std::string &string)
diff -Nur blackbox-0.70.1/src/BlackboxResource.cc blackbox-0.70.1-new/src/BlackboxResource.cc
--- blackbox-0.70.1/src/BlackboxResource.cc	2005-10-18 00:34:46.000000000 -0700
+++ blackbox-0.70.1-new/src/BlackboxResource.cc	2008-05-21 20:52:44.000000000 -0700
@@ -33,6 +33,8 @@
 #include <X11/Xutil.h>
 #include <X11/cursorfont.h>
 
+#include <cstring>
+
 
 BlackboxResource::BlackboxResource(const std::string& rc): rc_file(rc) {
   screen_resources = 0;
diff -Nur blackbox-0.70.1/src/Screen.cc blackbox-0.70.1-new/src/Screen.cc
--- blackbox-0.70.1/src/Screen.cc	2005-10-18 01:07:22.000000000 -0700
+++ blackbox-0.70.1-new/src/Screen.cc	2008-05-21 20:52:44.000000000 -0700
@@ -48,6 +48,8 @@
 #include <ctype.h>
 #include <dirent.h>
 
+#include <cstring>
+
 
 static bool running = true;
 static int anotherWMRunning(Display *, XErrorEvent *) {
diff -Nur blackbox-0.70.1/src/ScreenResource.cc blackbox-0.70.1-new/src/ScreenResource.cc
--- blackbox-0.70.1/src/ScreenResource.cc	2005-04-12 22:54:08.000000000 -0700
+++ blackbox-0.70.1-new/src/ScreenResource.cc	2008-05-21 20:52:44.000000000 -0700
@@ -33,6 +33,8 @@
 
 #include <assert.h>
 
+#include <cstring>
+
 
 static const int iconify_width  = 9;
 static const int iconify_height = 9;
diff -Nur blackbox-0.70.1/src/main.cc blackbox-0.70.1-new/src/main.cc
--- blackbox-0.70.1/src/main.cc	2005-01-03 02:42:57.000000000 -0700
+++ blackbox-0.70.1-new/src/main.cc	2008-05-21 20:52:44.000000000 -0700
@@ -36,6 +36,8 @@
 
 #include <stdio.h>
 
+#include <cstring>
+
 
 static void showHelp(int exitval) {
   // print version - this should not be localized!
diff -Nur blackbox-0.70.1/util/bsetbg blackbox-0.70.1-new/util/bsetbg
--- blackbox-0.70.1/util/bsetbg	2004-12-22 02:40:18.000000000 -0700
+++ blackbox-0.70.1-new/util/bsetbg	2008-05-21 20:52:44.000000000 -0700
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 # Copyright (c) 2000-2002 Timothy M. King (tmk@lordzork.com)
 #
diff -Nur blackbox-0.70.1/util/bsetroot.cc blackbox-0.70.1-new/util/bsetroot.cc
--- blackbox-0.70.1/util/bsetroot.cc	2005-03-15 00:01:37.000000000 -0700
+++ blackbox-0.70.1-new/util/bsetroot.cc	2008-05-21 20:52:44.000000000 -0700
@@ -27,11 +27,14 @@
 #include <Pen.hh>
 #include <Texture.hh>
 
+#include <cstdlib>
 #include <cctype>
 
 #include <X11/Xatom.h>
 #include <stdio.h>
 
+#include <cstring>
+
 
 // ignore all X errors
 static int x11_error(::Display *, XErrorEvent *)
