From: Fabian Vogt <fabian@ritter-vogt.de>
Subject: Don't require open264 when proprietary_codecs are supported

Amended on 2024-01-30: also disable h265

diff --git a/src/3rdparty/chromium/third_party/webrtc/webrtc.gni b/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
index 5a1c43c8888..d867f7e5330 100644
--- a/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
+++ b/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
@@ -186,8 +186,7 @@ declare_args() {
   #
   # Enabling H264 when building with MSVC is currently not supported, see
   # bugs.webrtc.org/9213#c13 for more info.
-  rtc_use_h264 =
-      proprietary_codecs && !is_android && !is_ios && !(is_win && !is_clang)
+  rtc_use_h264 = false

   # Enable this flag to make webrtc::Mutex be implemented by absl::Mutex.
   rtc_use_absl_mutex = false
diff --git a/src/core/config/common.pri b/src/core/config/common.pri
index fc46d55ce..27558a500 100644
--- a/src/core/config/common.pri
+++ b/src/core/config/common.pri
@@ -26,9 +26,6 @@ qtConfig(webengine-webrtc) {
 
 qtConfig(webengine-proprietary-codecs) {
     gn_args += proprietary_codecs=true ffmpeg_branding=\"Chrome\"
-    qtConfig(webengine-webrtc) {
-        gn_args += rtc_use_h264=true
-    }
 } else {
     gn_args += proprietary_codecs=false
 }
