--- qtwebengine-everywhere-src-5.13.0/src/core/config/linux.pri.orig	2019-08-06 08:23:45.385072740 +0300
+++ qtwebengine-everywhere-src-5.13.0/src/core/config/linux.pri	2019-08-06 08:23:51.085237082 +0300
@@ -162,7 +162,7 @@ host_build {
     qtConfig(webengine-system-harfbuzz): gn_args += use_system_harfbuzz=true
     !qtConfig(webengine-system-glib): gn_args += use_glib=false
     qtConfig(webengine-pulseaudio) {
-        gn_args += use_pulseaudio=true
+        gn_args += use_pulseaudio=true link_pulseaudio=true
     } else {
         gn_args += use_pulseaudio=false
     }
