--- src/core/gyp_run.pro.orig	2016-12-17 09:08:25.447058042 -0700
+++ src/core/gyp_run.pro	2016-12-17 09:11:34.499737847 -0700
@@ -80,7 +80,7 @@
         # we use arm_neon_optional for ARMv7 and newer and let chromium decide
         # about the mfpu option.
         contains(MFPU, ".*neon.*"): GYP_CONFIG += arm_fpu=\"$$MFPU\" arm_neon=1
-        else:!lessThan(MARMV, 7): GYP_CONFIG += arm_neon=0 arm_neon_optional=1
+        # else:!lessThan(MARMV, 7): GYP_CONFIG += arm_neon=0 arm_neon_optional=1
         else: GYP_CONFIG += arm_fpu=\"$$MFPU\" arm_neon=0 arm_neon_optional=0
     } else {
         # Chromium defaults to arm_neon=1, Qt does not.
