diff -ur qtwebengine-opensource-src-5.6.0-beta/src/core/gyp_run.pro qtwebengine-opensource-src-5.6.0-beta-no-neon/src/core/gyp_run.pro
--- qtwebengine-opensource-src-5.6.0-beta/src/core/gyp_run.pro	2015-12-14 16:27:24.000000000 +0100
+++ qtwebengine-opensource-src-5.6.0-beta-no-neon/src/core/gyp_run.pro	2016-01-10 17:48:45.689954050 +0100
@@ -74,7 +74,10 @@
         # we use arm_neon_optional for ARMv7 and newer and let chromium decide
         # about the mfpu option.
         contains(MFPU, "neon")|contains(MFPU, "neon-vfpv4"): GYP_CONFIG += arm_fpu=\"$$MFPU\" arm_neon=1
-        else:!lessThan(MARMV, 7): GYP_CONFIG += arm_neon=0 arm_neon_optional=1
+        # Disable NEON entirely for now, because it fails to build:
+        # https://bugzilla.redhat.com/show_bug.cgi?id=1282495
+        # (This line was also missing the required arm_fpu flag, which I added.)
+        # else:!lessThan(MARMV, 7): GYP_CONFIG += arm_fpu=\"$$MFPU\" arm_neon=0 arm_neon_optional=1
         else: GYP_CONFIG += arm_fpu=\"$$MFPU\" arm_neon=0 arm_neon_optional=0
     }
 
