https://bugzilla.mozilla.org/show_bug.cgi?id=1526653

diff --git a/js/src/wasm/WasmSignalHandlers.cpp.orig b/js/src/wasm/WasmSignalHandlers.cpp
index 636537f..0f3461a 100644
--- a/js/src/wasm/WasmSignalHandlers.cpp.orig
+++ b/js/src/wasm/WasmSignalHandlers.cpp
@@ -244,7 +244,7 @@ using mozilla::DebugOnly;
 // emulation here.
 
 #if defined(__linux__) && defined(__arm__)
-#  define WASM_EMULATE_ARM_UNALIGNED_FP_ACCESS
+//#  define WASM_EMULATE_ARM_UNALIGNED_FP_ACCESS
 #endif
 
 #ifdef WASM_EMULATE_ARM_UNALIGNED_FP_ACCESS
