--- a/js/moz.configure	2020-07-20 15:02:46.000000000 -0600
+++ b/js/moz.configure	2020-08-08 15:55:33.771501779 -0600
@@ -18,8 +18,6 @@
 option(env='JS_STANDALONE', default=building_js,
        help='Reserved for internal use')
 
-include('../build/moz.configure/rust.configure',
-        when='--enable-compile-environment')
 include('../build/moz.configure/bindgen.configure',
         when='--enable-compile-environment')
 
