--- lxpanel/panel.in	2008-04-20 12:09:34.000000000 +0200
+++ lxpanel/panel.in	2009-04-27 17:48:26.000000000 +0200
@@ -29,7 +29,7 @@
 Plugin {
     type = menu
     Config {
-        image=@prefix@/share/lxpanel/images/my-computer.png
+        image=@prefix@/share/lxde/images/lxde-icon.png
         system {
         }
         separator {
@@ -53,9 +53,6 @@
             id=pcmanfm.desktop
         }
         Button {
-            id=gnome-terminal.desktop
-        }
-        Button {
             id=firefox.desktop
         }
     }
