From b0110d28c675dfc68ae8567c0b1712b187c7e61a Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Thu, 17 May 2018 12:53:53 -0600
Subject: [PATCH 2/2] GCC 8 fix

---
 src/fbdev.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/fbdev.c b/src/fbdev.c
index 23ec855..76ca968 100644
--- a/src/fbdev.c
+++ b/src/fbdev.c
@@ -1093,7 +1093,7 @@ FBDevScreenInit(SCREEN_INIT_ARGS_DECL)
 	fPtr->CloseScreen = pScreen->CloseScreen;
 	pScreen->CloseScreen = FBDevCloseScreen;
 
-#if XV
+#ifdef XV
 	fPtr->SunxiVideo_private = NULL;
 	if (xf86ReturnOptValBool(fPtr->Options, OPTION_XV_OVERLAY, TRUE) &&
 	fPtr->sunxi_disp_private) {
@@ -1183,7 +1183,7 @@ FBDevCloseScreen(CLOSE_SCREEN_ARGS_DECL)
 	    fPtr->SunxiDispHardwareCursor_private = NULL;
 	}
 
-#if XV
+#ifdef XV
 	if (fPtr->SunxiVideo_private) {
 	    SunxiVideo_Close(pScreen);
 	    free(fPtr->SunxiVideo_private);
-- 
2.16.2

