diff --git a/PolymakeInterface/src/polymake_fan.cpp b/PolymakeInterface/src/polymake_fan.cpp
index e3914d7b..9e1b164b 100644
--- a/PolymakeInterface/src/polymake_fan.cpp
+++ b/PolymakeInterface/src/polymake_fan.cpp
@@ -93,7 +93,7 @@ Obj REAL_FAN_BY_RAYS_AND_CONES( Polymake_Data* data, Obj rays, Obj cones ){
           return NULL;
         }
 #endif
-        incMatr[i] += INT_INTOBJ( numb ) - 1;
+        incMatr[i] += (pm::Set<int>)pm::scalar2set(INT_INTOBJ( numb ) - 1);
       }
   }
 
diff --git a/PolymakeInterface/src/polymake_matroid.cpp b/PolymakeInterface/src/polymake_matroid.cpp
index 4f33795b..5ea2e8d2 100644
--- a/PolymakeInterface/src/polymake_matroid.cpp
+++ b/PolymakeInterface/src/polymake_matroid.cpp
@@ -52,7 +52,7 @@ Obj REAL_CREATE_MATROID_ABSTRACT( Polymake_Data* data, Obj size, Obj elements ){
       int current_length = LEN_PLIST( current_basis_list );
       
       for( int i=1;i<=current_length;i++){
-          incMatr[current_basis] += INT_INTOBJ( ELM_PLIST( current_basis_list, i ) ) - 1;
+          incMatr[current_basis] += (pm::Set<int>)pm::scalar2set(INT_INTOBJ( ELM_PLIST( current_basis_list, i ) ) - 1);
       }
       
   }
