--- configure.in	2008-02-26 16:30:05.000000000 -0700
+++ configure.in	2008-09-05 13:36:30.000000000 -0600
@@ -69,6 +69,16 @@
 gtksharp_prefix=`pkg-config --variable=prefix gtk-sharp-2.0`
 AC_SUBST(gtksharp_prefix)
 
+PKG_CHECK_MODULES(GNOMEPRINT_SHARP, gnome-print-sharp-2.18, use_gnomeprint=yes, use_gnomeprint=no)
+if test "x$use_gnomeprint" = "xyes" ; then
+GNOMEPRINT_SHARP_REQUIRES=gnome-print-sharp-2.18
+else
+GNOMEPRINT_SHARP_REQUIRES=
+fi
+AC_SUBST(GNOMEPRINT_SHARP_CFLAGS)
+AC_SUBST(GNOMEPRINT_SHARP_LIBS)
+AC_SUBST(GNOMEPRINT_SHARP_REQUIRES)
+
 PKG_CHECK_MODULES(GTKSOURCEVIEW, gtksourceview-1.0 >= 1.0.0)
 AC_SUBST(GTKSOURCEVIEW_LIBS)
 gtksourceview_prefix=`pkg-config --variable=prefix gtksourceview-1.0`
--- gtksourceview/Makefile.am	2008-02-26 16:30:05.000000000 -0700
+++ gtksourceview/Makefile.am	2008-09-05 13:14:53.000000000 -0600
@@ -2,7 +2,7 @@
 LIBRARY=	$(LIBRARY_NAME).dll
 APIS=		gtksourceview-api.xml
 
-ASSEMBLIES=	/pkg:gnome-sharp-2.0
+ASSEMBLIES=	$(GTK_SHARP_LIBS) $(GNOMEPRINT_SHARP_LIBS)
 
 EXTRA_DIST = $(sources) $(customs) Gtk.metadata gtksourceview-api.raw AssemblyInfo.cs.in
 
@@ -21,7 +21,7 @@
 	$(GAPI_FIXUP) --api=gtksourceview-api.xml --metadata=$(srcdir)/Gtk.metadata
 
 generated-stamp: $(APIS) $(customs) $(sources) Gtk.metadata
-	$(GAPI_CODEGEN) $(GTK_SHARP_CFLAGS) --generate $(APIS) \
+	$(GAPI_CODEGEN) $(GTK_SHARP_CFLAGS) $(GNOMEPRINT_SHARP_CFLAGS) --generate $(APIS) \
 	--outdir=generated --customdir=. --assembly-name=gtksourceview-sharp && touch generated-stamp
 
 sources = \
--- gtksourceview-sharp-2.0.pc.in	2008-02-26 16:30:05.000000000 -0700
+++ gtksourceview-sharp-2.0.pc.in	2008-09-05 13:22:33.000000000 -0600
@@ -6,5 +6,5 @@
 Name: GtkSourceView#
 Description: GtkSourceView# - gtksourceview .NET Binding
 Version: @VERSION@
-Requires: gnome-sharp-2.0
+Requires: gnome-sharp-2.0 @GNOMEPRINT_SHARP_REQUIRES@
 Libs: -r:${prefix}/lib/mono/gtksourceview-sharp-2.0/gtksourceview-sharp.dll
--- sample/Makefile.am	2008-02-26 16:30:05.000000000 -0700
+++ sample/Makefile.am	2008-09-05 13:35:34.000000000 -0600
@@ -10,5 +10,5 @@
 	$(CSC) -out:$@ $(srcdir)/SourceViewTest.cs -r:../gtksourceview/gtksourceview-sharp.dll -pkg:gtk-sharp-2.0
 
 $(PRINT): $(srcdir)/PrintSample.cs
-	$(CSC) -out:$@ $(srcdir)/PrintSample.cs -r:../gtksourceview/gtksourceview-sharp.dll -pkg:gtk-sharp-2.0 -pkg:gnome-sharp-2.0
+	$(CSC) -out:$@ $(srcdir)/PrintSample.cs -r:../gtksourceview/gtksourceview-sharp.dll -pkg:gtk-sharp-2.0 -pkg:gnome-sharp-2.0 $(GNOMEPRINT_SHARP_LIBS)
 
