diff -Nru xawtv-3.95/Makefile.in xawtv-3.95.my/Makefile.in
--- xawtv-3.95/Makefile.in	2004-04-22 09:52:29.000000000 -0500
+++ xawtv-3.95.my/Makefile.in	2006-06-04 19:05:24.000000000 -0500
@@ -49,7 +49,7 @@
 VBI_LIBS	:= @LIBZVBI@
 GL_LIBS		:= @LIBGL@
 DV_LIBS		:= @LIBDV@
-FS_LIBS		:= -L@x_libraries@ @FSLIB@
+FS_LIBS		:= @FSLIB@
 DLFLAGS		:= @DLFLAGS@
 
 # stuff configure has found
diff -Nru xawtv-3.95/common/channel.c xawtv-3.95.my/common/channel.c
--- xawtv-3.95/common/channel.c	2003-03-28 06:22:06.000000000 -0600
+++ xawtv-3.95.my/common/channel.c	2006-06-04 19:10:07.000000000 -0500
@@ -181,6 +181,9 @@
     {  -1, NULL,     },
 };
 
+extern struct STRTAB booltab[];
+extern struct STRTAB captab[];
+
 /* just malloc memory for a new channel ... */
 struct CHANNEL*
 add_channel(char *name)
diff -Nru xawtv-3.95/common/channel.h xawtv-3.95.my/common/channel.h
--- xawtv-3.95/common/channel.h	2003-02-14 08:14:04.000000000 -0600
+++ xawtv-3.95.my/common/channel.h	2006-06-04 19:08:13.000000000 -0500
@@ -90,8 +90,6 @@
 
 /* ----------------------------------------------------------------------- */
 
-extern struct STRTAB booltab[];
-extern struct STRTAB captab[];
 
 int str_to_int(char *str, struct STRTAB *tab);
 const char* int_to_str(int n, struct STRTAB *tab);
diff -Nru xawtv-3.95/common/commands.c xawtv-3.95.my/common/commands.c
--- xawtv-3.95/common/commands.c	2004-04-19 11:24:05.000000000 -0500
+++ xawtv-3.95.my/common/commands.c	2006-06-04 19:11:48.000000000 -0500
@@ -162,6 +162,9 @@
 
 static int keypad_state = -1;
 
+extern struct STRTAB booltab[];
+extern struct STRTAB captab[];
+
 /* ----------------------------------------------------------------------- */
 
 void add_attrs(struct ng_attribute *new)
diff -Nru xawtv-3.95/console/fs.h xawtv-3.95.my/console/fs.h
--- xawtv-3.95/console/fs.h	2003-02-14 08:14:04.000000000 -0600
+++ xawtv-3.95.my/console/fs.h	2006-06-04 19:00:20.000000000 -0500
@@ -1,5 +1,5 @@
 #ifndef X_DISPLAY_MISSING
-# include <FSlib.h>
+# include <X11/fonts/FSlib.h>
 
 struct fs_font {
     Font               font;
