--- old/src/inn-2.4.5/site/Makefile	2008-06-29 19:56:57.000000000 +0200
+++ new/src/inn-2.4.5/site/Makefile	2008-10-09 19:27:59.613208825 +0200
@@ -53,7 +53,7 @@
 
 PATH_ACTIVE		= ${PATHDB}/active
 PATH_ACTIVE_TIMES	= ${PATHDB}/active.times
-PATH_HISTORY		= ${PATHDB}/history
+
 PATH_NEWSGROUPS		= ${PATHDB}/newsgroups
 
 ##  Scripts from above, plus site-specific config files.
@@ -95,7 +95,7 @@
 ALL_INSTALLED	= $(MOST_INSTALLED) $(REST_INSTALLED)
 
 SPECIAL		= $D$(PATH_ACTIVE) $D$(PATH_ACTIVE_TIMES) \
-		$D$(PATH_NEWSGROUPS) $D$(PATH_HISTORY)
+		$D$(PATH_NEWSGROUPS)
 
 ##  Get new versions of everything from samples directory.
 all:		$(P) $(ALL) config
@@ -140,12 +140,12 @@
 	chown $(NEWSUSER) $@
 	chgrp $(NEWSGROUP) $@
 	chmod $(FILEMODE) $@
-$D$(PATH_HISTORY):
-	touch $@
-	chown $(NEWSUSER) $@
-	chgrp $(NEWSGROUP) $@
-	chmod $(FILEMODE) $@
-	$(PATHBIN)/makedbz -i -o
+
+
+
+
+
+
 
 ##  Remove files that are unchanged from the release version.
 clean:
