From 08b46bf4e228377f59d3ac377203fec383ffb825 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Thu, 30 Aug 2018 07:53:58 -0600
Subject: [PATCH] fix finding curl

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 1663b03e..bcd87e86 100644
--- a/meson.build
+++ b/meson.build
@@ -46,7 +46,7 @@ gstreamer_pbutils   = dependency('gstreamer-pbutils-1.0')
 gtk                 = dependency('gtk+-3.0', version: '>=3.22')
 gumbo               = dependency('gumbo')
 json_glib           = dependency('json-glib-1.0')
-libcurl             = c_compiler.find_library('libcurl')
+libcurl             = c_compiler.find_library('curl')
 libnotify           = dependency('libnotify')
 libpeas             = dependency('libpeas-1.0')
 libsecret           = dependency('libsecret-1')
-- 
2.18.0

