--- memory/mozjemalloc/jemalloc.c.orig	2016-03-21 19:00:39.971125884 -0600
+++ memory/mozjemalloc/jemalloc.c	2016-03-21 18:59:22.474081051 -0600
@@ -1089,7 +1089,7 @@
  * controlling the malloc behavior are defined as compile-time constants
  * for best performance and cannot be altered at runtime.
  */
-#if !defined(__ia64__) && !defined(__sparc__) && !defined(__mips__)
+#if !defined(__ia64__) && !defined(__sparc__) && !defined(__mips__) && !defined(__aarch64__)
 #define MALLOC_STATIC_SIZES 1
 #endif
 
