https://bugzilla.mozilla.org/show_bug.cgi?id=1526653

diff --git a/js/src/wasm/WasmSignalHandlers.cpp.orig b/js/src/wasm/WasmSignalHandlers.cpp
index d46b1c6..cf4c957 100644
--- a/js/src/wasm/WasmSignalHandlers.cpp.orig
+++ b/js/src/wasm/WasmSignalHandlers.cpp
@@ -228,7 +228,7 @@ using mozilla::DebugOnly;
 #  error "Don't know how to read/write to the thread state via the mcontext_t."
 #endif
 
-#if defined(__linux__) && defined(__arm__)
+#if 0 && defined(__linux__) && defined(__arm__)
 #  include <sys/user.h>
 #endif
 
@@ -456,7 +456,7 @@ struct AutoHandlingTrap {
   }
 };
 
-#if defined(__linux__) && defined(__arm__)
+#if 0 && defined(__linux__) && defined(__arm__)
 
 // Code to handle SIGBUS for unaligned floating point accesses on 32-bit ARM.
 
