--- b/gfx/ycbcr/ycbcr_to_rgb565.cpp	2011-11-04 21:33:59.000000000 +0000
+++ a/gfx/ycbcr/ycbcr_to_rgb565.cpp	2012-01-18 10:28:15.216874884 +0000
@@ -508,7 +508,7 @@
       source_uv_yoffs_q16 += 1<<(15+y_shift);
       if (yuv_type != YV24) {
         scale_row =
-#  if defined(MOZILLA_MAY_SUPPORT_NEON)
+#  if !defined(MOZILLA_MAY_SUPPORT_NEON)
          supports_neon() ? ScaleYCbCr42xToRGB565_BilinearY_Row_NEON :
 #  endif
          ScaleYCbCr42xToRGB565_BilinearY_Row_C;
@@ -580,7 +580,7 @@
   // Very fast.
   if (width <= 0 || height <= 0)
     return true;
-#  if defined(MOZILLA_MAY_SUPPORT_NEON)
+#  if !defined(MOZILLA_MAY_SUPPORT_NEON)
   if (filter != FILTER_NONE) {
     int source_dx_q16;
     int source_dy_q16;
@@ -647,7 +647,7 @@
   int y_shift;
   x_shift = yuv_type != YV24;
   y_shift = yuv_type == YV12;
-#  ifdef MOZILLA_MAY_SUPPORT_NEON
+#  ifndef MOZILLA_MAY_SUPPORT_NEON
   if (yuv_type != YV24 && supports_neon())
   {
     for (int i = 0; i < pic_height; i++) {
@@ -688,7 +688,7 @@
                                          int pic_height,
                                          YUVType yuv_type)
 {
-#  if defined(MOZILLA_MAY_SUPPORT_NEON)
+#  if !defined(MOZILLA_MAY_SUPPORT_NEON)
   return (yuv_type != YV24 && supports_neon());
 #  else
   return false;
