diff -Naur firefox-114.0.orig/taskcluster/docker/firefox-flatpak/org.mozilla.firefox.appdata.xml.in firefox-114.0/taskcluster/docker/firefox-flatpak/org.mozilla.firefox.appdata.xml.in
--- firefox-114.0.orig/taskcluster/docker/firefox-flatpak/org.mozilla.firefox.appdata.xml.in	2023-06-02 03:15:27.000000000 +0200
+++ firefox-114.0/taskcluster/docker/firefox-flatpak/org.mozilla.firefox.appdata.xml.in	2023-06-06 18:38:46.757054003 +0200
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <application>
   <id type="desktop">org.mozilla.firefox</id>
-  <launchable type="desktop-id">org.mozilla.firefox.desktop</launchable>
+  <launchable type="desktop-id">firefox.desktop</launchable>
   <name>Firefox</name>
   <developer_name>Mozilla</developer_name>
   <summary>Fast, Private &amp; Safe Web Browser</summary>
@@ -10,9 +10,6 @@
   <description>
     <p>When it comes to your life online, you have a choice: accept the factory settings or put your privacy first. When you choose Firefox as your default browser, you’re choosing to protect your data while supporting an independent tech company. Firefox is also the only major browser backed by a non-profit fighting to give you more openness, transparency and control of your life online. Join hundreds of millions of people who choose to protect what's important by choosing Firefox - a web browser designed to be fast, easy to use, customizable and private.</p>
   </description>
-  <releases>
-	<release version="$VERSION" date="$DATE"/>
-  </releases>
   <keywords>
     <keyword>mozilla</keyword>
     <keyword>internet</keyword>
diff -Naur firefox-114.0.orig/taskcluster/docker/firefox-flatpak/org.mozilla.firefox.desktop firefox-114.0/taskcluster/docker/firefox-flatpak/org.mozilla.firefox.desktop
--- firefox-114.0.orig/taskcluster/docker/firefox-flatpak/org.mozilla.firefox.desktop	2023-06-02 03:15:27.000000000 +0200
+++ firefox-114.0/taskcluster/docker/firefox-flatpak/org.mozilla.firefox.desktop	2023-06-06 18:38:46.757054003 +0200
@@ -151,7 +151,7 @@
 Keywords[zh_CN]=Internet;WWW;Browser;Web;Explorer;网页;浏览;上网;火狐;Firefox;ff;互联网;网站;
 Keywords[zh_TW]=Internet;WWW;Browser;Web;Explorer;網際網路;網路;瀏覽器;上網;網頁;火狐
 Exec=firefox %u
-Icon=org.mozilla.firefox
+Icon=firefox
 Terminal=false
 Type=Application
 MimeType=text/html;text/xml;application/xhtml+xml;application/vnd.mozilla.xul+xml;text/mml;x-scheme-handler/http;x-scheme-handler/https;
