diff -up firefox-45.0/firefox-45.0/js/src/jit/AtomicOperations.h.old firefox-45.0/firefox-45.0/js/src/jit/AtomicOperations.h
--- firefox-45.0/firefox-45.0/js/src/jit/AtomicOperations.h.old	2016-03-01 09:45:01.000000000 +0100
+++ firefox-45.0/firefox-45.0/js/src/jit/AtomicOperations.h	2016-03-03 13:13:34.915015391 +0100
@@ -305,7 +305,7 @@ AtomicOperations::isLockfree(int32_t siz
     || defined(__ppc__) || defined(__PPC__)
 # include "jit/none/AtomicOperations-ppc.h"
 #elif defined(JS_CODEGEN_NONE)
-# include "jit/none/AtomicOperations-none.h"
+# include "jit/none/AtomicOperations-ppc.h"
 #elif defined(JS_CODEGEN_X86) || defined(JS_CODEGEN_X64)
 # include "jit/x86-shared/AtomicOperations-x86-shared.h"
 #else
