diff -up firefox-46.0/media/webrtc/trunk/webrtc/build/common.gypi.arm firefox-46.0/media/webrtc/trunk/webrtc/build/common.gypi
--- firefox-46.0/media/webrtc/trunk/webrtc/build/common.gypi.arm	2016-04-25 12:03:12.486027089 +0200
+++ firefox-46.0/media/webrtc/trunk/webrtc/build/common.gypi	2016-04-25 12:05:55.714644873 +0200
@@ -312,20 +312,6 @@
         'defines': [
           'WEBRTC_ARCH_ARM',
         ],
-        'conditions': [
-          ['arm_version>=7', {
-            'defines': ['WEBRTC_ARCH_ARM_V7',
-                        'WEBRTC_BUILD_NEON_LIBS'],
-            'conditions': [
-              ['arm_neon==1', {
-                'defines': ['WEBRTC_ARCH_ARM_NEON',],
-              }],
-              ['arm_neon==0 and arm_neon_optional==1', {
-                'defines': ['WEBRTC_DETECT_ARM_NEON',],
-              }],
-            ],
-          }],
-        ],
       }],
       ['os_bsd==1', {
         'defines': [
