--- configure.orig	2015-08-07 06:32:41.988366320 -0600
+++ configure	2015-08-08 18:23:31.316475879 -0600
@@ -7137,14 +7137,7 @@
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
   if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_SIMD ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_SIMD 1
-EOF
-
-      HAVE_ARM_SIMD=1
+      HAVE_ARM_SIMD=
   fi
 
   echo $ac_n "checking ARM version support in compiler""... $ac_c" 1>&6
@@ -7176,27 +7169,13 @@
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
   if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_NEON 1
-EOF
+      HAVE_ARM_NEON=
 
-      HAVE_ARM_NEON=1
-
-                  if test -n "$ARM_ARCH"; then
+      if test -n "$ARM_ARCH"; then
           if test "$ARM_ARCH" -lt 7; then
               BUILD_ARM_NEON=
           else
-              cat >> confdefs.pytmp <<\EOF
-    (''' BUILD_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define BUILD_ARM_NEON 1
-EOF
-
-              BUILD_ARM_NEON=1
+              BUILD_ARM_NEON=
           fi
       fi
   fi
--- js/src/configure.orig	2015-08-07 06:33:02.108365042 -0600
+++ js/src/configure	2015-08-08 18:24:00.676479200 -0600
@@ -6267,14 +6267,7 @@
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
   if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_SIMD ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_SIMD 1
-EOF
-
-      HAVE_ARM_SIMD=1
+      HAVE_ARM_SIMD=
   fi
 
   echo $ac_n "checking ARM version support in compiler""... $ac_c" 1>&6
@@ -6306,27 +6299,13 @@
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
   if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_NEON 1
-EOF
-
-      HAVE_ARM_NEON=1
+      HAVE_ARM_NEON=
 
                   if test -n "$ARM_ARCH"; then
           if test "$ARM_ARCH" -lt 7; then
               BUILD_ARM_NEON=
           else
-              cat >> confdefs.pytmp <<\EOF
-    (''' BUILD_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define BUILD_ARM_NEON 1
-EOF
-
-              BUILD_ARM_NEON=1
+              BUILD_ARM_NEON=
           fi
       fi
   fi
--- mozglue/build/arm.h.orig	2015-08-08 18:21:03.686458059 -0600
+++ mozglue/build/arm.h	2015-08-08 18:21:16.781459724 -0600
@@ -69,12 +69,6 @@
 #    endif
 #  endif
 
-  // When using -mfpu=neon, gcc generates neon instructions.
-
-#  if defined(__ARM_NEON__)
-#    define MOZILLA_PRESUME_NEON 1
-#  endif
-
   // Currently we only have CPU detection for Linux via /proc/cpuinfo
 #  if defined(__linux__) || defined(ANDROID)
 #    define MOZILLA_ARM_HAVE_CPUID_DETECTION 1
--- build/gyp.mozbuild.orig	2015-08-08 20:00:38.026946903 -0600
+++ build/gyp.mozbuild	2015-08-08 20:00:45.321915167 -0600
@@ -40,7 +40,7 @@
 
      # turn off mandatory use of NEON and instead use NEON detection
     'arm_neon': 0,
-    'arm_neon_optional': 1,
+    'arm_neon_optional': 0,
 
     'moz_widget_toolkit_gonk': 0,
     'moz_webrtc_omx': 0,
