diff -urN a/build/gyp.mozbuild b/build/gyp.mozbuild
--- a/build/gyp.mozbuild	2016-05-12 11:13:12.000000000 -0600
+++ b/build/gyp.mozbuild	2016-06-08 06:53:25.183006599 -0600
@@ -42,7 +42,7 @@
 
      # turn off mandatory use of NEON and instead use NEON detection
     'arm_neon': 0,
-    'arm_neon_optional': 1,
+    'arm_neon_optional': 0,
 
     'moz_widget_toolkit_gonk': 0,
     'moz_webrtc_omx': 0,
diff -urN a/mozglue/build/arm.h b/mozglue/build/arm.h
--- a/mozglue/build/arm.h	2016-05-12 11:13:28.000000000 -0600
+++ b/mozglue/build/arm.h	2016-06-08 06:53:25.183006599 -0600
@@ -69,12 +69,6 @@
 #    endif
 #  endif
 
-  // When using -mfpu=neon, gcc generates neon instructions.
-
-#  if defined(__ARM_NEON__)
-#    define MOZILLA_PRESUME_NEON 1
-#  endif
-
   // Currently we only have CPU detection for Linux via /proc/cpuinfo
 #  if defined(__linux__) || defined(ANDROID)
 #    define MOZILLA_ARM_HAVE_CPUID_DETECTION 1
diff -urN a/old-configure b/old-configure
--- a/old-configure	2016-06-04 20:01:04.000000000 -0600
+++ b/old-configure	2016-06-08 06:53:22.463107876 -0600
@@ -7480,14 +7480,7 @@
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
   if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_SIMD ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_SIMD 1
-EOF
-
-      HAVE_ARM_SIMD=1
+      HAVE_ARM_SIMD=
   fi
 
   echo $ac_n "checking ARM version support in compiler""... $ac_c" 1>&6
@@ -7519,27 +7512,13 @@
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
   if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_NEON 1
-EOF
+      HAVE_ARM_NEON=
 
-      HAVE_ARM_NEON=1
-
-                  if test -n "$ARM_ARCH"; then
+      if test -n "$ARM_ARCH"; then
           if test "$ARM_ARCH" -lt 7; then
               BUILD_ARM_NEON=
           else
-              cat >> confdefs.pytmp <<\EOF
-    (''' BUILD_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define BUILD_ARM_NEON 1
-EOF
-
-              BUILD_ARM_NEON=1
+              BUILD_ARM_NEON=
           fi
       fi
   fi
