@@ -, +, @@ 
---
 layout/base/nsIPresShell.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
--- a/layout/base/nsIPresShell.h	
+++ a/layout/base/nsIPresShell.h	
@@ -224,7 +224,7 @@ public:
    * to the same aSize value.  AllocateFrame returns zero-filled memory.
    * AllocateFrame is infallible and will abort on out-of-memory.
    */
-  void* AllocateFrame(nsQueryFrame::FrameIID aID, size_t aSize)
+  void* __attribute__((optimize("no-lifetime-dse"))) AllocateFrame(nsQueryFrame::FrameIID aID, size_t aSize)
   {
     void* result = mFrameArena.AllocateByFrameID(aID, aSize);
     RecordAlloc(result);
-- 
