diff -wbBur FreeRDP-1.2.0-beta1-android9/channels/client/CMakeLists.txt FreeRDP-1.2.0-beta1-android9.my/channels/client/CMakeLists.txt
--- FreeRDP-1.2.0-beta1-android9/channels/client/CMakeLists.txt	2014-09-12 02:46:32.000000000 +0400
+++ FreeRDP-1.2.0-beta1-android9.my/channels/client/CMakeLists.txt	2015-02-25 14:05:27.840874974 +0300
@@ -24,7 +24,9 @@
 	${CMAKE_CURRENT_SOURCE_DIR}/addin.c
 	${CMAKE_CURRENT_SOURCE_DIR}/addin.h)
 
+if(CHANNEL_STATIC_CLIENT_ENTRIES)
 list(REMOVE_DUPLICATES CHANNEL_STATIC_CLIENT_ENTRIES)
+endif()
 
 foreach(STATIC_ENTRY ${CHANNEL_STATIC_CLIENT_ENTRIES})
 	foreach(STATIC_MODULE ${CHANNEL_STATIC_CLIENT_MODULES})
