diff -wbBur FreeRDP-1.1.0-beta-2013071101.org/channels/tsmf/client/ffmpeg/tsmf_ffmpeg.c FreeRDP-1.1.0-beta-2013071101/channels/tsmf/client/ffmpeg/tsmf_ffmpeg.c
--- FreeRDP-1.1.0-beta-2013071101.org/channels/tsmf/client/ffmpeg/tsmf_ffmpeg.c	2013-07-10 12:00:21.000000000 +0400
+++ FreeRDP-1.1.0-beta-2013071101/channels/tsmf/client/ffmpeg/tsmf_ffmpeg.c	2014-10-14 13:31:27.545141407 +0400
@@ -37,6 +37,8 @@
 #include "tsmf_constants.h"
 #include "tsmf_decoder.h"
 
+#define AVCODEC_MAX_AUDIO_FRAME_SIZE 192000
+
 /* Compatibility with older FFmpeg */
 #if LIBAVUTIL_VERSION_MAJOR < 50
 #define AVMEDIA_TYPE_VIDEO 0
@@ -48,7 +50,7 @@
 	ITSMFDecoder iface;
 
 	int media_type;
-	enum CodecID codec_id;
+	enum AVCodecID codec_id;
 	AVCodecContext* codec_context;
 	AVCodec* codec;
 	AVFrame* frame;
@@ -98,16 +100,6 @@
 	mdecoder->codec_context->channels = media_type->Channels;
 	mdecoder->codec_context->block_align = media_type->BlockAlign;
 
-#ifdef AV_CPU_FLAG_SSE2
-	mdecoder->codec_context->dsp_mask = AV_CPU_FLAG_SSE2 | AV_CPU_FLAG_MMX2;
-#else
-#if LIBAVCODEC_VERSION_MAJOR < 53
-	mdecoder->codec_context->dsp_mask = FF_MM_SSE2 | FF_MM_MMXEXT;
-#else
-	mdecoder->codec_context->dsp_mask = FF_MM_SSE2 | FF_MM_MMX2;
-#endif
-#endif
-
 	return TRUE;
 }
 
