diff -rup ./client/X11.old/xf_event.c ./client/X11/xf_event.c
--- ./client/X11.old/xf_event.c	2014-06-12 15:51:57.000000000 +0200
+++ ./client/X11/xf_event.c	2014-10-22 16:53:53.440644196 +0200
@@ -556,6 +556,7 @@ static BOOL xf_event_FocusOut(xfContext*
 	if (event->xfocus.mode == NotifyWhileGrabbed)
 		XUngrabKeyboard(xfc->display, CurrentTime);
 
+	xf_keyboard_release_all_keypress(xfc);
 	xf_keyboard_clear(xfc);
 
 	if (app)
diff -rup ./client/X11.old/xf_keyboard.c ./client/X11/xf_keyboard.c
--- ./client/X11.old/xf_keyboard.c	2014-06-12 15:51:57.000000000 +0200
+++ ./client/X11/xf_keyboard.c	2014-10-22 16:57:58.403749478 +0200
@@ -169,6 +169,12 @@ void xf_keyboard_release_all_keypress(xf
 		if (xfc->KeyboardState[keycode] != NoSymbol)
 		{
 			rdp_scancode = freerdp_keyboard_get_rdp_scancode_from_x11_keycode(keycode);
+
+			// release tab before releasing the windows key.
+			// this stops the start menu from opening on unfocus event.
+			if (rdp_scancode == RDP_SCANCODE_LWIN)
+				freerdp_input_send_keyboard_event_ex(xfc->instance->input, FALSE, RDP_SCANCODE_TAB);
+
 			freerdp_input_send_keyboard_event_ex(xfc->instance->input, FALSE, rdp_scancode);
 			xfc->KeyboardState[keycode] = NoSymbol;
 		}
