diff -rup FreeRDP-1.0.2-old/client/X11/xf_keyboard.c FreeRDP-1.0.2-new/client/X11/xf_keyboard.c
--- FreeRDP-1.0.2-old/client/X11/xf_keyboard.c	2013-01-02 22:46:59.000000000 +0100
+++ FreeRDP-1.0.2-new/client/X11/xf_keyboard.c	2014-06-24 17:45:44.062728740 +0200
@@ -169,6 +169,9 @@ void xf_kbd_focus_in(xfInfo* xfi)
 	/* on focus in send a tab up like mstsc.exe */
 	input->KeyboardEvent(input, KBD_FLAGS_RELEASE, 0x0F);
 
+	/* release windows key */
+	input->KeyboardEvent(input, KBD_FLAGS_EXTENDED|KBD_FLAGS_RELEASE, 0x5B);
+
 	/* synchronize toggle keys */
 	syncFlags = xf_kbd_get_toggle_keys_state(xfi);
 	input->SynchronizeEvent(input, syncFlags);
