diff -urN nuvexport-0.3/nuvexport nuvexport-0.3-fixed/nuvexport
--- nuvexport-0.3/nuvexport	2006-07-26 03:37:13.000000000 -0400
+++ nuvexport-0.3-fixed/nuvexport	2006-07-27 20:58:41.000000000 -0400
@@ -93,7 +93,7 @@
         push @Exporters, export::ffmpeg::ASF->new;
         push @Exporters, export::ffmpeg::MP3->new;
         push @Exporters, export::ffmpeg::PSP->new;
-        push @Exporters, export::ffmpeg::iPod->new;
+#        push @Exporters, export::ffmpeg::iPod->new;
     }
     elsif ($export_prog eq 'mencoder') {
         push @Exporters, export::mencoder::XviD->new;
