From 83df74ff4a39712b7853270cdc25e404f1b5097f Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Fri, 2 Oct 2020 11:10:28 -0600
Subject: [PATCH] disable neon

---
 src/include/OpenImageIO/simd.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/include/OpenImageIO/simd.h b/src/include/OpenImageIO/simd.h
index f82f8786..dd983580 100644
--- a/src/include/OpenImageIO/simd.h
+++ b/src/include/OpenImageIO/simd.h
@@ -206,7 +206,7 @@
 #  define OIIO_F16C_ENABLED 0
 #endif
 
-#if defined(__ARM_NEON__) && !defined(OIIO_NO_NEON)
+#if 0 && defined(__ARM_NEON__) && !defined(OIIO_NO_NEON)
 #  define OIIO_SIMD 4
 #  define OIIO_SIMD_NEON 1
 #  define OIIO_SIMD_MAX_SIZE_BYTES 16
-- 
2.36.1

