From c82893b59c33348057bf2b1640da110114a1a915 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Mon, 28 Aug 2017 08:42:23 +0200
Subject: [PATCH 1/5] pnm2png: add missing parenthesis

Signed-off-by: Christian Hesse <mail@eworm.de>
---
 contrib/pngminus/pnm2png.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/pngminus/pnm2png.c b/contrib/pngminus/pnm2png.c
index 1b550b8cf..d982c3435 100644
--- a/contrib/pngminus/pnm2png.c
+++ b/contrib/pngminus/pnm2png.c
@@ -376,7 +376,7 @@ BOOL pnm2png (FILE *pnm_file, FILE *png_file, FILE *alpha_file, BOOL interlace,
   /* row_bytes is the width x number of channels x (bit-depth / 8) */
     row_bytes = width * channels * ((bit_depth <= 8) ? 1 : 2);
 
-  if ((row_bytes == 0 || (size_t)height > ((size_t)(-1))/(size_t)row_bytes)
+  if ((row_bytes == 0 || (size_t)height > ((size_t)(-1))/(size_t)row_bytes))
   {
     /* too big */ 
     return FALSE;
From 484f1f94646dbc4ebdef8ef25b18b2d04a9e2f6d Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Mon, 28 Aug 2017 08:43:13 +0200
Subject: [PATCH 2/5] pnm2png: add missing semicolon

Signed-off-by: Christian Hesse <mail@eworm.de>
---
 contrib/pngminus/pnm2png.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/pngminus/pnm2png.c b/contrib/pngminus/pnm2png.c
index d982c3435..0ea9f7ae6 100644
--- a/contrib/pngminus/pnm2png.c
+++ b/contrib/pngminus/pnm2png.c
@@ -513,7 +513,7 @@ BOOL pnm2png (FILE *pnm_file, FILE *png_file, FILE *alpha_file, BOOL interlace,
   if (png_pixels != (unsigned char*) NULL)
     free (png_pixels);
 
-  PNG_UNUSED(raw) /* Quiet a Coverity defect */
+  PNG_UNUSED(raw); /* Quiet a Coverity defect */
 
   return TRUE;
 } /* end of pnm2png */
From ead5448352a4f8815fa4b6c389bc7962b810c1f9 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Mon, 28 Aug 2017 08:59:24 +0200
Subject: [PATCH 3/5] png2pnm: add missing semicolon

Signed-off-by: Christian Hesse <mail@eworm.de>
---
 contrib/pngminus/png2pnm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/pngminus/png2pnm.c b/contrib/pngminus/png2pnm.c
index 39a9f4673..ae6f299cf 100644
--- a/contrib/pngminus/png2pnm.c
+++ b/contrib/pngminus/png2pnm.c
@@ -443,7 +443,7 @@ BOOL png2pnm (FILE *png_file, FILE *pnm_file, FILE *alpha_file,
   if (png_pixels != (unsigned char*) NULL)
     free (png_pixels);
 
-  PNG_UNUSED(raw) /* to quiet a Coverity defect */
+  PNG_UNUSED(raw); /* to quiet a Coverity defect */
   return TRUE;
 
 } /* end of source */
From 7a3ee1fec241b1f27be4f107036043225e4bdf7b Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Mon, 28 Aug 2017 09:04:20 +0200
Subject: [PATCH 4/5] png2pnm: add missing header for PNG_UNUSED()

Signed-off-by: Christian Hesse <mail@eworm.de>
---
 contrib/pngminus/png2pnm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/contrib/pngminus/png2pnm.c b/contrib/pngminus/png2pnm.c
index ae6f299cf..f75c0cbef 100644
--- a/contrib/pngminus/png2pnm.c
+++ b/contrib/pngminus/png2pnm.c
@@ -22,6 +22,7 @@
 #include <fcntl.h>
 #endif
 #include <zlib.h>
+#include <pngpriv.h>
 
 #ifndef BOOL
 #define BOOL unsigned char
From 9d8c4a5b04b4201fa8659a3a76d48a560fa227b4 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Mon, 28 Aug 2017 09:05:05 +0200
Subject: [PATCH 5/5] pnm2png: add missing header for PNG_UNUSED()

Signed-off-by: Christian Hesse <mail@eworm.de>
---
 contrib/pngminus/pnm2png.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/contrib/pngminus/pnm2png.c b/contrib/pngminus/pnm2png.c
index 0ea9f7ae6..5f6cb7f2e 100644
--- a/contrib/pngminus/pnm2png.c
+++ b/contrib/pngminus/pnm2png.c
@@ -23,6 +23,7 @@
 #include <fcntl.h>
 #endif
 #include <zlib.h>
+#include <pngpriv.h>
 
 #ifndef BOOL
 #define BOOL unsigned char
