--- a/dcraw_api.cc	2015-06-16 04:58:38.000000000 +0100
+++ b/dcraw_api.cc	2019-06-30 16:37:05.503409567 +0100
@@ -689,3 +689,3 @@
             #pragma omp parallel for schedule(static) default(none) \
-            shared(h,dark,rgbWB)
+            firstprivate(black,pixels) shared(h,dark,rgbWB)
 #endif
@@ -702,3 +702,3 @@
             #pragma omp parallel for schedule(static) default(none) \
-            shared(h,dark,rgbWB)
+            firstprivate(black,pixels) shared(h,dark,rgbWB)
 #endif
--- a/dcraw_indi.c	2015-06-16 04:58:38.000000000 +0100
+++ b/dcraw_indi.c	2019-06-30 17:03:59.692710441 +0100
@@ -140,2 +140,3 @@
     default(none)					\
+    firstprivate(iheight,iwidth,noise,threshold)	\
     shared(nc,image,size,noise)				\
@@ -145,2 +146,3 @@
     default(none)					\
+    firstprivate(iheight,iwidth,noise,threshold)	\
     shared(nc,image,size)				\
@@ -416,2 +418,3 @@
     default(none)					\
+    firstprivate(colors,filters,height,width)			\
     shared(image,code,prow,pcol,h)			\
@@ -502,3 +502,4 @@
     default(none)					\
-    shared(image,dir,diff)				\
+    firstprivate(filters,height,width)	\
+    shared(image,dir,diff)				\
     private(row,col,i,d,c,pix,guess)
