--- SuiteSparse/SPQR/Lib/Makefile.orig	2016-09-24 09:36:59.588640845 +0000
+++ SuiteSparse/SPQR/Lib/Makefile	2016-09-24 09:37:14.248588339 +0000
@@ -13,7 +13,7 @@
 include ../../SuiteSparse_config/SuiteSparse_config.mk
 
 # SPQR depends on CHOLMOD, AMD, COLAMD, LAPACK, the BLAS and SuiteSparse_config
-LDLIBS += -lamd -lcolamd -lcholmod -lsuitesparseconfig $(LAPACK) $(BLAS)
+LDLIBS += -lamd -lcolamd -lcholmod -lsuitesparseconfig $(LAPACK) $(BLAS) $(TBB)
 
 # compile and install in SuiteSparse/lib
 library:
