--- SuiteSparse/Makefile.orig	2018-07-09 06:14:46.321600511 +0000
+++ SuiteSparse/Makefile	2018-07-09 06:14:56.631628163 +0000
@@ -13,7 +13,7 @@
 go: metis
 	( cd SuiteSparse_config && $(MAKE) )
 	( cd GraphBLAS && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
-	( cd Mongoose  && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
+	( cd Mongoose  && $(MAKE) library CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
 	( cd AMD && $(MAKE) )
 	( cd BTF && $(MAKE) )
 	( cd CAMD && $(MAKE) )
