diff -upr freeradius-server-3.0.25.orig/src/modules/rlm_python/configure freeradius-server-3.0.25/src/modules/rlm_python/configure
--- freeradius-server-3.0.25.orig/src/modules/rlm_python/configure	2021-10-07 22:46:04.000000000 +0300
+++ freeradius-server-3.0.25/src/modules/rlm_python/configure	2021-12-10 18:13:15.196401810 +0200
@@ -2945,7 +2945,7 @@ $as_echo "$as_me: Python sys.prefix \"${
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: Python sys.exec_prefix \"${PY_EXEC_PREFIX}\"" >&5
 $as_echo "$as_me: Python sys.exec_prefix \"${PY_EXEC_PREFIX}\"" >&6;}
 
-		PY_SYS_VERSION=`${PYTHON_BIN} -c 'import sys ; print(sys.version[0:3])'`
+		PY_SYS_VERSION=`${PYTHON_BIN} -c 'import sys ; print(".".join(map(str, sys.version_info[:2])))'`
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: Python sys.version \"${PY_SYS_VERSION}\"" >&5
 $as_echo "$as_me: Python sys.version \"${PY_SYS_VERSION}\"" >&6;}
 
