diff -urN a/qtbase/mkspecs/features/qt_functions.prf b/qtbase/mkspecs/features/qt_functions.prf
--- a/qtbase/mkspecs/features/qt_functions.prf	2013-07-02 01:09:32.000000000 -0600
+++ b/qtbase/mkspecs/features/qt_functions.prf	2013-07-06 14:32:15.335100188 -0600
@@ -256,7 +256,7 @@
         }
         equals(QMAKE_HOST.os, Windows) {
             deppath.name = PATH
-        } else:contains(QMAKE_HOST.os, Linux|FreeBSD) {
+        } else:contains(QMAKE_HOST.os, Linux|FreeBSD|GNU) {
             deppath.name = LD_LIBRARY_PATH
         } else:equals(QMAKE_HOST.os, Darwin) {
             contains(QT_CONFIG, qt_framework): \
