diff -urN a/qtbase/configure b/qtbase/configure
--- a/qtbase/configure	2013-07-02 01:09:32.000000000 -0600
+++ b/qtbase/configure	2013-07-06 14:25:04.356963083 -0600
@@ -2725,11 +2725,6 @@
 else
   XQMAKESPEC="$relpath/mkspecs/${XPLATFORM}"
 fi
-if [ "$PLATFORM" != "$XPLATFORM" ]; then
-    QT_CROSS_COMPILE=yes
-    QMAKE_CONFIG="$QMAKE_CONFIG cross_compile"
-    QTCONFIG_CONFIG="$QTCONFIG_CONFIG cross_compile"
-fi
 
 if [ "$BUILD_ON_MAC" = "yes" ]; then
    if [ `basename $QMAKESPEC` = "macx-xcode" ] || [ `basename $XQMAKESPEC` = "macx-xcode" ]; then
@@ -2997,11 +2992,6 @@
     if [ "$CFG_DEV" = "yes" ]; then
         CFG_BUILD_PARTS="$CFG_BUILD_PARTS tests"
     fi
-
-    # don't build tools by default when cross-compiling
-    if [ "$PLATFORM" != "$XPLATFORM" ]; then
-	CFG_BUILD_PARTS=`echo "$CFG_BUILD_PARTS" | sed "s, tools,,g"`
-    fi
 fi
 for nobuild in $CFG_NOBUILD_PARTS; do
     CFG_BUILD_PARTS=`echo "$CFG_BUILD_PARTS" | sed "s, $nobuild,,g"`
diff -urN a/qtbase/mkspecs/devices/common/linux_device_post.conf b/qtbase/mkspecs/devices/common/linux_device_post.conf
--- a/qtbase/mkspecs/devices/common/linux_device_post.conf	2013-07-02 01:09:32.000000000 -0600
+++ b/qtbase/mkspecs/devices/common/linux_device_post.conf	2013-07-06 14:25:04.356963083 -0600
@@ -1,15 +1,3 @@
-contains(DISTRO_OPTS, deb-multi-arch) {
-    QMAKE_LFLAGS   += -Wl,-rpath-link,$$[QT_SYSROOT]/usr/lib/$${GCC_MACHINE_DUMP} \
-                      -Wl,-rpath-link,$$[QT_SYSROOT]/lib/$${GCC_MACHINE_DUMP}
-}
-
-contains(DISTRO_OPTS, hard-float) {
-    COMPILER_FLAGS += -mfloat-abi=hard
-} else {
-    COMPILER_FLAGS += -mfloat-abi=softfp
-}
-
 QMAKE_CFLAGS       += $$COMPILER_FLAGS
 QMAKE_CXXFLAGS     += $$COMPILER_FLAGS
 
-deviceSanityCheckCompiler()
diff -urN a/qtbase/mkspecs/devices/common/linux_device_pre.conf b/qtbase/mkspecs/devices/common/linux_device_pre.conf
--- a/qtbase/mkspecs/devices/common/linux_device_pre.conf	2013-07-02 01:09:32.000000000 -0600
+++ b/qtbase/mkspecs/devices/common/linux_device_pre.conf	2013-07-06 14:25:04.356963083 -0600
@@ -11,14 +11,3 @@
 !load(device_config) {
     error(Could not successfully load device configuration)
 }
-
-# modifications to g++-unix.conf
-QMAKE_CC                = $${CROSS_COMPILE}gcc
-QMAKE_CXX               = $${CROSS_COMPILE}g++
-QMAKE_LINK              = $${QMAKE_CXX}
-QMAKE_LINK_SHLIB        = $${QMAKE_CXX}
-
-# modifications to linux.conf
-QMAKE_AR                = $${CROSS_COMPILE}ar cqs
-QMAKE_OBJCOPY           = $${CROSS_COMPILE}objcopy
-QMAKE_STRIP             = $${CROSS_COMPILE}strip
