--- qt-everywhere-opensource-src-5.3.1/qtbase/configure.orig	2014-07-19 13:59:34.000000000 +0200
+++ qt-everywhere-opensource-src-5.3.1/qtbase/configure	2014-07-19 13:53:37.000000000 +0200
@@ -4801,16 +4801,6 @@
     fi
 elif [ "$CFG_OPENGL" = "es2" ]; then
     #OpenGL ES 2.x
-    if [ -n "$PKG_CONFIG" ] && $PKG_CONFIG --exists glesv2 2>/dev/null; then
-        QMAKE_INCDIR_OPENGL_ES2=`$PKG_CONFIG --cflags-only-I glesv2 2>/dev/null | sed -e 's,^-I,,g' -e 's, -I, ,g'`
-        QMAKE_LIBDIR_OPENGL_ES2=`$PKG_CONFIG --libs-only-L glesv2 2>/dev/null | sed -e 's,^-L,,g' -e 's, -L, ,g'`
-        QMAKE_LIBS_OPENGL_ES2=`$PKG_CONFIG --libs glesv2 2>/dev/null`
-        QMAKE_CFLAGS_OPENGL_ES2=`$PKG_CONFIG --cflags glesv2 2>/dev/null`
-        QMakeVar set QMAKE_INCDIR_OPENGL_ES2 "`shellArgumentListToQMakeList "$QMAKE_INCDIR_OPENGL_ES2"`"
-        QMakeVar set QMAKE_LIBDIR_OPENGL_ES2 "`shellArgumentListToQMakeList "$QMAKE_LIBDIR_OPENGL_ES2"`"
-        QMakeVar set QMAKE_LIBS_OPENGL_ES2 "`shellArgumentListToQMakeList "$QMAKE_LIBS_OPENGL_ES2"`"
-    fi
-
     compileTest unix/opengles2 "OpenGL ES 2.x" $QMAKE_LIBS_OPENGL_ES2 $QMAKE_CFLAGS_OPENGL_ES2
     if [ $? != "0" ]; then
         echo "The OpenGL ES 2.0 functionality test failed!"
@@ -5130,14 +5120,6 @@
         exit 101
     fi
 
-    if [ -n "$PKG_CONFIG" ] && $PKG_CONFIG --exists egl 2>/dev/null; then
-        QMAKE_INCDIR_EGL=`$PKG_CONFIG --cflags-only-I egl 2>/dev/null | sed -e 's,^-I,,g' -e 's, -I, ,g'`
-        QMAKE_LIBS_EGL=`$PKG_CONFIG --libs egl 2>/dev/null`
-        QMAKE_CFLAGS_EGL=`$PKG_CONFIG --cflags egl 2>/dev/null`
-        QMakeVar set QMAKE_INCDIR_EGL "$QMAKE_INCDIR_EGL"
-        QMakeVar set QMAKE_LIBS_EGL "$QMAKE_LIBS_EGL"
-        QMakeVar set QMAKE_CFLAGS_EGL "`echo " $QMAKE_CFLAGS_EGL " | sed -e 's, -I[^ ]* , ,g;s,^ ,,;s, $,,'`"
-    fi       # detect EGL support
     if compileTest qpa/egl "EGL" $QMAKE_CFLAGS_EGL $QMAKE_LIBS_EGL; then
         CFG_EGL=yes
         if compileTest qpa/egl-x11 "EGL-X11" $QMAKE_CFLAGS_EGL $QMAKE_LIBS_EGL; then
--- qt-everywhere-opensource-src-5.3.1/qtbase/src/gui/Qt5GuiConfigExtras.cmake.in.orig	2014-07-29 19:23:45.000000000 +0200
+++ qt-everywhere-opensource-src-5.3.1/qtbase/src/gui/Qt5GuiConfigExtras.cmake.in	2014-07-29 19:41:12.000000000 +0200
@@ -96,8 +96,8 @@
         string(REGEX REPLACE "[^_A-Za-z0-9]" "_" _cmake_lib_name ${_lib})
         if (NOT TARGET Qt5::Gui_${_cmake_lib_name} AND NOT _Qt5Gui_${_cmake_lib_name}_LIBRARY_DONE)
             find_library(Qt5Gui_${_cmake_lib_name}_LIBRARY ${_lib}
+                HINTS \"${LibDir}\"
 !!IF !isEmpty(CROSS_COMPILE)
-                PATHS \"${LibDir}\"
 !!IF !mac
                 NO_DEFAULT_PATH
 !!ENDIF
