diff -urN a/qtbase/src/gui/painting/qpagedpaintdevice.h b/qtbase/src/gui/painting/qpagedpaintdevice.h
--- a/qtbase/src/gui/painting/qpagedpaintdevice.h	2013-07-02 01:09:54.000000000 -0600
+++ b/qtbase/src/gui/painting/qpagedpaintdevice.h	2013-07-06 14:30:35.874606990 -0600
@@ -44,6 +44,10 @@
 
 #include <QtGui/qpaintdevice.h>
 
+#if defined(B0)
+#undef B0 // Terminal hang-up.  We assume that you do not want that.
+#endif
+
 QT_BEGIN_NAMESPACE
 
 #if defined(B0)
