--- qtbase-opensource-src-5.0.1+dfsg.orig/src/gui/painting/qpagedpaintdevice.h	2013-01-29 12:03:00.000000000 -0700
+++ qtbase-opensource-src-5.0.1+dfsg/src/gui/painting/qpagedpaintdevice.h	2013-02-18 13:21:01.000000000 -0700
@@ -44,6 +44,10 @@
 
 #include <QtGui/qpaintdevice.h>
 
+#if defined(B0)
+#undef B0 // Terminal hang-up.  We assume that you do not want that.
+#endif
+
 QT_BEGIN_HEADER
 
 QT_BEGIN_NAMESPACE
