--- qt-everywhere-opensource-src-5.2.0-rc1/qtwebkit/Source/WebCore/DerivedSources.pri~	2013-11-29 16:57:47.573721157 +0000
+++ qt-everywhere-opensource-src-5.2.0-rc1/qtwebkit/Source/WebCore/DerivedSources.pri	2013-11-29 16:58:22.270157823 +0000
@@ -819,7 +819,7 @@
 inspectorValidate.output = InspectorProtocolVersion.h
 inspectorValidate.input = INSPECTOR_JSON
 inspectorValidate.script = $$PWD/inspector/generate-inspector-protocol-version
-inspectorValidate.commands = python $$inspectorValidate.script -o ${QMAKE_FILE_OUT} ${QMAKE_FILE_IN}
+inspectorValidate.commands = python2 $$inspectorValidate.script -o ${QMAKE_FILE_OUT} ${QMAKE_FILE_IN}
 inspectorValidate.depends = $$PWD/inspector/generate-inspector-protocol-version
 inspectorValidate.add_output_to_sources = false
 GENERATORS += inspectorValidate
@@ -827,7 +827,7 @@
 inspectorJSON.output = InspectorFrontend.cpp InspectorBackendDispatcher.cpp InspectorTypeBuilder.cpp
 inspectorJSON.input = INSPECTOR_JSON
 inspectorJSON.script = $$PWD/inspector/CodeGeneratorInspector.py
-inspectorJSON.commands = python $$inspectorJSON.script $$PWD/inspector/Inspector.json --output_h_dir ${QMAKE_FUNC_FILE_OUT_PATH} --output_cpp_dir ${QMAKE_FUNC_FILE_OUT_PATH}
+inspectorJSON.commands = python2 $$inspectorJSON.script $$PWD/inspector/Inspector.json --output_h_dir ${QMAKE_FUNC_FILE_OUT_PATH} --output_cpp_dir ${QMAKE_FUNC_FILE_OUT_PATH}
 inspectorJSON.depends = $$inspectorJSON.script
 GENERATORS += inspectorJSON
 
@@ -927,7 +927,7 @@
 entities.output = HTMLEntityTable.cpp
 entities.input = HTML_ENTITIES
 entities.script = $$PWD/html/parser/create-html-entity-table
-entities.commands = python $$entities.script -o ${QMAKE_FILE_OUT} $$HTML_ENTITIES
+entities.commands = python2 $$entities.script -o ${QMAKE_FILE_OUT} $$HTML_ENTITIES
 entities.clean = ${QMAKE_FILE_OUT}
 entities.depends = $$PWD/html/parser/create-html-entity-table
 GENERATORS += entities
--- qt-everywhere-opensource-src-5.2.0-rc1/qtwebkit/Source/JavaScriptCore/DerivedSources.pri~	2013-11-29 16:58:05.470269189 +0000
+++ qt-everywhere-opensource-src-5.2.0-rc1/qtwebkit/Source/JavaScriptCore/DerivedSources.pri	2013-11-29 16:58:44.276678608 +0000
@@ -83,14 +83,14 @@
 retgen.output = RegExpJitTables.h
 retgen.script = $$PWD/create_regex_tables
 retgen.input = retgen.script
-retgen.commands = python $$retgen.script > ${QMAKE_FILE_OUT}
+retgen.commands = python2 $$retgen.script > ${QMAKE_FILE_OUT}
 GENERATORS += retgen
 
 #GENERATOR: "KeywordLookup.h": decision tree used by the lexer
 klgen.output = KeywordLookup.h
 klgen.script = $$PWD/KeywordLookupGenerator.py
 klgen.input = KEYWORDLUT_FILES
-klgen.commands = python $$klgen.script ${QMAKE_FILE_NAME} > ${QMAKE_FILE_OUT}
+klgen.commands = python2 $$klgen.script ${QMAKE_FILE_NAME} > ${QMAKE_FILE_OUT}
 GENERATORS += klgen
 
 EXTRACTOR_BINARY = LLIntOffsetsExtractor$$BIN_EXTENSION
@@ -114,7 +114,7 @@
     disassembler.input = DISASSEMBLER_FILES
     disassembler.script = $$PWD/disassembler/udis86/itab.py
     disassembler.depends = $$DISASSEMBLER_DEPENDENCY
-    disassembler.commands = python $$disassembler.script ${QMAKE_FILE_NAME} --outputDir ${QMAKE_FUNC_FILE_OUT_PATH}
+    disassembler.commands = python2 $$disassembler.script ${QMAKE_FILE_NAME} --outputDir ${QMAKE_FUNC_FILE_OUT_PATH}
     disassembler.CONFIG += no_link
     GENERATORS += disassembler
 }
--- qt-everywhere-opensource-src-5.2.0-rc1/qtwebkit/Tools/qmake/mkspecs/features/configure.prf~	2013-11-29 16:59:04.859875494 +0000
+++ qt-everywhere-opensource-src-5.2.0-rc1/qtwebkit/Tools/qmake/mkspecs/features/configure.prf	2013-11-29 16:59:40.539638968 +0000
@@ -116,7 +116,7 @@
     production_build:blackberry|qnx {
         addReasonForSkippingBuild("Build not supported on BB10/QNX yet.")
     }
-    requiredPrograms = gperf python perl bison ruby flex
+    requiredPrograms = gperf python2 perl bison ruby flex
     for(program, requiredPrograms): \
         !programExistsInPath($$program): \
             addReasonForSkippingBuild("Missing $$program from PATH")
--- qt-everywhere-opensource-src-5.2.0-rc1/qtdeclarative/src/3rdparty/masm/masm.pri~	2013-11-29 17:03:43.228028589 +0000
+++ qt-everywhere-opensource-src-5.2.0-rc1/qtdeclarative/src/3rdparty/masm/masm.pri	2013-11-29 17:03:57.537933557 +0000
@@ -47,7 +47,7 @@
     udis86.output = udis86_itab.h
     udis86.input = ITAB
     udis86.CONFIG += no_link
-    udis86.commands = python $$PWD/disassembler/udis86/itab.py ${QMAKE_FILE_IN}
+    udis86.commands = python2 $$PWD/disassembler/udis86/itab.py ${QMAKE_FILE_IN}
     QMAKE_EXTRA_COMPILERS += udis86
 
     udis86_tab_cfile.target = $$OUT_PWD/udis86_itab.c
@@ -67,7 +67,7 @@
 retgen.script = $$PWD/create_regex_tables
 retgen.input = retgen.script
 retgen.CONFIG += no_link
-retgen.commands = python $$retgen.script > ${QMAKE_FILE_OUT}
+retgen.commands = python2 $$retgen.script > ${QMAKE_FILE_OUT}
 QMAKE_EXTRA_COMPILERS += retgen
 
 # Taken from WebKit/Tools/qmake/mkspecs/features/unix/default_post.prf
