--- a/tools/depends/target/ffmpeg/CMakeLists.txt	2019-07-18 11:05:21.000000000 -0400
+++ b/tools/depends/target/ffmpeg/CMakeLists.txt	2019-07-25 16:22:19.506991490 -0400
@@ -19,7 +19,7 @@ if(CROSSCOMPILING)
 endif()
 
 #if(CORE_PLATFORM_NAME STREQUAL rbpi)
-  string(CONCAT CMAKE_C_FLAGS ${CMAKE_C_FLAGS} " -I/opt/vc/include -I/opt/vc/include/interface/vcos/pthreads -I/opt/vc/include/interface/vmcs_host/linux")
+  string(CONCAT CMAKE_C_FLAGS ${CMAKE_C_FLAGS} " -I/build/kodi-rbp4/src/opt/vc/include -I/build/kodi-rbp4/src/opt/vc/include/interface/vcos/pthreads -I/build/kodi-rbp4/src/opt/vc/include/interface/vmcs_host/linux")
   list(APPEND ffmpeg_conf --enable-rpi --disable-ffmpeg --disable-ffprobe)
 #endif()
 
@@ -31,6 +31,8 @@ if(CMAKE_CXX_FLAGS)
   list(APPEND ffmpeg_conf --extra-cxxflags=${CMAKE_CXX_FLAGS})
 endif()
 
+string(CONCAT CMAKE_EXE_LINKER_FLAGS ${CMAKE_EXE_LINKER_FLAGS} " -L/build/kodi-rbp4/src/opt/vc/lib")
+
 if(CMAKE_EXE_LINKER_FLAGS)
   list(APPEND ffmpeg_conf --extra-ldflags=${CMAKE_EXE_LINKER_FLAGS})
 endif()
