--- openvpn-admin-1.9.2.orig/src/Configuration.cs	2005-11-22 04:46:04.000000000 -0200
+++ openvpn-admin-1.9.2/src/Configuration.cs	2006-02-26 02:16:06.225286712 -0300
@@ -29,6 +29,7 @@
 using System.Collections.Specialized;
 using System.Text.RegularExpressions;
 using Mono.Unix;
+using Mono.Unix.Native;
 
 namespace OpenVPN.Admin
 {
@@ -460,7 +461,7 @@
 			
 			// Only for Mono after 1.9.1 releases.
 			//Mono.Unix.Native.Syscall.kill(pid, Mono.Unix.Native.Signum.SIGKILL);
-			Mono.Unix.Syscall.kill(pid, Mono.Unix.Signum.SIGKILL);
+			Mono.Unix.Native.Syscall.kill(pid, Mono.Unix.Native.Signum.SIGKILL);
 
 			File.Delete(pidfile);
 			File.Delete(stafile);
