diff -rud slmodem-2.9.11-20060727-orig/modem/modem_main.c slmodem-2.9.11-20060727/modem/modem_main.c
--- slmodem-2.9.11-20060727-orig/modem/modem_main.c	2005-09-22 10:05:27.000000000 -0300
+++ slmodem-2.9.11-20060727/modem/modem_main.c	2007-07-10 14:37:33.000000000 -0300
@@ -379,7 +379,7 @@
 		ERR("cannot set periods for %s: %s\n", stream_name, snd_strerror(err));
 		return err;
 	}
-	if ( rsize != size ) {
+	if ( rsize < size ) {
 		ERR("period size %ld is not supported by %s (%ld).\n",
 		    size, stream_name, rsize);
 		return -1;		
