diff -ura sshfs-3.3.0.orig/meson.build sshfs-3.3.0.new/meson.build
--- sshfs-3.3.0.orig/meson.build	2017-09-20 14:08:27.000000000 -0300
+++ sshfs-3.3.0.new/meson.build	2017-09-25 11:19:19.946623793 -0300
@@ -58,7 +58,7 @@
 if rst2man.found()
     custom_target('manpages', input: [ 'sshfs.rst' ], output: [ 'sshfs.1' ],
                   command: [rst2man, '@INPUT@', '@OUTPUT@'], install: true,
-                  install_dir: join_paths(get_option('mandir'), '1'))
+                  install_dir: join_paths(get_option('mandir'), 'man1'))
 else
     message('rst2man not found, not building manual page.')
 endif
