diff -ura sshfs-3.3.0.orig/sshfs.c sshfs-3.3.0.new/sshfs.c
--- sshfs-3.3.0.orig/sshfs.c	2017-09-20 14:08:27.000000000 -0300
+++ sshfs-3.3.0.new/sshfs.c	2017-10-01 23:04:23.048755162 -0300
@@ -428,9 +428,15 @@
 	FUSE_OPT_KEY("writeback_cache=no", FUSE_OPT_KEY_DISCARD),
 	FUSE_OPT_KEY("unreliable_append", FUSE_OPT_KEY_DISCARD),
 
-	
-	FUSE_OPT_END
+	/* These may come in from /etc/fstab - we just ignore them */
+	FUSE_OPT_KEY("auto", FUSE_OPT_KEY_DISCARD),
+	FUSE_OPT_KEY("noauto", FUSE_OPT_KEY_DISCARD),
+	FUSE_OPT_KEY("user", FUSE_OPT_KEY_DISCARD),
+	FUSE_OPT_KEY("nouser", FUSE_OPT_KEY_DISCARD),
+	FUSE_OPT_KEY("users", FUSE_OPT_KEY_DISCARD),
+	FUSE_OPT_KEY("_netdev", FUSE_OPT_KEY_DISCARD),
 
+	FUSE_OPT_END
 };
 
 static struct fuse_opt workaround_opts[] = {
