From 37b325852f4d8256de9031170a3bef44bf86c7d6 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Thu, 17 May 2018 12:53:53 -0600
Subject: [PATCH 2/2] GCC 8 fix

---
 src/fbdev.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fbdev.c b/src/fbdev.c
index ee9bc46..ac5325f 100644
--- a/src/fbdev.c
+++ b/src/fbdev.c
@@ -1085,7 +1085,7 @@ FBDevScreenInit(SCREEN_INIT_ARGS_DECL)
 	fPtr->CloseScreen = pScreen->CloseScreen;
 	pScreen->CloseScreen = FBDevCloseScreen;
 
-#if XV
+#ifdef XV
 	XF86VideoAdaptorPtr *ptr;
 	int n = xf86XVListGenericAdaptors(pScrn,&ptr);
 	if (n) {
-- 
2.17.0

