From 39e711fbef271064e50cc5b704f159b46a34bff7 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 6 Oct 2019 15:30:56 -0600
Subject: [PATCH] Fix undeclared value in dbus/sigsegv.c

---
 dbus/sigsegv.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/dbus/sigsegv.c b/dbus/sigsegv.c
index de316429..75beea9d 100644
--- a/dbus/sigsegv.c
+++ b/dbus/sigsegv.c
@@ -64,9 +64,7 @@ static void signal_segv(int signum, siginfo_t* info, void*ptr) {
     const char *si_code_str;
     ucontext_t *ucontext = (ucontext_t*)ptr;
 
-#if defined(HAVE_UCONTEXT) && defined(HAVE_NGREG)
     size_t i;
-#endif
 #if defined(SIGSEGV_STACK_X86) || defined(SIGSEGV_STACK_IA64)
     int f = 0;
     Dl_info dlinfo;
-- 
2.23.0

