diff -urN a/common/jack/systemdeps.h b/common/jack/systemdeps.h
--- a/common/jack/systemdeps.h	2014-07-18 06:28:12.000000000 -0600
+++ b/common/jack/systemdeps.h	2017-02-26 20:45:07.087664517 -0700
@@ -120,9 +120,9 @@
 
 #endif /* __APPLE__ || __linux__ || __sun__ || sun */
 
-#if defined(__arm__)
+#if defined(__arm__) || defined(__aarch64__)
     #undef POST_PACKED_STRUCTURE
     #define POST_PACKED_STRUCTURE
-#endif /* __arm__ */
+#endif /* __arm__ || __aarch64__ */
 
 #endif /* __jack_systemdeps_h__ */
