diff -urN a/common/jack/systemdeps.h b/common/jack/systemdeps.h
--- a/common/jack/systemdeps.h	2014-07-18 06:28:12.000000000 -0600
+++ b/common/jack/systemdeps.h	2014-08-24 12:21:17.163543628 -0600
@@ -123,6 +123,8 @@
 #if defined(__arm__)
     #undef POST_PACKED_STRUCTURE
     #define POST_PACKED_STRUCTURE
+    #undef PRE_PACKED_STRUCTURE
+    #define PRE_PACKED_STRUCTURE
 #endif /* __arm__ */
 
 #endif /* __jack_systemdeps_h__ */
