diff --git a/common/jack/types.h b/common/jack/types.h
index 094d407..2dccf34 100644
--- a/common/jack/types.h
+++ b/common/jack/types.h
@@ -403,10 +403,8 @@ typedef void (*JackPortConnectCallback)(jack_port_id_t a, jack_port_id_t b, int
  * @param port the port that has been renamed
  * @param new_name the new name
  * @param arg pointer to a client supplied structure
- *
- * @return zero on success, non-zero on error
  */
-typedef int (*JackPortRenameCallback)(jack_port_id_t port, const char* old_name, const char* new_name, void *arg);
+typedef void (*JackPortRenameCallback)(jack_port_id_t port, const char* old_name, const char* new_name, void *arg);
 
 /**
  * Prototype for the client supplied function that is called
diff --git a/tests/test.cpp b/tests/test.cpp
index 750d152..8a8a811 100644
--- a/tests/test.cpp
+++ b/tests/test.cpp
@@ -186,11 +186,10 @@ void Jack_Client_Registration_Callback(const char* name, int val, void *arg)
 		client_register--;
 }
 
-int Jack_Port_Rename_Callback(jack_port_id_t port, const char* old_name, const char* new_name, void *arg)
+void Jack_Port_Rename_Callback(jack_port_id_t port, const char* old_name, const char* new_name, void *arg)
 {
      Log("Rename callback has been successfully called with old_name '%s' and new_name '%s'. (msg from callback)\n", old_name, new_name);
      port_rename_clbk = 1;
-     return 0;
 }
 
 int Jack_Update_Buffer_Size(jack_nframes_t nframes, void *arg)
