--- torrentsettings.py	2005-04-05 00:30:46.703711088 +0200
+++ torrentsettings.py.new	2005-04-05 00:33:45.610513120 +0200
@@ -34,12 +34,12 @@
         TorrentSettingsLayout.addMultiCellLayout(layout56,3,3,0,1)
 
         self.listBox1 = QListBox(self,"listBox1")
-        self.listBox1.setSizePolicy(QSizePolicy(1,7,0,0,self.listBox1.sizePolicy().hasHeightForWidth()))
+        self.listBox1.setSizePolicy(QSizePolicy(QSizePolicy.Minimum,QSizePolicy.Expanding,0,0,self.listBox1.sizePolicy().hasHeightForWidth()))
 
         TorrentSettingsLayout.addMultiCellWidget(self.listBox1,0,2,0,0)
 
         self.widgetStack1 = QWidgetStack(self,"widgetStack1")
-        self.widgetStack1.setSizePolicy(QSizePolicy(7,5,0,0,self.widgetStack1.sizePolicy().hasHeightForWidth()))
+        self.widgetStack1.setSizePolicy(QSizePolicy(QSizePolicy.Expanding,QSizePolicy.Preferred,0,0,self.widgetStack1.sizePolicy().hasHeightForWidth()))
 
         self.WStackPage = QWidget(self.widgetStack1,"WStackPage")
         WStackPageLayout = QGridLayout(self.WStackPage,1,1,11,6,"WStackPageLayout")
@@ -53,12 +53,12 @@
         WStackPageLayout.addWidget(self.textLabel1,0,0)
 
         self.WBind = QLineEdit(self.WStackPage,"WBind")
-        self.WBind.setSizePolicy(QSizePolicy(5,0,0,0,self.WBind.sizePolicy().hasHeightForWidth()))
+        self.WBind.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Fixed,0,0,self.WBind.sizePolicy().hasHeightForWidth()))
 
         WStackPageLayout.addWidget(self.WBind,0,1)
 
         self.WIP = QLineEdit(self.WStackPage,"WIP")
-        self.WIP.setSizePolicy(QSizePolicy(5,0,0,0,self.WIP.sizePolicy().hasHeightForWidth()))
+        self.WIP.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Fixed,0,0,self.WIP.sizePolicy().hasHeightForWidth()))
 
         WStackPageLayout.addWidget(self.WIP,1,1)
 
@@ -79,14 +79,14 @@
         WStackPageLayout.addWidget(self.textLabel4,3,0)
 
         self.WMinPort = QSpinBox(self.WStackPage,"WMinPort")
-        self.WMinPort.setSizePolicy(QSizePolicy(5,0,0,0,self.WMinPort.sizePolicy().hasHeightForWidth()))
+        self.WMinPort.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Fixed,0,0,self.WMinPort.sizePolicy().hasHeightForWidth()))
         self.WMinPort.setMaxValue(65535)
         self.WMinPort.setMinValue(1)
 
         WStackPageLayout.addWidget(self.WMinPort,2,1)
 
         self.WMaxPort = QSpinBox(self.WStackPage,"WMaxPort")
-        self.WMaxPort.setSizePolicy(QSizePolicy(5,0,0,0,self.WMaxPort.sizePolicy().hasHeightForWidth()))
+        self.WMaxPort.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Fixed,0,0,self.WMaxPort.sizePolicy().hasHeightForWidth()))
         self.WMaxPort.setMaxValue(65535)
         self.WMaxPort.setMinValue(1)
 
