diff -wbBur qtorrent-0.9.5/pyqtorrent/torrentmain.py qtorrent-0.9.5.my/pyqtorrent/torrentmain.py
--- qtorrent-0.9.5/pyqtorrent/torrentmain.py	2004-02-16 01:00:37.000000000 +0300
+++ qtorrent-0.9.5.my/pyqtorrent/torrentmain.py	2007-03-05 01:30:17.000000000 +0300
@@ -444,7 +444,7 @@
             torrent.start()
     
     def openFile(self):
-        fn = QFileDialog.getOpenFileName(None, "*.torrent")
+        fn = QFileDialog.getOpenFileName("*.torrent")
         if fn:
             self.newTorrent(str(fn))
     
