diff -Nura kdelibs-3.5.10/kdecore/kcrash.cpp kdelibs-3.5.10.new/kdecore/kcrash.cpp
--- kdelibs-3.5.10/kdecore/kcrash.cpp	2007-01-15 12:34:17.000000000 +0100
+++ kdelibs-3.5.10.new/kdecore/kcrash.cpp	2009-08-06 06:18:06.000000000 +0200
@@ -158,7 +158,7 @@
           int i = 0;
 
           // argument 0 has to be drkonqi
-          argv[i++] = "drkonqi";
+          argv[i++] = "/usr/lib/kde4/libexec/drkonqi";
 
 #if defined Q_WS_X11
           // start up on the correct display
@@ -325,7 +325,7 @@
       _exit(253);
     if(!geteuid() && setuid(getuid()) < 0)
       _exit(253);
-    execvp("drkonqi", const_cast< char** >( argv ));
+    execvp("/usr/lib/kde4/libexec/drkonqi", const_cast< char** >( argv ));
     _exit(errno);
   }
   else
diff -Nura kdelibs-3.5.10/kded/Makefile.am kdelibs-3.5.10.new/kded/Makefile.am
--- kdelibs-3.5.10/kded/Makefile.am	2005-10-10 17:06:29.000000000 +0200
+++ kdelibs-3.5.10.new/kded/Makefile.am	2009-08-06 06:21:37.000000000 +0200
@@ -58,7 +58,7 @@
 servicetype_DATA = kdedmodule.desktop
 servicetypedir = $(kde_servicetypesdir)
 
-xdg_menu_DATA = applications.menu
+xdg_menu_DATA = kde-applications.menu
 
 update_DATA = kded.upd
 updatedir = $(kde_datadir)/kconf_update
diff -Nura kdelibs-3.5.10/kded/Makefile.in kdelibs-3.5.10.new/kded/Makefile.in
--- kdelibs-3.5.10/kded/Makefile.in	2008-08-20 18:05:57.000000000 +0200
+++ kdelibs-3.5.10.new/kded/Makefile.in	2009-08-06 06:22:19.000000000 +0200
@@ -535,7 +535,7 @@
 
 servicetype_DATA = kdedmodule.desktop
 servicetypedir = $(kde_servicetypesdir)
-xdg_menu_DATA = applications.menu
+xdg_menu_DATA = kde-applications.menu
 update_DATA = kded.upd
 updatedir = $(kde_datadir)/kconf_update
 #>- all: all-am
@@ -1173,7 +1173,7 @@
 	-rm -f  kded.moc vfolder_menu.moc kdedmodule.moc kbuildsycoca.moc
 
 #>+ 2
-KDE_DIST=DESIGN kded.upd Makefile.in HOWTO vfolder_menu.h applications.menu kded.h kdedmodule.desktop README.kded kctimefactory.h Makefile.am 
+KDE_DIST=DESIGN kded.upd Makefile.in HOWTO vfolder_menu.h kde-applications.menu kded.h kdedmodule.desktop README.kded kctimefactory.h Makefile.am 
 
 #>+ 2
 docs-am:
diff -Nura kdelibs-3.5.10/kded/applications.menu kdelibs-3.5.10.new/kded/applications.menu
--- kdelibs-3.5.10/kded/applications.menu	2005-09-10 10:27:40.000000000 +0200
+++ kdelibs-3.5.10.new/kded/applications.menu	1970-01-01 01:00:00.000000000 +0100
@@ -1,469 +0,0 @@
- <!DOCTYPE Menu PUBLIC "-//freedesktop//DTD Menu 1.0//EN"
-  "http://www.freedesktop.org/standards/menu-spec/1.0/menu.dtd">
-
-<Menu>
-	<Name>Applications</Name>
-	<Directory>kde-main.directory</Directory>
-	<!-- Search the default locations -->
-	<KDELegacyDirs/>
-	<DefaultAppDirs/>
-	<DefaultDirectoryDirs/>
-	<DefaultLayout>
-		<Merge type="menus"/>
-		<Merge type="files"/>
-		<Separator/>
-		<Menuname>More</Menuname>
-	</DefaultLayout>
-	<Layout>
-		<Merge type="menus"/>
-		<Menuname>Applications</Menuname>
-		<Merge type="files"/>
-	</Layout>
-
-	<Menu>
-		<Name>Applications</Name>
-		<Directory>kde-unknown.directory</Directory>
-		<OnlyUnallocated/>
-		<Include>
-			<Not>
-				<!-- Don't list non-KDE core applications -->
-				<And>
-					<Category>Core</Category>
-					<Not><Category>KDE</Category></Not>
-				</And>
-				<!-- Don't list SUSE's YaST in here -->
-				<Category>X-SuSE-YaST</Category>
-			</Not>
-		</Include>
-	</Menu>
-       <Menu>
-                <Name>Development</Name>
-                <Directory>kde-development.directory</Directory>
-                <Menu>
-                        <Name>X-KDE-KDevelopIDE</Name>
-                        <Directory>kde-development-kdevelop.directory</Directory>
-                        <Include>
-                                <And>
-                                <Category>Development</Category>
-                                <Category>X-KDE-KDevelopIDE</Category>
-                                </And>
-                        </Include>
-                </Menu>
-		<Menu>
-			<Name>Translation</Name>
-			<Directory>kde-development-translation.directory</Directory>
-			<Include>
-				<And>
-				<Category>Development</Category>
-				<Category>Translation</Category>
-				</And>
-			</Include>
-		</Menu>
-		<Menu>
-			<Name>Web Development</Name>
-			<Directory>kde-development-webdevelopment.directory</Directory>
-			<Include>
-				<And>
-				<Category>Development</Category>
-				<Category>WebDevelopment</Category>
-				</And>
-			</Include>
-		</Menu>
-                <Include>
-                        <And>
-                        <Category>Development</Category>
-                        <Not><Category>X-KDE-KDevelopIDE</Category></Not>
-                        <Not><Category>Translation</Category></Not>
-                        <Not><Category>WebDevelopment</Category></Not>
-                        </And>
-                </Include>
-        </Menu>
-	<Menu>
-		<Name>Science</Name>
-		<Directory>kde-science.directory</Directory>
-		<Include>
-			<And><!-- Include /any/ Science app which is not an Education app -->
-				<Or>
-					<Category>Astronomy</Category>
-					<Category>Biology</Category>
-					<Category>Chemistry</Category>
-					<Category>Geology</Category>
-					<Category>MedicalSoftware</Category>
-					<Category>Physics</Category>
-					<Category>Math</Category>
-					<Category>Science</Category>
-				</Or>
-				<Not><Category>Education</Category></Not>
-			</And>
-		</Include>
-	</Menu>
-	<Menu>
-		<Name>Edutainment</Name>
-		<Directory>kde-edutainment.directory</Directory>
-		<Menu>
-			<Name>Languages</Name>
-			<Directory>kde-edu-languages.directory</Directory>
-			<Include>
-				<And>
-				<Category>Education</Category>
-				<Or>
-				<Category>Languages</Category>
-				<Category>X-KDE-Edu-Language</Category>
-				</Or>
-				</And>
-			</Include>
-		</Menu>
-		<Menu>
-			<Name>Mathematics</Name>
-			<Directory>kde-edu-mathematics.directory</Directory>
-			<Include>
-				<And>
-				<Category>Education</Category>
-				<Category>Math</Category>
-				</And>
-			</Include>
-		</Menu>
-		<Menu>
-			<Name>Miscellaneous</Name>
-			<Directory>kde-edu-miscellaneous.directory</Directory>
-			<Include>
-				<And>
-				<Category>Education</Category>
-				<Not>
-				<Category>Languages</Category>
-				<Category>X-KDE-Edu-Language</Category>
-				<Category>Math</Category>
-				<Category>Science</Category>
-				<Category>Teaching</Category>
-				<Category>X-KDE-Edu-Teaching</Category>
-				</Not>
-				</And>
-			</Include>
-		</Menu>
-		<Menu>
-			<Name>Science</Name>
-			<Directory>kde-edu-science.directory</Directory>
-			<Include>
-				<And>
-				<Category>Education</Category>
-				<Category>Science</Category>
-				</And>
-			</Include>
-		</Menu>
-		<Menu>
-			<Name>Tools</Name>
-			<Directory>kde-edu-tools.directory</Directory>
-			<Include>
-				<And>
-				<Category>Education</Category>
-				<Or>
-				<Category>Teaching</Category>
-				<Category>X-KDE-Edu-Teaching</Category>
-				</Or>
-				</And>
-			</Include>
-		</Menu>
-	</Menu>
-	<Menu>
-		<Name>Games</Name>
-		<Directory>kde-games.directory</Directory>
-		<Menu>
-			<Name>Arcade</Name>
-			<Directory>kde-games-arcade.directory</Directory>
-			<Include>
-				<And>
-				<Category>Game</Category>
-				<Category>ArcadeGame</Category>
-				</And>
-			</Include>
-		</Menu>
-		<Menu>
-			<Name>Board</Name>
-			<Directory>kde-games-board.directory</Directory>
-			<Include>
-				<And>
-				<Category>Game</Category>
-				<Category>BoardGame</Category>
-				</And>
-			</Include>
-		</Menu>
-		<Menu>
-			<Name>Card</Name>
-			<Directory>kde-games-card.directory</Directory>
-			<Include>
-				<And>
-				<Category>Game</Category>
-				<Category>CardGame</Category>
-				</And>
-			</Include>
-		</Menu>
-		<Menu>
-			<Name>Kidsgames</Name>
-			<Directory>kde-games-kids.directory</Directory>
-			<Include>
-				<And>
-				<Category>Game</Category>
-				<Or>
-				<Category>X-KDE-KidsGame</Category>
-				<Category>KidsGame</Category>
-				</Or>
-				</And>
-			</Include>
-		</Menu>
-		<Menu>
-			<Name>TacticStrategy</Name>
-			<Directory>kde-games-strategy.directory</Directory>
-			<Include>
-				<And>
-				<Category>Game</Category>
-				<Category>StrategyGame</Category>
-				</And>
-			</Include>
-		</Menu>
-		<Include>
-			<And>
-			<Category>Game</Category>
-			<Not>
-			<Category>ArcadeGame</Category>
-			<Category>BoardGame</Category>
-			<Category>CardGame</Category>
-			<Category>X-KDE-KidsGame</Category>
-			<Category>KidsGame</Category>
-			<Category>StrategyGame</Category>
-			</Not>
-			</And>
-		</Include>
-               <Menu>
-                        <Name>Toys</Name>
-                        <Directory>kde-toys.directory</Directory>
-                        <Include>
-                                 <Category>Amusement</Category>
-                        </Include>
-               </Menu>
-	</Menu>
-	<Menu>
-		<Name>Graphics</Name>
-		<Directory>kde-graphics.directory</Directory>
-		<Include>
-			<And>
-			<Category>Graphics</Category>
-			<Not><Category>X-KDE-More</Category></Not>
-			</And>
-		</Include>
-		<Menu>
-			<Name>More</Name>
-			<Directory>kde-more.directory</Directory>
-			<Include>
-				<And>
-				<Category>Graphics</Category>
-				<Category>X-KDE-More</Category>
-				</And>
-			</Include>
-		</Menu>
-	</Menu>
-	<Menu>
-		<Name>Internet</Name>
-		<Directory>kde-internet.directory</Directory>
-		<Include>
-			<And>
-			<Category>Network</Category>
-			<Not><Category>X-KDE-More</Category></Not>
-			</And>
-		</Include>
-		<Menu>
-		  <Name>Terminal</Name>
-		  <Directory>kde-internet-terminal.directory</Directory>
-		</Menu>
-		<Menu>
-			<Name>More</Name>
-			<Directory>kde-more.directory</Directory>
-			<Include>
-				<And>
-				<Category>Network</Category>
-				<Category>X-KDE-More</Category>
-				</And>
-			</Include>
-		</Menu>
-	</Menu>
-	<Menu>
-		<Name>Multimedia</Name>
-		<Directory>kde-multimedia.directory</Directory>
-		<Include>
-			<And>
-			<Category>AudioVideo</Category>
-			<Not><Category>X-KDE-More</Category></Not>
-			</And>
-		</Include>
-		<Menu>
-			<Name>More</Name>
-			<Directory>kde-more.directory</Directory>
-			<Include>
-				<And>
-				<Category>AudioVideo</Category>
-				<Category>X-KDE-More</Category>
-				</And>
-			</Include>
-		</Menu>
-	</Menu>
-	<Menu>
-		<Name>Office</Name>
-		<Directory>kde-office.directory</Directory>
-		<Layout>
-			<Merge type="menus"/>
-			<Filename>kde-koshell.desktop</Filename>
-			<Filename>kde-Kontact.desktop</Filename>
-			<Separator/>
-			<Filename>kde-kword.desktop</Filename>
-			<Filename>kde-kspread.desktop</Filename>
-			<Filename>kde-kpresenter.desktop</Filename>
-			<Merge type="files"/>
-			<Separator/>
-			<Menuname>More</Menuname>
-		</Layout>
-		<Include>
-                        <And>
-			<Category>Office</Category>
-                        <Not><Category>X-KDE-More</Category></Not>
-                        </And>
-                </Include>
-                <Menu>
-                        <Name>More</Name>
-                        <Directory>kde-more.directory</Directory>
-                        <Include>
-                                <And>
-                                <Category>Office</Category>
-                                <Category>X-KDE-More</Category>
-                                </And>
-                        </Include>
-                </Menu>
-	</Menu>
-	<Menu>
-		<Name>Settingsmenu</Name>
-		<Directory>kde-settingsmenu.directory</Directory>
-		<Include>
-			<Category>Settings</Category>
-		</Include>
-	</Menu>
-	<Menu>
-		<Name>System</Name>
-		<Directory>kde-system.directory</Directory>
-		<Include>
-			<And>
-			<Category>System</Category>
-			<Not><Category>X-KDE-More</Category></Not>
-			</And>
-		</Include>
-		<Menu>
-			<Name>More</Name>
-			<Directory>kde-more.directory</Directory>
-			<Include>
-				<And>
-				<Category>System</Category>
-				<Category>X-KDE-More</Category>
-				</And>
-			</Include>
-		</Menu>
-		<Menu>
-			<Name>ScreenSavers</Name>
-			<Directory>kde-system-screensavers.directory</Directory>
-		</Menu>
-		<Menu>
-			<Name>Terminal</Name>
-			<Directory>kde-system-terminal.directory</Directory>
-		</Menu>
-	</Menu>
-	<Menu>
-		<Name>Utilities</Name>
-		<Directory>kde-utilities.directory</Directory>
-		<Include>
-			<And>
-			<Category>Utility</Category>
-			<Not><Category>Accessibility</Category></Not>
-			<Not><Category>X-KDE-Utilities-Desktop</Category></Not>
-			<Not><Category>X-KDE-Utilities-File</Category></Not>
-			<Not><Category>X-KDE-Utilities-Peripherals</Category></Not>
-			<Not><Category>X-KDE-Utilities-PIM</Category></Not>
-			<Not><Category>X-KDE-More</Category></Not>
-			</And>
-		</Include>
-		<Menu>
-			<Name>Accessibility</Name>
-			<Directory>kde-utilities-accessibility.directory</Directory>
-			<Include>
-				<And>
-				<Category>Utility</Category>
-				<Category>Accessibility</Category>
-				</And>
-			</Include>
-		</Menu>
-		<Menu>
-			<Name>Desktop</Name>
-			<Directory>kde-utilities-desktop.directory</Directory>
-			<Include>
-				<And>
-				<Category>Utility</Category>
-				<Category>X-KDE-Utilities-Desktop</Category>
-				</And>
-			</Include>
-		</Menu>
-                <Menu>
-                        <Name>Editors</Name>
-                        <Directory>kde-editors.directory</Directory>
-                        <Include>
-                                <Category>TextEditor</Category>
-                        </Include>
-                </Menu>
-		<Menu>
-			<Name>File</Name>
-			<Directory>kde-utilities-file.directory</Directory>
-			<Include>
-				<And>
-				<Category>Utility</Category>
-				<Category>X-KDE-Utilities-File</Category>
-				</And>
-			</Include>
-		</Menu>
-		<Menu>
-			<Name>Peripherals</Name>
-			<Directory>kde-utilities-peripherals.directory</Directory>
-			<Include>
-				<And>
-				<Category>Utility</Category>
-				<Category>X-KDE-Utilities-Peripherals</Category>
-				</And>
-			</Include>
-		</Menu>
-		<Menu>
-			<Name>PIM</Name>
-			<Directory>kde-utilities-pim.directory</Directory>
-			<Include>
-				<And>
-				<Category>Utility</Category>
-				<Category>X-KDE-Utilities-PIM</Category>
-				</And>
-			</Include>
-		</Menu>
-		<Menu>
-			<Name>XUtilities</Name>
-			<Directory>kde-utilities-xutils.directory</Directory>
-		</Menu>
-		<Menu>
-			<Name>More</Name>
-			<Directory>kde-more.directory</Directory>
-			<Include>
-				<And>
-				<Category>Utility</Category>
-				<Category>X-KDE-More</Category>
-				</And>
-			</Include>
-		</Menu>
-	</Menu>
-	<Include>
-		<And>
-		<Category>KDE</Category>
-		<Category>Core</Category>
-		</And>
-	</Include>
-	<DefaultMergeDirs/>
-	<MergeFile>applications-kmenuedit.menu</MergeFile>
-</Menu>
diff -Nura kdelibs-3.5.10/kded/kbuildsycoca.cpp kdelibs-3.5.10.new/kded/kbuildsycoca.cpp
--- kdelibs-3.5.10/kded/kbuildsycoca.cpp	2007-05-14 09:52:43.000000000 +0200
+++ kdelibs-3.5.10.new/kded/kbuildsycoca.cpp	2009-08-06 06:23:48.000000000 +0200
@@ -379,7 +379,7 @@
      connect(g_vfolder, SIGNAL(newService(const QString &, KService **)),
              this, SLOT(slotCreateEntry(const QString &, KService **)));
              
-     VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu("applications.menu", true);
+     VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu("kde-applications.menu", true);
 
      KServiceGroup *entry = g_bsgf->addNew("/", kdeMenu->directoryFile, 0, false);
      entry->setLayoutInfo(kdeMenu->layoutList);
diff -Nura kdelibs-3.5.10/kded/kde-applications.menu kdelibs-3.5.10.new/kded/kde-applications.menu
--- kdelibs-3.5.10/kded/kde-applications.menu	1970-01-01 01:00:00.000000000 +0100
+++ kdelibs-3.5.10.new/kded/kde-applications.menu	2005-09-10 10:27:40.000000000 +0200
@@ -0,0 +1,469 @@
+ <!DOCTYPE Menu PUBLIC "-//freedesktop//DTD Menu 1.0//EN"
+  "http://www.freedesktop.org/standards/menu-spec/1.0/menu.dtd">
+
+<Menu>
+	<Name>Applications</Name>
+	<Directory>kde-main.directory</Directory>
+	<!-- Search the default locations -->
+	<KDELegacyDirs/>
+	<DefaultAppDirs/>
+	<DefaultDirectoryDirs/>
+	<DefaultLayout>
+		<Merge type="menus"/>
+		<Merge type="files"/>
+		<Separator/>
+		<Menuname>More</Menuname>
+	</DefaultLayout>
+	<Layout>
+		<Merge type="menus"/>
+		<Menuname>Applications</Menuname>
+		<Merge type="files"/>
+	</Layout>
+
+	<Menu>
+		<Name>Applications</Name>
+		<Directory>kde-unknown.directory</Directory>
+		<OnlyUnallocated/>
+		<Include>
+			<Not>
+				<!-- Don't list non-KDE core applications -->
+				<And>
+					<Category>Core</Category>
+					<Not><Category>KDE</Category></Not>
+				</And>
+				<!-- Don't list SUSE's YaST in here -->
+				<Category>X-SuSE-YaST</Category>
+			</Not>
+		</Include>
+	</Menu>
+       <Menu>
+                <Name>Development</Name>
+                <Directory>kde-development.directory</Directory>
+                <Menu>
+                        <Name>X-KDE-KDevelopIDE</Name>
+                        <Directory>kde-development-kdevelop.directory</Directory>
+                        <Include>
+                                <And>
+                                <Category>Development</Category>
+                                <Category>X-KDE-KDevelopIDE</Category>
+                                </And>
+                        </Include>
+                </Menu>
+		<Menu>
+			<Name>Translation</Name>
+			<Directory>kde-development-translation.directory</Directory>
+			<Include>
+				<And>
+				<Category>Development</Category>
+				<Category>Translation</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Web Development</Name>
+			<Directory>kde-development-webdevelopment.directory</Directory>
+			<Include>
+				<And>
+				<Category>Development</Category>
+				<Category>WebDevelopment</Category>
+				</And>
+			</Include>
+		</Menu>
+                <Include>
+                        <And>
+                        <Category>Development</Category>
+                        <Not><Category>X-KDE-KDevelopIDE</Category></Not>
+                        <Not><Category>Translation</Category></Not>
+                        <Not><Category>WebDevelopment</Category></Not>
+                        </And>
+                </Include>
+        </Menu>
+	<Menu>
+		<Name>Science</Name>
+		<Directory>kde-science.directory</Directory>
+		<Include>
+			<And><!-- Include /any/ Science app which is not an Education app -->
+				<Or>
+					<Category>Astronomy</Category>
+					<Category>Biology</Category>
+					<Category>Chemistry</Category>
+					<Category>Geology</Category>
+					<Category>MedicalSoftware</Category>
+					<Category>Physics</Category>
+					<Category>Math</Category>
+					<Category>Science</Category>
+				</Or>
+				<Not><Category>Education</Category></Not>
+			</And>
+		</Include>
+	</Menu>
+	<Menu>
+		<Name>Edutainment</Name>
+		<Directory>kde-edutainment.directory</Directory>
+		<Menu>
+			<Name>Languages</Name>
+			<Directory>kde-edu-languages.directory</Directory>
+			<Include>
+				<And>
+				<Category>Education</Category>
+				<Or>
+				<Category>Languages</Category>
+				<Category>X-KDE-Edu-Language</Category>
+				</Or>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Mathematics</Name>
+			<Directory>kde-edu-mathematics.directory</Directory>
+			<Include>
+				<And>
+				<Category>Education</Category>
+				<Category>Math</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Miscellaneous</Name>
+			<Directory>kde-edu-miscellaneous.directory</Directory>
+			<Include>
+				<And>
+				<Category>Education</Category>
+				<Not>
+				<Category>Languages</Category>
+				<Category>X-KDE-Edu-Language</Category>
+				<Category>Math</Category>
+				<Category>Science</Category>
+				<Category>Teaching</Category>
+				<Category>X-KDE-Edu-Teaching</Category>
+				</Not>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Science</Name>
+			<Directory>kde-edu-science.directory</Directory>
+			<Include>
+				<And>
+				<Category>Education</Category>
+				<Category>Science</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Tools</Name>
+			<Directory>kde-edu-tools.directory</Directory>
+			<Include>
+				<And>
+				<Category>Education</Category>
+				<Or>
+				<Category>Teaching</Category>
+				<Category>X-KDE-Edu-Teaching</Category>
+				</Or>
+				</And>
+			</Include>
+		</Menu>
+	</Menu>
+	<Menu>
+		<Name>Games</Name>
+		<Directory>kde-games.directory</Directory>
+		<Menu>
+			<Name>Arcade</Name>
+			<Directory>kde-games-arcade.directory</Directory>
+			<Include>
+				<And>
+				<Category>Game</Category>
+				<Category>ArcadeGame</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Board</Name>
+			<Directory>kde-games-board.directory</Directory>
+			<Include>
+				<And>
+				<Category>Game</Category>
+				<Category>BoardGame</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Card</Name>
+			<Directory>kde-games-card.directory</Directory>
+			<Include>
+				<And>
+				<Category>Game</Category>
+				<Category>CardGame</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Kidsgames</Name>
+			<Directory>kde-games-kids.directory</Directory>
+			<Include>
+				<And>
+				<Category>Game</Category>
+				<Or>
+				<Category>X-KDE-KidsGame</Category>
+				<Category>KidsGame</Category>
+				</Or>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>TacticStrategy</Name>
+			<Directory>kde-games-strategy.directory</Directory>
+			<Include>
+				<And>
+				<Category>Game</Category>
+				<Category>StrategyGame</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Include>
+			<And>
+			<Category>Game</Category>
+			<Not>
+			<Category>ArcadeGame</Category>
+			<Category>BoardGame</Category>
+			<Category>CardGame</Category>
+			<Category>X-KDE-KidsGame</Category>
+			<Category>KidsGame</Category>
+			<Category>StrategyGame</Category>
+			</Not>
+			</And>
+		</Include>
+               <Menu>
+                        <Name>Toys</Name>
+                        <Directory>kde-toys.directory</Directory>
+                        <Include>
+                                 <Category>Amusement</Category>
+                        </Include>
+               </Menu>
+	</Menu>
+	<Menu>
+		<Name>Graphics</Name>
+		<Directory>kde-graphics.directory</Directory>
+		<Include>
+			<And>
+			<Category>Graphics</Category>
+			<Not><Category>X-KDE-More</Category></Not>
+			</And>
+		</Include>
+		<Menu>
+			<Name>More</Name>
+			<Directory>kde-more.directory</Directory>
+			<Include>
+				<And>
+				<Category>Graphics</Category>
+				<Category>X-KDE-More</Category>
+				</And>
+			</Include>
+		</Menu>
+	</Menu>
+	<Menu>
+		<Name>Internet</Name>
+		<Directory>kde-internet.directory</Directory>
+		<Include>
+			<And>
+			<Category>Network</Category>
+			<Not><Category>X-KDE-More</Category></Not>
+			</And>
+		</Include>
+		<Menu>
+		  <Name>Terminal</Name>
+		  <Directory>kde-internet-terminal.directory</Directory>
+		</Menu>
+		<Menu>
+			<Name>More</Name>
+			<Directory>kde-more.directory</Directory>
+			<Include>
+				<And>
+				<Category>Network</Category>
+				<Category>X-KDE-More</Category>
+				</And>
+			</Include>
+		</Menu>
+	</Menu>
+	<Menu>
+		<Name>Multimedia</Name>
+		<Directory>kde-multimedia.directory</Directory>
+		<Include>
+			<And>
+			<Category>AudioVideo</Category>
+			<Not><Category>X-KDE-More</Category></Not>
+			</And>
+		</Include>
+		<Menu>
+			<Name>More</Name>
+			<Directory>kde-more.directory</Directory>
+			<Include>
+				<And>
+				<Category>AudioVideo</Category>
+				<Category>X-KDE-More</Category>
+				</And>
+			</Include>
+		</Menu>
+	</Menu>
+	<Menu>
+		<Name>Office</Name>
+		<Directory>kde-office.directory</Directory>
+		<Layout>
+			<Merge type="menus"/>
+			<Filename>kde-koshell.desktop</Filename>
+			<Filename>kde-Kontact.desktop</Filename>
+			<Separator/>
+			<Filename>kde-kword.desktop</Filename>
+			<Filename>kde-kspread.desktop</Filename>
+			<Filename>kde-kpresenter.desktop</Filename>
+			<Merge type="files"/>
+			<Separator/>
+			<Menuname>More</Menuname>
+		</Layout>
+		<Include>
+                        <And>
+			<Category>Office</Category>
+                        <Not><Category>X-KDE-More</Category></Not>
+                        </And>
+                </Include>
+                <Menu>
+                        <Name>More</Name>
+                        <Directory>kde-more.directory</Directory>
+                        <Include>
+                                <And>
+                                <Category>Office</Category>
+                                <Category>X-KDE-More</Category>
+                                </And>
+                        </Include>
+                </Menu>
+	</Menu>
+	<Menu>
+		<Name>Settingsmenu</Name>
+		<Directory>kde-settingsmenu.directory</Directory>
+		<Include>
+			<Category>Settings</Category>
+		</Include>
+	</Menu>
+	<Menu>
+		<Name>System</Name>
+		<Directory>kde-system.directory</Directory>
+		<Include>
+			<And>
+			<Category>System</Category>
+			<Not><Category>X-KDE-More</Category></Not>
+			</And>
+		</Include>
+		<Menu>
+			<Name>More</Name>
+			<Directory>kde-more.directory</Directory>
+			<Include>
+				<And>
+				<Category>System</Category>
+				<Category>X-KDE-More</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>ScreenSavers</Name>
+			<Directory>kde-system-screensavers.directory</Directory>
+		</Menu>
+		<Menu>
+			<Name>Terminal</Name>
+			<Directory>kde-system-terminal.directory</Directory>
+		</Menu>
+	</Menu>
+	<Menu>
+		<Name>Utilities</Name>
+		<Directory>kde-utilities.directory</Directory>
+		<Include>
+			<And>
+			<Category>Utility</Category>
+			<Not><Category>Accessibility</Category></Not>
+			<Not><Category>X-KDE-Utilities-Desktop</Category></Not>
+			<Not><Category>X-KDE-Utilities-File</Category></Not>
+			<Not><Category>X-KDE-Utilities-Peripherals</Category></Not>
+			<Not><Category>X-KDE-Utilities-PIM</Category></Not>
+			<Not><Category>X-KDE-More</Category></Not>
+			</And>
+		</Include>
+		<Menu>
+			<Name>Accessibility</Name>
+			<Directory>kde-utilities-accessibility.directory</Directory>
+			<Include>
+				<And>
+				<Category>Utility</Category>
+				<Category>Accessibility</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Desktop</Name>
+			<Directory>kde-utilities-desktop.directory</Directory>
+			<Include>
+				<And>
+				<Category>Utility</Category>
+				<Category>X-KDE-Utilities-Desktop</Category>
+				</And>
+			</Include>
+		</Menu>
+                <Menu>
+                        <Name>Editors</Name>
+                        <Directory>kde-editors.directory</Directory>
+                        <Include>
+                                <Category>TextEditor</Category>
+                        </Include>
+                </Menu>
+		<Menu>
+			<Name>File</Name>
+			<Directory>kde-utilities-file.directory</Directory>
+			<Include>
+				<And>
+				<Category>Utility</Category>
+				<Category>X-KDE-Utilities-File</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Peripherals</Name>
+			<Directory>kde-utilities-peripherals.directory</Directory>
+			<Include>
+				<And>
+				<Category>Utility</Category>
+				<Category>X-KDE-Utilities-Peripherals</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>PIM</Name>
+			<Directory>kde-utilities-pim.directory</Directory>
+			<Include>
+				<And>
+				<Category>Utility</Category>
+				<Category>X-KDE-Utilities-PIM</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>XUtilities</Name>
+			<Directory>kde-utilities-xutils.directory</Directory>
+		</Menu>
+		<Menu>
+			<Name>More</Name>
+			<Directory>kde-more.directory</Directory>
+			<Include>
+				<And>
+				<Category>Utility</Category>
+				<Category>X-KDE-More</Category>
+				</And>
+			</Include>
+		</Menu>
+	</Menu>
+	<Include>
+		<And>
+		<Category>KDE</Category>
+		<Category>Core</Category>
+		</And>
+	</Include>
+	<DefaultMergeDirs/>
+	<MergeFile>applications-kmenuedit.menu</MergeFile>
+</Menu>
