diff -Naur moc-2.4.4/decoder_plugins/ffmpeg/ffmpeg.c moc-2.4.4-new/decoder_plugins/ffmpeg/ffmpeg.c
--- moc-2.4.4/decoder_plugins/ffmpeg/ffmpeg.c	2008-04-09 14:44:00.000000000 -0400
+++ moc-2.4.4-new/decoder_plugins/ffmpeg/ffmpeg.c	2009-03-07 01:31:48.000000000 -0500
@@ -222,8 +222,8 @@
 {
 	struct ffmpeg_data *data = (struct ffmpeg_data *)prv_data;
 	int ret;
-	int data_size;
 	char avbuf[AVCODEC_MAX_AUDIO_FRAME_SIZE * sizeof(int16_t)];
+	int data_size = sizeof(avbuf);
 	AVPacket pkt;
 	uint8_t *pkt_data;
 	int pkt_size = 0;
@@ -270,7 +270,7 @@
 		while (pkt_size) {
 			int len;
 		
-			len = avcodec_decode_audio (data->enc, (int16_t *)avbuf,
+			len = avcodec_decode_audio2 (data->enc, (int16_t *)avbuf,
 					&data_size, pkt_data, pkt_size);
 			debug ("Decoded %dB", data_size);
 
