Only in netpbm-10.73.31/buildtools/: depend.mk
diff -aur netpbm-10.73.31.old/buildtools/stamp-date netpbm-10.73.31/buildtools/stamp-date
--- netpbm-10.73.31.old/buildtools/stamp-date	2020-05-30 17:34:29.833395465 +0200
+++ netpbm-10.73.31/buildtools/stamp-date	2020-05-30 17:48:52.910091897 +0200
@@ -9,7 +9,7 @@
 # documentation.  This software is provided "as is" without express or
 # implied warranty.
 #
-DATE=$(date)
+DATE="$(date --date="@${SOURCE_DATE_EPOCH:-$(date +%s)}")"
 LOGNAME_OR_UNKNOWN=${LOGNAME:-UNKNOWN}
 USER=${USER:-$LOGNAME_OR_UNKNOWN}
 if [ "$USER" = "UNKNOWN" ]; then
