diff -urN a/base-qt/CustomPixmap.cpp b/base-qt/CustomPixmap.cpp
--- a/base-qt/CustomPixmap.cpp	2014-09-26 02:05:51.000000000 -0600
+++ b/base-qt/CustomPixmap.cpp	2015-06-21 11:36:25.050199863 -0600
@@ -208,7 +208,7 @@
 
     Debug::Throw( "CustomPixmap::highlighted.\n" );
     if( opacity <= 0 ) return *this;
-    opacity = qMin( opacity, 1.0 );
+    opacity = qMin( (double)opacity, 1.0 );
 
     // compute mask
     QPixmap mask( *this );
diff -urN a/base-qt/PathEditor_p.h b/base-qt/PathEditor_p.h
--- a/base-qt/PathEditor_p.h	2014-09-12 02:43:11.000000000 -0600
+++ b/base-qt/PathEditor_p.h	2015-06-21 11:36:41.300204916 -0600
@@ -68,7 +68,7 @@
     { mouseOver_ = value; }
 
     //! border width
-    static const qreal BorderWidth;
+    static const double BorderWidth;
 
     protected:
 
