diff -urN a/base-qt/CustomPixmap.cpp b/base-qt/CustomPixmap.cpp
--- a/base-qt/CustomPixmap.cpp	2013-02-14 15:16:24.000000000 -0700
+++ b/base-qt/CustomPixmap.cpp	2013-05-11 10:58:03.071698156 -0600
@@ -169,7 +169,7 @@
 
     Debug::Throw( "CustomPixmap::highlighted.\n" );
     if( opacity <= 0 ) return *this;
-    opacity = qMin( opacity, 1.0 );
+    opacity = qMin( (double)opacity, 1.0 );
 
     // compute mask
     QPixmap mask( *this );
diff -urN a/base-qt/PathEditor_p.h b/base-qt/PathEditor_p.h
--- a/base-qt/PathEditor_p.h	2013-02-14 02:02:20.000000000 -0700
+++ b/base-qt/PathEditor_p.h	2013-05-11 11:04:16.119694794 -0600
@@ -67,7 +67,7 @@
     { mouseOver_ = value; }
 
     //! border width
-    static const qreal BorderWidth;
+    static const double BorderWidth;
 
     protected:
 
diff -urN a/base-qt/PlacesWidget_p.h b/base-qt/PlacesWidget_p.h
--- a/base-qt/PlacesWidget_p.h	2013-03-15 10:32:24.000000000 -0600
+++ b/base-qt/PlacesWidget_p.h	2013-05-11 11:18:43.734230867 -0600
@@ -131,7 +131,7 @@
     //@}
 
     //! border width
-    static const qreal BorderWidth;
+    static const double BorderWidth;
 
     protected:
 
