--- arch/arm/boot/dts/zynq-parallella.dts.orig	2014-03-18 15:44:28.575321598 -0600
+++ arch/arm/boot/dts/zynq-parallella.dts	2014-03-18 15:44:59.980206710 -0600
@@ -302,7 +302,7 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyPS0,115200 root=/dev/mmcblk0p2 rw earlyprintk rootfstype=ext4 rootwait devtmpfs.mount=0";
+		bootargs = "console=ttyPS0,115200 root=/dev/mmcblk0p2 rw earlyprintk rootwait";
 		linux,stdout-path = "/amba@0/uart@E0001000";
 	};
 
