--- a/src/libcec/CMakeLists.txt	2015-05-21 16:05:34.321349131 +0200
+++ b/src/libcec/CMakeLists.txt	2015-05-21 16:16:48.186086036 +0200
@@ -96,6 +96,9 @@
                 adapter/RPi/RPiCECAdapterMessageQueue.h
                 adapter/RPi/RPiCECAdapterCommunication.h
                 adapter/RPi/RPiCECAdapterDetection.h
+                adapter/IMX/AdapterMessageQueue.h
+                adapter/IMX/IMXCECAdapterCommunication.h
+                adapter/IMX/IMXCECAdapterDetection.h
                 CECInputBuffer.h
                 platform/util/baudrate.h
                 platform/util/edid.h
--- a/src/libcec/cmake/CheckPlatformSupport.cmake	2015-05-21 16:09:28.778514006 +0200
+++ b/src/libcec/cmake/CheckPlatformSupport.cmake	2015-05-21 16:19:42.733309082 +0200
@@ -10,6 +10,7 @@
 #	HAVE_RPI_API              1 if Raspberry Pi is supported
 #	HAVE_TDA995X_API          1 if TDA995X is supported
 #	HAVE_EXYNOS_API           1 if Exynos is supported
+#	HAVE_IMX6_API             1 if iMX.6 is supported
 #       HAVE_P8_USB_DETECT        1 if Pulse-Eight devices can be auto-detected
 #
 
@@ -108,6 +109,18 @@
   else()
     set(HAVE_EXYNOS_API 0)
   endif()
+
+  # i.MX6
+  if (${HAVE_IMX6_API})
+    set(LIB_INFO "${LIB_INFO}, 'i.MX6'")
+    set(HAVE_IMX6_API 1)
+    set(CEC_SOURCES_ADAPTER_IMX6 adapter/IMX/IMXCECAdapterCommunication.cpp
+                                 adapter/IMX/IMXCECAdapterDetection.cpp)
+    source_group("Source Files\\adapter\\IMX" FILES ${CEC_SOURCES_ADAPTER_IMX6})
+    list(APPEND CEC_SOURCES ${CEC_SOURCES_ADAPTER_IMX6})
+  else()
+    set(HAVE_IMX6_API 0)
+  endif()
 endif()
 
 # rt
--- a/src/libcec/cmake/DisplayPlatformSupport.cmake	2015-05-21 16:27:55.323461384 +0200
+++ b/src/libcec/cmake/DisplayPlatformSupport.cmake	2015-05-21 16:27:55.343471443 +0200
@@ -44,6 +44,12 @@
   message(STATUS "Exynos support:                         no")
 endif()
 
+if (HAVE_IMX6_API)
+  message(STATUS "i.MX6 support:                          yes")
+else()
+  message(STATUS "i.MX6 support:                          no")
+endif()
+
 if (HAVE_PYTHON)
   message(STATUS "Python support:                         version ${PYTHONLIBS_VERSION_STRING} (${PYTHON_VERSION})")
 else()
--- a/src/libcec/env.h.in	2015-05-28 08:24:20.283040169 +0200
+++ b/src/libcec/env.h.in	2015-05-28 08:24:20.287040168 +0200
@@ -63,6 +63,9 @@
 /* Define to 1 for TDA995x support */
 #cmakedefine HAVE_TDA995X_API @HAVE_TDA995X_API@
 
+/* Define to 1 for IMX6 support */
+#cmakedefine HAVE_IMX6_API @HAVE_IMX6_API@
+
 /* Define to 1 for Exynos support */
 #cmakedefine HAVE_TDA995X_API @HAVE_TDA995X_API@
 
