From 031922987a1fea903ecc908d1b40d696823f1f28 Mon Sep 17 00:00:00 2001
From: Brian Heim <brianlheim@gmail.com>
Date: Sun, 4 Mar 2018 18:48:42 -0500
Subject: [PATCH 2/2] set YAMLCPP_LIBRARY and YAMLCPP_INCLUDE_DIR if not using
 system

---
 external_libraries/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/external_libraries/CMakeLists.txt b/external_libraries/CMakeLists.txt
index 10c466fd4..df33dc6f1 100644
--- a/external_libraries/CMakeLists.txt
+++ b/external_libraries/CMakeLists.txt
@@ -106,6 +106,8 @@ if(NOT YAMLCPP_FOUND)
   add_library(yaml STATIC EXCLUDE_FROM_ALL ${CMAKE_CURRENT_BINARY_DIR}/libyamlcpp.cpp)
   target_include_directories(yaml PUBLIC ${CMAKE_SOURCE_DIR}/external_libraries/yaml-cpp/include boost)
   set_property( TARGET yaml PROPERTY FOLDER 3rdparty )
+  set(YAMLCPP_LIBRARY yaml)
+  set(YAMLCPP_INCLUDE_DIR ${CMAKE_SOURCE_DIR}/external_libraries/yaml-cpp/include)
 
   if(CMAKE_COMPILER_IS_GNUCXX)
     target_compile_options(yaml PRIVATE -Wno-deprecated-declarations)
-- 
2.16.2

