diff -upr supercollider-3.9.3.orig/server/supernova/utilities/time_tag.hpp supercollider-3.9.3/server/supernova/utilities/time_tag.hpp
--- supercollider-3.9.3.orig/server/supernova/utilities/time_tag.hpp	2018-03-28 23:05:07.000000000 +0300
+++ supercollider-3.9.3/server/supernova/utilities/time_tag.hpp	2018-06-03 01:32:16.824417338 +0300
@@ -230,7 +230,7 @@ public:
 #ifdef BOOST_DATE_TIME_POSIX_TIME_STD_CONFIG
         time_duration offset = seconds(get_secs() - ntp_offset) + nanoseconds(get_nanoseconds());
 #else
-        time_duration offset = seconds(get_secs() - ntp_offset) + microseconds(get_nanoseconds()/1000);
+        time_duration offset = seconds(get_secs() - ntp_offset) + microseconds(long(get_nanoseconds()/1000));
 #endif
         return ptime(base, offset);
     }
