--- server/plugins/DiskIO_UGens.cpp	2016-08-22 10:12:21.000000000 +0200
+++ server/plugins/DiskIO_UGens.cpp.fix	2017-12-06 23:45:39.043242894 +0100
@@ -27,6 +27,7 @@
 #include <sndfile.h>
 
 #include <atomic>
+#include <functional>
 #include <new>
 #include <SC_Lock.h>
 
--- lang/LangPrimSource/PyrSerialPrim.cpp	2016-08-22 10:12:21.000000000 +0200
+++ lang/LangPrimSource/PyrSerialPrim.cpp.fix	2017-12-07 00:06:12.756776492 +0100
@@ -27,6 +27,7 @@
 
 #include <errno.h>
 #include <fcntl.h>
+#include <functional>
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
