--- SuperCollider-Source/CMakeLists.txt.orig	2016-10-23 10:21:32.322819928 +0000
+++ SuperCollider-Source/CMakeLists.txt	2016-10-23 10:21:52.259731856 +0000
@@ -240,7 +240,7 @@
 if (CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG)
 	if (SSE)
 		set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}   -msse")
-		set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -msse")
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse")
 		if(NOT APPLE AND NOT CMAKE_COMPILER_IS_CLANG)
 			set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfpmath=sse")
 			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfpmath=sse")
