diff --git c/CMakeLists.txt w/CMakeLists.txt
index 30d9c8d1a..5e1e9cb89 100644
--- c/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -228,6 +228,7 @@ endif()
 
 option(SC_SYMLINK_CLASSLIB "Place a symlink of SCCLassLibrary instead of copying" OFF)
 
+option(SYSTEM_ABLETON_LINK "Use link from system" OFF)
 option(SYSTEM_BOOST   "Use boost libraries from system" OFF)
 option(SYSTEM_YAMLCPP "Use yaml-cpp library from system" OFF)
 
diff --git c/lang/CMakeLists.txt w/lang/CMakeLists.txt
index 9b5095d11..bb678c016 100644
--- c/lang/CMakeLists.txt
+++ w/lang/CMakeLists.txt
@@ -252,7 +252,12 @@ endif()
 
 if (SC_ABLETON_LINK)
 	message(STATUS "Compiling with Ableton Link support")
-	include(../external_libraries/link/AbletonLinkConfig.cmake)
+	if(SYSTEM_ABLETON_LINK)
+		find_package(AbletonLink NAMES AbletonLink ableton-link link REQUIRED)
+	else()
+		include(../external_libraries/link/AbletonLinkConfig.cmake)
+	endif()
+
 	target_link_libraries(libsclang Ableton::Link)
 
 	add_definitions(-DSC_ABLETON_LINK)
