diff -up firefox-146.0/third_party/rust/glslopt/glsl-optimizer/include/c11/threads_posix.h.build-c11-threads-avail firefox-146.0/third_party/rust/glslopt/glsl-optimizer/include/c11/threads_posix.h
--- firefox-146.0/third_party/rust/glslopt/glsl-optimizer/include/c11/threads_posix.h.build-c11-threads-avail	2025-12-05 18:43:20.000000000 +0100
+++ firefox-146.0/third_party/rust/glslopt/glsl-optimizer/include/c11/threads_posix.h	2025-12-11 13:30:50.463371937 +0100
@@ -66,7 +66,7 @@ typedef pthread_cond_t  cnd_t;
 typedef pthread_t       thrd_t;
 typedef pthread_key_t   tss_t;
 typedef pthread_mutex_t mtx_t;
-typedef pthread_once_t  once_flag;
+//typedef pthread_once_t  once_flag;
 
 
 /*
@@ -90,12 +90,13 @@ impl_thrd_routine(void *p)
 
 /*--------------- 7.25.2 Initialization functions ---------------*/
 // 7.25.2.1
-static inline void
+//
+/*static inline void
 call_once(once_flag *flag, void (*func)(void))
 {
     pthread_once(flag, func);
 }
-
+*/
 
 /*------------- 7.25.3 Condition variable functions -------------*/
 // 7.25.3.1
diff -up a/security/sandbox/chromium/sandbox/linux/system_headers/linux_seccomp.h b/security/sandbox/chromium/sandbox/linux/system_headers/linux_seccomp.h
--- a/security/sandbox/chromium/sandbox/linux/system_headers/linux_seccomp.h	2025-12-09 13:42:31.378958842 +0100
+++ b/security/sandbox/chromium/sandbox/linux/system_headers/linux_seccomp.h	2025-12-09 13:42:37.886937626 +0100
@@ -6,6 +6,7 @@
 #define SANDBOX_LINUX_SYSTEM_HEADERS_LINUX_SECCOMP_H_
 
 #include <stdint.h>
+#include <signal.h>
 #include <sys/ioctl.h>
 
 #include "build/build_config.h"
--- a/comm/third_party/rust/glslopt/glsl-optimizer/include/c11/threads_posix.h.orig
+++ b/comm/third_party/rust/glslopt/glsl-optimizer/include/c11/threads_posix.h
@@ -51,7 +51,6 @@ Configuration macro:
 #include <pthread.h>
 
 /*---------------------------- macros ----------------------------*/
-#define ONCE_FLAG_INIT PTHREAD_ONCE_INIT
 #ifdef INIT_ONCE_STATIC_INIT
 #define TSS_DTOR_ITERATIONS PTHREAD_DESTRUCTOR_ITERATIONS
 #else
@@ -66,7 +65,6 @@ typedef pthread_cond_t  cnd_t;
 typedef pthread_t       thrd_t;
 typedef pthread_key_t   tss_t;
 typedef pthread_mutex_t mtx_t;
-typedef pthread_once_t  once_flag;
 
 
 /*
@@ -88,15 +86,6 @@ impl_thrd_routine(void *p)
 }
 
 
-/*--------------- 7.25.2 Initialization functions ---------------*/
-// 7.25.2.1
-static inline void
-call_once(once_flag *flag, void (*func)(void))
-{
-    pthread_once(flag, func);
-}
-
-
 /*------------- 7.25.3 Condition variable functions -------------*/
 // 7.25.3.1
 static inline int
