--- a/comm/rust/sys_tray/src/linux/mod.rs
+++ b/comm/rust/sys_tray/src/linux/mod.rs
@@ -7,6 +7,7 @@
 use nserror::{nsresult, NS_OK};
 use std::os::raw::c_void;
 use std::rc::Rc;
+use std::thread;
 use system_tray::{SystemTray, TrayItem, XdgIcon};
 use xpcom::{nsIID, xpcom_method, RefPtr};
 
@@ -82,7 +83,10 @@
         let tray = SystemTray::new("Thunderbird", icon, "Thunderbird Daily").with_items(menus);
         let service = ksni::TrayService::new(tray);
         let handle = service.handle();
-        service.spawn_without_dbus_name();
+        thread::spawn(|| match service.run_without_dbus_name() {
+            Ok(_) => (),
+            Err(e) => log::error!("Spawning system tray FAILED: {e}"),
+        });
         LinuxSysTrayHandler::allocate(InitLinuxSysTrayHandler { handle })
     }
 
