
diff --git a/mailnews/local/src/nsMsgLocalStoreUtils.cpp b/mailnews/local/src/nsMsgLocalStoreUtils.cpp
--- a/mailnews/local/src/nsMsgLocalStoreUtils.cpp
+++ b/mailnews/local/src/nsMsgLocalStoreUtils.cpp
@@ -23,16 +23,19 @@ nsMsgLocalStoreUtils::AddDirectorySepara
   path->GetLeafName(leafName);
   leafName.AppendLiteral(FOLDER_SUFFIX);
   return path->SetLeafName(leafName);
 }
 
 bool
 nsMsgLocalStoreUtils::nsShouldIgnoreFile(nsAString& name)
 {
+  if (name.IsEmpty())
+    return true;
+
   char16_t firstChar = name.First();
   if (firstChar == '.' || firstChar == '#' ||
       name.CharAt(name.Length() - 1) == '~')
     return true;
 
   if (name.LowerCaseEqualsLiteral("msgfilterrules.dat") ||
       name.LowerCaseEqualsLiteral("rules.dat") ||
       name.LowerCaseEqualsLiteral("filterlog.html") ||




