--- a/comm/third_party/rust/encoding_rs/src/x_user_defined.rs	2026-04-22 09:34:19.814809516 +0100
+++ b/comm/third_party/rust/encoding_rs/src/x_user_defined.rs	2026-04-22 09:34:31.051967848 +0100
@@ -16,6 +16,7 @@
         use simd_funcs::*;
         use core::simd::u16x8;
         use core::simd::cmp::SimdPartialOrd;
+        use core::simd::Select;
 
         #[inline(always)]
         fn shift_upper(unpacked: u16x8) -> u16x8 {
