--- thunderbird-102.7.1/dom/media/webrtc/sdp/RsdparsaSdpGlue.cpp.gcc13-header-dependency	2023-01-24 04:23:43.000000000 +0100
+++ thunderbird-102.7.1/dom/media/webrtc/sdp/RsdparsaSdpGlue.cpp	2023-01-25 19:23:11.048662899 +0100
@@ -3,6 +3,7 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include <cstdint>
 #include <string>
 
 #include "sdp/RsdparsaSdpInc.h"
--- thunderbird-102.7.1/gfx/2d/Rect.h.gcc13-header-dependency	2023-01-24 04:23:44.000000000 +0100
+++ thunderbird-102.7.1/gfx/2d/Rect.h	2023-01-25 19:23:11.049662897 +0100
@@ -15,6 +15,7 @@
 #include "mozilla/Maybe.h"
 
 #include <cmath>
+#include <cstdint>
 
 namespace mozilla {
 
--- thunderbird-102.7.1/toolkit/components/telemetry/pingsender/pingsender.cpp.gcc13-header-dependency	2023-01-24 04:23:55.000000000 +0100
+++ thunderbird-102.7.1/toolkit/components/telemetry/pingsender/pingsender.cpp	2023-01-25 21:38:39.432188899 +0100
@@ -3,6 +3,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include <cstdint>
 #include <cstdlib>
 #include <cstring>
 #include <ctime>
