diff -ru thunderbird-148.0.1.orig/comm/third_party/rnp/src/lib/crypto/mem.cpp thunderbird-148.0.1/comm/third_party/rnp/src/lib/crypto/mem.cpp
--- thunderbird-148.0.1.orig/comm/third_party/rnp/src/lib/crypto/mem.cpp        2026-03-16 00:06:34.026592733 +0100
+++ thunderbird-148.0.1/comm/third_party/rnp/src/lib/crypto/mem.cpp     2026-03-16 00:07:24.336669665 +0100
@@ -25,6 +25,7 @@
  */
 
 #include <cstdio>
+#include <cstring>
 #include "mem.h"
 #include "logging.h"
 #include <botan/ffi.h>
