diff -u -rN thunderbird-52.1.1/mozilla/config/external/nss/crmf/moz.build thunderbird-52.1.1-nocrmf/mozilla/config/external/nss/crmf/moz.build
--- thunderbird-52.1.1/mozilla/config/external/nss/crmf/moz.build	2017-05-09 23:35:13.000000000 +0200
+++ thunderbird-52.1.1-nocrmf/mozilla/config/external/nss/crmf/moz.build	2017-06-14 01:51:36.686773132 +0200
@@ -8,7 +8,6 @@
 
 if CONFIG['MOZ_SYSTEM_NSS']:
     OS_LIBS += [l for l in CONFIG['NSS_LIBS'] if l.startswith('-L')]
-    OS_LIBS += ['-lcrmf']
 else:
     USE_LIBS += [
         # The dependency on nss is not real, but is required to force the
diff -u -rN thunderbird-52.1.1/mozilla/old-configure thunderbird-52.1.1-nocrmf/mozilla/old-configure
--- thunderbird-52.1.1/mozilla/old-configure	2017-05-09 23:35:35.000000000 +0200
+++ thunderbird-52.1.1-nocrmf/mozilla/old-configure	2017-06-14 01:50:28.726873231 +0200
@@ -10692,9 +10692,7 @@
 
 fi
 
-if test -n "$MOZ_SYSTEM_NSS"; then
-   NSS_LIBS="$NSS_LIBS -lcrmf"
-else
+if test -z "$MOZ_SYSTEM_NSS"; then
    NSS_CFLAGS="-I${DIST}/include/nss"
 fi
 
diff -u -rN thunderbird-52.1.1/mozilla/old-configure.in thunderbird-52.1.1-nocrmf/mozilla/old-configure.in
--- thunderbird-52.1.1/mozilla/old-configure.in	2017-05-09 23:35:22.000000000 +0200
+++ thunderbird-52.1.1-nocrmf/mozilla/old-configure.in	2017-06-14 01:50:50.953507079 +0200
@@ -2126,9 +2126,7 @@
     AM_PATH_NSS(3.28.4, [MOZ_SYSTEM_NSS=1], [AC_MSG_ERROR([you don't have NSS installed or your version is too old])])
 fi
 
-if test -n "$MOZ_SYSTEM_NSS"; then
-   NSS_LIBS="$NSS_LIBS -lcrmf"
-else
+if test -z "$MOZ_SYSTEM_NSS"; then
    NSS_CFLAGS="-I${DIST}/include/nss"
 fi
 
