--- mozilla/configure.orig	2014-02-16 19:18:34.782768334 -0700
+++ mozilla/configure	2014-02-16 19:18:53.287522296 -0700
@@ -6675,91 +6675,9 @@
 
 
 
-if test "$CPU_ARCH" = "arm"; then
-  echo $ac_n "checking for ARM SIMD support in compiler""... $ac_c" 1>&6
-echo "configure:6681: checking for ARM SIMD support in compiler" >&5
-  # We try to link so that this also fails when
-  # building with LTO.
-  cat > conftest.$ac_ext <<EOF
-#line 6685 "configure"
-#include "confdefs.h"
-
-int main() {
-asm("uqadd8 r1, r1, r2");
-; return 0; }
-EOF
-if { (eval echo configure:6692: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  result="yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  result="no"
-fi
-rm -f conftest*
-  echo "$ac_t"""$result"" 1>&6
-  if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_SIMD ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_SIMD 1
-EOF
-
-      HAVE_ARM_SIMD=1
-  fi
-
-  echo $ac_n "checking ARM version support in compiler""... $ac_c" 1>&6
-echo "configure:6715: checking ARM version support in compiler" >&5
-    ARM_ARCH=`${CC-cc} ${CFLAGS} -dM -E - < /dev/null | sed -n 's/.*__ARM_ARCH_\([0-9]*\).*/\1/p'`
-  echo "$ac_t"""$ARM_ARCH"" 1>&6
-
-  echo $ac_n "checking for ARM NEON support in compiler""... $ac_c" 1>&6
-echo "configure:6720: checking for ARM NEON support in compiler" >&5
-  # We try to link so that this also fails when
-  # building with LTO.
-  cat > conftest.$ac_ext <<EOF
-#line 6724 "configure"
-#include "confdefs.h"
-
-int main() {
-asm(".fpu neon\n vadd.i8 d0, d0, d0");
-; return 0; }
-EOF
-if { (eval echo configure:6731: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  result="yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  result="no"
-fi
-rm -f conftest*
-  echo "$ac_t"""$result"" 1>&6
-  if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_NEON 1
-EOF
-
-      HAVE_ARM_NEON=1
-
-                  if test -n "$ARM_ARCH"; then
-          if test "$ARM_ARCH" -lt 7; then
-              BUILD_ARM_NEON=0
-          else
-              BUILD_ARM_NEON=1
-          fi
-      fi
-  fi
-
-fi # CPU_ARCH = arm
-
-
+HAVE_ARM_NEON=0
+BUILD_ARM_NEON=0
+HAVE_ARM_SIMD=0
 
 
 
--- mozilla/js/src/configure.orig	2014-02-16 19:15:10.325465791 -0700
+++ mozilla/js/src/configure	2014-02-16 19:15:28.820223476 -0700
@@ -5900,90 +5900,9 @@
 
 
 
-if test "$CPU_ARCH" = "arm"; then
-  echo $ac_n "checking for ARM SIMD support in compiler""... $ac_c" 1>&6
-echo "configure:5906: checking for ARM SIMD support in compiler" >&5
-  # We try to link so that this also fails when
-  # building with LTO.
-  cat > conftest.$ac_ext <<EOF
-#line 5910 "configure"
-#include "confdefs.h"
-
-int main() {
-asm("uqadd8 r1, r1, r2");
-; return 0; }
-EOF
-if { (eval echo configure:5917: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  result="yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  result="no"
-fi
-rm -f conftest*
-  echo "$ac_t"""$result"" 1>&6
-  if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_SIMD ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_SIMD 1
-EOF
-
-      HAVE_ARM_SIMD=1
-  fi
-
-  echo $ac_n "checking ARM version support in compiler""... $ac_c" 1>&6
-echo "configure:5940: checking ARM version support in compiler" >&5
-    ARM_ARCH=`${CC-cc} ${CFLAGS} -dM -E - < /dev/null | sed -n 's/.*__ARM_ARCH_\([0-9]*\).*/\1/p'`
-  echo "$ac_t"""$ARM_ARCH"" 1>&6
-
-  echo $ac_n "checking for ARM NEON support in compiler""... $ac_c" 1>&6
-echo "configure:5945: checking for ARM NEON support in compiler" >&5
-  # We try to link so that this also fails when
-  # building with LTO.
-  cat > conftest.$ac_ext <<EOF
-#line 5949 "configure"
-#include "confdefs.h"
-
-int main() {
-asm(".fpu neon\n vadd.i8 d0, d0, d0");
-; return 0; }
-EOF
-if { (eval echo configure:5956: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  result="yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  result="no"
-fi
-rm -f conftest*
-  echo "$ac_t"""$result"" 1>&6
-  if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_NEON 1
-EOF
-
-      HAVE_ARM_NEON=1
-
-                  if test -n "$ARM_ARCH"; then
-          if test "$ARM_ARCH" -lt 7; then
-              BUILD_ARM_NEON=0
-          else
-              BUILD_ARM_NEON=1
-          fi
-      fi
-  fi
-
-fi # CPU_ARCH = arm
-
+HAVE_ARM_NEON=0
+BUILD_ARM_NEON=0
+HAVE_ARM_SIMD=0
 
 
 
