--- mozilla/configure.orig	2014-07-22 18:38:00.711218733 -0600
+++ mozilla/configure	2014-07-22 18:40:37.355699369 -0600
@@ -6970,98 +6970,10 @@
 
 
 
-if test "$CPU_ARCH" = "arm"; then
-  echo $ac_n "checking for ARM SIMD support in compiler""... $ac_c" 1>&6
-echo "configure:6976: checking for ARM SIMD support in compiler" >&5
-  # We try to link so that this also fails when
-  # building with LTO.
-  cat > conftest.$ac_ext <<EOF
-#line 6980 "configure"
-#include "confdefs.h"
-
-int main() {
-asm("uqadd8 r1, r1, r2");
-; return 0; }
-EOF
-if { (eval echo configure:6987: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  result="yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  result="no"
-fi
-rm -f conftest*
-  echo "$ac_t"""$result"" 1>&6
-  if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_SIMD ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_SIMD 1
-EOF
-
-      HAVE_ARM_SIMD=1
-  fi
-
-  echo $ac_n "checking ARM version support in compiler""... $ac_c" 1>&6
-echo "configure:7010: checking ARM version support in compiler" >&5
-    ARM_ARCH=`${CC-cc} ${CFLAGS} -dM -E - < /dev/null | sed -n 's/.*__ARM_ARCH_\([0-9][0-9]*\).*/\1/p'`
-  echo "$ac_t"""$ARM_ARCH"" 1>&6
-
-  echo $ac_n "checking for ARM NEON support in compiler""... $ac_c" 1>&6
-echo "configure:7015: checking for ARM NEON support in compiler" >&5
-  # We try to link so that this also fails when
-  # building with LTO.
-  cat > conftest.$ac_ext <<EOF
-#line 7019 "configure"
-#include "confdefs.h"
-
-int main() {
-asm(".fpu neon\n vadd.i8 d0, d0, d0");
-; return 0; }
-EOF
-if { (eval echo configure:7026: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  result="yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  result="no"
-fi
-rm -f conftest*
-  echo "$ac_t"""$result"" 1>&6
-  if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_NEON 1
-EOF
-
-      HAVE_ARM_NEON=1
-
-                  if test -n "$ARM_ARCH"; then
-          if test "$ARM_ARCH" -lt 7; then
-              BUILD_ARM_NEON=
-          else
-              cat >> confdefs.pytmp <<\EOF
-    (''' BUILD_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define BUILD_ARM_NEON 1
-EOF
-
-              BUILD_ARM_NEON=1
-          fi
-      fi
-  fi
-
-fi # CPU_ARCH = arm
-
 
+HAVE_ARM_NEON=0
+BUILD_ARM_NEON=0
+HAVE_ARM_SIMD=0
 
 
 
--- mozilla/js/src/configure.orig	2014-07-22 18:38:00.736218651 -0600
+++ mozilla/js/src/configure	2014-07-22 18:41:43.420479891 -0600
@@ -6153,99 +6153,9 @@
 
 
 
-if test "$CPU_ARCH" = "arm"; then
-  echo $ac_n "checking for ARM SIMD support in compiler""... $ac_c" 1>&6
-echo "configure:6159: checking for ARM SIMD support in compiler" >&5
-  # We try to link so that this also fails when
-  # building with LTO.
-  cat > conftest.$ac_ext <<EOF
-#line 6163 "configure"
-#include "confdefs.h"
-
-int main() {
-asm("uqadd8 r1, r1, r2");
-; return 0; }
-EOF
-if { (eval echo configure:6170: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  result="yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  result="no"
-fi
-rm -f conftest*
-  echo "$ac_t"""$result"" 1>&6
-  if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_SIMD ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_SIMD 1
-EOF
-
-      HAVE_ARM_SIMD=1
-  fi
-
-  echo $ac_n "checking ARM version support in compiler""... $ac_c" 1>&6
-echo "configure:6193: checking ARM version support in compiler" >&5
-    ARM_ARCH=`${CC-cc} ${CFLAGS} -dM -E - < /dev/null | sed -n 's/.*__ARM_ARCH_\([0-9][0-9]*\).*/\1/p'`
-  echo "$ac_t"""$ARM_ARCH"" 1>&6
-
-  echo $ac_n "checking for ARM NEON support in compiler""... $ac_c" 1>&6
-echo "configure:6198: checking for ARM NEON support in compiler" >&5
-  # We try to link so that this also fails when
-  # building with LTO.
-  cat > conftest.$ac_ext <<EOF
-#line 6202 "configure"
-#include "confdefs.h"
-
-int main() {
-asm(".fpu neon\n vadd.i8 d0, d0, d0");
-; return 0; }
-EOF
-if { (eval echo configure:6209: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  result="yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  result="no"
-fi
-rm -f conftest*
-  echo "$ac_t"""$result"" 1>&6
-  if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_NEON 1
-EOF
-
-      HAVE_ARM_NEON=1
-
-                  if test -n "$ARM_ARCH"; then
-          if test "$ARM_ARCH" -lt 7; then
-              BUILD_ARM_NEON=
-          else
-              cat >> confdefs.pytmp <<\EOF
-    (''' BUILD_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define BUILD_ARM_NEON 1
-EOF
-
-              BUILD_ARM_NEON=1
-          fi
-      fi
-  fi
-
-fi # CPU_ARCH = arm
-
-
-
+HAVE_ARM_NEON=0
+BUILD_ARM_NEON=0
+HAVE_ARM_SIMD=0
 
 
 
