--- mozilla/configure.orig	2016-04-19 12:42:57.280811615 -0600
+++ mozilla/configure	2016-04-19 12:44:20.697690476 -0600
@@ -7389,14 +7389,7 @@
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
   if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_SIMD ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_SIMD 1
-EOF
-
-      HAVE_ARM_SIMD=1
+      HAVE_ARM_SIMD=
   fi
 
   echo $ac_n "checking ARM version support in compiler""... $ac_c" 1>&6
@@ -7428,27 +7421,13 @@
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
   if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_NEON 1
-EOF
-
-      HAVE_ARM_NEON=1
+      HAVE_ARM_NEON=
 
                   if test -n "$ARM_ARCH"; then
           if test "$ARM_ARCH" -lt 7; then
               BUILD_ARM_NEON=
           else
-              cat >> confdefs.pytmp <<\EOF
-    (''' BUILD_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define BUILD_ARM_NEON 1
-EOF
-
-              BUILD_ARM_NEON=1
+              BUILD_ARM_NEON=
           fi
       fi
   fi
--- mozilla/js/src/configure.orig	2016-04-07 15:33:39.000000000 -0600
+++ mozilla/js/src/configure	2016-04-19 12:44:20.697690476 -0600
@@ -6578,14 +6578,7 @@
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
   if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_SIMD ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_SIMD 1
-EOF
-
-      HAVE_ARM_SIMD=1
+      HAVE_ARM_SIMD=
   fi
 
   echo $ac_n "checking ARM version support in compiler""... $ac_c" 1>&6
@@ -6617,27 +6610,13 @@
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
   if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_NEON 1
-EOF
-
-      HAVE_ARM_NEON=1
+      HAVE_ARM_NEON=
 
                   if test -n "$ARM_ARCH"; then
           if test "$ARM_ARCH" -lt 7; then
               BUILD_ARM_NEON=
           else
-              cat >> confdefs.pytmp <<\EOF
-    (''' BUILD_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define BUILD_ARM_NEON 1
-EOF
-
-              BUILD_ARM_NEON=1
+              BUILD_ARM_NEON=
           fi
       fi
   fi
