--- mozilla/configure.orig	2015-08-08 14:58:36.593927523 -0600
+++ mozilla/configure	2015-08-08 15:01:37.743857135 -0600
@@ -7121,14 +7121,7 @@
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
   if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_SIMD ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_SIMD 1
-EOF
-
-      HAVE_ARM_SIMD=1
+      HAVE_ARM_SIMD=
   fi
 
   echo $ac_n "checking ARM version support in compiler""... $ac_c" 1>&6
@@ -7160,27 +7153,13 @@
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
   if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_NEON 1
-EOF
-
-      HAVE_ARM_NEON=1
+      HAVE_ARM_NEON=
 
                   if test -n "$ARM_ARCH"; then
           if test "$ARM_ARCH" -lt 7; then
               BUILD_ARM_NEON=
           else
-              cat >> confdefs.pytmp <<\EOF
-    (''' BUILD_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define BUILD_ARM_NEON 1
-EOF
-
-              BUILD_ARM_NEON=1
+              BUILD_ARM_NEON=
           fi
       fi
   fi
--- mozilla/js/src/configure.orig	2015-08-08 14:58:44.998935015 -0600
+++ mozilla/js/src/configure	2015-08-08 15:02:01.643605841 -0600
@@ -6260,14 +6260,7 @@
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
   if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_SIMD ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_SIMD 1
-EOF
-
-      HAVE_ARM_SIMD=1
+      HAVE_ARM_SIMD=
   fi
 
   echo $ac_n "checking ARM version support in compiler""... $ac_c" 1>&6
@@ -6299,27 +6292,13 @@
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
   if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_NEON 1
-EOF
-
-      HAVE_ARM_NEON=1
+      HAVE_ARM_NEON=
 
                   if test -n "$ARM_ARCH"; then
           if test "$ARM_ARCH" -lt 7; then
               BUILD_ARM_NEON=
           else
-              cat >> confdefs.pytmp <<\EOF
-    (''' BUILD_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define BUILD_ARM_NEON 1
-EOF
-
-              BUILD_ARM_NEON=1
+              BUILD_ARM_NEON=
           fi
       fi
   fi
