--- ./mozilla/configure.orig	2015-06-11 19:05:34.195096563 -0600
+++ ./mozilla/configure	2015-06-11 19:06:44.405101697 -0600
@@ -7096,98 +7096,9 @@
 
 
 
-if test "$CPU_ARCH" = "arm"; then
-  echo $ac_n "checking for ARM SIMD support in compiler""... $ac_c" 1>&6
-echo "configure:7102: checking for ARM SIMD support in compiler" >&5
-  # We try to link so that this also fails when
-  # building with LTO.
-  cat > conftest.$ac_ext <<EOF
-#line 7106 "configure"
-#include "confdefs.h"
-
-int main() {
-asm("uqadd8 r1, r1, r2");
-; return 0; }
-EOF
-if { (eval echo configure:7113: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  result="yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  result="no"
-fi
-rm -f conftest*
-  echo "$ac_t"""$result"" 1>&6
-  if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_SIMD ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_SIMD 1
-EOF
-
-      HAVE_ARM_SIMD=1
-  fi
-
-  echo $ac_n "checking ARM version support in compiler""... $ac_c" 1>&6
-echo "configure:7136: checking ARM version support in compiler" >&5
-    ARM_ARCH=`${CC-cc} ${CFLAGS} -dM -E - < /dev/null | sed -n 's/.*__ARM_ARCH_\([0-9][0-9]*\).*/\1/p'`
-  echo "$ac_t"""$ARM_ARCH"" 1>&6
-
-  echo $ac_n "checking for ARM NEON support in compiler""... $ac_c" 1>&6
-echo "configure:7141: checking for ARM NEON support in compiler" >&5
-  # We try to link so that this also fails when
-  # building with LTO.
-  cat > conftest.$ac_ext <<EOF
-#line 7145 "configure"
-#include "confdefs.h"
-
-int main() {
-asm(".fpu neon\n vadd.i8 d0, d0, d0");
-; return 0; }
-EOF
-if { (eval echo configure:7152: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  result="yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  result="no"
-fi
-rm -f conftest*
-  echo "$ac_t"""$result"" 1>&6
-  if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_NEON 1
-EOF
-
-      HAVE_ARM_NEON=1
-
-                  if test -n "$ARM_ARCH"; then
-          if test "$ARM_ARCH" -lt 7; then
-              BUILD_ARM_NEON=
-          else
-              cat >> confdefs.pytmp <<\EOF
-    (''' BUILD_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define BUILD_ARM_NEON 1
-EOF
-
-              BUILD_ARM_NEON=1
-          fi
-      fi
-  fi
-
-fi # CPU_ARCH = arm
-
-
+HAVE_ARM_NEON=0
+BUILD_ARM_NEON=0
+HAVE_ARM_SIMD=0
 
 
 
--- ./mozilla/js/src/configure.orig	2015-06-11 19:05:34.155096560 -0600
+++ ./mozilla/js/src/configure	2015-06-11 19:07:17.065104453 -0600
@@ -6235,99 +6235,9 @@
 
 
 
-if test "$CPU_ARCH" = "arm"; then
-  echo $ac_n "checking for ARM SIMD support in compiler""... $ac_c" 1>&6
-echo "configure:6241: checking for ARM SIMD support in compiler" >&5
-  # We try to link so that this also fails when
-  # building with LTO.
-  cat > conftest.$ac_ext <<EOF
-#line 6245 "configure"
-#include "confdefs.h"
-
-int main() {
-asm("uqadd8 r1, r1, r2");
-; return 0; }
-EOF
-if { (eval echo configure:6252: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  result="yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  result="no"
-fi
-rm -f conftest*
-  echo "$ac_t"""$result"" 1>&6
-  if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_SIMD ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_SIMD 1
-EOF
-
-      HAVE_ARM_SIMD=1
-  fi
-
-  echo $ac_n "checking ARM version support in compiler""... $ac_c" 1>&6
-echo "configure:6275: checking ARM version support in compiler" >&5
-    ARM_ARCH=`${CC-cc} ${CFLAGS} -dM -E - < /dev/null | sed -n 's/.*__ARM_ARCH_\([0-9][0-9]*\).*/\1/p'`
-  echo "$ac_t"""$ARM_ARCH"" 1>&6
-
-  echo $ac_n "checking for ARM NEON support in compiler""... $ac_c" 1>&6
-echo "configure:6280: checking for ARM NEON support in compiler" >&5
-  # We try to link so that this also fails when
-  # building with LTO.
-  cat > conftest.$ac_ext <<EOF
-#line 6284 "configure"
-#include "confdefs.h"
-
-int main() {
-asm(".fpu neon\n vadd.i8 d0, d0, d0");
-; return 0; }
-EOF
-if { (eval echo configure:6291: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  result="yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  result="no"
-fi
-rm -f conftest*
-  echo "$ac_t"""$result"" 1>&6
-  if test "$result" = "yes"; then
-      cat >> confdefs.pytmp <<\EOF
-    (''' HAVE_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define HAVE_ARM_NEON 1
-EOF
-
-      HAVE_ARM_NEON=1
-
-                  if test -n "$ARM_ARCH"; then
-          if test "$ARM_ARCH" -lt 7; then
-              BUILD_ARM_NEON=
-          else
-              cat >> confdefs.pytmp <<\EOF
-    (''' BUILD_ARM_NEON ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define BUILD_ARM_NEON 1
-EOF
-
-              BUILD_ARM_NEON=1
-          fi
-      fi
-  fi
-
-fi # CPU_ARCH = arm
-
-
-
+HAVE_ARM_NEON=0
+BUILD_ARM_NEON=0
+HAVE_ARM_SIMD=0
 
 
 
