--- mozilla/config/autoconf.mk.in.orig	2008-12-05 22:26:33.000000000 +0000
+++ mozilla/config/autoconf.mk.in	2008-12-05 22:27:01.000000000 +0000
@@ -57,14 +57,14 @@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-includedir	= @includedir@/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+includedir	= @includedir@/$(MOZ_APP_NAME)-2.0
 libdir		= @libdir@
 datadir		= @datadir@
 mandir		= @mandir@
-idldir		= @datadir@/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+idldir		= @datadir@/idl/$(MOZ_APP_NAME)-2.0
 
-mozappdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-mredir		= $(libdir)/mre/mre-$(MOZ_APP_VERSION)
+mozappdir	= $(libdir)/$(MOZ_APP_NAME)-2.0
+mredir		= $(libdir)/mre/mre-2.0
 mrelibdir	= $(mredir)/lib
 
 DIST		= $(DEPTH)/dist
