--- mozilla/mail/app/profile/all-thunderbird.js.lang	2006-06-15 17:33:58.000000000 +0000
+++ mozilla/mail/app/profile/all-thunderbird.js	2006-06-15 17:34:50.000000000 +0000
@@ -245,6 +245,8 @@
 pref("font.language.group", "chrome://global/locale/intl.properties");
 pref("intl.menuitems.alwaysappendaccesskeys","chrome://global/locale/intl.properties");
 pref("intl.menuitems.insertseparatorbeforeaccesskeys","chrome://global/locale/intl.properties");
+// Use LANG environment variable to choose locale
+pref("intl.locale.matchOS", true);
 
 pref("signon.rememberSignons",              true);
 pref("signon.expireMasterPassword",         false);
