diff -upr comm-esr31.orig/config/baseconfig.mk comm-esr31/config/baseconfig.mk
--- comm-esr31.orig/config/baseconfig.mk	2014-07-22 09:44:23.000000000 +0300
+++ comm-esr31/config/baseconfig.mk	2014-07-22 09:46:30.000000000 +0300
@@ -1,6 +1,6 @@
 includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+installdir = $(libdir)/$(MOZ_APP_NAME)
 sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
 MOZILLA_SRCDIR = $(topsrcdir)/mozilla
 MOZDEPTH = $(DEPTH)/mozilla
diff -upr comm-esr31.orig/mozilla/config/baseconfig.mk comm-esr31/mozilla/config/baseconfig.mk
--- comm-esr31.orig/mozilla/config/baseconfig.mk	2014-07-22 09:44:22.000000000 +0300
+++ comm-esr31/mozilla/config/baseconfig.mk	2014-07-22 09:46:45.000000000 +0300
@@ -4,7 +4,7 @@
 # whether a normal build is happening or whether the check is running.
 includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+installdir = $(libdir)/$(MOZ_APP_NAME)
 sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
 ifndef TOP_DIST
 TOP_DIST = dist
