diff -ru thunderbird-128.7.0.orig/js/moz.configure thunderbird-128.7.0/js/moz.configure
--- thunderbird-128.7.0.orig/js/moz.configure	2025-02-11 23:28:09.244598770 +0100
+++ thunderbird-128.7.0/js/moz.configure	2025-02-11 23:30:01.066481545 +0100
@@ -1305,7 +1305,7 @@
     return enable_system_icu
 
 
-system_icu = pkg_check_modules("MOZ_ICU", "icu-i18n >= 73.1", when="--with-system-icu")
+system_icu = pkg_check_modules("MOZ_ICU", ["icu-i18n >= 73.1", "icu-uc >= 73.1"], when="--with-system-icu")
 
 
 @depends(enable_system_icu_option)

