--- a/xbmc/cores/VideoRenderers/OverlayRendererGL.cpp	2011-03-05 03:33:37.000000000 +0000
+++ b/xbmc/cores/VideoRenderers/OverlayRendererGL.cpp	2011-11-28 18:20:47.957645004 +0000
@@ -125,7 +125,11 @@
             , o->width * 4
             , &m_u, &m_v
             , GL_RGBA
+#ifndef HAS_GLES            
             , GL_BGRA
+#else
+	    , GL_RGBA
+#endif            
             , rgba);
   free(rgba);
 
@@ -199,7 +203,11 @@
             , o->width * 4
             , &m_u, &m_v
             , GL_RGBA
+#ifndef HAS_GLES            
             , GL_BGRA
+#else
+	    , GL_RGBA
+#endif            
             , rgba + min_x + min_y * o->width);
 
   free(rgba);
