diff --git a/xbmc/interfaces/python/xbmcmodule/xbmcplugin.cpp b/xbmc/interfaces/python/xbmcmodule/xbmcplugin.cpp
index a6623b8..1f88c94 100644
--- a/xbmc/interfaces/python/xbmcmodule/xbmcplugin.cpp
+++ b/xbmc/interfaces/python/xbmcmodule/xbmcplugin.cpp
@@ -68,7 +68,7 @@ namespace PYXBMC
     if (!PyArg_ParseTupleAndKeywords(
       args,
       kwds,
-      (char*)"iOO|bl",
+      (char*)"iOO|bi",
       (char**)keywords,
       &handle,
       &pURL,
@@ -120,7 +120,7 @@ namespace PYXBMC
     if (!PyArg_ParseTupleAndKeywords(
       args,
       kwds,
-      (char*)"iO|l",
+      (char*)"iO|i",
       (char**)keywords,
       &handle,
       &pItems,
