--- build/lib/socketio/handler.py	2016-03-22 19:48:41.844978501 +0000
+++ build/lib/socketio/handler.py.new	2016-03-22 19:48:12.018828806 +0000
@@ -74,7 +74,7 @@
             ("Access-Control-Allow-Origin", self.environ.get('HTTP_ORIGIN', '*')),
             ("Access-Control-Allow-Credentials", "true"),
             ("Access-Control-Allow-Methods", "POST, GET, OPTIONS"),
-            ("Access-Control-Max-Age", 3600),
+            ("Access-Control-Max-Age", "3600"),
             ("Content-Type", "text/plain"),
         ])
         self.result = [data]
