--- build/lib/socketio/transports.py	2016-03-22 19:48:21.948809209 +0000
+++ build/lib/socketio/transports.py.new	2016-03-22 19:48:06.983838743 +0000
@@ -21,7 +21,7 @@
             ("Access-Control-Allow-Origin", "*"),
             ("Access-Control-Allow-Credentials", "true"),
             ("Access-Control-Allow-Methods", "POST, GET, OPTIONS"),
-            ("Access-Control-Max-Age", 3600),
+            ("Access-Control-Max-Age", "3600"),
         ]
         self.handler = handler
         self.config = config
