From c359ae27d1c204cf2594c357f3abf22e343d6ec8 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Mon, 7 Aug 2017 23:10:58 -0600
Subject: [PATCH 1/4] rk3288: set CONFIG_SYS_TEXT_BASE for depthcharge

---
 include/configs/rk3288_common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/rk3288_common.h b/include/configs/rk3288_common.h
index c7757f023f..07e51bafb1 100644
--- a/include/configs/rk3288_common.h
+++ b/include/configs/rk3288_common.h
@@ -27,7 +27,7 @@
 /* Bootrom will load u-boot binary to 0x0 once return from SPL */
 #define CONFIG_SYS_TEXT_BASE		0x00000000
 #else
-#define CONFIG_SYS_TEXT_BASE		0x00100000
+#define CONFIG_SYS_TEXT_BASE		0x02000100
 #endif
 #define CONFIG_SYS_INIT_SP_ADDR		0x00100000
 #define CONFIG_SYS_LOAD_ADDR		0x00800800
-- 
2.14.1

