From 59bfd865fb7d23fd13e089098fda74ade692c5cf Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Wed, 6 Sep 2017 21:29:42 -0600
Subject: [PATCH 7/7] dtoc: python2 compatibility

---
 scripts/Makefile.spl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/Makefile.spl b/scripts/Makefile.spl
index 252f13826d..c7797f0e24 100644
--- a/scripts/Makefile.spl
+++ b/scripts/Makefile.spl
@@ -248,7 +248,7 @@ $(obj)/$(SPL_BIN)-pad.bin: $(obj)/$(SPL_BIN)
 $(obj)/$(SPL_BIN).dtb: dts/dt-spl.dtb FORCE
 	$(call if_changed,copy)
 
-pythonpath = PYTHONPATH=scripts/dtc/pylibfdt
+pythonpath = PYTHONPATH=scripts/dtc/pylibfdt $(PYTHON)
 
 quiet_cmd_dtocc = DTOC C  $@
 cmd_dtocc = $(pythonpath) $(srctree)/tools/dtoc/dtoc -d $(obj)/$(SPL_BIN).dtb -o $@ platdata
-- 
2.18.0

