diff -ur libexif-gtk-0.3.5.orig/libexif-gtk/Makefile.in libexif-gtk-0.3.5/libexif-gtk/Makefile.in
--- libexif-gtk-0.3.5.orig/libexif-gtk/Makefile.in	2007-10-30 12:57:42.000000000 +0000
+++ libexif-gtk-0.3.5/libexif-gtk/Makefile.in	2007-10-30 13:02:47.000000000 +0000
@@ -251,8 +251,7 @@
 	-I$(top_srcdir)/intl				\
 	-I$(top_srcdir)/gtk-extensions			\
 	$(LIBEXIF_GTK_CFLAGS)				\
-	-DG_LOG_DOMAIN=\"libexif\"			\
-	-DGTK_DISABLE_DEPRECATED
+	-DG_LOG_DOMAIN=\"libexif\"
 
 lib_LTLIBRARIES = libexif-gtk.la
 libexif_gtkincludedir = $(includedir)/libexif-gtk
