Index: libs/python/src/converter/builtin_converters.cpp
===================================================================
--- libs/python/src/converter/builtin_converters.cpp	(revision 67279)
+++ libs/python/src/converter/builtin_converters.cpp	(working copy)
@@ -431,7 +431,11 @@
           if (!result.empty())
           {
               int err = PyUnicode_AsWideChar(
+#if PY_VERSION_HEX >= 0x03020000
+                  intermediate
+#else
                   (PyUnicodeObject *)intermediate
+#endif
                 , &result[0]
                 , result.size());
 
