From b29603fa88a614835359168e70b5a44eae8e642c Mon Sep 17 00:00:00 2001
From: Rene Rivera <grafikrobot@gmail.com>
Date: Sat, 26 Sep 2020 18:02:59 -0500
Subject: [PATCH] Fix path to bootstrap for back compat.

fixes #650
--- a/src/engine/startup.cpp
+++ b/src/engine/startup.cpp
@@ -195,7 +195,7 @@ bool b2::startup::bootstrap(FRAME *frame)
     {
         const std::string path{
             b2::paths::normalize(
-                b2_exe_path + "/../../share/boost-build/" + boost_build_jam)};
+                b2_exe_path + "/../../share/boost-build/src/kernel/" + boost_build_jam)};
         if (b2::filesys::is_file(path))
             b2_file_path = path;
     }
