Index: boost/spirit/home/qi/nonterminal/detail/parameterized.hpp
===================================================================
--- boost/spirit/home/qi/nonterminal/detail/parameterized.hpp	(revision 68724)
+++ boost/spirit/home/qi/nonterminal/detail/parameterized.hpp	(revision 68725)
@@ -14,6 +14,7 @@
 
 #include <boost/ref.hpp>
 
+#include <boost/spirit/home/support/handles_container.hpp>
 #include <boost/spirit/home/qi/parser.hpp>
 
 namespace boost { namespace spirit { namespace qi
@@ -59,4 +60,16 @@ namespace boost { namespace spirit { nam
     };
 }}}
 
+namespace boost { namespace spirit { namespace traits
+{
+    ///////////////////////////////////////////////////////////////////////////
+    template <typename Subject, typename Params, typename Attribute
+      , typename Context, typename Iterator>
+    struct handles_container<qi::parameterized_nonterminal<Subject, Params>
+          , Attribute, Context, Iterator>
+      : handles_container<typename remove_const<Subject>::type
+        , Attribute, Context, Iterator> 
+    {};
+}}}
+
 #endif
Index: boost/spirit/home/karma/nonterminal/detail/parameterized.hpp
===================================================================
--- boost/spirit/home/karma/nonterminal/detail/parameterized.hpp	(revision 68724)
+++ boost/spirit/home/karma/nonterminal/detail/parameterized.hpp	(revision 68725)
@@ -14,6 +14,7 @@
 
 #include <boost/ref.hpp>
 
+#include <boost/spirit/home/support/handles_container.hpp>
 #include <boost/spirit/home/karma/generator.hpp>
 
 namespace boost { namespace spirit { namespace karma
@@ -60,4 +61,17 @@ namespace boost { namespace spirit { nam
     };
 }}}
 
+
+namespace boost { namespace spirit { namespace traits
+{
+    ///////////////////////////////////////////////////////////////////////////
+    template <typename Subject, typename Params, typename Attribute
+      , typename Context, typename Iterator>
+    struct handles_container<karma::parameterized_nonterminal<Subject, Params>
+          , Attribute, Context, Iterator>
+      : handles_container<typename remove_const<Subject>::type
+        , Attribute, Context, Iterator> 
+    {};
+}}}
+
 #endif
