diff -up boost/numeric/conversion/converter_policies.hpp\~ boost/numeric/conversion/converter_policies.hpp
--- boost/numeric/conversion/converter_policies.hpp~	2008-10-13 11:00:03.000000000 +0200
+++ boost/numeric/conversion/converter_policies.hpp	2011-07-22 11:46:40.961876274 +0200
@@ -20,6 +20,7 @@
 
 #include "boost/mpl/if.hpp"
 #include "boost/mpl/integral_c.hpp"
+#include "boost/throw_exception.hpp"
 
 namespace boost { namespace numeric
 {
@@ -159,9 +160,9 @@ struct def_overflow_handler
   void operator() ( range_check_result r ) // throw(negative_overflow,positive_overflow)
   {
     if ( r == cNegOverflow )
-      throw negative_overflow() ;
+      boost::throw_exception( negative_overflow() ) ;
     else if ( r == cPosOverflow )
-           throw positive_overflow() ;
+      boost::throw_exception( positive_overflow() ) ;
   }
 } ;
 

Diff finished.  Fri Jul 22 11:46:49 2011
