diff -Nur boost_1_41_0.old/boost/archive/shared_ptr_helper.hpp boost_1_41_0/boost/archive/shared_ptr_helper.hpp
--- boost_1_41_0.old/boost/archive/shared_ptr_helper.hpp	2009-09-20 05:18:42.000000000 +0300
+++ boost_1_41_0/boost/archive/shared_ptr_helper.hpp	2009-12-17 22:20:48.666281335 +0200
@@ -108,7 +108,7 @@
     struct non_polymorphic {
         static const boost::serialization::extended_type_info * 
         get_object_identifier(T & t){
-            return boost::serialization::singleton<
+            return & boost::serialization::singleton<
                 BOOST_DEDUCED_TYPENAME 
                 boost::serialization::type_info_implementation<T>::type
             >::get_const_instance();
