From a06634be9fe056b051e271ad4b83607d4b4ee2fc Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 15 Feb 2015 11:16:26 -0700
Subject: [PATCH] compat for newer xorg

---
 src/armsoc_dumb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/armsoc_dumb.c b/src/armsoc_dumb.c
index 8f29b7a..879d77c 100644
--- a/src/armsoc_dumb.c
+++ b/src/armsoc_dumb.c
@@ -27,6 +27,7 @@
 #include <errno.h>
 #include <unistd.h>
 
+#include <xorg-server.h>
 #include <xf86.h>
 #include <xf86drm.h>
 #include <xf86drmMode.h>
-- 
2.3.0

