From b8d09aae2902c60e89f48a1f45e822b64c9242a8 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 15 Feb 2015 11:16:26 -0700
Subject: [PATCH] compat for newer xorg

---
 src/armsoc_dumb.c     | 1 +
 src/drmmode_display.c | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/src/armsoc_dumb.c b/src/armsoc_dumb.c
index 8f29b7a..879d77c 100644
--- a/src/armsoc_dumb.c
+++ b/src/armsoc_dumb.c
@@ -27,6 +27,7 @@
 #include <errno.h>
 #include <unistd.h>
 
+#include <xorg-server.h>
 #include <xf86.h>
 #include <xf86drm.h>
 #include <xf86drmMode.h>
diff --git a/src/drmmode_display.c b/src/drmmode_display.c
index 85ce56f..1b47632 100644
--- a/src/drmmode_display.c
+++ b/src/drmmode_display.c
@@ -32,6 +32,10 @@
 #include "config.h"
 #endif
 
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
 #include "xf86DDC.h"
 #include "xf86RandR12.h"
 
-- 
2.5.2

