From f83caedccd856db42de89d265b96b912fa02fd69 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Thu, 14 Jun 2018 11:59:02 -0600
Subject: [PATCH 3/3] Initialize XRandR before EnterVT

---
 src/armsoc_driver.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/armsoc_driver.c b/src/armsoc_driver.c
index 7e70f34..5ffb64a 100644
--- a/src/armsoc_driver.c
+++ b/src/armsoc_driver.c
@@ -1105,6 +1105,9 @@ ARMSOCScreenInit(SCREEN_INIT_ARGS_DECL)
 	 */
 	pScrn->vtSema = TRUE;
 
+	/* Do some XRandR initialization. Return value is not useful */
+	(void)xf86CrtcScreenInit(pScreen);
+
 	/* Take over the virtual terminal from the console, set the
 	 * desired mode, etc.:
 	 */
@@ -1113,9 +1116,6 @@ ARMSOCScreenInit(SCREEN_INIT_ARGS_DECL)
 		goto fail6;
 	}
 
-	/* Do some XRandR initialization. Return value is not useful */
-	(void)xf86CrtcScreenInit(pScreen);
-
 	if (!miCreateDefColormap(pScreen)) {
 		ERROR_MSG("Cannot create colormap!");
 		goto fail7;
-- 
2.17.1

