diff -urN a/src/armsoc_driver.c b/src/armsoc_driver.c
--- a/src/armsoc_driver.c	2014-07-29 20:17:16.708546536 -0600
+++ b/src/armsoc_driver.c	2014-07-29 20:17:48.503440926 -0600
@@ -1060,7 +1060,7 @@
 	/* Initialize backing store: */
 	xf86SetBackingStore(pScreen);
 
-	fbdev = xf86GetOptValString(pARMSOC->pOptionInfo,
+	fbdev = (char *)xf86GetOptValString(pARMSOC->pOptionInfo,
 			OPTION_INIT_FROM_FBDEV);
 	if (fbdev && *fbdev != '\0') {
 		if (ARMSOCCopyFB(pScrn, fbdev)) {
diff -urN a/src/drmmode_display.c b/src/drmmode_display.c
--- a/src/drmmode_display.c	2014-07-29 20:17:16.708546536 -0600
+++ b/src/drmmode_display.c	2014-07-29 20:17:52.533427544 -0600
@@ -353,8 +353,8 @@
 	drmmode_crtc->last_good_y = crtc->y;
 	drmmode_crtc->last_good_rotation = crtc->rotation;
 	if (drmmode_crtc->last_good_mode) {
-		if (drmmode_crtc->last_good_mode->name)
-			free(drmmode_crtc->last_good_mode->name);
+		//if (drmmode_crtc->last_good_mode->name)
+		//	free(drmmode_crtc->last_good_mode->name);
 		free(drmmode_crtc->last_good_mode);
 	}
 	drmmode_crtc->last_good_mode = xf86DuplicateMode(&crtc->mode);
