From cef9e11371857a16fda242c179da1d8c5350486a Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sat, 7 Sep 2019 08:29:10 -0600
Subject: [PATCH] 32-bit fix

---
 user.cc | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/user.cc b/user.cc
index d2342aa..95e026e 100644
--- a/user.cc
+++ b/user.cc
@@ -261,14 +261,14 @@ bool initNsFromChild(nsjconf_t* nsjconf) {
 		return false;
 	}
 
-	LOG_D("setgroups(%lu, %s)", groups.size(), groupsString.c_str());
+	LOG_D("setgroups(%lu, %s)", (unsigned long)groups.size(), groupsString.c_str());
 	if (setgroups(groups.size(), groups.data()) == -1) {
 		/* Indicate error if specific groups were requested */
 		if (groups.size() > 0) {
-			PLOG_E("setgroups(%lu, %s) failed", groups.size(), groupsString.c_str());
+			PLOG_E("setgroups(%lu, %s) failed", (unsigned long)groups.size(), groupsString.c_str());
 			return false;
 		}
-		PLOG_D("setgroups(%lu, %s) failed", groups.size(), groupsString.c_str());
+		PLOG_D("setgroups(%lu, %s) failed", (unsigned long)groups.size(), groupsString.c_str());
 	}
 
 	if (!setResUid(nsjconf->uids[0].inside_id)) {
-- 
2.23.0

