commit 0ffabfea6c190d28e25587c95e282eb777b3feb2
Author: Phil Endecott <git@chezphil.org>
Date:   Mon Oct 31 21:21:41 2011 +0000

    Return ERESTARTSYS from IPU_GET_EVENT

Index: git/drivers/mxc/ipu/ipu_device.c
===================================================================
--- git.orig/drivers/mxc/ipu/ipu_device.c	2012-01-24 15:34:37.000000000 +0100
+++ git/drivers/mxc/ipu/ipu_device.c	2012-01-31 16:21:19.599303457 +0100
@@ -411,8 +411,10 @@
 
 			r = get_events(&info);
 			if (r == -1) {
-				wait_event_interruptible_timeout(waitq,
-						(pending_events != 0), 2 * HZ);
+				if (wait_event_interruptible_timeout(waitq,
+						(pending_events != 0), 2 * HZ)) < 0) {
+					return -ERESTARTSYS;
+				}
 				r = get_events(&info);
 			}
 			ret = -1;
