--- a/daemon/daemon.go	2014-08-28 21:22:18.805001026 -0400
+++ b/daemon/daemon.go	2014-08-28 21:23:52.030001071 -0400
@@ -1095,10 +1095,6 @@ func (daemon *Daemon) ImageGetCached(img
 }
 
 func checkKernelAndArch() error {
-	// Check for unsupported architectures
-	if runtime.GOARCH != "amd64" {
-		return fmt.Errorf("The Docker runtime currently only supports amd64 (not %s). This will change in the future. Aborting.", runtime.GOARCH)
-	}
 	// Check for unsupported kernel versions
 	// FIXME: it would be cleaner to not test for specific versions, but rather
 	// test for specific functionalities.
