From 204b0076d78108a11e0b60c0dc3a62bb8a6472b6 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Wed, 7 May 2014 15:44:57 -0600
Subject: [PATCH 2/3] Fix btrfs detection on ARM

---
 daemon/graphdriver/btrfs/btrfs.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/daemon/graphdriver/btrfs/btrfs.go b/daemon/graphdriver/btrfs/btrfs.go
index 4d19553..0b437b4 100644
--- a/daemon/graphdriver/btrfs/btrfs.go
+++ b/daemon/graphdriver/btrfs/btrfs.go
@@ -30,7 +30,7 @@ func Init(home string) (graphdriver.Driver, error) {
 		return nil, err
 	}
 
-	if buf.Type != 0x9123683E {
+	if buf.Type != -1859950530 {
 		return nil, fmt.Errorf("%s is not a btrfs filesystem", rootdir)
 	}
 
-- 
1.9.0

