From 88fd97ad84730327a5f56dc70af6b71547090c2c Mon Sep 17 00:00:00 2001
From: Vladimir Pouzanov <farcaller@gmail.com>
Date: Wed, 18 Jun 2014 21:16:41 +0100
Subject: [PATCH] Fix btrfs detection on ARM

---
 daemon/graphdriver/driver.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/daemon/graphdriver/driver.go b/daemon/graphdriver/driver.go
index 4536489..bd9e6df 100644
--- a/daemon/graphdriver/driver.go
+++ b/daemon/graphdriver/driver.go
@@ -10,10 +10,10 @@ import (
 	"github.com/dotcloud/docker/pkg/mount"
 )

-type FsMagic uint64
+type FsMagic int32

 const (
-	FsMagicBtrfs = FsMagic(0x9123683E)
+	FsMagicBtrfs = FsMagic(-1859950530)
 	FsMagicAufs  = FsMagic(0x61756673)
 )

--
1.8.5.2

