From 23c1b0f9a91dcf3c7dc83dc0af44826cb8e96562 Mon Sep 17 00:00:00 2001
From: Vladimir Pouzanov <farcaller@gmail.com>
Date: Sat, 22 Feb 2014 22:25:10 +0000
Subject: [PATCH] Golang sucks so much sometimes

---
 graphdriver/btrfs/btrfs.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/graphdriver/btrfs/btrfs.go b/graphdriver/btrfs/btrfs.go
index 592e058..03faf85 100644
--- a/graphdriver/btrfs/btrfs.go
+++ b/graphdriver/btrfs/btrfs.go
@@ -30,7 +30,7 @@ func Init(home string) (graphdriver.Driver, error) {
 		return nil, err
 	}

-	if buf.Type != 0x9123683E {
+	if buf.Type != -1859950530 {
 		return nil, fmt.Errorf("%s is not a btrfs filesystem", rootdir)
 	}

--
1.9.0
