diff --git a/engine/engine.go b/engine/engine.go
index ec880b9..f03d657 100644
--- a/engine/engine.go
+++ b/engine/engine.go
@@ -60,7 +60,7 @@ func (eng *Engine) Register(name string, handler Handler) error {
 // behavior.
 func New(root string) (*Engine, error) {
 	// Check for unsupported architectures
-	if runtime.GOARCH != "amd64" {
+	if runtime.GOARCH != "amd64" && runtime.GOARCH != "arm" {
 		return nil, fmt.Errorf("The docker runtime currently only supports amd64 (not %s). This will change in the future. Aborting.", runtime.GOARCH)
 	}
 	// Check for unsupported kernel versions
diff --git a/execdriver/lxc/lxc_init_linux.go b/execdriver/lxc/lxc_init_linux.go
index 7288f58..66871cd 100644
--- a/execdriver/lxc/lxc_init_linux.go
+++ b/execdriver/lxc/lxc_init_linux.go
@@ -1,4 +1,4 @@
-// +build amd64
+// +build amd64 arm
 
 package lxc
 
diff --git a/execdriver/lxc/lxc_init_unsupported.go b/execdriver/lxc/lxc_init_unsupported.go
index d68cb91..47a40c0 100644
--- a/execdriver/lxc/lxc_init_unsupported.go
+++ b/execdriver/lxc/lxc_init_unsupported.go
@@ -1,4 +1,4 @@
-// +build !linux !amd64
+// +build !linux !amd64,!arm
 
 package lxc
 
diff --git a/graphdriver/btrfs/btrfs.go b/graphdriver/btrfs/btrfs.go
index 592e058..e6570b0 100644
--- a/graphdriver/btrfs/btrfs.go
+++ b/graphdriver/btrfs/btrfs.go
@@ -1,4 +1,5 @@
-// +build linux,amd64
+// +build linux
+// +build amd64 arm
 
 package btrfs
 
@@ -30,7 +31,7 @@ func Init(home string) (graphdriver.Driver, error) {
 		return nil, err
 	}
 
-	if buf.Type != 0x9123683E {
+	if uint32(buf.Type) != 0x9123683E {
 		return nil, fmt.Errorf("%s is not a btrfs filesystem", rootdir)
 	}
 
diff --git a/graphdriver/btrfs/dummy_unsupported.go b/graphdriver/btrfs/dummy_unsupported.go
index 6c44615..4613bde 100644
--- a/graphdriver/btrfs/dummy_unsupported.go
+++ b/graphdriver/btrfs/dummy_unsupported.go
@@ -1,3 +1,3 @@
-// +build !linux !amd64
+// +build !linux !amd64,!arm
 
 package btrfs
diff --git a/pkg/graphdb/conn_linux.go b/pkg/graphdb/conn_linux.go
index 7a1ab8c..893b806 100644
--- a/pkg/graphdb/conn_linux.go
+++ b/pkg/graphdb/conn_linux.go
@@ -1,4 +1,4 @@
-// +build amd64
+// +build amd64 arm
 
 package graphdb
 
diff --git a/pkg/graphdb/conn_unsupported.go b/pkg/graphdb/conn_unsupported.go
index c2d6025..1da959f 100644
--- a/pkg/graphdb/conn_unsupported.go
+++ b/pkg/graphdb/conn_unsupported.go
@@ -1,4 +1,4 @@
-// +build !linux !amd64
+// +build !linux !amd64,!arm
 
 package graphdb
 
diff --git a/pkg/mount/flags_linux.go b/pkg/mount/flags_linux.go
index b7124b1..2e9ef05 100644
--- a/pkg/mount/flags_linux.go
+++ b/pkg/mount/flags_linux.go
@@ -1,4 +1,4 @@
-// +build amd64
+// +build amd64 arm
 
 package mount
 
diff --git a/pkg/mount/flags_unsupported.go b/pkg/mount/flags_unsupported.go
index c894efe..5688fdd 100644
--- a/pkg/mount/flags_unsupported.go
+++ b/pkg/mount/flags_unsupported.go
@@ -1,4 +1,4 @@
-// +build !linux !amd64
+// +build !linux !amd64,!arm
 
 package mount
 
diff --git a/pkg/mount/mounter_linux.go b/pkg/mount/mounter_linux.go
index 70b7798..1d72009 100644
--- a/pkg/mount/mounter_linux.go
+++ b/pkg/mount/mounter_linux.go
@@ -1,4 +1,4 @@
-// +build amd64
+// +build amd64 arm
 
 package mount
 
diff --git a/pkg/mount/mounter_unsupported.go b/pkg/mount/mounter_unsupported.go
index ee27b35..bf5f5cd 100644
--- a/pkg/mount/mounter_unsupported.go
+++ b/pkg/mount/mounter_unsupported.go
@@ -1,4 +1,4 @@
-// +build !linux !amd64
+// +build !linux !amd64,!arm
 
 package mount
 
diff --git a/pkg/netlink/netlink_linux.go b/pkg/netlink/netlink_linux.go
index 0ea5b4d..bb38427 100644
--- a/pkg/netlink/netlink_linux.go
+++ b/pkg/netlink/netlink_linux.go
@@ -1,4 +1,4 @@
-// +build amd64
+// +build amd64 arm
 
 package netlink
 
diff --git a/pkg/netlink/netlink_unsupported.go b/pkg/netlink/netlink_unsupported.go
index cd796b3..96721a9 100644
--- a/pkg/netlink/netlink_unsupported.go
+++ b/pkg/netlink/netlink_unsupported.go
@@ -1,4 +1,4 @@
-// +build !linux !amd64
+// +build !linux !amd64,!arm
 
 package netlink
 
diff --git a/reflink_copy_linux.go b/reflink_copy_linux.go
index 74a0cb9..8c1dfea 100644
--- a/reflink_copy_linux.go
+++ b/reflink_copy_linux.go
@@ -1,4 +1,4 @@
-// +build amd64
+// +build amd64 arm
 
 package docker
 
diff --git a/reflink_copy_unsupported.go b/reflink_copy_unsupported.go
index 271ed01..67ba8fc 100644
--- a/reflink_copy_unsupported.go
+++ b/reflink_copy_unsupported.go
@@ -1,4 +1,4 @@
-// +build !linux !amd64
+// +build !linux !amd64,!arm
 
 package docker
 
diff --git a/utils/uname_linux.go b/utils/uname_linux.go
index 2f4afb4..ac8c1cb 100644
--- a/utils/uname_linux.go
+++ b/utils/uname_linux.go
@@ -1,4 +1,4 @@
-// +build amd64
+// +build amd64 arm
 
 package utils
 
diff --git a/utils/uname_unsupported.go b/utils/uname_unsupported.go
index 57b82ec..b9b0207 100644
--- a/utils/uname_unsupported.go
+++ b/utils/uname_unsupported.go
@@ -1,4 +1,4 @@
-// +build !linux !amd64
+// +build !linux !amd64,!arm
 
 package utils
 
