--- source/linux/settings.cpp.old	2008-10-23 05:45:58.000000000 +0200
+++ source/linux/settings.cpp	2008-12-04 23:17:59.000000000 +0100
@@ -16,7 +16,7 @@
 using namespace LinuxNst;
 
 #define READ_SETTING(x) fread(&x, sizeof(x), 1, f)
-#define WRITE_SETTING(x) if (fwrite(&x, sizeof(x), 1, f) != sizeof(x)) std::cout << "Error writing setting!\n"
+#define WRITE_SETTING(x) if (fwrite(&x, sizeof(x), 1, f) != 1) std::cout << "Error writing setting!\n"
 
 static int rates[4] = { 11025, 22050, 44100, 48000 };
 
