diff -u sd_fuse/sd_fusing.sh sd_fuse/sd_fusing.sh
--- sd_fuse/sd_fusing.sh	2020-10-24 13:22:06.000000000 +0200
+++ sd_fuse/sd_fusing.sh	2020-12-15 10:59:14.400405631 +0100
@@ -18,9 +18,8 @@
 fi
 [ ! -f "$UBOOT" ] && abort "error: $UBOOT is not exist"
 
-sudo dd if="$UBOOT" of="$1" conv=fsync,notrunc bs=512 seek=1
+dd if="$UBOOT" of="$1" conv=fsync,notrunc bs=512 seek=1
 
 sync
 
-sudo eject "$1"
-echo "Finished."
+echo "Successfully wrote U-Boot to $1"
