diff -u include/configs/odroid-g12-common.h include/configs/odroid-g12-common.h
--- include/configs/odroid-g12-common.h	2020-10-24 13:22:06.000000000 +0200
+++ include/configs/odroid-g12-common.h	2020-12-15 22:16:15.250716582 +0100
@@ -96,7 +96,7 @@
 
 #define ENV_BOOT_ORDER_DEFAULT			"boot_order=mmc rawimage usb pxe spi\0"
 
-#define ENV_BOOTSCRIPTS_DEFAULT			"boot_scripts=boot.ini boot.scr\0"
+#define ENV_BOOTSCRIPTS_DEFAULT			"boot_scripts=/boot/boot.ini\0"
 
 #define ENV_BOOT_ATTEMPT_DEFAULT			\
 	"boot_attempt="					\
@@ -347,7 +347,7 @@
 #define CONFIG_DDR_PLL_BYPASS			0 //0:disable, 1:enable. ddr pll bypass function
 
 /* storage: emmc/nand/sd */
-#define CONFIG_ENV_IS_IN_MMC			1
+#define CONFIG_ENV_IS_NOWHERE			1
 #define 	CONFIG_ENV_OVERWRITE
 #define 	CONFIG_CMD_SAVEENV
 /* fixme, need fix*/
@@ -542,6 +542,11 @@
 #define CONFIG_CMD_UNZIP			1
 #define CONFIG_LZMA				1
 
+/* Arch Linux ARM */
+#define CONFIG_IDENT_STRING		" Arch Linux ARM"
+#define CONFIG_CMD_PART
+#define CONFIG_PARTITION_UUIDS
+
 /* other functions */
 #define CONFIG_NEED_BL301			1
 #define CONFIG_NEED_BL32			1
