From d072eef919589311595f482131e4fffef76b47a3 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Thu, 7 Mar 2019 19:31:55 -0700
Subject: [PATCH 2/2] arch linux arm modifications

---
 include/configs/odroid-g12-common.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/configs/odroid-g12-common.h b/include/configs/odroid-g12-common.h
index ca674a8f73..f732895071 100644
--- a/include/configs/odroid-g12-common.h
+++ b/include/configs/odroid-g12-common.h
@@ -459,6 +459,11 @@
 #define CONFIG_CMD_BDI				1
 #define CONFIG_CMD_FS_GENERIC			1
 
+/* Arch Linux ARM */
+#define CONFIG_IDENT_STRING			" Arch Linux ARM"
+#define CONFIG_CMD_PART
+#define CONFIG_PARTITION_UUIDS
+
 /*file system*/
 #define CONFIG_DOS_PARTITION			1
 #define CONFIG_EFI_PARTITION			1
-- 
2.20.1

