diff -urN a/Makefile.defs b/Makefile.defs
--- a/Makefile.defs	2014-07-02 07:52:34.000000000 -0600
+++ b/Makefile.defs	2014-07-27 20:14:16.713085258 -0600
@@ -84,7 +84,7 @@
 endif
 
 ARCH := $(shell $(GETARCH) |sed -e s/i.86/i386/ -e s/sun4u/sparc64/  \
-			-e s/armv4l/arm/ -e "s/Power Macintosh/ppc/" \
+			-e s/armv4l/arm/ -e s/armv5tel/arm/ -e s/armv6l/arm/ -e s/armv7l/arm/ -e "s/Power Macintosh/ppc/" \
 			-e s/sun4v/sparc64/ -e "s/cobalt/mips2/" \
 			-e s/amd64/x86_64/ )
 
@@ -943,7 +943,7 @@
 					-Wall   
 			#if gcc 4.x+
 ifeq			($(CC_CLASS), 4.x)
-					CFLAGS+=-mcpu=strongarm1100 -minline-all-stringops \
+					CFLAGS+=-minline-all-stringops \
 							-ftree-vectorize
 else
 			#if gcc 3.0+
