diff -Naur mozilla.orig/security/nss/cmd/shlibsign/sign.sh mozilla/security/nss/cmd/shlibsign/sign.sh
--- mozilla.orig/security/nss/cmd/shlibsign/sign.sh	2009-05-02 18:41:54.000000000 +0200
+++ mozilla/security/nss/cmd/shlibsign/sign.sh	2015-06-19 17:46:53.079361603 +0200
@@ -45,7 +45,7 @@
     SHLIB_PATH=${1}/lib:${4}:$SHLIB_PATH
     export SHLIB_PATH
     LD_LIBRARY_PATH=${1}/lib:${4}:$LD_LIBRARY_PATH
-    export LD_LIBRARY_PATH
+    #export LD_LIBRARY_PATH
     DYLD_LIBRARY_PATH=${1}/lib:${4}:$DYLD_LIBRARY_PATH
     export DYLD_LIBRARY_PATH
     LIBRARY_PATH=${1}/lib:${4}:$LIBRARY_PATH
