--- src/mozilla/gfx/src/gtk/nsFontMetricsXft.cpp.orig	2009-05-02 18:42:41.000000000 +0200
+++ src/mozilla/gfx/src/gtk/nsFontMetricsXft.cpp	2013-12-05 17:41:02.125268000 +0100
@@ -67,8 +67,8 @@
 #include "nsFontConfigUtils.h"
 
 #include <gdk/gdkx.h>
-#include <freetype/tttables.h>
-#include <freetype/freetype.h>
+#include <freetype2/freetype/tttables.h>
+#include <freetype2/freetype/freetype.h>
 
 #define FORCE_PR_LOG
 #include "prlog.h"
