--- intl/unicharutil/util/nsUnicharUtils.h	2011-05-10 17:02:12.472642196 +0200
+++ intl/unicharutil/util/nsUnicharUtils.h	2011-05-10 17:02:50.656034735 +0200
@@ -64,6 +64,9 @@
     : public nsStringComparator
   {
     public:
+      nsCaseInsensitiveStringComparator()
+      {
+      }
       virtual int operator()( const PRUnichar*, const PRUnichar*, PRUint32 aLength ) const;
       virtual int operator()( PRUnichar, PRUnichar ) const;
   };
--- toolkit/xre/nsAppRunner.cpp	                2011-05-10 17:26:05.853608464 +0200
+++ toolkit/xre/nsAppRunner.cpp	                2011-05-10 17:26:15.079221177 +0200
@@ -491,6 +491,7 @@
                      
 {
 public:
+  nsXULAppInfo() {}
   NS_DECL_ISUPPORTS_INHERITED
   NS_DECL_NSIXULAPPINFO
   NS_DECL_NSIXULRUNTIME
