--- .mozconfig	2009-04-26 16:19:33.000000000 +0100
+++ .mozconfig	1970-01-01 10:14:21.000000000 +0100
@@ -1,85 +1,80 @@
-# Pulling Composer from the 1.8 Mozilla branch (cvs)
-#$ cvs -d :pserver:anonymous@cvs-mirror.mozilla.org:/cvsroot co -r MOZILLA_1_8_BRANCH mozilla/client.mk
-#$ cvs -d :pserver:anonymous@cvs-mirror.mozilla.org:/cvsroot co -r FIREFOX_2_0_0_20_RELEASE mozilla/client.mk
-#$ cd mozilla
-#$ make -f client.mk checkout MOZ_CO_PROJECT=composer
-#$ find . -name CVS -prune -exec rm -rf {} \;
-#$ find . -name .cvsignore -prune -exec rm -rf {} \;
-
-# Building Composer on the 1.8 Mozilla branch
-# Note: on Ubuntu 8.04, requires libgtk2-dev, libxt-dev, libidl-dev
-#                                + dbus-glib-1-dev, curl (for Mozilla 1.9+)
-#$ make -f client.mk build
-#
+# Build configuration script for Debian
+
+# build in a separate directory (optional)
+#mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/../obj-@CONFIG_GUESS@
 
-export BUILD_OFFICIAL=1
 export MOZILLA_OFFICIAL=1
+export BUILD_OFFICIAL=1
+
+export MOZ_STANDALONE_COMPOSER=1
+mk_add_options MOZ_STANDALONE_COMPOSER=1
+
+# if you want don't want to use gcc and g or want to use
+# a specific version of the compilers, specify it in the
+# two following lines and uncomment them
+#export CC="gcc-3.0"
+#export CXX="g-3.0"
+
+# Uncomment the 3 following lines on Debian
+#export OS_LIBS="-lc -lpthread"
+#export CFLAGS="-DDEBIAN"
+#export CXXFLAGS="-DDEBIAN"
+mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/../obj-kompozer
 
-mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/../obj-@CONFIG_GUESS@
-#mk_add_options MOZ_CO_PROJECT=composer
+# build the standalone composer, obviously
+mk_add_options MOZ_CO_PROJECT=composer
 ac_add_options --enable-application=composer
 
+# change these two line to build in debug mode
 ac_add_options --enable-optimize 
+ac_add_options --with-pthreads
 ac_add_options --disable-debug
-#ac_add_options --disable-optimize 
-#ac_add_options --enable-debug
-
-#ac_add_options --disable-shared
-#ac_add_options --enable-static
-#ac_add_options --disable-libxul 
-#ac_add_options --disable-xpfe-components
-ac_add_options --enable-xpfe-components
-ac_add_options --disable-tests
-ac_add_options --disable-installer
-
-#ac_add_options --disable-libxul
 
-#ac_add_options --disable-crashreporter 
+# adapt the following lines according to your distribution
 
-#ac_add_options --enable-optimize 
-#ac_add_options --enable-default-toolkit=gtk2
-##ac_add_options --enable-default-toolkit=cairo-gtk2
-#ac_add_options --enable-xft
-#ac_add_options --disable-canvas
-##ac_add_options --disable-cairo
-
-#ac_add_options --enable-extensions="default inspector venkman"
-ac_add_options --enable-extensions="default spellcheck"
-#ac_add_options --enable-inspector-apis
-
-##ac_add_options  --enable-extensions=wallet,xml-rpc,xmlextras,pref,universalchardet,spellcheck
-##ac_add_options  --enable-extensions=xml-rpc,xmlextras,pref,universalchardet
-#ac_add_options --disable-extensions
-## (now by default) ac_add_options  --enable-necko-protocols=http,ftp,file,jar,viewsource,res,data
-ac_add_options  --enable-necko-protocols=http,ftp,file,jar,viewsource,res,data
-##ac_add_options  --enable-mathml
-#ac_add_options  --enable-svg
-
-#ac_add_options --disable-installer
-#ac_add_options --disable-javaxpcom
-#ac_add_options --disable-activex
-#ac_add_options --disable-activex-scripting
+ac_add_options --prefix=/usr
+ac_add_options --libdir=/usr/lib
+ac_add_options --disable-tests
+ac_add_options --enable-necko-protocols=http,ftp,file,jar,viewsource,res,data
 
-# Ubuntu 7.10 default flags: https://wiki.mozilla.org/Linux/Compiler_Options
 ac_add_options --enable-default-toolkit=gtk2
 ac_add_options --enable-xft
 ac_add_options --enable-pango
 ac_add_options --enable-postscript
 ac_add_options --disable-xprint
 
-ac_add_options --enable-mathml
+#ac_add_options --enable-mathml
 ac_add_options --enable-svg
-ac_add_options --enable-svg-renderer=cairo
+#ac_add_options --enable-svg-renderer=cairo
 ac_add_options --enable-system-cairo
 ac_add_options --enable-canvas
 
-ac_add_options --with-system-png=/usr
-ac_add_options --with-system-jpeg=/usr
-ac_add_options --with-system-zlib=/usr
+ac_add_options --with-system-png
+ac_add_options --with-system-jpeg
+ac_add_options --with-system-zlib
 
 ac_add_options --enable-gnomevfs
-ac_add_options --enable-xinerama
 ac_add_options --enable-single-profile
 ac_add_options --disable-profilesharing
 ac_add_options --enable-system-myspell
 
+ac_add_options --disable-installer
+ac_add_options --disable-xpfe-components
+ac_add_options --with-default-mozilla-five-home=/usr/lib/kompozer
+ac_add_options --with-distribution-id=aur.archlinux.org
+ac_add_options --without-system-nspr
+ac_add_options --without-system-nss
+
+#ac_add_options --disable-elf-dynstr-gc
+#ac_add_options --disable-gtktest
+ac_add_options --enable-strip
+#ac_add_options --disable-strip-lib
+ac_add_options --disable-updater
+
+ac_add_options --enable-xinerama
+ac_add_options --enable-xpcom-fastload        
+
+ac_cv_visibility_pragma=no
+
+
+
