From 43b4b7e0dbfab08cbbf3c54bbc5816153450363c Mon Sep 17 00:00:00 2001
From: John Galt <centromere@users.noreply.github.com>
Date: Thu, 2 Jun 2016 11:35:08 -0700
Subject: [PATCH] Added x86_64 constraint for rdrand

---
 cryptonite.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cryptonite.cabal b/cryptonite.cabal
index dcdfdf8..f7092fe 100644
--- a/cryptonite.cabal
+++ b/cryptonite.cabal
@@ -241,7 +241,7 @@ Library
   if arch(x86_64)
     CPP-options: -DARCH_X86_64
 
-  if flag(support_rdrand)
+  if flag(support_rdrand) && arch(x86_64)
     CPP-options:    -DSUPPORT_RDRAND
     Other-modules:  Crypto.Random.Entropy.RDRand
     c-sources:      cbits/cryptonite_rdrand.c
-- 
2.8.3

