--- dbus-qt3/connection.cpp	2005-04-18 05:19:17.000000000 -0600
+++ dbus-qt3/connection.cpp	2005-04-19 14:35:07.000000000 -0600
@@ -154,6 +154,7 @@
 
 void* Connection::virtual_hook( int, void*  )
 {
+  return (void *)NULL;
 }
 
 void Connection::dbus_connection_setup_with_qt_main (DBusConnection *connection)
--- dbus-qt3/message.cpp	2005-04-18 05:19:17.000000000 -0600
+++ dbus-qt3/message.cpp	2005-04-19 14:55:32.000000000 -0600
@@ -359,6 +359,7 @@
 Message Message::operator=( const Message& other )
 {
   //FIXME: ref the other.d->msg instead of copying it?
+  return (Message &)(d->msg);
 }
 /**
  * Destructs message.
@@ -508,42 +509,49 @@
   const dbus_bool_t right_size_bool = b;
   dbus_message_append_args( d->msg, DBUS_TYPE_BOOLEAN, &right_size_bool,
                             DBUS_TYPE_INVALID );
+  return (Message &)(d->msg);
 }
 
 Message& Message::operator<<( Q_INT8 byte )
 {
   dbus_message_append_args( d->msg, DBUS_TYPE_BYTE, &byte,
                             DBUS_TYPE_INVALID );
+  return (Message &)(d->msg);
 }
 
 Message& Message::operator<<( Q_INT32 num )
 {
   dbus_message_append_args( d->msg, DBUS_TYPE_INT32, &num,
                             DBUS_TYPE_INVALID );
+  return (Message &)(d->msg);
 }
 
 Message& Message::operator<<( Q_UINT32 num )
 {
   dbus_message_append_args( d->msg, DBUS_TYPE_UINT32, &num,
                             DBUS_TYPE_INVALID );
+  return (Message &)(d->msg);
 }
 
 Message& Message::operator<<( Q_INT64 num )
 {
   dbus_message_append_args( d->msg, DBUS_TYPE_INT64, &num,
                             DBUS_TYPE_INVALID );
+  return (Message &)(d->msg);
 }
 
 Message& Message::operator<<( Q_UINT64 num )
 {
   dbus_message_append_args( d->msg, DBUS_TYPE_UINT64, &num,
                             DBUS_TYPE_INVALID );
+  return (Message &)(d->msg);
 }
 
 Message& Message::operator<<( double num )
 {
   dbus_message_append_args( d->msg, DBUS_TYPE_DOUBLE, &num,
                             DBUS_TYPE_INVALID );
+  return (Message &)(d->msg);
 }
 
 Message& Message::operator<<( const QString& str )
@@ -551,11 +559,13 @@
   const char *u = str.utf8();
   dbus_message_append_args( d->msg, DBUS_TYPE_STRING, &u,
                             DBUS_TYPE_INVALID );
+  return (Message &)(d->msg);
 }
 
 Message& Message::operator<<( const QVariant& custom )
 {
   //FIXME: imeplement
+  return (Message &)(d->msg);
 }
 
 }

