diff -urN a/include/configs/udoo.h b/include/configs/udoo.h
--- a/include/configs/udoo.h	2014-05-18 16:50:00.204581579 -0600
+++ b/include/configs/udoo.h	2014-05-18 17:03:48.701528254 -0600
@@ -37,8 +37,8 @@
 
 /* SATA Configs */
 
+#ifdef CONFIG_MX6Q
 #define CONFIG_CMD_SATA
-#ifdef CONFIG_CMD_SATA
 #define CONFIG_DWC_AHSATA
 #define CONFIG_SYS_SATA_MAX_DEVICE	1
 #define CONFIG_DWC_AHSATA_PORT_ID	0
@@ -98,7 +98,13 @@
 #define CONFIG_CMD_FS_GENERIC
 #define CONFIG_DOS_PARTITION
 
+#define CONFIG_IDENT_STRING " Arch Linux ARM"
+
+#ifdef CONFIG_MX6Q
 #define CONFIG_DEFAULT_FDT_FILE		"imx6q-udoo.dtb"
+#else
+#define CONFIG_DEFAULT_FDT_FILE		"imx6dl-udoo.dtb"
+#endif
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"script=boot.scr\0" \
@@ -112,11 +118,11 @@
 	"boot_fdt=try\0" \
 	"ip_dyn=yes\0" \
 	"optargs=\0" \
-	"video=\0" \
+	"video=mxcfb0:dev=hdmi,1920x1080M@60,bpp=32\0" \
+	"memory=mem=1024M\0" \
 	"mmcdev=0\0" \
 	"mmcpart=1\0" \
-	"mmcroot=/dev/mmcblk0p2 ro\0" \
-	"mmcrootfstype=ext4 rootwait\0" \
+	"mmcroot=/dev/mmcblk0p1 rw rootwait\0" \
 	"update_sd_firmware_filename=u-boot.imx\0" \
 	"update_sd_firmware=" \
 		"if test ${ip_dyn} = yes; then " \
@@ -134,35 +140,24 @@
 	"mmcargs=setenv bootargs console=${console},${baudrate} " \
 		"${optargs} " \
 		"root=${mmcroot} " \
-		"rootfstype=${mmcrootfstype} " \
+		"${memory} " \
 		"video=${video}\0" \
-	"loadbootenv=load mmc ${mmcdev}:${mmcpart} ${loadaddr} uEnv.txt\0" \
+	"loadbootenv=load mmc ${mmcdev}:${mmcpart} ${loadaddr} /boot/uEnv.txt\0" \
 	"importbootenv=echo Importing environment from mmc (uEnv.txt)...; " \
 		"env import -t $loadaddr $filesize\0" \
 	"loadbootscript=" \
 		"fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
-	"loadimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} zImage\0" \
-	"loadzimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} zImage\0" \
-	"loadfdt=load mmc ${mmcdev}:${mmcpart} ${fdt_addr} /dtbs/${fdt_file}\0" \
-	"mmcboot=echo Booting from mmc ...; " \
+	"loaduimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} /boot/uImage\0" \
+	"loadzimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} /boot/zImage\0" \
+	"loadfdt=load mmc ${mmcdev}:${mmcpart} ${fdt_addr} /boot/dtbs/${fdt_file}\0" \
+	"mmcbootm=echo Booting from mmc ...; " \
+		"run mmcargs; " \
+		"bootm ${loadaddr};\0" \
+	"mmcbootz=echo Booting from mmc ...; " \
 		"run mmcargs; " \
 		"bootz ${loadaddr} - ${fdt_addr};\0" \
-	"mmcbootdefault=echo Booting from mmc ...; " \
-		"if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
-			"if run loadfdt; then " \
-				"bootz ${loadaddr} - ${fdt_addr}; " \
-			"else " \
-				"if test ${boot_fdt} = try; then " \
-					"bootz ${loadaddr}; " \
-				"else " \
-					"echo WARN: Cannot load the DT; " \
-				"fi; " \
-			"fi; " \
-		"else " \
-			"bootz ${loadaddr}; " \
-		"fi;\0" \
 	"netargs=setenv bootargs console=${console},${baudrate} " \
 		"root=/dev/nfs " \
 	"ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0" \
@@ -200,10 +195,14 @@
 			"echo Running uenvcmd ...;" \
 			"run uenvcmd;" \
 		"fi;" \
-		"echo Running default loadimage ...;" \
-		"if run loadimage; then " \
+		"echo Running default loaduimage ...;" \
+		"if run loaduimage; then " \
+			"run mmcbootm;" \
+		"fi;" \
+		"echo Running default loadzimage ...;" \
+		"if run loadzimage; then " \
 			"run loadfdt;" \
-			"run mmcboot;" \
+			"run mmcbootz;" \
 		"fi;" \
 	"fi;"
 
