diff -up qlandkartegt-1.8.1/3rdparty/map2rmp/CFileGenerator.cpp.char qlandkartegt-1.8.1/3rdparty/map2rmp/CFileGenerator.cpp
--- qlandkartegt-1.8.1/3rdparty/map2rmp/CFileGenerator.cpp.char	2016-02-15 17:17:32.000000000 +0100
+++ qlandkartegt-1.8.1/3rdparty/map2rmp/CFileGenerator.cpp	2016-02-15 17:27:02.000000000 +0100
@@ -57,7 +57,7 @@ extern "C"
 
 static std::vector<JOCTET> jpgbuf;
 
-static const char bmp2bit[] =
+static const signed char bmp2bit[] =
 {
     73, 99, 111, 110, 32, 102, 105, 108, 101, 32, 118, 101,
     114, 115, 105, 111, 110, 32, 49, 46, 48, 46, 48, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3, 0, 0, 0,
@@ -82,7 +82,7 @@ static const char bmp2bit[] =
     -1, -1, 0, -1, -1
 };
 
-static const char bmp4bit[] =
+static const signed char bmp4bit[] =
 {
     73, 99, 111, 110, 32, 102, 105, 108, 101, 32, 118, 101,
     114, 115, 105, 111, 110, 32, 49, 46, 48, 46, 48, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3, 0, 0, 0,
@@ -764,14 +764,14 @@ void CFileGenerator::writeBmp2Bit(QDataS
 {
     rmp.directory[INDEX_BMP2BIT].offset = stream.device()->pos();
     rmp.directory[INDEX_BMP2BIT].length = sizeof(bmp2bit);
-    stream.writeRawData(bmp2bit, sizeof(bmp2bit));
+    stream.writeRawData((const char *)bmp2bit, sizeof(bmp2bit));
 }
 
 void CFileGenerator::writeBmp4Bit(QDataStream& stream, rmp_file_t& rmp)
 {
     rmp.directory[INDEX_BMP4BIT].offset = stream.device()->pos();
     rmp.directory[INDEX_BMP4BIT].length = sizeof(bmp4bit);
-    stream.writeRawData(bmp4bit, sizeof(bmp4bit));
+    stream.writeRawData((const char *)bmp4bit, sizeof(bmp4bit));
 }
 
 void CFileGenerator::writeCvgMap(QDataStream& stream, rmp_file_t& rmp)
