diff -ruN dsss-0.75/rebuild/rebuild.conf/dmd-posix-tango dsss-0.75.arch/rebuild/rebuild.conf/dmd-posix-tango
--- dsss-0.75/rebuild/rebuild.conf/dmd-posix-tango	2008-01-10 08:23:06.000000000 +0100
+++ dsss-0.75.arch/rebuild/rebuild.conf/dmd-posix-tango	2008-09-04 15:07:13.635985756 +0200
@@ -39,7 +39,7 @@
 
 [link]
 oneatatime=yes
-cmd=dmd -L--start-group -L-lphobos $i -of$o
+cmd=dmd -L--start-group $i -of$o
 
 libdir=-L-L$i
 lib=-L-l$i
