diff -Naur Python-3.1-old/setup.py Python-3.1/setup.py
--- Python-3.1-old/setup.py	2009-05-24 03:13:14.000000000 +1000
+++ Python-3.1/setup.py	2009-06-28 20:53:11.000000000 +1000
@@ -911,7 +911,7 @@
                         if self.compiler.find_library_file(lib_dirs, 'ndbm'):
                             ndbm_libs = ['ndbm']
                         else:
-                            ndbm_libs = []
+                            ndbm_libs = ['gdbm', 'gdbm_compat']
                         print("building dbm using ndbm")
                         dbmext = Extension('_dbm', ['_dbmmodule.c'],
                                            define_macros=[
