diff --git a/libsepol/src/private.h b/libsepol/src/private.h
index b884c23b..447fa64b 100644
--- a/libsepol/src/private.h
+++ b/libsepol/src/private.h
@@ -14,7 +14,7 @@
 #endif
 
 #include <errno.h>
-#include <dso.h>
+#include "dso.h"
 
 #ifdef __APPLE__
 #define __BYTE_ORDER  BYTE_ORDER
diff --git a/libsepol/src/util.c b/libsepol/src/util.c
index b00251c6..68e7c6a0 100644
--- a/libsepol/src/util.c
+++ b/libsepol/src/util.c
@@ -27,7 +27,7 @@
 #include <sepol/policydb/flask_types.h>
 #include <sepol/policydb/policydb.h>
 #include <sepol/policydb/util.h>
-#include <dso.h>
+#include "dso.h"
 
 struct val_to_name {
 	unsigned int val;
