diff --git a/adb/adb.cpp b/adb/adb.cpp
index 24d4292d02..e4ad76b7c3 100644
--- a/adb/adb.cpp
+++ b/adb/adb.cpp
@@ -44,8 +44,6 @@
 #include <android-base/parsenetaddress.h>
 #include <android-base/stringprintf.h>
 #include <android-base/strings.h>
-#include <build/version.h>
-#include <platform_tools_version.h>
 
 #include "adb_auth.h"
 #include "adb_io.h"
@@ -66,10 +64,10 @@ std::string adb_version() {
     // Don't change the format of this --- it's parsed by ddmlib.
     return android::base::StringPrintf(
         "Android Debug Bridge version %d.%d.%d\n"
-        "Version %s-%s\n"
+        "Version %s-ArchLinux\n"
         "Installed as %s\n",
         ADB_VERSION_MAJOR, ADB_VERSION_MINOR, ADB_SERVER_VERSION,
-        PLATFORM_TOOLS_VERSION, android::build::GetBuildNumber().c_str(),
+        PLATFORM_TOOLS_VERSION,
         android::base::GetExecutablePath().c_str());
 }
 
diff --git a/adb/client/main.cpp b/adb/client/main.cpp
index 0c5c28f1b3..b6a1281521 100644
--- a/adb/client/main.cpp
+++ b/adb/client/main.cpp
@@ -118,10 +118,6 @@ int adb_server_main(int is_daemon, const std::string& socket_spec, int ack_reply
     init_transport_registration();
     init_reconnect_handler();
 
-    if (!getenv("ADB_MDNS") || strcmp(getenv("ADB_MDNS"), "0") != 0) {
-        init_mdns_transport_discovery();
-    }
-
     if (!getenv("ADB_USB") || strcmp(getenv("ADB_USB"), "0") != 0) {
         usb_init();
     } else {
diff --git a/adb/client/usb_libusb.cpp b/adb/client/usb_libusb.cpp
index 53f01a0ce5..5be0d91085 100644
--- a/adb/client/usb_libusb.cpp
+++ b/adb/client/usb_libusb.cpp
@@ -22,6 +22,7 @@
 #include <stdlib.h>
 
 #include <atomic>
+#include <condition_variable>
 #include <chrono>
 #include <condition_variable>
 #include <memory>
@@ -30,7 +31,7 @@
 #include <thread>
 #include <unordered_map>
 
-#include <libusb/libusb.h>
+#include <libusb-1.0/libusb.h>
 
 #include <android-base/file.h>
 #include <android-base/logging.h>
diff --git a/adb/client/usb_linux.cpp b/adb/client/usb_linux.cpp
index 81b8306436..b4f09caab8 100644
--- a/adb/client/usb_linux.cpp
+++ b/adb/client/usb_linux.cpp
@@ -29,6 +29,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/ioctl.h>
+#include <sys/sysmacros.h>
 #include <sys/time.h>
 #include <sys/sysmacros.h>
 #include <sys/types.h>
diff --git a/adb/fdevent/fdevent.h b/adb/fdevent/fdevent.h
index ccb0c9221c..2424252f5d 100644
--- a/adb/fdevent/fdevent.h
+++ b/adb/fdevent/fdevent.h
@@ -20,6 +20,7 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <atomic>
 #include <chrono>
 #include <deque>
 #include <functional>
diff --git a/adb/sysdeps/posix/network.cpp b/adb/sysdeps/posix/network.cpp
index c5c227568f..99686f15fb 100644
--- a/adb/sysdeps/posix/network.cpp
+++ b/adb/sysdeps/posix/network.cpp
@@ -22,6 +22,7 @@
 #include <sys/socket.h>
 
 #include <string>
+#include <string.h>
 
 #include <android-base/logging.h>
 #include <android-base/stringprintf.h>
diff --git a/adb/types.h b/adb/types.h
index cd1366dfbe..1eb7e4b6d7 100644
--- a/adb/types.h
+++ b/adb/types.h
@@ -22,6 +22,7 @@
 #include <type_traits>
 #include <utility>
 #include <vector>
+#include <cstring>
 
 #include <android-base/logging.h>
 
diff --git a/base/errors_unix.cpp b/base/errors_unix.cpp
index 296995efe2..48269b6750 100644
--- a/base/errors_unix.cpp
+++ b/base/errors_unix.cpp
@@ -17,6 +17,7 @@
 #include "android-base/errors.h"
 
 #include <errno.h>
+#include <string.h>
 
 namespace android {
 namespace base {
diff --git a/base/file.cpp b/base/file.cpp
index 3dfcfbb7ec..1b5c0203f0 100644
--- a/base/file.cpp
+++ b/base/file.cpp
@@ -26,6 +26,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <string.h>
 
 #include <memory>
 #include <mutex>
diff --git a/base/logging.cpp b/base/logging.cpp
index f89168c0fc..da522019dc 100644
--- a/base/logging.cpp
+++ b/base/logging.cpp
@@ -24,6 +24,7 @@
 #include <inttypes.h>
 #include <libgen.h>
 #include <time.h>
+#include <string.h>
 
 // For getprogname(3) or program_invocation_short_name.
 #if defined(__ANDROID__) || defined(__APPLE__)
diff --git a/fastboot/fastboot.cpp b/fastboot/fastboot.cpp
index 8923f409df..d245a577ff 100644
--- a/fastboot/fastboot.cpp
+++ b/fastboot/fastboot.cpp
@@ -58,9 +58,7 @@
 #include <android-base/stringprintf.h>
 #include <android-base/strings.h>
 #include <android-base/unique_fd.h>
-#include <build/version.h>
 #include <liblp/liblp.h>
-#include <platform_tools_version.h>
 #include <sparse/sparse.h>
 #include <ziparchive/zip_archive.h>
 
@@ -1749,7 +1747,7 @@ int FastBootTool::Main(int argc, char* argv[]) {
                 setvbuf(stdout, nullptr, _IONBF, 0);
                 setvbuf(stderr, nullptr, _IONBF, 0);
             } else if (name == "version") {
-                fprintf(stdout, "fastboot version %s-%s\n", PLATFORM_TOOLS_VERSION, android::build::GetBuildNumber().c_str());
+                fprintf(stdout, "fastboot version %s-ArchLinux\n", PLATFORM_TOOLS_VERSION);
                 fprintf(stdout, "Installed as %s\n", android::base::GetExecutablePath().c_str());
                 return 0;
 #if !defined(_WIN32)
diff --git a/fastboot/fs.cpp b/fastboot/fs.cpp
index 8c0aa6bf4e..3ad5b53568 100644
--- a/fastboot/fs.cpp
+++ b/fastboot/fs.cpp
@@ -117,7 +117,7 @@ static int generate_ext4_image(const char* fileName, long long partSize,
     static constexpr int block_size = 4096;
     const std::string exec_dir = android::base::GetExecutableDirectory();
 
-    const std::string mke2fs_path = exec_dir + "/mke2fs";
+    const std::string mke2fs_path = exec_dir + "/mke2fs.android";
     std::vector<const char*> mke2fs_args = {mke2fs_path.c_str(), "-t", "ext4", "-b"};
 
     std::string block_size_str = std::to_string(block_size);
diff --git a/fs_mgr/liblp/reader.cpp b/fs_mgr/liblp/reader.cpp
index dcee6d2d86..6fcd613ab6 100644
--- a/fs_mgr/liblp/reader.cpp
+++ b/fs_mgr/liblp/reader.cpp
@@ -19,6 +19,7 @@
 #include <stddef.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <cstring>
 
 #include <functional>
 
diff --git a/fs_mgr/liblp/writer.cpp b/fs_mgr/liblp/writer.cpp
index bffcb7e6e1..6b36a641a6 100644
--- a/fs_mgr/liblp/writer.cpp
+++ b/fs_mgr/liblp/writer.cpp
@@ -20,6 +20,7 @@
 #include <unistd.h>
 
 #include <string>
+#include <cstring>
 
 #include <android-base/file.h>
 #include <android-base/unique_fd.h>
diff --git a/libziparchive/zip_archive.cc b/libziparchive/zip_archive.cc
index c95b03542e..ae8ebb93a4 100644
--- a/libziparchive/zip_archive.cc
+++ b/libziparchive/zip_archive.cc
@@ -30,6 +30,7 @@
 #include <string.h>
 #include <time.h>
 #include <unistd.h>
+#include <string_view>
 
 #include <memory>
 #include <vector>
