diff --git a/contrib/android/perms.c b/contrib/android/perms.c
index 9ae8e586..433506fd 100644
--- a/contrib/android/perms.c
+++ b/contrib/android/perms.c
@@ -5,6 +5,7 @@
 #include "support/nls-enable.h"
 #include <time.h>
 #include <sys/stat.h>
+#include "private/android_filesystem_capability.h"
 
 #ifndef XATTR_SELINUX_SUFFIX
 # define XATTR_SELINUX_SUFFIX  "selinux"
diff --git a/lib/ext2fs/bitops.h b/lib/ext2fs/bitops.h
index 505b3c9c..7be5105f 100644
--- a/lib/ext2fs/bitops.h
+++ b/lib/ext2fs/bitops.h
@@ -233,11 +233,11 @@ extern errcode_t ext2fs_find_first_set_generic_bmap(ext2fs_generic_bitmap bitmap
 #if (__STDC_VERSION__ >= 199901L)
 #define _INLINE_ extern inline
 #else
-#define _INLINE_ inline
+#define _INLINE_ static inline
 #endif
 #else /* !INCLUDE_INLINE FUNCS */
 #if (__STDC_VERSION__ >= 199901L)
-#define _INLINE_ inline
+#define _INLINE_ static inline
 #else /* not C99 */
 #ifdef __GNUC__
 #define _INLINE_ extern __inline__
diff --git a/lib/ext2fs/ext2fs.h b/lib/ext2fs/ext2fs.h
index e153c817..dcae5947 100644
--- a/lib/ext2fs/ext2fs.h
+++ b/lib/ext2fs/ext2fs.h
@@ -1736,7 +1736,7 @@ extern const struct ext2_inode *ext2fs_const_inode(const struct ext2_inode_large
 #define _INLINE_ extern
 #else
 #if (__STDC_VERSION__ >= 199901L)
-#define _INLINE_ inline
+#define _INLINE_ static inline
 #else
 #ifdef __GNUC__
 #define _INLINE_ extern __inline__
