diff -ru qt3d-everywhere-src-6.1.1.orig/src/3rdparty/assimp/src/include/assimp/Importer.hpp qt3d-everywhere-src-6.1.1/src/3rdparty/assimp/src/include/assimp/Importer.hpp
--- qt3d-everywhere-src-6.1.1.orig/src/3rdparty/assimp/src/include/assimp/Importer.hpp	2021-06-07 08:29:23.902950055 +0000
+++ qt3d-everywhere-src-6.1.1/src/3rdparty/assimp/src/include/assimp/Importer.hpp	2021-06-07 08:30:51.924838527 +0000
@@ -59,6 +59,8 @@
 // Public ASSIMP data structures
 #include <assimp/types.h>
 
+#include <exception>
+
 namespace Assimp {
 // =======================================================================
 // Public interface to Assimp
diff -ru qt3d-everywhere-src-6.1.1.orig/src/3rdparty/assimp/src/code/Common/Importer.h qt3d-everywhere-src-6.1.1/src/3rdparty/assimp/src/code/Common/Importer.h
--- qt3d-everywhere-src-6.1.1.orig/src/3rdparty/assimp/src/code/Common/Importer.h	2021-06-07 08:34:37.598230479 +0000
+++ qt3d-everywhere-src-6.1.1/src/3rdparty/assimp/src/code/Common/Importer.h	2021-06-07 08:35:11.893602448 +0000
@@ -47,6 +47,7 @@
 #include <map>
 #include <vector>
 #include <string>
+#include <exception>
 #include <assimp/matrix4x4.h>
 
 struct aiScene;
