--- AssemblyInfo.cs.orig	2007-07-31 13:10:29.000000000 +0200
+++ AssemblyInfo.cs	2009-01-13 17:29:15.000000000 +0100
@@ -30,7 +30,7 @@
 // You can specify all the values or you can default the Revision and Build Numbers 
 // by using the '*' as shown below:
 
-[assembly: AssemblyVersion("2.1.5.*")]
+[assembly: AssemblyVersion("2.1.5.0")]
 
 //
 // In order to sign your assembly you must specify a key to use. Refer to the 
@@ -61,10 +61,10 @@
 [assembly: AssemblyKeyName("")]
 
 #if !WindowsCE
-[assembly: AssemblyKeyFile("..\\..\\FlickrNet.snk")]
+[assembly: AssemblyKeyFile("FlickrNet.snk")]
 [assembly: AllowPartiallyTrustedCallers()]
 [assembly: SecurityPermission(SecurityAction.RequestMinimum, Execution = true)]
 #endif
 
-[assembly: CLSCompliantAttribute(true)]
-[assembly: ComVisible(false)]
\ No newline at end of file
+[assembly: CLSCompliantAttribute(true)]
+[assembly: ComVisible(false)]
