diff -uprb graphviz-2.38.0.orig/plugin/gs/gvloadimage_gs.c graphviz-2.38.0/plugin/gs/gvloadimage_gs.c
--- graphviz-2.38.0.orig/plugin/gs/gvloadimage_gs.c	2014-04-13 23:40:25.000000000 +0300
+++ graphviz-2.38.0/plugin/gs/gvloadimage_gs.c	2015-11-11 00:08:32.916123704 +0200
@@ -72,11 +72,11 @@ static void gs_error(GVJ_t * job, const
 
     assert (err < 0);
 
-    if (err >= e_VMerror) 
+    if (err >= gs_error_VMerror)
 	errsrc = "PostScript Level 1"; 
-    else if (err >= e_unregistered)
+    else if (err >= gs_error_unregistered)
 	errsrc = "PostScript Level 2";
-    else if (err >= e_invalidid)
+    else if (err >= gs_error_invalidid)
 	errsrc = "DPS error";
     else
 	errsrc = "Ghostscript internal error";
