diff --git a/CMakeLists.txt b/CMakeLists.txt
index d51e6e3d..8bd39e95 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -610,7 +610,8 @@ add_library(${PROJECT_NAME}_static
   ${${PROJECT_NAME}_SOURCES}
   ${${PROJECT_NAME}_QM_FILES})
 target_link_libraries(${PROJECT_NAME}_static
-  ${ALL_LIBRARIES})
+  ${ALL_LIBRARIES}
+  atomic)
 
 add_executable(${PROJECT_NAME}
   WIN32
@@ -619,7 +620,8 @@ add_executable(${PROJECT_NAME}
   src/main.cpp)
 target_link_libraries(${PROJECT_NAME}
   ${PROJECT_NAME}_static
-  ${ALL_LIBRARIES})
+  ${ALL_LIBRARIES}
+  atomic)
 
 include(Testing)
 include(Installation)
