--- CMakeLists.txt.orig	2020-04-28 12:13:40.413921675 -0600
+++ CMakeLists.txt	2020-04-28 12:15:04.340850331 -0600
@@ -671,7 +671,8 @@
   ${${PROJECT_NAME}_RESOURCES})
 target_link_libraries(${PROJECT_NAME}_static
   ${CMAKE_REQUIRED_LIBRARIES}
-  ${ALL_LIBRARIES})
+  ${ALL_LIBRARIES}
+  atomic)
 
 add_executable(${PROJECT_NAME}
   WIN32
@@ -681,7 +682,8 @@
 target_link_libraries(${PROJECT_NAME}
   ${PROJECT_NAME}_static
   ${CMAKE_REQUIRED_LIBRARIES}
-  ${ALL_LIBRARIES})
+  ${ALL_LIBRARIES}
+  atomic)
 
 include(Testing)
 include(Installation)
