diff --git a/src/fs_btrfs.h b/src/fs_btrfs.h
index e2461f3..923e7f5 100644
--- a/src/fs_btrfs.h
+++ b/src/fs_btrfs.h
@@ -18,6 +18,8 @@
 #ifndef __FS_BTRFS_H__
 #define __FS_BTRFS_H__
 
+#include <linux/types.h>
+
 struct s_dico;
 struct s_strlist;
 
diff --git a/src/fs_xfs.h b/src/fs_xfs.h
index ee3c53d..17d0d5e 100644
--- a/src/fs_xfs.h
+++ b/src/fs_xfs.h
@@ -18,6 +18,8 @@
 #ifndef __FS_XFS_H__
 #define __FS_XFS_H__
 
+#include <linux/types.h>
+
 struct s_dico;
 struct s_strlist;
 
@@ -36,8 +38,6 @@ typedef uint16_t __u16;
 typedef int16_t __s16;
 typedef uint32_t __u32;
 typedef int32_t __s32;
-typedef uint64_t __u64;
-typedef int64_t __s64;
 
 typedef __uint32_t      xfs_agblock_t;  /* blockno in alloc. group */
 typedef __uint32_t      xfs_extlen_t;   /* extent length in blocks */
diff --git a/src/types.h b/src/types.h
index b69dfd7..a18e139 100644
--- a/src/types.h
+++ b/src/types.h
@@ -34,7 +34,6 @@ typedef int64_t s64;
 
 typedef uint16_t __le16;
 typedef uint32_t __le32;
-typedef uint64_t __le64;
 
 #define le8_to_cpu(v) (v)
 #define cpu_to_le8(v) (v)
