diff -u -r webkitgtk-2.18.5/Source/cmake/FindEnchant.cmake webkitgtk-2.18.5-enchant2/Source/cmake/FindEnchant.cmake
--- webkitgtk-2.18.5/Source/cmake/FindEnchant.cmake	2017-08-09 11:13:53.000000000 +0200
+++ webkitgtk-2.18.5-enchant2/Source/cmake/FindEnchant.cmake	2018-01-18 10:06:07.127310881 +0100
@@ -28,7 +28,7 @@
 # ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 find_package(PkgConfig)
-pkg_check_modules(PC_ENCHANT enchant)
+pkg_check_modules(PC_ENCHANT enchant-2)
 
 find_path(ENCHANT_INCLUDE_DIRS
     NAMES enchant.h
@@ -37,7 +37,7 @@
 )
 
 find_library(ENCHANT_LIBRARIES
-    NAMES enchant
+    NAMES enchant-2
     PATHS ${PC_ENCHANT_LIBDIR}
           ${PC_ENCHANT_LIBRARY_DIRS}
 )
