diff -u -r webkitgtk-2.18.4/Tools/gtk/generate-gtkdoc webkitgtk-2.18.4-gtkdoc/Tools/gtk/generate-gtkdoc
--- webkitgtk-2.18.4/Tools/gtk/generate-gtkdoc	2017-12-18 18:36:59.000000000 +0100
+++ webkitgtk-2.18.4-gtkdoc/Tools/gtk/generate-gtkdoc	2017-12-20 02:52:09.008775941 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # Copyright (C) 2011 Igalia S.L.
 #
 # This library is free software; you can redistribute it and/or
diff -u -r webkitgtk-2.18.4/Tools/gtk/gtkdoc.py webkitgtk-2.18.4-gtkdoc/Tools/gtk/gtkdoc.py
--- webkitgtk-2.18.4/Tools/gtk/gtkdoc.py	2017-08-09 11:13:53.000000000 +0200
+++ webkitgtk-2.18.4-gtkdoc/Tools/gtk/gtkdoc.py	2017-12-20 02:52:00.722090939 +0100
@@ -319,9 +319,9 @@
             ldflags = ' "-L%s" %s ' % (self.library_path, additional_ldflags) + ldflags
             current_ld_library_path = env.get('LD_LIBRARY_PATH')
             if current_ld_library_path:
-                env['RUN'] = 'LD_LIBRARY_PATH="%s:%s" ' % (self.library_path, current_ld_library_path)
+                env['RUN'] = 'env LD_LIBRARY_PATH="%s:%s" ' % (self.library_path, current_ld_library_path)
             else:
-                env['RUN'] = 'LD_LIBRARY_PATH="%s" ' % self.library_path
+                env['RUN'] = 'env LD_LIBRARY_PATH="%s" ' % self.library_path
 
         if ldflags:
             env['LDFLAGS'] = '%s %s' % (ldflags, env.get('LDFLAGS', ''))
