diff -urN webkitgtk-2.26.1.orig/Source/WebCore/dom/Document.cpp webkitgtk-2.26.1/Source/WebCore/dom/Document.cpp
--- webkitgtk-2.26.1.orig/Source/WebCore/dom/Document.cpp	2019-09-02 11:41:47.000000000 +0200
+++ webkitgtk-2.26.1/Source/WebCore/dom/Document.cpp	2019-10-08 16:24:57.649551493 +0200
@@ -4944,12 +4944,12 @@
     unsigned i = 0;
 
     UChar32 c;
-    U16_NEXT(characters, i, length, c)
+    U16_NEXT(characters, i, length, c);
     if (!isValidNameStart(c))
         return false;
 
     while (i < length) {
-        U16_NEXT(characters, i, length, c)
+        U16_NEXT(characters, i, length, c);
         if (!isValidNamePart(c))
             return false;
     }
@@ -5009,7 +5009,7 @@
 
     for (unsigned i = 0; i < length; ) {
         UChar32 c;
-        U16_NEXT(qualifiedName, i, length, c)
+        U16_NEXT(qualifiedName, i, length, c);
         if (c == ':') {
             if (sawColon)
                 return Exception { InvalidCharacterError };
diff -urN webkitgtk-2.26.1.orig/Source/WTF/wtf/URLHelpers.cpp webkitgtk-2.26.1/Source/WTF/wtf/URLHelpers.cpp
--- webkitgtk-2.26.1.orig/Source/WTF/wtf/URLHelpers.cpp	2019-08-28 13:46:29.000000000 +0200
+++ webkitgtk-2.26.1/Source/WTF/wtf/URLHelpers.cpp	2019-10-08 16:24:50.619850993 +0200
@@ -301,7 +301,7 @@
     Optional<UChar32> previousCodePoint;
     while (i < length) {
         UChar32 c;
-        U16_NEXT(buffer, i, length, c)
+        U16_NEXT(buffer, i, length, c);
         UErrorCode error = U_ZERO_ERROR;
         UScriptCode script = uscript_getScript(c, &error);
         if (error != U_ZERO_ERROR) {
