diff -u -r webkitgtk-2.10.8/Source/JavaScriptCore/runtime/Options.cpp webkitgtk-2.10.8-isnan/Source/JavaScriptCore/runtime/Options.cpp
--- webkitgtk-2.10.8/Source/JavaScriptCore/runtime/Options.cpp	2016-03-11 09:59:07.000000000 +0100
+++ webkitgtk-2.10.8-isnan/Source/JavaScriptCore/runtime/Options.cpp	2016-03-13 13:51:55.114836547 +0100
@@ -610,7 +610,7 @@
     case Options::Type::unsignedType:
         return m_entry.unsignedVal == other.m_entry.unsignedVal;
     case Options::Type::doubleType:
-        return (m_entry.doubleVal == other.m_entry.doubleVal) || (isnan(m_entry.doubleVal) && isnan(other.m_entry.doubleVal));
+        return (m_entry.doubleVal == other.m_entry.doubleVal) || (std::isnan(m_entry.doubleVal) && std::isnan(other.m_entry.doubleVal));
     case Options::Type::int32Type:
         return m_entry.int32Val == other.m_entry.int32Val;
     case Options::Type::optionRangeType:
