--- Source/WTF/wtf/CMakeLists.txt.orig	2018-02-19 02:36:10.000000000 -0700
+++ Source/WTF/wtf/CMakeLists.txt	2018-03-24 21:42:12.759186320 -0600
@@ -315,7 +315,6 @@
 
 set(WTF_SOURCES
     Assertions.cpp
-    Atomics.cpp
     AutomaticThread.cpp
     BitVector.cpp
     CPUTime.cpp
--- Source/JavaScriptCore/CMakeLists.txt.orig	2018-03-24 21:36:40.067891066 -0600
+++ Source/JavaScriptCore/CMakeLists.txt	2018-03-24 21:38:06.715707122 -0600
@@ -120,7 +120,7 @@
 # __atomic_fetch_add_8 is not available as a compiler intrinsic. It is
 # available on other platforms (including 32-bit Arm), so the link with
 # libatomic is only neede on MIPS.
-if (WTF_CPU_MIPS)
+if (WTF_CPU_MIPS OR WTF_CPU_ARM)
     list(APPEND JavaScriptCore_LIBRARIES
         -latomic
     )
