From 9154ea477728797b9d33b32fb36940485f6927fa Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Thu, 30 Aug 2018 07:04:58 +0200
Subject: [PATCH] Use ternary operator instead of std::max

---
 Source/JavaScriptCore/heap/RegisterState.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/JavaScriptCore/heap/RegisterState.h b/Source/JavaScriptCore/heap/RegisterState.h
index baaef4c7..b3037dba 100644
--- a/Source/JavaScriptCore/heap/RegisterState.h
+++ b/Source/JavaScriptCore/heap/RegisterState.h
@@ -166,7 +166,7 @@ using RegisterState = jmp_buf;
 #pragma warning(disable: 4611)
 #endif
 #define ALLOCATE_AND_GET_REGISTER_STATE(registers) \
-    alignas(std::max(alignof(void*), alignof(RegisterState))) RegisterState registers; \
+    alignas(alignof(void*) > alignof(RegisterState) ? alignof(void*) : alignof(RegisterState)) RegisterState registers; \
     setjmp(registers)
 
 #if COMPILER(MSVC)
-- 
2.17.1

