--- defs.h.org	2009-02-26 12:49:16.000000000 +0000
+++ defs.h	2009-02-26 12:50:04.000000000 +0000
@@ -17,10 +17,10 @@
 /******* General configuration *******/
 
 /* Name of configuration file (in Services directory) */
-#define IRCSERVICES_CONF        PROGRAM ".conf"
+#define IRCSERVICES_CONF        "/etc/ircd/" PROGRAM ".conf"
 
 /* Name of module configuration file (in Services directory) */
-#define MODULES_CONF            "modules.conf"
+#define MODULES_CONF            "/etc/ircd/modules.conf"
 
 /* Maximum number of parameters for a configuration directive */
 #define CONFIG_MAXPARAMS        8
