diff --git a/src/api/python/setup.py b/src/api/python/setup.py
index 325fb4230..f3b7287d7 100644
--- a/src/api/python/setup.py
+++ b/src/api/python/setup.py
@@ -270,16 +270,11 @@ def _copy_sources():
 
 class build(_build):
     def run(self):
-        if RELEASE_DIR is None:
-            self.execute(_configure_z3, (), msg="Configuring Z3")
-            self.execute(_build_z3, (), msg="Building Z3")
         self.execute(_copy_bins, (), msg="Copying binaries")
         _build.run(self)
 
 class develop(_develop):
     def run(self):
-        self.execute(_configure_z3, (), msg="Configuring Z3")
-        self.execute(_build_z3, (), msg="Building Z3")
         self.execute(_copy_bins, (), msg="Copying binaries")
         _develop.run(self)
 
