--- libtorrent-0.12.0.old/rak/functional.h	2008-03-09 15:59:49.000000000 +0100
+++ libtorrent-0.12.0/rak/functional.h	2008-03-09 16:00:11.000000000 +0100
@@ -37,6 +37,7 @@
 #ifndef RAK_FUNCTIONAL_H
 #define RAK_FUNCTIONAL_H
 
+#include <cstddef>
 #include <functional>
 
 namespace rak {
--- libtorrent-0.12.0.old/src/data/chunk.cc	2008-03-09 15:59:49.000000000 +0100
+++ libtorrent-0.12.0/src/data/chunk.cc	2008-03-09 16:00:11.000000000 +0100
@@ -36,6 +36,7 @@
 
 #include "config.h"
 
+#include <cstring>
 #include <algorithm>
 #include <functional>
 
--- libtorrent-0.12.0.old/src/data/chunk_list.h	2008-03-09 15:59:49.000000000 +0100
+++ libtorrent-0.12.0/src/data/chunk_list.h	2008-03-09 16:00:11.000000000 +0100
@@ -38,6 +38,7 @@
 #define LIBTORRENT_DATA_CHUNK_LIST_H
 
 #include <vector>
+#include <string>
 #include <rak/error_number.h>
 #include <rak/functional.h>
 
--- libtorrent-0.12.0.old/src/net/address_list.cc	2008-03-09 15:59:49.000000000 +0100
+++ libtorrent-0.12.0/src/net/address_list.cc	2008-03-09 16:00:11.000000000 +0100
@@ -36,6 +36,8 @@
 
 #include "config.h"
 
+#include <algorithm>
+
 #include <rak/functional.h>
 
 #include "download/download_info.h"  // for SocketAddressCompact
--- libtorrent-0.12.0.old/src/torrent/data/file_list_iterator.h	2008-03-09 15:59:49.000000000 +0100
+++ libtorrent-0.12.0/src/torrent/data/file_list_iterator.h	2008-03-09 16:00:11.000000000 +0100
@@ -37,6 +37,7 @@
 #ifndef LIBTORRENT_FILE_LIST_ITERATOR_H
 #define LIBTORRENT_FILE_LIST_ITERATOR_H
 
+#include <cstdlib>
 #include <torrent/common.h>
 #include <torrent/data/file_list.h>
 
--- libtorrent-0.12.0.old/src/torrent/exceptions.cc	2008-03-09 15:59:49.000000000 +0100
+++ libtorrent-0.12.0/src/torrent/exceptions.cc	2008-03-09 16:00:11.000000000 +0100
@@ -37,6 +37,7 @@
 #include "config.h"
 
 #include <cerrno>
+#include <cstring>
 
 #include "exceptions.h"
 
--- libtorrent-0.12.0.old/src/torrent/tracker_list.h	2008-03-09 15:59:49.000000000 +0100
+++ libtorrent-0.12.0/src/torrent/tracker_list.h	2008-03-09 16:01:01.000000000 +0100
@@ -38,6 +38,7 @@
 #define LIBTORRENT_TRACKER_LIST_H

 #include <algorithm>
+#include <string>
 #include <vector>
 #include <torrent/common.h>
 
--- libtorrent-0.12.0.old/src/utils/diffie_hellman.cc	2008-03-09 15:59:49.000000000 +0100
+++ libtorrent-0.12.0/src/utils/diffie_hellman.cc	2008-03-09 16:00:11.000000000 +0100
@@ -36,6 +36,7 @@
 
 #include "config.h"
 
+#include <cstring>
 #include <string>
 
 #ifdef USE_OPENSSL
