diff -urN libtorrent-0.12.4.orig/src/protocol/extensions.cc libtorrent-0.12.4/src/protocol/extensions.cc
--- libtorrent-0.12.4.orig/src/protocol/extensions.cc	2008-08-26 13:15:15.000000000 -0700
+++ libtorrent-0.12.4/src/protocol/extensions.cc	2009-02-27 07:53:45.000000000 -0800
@@ -36,6 +36,8 @@
 
 #include "config.h"
 
+#include <cstdio>
+
 #include <limits>
 #include <sstream>
 
diff -urN libtorrent-0.12.4.orig/src/protocol/handshake.cc libtorrent-0.12.4/src/protocol/handshake.cc
--- libtorrent-0.12.4.orig/src/protocol/handshake.cc	2008-11-10 18:39:23.000000000 -0800
+++ libtorrent-0.12.4/src/protocol/handshake.cc	2009-02-27 07:54:00.000000000 -0800
@@ -36,6 +36,8 @@
 
 #include "config.h"
 
+#include <cstdio>
+
 #include "download/download_info.h"
 #include "download/download_main.h"
 #include "net/throttle_list.h"
diff -urN libtorrent-0.12.4.orig/src/torrent/poll_epoll.cc libtorrent-0.12.4/src/torrent/poll_epoll.cc
--- libtorrent-0.12.4.orig/src/torrent/poll_epoll.cc	2008-10-28 04:49:47.000000000 -0700
+++ libtorrent-0.12.4/src/torrent/poll_epoll.cc	2009-02-27 07:52:16.000000000 -0800
@@ -38,6 +38,7 @@
 
 #include <cerrno>
 #include <cstring>
+#include <cstdio>
 
 #include <unistd.h>
 #include <torrent/exceptions.h>
diff -urN libtorrent-0.12.4.orig/src/tracker/tracker_dht.cc libtorrent-0.12.4/src/tracker/tracker_dht.cc
--- libtorrent-0.12.4.orig/src/tracker/tracker_dht.cc	2008-08-26 13:18:34.000000000 -0700
+++ libtorrent-0.12.4/src/tracker/tracker_dht.cc	2009-02-27 07:54:41.000000000 -0800
@@ -36,6 +36,7 @@
 
 #include "config.h"
 
+#include <cstdio>
 #include <sstream>
 
 #include "dht/dht_router.h"
diff -urN libtorrent-0.12.4.orig/src/tracker/tracker_udp.cc libtorrent-0.12.4/src/tracker/tracker_udp.cc
--- libtorrent-0.12.4.orig/src/tracker/tracker_udp.cc	2008-05-07 05:19:13.000000000 -0700
+++ libtorrent-0.12.4/src/tracker/tracker_udp.cc	2009-02-27 08:03:20.000000000 -0800
@@ -36,6 +36,8 @@
 
 #include "config.h"
 
+#include <cstdio>
+
 #include <sigc++/adaptors/bind.h>
 #include <torrent/connection_manager.h>
 
