--- configure.ac.orig	2015-07-02 00:19:04.000000000 +0200
+++ configure.ac	2015-07-13 20:32:07.158592915 +0200
@@ -1 +1 @@
-AC_INIT(libtorrent, 0.13.4, sundell.software@gmail.com)
+AC_INIT(libtorrent, 0.13.5, sundell.software@gmail.com)
@@ -6,2 +6,2 @@
-AC_DEFINE(PEER_NAME, "-lt0D40-", Identifier that is part of the default peer id)
-AC_DEFINE(PEER_VERSION, "lt\x0D\x40", 4 byte client and version identifier for DHT)
+AC_DEFINE(PEER_NAME, "-lt0D50-", Identifier that is part of the default peer id)
+AC_DEFINE(PEER_VERSION, "lt\x0D\x50", 4 byte client and version identifier for DHT)
