--- src/xsane-back-gtk.c	2009-06-24 00:12:18.000000000 +0200
+++ src/xsane-back-gtk,c.new	2009-06-24 00:13:41.000000000 +0200
@@ -2612,10 +2612,7 @@
       continue;
     }
 
-    if (!(opt->cap & SANE_CAP_ALWAYS_SETTABLE))
-    {
-      gtk_widget_set_sensitive(xsane.element[i].widget, sensitive);
-    }
+    gtk_widget_set_sensitive(xsane.element[i].widget, sensitive);
   }
 
   if (xsane.xsanemode_widget)
