diff -up libgag/include/StreamBackend.h.orig libgag/include/StreamBackend.h
--- libgag/include/StreamBackend.h.orig	2009-03-18 03:46:47.000000000 +0100
+++ libgag/include/StreamBackend.h	2009-06-20 14:21:10.000000000 +0200
@@ -43,7 +43,7 @@ namespace GAGCore
 		virtual void flush(void) = 0;
 		virtual void read(void *data, size_t size) = 0;
 		virtual void putc(int c) = 0;
-		virtual int getc(void) = 0;
+		virtual int get_c(void) = 0;
 		virtual void seekFromStart(int displacement) = 0;
 		virtual void seekFromEnd(int displacement) = 0;
 		virtual void seekRelative(int displacement) = 0;
@@ -67,7 +67,7 @@ namespace GAGCore
 		virtual void flush(void) { assert(fp); fflush(fp); }
 		virtual void read(void *data, size_t size) { assert(fp); fread(data, size, 1, fp); }
 		virtual void putc(int c) { assert(fp); fputc(c, fp); }
-		virtual int getc(void) { assert(fp); return fgetc(fp); }
+		virtual int get_c(void) { assert(fp); return fgetc(fp); }
 		virtual void seekFromStart(int displacement) { assert(fp); fseek(fp, displacement, SEEK_SET); }
 		virtual void seekFromEnd(int displacement) { assert(fp); fseek(fp, displacement, SEEK_END); }
 		virtual void seekRelative(int displacement) { assert(fp); fseek(fp, displacement, SEEK_CUR); }
@@ -92,7 +92,7 @@ namespace GAGCore
 		virtual void flush(void);
 		virtual void read(void *data, size_t size);
 		virtual void putc(int c);
-		virtual int getc(void);
+		virtual int get_c(void);
 		virtual void seekFromStart(int displacement);
 		virtual void seekFromEnd(int displacement);
 		virtual void seekRelative(int displacement);
@@ -117,7 +117,7 @@ namespace GAGCore
 		virtual void flush(void) { }
 		virtual void read(void *data, size_t size);
 		virtual void putc(int c);
-		virtual int getc(void);
+		virtual int get_c(void);
 		virtual void seekFromStart(int displacement);
 		virtual void seekFromEnd(int displacement);
 		virtual void seekRelative(int displacement);
diff -up libgag/src/StreamBackend.cpp.orig libgag/src/StreamBackend.cpp
--- libgag/src/StreamBackend.cpp.orig	2009-03-18 03:46:47.000000000 +0100
+++ libgag/src/StreamBackend.cpp	2009-06-20 14:21:53.000000000 +0200
@@ -75,9 +75,9 @@ namespace GAGCore
 		buffer->putc(c);
 	}
 	
-	int ZLibStreamBackend::getc(void)
+	int ZLibStreamBackend::get_c(void)
 	{
-		return buffer->getc();
+		return buffer->get_c();
 	}
 	
 	void ZLibStreamBackend::seekFromStart(int displacement)
@@ -147,7 +147,7 @@ namespace GAGCore
 		write(&ch, 1);
 	}
 	
-	int MemoryStreamBackend::getc(void)
+	int MemoryStreamBackend::get_c(void)
 	{
 		Uint8 ch;
 		read(&ch, 1);
diff -up libgag/src/Stream.cpp.orig libgag/src/Stream.cpp
--- libgag/src/Stream.cpp.orig	2009-03-18 03:46:47.000000000 +0100
+++ libgag/src/Stream.cpp	2009-06-20 14:21:35.000000000 +0200
@@ -60,7 +60,7 @@ namespace GAGCore
 		std::string s;
 		while (1)
 		{
-			int c = backend->getc();
+			int c = backend->get_c();
 			if(c=='\r')
 				continue;
 			if ((c >= 0) && (c != '\n'))
diff -up libgag/src/TextStream.cpp.orig libgag/src/TextStream.cpp
--- libgag/src/TextStream.cpp.orig	2009-03-18 03:46:47.000000000 +0100
+++ libgag/src/TextStream.cpp	2009-06-20 14:21:22.000000000 +0200
@@ -150,7 +150,7 @@ namespace GAGCore
 		
 		int nextChar(void)
 		{
-			next = stream->getc();
+			next = stream->get_c();
 			if 	(next == '\n')
 			{
 				column = 0;
