diff --git a/internal.h b/internal.h
index 40916a5..785be49 100644
--- a/internal.h
+++ b/internal.h
@@ -250,16 +250,8 @@ struct rb_subclass_entry {
     rb_subclass_entry_t *next;
 };
 
-#if defined(HAVE_LONG_LONG)
-typedef unsigned LONG_LONG rb_serial_t;
-#define SERIALT2NUM ULL2NUM
-#elif defined(HAVE_UINT64_T)
-typedef uint64_t rb_serial_t;
-#define SERIALT2NUM SIZET2NUM
-#else
 typedef unsigned long rb_serial_t;
 #define SERIALT2NUM ULONG2NUM
-#endif
 
 struct rb_classext_struct {
     struct st_table *iv_index_tbl;
-- 
1.9.0

