--- Java/Makefile.PL.old	2006-03-15 01:11:18.000000000 +0100
+++ Java/Makefile.PL	2006-03-15 01:16:08.000000000 +0100
@@ -61,8 +61,7 @@
 	"      PerlInterpreter.\n" ;
 
 my $build_jni_by_dflt = Inline::Java::Portable::portable("BUILD_JNI_BY_DFLT") ;
-if (($build_jni || AskYN("Do you wish to build the JNI extension?", 
-	($build_jni_by_dflt ? 'y' : 'n')))){
+if ($build_jni){
 	print "\nBuilding JNI extension.\n\n" ;
 
 	$jdk_dir = Inline::Java::get_default_j2sdk() ;
@@ -139,9 +138,9 @@
 on all platforms. See documentation for more details.
 Note: PerlNatives requires J2SDK 1.4 or greater.
 TXT
-			if (AskYN("Do you wish to build the PerlNatives extension?", 'n')){
+#			if (AskYN("Do you wish to build the PerlNatives extension?", 'n')){
 				push @{$DIR}, 'PerlNatives' ;
-			}
+#			}
 			print "\n" ;
 
 			print <<TXT;
@@ -150,9 +149,9 @@
 may not build or work properly on all platforms. See documentation for 
 more details.
 TXT
-			if (AskYN("Do you wish to build the PerlInterpreter extension?", 'n')){
+#			if (AskYN("Do you wish to build the PerlInterpreter extension?", 'n')){
 				push @{$DIR}, 'PerlInterpreter' ;
-			}
+#			}
 			print "\n" ;
 
 			WriteMakefile(
