diff -wbBur PerlPanel.org/lib/PerlPanel.pm PerlPanel/lib/PerlPanel.pm
--- PerlPanel.org/lib/PerlPanel.pm	2005-02-21 18:58:50.000000000 +0300
+++ PerlPanel/lib/PerlPanel.pm	2008-07-29 18:12:46.000000000 +0400
@@ -269,10 +269,7 @@
 
 	$self->{icon_theme}->set_custom_theme($theme);
 
-	if ($VERSION !~ /^[\d\.]$/) {
-		# we're in sandbox mode
 		$self->{icon_theme}->prepend_search_path(sprintf('%s/share/icons', $PREFIX));
-	}
 	$self->{icon_theme}->prepend_search_path(sprintf('%s/.%s/icon-files', $ENV{HOME}, lc($NAME)));
 	$self->{icon_theme}->prepend_search_path(sprintf('%s/.local/share/icons', $ENV{HOME}));
 
diff -wbBur PerlPanel.org/src/perlpanel-applet-install PerlPanel/src/perlpanel-applet-install
--- PerlPanel.org/src/perlpanel-applet-install	2004-10-20 03:09:51.000000000 +0400
+++ PerlPanel/src/perlpanel-applet-install	2008-07-29 18:12:26.000000000 +0400
@@ -74,19 +74,11 @@
 
 } else {
 	my %args;
-	if ($user > 0) {
-		%args = (
-			datadir	=> sprintf('%s/.local/share', $ENV{HOME}),
-			libdir	=> sprintf('%s/.%s/applets', $ENV{HOME}, lc($PerlPanel::NAME)),
-			regfile	=> sprintf('%s/.%s/applet.registry', $ENV{HOME}, lc($PerlPanel::NAME)),
-		);
-	} else {
 		%args = (
 			datadir	=> sprintf('%s/share', $PerlPanel::PREFIX),
 			libdir	=> sprintf('%s/%s/Applet', $PerlPanel::LIBDIR, $PerlPanel::NAME),
 			regfile	=> sprintf('%s/share/%s/applet.registry', $PerlPanel::PREFIX, lc($PerlPanel::NAME)),
 		);
-	}
 	print Dumper(\%args);
 	my ($code, $error) = PerlPanel::install_applet($file, %args);
 	if ($code == 1) {
