diff --git i/contrib/systemd/syslog-ng@.service w/contrib/systemd/syslog-ng@.service
index 349d23f2d..b0ade77e9 100644
--- i/contrib/systemd/syslog-ng@.service
+++ w/contrib/systemd/syslog-ng@.service
@@ -8,9 +8,8 @@ After=network.target network-online.target
 [Service]
 Type=notify
 EnvironmentFile=-/etc/default/syslog-ng@%i
-EnvironmentFile=-/etc/sysconfig/syslog-ng@%i
-ExecStart=/usr/sbin/syslog-ng -F $OTHER_OPTIONS --cfgfile $CONFIG_FILE --control $CONTROL_FILE --persist-file $PERSIST_FILE --pidfile $PID_FILE
-ExecReload=/bin/kill -HUP $MAINPID
+ExecStart=/usr/bin/syslog-ng -F $OTHER_OPTIONS --cfgfile $CONFIG_FILE --control $CONTROL_FILE --persist-file $PERSIST_FILE --pidfile $PID_FILE
+ExecReload=/usr/bin/kill -HUP $MAINPID
 StandardOutput=journal
 StandardError=journal
 Restart=on-failure
diff --git i/contrib/systemd/syslog-ng@default w/contrib/systemd/syslog-ng@default
index 0ccc2b9be..48316a15a 100644
--- i/contrib/systemd/syslog-ng@default
+++ w/contrib/systemd/syslog-ng@default
@@ -1,5 +1,5 @@
 CONFIG_FILE=/etc/syslog-ng/syslog-ng.conf
 PERSIST_FILE=/var/lib/syslog-ng/syslog-ng.persist
-CONTROL_FILE=/var/run/syslog-ng.ctl
-PID_FILE=/var/run/syslog-ng.pid
+CONTROL_FILE=/run/syslog-ng.ctl
+PID_FILE=/run/syslog-ng.pid
 OTHER_OPTIONS="--enable-core"
