diff --git i/scl/syslog-ng.conf w/scl/syslog-ng.conf
index e2d630200..66c7bcdaa 100644
--- i/scl/syslog-ng.conf
+++ w/scl/syslog-ng.conf
@@ -28,11 +28,249 @@ destination d_local {
 	file("/var/log/messages");
 	file("/var/log/messages-kv.log" template("$ISODATE $HOST $(format-welf --scope all-nv-pairs)\n") frac-digits(3));
 };
+destination d_acpid {
+	file("/var/log/acpid.log");
+};
+destination d_authlog {
+	file("/var/log/auth.log");
+};
+destination d_console {
+	usertty("root");
+};
+destination d_cron {
+	file("/var/log/crond.log");
+};
+destination d_daemon {
+	file("/var/log/daemon.log");
+};
+destination d_debug {
+	file("/var/log/debug.log");
+};
+destination d_errors {
+	file("/var/log/errors.log");
+};
+destination d_everything {
+	file("/var/log/everything.log");
+};
+destination d_iptables {
+	file("/var/log/iptables.log");
+};
+destination d_kernel {
+	file("/var/log/kernel.log");
+};
+destination d_lpr {
+	file("/var/log/lpr.log");
+};
+destination d_mail {
+	file("/var/log/mail.log");
+};
+destination d_messages {
+	file("/var/log/messages.log");
+};
+destination d_news {
+	file("/var/log/news.log");
+};
+destination d_ppp {
+	file("/var/log/ppp.log");
+};
+destination d_syslog {
+	file("/var/log/syslog.log");
+};
+# Log everything to tty12
+destination d_tty12 {
+	file("/dev/tty12");
+};
+destination d_user {
+	file("/var/log/user.log");
+};
+destination d_uucp {
+	file("/var/log/uucp.log");
+};
+
+filter f_acpid {
+	program("acpid");
+};
+filter f_auth {
+	facility(auth);
+};
+filter f_authpriv {
+	facility(auth, authpriv);
+};
+filter f_crit {
+	level(crit);
+};
+filter f_cron {
+	facility(cron);
+};
+filter f_daemon {
+	facility(daemon);
+};
+filter f_debug {
+	not facility(auth, authpriv, news, mail);
+};
+filter f_emergency {
+	level(emerg);
+};
+filter f_err {
+	level(err);
+};
+filter f_everything {
+	level(debug..emerg) and not facility(auth, authpriv);
+};
+filter f_info {
+	level(info);
+};
+filter f_iptables {
+	match("IN=" value("MESSAGE")) and match("OUT=" value("MESSAGE"));
+};
+filter f_kernel {
+	facility(kern) and not filter(f_iptables);
+};
+filter f_lpr {
+	facility(lpr);
+};
+filter f_mail {
+	facility(mail);
+};
+filter f_messages {
+	level(info..warn) and not facility(auth, authpriv, mail, news, cron) and not program(syslog-ng) and not filter(f_iptables);
+};
+filter f_news {
+	facility(news);
+};
+filter f_notice {
+	level(notice);
+};
+filter f_ppp {
+	facility(local2);
+};
+filter f_syslog {
+	program(syslog-ng);
+};
+filter f_user {
+	facility(user);
+};
+filter f_uucp {
+	facility(uucp);
+};
+filter f_warn {
+	level(warn);
+};
 
 log {
 	source(s_local);
 
 	# uncomment this line to open port 514 to receive messages
 	#source(s_network);
-	destination(d_local);
+	# destination(d_local);
+};
+log {
+	source(s_local);
+	# filter(f_acpid);
+	# destination(d_acpid);
+};
+log {
+	source(s_local);
+	# filter(f_authpriv);
+	# destination(d_authlog);
+};
+log {
+	source(s_local);
+	# filter(f_cron);
+	# destination(d_cron);
+};
+log {
+	source(s_local);
+	# filter(f_daemon);
+	# destination(d_daemon);
+};
+log {
+	source(s_local);
+# 	filter(f_debug);
+# 	destination(d_debug);
+};
+log {
+	source(s_local);
+	# filter(f_emergency);
+	# destination(d_console);
+};
+log {
+	source(s_local);
+	# filter(f_err);
+	# destination(d_errors);
+};
+log {
+	source(s_local);
+	# filter(f_everything);
+	# destination(d_everything);
+};
+log {
+	source(s_local);
+	# filter(f_iptables);
+	# destination(d_iptables);
+};
+log {
+	source(s_local);
+	# filter(f_kernel);
+	# destination(d_kernel);
+};
+log {
+	source(s_local);
+	# filter(f_lpr);
+	# destination(d_lpr);
+};
+log {
+	source(s_local);
+	# filter(f_mail);
+	# destination(d_mail);
+};
+log {
+	source(s_local);
+	# filter(f_messages);
+	# destination(d_messages);
+};
+log {
+	source(s_local);
+	# filter(f_news);
+	# destination(d_news);
+};
+log {
+	source(s_local);
+	# filter(f_ppp);
+	# destination(d_ppp);
+};
+log {
+	source(s_local);
+	# filter(f_user);
+	# destination(d_user);
+};
+log {
+	source(s_local);
+	# filter(f_uucp);
+	# destination(d_uucp);
+};
+log {
+	source(s_local);
+	# filter(f_syslog);
+	# destination(d_syslog);
+};
+# Log everything to tty12
+log {
+	source(s_local);
+	# destination(d_tty12);
+};
+
+options {
+	chain_hostnames(off);
+	create_dirs(no);
+	dns_cache(no);
+	flush_lines(0);
+	group("log");
+	keep_hostname(yes);
+	log_fifo_size(10000);
+	perm(0640);
+	stats(freq(0));
+	time_reopen(10);
+	use_dns(no);
+	use_fqdn(no);
 };
