diff -u -r NetworkManager-0.9.2.0/src/dns-manager/nm-dns-dnsmasq.c NetworkManager-0.9.2.0-dnsmasq/src/dns-manager/nm-dns-dnsmasq.c
--- NetworkManager-0.9.2.0/src/dns-manager/nm-dns-dnsmasq.c	2011-10-10 23:38:20.000000000 +0200
+++ NetworkManager-0.9.2.0-dnsmasq/src/dns-manager/nm-dns-dnsmasq.c	2012-03-06 09:07:24.137460927 +0100
@@ -52,8 +52,11 @@
 find_dnsmasq (void)
 {
 	static const char *paths[] = {
+		"/usr/local/bin/dnsmasq",
 		"/usr/local/sbin/dnsmasq",
+		"/usr/bin/dnsmasq",
 		"/usr/sbin/dnsmasq",
+		"/bin/dnsmasq",
 		"/sbin/dnsmasq",
 		NULL
 	};
diff -u -r NetworkManager-0.9.2.0/src/dnsmasq-manager/nm-dnsmasq-manager.c NetworkManager-0.9.2.0-dnsmasq/src/dnsmasq-manager/nm-dnsmasq-manager.c
--- NetworkManager-0.9.2.0/src/dnsmasq-manager/nm-dnsmasq-manager.c	2011-04-19 07:06:22.000000000 +0200
+++ NetworkManager-0.9.2.0-dnsmasq/src/dnsmasq-manager/nm-dnsmasq-manager.c	2012-03-06 09:07:58.603851818 +0100
@@ -170,8 +170,11 @@
 nm_find_dnsmasq (void)
 {
 	static const char *dnsmasq_binary_paths[] = {
+		"/usr/local/bin/dnsmasq",
 		"/usr/local/sbin/dnsmasq",
+		"/usr/bin/dnsmasq",
 		"/usr/sbin/dnsmasq",
+		"/bin/dnsmasq",
 		"/sbin/dnsmasq",
 		NULL
 	};
