Index: FreeCAD-0.19.2/CMakeLists.txt
===================================================================
--- FreeCAD-0.19.2.orig/CMakeLists.txt
+++ FreeCAD-0.19.2/CMakeLists.txt
@@ -46,6 +46,7 @@ if(NOT FREECAD_LIBPACK_USE OR FREECAD_LI
     SetupPCL()
     SetupPybind11()
     SetupBoost()
+	set(Boost_LIBRARIES_App ${Boost_LIBRARIES})
     SetupXercesC()
     find_package(ZLIB REQUIRED)
     find_package(PyCXX REQUIRED)
Index: FreeCAD-0.19.2/src/App/CMakeLists.txt
===================================================================
--- FreeCAD-0.19.2.orig/src/App/CMakeLists.txt
+++ FreeCAD-0.19.2/src/App/CMakeLists.txt
@@ -59,7 +59,7 @@ include_directories(
 
 set(FreeCADApp_LIBS
     FreeCADBase
-    ${Boost_LIBRARIES}
+    ${Boost_LIBRARIES_App}
 )
 
 if (BUILD_QT5)
