From 623ea5c4d2871aba2e3e5114ad7d58d6f910a2f3 Mon Sep 17 00:00:00 2001
From: Caolán McNamara <caolanm@redhat.com>
Date: Thu, 12 Jan 2023 20:55:27 +0000
Subject: [PATCH] Related: tdf#151898 fix gtk4 build in 7-4

Change-Id: I71b59b7d2eac406817c79bf6911736ad4510771c
---

diff --git a/vcl/unx/gtk3/gtkinst.cxx b/vcl/unx/gtk3/gtkinst.cxx
index 8ba6409..b697b0b 100644
--- a/vcl/unx/gtk3/gtkinst.cxx
+++ b/vcl/unx/gtk3/gtkinst.cxx
@@ -5114,7 +5114,7 @@
 
     void picture_set_from_xgraphic(GtkPicture* pPicture, const css::uno::Reference<css::graphic::XGraphic>& rPicture)
     {
-        if (auto xTempFile = getImageFile(rPicture, false))
+        if (auto xTempFile = getImageFile(rPicture))
             gtk_picture_set_filename(pPicture, OUStringToOString(xTempFile->GetFileName(), osl_getThreadTextEncoding()).getStr());
         else
             gtk_picture_set_pixbuf(pPicture, nullptr);
