diff --git a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
index 56e9afe18..f2ed6e8ee 100644
--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
+++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
@@ -1117,7 +1117,7 @@ void PDFOutDev::drawChar(GfxState *state, double x, double y,
 #if POPPLER_CHECK_VERSION(26, 4, 0)
 void PDFOutDev::drawString(GfxState*, const std::string& /*s*/)
 #elif POPPLER_CHECK_VERSION(0, 64, 0)
-void PDFOutDev::drawString(GfxState*, const GooString* /*s*/)
+void PDFOutDev::drawString(GfxState*, const std::string & /*s*/)
 #else
 void PDFOutDev::drawString(GfxState*, GooString* /*s*/)
 #endif
diff --git a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
index aa8d5445e..5f644445e 100644
--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
+++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
@@ -263,7 +263,7 @@ void setDefaultCTM(const std::array<double, 6> &pMat) override;
 #if POPPLER_CHECK_VERSION(26, 4, 0)
         virtual void drawString(GfxState *state, const std::string &s) override;
 #elif POPPLER_CHECK_VERSION(0, 64, 0)
-        virtual void drawString(GfxState *state, const GooString *s) override;
+        virtual void drawString(GfxState *state, const std::string &s) override;
 #else
         virtual void drawString(GfxState *state, GooString *s) override;
 #endif
