diff --git a/include/vcl/font/Feature.hxx b/include/vcl/font/Feature.hxx
index 67c24dd44..2696fc2c0 100644
--- a/include/vcl/font/Feature.hxx
+++ b/include/vcl/font/Feature.hxx
@@ -14,6 +14,7 @@
 #include <rtl/character.hxx>
 #include <rtl/ustring.hxx>
 #include <unotools/resmgr.hxx>
+#include <cstdint>
 #include <vector>
 
 namespace vcl::font
diff --git a/starmath/inc/mathml/def.hxx b/starmath/inc/mathml/def.hxx
index 380736711..6952fe311 100644
--- a/starmath/inc/mathml/def.hxx
+++ b/starmath/inc/mathml/def.hxx
@@ -10,6 +10,7 @@
 #pragma once
 
 #include <tools/color.hxx>
+#include <cstdint>
 
 /* All possible data needed to do the job outside mathml limits */
 // Ml prefix means it is part of mathml standard
diff --git a/svl/source/notify/broadcast.cxx b/svl/source/notify/broadcast.cxx
index 6b323a669..0c8cb0b22 100644
--- a/svl/source/notify/broadcast.cxx
+++ b/svl/source/notify/broadcast.cxx
@@ -22,6 +22,7 @@
 #include <svl/hint.hxx>
 #include <o3tl/safeint.hxx>
 #include <cassert>
+#include <cstdint>
 #include <algorithm>
 
 /**
