--- libreoffice-25.2.6.2/configure.ac	2025-09-17 20:24:49.957639858 +0200
+++ libreoffice-25.2.6.2/configure.ac.new	2025-09-17 20:27:53.709450712 +0200
@@ -11420,7 +11420,7 @@
 dnl ===================================================================
 dnl Orcus
 dnl ===================================================================
-libo_CHECK_SYSTEM_MODULE([orcus],[ORCUS],[liborcus-0.20 >= 0.20.0])
+libo_CHECK_SYSTEM_MODULE([orcus],[ORCUS],[liborcus-0.21 >= 0.21.0])
 if test "$with_system_orcus" != "yes"; then
     if test "$SYSTEM_BOOST" = "TRUE"; then
         dnl Link with Boost.System
