--- sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx	2018-01-30 18:22:07.000000000 +0100
+++ sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx.new	2018-03-20 11:38:16.558290319 +0100
@@ -32,10 +32,12 @@
 #pragma warning(push, 1)
 #endif
 
-// sigh, UTF8.h was removed in poppler-0.21.0 and put back in 0.21.1
+// sigh, UTF8.h was removed in poppler-0.21.0 and put back in 0.21.1, then renamed to UnicodeMapFuncs.h in 0.62.0
 // FIXME: we can't use #if POPPLER_CHECK_VERSION(0, 21, 0) && !POPPLER_CHECK_VERSION(0, 21, 1)
 //        because the internal poppler does not provide poppler-version.h and the macro always returns 0
-#if POPPLER_CHECK_VERSION(0, 21, 1)
+#if POPPLER_CHECK_VERSION(0, 62, 0)
+#include <UnicodeMapFuncs.h>
+#elif POPPLER_CHECK_VERSION(0, 21, 1)
 #include "UTF8.h"
 #elif POPPLER_CHECK_VERSION(0, 21, 0)
 #include "UTF.h"
@@ -918,7 +920,11 @@
             );
 
     // silence spurious warning
+#if POPPLER_CHECK_VERSION(0, 62, 0)
+    (void)&mapUTF16;
+#else
     (void)&mapUCS2;
+#endif
 
     char buf[9];
     for( int i=0; i<uLen; ++i )
