From 04b5a1ddac1dd90c5a392f23aaf7df42eae32918 Mon Sep 17 00:00:00 2001
From: Gaurav Sood <gaurav-sood@invalid-email.launchpad.net>
Date: Thu, 4 May 2017 18:32:08 +0200
Subject: Fix handling of unicode filenames in sixext.py

LP: #1480152
Closes: #861731
---
 base/sixext.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/base/sixext.py b/base/sixext.py
index 0bf4fc4f4..311bf72c6 100644
--- a/base/sixext.py
+++ b/base/sixext.py
@@ -110,11 +110,11 @@ if PY3:
 
 
     def to_string_utf8(s):
-        return s.decode("utf-8")
+        return s.decode("utf-8", 'ignore')
 
 
     def to_string_latin(s):
-        return s.decode("latin-1")
+        return s.decode("latin-1", 'ignore')
 
 
     def to_unicode(s, enc=None):
