--- hplip-3.26.4/hp-uiscan.desktop.in	2026-05-20 18:43:44.000000000 +0200
+++ hplip-3.26.4/hp-uiscan.desktop.in.new	2026-05-20 20:41:21.777397333 +0200
@@ -1,8 +1,11 @@
-#!/usr/bin/env xdg-open
 [Desktop Entry]
 Version=1.0
 Type=Application
 Terminal=false
-Name=hp-uiscan
+Name=HP Scan
+GenericName=Scanner
+Comment=HP Printer/Scanner Utility - Scan
 Exec=/usr/bin/hp-uiscan
 Icon=/usr/share/icons/Humanity/devices/48/printer.svg
+Categories=Graphics;Scanning;
+Keywords=scan;scanner;hp;hplip;printer;
