--- a/scan.py	2026-03-14 00:34:53.088030280 +0300
+++ b/scan.py	2026-03-14 00:46:23.890189400 +0300
@@ -1716,8 +1716,9 @@
         #resize the image here to the original scan area size, 
         #so that the output image size matches with the input scan area
         #for Flatbad 
-        if im:    
-            im = imageprocessing.resize_to_scan_area(im,PAGE_SIZES[size],res)
+        if im and page_size in PAGE_SIZES:
+            im = imageprocessing.resize_to_scan_area(im, PAGE_SIZES[page_size], res)
+
         """
         #for ADF
         for Image_file in adf_page_files:
