diff --git a/base/device.py b/base/device.py
index 2fdb29c..ca0d97e 100644
--- a/base/device.py
+++ b/base/device.py
@@ -2526,9 +2526,9 @@ Content-length: %d\r
                 data = None
 
             log.debug("Opening: %s" % url2)
-            opener = LocalOpener({})
+            opener = LocalOpener()
             try:
-                f = opener.open(url2, data)
+                f = opener.open_hp(url2, data)
                 
             except Error:
                 log.error("Status read failed: %s" % url2)
@@ -2548,7 +2548,7 @@ Content-length: %d\r
         try:
             url2 = "%s&loc=%s" % (self.device_uri.replace('hpfax:', 'hp:'), url)
             data = self
-            opener = LocalOpenerEWS_LEDM({})
+            opener = LocalOpenerEWS_LEDM()
             try:
                 if footer:
                     return opener.open_hp(url2, data, footer)
@@ -2563,7 +2563,7 @@ Content-length: %d\r
         try:
             url2 = "%s&loc=%s" % (self.device_uri.replace('hpfax:', 'hp:'), url)
             data = self
-            opener = LocalOpener_LEDM({})
+            opener = LocalOpener_LEDM()
             try:
                 if footer:
                     return opener.open_hp(url2, data, footer)
@@ -2580,7 +2580,7 @@ Content-length: %d\r
             url="/cdm/supply/v1/suppliesPublic"
             url2 = "%s&loc=%s" % (self.device_uri.replace('hpfax:', 'hp:'), url)
             data = self
-            opener = LocalOpener_CDM({})
+            opener = LocalOpener_CDM()
             try:
                 if footer:
                     return opener.open_hp(url2, data, footer)
@@ -2688,7 +2688,7 @@ Content-length: %d\r
 
 
 # URLs: hp:/usb/HP_LaserJet_3050?serial=00XXXXXXXXXX&loc=/hp/device/info_device_status.xml
-class LocalOpener(urllib_request.URLopener):
+class LocalOpener():
     def open_hp(self, url, dev):
         log.debug("open_hp(%s)" % url)
 
@@ -2718,7 +2718,7 @@ class LocalOpener(urllib_request.URLopener):
             return response#.fp
 
 # URLs: hp:/usb/HP_OfficeJet_7500?serial=00XXXXXXXXXX&loc=/hp/device/info_device_status.xml
-class LocalOpenerEWS_LEDM(urllib_request.URLopener):
+class LocalOpenerEWS_LEDM():
     def open_hp(self, url, dev, foot=""):
         log.debug("open_hp(%s)" % url)
 
@@ -2744,7 +2744,7 @@ class LocalOpenerEWS_LEDM(urllib_request.URLopener):
 
 
 # URLs: hp:/usb/HP_OfficeJet_7500?serial=00XXXXXXXXXX&loc=/hp/device/info_device_status.xml
-class LocalOpener_LEDM(urllib_request.URLopener):
+class LocalOpener_LEDM():
     def open_hp(self, url, dev, foot=""):
         log.debug("open_hp(%s)" % url)
 
@@ -2771,7 +2771,7 @@ class LocalOpener_LEDM(urllib_request.URLopener):
 
 
 # URLs: hp:/usb/HP_OfficeJet_7500?serial=00XXXXXXXXXX&loc=/hp/device/info_device_status.xml
-class LocalOpener_CDM(urllib_request.URLopener):
+class LocalOpener_CDM():
     def open_hp(self, url, dev, foot=""):
         log.debug("open_hp(%s)" % url)
         match_obj = http_pat_url.search(url)
