diff -Nru hplip-3.23.12.orig/prnt/hpps/pserror.c hplip-3.23.12/prnt/hpps/pserror.c
--- hplip-3.23.12.orig/prnt/hpps/pserror.c	2023-11-27 15:06:40.000000000 +0000
+++ hplip-3.23.12/prnt/hpps/pserror.c	2024-06-06 07:28:05.765351553 +0000
@@ -24,7 +24,7 @@
 void message(int flags, char *format, ...)
 {
   va_list args ;
-  static column = 0 ;		/* current screen column for message wrap */
+  static int column = 0 ;	/* current screen column for message wrap */
   char msgbuf[MAX_MESSAGE] ;	/* buffer in which to put the message */
   char *bufptr = msgbuf ;	/* message buffer pointer */
 
