--- hplip-3.20.5/createPPD.sh	2020-05-11 15:34:06.000000000 +0200
+++ hplip-3.20.5.old/createPPD.sh	2020-05-27 12:02:44.089477299 +0200
@@ -54,15 +54,15 @@
 	else
 		cp prnt/ps/hp-postscript*.ppd ppd/classppd/ps
 		for i in ppd/classppd/ps/*.ppd; do
-			gzip -c "$i" > "$i.gz"
+			gzip -nc "$i" > "$i.gz"
 		done
 	fi
         #cp prnt/ps/hp-postscript*.ppd.gz ppd/classppd/ps
 	for i in ppd/classppd/hpcups/*.ppd; do
-		gzip -c "$i" > "$i.gz"
+		gzip -nc "$i" > "$i.gz"
 	done
 	#for i in ppd/classppd/ps/*.ppd; do
-	#	gzip -c "$i" > "$i.gz"
+	#	gzip -nc "$i" > "$i.gz"
 	#done
 
 	cat > class_cups_drv.inc <<EOF
@@ -81,14 +81,14 @@
 install -d ppd/hpcups
 ppdc -d ppd/hpcups prnt/drv/hpcups.drv
 for i in ppd/hpcups/*.ppd; do
-      gzip -c "$i" > "$i.gz"
+      gzip -nc "$i" > "$i.gz"
 done
 
 install -d ppd/hpijs
 ppdc -d ppd/hpijs prnt/drv/hpijs.drv
 echo "gzip ppd files..."
 for i in ppd/hpijs/*.ppd; do
-      gzip -c "$i" > "$i.gz"
+      gzip -nc "$i" > "$i.gz"
 done
 echo "building foomatic_drv.inc..."
 ./hpijs-drv > foomatic_drv.inc $shadow
