--- hplip-3.22.4/createPPD.sh	2022-04-26 06:16:59.000000000 +0200
+++ hplip-3.22.4/createPPD.sh.new	2022-04-30 14:42:35.877365900 +0200
@@ -55,16 +55,16 @@
 	else
 		cp prnt/ps/hp-postscript*.ppd ppd/classppd/ps
 		for i in ppd/classppd/ps/*.ppd; do
-			gzip -c "$i" > "$i.gz"
+			gzip -nc "$i" > "$i.gz"
 		done
 	fi
 	#cp prnt/spd/*.ppd ppd/classppd/hpcups/.
     #cp prnt/ps/hp-postscript*.ppd.gz ppd/classppd/ps
 	for i in ppd/classppd/hpcups/*.ppd; do
-		gzip -c "$i" > "$i.gz"
+		gzip -nc "$i" > "$i.gz"
 	done
 	#for i in ppd/classppd/ps/*.ppd; do
-	#	gzip -c "$i" > "$i.gz"
+	#	gzip -nc "$i" > "$i.gz"
 	#done
 
 	cat > class_cups_drv.inc <<EOF
@@ -83,14 +83,14 @@
 install -d ppd/hpcups
 ppdc -d ppd/hpcups prnt/drv/hpcups.drv
 for i in ppd/hpcups/*.ppd; do
-      gzip -c "$i" > "$i.gz"
+      gzip -nc "$i" > "$i.gz"
 done
 
 install -d ppd/hpijs
 ppdc -d ppd/hpijs prnt/drv/hpijs.drv
 echo "gzip ppd files..."
 for i in ppd/hpijs/*.ppd; do
-      gzip -c "$i" > "$i.gz"
+      gzip -nc "$i" > "$i.gz"
 done
 echo "building foomatic_drv.inc..."
 ./hpijs-drv > foomatic_drv.inc $shadow
