From dce379fc81564dea21306935463fc42419a3e089 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sat, 10 Jan 2015 11:51:52 -0700
Subject: [PATCH 3/3] environment fixes

---
 include/configs/hardkernel/odroidc.h | 34 ++++++++++++++++------------------
 1 file changed, 16 insertions(+), 18 deletions(-)

diff --git a/include/configs/hardkernel/odroidc.h b/include/configs/hardkernel/odroidc.h
index 2099f3b..2bc8b2c 100644
--- a/include/configs/hardkernel/odroidc.h
+++ b/include/configs/hardkernel/odroidc.h
@@ -119,21 +119,12 @@
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
         "boardname=ODROIDC\0" \
-        "bootargs=root=" MK_STR(CONFIG_DEFAULT_ROOT) " rw " \
-                "console=" CONFIG_CONSOLE_PROTOCOL " no_console_suspend\0" \
-        "bootcmd="\
-                "cfgload;" \
-                "setenv bootargs root=" MK_STR(CONFIG_DEFAULT_ROOT) " rw " \
-                        "console=" CONFIG_CONSOLE_PROTOCOL " no_console_suspend " \
-                        "vdaccfg=${vdac_config} " \
-                        "logo=osd1,loaded,${fb_addr},${outputmode},full " \
-                        "hdmimode=${hdmimode} " \
-                        "cvbsmode=${cvbsmode} " \
-                        "hdmitx=${cecconfig} " \
-                        "androidboot.serialno=${fbt_id#}; " \
-                "movi read boot 0 0x12000000;" \
-                "movi read dtb 0 0x12800000;" \
-                "bootm 0x12000000 - 0x12800000\0" \
+        "optargs=\0" \
+        "mmcargs=setenv bootargs console=ttyS0,115200n8 no_console_suspend " \
+                "root=/dev/mmcblk0p1 rw rootwait " \
+                "vdaccfg=0xa000 logo=osd1,loaded,0x7900000,720p,full dmfc=3 cvbsmode=576cvbs " \
+                "hdmimode=${m} m_bpp=${m_bpp} vout=${vout_mode} ${disableuhs} hdmitx=cec1 " \
+                "${optargs}\0" \
         "bootm_low=0x00000000\0" \
         "bootm_size=0x80000000\0" \
         "bootpath=u-boot.bin\0" \
@@ -163,9 +154,16 @@
                 "movi read logo 0 ${loadaddr_logo};" \
                 "bmp display ${loadaddr_logo}; bmp scale\0" \
         "vdac_config=0x10\0" \
-        "video_dev=tvout\0"
-
-#define CONFIG_BOOTCOMMAND              "bootm"
+        "video_dev=tvout\0" \
+        "loadimage=ext4load mmc 0:1 0x21000000 /boot/uImage\0" \
+        "loadfdt=ext4load mmc 0:1 0x21800000 /boot/dtbs/meson8b_odroidc.dtb\0"
+
+#define CONFIG_BOOTCOMMAND \
+        "cfgload;" \
+        "run loadimage;" \
+        "run loadfdt;" \
+        "run mmcargs;" \
+        "bootm 0x21000000 - 0x21800000;"
 
 #define CONFIG_AUTO_COMPLETE            1
 #define CONFIG_ENV_SIZE                 (32 * 1024)     // unit: bytes
-- 
2.2.1

