--- a/Makefile	2011-07-01 11:18:57.000000000 +0800
+++ b/Makefile	2014-07-13 01:21:41.887559397 +0800
@@ -128,7 +128,7 @@
 ifeq ($(TARGET_OS):$(TARGET_ARCH),linux:i386)
 npviewer_SOURCES += $(SRC_PATH)/src/cxxabi-compat.cpp
 npviewer_OBJECTS += npviewer-cxxabi-compat.o
-npviewer_LIBS    += -lsupc++
+npviewer_LIBS    += -lsupc++ -lgthread-2.0
 endif
 ifeq ($(TARGET_OS):$(TARGET_ARCH),solaris:i386)
 npviewer_LIBS    += $(libsocket_LIBS)
@@ -142,7 +142,7 @@
 npplayer_LDFLAGS  = $(LDFLAGS)
 npplayer_LDFLAGS += $(libpthread_LDFLAGS)
 npplayer_LIBS     = $(GTK_LIBS) $(GLIB_LIBS) $(CURL_LIBS) $(X_LIBS)
-npplayer_LIBS    += $(libpthread_LIBS) $(libsocket_LIBS)
+npplayer_LIBS    += $(libpthread_LIBS) $(libsocket_LIBS) -ldl -lgthread-2.0
 
 libnoxshm_LIBRARY = libnoxshm.so
 libnoxshm_RAWSRCS = libnoxshm.c
