--- tachyon/unix/Make-arch.orig	2019-08-24 23:13:05.136655180 +0000
+++ tachyon/unix/Make-arch	2019-08-24 23:13:36.057062493 +0000
@@ -1217,12 +1217,12 @@
 	$(MAKE) all \
 	"ARCH = linux-64-thr" \
 	"CC = gcc" \
-	"CFLAGS = -m64 -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux -DLP64 -DTHR -D_REENTRANT $(MISCFLAGS)" \
+	"CFLAGS = $(CFLAGS) -DLinux -DLP64 -DTHR -D_REENTRANT $(MISCFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = strip" \
 	"RANLIB = ranlib" \
-	"LIBS = -L. -ltachyon $(MISCLIB) -lm -lpthread"
+	"LIBS = -L. -ltachyon $(MISCLIB) -lm -lpthread" "LDFLAGS = $(LDFLAGS)"
 
 # Linux x86, on AMD Athlon, using gcc
 linux-p4:
