From d3e9ed9d66d16bd9beb17f9ab7970499ef039adb Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sat, 27 Jun 2015 12:37:40 -0600
Subject: [PATCH] arch linux arm modifications

---
 include/configs/usbarmory.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/usbarmory.h b/include/configs/usbarmory.h
index 23092db..995532a 100644
--- a/include/configs/usbarmory.h
+++ b/include/configs/usbarmory.h
@@ -23,6 +23,8 @@
 
 #include <config_distro_defaults.h>
 
+#define CONFIG_IDENT_STRING " Arch Linux ARM"
+
 /* U-Boot environment */
 #define CONFIG_ENV_OFFSET	(6 * 64 * 1024)
 #define CONFIG_ENV_SIZE		(8 * 1024)
-- 
2.10.0

