From bc3a7ffa4219dd717b88cf2b38d7476bb9fa2efb Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sat, 27 Jun 2015 12:37:40 -0600
Subject: [PATCH] arch linux arm modifications

---
 include/configs/usbarmory.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/usbarmory.h b/include/configs/usbarmory.h
index 8568663..8abae4b 100644
--- a/include/configs/usbarmory.h
+++ b/include/configs/usbarmory.h
@@ -22,6 +22,8 @@
 
 #include <config_distro_defaults.h>
 
+#define CONFIG_IDENT_STRING " Arch Linux ARM"
+
 /* U-Boot commands */
 
 /* U-Boot environment */
-- 
2.9.0

