diff --git a/include/configs/usbarmory.h b/include/configs/usbarmory.h
index e00ec7b..cfd457b 100644
--- a/include/configs/usbarmory.h
+++ b/include/configs/usbarmory.h
@@ -28,6 +28,8 @@
 #define CONFIG_CMD_FAT
 #define CONFIG_CMD_MEMTEST
 #define CONFIG_CMD_EXT2
+#define CONFIG_CMD_EXT4
+#define CONFIG_CMD_FS_GENERIC
 #undef CONFIG_CMD_IMLS
 
 /* U-Boot environment */
@@ -62,6 +64,10 @@
 #define CONFIG_MMC
 #define CONFIG_GENERIC_MMC
 #define CONFIG_DOS_PARTITION
+#define CONFIG_CMD_GPT
+#define CONFIG_EFI_PARTITION
+#define CONFIG_PARTITION_UUIDS
+#define CONFIG_CMD_PART
 
 /* USB */
 #define CONFIG_CMD_USB
@@ -87,11 +93,65 @@
 #define CONFIG_SYS_TEXT_BASE	0x77800000
 #define CONFIG_SYS_LOAD_ADDR	CONFIG_LOADADDR
 #define CONFIG_HOSTNAME		usbarmory
-#define CONFIG_BOOTARGS \
-	"console=ttymxc0,115200 root=/dev/mmcblk0p1 rootwait rw"
+#define CONFIG_IDENT_STRING " Arch Linux ARM"
+#define CONFIG_SUPPORT_RAW_INITRD
+#define CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG
+#define CONFIG_EXTRA_ENV_SETTINGS \
+    "loadaddr=0x70800000\0" \
+    "fdtaddr=0x71000000\0" \
+    "rdaddr=0x71100000\0" \
+    "boot_fdt=try\0" \
+    "bootpart=0:1\0" \
+    "bootdir=/boot\0" \
+    "bootfile=zImage\0" \
+    "fdtdir=/boot/dtbs\0" \
+    "fdtfile=imx53-usbarmory.dtb\0" \
+    "console=ttymxc0,115200n8\0" \
+    "partitions=" \
+        "uuid_disk=${uuid_gpt_disk};" \
+        "name=rootfs,start=2MiB,size=-,uuid=${uuid_gpt_rootfs}\0" \
+    "optargs=\0" \
+    "bootenv=uEnv.txt\0" \
+    "mmcdev=0\0" \
+    "mmcroot=/dev/mmcblk0p1\0" \
+    "mmcrootextra=rw fixrtc rootwait\0" \
+    "mmcargs=setenv booatrgs console=${console} " \
+        "${optargs} " \
+        "root=${mmcroot} ${mmcrootextra}\0" \
+    "loadramdisk=load mmc ${bootpart} ${rdaddr} ${bootdir}/${rdfile}\0" \
+    "loadimage=load mmc ${bootpart} ${loadaddr} ${bootdir}/${bootfile}\0" \
+    "loadfdt=load mmc ${bootpart} ${fdtaddr} ${fdtdir}/${fdtfile}\0" \
+    "loadbootenv=load mmc ${mmcdev} ${loadaddr} ${bootdir}/${bootenv}\0" \
+    "importbootenv=echo Importing environment from mmc ...; "\
+        "env import -t $loadaddr $filesize\0" \
+    "mmcboot=mmc dev ${mmcdev};" \
+        "if mmc rescan; then " \
+            "echo SD/MMC found on device ${mmcdev};" \
+            "if run loadbootenv; then " \
+                "echo Loaded environment from ${bootenv};" \
+                "run importbootenv;" \
+            "fi;"\
+            "if test -n $uenvcmd; then " \
+                "echo Running uenvcmd ...;" \
+                "run uenvcmd;" \
+            "fi;" \
+            "if run loadimage; then " \
+                "run mmcargs;" \
+                "if run loadfdt; then " \
+                    "if run loadramdisk; then " \
+                        "bootz ${loadaddr} ${rdaddr}:${filesize} ${fdtaddr};" \
+                    "else " \
+                        "bootz ${loadaddr} - ${fdtaddr};" \
+                    "fi;" \
+                "else " \
+                    "echo Kernel found, but no device tree found;" \
+                "fi;" \
+            "fi;" \
+            "echo No Kernel found;" \
+        "fi;"
+
 #define CONFIG_BOOTCOMMAND \
-	"ext2load mmc 0:1 0x70800000 /boot/uImage; ext2load mmc 0:1" \
-	"0x71000000 /boot/imx53-usbarmory.dtb; bootm 0x70800000 - 0x71000000"
+    "run mmcboot; "
 
 /* Physical Memory Map */
 #define CONFIG_NR_DRAM_BANKS		1
