diff -ruN a/arch/arm/mach-meson8b/hdmi_tx_hw/hdmi_tx_hw.c b/arch/arm/mach-meson8b/hdmi_tx_hw/hdmi_tx_hw.c
--- a/arch/arm/mach-meson8b/hdmi_tx_hw/hdmi_tx_hw.c	2021-02-03 08:03:56.000000000 -0700
+++ b/arch/arm/mach-meson8b/hdmi_tx_hw/hdmi_tx_hw.c	2021-03-26 11:29:07.462226352 -0600
@@ -217,8 +217,10 @@
     //udelay(us);
     if(delay_flag&0x1)
         mdelay((us+999)/1000);
+    else if (us >= 1000)
+        mdelay(us/1000);
     else
-    udelay(us);
+        udelay(us);
 } /* delay_us */
 
 static irqreturn_t intr_handler(int irq, void *dev_instance)
