From 2276b50d210fea2d61d7a32c1d16c69a7b3f32ef Mon Sep 17 00:00:00 2001
From: David Runge <dvzrv@archlinux.org>
Date: Sun, 24 May 2026 23:42:51 +0200
Subject: [PATCH 2/4] Adjust hardcoded font paths for ttf-dejavu

Signed-off-by: David Runge <dvzrv@archlinux.org>
---
 modules/visualization/projectm.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/visualization/projectm.cpp b/modules/visualization/projectm.cpp
index 208fa22a2e..56df7b7d72 100644
--- a/modules/visualization/projectm.cpp
+++ b/modules/visualization/projectm.cpp
@@ -82,8 +82,8 @@ static void Close        ( vlc_object_t * );
 # define FONT_PATH_MENU "C:\\WINDOWS\\Fonts\\arial.ttf"
 # define PRESET_PATH    NULL
 #else
-# define FONT_PATH      "/usr/share/fonts/truetype/ttf-dejavu/DejaVuSans.ttf"
-# define FONT_PATH_MENU "/usr/share/fonts/truetype/ttf-dejavu/DejaVuSansMono.ttf"
+# define FONT_PATH      "/usr/share/fonts/TTF/DejaVuSans.ttf"
+# define FONT_PATH_MENU "/usr/share/fonts/TTF/DejaVuSansMono.ttf"
 # define PRESET_PATH    "/usr/share/projectM/presets"
 #endif
 
