diff -u -r vlc-3.0.8/modules/codec/aom.c vlc-3.0.8-aom/modules/codec/aom.c
--- vlc-3.0.8/modules/codec/aom.c	2019-01-09 16:02:11.000000000 +0000
+++ vlc-3.0.8-aom/modules/codec/aom.c	2020-02-22 11:44:27.188620819 +0000
@@ -98,7 +98,6 @@
 { VLC_CODEC_I444, AOM_IMG_FMT_I444, 8, 0 },
 
 { VLC_CODEC_YV12, AOM_IMG_FMT_YV12, 8, 0 },
-{ VLC_CODEC_YUVA, AOM_IMG_FMT_444A, 8, 0 },
 
 { VLC_CODEC_GBR_PLANAR, AOM_IMG_FMT_I444, 8, 1 },
 { VLC_CODEC_GBR_PLANAR_10L, AOM_IMG_FMT_I44416, 10, 1 },
