From 9288c513fd7288ea85cd162aa610ca6d7e083992 Mon Sep 17 00:00:00 2001
From: anthraxx <levente@leventepolyak.net>
Date: Fri, 11 Jan 2018 23:57:53 +0100
Subject: [PATCH] avoid defining getter functions present in libupnp 1.6.24

---
 modules/services_discovery/upnp.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/services_discovery/upnp.cpp b/modules/services_discovery/upnp.cpp
index b265113..f18afd6 100644
--- a/modules/services_discovery/upnp.cpp
+++ b/modules/services_discovery/upnp.cpp
@@ -50,6 +50,7 @@ typedef Upnp_Action_Complete UpnpActionComplete;
 typedef Upnp_Event UpnpEvent;
 typedef Upnp_Event_Subscribe UpnpEventSubscribe;
 
+#if UPNP_VERSION < 10624
 static const char* UpnpDiscovery_get_Location_cstr( const UpnpDiscovery* p_discovery )
 {
   return p_discovery->Location;
@@ -74,6 +75,7 @@ static const char* UpnpEventSubscribe_get_SID_cstr( const UpnpEventSubscribe* p_
 {
   return p_s->Sid;
 }
+#endif
 #else
 typedef const void* UpnpEventPtr;
 #endif
-- 
2.15.1

