diff -ru vlc-3.0.13.orig/configure.ac vlc-3.0.13/configure.ac
--- vlc-3.0.13.orig/configure.ac        2021-04-27 15:18:39.000000000 +0200
+++ vlc-3.0.13/configure.ac     2021-05-03 16:10:00.342592591 +0200
@@ -4016,7 +4016,7 @@
 dnl
 dnl  SRT plugin
 dnl
-PKG_ENABLE_MODULES_VLC([SRT], [access_srt access_output_srt], [srt >= 1.2.2 srt < 1.3.0], [SRT input/output plugin], [auto], [], [], [-DENABLE_SRT])
+PKG_ENABLE_MODULES_VLC([SRT], [access_srt access_output_srt], [srt >= 1.2.2], [SRT input/output plugin], [auto], [], [], [-DENABLE_SRT])
 
 EXTEND_HELP_STRING([Visualisations and Video filter plugins:])
 dnl
diff -ruN a/modules/access/srt.c b/modules/access/srt.c
--- a/modules/access/srt.c	2018-04-23 10:03:39.000000000 +0200
+++ b/modules/access/srt.c	2020-09-26 10:35:18.830149487 +0200
@@ -165,7 +165,7 @@
 
     /* Set latency */
     i_latency = var_InheritInteger( p_stream, "latency" );
-    srt_setsockopt( p_sys->sock, 0, SRTO_TSBPDDELAY,
+    srt_setsockopt( p_sys->sock, 0, SRTO_LATENCY,
         &i_latency, sizeof( int ) );
 
     psz_passphrase = var_InheritString( p_stream, "passphrase" );
diff -ruN a/modules/access_output/srt.c b/modules/access_output/srt.c
--- a/modules/access_output/srt.c	2018-04-06 11:22:51.000000000 +0200
+++ b/modules/access_output/srt.c	2020-09-26 10:35:03.386828120 +0200
@@ -162,7 +162,7 @@
 
     /* Set latency */
     i_latency = var_InheritInteger( p_access, "latency" );
-    srt_setsockopt( p_sys->sock, 0, SRTO_TSBPDDELAY,
+    srt_setsockopt( p_sys->sock, 0, SRTO_LATENCY,
         &i_latency, sizeof( int ) );
 
     if ( psz_passphrase != NULL && psz_passphrase[0] != '\0')
