From ea444abae1d09b751314b24f6352e55be335621d Mon Sep 17 00:00:00 2001
From: Jean-Baptiste Kempf <jb@videolan.org>
Date: Thu, 25 Jul 2013 01:14:32 +0200
Subject: [PATCH] Fix live555 breakage in last update

---
 modules/demux/live555.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/modules/demux/live555.cpp b/modules/demux/live555.cpp
index 9e02630..7a1ab16 100644
--- a/modules/demux/live555.cpp
+++ b/modules/demux/live555.cpp
@@ -243,7 +243,11 @@ public:
                    char const* applicationName, portNumBits tunnelOverHTTPPortNum,
                    demux_sys_t *p_sys) :
                    RTSPClient( env, rtspURL, verbosityLevel, applicationName,
-                   tunnelOverHTTPPortNum )
+                   tunnelOverHTTPPortNum
+#if LIVEMEDIA_LIBRARY_VERSION_INT >= 1373932800
+                   , -1
+#endif
+                   )
     {
         this->p_sys = p_sys;
     }
-- 
1.8.3.4

