diff -u -r --new-file psutils.orig/Makefile.unix psutils/Makefile.unix
--- psutils.orig/Makefile.unix	Tue Mar 11 14:52:59 1997
+++ psutils/Makefile.unix	Thu Jan 18 20:53:36 2001
@@ -19,16 +19,16 @@
 # psnup puts multiple logical pages on one physical page
 # psresize scales and moves pages to fit on different paper sizes
 
-PAPER=a4
+PAPER=letter
 
 # Makefile for PSUtils under Unix
 
 OS = UNIX
 
-BINDIR = /usr/local/bin
+BINDIR = /usr/bin
 SCRIPTDIR = $(BINDIR)
-INCLUDEDIR = /usr/local/share/psutils
-PERL = /usr/local/bin/perl
+INCLUDEDIR = /usr/share/psutils
+PERL = /usr/bin/perl
 
 BINMODE = 0755
 MANMODE = 0644
@@ -36,7 +36,7 @@
 INSTALL = install -c -m $(BINMODE)
 INSTALLMAN = install -c -m $(MANMODE)
 MANEXT = 1
-MANDIR = /usr/local/share/man/man$(MANEXT)
+MANDIR = /usr/man/man$(MANEXT)
 
 CC = gcc
 CFLAGS = -DPAPER=\"$(PAPER)\" -DUNIX -O -Wall
