diff -Naur SDL-1.2.13-old/configure SDL-1.2.13/configure
--- SDL-1.2.13-old/configure	2007-12-31 15:09:39.000000000 +1000
+++ SDL-1.2.13/configure	2008-10-20 22:22:36.000000000 +1000
@@ -25382,7 +25382,7 @@
   enable_esd_shared=yes
 fi
 
-            esd_lib_spec=`echo $ESD_LIBS | sed 's/.*-L\([^ ]*\).*/\1\/libesd.so.*/'`
+            esd_lib_spec="/usr/lib/libesd.so.*"
             esd_lib=`ls -- $esd_lib_spec | sed 's/.*\/\(.*\)/\1/; q'`
             echo "-- $esd_lib_spec -> $esd_lib"
 
