diff -Nru swh-plugins-0.4.15.orig/mbeq_1197.c swh-plugins-0.4.15/mbeq_1197.c
--- swh-plugins-0.4.15.orig/mbeq_1197.c	2006-08-09 01:49:15.000000000 +1000
+++ swh-plugins-0.4.15/mbeq_1197.c	2011-06-05 10:57:17.929107102 +1000
@@ -457,7 +457,8 @@
 	#endif
 	
 	                // Multiply the bins magnitudes by the coeficients
-	                for (i = 0; i < FFT_LENGTH/2; i++) {
+	                comp[0] *= coefs[0];
+	                for (i = 1; i < FFT_LENGTH/2; i++) {
 	                        comp[i] *= coefs[i];
 	                        comp[FFT_LENGTH-i] *= coefs[i];
 	                }
diff -Nru swh-plugins-0.4.15.orig/mbeq_1197.so.c swh-plugins-0.4.15/mbeq_1197.so.c
--- swh-plugins-0.4.15.orig/mbeq_1197.so.c	2003-10-22 02:13:40.000000000 +1000
+++ swh-plugins-0.4.15/mbeq_1197.so.c	2011-06-05 10:57:52.265774336 +1000
@@ -446,7 +446,8 @@
 	#endif
 	
 	                // Multiply the bins magnitudes by the coeficients
-	                for (i = 0; i < FFT_LENGTH/2; i++) {
+	                comp[0] *= coefs[0];
+	                for (i = 1; i < FFT_LENGTH/2; i++) {
 	                        comp[i] *= coefs[i];
 	                        comp[FFT_LENGTH-i] *= coefs[i];
 	                }
diff -Nru swh-plugins-0.4.15.orig/mbeq_1197.xml swh-plugins-0.4.15/mbeq_1197.xml
--- swh-plugins-0.4.15.orig/mbeq_1197.xml	2005-06-01 19:51:58.000000000 +1000
+++ swh-plugins-0.4.15/mbeq_1197.xml	2011-06-05 10:57:39.335774121 +1000
@@ -171,7 +171,8 @@
 #endif
 
 		// Multiply the bins magnitudes by the coeficients
-		for (i = 0; i < FFT_LENGTH/2; i++) {
+		comp[0] *= coefs[0];
+		for (i = 1; i < FFT_LENGTH/2; i++) {
 			comp[i] *= coefs[i];
 			comp[FFT_LENGTH-i] *= coefs[i];
 		}
