diff -wbBur mod_accounting-0.5/Makefile mod_accounting-0.5.my/Makefile
--- mod_accounting-0.5/Makefile	2001-12-30 17:11:43 +0300
+++ mod_accounting-0.5.my/Makefile	2006-04-10 13:22:00 +0400
@@ -5,13 +5,14 @@
 ##
 
 #   the used tools
-APXS=apxs
-APACHECTL=apachectl
+APXS=apxs13
+APACHECTL=apachectl13
 
 #   here's what you may need to change
-DEF=-DNEED_POSTGRES -DNEED_MYSQL
-INC=-I/usr/local/pgsql/include/ -I/usr/local/mysql/include/
-LIB=-L/usr/local/pgsql/lib -lpq -L/usr/local/mysql/lib/mysql/ -lmysqlclient
+DEF=-DNEED_MYSQL
+#INC=-I/usr/local/pgsql/include/ -I/usr/local/mysql/include/
+#LIB=-L/usr/local/mysql/lib/mysql/ -lmysqlclient
+LIB=-lmysqlclient
 
 #   the default target
 all: mod_accounting.so
diff -wbBur mod_accounting-0.5/mod_accounting.c mod_accounting-0.5.my/mod_accounting.c
--- mod_accounting-0.5/mod_accounting.c	2002-09-08 19:17:22 +0400
+++ mod_accounting-0.5.my/mod_accounting.c	2006-04-10 13:29:27 +0400
@@ -218,8 +218,10 @@
 	char			 *ptr = arg;
 
 	while( *ptr )
-		*ptr++ = tolower( *ptr );
-
+	{
+		*ptr = tolower( *ptr );
+		*ptr++;
+	}
 	// let's see if we have the requested driver
 	for( i = 0; i < DB_MAX; i++ )
 		if( !strcmp( DBDrivers[ i ].ID, arg )) {
