diff -Naur abuse-0.7.1.orig/Makefile.in abuse-0.7.1/Makefile.in
--- abuse-0.7.1.orig/Makefile.in	2008-03-02 15:25:05.000000000 -0500
+++ abuse-0.7.1/Makefile.in	2008-08-11 23:34:21.000000000 -0400
@@ -201,7 +201,7 @@
 top_srcdir = @top_srcdir@
 SUBDIRS = src
 man_MANS = abuse.6
-datafiledir = $(datadir)/games/abuse
+datafiledir = $(datadir)
 datafile_DATA = abuse.bmp abuse.lsp abuse.png
 EXTRA_DIST = bootstrap abuse.6 abuse.bmp abuse.lsp abuse.png
 all: config.h
diff -Naur abuse-0.7.1.orig/configure abuse-0.7.1/configure
--- abuse-0.7.1.orig/configure	2008-03-02 15:25:03.000000000 -0500
+++ abuse-0.7.1/configure	2008-08-11 23:33:49.000000000 -0400
@@ -21887,7 +21887,7 @@
 
 X_LIBS=-L$x_libraries
 
-EXPDATADIR="-DEXPDATADIR=\\\"$datadir/games/abuse\\\""
+EXPDATADIR="-DEXPDATADIR=\\\"$datadir\\\""
 
 
 { echo "$as_me:$LINENO: checking for XShmAttach in -lXext" >&5
diff -Naur abuse-0.7.1.orig/src/sdlport/setup.cpp abuse-0.7.1/src/sdlport/setup.cpp
--- abuse-0.7.1.orig/src/sdlport/setup.cpp	2008-03-02 14:42:24.000000000 -0500
+++ abuse-0.7.1/src/sdlport/setup.cpp	2008-08-11 23:33:49.000000000 -0400
@@ -89,7 +89,7 @@
     if( (fd = fopen( rcfile, "w" )) != NULL )
     {
         fputs( "; Abuse-SDL Configuration file\n\n", fd );
-        fputs( "; Location of the datafiles\ndatadir=/var/games/abuse\n\n", fd );
+        fputs( "; Location of the datafiles\ndatadir=/usr/share/abuse/orig\n\n", fd );
         fputs( "; Startup fullscreen\nfullscreen=0\n\n", fd );
         fputs( "; Use DoubleBuffering\ndoublebuf=0\n\n", fd );
         fputs( "; Use mono audio only\nmono=0\n\n", fd );
