From 5af344398ccc2c53addf100ddee45a0d115195ea Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Wed, 22 Mar 2017 23:07:06 -0600
Subject: [PATCH] arch linux arm modifications

---
 arch/arm/dts/dragonboard410c.dts | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/dts/dragonboard410c.dts b/arch/arm/dts/dragonboard410c.dts
index 7746622dda..55deee4887 100644
--- a/arch/arm/dts/dragonboard410c.dts
+++ b/arch/arm/dts/dragonboard410c.dts
@@ -143,6 +143,14 @@
 			gpios = <&pm8916_gpios 1 0>;
 		};
 	};
+
+	wifi {
+		compatible = "qcom,wcnss-wlan";
+	};
+
+	bt {
+		compatible = "qcom,wcnss-bt";
+	};
 };
 
 #include "dragonboard410c-uboot.dtsi"
-- 
2.12.0

