From 8c190679b688cf186e132a9f26052ac26ab717c3 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sat, 5 Mar 2016 12:27:57 -0700
Subject: [PATCH] arch linux arm modifications

---
 include/configs/dragonboard410c.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/dragonboard410c.h b/include/configs/dragonboard410c.h
index 4b00922..15350c1 100644
--- a/include/configs/dragonboard410c.h
+++ b/include/configs/dragonboard410c.h
@@ -12,7 +12,7 @@
 #include <linux/sizes.h>
 #include <asm/arch/sysmap-apq8016.h>
 
-#define CONFIG_IDENT_STRING		"\nQualcomm-DragonBoard 410C"
+#define CONFIG_IDENT_STRING		" Arch Linux ARM"
 
 #define CONFIG_MISC_INIT_R /* To stop autoboot */
 
-- 
2.9.0

