diff -aur linux-ae5b5f99f06f936382edda30522787bdde1248bd-pristine/arch/arm/mach-exynos/exynos-ss.c 
linux-ae5b5f99f06f936382edda30522787bdde1248bd-new/arch/arm/mach-exynos/exynos-ss.c
--- linux-ae5b5f99f06f936382edda30522787bdde1248bd-pristine/arch/arm/mach-exynos/exynos-ss.c	2016-06-15 
21:48:59.000000000 +0000
+++ linux-ae5b5f99f06f936382edda30522787bdde1248bd-new/arch/arm/mach-exynos/exynos-ss.c	2016-08-17 
16:01:16.095003127 +0000
@@ -238,6 +238,13 @@
 	struct exynos_ss_hook info_hook;
 };
 
+enum ess_cause_emerg_events {
+	CAUSE_INVALID_DUMP = 0x00000000,
+	CAUSE_KERNEL_PANIC = 0x00000001,
+	CAUSE_FORCE_DUMP   = 0x0000000D,
+	CAUSE_FORCE_REBOOT = 0x000000FF,
+};
+
 extern void (*arm_pm_restart)(char str, const char *cmd);
 #if LINUX_VERSION_CODE <= KERNEL_VERSION(3,5,00)
 extern void register_hook_logbuf(void (*)(const char));
@@ -493,13 +500,6 @@
 }
 #endif
 
-enum ess_cause_emerg_events {
-	CAUSE_INVALID_DUMP = 0x00000000,
-	CAUSE_KERNEL_PANIC = 0x00000001,
-	CAUSE_FORCE_DUMP   = 0x0000000D,
-	CAUSE_FORCE_REBOOT = 0x000000FF,
-};
-
 static void exynos_ss_scratch_reg(unsigned int val)
 {
 	__raw_writel(val, S5P_VA_SS_SCRATCH);

