diff -ruN a/cmake/modules/FindWine.cmake b/cmake/modules/FindWine.cmake
--- a/cmake/modules/FindWine.cmake	2018-11-03 02:43:42.000000000 +0100
+++ b/cmake/modules/FindWine.cmake	2021-06-03 22:37:41.258182544 +0200
@@ -31,7 +31,7 @@
 		# WineHQ (/opt/wine-stable, /opt/wine-devel, /opt/wine-staging)
 		STRING(REPLACE "/lib64/wine/" "/lib/wine/" FLAG "${FLAG}")
 
-		STRING(REGEX REPLACE "/wine/libwinecrt0.a.*" "" WINE_LIBRARY_FIX "${FLAG}")
+		STRING(REGEX REPLACE "libwinecrt0.a.*" "" WINE_LIBRARY_FIX "${FLAG}")
 		SET(WINE_LIBRARY_FIX "${WINE_LIBRARY_FIX}/")
 	ENDIF()
 ENDFOREACH()
diff -ruN a/plugins/vst_base/CMakeLists.txt b/plugins/vst_base/CMakeLists.txt
--- a/plugins/vst_base/CMakeLists.txt	2018-11-03 02:43:42.000000000 +0100
+++ b/plugins/vst_base/CMakeLists.txt	2021-06-03 22:37:45.881530269 +0200
@@ -38,7 +38,7 @@
 IF(LMMS_HOST_X86_64)
 	SET(EXTRA_FLAGS -m32)
 	IF(WINE_LIBRARY_FIX)
-		SET(EXTRA_FLAGS ${EXTRA_FLAGS} -nodefaultlibs ${WINE_LIBRARY_FIX}wine/libwinecrt0.a -L${WINE_LIBRARY_FIX}wine/ -luser32 -lkernel32 -lgdi32)
+		SET(EXTRA_FLAGS ${EXTRA_FLAGS} -nodefaultlibs ${WINE_LIBRARY_FIX}libwinecrt0.a -L${WINE_LIBRARY_FIX} -luser32 -lkernel32 -lgdi32)
 	ENDIF()
 ENDIF(LMMS_HOST_X86_64)
 
