diff -up tuxcmd-0.6.70/ULibc.pas.old tuxcmd-0.6.70/ULibc.pas
--- tuxcmd-0.6.70/ULibc.pas.old	2009-11-15 11:00:35.000000000 -0500
+++ tuxcmd-0.6.70/ULibc.pas	2013-09-27 11:59:23.651774490 -0400
@@ -28,6 +28,7 @@ interface
 const GLIBC_LIB = 'libc.so.6';
       DL_LIB = 'libdl.so.2';
       PTHREAD_LIB = 'libpthread.so.0';
+      GCC_S_LIB = 'libgcc_s.so.1';
 
 type
 {$IFDEF KYLIX}
@@ -892,6 +893,9 @@ function getegid: __gid_t; cdecl; extern
 function access(pathname: PChar; mode: Longint): Longint; cdecl; external GLIBC_LIB name 'access';
 function euidaccess(pathname: PChar; mode: Longint): Longint; cdecl; external GLIBC_LIB name 'euidaccess';
 
+{$IFDEF CPUARM}
+function __aeabi_unwind_cpp_pr0: integer; cdecl; external GCC_S_LIB name '__aeabi_unwind_cpp_pr0';
+{$ENDIF}
 
 type pthread_t = {$ifdef cpu64}QWord{$else}DWord{$endif};
 
