Index: mhwaveedit-1.4.16/configure.in
===================================================================
--- mhwaveedit-1.4.16.orig/configure.in
+++ mhwaveedit-1.4.16/configure.in
@@ -282,7 +282,7 @@ AC_DEFUN([AC_WITHOUT_PULSE],[
 	fi
 ])
 
-AC_ARG_WITH(pulse,[  --without-pulse          Don't compile PulseAudio driver],AC_WITHOUT_JACK)
+AC_ARG_WITH(pulse,[  --without-pulse          Don't compile PulseAudio driver],AC_WITHOUT_PULSE)
 
 if test "x$MHW_SKIP_PULSETEST" = "xno"
 then
