diff -urN a/arch/arm/mach-mmp/gplugd.c b/arch/arm/mach-mmp/gplugd.c
--- a/arch/arm/mach-mmp/gplugd.c	2011-10-24 07:10:05.000000000 +0000
+++ b/arch/arm/mach-mmp/gplugd.c	2012-03-25 03:41:20.040796509 +0000
@@ -188,7 +188,7 @@
 	pxa168_add_eth(&gplugd_eth_platform_data);
 }
 
-MACHINE_START(SHEEVAD, "PXA168-based GuruPlug Display (gplugD) Platform")
+MACHINE_START(GPLUGD, "PXA168-based GuruPlug Display (gplugD) Platform")
 	.map_io		= mmp_map_io,
 	.nr_irqs	= IRQ_BOARD_START,
 	.init_irq       = pxa168_init_irq,
