--- wvstreams-4.4.1.orig/utils/wvcrash.cc
+++ wvstreams-4.4.1/utils/wvcrash.cc
@@ -385,10 +385,10 @@
 #if WVCRASH_USE_SIGALTSTACK
     struct sigaction act;
     
+    memset(&act,0,sizeof(act));
     act.sa_handler = wvcrash;
     sigfillset(&act.sa_mask);
     act.sa_flags = SA_ONSTACK | SA_RESTART;
-    act.sa_restorer = NULL;
     
     if (sigaction(sig, &act, NULL))
         fprintf(stderr, "Failed to setup wvcrash handler for signal %d: %s\n",
--- wvstreams-4.4.1.orig/include/wvserialize.h
+++ wvstreams-4.4.1/include/wvserialize.h
@@ -349,7 +349,7 @@
  * except as the last character.
  */
 template <>
-extern WvString _wv_deserialize<WvString>(WvBuf &buf);
+WvString _wv_deserialize<WvString>(WvBuf &buf);
 
 
 /** Deserialize a WvBuf. */
--- wvstreams-4.4.1.orig/include/uniconfkey.h
+++ wvstreams-4.4.1/include/uniconfkey.h
@@ -327,7 +327,7 @@
      */
     UniConfKey last(int n = 1) const
     {
-        return range(numsegments() - n, INT_MAX);
+        return range(numsegments() - n, __INT_MAX__);
     }
 
     /**
@@ -338,7 +338,7 @@
      */
     UniConfKey removefirst(int n = 1) const
     {
-        return range(n, INT_MAX);
+        return range(n, __INT_MAX__);
     }
 
     /**
