diff -Nru wvstreams-4.3.orig/utils/wvcrash.cc wvstreams-4.3/utils/wvcrash.cc
--- wvstreams-4.3.orig/utils/wvcrash.cc	2007-02-07 21:06:12.000000000 +0200
+++ wvstreams-4.3/utils/wvcrash.cc	2007-03-12 14:23:47.000000000 +0200
@@ -91,8 +91,8 @@
     char digits[] = "0123456789ABCDEF";
     
     write(fd, "0x", 2);
-    for (int shift=28; shift>=0; shift-=4)
-        write(fd, &digits[(((unsigned)addr)>>shift)&0xF], 1);
+    for (int shift = (sizeof(addr) << 3) - 4; shift>=0; shift-=4)
+        write(fd, &digits[(((ptrdiff_t)addr)>>shift)&0xF], 1);
 }
 
 
