--- Make.common	2024-04-07 00:18:25.000000000 +0300
+++ /dev/shm/Make.common	2025-06-22 21:24:08.982974956 +0300
@@ -39,7 +39,7 @@
 # version or later and compiling for ARM64; but set it to "n" if using an
 # earlier version of GNU-EFI on ARM64. This option has no effect on IA32/x86
 # or X64/AMD64/x86-64 systems.
-GNUEFI_ARM64_TARGET_SUPPORT ?= n
+GNUEFI_ARM64_TARGET_SUPPORT ?= y
 
 HOSTARCH        = $(shell uname -m | sed s,i[3456789]86,ia32,)
 ARCH            ?= $(HOSTARCH)
@@ -54,12 +54,12 @@
 GENFW           = $(TIANOBASE)/BaseTools/Source/C/bin/GenFw
 prefix          = /usr/bin/
 ifeq ($(ARCH),aarch64)
-  CC            = $(prefix)aarch64-linux-gnu-gcc
-  AS            = $(prefix)aarch64-linux-gnu-as
-  LD            = $(prefix)aarch64-linux-gnu-ld
-  AR            = $(prefix)aarch64-linux-gnu-ar
-  RANLIB        = $(prefix)aarch64-linux-gnu-ranlib
-  OBJCOPY       = $(prefix)aarch64-linux-gnu-objcopy
+  CC            = $(prefix)gcc
+  AS            = $(prefix)as
+  LD            = $(prefix)ld
+  AR            = $(prefix)ar
+  RANLIB        = $(prefix)ranlib
+  OBJCOPY       = $(prefix)objcopy
 else
   CC            = $(prefix)gcc
   AS            = $(prefix)as
