diff -Nru grub-1.96.orig/kern/i386/pc/startup.S grub-1.96/kern/i386/pc/startup.S
--- grub-1.96.orig/kern/i386/pc/startup.S	2008-02-03 14:11:32.000000000 +0100
+++ grub-1.96_20081229/kern/i386/pc/startup.S	2008-02-17 06:22:54.000000000 +0100
@@ -594,15 +594,15 @@
 
 	call	EXT_C(grub_dl_unload_all)
 
+	/* Turn off Gate A20 */
+	xorl	%eax, %eax
+	call	EXT_C(grub_gate_a20)       
+
 	/* set up to pass boot drive */
 	popl	%edx
 
 	/* ESI must point to a partition table entry */
 	popl	%esi
-
-	/* Turn off Gate A20 */
-	xorl	%eax, %eax
-	call	EXT_C(grub_gate_a20)
 	
 	call	prot_to_real
 	.code16
