diff -ruN a/scripts/wrap/__main__.py b/scripts/wrap/__main__.py
--- a/scripts/wrap/__main__.py	2024-01-10 16:00:05.000000000 +0100
+++ b/scripts/wrap/__main__.py	2024-01-26 10:53:09.531262564 +0100
@@ -1633,6 +1633,7 @@
                                     -I {include2}
                                     {cpp_files_text}
                                     {link_l_flags(libmupdf)}
+                                    {os.environ.get("LDFLAGS") or ""}
                                 ''')
                                 )
                         jlib.build(
