From b1a37841d7f337458301064fef27d96c4d2ceefd Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Thu, 27 Feb 2020 18:26:42 -0700
Subject: [PATCH] remove manuals

---
 Makefile | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 99af8c4..76c5037 100644
--- a/Makefile
+++ b/Makefile
@@ -33,9 +33,7 @@ all: \
 data: \
 	$(BUILDDIR)/GeoIP.conf \
 	$(BUILDDIR)/GeoIP.conf.md \
-	$(BUILDDIR)/geoipupdate.md \
-	$(BUILDDIR)/GeoIP.conf.5 \
-	$(BUILDDIR)/geoipupdate.1
+	$(BUILDDIR)/geoipupdate.md
 
 $(BUILDDIR):
 	mkdir -p $(BUILDDIR)
-- 
2.24.1

