diff -uar streamtuner-0.99.99-orig/src/plugins/shoutcast/shoutcast.c streamtuner-0.99.99-patched/src/plugins/shoutcast/shoutcast.c
--- streamtuner-0.99.99-orig/src/plugins/shoutcast/shoutcast.c	2008-09-12 18:04:00.229785732 -0400
+++ streamtuner-0.99.99-patched/src/plugins/shoutcast/shoutcast.c	2008-09-12 18:05:56.716440720 -0400
@@ -37,7 +37,7 @@
 
 /*** cpp *********************************************************************/
 
-#define SHOUTCAST_ROOT		"http://www.shoutcast.com/"
+#define SHOUTCAST_ROOT		"http://classic.shoutcast.com/"
 
 #define MAX_STREAMS_PER_PAGE	100 /* enforced by SHOUTcast */
 
