diff --git a/autobahn/wamp/test/test_user_handler_errors.py b/autobahn/wamp/test/test_user_handler_errors.py
index 35570bad..8f1b0181 100644
--- a/autobahn/wamp/test/test_user_handler_errors.py
+++ b/autobahn/wamp/test/test_user_handler_errors.py
@@ -108,7 +108,7 @@ if os.environ.get('USE_TWISTED', False):
         # twice)...but that would mean switching all test-running over
         # to py-test
 
-        skip = True
+        skip = 'True'
 
         def test_on_join(self):
             session = MockApplicationSession()
