diff -urN a/inittab b/inittab
--- a/inittab	2012-06-29 07:26:59.000000000 -0600
+++ b/inittab	2012-07-15 16:02:29.969505191 -0600
@@ -24,12 +24,13 @@
 su:S:wait:/sbin/sulogin -p
 
 # -8 options fixes umlauts problem on login
+s0:12345:respawn:/sbin/agetty -8 -s 115200 @TTY@ linux
 c1:2345:respawn:/sbin/agetty -8 -s 38400 tty1 linux
 c2:2345:respawn:/sbin/agetty -8 -s 38400 tty2 linux
-c3:2345:respawn:/sbin/agetty -8 -s 38400 tty3 linux
-c4:2345:respawn:/sbin/agetty -8 -s 38400 tty4 linux
-c5:2345:respawn:/sbin/agetty -8 -s 38400 tty5 linux
-c6:2345:respawn:/sbin/agetty -8 -s 38400 tty6 linux
+#c3:2345:respawn:/sbin/agetty -8 -s 38400 tty3 linux
+#c4:2345:respawn:/sbin/agetty -8 -s 38400 tty4 linux
+#c5:2345:respawn:/sbin/agetty -8 -s 38400 tty5 linux
+#c6:2345:respawn:/sbin/agetty -8 -s 38400 tty6 linux
 
 # Serial Virtual Console for KVM and others VMs
 #s0:2345:respawn:/sbin/agetty -8 -s 9600 ttyS0 linux
diff -urN a/rc.shutdown b/rc.shutdown
--- a/rc.shutdown	2012-06-29 07:26:59.000000000 -0600
+++ b/rc.shutdown	2012-07-15 16:02:29.969505191 -0600
@@ -76,6 +76,10 @@
 
 run_hook shutdown_poweroff
 
+# ALARM: Turn LEDs off
+
+/usr/bin/find /sys/ -path *led* -name *trigger* -exec sh -c '/bin/echo none > {}' \;
+
 if [[ -x /run/initramfs/shutdown ]]; then
 
 	# decide what we want to do
diff -urN a/rc.sysinit b/rc.sysinit
--- a/rc.sysinit	2012-06-29 07:26:59.000000000 -0600
+++ b/rc.sysinit	2012-07-15 19:59:55.346823548 -0600
@@ -13,8 +13,8 @@
 fi
 
 echo " "
-printhl "Arch Linux\n"
-printhl "${C_H2}http://www.archlinux.org"
+printhl "Arch Linux ARM\n"
+printhl "${C_H2}http://www.archlinuxarm.org"
 printsep
 
 # mount the api filesystems
@@ -109,7 +109,7 @@
 fsck_reboot $fsckret
 
 status "Remounting Root and API filesystems" \
-	/usr/lib/systemd/systemd-remount-fs
+	mount -o remount,rw /
 
 # now mount all the local filesystems
 run_hook sysinit_premount
