diff -wbBur netbrake/libnetbrake.c netbrake.my/libnetbrake.c
--- netbrake/libnetbrake.c	2003-12-04 09:33:00.000000000 +0300
+++ netbrake.my/libnetbrake.c	2007-10-11 13:29:22.000000000 +0400
@@ -236,7 +236,7 @@
 	return retval;
 }
 
-int recv(int s, void *buf, size_t len, int flags)
+ssize_t recv(int s, void *buf, size_t len, int flags)
 {
 	int retval;
 
@@ -245,7 +245,7 @@
 	return retval;
 }
 
-int recvfrom(int s, void *buf, size_t len, int flags,
+ssize_t recvfrom(int s, void *buf, size_t len, int flags,
 	     struct sockaddr *from, socklen_t *fromlen)
 {
 	int retval;
@@ -255,7 +255,7 @@
 	return retval;
 }
 
-int send(int s, const void *msg, size_t len , int flags)
+ssize_t send(int s, const void *msg, size_t len , int flags)
 {
 	int retval;
 
@@ -264,7 +264,7 @@
 	return retval;
 }
 
-int sendto(int s, const void *msg, size_t len, int flags,
+ssize_t sendto(int s, const void *msg, size_t len, int flags,
 	   const struct sockaddr *to, socklen_t tolen)
 {
 	int retval;
