--- config.default.orig	2011-11-10 21:48:01.000000000 +0100
+++ config.default	2011-11-10 21:51:58.383221979 +0100
@@ -6,18 +6,18 @@
 #BACKEND = opengl
 #BACKEND = sdl
 #BACKEND = sdl2
-#BACKEND = mixer_sdl
+BACKEND = mixer_sdl
 #BACKEND = posix
 
 #COLOUR_DEPTH = 0
-#COLOUR_DEPTH = 16
+COLOUR_DEPTH = 16
 
 #OSTYPE = amiga
 #OSTYPE = beos
 #OSTYPE = cygwin
 #OSTYPE = freebsd
 #OSTYPE = haiku
-#OSTYPE = linux
+OSTYPE = linux
 #OSTYPE = mingw
 #OSTYPE = mac
 
@@ -37,6 +37,6 @@
 #SDL_CONFIG     = sdl-config
 #SDL2_CONFIG    = sdl2-config
 
-#VERBOSE = 1
+VERBOSE = 1
 
 # The following useful conditional compilation flags exist
 #
