diff -Naur simutrans.orig/Makefile simutrans/Makefile
--- simutrans.orig/Makefile	2014-11-20 17:50:34.948410000 +0100
+++ simutrans/Makefile	2014-11-20 17:51:43.936577860 +0100
@@ -76,15 +76,6 @@
 SDL_CONFIG     ?= sdl-config
 SDL2_CONFIG    ?= sdl2-config
 
-ifneq ($(OPTIMISE),)
-    CFLAGS += -O3
-  ifeq ($(findstring $(OSTYPE), amiga haiku mac),)
-    CFLAGS += -minline-all-stringops
-  endif
-else
-  CFLAGS += -O
-endif
-
 ifdef DEBUG
   ifeq ($(shell expr $(DEBUG) \>= 1), 1)
     CFLAGS += -g -DDEBUG
diff -Naur simutrans.orig/makeobj/Makefile simutrans/makeobj/Makefile
--- simutrans.orig/makeobj/Makefile	2014-11-20 17:50:34.945077000 +0100
+++ simutrans/makeobj/Makefile	2014-11-20 17:51:42.553214510 +0100
@@ -42,17 +42,6 @@
   CXXFLAGS   += -DUSE_HW -DUSE_C
 endif
 
-ifneq ($(OPTIMISE),)
-  CFLAGS   += -O3 -fno-schedule-insns
-  CXXFLAGS += -O3 -fno-schedule-insns
-  ifneq ($(OSTYPE),mac)
-    CFLAGS  += -minline-all-stringops
-  endif
-else
-  CFLAGS   += -O
-  CXXFLAGS += -O
-endif
-
 ifdef DEBUG
   ifeq ($(shell expr $(DEBUG) \>= 1), 1)
     CFLAGS   += -g -DDEBUG
