diff -Nur rawstudio-1.2.orig/src/dcraw.cc rawstudio-1.2/src/dcraw.cc
--- rawstudio-1.2.orig/src/dcraw.cc	2009-04-10 00:21:45.000000000 +0300
+++ rawstudio-1.2/src/dcraw.cc	2009-07-16 00:05:28.169155134 +0300
@@ -8510,7 +8510,7 @@
   argv[argc] = "";
   for (arg=1; (((opm = argv[arg][0]) - 2) | 2) == '+'; ) {
     opt = argv[arg++][1];
-    if ((cp = strchr (sp="nbrkStqmHACg", opt)))
+    if ((cp = (char *)strchr (sp="nbrkStqmHACg", opt)))
       for (i=0; i < "114111111422"[cp-sp]-'0'; i++)
 	if (!isdigit(argv[arg+i][0])) {
 	  dcraw_message (DCRAW_ERROR,_("Non-numeric argument to \"-%c\"\n"), opt);
