--- ndiswrapper-1.56.orig/driver/loader.c	2010-02-11 03:09:32.000000000 +0100
+++ ndiswrapper-1.56/driver/loader.c	2010-10-24 01:38:13.000000000 +0200
@@ -750,13 +750,13 @@
 }
 
 /* called with loader_mutex is down */
-static int wrapper_ioctl(struct inode *inode, struct file *file,
+static long wrapper_unlocked_ioctl(struct file *file,
 			 unsigned int cmd, unsigned long arg)
 {
 	struct load_driver *load_driver;
 	struct load_device load_device;
 	struct load_driver_file load_bin_file;
-	int ret;
+	long ret;
 	void __user *addr = (void __user *)arg;
 
 	ENTER1("cmd: %u", cmd);
@@ -831,7 +831,7 @@
 
 static struct file_operations wrapper_fops = {
 	.owner          = THIS_MODULE,
-	.ioctl		= wrapper_ioctl,
+	.unlocked_ioctl	= wrapper_unlocked_ioctl,
 	.release	= wrapper_ioctl_release,
 };
 
