diff -Nur -x '*.orig' -x '*~' bluez-gnome-1.4/analyzer/bluetooth-analyzer.desktop.in bluez-gnome-1.4.new/analyzer/bluetooth-analyzer.desktop.in
--- bluez-gnome-1.4/analyzer/bluetooth-analyzer.desktop.in	2008-08-24 11:05:50.000000000 -0500
+++ bluez-gnome-1.4.new/analyzer/bluetooth-analyzer.desktop.in	2008-09-26 15:59:26.000000000 -0500
@@ -9,3 +9,4 @@
 Type=Application
 Categories=System;Monitor;
 OnlyShowIn=GNOME;
+NoDisplay=true
