diff --git a/sendto/main.c b/sendto/main.c
index 044bbbc..f0c435b 100644
diff -Nur -x '*.orig' -x '*~' bluez-gnome-1.8/sendto/main.c bluez-gnome-1.8.new/sendto/main.c
--- bluez-gnome-1.8/sendto/main.c	2008-10-04 14:32:56.000000000 -0500
+++ bluez-gnome-1.8.new/sendto/main.c	2008-10-06 01:01:19.000000000 -0500
@@ -790,6 +790,7 @@
 		dbus_g_proxy_begin_call(proxy, "CreateBluetoothSession",
 					create_notify, NULL, NULL,
 					G_TYPE_STRING, option_device,
+					G_TYPE_STRING, "00:00:00:00:00:00",
 					G_TYPE_STRING, "opp", G_TYPE_INVALID);
 	} else {
 		GHashTable *hash = NULL;
