--- st7554.c~	2005-11-01 16:31:45.000000000 +0100
+++ st7554.c	2006-03-12 14:57:35.000000000 +0100
@@ -982,7 +982,6 @@
 
 
 static struct file_operations st7554_fops = {
-        .owner =   THIS_MODULE,
         .llseek =  no_llseek,
         .read =    st7554_read,
         .write =   st7554_write,
@@ -1150,7 +1149,6 @@
 MODULE_DEVICE_TABLE (usb, st7554_ids);
 
 static struct usb_driver st7554_usb_driver = {
-	.owner =       THIS_MODULE,
 	.name =	       "ST7554 USB Modem",
 	.probe =       st7554_probe,
 	.disconnect =  st7554_disconnect,
