--- drivers/amrmo_init.c~	2006-07-07 11:58:31.000000000 +0200
+++ drivers/amrmo_init.c	2006-07-07 12:00:13.000000000 +0200
@@ -641,7 +641,6 @@
 #endif
 #else
 	class_simple_device_add(amrmo_class, MKDEV(AMRMO_MAJOR, i), NULL, "slamr%d", i);
-	devfs_mk_cdev(MKDEV(AMRMO_MAJOR,i), S_IFCHR|S_IRUSR|S_IWUSR, "slamr%d", i);
 #endif
 	return 0;
 
@@ -673,7 +672,6 @@
 #endif
 #else
 	class_simple_device_remove(MKDEV(AMRMO_MAJOR, amrmo->num));
-	devfs_remove("slamr%d", amrmo->num);
 #endif
 	amrmo_table[amrmo->num] = NULL;
 	amrmo_card_disable(amrmo->card);
--- drivers/amrmo_init.c~	2006-07-07 12:02:49.000000000 +0200
+++ drivers/amrmo_init.c	2006-07-07 12:03:11.000000000 +0200
@@ -53,7 +53,6 @@
 #include <linux/poll.h>
 #include <linux/fs.h>
 #include <asm/uaccess.h>
-#include <linux/devfs_fs_kernel.h>
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,5,0)
 #define OLD_KERNEL 1
--- drivers/st7554.c~	2006-07-07 12:03:36.000000000 +0200
+++ drivers/st7554.c	2006-07-07 12:04:32.000000000 +0200
@@ -55,7 +55,6 @@
 #include <linux/poll.h>
 #include <linux/usb.h>
 #include <linux/device.h>
-#include <linux/devfs_fs_kernel.h>
 
 #include <modem_defs.h>
 
@@ -1281,7 +1280,6 @@
 
 	usb_set_intfdata(interface, s );
 	class_simple_device_add(st7554_class, MKDEV(243, i), NULL, "slusb%d", i);
-	devfs_mk_cdev(MKDEV(243,i),S_IFCHR|S_IRUSR|S_IWUSR,"slusb%d",i);
 
 	USB_INFO(KERN_INFO "slusb: slusb%d is found.\n", s->minor);
 
@@ -1310,7 +1308,6 @@
         }
 
 	class_simple_device_remove(MKDEV(243, s->minor));
- 	devfs_remove("slusb%d",s->minor);
 
 	st7554_stop(s);
 	down(&open_sem);
