--- ungrab-winmodem/ungrab-winmodem.c~	2005-10-09 22:54:03.000000000 +0200
+++ ungrab-winmodem/ungrab-winmodem.c	2006-10-17 19:07:57.000000000 +0200
@@ -11,7 +11,6 @@
  *
  */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/version.h>
 #include <linux/init.h>
--- slmodem-2.9.11-20061021/drivers/amrmo_init.c~	2006-10-17 19:08:16.000000000 +0200
+++ slmodem-2.9.11-20061021/drivers/amrmo_init.c	2006-10-17 19:08:27.000000000 +0200
@@ -43,7 +43,6 @@
 
 /*****************************************************************************/
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/version.h>
 #include <linux/init.h>
