--- slmodem-2.9.11-20061021/drivers/amrmo_init.c~	2005-10-09 23:14:13.000000000 +0200
+++ slmodem-2.9.11-20061021/drivers/amrmo_init.c	2006-04-09 15:35:27.000000000 +0200
@@ -701,7 +701,7 @@
  *  module stuff
  */
 
-MODULE_PARM(debug,"i");
+module_param(debug, int, 0);
 MODULE_PARM_DESC(debug,"debug level: 0-3 (default=0)");
 
 MODULE_AUTHOR("Smart Link Ltd.");
--- slmodem-2.9.11-20061021/drivers/st7554.c~	2006-04-09 15:37:14.000000000 +0200
+++ slmodem-2.9.11-20061021/drivers/st7554.c	2006-04-09 15:37:45.000000000 +0200
@@ -76,7 +76,7 @@
 #define USB_DBG_URB(fmt...) // USB_DBG(fmt)
 
 static int debug = 0;
-MODULE_PARM(debug,"i");
+module_param(debug, int, 0);
 MODULE_PARM_DESC(debug,"Debug level: 0-3 (default=0)");
 
 /* st7554 IDs */
