diff -aur old/initscript/archlinux/avahi-daemon.in new/initscript/archlinux/avahi-daemon.in
--- old/initscript/archlinux/avahi-daemon.in	2011-06-24 03:07:00.916170590 +0200
+++ new/initscript/archlinux/avahi-daemon.in	2011-06-24 03:16:32.220596377 +0200
@@ -33,6 +33,7 @@
 
 case "$1" in
     start)
+        ck_daemon dbus && { echo -n "Start dbus first." >&2; stat_die; }
         stat_busy "Starting $DESC"
         $DAEMON -D > /dev/null 2>&1
         if [ $? -gt 0 ]; then
