diff --exclude-from=/opt/home/dang/.scripts/diffrc -up -ruN seahorse-2.26.2.orig/pgp/seahorse-gpg-op.c seahorse-2.26.2/pgp/seahorse-gpg-op.c
--- seahorse-2.26.2.orig/pgp/seahorse-gpg-op.c	2009-04-20 11:35:00.000000000 -0400
+++ seahorse-2.26.2/pgp/seahorse-gpg-op.c	2009-07-10 09:43:31.550476889 -0400
@@ -18,7 +18,9 @@
  * 59 Temple Place, Suite 330,
  * Boston, MA 02111-1307, USA.
  */
- 
+
+#include "config.h"
+
 #include <glib.h>
 #include <gpgme.h>
 #include <string.h>
diff --exclude-from=/opt/home/dang/.scripts/diffrc -up -ruN seahorse-2.26.2.orig/pgp/seahorse-gpg-options.c seahorse-2.26.2/pgp/seahorse-gpg-options.c
--- seahorse-2.26.2.orig/pgp/seahorse-gpg-options.c	2009-04-20 11:35:00.000000000 -0400
+++ seahorse-2.26.2/pgp/seahorse-gpg-options.c	2009-07-10 09:36:16.012409729 -0400
@@ -19,6 +19,11 @@
  * Boston, MA 02111-1307, USA.
  */
 
+#include "config.h"
+
+#include "seahorse-util.h"
+#include "seahorse-context.h"
+
 #include <sys/param.h>
 #include <sys/wait.h>
 #include <sys/file.h>
@@ -31,9 +36,6 @@
 #include <unistd.h>
 #include <sys/stat.h>
 
-#include "seahorse-util.h"
-#include "seahorse-context.h"
-
 #include "pgp/seahorse-gpgme.h"
 #include "pgp/seahorse-gpg-options.h"
 
