--- par2cmdline-0.4.orig/par2creatorsourcefile.cpp	2004-04-15 14:45:23.000000000 +0100
+++ par2cmdline-0.4/par2creatorsourcefile.cpp	2005-09-29 02:27:43.000000000 +0100
@@ -213,11 +213,12 @@
         }
       }
 
+      offset += want;
+
       if (noiselevel > CommandLine::nlQuiet)
       {
         // Display progress
         u32 oldfraction = (u32)(1000 * offset / filesize);
-        offset += want;
         u32 newfraction = (u32)(1000 * offset / filesize);
         if (oldfraction != newfraction)
         {
