--- ../orig/par2cmdline-0.4/par2cmdline.h	2004-04-15 15:30:02.000000000 +0200
+++ ./par2cmdline.h	2007-03-24 13:25:43.000000000 +0100
@@ -132,6 +132,10 @@
 typedef unsigned long long u64;
 #endif
 
+// FIXME: this is not done right w/ autoconf
+#include <sys/ioctl.h>
+#include <sys/mount.h>
+
 #if HAVE_SYS_STAT_H
 #  include <sys/stat.h>
 #endif
@@ -179,6 +183,8 @@
 #include <ctype.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/ioctl.h>
+#include <sys/mount.h>
 #include <dirent.h>
 #include <assert.h>
 
