--- commandline.cpp.orig	2006-04-06 21:41:27.000000000 -0700
+++ commandline.cpp	2006-04-07 00:12:29.000000000 -0700
@@ -550,6 +550,20 @@
         }
         else
         {
+          //start of shell expanded * patch. -- Michael Evans
+          //The shell might expaned * so, if we have our name and we're creating, then filter for files...
+          if ((parfilename.length() != 0) && (operation == opCreate))
+          {
+            struct stat st;
+            if (!(stat(argv[0], &st) == 0 && S_ISREG(st.st_mode)))
+            {
+              cerr << "Skipping non-regular file: " << argv[0] << endl;
+              argc--;
+              argv++;
+              options = false;
+              continue;
+            }
+          }//end of shell expanded * patch. -- Michael Evans
           filenames = new list<string>;
           filenames->push_back(argv[0]);
         }
