--- a/xbmc/screensavers/rsxs-0.9/lib/argp.h.orig	2015-03-26 13:25:20.000000000 -0500
+++ b/xbmc/screensavers/rsxs-0.9/lib/argp.h	2015-05-19 16:05:20.992493851 -0500
@@ -559,7 +559,7 @@
 # endif
 
 # ifndef ARGP_EI
-#  define ARGP_EI extern __inline__
+#  define ARGP_EI extern __inline__ __attribute__ ((gnu_inline)) 
 # endif
 
 ARGP_EI void
--- a/xbmc/screensavers/rsxs-0.9/lib/argp-fmtstream.h.orig	2015-03-26 13:25:20.000000000 -0500
+++ b/xbmc/screensavers/rsxs-0.9/lib/argp-fmtstream.h	2015-05-19 16:02:25.775456866 -0500
@@ -198,11 +198,7 @@
 #endif
 
 #ifndef ARGP_FS_EI
-#ifdef __clang__
 #define ARGP_FS_EI extern inline __attribute__ ((__gnu_inline__))
-#else
-#define ARGP_FS_EI extern inline
-#endif
 #endif
 
 ARGP_FS_EI size_t
