diff --git i/src/internal_modules/roc_sndio/driver.h w/src/internal_modules/roc_sndio/driver.h
index ff2406f9..dc5bc521 100644
--- i/src/internal_modules/roc_sndio/driver.h
+++ w/src/internal_modules/roc_sndio/driver.h
@@ -21,7 +21,7 @@ namespace sndio {
 class IBackend;
 
 //! Maximum number of drivers.
-static const size_t MaxDrivers = 128;
+static const size_t MaxDrivers = 256;
 
 //! Driver type.
 enum DriverType {
