diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2015-04-23 04:08:00.000000000 -0600
+++ b/CMakeLists.txt	2015-05-03 17:59:09.416829142 -0600
@@ -414,7 +414,7 @@
 endif()
 
 if(KWIN_HAVE_EGL)
-    set(kwin_KDEINIT_SRCS ${kwin_KDEINIT_SRCS} abstract_egl_backend.cpp eglonxbackend.cpp)
+    set(kwin_KDEINIT_SRCS ${kwin_KDEINIT_SRCS} abstract_egl_backend.cpp)
 endif()
 
 if(HAVE_WAYLAND)
diff -urN a/scene_opengl.cpp b/scene_opengl.cpp
--- a/scene_opengl.cpp	2015-04-23 04:06:05.000000000 -0600
+++ b/scene_opengl.cpp	2015-05-03 17:59:40.756785872 -0600
@@ -28,7 +28,6 @@
 *********************************************************************/
 #include "scene_opengl.h"
 #ifdef KWIN_HAVE_EGL
-#include "eglonxbackend.h"
 #endif // KWIN_HAVE_EGL
 #ifndef KWIN_HAVE_OPENGLES
 #include "glxbackend.h"
@@ -501,7 +500,7 @@
         } else
 #endif // HAVE_WAYLAND
         {
-            backend = new EglOnXBackend();
+            backend = NULL;
         }
 #endif // KWIN_HAVE_EGL
         break;
