diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4058b34..a1f2060 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -397,10 +397,6 @@ if(KWIN_BUILD_ACTIVITIES)
     )
 endif()
 
-if(KWIN_HAVE_EGL)
-    set(kwin_KDEINIT_SRCS ${kwin_KDEINIT_SRCS} eglonxbackend.cpp)
-endif()
-
 if(HAVE_WAYLAND)
     set(kwin_KDEINIT_SRCS
             ${kwin_KDEINIT_SRCS}
diff --git a/scene_opengl.cpp b/scene_opengl.cpp
index bacc980..e19bb00 100644
--- a/scene_opengl.cpp
+++ b/scene_opengl.cpp
@@ -28,7 +28,6 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *********************************************************************/
 #include "scene_opengl.h"
 #ifdef KWIN_HAVE_EGL
-#include "eglonxbackend.h"
 // for Wayland
 #if HAVE_WAYLAND_EGL
 #include "egl_wayland_backend.h"
@@ -437,10 +436,10 @@ SceneOpenGL *SceneOpenGL::createScene()
         if (kwinApp()->shouldUseWaylandForCompositing()) {
             backend = new EglWaylandBackend();
         } else {
-            backend = new EglOnXBackend();
+            backend = NULL;
         }
 #else
-        backend = new EglOnXBackend();
+        backend = NULL;
 #endif
 #endif
         break;
