diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2015-02-21 13:23:27.000000000 -0700
+++ b/CMakeLists.txt	2015-03-01 16:00:41.823988772 -0700
@@ -393,10 +393,6 @@
     )
 endif()
 
-if(KWIN_HAVE_EGL)
-    set(kwin_KDEINIT_SRCS ${kwin_KDEINIT_SRCS} eglonxbackend.cpp)
-endif()
-
 if(HAVE_WAYLAND)
     set(kwin_KDEINIT_SRCS
             ${kwin_KDEINIT_SRCS}
diff -urN a/scene_opengl.cpp b/scene_opengl.cpp
--- a/scene_opengl.cpp	2015-02-21 13:21:17.000000000 -0700
+++ b/scene_opengl.cpp	2015-03-01 16:00:41.838988770 -0700
@@ -28,7 +28,6 @@
 *********************************************************************/
 #include "scene_opengl.h"
 #ifdef KWIN_HAVE_EGL
-#include "eglonxbackend.h"
 // for Wayland
 #if HAVE_WAYLAND_EGL
 #include "egl_wayland_backend.h"
@@ -488,10 +487,10 @@
         if (kwinApp()->shouldUseWaylandForCompositing()) {
             backend = new EglWaylandBackend();
         } else {
-            backend = new EglOnXBackend();
+            backend = NULL;
         }
 #else
-        backend = new EglOnXBackend();
+        backend = NULL;
 #endif
 #endif
         break;
