diff --git a/CMakeLists.txt b/CMakeLists.txt
index c4745df..4058b34 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -222,10 +222,12 @@ set(KWIN_BUILD_OPENGL FALSE)
 set(KWIN_BUILD_OPENGLES FALSE)
 if(OPENGL_FOUND AND (${Qt5Gui_OPENGL_IMPLEMENTATION} STREQUAL "GL"))
     set(KWIN_BUILD_OPENGL TRUE)
+    include_directories(${OPENGL_INCLUDE_DIR})
     message("Building KWin with OpenGL support")
 endif()
 if(OPENGLES_FOUND AND (${Qt5Gui_OPENGL_IMPLEMENTATION} STREQUAL "GLESv2"))
     set(KWIN_BUILD_OPENGLES TRUE)
+    include_directories(${OPENGLES_INCLUDE_DIR} ${OPENGLES_EGL_INCLUDE_DIR})
     message("Building KWin with OpenGL ES 2 support")
 endif()
 
diff --git a/libkwineffects/kwinglplatform.cpp b/libkwineffects/kwinglplatform.cpp
index 548d459..5e5b25b 100644
--- a/libkwineffects/kwinglplatform.cpp
+++ b/libkwineffects/kwinglplatform.cpp
@@ -25,6 +25,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <QStringList>
 #include <QDebug>
 
+#include <X11/Xlib.h>
+
 #include <sys/utsname.h>
 
 #include <iostream>
diff --git a/libkwineffects/kwingltexture.cpp b/libkwineffects/kwingltexture.cpp
index 97d2747..1dcf0e3 100644
--- a/libkwineffects/kwingltexture.cpp
+++ b/libkwineffects/kwingltexture.cpp
@@ -186,7 +186,7 @@ void GLTexturePrivate::cleanup()
 bool GLTexture::isNull() const
 {
     Q_D(const GLTexture);
-    return None == d->m_texture;
+    return 0 == d->m_texture;
 }
 
 QSize GLTexture::size() const
