diff -urN a/libkwineffects/kwinglplatform.cpp b/libkwineffects/kwinglplatform.cpp
--- a/libkwineffects/kwinglplatform.cpp	2015-01-26 03:27:31.000000000 -0700
+++ b/libkwineffects/kwinglplatform.cpp	2015-01-27 05:56:33.605773992 -0700
@@ -25,6 +25,8 @@
 #include <QStringList>
 #include <QDebug>
 
+#include <X11/Xlib.h>
+
 #include <sys/utsname.h>
 
 #include <iostream>
diff -urN a/libkwineffects/kwingltexture.cpp b/libkwineffects/kwingltexture.cpp
--- a/libkwineffects/kwingltexture.cpp	2015-01-26 03:27:31.000000000 -0700
+++ b/libkwineffects/kwingltexture.cpp	2015-01-27 05:56:33.605773992 -0700
@@ -315,7 +315,7 @@
 bool GLTexture::isNull() const
 {
     Q_D(const GLTexture);
-    return None == d->m_texture;
+    return 0 == d->m_texture;
 }
 
 QSize GLTexture::size() const
