diff -urN qt-x11-free-3.3.8.orig/src/tools/qglobal.h qt-x11-free-3.3.8/src/tools/qglobal.h
--- qt-x11-free-3.3.8.orig/src/tools/qglobal.h	2011-01-20 21:18:42.000000000 -0700
+++ qt-x11-free-3.3.8/src/tools/qglobal.h	2011-01-20 21:20:34.000000000 -0700
@@ -320,6 +320,9 @@
 #    if __GNUC__ == 3 && __GNUC_MINOR__ >= 4
 #      define Q_NO_PACKED_REFERENCE
 #    endif
+#    if __GNUC__ == 4 && __GNUC_MINOR__ >= 0
+#      define Q_NO_PACKED_POINTERS
+#    endif
 #  endif
 #  if !defined(__EXCEPTIONS)
 #    define Q_NO_EXCEPTIONS
diff -urN qt-x11-free-3.3.8.orig/src/tools/qstring.h qt-x11-free-3.3.8/src/tools/qstring.h
--- qt-x11-free-3.3.8.orig/src/tools/qstring.h	2011-01-20 21:18:43.000000000 -0700
+++ qt-x11-free-3.3.8/src/tools/qstring.h	2011-01-20 21:21:48.000000000 -0700
@@ -195,6 +195,8 @@
     ushort unicode() const { return ucs; }
 #ifdef Q_NO_PACKED_REFERENCE
     ushort &unicode() { return *(&ucs); }
+#elif defined Q_NO_PACKED_POINTERS
+    ushort &unicode() { ushort& tmp = ucs; return tmp; }
 #else
     ushort &unicode() { return ucs; }
 #endif
