diff --git a/tests/Makefile.am b/tests/Makefile.am
index 2872cb1aa..247dfd3d8 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -530,10 +530,8 @@
 indirect_tests += gnutls_ktls
 dist_check_SCRIPTS += ktls.sh
 
-indirect_tests += ktls_keyupdate
 ktls_keyupdate_SOURCES = tls13/key_update.c
 ktls_keyupdate_CFLAGS = -DUSE_KTLS
-dist_check_SCRIPTS += ktls_keyupdate.sh
 endif
 
 dist_check_SCRIPTS += dtls/dtls.sh dtls/dtls-resume.sh #dtls/dtls-nb
