From b64e8e1f72a1c41d3df346c17efd49bcd65b1e8b Mon Sep 17 00:00:00 2001
From: Andreas Metzler <ametzler@bebt.de>
Date: Thu, 15 Aug 2024 12:42:56 +0200
Subject: [PATCH 1/2] Also set ENABLE_DSA for tests in cert-tests subdirectory.

Signed-off-by: Andreas Metzler <ametzler@bebt.de>
---
 tests/cert-tests/Makefile.am | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tests/cert-tests/Makefile.am b/tests/cert-tests/Makefile.am
index ea5673d37e..5a695dbebc 100644
--- a/tests/cert-tests/Makefile.am
+++ b/tests/cert-tests/Makefile.am
@@ -180,6 +180,12 @@ if WINDOWS
 TESTS_ENVIRONMENT += WINDOWS=1
 endif
 
+if ENABLE_DSA
+TESTS_ENVIRONMENT += ENABLE_DSA=1
+else
+TESTS_ENVIRONMENT += ENABLE_DSA=0
+endif
+
 if ENABLE_GOST
 TESTS_ENVIRONMENT += ENABLE_GOST=1
 else
-- 
GitLab


From f3e8eac0586a19f4dafd89f68006a536b826e65a Mon Sep 17 00:00:00 2001
From: Andreas Metzler <ametzler@bebt.de>
Date: Thu, 15 Aug 2024 16:22:02 +0200
Subject: [PATCH 2/2] revert back to datefudge for "openssl ocsp".

openssl's -attime only changes the verification logic but not the
generation.

Broken by: d1bc7f644422c4d87edfcd9fafe7f292a1a3a6de

Signed-off-by: Andreas Metzler <ametzler@bebt.de>
---
 tests/ocsp-tests/ocsp-must-staple-connection.sh | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/tests/ocsp-tests/ocsp-must-staple-connection.sh b/tests/ocsp-tests/ocsp-must-staple-connection.sh
index bfa785e040..94d41ce245 100755
--- a/tests/ocsp-tests/ocsp-must-staple-connection.sh
+++ b/tests/ocsp-tests/ocsp-must-staple-connection.sh
@@ -48,6 +48,8 @@ fi
 
 . "${srcdir}/scripts/common.sh"
 
+skip_if_no_datefudge
+
 eval "${GETPORT}"
 # Port for gnutls-serv
 TLS_SERVER_PORT=$PORT
@@ -69,7 +71,6 @@ fi
 
 CERTDATE="2016-04-28 00:00:00"
 TESTDATE="2016-04-29 00:00:00"
-EPOCHTESTDATE=1461888000
 EXP_OCSP_DATE="2016-03-27 00:00:00"
 
 OCSP_PID=""
@@ -129,8 +130,8 @@ cp "${srcdir}/ocsp-tests/certs/ocsp_index.txt.attr" ${ATTRFILE}
 # SO_REUSEADDR usage.
 PORT=${OCSP_PORT}
 launch_bare_server \
-	  "${OPENSSL}" ocsp -attime "${EPOCHTESTDATE}" \
-	  -index "${INDEXFILE}" -text \
+	  "$FAKETIME" "${TESTDATE}" \
+	  "${OPENSSL}" ocsp -index "${INDEXFILE}" -text \
 	  -port "${OCSP_PORT}" \
 	  -rsigner "${srcdir}/ocsp-tests/certs/ocsp-server.pem" \
 	  -rkey "${srcdir}/ocsp-tests/certs/ocsp-server.key" \
-- 
GitLab

