diff -Naur rxtx-2.2pre2.orig/configure.in rxtx-2.2pre2/configure.in
--- rxtx-2.2pre2.orig/configure.in	2008-12-31 03:28:37.000000000 +0100
+++ rxtx-2.2pre2/configure.in	2018-04-25 07:50:42.382848449 +0200
@@ -483,8 +483,7 @@
 		;;
 	*)
 		find_java_native_include jni_md.h
-		JAVAH="\$(JPATH)/bin/javah -classpath \$(CLASSPATH) -d \$(DEST) -jni"
-		JAVAC="\$(JPATH)/bin/javac -classpath \$(CLASSPATH) -d \$(TOP)/ -O -source 1.3 -target 1.3"
+		JAVAC="\$(JPATH)/bin/javac -classpath \$(CLASSPATH) -d \$(TOP)/ -h \$(DEST) -O -source 6 -target 1.6"
 		JAR="\$(JPATH)/bin/jar"
 		JAVADOC="\$(JPATH)/bin/javadoc"
 		CFLAGS=$CFLAGS" -D_BSD_SOURCE"
diff -Naur rxtx-2.2pre2.orig/Makefile.am rxtx-2.2pre2/Makefile.am
--- rxtx-2.2pre2.orig/Makefile.am	2008-09-14 23:36:27.000000000 +0200
+++ rxtx-2.2pre2/Makefile.am	2018-04-25 07:50:01.198326446 +0200
@@ -111,11 +111,6 @@
 $(CLASSES): $(javafiles) $(TOP)/Makefile
 	$(JAVAH_FIX)
 	$(JAVAC) $(javafiles)
-	$(JAVAH) `(for i in $(javahfiles);do \
-		if grep > /dev/null native.*\( $(SPATH)/gnu/io/$$i.java; then \
-			echo $(CLASSDIR).$$i; \
-		fi; \
-	done)`
 	$(JAR) cf $(JARTARGET) $(CLASSTOP)/*
 
 #  install librxtxSerial.so into the proper directory and copy $(JARTARGET) to its
