Description: sys/io.h does not exist on all systems
Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=609152
Author: Scott Howard <showard@debian.org>

Index: rxtx/configure.in
===================================================================
--- rxtx.orig/configure.in	2011-02-23 23:01:09.596613286 -0500
+++ rxtx/configure.in	2011-02-23 23:02:41.269525630 -0500
@@ -71,6 +71,7 @@
 AC_CHECK_HEADERS(sys/time.h)
 AC_CHECK_HEADERS(sys/signal.h)
 AC_CHECK_HEADERS(signal.h)
+AC_CHECK_HEADERS(sys/io.h)
 AC_CHECK_HEADERS(termios.h)
 AC_CHECK_HEADERS(grp.h)
 AC_CHECK_HEADERS(pwd.h)
Index: rxtx/src/RawImp.c
===================================================================
--- rxtx.orig/src/RawImp.c	2011-02-23 23:04:29.641932242 -0500
+++ rxtx/src/RawImp.c	2011-02-23 23:05:40.487353587 -0500
@@ -99,11 +99,9 @@
 #	include <linux/serial.h>
 #	include <linux/version.h>
 #endif
-#ifndef __APPLE__  /* dima */
-#ifndef PPC
-#include <sys/io.h>
-#endif /* PPC */
-#endif /* dima */
+#ifdef HAVE_SYS_IO_H
+#   include <sys/io.h>
+#endif
 
 extern int errno;
 #include "I2CImp.h"
