diff -Naur rxtx-2.2pre2.orig/src/I2CImp.c rxtx-2.2pre2/src/I2CImp.c
--- rxtx-2.2pre2.orig/src/I2CImp.c	2007-04-26 07:26:05.000000000 +0200
+++ rxtx-2.2pre2/src/I2CImp.c	2013-08-10 13:28:54.690313710 +0200
@@ -125,7 +125,7 @@
 	sigaction( SIGIO, NULL, &handler );
 	if( !handler.sa_handler ) signal( SIGIO, SIG_IGN );
 #endif /* !__FreeBSD__ */
-#if defined(__linux__) 
+#if defined(DEBUG) && defined(__linux__) && defined(UTS_RELEASE) 
 	/* Lets let people who upgraded kernels know they may have problems */
 	if (uname (&name) == -1)
 	{
diff -Naur rxtx-2.2pre2.orig/src/RawImp.c rxtx-2.2pre2/src/RawImp.c
--- rxtx-2.2pre2.orig/src/RawImp.c	2007-11-18 23:32:42.000000000 +0100
+++ rxtx-2.2pre2/src/RawImp.c	2013-08-10 13:28:54.703647364 +0200
@@ -270,7 +270,7 @@
 	sigaction( SIGIO, NULL, &handler );
 	if( !handler.sa_handler ) signal( SIGIO, SIG_IGN );
 #endif /* !__FreeBSD__ */
-#if defined(__linux__) 
+#if defined(DEBUG) && defined(__linux__) && defined(UTS_RELEASE) 
 	/* Lets let people who upgraded kernels know they may have problems */
 	if (uname (&name) == -1)
 	{
diff -Naur rxtx-2.2pre2.orig/src/RS485Imp.c rxtx-2.2pre2/src/RS485Imp.c
--- rxtx-2.2pre2.orig/src/RS485Imp.c	2007-04-26 07:26:05.000000000 +0200
+++ rxtx-2.2pre2/src/RS485Imp.c	2013-08-10 13:28:54.706980778 +0200
@@ -125,7 +125,7 @@
 	sigaction( SIGIO, NULL, &handler );
 	if( !handler.sa_handler ) signal( SIGIO, SIG_IGN );
 #endif /* !__FreeBSD__ */
-#if defined(__linux__) 
+#if defined(DEBUG) && defined(__linux__) && defined(UTS_RELEASE) 
 	/* Lets let people who upgraded kernels know they may have problems */
 	if (uname (&name) == -1)
 	{
