diff -Nur klibc-1.5.orig/klcc/klcc.in klibc-1.5/klcc/klcc.in
--- klibc-1.5.orig/klcc/klcc.in	2007-03-04 02:52:10.000000000 +0100
+++ klibc-1.5/klcc/klcc.in	2007-10-31 09:20:34.000000000 +0100
@@ -113,7 +113,7 @@
 
 $save_temps = 0;		# The -save-temps option
 $verbose = 0;			# The -v option
-$shared = 0;	   		# Are we compiling shared?
+$shared = 1;	   		# Are we compiling shared?
 $debugging = 0;	   		# -g or -p option present?
 $strip = 0;			# -s option present?
 undef $output;			# -o option present?
