diff -Nur klibc-1.5.14.orig/usr/include/arch/x86_64/sys/io.h klibc-1.5.14/usr/include/arch/x86_64/sys/io.h
--- klibc-1.5.14.orig/usr/include/arch/x86_64/sys/io.h	2008-07-30 23:06:03.000000000 +0200
+++ klibc-1.5.14/usr/include/arch/x86_64/sys/io.h	2008-09-18 14:34:20.000000000 +0200
@@ -60,21 +60,21 @@
 {
 	unsigned char __v;
 	asm volatile ("inb %1,%0" : "=a" (__v) : "dN"(__p));
-	return v;
+	return __v;
 }
 
 static __inline__ unsigned short inw(unsigned short __p)
 {
 	unsigned short __v;
 	asm volatile ("inw %1,%0" : "=a" (__v) : "dN"(__p));
-	return v;
+	return __v;
 }
 
 static __inline__ unsigned int inl(unsigned short __p)
 {
 	unsigned int __v;
 	asm volatile ("inl %1,%0" : "=a" (__v) : "dN"(__p));
-	return v;
+	return __v;
 }
 
 /* String I/O macros */
