--- everygui/setup.py	2006-01-10 00:59:47.000000000 -0800
+++ everygui-new/setup.py	2006-01-26 22:29:22.000000000 -0800
@@ -1,32 +1,31 @@
-#!/usr/bin/env python
-
-# build with python setup.py
-
+#!/usr/bin/env python
+
+# build with python setup.py
+
 # setup.py
 import os
-import sys
-from distutils.core import setup
-import glob
+import sys
+from distutils.core import setup
+import glob
 
 try:
 	import gtk
 except:
 	sys.exit("pyGTK version 2.6 or greater is required to run everygui. http://www.pygtk.org")
 
-
-setup(
-	name='everygui',
-	version='0.99.b',
-	description='GTK+ interface for "everything", loosely speaking',
-	author='Jeffrey Bakker',
+
+setup(
+	name='everygui',
+	version='0.99.b',
+	description='GTK+ interface for "everything", loosely speaking',
+	author='Jeffrey Bakker',
 	author_email='seeplusplus@users.sf.net',
 	url='http://everygui.sourceforge.net',
-	license = 'GPL',
+	license = 'GPL',
+
 	data_files=[("share/everygui/ui", glob.glob("ui/*.glade")),
 			("bin", ["everygui", "egdesign"]),
-			("src/everygui", glob.glob("src/*")),
+			("lib/python2.6/site-packages/everygui", glob.glob("src/*")),
 			("share/everygui/docs", ["COPYING","LGPL","README","ChangeLog","AUTHORS","TODO"]),
-			("share/everygui/egconfig", glob.glob("egconfig/*.egd"))]
+			("share/everygui/egconfig", glob.glob("egconfig/*.egd"))]
 	)
-
-os.chmod("/usr/share/everygui/egconfig", 0777)
