diff -urN a/sql-common/client_plugin.c b/sql-common/client_plugin.c
--- a/sql-common/client_plugin.c	2011-03-31 07:36:18.000000000 -0600
+++ b/sql-common/client_plugin.c	2011-04-07 11:59:48.000000000 -0600
@@ -242,9 +242,9 @@
   initialized= 1;
 
   pthread_mutex_lock(&LOCK_load_client_plugin);
-
+  va_list a = {0};
   for (builtin= mysql_client_builtins; *builtin; builtin++)
-    add_plugin(&mysql, *builtin, 0, 0, 0);
+    add_plugin(&mysql, *builtin, 0, 0, a);
 
   pthread_mutex_unlock(&LOCK_load_client_plugin);
 
@@ -301,8 +301,10 @@
                              plugin->name, "it is already loaded");
     plugin= NULL;
   }
-  else
-    plugin= add_plugin(mysql, plugin, 0, 0, 0);
+  else {
+    va_list a = {0};
+    plugin= add_plugin(mysql, plugin, 0, 0, a);
+  }
 
   pthread_mutex_unlock(&LOCK_load_client_plugin);
   return plugin;
