diff -Naur mysql-5.1.32/Makefile.in mysql-5.1.32-novo/Makefile.in
--- mysql-5.1.32/Makefile.in	2009-02-13 23:03:22.000000000 -0200
+++ mysql-5.1.32-novo/Makefile.in	2009-03-24 04:48:19.000000000 -0300
@@ -396,8 +396,9 @@
 SUBDIRS = . include @docs_dirs@ @zlib_dir@ \
 			@readline_topdir@ sql-common scripts \
 			@pstack_dir@ \
-			@sql_union_dirs@ unittest storage plugin \
-			@sql_server@ @man_dirs@ tests \
+			strings mysys dbug storage regex extra vio \
+			sql unittest plugin \
+			@sql_server@ \
 			netware @libmysqld_dirs@ \
 			mysql-test support-files sql-bench @tools_dirs@ \
 			win

