diff -ruN -x .gitignore -x .cvsignore xf86-input-ur98-d7dc1bb24a90d7c8275ec51062b7f6f95b6a7760/README xf86-input-ur98-4250b8149d1d021d4e1315e879abfb0774981abc/README
--- xf86-input-ur98-d7dc1bb24a90d7c8275ec51062b7f6f95b6a7760/README	1970-01-01 00:00:00.000000000 +0000
+++ xf86-input-ur98-4250b8149d1d021d4e1315e879abfb0774981abc/README	2009-01-31 04:30:00.000000000 +0000
@@ -0,0 +1,20 @@
+UR-98 - UR98 (TR88L803) head tracker driver for the Xorg X server
+
+Please submit bugs & patches to the Xorg bugzilla:
+
+        https://bugs.freedesktop.org/enter_bug.cgi?product=xorg
+
+All questions regarding this software should be directed at the
+Xorg mailing list:
+
+        http://lists.freedesktop.org/mailman/listinfo/xorg
+
+The master development code repository can be found at:
+
+        git://anongit.freedesktop.org/git/xorg/driver/xf86-input-ur98
+
+        http://cgit.freedesktop.org/xorg/driver/xf86-input-ur98
+
+For more information on the git code manager, see:
+
+        http://wiki.x.org/wiki/GitPage
diff -ruN -x .gitignore -x .cvsignore xf86-input-ur98-d7dc1bb24a90d7c8275ec51062b7f6f95b6a7760/man/ur98.man xf86-input-ur98-4250b8149d1d021d4e1315e879abfb0774981abc/man/ur98.man
--- xf86-input-ur98-d7dc1bb24a90d7c8275ec51062b7f6f95b6a7760/man/ur98.man	2006-04-07 18:13:50.000000000 +0000
+++ xf86-input-ur98-4250b8149d1d021d4e1315e879abfb0774981abc/man/ur98.man	2009-01-31 04:30:00.000000000 +0000
@@ -125,7 +125,7 @@
 This driver is currently Linux specific.
 .PP
 .SH "SEE ALSO"
-__xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), xorgconfig(__appmansuffix__), Xserver(__appmansuffix__), X(__miscmansuffix__).
+__xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), Xserver(__appmansuffix__), X(__miscmansuffix__).
 .SH AUTHORS
 Authors include...
  Alan Cox
diff -ruN -x .gitignore -x .cvsignore xf86-input-ur98-d7dc1bb24a90d7c8275ec51062b7f6f95b6a7760/src/xf86Ur-98.c xf86-input-ur98-4250b8149d1d021d4e1315e879abfb0774981abc/src/xf86Ur-98.c
--- xf86-input-ur98-d7dc1bb24a90d7c8275ec51062b7f6f95b6a7760/src/xf86Ur-98.c	2006-04-07 18:13:50.000000000 +0000
+++ xf86-input-ur98-4250b8149d1d021d4e1315e879abfb0774981abc/src/xf86Ur-98.c	2009-01-31 04:30:00.000000000 +0000
@@ -49,7 +49,9 @@
 	"xf86CollectInputOptions",
 	"xf86ErrorFVerb",
 	"xf86FindOptionValue",
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
 	"xf86GetMotionEvents",
+#endif
 	"xf86GetVerbosity",
 	"xf86MotionHistoryAllocate",
 	"xf86NameCmp",
@@ -316,7 +318,10 @@
 		 * screen to fit one meter.
 		 */
 	
-		if (InitValuatorClassDeviceStruct(dev, priv->axes, xf86GetMotionEvents,
+		if (InitValuatorClassDeviceStruct(dev, priv->axes,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
+				xf86GetMotionEvents,
+#endif
 				local->history_size, Absolute) == FALSE) {
 			ErrorF("Unable to allocate ValuatorClassDeviceStruct\n");
 			return !Success;
