From c950a2fe53e8f3269b2cb969fca2c37397732216 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Wed, 12 Oct 2016 06:40:31 -0600
Subject: [PATCH] ARM fix

---
 libide/highlighting/ide-highlight-index.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libide/highlighting/ide-highlight-index.c b/libide/highlighting/ide-highlight-index.c
index 1e90686..711bd89 100644
--- a/libide/highlighting/ide-highlight-index.c
+++ b/libide/highlighting/ide-highlight-index.c
@@ -31,6 +31,11 @@ G_DEFINE_BOXED_TYPE (IdeHighlightIndex, ide_highlight_index,
 
 EGG_DEFINE_COUNTER (instances, "IdeHighlightIndex", "Instances", "Number of indexes")
 
+#ifndef PAGE_SIZE
+#include <unistd.h>
+#define PAGE_SIZE sysconf(_SC_PAGESIZE)
+#endif
+
 struct _IdeHighlightIndex
 {
   volatile gint  ref_count;
-- 
2.10.0

