diff -urN a/libide/ide-highlight-index.c b/libide/ide-highlight-index.c
--- a/libide/ide-highlight-index.c	2015-04-08 00:43:25.000000000 -0600
+++ b/libide/ide-highlight-index.c	2015-04-14 13:18:59.080511371 -0600
@@ -28,6 +28,11 @@
 G_DEFINE_BOXED_TYPE (IdeHighlightIndex, ide_highlight_index,
                      ide_highlight_index_ref, ide_highlight_index_unref)
 
+#ifndef PAGE_SIZE
+#include <unistd.h>
+#define PAGE_SIZE sysconf(_SC_PAGESIZE)
+#endif
+
 struct _IdeHighlightIndex
 {
   volatile gint  ref_count;
