From f77c428772273db365bba4d21b6913196b7947ad Mon Sep 17 00:00:00 2001
From: Ignacio Casal Quinteiro <icq@gnome.org>
Date: Fri, 17 Jul 2015 11:58:34 +0200
Subject: git: pass NULL to the new binary callback


diff --git a/libide/git/ide-git-buffer-change-monitor.c b/libide/git/ide-git-buffer-change-monitor.c
index a6a306d..7763c76 100644
--- a/libide/git/ide-git-buffer-change-monitor.c
+++ b/libide/git/ide-git-buffer-change-monitor.c
@@ -618,7 +618,7 @@ ide_git_buffer_change_monitor_calculate_threaded (IdeGitBufferChangeMonitor  *se
   data = g_bytes_get_data (diff->content, &data_len);
 
   ggit_diff_blob_to_buffer (diff->blob, relative_path, data, data_len, relative_path,
-                            NULL, NULL, NULL, diff_line_cb, (gpointer)diff->state, error);
+                            NULL, NULL, NULL, NULL, diff_line_cb, (gpointer)diff->state, error);
 
   return ((*error) == NULL);
 }
-- 
cgit v0.10.2

