From 48eeb9f7826e3ec492262f9ddb5d2b60a429740b Mon Sep 17 00:00:00 2001
From: Christian Hergert <chergert@redhat.com>
Date: Thu, 15 Oct 2015 08:36:52 -0700
Subject: build: install libide-1.0.pc into private directory

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 48e783e..90267e6 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,6 @@
 SUBDIRS = . build contrib data libide src plugins po tools tests doc
 
-pkgconfigdir = $(libdir)/pkgconfig
+pkgconfigdir = $(libdir)/gnome-builder/pkgconfig
 pkgconfig_DATA = $(top_builddir)/data/libide-1.0.pc
 
 INTLTOOL_FILES = \
-- 
cgit v0.11.2

