diff --git i/meson.build w/meson.build
index 71e4be94c..a6257e535 100644
--- i/meson.build
+++ w/meson.build
@@ -284,7 +284,7 @@ libjsonrpc_glib_dep = dependency('jsonrpc-glib-1.0', version: '>= 3.29.91')
 libm_dep = cc.find_library('m', required: false)
 libpangoft2_dep = dependency('pangoft2', version: '>= 1.38.0')
 libpeas_dep = dependency('libpeas-1.0', version: '>= 1.22.0')
-libportal_dep = dependency('libportal', version: '>= 0.3', required: false)
+libportal_dep = dependency('libportal-gtk3', version: '>= 0.5', required: false)
 libtemplate_glib_dep = dependency('template-glib-1.0', version: '>= 3.28.0')
 libvte_dep = dependency('vte-2.91', version: '>= 0.40.2')
 libwebkit_dep = dependency('webkit2gtk-4.0', version: '>= 2.26', required: false)
diff --git i/src/gstyle/gstyle-color-panel.c w/src/gstyle/gstyle-color-panel.c
index 15742a7be..a6295c168 100644
--- i/src/gstyle/gstyle-color-panel.c
+++ w/src/gstyle/gstyle-color-panel.c
@@ -30,7 +30,7 @@
 #include "gstyle-color.h"
 
 #include <libportal/portal.h>
-#include <libportal/portal-gtk3.h>
+#include <libportal-gtk3/portal-gtk3.h>
 
 #define HSV_TO_SCALE_FACTOR (1.0 / 256.0)
 #define CIELAB_L_TO_SCALE_FACTOR (100.0 / 256.0)
diff --git i/src/plugins/devhelp/gbp-devhelp-page.c w/src/plugins/devhelp/gbp-devhelp-page.c
index 5d5c52b0d..2ac788414 100644
--- i/src/plugins/devhelp/gbp-devhelp-page.c
+++ w/src/plugins/devhelp/gbp-devhelp-page.c
@@ -25,7 +25,7 @@
 #include <webkit2/webkit2.h>
 
 #include <libportal/portal.h>
-#include <libportal/portal-gtk3.h>
+#include <libportal-gtk3/portal-gtk3.h>
 
 #include "gbp-devhelp-page.h"
 #include "gbp-devhelp-search.h"
diff --git i/src/plugins/open-with-external/gbp-owe-workbench-addin.c w/src/plugins/open-with-external/gbp-owe-workbench-addin.c
index 51d2da844..816fe643f 100644
--- i/src/plugins/open-with-external/gbp-owe-workbench-addin.c
+++ w/src/plugins/open-with-external/gbp-owe-workbench-addin.c
@@ -23,7 +23,7 @@
 #include "gbp-owe-workbench-addin.h"
 
 #include <libportal/portal.h>
-#include <libportal/portal-gtk3.h>
+#include <libportal-gtk3/portal-gtk3.h>
 
 struct _GbpOweWorkbenchAddin
 {
