diff -wbBur xchatosd-5.19/my_xosd.h xchatosd-5.19.my/my_xosd.h
--- xchatosd-5.19/my_xosd.h	2005-01-20 23:38:28.000000000 +0300
+++ xchatosd-5.19.my/my_xosd.h	2006-11-22 17:01:34.000000000 +0300
@@ -8,8 +8,8 @@
 
 class XOsd {
 	public:
-		XOsd::XOsd(int);
-		XOsd::~XOsd();
+		XOsd(int);
+		~XOsd();
 		int setPos(string);
 		int setVOffset(int);
 		int setHOffset(int);
diff -wbBur xchatosd-5.19/xchatosd_conf.h xchatosd-5.19.my/xchatosd_conf.h
--- xchatosd-5.19/xchatosd_conf.h	2005-01-28 23:54:58.000000000 +0300
+++ xchatosd-5.19.my/xchatosd_conf.h	2006-11-22 17:01:52.000000000 +0300
@@ -11,8 +11,8 @@
 
 class XOsdConf {
 	public:
-		XOsdConf::XOsdConf();
-		XOsdConf::~XOsdConf();
+		XOsdConf();
+		~XOsdConf();
 		int read(string);
 		int write(string);
 		int getiValue(string);
