diff -Naur motor-3.4.0.orig/kkconsui/src/texteditor.cc motor-3.4.0.new/kkconsui/src/texteditor.cc
--- motor-3.4.0.orig/kkconsui/src/texteditor.cc	2003-09-09 17:51:33.000000000 -0400
+++ motor-3.4.0.new/kkconsui/src/texteditor.cc	2007-06-29 15:55:13.000000000 -0400
@@ -1940,7 +1940,7 @@
 }
 
 int texteditor::findint(void *p1, void *p2) {
-    return *(int *) p1 != (int) p2;
+    return *(int *) p1 != *(int *) p2;
 }
 
 int texteditor::findhighline(void *p1, void *p2) {
diff -Naur motor-3.4.0.orig/kkstrtext/kkstrtext.cc motor-3.4.0.new/kkstrtext/kkstrtext.cc
--- motor-3.4.0.orig/kkstrtext/kkstrtext.cc	2005-01-31 19:13:24.000000000 -0500
+++ motor-3.4.0.new/kkstrtext/kkstrtext.cc	2007-06-29 15:53:39.000000000 -0400
@@ -431,7 +431,7 @@
 }
 
 int intcompare(void *s1, void *s2) {
-    return (int) s1 != (int) s2;
+    return *(int *) s1 != *(int *) s2;
 }
 
 string i2str(int i) {
diff -Naur motor-3.4.0.orig/src/ui/ncurses/ncursesui.cc motor-3.4.0.new/src/ui/ncurses/ncursesui.cc
--- motor-3.4.0.orig/src/ui/ncurses/ncursesui.cc	2004-11-14 16:04:08.000000000 -0500
+++ motor-3.4.0.new/src/ui/ncurses/ncursesui.cc	2007-06-29 15:57:40.000000000 -0400
@@ -1242,7 +1242,7 @@
 	populatesettingstree(*db.gettree());
         if(fin = !db.open(n, b, &p)) break;
 
-        switch((int) p) {
+        switch(*(int *) p) {
             case 10:
 		if(input(motorui::text, buf = project.getversion(),
 		_("version: ")) == motorui::yes) {
@@ -1368,7 +1368,7 @@
 	    id = tree.getid(i);
 
 	    if(tree.isnode(id))
-		if(sid = (int) tree.getref(id))
+		if(sid = *(int *) tree.getref(id))
 		    if(!tree.isnodeopen(id))
 			cnodes.push_back(treeshit[sid-1].second->gettagname());
 	}
diff -Naur motor-3.4.0.orig/src/ui/ncurses/uitagbrowser.cc motor-3.4.0.new/src/ui/ncurses/uitagbrowser.cc
--- motor-3.4.0.orig/src/ui/ncurses/uitagbrowser.cc	2004-07-22 20:18:16.000000000 -0400
+++ motor-3.4.0.new/src/ui/ncurses/uitagbrowser.cc	2007-06-29 15:59:42.000000000 -0400
@@ -95,7 +95,7 @@
     cm.close();
 
     if(n) {
-	int c = (int) cm.getref(n-1);
+	int c = *(int *) cm.getref(n-1);
 
 	if(c != tagbrowser.getscope()) {
 	    tagbrowser.setscope((motortagbrowser::viewscope) c);
@@ -142,7 +142,7 @@
 	switch(cm.open()) {
 	    case -2:
 		iter = false;
-		c = (char) (int) cm.getref(cm.getpos());
+		c = (char) *(int *) cm.getref(cm.getpos());
 
 		while((pos = nfilter.find(c)) != -1) {
 		    nfilter.erase(pos, 1);
@@ -240,7 +240,7 @@
     m.setpos(mpos);
 
     if(r = (n = m.open())) {
-        k = (int) m.getref(n-1);
+        k = *(int *) m.getref(n-1);
         mpos = n-1;
     }
 
diff -Naur motor-3.4.0.orig/src/ui/ncurses/uivcs.cc motor-3.4.0.new/src/ui/ncurses/uivcs.cc
--- motor-3.4.0.orig/src/ui/ncurses/uivcs.cc	2005-02-12 09:58:28.000000000 -0500
+++ motor-3.4.0.new/src/ui/ncurses/uivcs.cc	2007-06-29 16:00:41.000000000 -0400
@@ -98,7 +98,7 @@
 	switch(b) {
 	    case 0:
 		if(!i) {
-                    i = (int) tree.getref(tree.getid(n));
+                    i = *(int *) tree.getref(tree.getid(n));
 		    ic = ch.begin()+i-1;
 
                     found = (ik = find(naffect.begin(), naffect.end(),
