From d8e552de80f4a7d2834843b808369315cb5b4b40 Mon Sep 17 00:00:00 2001
From: Daniel Winzen <d@winzen4.de>
Date: Thu, 15 Jan 2015 22:15:04 +0100
Subject: [PATCH] Fix calcSlope1 was not declared errors

---
 plugins/monstro/Monstro.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/plugins/monstro/Monstro.cpp b/plugins/monstro/Monstro.cpp
index dc51619..d93af44 100644
--- a/plugins/monstro/Monstro.cpp
+++ b/plugins/monstro/Monstro.cpp
@@ -784,7 +784,7 @@ inline void MonstroSynth::updateModulators( float * env1, float * env2, float *
 			}
 			else if( m_env_phase[i] < 2.0f ) // attack phase
 			{
-				env[i][f] = calcSlope1( fraction( m_env_phase[i] ) );
+				env[i][f] = calcSlope( i, fraction( m_env_phase[i] ) );
 				m_env_phase[i] = qMin( 2.0f, m_env_phase[i] + m_env_att[i] );
 			}
 			else if( m_env_phase[i] < 3.0f ) // hold phase
@@ -794,7 +794,7 @@ inline void MonstroSynth::updateModulators( float * env1, float * env2, float *
 			}
 			else if( m_env_phase[i] < 4.0f ) // decay phase
 			{
-				const sample_t s = calcSlope1( 1.0f - fraction( m_env_phase[i] ) );
+				const sample_t s = calcSlope( i, 1.0f - fraction( m_env_phase[i] ) );
 				if( s <= m_env_sus[i] )
 				{
 					env[i][f] = m_env_sus[i];
@@ -808,7 +808,7 @@ inline void MonstroSynth::updateModulators( float * env1, float * env2, float *
 			}
 			else if( m_env_phase[i] < 5.0f ) // release phase
 			{
-				env[i][f] = calcSlope1( 1.0f - fraction( m_env_phase[i] ) );
+				env[i][f] = calcSlope( i, 1.0f - fraction( m_env_phase[i] ) );
 				m_env_phase[i] += m_env_rel[i];
 			}
 			else env[i][f] = 0.0f;
-- 
2.2.2

