diff --git a/plugins/carlabase/carla.h b/plugins/carlabase/carla.h
index fb54e2271..beb6d935f 100644
--- a/plugins/carlabase/carla.h
+++ b/plugins/carlabase/carla.h
@@ -34,10 +34,10 @@
     #include "CarlaBackend.h"
     #include "CarlaNative.h"
     #include "CarlaUtils.h"
-    CARLA_EXPORT
+    CARLA_API_EXPORT
     const NativePluginDescriptor* carla_get_native_patchbay_plugin();
 
-    CARLA_EXPORT
+    CARLA_API_EXPORT
     const NativePluginDescriptor* carla_get_native_rack_plugin();
 #endif
 
