From 9cc1a5931c34535436a2fb82e51031121b3a53d1 Mon Sep 17 00:00:00 2001
From: Lukas W <lukaswhl@gmail.com>
Date: Thu, 22 Jan 2015 12:39:20 +0100
Subject: [PATCH] Make -Werror optional

---
 CMakeLists.txt                | 3 ++-

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4cb7407..a526df3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -358,7 +358,8 @@ CONFIGURE_FILE("${CMAKE_SOURCE_DIR}/plugins/zynaddsubfx/zynaddsubfx.rc.in" "${CM
 
 # set compiler flags
 SET(WERROR_FLAGS "-Wall -Werror=unused-function -Wno-sign-compare -Wno-strict-overflow")
-IF(NOT LMMS_BUILD_APPLE)
+OPTION(USE_WERROR "Add -werror to the build flags. Stops the build on warnings" OFF)
+IF(${USE_WERROR})
    SET(WERROR_FLAGS "${WERROR_FLAGS} -Werror")
 ENDIF()
 
