--- src/kile/kilestdtools.rc #809350:809351
@@ -96,29 +96,23 @@

 [Tool/DVItoPDF/Default]
 class=Convert
-command=dvipdfm
+command=dvipdfmx
 options='%S.dvi'
 from=dvi
 to=pdf
 type=Process

-[Tool/DVItoPDF/Modern]
+[Tool/DVItoPDF/Black and White]
 class=Convert
 command=dvipdfmx
-options='%S.dvi'
-from=dvi
-to=pdf
-type=Process
-
-[Tool/DVItoPDF/Black and White]
-command=dvipdfm
 options=-c '%S.dvi'
 from=dvi
 to=pdf
 type=Process

 [Tool/DVItoPDF/Landscape]
-command=dvipdfm
+class=Convert
+command=dvipdfmx
 options=-l '%S.dvi'
 from=dvi
 to=pdf
@@ -159,10 +153,21 @@
 jumpToFirstError=yes
 autoRun=yes

+[Tool/LaTeX/Draft]
+class=LaTeX
+command=latex
+options=-draftmode -interaction=nonstopmode '%source'
+from=
+to=dvi
+type=Process
+checkForRoot=yes
+jumpToFirstError=yes
+autoRun=yes
+
 [Tool/LaTeX/Modern]
 class=LaTeX
 command=latex
-options=-src -interaction=nonstopmode '%source'
+options=-src-specials -interaction=nonstopmode '%source'
 from=
 to=dvi
 type=Process
@@ -202,7 +207,7 @@
 to=html
 type=Process

-[Tool/MakeIndex/Default]
+[Tool/MakeIndex/makeindex]
 class=Compile
 command=makeindex
 options='%S.idx'
@@ -210,6 +215,14 @@
 to=idx
 type=Process

+[Tool/MakeIndex/xindy]
+class=Compile
+command=xindy
+options='%S.idx'
+from=idx
+to=idx
+type=Process
+
 [Tool/MetaPost/Default]
 class=Compile
 command=mpost
@@ -251,7 +264,7 @@
 [Tool/PDFLaTeX/Modern]
 class=LaTeX
 command=pdflatex
-options=-src -interaction=nonstopmode '%source'
+options=-src-specials -interaction=nonstopmode '%source'
 from=
 to=pdf
 type=Process
@@ -259,6 +272,94 @@
 jumpToFirstError=yes
 autoRun=yes

+[Tool/ConTeXt/PDF]
+class=LaTeX
+command=texexec
+options=--pdf --nonstopmode '%source'
+from=
+to=pdf
+type=Process
+checkForRoot=no
+jumpToFirstError=no
+autoRun=no
+
+[Tool/ConTeXt/PDF Draft]
+class=LaTeX
+command=pdflatex
+options=--pdf --draft --nonstopmode '%source'
+from=
+to=pdf
+type=Process
+checkForRoot=yes
+jumpToFirstError=yes
+autoRun=yes
+
+[Tool/ConTeXt/DVI and PS]
+class=LaTeX
+command=texexec
+options=--dvi --nonstopmode '%source'
+from=
+to=pdf
+type=Process
+checkForRoot=no
+jumpToFirstError=no
+autoRun=no
+
+[Tool/ConTeXt/DVI and PS Draft]
+class=LaTeX
+command=texexec
+options=--dvi --draft --nonstopmode '%source'
+from=
+to=pdf
+type=Process
+checkForRoot=yes
+jumpToFirstError=yes
+autoRun=yes
+
+[Tool/PDFLaTeX/Modern]
+class=LaTeX
+command=pdflatex
+options=-src-specials -interaction=nonstopmode '%source'
+from=
+to=pdf
+type=Process
+checkForRoot=yes
+jumpToFirstError=yes
+autoRun=yes
+
+[Tool/XeTeX/Default]
+class=LaTeX
+command=xetex
+options=-interaction=nonstopmode '%source'
+from=
+to=pdf
+type=Process
+checkForRoot=yes
+jumpToFirstError=yes
+autoRun=yes
+
+[Tool/XeTeX/Draft]
+class=LaTeX
+command=xetex
+options=-draftmode -interaction=nonstopmode '%source'
+from=
+to=pdf
+type=Process
+checkForRoot=yes
+jumpToFirstError=yes
+autoRun=yes
+
+[Tool/XeTeX/Modern]
+class=LaTeX
+command=xetex
+options=-src-specials -interaction=nonstopmode '%source'
+from=
+to=pdf
+type=Process
+checkForRoot=yes
+jumpToFirstError=yes
+autoRun=yes
+
 [Tool/PreviewLaTeX/Default]
 autoRun=no
 checkForRoot=no
@@ -309,11 +410,6 @@
 type=Sequence
 sequence=LaTeX,DVItoPDF,ViewPDF

-[Tool/QuickBuild/LaTeX+BibTeX+ViewDVI]
-class=Sequence
-type=Sequence
-sequence=LaTeX,BibTeX,LaTeX,LaTeX,ViewDVI
-
 [Tool/QuickBuild/PDFLaTeX+ViewPDF]
 class=Sequence
 type=Sequence
@@ -340,7 +436,7 @@
 [Tool/TeX/Modern]
 class=Compile
 command=tex
-options=-src -interaction=nonstopmode '%source'
+options=-src-specials -interaction=nonstopmode '%source'
 from=
 to=dvi
 type=Process
@@ -364,7 +460,7 @@
 [Tool/PDFTeX/Modern]
 class=Compile
 command=pdftex
-options=-src -interaction=nonstopmode '%source'
+options=-src-specials -interaction=nonstopmode '%source'
 from=
 to=pdf
 type=Process
@@ -461,120 +557,90 @@
 from=html
 to=html

-[Tool/ViewDVI/Embedded Viewer]
-class=View
-type=Part
-className=KViewPart
-libName=kviewerpart
-from=dvi
-to=dvi
-state=Viewer
-libOptions=dvi
-
-[Tool/ViewDVI/KDVI]
-class=View
-type=Process
-command=kdvi
-options='%target'
-from=dvi
-to=dvi
-
-[Tool/ViewDVI/KDVI Unique]
-class=View
-type=Process
-command=kdvi
-options=--unique '%target'
-from=dvi
-to=dvi
-
 [Tool/ForwardDVI/Embedded Viewer]
 type=Part
 class=ForwardDVI
-className=KViewPart
-libName=kviewerpart
+className=Okular::Part
+libName=libokular
 from=dvi
 to=dvi
 state=Viewer
 libOptions=dvi

-[Tool/ForwardDVI/KDVI]
+[Tool/ForwardDVI/Okular]
 class=ForwardDVI
 type=Process
-command=kdvi
+command=okular
 options=--unique '%target'
 from=dvi
 to=dvi

-[Tool/ViewPDF/Embedded Viewer]
+[Tool/ViewDVI/Embedded Viewer]
 class=View
-className=KGVPart
-libName=libkghostviewpart
+className=Okular::Part
+libName=libokular
 state=Viewer
-from=pdf
-to=pdf
 type=Part

-[Tool/ViewPDF/KGhostView]
+[Tool/ViewDVI/Okular]
 class=View
-command=kghostview
+command=okular
 options='%target'
+from=dvi
+to=dvi
 state=Viewer
-from=pdf
-to=pdf
 type=Process

-[Tool/ViewPDF/KPDF]
+[Tool/ViewPS/GV]
 class=View
-command=kpdf
+command=gv
 options='%target'
 state=Viewer
-from=pdf
-to=pdf
+from=ps
+to=ps
 type=Process

-[Tool/ViewPDF/Embedded Viewer]
+[Tool/ViewPS/Okular]
 class=View
-className=KPDF::Part
-libName=libkpdfpart
-state=Viewer
-from=pdf
-to=pdf
-type=Part
-
-[Tool/ViewPDF/Acroread]
-class=View
-command=acroread
+command=okular
 options='%target'
+from=ps
+to=ps
 state=Viewer
-from=pdf
-to=pdf
 type=Process

 [Tool/ViewPS/Embedded Viewer]
 class=View
-className=KGVPart
-libName=libkghostviewpart
+className=Okular::Part
+libName=libokular
 state=Viewer
-from=ps
-to=ps
 type=Part

-[Tool/ViewPS/KGhostView]
+[Tool/ViewPDF/Okular]
 class=View
-command=kghostview
+command=okular
 options='%target'
+from=pdf
+to=pdf
 state=Viewer
-from=ps
-to=ps
 type=Process

-[Tool/ViewPS/GV]
+[Tool/ViewPDF/Embedded Viewer]
 class=View
-command=gv
+className=Okular::Part
+libName=libokular
+from=pdf
+to=pdf
+state=Viewer
+type=Part
+
+[Tool/ViewPDF/Acroread]
+class=View
+command=acroread
 options='%target'
 state=Viewer
-from=ps
-to=ps
+from=pdf
+to=pdf
 type=Process

 [Tool/Lilypond/PDF]
@@ -628,49 +694,52 @@
 DVItoPS=Default
 LaTeX=Default
 LaTeX to Web=latex2html
-MakeIndex=Default
+MakeIndex=makeindex
 MetaPost=Default
 PDFLaTeX=Default
 PreviewLaTeX=Default
 PreviewPDFLaTeX=Default
 PStoPDF=Default
-QuickBuild=LaTeX+ViewDVI
+QuickBuild=PDFLaTeX+ViewPDF
 TeX=Default
 PDFTeX=Default
 ViewBib=KBibTex
+ViewPDF=Okular
+ViewPS=Okular
+ViewDVI=Okular
 ViewHTML=Konqueror
-ViewDVI=KDVI
-ForwardDVI=KDVI
-ViewPDF=KPDF
-ViewPS=KGhostView
+ForwardDVI=Okular
 Lilypond=PDF

 [ToolsGUI]
 Lilypond=Compile,lilypond
 Archive=Archive,package
+Asymptote=Compile,asymptote
 BibTeX=Compile,bibtex
 Convert=Convert,convert
 DVItoPDF=Convert,dvipdf
 DVItoPNG=Convert,dvipng
 DVItoPS=Convert,dvips
-LaTeX=Compile,latex
 LaTeX to Web=Compile,l2h
 DBLaTeX=Compile,dblatex
 MakeIndex=Compile,makeidx
 MetaPost=Compile,mpost
-PDFLaTeX=Compile,pdflatex
 PreviewLaTeX=none,none
 PreviewPDFLaTeX=none,none
 PStoPDF=Convert,ps2pdf
 QuickBuild=Quick,quick
+LaTeX=Compile,latex
 TeX=Compile,texcompiler
 PDFTeX=Compile,pdftex
+PDFLaTeX=Compile,pdflatex
+ConTeXt=Compile,latex
+XeTeX=Compile,latex
 ViewBib=View,viewbib
+ViewDVI=View,okular
+ViewPDF=View,okular
+ViewPS=View,okular
 ViewHTML=View,viewhtml
-ViewDVI=View,viewdvi
 ForwardDVI=View,dvisearch
-ViewPDF=View,viewpdf
-ViewPS=View,viewps

 [Shortcuts]
 tool_QuickBuild=Alt+1
--- src/kile/quickpreview.cpp #809350:809351
@@ -38,12 +38,11 @@
 QuickPreview::QuickPreview(KileInfo *ki) : m_ki(ki), m_running(0), m_tempDir(NULL)
 {
 	m_taskList << i18n("LaTeX ---> DVI")
-	           << i18n("LaTeX ---> DVI (KDVI)")
+	           << i18n("LaTeX ---> DVI (Okular)")
 	           << i18n("LaTeX ---> PS")
-	           << i18n("LaTeX ---> PS (KGhostView)")
+	           << i18n("LaTeX ---> PS (Okular)")
 	           << i18n("PDFLaTeX ---> PDF")
-	           << i18n("PDFLaTeX ---> PDF (KGhostView)")
-	           << i18n("PDFLaTeX ---> PDF (KPDF)")
+	           << i18n("PDFLaTeX ---> PDF (Okular)")
 	           ;
 }

@@ -147,12 +146,11 @@
 {
 	tasklist.clear();
 	tasklist << "Tool/ViewDVI/Embedded Viewer=" + m_taskList[0]
-	         << "Tool/ViewDVI/KDVI Unique="     + m_taskList[1]
+	         << "Tool/ViewDVI/Okular="     + m_taskList[1]
 	         << "Tool/ViewPS/Embedded Viewer="  + m_taskList[2]
-	         << "Tool/ViewPS/KGhostView="       + m_taskList[3]
+	         << "Tool/ViewPS/Okular="       + m_taskList[3]
 	         << "Tool/ViewPDF/Embedded Viewer=" + m_taskList[4]
-	         << "Tool/ViewPDF/KGhostView="      + m_taskList[5]
-	         << "Tool/ViewPDF/KPDF="            + m_taskList[6]
+	         << "Tool/ViewPDF/Okular="      + m_taskList[5]
 	         ;
 }

@@ -166,12 +164,11 @@
 	// define possible tools
 	QMap <QString,QString> map;
 	map[m_taskList[0]] = "PreviewLaTeX,,,ViewDVI,Embedded Viewer,dvi"; 
-	map[m_taskList[1]] = "PreviewLaTeX,,,ViewDVI,KDVI Unique,dvi";
+	map[m_taskList[1]] = "PreviewLaTeX,,,ViewDVI,Okular,dvi"; 
 	map[m_taskList[2]] = "PreviewLaTeX,DVItoPS,Default,ViewPS,Embedded Viewer,ps";
-	map[m_taskList[3]] = "PreviewLaTeX,DVItoPS,Default,ViewPS,KGhostView,ps";
-	map[m_taskList[4]] = "PreviewPDFLaTeX,,,ViewPDF,KPDF (embedded),pdf"; 
-	map[m_taskList[5]] = "PreviewPDFLaTeX,,,ViewPDF,KGhostView,pdf";
-	map[m_taskList[6]] = "PreviewPDFLaTeX,,,ViewPDF,KPDF,pdf";
+	map[m_taskList[3]] = "PreviewLaTeX,DVItoPS,Default,ViewPS,Okular,ps";
+	map[m_taskList[4]] = "PreviewPDFLaTeX,,,ViewPDF,Embedded Viewer,pdf"; 
+	map[m_taskList[5]] = "PreviewPDFLaTeX,,,ViewPDF,Okular,pdf";

 	QString previewtask = KileConfig::previewTask();
 	if(!map.contains(previewtask)) {
--- src/kile/test/runTests.sh #809350:809351
@@ -128,8 +128,8 @@
 echo "starting test: DVItoPDF"
 setTool DVItoPDF
 setKey mustpass ""
-setKey executable dvipdfm
-setKey where `which dvipdfm`
+setKey executable dvipdfmx
+setKey where `which dvipdfmx`
 if [ -r $testFileBase.dvi ]; then performKileTest kile "run DVItoPDF"; fi

 echo "starting test: PStoPDF"
@@ -145,6 +145,7 @@
 setKey mustpass ""
 setKey executable bibtex
 setKey where `which bibtex`
+
 if [ -r "test.dvi" ] #LaTeX is working
 then
 	testFileBase=test_bib
@@ -161,6 +162,7 @@
 setKey mustpass ""
 setKey executable makeindex
 setKey where `which makeindex`
+
 if [ -r "test.dvi" ] #LaTeX is working
 then
 	testFileBase=test_index
@@ -172,24 +174,12 @@
 	$closeDoc
 fi

-echo "starting test: KDVI"
-setTool KDVI
+echo "starting test: Okular"
+setTool Okular
 setKey mustpass "where"
-setKey executable kdvi
-setKey where `which kdvi`
+setKey executable okular
+setKey where `which okular`

-echo "starting test: KGhostView"
-setTool KGhostView
-setKey mustpass ""
-setKey executable kghostview
-setKey where `which kghostview`
-
-echo "starting test: KPDF"
-setTool KPDF
-setKey mustpass ""
-setKey executable kpdf
-setKey where `which kpdf`
-
 echo "starting test: Acroread"
 setTool Acroread
 setKey mustpass ""


