--- mtaskbar-0.7/mtaskbar/src/mtaskbar.desktop	2005-03-30 22:38:07.000000000 +0000
+++ mtaskbar-0.7-orig/mtaskbar/src/mtaskbar.desktop	2005-03-30 22:40:14.000000000 +0000
@@ -1,13 +1,14 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Taskbar v2
+Name[pl]=Taskbar v2
 Icon=taskbarapplet
 
 X-KDE-Library=libmtaskbar
 X-KDE-UniqueApplet=true 
+Type=Plugin
 
-Name[xx]=xxTaskbar v2xx
-
 Comment=An extended taskbar panel applet.
-omment[xx]=xxAn extended taskbar panel applet.xx
+Comment[pl]=Rozbudowany aplet panelu systemowego
+Categories=KDE;System;
