diff -urN ./mtaskbar-0.7.orig/mtaskbar/src/mtaskcontainer.h ./mtaskbar-0.7/mtaskbar/src/mtaskcontainer.h
--- ./mtaskbar-0.7.orig/mtaskbar/src/mtaskcontainer.h	2004-11-08 14:14:16.000000000 +0100
+++ ./mtaskbar-0.7/mtaskbar/src/mtaskcontainer.h	2006-05-30 23:26:29.542823000 +0200
@@ -69,7 +69,7 @@
 	virtual bool event(QEvent *e);
 	MTaskBar * m_taskbar;
 	virtual void popupMenu( TaskBar::Action action );
-	virtual void MTaskContainer::drawShadowText(QPainter  &p, QRect tr, int tf, const QString & str, int len = -1, QRect * brect = 0, QTextParag ** internal = 0 );
+	virtual void drawShadowText(QPainter  &p, QRect tr, int tf, const QString & str, int len = -1, QRect * brect = 0, QTextParag ** internal = 0 );
 
 private:
 	bool m_aboutToCloseAnim; // is true when we are closing the anim widget. During thatt operation we don't do any paint operations since it would flicker like nuts (workaroun for a maybe internal Qt bug)
