diff -urN a/type_compiler.cpp b/type_compiler.cpp
--- a/type_compiler.cpp	2011-05-20 14:33:31.000000000 -0600
+++ b/type_compiler.cpp	2011-08-14 16:39:02.000000000 -0600
@@ -191,6 +191,8 @@
 
 void TypeCompiler::visitPtrOperator(PtrOperatorAST* node)
 {
+    if ( ! m_session->token_stream ) return;
+    if ( ! token_text(m_session->token_stream->kind(node->op)) ) return;
     if (token_text(m_session->token_stream->kind(node->op))[0] == '*') {
         QPair<bool, bool> cv = m_visitor->parseCv(node->cv);
         pointerDepth.append(cv.first);
