--- setup.cfg	2006-11-15 20:26:26.000000000 +0300
+++ setup.cfg.my	2007-06-25 16:19:24.000000000 +0400
@@ -7,14 +7,14 @@
 # for wrapping OpenLDAP 2 libs
 [_ldap]
 
-library_dirs = /usr/local/openldap-2.3/lib
-include_dirs = /usr/local/openldap-2.3/include /usr/include/sasl
+library_dirs = /usr/lib
+include_dirs = /usr/include /usr/include/sasl
 
 extra_compile_args = 
 extra_objects = 
 
 # Example for sparse build: basic functionality
-#libs = ldap lber
+libs = ldap lber
 
 # Some Linux systems might need explicitly linking with -lresolv
 #libs = ldap lber resolv
@@ -31,10 +31,10 @@
 optimize = 1
 
 # Linux distributors/packagers should adjust these settings
-[bdist_rpm]
-provides = python-ldap
-requires = python libldap.so.2
+#[bdist_rpm]
+#provides = python-ldap
+#requires = python libldap.so.2
 # distribution_name = 
-release = 0
-packager = Michael Stroeder <michael@stroeder.com>
-doc_files = CHANGES README INSTALL TODO Demo/
+#release = 0
+#packager = Michael Stroeder <michael@stroeder.com>
+#doc_files = CHANGES README INSTALL TODO Demo/
