diff -wbBur pwmanager-1.2.4/pwmanager/base64.cpp pwmanager-1.2.4.my/pwmanager/base64.cpp
--- pwmanager-1.2.4/pwmanager/base64.cpp	2005-11-06 01:16:51.000000000 +0000
+++ pwmanager-1.2.4.my/pwmanager/base64.cpp	2009-01-28 16:02:43.000000000 +0000
@@ -41,6 +41,7 @@
  *                                                                         *
  ***************************************************************************/
 
+#include <stdlib.h>
 #include "base64.h"
 #include "pwmexception.h"
 
diff -wbBur pwmanager-1.2.4/pwmanager/binentrygen.cpp pwmanager-1.2.4.my/pwmanager/binentrygen.cpp
--- pwmanager-1.2.4/pwmanager/binentrygen.cpp	2005-11-06 01:16:51.000000000 +0000
+++ pwmanager-1.2.4.my/pwmanager/binentrygen.cpp	2009-01-28 16:03:08.000000000 +0000
@@ -8,7 +8,7 @@
  *   as published by the Free Software Foundation.                         *
  *                                                                         *
  ***************************************************************************/
-
+#include <stdlib.h>
 #include "binentrygen.h"
 #include "base64.h"
 #include "pwmexception.h"
diff -wbBur pwmanager-1.2.4/pwmanager/compressbzip2.cpp pwmanager-1.2.4.my/pwmanager/compressbzip2.cpp
--- pwmanager-1.2.4/pwmanager/compressbzip2.cpp	2005-11-06 01:16:50.000000000 +0000
+++ pwmanager-1.2.4.my/pwmanager/compressbzip2.cpp	2009-01-28 16:03:47.000000000 +0000
@@ -10,7 +10,7 @@
  ***************************************************************************/
 
 #include "compressbzip2.h"
-
+#include <string.h>
 #include <stdlib.h>
 
 #define BZ_NO_STDIO
diff -wbBur pwmanager-1.2.4/pwmanager/main.cpp pwmanager-1.2.4.my/pwmanager/main.cpp
--- pwmanager-1.2.4/pwmanager/main.cpp	2005-11-06 01:16:51.000000000 +0000
+++ pwmanager-1.2.4.my/pwmanager/main.cpp	2009-01-28 16:05:16.000000000 +0000
@@ -9,6 +9,7 @@
  *                                                                         *
  ***************************************************************************/
 
+#include <stdlib.h>
 #include <kcmdlineargs.h>
 #include <kaboutdata.h>
 #include <klocale.h>
diff -wbBur pwmanager-1.2.4/pwmanager/serializer.cpp pwmanager-1.2.4.my/pwmanager/serializer.cpp
--- pwmanager-1.2.4/pwmanager/serializer.cpp	2005-11-06 01:16:50.000000000 +0000
+++ pwmanager-1.2.4.my/pwmanager/serializer.cpp	2009-01-28 16:06:32.000000000 +0000
@@ -9,6 +9,7 @@
  *                                                                         *
  ***************************************************************************/
 
+#include <stdlib.h>
 #include "serializer.h"
 #include "configuration.h"
 #include "pwmexception.h"
