diff -urN a/include/configs/omap3_beagle.h b/include/configs/omap3_beagle.h
--- a/include/configs/omap3_beagle.h	2014-05-17 14:05:55.202683461 -0600
+++ b/include/configs/omap3_beagle.h	2014-05-17 14:06:11.697622480 -0600
@@ -192,6 +192,8 @@
 /* Environment information */
 #define CONFIG_BOOTDELAY		1
 
+#define CONFIG_IDENT_STRING " Arch Linux ARM"
+
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"loadaddr=0x80300000\0" \
 	"rdaddr=0x81000000\0" \
@@ -200,9 +202,9 @@
 	"usbtty=cdc_acm\0" \
 	"bootfile=zImage\0" \
 	"ramdisk=ramdisk.gz\0" \
-	"bootdir=\0" \
-	"bootpart=0:1\0" \
-	"fdtdir=/dtbs\0" \
+	"bootdir=/boot\0" \
+	"bootpart=0:2\0" \
+	"fdtdir=/boot/dtbs\0" \
 	"console=ttyO2,115200n8\0" \
 	"mpurate=auto\0" \
 	"buddy=none\0" \
@@ -213,8 +215,7 @@
 	"dvimode=640x480MR-16@60\0" \
 	"defaultdisplay=dvi\0" \
 	"mmcdev=0\0" \
-	"mmcroot=/dev/mmcblk0p2 ro\0" \
-	"mmcrootfstype=ext4 rootwait fixrtc\0" \
+	"mmcroot=/dev/mmcblk0p2 rw rootwait fixrtc\0" \
 	"nandroot=ubi0:rootfs ubi.mtd=4\0" \
 	"nandrootfstype=ubifs\0" \
 	"ramroot=/dev/ram0 rw ramdisk_size=65536 initrd=0x81000000,64M\0" \
@@ -228,7 +229,6 @@
 		"omapfb.mode=dvi:${dvimode} " \
 		"omapdss.def_disp=${defaultdisplay} " \
 		"root=${mmcroot} " \
-		"rootfstype=${mmcrootfstype} " \
 		"${musb}\0" \
 	"nandargs=setenv bootargs console=${console} " \
 		"${optargs} " \
@@ -257,7 +257,7 @@
 		"if test $beaglerev = xMAB; then " \
 			"fdt addr ${fdtaddr}; fdt resize; fdt set /hsusb2_power_reg enable-active-high; fi; \0" \
 	"bootenv=uEnv.txt\0" \
-	"loadbootenv=load mmc ${bootpart} ${loadaddr} ${bootenv}\0" \
+	"loadbootenv=load mmc ${bootpart} ${loadaddr} ${bootdir}/${bootenv}\0" \
 	"importbootenv=echo Importing environment from mmc ...; " \
 		"env import -t $loadaddr $filesize\0" \
 	"ramargs=setenv bootargs console=${console} " \
