Index: lastfm-1.5.1.31879.dfsg/src/Makefile
===================================================================
--- lastfm-1.5.1.31879.dfsg.orig/src/Makefile
+++ lastfm-1.5.1.31879.dfsg/src/Makefile
@@ -959,6 +959,7 @@ compiler_uic_clean:
 ../build/ui_playcontrols.h: playcontrols.ui \
 		volumeslider.h
 	/usr/bin/uic playcontrols.ui -o ../build/ui_playcontrols.h
+	sed -i "s/phonon\/Phonon::VolumeSlider/volumeslider.h/" ../build/ui_playcontrols.h
 
 ../build/ui_failedlogindialog.h: failedlogindialog.ui
 	/usr/bin/uic failedlogindialog.ui -o ../build/ui_failedlogindialog.h
