diff -wbBur glipper-0.95.1/configure glipper-0.95.1.my/configure
--- glipper-0.95.1/configure	2006-11-21 03:42:45.000000000 +0300
+++ glipper-0.95.1.my/configure	2007-08-09 13:27:44.000000000 +0400
@@ -557,10 +557,10 @@
 SHELL=${CONFIG_SHELL-/bin/sh}
 
 # Identity of this package.
-PACKAGE_NAME='glipper'
-PACKAGE_TARNAME='glipper'
+PACKAGE_NAME='glipper-old'
+PACKAGE_TARNAME='glipper-old'
 PACKAGE_VERSION='0.95.1'
-PACKAGE_STRING='glipper 0.95.1'
+PACKAGE_STRING='glipper-old 0.95.1'
 PACKAGE_BUGREPORT='svenrech@gmx.de'
 
 ac_unique_file="src/main.c"
@@ -2134,7 +2134,7 @@
 
 
 # Define the identity of the package.
- PACKAGE='glipper'
+ PACKAGE='glipper-old'
  VERSION='0.95.1'
 
 
@@ -2301,7 +2301,7 @@
 # GETTEXT
 ###########################################################################
 
-GETTEXT_PACKAGE="glipper"
+GETTEXT_PACKAGE="glipper-old"
 
 
 cat >>confdefs.h <<_ACEOF
diff -wbBur glipper-0.95.1/configure.in glipper-0.95.1.my/configure.in
--- glipper-0.95.1/configure.in	2006-11-21 03:59:24.000000000 +0300
+++ glipper-0.95.1.my/configure.in	2007-08-09 13:35:04.000000000 +0400
@@ -5,7 +5,7 @@
 AC_PREREQ(2.5)
 
 # Set VERSION number here
-AC_INIT(glipper, [0.95.1], [svenrech@gmx.de])
+AC_INIT(glipper-old, [0.95.1], [svenrech@gmx.de])
 AC_CONFIG_SRCDIR([src/main.c])
 AC_CONFIG_HEADER([config.h])
 AM_INIT_AUTOMAKE
@@ -29,7 +29,7 @@
 # GETTEXT
 ###########################################################################
 
-GETTEXT_PACKAGE="glipper"
+GETTEXT_PACKAGE="glipper-old"
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [gettext package])
 
diff -wbBur glipper-0.95.1/data/glipper.desktop.in glipper-0.95.1.my/data/glipper.desktop.in
--- glipper-0.95.1/data/glipper.desktop.in	2006-06-26 23:46:04.000000000 +0400
+++ glipper-0.95.1.my/data/glipper.desktop.in	2007-08-09 13:04:01.000000000 +0400
@@ -3,9 +3,9 @@
 _Name=Glipper
 _Comment=Clipboard Manager
 Categories=GTK;GNOME;Application;Utility;
-TryExec=glipper
-Exec=glipper
-Icon=glipper.png
+TryExec=glipper-old
+Exec=glipper-old
+Icon=glipper-old.png
 Terminal=false
 Type=Application
 StartupNotify=false
diff -wbBur glipper-0.95.1/data/Makefile.am glipper-0.95.1.my/data/Makefile.am
--- glipper-0.95.1/data/Makefile.am	2006-06-26 23:46:08.000000000 +0400
+++ glipper-0.95.1.my/data/Makefile.am	2007-08-09 13:33:06.000000000 +0400
@@ -1,13 +1,13 @@
 desktopdir = $(datadir)/applications
-desktop_in_files = glipper.desktop.in
+desktop_in_files = glipper-old.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 @INTLTOOL_DESKTOP_RULE@
 
 icondir = $(datadir)/pixmaps
-icon_DATA = glipper.png
+icon_DATA = glipper-old.png
 
-gladedir = $(datadir)/glipper
-glade_DATA = glipper-properties.glade
+gladedir = $(datadir)/glipper-old
+glade_DATA = glipper-old-properties.glade
 
 EXTRA_DIST =            \
 	$(desktop_in_files) \
diff -wbBur glipper-0.95.1/data/Makefile.in glipper-0.95.1.my/data/Makefile.in
--- glipper-0.95.1/data/Makefile.in	2006-11-21 03:42:44.000000000 +0300
+++ glipper-0.95.1.my/data/Makefile.in	2007-08-09 13:16:54.000000000 +0400
@@ -214,12 +214,12 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 desktopdir = $(datadir)/applications
-desktop_in_files = glipper.desktop.in
+desktop_in_files = glipper-old.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 icondir = $(datadir)/pixmaps
-icon_DATA = glipper.png
-gladedir = $(datadir)/glipper
-glade_DATA = glipper-properties.glade
+icon_DATA = glipper-old.png
+gladedir = $(datadir)/glipper-old
+glade_DATA = glipper-old-properties.glade
 EXTRA_DIST = \
 	$(desktop_in_files) \
 	$(icon_DATA)        \
diff -wbBur glipper-0.95.1/help/Makefile.am glipper-0.95.1.my/help/Makefile.am
--- glipper-0.95.1/help/Makefile.am	2006-11-15 00:00:59.000000000 +0300
+++ glipper-0.95.1.my/help/Makefile.am	2007-08-09 13:33:10.000000000 +0400
@@ -1,7 +1,7 @@
 include $(top_srcdir)/gnome-doc-utils.make
 dist-hook: doc-dist-hook
 
-DOC_MODULE = glipper
+DOC_MODULE = glipper-old
 DOC_ENTITIES = 
 DOC_INCLUDES = 
 DOC_FIGURES = 
diff -wbBur glipper-0.95.1/help/Makefile.in glipper-0.95.1.my/help/Makefile.in
--- glipper-0.95.1/help/Makefile.in	2006-11-21 03:42:44.000000000 +0300
+++ glipper-0.95.1.my/help/Makefile.in	2007-08-09 13:04:40.000000000 +0400
@@ -374,7 +374,7 @@
 _clean_dsk = $(if $(_DOC_DSK_IN),clean-doc-dsk)
 _clean_lc = $(if $(_DOC_REAL_LINGUAS),clean-doc-lc)
 _clean_dir = $(if $(DOC_MODULE),clean-doc-dir)
-DOC_MODULE = glipper
+DOC_MODULE = glipper-old
 DOC_ENTITIES = 
 DOC_INCLUDES = 
 DOC_FIGURES = 
diff -wbBur glipper-0.95.1/po/POTFILES.in glipper-0.95.1.my/po/POTFILES.in
--- glipper-0.95.1/po/POTFILES.in	2006-06-26 23:46:14.000000000 +0400
+++ glipper-0.95.1.my/po/POTFILES.in	2007-08-09 13:17:22.000000000 +0400
@@ -1,4 +1,4 @@
 src/main.c
 src/preferences.c
-data/glipper-properties.glade
-data/glipper.desktop.in
+data/glipper-old-properties.glade
+data/glipper-old.desktop.in
diff -wbBur glipper-0.95.1/src/main.c glipper-0.95.1.my/src/main.c
--- glipper-0.95.1/src/main.c	2006-11-21 03:05:53.000000000 +0300
+++ glipper-0.95.1.my/src/main.c	2007-08-09 13:46:53.000000000 +0400
@@ -141,7 +141,7 @@
 		GError* pix_error = NULL;
 
 		menuHeader = gtk_image_menu_item_new_with_label("");
-		pixbuf = gdk_pixbuf_new_from_file(PIXMAPDIR"/glipper.png", &pix_error);
+		pixbuf = gdk_pixbuf_new_from_file(PIXMAPDIR"/glipper-old.png", &pix_error);
 
 		//In case we cannot load icon, display error message and exit
 		if (pix_error != NULL)
@@ -312,7 +312,7 @@
 
 	error = NULL;
 	//gnome_help_display ("glipper", NULL, &error);
-        gnome_help_display_desktop(NULL, "glipper", "glipper", 
+        gnome_help_display_desktop(NULL, "glipper-old", "glipper-old", 
                                    section, &error);
 
 	if (error)
@@ -337,7 +337,7 @@
 
 	//Load trayicon
 	TrayIcon = (GtkWidget*)egg_tray_icon_new("GLIPPER");
-	pixbuf = gdk_pixbuf_new_from_file(PIXMAPDIR"/glipper.png", &pix_error);
+	pixbuf = gdk_pixbuf_new_from_file(PIXMAPDIR"/glipper-old.png", &pix_error);
 
 	//In case we cannot load icon, display error message and exit
 	if (pix_error != NULL)
@@ -392,7 +392,7 @@
 		"license", license,
 		"name", "Glipper",
 		"comments", _("Clipboardmanager for Gnome"),
-		"logo", gdk_pixbuf_new_from_file(PIXMAPDIR"/glipper.png", &pix_error),
+		"logo", gdk_pixbuf_new_from_file(PIXMAPDIR"/glipper-old.png", &pix_error),
 		"website", "http://glipper.sourceforge.net/",
 		"version", VERSION,
 		 NULL);
@@ -442,7 +442,7 @@
 //trys to open (or create) a file in "~/.glipper" for writing purposes:
 FILE* writeGlipperFile(char* filename)
 {
-	gchar* directory = g_build_path("/", g_get_home_dir(), ".glipper", NULL);
+	gchar* directory = g_build_path("/", g_get_home_dir(), ".glipper-old", NULL);
 	gchar* path = g_build_filename(directory, filename, NULL);
 	FILE* file = fopen(path, "w");
 	if (file == NULL)
@@ -453,7 +453,7 @@
 				g_warning("Can't open or create file %s!", file);
 		}
 		else
-			g_warning ("Can't create directory '.glipper' in user's home directory!");
+			g_warning ("Can't create directory '.glipper-old' in user's home directory!");
 	g_free(directory);
 	g_free(path);
 	return file;
@@ -480,7 +480,7 @@
 
 void readHistory()
 {
-	gchar* path= g_build_filename(g_get_home_dir(), ".glipper/history", NULL);
+	gchar* path= g_build_filename(g_get_home_dir(), ".glipper-old/history", NULL);
 	FILE* histFile = fopen(path, "r");
 	g_free(path);
 	if (histFile != 0)
@@ -501,7 +501,7 @@
 
 void readPreferences()
 {
-	gchar* path= g_build_filename(g_get_home_dir(), ".glipper/prefs", NULL);
+	gchar* path= g_build_filename(g_get_home_dir(), ".glipper-old/prefs", NULL);
 	FILE* prefFile = fopen(path, "r");
 	g_free(path);
 	free(keyComb);
@@ -599,7 +599,7 @@
 
 	//Init GTK+ (and optionally GNOME libs)
 #ifndef DISABLE_GNOME
-	gnome_program_init("glipper", VERSION, NULL, argc, argv,
+	gnome_program_init("glipper-old", VERSION, NULL, argc, argv,
 		NULL, NULL, NULL);
 #endif /*DISABLE_GNOME*/
 	gtk_init (&argc, &argv);
diff -wbBur glipper-0.95.1/src/Makefile.am glipper-0.95.1.my/src/Makefile.am
--- glipper-0.95.1/src/Makefile.am	2006-11-14 23:28:11.000000000 +0300
+++ glipper-0.95.1.my/src/Makefile.am	2007-08-09 13:32:44.000000000 +0400
@@ -1,6 +1,6 @@
 AM_CFLAGS = $(FR_CFLAGS) -I$(top_srcdir) \
 	-DGLIPPERLOCALEDIR=\""$(glipperlocaledir)"\" -DPIXMAPDIR=\""$(datadir)/pixmaps"\" \
-	-DGLADEDIR=\""$(datadir)/glipper"\" 
+	-DGLADEDIR=\""$(datadir)/glipper-old"\" 
 
 noinst_PROGRAMS=glipper
 bin_PROGRAMS=glipper
diff -wbBur glipper-0.95.1/src/Makefile.in glipper-0.95.1.my/src/Makefile.in
--- glipper-0.95.1/src/Makefile.in	2006-11-21 03:42:44.000000000 +0300
+++ glipper-0.95.1.my/src/Makefile.in	2007-08-09 13:03:48.000000000 +0400
@@ -223,7 +223,7 @@
 target_alias = @target_alias@
 AM_CFLAGS = $(FR_CFLAGS) -I$(top_srcdir) \
 	-DGLIPPERLOCALEDIR=\""$(glipperlocaledir)"\" -DPIXMAPDIR=\""$(datadir)/pixmaps"\" \
-	-DGLADEDIR=\""$(datadir)/glipper"\" 
+	-DGLADEDIR=\""$(datadir)/glipper-old"\" 
 
 glipper_SOURCES = \
 	main.c main.h preferences.c preferences.h \
diff -wbBur glipper-0.95.1/src/preferences.c glipper-0.95.1.my/src/preferences.c
--- glipper-0.95.1/src/preferences.c	2006-07-15 17:27:49.000000000 +0400
+++ glipper-0.95.1.my/src/preferences.c	2007-08-09 13:17:31.000000000 +0400
@@ -8,7 +8,7 @@
 #include "main.h"
 #include "utils/glipper-i18n.h"
 
-#define GLADE_XML_FILE "glipper-properties.glade"
+#define GLADE_XML_FILE "glipper-old-properties.glade"
 
 GtkWidget* historyLength;
 GtkWidget* itemLength;
