From da6193b1fcc1dfab27f4c36917864f2f2c41cf3e Mon Sep 17 00:00:00 2001
From: Matt Turner <mattst88@gmail.com>
Date: Sun, 27 May 2012 17:01:57 +0000
Subject: mmx: add missing _mm_empty calls

Fixes spurious test failures on x86-32.
---
diff --git a/pixman/pixman-mmx.c b/pixman/pixman-mmx.c
index bb125bf..6e292c6 100644
--- a/pixman/pixman-mmx.c
+++ b/pixman/pixman-mmx.c
@@ -2232,6 +2232,8 @@ mmx_composite_src_x888_0565 (pixman_implementation_t *imp,
	    w--;
	}
     }
+
+    _mm_empty ();
 }

 static void
@@ -3542,6 +3544,7 @@ mmx_fetch_x8r8g8b8 (pixman_iter_t *iter, const uint32_t *mask)
	w--;
     }

+    _mm_empty ();
     return iter->buffer;
 }

@@ -3585,6 +3588,7 @@ mmx_fetch_r5g6b5 (pixman_iter_t *iter, const uint32_t *mask)
	w--;
     }

+    _mm_empty ();
     return iter->buffer;
 }

@@ -3630,6 +3634,7 @@ mmx_fetch_a8 (pixman_iter_t *iter, const uint32_t *mask)
	w--;
     }

+    _mm_empty ();
     return iter->buffer;
 }

--
cgit v0.9.0.2-2-gbebe
