diff -urN js-1.8.5.orig/js/src/configure js-1.8.5/js/src/configure
--- js-1.8.5.orig/js/src/configure	2011-03-31 13:08:36.000000000 -0600
+++ js-1.8.5/js/src/configure	2011-11-29 18:13:45.000000000 -0700
@@ -13322,13 +13322,9 @@
   case "$target_cpu" in
     arm*)
       if test "$GNU_CC"; then
-        cat >> confdefs.h <<\EOF
-#define MOZ_ARM_ARCH 1
-EOF
-
-        CFLAGS="$CFLAGS -march=armv7-a -marm -mfloat-abi=softfp $MOZ_ARM_VFP_FLAGS"
-        CXXFLAGS="$CXXFLAGS -march=armv7-a -marm -mfloat-abi=softfp $MOZ_ARM_VFP_FLAGS"
-        ASFLAGS="$ASFLAGS -march=armv7-a -marm -mfloat-abi=softfp $MOZ_ARM_VFP_FLAGS"
+        CFLAGS="$CFLAGS"
+        CXXFLAGS="$CXXFLAGS"
+        ASFLAGS="$ASFLAGS"
       else
         { echo "configure: error: --with-cpu-arch=armv7 is not supported for non-GNU toolchains" 1>&2; exit 1; }
       fi
@@ -13341,9 +13337,9 @@
   case "$target_cpu" in
     arm*)
       if test "$GNU_CC"; then
-        CFLAGS="$CFLAGS -march=armv5te -mthumb-interwork -msoft-float"
-        CXXFLAGS="$CXXFLAGS -march=armv5te -mthumb-interwork -msoft-float"
-        ASFLAGS="$ASFLAGS -march=armv5te -mthumb-interwork -msoft-float"
+        CFLAGS="$CFLAGS"
+        CXXFLAGS="$CXXFLAGS"
+        ASFLAGS="$ASFLAGS"
       fi
       ;;
   esac
