diff -Naur xzgv-0.9.1-old/Makefile xzgv-0.9.1/Makefile
--- xzgv-0.9.1-old/Makefile	2009-05-01 07:58:36.000000000 +1000
+++ xzgv-0.9.1/Makefile	2009-08-06 23:58:04.000000000 +1000
@@ -23,7 +23,7 @@
 
 
 
-all: src man
+all: src man info
 
 src: xzgv
 
diff -Naur xzgv-0.9.1-old/doc/Makefile xzgv-0.9.1/doc/Makefile
--- xzgv-0.9.1-old/doc/Makefile	2007-12-24 06:49:03.000000000 +1000
+++ xzgv-0.9.1/doc/Makefile	2009-08-07 00:01:20.000000000 +1000
@@ -39,7 +39,6 @@
 install: installdirs
 	install xzgv.1 $(MANDIR)
 	install -m 644 xzgv.info.gz $(INFODIR)
-	install-info --infodir=$(INFODIR) xzgv.info.gz
 
 uninstall:
 	$(RM) $(MANDIR)/xzgv.1
