diff --unified --recursive --text --new-file colm-0.14.7.orig/test/colm.d/ext1.lm colm-0.14.7/test/colm.d/ext1.lm
--- colm-0.14.7.orig/test/colm.d/ext1.lm	2026-04-03 23:33:35.063223265 +0200
+++ colm-0.14.7/test/colm.d/ext1.lm	2026-04-03 23:33:57.910836560 +0200
@@ -24,7 +24,7 @@
 	head_t *h = string_alloc_full( prg, strc, strlen( strc ) );
 	tree_t *s = construct_string( prg, h );
 	colm_tree_upref( prg, s );
-	colm_tree_downref( prg, sp, a1 );
+	colm_tree_downref( prg, sp, (tree_t*)a1 );
 	return (str_t*)s;
 }
 ##### EXP #####
