diff -Nur hibernate-script-1.96.orig/common.conf hibernate-script-1.96/common.conf
--- hibernate-script-1.96.orig/common.conf	2007-07-16 14:32:41.000000000 +0200
+++ hibernate-script-1.96/common.conf	2007-08-26 17:44:47.000000000 +0200
@@ -14,7 +14,7 @@
 # AlwaysForce yes
 # AlwaysKill yes
 # HibernateVT 15
-# Distribution debian (or fedora/gentoo/mandrake/redhat/slackware/suse)
+Distribution arch
 # XDisplay :0
 
 ##############################################################################
diff -Nur hibernate-script-1.96.orig/scriptlets.d/hardware_tweaks hibernate-script-1.96/scriptlets.d/hardware_tweaks
--- hibernate-script-1.96.orig/scriptlets.d/hardware_tweaks	2007-07-16 14:33:09.000000000 +0200
+++ hibernate-script-1.96/scriptlets.d/hardware_tweaks	2007-08-26 17:44:47.000000000 +0200
@@ -86,7 +86,7 @@
     local action
     action=start
     [ "$DISTRIBUTION" = "gentoo" ] && action=restart
-    /etc/init.d/915resolution $action || return 1
+    /etc/rc.d/915resolution $action || return 1
     return 0
 }
 
diff -Nur hibernate-script-1.96.orig/scriptlets.d/network hibernate-script-1.96/scriptlets.d/network
--- hibernate-script-1.96.orig/scriptlets.d/network	2007-07-16 14:33:09.000000000 +0200
+++ hibernate-script-1.96/scriptlets.d/network	2007-08-26 17:44:47.000000000 +0200
@@ -90,6 +90,14 @@
 NetworkDetectDistro() {
     # Use either a given $DISTRIBUTION or autodetect one.
     case "$DISTRIBUTION" in
+	arch)
+	    network_ifup() {
+	    	/etc/rc.d/network ifup $1
+	    }
+	    network_ifdown() {
+	    	/etc/rc.d/network ifdown $1
+	    }
+	    ;;
 	gentoo)
 	    network_ifup() {
 	    	[ -x "/etc/init.d/net.$1" ] && /etc/init.d/net.$1 start
diff -Nur hibernate-script-1.96.orig/scriptlets.d/services hibernate-script-1.96/scriptlets.d/services
--- hibernate-script-1.96.orig/scriptlets.d/services	2007-07-16 14:33:09.000000000 +0200
+++ hibernate-script-1.96/scriptlets.d/services	2007-08-26 17:44:47.000000000 +0200
@@ -75,7 +75,7 @@
 	debian|ubuntu|redhat|fedora)
 	    INITDIR=/etc/init.d
 	    ;;
-	slackware)
+	slackware|arch)
 	    INITDIR=/etc/rc.d
 	    ;;
     	*)
