diff -U 3 -b -d -p -r -- a/glx/glxdri.c b/glx/glxdri.c
--- a/glx/glxdri.c	2013-01-24 22:14:35.216092949 +0100
+++ b/glx/glxdri.c	2013-01-24 22:13:48.499427991 +0100
@@ -971,6 +971,8 @@ __glXDRIscreenProbe(ScreenPtr pScreen)
     size_t buffer_size;
     ScrnInfoPtr pScrn = xf86ScreenToScrn(pScreen);
 
+    framebuffer.base = NULL;
+
     if (!xf86LoaderCheckSymbol("DRIQueryDirectRenderingCapable") ||
         !DRIQueryDirectRenderingCapable(pScreen, &isCapable) || !isCapable) {
         LogMessage(X_INFO,