From a38ca0063c91ef221f20bf0c4ac0dc84ce07d557 Mon Sep 17 00:00:00 2001
From: Benjamin Close <Benjamin.Close@clearchain.com>
Date: Fri, 13 Feb 2009 04:14:53 +0000
Subject: xext: Use proto header rather than the Xext include file, this prevents userspace being pulled in causing issues

Signed-off-by: Benjamin Close <Benjamin.Close@clearchain.com>
Acked-by: Peter Hutterer <peter.hutterer@who-t.net>
Acked-by: Daniel Stone <daniel@fooishbar.org>
---
diff --git a/Xext/panoramiX.h b/Xext/panoramiX.h
index 700c5a4..d413ff9 100644
--- a/Xext/panoramiX.h
+++ b/Xext/panoramiX.h
@@ -41,9 +41,8 @@ Equipment Corporation.
 #define _PANORAMIX_H_
 
 #define _PANORAMIX_SERVER
-#include <X11/extensions/panoramiXext.h>
+#include <X11/extensions/panoramiXproto.h>
 #undef _PANORAMIX_SERVER
-
 #include "gcstruct.h"
 
 
--
cgit v0.8.2
