--- lmctl-org/src/lmctl.c	2004-09-08 07:59:55.000000000 +0000
+++ lmctl-0.3.1/src/lmctl.c	2005-06-29 14:03:50.000000000 +0000
@@ -99,6 +99,8 @@
 	{0xc00e, "Wheel Mouse Optical",                    "M-BJ58",      0, 1, 0, 0},
 	{0xc00f, "MouseMan Traveler",                      "M-BJ79",      0, 1, 0, 0},
 	{0xc012, "MouseMan Dual Optical",                  "M-BL63B",     0, 1, 0, 0},
+	{0xc01d, "MX510 Optical Mouse",                    "M-BS81A",     0, 1, 1, 0},
+	{0xc01e, "MX518 Optical Mouse",                    "M-BS81A",     0, 1, 0, 0},
 	{0xc024, "MX300 Optical Mouse",                    "M-BP82",      0, 1, 0, 0},
 	{0xc025, "MX500 Optical Mouse",                    "M-BP81A",     0, 1, 1, 0},
 	{0xc031, "iFeel Mouse (silver)",                   "M-UT58A",     0, 1, 0, 0},
@@ -270,7 +272,7 @@
 			exit(1);
 		}
 		
-		assert ((buf [0] == 3) || (buf [0] == 4));
+		assert ((buf [0] == 3) || (buf [0] == 4) || (buf [0] == 5) || (buf [0] == 6));
 
 		printf ("\tResolution (RES): %d cpi\n", (buf [0] - 2) * 400);
 	}
@@ -355,7 +357,7 @@
 	unsigned char buf [80];
 	if (!set_res) return;
 
-	assert ((set_res == 400) || (set_res == 800));
+	assert ((set_res == 400) || (set_res == 800) || (set_res == 1200)  || (set_res == 1600));
 	if (usb_control_msg (handle, 0x40, 0x02, 0x000e, (set_res / 400) + 2,
 			     buf, 0, TIMEOUT) != 0) {
 		perror("Writing to USB device");
