--- lmctl-org/src/cmdline.c	2004-09-07 22:40:15.000000000 +0000
+++ lmctl-0.3.1/src/cmdline.c	2005-06-29 13:36:33.000000000 +0000
@@ -128,6 +128,10 @@
 	  "Set matching devices to a resolution of 400cpi"},
 	{ "800",         '8',     NULL,             0,
 	  "Set matching devices to a resolution of 800cpi"},
+	{ "1200",        'm',     NULL,             0,
+	  "Set matching devices to a medium resolution of 1200cpi"},
+	{ "1600",        'h',     NULL,             0,
+	  "Set matching devices to a high resolution of 1600cpi"},
 
 	{ NULL, 0, NULL, 0, _("SmartScroll/Cruise Control (SMS command set)") },
 
@@ -227,13 +231,22 @@
 		command = cmd_set;
 		set_res = (int) (key - '0') * 100;
 		break;
+	
+	case 'm':
+		command = cmd_set;
+		set_res = (int) (12) * 100;
+		break;
+	case 'h':
+		command = cmd_set;
+		set_res = (int) (16) *100;
+		break;
 
 	case '1':		/* set the channel */
 	case '2':
 		command = cmd_set;
 		set_channel = (int) (key - '0');
 		break;
-
+	
 	case 'u':
 		command = cmd_set;
 		set_unlock = 1;
