diff -upr xf86-video-dove-0.3.4-a/src/dovefb.c xf86-video-dove-0.3.4-b/src/dovefb.c
--- xf86-video-dove-0.3.4-a/src/dovefb.c	2013-03-19 12:07:16.840080915 +0000
+++ xf86-video-dove-0.3.4-b/src/dovefb.c	2013-03-19 12:08:57.699645908 +0000
@@ -17,7 +17,6 @@
 #include "xf86_OSproc.h"
 
 #include "mipointer.h"
-#include "mibstore.h"
 #include "micmap.h"
 #include "colormapst.h"
 #include "xf86cmap.h"
@@ -1507,7 +1506,6 @@ FBDevScreenInit(SCREEN_INIT_ARGS_DECL)
 	}
 
 	xf86SetBlackWhitePixels(pScreen);
-	miInitializeBackingStore(pScreen);
 	xf86SetBackingStore(pScreen);
 
 	/* software cursor */
diff -upr xf86-video-dove-0.3.4-a/src/dovefb_driver.h xf86-video-dove-0.3.4-b/src/dovefb_driver.h
--- xf86-video-dove-0.3.4-a/src/dovefb_driver.h	2013-03-19 12:07:16.850075916 +0000
+++ xf86-video-dove-0.3.4-b/src/dovefb_driver.h	2013-03-19 12:08:16.500247765 +0000
@@ -18,7 +18,6 @@
 
 #include "mipointer.h"
 #include "micmap.h"
-#include "mibstore.h"
 
 #include "colormapst.h"
 #include "xf86cmap.h"
diff -upr xf86-video-dove-0.3.4-a/src/dovefb_xv.c xf86-video-dove-0.3.4-b/src/dovefb_xv.c
--- xf86-video-dove-0.3.4-a/src/dovefb_xv.c	2011-03-07 17:52:09.000000000 +0000
+++ xf86-video-dove-0.3.4-b/src/dovefb_xv.c	2013-03-19 12:08:16.510242929 +0000
@@ -9,7 +9,6 @@
 #include "xf86.h"
 #include "xf86_OSproc.h"
 #include "mipointer.h"
-#include "mibstore.h"
 #include "micmap.h"
 #include "colormapst.h"
 #include "xf86cmap.h"
