diff -upr chromeos-3.4.orig/kernel/printk.c chromeos-3.4/kernel/printk.c
--- chromeos-3.4.orig/kernel/printk.c	2013-02-17 20:37:48.018383118 -0500
+++ chromeos-3.4/kernel/printk.c	2013-02-17 20:38:43.679214839 -0500
@@ -62,7 +62,7 @@ void asmlinkage __attribute__((weak)) ea
 
 /* We show everything that is MORE important than this.. */
 #define MINIMUM_CONSOLE_LOGLEVEL 1 /* Minimum loglevel we let people use */
-#define DEFAULT_CONSOLE_LOGLEVEL 7 /* anything MORE serious than KERN_DEBUG */
+#define DEFAULT_CONSOLE_LOGLEVEL 4 /* anything MORE serious than KERN_DEBUG */
 
 DECLARE_WAIT_QUEUE_HEAD(log_wait);
 
