--- icqt.py.org	2009-01-12 19:55:44.000000000 +0300
+++ icqt.py	2009-01-23 12:57:41.000000000 +0300
@@ -321,7 +321,10 @@
 			self.oscarcon.legacyList.delCustomStatus(user.name, savemask=mask)
 			
 		LogEvent(WARN, self.session.jabberID, "Status message: %s" % status)
+		try:
-		status = status.encode("utf-8", "replace")
+			status = status.encode("utf-8", "replace")
+                except UnicodeDecodeError:
+                        status = "unicode error"
 		# status = status.encode(config.encoding, "replace")
 		
 		if config.xstatusessupport:		
