--- PolicyKit-0.8.orig/polkitd/org.freedesktop.PolicyKit.conf.in	2008-12-08 10:55:12.000000000 -0500
+++ PolicyKit-0.8/polkitd/org.freedesktop.PolicyKit.conf.in	2008-12-08 12:05:33.000000000 -0500
@@ -8,4 +8,9 @@
   <policy user="@polkituser@">
     <allow own="org.freedesktop.PolicyKit"/>
   </policy>
+
+  <!-- any user can talk to the service (fd.o #18948) -->
+  <policy context="default">
+    <allow send_destination="org.freedesktop.PolicyKit"/>
+  </policy>
 </busconfig>
