--- Makefile.inc.orig	2005-04-28 02:21:35.000000000 +0200
+++ Makefile.inc	2008-10-22 20:26:28.000000000 +0200
@@ -28,8 +28,8 @@
 #it will try OSS if JACK fails. At runtime you can set the OSS by default by command-line
 #parameters (run 'zynaddsubfx --help' for help) 
 
-#LINUX_AUDIOOUT=OSS_AND_JACK
-LINUX_AUDIOOUT=OSS
+LINUX_AUDIOOUT=OSS_AND_JACK
+#LINUX_AUDIOOUT=OSS
 #LINUX_AUDIOOUT=NONE
 #LINUX_AUDIOOUT=JACK
 #LINUX_AUDIOOUT=JACK_RT     JACK_RT support is broken
