diff --git i/src/mruby-widget-lib/src/gem.c w/src/mruby-widget-lib/src/gem.c
index 8e3d721..7f200e3 100644
--- i/src/mruby-widget-lib/src/gem.c
+++ w/src/mruby-widget-lib/src/gem.c
@@ -11,8 +11,8 @@
 #ifdef WIN32
 #include <windows.h>
 #endif
+#include <rtosc/rtosc.h>
 #include "../../../deps/pugl/pugl/pugl.h"
-#include "../../../deps/rtosc/include/rtosc/rtosc.h"
 #include "../../../src/osc-bridge/src/bridge.h"
 #include "../../../deps/mruby-nanovg/src/gl_core.3.2.h"
 
