diff -Naur occt-V7_3_0/adm/templates/OpenCASCADEConfig.cmake.in occt-V7_3_0_new/adm/templates/OpenCASCADEConfig.cmake.in
--- occt-V7_3_0/adm/templates/OpenCASCADEConfig.cmake.in	2018-05-29 12:14:02.000000000 +0200
+++ occt-V7_3_0.new/adm/templates/OpenCASCADEConfig.cmake.in	2018-05-29 12:14:02.000000000 +0200
@@ -26,7 +26,7 @@
 if (OpenCASCADE_INSTALL_PREFIX MATCHES "/cmake$")
   get_filename_component (OpenCASCADE_INSTALL_PREFIX "${OpenCASCADE_INSTALL_PREFIX}" PATH)
 endif()
-if (OpenCASCADE_INSTALL_PREFIX MATCHES "/lib$")
+if (OpenCASCADE_INSTALL_PREFIX MATCHES "/lib(64)?$")
   get_filename_component (OpenCASCADE_INSTALL_PREFIX "${OpenCASCADE_INSTALL_PREFIX}" PATH)
 endif()
 
