diff --git a/src/planar.c b/src/planar.c
index 1cdf94a7..9d3e86ab 100644
--- a/src/planar.c
+++ b/src/planar.c
@@ -39,9 +39,7 @@
 #include "c/graphK4Search.h"
 #else
 #include "planarity/graph.h"
-#include "planarity/graphK23Search.h"
-#include "planarity/graphK33Search.h"
-#include "planarity/graphK4Search.h"
+#include "planarity/graphLib.h"
 #endif
 #if defined(__clang__)
 #pragma clang diagnostic pop
