diff -ru digraphs.orig/configure digraphs/configure
--- digraphs.orig/configure	2026-06-17 23:19:35.565131292 +0200
+++ digraphs/configure	2026-06-17 23:20:11.313129846 +0200
@@ -5380,7 +5380,7 @@
           #else
           #error too old
           #endif
-          int main(void) { gp_InitGraph(0, 0); }
+          int main(void) { gp_EnsureVertexCapacity(0, 0); }
 
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
diff -ru digraphs.orig/m4/ax_check_planarity.m4 digraphs/m4/ax_check_planarity.m4
--- digraphs.orig/m4/ax_check_planarity.m4	2026-06-17 23:19:35.548494137 +0200
+++ digraphs/m4/ax_check_planarity.m4	2026-06-17 23:20:23.807332022 +0200
@@ -23,7 +23,7 @@
           #else
           #error too old
           #endif
-          int main(void) { gp_InitGraph(0, 0); }
+          int main(void) { gp_EnsureVertexCapacity(0, 0); }
         ])], [
           AC_MSG_RESULT([yes])
         ], [
diff -ru digraphs.orig/src/planar.c digraphs/src/planar.c
--- digraphs.orig/src/planar.c	2026-06-17 23:39:57.575009123 +0200
+++ digraphs/src/planar.c	2026-06-17 23:28:10.272507602 +0200
@@ -38,7 +38,7 @@
 #include "c/graphLib/homeomorphSearch/graphK33Search.h"
 #include "c/graphLib/homeomorphSearch/graphK4Search.h"
 #else
-#include "planarity/graph.h"
+#include "planarity/graphLib.h"
 #include "planarity/c/graphLib/homeomorphSearch/graphK23Search.h"
 #include "planarity/c/graphLib/homeomorphSearch/graphK33Search.h"
 #include "planarity/c/graphLib/homeomorphSearch/graphK4Search.h"
@@ -169,13 +169,13 @@
 
   graphP theGraph = gp_New();
 
-  if (gp_InitGraph(theGraph, V) != OK) {
+  if (gp_EnsureVertexCapacity(theGraph, V) != OK) {
     gp_Free(&theGraph);
     ErrorQuit("Digraphs: boyers_planarity_check (C): invalid number of nodes!",
               0L,
               0L);
     return 0L;
-  } else if (gp_EnsureArcCapacity(theGraph, 2 * E) != OK) {
+  } else if (gp_EnsureEdgeCapacity(theGraph, 2 * E) != OK) {
     gp_Free(&theGraph);
     ErrorQuit("Digraphs: boyers_planarity_check (C): invalid number of edges!",
               0L,
@@ -185,13 +185,13 @@
 
   switch (flags) {
     case EMBEDFLAGS_SEARCHFORK33:
-      gp_AttachK33Search(theGraph);
+      gp_ExtendWith_K33Search(theGraph);
       break;
     case EMBEDFLAGS_SEARCHFORK23:
-      gp_AttachK23Search(theGraph);
+      gp_ExtendWith_K23Search(theGraph);
       break;
     case EMBEDFLAGS_SEARCHFORK4:
-      gp_AttachK4Search(theGraph);
+      gp_ExtendWith_K4Search(theGraph);
       break;
     default:
       break;
@@ -203,7 +203,7 @@
   // Construct the antisymmetric digraph with no loops
   for (Int v = 1; v <= LEN_LIST(out); ++v) {
     DIGRAPHS_ASSERT(gp_VertexInRange(theGraph, v));
-    gp_SetVertexIndex(theGraph, v, v);
+    gp_SetIndex(theGraph, v, v);
     Obj const out_v = ELM_LIST(out, v);
     for (Int w = 1; w <= LEN_LIST(out_v); ++w) {
       DIGRAPHS_ASSERT(gp_VertexInRange(theGraph, w));
