diff -ubr src/pcmanfm-0.5.1.orig/src/vfs/vfs-utils.c src/pcmanfm-0.5.1/src/vfs/vfs-utils.c
--- src/pcmanfm-0.5.1.orig/src/vfs/vfs-utils.c	2009-04-20 19:39:39.000000000 +0200
+++ src/pcmanfm-0.5.1/src/vfs/vfs-utils.c	2009-07-02 22:13:07.000000000 +0200
@@ -37,7 +37,10 @@
     if( G_LIKELY( file ) )
         icon = gdk_pixbuf_new_from_file( file, NULL );
     else
+    {
         icon = gtk_icon_info_get_builtin_pixbuf( inf );
+        g_object_ref( icon );
+    }
     gtk_icon_info_free( inf );
 
     if( G_LIKELY( icon ) )  /* scale down the icon if it's too big */
