diff -aur psi-0.12.old/src/systeminfo.cpp psi-0.12/src/systeminfo.cpp
--- psi-0.12.old/src/systeminfo.cpp	2007-04-08 10:37:46.000000000 +0200
+++ psi-0.12/src/systeminfo.cpp	2009-05-11 15:09:14.000000000 +0200
@@ -89,7 +89,7 @@
 		{ LinuxSlackware,	OsAppendFile,	"/etc/slackware-version",	"Slackware Linux"	},
 		{ LinuxPLD,		OsUseFile,	"/etc/pld-release",		"PLD Linux"		},
 		{ LinuxAurox,		OsUseName,	"/etc/aurox-release",		"Aurox Linux"		},
-		{ LinuxArch,		OsUseFile,	"/etc/arch-release",		"Arch Linux"		},
+		{ LinuxArch,		OsUseName,	"/etc/arch-release",		"Arch Linux"		},
 		{ LinuxLFS,		OsAppendFile,	"/etc/lfs-release",		"LFS Linux"		},
 
 		// untested
