--- e2fsprogs-1.40.4/misc/Makefile	2008-01-16 14:52:53.000000000 +0000
+++ e2fsprogs-1.40.4/misc/Makefile.neu	2008-01-16 14:53:43.000000000 +0000
@@ -376,6 +376,10 @@
 	@echo "	LD $@"
 	@$(CC) $(ALL_LDFLAGS) -o fsck $(FSCK_OBJS) $(LIBS_BLKID) $(LIBINTL)
 
+fsck.static: $(FSCK_OBJS) $(DEPSTATIC_LIBBLKID)
+	@echo "	LD $@"
+	@$(CC) $(ALL_LDFLAGS) -static -o fsck.static $(FSCK_OBJS) $(STATIC_LIBBLKID) $(DEPSTATIC_LIBUUID) $(LIBINTL)
+
 badblocks: $(BADBLOCKS_OBJS) $(DEPLIBS)
 	@echo "	LD $@"
 	@$(CC) $(ALL_LDFLAGS) -o badblocks $(BADBLOCKS_OBJS) $(LIBS) $(LIBINTL)
