--- gnomebaker-0.6.1/src/main.c.orig	2007-04-14 19:35:21.000000000 +0000
+++ gnomebaker-0.6.1/src/main.c	2007-04-14 19:36:16.000000000 +0000
@@ -50,6 +50,9 @@
 main(gint argc, gchar *argv[])
 {
 	GError *error = NULL;
+
+	g_thread_init(NULL);
+	
 	GOptionContext *context = g_option_context_new(_(" - GNOME CD/DVD burning application"));
 	/* add main entries */
 	g_option_context_add_main_entries(context, entries, GETTEXT_PACKAGE);
