diff -urN a/src/3rdparty/libwebp.pri b/src/3rdparty/libwebp.pri
--- a/src/3rdparty/libwebp.pri	2016-02-02 05:12:49.000000000 -0700
+++ b/src/3rdparty/libwebp.pri	2016-03-27 14:25:43.385199707 -0600
@@ -82,7 +82,7 @@
     INCLUDEPATH += $$NDK_ROOT/sources/android/cpufeatures
 }
 
-equals(QT_ARCH, arm) {
+contains(QT_ARCH, "arm.*") {
     SOURCES_FOR_NEON += \
         $$PWD/libwebp/src/dsp/dec_neon.c \
         $$PWD/libwebp/src/dsp/enc_neon.c \
