From af09900bad738208b3228d9fed8bd3a0619e2638 Mon Sep 17 00:00:00 2001
From: Mike Swanson <mikeonthecomputer@gmail.com>
Date: Sat, 13 Jun 2015 13:35:54 -0700
Subject: [PATCH 03/13] setup.desktop: Add a missing semicolon to Categories

---
 src/setup/setup.desktop.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/setup/setup.desktop.in b/src/setup/setup.desktop.in
index 91decb4..cf127d8 100644
--- a/src/setup/setup.desktop.in
+++ b/src/setup/setup.desktop.in
@@ -4,4 +4,4 @@ Exec=@PROGRAM_PREFIX@setup
 Icon=@PROGRAM_PREFIX@setup
 Type=Application
 Comment=Setup tool for @PACKAGE_SHORTNAME@
-Categories=Settings
+Categories=Settings;
-- 
2.5.0

