From 7c192af0d17f3f45d4208bc15cb1225b811aab74 Mon Sep 17 00:00:00 2001
From: Fabian Greffrath <fabian@greffrath.com>
Date: Mon, 15 Jun 2015 09:16:44 +0200
Subject: [PATCH 04/13] Add Keywords to the setup.desktop file

---
 src/setup/setup.desktop.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/setup/setup.desktop.in b/src/setup/setup.desktop.in
index cf127d8..2a5e71f 100644
--- a/src/setup/setup.desktop.in
+++ b/src/setup/setup.desktop.in
@@ -5,3 +5,4 @@ Icon=@PROGRAM_PREFIX@setup
 Type=Application
 Comment=Setup tool for @PACKAGE_SHORTNAME@
 Categories=Settings;
+Keywords=first;person;shooter;doom;heretic;hexen;strife;vanilla;
-- 
2.5.0

