From 750aa95aad7d924b879dce60aa8abdd9c281af6e Mon Sep 17 00:00:00 2001
From: linguica <github@andrewstine.org>
Date: Wed, 22 Jul 2015 11:06:50 -0700
Subject: [PATCH 07/13] Added deh_bexstr.c to MSVC project

---
 msvc/doom.vcproj | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/msvc/doom.vcproj b/msvc/doom.vcproj
index 7f98d3a..7446ede 100644
--- a/msvc/doom.vcproj
+++ b/msvc/doom.vcproj
@@ -859,6 +859,10 @@
 					>
 				</File>
 				<File
+					RelativePath="..\src\doom\deh_bexstr.c"
+					>
+				</File>
+				<File
 					RelativePath="..\src\doom\deh_cheat.c"
 					>
 				</File>
-- 
2.5.0

