From 525a7b0d071b146221498a13275c5db206a437e3 Mon Sep 17 00:00:00 2001
From: Mike Swanson <mikeonthecomputer@gmail.com>
Date: Fri, 14 Aug 2015 00:42:31 -0700
Subject: [PATCH 10/13] setup-manifest: Add compatibility GUID for Windows 10

---
 src/setup/setup-manifest.xml.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/setup/setup-manifest.xml.in b/src/setup/setup-manifest.xml.in
index 7db56e8..111ad0f 100644
--- a/src/setup/setup-manifest.xml.in
+++ b/src/setup/setup-manifest.xml.in
@@ -32,6 +32,7 @@
       <supportedOS Id="{e2011457-1546-43c5-a5fe-008deee3d3f0}"/> <!-- Vista -->
       <supportedOS Id="{4a2f28e3-53b9-4441-ba9c-d69d4a4a6e38}"/> <!-- 8 -->
       <supportedOS Id="{1f676c76-80e1-4239-95bb-83d0f6d0da78}"/> <!-- 8.1 -->
+      <supportedOS Id="{8e0f7a12-bfb3-4fe8-b9a5-48fd50a15a9a}"/> <!-- 10 -->
     </application>
   </compatibility>
 
-- 
2.5.0

