--- images-old/Makefile	2008-10-22 23:16:32.000000000 -0700
+++ images/Makefile	2008-10-22 23:15:16.000000000 -0700
@@ -223,7 +223,7 @@
 top_builddir = ..
 top_srcdir = ..
 SYNFIG = synfig
-EXT = png
+EXT = mng
 EXTRA_DIST = installer_logo.sif installer_logo_osx.sif \
 	splash_screen.sifz splash_screen-0.61.08.sif \
 	splash_screen-0.61.07.sif about_icon.sif angle_icon.sif \
@@ -650,31 +650,37 @@
 	$(SYNFIG) -q $< -o $@ --time 0
 	echo "  File \"images\\$@\"" >>./images.nsh
 	echo "  Delete \"\$$INSTDIR\\share\\pixmaps\\$@\"" >>./unimages.nsh
+	mogrify -format png $@
 
 .sifz.$(EXT):
 	$(SYNFIG) -q $< -o $@ --time 0
 	echo "  File \"images\\$@\"" >>./images.nsh
 	echo "  Delete \"\$$INSTDIR\\share\\pixmaps\\$@\"" >>./unimages.nsh
+	mogrify -format png $@
 
 keyframe_lock_all.$(EXT): $(srcdir)/keyframe_lock_icon.sif
 	$(SYNFIG) -q $< -o $@ --time 0
 	echo "  File \"images\\$@\"" >>./images.nsh
 	echo "  Delete \"\$$INSTDIR\\share\\pixmaps\\$@\"" >>./unimages.nsh
+	mogrify -format png $@
 
 keyframe_lock_past.$(EXT): $(srcdir)/keyframe_lock_icon.sif
 	$(SYNFIG) -q $< -o $@ --time 0 -c PastOnly
 	echo "  File \"images\\$@\"" >>./images.nsh
 	echo "  Delete \"\$$INSTDIR\\share\\pixmaps\\$@\"" >>./unimages.nsh
+	mogrify -format png $@
 
 keyframe_lock_future.$(EXT): $(srcdir)/keyframe_lock_icon.sif
 	$(SYNFIG) -q $< -o $@ --time 0 -c FutureOnly
 	echo "  File \"images\\$@\"" >>./images.nsh
 	echo "  Delete \"\$$INSTDIR\\share\\pixmaps\\$@\"" >>./unimages.nsh
+	mogrify -format png $@
 
 keyframe_lock_none.$(EXT): $(srcdir)/keyframe_lock_icon.sif
 	$(SYNFIG) -q $< -o $@ --time 0 -c Disabled
 	echo "  File \"images\\$@\"" >>./images.nsh
 	echo "  Delete \"\$$INSTDIR\\share\\pixmaps\\$@\"" >>./unimages.nsh
+	mogrify -format png $@
 
 #.sif.png:
 #	$(SYNFIG) -q $< -o $@ --time 0
