From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@ltnglobal.com>
Date: Mon, 26 Oct 2020 16:24:08 +0100
Subject: [PATCH] opencv: allow compilation against 4.5.x

---
 ext/opencv/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/opencv/meson.build b/ext/opencv/meson.build
index 074272a9c..e3d1865da 100644
--- a/ext/opencv/meson.build
+++ b/ext/opencv/meson.build
@@ -66,7 +66,7 @@ if opencv_found
     endif
   endforeach
 else
-  opencv_dep = dependency('opencv4', version : ['>= 4.0.0', '< 4.5.0'], required : false)
+  opencv_dep = dependency('opencv4', version : ['>= 4.0.0', '< 4.6.0'], required : false)
   opencv_found = opencv_dep.found()
   if opencv_found
     foreach h : libopencv4_headers
