diff --color -aur OpenSubdiv-3_4_4-old/CMakeLists.txt OpenSubdiv-3_4_4-new/CMakeLists.txt
--- OpenSubdiv-3_4_4-old/CMakeLists.txt	2021-06-14 16:59:54.699746298 +0300
+++ OpenSubdiv-3_4_4-new/CMakeLists.txt	2021-06-14 17:01:32.976412136 +0300
@@ -586,8 +586,12 @@
         if (NOT DEFINED OSD_CUDA_NVCC_FLAGS)
             if (CUDA_VERSION_MAJOR LESS 6)
                 set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_11 )
-            else()
+            elseif (CUDA_VERSION_MAJOR LESS 9)
                 set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_20 )
+            elseif (CUDA_VERSION_MAJOR LESS 11)
+                set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_35 )
+            else()
+                set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_53 )
             endif()
         endif()
     endif()
