diff -Nur pdns-2.9.18.orig/modules/gmysqlbackend/smysql.cc pdns-2.9.18/modules/gmysqlbackend/smysql.cc
--- pdns-2.9.18.orig/modules/gmysqlbackend/smysql.cc	2005-04-23 23:40:15.000000000 +0200
+++ pdns-2.9.18/modules/gmysqlbackend/smysql.cc	2005-07-17 10:38:35.363597586 +0200
@@ -15,6 +15,9 @@
 	       const string &password)
 {
   mysql_init(&d_db);
+#if MYSQL_VERSION_ID > 32349
+  mysql_options(&d_db, MYSQL_READ_DEFAULT_GROUP, "client");
+#endif
   if (!mysql_real_connect(&d_db, host.empty() ? 0 : host.c_str(), 
 			  user.empty() ? 0 : user.c_str(), 
 			  password.empty() ? 0 : password.c_str(),
