diff -ur pdns-2.9.22.orig/pdns/dnsparser.cc pdns-2.9.22/pdns/dnsparser.cc
--- pdns-2.9.22.orig/pdns/dnsparser.cc	2008-11-15 22:32:46.000000000 +0200
+++ pdns-2.9.22/pdns/dnsparser.cc	2009-07-06 22:09:02.000000000 +0300
@@ -16,6 +16,7 @@
     Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */
 
+#include <stdio.h>
 #include "dnsparser.hh"
 #include "dnswriter.hh"
 #include <boost/lexical_cast.hpp>
diff -ur pdns-2.9.22.orig/pdns/rcpgenerator.cc pdns-2.9.22/pdns/rcpgenerator.cc
--- pdns-2.9.22.orig/pdns/rcpgenerator.cc	2008-11-15 23:20:27.000000000 +0200
+++ pdns-2.9.22/pdns/rcpgenerator.cc	2009-07-06 22:07:26.000000000 +0300
@@ -16,6 +16,7 @@
     Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */
 
+#include <stdio.h>
 #include "rcpgenerator.hh"
 #include "dnsparser.hh"
 #include "misc.hh"
