--- parted-1.7.1/libparted/fs/hfs/probe.c	2006-05-25 13:28:58.000000000 -0400
+++ parted-1.7.1.new/libparted/fs/hfs/probe.c	2006-06-09 16:58:35.000000000 -0400
@@ -42,8 +42,9 @@
 {
 	PedDevice* dev;
 
+	dev = geom->dev;
 	PED_ASSERT (geom != NULL, return 0);
-	PED_ASSERT ((dev = geom->dev) != NULL, return 0);
+	PED_ASSERT (dev != NULL, return 0);
 
 	if (dev->sector_size != PED_SECTOR_SIZE_DEFAULT) {
 		ped_exception_throw (
