--- macros/zeroconf.m4.orig	2014-06-08 21:23:46.000000000 +0400
+++ macros/zeroconf.m4	2014-06-08 21:36:25.000000000 +0400
@@ -25,23 +25,7 @@
 			zeroconf_dir="$zeroconf"
 		fi
 
-        # mDNS support using mDNSResponder
-        AC_CHECK_HEADER(
-            dns_sd.h,
-            AC_CHECK_LIB(
-                dns_sd,
-                DNSServiceRegister,
-                AC_DEFINE(USE_ZEROCONF, 1, [Use DNS-SD registration]))
-        )
-
-        if test "$ac_cv_lib_dns_sd_DNSServiceRegister" = yes; then
-            ZEROCONF_LIBS="-ldns_sd"
-            AC_DEFINE(HAVE_MDNS, 1, [Use mDNSRespnder/DNS-SD registration])
-            found_zeroconf=yes
-        fi
-
         # mDNS support using Avahi
-        if test x"$found_zeroconf" != x"yes" ; then
             AC_CHECK_HEADER(
                 avahi-client/client.h,
                 AC_CHECK_LIB(
@@ -62,10 +46,26 @@
                 found_zeroconf=yes
                 ;;
             esac
+
+        # mDNS support using mDNSResponder
+        if test x"$found_zeroconf" != x"yes" ; then
+            AC_CHECK_HEADER(
+                dns_sd.h,
+                AC_CHECK_LIB(
+                    dns_sd,
+                    DNSServiceRegister,
+                    AC_DEFINE(USE_ZEROCONF, 1, [Use DNS-SD registration]))
+            )
+
+            if test "$ac_cv_lib_dns_sd_DNSServiceRegister" = yes; then
+                ZEROCONF_LIBS="-ldns_sd"
+                AC_DEFINE(HAVE_MDNS, 1, [Use mDNSRespnder/DNS-SD registration])
+                found_zeroconf=yes
+            fi
+        fi
 	    	CPPFLAGS="$savedcppflags"
 		    LDFLAGS="$savedldflags"
     	fi
-	fi
 
 	netatalk_cv_zeroconf=no
 	AC_MSG_CHECKING([whether to enable Zerconf support])
