--- kdeutils-4.3.2/printer-applet/printer-applet.py~	2009-10-07 19:01:11.156834631 +0200
+++ kdeutils-4.3.2/printer-applet/printer-applet.py	2009-10-07 19:01:25.830154976 +0200
@@ -221,7 +221,7 @@
         self.jobs = {}
         self.jobiters = {}
         self.will_update_job_creation_times = False # whether timeout is set FIXME now job_creation_times_timer
-        self.update_job_creation_times_timer = QTimer(self)
+        self.update_job_creation_times_timer = QTimer()
         self.connect(self.update_job_creation_times_timer, SIGNAL("timeout()"), self.update_job_creation_times)
         self.statusbar_set = False
         self.connecting_to_device = {} # dict of printer->time first seen
