#DPATCHLEVEL=1
diff -Naur arj-3.10.22.orig/gnu/makefile.in arj-3.10.22/gnu/makefile.in
--- arj-3.10.22.orig/gnu/makefile.in	2004-04-17 14:28:06.000000000 +0300
+++ arj-3.10.22/gnu/makefile.in	2005-08-04 21:50:24.000000000 +0300
@@ -192,6 +192,15 @@
 dispose:
 
 #
+# XXX: Do not use -O2, it removes the static const variable with gcc 4.x
+#
+
+INTEGR_DIRS = $(ARJ_DIR) $(REARJ_DIR) $(ARJCRYPT_DIR) $(REGISTER_DIR)
+
+$(patsubst %,%/integr.o, $(INTEGR_DIRS)): $(SRC_DIR)/integr.c
+	$(CC) -Wall -g -c -o$@ $<
+
+#
 # The tools
 #
 
