From d2a6ae09fd7d021612fcf5b7f72258d889fe293a Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 3 May 2020 13:15:03 -0600
Subject: [PATCH] Don't USE_ARM_GCM

---
 lib/freebl/gcm.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/lib/freebl/gcm.c b/lib/freebl/gcm.c
index a2f63a6a8..f10a9a08f 100644
--- a/lib/freebl/gcm.c
+++ b/lib/freebl/gcm.c
@@ -22,10 +22,6 @@
 #if defined(__aarch64__) && defined(IS_LITTLE_ENDIAN) && \
     (defined(__clang__) || defined(__GNUC__) && __GNUC__ > 6)
 #define USE_ARM_GCM
-#elif defined(__arm__) && defined(IS_LITTLE_ENDIAN) && \
-    !defined(NSS_DISABLE_ARM32_NEON)
-/* We don't test on big endian platform, so disable this on big endian. */
-#define USE_ARM_GCM
 #endif
 
 #if defined(__ARM_NEON) || defined(__ARM_NEON__)
-- 
2.51.2

