From 724bf0ea861fba941b0f7be3256e8b26dbcaa154 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Mon, 21 Oct 2019 12:14:13 -0600
Subject: [PATCH] Remove ARM AES from freebl gyp

---
 lib/freebl/freebl.gyp | 68 -------------------------------------------
 1 file changed, 68 deletions(-)

diff --git a/lib/freebl/freebl.gyp b/lib/freebl/freebl.gyp
index 39506921c..8eb5162eb 100644
--- a/lib/freebl/freebl.gyp
+++ b/lib/freebl/freebl.gyp
@@ -116,51 +116,6 @@
         }]
       ]
     },
-    {
-      'target_name': 'gcm-aes-aarch64_c_lib',
-      'type': 'static_library',
-      'sources': [
-        'gcm-aarch64.c'
-      ],
-      'dependencies': [
-        '<(DEPTH)/exports.gyp:nss_exports'
-      ],
-      'cflags': [
-        '-march=armv8-a+crypto'
-      ],
-      'cflags_mozilla': [
-        '-march=armv8-a+crypto'
-      ]
-    },
-    {
-      'target_name': 'armv8_c_lib',
-      'type': 'static_library',
-      'sources': [
-        'aes-armv8.c',
-      ],
-      'dependencies': [
-        '<(DEPTH)/exports.gyp:nss_exports'
-      ],
-      'conditions': [
-        [ 'target_arch=="arm"', {
-          'cflags': [
-            '-march=armv8-a',
-            '-mfpu=crypto-neon-fp-armv8'
-          ],
-          'cflags_mozilla': [
-            '-march=armv8-a',
-            '-mfpu=crypto-neon-fp-armv8'
-          ],
-        }, 'target_arch=="arm64" or target_arch=="aarch64"', {
-          'cflags': [
-            '-march=armv8-a+crypto'
-          ],
-          'cflags_mozilla': [
-            '-march=armv8-a+crypto'
-          ],
-        }]
-      ]
-    },
     {
       'target_name': 'freebl',
       'type': 'static_library',
@@ -189,15 +144,6 @@
           'dependencies': [
             'gcm-aes-x86_c_lib',
           ],
-        }, 'disable_arm_hw_aes==0 and (target_arch=="arm" or target_arch=="arm64" or target_arch=="aarch64")', {
-          'dependencies': [
-            'armv8_c_lib'
-          ],
-        }],
-        [ 'target_arch=="arm64" or target_arch=="aarch64"', {
-          'dependencies': [
-            'gcm-aes-aarch64_c_lib',
-          ],
         }],
         [ 'OS=="linux"', {
           'defines!': [
@@ -235,15 +181,6 @@
           'dependencies': [
             'gcm-aes-x86_c_lib',
           ]
-        }, 'target_arch=="arm" or target_arch=="arm64" or target_arch=="aarch64"', {
-          'dependencies': [
-            'armv8_c_lib',
-          ],
-        }],
-        [ 'target_arch=="arm64" or target_arch=="aarch64"', {
-          'dependencies': [
-            'gcm-aes-aarch64_c_lib',
-          ],
         }],
         [ 'OS!="linux"', {
           'conditions': [
@@ -468,11 +405,6 @@
               'ARMHF',
             ],
           }],
-          [ 'disable_arm_hw_aes==0 and (target_arch=="arm" or target_arch=="arm64" or target_arch=="aarch64")', {
-            'defines': [
-              'USE_HW_AES',
-            ],
-          }],
         ],
       }],
     ],
-- 
2.23.0

