From 2c0b3e4dd3a9552a91ec641fe8c8653575200221 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Fri, 24 Jan 2020 17:43:14 -0700
Subject: [PATCH 1/2] Don't USE_ARM_GCM

---
 lib/freebl/gcm.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/lib/freebl/gcm.c b/lib/freebl/gcm.c
index 2a42f74c0..e0051fa11 100644
--- a/lib/freebl/gcm.c
+++ b/lib/freebl/gcm.c
@@ -21,9 +21,6 @@
 #if defined(__aarch64__) && defined(IS_LITTLE_ENDIAN) && \
     (defined(__clang__) || defined(__GNUC__) && __GNUC__ > 6)
 #define USE_ARM_GCM
-#elif defined(__arm__) && defined(IS_LITTLE_ENDIAN)
-/* We don't test on big endian platform, so disable this on big endian. */
-#define USE_ARM_GCM
 #endif
 
 /* Forward declarations */
-- 
2.24.1

