From ef3212dda464008536d125c19924a2c27c8ea082 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 3 May 2020 13:08:54 -0600
Subject: [PATCH 2/2] fix armv8_c_lib dependency

---
 lib/freebl/freebl.gyp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/freebl/freebl.gyp b/lib/freebl/freebl.gyp
index d780b2217..2dc5c851f 100644
--- a/lib/freebl/freebl.gyp
+++ b/lib/freebl/freebl.gyp
@@ -453,7 +453,7 @@
           'dependencies': [
             'gcm-aes-x86_c_lib',
           ]
-        }, 'target_arch=="arm" or target_arch=="arm64" or target_arch=="aarch64"', {
+        }, 'disable_arm_hw_aes==0 and (target_arch=="arm" or target_arch=="arm64" or target_arch=="aarch64")', {
           'dependencies': [
             'armv8_c_lib',
           ],
-- 
2.24.1

