From 506dcf5e09690ac1d5ce02589a247a5e786aa0a7 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 3 May 2020 13:08:54 -0600
Subject: [PATCH 2/2] fix armv8_c_lib dependency

---
 lib/freebl/freebl.gyp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/freebl/freebl.gyp b/lib/freebl/freebl.gyp
index 3d9969435..dea80a35d 100644
--- a/lib/freebl/freebl.gyp
+++ b/lib/freebl/freebl.gyp
@@ -603,7 +603,7 @@
             'intel-gcm-wrap-nodepend_c_lib',
           ]
         }],
-        [ 'target_arch=="arm" or target_arch=="arm64" or target_arch=="aarch64"', {
+        [ '(disable_arm_hw_aes==0 or disable_arm_hw_sha1==0 or disable_arm_hw_sha2==0) and (target_arch=="arm" or target_arch=="arm64" or target_arch=="aarch64")', {
           'dependencies': [
             'armv8_c_lib',
           ],
-- 
2.51.2

