From 17aba1ccad12ea361d04fd60cda1b12f75cbd8ac Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 3 May 2020 13:08:54 -0600
Subject: [PATCH 2/2] fix armv8_c_lib dependency

---
 lib/freebl/freebl.gyp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/freebl/freebl.gyp b/lib/freebl/freebl.gyp
index dd9446da6..779c8951e 100644
--- a/lib/freebl/freebl.gyp
+++ b/lib/freebl/freebl.gyp
@@ -452,7 +452,7 @@
           'dependencies': [
             'gcm-aes-x86_c_lib',
           ]
-        }, 'target_arch=="arm" or target_arch=="arm64" or target_arch=="aarch64"', {
+        }, 'disable_arm_hw_aes==0 and (target_arch=="arm" or target_arch=="arm64" or target_arch=="aarch64")', {
           'dependencies': [
             'armv8_c_lib',
           ],
-- 
2.26.2

