From ead322cc7a7c401f2d10f2585ec021ed7365aa73 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Fri, 1 May 2020 20:17:39 -0600
Subject: [PATCH 2/2] disable neon

---
 meson.build | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/meson.build b/meson.build
index 74142587..e7844cd0 100644
--- a/meson.build
+++ b/meson.build
@@ -100,9 +100,6 @@ have_neon = false
 if host_machine.cpu_family() == 'aarch64'
   neon_args = []
   have_neon = true
-elif cc.has_argument('-mfpu=neon')
-  neon_args = ['-mfpu=neon']
-  have_neon = true
 endif
 
 
-- 
2.26.2

