diff --git i/src/gst/gstpipewiredeviceprovider.c w/src/gst/gstpipewiredeviceprovider.c
index fefdb57b..f41ea524 100644
--- i/src/gst/gstpipewiredeviceprovider.c
+++ w/src/gst/gstpipewiredeviceprovider.c
@@ -279,12 +279,6 @@ get_core_info (struct pw_remote          *remote,
 
     GST_DEBUG_OBJECT (provider, "have hidden providers: %s", value);
 
-    for (i = 0; monitors[i]; i++) {
-      if (strcmp (monitors[i], "v4l2") == 0)
-        gst_device_provider_hide_provider (provider, "v4l2deviceprovider");
-      else if (strcmp (monitors[i], "alsa") == 0)
-        gst_device_provider_hide_provider (provider, "pulsedeviceprovider");
-    }
     g_strfreev (monitors);
   }
 }
