diff -ru libqtxdg-4.2.0.orig/CMakeLists.txt libqtxdg-4.2.0/CMakeLists.txt
--- libqtxdg-4.2.0.orig/CMakeLists.txt	2025-10-05 10:33:04.762455622 +0200
+++ libqtxdg-4.2.0/CMakeLists.txt	2025-10-05 10:34:45.318683750 +0200
@@ -26,6 +26,9 @@
 find_package(Qt6 ${QT_MINIMUM_VERSION} CONFIG REQUIRED Widgets Svg Xml DBus)
 find_package(GLIB ${GLIB_MINIMUM_VERSION} REQUIRED COMPONENTS gobject gio gio-unix)
 find_package(XTerm)
+if(Qt6_VERSION VERSION_GREATER_EQUAL 6.10)
+  find_package(Qt6 COMPONENTS GuiPrivate REQUIRED)
+endif()
 
 include(GNUInstallDirs)             # Standard directories for installation
 include(CMakePackageConfigHelpers)
diff -ru libqtxdg-4.2.0.orig/src/xdgiconloader/CMakeLists.txt libqtxdg-4.2.0/src/xdgiconloader/CMakeLists.txt
--- libqtxdg-4.2.0.orig/src/xdgiconloader/CMakeLists.txt	2025-10-05 10:33:04.772910866 +0200
+++ libqtxdg-4.2.0/src/xdgiconloader/CMakeLists.txt	2025-10-05 10:38:24.333285939 +0200
@@ -54,6 +54,9 @@
         Qt6::Gui
         Qt6::Svg
 )
+if(Qt6_VERSION VERSION_GREATER_EQUAL 6.10)
+  target_link_libraries(${QTXDGX_ICONLOADER_LIBRARY_NAME} PUBLIC Qt6::GuiPrivate)
+endif()
 
 set_target_properties(${QTXDGX_ICONLOADER_LIBRARY_NAME}
     PROPERTIES
