--- src/fr-command-rpm.c.orig	2009-01-15 13:35:12.000000000 +0100
+++ src/fr-command-rpm.c	2009-01-15 13:38:17.000000000 +0100
@@ -175,7 +175,7 @@
 
 	fr_process_begin_command (comm->process, "sh");
 	fr_process_add_arg (comm->process, "-c");
-	fr_process_add_arg_concat (comm->process, "rpm2cpio ", comm->e_filename, " | cpio -itv", NULL);
+	fr_process_add_arg_concat (comm->process, "rpm2cpio ", comm->e_filename, " | bsdtar -tvf -", NULL);
 	fr_process_end_command (comm->process);
 	fr_process_start (comm->process);
 }
@@ -200,7 +200,7 @@
 
 	cmd = g_string_new ("rpm2cpio ");
 	g_string_append (cmd, comm->e_filename);
-	g_string_append (cmd, " | cpio -idu ");
+	g_string_append (cmd, " | bsdtar -xf - ");
 	for (scan = file_list; scan; scan = scan->next) {
 		char *filename = g_shell_quote (scan->data);
 		g_string_append (cmd, filename);
