From 3f7dd8f6fa2a9e6bd1b57830df24fb83f6b4a951 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Wed, 23 May 2018 19:47:32 -0600
Subject: [PATCH] fix ARM build

---
 GNUmakefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/GNUmakefile b/GNUmakefile
index 124e9cb4..dd7ef190 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -57,8 +57,8 @@ changelogfmt:
 
 # linux builds a linux package independent of the source platform
 linux:
-	mkdir -p pkg/linux_amd64/
-	GOOS=linux GOARCH=amd64 go build -buildmode=pie -ldflags '$(GOLDFLAGS)' -tags '$(GOTAGS)' -o pkg/linux_amd64/consul
+	mkdir -p pkg/$(GOOS)_$(GOARCH)
+	GOOS=linux GOARCH=$(GOARCH) go build -buildmode=pie -ldflags '$(GOLDFLAGS)' -tags '$(GOTAGS)' -o pkg/$(GOOS)_$(GOARCH)/consul
 
 # dist builds binaries for all platforms and packages them for distribution
 dist:
-- 
2.17.0

