From ec190de75ad612a93561d65a11168a544d334bac Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 26 Jun 2018 06:43:01 -0600
Subject: [PATCH] fix ARM build

---
 GNUmakefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/GNUmakefile b/GNUmakefile
index bee271760..7786ac50f 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -137,7 +137,7 @@ changelogfmt:
 
 # linux builds a linux package independent of the source platform
 linux:
-	@$(SHELL) $(CURDIR)/build-support/scripts/build-local.sh -o linux -a amd64
+	@$(SHELL) $(CURDIR)/build-support/scripts/build-local.sh -o linux -a $(GOARCH)
 
 # dist builds binaries for all platforms and packages them for distribution
 dist:
-- 
2.21.0

