--- a/./logger/syslog_test.go	2018-11-14 23:37:47.000000000 +0100
+++ b/./logger/syslog_test.go	2019-01-04 14:16:21.575451546 +0100
@@ -16,6 +16,9 @@ func TestSyslogFilter(t *testing.T) {
 	if os.Getenv("TRAVIS") == "true" {
 		t.Skip("Syslog not supported on travis-ci")
 	}
+	if os.Getenv("CONSUL_TEST_SKIP_SYSLOG") == "true" {
+		t.Skip("Skipping test due to env var CONSUL_TEST_SKIP_SYSLOG being set")
+	}
 
 	l, err := gsyslog.NewLogger(gsyslog.LOG_NOTICE, "LOCAL0", "consul")
 	if err != nil {
