--- a/version/version.go	2018-05-11 20:57:42.000000000 +0200
+++ b/version/version.go	2018-05-22 14:05:53.639672108 +0200
@@ -1,7 +1,6 @@
 package version
 
 import (
-	"fmt"
 	"strings"
 )
 
@@ -27,21 +26,6 @@
 // for displaying to humans.
 func GetHumanVersion() string {
 	version := Version
-	if GitDescribe != "" {
-		version = GitDescribe
-	}
-
-	release := VersionPrerelease
-	if GitDescribe == "" && release == "" {
-		release = "dev"
-	}
-	if release != "" {
-		version += fmt.Sprintf("-%s", release)
-		if GitCommit != "" {
-			version += fmt.Sprintf(" (%s)", GitCommit)
-		}
-	}
-
 	// Strip off any single quotes added by the git information.
 	return strings.Replace(version, "'", "", -1)
 }
