--- a/GNUmakefile	2018-05-22 13:25:37.236071564 +0200
+++ b/GNUmakefile	2018-05-22 13:26:44.852192711 +0200
@@ -58,7 +58,7 @@
 # linux builds a linux package independent of the source platform
 linux:
 	mkdir -p pkg/linux_amd64/
-	GOOS=linux GOARCH=amd64 go build -ldflags '$(GOLDFLAGS)' -tags '$(GOTAGS)' -o pkg/linux_amd64/consul
+	GOOS=linux GOARCH=amd64 go build -buildmode=pie -ldflags '$(GOLDFLAGS)' -tags '$(GOTAGS)' -o pkg/linux_amd64/consul
 
 # dist builds binaries for all platforms and packages them for distribution
 dist:
