Index: src/p80211/p80211netdev.c
===================================================================
--- src/p80211/p80211netdev.c	(revision 1865)
+++ src/p80211/p80211netdev.c	(working copy)
@@ -947,7 +947,9 @@
 #endif
 #endif
 	
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,24) )
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26) )
+		dev_net_set(dev, &init_net);
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,24) )
 		dev->nd_net = &init_net;
 #endif