diff -ur linux-wlan-ng-0.2.1pre22/man/Makefile linux-wlan-ng-0.2.1pre22-new/man/Makefile
--- linux-wlan-ng-0.2.1pre22/man/Makefile	2004-08-04 14:09:27.000000000 -0700
+++ linux-wlan-ng-0.2.1pre22-new/man/Makefile	2004-10-18 12:23:19.000000000 -0700
@@ -48,8 +48,8 @@
 	@echo "Nothing to do..."
 
 install:
-	mkdir -p $(TARGET_ROOT_ON_HOST)/usr/local/man/man1
-	cp *.1 	$(TARGET_ROOT_ON_HOST)/usr/local/man/man1
+	mkdir -p $(TARGET_ROOT_ON_HOST)/usr/share/man/man1
+	cp *.1 	$(TARGET_ROOT_ON_HOST)/usr/share/man/man1
 
 clean:
 	rm -f core core.* *.o .*.o *.s *.a .depend tmp_make *~ tags
