diff -upNr nvclock0.8b4.orign/src/Makefile.in nvclock0.8b4/src/Makefile.in
--- nvclock0.8b4.orign/src/Makefile.in	2009-01-03 18:02:54.000000000 +0100
+++ nvclock0.8b4/src/Makefile.in	2009-01-07 11:37:39.000000000 +0100
@@ -50,8 +50,8 @@ distclean: distclean-recursive clean-app
 install: install-recursive install-app
 install-app:
 	mkdir -p $(bindir)
-	$(INSTALL) -c nvclock $(DESTDIR)$(bindir)/nvclock
-	$(INSTALL) -c smartdimmer $(DESTDIR)$(bindir)/smartdimmer
+	$(INSTALL) -c nvclock $(bindir)/nvclock
+	$(INSTALL) -c smartdimmer $(bindir)/smartdimmer
 
 
 uninstall: uninstall-recursive uninstall-app


