diff -Nur cryptsetup-luks-1.0.4.orig/src/cryptsetup.c cryptsetup-luks-1.0.4/src/cryptsetup.c
--- cryptsetup-luks-1.0.4.orig/src/cryptsetup.c	2006-10-04 15:47:00.000000000 +0200
+++ cryptsetup-luks-1.0.4/src/cryptsetup.c	2006-12-16 15:54:12.000000000 +0100
@@ -249,7 +249,8 @@
 	if (opt_readonly)
 		options.flags |= CRYPT_FLAG_READONLY;
 	r = crypt_luksOpen(&options);
-	show_status(-r);
+	if(r)
+		show_status(-r);
 	return r;
 }
 
