--- a/cmake/findDependencies.cmake
+++ b/cmake/findDependencies.cmake
@@ -48,7 +48,9 @@ endif()
 
 if (NOT USE_BUNDLED_TINYXML2)
     find_package(tinyxml2 QUIET)
-    if (NOT tinyxml2_FOUND)
+    if (tinyxml2_FOUND)
+        set(tinyxml2_LIBRARIES "tinyxml2")
+    else()
         find_library(tinyxml2_LIBRARIES tinyxml2)
         if (NOT tinyxml2_LIBRARIES)
             message(FATAL_ERROR "tinyxml2 has not been found")
