From fffe7737f11b844c3007b588310c927be437df86 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 23 Apr 2017 10:15:01 -0600
Subject: [PATCH] Link against GLESv2 for Linux

Signed-off-by: Kevin Mihelich <kevin@archlinuxarm.org>
---
 analitzaplot/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/analitzaplot/CMakeLists.txt b/analitzaplot/CMakeLists.txt
index 560401b4..0c1256af 100644
--- a/analitzaplot/CMakeLists.txt
+++ b/analitzaplot/CMakeLists.txt
@@ -63,7 +63,7 @@ target_link_libraries ( AnalitzaPlot
   Analitza
 )
 
-if(CMAKE_SYSTEM_NAME STREQUAL Android)
+if(CMAKE_SYSTEM_NAME STREQUAL Android OR CMAKE_SYSTEM_NAME STREQUAL Linux)
     target_link_libraries(AnalitzaPlot GLESv2)
 elseif(WIN32)
     target_link_libraries(AnalitzaPlot ${GLEW_LIBRARIES})
-- 
2.12.2

