diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2015-01-05 13:24:31.000000000 -0700
+++ b/CMakeLists.txt	2015-01-24 10:23:32.169332291 -0700
@@ -29,14 +29,7 @@
                         VERSION_HEADER "${CMAKE_CURRENT_BINARY_DIR}/analitza_version.h"
                         PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/Analitza5ConfigVersion.cmake")
 
-find_package(OpenGL)
-set_package_properties(OpenGL PROPERTIES TYPE OPTIONAL
-                                         PURPOSE "Support for 3D graphs in Analitza"
-                                         URL "http://opengl.org")
-if(OPENGL_FOUND)
-    find_package(Qt5OpenGL ${QT_REQUIRED_VERSION} REQUIRED NO_MODULE)
-endif()
-set(HAVE_OPENGL ${Qt5OpenGL_FOUND})
+set(HAVE_OPENGL FALSE)
 
 find_package(Eigen3) # find and setup Eigen3 if available
 set(HAVE_EIGEN3 ${EIGEN3_FOUND})
