diff -up analitza-4.10.1/CMakeLists.txt.opengl_optional analitza-4.10.1/CMakeLists.txt
--- analitza-4.10.1/CMakeLists.txt.opengl_optional	2013-03-01 00:39:17.251723086 -0600
+++ analitza-4.10.1/CMakeLists.txt	2013-03-06 21:48:38.705252164 -0600
@@ -13,8 +13,11 @@ macro_bool_to_01(READLINE_FOUND HAVE_REA
 macro_log_feature(READLINE_FOUND "GNU Readline" "Allows KAlgebra to provide a console interface." "http://cnswww.cns.cwru.edu/php/chet/readline/rltop.html" FALSE "" "")
 
 macro_optional_find_package(OpenGL)
-macro_bool_to_01(OPENGL_FOUND HAVE_OPENGL)
-macro_log_feature(OPENGL_FOUND "OpenGL" "Support for 3D graphs in Analitza" "http://opengl.org" FALSE "" "")
+option(SHOULD_BUILD_OPENGL "Build OpenGL support" ON)
+if(SHOULD_BUILD_OPENGL)
+  macro_bool_to_01(OPENGL_FOUND HAVE_OPENGL)
+endif(SHOULD_BUILD_OPENGL)
+macro_log_feature(HAVE_OPENGL "OpenGL" "Support for 3D graphs in Analitza" "http://opengl.org" FALSE "" "")
 
 include_directories(.)
 add_definitions (-DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS)
