diff -urN a/setup.py b/setup.py
--- a/setup.py	2014-01-20 05:05:14.000000000 -0700
+++ b/setup.py	2014-02-07 06:45:06.632571190 -0700
@@ -734,9 +734,6 @@
     inc_dirs += glob.glob('c-blosc/internal-complibs/*')
     # ...and the macros for all the compressors supported
     def_macros += [('HAVE_LZ4', 1), ('HAVE_SNAPPY', 1), ('HAVE_ZLIB', 1)]
-    # Add -msse2 flag for optimizing shuffle in include Blosc
-    if os.name == 'posix':
-        CFLAGS.append("-msse2")
 else:
     ADDLIBS += ['blosc']
 
