--- setup.py.old	2016-05-24 13:16:36.129769271 +0200
+++ setup.py	2016-05-24 13:17:07.380243229 +0200
@@ -552,6 +552,11 @@
         libdir = compiler.has_function(package.target_function,
                                        libraries=(package.library_name,))
 
+    # force location of hdf5_18
+    if package==hdf5_package:
+        hdrdir = '/usr/include/hdf5_18'
+        libdir = '/usr/lib/hdf5_18'    
+
     if not (hdrdir and libdir):
         if package.tag in ['HDF5']:  # these are compulsory!
             pname, ptag = package.name, package.tag
