From 9ed900289d43c849da0ddaad60254a350e73d0cf Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 23 Oct 2016 10:56:44 -0600
Subject: [PATCH 1/2] Add aarch64 GN support

---
 tools/gn/args.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tools/gn/args.cc b/tools/gn/args.cc
index 9aaafb0..7ddfba4 100644
--- a/tools/gn/args.cc
+++ b/tools/gn/args.cc
@@ -268,6 +268,7 @@ void Args::SetSystemVarsLocked(Scope* dest) const {
   static const char kX86[] = "x86";
   static const char kX64[] = "x64";
   static const char kArm[] = "arm";
+  static const char kArm64[] = "arm64";
   static const char kMips[] = "mipsel";
   const char* arch = nullptr;
 
@@ -280,6 +281,8 @@ void Args::SetSystemVarsLocked(Scope* dest) const {
     arch = kX64;
   else if (os_arch.substr(0, 3) == "arm")
     arch = kArm;
+  else if (os_arch == "aarch64")
+    arch = kArm64;
   else if (os_arch == "mips")
     arch = kMips;
   else
-- 
2.10.2

