From 99b6cda245256c0ccb5c50e4bf7a4c6256ec5c2c Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 23 Oct 2016 10:57:25 -0600
Subject: [PATCH] ARM toolchain fixes

---
 build/toolchain/linux/BUILD.gn | 24 ++++++++++--------------
 1 file changed, 10 insertions(+), 14 deletions(-)

diff --git a/build/toolchain/linux/BUILD.gn b/build/toolchain/linux/BUILD.gn
index 86cd7dabc6e8..4b1e73b1aca3 100644
--- a/build/toolchain/linux/BUILD.gn
+++ b/build/toolchain/linux/BUILD.gn
@@ -22,15 +22,13 @@ clang_toolchain("clang_arm64") {
 }
 
 gcc_toolchain("arm64") {
-  toolprefix = "aarch64-linux-gnu-"
-
-  cc = "${toolprefix}gcc"
-  cxx = "${toolprefix}g++"
+  cc = "gcc"
+  cxx = "g++"
 
-  ar = "${toolprefix}ar"
+  ar = "ar"
   ld = cxx
-  readelf = "${toolprefix}readelf"
-  nm = "${toolprefix}nm"
+  readelf = "readelf"
+  nm = "nm"
 
   toolchain_args = {
     current_cpu = "arm64"
@@ -40,15 +38,13 @@ gcc_toolchain("arm64") {
 }
 
 gcc_toolchain("arm") {
-  toolprefix = "arm-linux-gnueabihf-"
-
-  cc = "${toolprefix}gcc"
-  cxx = "${toolprefix}g++"
+  cc = "gcc"
+  cxx = "g++"
 
-  ar = "${toolprefix}ar"
+  ar = "ar"
   ld = cxx
-  readelf = "${toolprefix}readelf"
-  nm = "${toolprefix}nm"
+  readelf = "readelf"
+  nm = "nm"
 
   toolchain_args = {
     current_cpu = "arm"
-- 
2.12.0

