From 69d032371b025928e54c4304b8443fed7b12c08c Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 23 Oct 2016 10:57:25 -0600
Subject: [PATCH 3/3] ARM toolchain fixes

---
 build/toolchain/linux/BUILD.gn | 28 +++++++++++++++++++++-------
 1 file changed, 21 insertions(+), 7 deletions(-)

diff --git a/build/toolchain/linux/BUILD.gn b/build/toolchain/linux/BUILD.gn
index 548722d..39c5b08 100644
--- a/build/toolchain/linux/BUILD.gn
+++ b/build/toolchain/linux/BUILD.gn
@@ -22,15 +22,13 @@ clang_toolchain("clang_arm64") {
 }
 
 gcc_toolchain("arm") {
-  toolprefix = "arm-linux-gnueabihf-"
-
-  cc = "${toolprefix}gcc"
-  cxx = "${toolprefix}g++"
+  cc = "gcc"
+  cxx = "g++"
 
-  ar = "${toolprefix}ar"
+  ar = "ar"
   ld = cxx
-  readelf = "${toolprefix}readelf"
-  nm = "${toolprefix}nm"
+  readelf = "readelf"
+  nm = "nm"
 
   toolchain_args = {
     current_cpu = "arm"
@@ -39,6 +37,22 @@ gcc_toolchain("arm") {
   }
 }
 
+gcc_toolchain("arm64") {
+  cc = "gcc"
+  cxx = "g++"
+
+  ar = "ar"
+  ld = cxx
+  readelf = "readelf"
+  nm = "nm"
+
+  toolchain_args = {
+    current_cpu = "arm64"
+    current_os = "linux"
+    is_clang = false
+  }
+}
+
 clang_toolchain("clang_x86") {
   toolchain_args = {
     current_cpu = "x86"
-- 
2.10.0

