--- a/ui/ozone/platform/wayland/host/wayland_buffer_manager_host.cc
+++ b/ui/ozone/platform/wayland/host/wayland_buffer_manager_host.cc
@@ -102,8 +102,7 @@
 }
 
 bool WaylandBufferManagerHost::SupportsAcquireFence() const {
-  return connection_->SupportsExplicitSync() ||
-         connection_->UseImplicitSyncInterop();
+  return false;
 }
 
 bool WaylandBufferManagerHost::SupportsViewporter() const {
