--- ppapi/native_client/src/untrusted/pnacl_support_extension/pnacl_support_extension.gyp.orig	2013-03-27 19:49:17.000000000 +0100
+++ ppapi/native_client/src/untrusted/pnacl_support_extension/pnacl_support_extension.gyp	2013-03-27 19:49:59.000000000 +0100
@@ -13,7 +13,7 @@
     'target_name': 'pnacl_support_extension',
     'type': 'none',
     'conditions': [
-      ['disable_nacl==0 and disable_nacl_untrusted==0', {
+      ['disable_nacl==0 and disable_pnacl==0 and disable_nacl_untrusted==0', {
         'dependencies': [
           '../../../../../ppapi/native_client/src/untrusted/pnacl_irt_shim/pnacl_irt_shim.gyp:pnacl_irt_shim',
           '../../../../../native_client/tools.gyp:prep_toolchain',
