--- chromium-browser-26.0.1410.19/src/build/linux/system.gyp~	2013-03-04 00:04:06.000000000 +0200
+++ chromium-browser-26.0.1410.19/src/build/linux/system.gyp	2013-03-04 13:45:31.822101677 +0200
@@ -494,7 +494,7 @@
                      '--name', 'LibSpeechdLoader',
                      '--output-h', '<(output_h)',
                      '--output-cc', '<(output_cc)',
-                     '--header', '<libspeechd.h>',
+                     '--header', '<speech-dispatcher/libspeechd.h>',
                      '--link-directly=<(linux_link_libspeechd)',
                      'spd_open',
                      'spd_say',
