--- third_party/icu/icu.gyp.orig	2014-03-16 17:49:02.561203814 -0500
+++ third_party/icu/icu.gyp	2014-03-16 17:49:23.191037086 -0500
@@ -30,7 +30,7 @@
       }],
       ['(OS=="linux" or OS=="freebsd" or OS=="openbsd" or OS=="solaris" \
          or OS=="netbsd" or OS=="mac" or OS=="android") and \
-        (target_arch=="arm" or target_arch=="ia32" or \
+        (target_arch=="ia32" or \
          target_arch=="mipsel")', {
         'target_conditions': [
           ['_toolset=="host"', {
