Index: src/chrome/VERSION
===================================================================
--- src/chrome/VERSION	(revision 244501)
+++ src/chrome/VERSION	(revision 244503)
@@ -1,4 +1,4 @@
 MAJOR=32
 MINOR=0
 BUILD=1700
-PATCH=76
+PATCH=77
Index: src/chrome/browser/browser_process_impl.cc
===================================================================
--- src/chrome/browser/browser_process_impl.cc	(revision 244501)
+++ src/chrome/browser/browser_process_impl.cc	(revision 244503)
@@ -882,7 +882,6 @@
       prefs::kMetricsReportingEnabled,
       base::Bind(&BrowserProcessImpl::ApplyMetricsReportingPolicy,
                  base::Unretained(this)));
-  ApplyMetricsReportingPolicy();
 #endif
 
   int max_per_proxy = local_state_->GetInteger(prefs::kMaxConnectionsPerProxy);
@@ -912,6 +911,10 @@
   if (local_state_->IsManagedPreference(prefs::kDefaultBrowserSettingEnabled))
     ApplyDefaultBrowserPolicy();
 
+#if !defined(OS_CHROMEOS) && !defined(OS_ANDROID) && !defined(OS_IOS)
+  ApplyMetricsReportingPolicy();
+#endif
+
 #if defined(ENABLE_PLUGINS)
   PluginService* plugin_service = PluginService::GetInstance();
   plugin_service->SetFilter(ChromePluginServiceFilter::GetInstance());
@@ -1037,11 +1040,11 @@
 
 void BrowserProcessImpl::ApplyMetricsReportingPolicy() {
 #if !defined(OS_CHROMEOS) && !defined(OS_ANDROID) && !defined(OS_IOS)
-  BrowserThread::PostTask(
+  CHECK(BrowserThread::PostTask(
       BrowserThread::FILE, FROM_HERE,
       base::Bind(
           base::IgnoreResult(&GoogleUpdateSettings::SetCollectStatsConsent),
-          local_state()->GetBoolean(prefs::kMetricsReportingEnabled)));
+          local_state()->GetBoolean(prefs::kMetricsReportingEnabled))));
 #endif
 }
 
