--- third_party/icu/icu.gyp.orig	2014-05-20 19:17:22.879538568 -0600
+++ third_party/icu/icu.gyp	2014-05-20 19:19:42.204032167 -0600
@@ -30,7 +30,7 @@
       }],
       ['(OS=="linux" or OS=="freebsd" or OS=="openbsd" or OS=="solaris" \
          or OS=="netbsd" or OS=="mac" or OS=="android" or OS=="qnx") and \
-        (target_arch=="arm" or target_arch=="ia32" or \
+        (target_arch=="ia32" or \
          target_arch=="mipsel")', {
         'target_conditions': [
           ['_toolset=="host"', {
