Index: content/child/npapi/plugin_url_fetcher.cc
===================================================================
--- content/child/npapi/plugin_url_fetcher.cc	(revision 233351)
+++ content/child/npapi/plugin_url_fetcher.cc	(working copy)
@@ -145,7 +145,7 @@
 
 void PluginURLFetcher::URLRedirectResponse(bool allow) {
   if (allow) {
-    bridge_->SetDefersLoading(true);
+    bridge_->SetDefersLoading(false);
   } else {
     bridge_->Cancel();
     plugin_stream_->DidFail(resource_id_);  // That will delete |this|.
