diff -upr chromium-26.0.1410.43.orig/media/audio/audio_util.cc chromium-26.0.1410.43/media/audio/audio_util.cc
--- chromium-26.0.1410.43.orig/media/audio/audio_util.cc	2013-03-29 02:10:07.000000000 +0200
+++ chromium-26.0.1410.43/media/audio/audio_util.cc	2013-03-29 02:21:41.000000000 +0200
@@ -201,8 +201,6 @@ size_t GetAudioHardwareBufferSize() {
   HRESULT hr = CoreAudioUtil::GetPreferredAudioParameters(eRender, eConsole,
                                                           &params);
   return FAILED(hr) ? kFallbackBufferSize : params.frames_per_buffer();
-#elif defined(OS_LINUX)
-  return 512;
 #else
   return 2048;
 #endif
