diff -upr chromium-20.0.1132.57.orig/third_party/WebKit/Source/WebCore/WebCore.gyp/scripts/rule_bison.py chromium-20.0.1132.57/third_party/WebKit/Source/WebCore/WebCore.gyp/scripts/rule_bison.py
--- chromium-20.0.1132.57.orig/third_party/WebKit/Source/WebCore/WebCore.gyp/scripts/rule_bison.py	2012-07-10 10:51:56.000000000 +0300
+++ chromium-20.0.1132.57/third_party/WebKit/Source/WebCore/WebCore.gyp/scripts/rule_bison.py	2012-07-24 20:20:48.316103484 +0300
@@ -87,7 +87,7 @@ assert outputHTmp != None

 # Read the header file in under the generated name and remove it.
 outputHFile = open(outputHTmp)
-outputHContents = outputHFile.read()
+outputHContents = outputHFile.readlines()
 outputHFile.close()
 os.unlink(outputHTmp)

@@ -97,6 +97,8 @@ outputH = os.path.join(outputDir, inputR
 outputHFile = open(outputH, 'w')
 print >>outputHFile, '#ifndef %sH' % inputRoot
 print >>outputHFile, '#define %sH' % inputRoot
-print >>outputHFile, outputHContents
+print >>outputHFile, ''.join(filter(
+	lambda line: not line.startswith('int %sparse' % prefix),
+	outputHContents))
 print >>outputHFile, '#endif'
 outputHFile.close()
