--- a/build/common.gypi	2013-01-09 07:02:00.000000000 -0700
+++ b/build/common.gypi	2013-01-11 20:09:13.380197474 -0700
@@ -594,7 +594,7 @@
           'linux_use_gold_flags%': 0,
           # sysroot needs to be an absolute path otherwise it generates
           # incorrect results when passed to pkg-config
-          'sysroot%': '<!(cd <(DEPTH) && pwd -P)/arm-sysroot',
+          #'sysroot%': '<!(cd <(DEPTH) && pwd -P)/arm-sysroot',
         }], # OS=="linux" and target_arch=="arm" and chromeos==0
       ],
 
