--- a/third_party/WebKit/Source/platform/image-decoders/png/PNGImageDecoder.cppg	2016-05-01 15:22:46.103607522 +0800
+++ b/third_party/WebKit/Source/platform/image-decoders/png/PNGImageDecoder.cpp	2016-05-01 15:25:14.459753876 +0800
@@ -234,7 +234,7 @@
 #endif
             png_uint_32 profileLength = 0;
             if (png_get_iCCP(png, info, &profileName, &compressionType, &profile, &profileLength)) {
-                setColorProfileAndTransform(profile, profileLength, imageHasAlpha, false /* useSRGB */);
+                setColorProfileAndTransform((const char*)profile, profileLength, imageHasAlpha, false /* useSRGB */);
             }
         }
 #endif // PNG_iCCP_SUPPORTED
