--- skia/skia.gyp.orig	2013-08-21 15:19:34.068777150 -0600
+++ skia/skia.gyp	2013-08-21 15:20:16.163601088 -0600
@@ -753,6 +753,7 @@
             '../third_party/skia/src/opts/SkBlitRow_opts_arm.cpp',
             '../third_party/skia/src/opts/SkBlitRow_opts_arm.h',
             '../third_party/skia/src/opts/opts_check_arm.cpp',
+            '../third_party/skia/src/opts/memset.arm.S',
           ],
         }],
         [ 'target_arch == "mipsel"',{
