--- skia/skia_library_opts.gyp.orig	2014-01-15 18:36:06.947606588 -0700
+++ skia/skia_library_opts.gyp	2014-01-15 18:38:07.942123899 -0700
@@ -117,6 +117,7 @@
             '../third_party/skia/src/opts/SkBlitRow_opts_arm.cpp',
             '../third_party/skia/src/opts/SkBlitRow_opts_arm.h',
             '../third_party/skia/src/opts/opts_check_arm.cpp',
+            '../third_party/skia/src/opts/memset.arm.S',
             '../third_party/skia/src/opts/SkXfermode_opts_none.cpp',
             '../third_party/skia/src/opts/SkMorphology_opts_none.cpp',
           ],
