--- skia/skia_library_opts.gyp.orig	2013-10-02 17:58:46.454702283 -0600
+++ skia/skia_library_opts.gyp	2013-10-02 17:58:17.649830405 -0600
@@ -110,6 +110,7 @@
             '../third_party/skia/src/opts/SkBlitRow_opts_arm.cpp',
             '../third_party/skia/src/opts/SkBlitRow_opts_arm.h',
             '../third_party/skia/src/opts/opts_check_arm.cpp',
+            '../third_party/skia/src/opts/memset.arm.S',
           ],
         }],
         [ 'target_arch == "mipsel"',{
