--- ./third_party/ffmpeg/chromium/scripts/build_ffmpeg.sh.orig	2013-10-18 18:18:28.032701497 -0600
+++ ./third_party/ffmpeg/chromium/scripts/build_ffmpeg.sh	2013-10-18 18:37:10.817807244 -0600
@@ -310,13 +310,12 @@
     # much smaller than optimized arm builds, hence we go with the global
     # CrOS settings.
     add_flag_common --enable-armv6
-    add_flag_common --enable-armv6t2
+    add_flag_common --disable-armv6t2
     add_flag_common --enable-vfp
-    add_flag_common --enable-thumb
+    add_flag_common --disable-thumb
     add_flag_common --disable-neon
-    add_flag_common --extra-cflags=-march=armv7-a
-    add_flag_common --extra-cflags=-mtune=cortex-a8
-    add_flag_common --extra-cflags=-mfpu=vfpv3-d16
+    add_flag_common --extra-cflags=-march=armv6
+    add_flag_common --extra-cflags=-mfpu=vfp
     # NOTE: softfp/hardfp selected at gyp time.
     add_flag_common --extra-cflags=-mfloat-abi=hard
   elif [ "$TARGET_ARCH" = "arm-neon" ]; then
--- ./third_party/ffmpeg/ffmpeg.gyp.orig	2013-10-18 18:07:27.025272782 -0600
+++ ./third_party/ffmpeg/ffmpeg.gyp	2013-10-18 18:38:58.462388238 -0600
@@ -234,9 +234,7 @@
                 '-marm',
               ],
               'cflags': [
-                '-mthumb',
-                '-march=armv7-a',
-                '-mtune=cortex-a8',
+                '-march=armv6',
               ],
               # On arm we use gcc to compile the assembly.
               'sources': [
@@ -245,7 +243,7 @@
               'conditions': [
                 ['arm_neon == 0', {
                   'cflags': [
-                    '-mfpu=vfpv3-d16',
+                    '-mfpu=vfp',
                   ],
                 }, {
                   'cflags': [
