diff --unified --recursive --text chromium-86.0.4240.75-unmodified/third_party/widevine/cdm/widevine.gni chromium-86.0.4240.75/third_party/widevine/cdm/widevine.gni
--- chromium-86.0.4240.75-unmodified/third_party/widevine/cdm/widevine.gni	2020-10-13 08:34:52.557047763 +0000
+++ chromium-86.0.4240.75/third_party/widevine/cdm/widevine.gni	2020-10-13 08:18:08.646989774 +0000
@@ -23,7 +23,7 @@
 # supported via Android MediaDrm API.
 library_widevine_cdm_available =
     (is_chromeos && (target_cpu == "x64" || target_cpu == "arm")) ||
-    (is_linux && (target_cpu == "x86" || target_cpu == "x64")) ||
+    (is_linux && (target_cpu == "x86" || target_cpu == "x64" || target_cpu == "arm")) ||
     (is_mac && (target_cpu == "x64" || target_cpu == "arm64")) ||
     (is_win && (target_cpu == "x86" || target_cpu == "x64"))
 
