diff --git a/src/google/protobuf/message_lite.h b/src/google/protobuf/message_lite.h
index 25264ef5c..3c8b10d3c 100644
--- a/src/google/protobuf/message_lite.h
+++ b/src/google/protobuf/message_lite.h
@@ -293,9 +293,8 @@ template <typename T>
 using MessageTraits = decltype(MessageTraitsImpl::value<T>);
 
 struct EnumTraitsImpl {
-  struct Undefined;
   template <typename T>
-  static Undefined value;
+  static std::enable_if_t<sizeof(T) != 0> value;
 };
 template <typename T>
 using EnumTraits = decltype(EnumTraitsImpl::value<T>);
