diff -ruN a/python/setup.py b/python/setup.py
--- a/python/setup.py	2022-12-13 01:03:12.000000000 +0100
+++ b/python/setup.py	2023-04-06 22:12:36.199029178 +0200
@@ -220,10 +220,10 @@
   return False
 
 
-if __name__ == '__main__':
+if True:
   ext_module_list = []
   warnings_as_errors = '--warnings_as_errors'
-  if GetOptionFromArgv('--cpp_implementation'):
+  if True:
     # Link libprotobuf.a and libprotobuf-lite.a statically with the
     # extension. Note that those libraries have to be compiled with
     # -fPIC for this to work.
