diff --git a/cmake/installed_include_golden.txt b/cmake/installed_include_golden.txt
index 20ad3219b6..0562066bd7 100644
--- a/cmake/installed_include_golden.txt
+++ b/cmake/installed_include_golden.txt
@@ -35,6 +35,8 @@ google/protobuf/compiler/php/php_generator.h
 google/protobuf/compiler/plugin.h
 google/protobuf/compiler/plugin.pb.h
 google/protobuf/compiler/plugin.proto
+google/protobuf/compiler/python/generator.h
+google/protobuf/compiler/python/pyi_generator.h
 google/protobuf/compiler/retention.h
 google/protobuf/compiler/scc.h
 google/protobuf/compiler/subprocess.h
diff --git a/src/file_lists.cmake b/src/file_lists.cmake
index d74a982f79..36984247bf 100644
--- a/src/file_lists.cmake
+++ b/src/file_lists.cmake
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/php/php_generator.h
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/plugin.h
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/plugin.pb.h
+  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/python/generator.h
+  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/python/pyi_generator.h
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/retention.h
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/scc.h
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/subprocess.h
