--- krusader-2.0.0/krusader/VFS/kiojobwrapper.cpp~  2009-04-11 12:52:31.318455396 +0200
+++ krusader-2.0.0/krusader/VFS/kiojobwrapper.cpp   2009-04-11 12:55:12.005493271 +0200
@@ -28,6 +28,7 @@
  *                                                                         *
  ***************************************************************************/

+#include <stdio.h>
 #include "kiojobwrapper.h"
 #include <qevent.h>
 #include <kurl.h>
