diff -Nur -x '*.orig' -x '*~' gconf-2.25.2/defaults/gconf-defaults.c gconf-2.25.2.new/defaults/gconf-defaults.c
--- gconf-2.25.2/defaults/gconf-defaults.c	2009-02-17 01:15:19.000000000 +0100
+++ gconf-2.25.2.new/defaults/gconf-defaults.c	2009-02-17 11:12:39.000000000 +0100
@@ -507,7 +507,7 @@
 	else {
 		annotation_key = "org.gnome.gconf.defaults.set-system.prefix"; 
 		default_action = "org.gnome.gconf.defaults.set-system";
-		dest_address = "xml:merged:" SYSGCONFDIR "/gconf.xml.system";
+		dest_address = "xml:merged:" SYSGCONFDIR "/gconf.xml.defaults";
 	}
 
 	for (i = 0; includes[i]; i++) {
