diff -ur fcron-3.0.1.orig/configure.in fcron-3.0.1/configure.in
--- fcron-3.0.1.orig/configure.in	2006-01-10 01:21:24.000000000 +0100
+++ fcron-3.0.1/configure.in	2006-05-06 18:48:04.000000000 +0200
@@ -259,31 +259,15 @@
     AC_MSG_ERROR(Need ETCDIR.)
     ;;
   yes)
-    if test -d "/etc"; then
-      etcdir=/etc
-      AC_MSG_RESULT(/etc)
-    else
-      AC_MSG_ERROR([
-Directory /etc does not exist: try option --with-etcdir=PATH])
-    fi
+    AC_MSG_RESULT(/etc)
     ;;
   *)
-    if test -d "$withval"; then
-      etcdir="$withval"
-      AC_MSG_RESULT($withval)
-    else
-      AC_MSG_ERROR([
-Directory $withval does not exist])
-    fi
+    etcdir="$withval"
+    AC_MSG_RESULT($withval)
     ;;
   esac ],
-  if test -d "/etc"; then
-    etcdir="/etc"
-    AC_MSG_RESULT(/etc)
-  else
-    AC_MSG_ERROR([
-Directory /etc does not exist: try option --with-etcdir=PATH])
-  fi
+  etcdir="/etc"
+  AC_MSG_RESULT(/etc)
 )
 ETC="$etcdir"
 AC_DEFINE_UNQUOTED(ETC, "$etcdir")
