diff -uNr soundkonverter-0.3.10b/src/metadata/audible/audibleproperties.cpp soundkonverter-0.3.10/src/metadata/audible/audibleproperties.cpp
--- soundkonverter-0.3.10b/src/metadata/audible/audibleproperties.cpp	2009-08-06 03:28:50.000000000 +0200
+++ soundkonverter-0.3.10/src/metadata/audible/audibleproperties.cpp	2009-08-06 12:48:48.000000000 +0200
@@ -22,7 +22,7 @@
  *   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,            *
  *   MA  02110-1301  USA                                                   *
  ***************************************************************************/
-
+#include<stdio.h>
 #include "audibleproperties.h"
 
 #include <taglib/tstring.h>
diff -uNr soundkonverter-0.3.10b/src/metadata/audible/audibleproperties.cpp~ soundkonverter-0.3.10/src/metadata/audible/audibleproperties.cpp~
--- soundkonverter-0.3.10b/src/metadata/audible/audibleproperties.cpp~	1970-01-01 01:00:00.000000000 +0100
+++ soundkonverter-0.3.10/src/metadata/audible/audibleproperties.cpp~	2007-12-17 16:53:19.000000000 +0100
@@ -0,0 +1,84 @@
+/***************************************************************************
+    copyright            : (C) 2005 by Martin Aumueller
+    email                : aumuell@reserv.at
+
+    copyright            : (C) 2005 by Andy Leadbetter
+    email                : andrew.leadbetter@gmail.com
+                           (original mp4 implementation)
+ ***************************************************************************/
+
+/***************************************************************************
+ *   This library is free software; you can redistribute it and/or modify  *
+ *   it  under the terms of the GNU Lesser General Public License version  *
+ *   2.1 as published by the Free Software Foundation.                     *
+ *                                                                         *
+ *   This library is distributed in the hope that it will be useful, but   *
+ *   WITHOUT ANY WARRANTY; without even the implied warranty of            *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU     *
+ *   Lesser General Public License for more details.                       *
+ *                                                                         *
+ *   You should have received a copy of the GNU Lesser General Public      *
+ *   License along with this library; if not, write to the Free Software   *
+ *   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,            *
+ *   MA  02110-1301  USA                                                   *
+ ***************************************************************************/
+
+#include "audibleproperties.h"
+
+#include <taglib/tstring.h>
+
+#include "taglib_audiblefile.h"
+
+#include <netinet/in.h> // ntohl
+
+using namespace TagLib;
+
+
+////////////////////////////////////////////////////////////////////////////////
+// public members
+////////////////////////////////////////////////////////////////////////////////
+
+Audible::Properties::Properties(Properties::ReadStyle style) : AudioProperties(style)
+{
+    m_length = 0;
+    m_bitrate = 0;
+    m_sampleRate = 0;
+    m_channels = 0;
+}
+
+Audible::Properties::~Properties()
+{
+}
+
+int Audible::Properties::length() const
+{
+    return m_length;
+}
+
+int Audible::Properties::bitrate() const
+{
+    return m_bitrate;
+}
+
+int Audible::Properties::sampleRate() const
+{
+    return m_sampleRate;
+}
+
+int Audible::Properties::channels() const
+{
+    return m_channels;
+}
+
+#define LENGTH_OFF 61
+
+void Audible::Properties::readAudibleProperties( FILE *fp, int off )
+{
+    fseek(fp, off+LENGTH_OFF, SEEK_SET );
+    fread(&m_length, sizeof(m_length), 1, fp);
+    m_length = ntohl(m_length);
+    //fprintf(stderr, "len (sec): %d\n", m_length);
+    m_bitrate = 0;
+    m_sampleRate = 0;
+    m_channels = 1;
+}
diff -uNr soundkonverter-0.3.10b/src/metadata/audible/audibletag.cpp soundkonverter-0.3.10/src/metadata/audible/audibletag.cpp
--- soundkonverter-0.3.10b/src/metadata/audible/audibletag.cpp	2009-08-06 03:28:50.000000000 +0200
+++ soundkonverter-0.3.10/src/metadata/audible/audibletag.cpp	2009-08-06 03:30:14.000000000 +0200
@@ -28,6 +28,7 @@
 #include <taglib/tag.h>
 
 #include <netinet/in.h> // ntohl
+#include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
diff -uNr soundkonverter-0.3.10b/src/metadata/audible/taglib_audiblefile.cpp soundkonverter-0.3.10/src/metadata/audible/taglib_audiblefile.cpp
--- soundkonverter-0.3.10b/src/metadata/audible/taglib_audiblefile.cpp	2009-08-06 03:28:50.000000000 +0200
+++ soundkonverter-0.3.10/src/metadata/audible/taglib_audiblefile.cpp	2009-08-06 03:29:44.000000000 +0200
@@ -28,7 +28,7 @@
 #include "audibletag.h"
 #include <taglib/tfile.h>
 #include <taglib/audioproperties.h>
-
+#include<stdio.h>
 namespace TagLib {
 ////////////////////////////////////////////////////////////////////////////////
 // public members
diff -uNr soundkonverter-0.3.10b/src/metadata/audible/taglib_audiblefiletyperesolver.cpp soundkonverter-0.3.10/src/metadata/audible/taglib_audiblefiletyperesolver.cpp
--- soundkonverter-0.3.10b/src/metadata/audible/taglib_audiblefiletyperesolver.cpp	2009-08-06 03:28:50.000000000 +0200
+++ soundkonverter-0.3.10/src/metadata/audible/taglib_audiblefiletyperesolver.cpp	2009-08-06 03:29:57.000000000 +0200
@@ -21,7 +21,7 @@
 
 #include "taglib_audiblefiletyperesolver.h"
 #include "taglib_audiblefile.h"
-
+#include<stdio.h>
 #include <string.h>
 
 TagLib::File *AudibleFileTypeResolver::createFile(const char *fileName,
diff -uNr soundkonverter-0.3.10b/src/metadata/wav/wavfile.cpp soundkonverter-0.3.10/src/metadata/wav/wavfile.cpp
--- soundkonverter-0.3.10b/src/metadata/wav/wavfile.cpp	2009-08-06 03:28:50.000000000 +0200
+++ soundkonverter-0.3.10/src/metadata/wav/wavfile.cpp	2009-08-06 03:30:49.000000000 +0200
@@ -22,7 +22,7 @@
  *   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,            *
  *   MA  02110-1301  USA                                                   *
  ***************************************************************************/
-
+#include<stdio.h>
 #include "wavfile.h"
 
 #include <taglib/tfile.h>
diff -uNr soundkonverter-0.3.10b/src/metadata/wav/wavfiletyperesolver.cpp soundkonverter-0.3.10/src/metadata/wav/wavfiletyperesolver.cpp
--- soundkonverter-0.3.10b/src/metadata/wav/wavfiletyperesolver.cpp	2009-08-06 03:28:50.000000000 +0200
+++ soundkonverter-0.3.10/src/metadata/wav/wavfiletyperesolver.cpp	2009-08-06 03:30:57.000000000 +0200
@@ -21,7 +21,7 @@
 
 #include "wavfiletyperesolver.h"
 #include "wavfile.h"
-
+#include<stdio.h>
 #include <string.h>
 
 TagLib::File *WavFileTypeResolver::createFile(const char *fileName,
diff -uNr soundkonverter-0.3.10b/src/metadata/wav/wavproperties.cpp soundkonverter-0.3.10/src/metadata/wav/wavproperties.cpp
--- soundkonverter-0.3.10b/src/metadata/wav/wavproperties.cpp	2009-08-06 03:28:50.000000000 +0200
+++ soundkonverter-0.3.10/src/metadata/wav/wavproperties.cpp	2009-08-06 03:30:44.000000000 +0200
@@ -26,7 +26,7 @@
 #include "wavproperties.h"
 
 #include <taglib/tstring.h>
-
+#include<stdio.h>
 #include "wavfile.h"
 
 #include <netinet/in.h> // ntohl
