diff -wbBur pppusage-0.2.5/Makefile.PL pppusage-0.2.5.my/Makefile.PL
--- pppusage-0.2.5/Makefile.PL	2003-12-20 00:56:32.000000000 +0300
+++ pppusage-0.2.5.my/Makefile.PL	2007-01-19 18:27:14.000000000 +0300
@@ -107,7 +107,7 @@
 		bin/pppusage.raw > bin/pppusage
 
 inst_conf: etc/pppusagerc
-	test -f $(SYSCONFDIR)/pppusagerc || cp etc/pppusagerc $(SYSCONFDIR)/
+	test -f $(SYSCONFDIR)/pppusagerc || install -D -m0644 etc/pppusagerc $(DESTDIR)$(SYSCONFDIR)/pppusagerc
 
 clean_bin:
 	rm -f bin/pppusage
