diff -Naur xf86-input-tslib-0.0.5/src/tslib.c xf86-input-tslib-0.0.5.new/src/tslib.c
--- xf86-input-tslib-0.0.5/src/tslib.c	2008-08-19 06:48:11.000000000 +0000
+++ xf86-input-tslib-0.0.5.new/src/tslib.c	2009-04-21 03:15:49.000000000 +0000
@@ -206,7 +206,9 @@
 
 		if (InitValuatorClassDeviceStruct(device,
 						  2,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION)<3
 						  xf86GetMotionEvents,
+#endif
 						  0, Absolute) == FALSE) {
 			ErrorF("unable to allocate Valuator class device\n");
 			return !Success;
