diff -dur lxlauncher-0.2.1.orig/src/Makefile.in lxlauncher-0.2.1/src/Makefile.in
--- lxlauncher-0.2.1.orig/src/Makefile.in	2009-07-07 14:46:41.000000000 +0200
+++ lxlauncher-0.2.1/src/Makefile.in	2010-02-17 13:34:43.000000000 +0100
@@ -101,7 +101,7 @@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 INSTOBJEXT = @INSTOBJEXT@
 INTLLIBS = @INTLLIBS@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ -lX11
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LTLIBOBJS = @LTLIBOBJS@
