From cb99b126dd90a8460c5bd4a837fdb7505658ba52 Mon Sep 17 00:00:00 2001
From: martyj19 <martyj19@comcast.net>
Date: Mon, 13 Jul 2009 16:46:09 +0000
Subject: [PATCH 02/50] Fix SUSE lint warnings

---
 src/lxlauncher.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/lxlauncher.c b/src/lxlauncher.c
index 7dab830..034d308 100644
--- a/src/lxlauncher.c
+++ b/src/lxlauncher.c
@@ -311,6 +311,7 @@ static gboolean on_scroll( GtkAdjustment* adj, PageData* data )
 {
     // Dirty hacks used to force pseudo-transparent background
     gtk_widget_queue_draw( data->table );
+    return TRUE;
 }
 
 // Dirty hacks used to reduce unnecessary redrew during scroll
@@ -674,6 +675,7 @@ gchar* get_xdg_config_file(const char *name) {
 	}
 	free(file);
     }
+    return NULL;
 }
 
 int main(int argc, char** argv)
-- 
1.6.6

