From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Mon, 6 Jun 2022 00:30:08 +0200
Subject: [PATCH] meson: Allow building with system orc

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 9ea00429bcb5..0481dbcbe512 100644
--- a/meson.build
+++ b/meson.build
@@ -112,7 +112,7 @@ elif build_system == 'darwin'
   subproject('macos-bison-binary')
 endif
 
-orc_subproject = subproject('orc', required: get_option('orc'))
+orc_subproject = subproject('orc', required: false)
 
 foreach custom_subproj: get_option('custom_subprojects').split(',')
     if custom_subproj != ''
