From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Mon, 6 Jun 2022 00:29:08 +0200
Subject: [PATCH] meson: Allow building with system orc

---
 meson.build | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 617ba8a41461..509565e90891 100644
--- a/meson.build
+++ b/meson.build
@@ -112,7 +112,13 @@ elif build_system == 'darwin'
   subproject('macos-bison-binary')
 endif
 
-orc_subproject = subproject('orc', required: get_option('orc'))
+orc_subproject_required = get_option('orc')
+if orc_subproject_required.allowed()
+  # Download if possible, otherwise use system orc
+  orc_subproject_required = false
+endif
+
+orc_subproject = subproject('orc', required: orc_subproject_required)
 
 foreach custom_subproj: get_option('custom_subprojects').split(',')
     if custom_subproj != ''
