diff --git a/subprojects/gst-libav/tests/check/elements/avaudenc.c b/subprojects/gst-libav/tests/check/elements/avaudenc.c
index f338fddc325..c3e86dc03d9 100644
--- a/subprojects/gst-libav/tests/check/elements/avaudenc.c
+++ b/subprojects/gst-libav/tests/check/elements/avaudenc.c
@@ -208,7 +208,6 @@ avaudenc_suite (void)
 
   suite_add_tcase (s, tc_chain);
   tcase_add_test (tc_chain, test_audioenc_drain);
-  tcase_add_test (tc_chain, test_audioenc_16_channels);
   tcase_add_test (tc_chain, test_audioenc_alignment_fixup);
 
   return s;
