diff -u -r tellico-1.3.5.orig/src/translators/pilotdb/libflatfile/DB.cpp tellico-1.3.5/src/translators/pilotdb/libflatfile/DB.cpp
--- tellico-1.3.5.orig/src/translators/pilotdb/libflatfile/DB.cpp	2009-02-14 13:24:52.000000000 -0700
+++ tellico-1.3.5/src/translators/pilotdb/libflatfile/DB.cpp	2009-03-05 03:54:57.000000000 -0700
@@ -242,7 +242,7 @@
            kdDebug() << "list view is corrupt" << endl;
         }
         // Determine the length of the name string.
-        pi_char_t* null_ptr = reinterpret_cast<pi_char_t*>
+        const pi_char_t* null_ptr = reinterpret_cast<const pi_char_t*>
             (memchr(chunk.data() + 4, 0, 32));
         if (null_ptr)
             lv.name = std::string((char *) (chunk.data() + 4),
Only in tellico-1.3.5/src/translators/pilotdb/libflatfile: DB.cpp.orig
