diff -Naur xfmedia-0.9.1.orig/src/remote.c xfmedia-0.9.1/src/remote.c
--- xfmedia-0.9.1.orig/src/remote.c	2005-09-05 00:56:37.000000000 -0400
+++ xfmedia-0.9.1/src/remote.c	2006-04-08 00:27:18.000000000 -0400
@@ -55,6 +55,10 @@
 #define DBUS_REQUEST_NAME_REPLY_PRIMARY_OWNER DBUS_SERVICE_REPLY_PRIMARY_OWNER
 #endif
 
+#ifndef DBUS_NAME_FLAG_PROHIBIT_REPLACEMENT
+#define DBUS_NAME_FLAG_PROHIBIT_REPLACEMENT DBUS_NAME_FLAG_DO_NOT_QUEUE
+#endif
+
 #endif  /* HAVE_DBUS */
 
 #include <libxfce4util/libxfce4util.h>
