--- js/src/ion/shared/IonAssemblerBufferWithConstantPools.h.orig	2013-06-21 16:17:35.029032287 -0500
+++ js/src/ion/shared/IonAssemblerBufferWithConstantPools.h	2013-06-21 16:18:05.842731689 -0500
@@ -675,7 +675,7 @@
         JS_ASSERT(perforatedNode != NULL);
         if (numDumps >= (1<<logBasePoolInfo) && (numDumps & (numDumps-1)) == 0) {
             // need to resize.
-            poolInfo = static_cast<PoolInfo*>(realloc_(poolInfo, sizeof(PoolInfo) * numDumps,
+            poolInfo = static_cast<PoolInfo*>(this->realloc_(poolInfo, sizeof(PoolInfo) * numDumps,
                                                        sizeof(PoolInfo) * numDumps * 2));
             if (poolInfo == NULL) {
                 this->fail_oom();
