diff -r f1af606531f5 xulrunner/installer/Makefile.in
--- a/xulrunner/installer/Makefile.in	Sat Nov 22 20:22:22 2008 +0100
+++ b/xulrunner/installer/Makefile.in	Mon Dec 01 22:30:12 2008 -0500
@@ -116,6 +118,7 @@
 	-e "s|%includedir%|$(includedir)|" \
 	-e "s|%idldir%|$(idldir)|" \
 	-e "s|%sdkdir%|$(sdkdir)|" \
+	-e "s|%libdir%|$(installdir)|" \
 	-e "s|%MOZ_APP_NAME%|$(MOZ_APP_NAME)|" \
 	-e "s|%MOZILLA_VERSION%|$(MOZ_APP_VERSION)|" \
 	-e "s|%WCHAR_CFLAGS%|$(WCHAR_CFLAGS)|" \
diff -r f1af606531f5 xulrunner/installer/libxul-embedding-unstable.pc.in
--- a/xulrunner/installer/libxul-embedding-unstable.pc.in	Sat Nov 22 20:22:22 2008 +0100
+++ b/xulrunner/installer/libxul-embedding-unstable.pc.in	Mon Dec 01 22:30:12 2008 -0500
@@ -1,11 +1,12 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
 includedir=%includedir%
-idldir=%idldir%
+idldir=%idldir%/unstable
 includetype=unstable
 
 Name: libxul-embedding-unstable
 Description: Static library for version-independent embedding of the Mozilla runtime (unstable API)
 Version: %MOZILLA_VERSION%
+Requires: %NSPR_NAME% >= %NSPR_VERSION%
 Libs: -L${sdkdir}/lib -lxpcomglue
 Cflags: -DXPCOM_GLUE -I${includedir}/${includetype} %WCHAR_CFLAGS%
diff -r f1af606531f5 xulrunner/installer/libxul-embedding.pc.in
--- a/xulrunner/installer/libxul-embedding.pc.in	Sat Nov 22 20:22:22 2008 +0100
+++ b/xulrunner/installer/libxul-embedding.pc.in	Mon Dec 01 22:30:12 2008 -0500
@@ -1,7 +1,7 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
 includedir=%includedir%
-idldir=%idldir%
+idldir=%idldir%/stable
 includetype=stable
 
 # Note: the default pkg-config cflags will only list include directories
@@ -11,5 +11,6 @@
 Name: libxul-embedding
 Description: Static library for version-independent embedding of the Mozilla runtime
 Version: %MOZILLA_VERSION%
-Libs: -L${sdkdir}/lib -lxpcomglue
+Requires: %NSPR_NAME% >= %NSPR_VERSION%
+Libs: -L${sdkdir}/lib -lxpcomglue
 Cflags: -DXPCOM_GLUE -I${includedir}/${includetype} %WCHAR_CFLAGS%
diff -r f1af606531f5 xulrunner/installer/libxul-unstable.pc.in
--- a/xulrunner/installer/libxul-unstable.pc.in	Sat Nov 22 20:22:22 2008 +0100
+++ b/xulrunner/installer/libxul-unstable.pc.in	Mon Dec 01 22:30:12 2008 -0500
@@ -1,7 +1,8 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
-idldir=%idldir%
+idldir=%idldir%/unstable
 includetype=unstable
 
 Name: libxul
diff -r f1af606531f5 xulrunner/installer/libxul.pc.in
--- a/xulrunner/installer/libxul.pc.in	Sat Nov 22 20:22:22 2008 +0100
+++ b/xulrunner/installer/libxul.pc.in	Mon Dec 01 22:30:12 2008 -0500
@@ -1,7 +1,8 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
-idldir=%idldir%
+idldir=%idldir%/stable
 includetype=stable
 
 # Note: the default pkg-config includes will only list include directories
diff -r f1af606531f5 xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in
--- a/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in	Sat Nov 22 20:22:22 2008 +0100
+++ b/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in	Mon Dec 01 22:30:12 2008 -0500
@@ -1,5 +1,6 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
 
 Name: mozilla-gtkembedmoz
diff -r f1af606531f5 xulrunner/installer/mozilla-gtkmozembed.pc.in
--- a/xulrunner/installer/mozilla-gtkmozembed.pc.in	Sat Nov 22 20:22:22 2008 +0100
+++ b/xulrunner/installer/mozilla-gtkmozembed.pc.in	Mon Dec 01 22:30:12 2008 -0500
@@ -1,5 +1,6 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
 
 Name: mozilla-gtkembedmoz
diff -r f1af606531f5 xulrunner/installer/mozilla-js.pc.in
--- a/xulrunner/installer/mozilla-js.pc.in	Sat Nov 22 20:22:22 2008 +0100
+++ b/xulrunner/installer/mozilla-js.pc.in	Mon Dec 01 22:30:12 2008 -0500
@@ -6,5 +6,5 @@
 Description: The Mozilla JavaScript Library
 Version: %MOZILLA_VERSION%
 Requires: %NSPR_NAME% >= %NSPR_VERSION%
-Libs: -L${sdkdir}/lib -lmozjs
-Cflags: -I${includedir}/stable -DXP_UNIX -DJS_THREADSAFE
+Libs: -L${sdkdir}/lib -lmozjs
+Cflags: -I${includedir}/stable -I${includedir}/js -DXP_UNIX -DJS_THREADSAFE
diff -r f1af606531f5 xulrunner/installer/mozilla-plugin.pc.in
--- a/xulrunner/installer/mozilla-plugin.pc.in	Sat Nov 22 20:22:22 2008 +0100
+++ b/xulrunner/installer/mozilla-plugin.pc.in	Mon Dec 01 22:30:12 2008 -0500
@@ -6,4 +6,4 @@
 Description: Mozilla Plug-In API
 Version: %MOZILLA_VERSION%
 Requires: %NSPR_NAME% >= %NSPR_VERSION%
-Cflags: -I${includedir}/stable -DXP_UNIX %ojidef%
+Cflags: -I${includedir}/stable -DXP_UNIX %ojidef% -I${includedir}/java -I${includedir}/plugin
