diff -Nur mozilla-release.orig/browser/installer/Makefile.in mozilla-release/browser/installer/Makefile.in
--- mozilla-release.orig/browser/installer/Makefile.in	2012-07-17 16:19:29.480356991 +0000
+++ mozilla-release/browser/installer/Makefile.in	2012-07-17 17:32:41.250937293 +0000
@@ -17,7 +17,7 @@
 
 # Some files have been already bundled with xulrunner
 ifndef MOZ_MULET
-MOZ_PKG_FATAL_WARNINGS = 1
+MOZ_PKG_FATAL_WARNINGS = 0
 endif
 
 DEFINES += -DAB_CD=$(AB_CD) -DMOZ_APP_NAME=$(MOZ_APP_NAME) -DPREF_DIR=$(PREF_DIR)
