diff -up mozilla/toolkit/mozapps/update/src/updater/module.ver mozilla/toolkit/mozapps/update/src/updater/module
diff -up mozilla/xulrunner/installer/Makefile.in.ver mozilla/xulrunner/installer/Makefile.in
--- mozilla/xulrunner/installer/Makefile.in.ver	2007-12-14 09:51:34.000000000 +0100
+++ mozilla/xulrunner/installer/Makefile.in	2007-12-14 09:52:03.000000000 +0100
@@ -44,6 +44,8 @@ VPATH		= @srcdir@
 
 include $(DEPTH)/config/autoconf.mk
 
+MOZ_APP_VERSION="1.9.1"
+
 NO_PKG_FILES = \
 	xulrunner-config \
 	regchrome* \
