diff -Naur exaile_0.2.13-old/Makefile exaile_0.2.13/Makefile
--- exaile_0.2.13-old/Makefile	2008-04-02 11:42:17.000000000 +1000
+++ exaile_0.2.13/Makefile	2008-08-22 23:37:24.000000000 +1000
@@ -68,6 +68,8 @@
 	cd $(DESTDIR)$(PREFIX)/bin && \
 	  /bin/echo -e \
 	    "#!/bin/sh\n" \
+	    "export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/xulrunner-1.9\n" \
+	    "export MOZILLA_FIVE_HOME=/usr/lib/xulrunner-1.9\n" \
 	    "cd $(PREFIX)/share/exaile\n" \
 	    "exec python $(PREFIX)$(LIBDIR)/exaile/exaile.py \"\$$@\"" \
 	    > exaile && \
