From ede87f48af3f9eda4fc53f0d12ff1294e07d4e3b Mon Sep 17 00:00:00 2001
From: Andrew Stitcher <astitcher@apache.org>
Date: Thu, 20 Aug 2020 15:47:44 -0400
Subject: [PATCH] PROTON-2268: Fix the fix so it works everywhere

---
 c/tests/threaderciser.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/c/tests/threaderciser.c b/c/tests/threaderciser.c
index 5f2bb626..f787bda6 100644
--- a/c/tests/threaderciser.c
+++ b/c/tests/threaderciser.c
@@ -57,6 +57,7 @@
 #include <proton/netaddr.h>
 #include <proton/proactor.h>
 
+#include <inttypes.h>
 #include <stdarg.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -94,7 +95,7 @@ static void debug_impl(const char *fmt, ...) {
   char msg[256];
   char *i = msg;
   char *end = i + sizeof(msg);
-  i += assert_no_err(snprintf(i, end-i, "(%lx) ", (uintptr_t) pthread_self()));
+  i += assert_no_err(snprintf(i, end-i, "(%" PRIxPTR ") ", (uintptr_t) pthread_self()));
   if (i < end) {
     va_list ap;
     va_start(ap, fmt);
-- 
2.28.0

