diff -Nru speech_tools.orig/config/compilers/jdk.mak speech_tools/config/compilers/jdk.mak
--- speech_tools.orig/config/compilers/jdk.mak	2001-04-04 06:55:32.000000000 -0500
+++ speech_tools/config/compilers/jdk.mak	2005-05-30 19:50:54.000000000 -0500
@@ -52,7 +52,7 @@
 endif
 endif
 
-JAVAFLAGS = -depend
+JAVAFLAGS =
 
 DEBUG_JAVAFLAGS = -g
 
diff -Nru speech_tools.orig/config/systems/Linux.mak speech_tools/config/systems/Linux.mak
--- speech_tools.orig/config/systems/Linux.mak	2001-04-04 06:55:32.000000000 -0500
+++ speech_tools/config/systems/Linux.mak	2005-05-30 19:52:06.000000000 -0500
@@ -40,13 +40,13 @@
 
 include $(EST)/config/systems/default.mak
 
-DEFAULT_JAVA_HOME=/usr/lib/jdk-1.1.6
-JAVA=/usr/bin/java
-JAVAC=/usr/bin/javac
-JAVAH=/usr/bin/javah
+DEFAULT_JAVA_HOME=/opt/java
+JAVA=/opt/java/bin/java
+JAVAC=/opt/java/bin/javac
+JAVAH=/opt/java/bin/javah
 
 TCL_LIBRARY = -ltcl
-OS_LIBS = -ldl 
+OS_LIBS = -ldl -lncurses 
 
 ## the native audio module for this type of system
 NATIVE_AUDIO_MODULE = LINUX16
diff -Nru speech_tools.orig/config/systems/default.mak speech_tools/config/systems/default.mak
--- speech_tools.orig/config/systems/default.mak	2004-01-30 18:04:02.000000000 -0600
+++ speech_tools/config/systems/default.mak	2005-05-30 19:50:54.000000000 -0500
@@ -40,7 +40,7 @@
 ###########################################################################
 ## Installation directories
 
-INSTALL_PREFIX=/usr/local
+INSTALL_PREFIX=/usr
 
 BINDIR=$(INSTALL_PREFIX)/bin
 LIBDIR=$(INSTALL_PREFIX)/lib
@@ -63,8 +63,8 @@
 ###########################################################################
 ## Where to find Enlightenment Speech Demon
 
-ESD_INCLUDE = /usr/local/include
-ESD_LIB = /usr/local/lib
+ESD_INCLUDE = /usr/include
+ESD_LIB = /usr/lib
 
 ###########################################################################
 ## Where to find X11
@@ -75,14 +75,14 @@
 ###########################################################################
 ## TCL support
 
-TCL_INCLUDE = /usr/local/include
-TCL_LIB     = /usr/local/lib
-TCL_LIBRARY = -ltcl7.6
+TCL_INCLUDE = /usr/include
+TCL_LIB     = /usr/lib
+TCL_LIBRARY = -ltcl
 
 ###########################################################################
 ## Efence library for malloc debugging
 
-EFENCE_LIB = /usr/local/lib
+EFENCE_LIB = /usr/lib
 
 ###########################################################################
 ## Commands.
--- a/config/config.in	2006-07-08 13:06:18.000000000 -0500
+++ b/config/config.in.new	2007-06-27 10:35:51.000000000 -0500
@@ -15,7 +15,7 @@
 ## You may need to set this explicitly if automounter or NFS
 ## side effects cause problems
 
-EST_HOME := $(shell (cd $(EST); pwd))
+EST_HOME := /usr
 
 ###########################################################################
 ## System type. 
@@ -28,14 +28,14 @@
 ## 
 ## Examples: sparc_SunOS5 intel_Linux2.0
 
-SYSTEM_TYPE=$(MACHINETYPE)_$(OSTYPE)$(OSREV)
+SYSTEM_TYPE=Linux
 
 ###########################################################################
 ## Compiler.
 ## The definitions are in compilers/$(COMPILER).mak
 ## Examples: gcc suncc egcs gcc28
 
-COMPILER=@COMPILERTYPE@
+COMPILER=gcc32
 
 ###########################################################################
 ## Java system to use if you include the Java interface.
@@ -61,7 +61,7 @@
 # VERBOSE=1
 #DEBUG=1
 # PROFILE=gprof
-#SHARED=2
+SHARED=1
 
 ## Directory specific selections which override the above
 
@@ -94,7 +94,7 @@
 ## It may not work under all systems, so may be optionally omitted.
 INCLUDE_MODULES += EDITLINE
 
-TERMCAPLIB = @TERMCAPLIB@
+TERMCAPLIB = 
 # speech recognition
 #INCLUDE_MODULES += ASR
 
@@ -165,8 +165,8 @@
 CONFIG_WRAPPER_LANGUAGES = PYTHON PERL5
 
 # Language specific includes should be set to correct site paths
-CONFIG_PYTHON_INCLUDES= -I/usr/include/python2.2/
-CONFIG_PERL_INCLUDES= -I/usr/lib/perl5/5.8.3/i386-linux-thread-multi/CORE/
+CONFIG_PYTHON_INCLUDES= -I/usr/include/python2.5/
+CONFIG_PERL_INCLUDES= -I/usr/lib/perl5/5.8.3/i686-linux-thread-multi/CORE/
 #OR OLD#CONFIG_PERL_INCLUDES= -I/usr/lib/perl5/5.6.1/i386-linux/CORE/
 ##
 ###################################################################
--- a/config/compilers/gcc_defaults.mak	2006-08-05 10:37:59.000000000 -0500
+++ b/config/compilers/gcc_defaults.mak.new	2007-06-27 10:43:51.000000000 -0500
@@ -76,9 +76,9 @@
 PROFILE_gprof_CXXFLAGS  = -pg
 PROFILE_gprof_LINKFLAGS = -pg
 
-SHARED_CCFLAGS  = -fPIC -fno-shared-data
-SHARED_CXXFLAGS  = -fPIC -fno-shared-data
-SHARED_LINKFLAGS = -fno-shared-data
+SHARED_CCFLAGS  = -fPIC 
+SHARED_CXXFLAGS  = -fPIC 
+SHARED_LINKFLAGS = 
 
 ifndef GCC_MAKE_SHARED_LIB
     MAKE_SHARED_LIB = $(CXX) -shared -fno-shared-data -o XXX
