diff -Naur festival-orig/src/modules/Text/text_modes.cc festival/src/modules/Text/text_modes.cc
--- festival-orig/src/modules/Text/text_modes.cc	2004-09-30 07:04:18.000000000 -0600
+++ festival/src/modules/Text/text_modes.cc	2008-01-07 22:32:44.000000000 -0600
@@ -60,7 +60,7 @@
 {
 
     volatile EST_String tmpname = make_tmp_filename();
-    volatile EST_String inname = get_c_string(filename);
+    volatile EST_String inname(get_c_string(filename));
     volatile EST_String filter;
     volatile EST_TokenStream ts;
     volatile LISP func;
