diff -Nru festival.orig/config/config.in festival/config/config.in
--- festival.orig/config/config.in	2004-06-21 15:52:42.000000000 -0500
+++ festival/config/config.in	2005-05-30 19:50:45.000000000 -0500
@@ -20,7 +20,7 @@
 ## You may need to set this explicitly if automounter or NFS
 ## side effects cause problems
 
-FESTIVAL_HOME := $(shell (cd $(TOP); pwd))
+FESTIVAL_HOME := /usr
 
 ###########################################################################
 ## Feature selection.
diff -Nru festival.orig/config/project.mak festival/config/project.mak
--- festival.orig/config/project.mak	2004-07-12 10:39:37.000000000 -0500
+++ festival/config/project.mak	2005-05-30 19:50:45.000000000 -0500
@@ -112,6 +112,5 @@
 DOCXX_DIRS = $(TOP)/src 
 MODULE_TO_DOCXX = perl $(TOP)/src/modules/utilities/extract_module_doc++.prl
 
-FTLIBDIR = $(FESTIVAL_HOME)/lib 
-
+FTLIBDIR = /usr/share/festival
 
diff -Nru festival.orig/config/systems/Linux.mak festival/config/systems/Linux.mak
--- festival.orig/config/systems/Linux.mak	2001-04-04 06:55:32.000000000 -0500
+++ festival/config/systems/Linux.mak	2005-05-30 19:51:28.000000000 -0500
@@ -40,13 +40,13 @@
 
 include $(EST)/config/systems/default.mak
 
-DEFAULT_JAVA_HOME=/usr/lib/jdk-1.1.6
-JAVA=/usr/bin/java
-JAVAC=/usr/bin/javac
-JAVAH=/usr/bin/javah
+DEFAULT_JAVA_HOME=/opt/java
+JAVA=/opt/java/bin/java
+JAVAC=/opt/java/bin/javac
+JAVAH=/opt/java/bin/javah
 
 TCL_LIBRARY = -ltcl
-OS_LIBS = -ldl 
+OS_LIBS = -ldl -lncurses
 
 ## the native audio module for this type of system
 NATIVE_AUDIO_MODULE = LINUX16
diff -Nru festival.orig/config/systems/default.mak festival/config/systems/default.mak
--- festival.orig/config/systems/default.mak	2001-04-04 06:55:32.000000000 -0500
+++ festival/config/systems/default.mak	2005-05-30 19:50:45.000000000 -0500
@@ -40,7 +40,7 @@
 ###########################################################################
 ## Installation directories
 
-INSTALL_PREFIX=/usr/local
+INSTALL_PREFIX=/usr
 
 BINDIR=$(INSTALL_PREFIX)/bin
 LIBDIR=$(INSTALL_PREFIX)/lib
@@ -63,8 +63,8 @@
 ###########################################################################
 ## Where to find Enlightenment Speech Demon
 
-ESD_INCLUDE = /usr/local/include
-ESD_LIB = /usr/local/lib
+ESD_INCLUDE = /usr/include
+ESD_LIB = /usr/lib
 
 ###########################################################################
 ## Where to find X11
@@ -75,14 +75,14 @@
 ###########################################################################
 ## TCL support
 
-TCL_INCLUDE = /usr/local/include
-TCL_LIB     = /usr/local/lib
-TCL_LIBRARY = -ltcl7.6
+TCL_INCLUDE = /usr/include
+TCL_LIB     = /usr/lib
+TCL_LIBRARY = -ltcl
 
 ###########################################################################
 ## Efence library for malloc debugging
 
-EFENCE_LIB = /usr/local/lib
+EFENCE_LIB = /usr/lib
 
 ###########################################################################
 ## Commands.
