--- speech_tools/config/compilers/egcs.mak	2001-04-04 13:55:32.000000000 +0200
+++ speech_tools/config/compilers/egcs.mak	2007-08-28 23:08:01.000000000 +0200
@@ -54,7 +54,7 @@
 CXX=$(EGCS_CXX)
 
 ifndef EGCS_MAKE_SHARED_LIB
-    MAKE_SHARED_LIB = gcc -shared -fno-shared-data -o XXX -Xlinker -h -Xlinker XXX
+    MAKE_SHARED_LIB = gcc -shared -o XXX -Xlinker -h -Xlinker XXX
 else
     MAKE_SHARED_LIB = $(GCC_MAKE_SHARED_LIB)
 endif
--- speech_tools/config/compilers/gcc_defaults.mak	2006-08-05 17:37:59.000000000 +0200
+++ speech_tools/config/compilers/gcc_defaults.mak	2007-08-28 23:08:30.000000000 +0200
@@ -76,12 +76,12 @@
 PROFILE_gprof_CXXFLAGS  = -pg
 PROFILE_gprof_LINKFLAGS = -pg
 
-SHARED_CCFLAGS  = -fPIC -fno-shared-data
-SHARED_CXXFLAGS  = -fPIC -fno-shared-data
-SHARED_LINKFLAGS = -fno-shared-data
+SHARED_CCFLAGS  = -fPIC
+SHARED_CXXFLAGS  = -fPIC
+SHARED_LINKFLAGS =
 
 ifndef GCC_MAKE_SHARED_LIB
-    MAKE_SHARED_LIB = $(CXX) -shared -fno-shared-data -o XXX
+    MAKE_SHARED_LIB = $(CXX) -shared -o XXX
 else
     MAKE_SHARED_LIB = $(GCC_MAKE_SHARED_LIB)
 endif
--- speech_tools/doc/estjavaapp.sgml	2001-04-04 13:55:32.000000000 +0200
+++ speech_tools/doc/estjavaapp.sgml	2007-08-28 23:08:43.000000000 +0200
@@ -105,7 +105,7 @@
 	&shellprompt; <userinput>cd tmpdir</userinput>
 	&shellprompt; <userinput>ar x ../libgcc.a</userinput>
 	&shellprompt; <userinput>cd ..</userinput>
-	&shellprompt; <userinput>gcc -shared -fno-shared-data -o libgcc.so.2.7.2 tmpdir/*.o</userinput>
+	&shellprompt; <userinput>gcc -shared -o libgcc.so.2.7.2 tmpdir/*.o</userinput>
 </screen>
 </para>
 </sect1>
--- speech_tools/stats/EST_DProbDist.cc	2004-09-30 14:53:36.000000000 +0200
+++ speech_tools/stats/EST_DProbDist.cc	2007-08-28 23:21:17.000000000 +0200
@@ -310,7 +310,7 @@
     if (type == tprob_discrete)
 	return 0;
     else
-	return (int)scounts.list.head();
+	return (long)scounts.list.head();
 }
 
 int EST_DiscreteProbDistribution::item_end(int idx) const
@@ -326,7 +326,7 @@
     if (type == tprob_discrete)
 	return ++idx;
     else
-	return (int)next((EST_Litem *)idx);
+	return (long)next((EST_Litem *)idx);
 }
 
 const EST_String &EST_DiscreteProbDistribution::item_name(int idx) const
