diff -Naur speech_tools-orig/base_class/EST_TSimpleMatrix.cc speech_tools/base_class/EST_TSimpleMatrix.cc
--- speech_tools-orig/base_class/EST_TSimpleMatrix.cc	2004-09-30 06:53:35.000000000 -0600
+++ speech_tools/base_class/EST_TSimpleMatrix.cc	2008-01-05 11:00:07.000000000 -0600
@@ -42,6 +42,7 @@
 
 #include "EST_TSimpleMatrix.h"
 #include "EST_TVector.h"
+#include <cstring>
 #include <fstream>
 #include <iostream>
 #include "EST_cutils.h"
diff -Naur speech_tools-orig/base_class/EST_TSimpleVector.cc speech_tools/base_class/EST_TSimpleVector.cc
--- speech_tools-orig/base_class/EST_TSimpleVector.cc	2006-07-06 06:57:18.000000000 -0600
+++ speech_tools/base_class/EST_TSimpleVector.cc	2008-01-05 11:00:32.000000000 -0600
@@ -42,6 +42,7 @@
 
 #include "EST_TSimpleVector.h"
 #include "EST_matrix_support.h"
+#include <cstring>
 #include <fstream>
 #include "EST_cutils.h"
 
