diff -NaurwB bochs-2.3.7.orig/cpu/ia_opcodes.h bochs-2.3.7/cpu/ia_opcodes.h
--- bochs-2.3.7.orig/cpu/ia_opcodes.h	2008-05-30 22:35:08.000000000 +0200
+++ bochs-2.3.7/cpu/ia_opcodes.h	2008-06-04 14:56:46.000000000 +0200
@@ -891,7 +891,7 @@
 bx_define_opcode(BX_IA_PF2ID_PqQq, BX_CPU_C::PF2ID_PqQq)
 bx_define_opcode(BX_IA_PF2IW_PqQq, BX_CPU_C::PF2IW_PqQq)
 bx_define_opcode(BX_IA_PFACC_PqQq, BX_CPU_C::PFACC_PqQq)
-bx_define_opcode(BX_IA_PFADD_PqQq, BX_CPU_C::BX_PFADD_PqQq)
+bx_define_opcode(BX_IA_PFADD_PqQq, BX_CPU_C::PFADD_PqQq)
 bx_define_opcode(BX_IA_PFCMPEQ_PqQq, BX_CPU_C::PFCMPEQ_PqQq)
 bx_define_opcode(BX_IA_PFCMPGE_PqQq, BX_CPU_C::PFCMPGE_PqQq)
 bx_define_opcode(BX_IA_PFCMPGT_PqQq, BX_CPU_C::PFCMPGT_PqQq)
diff -NaurwB bochs-2.3.7.orig/iodev/iodebug.h bochs-2.3.7/iodev/iodebug.h
--- bochs-2.3.7.orig/iodev/iodebug.h	2008-05-01 22:46:58.000000000 +0200
+++ bochs-2.3.7/iodev/iodebug.h	2008-06-04 14:45:50.000000000 +0200
@@ -18,7 +18,7 @@
   virtual void init(void);
   virtual void reset (unsigned type) {}
   static void mem_write(BX_CPU_C *cpu, bx_phy_address addr, unsigned len, void *data);
-  static void mem_read(BX_CPU_C *cpu, bx_phy_addressu addr, unsigned len, void *data);
+  static void mem_read(BX_CPU_C *cpu, bx_phy_address addr, unsigned len, void *data);
 
 private:
   static Bit32u read_handler(void *this_ptr, Bit32u address, unsigned io_len);
