diff -Naur dvd+rw-tools-7.0/growisofs.c dvd+rw-tools-7.0_new/growisofs.c
--- dvd+rw-tools-7.0/growisofs.c	2006-09-24 19:28:53.000000000 +0200
+++ dvd+rw-tools-7.0_new/growisofs.c	2008-01-08 00:56:57.000000000 +0100
@@ -418,6 +418,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <assert.h>
+#include <limits.h>
 #include "mp.h"
 
 #if defined(__unix) || defined(__unix__)
diff -Naur dvd+rw-tools-7.0/transport.hxx dvd+rw-tools-7.0_new/transport.hxx
--- dvd+rw-tools-7.0/transport.hxx	2006-09-23 13:29:07.000000000 +0200
+++ dvd+rw-tools-7.0_new/transport.hxx	2008-01-08 00:57:36.000000000 +0100
@@ -137,6 +137,7 @@
 #include <sys/wait.h>
 #include <sys/utsname.h>
 #include <scsi/sg.h>
+#include <limits.h>
 #if !defined(SG_FLAG_LUN_INHIBIT)
 # if defined(SG_FLAG_UNUSED_LUN_INHIBIT)
 #  define SG_FLAG_LUN_INHIBIT SG_FLAG_UNUSED_LUN_INHIBIT
