--- Makefile	2001-11-04 19:39:30.000000000 -0500
+++ Makefile.new	2004-01-21 09:52:17.000000000 -0500
@@ -10,15 +10,9 @@
 CC		= gcc
 ARCH           = $(shell uname -m)
 #
-ifeq "$(ARCH)" "alpha"
-  CFLAGS       = $(RPM_OPT_FLAGS) -Wall
-else
-ifeq "$(ARCH)" "ppc"
-  CFLAGS       = $(RPM_OPT_FLAGS) -fsigned-char -Wall
-else
-  CFLAGS       = $(RPM_OPT_FLAGS) -m486 -Wall
-endif
-endif
+
+CFLAGS		= #CFLAGS#
+
 LFLAGS         = -Xlinker -warn-common
 
 LIBS		= -lgdbm
@@ -26,10 +20,10 @@
 LLIBS		=	
 
 # where things go
-BINDIR		= /usr/bin
+BINDIR		= STARTDIR/pkg/usr/bin
 VACATION        = $(BINDIR)/vacation
 VACLOOK         = $(BINDIR)/vaclook
-MANDIR		= /usr/man/man
+MANDIR		= STARTDIR/pkg/usr/man/man
 MANEXT1		= 1
 
 VERSION 	= 1
@@ -53,7 +47,7 @@
 
 install:  all
 	install -s -m 755 $(BIN) $(VACATION)
-	install -s -m 755 vaclook $(VACLOOK)
+	install -m 755 vaclook $(VACLOOK)
 	install -m 444 vacation.man $(MANDIR)$(MANEXT1)/vacation.$(MANEXT1)
 	install -m 444 vaclook.man $(MANDIR)$(MANEXT1)/vaclook.$(MANEXT1)
 
