diff --git a/setup.py b/setup.py
index 6ff80ee..1eab62e 100644
--- a/setup.py
+++ b/setup.py
@@ -515,7 +515,7 @@ def cython_extensions_and_necessity():
                 + list(CYTHON_HELPER_C_FILES)
                 + core_c_files
                 + asm_files
-                + ["third_party/abseil-cpp/absl/log/initialize.cc"]
+                + ([] if BUILD_WITH_SYSTEM_ABSL else ["third_party/abseil-cpp/absl/log/initialize.cc"])
             ),
             include_dirs=list(EXTENSION_INCLUDE_DIRECTORIES),
             libraries=list(EXTENSION_LIBRARIES),
