--- ufraw-0.15.old/dcraw.cc	2008-12-23 07:19:41.000000000 +0000
+++ ufraw-0.15/dcraw.cc	2009-05-28 15:25:51.000000000 +0100
@@ -8401,7 +8401,7 @@
   argv[argc] = "";
   for (arg=1; (((opm = argv[arg][0]) - 2) | 2) == '+'; ) {
     opt = argv[arg++][1];
-    if ((cp = strchr (sp="nbrkStqmHAC", opt)))
+    if ((cp = (char*) strchr (sp="nbrkStqmHAC", opt)))
       for (i=0; i < "11411111142"[cp-sp]-'0'; i++)
 	if (!isdigit(argv[arg+i][0])) {
 	  dcraw_message (DCRAW_ERROR,_("Non-numeric argument to \"-%c\"\n"), opt);

