--- ./google-breakpad/src/processor/postfix_evaluator-inl.h~	2009-02-26 19:56:48.000000000 -0500
+++ ./google-breakpad/src/processor/postfix_evaluator-inl.h	2009-06-11 17:27:18.451971694 -0400
@@ -37,7 +37,7 @@
 #ifndef PROCESSOR_POSTFIX_EVALUATOR_INL_H__
 #define PROCESSOR_POSTFIX_EVALUATOR_INL_H__
 
-
+#include <cstdio>
 #include <sstream>
 
 #include "processor/postfix_evaluator.h"
--- ./google-breakpad/src/client/linux/handler/exception_handler.cc~	2009-02-26 19:56:49.000000000 -0500
+++ ./google-breakpad/src/client/linux/handler/exception_handler.cc	2009-06-11 20:54:54.667061888 -0400
@@ -44,6 +44,8 @@
 #include <ctime>
 #include <linux/limits.h>
 
+#include <cstdio>
+
 #include "client/linux/handler/exception_handler.h"
 #include "common/linux/guid_creator.h"
 #include "google_breakpad/common/minidump_format.h"
--- ./google-breakpad/src/client/linux/handler/minidump_generator.cc~	2009-02-26 19:56:49.000000000 -0500
+++ ./google-breakpad/src/client/linux/handler/minidump_generator.cc	2009-06-11 21:06:14.963661633 -0400
@@ -39,6 +39,7 @@
 #include <sys/wait.h>
 #include <string.h>
 
+#include <cstdio>
 #include <cstdlib>
 #include <cstring>
 #include <ctime>
