diff --git a/admin/iprange/Makefile b/admin/iprange/Makefile
index 13598d8..eb89fb9 100644
--- a/admin/iprange/Makefile
+++ b/admin/iprange/Makefile
@@ -7,7 +7,7 @@ OBJS     = iprange ip6range
 all: $(OBJS)
 
 %: %.c
-	$(CC) $(LDFLAGS) $(OPTIMIZE) -o $@ $^
+	$(CC) $(OPTIMIZE) $(CFLAGS) $(LDFLAGS) -o $@ $^
 
 clean:
 	rm -f $(OBJS) *.o *.a *~
diff --git a/dev/poll/Makefile b/dev/poll/Makefile
index fdee514..9be926c 100644
--- a/dev/poll/Makefile
+++ b/dev/poll/Makefile
@@ -5,7 +5,7 @@ INCLUDE  =
 OBJS     = poll
 
 poll: poll.c
-	$(CC) $(OPTIMIZE) $(DEFINE) $(INCLUDE) -o $@ $^
+	$(CC) $(OPTIMIZE) $(CFLAGS) $(LDFLAGS) $(DEFINE) $(INCLUDE) -o $@ $^
 
 clean:
 	rm -f $(OBJS) *.[oas] *~
diff --git a/dev/tcploop/Makefile b/dev/tcploop/Makefile
index 42a6259..e8a6ff7 100644
--- a/dev/tcploop/Makefile
+++ b/dev/tcploop/Makefile
@@ -5,7 +5,7 @@ INCLUDE  =
 OBJS     = tcploop
 
 tcploop: tcploop.c
-	$(CC) $(OPTIMIZE) $(DEFINE) $(INCLUDE) -o $@ $^
+	$(CC) $(OPTIMIZE) $(CFLAGS) $(LDFLAGS) $(DEFINE) $(INCLUDE) -o $@ $^
 
 clean:
 	rm -f $(OBJS) *.[oas] *~
