diff --git a/contrib/halog/Makefile b/contrib/halog/Makefile
index bf67c1f..e98a3fa 100644
--- a/contrib/halog/Makefile
+++ b/contrib/halog/Makefile
@@ -16,7 +16,7 @@ DEFINE   =
 OBJS     = halog
 
 halog: halog.c fgets2.c
-	$(CC) $(OPTIMIZE) $(DEFINE) -o $@ $(INCLUDE) $(SRC)/ebtree.c $(SRC)/eb32tree.c $(SRC)/eb64tree.c $(SRC)/ebmbtree.c $(SRC)/ebsttree.c $(SRC)/ebistree.c $(SRC)/ebimtree.c $^
+	$(CC) $(OPTIMIZE) $(CFLAGS) $(LDFLAGS) $(DEFINE) -o $@ $(INCLUDE) $(SRC)/ebtree.c $(SRC)/eb32tree.c $(SRC)/eb64tree.c $(SRC)/ebmbtree.c $(SRC)/ebsttree.c $(SRC)/ebistree.c $(SRC)/ebimtree.c $^
 
 clean:
 	rm -f $(OBJS) *.[oas]
diff --git a/contrib/ip6range/Makefile b/contrib/ip6range/Makefile
index 55849b6..1276b97 100644
--- a/contrib/ip6range/Makefile
+++ b/contrib/ip6range/Makefile
@@ -7,7 +7,7 @@ OBJS     = ip6range
 all: $(OBJS)
 
 %: %.c
-	$(CC) $(LDFLAGS) $(OPTIMIZE) -o $@ $^
+	$(CC) $(CFLAGS) $(LDFLAGS) $(OPTIMIZE) -o $@ $^
 
 clean:
 	rm -f $(OBJS) *.o *.a *~
diff --git a/contrib/iprange/Makefile b/contrib/iprange/Makefile
index b3c72ed..e92baf3 100644
--- a/contrib/iprange/Makefile
+++ b/contrib/iprange/Makefile
@@ -7,7 +7,7 @@ OBJS     = iprange
 all: $(OBJS)
 
 %: %.c
-	$(CC) $(LDFLAGS) $(OPTIMIZE) -o $@ $^
+	$(CC) $(CFLAGS) $(LDFLAGS) $(OPTIMIZE) -o $@ $^
 
 clean:
 	rm -f $(OBJS) *.o *.a *~
