diff -urN a/lib/Support/CMakeLists.txt b/lib/Support/CMakeLists.txt
--- a/lib/Support/CMakeLists.txt	2015-12-14 17:59:19.000000000 -0700
+++ b/lib/Support/CMakeLists.txt	2016-05-12 20:11:40.042874505 -0600
@@ -89,7 +89,6 @@
   StringRef.cpp
   SystemUtils.cpp
   TargetParser.cpp
-  ThreadPool.cpp
   Timer.cpp
   ToolOutputFile.cpp
   Triple.cpp
diff -urN a/unittests/Support/CMakeLists.txt b/unittests/Support/CMakeLists.txt
--- a/unittests/Support/CMakeLists.txt	2015-12-22 10:36:17.000000000 -0700
+++ b/unittests/Support/CMakeLists.txt	2016-05-12 21:31:22.669908242 -0600
@@ -41,7 +41,6 @@
   SwapByteOrderTest.cpp
   TargetRegistry.cpp
   ThreadLocalTest.cpp
-  ThreadPool.cpp
   TimerTest.cpp
   TimeValueTest.cpp
   TrailingObjectsTest.cpp
