From 44dd88d1e683d1695524b0a52712146a2713f031 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 12 Jun 2018 17:37:24 -0600
Subject: [PATCH] fix armv5 build

---
 src/system/sysutils.cpp | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/src/system/sysutils.cpp b/src/system/sysutils.cpp
index 64f18bb..c11c2d7 100644
--- a/src/system/sysutils.cpp
+++ b/src/system/sysutils.cpp
@@ -189,20 +189,6 @@ void system_specific_initialise()
     fesetenv(FE_DFL_DISABLE_DENORMS_ENV);
 #endif
 #endif
-#if defined __ARMEL__
-    // ARM32
-    static const unsigned int x = 0x04086060;
-    static const unsigned int y = 0x03000000;
-    int r;
-    asm volatile (
-        "fmrx	%0, fpscr   \n\t"
-        "and	%0, %0, %1  \n\t"
-        "orr	%0, %0, %2  \n\t"
-        "fmxr	fpscr, %0   \n\t"
-        : "=r"(r)
-        : "r"(x), "r"(y)
-	);
-#endif
 }
 
 void system_specific_application_initialise()
-- 
2.16.2

