Description: Patch fixing build on armel 
Author: Jaromír Mikeš <mira.mikes@seznam.cz>
Forwarded: yes

Index: rubberband/src/system/sysutils.cpp
===================================================================
--- rubberband.orig/src/system/sysutils.cpp	2013-06-05 14:18:50.718456311 +0200
+++ rubberband/src/system/sysutils.cpp	2013-06-05 14:22:36.759577185 +0200
@@ -219,19 +219,6 @@
     fesetenv(FE_DFL_DISABLE_SSE_DENORMS_ENV);
 #endif
 #endif
-#if defined __ARMEL__
-    static const unsigned int x = 0x04086060;
-    static const unsigned int y = 0x03000000;
-    int r;
-    asm volatile (
-        "fmrx	%0, fpscr   \n\t"
-        "and	%0, %0, %1  \n\t"
-        "orr	%0, %0, %2  \n\t"
-        "fmxr	fpscr, %0   \n\t"
-        : "=r"(r)
-        : "r"(x), "r"(y)
-	);
-#endif
 }
 
 void system_specific_application_initialise()
