--- utils/ndiswrapper~	2009-06-21 09:52:13.000000000 +0200
+++ utils/ndiswrapper	2009-06-21 09:52:13.000000000 +0200
@@ -57,7 +57,7 @@
 if (`uname -r` =~ /(\d+)\.(\d+)\.(\d+)/) {
     if ($2 > 4) {
 	if (-d "/etc/modprobe.d") {
-	    $modconf = "/etc/modprobe.d/ndiswrapper";
+	    $modconf = "/etc/modprobe.d/ndiswrapper.conf";
 	} else {
 	    $modconf = "/etc/modprobe.conf";
 	}
@@ -786,7 +786,7 @@
     if (-d "/etc/modprobe.d") {
 	$device_map = $modconf;
     } elsif (-d "/etc/modules.d") {
-	$device_map = "/etc/modules.d/ndiswrapper";
+	$device_map = "/etc/modules.d/ndiswrapper.conf";
     } else {
 	$device_map = "/etc/ndiswrapper/ndiswrapper";
     }
