diff --git a/libs/lua/sol/sol.hpp b/libs/lua/sol/sol.hpp
index c25fd549..2ca60e5f 100644
--- a/libs/lua/sol/sol.hpp
+++ b/libs/lua/sol/sol.hpp
@@ -6056,7 +6056,8 @@ namespace sol {
 			static_assert(std::is_constructible<T, Args&&...>::value, "T must be constructible with Args");
 
 			*this = nullopt;
-			this->construct(std::forward<Args>(args)...);
+			new (static_cast<void*>(this)) optional(std::in_place, std::forward<Args>(args)...);
+			return **this;
 		}
 
 		/// Swaps this optional with the other.
