diff -urN a/src/Plugins/Qt/qt_renderer.cpp b/src/Plugins/Qt/qt_renderer.cpp
--- a/src/Plugins/Qt/qt_renderer.cpp	2013-11-03 16:09:35.000000000 -0700
+++ b/src/Plugins/Qt/qt_renderer.cpp	2014-02-03 13:45:42.670490781 -0700
@@ -271,7 +271,7 @@
   if ((N(y) != n) || (n<2)) return;
   STACK_NEW_ARRAY (pnt, QPointF, n);
   for (i=0; i<n; i++)
-    decode (x[i], y[i], pnt[i].rx(), pnt[i].ry());
+    decode (x[i], y[i], (double&)pnt[i].rx(), (double&)pnt[i].ry());
   painter->setRenderHints (QPainter::Antialiasing);
   painter->drawPolyline (pnt, n);
   STACK_DELETE_ARRAY (pnt);
