diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2017-12-21 13:27:33.000000000 -0700
+++ b/CMakeLists.txt	2018-03-18 16:15:51.243005972 -0600
@@ -279,14 +279,14 @@
 CHECK_C_SOURCE_COMPILES(  "#include <guile/gh.h>
   #include <libguile.h>
   void print_string (SCM s) {
-    int len_r;
+    size_t len_r;
     char* r= gh_scm2newstr (s, &len_r); } int main() { return 0; }
 " _guile_str_size_test)
 
 IF(_guile_str_size_test)
-  SET(guile_str_size_t int)
-ELSE(_guile_str_size_test)
   SET(guile_str_size_t size_t)
+ELSE(_guile_str_size_test)
+  SET(guile_str_size_t int)
 ENDIF(_guile_str_size_test)
 
 MESSAGE(STATUS "Checking whether ... arguments behave correctly")
