--- texmacs/src/CMakeLists.txt.orig	2018-07-14 09:20:14.114218215 +0000
+++ texmacs/src/CMakeLists.txt	2018-07-14 09:21:07.651031679 +0000
@@ -1,6 +1,8 @@
 
 add_library(texmacs_body ${TeXmacs_All_SRCS})
 
+target_link_libraries (texmacs_body ${TeXmacs_Libraries})
+
 add_executable (${TeXmacs_binary_name}
   ./Texmacs/Texmacs/texmacs.cpp
 )
@@ -19,4 +21,4 @@
     RUNTIME_OUTPUT_DIRECTORY ${TEXMACS_BINARY_DIR}
 )
 
-install(TARGETS ${TeXmacs_binary_name} DESTINATION lib/texmacs/TeXmacs/bin)
\ No newline at end of file
+install(TARGETS ${TeXmacs_binary_name} DESTINATION lib/texmacs/TeXmacs/bin)
