diff --git a/src/Plugins/Qt/qt_renderer.cpp b/src/Plugins/Qt/qt_renderer.cpp
index d26769a..fef5987 100755
--- a/src/Plugins/Qt/qt_renderer.cpp
+++ b/src/Plugins/Qt/qt_renderer.cpp
@@ -273,8 +273,12 @@ qt_renderer_rep::lines (array<SI> x, array<SI> y) {
   int i, n= N(x);
   if ((N(y) != n) || (n<2)) return;
   STACK_NEW_ARRAY (pnt, QPointF, n);
-  for (i=0; i<n; i++)
-    decode (x[i], y[i], pnt[i].rx(), pnt[i].ry());
+  for (i=0; i<n; i++) {
+    SI xx= x[i], yy= y[i];
+    decode (xx, yy);
+    pnt[i].rx()= xx;
+    pnt[i].ry()= yy;
+  }
 
   QPen p= painter->pen();
   p.setCapStyle (pen->get_cap () == cap_round? Qt::RoundCap: Qt::SquareCap);
