diff -urN old/arch/arm/mach-kirkwood/rd88f6281-setup.c new/arch/arm/mach-kirkwood/rd88f6281-setup.c
--- old/arch/arm/mach-kirkwood/rd88f6281-setup.c	2010-01-06 09:21:15.000000000 +0800
+++ new/arch/arm/mach-kirkwood/rd88f6281-setup.c	2010-01-06 10:17:52.000000000 +0800
@@ -11,7 +11,7 @@
 // linux/arch/arm/mach-kirkwood/rd88f6281-setup.c
 // Project: iConnect (NAS PLUG)
 // function: 
-// version: 1.0 (2009/11/11)
+// version: 1.1 (2010/01/06)
 // Modifier: Ken Cheng (kencheng@mapower.com.tw)
 //***************************************************************************************
 
@@ -90,6 +90,11 @@
 
 static struct gpio_led rd88f6281_led_pins[] = {
 	{
+		.name			= "led_level",
+		.gpio			= 41,
+		.default_trigger	= "default-on",
+	},
+	{
 		.name			= "power_blue_led",
 		.gpio			= 42,
 		.default_trigger	= "timer",
@@ -197,6 +202,7 @@
 static unsigned int rd88f6281_mpp_config[] __initdata = {
 	MPP12_GPIO,
 	MPP35_GPIO,
+	MPP41_GPIO,
 	MPP42_GPIO,
 	MPP43_GPIO,
 	MPP44_GPIO,
