diff -urpN linux-2.6.30_orig/arch/arm/boot/Makefile linux-2.6.30/arch/arm/boot/Makefile
--- linux-2.6.30_orig/arch/arm/boot/Makefile	2009-08-28 11:20:52.000000000 -0400
+++ linux-2.6.30/arch/arm/boot/Makefile	2009-08-28 11:22:45.000000000 -0400
@@ -60,7 +60,7 @@ $(obj)/zImage:	$(obj)/compressed/vmlinux
 endif
 
 quiet_cmd_uimage = UIMAGE  $@
-      cmd_uimage = $(CONFIG_SHELL) $(MKIMAGE) -A arm -O linux -T kernel \
+      cmd_uimage = $(MKIMAGE) -A arm -O linux -T kernel \
 		   -C none -a $(LOADADDR) -e $(LOADADDR) \
 		   -n 'Linux-$(KERNELRELEASE)' -d $< $@
 
