diff -bur snd-14.1.orig/snd-ladspa.c snd-14.1/snd-ladspa.c
--- snd-14.1.orig/snd-ladspa.c	2013-10-28 16:07:53.260612114 +0800
+++ snd-14.1/snd-ladspa.c	2013-10-28 16:11:18.792831983 +0800
@@ -1191,7 +1191,7 @@
   if (descriptor->connect_port) 
     descriptor->connect_port(XEN_TO_C_Ladspa_Handle(ptr),
 			     XEN_TO_C_ULONG(port),
-			     double_to_float(samples->data, samples->length)
+			     double_to_float(mus_vct_data(samples), mus_vct_length(samples))
 			     );
   return(XEN_FALSE);
 }
