diff -ur snd-14.1.orig/vct.h snd-14.1/vct.h
--- snd-14.1.orig/vct.h	2013-10-28 02:22:35.073798054 +0800
+++ snd-14.1/vct.h	2013-10-28 02:45:27.166364791 +0800
@@ -1,11 +1,11 @@
 #ifndef VCT_H
 #define VCT_H
 
-#if HAVE_SCHEME
-  typedef struct s7_cell vct;
-#else
-typedef struct vct vct;
-#endif
+typedef struct {
+  mus_long_t length;
+  mus_float_t *data;
+  bool dont_free;
+} vct;
 
 #ifdef __cplusplus
 extern "C" {
