From c93db056e57b2eb58975f8a0ad329f75dcd177b8 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sat, 14 Jan 2017 14:08:30 -0700
Subject: [PATCH] disable pthread warmup on ARM

---
 ntpd/ntpd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ntpd/ntpd.c b/ntpd/ntpd.c
index 2c7f02ec5..8746eff4c 100644
--- a/ntpd/ntpd.c
+++ b/ntpd/ntpd.c
@@ -34,7 +34,7 @@
 # ifdef HAVE_PTHREAD_H
 #  include <pthread.h>
 # endif
-# if defined(linux)
+# if defined(linux) && !defined(__arm__)
 #  define NEED_PTHREAD_WARMUP
 # endif
 #endif
-- 
2.11.0

