diff -Naur CriticalMass-1.0.2-old/tinyxml/tinyxml.cpp CriticalMass-1.0.2/tinyxml/tinyxml.cpp
--- CriticalMass-1.0.2-old/tinyxml/tinyxml.cpp	2003-03-03 12:34:58.000000000 +1000
+++ CriticalMass-1.0.2/tinyxml/tinyxml.cpp	2008-04-01 22:26:47.000000000 +1000
@@ -22,7 +22,8 @@
 */
 
 #include "tinyxml.h"
-
+#include <cstring>
+#include <cstdlib>
 
 TiXmlNode::TiXmlNode( NodeType _type )
 {
diff -Naur CriticalMass-1.0.2-old/tinyxml/tinyxmlparser.cpp CriticalMass-1.0.2/tinyxml/tinyxmlparser.cpp
--- CriticalMass-1.0.2-old/tinyxml/tinyxmlparser.cpp	2003-03-03 12:34:58.000000000 +1000
+++ CriticalMass-1.0.2/tinyxml/tinyxmlparser.cpp	2008-04-01 22:27:02.000000000 +1000
@@ -24,7 +24,7 @@
 
 #include "tinyxml.h"
 #include <ctype.h>
-
+#include <cstring>
 const char* TiXmlBase::SkipWhiteSpace( const char* p )
 {
 	while ( p && *p && 
diff -Naur CriticalMass-1.0.2-old/utils/ResourceManager.cpp CriticalMass-1.0.2/utils/ResourceManager.cpp
--- CriticalMass-1.0.2-old/utils/ResourceManager.cpp	2005-01-02 12:59:29.000000000 +1000
+++ CriticalMass-1.0.2/utils/ResourceManager.cpp	2008-04-01 22:27:19.000000000 +1000
@@ -15,6 +15,7 @@
 #include <iomanip>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <cstring>
 
 #include <Trace.hpp>
 #include <ResourceManager.hpp>
diff -Naur CriticalMass-1.0.2-old/utils/Value.hpp CriticalMass-1.0.2/utils/Value.hpp
--- CriticalMass-1.0.2-old/utils/Value.hpp	2004-12-18 12:41:24.000000000 +1000
+++ CriticalMass-1.0.2/utils/Value.hpp	2008-04-01 22:27:39.000000000 +1000
@@ -17,6 +17,7 @@
 
 #include <stdio.h>
 #include <string>
+#include <cstdlib>
 
 #include <Trace.hpp>
 
diff -Naur CriticalMass-1.0.2-old/utils/zStream.cpp CriticalMass-1.0.2/utils/zStream.cpp
--- CriticalMass-1.0.2-old/utils/zStream.cpp	2005-08-01 06:06:14.000000000 +1000
+++ CriticalMass-1.0.2/utils/zStream.cpp	2008-04-01 22:28:03.000000000 +1000
@@ -17,6 +17,7 @@
 #include <zStream.hpp>
 #include <zStreamBufferImplLZMA.hpp>
 #include <zStreamBufferImplZLib.hpp>
+#include <cstdlib>
 
 ofstream &operator<<( ofstream &outfile, Uint32 i)
 {       
diff -Naur CriticalMass-1.0.2-old/utils/zStreamBufferImplZLib.cpp CriticalMass-1.0.2/utils/zStreamBufferImplZLib.cpp
--- CriticalMass-1.0.2-old/utils/zStreamBufferImplZLib.cpp	2005-08-01 06:06:14.000000000 +1000
+++ CriticalMass-1.0.2/utils/zStreamBufferImplZLib.cpp	2008-04-01 22:28:23.000000000 +1000
@@ -14,6 +14,7 @@
 //
 #include <Trace.hpp>
 #include <zStreamBufferImplZLib.hpp>
+#include <cstdlib>
 
 bool ziStreamBufferImplZLib::init( void)
 {
