diff -Naur ImageMagick-6.2.7.orig/coders/png.c ImageMagick-6.2.7/coders/png.c
--- ImageMagick-6.2.7.orig/coders/png.c	2006-05-21 12:35:55.784852536 +0200
+++ ImageMagick-6.2.7/coders/png.c	2006-05-21 12:36:50.698504392 +0200
@@ -164,6 +164,8 @@
 #define PNG_MAX_UINT (png_uint_32) 0x7fffffffL
 #endif
 
+#undef PNG_ASSEMBLER_CODE_SUPPORTED
+
 /*
   Constant strings for known chunk types.  If you need to add a chunk,
   add a string holding the name here.   To make the code more
