diff -ruN stfl-0.21-orig/Makefile.cfg stfl-0.21/Makefile.cfg
--- stfl-0.21-orig/Makefile.cfg	2009-05-31 20:20:39.000000000 +0200
+++ stfl-0.21/Makefile.cfg	2009-07-11 14:26:51.000000000 +0200
@@ -19,35 +19,35 @@
 #
 
 export libdir ?= lib
-export prefix ?= /usr/local
+export prefix ?= /usr
 export DESTDIR ?= /
 
 ifneq ($(shell spl-config --cflags 2>/dev/null),)
-FOUND_SPL = 1
+FOUND_SPL = 0
 else
 FOUND_SPL = 0
 endif
 
 ifneq ($(shell swig -help 2>/dev/null),)
-FOUND_SWIG = 1
+FOUND_SWIG = 0
 else
 FOUND_SWIG = 0
 endif
 
 ifneq ($(shell perl -le 'print 1' 2>/dev/null),)
-FOUND_PERL5 = 1
+FOUND_PERL5 = 0
 else
 FOUND_PERL5 = 0
 endif
 
 ifneq ($(shell python -c 'print 1' 2>/dev/null),)
-FOUND_PYTHON = 1
+FOUND_PYTHON = 0
 else
 FOUND_PYTHON = 0
 endif
 
 ifneq ($(shell ruby -e 'puts 1' 2>/dev/null),)
-FOUND_RUBY = 1
+FOUND_RUBY = 0
 else
 FOUND_RUBY = 0
 endif
diff -ruN stfl-0.21-orig/stfl_internals.h stfl-0.21/stfl_internals.h
--- stfl-0.21-orig/stfl_internals.h	2009-06-19 19:27:33.000000000 +0200
+++ stfl-0.21/stfl_internals.h	2009-07-11 14:20:22.000000000 +0200
@@ -28,7 +28,7 @@
 #endif
 
 #include "stfl.h"
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #include <pthread.h>
 
 struct stfl_widget_type;
