Index: src/perl/urxvt-popup
===================================================================
RCS file: /schmorpforge/rxvt-unicode/src/perl/urxvt-popup,v
retrieving revision 1.15
diff -u -r1.15 urxvt-popup
--- src/perl/urxvt-popup	24 Dec 2007 08:21:40 -0000	1.15
+++ src/perl/urxvt-popup	2 Aug 2009 08:22:52 -0000
@@ -120,7 +120,7 @@
    $self->refresh;
 
    # might fail, but try anyways
-   $self->grab ($self->{data}{event}{time}, 1)
+   $self->grab ($self->{data}{event}{time})
       and $self->allow_events_async;
 
    on_button_press $self, $self->{data}{event} if $self->{data}{event}{button};
@@ -132,7 +132,7 @@
    my ($self, $event) = @_;
 
    # should definitely not fail
-   $self->grab ($self->{data}{event}{time}, 1)
+   $self->grab ($self->{data}{event}{time})
       and $self->allow_events_async;
 }
