--- kdelibs-4.3.98/kded/applications.menu	2010-01-31 19:28:11.000000000 +0000
+++ kdelibs-4.3.98/kded/applications.menu	2010-01-31 22:25:53.556043077 +0000
@@ -16,11 +16,19 @@
 	</DefaultLayout>
 	<Layout>
 		<Merge type="menus"/>
+		<Menuname>Arch Linux</Menuname>
 		<Menuname>Applications</Menuname>
 		<Merge type="files"/>
 	</Layout>
 
 	<Menu>
+		<Name>Arch Linux</Name>                                                                                                   
+		<Directory>Archlinux.directory</Directory>                                                                               
+		<Include>                                                                                                                
+			<Category>Archlinux</Category>                                                                                           
+		</Include>
+	</Menu>
+	<Menu>
 		<Name>Applications</Name>
 		<Directory>kde-unknown.directory</Directory>
 		<OnlyUnallocated/>
