--- kdelibs-4.3.0/kded/applications.menu~	2009-08-04 09:25:19.000000000 +0200
+++ kdelibs-4.3.0/kded/applications.menu	2009-08-04 09:25:19.000000000 +0200
@@ -21,6 +21,13 @@
 	</Layout>
 
 	<Menu>
+		<Name>Archlinux</Name>                                     
+		<Directory>Archlinux.directory</Directory>                 
+		<Include>                                                  
+		<Category>Archlinux</Category>                     
+		</Include> 
+	</Menu>
+	<Menu>
 		<Name>Applications</Name>
 		<Directory>kde-unknown.directory</Directory>
 		<OnlyUnallocated/>
