diff --git a/plasma/corona.cpp b/plasma/corona.cpp
index 5c90aa4..7dff075 100644
--- a/plasma/corona.cpp
+++ b/plasma/corona.cpp
@@ -388,7 +388,7 @@ void Corona::addOffscreenWidget(QGraphicsWidget *widget)
     }
 
     d->offscreenWidgets[i] = widget;
-    widget->setPos((-i - 1) * QWIDGETSIZE_MAX, -QWIDGETSIZE_MAX);
+    widget->setPos((-i - 1) * 2000, -2000);
 
     QGraphicsWidget *pw = widget->parentWidget();
     widget->setParentItem(0);

