diff -Nura kdelibs-4.3.0.orig/kded/CMakeLists.txt kdelibs-4.3.0/kded/CMakeLists.txt
--- kdelibs-4.3.0.orig/kded/CMakeLists.txt	2008-12-10 17:12:28.000000000 +0100
+++ kdelibs-4.3.0/kded/CMakeLists.txt	2009-07-30 13:22:06.000000000 +0200
@@ -69,7 +69,7 @@
 if (WIN32)
 install( FILES applications.menu  DESTINATION  ${SHARE_INSTALL_PREFIX}/xdg/menus )
 else (WIN32)
-install( FILES applications.menu  DESTINATION  ${SYSCONF_INSTALL_DIR}/xdg/menus )
+install( FILES applications.menu  DESTINATION  ${SYSCONF_INSTALL_DIR}/xdg/menus RENAME kde-applications.menu )
 endif (WIN32)
 install( FILES kdedmodule.desktop DESTINATION  ${SERVICETYPES_INSTALL_DIR} )
 install( FILES kded.upd           DESTINATION  ${DATA_INSTALL_DIR}/kconf_update )
diff -Nura kdelibs-4.3.0.orig/kded/kbuildsycoca.cpp kdelibs-4.3.0/kded/kbuildsycoca.cpp
--- kdelibs-4.3.0.orig/kded/kbuildsycoca.cpp	2009-03-10 13:26:04.000000000 +0100
+++ kdelibs-4.3.0/kded/kbuildsycoca.cpp	2009-07-30 13:22:06.000000000 +0200
@@ -306,7 +306,7 @@
      connect(g_vfolder, SIGNAL(newService(const QString &, KService::Ptr *)),
              this, SLOT(slotCreateEntry(const QString &, KService::Ptr *)));
 
-     VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu("applications.menu", true);
+     VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu("kde-applications.menu", true);
 
      KServiceGroup::Ptr entry = g_bsgf->addNew("/", kdeMenu->directoryFile, KServiceGroup::Ptr(), false);
      entry->setLayoutInfo(kdeMenu->layoutList);
