diff -up kdelibs-4.14.16/CMakeLists.txt.webkit kdelibs-4.14.16/CMakeLists.txt
--- kdelibs-4.14.16/CMakeLists.txt.webkit	2016-01-07 00:02:22.000000000 +0100
+++ kdelibs-4.14.16/CMakeLists.txt	2016-01-28 13:03:53.556194927 +0100
@@ -356,7 +356,6 @@ if(NOT WINCE)
 add_subdirectory( plasma )
 endif(NOT WINCE)
 add_subdirectory( kunitconversion )
-add_subdirectory( kdewebkit )
 add_subdirectory( includes )
 
 add_subdirectory( experimental )
diff -up kdelibs-4.14.16/kdewidgets/CMakeLists.txt.webkit kdelibs-4.14.16/kdewidgets/CMakeLists.txt
--- kdelibs-4.14.16/kdewidgets/CMakeLists.txt.webkit	2016-01-28 13:03:53.527196020 +0100
+++ kdelibs-4.14.16/kdewidgets/CMakeLists.txt	2016-01-28 13:03:53.556194927 +0100
@@ -88,41 +88,6 @@ if(QT_QTDESIGNER_FOUND)
     install(TARGETS kdedeprecated  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
   endif(NOT KDE_NO_DEPRECATED)
 
-
-  # kdewebkit widgets
-  include_directories(
-    ${CMAKE_SOURCE_DIR}/kdewebkit
-  )
-
-  add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/kdewebkitwidgets.cpp
-    COMMAND "${MAKEKDEWIDGETS_EXECUTABLE}" -o ${CMAKE_CURRENT_BINARY_DIR}/kdewebkitwidgets.cpp ${CMAKE_CURRENT_SOURCE_DIR}/kdewebkit.widgets
-    MAIN_DEPENDENCY kdewebkit.widgets DEPENDS makekdewidgets)
-
-  set(kdewebkitwidgets_PART_SRCS
-     classpreviews.cpp
-     ${CMAKE_CURRENT_BINARY_DIR}/kdewebkitwidgets.cpp
-  )
-
-  qt4_generate_moc(${CMAKE_CURRENT_BINARY_DIR}/kdewebkitwidgets.cpp ${CMAKE_CURRENT_BINARY_DIR}/kdewebkitwidgets.moc)
-
-
-  qt4_add_resources(kdewebkitwidgets_PART_SRCS kdewebkitwidgets.qrc)
-
-  kde4_add_plugin(kdewebkitwidgets ${kdewebkitwidgets_PART_SRCS})
-
-  target_link_libraries(kdewebkitwidgets ${KDE4_KDEUI_LIBS} ${KDE4_KDEWEBKIT_LIBS} ${QT_QTWEBKIT_LIBRARY})
-  if(NOT WIN32)
-    set_target_properties(kdewebkitwidgets PROPERTIES
-                                     INSTALL_RPATH_USE_LINK_PATH TRUE
-                                     SKIP_BUILD_RPATH TRUE
-                                     BUILD_WITH_INSTALL_RPATH TRUE
-                                     INSTALL_RPATH ${LIB_INSTALL_DIR}
-                         )
-  endif(NOT WIN32)
-
-  install(TARGETS kdewebkitwidgets  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
-
-
   if (QT_QT3SUPPORT_FOUND)
 
      include_directories(
diff -up kdelibs-4.14.16/plasma/CMakeLists.txt.webkit kdelibs-4.14.16/plasma/CMakeLists.txt
--- kdelibs-4.14.16/plasma/CMakeLists.txt.webkit	2016-01-07 00:02:22.000000000 +0100
+++ kdelibs-4.14.16/plasma/CMakeLists.txt	2016-01-28 13:26:55.730137496 +0100
@@ -11,6 +11,7 @@ if(KDE_PLATFORM_FEATURE_BINARY_COMPATIBL
 endif(KDE_PLATFORM_FEATURE_BINARY_COMPATIBLE_FEATURE_REDUCTION)
 
 set(PLASMA_NO_PACKAGEKIT TRUE)
+set(PLASMA_NO_KDEWEBKIT TRUE)
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}
                     ${KDE4_KDECORE_INCLUDES}
@@ -121,7 +122,6 @@ set(plasma_LIB_SRCS
     framesvg.cpp
     plasma.cpp
     popupapplet.cpp
-    private/animablegraphicswebview.cpp
     private/applethandle.cpp
     private/associatedapplicationmanager.cpp
     private/componentinstaller.cpp
@@ -211,7 +211,6 @@ set(plasma_LIB_SRCS
     widgets/textbrowser.cpp
     widgets/treeview.cpp
     widgets/textedit.cpp
-    widgets/webview.cpp
 
     #Temporary QtJolie branch
     private/qtjolie-branch/qtjolie/abstractadaptor.cpp
@@ -275,7 +274,7 @@ endif(PHONON_FOUND)
 
 kde4_add_library(plasma ${LIBRARY_TYPE} ${plasma_LIB_SRCS})
 
-target_link_libraries(plasma LINK_PRIVATE ${QT_QTUITOOLS_LIBRARY} ${QT_QTWEBKIT_LIBRARY}
+target_link_libraries(plasma LINK_PRIVATE ${QT_QTUITOOLS_LIBRARY}
                              ${QT_QTSCRIPT_LIBRARY} ${QT_QTNETWORK_LIBRARY} ${QT_QTXML_LIBRARY} ${QT_QTSQL_LIBRARY} ${QT_QTDECLARATIVE_LIBRARY}
                              ${KDE4_KDEUI_LIBS} ${KDE4_KDNSSD_LIBS} ${KDE4_THREADWEAVER_LIBS} ${PLASMA_EXTRA_LIBS})
 
@@ -415,7 +414,6 @@ install(FILES
     widgets/textbrowser.h
     widgets/treeview.h
     widgets/textedit.h
-    widgets/webview.h
     DESTINATION ${INCLUDE_INSTALL_DIR}/plasma/widgets COMPONENT Devel)
 
 install(FILES
