diff -wbBur FreeMat-3.6/libs/libFreeMat/Malloc.cpp FreeMat-3.6.my/libs/libFreeMat/Malloc.cpp
--- FreeMat-3.6/libs/libFreeMat/Malloc.cpp	2008-03-14 18:43:10.000000000 +0300
+++ FreeMat-3.6.my/libs/libFreeMat/Malloc.cpp	2008-04-25 21:29:19.000000000 +0400
@@ -21,6 +21,7 @@
 #include "Exception.hpp"
 #include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 
 
 void* Malloc(int count) {
diff -wbBur FreeMat-3.6/libs/libFreeMat/Token.cpp FreeMat-3.6.my/libs/libFreeMat/Token.cpp
--- FreeMat-3.6/libs/libFreeMat/Token.cpp	2008-03-14 18:43:10.000000000 +0300
+++ FreeMat-3.6.my/libs/libFreeMat/Token.cpp	2008-04-25 21:31:33.000000000 +0400
@@ -2,6 +2,7 @@
 #include "Serialize.hpp"
 #include <iostream>
 #include <errno.h>
+#include <limits.h>
 
 // These must appear as sequential token numbers
 string fm_reserved[22] = {
diff -wbBur FreeMat-3.6/libs/libFreeMat/Types.hpp FreeMat-3.6.my/libs/libFreeMat/Types.hpp
--- FreeMat-3.6/libs/libFreeMat/Types.hpp	2008-03-14 18:43:10.000000000 +0300
+++ FreeMat-3.6.my/libs/libFreeMat/Types.hpp	2008-04-25 21:28:26.000000000 +0400
@@ -20,7 +20,11 @@
 #ifndef __Types_hpp__
 #define __Types_hpp__
 
+#include <algorithm>
 #include <string>
+#include <string.h>
+using namespace std;
+
 #include <QVector>
 #include <QtGlobal>
 #include <QList>
