--- beep/vis.c	2004-12-04 10:04:29.000000000 +0100
+++ beep/vis.c	2005-02-17 12:02:25.391044008 +0100
@@ -223,7 +223,8 @@ vis_draw(Widget * w)
     /* FIXME: The check "shouldn't" be neccessary? */
     /*	if (GTK_IS_WINDOW(vis->vs_window)) { */
     GDK_THREADS_ENTER();
-    gdk_draw_indexed_image(vis->vs_window, vis->vs_widget.gc,
+    if (GDK_IS_DRAWABLE(vis->vs_window))
+        gdk_draw_indexed_image(vis->vs_window, vis->vs_widget.gc,
                            vis->vs_widget.x, vis->vs_widget.y,
                            vis->vs_widget.width, vis->vs_widget.height,
                            GDK_RGB_DITHER_NORMAL, (guchar *) rgb_data,
