From 98a9974f550dfb5c8f66213774e4ef95848127e1 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Mon, 23 Jul 2018 11:04:55 +0200
Subject: [PATCH 2/4] disable-unsupported-operating-system-warning

Signed-off-by: Christian Hesse <mail@eworm.de>
---
 backend/wbprivate/workbench/wb_context.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/backend/wbprivate/workbench/wb_context.cpp b/backend/wbprivate/workbench/wb_context.cpp
index e80501a..a4047b1 100644
--- a/backend/wbprivate/workbench/wb_context.cpp
+++ b/backend/wbprivate/workbench/wb_context.cpp
@@ -1011,9 +1011,11 @@ void WBContext::init_finish_(WBOptions *options) {
 
   _frontendCallbacks->show_status_text(_("Ready."));
 
+#if 0
   // Avoid our runtime tests to lock up when a modal warning dialog is displayed.
   if (options->open_at_startup_type != "run-script")
     warnIfRunningOnUnsupportedOS();
+#endif
 
   try {
     // execute action requested from command line
