diff --git a/library/forms/swig/cairo.i b/library/forms/swig/cairo.i
index a02efdcd1..d9091fca0 100644
--- a/library/forms/swig/cairo.i
+++ b/library/forms/swig/cairo.i
@@ -125,7 +125,7 @@
 
 %typemap(argout) cairo_text_extents_t *extents {
       PyObject *o= SWIG_NewPointerObj(new cairo_text_extents_t(*$1), SWIGTYPE_p_cairo_text_extents_t, 0 |  0 );
-      $result= SWIG_Python_AppendOutput($result, o);
+      $result= SWIG_Python_AppendOutput($result, o, 0);
 }
 
 %typemap(in) const char* (std::string s) {
diff --git a/library/forms/swig/mforms.i b/library/forms/swig/mforms.i
index ac4c6e36a..e46459966 100644
--- a/library/forms/swig/mforms.i
+++ b/library/forms/swig/mforms.i
@@ -722,7 +722,7 @@ inline boost::function<void (mforms::TextEntryAction)> pycall_void_entryaction_f
 
 %typemap(argout) std::string &ret_password {
     PyObject *o= PyUnicode_DecodeUTF8(($1)->data(), ($1)->size(), NULL);
-    $result= SWIG_Python_AppendOutput($result, o);
+    $result= SWIG_Python_AppendOutput($result, o, 0);
 }
 
 %typemap(in,numinputs=0) std::string &ret_password(std::string temp) {
@@ -731,7 +731,7 @@ inline boost::function<void (mforms::TextEntryAction)> pycall_void_entryaction_f
 
 %typemap(argout) std::string &ret_value {
     PyObject *o= PyUnicode_DecodeUTF8(($1)->data(), ($1)->size(), NULL);
-    $result= SWIG_Python_AppendOutput($result, o);
+    $result= SWIG_Python_AppendOutput($result, o, 0);
 }
 
 %typemap(in,numinputs=0) std::string &ret_value(std::string temp) {
@@ -741,7 +741,7 @@ inline boost::function<void (mforms::TextEntryAction)> pycall_void_entryaction_f
 
 %typemap(argout) bool &ret_store {
     if (*$1) Py_INCREF(Py_True); else Py_INCREF(Py_False);
-    $result= SWIG_Python_AppendOutput($result, *$1 ? Py_True : Py_False);
+    $result= SWIG_Python_AppendOutput($result, *$1 ? Py_True : Py_False, 0);
 }
 
 %typemap(in,numinputs=0) bool &ret_store(bool temp) {
