diff -ru mysql-connector-c++-9.3.0-src.orig/include/mysqlx/common/value.h mysql-connector-c++-9.3.0-src/include/mysqlx/common/value.h
--- mysql-connector-c++-9.3.0-src.orig/include/mysqlx/common/value.h	2025-05-18 12:30:03.916477018 +0200
+++ mysql-connector-c++-9.3.0-src/include/mysqlx/common/value.h	2025-05-18 12:30:25.042409140 +0200
@@ -35,6 +35,7 @@
 #include "api.h"
 #include "error.h"
 #include "util.h"
+#include <cstdint>
 
 PUSH_SYS_WARNINGS
 #include <string>
