diff --git a/ext/scintilla/qt/ScintillaEditPy/typesystem_ScintillaEdit.xml.template b/ext/scintilla/qt/ScintillaEditPy/typesystem_ScintillaEdit.xml.template
index 6062955..c70ff8f 100644
--- a/ext/scintilla/qt/ScintillaEditPy/typesystem_ScintillaEdit.xml.template
+++ b/ext/scintilla/qt/ScintillaEditPy/typesystem_ScintillaEdit.xml.template
@@ -17,7 +17,7 @@
 	     greater than max value of an int -->
 		<modify-function signature="set_margin_mask_n(sptr_t, sptr_t)">
 			<inject-code>
-				int margin = PyInt_AsLong(%PYARG_1);
+				int margin = PyLong_AsLong(%PYARG_1);
 				if (margin == -1 &amp;&amp; PyErr_Occurred())
 					return NULL;
 				unsigned long mask = PyLong_AsUnsignedLongMask(%PYARG_2);
@@ -30,16 +30,16 @@
 		</modify-function>
 		<modify-function signature="get_margin_mask_n(sptr_t) const">
 			<inject-code>
-				int margin = PyInt_AsLong(%PYARG_1);
+				int margin = PyLong_AsLong(%PYARG_1);
 				if (margin == -1 &amp;&amp; PyErr_Occurred())
 					return NULL;
 				unsigned int mask = (unsigned int)%CPPSELF-&gt;get_margin_mask_n(margin);
-				%PYARG_0 = PyInt_FromSize_t(mask);
+				%PYARG_0 = PyLong_FromSize_t(mask);
 			</inject-code>
 		</modify-function>
 		<modify-function signature="setMarginMaskN(sptr_t, sptr_t)">
 			<inject-code>
-				int margin = PyInt_AsLong(%PYARG_1);
+				int margin = PyLong_AsLong(%PYARG_1);
 				if (margin == -1 &amp;&amp; PyErr_Occurred())
 					return NULL;
 				unsigned long mask = PyLong_AsUnsignedLongMask(%PYARG_2);
@@ -52,11 +52,11 @@
 		</modify-function>
 		<modify-function signature="marginMaskN(sptr_t) const">
 			<inject-code>
-				int margin = PyInt_AsLong(%PYARG_1);
+				int margin = PyLong_AsLong(%PYARG_1);
 				if (margin == -1 &amp;&amp; PyErr_Occurred())
 					return NULL;
 				unsigned int mask = (unsigned int)%CPPSELF-&gt;marginMaskN(margin);
-				%PYARG_0 = PyInt_FromSize_t(mask);
+				%PYARG_0 = PyLong_FromSize_t(mask);
 			</inject-code>
 		</modify-function>
 
