diff --git a/library/grt/src/grt.h b/library/grt/src/grt.h
index bc5e352128..c0fd68f6a0 100644
--- a/library/grt/src/grt.h
+++ b/library/grt/src/grt.h
@@ -1385,7 +1385,7 @@ namespace grt {
     }
 
   protected:
-    explicit ListRef<internal::Integer>(const ValueRef &lvalue) : BaseListRef(lvalue) {
+    explicit ListRef(const ValueRef &lvalue) : BaseListRef(lvalue) {
       if (lvalue.is_valid() && content().content_type() != IntegerType)
         throw type_error(IntegerType, content().content_type(), ListType);
     }
@@ -1813,7 +1813,7 @@ namespace grt {
     }
 
   protected:
-    explicit ListRef<internal::Dict>(const ValueRef &lvalue) : BaseListRef(lvalue) {
+    explicit ListRef(const ValueRef &lvalue) : BaseListRef(lvalue) {
       if (lvalue.is_valid() && content().content_type() != DictType)
         throw type_error(DictType, content().content_type(), ListType);
     }
