Index: trunk/gdal/apps/gdalserver.c
===================================================================
--- a/trunk/gdal/apps/gdalserver.c
+++ b/trunk/gdal/apps/gdalserver.c
@@ -28,7 +28,6 @@
  ****************************************************************************/

-#if defined(__STDC_VERSION__)
-#define _XOPEN_SOURCE
-#endif
+// So that __USE_XOPEN2K is defined to have getaddrinfo
+#define _XOPEN_SOURCE 600

 #include "cpl_port.h"
