diff -Nur gdal-1.11.2/frmts/pdf/pdfdataset.cpp gdal-1.11.2a/frmts/pdf/pdfdataset.cpp
--- gdal-1.11.2/frmts/pdf/pdfdataset.cpp	2015-02-10 13:11:19.000000000 +0100
+++ gdal-1.11.2a/frmts/pdf/pdfdataset.cpp	2015-03-02 18:38:25.170155277 +0100
@@ -113,7 +113,7 @@
                          GBool allowAntialiasA = gTrue) :
                 SplashOutputDev(colorModeA, bitmapRowPadA,
                                 reverseVideoA, paperColorA,
-                                bitmapTopDownA, allowAntialiasA),
+                                bitmapTopDownA),
                 bEnableVector(TRUE),
                 bEnableText(TRUE),
                 bEnableBitmap(TRUE) {}
