diff --git xbmc/windowing/egl/EGLNativeTypeIMX.cpp xbmc/windowing/egl/EGLNativeTypeIMX.cpp
index 4ffa708..e85a633 100644
--- xbmc/windowing/egl/EGLNativeTypeIMX.cpp
+++ xbmc/windowing/egl/EGLNativeTypeIMX.cpp
@@ -48,11 +48,15 @@ bool CEGLNativeTypeIMX::CheckCompatibility()
 {
   std::string strName;
   std::string str2 ("mxc_sdc_fb");
+  std::string str3 ("platform:fb.24");
   get_sysfs_str("/sys/class/graphics/fb0/device/modalias", strName);
   StringUtils::Trim(strName);
   size_t found = strName.find(str2);
   if (found!=std::string::npos)
     return true;
+  found = strName.find(str3);
+  if (found!=std::string::npos)
+    return true;
   return false;
 }

