diff -aur vde2-2.2.2-orig/src/kvde_switch/consmgmt.c vde2-2.2.2/src/kvde_switch/consmgmt.c
--- vde2-2.2.2-orig/src/kvde_switch/consmgmt.c	2008-06-15 13:49:19.000000000 -0300
+++ vde2-2.2.2/src/kvde_switch/consmgmt.c	2008-09-12 18:19:00.935408481 -0300
@@ -13,6 +13,7 @@
 #include <syslog.h>
 #include <stdlib.h>
 #include <libgen.h>
+#include <sys/param.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/ioctl.h>
diff -aur vde2-2.2.2-orig/src/kvde_switch/datasock.c vde2-2.2.2/src/kvde_switch/datasock.c
--- vde2-2.2.2-orig/src/kvde_switch/datasock.c	2008-06-15 13:49:19.000000000 -0300
+++ vde2-2.2.2/src/kvde_switch/datasock.c	2008-09-12 18:17:36.727494138 -0300
@@ -15,6 +15,7 @@
 #include <stdlib.h>
 #include <stdint.h>
 #include <libgen.h>
+#include <sys/param.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/ioctl.h>
diff -aur vde2-2.2.2-orig/src/slirpvde/slirpvde.c vde2-2.2.2/src/slirpvde/slirpvde.c
--- vde2-2.2.2-orig/src/slirpvde/slirpvde.c	2008-06-15 13:49:15.000000000 -0300
+++ vde2-2.2.2/src/slirpvde/slirpvde.c	2008-09-12 18:21:50.872173559 -0300
@@ -12,6 +12,7 @@
 #include <unistd.h>
 #include <stdint.h>
 #include <libgen.h>
+#include <sys/param.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/un.h>
diff -aur vde2-2.2.2-orig/src/vde_autolink.c vde2-2.2.2/src/vde_autolink.c
--- vde2-2.2.2-orig/src/vde_autolink.c	2008-03-10 06:10:34.000000000 -0300
+++ vde2-2.2.2/src/vde_autolink.c	2008-09-12 18:21:19.982059871 -0300
@@ -26,6 +26,7 @@
 #include <limits.h>
 #include <string.h>
 #include <time.h>
+#include <sys/param.h>
 #include <sys/wait.h>
 #include <sys/socket.h>
 #include <sys/un.h>
diff -aur vde2-2.2.2-orig/src/vde_plug2tap.c vde2-2.2.2/src/vde_plug2tap.c
--- vde2-2.2.2-orig/src/vde_plug2tap.c	2008-03-10 06:10:46.000000000 -0300
+++ vde2-2.2.2/src/vde_plug2tap.c	2008-09-12 18:20:08.629583649 -0300
@@ -10,6 +10,7 @@
 #include <syslog.h>
 #include <errno.h>
 #include <string.h>
+#include <sys/param.h>
 #include <sys/ioctl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
diff -aur vde2-2.2.2-orig/src/vdetaplib/libvdetap.c vde2-2.2.2/src/vdetaplib/libvdetap.c
--- vde2-2.2.2-orig/src/vdetaplib/libvdetap.c	2008-03-10 06:10:43.000000000 -0300
+++ vde2-2.2.2/src/vdetaplib/libvdetap.c	2008-09-12 18:21:03.755406693 -0300
@@ -4,6 +4,7 @@
 #define _GNU_SOURCE
 #include <dlfcn.h>
 #include <stdio.h>
+#include <sys/param.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/syscall.h>
diff -aur vde2-2.2.2-orig/src/wirefilter.c vde2-2.2.2/src/wirefilter.c
--- vde2-2.2.2-orig/src/wirefilter.c	2008-07-07 02:56:22.000000000 -0300
+++ vde2-2.2.2/src/wirefilter.c	2008-09-12 18:21:36.019384936 -0300
@@ -26,6 +26,7 @@
 #include <signal.h>
 #include <math.h>
 #include <stdarg.h>
+#include <sys/param.h>
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/stat.h>
