--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -657,6 +657,7 @@ if (TARGET TBB::tbbmalloc)
 
             enable_language(C)
             tbb_add_c_test(SUBDIR tbbmalloc NAME test_malloc_pure_c DEPENDENCIES TBB::tbbmalloc)
+            set_target_properties(test_malloc_pure_c PROPERTIES LINKER_LANGUAGE CXX)
         endif()
         # ----------------------------------------------------------------------------------------
         # Whitebox testing
