From ffb46846f6124ba7e28ba0a27c5457b5891dbdf9 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 21 Aug 2018 06:51:04 -0600
Subject: [PATCH] link atomic

---
 src/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 463e7a9..b7a641a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -35,7 +35,7 @@ add_library(kdiff3part MODULE ${kdiff3part_PART_SRCS})
 
 set_target_properties(kdiff3part PROPERTIES DEFINE_SYMBOL KDIFF3_PART)
 target_compile_features(kdiff3part PRIVATE ${needed_features})
-target_link_libraries(kdiff3part ${KDiff3_LIBRARIES} KF5::Parts KF5::Crash)
+target_link_libraries(kdiff3part ${KDiff3_LIBRARIES} KF5::Parts KF5::Crash atomic)
 target_compile_definitions(kdiff3part PRIVATE -DTRANSLATION_DOMAIN=\"kdiff3\")
 
 install(TARGETS kdiff3part DESTINATION ${KDE_INSTALL_PLUGINDIR}/kf5/parts )
@@ -51,7 +51,7 @@ set(kdiff3_SRCS
 
 add_executable(kdiff3 ${kdiff3_SRCS})
 
-target_link_libraries(kdiff3 KF5::ConfigCore KF5::ConfigGui KF5::Parts KF5::Crash ${KDiff3_LIBRARIES} )
+target_link_libraries(kdiff3 KF5::ConfigCore KF5::ConfigGui KF5::Parts KF5::Crash ${KDiff3_LIBRARIES} atomic)
 target_compile_features(kdiff3 PRIVATE ${needed_features})
 
 install(TARGETS kdiff3 ${INSTALL_TARGETS_DEFAULT_ARGS})
-- 
2.21.0

