From 2ef432401be01de5ebb9e4545d497d2cacab9168 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 21 Aug 2018 06:51:04 -0600
Subject: [PATCH] link atomic

---
 src/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4f9adab..8165292 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -27,7 +27,7 @@ add_library(kdiff3part MODULE ${kdiff3part_PART_SRCS})
 
 set_target_properties(kdiff3part PROPERTIES DEFINE_SYMBOL KDIFF3_PART)
 target_compile_features(kdiff3part PRIVATE ${needed_features})
-target_link_libraries(kdiff3part ${KDiff3_LIBRARIES} KF5::Parts)
+target_link_libraries(kdiff3part ${KDiff3_LIBRARIES} KF5::Parts atomic)
 
 install(TARGETS kdiff3part DESTINATION ${KDE_INSTALL_PLUGINDIR}/kf5/parts )
 
@@ -42,7 +42,7 @@ set(kdiff3_SRCS
 
 add_executable(kdiff3 ${kdiff3_SRCS})
 
-target_link_libraries(kdiff3 KF5::ConfigCore KF5::ConfigGui KF5::Parts ${KDiff3_LIBRARIES} )
+target_link_libraries(kdiff3 KF5::ConfigCore KF5::ConfigGui KF5::Parts ${KDiff3_LIBRARIES} atomic)
 target_compile_features(kdiff3 PRIVATE ${needed_features})
 
 install(TARGETS kdiff3 ${INSTALL_TARGETS_DEFAULT_ARGS})
-- 
2.18.0

