diff -NaurwB librpcsecgss-0.18.orig/configure.in librpcsecgss-0.18/configure.in
--- librpcsecgss-0.18.orig/configure.in	2008-04-09 00:05:40.000000000 +0200
+++ librpcsecgss-0.18/configure.in	2008-06-12 19:05:51.000000000 +0200
@@ -12,10 +12,15 @@
 AC_PROG_RANLIB
 
 # Checks for libraries.
-PKG_CHECK_MODULES([GSSGLUE], [libgssglue >= 0.1], [],
+PKG_CHECK_MODULES([GSSGLUE], [libgssglue >= 0.1],
+  [echo GSSGLUE found; GSSAPI_IMPLEMENTATION=libgssglue],
+  [PKG_CHECK_MODULES([GSSGLUE], [heimdal-gssapi],
+    [echo HEIMDAL found; GSSAPI_IMPLEMENTATION=heimdal-gssapi],
   [AC_MSG_ERROR([Unable to locate information required to use libgssglue.
   If you have pkgconfig installed, you might try setting environment
-  variable PKG_CONFIG_PATH to /usr/local/lib/pkgconfig])])
+    variable PKG_CONFIG_PATH to /usr/local/lib/pkgconfig])])])
+
+AC_SUBST([GSSAPI_IMPLEMENTATION])
 
 # Checks for header files.
 AC_HEADER_STDC
diff -NaurwB librpcsecgss-0.18.orig/librpcsecgss.pc.in librpcsecgss-0.18/librpcsecgss.pc.in
--- librpcsecgss-0.18.orig/librpcsecgss.pc.in	2007-09-06 17:39:04.000000000 +0200
+++ librpcsecgss-0.18/librpcsecgss.pc.in	2008-06-12 19:06:40.000000000 +0200
@@ -5,7 +5,7 @@
 
 Name: librpcsecgss
 Description: Library that implements rpcsec_gss interface.
-Requires: libgssglue
+Requires: @GSSAPI_IMPLEMENTATION@
 Version: @PACKAGE_VERSION@
 Libs: -L@libdir@ -lrpcsecgss
 Cflags: -I@includedir@/rpcsecgss
