--- memory.cpp	2003/10/27 18:51:22	1.1.1.1
+++ memory.cpp	2009/07/15 21:49:27	1.1.1.1.2.1
@@ -1,7 +1,7 @@
 //=========================================================
 //  MusE
 //  Linux Music Editor
-//  $Id: memory.cpp,v 1.1.1.1 2003/10/27 18:51:22 wschweer Exp $
+//  $Id: memory.cpp,v 1.1.1.1.2.1 2009/07/15 21:49:27 spamatica Exp $
 //
 //  (C) Copyright 2003 Werner Schweer (ws@seh.de)
 //=========================================================
@@ -48,7 +48,14 @@
       {
 //      printf("grow memory idx %d\n", idx);
 
-      int esize = (idx+1) * sizeof(int);
+//      int esize = (idx+1) * sizeof(int);
+      int esize = (idx+1) * sizeof(int)+2;
+      // (rj) changed to +2 due to bug 2819312 
+      // "1.0rc3 malloc problem w/glibc-2.10 on x86_64 (with fix)"
+      // the change apparently allows muse to launch with this 
+      // configuration. It is however uncertain that this is the right fix. 
+      // Probably it means that nelem will not become bigger than
+      // the allowed number of elements
 
       Chunk* n    = new Chunk;
       n->next     = chunks[idx];
