diff -ur PyQwt-5.2.0.orig/qt4lib/PyQt4/Qwt5/qplt.py PyQwt-5.2.0/qt4lib/PyQt4/Qwt5/qplt.py
--- PyQwt-5.2.0.orig/qt4lib/PyQt4/Qwt5/qplt.py	2014-01-31 15:56:00.000000000 +0100
+++ PyQwt-5.2.0/qt4lib/PyQt4/Qwt5/qplt.py	2014-01-31 15:57:03.000000000 +0100
@@ -204,7 +204,7 @@
                 self.setTitle(text)
             elif isinstance(item, int):
                 self.setZoomerMouseEventSet(item)
-            elif (isinstance(item, tuple) and len(tuple) == 2
+            elif (isinstance(item, tuple) and len(item) == 2
                   and isinstance(item[0], int) and isinstance(item[1], int)):
                 self.size = item
             elif isinstance(item, QWidget): # accept a parent silently
