--- ./config.h.org	2005-04-11 10:30:55.000000000 -0700
+++ ./config.h	2006-04-26 17:42:04.000000000 -0700
@@ -110,10 +110,12 @@
 /* dirs */
 #define DPATH		IRCD_PREFIX                                                     
 #define BINPATH		IRCD_PREFIX "/bin/"
-#define ETCPATH		IRCD_PREFIX "/etc"
-#define LOGPATH		IRCD_PREFIX "/logs"
+#define ETCPATH "/etc/ircd"
+#define LOGPATH "/var/log/ircd"
 
 /* files */
+#define	PPATH		"/var/run/ircd/ircd.pid"
+
 #define SPATH		BINPATH "/ircd"			/* ircd executable */
 
 #define CPATH		ETCPATH "/ircd.conf"		/* ircd.conf file */
@@ -125,7 +127,6 @@
 #define	SMPATH		ETCPATH "/ircd.smotd"		/* Short Message Of The Day */
 #define OMPATH		ETCPATH "/ircd.opermotd"	/* Operator Message Of The Day */
 #define RPATH		ETCPATH "/ircd.rules"		/* Rules File */
-#define	PPATH		ETCPATH "/ircd.pid"		/* IRCd PID */
 #define IRCDSSL_CPATH 	ETCPATH "/ircd.crt"		/* Client SSL */
 #define IRCDSSL_KPATH 	ETCPATH "/ircd.key"		/* Client SSL */
 
