From b268e9d9ca0fa04c85fbbedd54e57eb429c787e8 Mon Sep 17 00:00:00 2001
From: Sergey Astanin <s.astanin@gmail.com>
Date: Mon, 20 Jul 2009 15:51:04 +0000
Subject: add keymap for Samsung X22

Reported on hal ML.
---
diff --git a/fdi/information/10freedesktop/30-keymap-misc.fdi b/fdi/information/10freedesktop/30-keymap-misc.fdi
index c4be0aa..c4e527e 100644
--- a/fdi/information/10freedesktop/30-keymap-misc.fdi
+++ b/fdi/information/10freedesktop/30-keymap-misc.fdi
@@ -134,7 +134,7 @@
       </match>
 
       <match key="/org/freedesktop/Hal/devices/computer:system.hardware.vendor" prefix_ncase="samsung">
-	<match key="/org/freedesktop/Hal/devices/computer:system.hardware.product" contains_outof="NC10;SP55S;SQ45S70S;SX60P;SX30S;R59P/R60P/R61P;Q210;Q310;X05;P560;R560">
+	<match key="/org/freedesktop/Hal/devices/computer:system.hardware.product" contains_outof="NC10;SP55S;SQ45S70S;SX60P;SX30S;SX22S;R59P/R60P/R61P;Q210;Q310;X05;P560;R560">
           <append key="input.keymap.data" type="strlist">e002:switchvideomode</append> <!-- Fn+F4 CRT/LCD (high keycode: "displaytoggle") -->
           <append key="input.keymap.data" type="strlist">e003:battery</append> <!-- Fn+F2 battery icon -->
           <append key="input.keymap.data" type="strlist">e004:prog1</append> <!-- Fn+F5 backlight on/off -->
--
cgit v0.8.2
From 91d54afdaedcafa605b92e9c17b965513eeb64ba Mon Sep 17 00:00:00 2001
From: Arian Sanusi <arian@sanusi.de>
Date: Fri, 31 Jul 2009 20:02:16 +0000
Subject: suspend quirks for HP tc4x00 series

As reported on hal ML.
---
diff --git a/fdi/information/10freedesktop/20-video-quirk-pm-hp.fdi b/fdi/information/10freedesktop/20-video-quirk-pm-hp.fdi
index cd8f41e..4a3016c 100644
--- a/fdi/information/10freedesktop/20-video-quirk-pm-hp.fdi
+++ b/fdi/information/10freedesktop/20-video-quirk-pm-hp.fdi
@@ -121,7 +121,7 @@
         <merge key="power_management.quirk.s3_bios" type="bool">true</merge>
         <merge key="power_management.quirk.vbemode_restore" type="bool">true</merge>
       </match>
-      <match key="system.hardware.product" contains_outof="nx6120;nc4200;HP 530 Notebook PC">
+      <match key="system.hardware.product" contains_outof="tc4400;nc4400;tc4200;nx6120;nc4200;HP 530 Notebook PC">
         <merge key="power_management.quirk.vbe_post" type="bool">true</merge>
       </match>
 
--
cgit v0.8.2
