diff -Naur qalculate-kde-0.9.6/configure.in qalculate-kde-0.9.6.new/configure.in
--- qalculate-kde-0.9.6/configure.in	2007-06-17 07:33:58.000000000 -0400
+++ qalculate-kde-0.9.6.new/configure.in	2008-06-15 07:20:32.000000000 -0400
@@ -73,11 +73,7 @@
 
 AC_LANG([C++])
 
-AC_PATH_CLN(1.1.0, [
-     	LIBS="$LIBS $CLN_LIBS"
-     	CPPFLAGS="$CPPFLAGS $CLN_CPPFLAGS"
-  	AC_DEFINE(HAVE_LIBCLN,1,[If CLN can be used])
-], AC_MSG_ERROR([No suitable installed version of CLN could be found.]))
+PKG_CHECK_MODULES([CLN], [cln >= 1.1.0])
 
 PKG_CHECK_MODULES(QALCULATE, [
 	libxml-2.0
diff -Naur qalculate-kde-0.9.6/configure.in.in qalculate-kde-0.9.6.new/configure.in.in
--- qalculate-kde-0.9.6/configure.in.in	2007-05-18 04:10:17.000000000 -0400
+++ qalculate-kde-0.9.6.new/configure.in.in	2008-06-15 07:20:47.000000000 -0400
@@ -8,11 +8,7 @@
 
 AC_LANG([C++])
 
-AC_PATH_CLN(1.1.0, [
-     	LIBS="$LIBS $CLN_LIBS"
-     	CPPFLAGS="$CPPFLAGS $CLN_CPPFLAGS"
-  	AC_DEFINE(HAVE_LIBCLN,1,[If CLN can be used])
-], AC_MSG_ERROR([No suitable installed version of CLN could be found.]))
+PKG_CHECK_MODULES([CLN], [cln >= 1.1.0])
 
 PKG_CHECK_MODULES(QALCULATE, [
 	libxml-2.0
