--- util-linux-ng-2.16/fsck/Makefile.am	2009-07-04 01:20:01.000000000 +0200
+++ util-linux-ng-2.16.new/fsck/Makefile.am	2009-08-01 14:16:31.000000000 +0200
@@ -1,17 +1,25 @@
 include $(top_srcdir)/config/include-Makefile.am
 
-sbin_PROGRAMS = fsck
+sbin_PROGRAMS = fsck fsck.static
 dist_man_MANS = fsck.8
 
 fsck_SOURCES = base_device.c fsck.c fsck.h ../lib/ismounted.c \
 	       ../lib/fsprobe.c ../lib/canonicalize.c
+fsck_static_SOURCES = $(fsck_SOURCES)
 fsck_LDADD =
 fsck_CFLAGS =
+fsck_static_LDADD =
+fsck_static_CFLAGS =
+fsck_static_LDFLAGS = -all-static
 
 if BUILD_LIBBLKID
 fsck_LDADD += $(ul_libblkid_la)
 fsck_CFLAGS += -I$(ul_libblkid_srcdir)
+fsck_static_LDADD += $(ul_libblkid_la)
+fsck_static_CFLAGS += -I$(ul_libblkid_srcdir)
 else
 fsck_LDADD += $(BLKID_LIBS)
 fsck_CFLAGS += $(BLKID_CFLAGS)
+fsck_static_LDADD += $(BLKID_LIBS)
+fsck_static_CFLAGS += $(BLKID_CFLAGS)
 endif
