diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index a0c41ac..e9a94f6 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -157,7 +157,7 @@
 #define CONFIG_SYS_NO_FLASH
 
 #define CONFIG_SYS_MONITOR_LEN		(512 << 10)	/* 512 KiB */
-#define CONFIG_IDENT_STRING		" Allwinner Technology"
+#define CONFIG_IDENT_STRING		" Arch Linux ARM"
 
 #define CONFIG_ENV_OFFSET		(544 << 10) /* (8 + 24 + 512) KiB */
 #define CONFIG_ENV_SIZE			(128 << 10)	/* 128 KiB */
@@ -215,38 +215,28 @@
 	"bootscr=boot.scr\0" \
 	"script=script.bin\0" \
 	"loadbootscr=" \
-	  "fatload $device $partition $scriptaddr ${bootscr}" \
-	  " || " \
-	  "ext2load $device $partition $scriptaddr boot/${bootscr}" \
+	  "load $device $partition $scriptaddr boot/${bootscr}" \
 	  " ||" \
-	  "ext2load $device $partition $scriptaddr ${bootscr}" \
+	  "load $device $partition $scriptaddr ${bootscr}" \
 	  "\0" \
 	"loadbootenv=" \
-	  "fatload $device $partition $scriptaddr ${bootenv}" \
-	  " || " \
-	  "ext2load $device $partition $scriptaddr boot/${bootenv}" \
+	  "load $device $partition $scriptaddr boot/${bootenv}" \
 	  " || " \
-	  "ext2load $device $partition $scriptaddr ${bootenv}" \
+	  "load $device $partition $scriptaddr ${bootenv}" \
 	  "\0" \
 	"loadkernel=" \
 	  "if "\
 	    "bootpath=/boot/" \
 	    " && " \
-	    "ext2load $device $partition 0x43000000 ${bootpath}${script}" \
-	    " && " \
-	    "ext2load $device $partition 0x48000000 ${bootpath}${kernel}" \
-	  ";then true; elif " \
-	    "bootpath=/" \
-	    " && " \
-	    "fatload $device $partition 0x43000000 ${script}" \
+	    "load $device $partition 0x43000000 ${bootpath}${script}" \
 	    " && " \
-	    "fatload $device $partition 0x48000000 ${kernel}" \
+	    "load $device $partition 0x48000000 ${bootpath}${kernel}" \
 	  ";then true; elif " \
 	    "bootpath=/" \
 	    " && " \
-	    "ext2load $device $partition 0x43000000 ${bootpath}${script}" \
+	    "load $device $partition 0x43000000 ${bootpath}${script}" \
 	    " && " \
-	    "ext2load $device $partition 0x48000000 ${bootpath}${kernel}" \
+	    "load $device $partition 0x48000000 ${bootpath}${kernel}" \
 	  ";then true; else "\
 	    "false" \
 	  ";fi" \
@@ -272,7 +262,7 @@
 	  "\0" \
 	""
 
-#define CONFIG_BOOTDELAY	3
+#define CONFIG_BOOTDELAY	1
 #define CONFIG_SYS_BOOT_GET_CMDLINE
 #define CONFIG_AUTO_COMPLETE
 
@@ -287,6 +277,7 @@
 #define CONFIG_FAT_WRITE	/* enable write access */
 #define CONFIG_CMD_EXT2		/* with this we can access ext2 bootfs */
 #define CONFIG_CMD_EXT4		/* with this we can access ext4 bootfs */
+#define CONFIG_CMD_FS_GENERIC
 
 #define CONFIG_SPL_FRAMEWORK
 #define CONFIG_SPL_LIBCOMMON_SUPPORT
