diff -urN a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
--- a/include/configs/sunxi-common.h	2013-07-26 07:45:03.000000000 -0600
+++ b/include/configs/sunxi-common.h	2013-07-30 21:50:31.691055539 -0600
@@ -148,7 +148,7 @@
 #define CONFIG_SYS_NO_FLASH
 
 #define CONFIG_SYS_MONITOR_LEN		(512 << 10)	/* 512 KiB */
-#define CONFIG_IDENT_STRING		" Allwinner Technology"
+#define CONFIG_IDENT_STRING		" Arch Linux ARM"
 
 #define CONFIG_ENV_OFFSET		(544 << 10) /* (8 + 24 + 512) KiB */
 #define CONFIG_ENV_SIZE			(128 << 10)	/* 128 KiB */
@@ -191,13 +191,7 @@
 	"device=mmc\0" \
 	"partition=0:1\0" \
 	"setargs=" \
-	  "if test -z \\\\\"$root\\\\\"; then"\
-	    " if test \\\\\"$bootpath\\\\\" = \"boot/\"; then"\
-	      " root=\"/dev/mmcblk0p1 rootwait\";"\
-	    " else" \
-	      " root=\"/dev/mmcblk0p2 rootwait\";"\
-	    " fi;"\
-	  " fi;"\
+	  " root=\"/dev/mmcblk0p1 rw rootwait\";"\
 	  " setenv bootargs console=${console} root=${root}" \
 	  " loglevel=${loglevel} ${panicarg} ${extraargs}" \
 	  "\0" \
@@ -205,38 +199,20 @@
 	"bootenv=uEnv.txt\0" \
 	"bootscr=boot.scr\0" \
 	"loadbootscr=" \
-	  "fatload $device $partition $scriptaddr ${bootscr}" \
+	  "load $device $partition $scriptaddr boot/${bootscr}" \
 	  " || " \
-	  "ext2load $device $partition $scriptaddr boot/${bootscr}" \
-	  " ||" \
-	  "ext2load $device $partition $scriptaddr ${bootscr}" \
+	  "load $device $partition $scriptaddr ${bootscr}" \
 	  "\0" \
 	"loadbootenv=" \
-	  "fatload $device $partition $scriptaddr ${bootenv}" \
+	  "load $device $partition $scriptaddr boot/${bootenv}" \
 	  " || " \
-	  "ext2load $device $partition $scriptaddr boot/${bootenv}" \
-	  " || " \
-	  "ext2load $device $partition $scriptaddr ${bootenv}" \
+	  "load $device $partition $scriptaddr ${bootenv}" \
 	  "\0" \
 	"loadkernel=" \
 	  "if "\
-	    "bootpath=/boot/" \
-	    " && " \
-	    "ext2load $device $partition 0x43000000 ${bootpath}script.bin" \
-	    " && " \
-	    "ext2load $device $partition 0x48000000 ${bootpath}${kernel}" \
-	  ";then true; elif " \
-	    "bootpath=/" \
-	    " && " \
-	    "fatload $device $partition 0x43000000 script.bin" \
-	    " && " \
-	    "fatload $device $partition 0x48000000 ${kernel}" \
-	  ";then true; elif " \
-	    "bootpath=/" \
-	    " && " \
-	    "ext2load $device $partition 0x43000000 ${bootpath}script.bin" \
+	    "load $device $partition 0x43000000 /boot/script.bin" \
 	    " && " \
-	    "ext2load $device $partition 0x48000000 ${bootpath}${kernel}" \
+	    "load $device $partition 0x48000000 /boot/${kernel}" \
 	  ";then true; else "\
 	    "false" \
 	  ";fi" \
@@ -262,7 +238,7 @@
 	  "\0" \
 	""
 
-#define CONFIG_BOOTDELAY	3
+#define CONFIG_BOOTDELAY	1
 #define CONFIG_SYS_BOOT_GET_CMDLINE
 #define CONFIG_AUTO_COMPLETE
 
@@ -277,6 +253,7 @@
 #define CONFIG_FAT_WRITE	/* enable write access */
 #define CONFIG_CMD_EXT2		/* with this we can access ext2 bootfs */
 #define CONFIG_CMD_EXT4		/* with this we can access ext4 bootfs */
+#define CONFIG_CMD_FS_GENERIC
 #define CONFIG_CMD_ZFS		/* with this we can access ZFS bootfs */
 
 #define CONFIG_SPL_FRAMEWORK
