diff -ruN u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0.orig/arch/arm/cpu/armv7/sunxi/board.c u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0/arch/arm/cpu/armv7/sunxi/board.c
--- u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0.orig/arch/arm/cpu/armv7/sunxi/board.c	2014-07-03 11:47:02.000000000 +0200
+++ u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0/arch/arm/cpu/armv7/sunxi/board.c	2014-07-07 23:43:08.022343315 +0200
@@ -148,6 +148,10 @@
 	}
 #endif
 
+#ifdef CONFIG_BANANAPI
+      gpio_direction_output(SUNXI_GPH(23), 1);
+#endif
+
 #ifdef CONFIG_SUNXI_GMAC
 	rc = sunxi_gmac_initialize(bis);
 	if (rc < 0) {
diff -ruN u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0.orig/board/sunxi/dram_bananapi.c u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0/board/sunxi/dram_bananapi.c
--- u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0.orig/board/sunxi/dram_bananapi.c	1970-01-01 01:00:00.000000000 +0100
+++ u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0/board/sunxi/dram_bananapi.c	2014-07-07 23:41:57.491255249 +0200
@@ -0,0 +1,31 @@
+/* this file is generated, don't edit it yourself */
+
+#include <common.h>
+#include <asm/arch/dram.h>
+
+static struct dram_para dram_para = {
+	.clock = 480,
+	.type = 3,
+	.rank_num = 1,
+	.density = 4096,
+	.io_width = 16,
+	.bus_width = 32,
+	.cas = 9,
+	.zq = 0x7f,
+	.odt_en = 0,
+	.size = 1024,
+	.tpr0 = 0x42d899b7,
+	.tpr1 = 0xa090,
+	.tpr2 = 0x22a00,
+	.tpr3 = 0x0,
+	.tpr4 = 0x1,
+	.tpr5 = 0x0,
+	.emr1 = 0x4,
+	.emr2 = 0x10,
+	.emr3 = 0x0,
+};
+
+unsigned long sunxi_dram_init(void)
+{
+	return dramc_init(&dram_para);
+}
diff -ruN u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0.orig/board/sunxi/Makefile u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0/board/sunxi/Makefile
--- u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0.orig/board/sunxi/Makefile	2014-07-03 11:47:02.000000000 +0200
+++ u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0/board/sunxi/Makefile	2014-07-07 23:41:57.491255249 +0200
@@ -22,6 +22,7 @@
 # This is not a typo, uses the same mem settings as the a10s-olinuxino-m
 obj-$(CONFIG_AUXTEK_T004)	+= dram_a10s_olinuxino_m.o
 obj-$(CONFIG_BA10_TV_BOX)	+= dram_sun4i_384_1024_iow8.o
+obj-$(CONFIG_BANANAPI)		+= dram_bananapi.o
 obj-$(CONFIG_COBY_MID7042)	+= dram_sun4i_408_1024_iow16.o
 obj-$(CONFIG_COBY_MID8042)	+= dram_sun4i_360_1024_iow16.o
 obj-$(CONFIG_COBY_MID9742)	+= dram_sun4i_408_1024_iow16.o
diff -ruN u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0.orig/boards.cfg u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0/boards.cfg
--- u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0.orig/boards.cfg	2014-07-03 11:47:02.000000000 +0200
+++ u-boot-sunxi-38f2efaf1157af117edcff5874568df0731e1ec0/boards.cfg	2014-07-07 23:41:57.494588474 +0200
@@ -387,6 +387,7 @@
 Active  arm         armv7          sunxi       -               sunxi               Auxtek-T003                          sun5i:AUXTEK_T003,SPL,AXP152_POWER,STATUSLED=34                                                                                   -
 Active  arm         armv7          sunxi       -               sunxi               Auxtek-T004                          sun5i:AUXTEK_T004,SPL,AXP152_POWER,STATUSLED=34                                                                                   -
 Active  arm         armv7          sunxi       -               sunxi               ba10_tv_box                          sun4i:BA10_TV_BOX,SPL,SUNXI_EMAC                                                                                                  -
+Active  arm         armv7          sunxi       -               sunxi               bananapi                             sun7i:BANANAPI,SPL,SUNXI_GMAC,RGMII,STATUSLED=248,FAST_MBUS                                                                        -
 Active  arm         armv7          sunxi       -               sunxi               Coby_MID7042                         sun4i:COBY_MID7042,SPL                                                                                                            -
 Active  arm         armv7          sunxi       -               sunxi               Coby_MID8042                         sun4i:COBY_MID8042,SPL                                                                                                            -
 Active  arm         armv7          sunxi       -               sunxi               Coby_MID9742                         sun4i:COBY_MID9742,SPL                                                                                                            -
