--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -116,7 +116,17 @@
 #include <config_distro_defaults.h>
 #include <config_distro_bootcmd.h>
 
-#define CONFIG_DEFAULT_FDT_FILE		"sun7i-a20-cubietruck.dtb"
+#if defined(CONFIG_CUBIEBOARD2)
+  #define CONFIG_DEFAULT_FDT_FILE	"sun7i-a20-cubieboard2.dtb"
+#elif defined(CONFIG_CUBIETRUCK)
+  #define CONFIG_DEFAULT_FDT_FILE	"sun7i-a20-cubietruck.dtb"
+#elif defined(CONFIG_A20_OLINUXINO_M)
+  #define CONFIG_DEFAULT_FDT_FILE	"sun7i-a20-olinuxino-micro.dtb"
+#elif defined(CONFIG_HACKBERRY)
+  #define CONFIG_DEFAULT_FDT_FILE	"sun4i-a10-hackberry.dtb"
+#else
+  #define CONFIG_DEFAULT_FDT_FILE	""
+#endif
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"kernel_addr_r=0x42000000\0" \
