diff -u -r firefox-52.9.0esr/js/src/jsapi-tests/moz.build firefox-52.9.0esr-noerror/js/src/jsapi-tests/moz.build
--- firefox-52.9.0esr/js/src/jsapi-tests/moz.build	2017-04-11 02:13:16.000000000 +0000
+++ firefox-52.9.0esr-noerror/js/src/jsapi-tests/moz.build	2019-06-20 20:10:31.092230735 +0000
@@ -147,7 +147,7 @@
 OS_LIBS += CONFIG['MOZ_ZLIB_LIBS']
 
 if CONFIG['GNU_CXX']:
-    CXXFLAGS += ['-Wno-shadow', '-Werror=format']
+    CXXFLAGS += ['-Wno-shadow']
 
 # This is intended as a temporary workaround to enable VS2015.
 if CONFIG['_MSC_VER']:
