From 71fa379cb561f66d60e3507cd823ce3b438b4454 Mon Sep 17 00:00:00 2001
From: warped-rudi <r.ihle@s-t.de>
Date: Fri, 19 Oct 2012 07:56:19 +0200
Subject: [PATCH] Added adapter ID interface

---
 src/lib/adapter/CuBox/NxpCECAdapterCommunication.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/lib/adapter/CuBox/NxpCECAdapterCommunication.h b/src/lib/adapter/CuBox/NxpCECAdapterCommunication.h
index 066c032..42e611a 100644
--- a/src/lib/adapter/CuBox/NxpCECAdapterCommunication.h
+++ b/src/lib/adapter/CuBox/NxpCECAdapterCommunication.h
@@ -39,6 +39,8 @@
 #include "lib/adapter/AdapterCommunication.h"
 #include <map>
 
+#define NXP_ADAPTER_VID 0x0471
+#define NXP_ADAPTER_PID 0x1001
 
 namespace PLATFORM
 {
@@ -84,6 +86,8 @@ namespace CEC
     cec_vendor_id GetVendorId(void);
     bool SupportsSourceLogicalAddress(const cec_logical_address address) { return address > CECDEVICE_TV && address <= CECDEVICE_BROADCAST; }
     cec_adapter_type GetAdapterType(void) { return ADAPTERTYPE_TDA995x; }
+    uint16_t GetAdapterVendorId(void) const { return NXP_ADAPTER_VID; }
+    uint16_t GetAdapterProductId(void) const { return NXP_ADAPTER_PID; }
     void HandleLogicalAddressLost(cec_logical_address oldAddress);
     ///}
 
-- 
1.7.12

