diff -Naur mailx-8.1.1-fixed-orig/Makefile mailx-8.1.1-fixed/Makefile
--- mailx-8.1.1-fixed-orig/Makefile	Thu Mar 16 09:30:36 2000
+++ mailx-8.1.1-fixed/Makefile	Wed Feb 27 12:50:09 2002
@@ -7,18 +7,20 @@
 BINOWN = root
 BINGRP = mail
 
+DESTDIR = 
+
 all:
 	gcc $(SRCS) -o $(PROG)
 
 install:
-	mkdir -p /usr/share/misc
-	mkdir -p /usr/share/man/man1
-	mkdir -p /etc
-	mkdir -p /usr/bin
-	install -o $(BINOWN) -g $(BINGRP) -m 2755 $(PROG) /usr/bin
-	install -o root -g root -m 644 mail.1 /usr/share/man/man1
+	mkdir -p $(DESTDIR)/usr/share/misc
+	mkdir -p $(DESTDIR)/usr/man/man1
+	mkdir -p $(DESTDIR)/etc
+	mkdir -p $(DESTDIR)/usr/bin
+	install -o $(BINOWN) -g $(BINGRP) -m 2755 $(PROG) $(DESTDIR)/usr/bin
+	install -o root -g root -m 644 mail.1 $(DESTDIR)/usr/man/man1
 	cd misc; install -c -o ${BINOWN} -g ${BINGRP} \
-	    -m 444 ${SFILES} /usr/share/misc
+	    -m 444 ${SFILES} $(DESTDIR)/usr/share/misc
 	cd misc; install -c -o root -g root \
-	    -m 644 ${EFILES} /etc
+	    -m 644 ${EFILES} $(DESTDIR)/etc
 
