--- tw.color.c.orig	2009-07-27 01:45:38.000000000 -0400
+++ tw.color.c	2009-07-27 01:47:55.000000000 -0400
@@ -86,6 +86,8 @@
     VAR(NOS, "ow", ""),		/* Other writable dir (o+w) but not sticky */
     VAR(NOS, "st", ""),		/* Sticky dir (+t) but not other writable */
     VAR(NOS, "rs", "0"),	/* Reset to normal color */
+    VAR(NOS, "hl", ""),		/* Regular file with more than one link */
+    VAR(NOS, "ca", ""),		/* File with capability */
 };
 
 enum FileType {
