diff -aur d4x-2.5.7.1/configure d4x-2.5.7.1/configure
--- d4x-2.5.7.1/configure	2006-04-12 18:48:29.000000000 +0200
+++ d4x-2.5.7.1/configure	2008-06-03 23:44:14.000000000 +0200
@@ -18396,12 +18396,6 @@
 #define _REENTRANT
 _ACEOF
 
-
-cat >>confdefs.h <<\_ACEOF
-#define GTK_DISABLE_DEPRECATED
-_ACEOF
-
-
 cat >>confdefs.h <<\_ACEOF
 #define _FILE_OFFSET_BITS 64
 _ACEOF
diff -aur d4x-2.5.7.1/configure.ac d4x-2.5.7.1/configure.ac
--- d4x-2.5.7.1/configure.ac	2005-11-05 00:35:38.000000000 +0100
+++ d4x-2.5.7.1/configure.ac	2008-06-03 23:43:01.000000000 +0200
@@ -298,7 +298,6 @@
 
 # Other needed defines
 AC_DEFINE(_REENTRANT, , [Enable reenrant code of glibc])
-AC_DEFINE(GTK_DISABLE_DEPRECATED, , [Enable warnings about deprecated functions usage])
 AC_DEFINE(_FILE_OFFSET_BITS, 64, [To support files over 2GB])
 
 # Prepare all requiered flags
diff -aur d4x-2.5.7.1/main/addr.cc d4x-2.5.7.1/main/addr.cc
--- d4x-2.5.7.1/main/addr.cc	2006-04-05 20:19:36.000000000 +0200
+++ d4x-2.5.7.1/main/addr.cc	2008-06-03 23:42:00.000000000 +0200
@@ -181,7 +181,6 @@
 			file=path.substr(p+1);
 			path=path.substr(0,p);
 		};
-		path=d4x::Path("/")/path;
 		
 		if (!port)
 			port=get_port_by_proto(proto);
diff -aur d4x-2.5.7.1/main/face/list.cc d4x-2.5.7.1/main/face/list.cc
--- d4x-2.5.7.1/main/face/list.cc	2006-03-20 01:29:41.000000000 +0100
+++ d4x-2.5.7.1/main/face/list.cc	2008-06-03 23:42:18.000000000 +0200
@@ -8,7 +8,6 @@
  *	but WITHOUT ANY WARRANTY; without even the implied warranty of
  *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
  */
-#define GTK_DISABLE_DEPRECATED
 
 #include <stdio.h>
 #include <gtk/gtk.h>
