Index: art_misc.h
===================================================================
--- art_misc.h	(revision 277)
+++ art_misc.h	(working copy)
@@ -34,9 +34,15 @@
 #include <libart_lgpl/art_config.h>
 #endif
 
+#ifdef __cplusplus
+extern "C" {
+#endif
 void *art_alloc(size_t size);
 void art_free(void *ptr);
 void *art_realloc(void *ptr, size_t size);
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
 
 /* These aren't, strictly speaking, configuration macros, but they're
    damn handy to have around, and may be worth playing with for
Index: ChangeLog
===================================================================
--- ChangeLog	(revision 277)
+++ ChangeLog	(working copy)
@@ -1,3 +1,10 @@
+2007-03-01  Frederic Crozat  <fcrozat@mandriva.com>
+
+	reviewed by: Dom Lachowicz <cinamod@hotmail.com>
+
+	* art_misc.h: Fix header when included in C++.
+	Patch from Goetz Waschk and Laurent Montel.
+
 2007-02-28  Kjartan Maraas  <kmaraas@gnome.org>
 
 	configure.in: Release 2.3.19
