diff -wbBur pacman-1_002/ghost.cc pacman-1_002.my/ghost.cc
--- pacman-1_002/ghost.cc	1995-07-21 19:35:30.000000000 +0400
+++ pacman-1_002.my/ghost.cc	2007-07-03 19:18:56.000000000 +0400
@@ -165,7 +165,7 @@
 	    }
 }
 
-Ghost::go(Pacman *pac) {	//go for pacman: do ghost code
+int Ghost::go(Pacman *pac) {	//go for pacman: do ghost code
 typ w;				//what type is at next coordinates
 int i=1;			//moved or not?
 int xx,yy;			//coordinates
diff -wbBur pacman-1_002/ghost.h pacman-1_002.my/ghost.h
--- pacman-1_002/ghost.h	1995-04-07 21:02:34.000000000 +0400
+++ pacman-1_002.my/ghost.h	2007-07-03 19:19:01.000000000 +0400
@@ -31,7 +31,7 @@
 Ghost(COLOURTYPE,int,int);		//constructor with coordinates
 ~Ghost();				//destructor
 void draw(void);			//draw the ghost
-go(Pacman*);				//go for pacman: do ghost code
+int go(Pacman*);				//go for pacman: do ghost code
 typ is_a(void) {return classGhost;}    
 void eat(void);				//when pacman eats ghost
 void getxy(int*,int*);
diff -wbBur pacman-1_002/point.cc pacman-1_002.my/point.cc
--- pacman-1_002/point.cc	1995-02-28 13:41:08.000000000 +0300
+++ pacman-1_002.my/point.cc	2007-07-03 19:19:16.000000000 +0400
@@ -1,7 +1,7 @@
 #include"point.h"
 Point::Point(int xx=0,int yy=0) { x=xx; y=yy; }; 
-Point::val_x() { return x; };
-Point::val_y() { return y; };
+int Point::val_x() { return x; };
+int Point::val_y() { return y; };
 void Point::set_x(int xx) { x=xx; };
 void Point::set_y(int yy) { y=yy; };
 void Point::set_xy(int xx,int yy) { x=xx; y=yy; };
