diff -ruN supertux-0.3.1.orig/src/console.hpp supertux-0.3.1/src/console.hpp
--- supertux-0.3.1.orig/src/console.hpp	2008-01-03 21:59:27.000000000 +0100
+++ supertux-0.3.1/src/console.hpp	2008-08-23 11:57:47.000000000 +0200
@@ -27,6 +27,8 @@
 #include <sstream>
 #include <iostream>
 #include <squirrel.h>
+#include <memory>
+#include <cstdlib>
 
 class Console;
 class ConsoleStreamBuffer;
diff -ruN supertux-0.3.1.orig/src/lisp/lexer.cpp supertux-0.3.1/src/lisp/lexer.cpp
--- supertux-0.3.1.orig/src/lisp/lexer.cpp	2008-01-03 21:59:26.000000000 +0100
+++ supertux-0.3.1/src/lisp/lexer.cpp	2008-08-23 13:45:58.000000000 +0200
@@ -21,6 +21,7 @@
 #include <sstream>
 #include <stdexcept>
 #include <iostream>
+#include <cstring>
 
 #include "lexer.hpp"
 
diff -ruN supertux-0.3.1.orig/src/sprite/sprite_manager.hpp supertux-0.3.1/src/sprite/sprite_manager.hpp
--- supertux-0.3.1.orig/src/sprite/sprite_manager.hpp	2008-01-03 21:59:26.000000000 +0100
+++ supertux-0.3.1/src/sprite/sprite_manager.hpp	2008-08-23 13:29:34.000000000 +0200
@@ -21,6 +21,7 @@
 #define SUPERTUX_SPRITE_MANAGER_H
 
 #include <map>
+#include <string>
 
 class SpriteData;
 class Sprite;
diff -ruN supertux-0.3.1.orig/src/textscroller.hpp supertux-0.3.1/src/textscroller.hpp
--- supertux-0.3.1.orig/src/textscroller.hpp	2008-01-03 21:59:27.000000000 +0100
+++ supertux-0.3.1/src/textscroller.hpp	2008-08-23 13:34:35.000000000 +0200
@@ -24,6 +24,7 @@
 #include <vector>
 #include <string>
 #include <map>
+#include <memory>
 
 #include "screen.hpp"
 #include "math/vector.hpp"
diff -ruN supertux-0.3.1.orig/src/title.cpp supertux-0.3.1/src/title.cpp
--- supertux-0.3.1.orig/src/title.cpp	2008-01-03 21:59:28.000000000 +0100
+++ supertux-0.3.1/src/title.cpp	2008-08-23 12:05:19.000000000 +0200
@@ -32,6 +32,7 @@
 #include <SDL.h>
 #include <SDL_image.h>
 #include <physfs.h>
+#include <algorithm>
 
 #include "title.hpp"
 #include "mainloop.hpp"
diff -ruN supertux-0.3.1.orig/src/video/sdl_texture.hpp supertux-0.3.1/src/video/sdl_texture.hpp
--- supertux-0.3.1.orig/src/video/sdl_texture.hpp	2008-01-03 21:59:22.000000000 +0100
+++ supertux-0.3.1/src/video/sdl_texture.hpp	2008-08-23 13:37:14.000000000 +0200
@@ -23,6 +23,7 @@
 #include <config.h>
 
 #include <SDL.h>
+#include <algorithm>
 
 #include "texture.hpp"
 #include "color.hpp"
