diff -ruN supertux-0.3.1.orig/src/console.hpp supertux-0.3.1/src/console.hpp
--- supertux-0.3.1.orig/src/console.hpp	2008-01-03 20:59:27.000000000 +0000
+++ supertux-0.3.1/src/console.hpp	2009-05-17 19:43:26.914307646 +0000
@@ -27,6 +27,8 @@
 #include <sstream>
 #include <iostream>
 #include <squirrel.h>
+#include <memory>
+#include <cstdlib>
 
 class Console;
 class ConsoleStreamBuffer;
diff -ruN supertux-0.3.1.orig/src/lisp/lexer.cpp supertux-0.3.1/src/lisp/lexer.cpp
--- supertux-0.3.1.orig/src/lisp/lexer.cpp	2008-01-03 20:59:26.000000000 +0000
+++ supertux-0.3.1/src/lisp/lexer.cpp	2009-05-17 19:43:27.037705317 +0000
@@ -21,6 +21,7 @@
 #include <sstream>
 #include <stdexcept>
 #include <iostream>
+#include <cstring>
 
 #include "lexer.hpp"
 
diff -ruN supertux-0.3.1.orig/src/lisp/lisp.cpp supertux-0.3.1/src/lisp/lisp.cpp
--- supertux-0.3.1.orig/src/lisp/lisp.cpp	2008-01-03 20:59:26.000000000 +0000
+++ supertux-0.3.1/src/lisp/lisp.cpp	2009-05-17 19:44:29.305165638 +0000
@@ -18,7 +18,7 @@
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
 #include <config.h>
-
+#include <cstdio>
 #include "lisp.hpp"
 
 namespace lisp
diff -ruN supertux-0.3.1.orig/src/random_generator.cpp supertux-0.3.1/src/random_generator.cpp
--- supertux-0.3.1.orig/src/random_generator.cpp	2008-01-03 20:59:27.000000000 +0000
+++ supertux-0.3.1/src/random_generator.cpp	2009-05-17 19:44:05.851002254 +0000
@@ -39,6 +39,7 @@
 #include <stdexcept>
 #include <time.h>
 #include <cassert>
+#include <cstdio>
 #include "random_generator.hpp"
 
 RandomGenerator systemRandom;               // global random number generator
diff -ruN supertux-0.3.1.orig/src/sprite/sprite_manager.hpp supertux-0.3.1/src/sprite/sprite_manager.hpp
--- supertux-0.3.1.orig/src/sprite/sprite_manager.hpp	2008-01-03 20:59:26.000000000 +0000
+++ supertux-0.3.1/src/sprite/sprite_manager.hpp	2009-05-17 19:43:27.037705317 +0000
@@ -21,6 +21,7 @@
 #define SUPERTUX_SPRITE_MANAGER_H
 
 #include <map>
+#include <string>
 
 class SpriteData;
 class Sprite;
diff -ruN supertux-0.3.1.orig/src/textscroller.hpp supertux-0.3.1/src/textscroller.hpp
--- supertux-0.3.1.orig/src/textscroller.hpp	2008-01-03 20:59:27.000000000 +0000
+++ supertux-0.3.1/src/textscroller.hpp	2009-05-17 19:43:27.037705317 +0000
@@ -24,6 +24,7 @@
 #include <vector>
 #include <string>
 #include <map>
+#include <memory>
 
 #include "screen.hpp"
 #include "math/vector.hpp"
diff -ruN supertux-0.3.1.orig/src/title.cpp supertux-0.3.1/src/title.cpp
--- supertux-0.3.1.orig/src/title.cpp	2008-01-03 20:59:28.000000000 +0000
+++ supertux-0.3.1/src/title.cpp	2009-05-17 19:43:27.041000359 +0000
@@ -32,6 +32,7 @@
 #include <SDL.h>
 #include <SDL_image.h>
 #include <physfs.h>
+#include <algorithm>
 
 #include "title.hpp"
 #include "mainloop.hpp"
diff -ruN supertux-0.3.1.orig/src/video/sdl_texture.hpp supertux-0.3.1/src/video/sdl_texture.hpp
--- supertux-0.3.1.orig/src/video/sdl_texture.hpp	2008-01-03 20:59:22.000000000 +0000
+++ supertux-0.3.1/src/video/sdl_texture.hpp	2009-05-17 19:43:27.041000359 +0000
@@ -23,6 +23,7 @@
 #include <config.h>
 
 #include <SDL.h>
+#include <algorithm>
 
 #include "texture.hpp"
 #include "color.hpp"
