--- configure.in.orig	2008-03-24 14:36:16.000000000 +0000
+++ configure.in	2008-03-24 14:36:24.000000000 +0000
@@ -37,7 +37,7 @@
 AC_SUBST(MONODOC_LIBS)
 
 
-PKG_CHECK_MODULES(GTK_SHARP, gtk-sharp-2.0 glade-sharp-2.0 gconf-sharp-2.0)
+PKG_CHECK_MODULES(GTK_SHARP, gtk-sharp-2.0 glade-sharp-2.0 gconf-sharp-2.0 gnome-sharp-2.0)
 #PKG_CHECK_MODULES(GTK_SHARP, gtk-sharp glade-sharp gconf-sharp)
 AC_SUBST(GTK_SHARP_LIBS)
 
