--- main.cpp	2009-01-29 19:02:25.000000000 +0100
+++ main.cpp	2009-01-29 19:02:25.000000000 +0100
@@ -21,6 +21,7 @@
 #include <qtextcodec.h>
 #include "mainwindow.h"
 #include "channel.h"
+#include <stdlib.h>
 
 int
 main (int argc, char **argv)
--- record.cpp	2009-01-29 19:07:33.000000000 +0100
+++ record.cpp	2009-01-29 19:07:33.000000000 +0100
@@ -29,7 +29,7 @@
 #include "menubar.h"
 #include "utils.h"
 #include "mystatusbar.h"
-
+#include <stdlib.h>
 
 Record::Record (QWidget * parent, const char *name):
 MyHBox (parent, name),
--- sopfork.cpp	2009-01-29 19:12:55.000000000 +0100
+++ sopfork.cpp	2009-01-29 19:12:55.000000000 +0100
@@ -27,6 +27,7 @@
 #include "mystatusbar.h"
 #include "record.h"
 #include "mainwindow.h"
+#include <stdlib.h>
 
 SopFork::SopFork (QWidget * widget, QString url, QString name, QString type,
 		  QWidget * parent, const char *name2):
