diff -urN libtrash-2.1/src/Makefile libtrash-2.1-new/src/Makefile
--- libtrash-2.1/src/Makefile	2003-06-24 16:35:25.000000000 -0700
+++ libtrash-2.1-new/src/Makefile	2003-07-13 01:35:35.000000000 -0700
@@ -29,7 +29,6 @@
 	TRASH_OFF=YES install libtrash.so.${VERSION} ${INSTLIBDIR}/libtrash.so.${VERSION}
 	TRASH_OFF=YES ln -sf libtrash.so.${VERSION} ${INSTLIBDIR}/libtrash.so
 	TRASH_OFF=YES install libtrash.conf.sys --mode a=r ${SYSCONFFILE}
-	ldconfig	
 
 clean:
 	rm -f *~
