diff -aur wicd-old/etc/acpi/resume.d/80-wicd-connect.sh wicd-new/etc/acpi/resume.d/80-wicd-connect.sh
--- wicd-old/etc/acpi/resume.d/80-wicd-connect.sh	2007-07-25 20:34:25.000000000 +0530
+++ wicd-new/etc/acpi/resume.d/80-wicd-connect.sh	2007-12-20 07:19:00.000000000 +0530
@@ -1,4 +1,4 @@
 #!/bin/sh
 # Bring wifi network interface back up.
 
-/opt/wicd/autoconnect.py
+/usr/lib/wicd/autoconnect.py
diff -aur wicd-old/etc/init.d/wicd wicd-new/etc/init.d/wicd
--- wicd-old/etc/init.d/wicd	2007-07-25 20:34:25.000000000 +0530
+++ wicd-new/etc/init.d/wicd	2007-12-20 07:19:15.000000000 +0530
@@ -5,7 +5,7 @@
 	echo "Stopping any running daemons..."
 	killall daemon.py 2> /dev/null
 	echo "Starting wicd daemon..."
-	/opt/wicd/daemon.py 2> /dev/null
+	/usr/lib/wicd/daemon.py 2> /dev/null
 fi
 
 if [[ $1 = "stop" ]]
diff -aur wicd-old/opt/wicd/dapper.py wicd-new/opt/wicd/dapper.py
--- wicd-old/opt/wicd/dapper.py	2007-12-01 19:47:51.000000000 +0530
+++ wicd-new/opt/wicd/dapper.py	2007-12-20 07:20:35.000000000 +0530
@@ -76,7 +76,7 @@
     if wired.CheckPluggedIn() == True and wired_ip != None:
         # Only set image/tooltip if it hasn't been set already
         if stillWired == False:
-            pic.set_from_file("images/wired.png")
+            pic.set_from_file("/usr/share/wicd/images/wired.png")
             tooltip.set_tip(eb,language['connected_to_wired'].replace('$A',
                                                                       wired_ip))
             stillWired = True
@@ -85,7 +85,7 @@
         # Check to see if we were using a wired connection that has now become
         # unplugged or disabled.
         if stillWired == True:
-            pic.set_from_file("images/no-signal.png")
+            pic.set_from_file("/usr/share/wicd/images/no-signal.png")
             tooltip.set_tip(eb,language['not_connected'])
         stillWired = False
 
@@ -124,28 +124,28 @@
             
             if daemon.GetSignalDisplayType()== 0:
                 if wireless_signal > 75:
-                    pic.set_from_file("images/high-signal" + lock + ".png")
+                    pic.set_from_file("/usr/share/wicd/images/high-signal" + lock + ".png")
                 elif wireless_signal > 50:
-                    pic.set_from_file("images/good-signal" + lock + ".png")
+                    pic.set_from_file("/usr/share/wicd/images/good-signal" + lock + ".png")
                 elif wireless_signal > 25:
-                    pic.set_from_file("images/low-signal" + lock + ".png")
+                    pic.set_from_file("/usr/share/wicd/images/low-signal" + lock + ".png")
                 elif wireless_signal > 0:
-                    pic.set_from_file("images/bad-signal" + lock + ".png")
+                    pic.set_from_file("/usr/share/wicd/images/bad-signal" + lock + ".png")
                 elif wireless_signal == 0:
-                    pic.set_from_file("images/no-signal.png")
+                    pic.set_from_file("/usr/share/wicd/images/no-signal.png")
                     auto_reconnect()
             else:
                 if wireless_signal >= -60:
-                    pic.set_from_file("images/high-signal" + lock + ".png")
+                    pic.set_from_file("/usr/share/wicd/images/high-signal" + lock + ".png")
                 elif wireless_signal >= -70:
-                    pic.set_from_file("images/good-signal" + lock + ".png")
+                    pic.set_from_file("/usr/share/wicd/images/good-signal" + lock + ".png")
                 elif wireless_signal >= -80:
-                    pic.set_from_file("images/low-signal" + lock + ".png")
+                    pic.set_from_file("/usr/share/wicd/images/low-signal" + lock + ".png")
                 else:
-                    pic.set_from_file("images/bad-signal" + lock + ".png")
+                    pic.set_from_file("/usr/share/wicd/images/bad-signal" + lock + ".png")
 
         elif wireless_ip is None and wired_ip is None:
-            pic.set_from_file("images/no-signal")
+            pic.set_from_file("/usr/share/wicd/images/no-signal")
             tooltip.set_tip(eb,language['not_connected'])
             auto_reconnect()
 
@@ -247,7 +247,7 @@
 
 gobject.timeout_add(3000,set_signal_image)
 tooltip.set_tip(eb, "Wicd Systray")
-pic.set_from_file("images/no-signal.png")
+pic.set_from_file("/usr/share/wicd/images/no-signal.png")
 
 eb.connect('button_press_event',tray_clicked)
 eb.add(pic)
diff -aur wicd-old/opt/wicd/edgy.py wicd-new/opt/wicd/edgy.py
--- wicd-old/opt/wicd/edgy.py	2007-12-01 19:47:51.000000000 +0530
+++ wicd-new/opt/wicd/edgy.py	2007-12-20 07:20:35.000000000 +0530
@@ -158,7 +158,7 @@
     if wired.CheckPluggedIn() == True and wired_ip != None:
         # Only set image/tooltip if it hasn't been set already
         if stillWired == False:
-            tr.set_from_file("images/wired.png")
+            tr.set_from_file("/usr/share/wicd/images/wired.png")
             tr.set_tooltip(language['connected_to_wired'].replace('$A',
                                                                   wired_ip))
             stillWired = True
@@ -167,7 +167,7 @@
         # Check to see if we were using a wired connection that has now become
         # unplugged or disabled.
         if stillWired == True:
-            tr.set_from_file("images/no-signal.png")
+            tr.set_from_file("/usr/share/wicd/images/no-signal.png")
             tr.set_tooltip(language['not_connected'])
         stillWired = False
 
@@ -209,35 +209,35 @@
                            
             if daemon.GetSignalDisplayType()== 0:
                 if wireless_signal > 75:
-                    tr.set_from_file("images/" + data_string + "high-signal" + lock + ".png")
+                    tr.set_from_file("/usr/share/wicd/images/" + data_string + "high-signal" + lock + ".png")
                 elif wireless_signal > 50:
-                    tr.set_from_file("images/" + data_string + "good-signal" + lock + ".png")
+                    tr.set_from_file("/usr/share/wicd/images/" + data_string + "good-signal" + lock + ".png")
                 elif wireless_signal > 25:
-                    tr.set_from_file("images/" + data_string + "low-signal" + lock + ".png")
+                    tr.set_from_file("/usr/share/wicd/images/" + data_string + "low-signal" + lock + ".png")
                 elif wireless_signal > 0:
-                    tr.set_from_file("images/" + data_string + "bad-signal" + lock + ".png")
+                    tr.set_from_file("/usr/share/wicd/images/" + data_string + "bad-signal" + lock + ".png")
                 elif wireless_signal == 0:
-                    tr.set_from_file("images/no-signal.png")
+                    tr.set_from_file("/usr/share/wicd/images/no-signal.png")
                     auto_reconnect()
             else:
                 if wireless_signal >= -60:
-                    tr.set_from_file("images/" + data_string + "high-signal" + lock + ".png")
+                    tr.set_from_file("/usr/share/wicd/images/" + data_string + "high-signal" + lock + ".png")
                 elif wireless_signal >= -70:
-                    tr.set_from_file("images/" + data_string + "good-signal" + lock + ".png")
+                    tr.set_from_file("/usr/share/wicd/images/" + data_string + "good-signal" + lock + ".png")
                 elif wireless_signal >= -80:
-                    tr.set_from_file("images/" + data_string + "low-signal" + lock + ".png")
+                    tr.set_from_file("/usr/share/wicd/images/" + data_string + "low-signal" + lock + ".png")
                 else:
-                    tr.set_from_file("images/bad-signal" + lock + ".png")
+                    tr.set_from_file("/usr/share/wicd/images/bad-signal" + lock + ".png")
 
         elif wireless_ip is None and wired_ip is None:
-            tr.set_from_file("images/no-signal.png")
+            tr.set_from_file("/usr/share/wicd/images/no-signal.png")
             tr.set_tooltip(language['not_connected'])
             auto_reconnect()
         
         elif tr.current_icon_path.startswith(data_string) == False:
             print ":"+tr.current_icon_path
             print "|"+data_string+tr.current_icon_path[tr.current_icon_path.find('-')+1:]
-            tr.set_from_file("images/"+data_string+tr.current_icon_path[tr.current_icon_path.find('-')+1:])
+            tr.set_from_file("/usr/share/wicd/images/"+data_string+tr.current_icon_path[tr.current_icon_path.find('-')+1:])
 
     if not daemon.GetDebugMode():
         config.EnableLogging()
@@ -301,11 +301,11 @@
         self.manager.add_ui_from_string(menu)
         self.menu = self.manager.get_widget('/Menubar/Menu/About').props.parent
         self.current_icon_path = ''
-        self.set_from_file("images/no-signal.png")
+        self.set_from_file("/usr/share/wicd/images/no-signal.png")
         self.set_visible(True)
         self.connect('activate', self.on_activate)
         self.connect('popup-menu', self.on_popup_menu)
-        self.set_from_file("images/no-signal.png")
+        self.set_from_file("/usr/share/wicd/images/no-signal.png")
         self.set_tooltip("Initializing wicd...")
 
         wireless.SetForcedDisconnect(False)
diff -aur wicd-old/opt/wicd/gui.py wicd-new/opt/wicd/gui.py
--- wicd-old/opt/wicd/gui.py	2007-12-01 19:47:51.000000000 +0530
+++ wicd-new/opt/wicd/gui.py	2007-12-20 07:20:35.000000000 +0530
@@ -350,24 +350,24 @@
             # "converted" to strength bars, so suggestions from people
             # for a better way would be welcome.
             if dbm_strength >= -60:
-                self.image.set_from_file('images/signal-100.png')
+                self.image.set_from_file('/usr/share/wicd/images/signal-100.png')
             elif dbm_strength >= -70:
-                self.image.set_from_file('images/signal-75.png')
+                self.image.set_from_file('/usr/share/wicd/images/signal-75.png')
             elif dbm_strength >= -80:
-                self.image.set_from_file('images/signal-50.png')
+                self.image.set_from_file('/usr/share/wicd/images/signal-50.png')
             else:
-                self.image.set_from_file('images/signal-25.png') 
+                self.image.set_from_file('/usr/share/wicd/images/signal-25.png') 
 
         else:
             # Uses normal link quality, should be fine in most cases
             if strength > 75:
-                self.image.set_from_file('images/signal-100.png')
+                self.image.set_from_file('/usr/share/wicd/images/signal-100.png')
             elif strength > 50:
-                self.image.set_from_file('images/signal-75.png')
+                self.image.set_from_file('/usr/share/wicd/images/signal-75.png')
             elif strength > 25:
-                self.image.set_from_file('images/signal-50.png')
+                self.image.set_from_file('/usr/share/wicd/images/signal-50.png')
             else:
-                self.image.set_from_file('images/signal-25.png')
+                self.image.set_from_file('/usr/share/wicd/images/signal-25.png')
         self.expander.setSignalStrength(strength, dbm_strength)
 
     def setMACAddress(self,address):
