diff -aur wicd-old/opt/wicd/gui.py wicd-new/opt/wicd/gui.py
--- wicd-old/opt/wicd/gui.py	2008-01-03 23:18:40.000000000 +0530
+++ wicd-new/opt/wicd/gui.py	2008-01-09 00:27:29.000000000 +0530
@@ -44,7 +44,7 @@
 #which is also under GPLv2
 
 #Get the local directory since we are not installing anything
-local_path = os.path.realpath(os.path.dirname(sys.argv[0])) + '/translations'
+local_path = ('/usr/share/locale')
 # Init the list of languages to support
 langs = list()
 #Check the default locale
