diff -Naur fam-2.7.0-old/include/BTree.h fam-2.7.0/include/BTree.h
--- fam-2.7.0-old/include/BTree.h	2003-01-20 14:22:30.000000000 +1000
+++ fam-2.7.0/include/BTree.h	2009-06-14 21:21:41.000000000 +1000
@@ -24,6 +24,7 @@
 #define BTree_included
 
 #include "Boolean.h"
+#include <cstdlib>
 
 //  This is an in-core B-Tree implementation.
 //
diff -Naur fam-2.7.0-old/lib/Client.c++ fam-2.7.0/lib/Client.c++
--- fam-2.7.0-old/lib/Client.c++	2003-01-19 00:18:12.000000000 +1000
+++ fam-2.7.0/lib/Client.c++	2009-06-14 21:22:33.000000000 +1000
@@ -34,7 +34,7 @@
 #include <syslog.h>
 #include <errno.h>
 
-#include <iostream.h>
+#include <iostream>
 
 #include "fam.h"
 #include "Client.h"
diff -Naur fam-2.7.0-old/src/DNotify.c++ fam-2.7.0/src/DNotify.c++
--- fam-2.7.0-old/src/DNotify.c++	2009-06-14 21:14:34.000000000 +1000
+++ fam-2.7.0/src/DNotify.c++	2009-06-14 21:23:23.000000000 +1000
@@ -31,6 +31,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <libgen.h>
+#include <cstdlib>
 
 #include "DNotify.h"
 
diff -Naur fam-2.7.0-old/src/IMon.c++ fam-2.7.0/src/IMon.c++
--- fam-2.7.0-old/src/IMon.c++	2003-01-19 00:18:12.000000000 +1000
+++ fam-2.7.0/src/IMon.c++	2009-06-14 21:24:11.000000000 +1000
@@ -40,7 +40,7 @@
 #include "Interest.h"
 #include "Log.h"
 #include "Scheduler.h"
-#include "alloc.h"
+#include <memory>
 
 int		   IMon::imonfd = -2;
 IMon::EventHandler IMon::ehandler = NULL;
diff -Naur fam-2.7.0-old/src/Interest.h fam-2.7.0/src/Interest.h
--- fam-2.7.0-old/src/Interest.h	2009-06-14 21:14:34.000000000 +1000
+++ fam-2.7.0/src/Interest.h	2009-06-14 21:25:25.000000000 +1000
@@ -29,6 +29,7 @@
 #include <netinet/in.h>  //  for in_addr
 
 #include "Boolean.h"
+#include <cstdlib>
 
 class Event;
 class FileSystem;
diff -Naur fam-2.7.0-old/src/NFSFileSystem.c++ fam-2.7.0/src/NFSFileSystem.c++
--- fam-2.7.0-old/src/NFSFileSystem.c++	2003-01-19 00:18:12.000000000 +1000
+++ fam-2.7.0/src/NFSFileSystem.c++	2009-06-14 21:26:16.000000000 +1000
@@ -97,7 +97,7 @@
 
     attr_cache_timeout = ACREGMAX;
 
-    char * p;
+    const char * p;
         
     if (strstr(opt, "noac")) {
         f_noac = true;
