--- src/traywindow.cpp.orig	2006-07-23 13:26:21.236380500 +0100
+++ src/traywindow.cpp	2006-07-23 13:26:38.625467250 +0100
@@ -107,7 +107,7 @@
    delete mDisconnectMenuItems[ manager->id() ];
 
    /* I think (int) manager is unique and static enough to be used as id for the menu */
-   mContextMenu->removeItem( ( int ) manager );
+   mContextMenu->removeItem( ( long ) manager );
    mManagerMenus.remove( manager->id() );
    mOvpnManagers.remove( manager );
 }
@@ -125,7 +125,7 @@
 
    mManagerMenus[ manager->id() ] = new KPopupMenu( mContextMenu );
    /* I think (int) manager is unique and static enough to be used as id for the menu */
-   mContextMenu->insertItem( manager->id() , mManagerMenus[ manager->id() ] , ( int ) manager, 5 );
+   mContextMenu->insertItem( manager->id() , mManagerMenus[ manager->id() ] , ( long ) manager, 5 );
 
    mManagerMenus[ manager->id() ] ->insertTitle( i18n( "Status: Unavailable" ), 0 );
 
