diff -wbBur pdmenu.org/src/inputbox.c pdmenu/src/inputbox.c
--- pdmenu.org/src/inputbox.c	2006-01-25 07:58:35.000000000 +0300
+++ pdmenu/src/inputbox.c	2007-06-25 18:59:12.000000000 +0400
@@ -110,7 +112,7 @@
       ret->ignore=1;
       return ret;
     default:
-      if ((key>=32) && (key<127) && (curspos<SLtt_Screen_Cols -6)) {
+      if ((key>=32) && (curspos<SLtt_Screen_Cols-6)) {
 	ret->value[curspos++]=key;
 	ret->value[curspos]='\0';
       }
diff -wbBur pdmenu.org/src/pdmenu.c pdmenu/src/pdmenu.c
--- pdmenu.org/src/pdmenu.c	2006-01-25 07:58:35.000000000 +0300
+++ pdmenu/src/pdmenu.c	2007-06-25 18:16:46.000000000 +0400
@@ -187,7 +187,9 @@
   Menu_Type *m;
   int ret;
 
-  setlocale(LC_ALL, "");
+//  setlocale(LC_ALL, "");
+  SLutf8_enable(1);
+
   bindtextdomain(PACKAGE,LOCALEDIR);
   textdomain(PACKAGE);
   
