--- src/search.c.orig	2008-11-07 00:38:18.000000000 +0100
+++ src/search.c	2008-11-07 00:50:51.000000000 +0100
@@ -83,6 +83,8 @@
 	dlclose(pcFilename);
 	free(pcFilename);
       }
+    } else {
+      free(pcFilename);
     }
   }
 }
@@ -99,13 +101,10 @@
 
   pcLADSPAPath = getenv("LADSPA_PATH");
   if (!pcLADSPAPath) {
-    fprintf(stderr,
-	    "Warning: You do not have a LADSPA_PATH "
-	    "environment variable set.\n");
-    return;
+    pcStart = "/usr/lib/ladspa/";
+  } else {
+    pcStart = pcLADSPAPath;
   }
-  
-  pcStart = pcLADSPAPath;
   while (*pcStart != '\0') {
     pcEnd = pcStart;
     while (*pcEnd != ':' && *pcEnd != '\0')
