diff -u libdownload-1.1/Makefile libdownload-1.1-new/Makefile
--- libdownload-1.1/Makefile	2006-11-17 10:19:53.000000000 -0500
+++ libdownload-1.1-new/Makefile	2007-08-17 19:53:21.000000000 -0400
@@ -3,7 +3,7 @@
 DEBUG = true
 ENABLE_HTTPS = true
 
-CFLAGS   = -O2 -pipe -I. -DINET6 -D_GNU_SOURCE
+CFLAGS   = -O2 -pipe -I. -DINET6 -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64
 WARNINGS = -Wall -Wstrict-prototypes -Wsign-compare -Wchar-subscripts \
            -Wpointer-arith -Wcast-align -Wsign-compare
 CFLAGS  += $(WARNINGS)
