diff -up ksirk-15.12.1/ksirk/iris/src/CMakeLists.txt.iris_base64 ksirk-15.12.1/ksirk/iris/src/CMakeLists.txt
--- ksirk-15.12.1/ksirk/iris/src/CMakeLists.txt.iris_base64	2016-01-28 14:00:49.804584004 -0600
+++ ksirk-15.12.1/ksirk/iris/src/CMakeLists.txt	2016-01-28 14:02:19.778300698 -0600
@@ -29,7 +29,6 @@ set(libiris_SRCS
 	libidn/rfc3454.c
 	libidn/stringprep.c
   xmpp/base/randomnumbergenerator.cpp
-  xmpp/base64/base64.cpp
   xmpp/cutestuff/bsocket.cpp
 	xmpp/cutestuff/bytestream.cpp
 	xmpp/cutestuff/httpconnect.cpp
@@ -104,7 +103,6 @@ irisnet/corelib/objectsession.h
 irisnet/noncore/processquit.h
 jdns/qjdns.h
 xmpp/base/randrandomnumbergenerator.h
-xmpp/base64/base64.h
 xmpp/cutestuff/bsocket.h
 xmpp/cutestuff/bytestream.h
 xmpp/cutestuff/httpconnect.h
diff -up ksirk-15.12.1/ksirk/iris/src/xmpp/modules.pri.iris_base64 ksirk-15.12.1/ksirk/iris/src/xmpp/modules.pri
--- ksirk-15.12.1/ksirk/iris/src/xmpp/modules.pri.iris_base64	2015-12-15 13:27:19.000000000 -0600
+++ ksirk-15.12.1/ksirk/iris/src/xmpp/modules.pri	2016-01-28 14:00:49.808584036 -0600
@@ -3,4 +3,4 @@ IRIS_XMPP_JID_MODULE = $$PWD/jid/jid.pri
 IRIS_XMPP_SASL_MODULE = $$PWD/sasl/sasl.pri
 IRIS_XMPP_BASE_MODULE = $$PWD/base/base.pri
 IRIS_XMPP_ZLIB_MODULE = $$PWD/zlib/zlib.pri
-IRIS_XMPP_BASE64_MODULE = $$PWD/base64/base64.pri
+#IRIS_XMPP_BASE64_MODULE = $$PWD/base64/base64.pri
diff -up ksirk-15.12.1/ksirk/iris/src/xmpp/sasl/digestmd5response.cpp.iris_base64 ksirk-15.12.1/ksirk/iris/src/xmpp/sasl/digestmd5response.cpp
--- ksirk-15.12.1/ksirk/iris/src/xmpp/sasl/digestmd5response.cpp.iris_base64	2015-12-15 13:27:19.000000000 -0600
+++ ksirk-15.12.1/ksirk/iris/src/xmpp/sasl/digestmd5response.cpp	2016-01-28 14:00:49.808584036 -0600
@@ -11,7 +11,7 @@
 
 #include "xmpp/sasl/digestmd5proplist.h"
 #include "xmpp/base/randomnumbergenerator.h"
-#include "xmpp/base64/base64.h"
+//#include "xmpp/base64/base64.h"
 
 namespace XMPP {
 
@@ -33,7 +33,7 @@ DIGESTMD5Response::DIGESTMD5Response(con
 	for(int n = 0; n < (int)a.size(); ++n) {
 		a[n] = (char) rand.generateNumberBetween(0, 255); 
 	}
-	QByteArray cnonce = Base64::encode(a).toLatin1();
+	QByteArray cnonce = a.toBase64();
 
 	// make other variables
 	if (realm.isEmpty()) {
