diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2013-01-23 14:49:42.000000000 -0700
+++ b/CMakeLists.txt	2013-02-09 11:14:13.362993722 -0700
@@ -101,6 +101,8 @@
     ${KDEPIMLIBS_INCLUDE_DIRS}
 )
 
+# Pull in ARM configuration options if needed
+STRING(REGEX MATCH "PyQt_qreal_double" SIP_ARM_HACK ${PYQT4_SIP_FLAGS} "")
 set(SIP_INCLUDES ${PYQT4_SIP_DIR} sip)
 set(SIP_CONCAT_PARTS 8)
 if (WIN32)
@@ -108,7 +110,7 @@
 else ()
     set(SIP_TAGS ALL WS_X11 ${PYQT4_VERSION_TAG})
 endif ()
-set(SIP_DISABLE_FEATURES VendorID PyQt_NoPrintRangeBug)
+set(SIP_DISABLE_FEATURES VendorID PyQt_NoPrintRangeBug ${SIP_ARM_HACK})
 
 # Use an extra option when compiling on Python 3.
 if (PYTHON_VERSION_MAJOR GREATER 2)
