--- pykde4-4.11.1.orig/CMakeLists.txt	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/CMakeLists.txt	2013-09-08 17:29:29.755854280 -0400
@@ -124,20 +124,20 @@
 if (PYTHON_VERSION_MAJOR GREATER 2)
     if(PYQT4_VERSION STRGREATER "040904")
 	    # Disable for newer PyQt
-	    set(SIP_EXTRA_OPTIONS -g -x PyKDE_QVector)
+	    set(SIP_EXTRA_OPTIONS -P -g -x PyKDE_QVector)
     else ()
 	    set(SIP_EXTRA_OPTIONS -g)
     endif()
 else (PYTHON_VERSION_MAJOR GREATER 2)
     if(PYQT4_VERSION STRGREATER "040904")
 	# Disable for newer PyQt
-    	set(SIP_EXTRA_OPTIONS -g -x PyKDE_QVector -x Py_v3)
+    	set(SIP_EXTRA_OPTIONS -P -g -x PyKDE_QVector -x Py_v3)
     else ()
 	    set(SIP_EXTRA_OPTIONS -g -x Py_v3)
     endif()
 endif ()
 
-add_definitions(-D_REENTRANT -DQT_CORE_LIB -DQT_GUI_LIB -DUSING_SOPRANO_NRLMODEL_UNSTABLE_API)
+add_definitions(-D_REENTRANT -DQT_CORE_LIB -DQT_GUI_LIB -DUSING_SOPRANO_NRLMODEL_UNSTABLE_API -DSIP_PROTECTED_IS_PUBLIC -Dprotected=public)
 
 # Do not use --no-undefined for python modules.
 STRING(REPLACE -Wl,--no-undefined "" CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS}")
--- pykde4-4.11.1.orig/sip/akonadi/entitytreemodel.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/akonadi/entitytreemodel.sip	2013-09-08 17:29:29.755854280 -0400
@@ -122,5 +122,12 @@
     };
     bool                    includeUnsubscribed () const;
     void                    setIncludeUnsubscribed (bool show);
+//force
+private:
+    virtual bool removeColumns(int, int, const QModelIndex&);
+    virtual bool removeRows(int, int, const QModelIndex&);
+    virtual bool insertColumns(int, int, const QModelIndex&);
+    virtual bool insertRows(int, int, const QModelIndex&);
+//end
 };
 };
--- pykde4-4.11.1.orig/sip/akonadi/favoritecollectionsmodel.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/akonadi/favoritecollectionsmodel.sip	2013-09-08 17:29:29.759854280 -0400
@@ -42,5 +42,9 @@
     virtual bool            dropMimeData (const QMimeData* data, Qt::DropAction action, int row, int column, const QModelIndex& parent);
     virtual QStringList     mimeTypes () const;
     virtual Qt::ItemFlags   flags (const QModelIndex& index) const;
+//force
+private:
+    virtual void setSourceModel(QAbstractItemModel *);
+//end
 };
 };
--- pykde4-4.11.1.orig/sip/kdeui/highlighter.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/kdeui/highlighter.sip	2013-09-08 17:29:29.759854280 -0400
@@ -70,6 +70,11 @@
 public:
     ~Highlighter ();
     bool                    checkerEnabledByDefault () const;
+//force
+private:
+        virtual void connectNotify(const char* signal);
+        virtual void disconnectNotify(const char* signal);
+//end
 };
 // Highlighter
 
--- pykde4-4.11.1.orig/sip/kdeui/kdatetable.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/kdeui/kdatetable.sip	2013-09-08 17:29:29.759854280 -0400
@@ -61,8 +61,10 @@
 
 public:
                             KDateValidator (QWidget* parent /TransferThis/ = 0);
-    virtual QValidator::State  validate (QString& text, int& e /In/) const;
-    virtual void            fixup (QString& input) const;
+    virtual QValidator::State validate(QString & /In,Out/, int & /In,Out/) const = 0 /API=QString:2 - /;
+    virtual QValidator::State validate(QString & /Constrained/, int & /In,Out/) const = 0 /API=QString: - 2/;
+    virtual void fixup(QString &input /In,Out/) const /API=QString:2 - /;
+    virtual void fixup(QString & /Constrained/) const /API=QString: - 2/;
     QValidator::State       date (const QString& text, QDate& date) const;
 };
 // KDateValidator
--- pykde4-4.11.1.orig/sip/kdeui/knumvalidator.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/kdeui/knumvalidator.sip	2013-09-08 17:29:29.759854280 -0400
@@ -30,8 +30,10 @@
 public:
     explicit                KIntValidator (QWidget* parent /TransferThis/, int base = 10);
                             KIntValidator (int bottom, int top, QWidget* parent /TransferThis/, int base = 10);
-    virtual QValidator::State  validate (QString&, int& /In/) const;
-    virtual void            fixup (QString&) const;
+    virtual QValidator::State validate(QString & /In,Out/, int & /In,Out/) const = 0 /API=QString:2 - /;
+    virtual QValidator::State validate(QString & /Constrained/, int & /In,Out/) const = 0 /API=QString: - 2/;
+    virtual void fixup(QString &input /In,Out/) const /API=QString:2 - /;
+    virtual void fixup(QString & /Constrained/) const /API=QString: - 2/;
     virtual void            setRange (int bottom, int top);
     virtual void            setBase (int base);
     virtual int             bottom () const;
@@ -53,8 +55,10 @@
     explicit                KFloatValidator (QWidget* parent /TransferThis/);
                             KFloatValidator (double bottom, double top, QWidget* parent /TransferThis/);
                             KFloatValidator (double bottom, double top, bool localeAware, QWidget* parent /TransferThis/);
-    virtual QValidator::State  validate (QString&, int& /In/) const;
-    virtual void            fixup (QString&) const;
+    virtual QValidator::State validate(QString & /In,Out/, int & /In,Out/) const = 0 /API=QString:2 - /;
+    virtual QValidator::State validate(QString & /Constrained/, int & /In,Out/) const = 0 /API=QString: - 2/;
+    virtual void fixup(QString &input /In,Out/) const /API=QString:2 - /;
+    virtual void fixup(QString & /Constrained/) const /API=QString: - 2/;
     virtual void            setRange (double bottom, double top);
     virtual double          bottom () const;
     virtual double          top () const;
@@ -75,7 +79,8 @@
 public:
     explicit                KDoubleValidator (QObject* parent /TransferThis/);
                             KDoubleValidator (double bottom, double top, int decimals, QObject* parent /TransferThis/);
-    virtual QValidator::State  validate (QString& input, int& pos /In/) const;
+    virtual QValidator::State validate(QString & /In,Out/, int & /In,Out/) const = 0 /API=QString:2 - /;
+    virtual QValidator::State validate(QString & /Constrained/, int & /In,Out/) const = 0 /API=QString: - 2/;
     bool                    acceptLocalizedNumbers () const;
     void                    setAcceptLocalizedNumbers (bool accept);
     ~KDoubleValidator ();
--- pykde4-4.11.1.orig/sip/kdeui/kpixmapregionselectorwidget.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/kdeui/kpixmapregionselectorwidget.sip	2013-09-08 17:29:29.759854280 -0400
@@ -57,6 +57,7 @@
 private:
 //force
                             KPixmapRegionSelectorWidget (const KPixmapRegionSelectorWidget&);
+    virtual bool eventFilter(QObject*, QEvent*);
 //end
 public:
     ~KPixmapRegionSelectorWidget ();
--- pykde4-4.11.1.orig/sip/kdeui/kpixmapsequenceoverlaypainter.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/kdeui/kpixmapsequenceoverlaypainter.sip	2013-09-08 17:29:29.763854280 -0400
@@ -39,4 +39,8 @@
     void                    setOffset (const QPoint& offset);
     void                    start ();
     void                    stop ();
+//force
+private:
+    virtual bool eventFilter(QObject*, QEvent*);
+//end
 };
--- pykde4-4.11.1.orig/sip/kdeui/krecursivefilterproxymodel.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/kdeui/krecursivefilterproxymodel.sip	2013-09-08 17:29:29.763854280 -0400
@@ -31,4 +31,8 @@
     virtual bool            acceptRow (int sourceRow, const QModelIndex& sourceParent) const;
 public:
     virtual QModelIndexList  match (const QModelIndex& start, int role, const QVariant& value, int hits = 1, Qt::MatchFlags flags = Qt::MatchFlags(Qt::MatchStartsWith|Qt::MatchWrap)) const;
+//force
+private:
+    virtual bool filterAcceptsRow(int, const QModelIndex&) const;
+//end
 };
--- pykde4-4.11.1.orig/sip/kdeui/kstringvalidator.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/kdeui/kstringvalidator.sip	2013-09-08 17:29:29.763854280 -0400
@@ -35,8 +35,10 @@
     bool                    isFixupEnabled () const;
     void                    setStringList (const QStringList& list);
     QStringList             stringList () const;
-    virtual QValidator::State  validate (QString& input, int& pos /Out/) const;
-    virtual void            fixup (QString& input) const;
+    virtual QValidator::State validate(QString & /In,Out/, int & /In,Out/) const = 0 /API=QString:2 - /;
+    virtual QValidator::State validate(QString & /Constrained/, int & /In,Out/) const = 0 /API=QString: - 2/;
+    virtual void fixup(QString &input /In,Out/) const /API=QString:2 - /;
+    virtual void fixup(QString & /Constrained/) const /API=QString: - 2/;
     ~KStringListValidator ();
 };
 // KStringListValidator
@@ -51,8 +53,10 @@
 
 public:
     explicit                KMimeTypeValidator (QObject* parent /TransferThis/ = 0);
-    virtual QValidator::State  validate (QString& input, int& pos /Out/) const;
-    virtual void            fixup (QString& input) const;
+    virtual QValidator::State validate(QString & /In,Out/, int & /In,Out/) const = 0 /API=QString:2 - /;
+    virtual QValidator::State validate(QString & /Constrained/, int & /In,Out/) const = 0 /API=QString: - 2/;
+    virtual void fixup(QString &input /In,Out/) const /API=QString:2 - /;
+    virtual void fixup(QString & /Constrained/) const /API=QString: - 2/;
     ~KMimeTypeValidator ();
 };
 // KMimeTypeValidator
--- pykde4-4.11.1.orig/sip/kdeui/ktogglefullscreenaction.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/kdeui/ktogglefullscreenaction.sip	2013-09-08 17:29:29.763854280 -0400
@@ -37,7 +37,10 @@
     bool                    eventFilter (QObject* object, QEvent* event);
 public:
     ~KToggleFullScreenAction ();
+
+//force
+private:
+    virtual void slotToggled(bool);
+//end
 };
 // KToggleFullScreenAction
-
-
--- pykde4-4.11.1.orig/sip/kdeui/ktoggletoolbaraction.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/kdeui/ktoggletoolbaraction.sip	2013-09-08 17:29:29.763854280 -0400
@@ -33,6 +33,10 @@
     KToolBar*               toolBar ();
     virtual bool            eventFilter (QObject* watched, QEvent* event);
     ~KToggleToolBarAction ();
+//force
+private:
+    virtual void slotToggled( bool checked );
+//end
 };
 // KToggleToolBarAction
 
--- pykde4-4.11.1.orig/sip/kio/kdirmodel.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/kio/kdirmodel.sip	2013-09-08 17:29:29.767854280 -0400
@@ -105,6 +105,13 @@
     ~KDirModel ();
     void                    setJobTransfersVisible (bool value);
     bool                    jobTransfersVisible () const;
+//force
+private:
+    virtual bool removeColumns(int, int, const QModelIndex&);
+    virtual bool removeRows(int, int, const QModelIndex&);
+    virtual bool insertColumns(int, int, const QModelIndex&);
+    virtual bool insertRows(int, int, const QModelIndex&);
+//end
 };
 // KDirModel
 
--- pykde4-4.11.1.orig/sip/kio/kshellcompletion.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/kio/kshellcompletion.sip	2013-09-08 17:29:29.767854280 -0400
@@ -39,5 +39,3 @@
     ~KShellCompletion ();
 };
 // KShellCompletion
-
-
--- pykde4-4.11.1.orig/sip/kio/metainfojob.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/kio/metainfojob.sip	2013-09-08 17:29:29.767854280 -0400
@@ -51,7 +51,7 @@
 private:
 //force
                             MetaInfoJob ();
-
+    virtual void start();
 //end
 public:
     ~MetaInfoJob ();
--- pykde4-4.11.1.orig/sip/plasma/applet.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/plasma/applet.sip	2013-09-08 17:29:29.767854280 -0400
@@ -217,6 +217,10 @@
     void                    showConfigurationInterface (QWidget* widget);
     bool                    isUserConfiguring () const;
     explicit                Applet (const KPluginInfo& info, QGraphicsItem* parent = 0, uint appletId = 0);
+//force
+private:
+    virtual void paint(QPainter*, const QStyleOptionGraphicsItem*, QWidget*);
+//end
 };
 // Applet
 
--- pykde4-4.11.1.orig/sip/plasma/glapplet.sip	2013-09-08 17:29:29.775854281 -0400
+++ pykde4-4.11.1/sip/plasma/glapplet.sip	2013-09-08 17:29:29.771854281 -0400
@@ -52,6 +52,10 @@
     ~GLApplet ();
     GLuint                  bindTexture (const QImage& image, GLenum target = GL_TEXTURE_2D);
     void                    deleteTexture (GLuint texture_id);
+
+private:
+    virtual void paint(QPainter*, const QStyleOptionGraphicsItem*, QWidget*);
+    virtual void paintInterface(QPainter*, const QStyleOptionGraphicsItem*, const QRect&);
 //end
 };
 // GLApplet
