diff -Naur pykde4-4.4.3.orig/python/pykde4/CMakeLists.txt pykde4-4.4.3/python/pykde4/CMakeLists.txt
--- pykde4-4.4.3.orig/python/pykde4/CMakeLists.txt	2010-03-29 14:57:40.000000000 +0200
+++ pykde4-4.4.3/python/pykde4/CMakeLists.txt	2010-05-31 15:49:17.095615617 +0200
@@ -75,6 +75,9 @@
     ${KDEPIMLIBS_INCLUDE_DIRS}
 )
 
+# Pull in ARM configuration options if needed
+STRING(REGEX MATCH "PyQt_qreal_double" SIP_ARM_HACK ${PYQT4_SIP_FLAGS} "")
+
 SET(SIP_INCLUDES ${PYQT4_SIP_DIR} sip)
 SET(SIP_CONCAT_PARTS 8)
 IF (WIN32)
@@ -82,7 +85,7 @@
 ELSE (WIN32)
     SET(SIP_TAGS ALL WS_X11 ${PYQT4_VERSION_TAG})
 ENDIF (WIN32)
-SET(SIP_DISABLE_FEATURES VendorID PyQt_NoPrintRangeBug)
+SET(SIP_DISABLE_FEATURES VendorID PyQt_NoPrintRangeBug ${SIP_ARM_HACK})
 
 # Use an extra option when compiling on Python 3.
 IF (PYTHON_3)
