diff -up pykde4-4.12.3/sip/phonon/backendcapabilities.sip.sip4_15_5 pykde4-4.12.3/sip/phonon/backendcapabilities.sip
--- pykde4-4.12.3/sip/phonon/backendcapabilities.sip.sip4_15_5  2014-01-18 13:18:35.000000000 -0600
+++ pykde4-4.12.3/sip/phonon/backendcapabilities.sip    2014-03-16 16:56:36.865132943 -0500
@@ -26,7 +26,7 @@ namespace Phonon
 {
 namespace BackendCapabilities
 {
-class Notifier : QObject /NoDefaultCtors,PyQt4NoQMetaObject/
+class Notifier : QObject /NoDefaultCtors,PyQtNoQMetaObject/
 {
 
 signals:
