diff -aur jeuclid-3.1.4.orig/bin/mml2xxx jeuclid-3.1.4.new/bin/mml2xxx
--- jeuclid-3.1.4.orig/bin/mml2xxx	2009-02-21 12:56:50.000000000 +0100
+++ jeuclid-3.1.4.new/bin/mml2xxx	2009-02-21 13:27:09.000000000 +0100
@@ -18,9 +18,9 @@
 #   Copyright (c) 2001-2002 The Apache Software Foundation.  All rights
 #   reserved.
 
-BASEDIR=`dirname $0`/..
-BASEDIR=`(cd "$BASEDIR"; pwd)`
-
+#BASEDIR=`dirname $0`/..
+#BASEDIR=`(cd "$BASEDIR"; pwd)`
+BASEDIR=/usr/share/java/jeuclid
 
 
 # OS specific support.  $var _must_ be set to either true or false.
@@ -96,5 +96,5 @@
   -Dapp.pid="$$" \
   -Dapp.repo="$REPO" \
   -Dbasedir="$BASEDIR" \
-  net.sourceforge.jeuclid.app.Mml2xxx \
+    Mml2xxx \
   "$@"
