Index: ntp.c
===================================================================
RCS file: /storage/1/mirror/openbsd/src/usr.sbin/ntpd/ntp.c,v
retrieving revision 1.91
diff -u -r1.91 ntp.c
--- ntp.c       1 Jul 2006 18:52:46 -0000       1.91
+++ ntp.c       15 Jul 2006 07:50:05 -0000
@@ -1,4 +1,4 @@
-/*     $OpenBSD: ntp.c,v 1.91 2006-07-01 18:52:46 otto Exp $ */
+/*     $OpenBSD: ntp.c,v 1.91 2006/07/01 18:52:46 otto Exp $ */

 /*
  * Copyright (c) 2003, 2004 Henning Brauer <[EMAIL PROTECTED]>
@@ -315,8 +315,10 @@
                for (s = TAILQ_FIRST(&conf->ntp_sensors); s != NULL;
                    s = next_s) {
                        next_s = TAILQ_NEXT(s, entry);
-                       if (s->next <= time(NULL))
+                       if (s->next <= time(NULL)) {
                                sensor_query(s);
+                               priv_adjtime();
+                       }
                }
        }
