--- desktop/source/migration/wizard.cxx
+++ desktop/source/migration/wizard.cxx
@@ -121,6 +121,7 @@
     ,m_bAutomaticUpdChk(sal_True)
     ,m_aLicensePath( rLicensePath )
 {
+	return;
     // ---
     // FreeResource();
 //	enableState(STATE_USER, sal_False);
@@ -383,7 +384,7 @@
 
 short FirstStartWizard::Execute()
 {
-    return svt::RoadmapWizard::Execute();
+    return sal_True || svt::RoadmapWizard::Execute();
 }
 
 static OUString _makeDateTimeString (const DateTime& aDateTime, sal_Bool bUTC = sal_False)
