index c5249f0..69946de 100644
--- a/ld/testsuite/ld-plugin/lto.exp
+++ b/ld/testsuite/ld-plugin/lto.exp
@@ -57,7 +57,7 @@ set lto_link_tests {
    "" "-O2"
    {lto-3c.c} {} "libdummy.a"}
   {"Build liblto-3.a"
-   "" "-flto"
+   "" "-flto -ffat-lto-objects"
    {lto-3b.c} {} "liblto-3.a"}
   {"Build libdummy.a 5a"
    "" "-flto"
@@ -111,7 +111,7 @@ set lto_link_tests {
    "" ""
    {pr12758a.s} {} "libdummy.a"}
   {"Build libpr12758.a"
-   "" "-flto -O2"
+   "" "-flto -O2 -ffat-lto-objects"
    {pr12758b.c} {} "libpr12758.a"}
   {"PR ld/12758"
    "-O2 -Wl,-e,foo -nostdlib -flto -fuse-linker-plugin tmpdir/pr12758a.o -Wl,--start-group tmpdir/libpr12758.a -Wl,--end-group" ""
@@ -120,13 +120,13 @@ set lto_link_tests {
    "" ""
    {pr12760a.c} {} "libdummy.a"}
   {"Build libpr12760.a"
-   "" "-flto -O2"
+   "" "-flto -O2 -ffat-lto-objects"
    {pr12760b.c} {} "libpr12760.a"}
   {"PR ld/12760"
    "-O2 -Wl,-e,foo -nostdlib -flto -fuse-linker-plugin tmpdir/pr12760a.o -Wl,--start-group tmpdir/libpr12760.a -Wl,--end-group" ""
    {dummy.c} {} "pr12760.exe" "c" "warning: Bad bar"}
   {"Build libpr13183.a"
-   "-T" "-flto -O2"
+   "-T" "-flto -O2 -ffat-lto-objects"
    {pr13183a.c} {} "libpr13183.a"}
   {"Build libdummy.a PR ld/13183"
    "" "-flto -O2"
diff --git a/ld/testsuite/lib/ld-lib.exp b/ld/testsuite/lib/ld-lib.exp
index 8df42ae..175449a 100644
--- a/ld/testsuite/lib/ld-lib.exp
+++ b/ld/testsuite/lib/ld-lib.exp
@@ -1643,7 +1643,7 @@ proc check_lto_available { } {
 	set f [open $src "w"]
 	puts $f ""
 	close $f
-	set status [remote_exec host $CC "-shared -B[pwd]/tmpdir/ld/ -flto -fuse-linker-plugin $src -o $output"]
+	set status [remote_exec host $CC "-shared -B[pwd]/tmpdir/ld/ -flto -ffat-lto-objects -fuse-linker-plugin $src -o $output"]
 	if { [lindex $status 0] == 0 } {
 	    set lto_available_saved 1
 	} else {
-- 
1.7.1

