--- Makefile	2004-10-09 01:41:12.000000000 +0300
+++ Makefile	2005-03-04 05:51:21.000000000 +0200
@@ -6,15 +6,15 @@
 # Settings
 
 ENABLE_AUDIO=true
-ENABLE_OPENGL=false
+ENABLE_OPENGL=true
 ENABLE_DGA=false
 
 DEBUG_MODE=false
 
 # Unix/Linux settings
-PREFIX=/usr/local
+PREFIX=/usr
 DATADIR=$(PREFIX)/share/games/flobopuyo
-INSTALL_BINDIR=$(DESTDIR)/$(PREFIX)/games
+INSTALL_BINDIR=$(DESTDIR)/$(PREFIX)/bin
 INSTALL_DATADIR=$(DESTDIR)/$(DATADIR)
 
 # Mac settings
