diff -urN a/src/include/thread.h b/src/include/thread.h
--- a/src/include/thread.h	2013-07-08 13:47:25.000000000 -0600
+++ b/src/include/thread.h	2013-07-13 14:30:25.709073357 -0600
@@ -323,8 +323,12 @@
 #if defined __arm__ || defined __s390__
         __asm__ __volatile__("NOP;");
 #else
+#if defined __arm__
+        __asm__ __volatile__("NOP;");
+#else
         __asm__ __volatile__("pause;");
 #endif
+#endif
     }
 #elif USE_TBB
     __TBB_Pause(delay);
@@ -447,7 +451,12 @@
 
     // Disallow copy construction by making private and unimplemented.
     atomic (atomic const &);
+
+#if defined __arm__
+} __attribute__((aligned(8)));
+#else
 };
+#endif
 
 
 #endif /* ! USE_TBB_ATOMIC */
