diff --git a/src/libOpenImageIO/CMakeLists.txt b/src/libOpenImageIO/CMakeLists.txt
index cedb2f5b..98b8d1fc 100644
--- a/src/libOpenImageIO/CMakeLists.txt
+++ b/src/libOpenImageIO/CMakeLists.txt
@@ -100,7 +100,7 @@ endif ()
 target_link_libraries (OpenImageIO
                        ${VISIBILITY_COMMAND} ${VISIBILITY_MAP_COMMAND}
                        ${format_plugin_libs} # Add all the target link libraries from the plugins
-                       ${Boost_LIBRARIES})
+                       ${Boost_LIBRARIES} atomic)
 
 
 # Include OpenColorIO if using it
diff --git a/src/libutil/CMakeLists.txt b/src/libutil/CMakeLists.txt
index cb00a42c..ef847c20 100644
--- a/src/libutil/CMakeLists.txt
+++ b/src/libutil/CMakeLists.txt
@@ -40,7 +40,7 @@ if (OIIO_BUILD_TESTS)
 
     add_executable (atomic_test atomic_test.cpp)
     set_target_properties (atomic_test PROPERTIES FOLDER "Unit Tests")
-    target_link_libraries (atomic_test OpenImageIO_Util ${Boost_LIBRARIES} ${CMAKE_DL_LIBS})
+    target_link_libraries (atomic_test OpenImageIO_Util ${Boost_LIBRARIES} ${CMAKE_DL_LIBS} atomic)
     add_test (unit_atomic atomic_test)
 
     add_executable (array_view_test array_view_test.cpp)
