diff -urN a/src/libOpenImageIO/CMakeLists.txt b/src/libOpenImageIO/CMakeLists.txt
--- a/src/libOpenImageIO/CMakeLists.txt	2018-03-29 18:30:02.061880022 -0600
+++ b/src/libOpenImageIO/CMakeLists.txt	2018-03-29 18:31:08.701755694 -0600
@@ -109,7 +109,7 @@
                        ${SANITIZE_LIBRARIES}
                        ${format_plugin_libs} # Add all the target link libraries from the plugins
                        ${Boost_LIBRARIES}
-                       ${CMAKE_DL_LIBS})
+                       ${CMAKE_DL_LIBS} atomic)
 
 
 # Include OpenColorIO if using it
diff -urN a/src/libutil/CMakeLists.txt b/src/libutil/CMakeLists.txt
--- a/src/libutil/CMakeLists.txt	2018-03-29 18:30:01.021881953 -0600
+++ b/src/libutil/CMakeLists.txt	2018-03-29 18:30:19.753847129 -0600
@@ -46,7 +46,7 @@
 
     add_executable (atomic_test atomic_test.cpp)
     set_target_properties (atomic_test PROPERTIES FOLDER "Unit Tests")
-    target_link_libraries (atomic_test OpenImageIO_Util ${Boost_LIBRARIES} ${CMAKE_DL_LIBS})
+    target_link_libraries (atomic_test OpenImageIO_Util ${Boost_LIBRARIES} ${CMAKE_DL_LIBS} atomic)
     add_test (unit_atomic atomic_test)
 
     add_executable (array_view_test array_view_test.cpp)
