diff --git a/src/include/OpenImageIO/fmath.h b/src/include/OpenImageIO/fmath.h
index 8db8442..751cd07 100644
--- a/src/include/OpenImageIO/fmath.h
+++ b/src/include/OpenImageIO/fmath.h
@@ -1298,9 +1298,9 @@ OIIO_NAMESPACE_EXIT
 
 #ifndef _MSC_VER
  // Some systems have isnan, isinf and isfinite in the std namespace.
- using std::isnan;
- using std::isinf;
- using std::isfinite;
+ using ::isnan;
+ using ::isinf;
+ using OIIO::isfinite;
 #endif
 
 #if (defined(__FreeBSD__) && (__FreeBSD_version < 803000))

