diff -u -r avidemux_2.7.4/bootStrap.bash avidemux_2.7.4-fix/bootStrap.bash
--- avidemux_2.7.4/bootStrap.bash	2019-08-15 07:25:40.000000000 +0000
+++ avidemux_2.7.4-fix/bootStrap.bash	2020-06-05 13:40:10.633140851 +0000
@@ -66,7 +66,7 @@
         fi
         cd $BUILDDIR 
         cmake $COMPILER $PKG $FAKEROOT $QT_FLAVOR -DCMAKE_EDIT_COMMAND=vim $INSTALL_PREFIX $EXTRA $BUILD_QUIRKS $ASAN $DEBUG -G "$BUILDER" $SOURCEDIR || fail cmakeZ
-        make  $PARAL >& /tmp/log$BUILDDIR || fail "make, result in /tmp/log$BUILDDIR"
+        make  $PARAL || fail make
 	if  [ "x$PKG" != "x" ] ; then
           $FAKEROOT_COMMAND make package DESTDIR=$FAKEROOT_DIR/tmp || fail package
 	fi
