diff -u -r avidemux_2.7.4/avidemux/qt4/ADM_UIs/src/T_RubberControl.cpp avidemux_2.7.4-fix/avidemux/qt4/ADM_UIs/src/T_RubberControl.cpp
--- avidemux_2.7.4/avidemux/qt4/ADM_UIs/src/T_RubberControl.cpp	2019-08-15 07:25:39.000000000 +0000
+++ avidemux_2.7.4-fix/avidemux/qt4/ADM_UIs/src/T_RubberControl.cpp	2020-06-05 13:48:28.014061206 +0000
@@ -8,6 +8,7 @@
 ***************************************************************************///
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QRubberBand>
 #include <QBoxLayout>
 #include <QSizeGrip>
