diff -ru avidemux_2.8.1.orig/avidemux_plugins/ADM_videoEncoder/x265/ADM_x265.cpp avidemux_2.8.1/avidemux_plugins/ADM_videoEncoder/x265/ADM_x265.cpp
--- avidemux_2.8.1.orig/avidemux_plugins/ADM_videoEncoder/x265/ADM_x265.cpp	2024-10-12 10:07:58.005741891 +0200
+++ avidemux_2.8.1/avidemux_plugins/ADM_videoEncoder/x265/ADM_x265.cpp	2024-10-12 11:12:13.519781376 +0200
@@ -205,8 +205,8 @@
     //
       x265_nal          *nal;
       uint32_t          nbNal = 0;
-      x265_picture      pic_out;
-      api->picture_init(&param,&pic_out);
+      x265_picture      *pic_out;
+      api->picture_init(&param,pic_out);
       out->flags = 0;
       
         int er;
@@ -236,7 +236,7 @@
 
 
     // 3-encode
-    if(false==postAmble(out,nbNal,nal,&pic_out))
+    if(false==postAmble(out,nbNal,nal,pic_out))
     {
         ADM_warning("[x265] postAmble failed\n");
         return false;     
