diff -aur CuraEngine-3.4.1/CMakeLists.txt CuraEngine-3.4.1.old/CMakeLists.txt
--- CuraEngine-3.4.1/CMakeLists.txt	2018-07-09 10:55:15.000000000 +0200
+++ CuraEngine-3.4.1.old/CMakeLists.txt	2018-09-07 20:16:42.621995976 +0200
@@ -252,10 +252,6 @@
 endif()
 
 
-add_custom_command(TARGET CuraEngine POST_BUILD
-                   COMMAND ${CMAKE_COMMAND} -E copy_directory
-                       ${CMAKE_SOURCE_DIR}/resources $<TARGET_FILE_DIR:CuraEngine>)
-
 # Installing CuraEngine.
 include(GNUInstallDirs)
 install(TARGETS CuraEngine DESTINATION ${CMAKE_INSTALL_BINDIR})
