--- fritz.pcmcia/src/fcpcmcia_cs.c~	2005-08-12 18:25:17.000000000 +0200
+++ fritz.pcmcia/src/fcpcmcia_cs.c	2005-08-12 18:25:17.000000000 +0200
@@ -233,12 +233,14 @@
     link->next = dev_list;
     dev_list = link;
     client_reg.dev_info = &dev_info;
+#if 0
     client_reg.Attributes = INFO_IO_CLIENT | INFO_CARD_SHARE;
     client_reg.EventMask =
 	CS_EVENT_CARD_INSERTION | CS_EVENT_CARD_REMOVAL |
 	CS_EVENT_RESET_PHYSICAL | CS_EVENT_CARD_RESET |
 	CS_EVENT_PM_SUSPEND | CS_EVENT_PM_RESUME;
     client_reg.event_handler = &cs_event;
+#endif
     client_reg.Version = 0x0210;
     client_reg.event_callback_args.client_data = link;
     ret = pcmcia_register_client(&link->handle, &client_reg);
@@ -545,6 +547,13 @@
     return 0;
 } /* cs_event */
 
+static struct pcmcia_device_id fcpcmcia_ids[] = {
+	PCMCIA_DEVICE_PROD_ID12("AVM", "ISDN A", 0x95d42008, 0xadc9d4bb),
+	PCMCIA_DEVICE_PROD_ID12("ISDN", "CARD", 0x8d9761c8, 0x01c5aa7b),
+	PCMCIA_DEVICE_NULL
+};
+MODULE_DEVICE_TABLE(pcmcia, fcpcmcia_ids);
+
 static struct pcmcia_driver cs_driver = {
 	.owner	= THIS_MODULE,
 	.drv	= {
@@ -552,6 +561,8 @@
 	},
 	.attach	= cs_attach,
 	.detach	= cs_detach,
+	.event	= cs_event,
+	.id_table = fcpcmcia_ids, 
 };
 
 static int __init cs_init(void)
