--- driver.c~	2005-07-07 00:00:00.000000000 +0200
+++ driver.c	2007-01-08 22:33:51.000000000 +0100
@@ -901,10 +901,10 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-int driver_init (void) {
+int avm_init (void) {
 
 	return (NULL != (capi_lib = link_library (NULL)));
-} /* driver_init */
+} /* avm_init */
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
--- main.c~	2007-01-08 22:18:57.000000000 +0100
+++ main.c	2007-01-08 22:34:33.000000000 +0100
@@ -204,7 +204,7 @@
 	NOTE(PRODUCT_LOGO " found: port 0x%04x, irq %u\n", pars.port, pars.irq);
 
 	NOTE("Loading...\n");
-	if (!driver_init ()) {
+	if (!avm_init ()) {
 		ERROR("Error: Driver library not available.\n");
 		ERROR("Not loaded.\n");
 		return res;
@@ -269,7 +269,7 @@
 	NOTE (PRODUCT_LOGO " found: port 0x%04x, irq %u\n", pars.port, pars.irq);
 
 	NOTE("Loading...\n");
-	if (!driver_init ()) {
+	if (!avm_init ()) {
 		ERROR("Error: Driver library not available.\n");
 		ERROR("Not loaded.\n");
 		return -EBUSY;
@@ -377,7 +377,7 @@
 	}
 	return err;
 #elif defined (__fcpcmcia__) || defined (__fcclassic__)
-	if (!driver_init ()) {
+	if (!avm_init ()) {
 		ERROR("Error: Driver library not available.\n");
 		ERROR("Not loaded.\n");
 		return -EBUSY;
--- driver.h~	2007-01-08 22:18:57.000000000 +0100
+++ driver.h	2007-01-08 22:34:12.000000000 +0100
@@ -97,7 +97,7 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-extern int driver_init (void);
+extern int avm_init (void);
 extern void driver_exit (void);
 
 /*---------------------------------------------------------------------------*\
