diff -ruN a/cloudinit/distros/parsers/sys_conf.py b/cloudinit/distros/parsers/sys_conf.py
--- a/cloudinit/distros/parsers/sys_conf.py	2022-02-15 19:02:23.000000000 +0100
+++ b/cloudinit/distros/parsers/sys_conf.py	2022-02-16 00:09:13.366255298 +0100
@@ -107,7 +107,7 @@
         return "%s%s%s%s%s" % (
             indent_string,
             key,
-            self._a_to_u("="),
+            "=",
             val,
             cmnt,
         )
