From fbc0cc200f1cc2b0fd8abc6de1dbd053d45f3321 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 2 Apr 2017 01:10:04 -0600
Subject: [PATCH] fix drm_fourcc.h include path

Signed-off-by: Kevin Mihelich <kevin@archlinuxarm.org>
---
 common/fourcc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/fourcc.h b/common/fourcc.h
index efe2d62..74f5c6d 100644
--- a/common/fourcc.h
+++ b/common/fourcc.h
@@ -1,7 +1,7 @@
 #ifndef COMMON_FOURCC_H
 #define COMMON_FOURCC_H
 
-#include <drm/drm_fourcc.h>
+#include <drm_fourcc.h>
 /* pick up on the Xorg fourcc.h. */
 #include <fourcc.h>
 
-- 
2.12.0

