diff -Nru cinepaint-0.22-1.orig/plug-ins/collect/collect.cpp cinepaint-0.22-1/plug-ins/collect/collect.cpp
--- cinepaint-0.22-1.orig/plug-ins/collect/collect.cpp	2007-03-30 11:11:33.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/collect/collect.cpp	2009-05-21 14:54:45.000000000 +0200
@@ -301,7 +301,7 @@
             return -1;
 
           // renaming the layer to the original filename
-          gimp_layer_set_name (layers[0], strrchr(fc->value(1),'/')+1);
+          gimp_layer_set_name (layers[0], strrchr((char *)fc->value(1),'/')+1);
           GPrecisionType image_base_prec = gimp_drawable_precision (layers[0]);
           int base_gray = gimp_drawable_gray (layers[0]);
 
@@ -341,7 +341,7 @@
                     #endif
                     gimp_image_add_layer (image_ID, layers[0], 0); DBG
                     // set layer name to filename
-                    gimp_layer_set_name(layers[0], strrchr(fc->value(i),'/')+1);
+                    gimp_layer_set_name(layers[0], strrchr((char *)fc->value(i),'/')+1);
 
                     // searching max dimensions
                     if (gimp_image_width(load_image_ID) > max_w)

