--- ./plug-ins/icc_examin/icc_examin/icc_oyranos.h.orig	2008-06-27 19:28:52.000000000 +0200
+++ ./plug-ins/icc_examin/icc_examin/icc_oyranos.h	2008-06-27 19:31:34.000000000 +0200
@@ -142,9 +142,6 @@
 
     // colour transformations
     ColourTransformKey erzeugeTrafo (
-                                  const char* eingangs_profil__geraet,
-                                  int         byte,
-                                  int         kanaele,
                                   const char* ausgangs_profil__geraet,
                                   int         byte,
                                   int         kanaele,
@@ -152,9 +149,6 @@
                                   const char* cmm, // 4 bytes 'lcms' 'APPL'
                                   int         cmm_optionen); // BPC, precission
     ColourTransformKey erzeugeTrafo (
-                                  const char* eingangs_profil__geraet,
-                                  int         byte,
-                                  int         kanaele,
                                   const char* ausgangs_profil__geraet,
                                   int         byte,
                                   int         kanaele,
@@ -163,9 +157,6 @@
                                   int         cmm_optionen,
                                   std::list<const char*> &profile );
     ColourTransformKey erzeugeTrafo (
-                                  const char* eingangs_profil__geraet,
-                                  int         byte,
-                                  int         kanaele,
                                   const char* ausgangs_profil__geraet,
                                   int         byte,
                                   int         kanaele,
@@ -181,9 +172,6 @@
     std::string cmm;
   private:
     ColourTransformKey erzeugeSchluessel_ (
-                                  const char* eingangs_profil__geraet,
-                                  int         byte,
-                                  int         kanaele,
                                   const char* ausgangs_profil__geraet,
                                   int         byte,
                                   int         kanaele,
