Index: cinepaint-0.22-1/plug-ins/bracketing_to_hdr/br_core/ResponseSolver.hpp
===================================================================
--- cinepaint-0.22-1.orig/plug-ins/bracketing_to_hdr/br_core/ResponseSolver.hpp
+++ cinepaint-0.22-1/plug-ins/bracketing_to_hdr/br_core/ResponseSolver.hpp
@@ -34,6 +34,7 @@
 
 
 #include <fstream>
+#include <cstring>
 #include "TNT/tnt_misc.hpp"     // proof_least_square_fit()
 #include "TNT/tnt_stopwatch.hpp"
 #include "TNT/jama_qr.hpp"      // JAMA::QR
Index: cinepaint-0.22-1/plug-ins/bracketing_to_hdr/br_core/br_Image.cpp
===================================================================
--- cinepaint-0.22-1.orig/plug-ins/bracketing_to_hdr/br_core/br_Image.cpp
+++ cinepaint-0.22-1/plug-ins/bracketing_to_hdr/br_core/br_Image.cpp
@@ -26,6 +26,7 @@
 */
 #include <iostream>
 #include <cmath>                    // pow()
+#include <cstring>
 
 #include "br_types.hpp"             // ImageID
 #include "br_enums.hpp"             // DataType, ReportWhat
Index: cinepaint-0.22-1/plug-ins/icc_examin/icc_examin/icc_oyranos_extern.cpp
===================================================================
--- cinepaint-0.22-1.orig/plug-ins/icc_examin/icc_examin/icc_oyranos_extern.cpp
+++ cinepaint-0.22-1/plug-ins/icc_examin/icc_examin/icc_oyranos_extern.cpp
@@ -29,6 +29,7 @@
 
 #include "icc_utils.h"
 #include "icc_fenster.h"
+#include <cstdlib>
 
 using namespace icc_examin_ns;
 
Index: cinepaint-0.22-1/plug-ins/icc_examin/icc_examin/icc_speicher.h
===================================================================
--- cinepaint-0.22-1.orig/plug-ins/icc_examin/icc_examin/icc_speicher.h
+++ cinepaint-0.22-1/plug-ins/icc_examin/icc_examin/icc_speicher.h
@@ -33,6 +33,8 @@
 
 #include "icc_utils.h"
 #include <string>
+#include <cstdlib>
+#include <cstring>
 
 #define DBG_SPEICHER_START   if(icc_debug >= 3) DBG_PROG_START
 #define DBG_SPEICHER_ENDE    if(icc_debug >= 3) DBG_PROG_ENDE
Index: cinepaint-0.22-1/plug-ins/icc_examin/icc_examin/threads.cpp
===================================================================
--- cinepaint-0.22-1.orig/plug-ins/icc_examin/icc_examin/threads.cpp
+++ cinepaint-0.22-1/plug-ins/icc_examin/icc_examin/threads.cpp
@@ -42,6 +42,7 @@
 #include <iostream>
 #  if HAVE_PTHREAD_H
 // Use POSIX threading...
+#include <climits>
 
 int fl_create_thread(Fl_Thread& t, void *(*f) (void *), void* p)
 {
Index: cinepaint-0.22-1/plug-ins/pdf/pdf.cpp
===================================================================
--- cinepaint-0.22-1.orig/plug-ins/pdf/pdf.cpp
+++ cinepaint-0.22-1/plug-ins/pdf/pdf.cpp
@@ -39,6 +39,7 @@
 
 #include "pdf.h"
 #include "pdf_dialog.h"
+#include <cstring>
 
 extern "C" {
 #include <gtk/gtk.h>
