diff -ruN owfs-orig/module/owshell/src/c/Makefile.am owfs/module/owshell/src/c/Makefile.am
--- owfs-orig/module/owshell/src/c/Makefile.am 2015-01-11 03:31:32.000000000 +0200
+++ owfs/module/owshell/src/c/Makefile.am 2015-03-23 18:47:03.211445494 +0200
@@ -41,6 +41,7 @@
	-Wcast-align \
	-Wstrict-prototypes \
	-Wredundant-decls \
+	${LIBUSB_CFLAGS} \
	${EXTRACFLAGS}

	LDADD = ${LD_EXTRALIBS} ${OSLIBS}
diff -ruN owfs-orig/module/owtcl/Makefile.am owfs/module/owtcl/Makefile.am
--- owfs-orig/module/owtcl/Makefile.am 2015-01-11 03:31:32.000000000 +0200
+++ owfs/module/owtcl/Makefile.am 2015-03-23 18:46:52.681151547 +0200
@@ -39,6 +39,7 @@
	-Wcast-align \
	-Wstrict-prototypes \
	-Wredundant-decls \
+	${LIBUSB_CFLAGS} \
	${EXTRACFLAGS} \
	${PTHREAD_CFLAGS} \
	$(TCL_DEFS) \
