diff -Naur a/src/scripts/systemd/owfs.service.in b/src/scripts/systemd/owfs.service.in
--- a/src/scripts/systemd/owfs.service.in	2015-01-11 01:31:33.000000000 +0000
+++ b/src/scripts/systemd/owfs.service.in	2016-05-25 12:19:24.082429561 +0100
@@ -3,7 +3,7 @@
 Documentation=man:owfs(1)
 
 [Service]
-ExecStart=@bindir@/owfs --server=127.0.0.1 --allow_other /run/owfs
+ExecStart=@bindir@/owfs --error_print=2 --server=127.0.0.1 --allow_other /run/owfs
 ExecStop=/usr/bin/umount /run/owfs
 RuntimeDirectory=owfs
 
