--- nxcomp.old/NXproto.h	2006-06-19 19:55:56.000000000 +0200
+++ nxcomp/NXproto.h	2007-02-12 18:17:41.000000000 +0100
@@ -26,6 +26,30 @@
 #include <X11/Xmd.h>
 #include <X11/Xproto.h>
 
+/*
+  Copied from Xproto.h
+ */
+
+/* For the purpose of the structure definitions in this file,
+we must redefine the following types in terms of Xmd.h's types, which may
+include bit fields.  All of these are #undef'd at the end of this file,
+restoring the definitions in X.h.  */
+
+#define Window CARD32
+#define Drawable CARD32
+#define Font CARD32
+#define Pixmap CARD32
+#define Cursor CARD32
+#define Colormap CARD32
+#define GContext CARD32
+#define Atom CARD32
+#define VisualID CARD32
+#define Time CARD32
+#define KeyCode CARD8
+#define KeySym CARD32
+
+/* End copied from Xproto.h */
+
 #define sz_xNXGetControlParametersReq           4
 #define sz_xNXGetCleanupParametersReq           4
 #define sz_xNXGetImageParametersReq             4
@@ -343,9 +367,9 @@
     CARD8    dstDepth;
     CARD32   srcLength B32;
     CARD32   dstLength B32;
-    INT16    srcX B16, srcY B16;
+    CARD16    srcX B16, srcY B16;
     CARD16   srcWidth B16, srcHeight B16;
-    INT16    dstX B16, dstY B16;
+    CARD16    dstX B16, dstY B16;
     CARD16   dstWidth B16, dstHeight B16;
 } xNXPutPackedImageReq;
 
@@ -463,4 +487,20 @@
 }
 #endif
 
+/* Copied from Xproto.h */
+
+/* restore these definitions back to the typedefs in X.h */
+#undef Window
+#undef Drawable
+#undef Font
+#undef Pixmap
+#undef Cursor
+#undef Colormap
+#undef GContext
+#undef Atom
+#undef VisualID
+#undef Time
+#undef KeyCode
+#undef KeySym
+
 #endif /* NXproto_H */
