Index: ogr/ogrsf_frmts/couchdb/ogrcouchdbdatasource.cpp
===================================================================
--- ogr/ogrsf_frmts/couchdb/ogrcouchdbdatasource.cpp	(revision 27044)
+++ ogr/ogrsf_frmts/couchdb/ogrcouchdbdatasource.cpp	(revision 27267)
@@ -1079,5 +1079,5 @@
         CPLError( CE_Failure, CPLE_AppDefined,
                     "JSON parsing error: %s (at offset %d)",
-                    json_tokener_errors[jstok->err], jstok->char_offset);
+                    json_tokener_error_desc(jstok->err), jstok->char_offset);
 
         json_tokener_free(jstok);
Index: ogr/ogrsf_frmts/gme/ogrgmejson.cpp
===================================================================
--- ogr/ogrsf_frmts/gme/ogrgmejson.cpp	(revision 27208)
+++ ogr/ogrsf_frmts/gme/ogrgmejson.cpp	(revision 27267)
@@ -515,5 +515,5 @@
             CPLError( CE_Failure, CPLE_AppDefined,
                       "JSON parsing error: %s (at offset %d)",
-                          json_tokener_errors[jstok->err], jstok->char_offset);
+                          json_tokener_error_desc(jstok->err), jstok->char_offset);
 
             json_tokener_free(jstok);
Index: ogr/ogrsf_frmts/geojson/ogrgeojsonreader.cpp
===================================================================
--- ogr/ogrsf_frmts/geojson/ogrgeojsonreader.cpp	(revision 27044)
+++ ogr/ogrsf_frmts/geojson/ogrgeojsonreader.cpp	(revision 27267)
@@ -78,5 +78,5 @@
             CPLError( CE_Failure, CPLE_AppDefined,
                       "GeoJSON parsing error: %s (at offset %d)",
-            	      json_tokener_errors[jstok->err], jstok->char_offset);
+            	      json_tokener_error_desc(jstok->err), jstok->char_offset);
             
             json_tokener_free(jstok);
@@ -1517,5 +1517,5 @@
             CPLError( CE_Failure, CPLE_AppDefined,
                       "GeoJSON parsing error: %s (at offset %d)",
-                      json_tokener_errors[jstok->err], jstok->char_offset);
+                      json_tokener_error_desc(jstok->err), jstok->char_offset);
             json_tokener_free(jstok);
             return NULL;
Index: ogr/ogrsf_frmts/geojson/ogrtopojsonreader.cpp
===================================================================
--- ogr/ogrsf_frmts/geojson/ogrtopojsonreader.cpp	(revision 27044)
+++ ogr/ogrsf_frmts/geojson/ogrtopojsonreader.cpp	(revision 27267)
@@ -75,5 +75,5 @@
             CPLError( CE_Failure, CPLE_AppDefined,
                       "TopoJSON parsing error: %s (at offset %d)",
-            	      json_tokener_errors[jstok->err], jstok->char_offset);
+            	      json_tokener_error_desc(jstok->err), jstok->char_offset);
             
             json_tokener_free(jstok);
Index: ogr/ogrsf_frmts/geojson/ogresrijsonreader.cpp
===================================================================
--- ogr/ogrsf_frmts/geojson/ogresrijsonreader.cpp	(revision 27050)
+++ ogr/ogrsf_frmts/geojson/ogresrijsonreader.cpp	(revision 27267)
@@ -79,5 +79,5 @@
             CPLError( CE_Failure, CPLE_AppDefined,
                       "ESRIJSON parsing error: %s (at offset %d)",
-            	      json_tokener_errors[jstok->err], jstok->char_offset);
+            	      json_tokener_error_desc(jstok->err), jstok->char_offset);
             
             json_tokener_free(jstok);
Index: ogr/ogrsf_frmts/cartodb/ogrcartodbdatasource.cpp
===================================================================
--- ogr/ogrsf_frmts/cartodb/ogrcartodbdatasource.cpp	(revision 27044)
+++ ogr/ogrsf_frmts/cartodb/ogrcartodbdatasource.cpp	(revision 27267)
@@ -505,5 +505,5 @@
         CPLError( CE_Failure, CPLE_AppDefined,
                     "JSON parsing error: %s (at offset %d)",
-                    json_tokener_errors[jstok->err], jstok->char_offset);
+                    json_tokener_error_desc(jstok->err), jstok->char_offset);
         json_tokener_free(jstok);
         CPLHTTPDestroyResult(psResult);
Index: frmts/mbtiles/mbtilesdataset.cpp
===================================================================
--- frmts/mbtiles/mbtilesdataset.cpp	(revision 27215)
+++ frmts/mbtiles/mbtilesdataset.cpp	(revision 27267)
@@ -619,5 +619,5 @@
         CPLError( CE_Failure, CPLE_AppDefined,
                     "JSON parsing error: %s (at offset %d)",
-                    json_tokener_errors[jstok->err],
+                    json_tokener_error_desc(jstok->err),
                     jstok->char_offset);
         json_tokener_free(jstok);
