--- swig/python/GNUmakefile.orig	2008-02-02 17:00:52.000000000 -0800
+++ swig/python/GNUmakefile	2008-02-02 17:02:03.000000000 -0800
@@ -51,9 +51,9 @@
 install:
 
 ifeq ($(PY_HAVE_SETUPTOOLS),1)
-	$(PYTHON) setup.py install 
+	$(PYTHON) setup.py install --root=$(DESTDIR) 
 else
-	$(PYTHON) setup.py install --prefix=$(DESTDIR)$(prefix)
+	$(PYTHON) setup.py install --root=$(DESTDIR) --prefix=$(prefix)
 endif
 
 	for f in $(SCRIPTS) ; do $(INSTALL) ./scripts/$$f $(DESTDIR)$(INST_BIN) ; done
