From fe52a09cdd6e4d70c2a6523a2caaa5e3ce33b3c9 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 15 Nov 2015 18:24:15 -0700
Subject: [PATCH] arch linux arm modifications

---
 include/configs/odroid_xu3.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/odroid_xu3.h b/include/configs/odroid_xu3.h
index 5196d58..294a657 100644
--- a/include/configs/odroid_xu3.h
+++ b/include/configs/odroid_xu3.h
@@ -12,7 +12,7 @@
 #include <configs/exynos5-common.h>
 
 #undef CONFIG_ENV_IS_IN_SPI_FLASH
-#define CONFIG_IDENT_STRING		" for ODROID-XU3"
+#define CONFIG_IDENT_STRING		" Arch Linux ARM"
 
 #define CONFIG_BOARD_COMMON
 
-- 
2.9.0

