diff --git a/meson.build b/meson.build
index 3858cbe..d407be7 100644
--- a/meson.build
+++ b/meson.build
@@ -48,6 +48,8 @@ gio_unix_dep = dependency('gio-unix-2.0', version: '>=' + glib_min_version)
 
 xsltproc = find_program('xsltproc', required: get_option('gtk-doc') or get_option('man'))
 
+libm = cc.find_library('m', required : false)
+
 # Resolve OS backend
 os_backend = get_option('os_backend')
 if os_backend == 'auto'
diff --git a/src/meson.build b/src/meson.build
index d0d8141..0732370 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -6,7 +6,7 @@ upowerd_deps = declare_dependency(
         include_directories('../dbus'),
     ],
     dependencies: [
-        glib_dep, gobject_dep, gio_dep, gio_unix_dep, libupower_glib_dep, upowerd_dbus_dep
+        glib_dep, gobject_dep, gio_dep, gio_unix_dep, libupower_glib_dep, upowerd_dbus_dep, libm
     ],
     compile_args: [
         '-DUP_COMPILATION',
