--- source/creator/creator.c	2007-01-28 05:45:58.000000000 +0000
+++ source/creator/creator.c.new	2007-02-24 21:37:25.000000000 +0000
@@ -241,7 +241,7 @@
 		having to fix all faulty saved 64 bits files (ton) */
 	if(sizeof(void *)==8) {
 		printf("64 bits compiles will give incorrectly saved .blend files. Do not use it. For testing purposes please remove this line from creator.c\n"); 
-		exit(0);
+	//	exit(0);
 	}
 #else
 	if(sizeof(void *)==8) {
