diff -ruNa a/xdelta3-main.h b/xdelta3-main.h
--- a/xdelta3-main.h    2009-01-30 05:59:02.000000000 +0100
+++ b/xdelta3-main.h    2009-05-13 12:43:00.000000000 +0200
@@ -355,6 +355,7 @@
     RD_NONEXTERNAL },
   { "bzip2",    "-cf",   "bzip2",      "-dcf",   "B", "BZh",          3, 0 },
   { "gzip",     "-cf",   "gzip",       "-dcf",   "G", "\037\213",     2, 0 },
+  { "xz",       "-cf",   "xz",         "-dcf",   "Y", "\xfd\x37\x7a\x58\x5a\x00",     2, 0 },
   { "compress", "-cf",   "uncompress", "-cf",    "Z", "\037\235",     2, 0 },
 
   /* TODO: add commandline support for magic-less formats */

