diff --git a/src/base/atomicops.h b/src/base/atomicops.h
index dac95be..554d0a1 100644
--- a/src/base/atomicops.h
+++ b/src/base/atomicops.h
@@ -112,7 +112,7 @@
 #include "base/atomicops-internals-gcc.h"
 #elif defined(__MACH__) && defined(__APPLE__)
 #include "base/atomicops-internals-macosx.h"
-#elif defined(__GNUC__) && defined(ARMV6)
+#elif defined(__GNUC__) && defined(ARMV7)
 #include "base/atomicops-internals-arm-v6plus.h"
 #elif defined(ARMV3)
 #include "base/atomicops-internals-arm-generic.h"
