diff -Nru aumix-2.7/Makefile.in aumix-2.7-new/Makefile.in
--- aumix-2.7/Makefile.in	Tue Jun 27 07:51:42 2000
+++ aumix-2.7-new/Makefile.in	Fri Oct  5 18:29:37 2001
@@ -102,7 +102,7 @@
 all: all-redirect
 .SUFFIXES:
 $(srcdir)/Makefile.in: Makefile.am $(top_srcdir)/configure.in $(ACLOCAL_M4) 
-	cd $(top_srcdir) && $(AUTOMAKE) --gnu --include-deps Makefile
+	cd $(top_srcdir) 
 
 Makefile: $(srcdir)/Makefile.in  $(top_builddir)/config.status
 	cd $(top_builddir) \
@@ -114,7 +114,7 @@
 config.status: $(srcdir)/configure $(CONFIG_STATUS_DEPENDENCIES)
 	$(SHELL) ./config.status --recheck
 $(srcdir)/configure: $(srcdir)/configure.in $(ACLOCAL_M4) $(CONFIGURE_DEPENDENCIES)
-	cd $(srcdir) && $(AUTOCONF)
+	cd $(srcdir) 
 
 config.h: stamp-h
 	@if test ! -f $@; then \
@@ -132,7 +132,7 @@
 		$(MAKE) $(srcdir)/stamp-h.in; \
 	else :; fi
 $(srcdir)/stamp-h.in: $(top_srcdir)/configure.in $(ACLOCAL_M4) acconfig.h
-	cd $(top_srcdir) && $(AUTOHEADER)
+	cd $(top_srcdir) 
 	@echo timestamp > $(srcdir)/stamp-h.in 2> /dev/null
 
 mostlyclean-hdr:
diff -Nru aumix-2.7/src/Makefile.in aumix-2.7-new/src/Makefile.in
--- aumix-2.7/src/Makefile.in	Tue Jun 27 07:51:42 2000
+++ aumix-2.7-new/src/Makefile.in	Fri Oct  5 18:30:02 2001
@@ -123,8 +123,6 @@
 all: all-redirect
 .SUFFIXES:
 .SUFFIXES: .S .c .o .s
-$(srcdir)/Makefile.in: Makefile.am $(top_srcdir)/configure.in $(ACLOCAL_M4) 
-	cd $(top_srcdir) && $(AUTOMAKE) --gnu --include-deps src/Makefile
 
 Makefile: $(srcdir)/Makefile.in  $(top_builddir)/config.status
 	cd $(top_builddir) \

