diff -r 1a327a09262e netx/net/sourceforge/jnlp/runtime/Boot.java
--- a/netx/net/sourceforge/jnlp/runtime/Boot.java   Mon Jun 10 13:22:53 2013 +0200
+++ b/netx/net/sourceforge/jnlp/runtime/Boot.java   Wed Jun 19 10:32:32 2013 +0200
@@ -35,6 +35,8 @@
 import net.sourceforge.jnlp.cache.UpdatePolicy;
 import net.sourceforge.jnlp.security.viewer.CertificateViewer;
 import net.sourceforge.jnlp.services.ServiceUtil;
+import sun.awt.AppContext;
+import sun.awt.SunToolkit;
 
 /**
  * This is the main entry point for the JNLP client.  The main
@@ -113,6 +115,9 @@
      * Launch the JNLP file specified by the command-line arguments.
      */
     public static void main(String[] argsIn) {
+        if (AppContext.getAppContext() == null) {
+            SunToolkit.createNewAppContext();
+        }
         args = argsIn;
 
         if (null != getOption("-viewer")) {
diff -r 1a327a09262e netx/net/sourceforge/jnlp/runtime/JNLPRuntime.java
--- a/netx/net/sourceforge/jnlp/runtime/JNLPRuntime.java    Mon Jun 10 13:22:53 2013 +0200
+++ b/netx/net/sourceforge/jnlp/runtime/JNLPRuntime.java    Wed Jun 19 10:32:32 2013 +0200
@@ -234,7 +234,7 @@
         try {
             UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
         } catch (Exception e) {
-            // ignore it
+            e.printStackTrace();
         }
 
         doMainAppContextHacks();
diff -r 1a327a09262e plugin/icedteanp/java/sun/applet/PluginMain.java
--- a/plugin/icedteanp/java/sun/applet/PluginMain.java  Mon Jun 10 13:22:53 2013 +0200
+++ b/plugin/icedteanp/java/sun/applet/PluginMain.java  Wed Jun 19 10:32:32 2013 +0200
@@ -72,6 +72,8 @@
 import java.net.ProxySelector;
 import java.util.Enumeration;
 import java.util.Properties;
+import sun.awt.AppContext;
+import sun.awt.SunToolkit;
 
 import net.sourceforge.jnlp.config.DeploymentConfiguration;
 import net.sourceforge.jnlp.runtime.JNLPRuntime;
@@ -94,6 +96,9 @@
      */
     public static void main(String args[])
             throws IOException {
+        if (AppContext.getAppContext() == null) {
+            SunToolkit.createNewAppContext();
+        }
         if (args.length != 2 || !(new File(args[0]).exists()) || !(new File(args[1]).exists())) {
             System.err.println("Invalid pipe names provided. Refusing to proceed.");
             System.exit(1);
