From a198443121aa446161d04d04b669218f58f7b7e6 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Mon, 19 Oct 2015 19:04:18 -0600
Subject: [PATCH 2/2] fix for gcc 5

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 11062f0..d0d73d4 100644
--- a/Makefile
+++ b/Makefile
@@ -214,7 +214,7 @@ GC_SECTIONS=--gc-sections
 NOSTDINC_FLAGS=-nostdinc -isystem $(shell $(CC) -print-file-name=include)
 
 CPPFLAGS=$(NOSTDINC_FLAGS) -ffunction-sections -g -Os -Wall \
-	-fno-stack-protector -fno-common \
+	-fno-stack-protector -fno-common -fgnu89-inline \
 	-I$(INCL) -Iinclude -Ifs/include \
 	-DAT91BOOTSTRAP_VERSION=\"$(VERSION)$(REV)$(SCMINFO)\" -DCOMPILE_TIME="\"$(DATE)\""
 
-- 
2.6.1

