--- screem-0.16.1/src/screem-dbus.c.orig	2007-03-24 16:11:20.000000000 +0000
+++ screem-0.16.1/src/screem-dbus.c	2007-03-24 16:21:33.000000000 +0000
@@ -129,19 +129,9 @@ static void screem_dbus_finalize( GObjec
 	
 #ifdef HAVE_DBUS
 	if( priv->conn ) {
-#if DBUS_VERSION > 33000
-		dbus_connection_close( priv->conn );
-#else
-		dbus_connection_disconnect( priv->conn );
-#endif
 		dbus_connection_unref( priv->conn );
 	}
 	if( priv->sconn ) {
-#if DBUS_VERSION > 33000
-		dbus_connection_close( priv->sconn );
-#else
-		dbus_connection_disconnect( priv->sconn );
-#endif
 		dbus_connection_unref( priv->sconn );
 	}
 #endif
