From e5071421ad33702cdc2502fc0ce734c0968d8ad6 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 13 Oct 2015 20:45:53 -0600
Subject: [PATCH] arm fixes

---
 configure                       | 5 ++++-
 deps/v8/gypfiles/toolchain.gypi | 4 ++--
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/configure b/configure
index d926cec..f3fad3e 100755
--- a/configure
+++ b/configure
@@ -738,8 +738,11 @@ def configure_arm(o):
   if is_arch_armv7():
     arm_fpu = 'vfpv3'
     o['variables']['arm_version'] = '7'
+  elif is_arch_armv6():
+    o['variables']['arm_version'] = '6'
   else:
-    o['variables']['arm_version'] = '6' if is_arch_armv6() else 'default'
+    o['variables']['arm_fpu'] = ''
+    o['variables']['arm_version'] = '5'
 
   o['variables']['arm_thumb'] = 0      # -marm
   o['variables']['arm_float_abi'] = arm_float_abi
diff --git a/deps/v8/gypfiles/toolchain.gypi b/deps/v8/gypfiles/toolchain.gypi
index 95eb1d9..6416a1e 100644
--- a/deps/v8/gypfiles/toolchain.gypi
+++ b/deps/v8/gypfiles/toolchain.gypi
@@ -207,7 +207,7 @@
                   }],
                   [ 'arm_version==7 or arm_version=="default"', {
                     'conditions': [
-                      [ 'arm_fpu!="default"', {
+                      [ 'arm_fpu!="default" and arm_fpu!=""', {
                         'cflags': ['-mfpu=<(arm_fpu)',],
                       }],
                     ],
@@ -250,7 +250,7 @@
                   }],
                   [ 'arm_version==7 or arm_version=="default"', {
                     'conditions': [
-                      [ 'arm_fpu!="default"', {
+                      [ 'arm_fpu!="default" and arm_fpu!=""', {
                         'cflags': ['-mfpu=<(arm_fpu)',],
                       }],
                     ],
-- 
2.10.2

