From 3248eca016badde88fe1f8cb97c08344ee182a23 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 13 Oct 2015 20:45:53 -0600
Subject: [PATCH] arm fixes

---
 configure                    | 5 ++++-
 deps/v8/build/toolchain.gypi | 4 ++--
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/configure b/configure
index b4936f1..53192eb 100755
--- a/configure
+++ b/configure
@@ -626,8 +626,11 @@ def configure_arm(o):
   if is_arch_armv7():
     arm_fpu = 'vfpv3'
     o['variables']['arm_version'] = '7'
+  elif is_arch_armv6():
+    o['variables']['arm_version'] = '6'
   else:
-    o['variables']['arm_version'] = '6' if is_arch_armv6() else 'default'
+    o['variables']['arm_fpu'] = ''
+    o['variables']['arm_version'] = '5'
 
   o['variables']['arm_thumb'] = 0      # -marm
   o['variables']['arm_float_abi'] = arm_float_abi
diff --git a/deps/v8/build/toolchain.gypi b/deps/v8/build/toolchain.gypi
index 4dbf42b..8d3e968 100644
--- a/deps/v8/build/toolchain.gypi
+++ b/deps/v8/build/toolchain.gypi
@@ -203,7 +203,7 @@
                   }],
                   [ 'arm_version==7 or arm_version=="default"', {
                     'conditions': [
-                      [ 'arm_fpu!="default"', {
+                      [ 'arm_fpu!="default" and arm_fpu!=""', {
                         'cflags': ['-mfpu=<(arm_fpu)',],
                       }],
                     ],
@@ -246,7 +246,7 @@
                   }],
                   [ 'arm_version==7 or arm_version=="default"', {
                     'conditions': [
-                      [ 'arm_fpu!="default"', {
+                      [ 'arm_fpu!="default" and arm_fpu!=""', {
                         'cflags': ['-mfpu=<(arm_fpu)',],
                       }],
                     ],
-- 
2.6.1

