--- node.gyp.orig	2019-08-24 10:05:57.160565594 -0600
+++ node.gyp	2019-08-24 10:07:00.240470669 -0600
@@ -311,6 +311,12 @@
         'deps/v8/include'
       ],
 
+      'link_settings': {
+        'libraries': [
+          '-latomic',
+        ],
+      },
+
       'sources': [
         'src/node_main.cc'
       ],
--- tools/v8_gypfiles/v8.gyp.orig	2019-08-24 12:42:33.041596717 -0600
+++ tools/v8_gypfiles/v8.gyp	2019-08-24 13:02:49.504187934 -0600
@@ -858,7 +858,7 @@
         }],
         # Platforms that don't have Compare-And-Swap (CAS) support need to link atomic library
         # to implement atomic memory access
-        ['v8_current_cpu in ["mips", "mipsel", "mips64", "mips64el", "ppc", "ppc64", "s390x"]', {
+        ['v8_current_cpu in ["mips", "mipsel", "mips64", "mips64el", "ppc", "ppc64", "s390x", "arm"]', {
           'link_settings': {
             'libraries': ['-latomic', ],
           },
