--- node.gyp.orig	2019-10-14 05:32:27.370777519 -0600
+++ node.gyp	2019-10-14 05:39:18.842470025 -0600
@@ -314,6 +314,12 @@
         'deps/v8/include'
       ],
 
+      'link_settings': {
+        'libraries': [
+          '-latomic',
+        ],
+      },
+
       'sources': [
         'src/node_main.cc'
       ],
--- tools/v8_gypfiles/v8.gyp.orig	2019-10-14 05:32:59.750753410 -0600
+++ tools/v8_gypfiles/v8.gyp	2019-10-14 05:39:18.842470025 -0600
@@ -795,7 +795,7 @@
         }],
         # Platforms that don't have Compare-And-Swap (CAS) support need to link atomic library
         # to implement atomic memory access
-        ['v8_current_cpu in ["mips", "mipsel", "mips64", "mips64el", "ppc"]', {
+        ['v8_current_cpu in ["mips", "mipsel", "mips64", "mips64el", "ppc", "arm"]', {
           'link_settings': {
             'libraries': ['-latomic', ],
           },
