--- node.gyp.orig	2018-08-16 12:34:32.110341767 -0600
+++ node.gyp	2018-08-16 12:35:47.914941800 -0600
@@ -320,6 +320,12 @@
         '<(SHARED_INTERMEDIATE_DIR)' # for node_natives.h
       ],
 
+      'link_settings': {
+        'libraries': [
+          '-latomic',
+        ],
+      },
+
       'sources': [
         'src/async_wrap.cc',
         'src/bootstrapper.cc',
--- deps/v8/gypfiles/v8.gyp.orig	2018-10-29 07:54:38.292831964 -0600
+++ deps/v8/gypfiles/v8.gyp	2018-10-29 07:56:38.192570653 -0600
@@ -483,6 +483,11 @@
         '<(DEPTH)',
         '<(SHARED_INTERMEDIATE_DIR)'
       ],
+      'link_settings': {
+        'libraries': [
+          '-latomic',
+        ],
+      },
       'sources': [
         '<@(inspector_all_sources)',
         '../include//v8-inspector-protocol.h',
