diff --git a/projects/libcxx/include/atomic.orig b/projects/libcxx/include/atomic
index 0fc799a..a838635 100644
--- a/projects/libcxx/include/atomic.orig
+++ b/projects/libcxx/include/atomic
@@ -2791,10 +2791,15 @@ typedef conditional<_LIBCPP_CONTENTION_LOCK_FREE, __cxx_contention_t, char>::typ
 typedef conditional<_LIBCPP_CONTENTION_LOCK_FREE, __cxx_contention_t, unsigned char>::type      __libcpp_unsigned_lock_free;
 #else
     // No signed/unsigned lock-free types
+#define LOCK_FREE_NOT_AVAILABLE
 #endif
 
+#ifndef LOCK_FREE_NOT_AVAILABLE
 typedef atomic<__libcpp_signed_lock_free> atomic_signed_lock_free;
 typedef atomic<__libcpp_unsigned_lock_free> atomic_unsigned_lock_free;
+#endif
+
+#undef LOCK_FREE_NOT_AVAILABLE
 
 #define ATOMIC_FLAG_INIT {false}
 #define ATOMIC_VAR_INIT(__v) {__v}
