diff --git a/drivers/net/wireless/ti/wl12xx/wl12xx.h b/drivers/net/wireless/ti/wl12xx/wl12xx.h
index 634ad00..310df52 100644
--- a/drivers/net/wireless/ti/wl12xx/wl12xx.h
+++ b/drivers/net/wireless/ti/wl12xx/wl12xx.h
@@ -32,16 +32,16 @@
 
 /* minimum FW required for driver for wl127x */
 #define WL127X_CHIP_VER		6
-#define WL127X_IFTYPE_VER	3
+#define WL127X_IFTYPE_VER	WLCORE_FW_VER_IGNORE
 #define WL127X_MAJOR_VER	10
-#define WL127X_SUBTYPE_VER	2
+#define WL127X_SUBTYPE_VER	WLCORE_FW_VER_IGNORE
 #define WL127X_MINOR_VER	115
 
 /* minimum FW required for driver for wl128x */
 #define WL128X_CHIP_VER		7
-#define WL128X_IFTYPE_VER	3
+#define WL128X_IFTYPE_VER	WLCORE_FW_VER_IGNORE
 #define WL128X_MAJOR_VER	10
-#define WL128X_SUBTYPE_VER	2
+#define WL128X_SUBTYPE_VER	WLCORE_FW_VER_IGNORE
 #define WL128X_MINOR_VER	115
 
 #define WL12XX_AGGR_BUFFER_SIZE	(4 * PAGE_SIZE)
