--- drivers/net/wireless/ti/wl18xx/wl18xx.h.orig	2013-06-08 00:09:45.979268733 -0600
+++ drivers/net/wireless/ti/wl18xx/wl18xx.h	2013-06-08 00:10:03.334181254 -0600
@@ -27,8 +27,8 @@
 /* minimum FW required for driver */
 #define WL18XX_CHIP_VER		8
 #define WL18XX_IFTYPE_VER	5
-#define WL18XX_MAJOR_VER	0
-#define WL18XX_SUBTYPE_VER	0
+#define WL18XX_MAJOR_VER	WLCORE_FW_VER_IGNORE
+#define WL18XX_SUBTYPE_VER	WLCORE_FW_VER_IGNORE
 #define WL18XX_MINOR_VER	28
 
 #define WL18XX_CMD_MAX_SIZE          740
