--- src/optimize_usage.h.orig	2007-04-22 15:40:51.000000000 +0000
+++ src/optimize_usage.h	2007-04-22 15:41:25.000000000 +0000
@@ -16,7 +16,7 @@
 ** Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 */
 
-extern GtkWidget *opt_try_harder;
+static GtkWidget *opt_try_harder;
 extern long int opt_cdsize;
 
 void open_optimize_size_win(GtkWidget *, gpointer);
--- src/rip_audio.h.orig	2007-04-22 15:41:45.000000000 +0000
+++ src/rip_audio.h	2007-04-22 15:41:59.000000000 +0000
@@ -20,4 +20,4 @@
 
 extern char *cdda2wav_path, *cdda_device_path, *cdparanoia_path;
 extern char *def_rip_path;
-extern GtkWidget *rip_path_entry;
+static GtkWidget *rip_path_entry;
