diff -upr cherokee-0.98.1.orig/cherokee.conf.sample.pre cherokee-0.98.1/cherokee.conf.sample.pre
--- cherokee-0.98.1.orig/cherokee.conf.sample.pre	2009-01-23 21:18:41.000000000 +0200
+++ cherokee-0.98.1/cherokee.conf.sample.pre	2009-02-20 17:03:07.000000000 +0200
@@ -26,10 +26,10 @@ vserver!1!directory_index = index.php,in

 vserver!1!logger = combined
 vserver!1!logger!access!type = file
-vserver!1!logger!access!filename = %localstatedir%/log/cherokee.access
+vserver!1!logger!access!filename = %localstatedir%/log/cherokee/cherokee.access
 vserver!1!logger!access!buffsize = 16384
 vserver!1!logger!error!type = file
-vserver!1!logger!error!filename = %localstatedir%/log/cherokee.error
+vserver!1!logger!error!filename = %localstatedir%/log/cherokee/cherokee.error

 vserver!1!rule!1!match = default
 vserver!1!rule!1!handler = common
