diff --git a/include/configs/tbs2910.h b/include/configs/tbs2910.h
index 7089378..d91eed4 100644
--- a/include/configs/tbs2910.h
+++ b/include/configs/tbs2910.h
@@ -227,13 +227,15 @@
 	"bootargs_mmc1=console=ttymxc0,115200 di0_primary console=tty1\0" \
 	"bootargs_mmc2=video=mxcfb0:dev=hdmi,1920x1080M@60 " \
 			"video=mxcfb1:off video=mxcfb2:off fbmem=28M\0" \
-	"bootargs_mmc3=root=/dev/mmcblk0p1 rootwait consoleblank=0 quiet\0" \
+	"bootargs_mmc3=root=/dev/mmcblk0p1 rw rootwait consoleblank=0\0" \
 	"bootargs_mmc=setenv bootargs ${bootargs_mmc1} ${bootargs_mmc2} " \
 			"${bootargs_mmc3}\0" \
 	"bootargs_upd=setenv bootargs console=ttymxc0,115200 " \
 			"rdinit=/sbin/init enable_wait_mode=off\0" \
-	"bootcmd_mmc=run bootargs_mmc; mmc dev 2; " \
-			"mmc read 0x10800000 0x800 0x4000; bootm\0" \
+	"bootcmd_mmc=run bootargs_mmc; mmc dev 1; " \
+			"load mmc 1 0x10800000 /boot/zImage; " \
+			"load mmc 1 0x11000000 /boot/dtbs/imx6q-tbs2910.dtb; " \
+			"bootz 0x10800000 - 0x11000000\0" \
 	"bootcmd_up1=load mmc 1 0x10800000 uImage\0" \
 	"bootcmd_up2=load mmc 1 0x10d00000 uramdisk.img; " \
 			"run bootargs_upd; " \
