--- php.ini-production.orig
+++ php.ini-production
@@ -756,7 +756,7 @@
 
 ; Directory in which the loadable extensions (modules) reside.
 ; http://php.net/extension-dir
-;extension_dir = "./"
+extension_dir = "/usr/lib/php7/modules/"
 ; On windows:
 ;extension_dir = "ext"
 
@@ -904,49 +904,45 @@
 ; deprecated in a future PHP major version. So, when it is possible, please
 ; move to the new ('extension=<ext>) syntax.
 ;
-; Notes for Windows environments :
-;
-; - Many DLL files are located in the extensions/ (PHP 4) or ext/ (PHP 5+)
-;   extension folders as well as the separate PECL DLL download (PHP 5+).
-;   Be sure to appropriately set the extension_dir directive.
-;
+;extension=bcmath
 ;extension=bz2
-;extension=curl
+;extension=calendar
+extension=curl
+;extension=dba
+;extension=enchant
+;extension=exif
 ;extension=ffi
 ;extension=ftp
-;extension=fileinfo
-;extension=gd2
+;extension=gd
 ;extension=gettext
 ;extension=gmp
-;extension=intl
+;extension=iconv
 ;extension=imap
+;extension=intl
 ;extension=ldap
-;extension=mbstring
-;extension=exif      ; Must be after mbstring as it depends on it
 ;extension=mysqli
-;extension=oci8_12c  ; Use with Oracle Database 12c Instant Client
 ;extension=odbc
-;extension=openssl
-;extension=pdo_firebird
+;zend_extension=opcache
+;extension=pdo_dblib
 ;extension=pdo_mysql
-;extension=pdo_oci
 ;extension=pdo_odbc
 ;extension=pdo_pgsql
 ;extension=pdo_sqlite
 ;extension=pgsql
+;extension=pspell
 ;extension=shmop
-
-; The MIBS data available in the PHP distribution must be installed.
-; See http://www.php.net/manual/en/snmp.installation.php
 ;extension=snmp
-
 ;extension=soap
 ;extension=sockets
 ;extension=sodium
 ;extension=sqlite3
+;extension=sysvmsg
+;extension=sysvsem
+;extension=sysvshm
 ;extension=tidy
 ;extension=xmlrpc
 ;extension=xsl
+extension=zip
 
 ;;;;;;;;;;;;;;;;;;;
 ; Module Settings ;
