--- a/pm/sleep.d/98smart-kernel-video.old	2010-01-07 08:31:25.275911479 -0800
+++ a/pm/sleep.d/98smart-kernel-video	    2010-01-07 08:32:01.596732967 -0800
@@ -32,7 +32,7 @@
 {
     # if we are running with a KMS-enabled video driver, we should not
     # attempt to run any quirks
-    grep -q -E '(noveau|drm)fb' /proc/fb || return 1
+    grep -q -E '(nouveau|drm)fb' /proc/fb || return 1
     remove_all_video_quirks
     add_parameters --quirk-no-chvt
 }
