also find gzipped modules
--- rtslib/rtslib/utils.py	2011-10-06 00:46:09.000000000 +0200
+++ rtslib/rtslib/utils.py	2011-10-06 01:36:26.000000000 +0200
@@ -604,8 +604,8 @@
     '''
     kver = os.uname()[2]
     depfile = "/lib/modules/%s/modules.dep" % kver
-    return [module.split(".")[0] for module in
-            re.findall(r"[a-zA-Z0-9_-]+\.ko:", fread(depfile))]
+    return [module[0] for module in
+            re.findall(r"([a-zA-Z0-9_-]+)\.ko(.gz)?:", fread(depfile))]
 
 def list_loaded_kernel_modules():
     '''
