--- vnc_unixsrc/Xvnc/config/cf/linux.cf.orig	2010-09-21 20:00:26.000000000 +0000
+++ vnc_unixsrc/Xvnc/config/cf/linux.cf	2010-09-21 20:02:08.000000000 +0000
@@ -249,42 +249,36 @@
 
 #ifdef i386Architecture
 #define OptimizedCDebugFlags	DefaultGcc2i386Opt
-#define LinuxMachineDefines	-D__i386__
 #define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME -DPART_NET
 #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
 #endif /* i386Architecture */
 
 #ifdef x86_64Architecture
 #define OptimizedCDebugFlags	-O2
-#define LinuxMachineDefines	-D__x86_64__
 #define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME -DPART_NET
 #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines -D_XSERVER64
 #endif /* x86_64Architecture */
 
 #ifdef AlphaArchitecture
 #define OptimizedCDebugFlags	-O2
-#define LinuxMachineDefines	-D__alpha__
 #define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME -DPART_NET
 #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines -D_XSERVER64
 #endif /* AlphaArchitecture */
 
 #ifdef Mc68020Architecture
 #define OptimizedCDebugFlags	-O2
-#define LinuxMachineDefines 	-D__mc68000__
 #define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME -DPART_NET
 #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
 #endif /* Mc68020Architecture */
 
 #ifdef s390Architecture
 #define OptimzedCDebugFlags	-O2
-#define LinuxMachineDefines	-D__s390__
 #define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME -DPART_NET
 #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
 #endif /* s390Architecture */
 
 #ifdef s390xArchitecture
 #define OptimzedCDebugFlags	-O2
-#define LinuxMachineDefines	-D__s390x__
 #define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME -DPART_NET
 #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
 #endif /* s390xArchitecture */
@@ -292,7 +286,6 @@
 #ifdef PowerPCArchitecture
 #define DefaultCCOptions        -fsigned-char
 #define OptimizedCDebugFlags	-O2
-#define LinuxMachineDefines 	-D__powerpc__
 #define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME -DPART_NET
 #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
 #endif /* PowerPCArchitecture */
@@ -300,14 +293,12 @@
 #ifdef ArmArchitecture
 #define DefaultCCOptions	-fsigned-char
 #define OptimizedCDebugFlags	-O2
-#define LinuxMachineDefines	-D__arm__
 #define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME -DPART_NET
 #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
 #endif /* Arm */
 
 #ifdef SparcArchitecture
 #define OptimizedCDebugFlags	-O2
-#define LinuxMachineDefines 	-D__sparc__
 #define ServerOSDefines		-DDDXOSINIT
 #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
 #define AsVISOption		-Av9a
@@ -327,10 +318,10 @@
    Always define _REENTRANT since we may use threaded libraries */
 
 /* REMOVE
-#define StandardDefines		-Dlinux LinuxMachineDefines LinuxSourceDefines
+#define StandardDefines		-Dlinux LinuxSourceDefines
 */
 
-#define StandardDefines		-Dlinux LinuxMachineDefines \
+#define StandardDefines		-Dlinux \
 				LinuxSourceDefines -D_REENTRANT
 
 /* END ORL VNC modification
