From b3cb4617e396829c1344cb033c75062ada792bd4 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Mon, 9 Jan 2017 06:28:14 -0700
Subject: [PATCH] AArch64 cpuid fix

Signed-off-by: Kevin Mihelich <kevin@archlinuxarm.org>
---
 libsent/src/phmm/calc_dnn.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libsent/src/phmm/calc_dnn.c b/libsent/src/phmm/calc_dnn.c
index c7a8a53..34dcd1b 100644
--- a/libsent/src/phmm/calc_dnn.c
+++ b/libsent/src/phmm/calc_dnn.c
@@ -12,7 +12,7 @@
 #ifdef _WIN32
 #include <intrin.h>
 #else
-#if defined(__arm__) || TARGET_OS_IPHONE
+#if defined(__arm__) || defined(__aarch64__) || TARGET_OS_IPHONE
 #else
 #include <cpuid.h>
 #endif
-- 
2.10.2

