From 06b026817594649a0e12f3687d5a4e400e3a91c4 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Mon, 14 Nov 2016 20:51:49 -0700
Subject: [PATCH] AArch64 cpuid fix

Signed-off-by: Kevin Mihelich <kevin@archlinuxarm.org>
---
 libsent/src/phmm/calc_dnn.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libsent/src/phmm/calc_dnn.c b/libsent/src/phmm/calc_dnn.c
index bfd741d..27d3c7c 100644
--- a/libsent/src/phmm/calc_dnn.c
+++ b/libsent/src/phmm/calc_dnn.c
@@ -12,7 +12,7 @@
 #ifdef _WIN32
 #include <intrin.h>
 #else
-#ifndef __arm__
+#if !defined(__arm__) && !defined(__aarch64__)
 #include <cpuid.h>
 #endif
 #endif	/* _WIN32 */
-- 
2.10.2

