diff -ruN julius-4.4-orig/jcontrol/configure julius-4.4/jcontrol/configure
--- julius-4.4-orig/jcontrol/configure	2016-09-04 16:03:39.610663560 +0300
+++ julius-4.4/jcontrol/configure	2016-09-04 16:05:14.054157193 +0300
@@ -380,13 +380,6@@
     ;;
 
   *)
-    if test -n "`echo $ac_option| sed 's/[-a-z0-9.]//g'`"; then
-      echo "configure: warning: $ac_option: invalid host type" 1>&2
-    fi
-    if test "x$nonopt" != xNONE; then
-      { echo "configure: error: can only configure for one host and one target at a time" 1>&2; exit 1; }
-    fi
-    nonopt="$ac_option"
     ;;
 
   esac
diff -ruN julius-4.4-orig/mkgshmm/configure julius-4.4/mkgshmm/configure
--- julius-4.4-orig/mkgshmm/configure	2016-09-04 16:03:39.660661881 +0300
+++ julius-4.4/mkgshmm/configure	2016-09-04 16:04:55.021463305 +0300
@@ -380,13 +380,6 @@
     ;;
 
   *)
-    if test -n "`echo $ac_option| sed 's/[-a-z0-9.]//g'`"; then
-      echo "configure: warning: $ac_option: invalid host type" 1>&2
-    fi
-    if test "x$nonopt" != xNONE; then
-      { echo "configure: error: can only configure for one host and one target at a time" 1>&2; exit 1; }
-    fi
-    nonopt="$ac_option"
     ;;
 
   esac
diff -ruN julius-4.4-orig/libjulius/Makefile.in julius-4.4/libjulius/Makefile.in
--- julius-4.4-orig/libjulius/Makefile.in	2016-09-04 16:35:54.009006591 +0300
+++ julius-4.4/libjulius/Makefile.in	2016-09-04 16:35:13.733693061 +0300
@@ -94,12 +94,12 @@
 
 install.lib: $(TARGET)
 	${INSTALL} -d ${libdir}
 	${INSTALL_DATA} $(TARGET) ${libdir}
-	${INSTALL_DATA} $(PKGCONF_FILE) ${pkgconfigdir}
+	${INSTALL_DATA} -D $(PKGCONF_FILE) ${pkgconfigdir}/libjulius.pc
 
 install.include:
 	${INSTALL} -d ${includedir}/julius
	${INSTALL_DATA} include/julius/*.h ${includedir}/julius 
 
 install.bin: libjulius-config-dist
 	${INSTALL} -d ${bindir}
diff -ruN julius-4.4.1-orig/libsent/Makefile.in julius-4.4.1/libsent/Makefile.in
--- julius-4.4.1-orig/libsent/Makefile.in	2016-09-07 18:01:52.948127111 +0300
+++ julius-4.4.1/libsent/Makefile.in	2016-09-07 18:03:50.985347850 +0300
@@ -152,8 +152,8 @@
 
 install.lib: $(TARGET)
 	${INSTALL} -d ${libdir}
 	${INSTALL_DATA} $(TARGET) ${libdir}
-	${INSTALL_DATA} $(PKGCONF_FILE) ${pkgconfigdir}
+	${INSTALL_DATA} -D $(PKGCONF_FILE) ${pkgconfigdir}/libsent.pc
 
 install.include:
 	${INSTALL} -d ${includedir}/sent
