From 8e247bed77173287efa8f572a58e951a460700d0 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Fri, 9 Jun 2017 12:02:03 -0600
Subject: [PATCH] remove -msse2

---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e6d19e53..21151add 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,6 @@ IF(CMAKE_COMPILER_IS_GNUCC)
 	SET(CMAKE_BUILD_TYPE_DebugFast)
 	SET(CMAKE_CXX_FLAGS_DEBUGFAST "-g -O0")
 	SET(CMAKE_C_FLAGS_DEBUGFAST "-g -O0")
-	SET(CMAKE_C_FLAGS "-msse2")
 	if(NOT CMAKE_BUILD_TYPE)
 	        SET(CMAKE_BUILD_TYPE "DebugFast")
 	endif(NOT CMAKE_BUILD_TYPE)
-- 
2.12.0

