diff -Nur kazehakase-0.5.7.old/src/widget/kz-entry.c kazehakase-0.5.7/src/widget/kz-entry.c
--- kazehakase-0.5.7.old/src/widget/kz-entry.c	2009-08-29 01:15:11.000000000 +0300
+++ kazehakase-0.5.7/src/widget/kz-entry.c	2009-09-06 13:26:28.979887150 +0300
@@ -514,7 +514,7 @@
       
 		if (entry->visible)
 		{
-			g_string_prepend_len (tmp_string, entry->text, entry->x_n_bytes);
+			g_string_prepend_len (tmp_string, entry->text, -1);
 			g_string_insert (tmp_string, cursor_index, preedit_string);
 		}
 		else
@@ -523,7 +523,7 @@
 			gint preedit_len_chars;
 			gunichar invisible_char;
           
-			ch_len = g_utf8_strlen (entry->text, entry->x_n_bytes);
+			ch_len = g_utf8_strlen (entry->text, -1);
 			preedit_len_chars = g_utf8_strlen (preedit_string, -1);
 			ch_len += preedit_len_chars;
 
@@ -557,7 +557,7 @@
 	{
 		if (entry->visible)
 		{
-		pango_layout_set_text (layout, entry->text, entry->x_n_bytes);
+		pango_layout_set_text (layout, entry->text, -1);
 		}
 		else
 		{
