diff -cr zsnes_1_51/src/parsegen.cpp zsnes_1_51-patched/src/parsegen.cpp
*** zsnes_1_51/src/parsegen.cpp	2007-10-31 01:30:26.000000000 -0300
--- zsnes_1_51-patched/src/parsegen.cpp	2008-07-13 18:28:53.000000000 -0300
***************
*** 19,24 ****
--- 19,27 ----
  Config file handler creator by Nach (C) 2005-2007
  */
  
+ #include <cstring>
+ #include <cstdlib>
+ 
  #if !defined(__GNUC__) && !defined(_MSC_VER)
  #error You are using an unsupported compiler
  #endif
***************
*** 1822,1828 ****
    }
  }
  
! int main(size_t argc, const char *const *const argv)
  {
    const char *cheader_file = 0;
    bool compile = false;
--- 1825,1831 ----
    }
  }
  
! int main(int argc, const char *const *const argv)
  {
    const char *cheader_file = 0;
    bool compile = false;
diff -cr zsnes_1_51/src/tools/depbuild.cpp zsnes_1_51-patched/src/tools/depbuild.cpp
*** zsnes_1_51/src/tools/depbuild.cpp	2006-12-27 08:04:05.000000000 -0300
--- zsnes_1_51-patched/src/tools/depbuild.cpp	2008-07-13 18:30:46.000000000 -0300
***************
*** 183,189 ****
    }
  }
  
! int main(size_t argc, const char *const *const argv)
  {
    if (argc < 5)
    {
--- 183,189 ----
    }
  }
  
! int main(int argc, const char *const *const argv)
  {
    if (argc < 5)
    {
diff -cr zsnes_1_51/src/tools/strutil.h zsnes_1_51-patched/src/tools/strutil.h
*** zsnes_1_51/src/tools/strutil.h	2006-12-27 08:04:05.000000000 -0300
--- zsnes_1_51-patched/src/tools/strutil.h	2008-07-13 18:30:27.000000000 -0300
***************
*** 15,20 ****
--- 15,23 ----
  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
  
+ #include <cstring>
+ #include <cstdlib>
+ 
  /*
  This is part of a toolkit used to assist in ZSNES development
  */
