diff -urN guile-1.8.7.orig/libguile/guile-snarf-docs.in guile-1.8.7/libguile/guile-snarf-docs.in
--- guile-1.8.7.orig/libguile/guile-snarf-docs.in	2009-07-03 16:19:00.000000000 -0600
+++ guile-1.8.7/libguile/guile-snarf-docs.in	2011-01-21 18:05:47.000000000 -0700
@@ -23,4 +23,4 @@
 ## Let the user override the preprocessor autoconf found.
 test -n "${CPP+set}" || CPP="@CPP@"
 
-${CPP} -DSCM_MAGIC_SNARF_DOCS "$@"
+${CPP} -P -DSCM_MAGIC_SNARF_DOCS "$@"
