From 62f510702ef9c34ac50f8d8601a4290ab558464c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=96mer=20Fad=C4=B1l=20Usta?= <omerusta@gmail.com>
Date: Sun, 6 Jun 2021 08:41:05 +0000
Subject: [PATCH] Update Cmake and deps, Fix Freetype and FontConfig Linkage

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b237f68fcb6..cdcc02f0236 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -599,8 +593,8 @@ if(NOT WIN32 AND NOT APPLE)
 	)
 endif()
 
-if(NOT FONTCONFIG_FOUND OR NOT FREETYPE_FOUND)
-    set(FONTCONFIG_INCLUDE_DIR "")
+if(NOT Fontconfig_FOUND OR NOT FREETYPE_FOUND)
+    set(Fontconfig_INCLUDE_DIRS "")
     set(FREETYPE_INCLUDE_DIRS "")
 else()
     add_definitions( -DSHOULD_BUILD_FONT_CONVERSION )
diff --git a/libs/text/CMakeLists.txt b/libs/text/CMakeLists.txt
index 16d7e066fa2..429afe79704 100644
--- a/libs/text/CMakeLists.txt
+++ b/libs/text/CMakeLists.txt
@@ -152,11 +152,11 @@ if( SHOULD_BUILD_FEATURE_RDF )
 endif()
 
 if( FONTCONFIG_FOUND )
-    target_link_libraries(kotext PRIVATE ${FONTCONFIG_LIBRARIES})
+    target_link_libraries(kotext PRIVATE Fontconfig::Fontconfig)
 endif()
 
 if( FREETYPE_FOUND )
-    target_link_libraries(kotext PRIVATE ${FREETYPE_LIBRARIES})
+    target_link_libraries(kotext PRIVATE Freetype::Freetype)
 endif()
 
 
-- 
GitLab
