diff -urN a/krita/image/kis_fixed_point_maths.h b/krita/image/kis_fixed_point_maths.h
--- a/krita/image/kis_fixed_point_maths.h	2013-07-29 00:27:14.000000000 -0600
+++ b/krita/image/kis_fixed_point_maths.h	2013-08-07 11:11:37.025564690 -0600
@@ -34,7 +34,7 @@
         d = iValue << 8;
     }
 
-    KisFixedPoint(qreal fValue) {
+    KisFixedPoint(double fValue) {
         d = fValue * (1 << 8);
     }
 
diff -urN a/krita/image/tests/kis_filter_weights_buffer_test.cpp b/krita/image/tests/kis_filter_weights_buffer_test.cpp
--- a/krita/image/tests/kis_filter_weights_buffer_test.cpp	2013-07-29 00:27:14.000000000 -0600
+++ b/krita/image/tests/kis_filter_weights_buffer_test.cpp	2013-08-07 10:44:47.563284702 -0600
@@ -38,7 +38,7 @@
     KisFixedPoint fp2;
 
     const int startIndex = 1;
-    const int endIndex = 255 * qMin(scale, 1.0);
+    const int endIndex = 255 * qMin((double)scale, 1.0);
 
     fp1.from256Frac(startIndex);
     fp2.from256Frac(endIndex);
diff -urN a/krita/ui/kis_image_view_converter.cpp b/krita/ui/kis_image_view_converter.cpp
--- a/krita/ui/kis_image_view_converter.cpp	2013-07-29 00:27:15.000000000 -0600
+++ b/krita/ui/kis_image_view_converter.cpp	2013-08-07 10:44:47.563284702 -0600
@@ -40,7 +40,7 @@
 
 // remember here; document is postscript points;  view is krita pixels.
 
-void KisImageViewConverter::zoom(qreal *zoomX, qreal *zoomY) const
+void KisImageViewConverter::zoom(double *zoomX, double *zoomY) const
 {
     Q_ASSERT(zoomX);
     Q_ASSERT(zoomY);
diff -urN a/krita/ui/kis_image_view_converter.h b/krita/ui/kis_image_view_converter.h
--- a/krita/ui/kis_image_view_converter.h	2013-07-29 00:27:15.000000000 -0600
+++ b/krita/ui/kis_image_view_converter.h	2013-08-07 10:44:47.563284702 -0600
@@ -52,7 +52,7 @@
     using KoViewConverter::viewToDocument;
 
     /// reimplemented from superclass
-    void zoom(qreal *zoomX, qreal *zoomY) const;
+    void zoom(double *zoomX, double *zoomY) const;
 
     qreal documentToViewX(qreal documentX) const;
     qreal documentToViewY(qreal documentY) const;
diff -urN a/krita/ui/tests/kis_zoom_and_pan_test.cpp b/krita/ui/tests/kis_zoom_and_pan_test.cpp
--- a/krita/ui/tests/kis_zoom_and_pan_test.cpp	2013-07-29 00:27:15.000000000 -0600
+++ b/krita/ui/tests/kis_zoom_and_pan_test.cpp	2013-08-07 10:44:47.568284679 -0600
@@ -218,7 +218,7 @@
     qreal oldPreferredCenterFractionY = 1.0 * oldPreferredCenter.y() / oldDocumentSize.height();
 
     qreal roundingTolerance =
-        qMax(1.0, qMax(oldPreferredCenterFractionX, oldPreferredCenterFractionY) / k);
+        qMax((float)1.0, qMax(oldPreferredCenterFractionX, oldPreferredCenterFractionY) / k);
 
     /**
      * In the computation of the offset two roundings happen:
