diff -urN a/krita/image/kis_fixed_point_maths.h b/krita/image/kis_fixed_point_maths.h
--- a/krita/image/kis_fixed_point_maths.h	2014-03-02 07:46:08.000000000 -0700
+++ b/krita/image/kis_fixed_point_maths.h	2014-03-05 12:33:56.583956529 -0700
@@ -34,7 +34,7 @@
         d = iValue << 8;
     }
 
-    KisFixedPoint(qreal fValue) {
+    KisFixedPoint(double fValue) {
         d = fValue * (1 << 8);
     }
 
diff -urN a/krita/ui/kis_image_view_converter.cpp b/krita/ui/kis_image_view_converter.cpp
--- a/krita/ui/kis_image_view_converter.cpp	2014-03-02 07:46:08.000000000 -0700
+++ b/krita/ui/kis_image_view_converter.cpp	2014-03-05 12:33:56.583956529 -0700
@@ -40,7 +40,7 @@
 
 // remember here; document is postscript points;  view is krita pixels.
 
-void KisImageViewConverter::zoom(qreal *zoomX, qreal *zoomY) const
+void KisImageViewConverter::zoom(double *zoomX, double *zoomY) const
 {
     Q_ASSERT(zoomX);
     Q_ASSERT(zoomY);
diff -urN a/krita/ui/kis_image_view_converter.h b/krita/ui/kis_image_view_converter.h
--- a/krita/ui/kis_image_view_converter.h	2014-03-02 07:46:08.000000000 -0700
+++ b/krita/ui/kis_image_view_converter.h	2014-03-05 12:33:56.588956485 -0700
@@ -52,7 +52,7 @@
     using KoViewConverter::viewToDocument;
 
     /// reimplemented from superclass
-    void zoom(qreal *zoomX, qreal *zoomY) const;
+    void zoom(double *zoomX, double *zoomY) const;
 
     qreal documentToViewX(qreal documentX) const;
     qreal documentToViewY(qreal documentY) const;
