diff -urN a/krita/ui/kis_image_view_converter.cpp b/krita/ui/kis_image_view_converter.cpp
--- a/krita/ui/kis_image_view_converter.cpp	2013-04-08 03:23:17.000000000 -0500
+++ b/krita/ui/kis_image_view_converter.cpp	2013-05-21 17:20:34.227367776 -0500
@@ -40,7 +40,7 @@
 
 // remember here; document is postscript points;  view is krita pixels.
 
-void KisImageViewConverter::zoom(qreal *zoomX, qreal *zoomY) const
+void KisImageViewConverter::zoom(double *zoomX, double *zoomY) const
 {
     Q_ASSERT(zoomX);
     Q_ASSERT(zoomY);
diff -urN a/krita/ui/kis_image_view_converter.h b/krita/ui/kis_image_view_converter.h
--- a/krita/ui/kis_image_view_converter.h	2013-04-08 03:23:17.000000000 -0500
+++ b/krita/ui/kis_image_view_converter.h	2013-05-21 17:19:52.171064761 -0500
@@ -52,7 +52,7 @@
     using KoViewConverter::viewToDocument;
 
     /// reimplemented from superclass
-    void zoom(qreal *zoomX, qreal *zoomY) const;
+    void zoom(double *zoomX, double *zoomY) const;
 
     qreal documentToViewX(qreal documentX) const;
     qreal documentToViewY(qreal documentY) const;
diff -urN a/krita/ui/tests/kis_zoom_and_pan_test.cpp b/krita/ui/tests/kis_zoom_and_pan_test.cpp
--- a/krita/ui/tests/kis_zoom_and_pan_test.cpp	2013-04-08 03:24:15.000000000 -0500
+++ b/krita/ui/tests/kis_zoom_and_pan_test.cpp	2013-05-21 19:09:55.408078960 -0500
@@ -220,7 +220,7 @@
     qreal oldPreferredCenterFractionY = 1.0 * oldPreferredCenter.y() / oldDocumentSize.height();
 
     qreal roundingTolerance =
-        qMax(1.0, qMax(oldPreferredCenterFractionX, oldPreferredCenterFractionY) / k);
+        qMax((float)1.0, qMax(oldPreferredCenterFractionX, oldPreferredCenterFractionY) / k);
 
     /**
      * In the computation of the offset two roundings happen:
