Index: calligra-2.5.3/krita/ui/canvas/kis_coordinates_converter.cpp
===================================================================
--- calligra-2.5.3.orig/krita/ui/canvas/kis_coordinates_converter.cpp	2012-10-09 10:25:56.706529477 +0100
+++ calligra-2.5.3/krita/ui/canvas/kis_coordinates_converter.cpp	2012-10-09 10:27:14.963095078 +0100
@@ -73,8 +73,8 @@
 
     QPointF range = maxOffset - minOffset;
 
-    range.rx() = qMin(range.x(), 0.0);
-    range.ry() = qMin(range.y(), 0.0);
+    range.rx() = qMin(range.x(), (qreal)0.0);
+    range.ry() = qMin(range.y(), (qreal)0.0);
 
     range /= 2;
 
