diff -urN a/krita/image/kis_circle_mask_generator.cpp b/krita/image/kis_circle_mask_generator.cpp
--- a/krita/image/kis_circle_mask_generator.cpp	2015-02-24 05:53:08.000000000 -0700
+++ b/krita/image/kis_circle_mask_generator.cpp	2015-03-03 16:52:50.040204434 -0700
@@ -122,7 +122,7 @@
 void KisCircleMaskGenerator::setSoftness(qreal softness)
 {
     KisMaskGenerator::setSoftness(softness);
-    qreal safeSoftnessCoeff = 1.0 / qMax(0.01, softness);
+    qreal safeSoftnessCoeff = 1.0 / qMax(0.01, (double)softness);
 
     d->transformedFadeX = d->xfadecoef * safeSoftnessCoeff;
     d->transformedFadeY = d->yfadecoef * safeSoftnessCoeff;
diff -urN a/krita/image/kis_rect_mask_generator.cpp b/krita/image/kis_rect_mask_generator.cpp
--- a/krita/image/kis_rect_mask_generator.cpp	2015-02-24 05:53:08.000000000 -0700
+++ b/krita/image/kis_rect_mask_generator.cpp	2015-03-03 16:52:19.980206593 -0700
@@ -72,7 +72,7 @@
 void KisRectangleMaskGenerator::setSoftness(qreal softness)
 {
     KisMaskGenerator::setSoftness(softness);
-    qreal safeSoftnessCoeff = 1.0 / qMax(0.01, softness);
+    qreal safeSoftnessCoeff = 1.0 / qMax(0.01, (double)softness);
 
     d->transformedFadeX = d->xfadecoeff * safeSoftnessCoeff;
     d->transformedFadeY = d->yfadecoeff * safeSoftnessCoeff;
