diff -upr calligra-3.1.0.orig/filters/karbon/pdf/SvgOutputDev.cpp calligra-3.1.0/filters/karbon/pdf/SvgOutputDev.cpp
--- calligra-3.1.0.orig/filters/karbon/pdf/SvgOutputDev.cpp	2019-11-12 14:55:33.418358543 +0200
+++ calligra-3.1.0/filters/karbon/pdf/SvgOutputDev.cpp	2019-11-12 14:59:26.843199351 +0200
@@ -405,7 +405,7 @@ void SvgOutputDev::drawString(GfxState *
     const char * p = s->c_str();
     int len = s->getLength();
     CharCode code;
-    Unicode *u = nullptr;
+    const Unicode *u = nullptr;
     int uLen;
     double dx, dy, originX, originY;
     while (len > 0) {
