diff -up calligra-2.9.11/libs/flake/KoSnapStrategy.cpp.gcc6_arm calligra-2.9.11/libs/flake/KoSnapStrategy.cpp
--- calligra-2.9.11/libs/flake/KoSnapStrategy.cpp.gcc6_arm	2016-02-02 13:53:15.000000000 -0600
+++ calligra-2.9.11/libs/flake/KoSnapStrategy.cpp	2016-02-08 09:12:39.452896965 -0600
@@ -181,7 +181,7 @@ bool ExtensionSnapStrategy::snap(const Q
     Q_ASSERT(isfinite(maxSnapDistance));
 
     const qreal maxDistance = maxSnapDistance * maxSnapDistance;
-    qreal minDistances[2] = { HUGE_VAL, HUGE_VAL };
+    qreal minDistances[2] = { (qreal)HUGE_VAL, (qreal)HUGE_VAL };
 
     QPointF snappedPoints[2] = { mousePosition, mousePosition };
     QPointF startPoints[2];
