commit 58db794d10ab882a2ca9683476184758a58e4a2a
Author: Marijn Kruisselbrink <mkruisselbrink@kde.org>
Date:   Tue May 8 02:14:15 2012 +0200

    compile on arm

diff --git a/plugins/pathshapes/rectangle/RectangleShape.cpp b/plugins/pathshapes/rectangle/RectangleShape.cpp
index 7c42b70..e83874d 100644
--- a/plugins/pathshapes/rectangle/RectangleShape.cpp
+++ b/plugins/pathshapes/rectangle/RectangleShape.cpp
@@ -60,9 +60,9 @@ bool RectangleShape::loadOdf(const KoXmlElement &element, KoShapeLoadingContext
     } else {
         QString cornerRadius = element.attributeNS(KoXmlNS::draw, "corner-radius", "");
         if (! cornerRadius.isEmpty()) {
-            float radius = KoUnit::parseValue(cornerRadius);
-            m_cornerRadiusX = qMin(radius / (0.5 * size().width()) * 100, qreal(100));
-            m_cornerRadiusY = qMin(radius / (0.5 * size().height()) * 100, qreal(100));
+            qreal radius = KoUnit::parseValue(cornerRadius);
+            m_cornerRadiusX = qMin<qreal>(radius / (0.5 * size().width()) * 100, qreal(100));
+            m_cornerRadiusY = qMin<qreal>(radius / (0.5 * size().height()) * 100, qreal(100));
         }
     }
 
