diff --git a/CMakeLists.txt b/CMakeLists.txt
index 878caf74af4..4df52e9fe82 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -803,7 +803,7 @@ calligra_drop_product_on_bad_condition( FEATURE_RDF
     )
 
 calligra_drop_product_on_bad_condition( PART_SHEETS
-    EIGEN3_FOUND "Eigen devel not found"
+    Eigen3_FOUND "Eigen devel not found"
     )
 
 calligra_drop_product_on_bad_condition( OKULAR_GENERATOR_ODP
