diff -urN a/krita/plugins/paintops/chalk/kis_chalk_paintop_settings.cpp b/krita/plugins/paintops/chalk/kis_chalk_paintop_settings.cpp
--- a/krita/plugins/paintops/chalk/kis_chalk_paintop_settings.cpp	2015-08-29 08:39:29.000000000 -0600
+++ b/krita/plugins/paintops/chalk/kis_chalk_paintop_settings.cpp	2015-09-03 20:34:54.794178853 -0600
@@ -51,7 +51,7 @@
 
         QPainterPath tiltLine;
         QLineF tiltAngle(QPointF(0.0,0.0), QPointF(0.0,size));
-        tiltAngle.setLength(qMax(size*0.5, qreal(50.0)) * (1 - info.tiltElevation(info, 60.0, 60.0, true)));
+        tiltAngle.setLength(qMax(size*0.5, 50.0) * (1 - info.tiltElevation(info, 60.0, 60.0, true)));
         tiltAngle.setAngle((360.0 - fmod(KisPaintInformation::tiltDirection(info, true) * 360.0 + 270.0, 360.0))-3.0);
         tiltLine.moveTo(tiltAngle.p1());
         tiltLine.lineTo(tiltAngle.p2());
diff -urN a/krita/plugins/paintops/libpaintop/kis_brush_based_paintop_settings.cpp b/krita/plugins/paintops/libpaintop/kis_brush_based_paintop_settings.cpp
--- a/krita/plugins/paintops/libpaintop/kis_brush_based_paintop_settings.cpp	2015-08-29 08:39:29.000000000 -0600
+++ b/krita/plugins/paintops/libpaintop/kis_brush_based_paintop_settings.cpp	2015-09-03 20:34:32.054165712 -0600
@@ -73,7 +73,7 @@
         QPainterPath realOutline = brush->outline();
         QPainterPath tiltLine;
         QLineF tiltAngle(realOutline.boundingRect().center(), realOutline.boundingRect().topLeft());
-        tiltAngle.setLength(qMax(realOutline.boundingRect().width()*0.5, qreal(50.0)) * (1 - info.tiltElevation(info, 60.0, 60.0, true)));
+        tiltAngle.setLength(qMax(realOutline.boundingRect().width()*0.5, 50.0) * (1 - info.tiltElevation(info, 60.0, 60.0, true)));
         tiltAngle.setAngle((360.0 - fmod(KisPaintInformation::tiltDirection(info, true) * 360.0 + 270.0, 360.0))-3.0);
         tiltLine.moveTo(tiltAngle.p1());
         tiltLine.lineTo(tiltAngle.p2());
diff -urN a/krita/plugins/paintops/sketch/kis_sketch_paintop_settings.cpp b/krita/plugins/paintops/sketch/kis_sketch_paintop_settings.cpp
--- a/krita/plugins/paintops/sketch/kis_sketch_paintop_settings.cpp	2015-08-29 08:39:29.000000000 -0600
+++ b/krita/plugins/paintops/sketch/kis_sketch_paintop_settings.cpp	2015-09-03 22:14:28.339044535 -0600
@@ -62,7 +62,7 @@
         path = ellipseOutline(diameter, diameter, 1.0, 0.0/*brush->scale(), brush->angle()*/);
         QPainterPath tiltLine;
         QLineF tiltAngle(path.boundingRect().center(), path.boundingRect().topLeft());
-        tiltAngle.setLength(qMax(diameter*0.5, qreal(50.0)) * (1 - info.tiltElevation(info, 60.0, 60.0, true)));
+        tiltAngle.setLength(qMax(diameter*0.5, 50.0) * (1 - info.tiltElevation(info, 60.0, 60.0, true)));
         tiltAngle.setAngle((360.0 - fmod(KisPaintInformation::tiltDirection(info, true) * 360.0 + 270.0, 360.0))-3.0);
         tiltLine.moveTo(tiltAngle.p1());
         tiltLine.lineTo(tiltAngle.p2());
