--- plugins/menu.c.old	2007-07-31 02:38:54.000000000 +0300
+++ plugins/menu.c	2008-01-16 16:20:56.000000000 +0200
@@ -92,8 +92,8 @@
     items = gtk_container_get_children(GTK_CONTAINER(menu_shell));
     for(i=0; items; items=items->next, i++)  {
         cmpname = (gchar *)g_object_get_data(G_OBJECT(items->data), "item-name");
-        if(cmpname && g_ascii_strcasecmp(name, cmpname) < 0)
-            break;
+        if(cmpname && g_ascii_strcasecmp(name, cmpname) == 0)
+	     return -1;
     }
     gtk_menu_shell_insert(menu_shell, mi, i);
     return i;
