diff --git a/include/VideoFileSource.h b/include/VideoFileSource.h
index 0fda14e..8c15178 100644
--- a/include/VideoFileSource.h
+++ b/include/VideoFileSource.h
@@ -50,6 +50,8 @@
 #if defined(HAS_OPENCV2_HEADERS) || defined(OPENCV2_HEADERS)
 #include <opencv2/core/core_c.h>
 #include <opencv2/highgui/highgui_c.h>
+#include <opencv2/videoio.hpp>
+#include <opencv2/videoio/videoio_c.h>
 #else
 #include <cv.h>
 #include <highgui.h>
diff --git a/include/WebcamSource.h b/include/WebcamSource.h
index 8c1911a..9194385 100644
--- a/include/WebcamSource.h
+++ b/include/WebcamSource.h
@@ -51,6 +51,8 @@
 #if defined(HAS_OPENCV2_HEADERS) || defined(OPENCV2_HEADERS)
 #include <opencv2/core/core_c.h>
 #include <opencv2/highgui/highgui_c.h>
+#include <opencv2/videoio.hpp>
+#include <opencv2/videoio/videoio_c.h>
 #else
 #include <cv.h>
 #include <highgui.h>
diff --git a/zart.pro b/zart.pro
index d98a192..ddb2d3b 100644
--- a/zart.pro
+++ b/zart.pro
@@ -19,7 +19,7 @@ greaterThan(QT_MAJOR_VERSION, 4): CONFIG += c++11
 CONFIG	+= warn_on
 QT_CONFIG -= no-pkg-config
 CONFIG += link_pkgconfig
-PKGCONFIG += opencv fftw3 zlib
+PKGCONFIG += opencv4 fftw3 zlib
 # LIBS += -lfftw3_threads
 DEFINES += cimg_use_fftw3 cimg_use_zlib
 
