--- lib/CPAN/FirstTime.pm.orig	2007-12-24 23:21:42.000000000 -0500
+++ lib/CPAN/FirstTime.pm	2007-12-24 23:26:04.000000000 -0500
@@ -940,7 +940,7 @@
     }
 
     if (!$matcher or 'makepl_arg make_arg' =~ /$matcher/) {
-        my_dflt_prompt(makepl_arg => "", $matcher);
+        my_dflt_prompt(makepl_arg => "INSTALLDIRS=site", $matcher);
         my_dflt_prompt(make_arg => "", $matcher);
     }
 
