diff -urN a/asmcomp/CSEgen.ml b/asmcomp/CSEgen.ml
--- a/asmcomp/CSEgen.ml	2014-10-09 03:21:33.000000000 -0600
+++ b/asmcomp/CSEgen.ml	2014-12-06 14:17:49.879574784 -0700
@@ -216,7 +216,8 @@
 (* Perform CSE on the given instruction [i] and its successors.
    [n] is the value numbering current at the beginning of [i]. *)
 
-method private cse n i =
+method private cse n i = i
+(*
   match i.desc with
   | Iend | Ireturn | Iop(Itailcall_ind) | Iop(Itailcall_imm _)
   | Iexit _ | Iraise _ ->
@@ -315,6 +316,7 @@
       {i with desc = Itrywith(self#cse n body,
                               self#cse empty_numbering handler);
               next = self#cse empty_numbering i.next}
+*)
 
 method fundecl f =
   {f with fun_body = self#cse empty_numbering f.fun_body}
