diff -Naur mozilla-old/calendar/sunbird/app/mozilla.in mozilla/calendar/sunbird/app/mozilla.in
--- mozilla-old/calendar/sunbird/app/mozilla.in	2008-01-11 13:20:02.000000000 +1000
+++ mozilla/calendar/sunbird/app/mozilla.in	2008-09-24 20:37:12.000000000 +1000
@@ -250,14 +250,21 @@
 
 export MRE_HOME
 
+## Get system language
+SHORTLOCALE=$(echo $LANG | sed "s|_\([^.]*\).*||g")
+LONGLOCALE=$(echo $LANG | sed "s|_\([^.]*\).*|-\1|g")
+LOCALEPARAM=""
+[ -f ${moz_libdir}/chrome/${SHORTLOCALE}.jar ] && LOCALEPARAM="-UILocale ${SHORTLOCALE}"
+[ -f ${moz_libdir}/chrome/${LONGLOCALE}.jar ] && LOCALEPARAM="-UILocale ${LONGLOCALE}"
+
 ## Start addon scripts
 moz_pis_startstop_scripts "start"
 
 if [ $debugging = 1 ]
 then
-  echo $dist_bin/run-mozilla.sh $script_args $dist_bin/$MOZILLA_BIN "$@"
+  echo $dist_bin/run-mozilla.sh $script_args $dist_bin/$MOZILLA_BIN "$@" ${LOCALEPARAM}
 fi
-"$dist_bin/run-mozilla.sh" $script_args "$dist_bin/$MOZILLA_BIN" "$@"
+"$dist_bin/run-mozilla.sh" $script_args "$dist_bin/$MOZILLA_BIN" "$@" ${LOCALEPARAM}
 exitcode=$?
 
 ## Stop addon scripts
