diff -up mozilla/toolkit/xre/nsAppRunner.cpp.old mozilla/toolkit/xre/nsAppRunner.cpp
--- mozilla/toolkit/xre/nsAppRunner.cpp.old	2008-12-10 20:07:01.000000000 +0100
+++ mozilla/toolkit/xre/nsAppRunner.cpp	2009-02-27 17:17:59.000000000 +0100
@@ -1475,7 +1475,7 @@ XRE_GetBinaryPath(const char* argv0, nsI
   if (NS_FAILED(rv))
     return rv;
 
-#elif
+#else
 #error Oops, you need platform-specific code here
 #endif
 
