--- a/par2creatorsourcefile.cpp
+++ b/par2creatorsourcefile.cpp
@@ -319,7 +319,7 @@ void Par2CreatorSourceFile::UpdateHashes(u32 blocknumber, const void *buffer, si
 
 
   // Update the full file hash, but don't go beyond the end of the file
-  if (length > filesize - blocknumber * length)
+  if ((u64)length > filesize - blocknumber * (u64)length)
   {
     length = (size_t)(filesize - blocknumber * (u64)length);
   }
