diff -urN vnc_unixsrc/vncserver vnc_unixsrc.fixed/vncserver
--- vnc_unixsrc/vncserver	2009-02-11 23:27:18.000000000 -0500
+++ vnc_unixsrc.fixed/vncserver	2009-04-23 21:46:46.000000000 -0400
@@ -34,7 +34,8 @@
 $desktopName = "X";
 $vncClasses = "/usr/local/vnc/classes";
 $vncUserDir = "$ENV{HOME}/.vnc";
-$fontPath = "unix/:7100";
+$fontPath = "/usr/share/fonts/misc/,/usr/share/fonts/75dpi/,/usr/share/fonts/100dpi/,/usr/share/fonts/Type1/";
+$colorPath = "/usr/share/X11/rgb";
 $authType = "-rfbauth $vncUserDir/passwd";
 
 # Read configuration from the system-wide and user files if present.
