diff -ru vnc_unixsrc.orig//Xvnc/configure vnc_unixsrc/Xvnc/configure
--- vnc_unixsrc.orig//Xvnc/configure	2010-12-01 01:24:05.000000000 +0000
+++ vnc_unixsrc/Xvnc/configure	2010-12-01 01:20:19.000000000 +0000
@@ -39,6 +39,7 @@
 echo "#define OSMinorVersion $MINOR" >> config/cf/platform.def
 echo "#define OSTeenyVersion $TEENY" >> config/cf/platform.def
 echo "#define LinuxCLibMajorVersion 6" >> config/cf/platform.def
+echo "#define ArmArchitecture 1" >> config/cf/platform.def
 
 if [ "$SYSNAME" = AIX ]
 then
