From f1c2966fe1b01b98f405b8bf9aeebec552bbfec3 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Wed, 3 Feb 2016 19:29:25 -0700
Subject: [PATCH 1/4] remove cross compiling

---
 Makefile | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/Makefile b/Makefile
index 1da09c2..2968eb4 100644
--- a/Makefile
+++ b/Makefile
@@ -244,12 +244,6 @@ export	HOSTARCH HOSTOS
 # set default to nothing for native builds
 ifeq ($(HOSTARCH),$(ARCH))
 CROSS_COMPILE ?=
-else
-ifeq ($(CROSS_COMPILE),)
-CROSS_COMPILE = /opt/toolchains/gcc-linaro-aarch64-none-elf-4.9-2014.09_linux/bin/aarch64-none-elf-
-export CROSS_COMPILE
-endif
-
 endif
 
 KCONFIG_CONFIG	?= .config
-- 
2.7.1

