diff -upr openvdb-5.1.0.orig/openvdb/CMakeLists.txt openvdb-5.1.0/openvdb/CMakeLists.txt
--- openvdb-5.1.0.orig/openvdb/CMakeLists.txt	2018-06-03 02:59:00.521389220 +0300
+++ openvdb-5.1.0/openvdb/CMakeLists.txt	2018-06-03 03:07:55.094264747 +0300
@@ -275,6 +275,7 @@ IF (NOT WIN32)
   TARGET_LINK_LIBRARIES ( vdb_view
     openvdb_shared
     ${Boost_THREAD_LIBRARY}
+    ${CMAKE_THREAD_LIBS_INIT}
     ${OPENGL_gl_LIBRARY}
     ${OPENGL_glu_LIBRARY}
     ${COCOA_LIBRARY}
