diff -urN a/configure.py b/configure.py
--- a/configure.py	2014-03-14 08:38:18.000000000 -0600
+++ b/configure.py	2014-03-28 14:55:32.379777351 -0600
@@ -492,11 +492,8 @@
 #endif
 
 #if QT_VERSION < 0x050200
-// This is the test used in qglobal.h in Qt prior to v5.2.  In v5.2 and later
-// qreal is always double.
-#if defined(QT_NO_FPU) || defined(Q_PROCESSOR_ARM) || defined(Q_OS_WINCE)
-    out << "PyQt_qreal_double\\n";
-#endif
+    if (sizeof (qreal) != sizeof (double))
+        out << "PyQt_qreal_double\\n";
 #endif
 
     return 0;
