Description: check if qreal is double based on sizeof
Author: Dmitry Shachnev <mitya57@gmail.com>
Forwarded: no
Last-Update: 2014-01-29

--- a/configure.py
+++ b/configure.py
@@ -481,10 +481,8 @@
     out << "PyQt_Desktop_OpenGL\\n";
 #endif
 
-// This is the test used in qglobal.h.
-#if defined(QT_NO_FPU) || defined(Q_PROCESSOR_ARM) || defined(Q_OS_WINCE)
-    out << "PyQt_qreal_double\\n";
-#endif
+    if (sizeof (qreal) != sizeof (double))
+        out << "PyQt_qreal_double\\n";
 
     return 0;
 }

