--- PyQt5_gpl-5.7.1/configure.py.orig	2017-01-23 20:42:47.791266791 +0100
+++ PyQt5_gpl-5.7.1/configure.py	2017-01-23 20:43:02.754828184 +0100
@@ -723,7 +723,7 @@
         lines = f.read().split('\n')
         f.close()
 
-        self.qt_licensee = lines[0]
+        self.qt_licensee = 'Open Source'
         self.qt_shared = (lines[1] == 'shared')
         self.pyqt_disabled_features = lines[2:-1]
 
