diff -aur irqbalance-1.5.0/misc/irqbalance.service irqbalance-1.5.0.new/misc/irqbalance.service
--- irqbalance-1.5.0/misc/irqbalance.service	2018-10-27 11:36:59.000000000 +0200
+++ irqbalance-1.5.0.new/misc/irqbalance.service	2019-02-19 21:49:12.093453013 +0100
@@ -5,8 +5,8 @@
 ConditionVirtualization=!container
 
 [Service]
-EnvironmentFile=/path/to/irqbalance.env
-ExecStart=/usr/sbin/irqbalance --foreground $IRQBALANCE_ARGS
+EnvironmentFile=/etc/irqbalance.env
+ExecStart=/usr/bin/irqbalance --foreground $IRQBALANCE_ARGS
 CapabilityBoundingSet=
 NoNewPrivileges=yes
 ReadOnlyPaths=/
