diff --git a/configure.ac b/configure.ac
index 38a1bee..5e9cc03 100644
--- a/configure.ac
+++ b/configure.ac
@@ -53,9 +53,6 @@ AC_ARG_WITH([systemd],
 AS_IF(
   [test "x$with_systemd" = xyes], [
     PKG_CHECK_MODULES([SYSTEMD], [libsystemd], [journal_lib=yes], [journal_lib=no])
-    AS_IF([test "x$journal_lib" != "xyes"], [
-      PKG_CHECK_MODULES([SYSTEMD], [libsystemd-journal], [journal_lib=yes])
-    ])
     AC_DEFINE(HAVE_LIBSYSTEMD, 1, [systemd support])
     AC_CHECK_LIB([systemd], [sd_journal_print_with_location])
     AC_CHECK_LIB([systemd], [sd_journal_print])
diff --git a/misc/irqbalance.service b/misc/irqbalance.service
index e9a5527..a8451d9 100644
--- a/misc/irqbalance.service
+++ b/misc/irqbalance.service
@@ -3,8 +3,8 @@ Description=irqbalance daemon
 ConditionVirtualization=!container
 
 [Service]
-EnvironmentFile=/path/to/irqbalance.env
-ExecStart=/usr/sbin/irqbalance --foreground $IRQBALANCE_ARGS
+EnvironmentFile=/etc/irqbalance.env
+ExecStart=/usr/bin/irqbalance --foreground $IRQBALANCE_ARGS
 
 [Install]
 WantedBy=multi-user.target
