diff --git a/procinterrupts.c b/procinterrupts.c
index 63101d6..b70bfc2 100644
--- a/procinterrupts.c
+++ b/procinterrupts.c
@@ -80,7 +80,7 @@ static int check_platform_device(char *name, struct irq_info *info)
 			rc = 0;
 			goto out;
 		} else if (!strncmp(ent->d_name, "net", strlen("net"))) {
-			info->IRQ_TYPE_LEGACY;
+			info->type = IRQ_TYPE_LEGACY;
 			info->class = IRQ_ETH;
 			rc = 0;
 			goto out;
@@ -94,7 +94,7 @@ static int check_platform_device(char *name, struct irq_info *info)
 
 out:
 	closedir(dirfd);
-	log(TO_ALL, LOG_DEBUG, "IRQ %s is of type %d and class %d\n", name, info->type, info->class)
+	log(TO_ALL, LOG_DEBUG, "IRQ %s is of type %d and class %d\n", name, info->type, info->class);
 	return rc;
 
 }
