From d17076c50ab71a73443bbabbc24d06064e7c7572 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sat, 19 Aug 2017 15:28:38 -0600
Subject: [PATCH] ATF: set -fno-stack-protector

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 87c307de..acc5791b 100644
--- a/Makefile
+++ b/Makefile
@@ -231,7 +231,7 @@ CFLAGS			+= 	-nostdinc -pedantic -ffreestanding -Wall	\
 				-Werror -Wmissing-include-dirs			\
 				-mgeneral-regs-only -std=c99 -c -Os		\
 				${DEFINES} ${INCLUDES}
-CFLAGS			+=	-ffunction-sections -fdata-sections
+CFLAGS			+=	-ffunction-sections -fdata-sections -fno-stack-protector
 LDFLAGS			+=	--fatal-warnings -O1
 LDFLAGS			+=	--gc-sections
 LDFLAGS			+=	$(if $(shell $(LD) -v --fix-cortex-a53-843419 > /dev/null 2>&1 && echo 1),--fix-cortex-a53-843419)
-- 
2.14.1

