diff -ru giac-2.0.0.orig/src/optimization.cc giac-2.0.0/src/optimization.cc
--- giac-2.0.0.orig/src/optimization.cc	2026-09-28 14:12:40.222993300 +0200
+++ giac-2.0.0/src/optimization.cc	2026-09-28 14:13:39.917307773 +0200
@@ -1271,7 +1271,7 @@
     vector<ulong> sets(comb(n,m).val);
     int i=0;
     for (ulong k=1;k<N;++k) {
-        bitset<32> b(k);
+        std::bitset<32> b(k);
         if (b.count()==(size_t)m)
             sets[i++]=k;
     }
