diff -ruN a/auxiliary_files/mumble-server.ini b/auxiliary_files/mumble-server.ini
--- a/auxiliary_files/mumble-server.ini	2023-01-18 07:54:18.355957000 +0100
+++ b/auxiliary_files/mumble-server.ini	2023-01-22 18:09:47.905232642 +0100
@@ -13,7 +13,7 @@
 
 ; Path to database. If blank, will search for
 ; mumble-server.sqlite in default locations or create it if not found.
-database=
+database=/var/lib/mumble-server/mumble-server.sqlite
 
 ; The server defaults to using SQLite with its default rollback journal.
 ; In some situations, using SQLite's write-ahead log (WAL) can be
@@ -83,7 +83,7 @@
 ; logs to the file 'mumble-server.log'. If you leave this field blank
 ; on Unix-like systems, the server will force itself into foreground
 ; mode which logs to the console.
-;logfile=mumble-server.log
+logfile=
 
 ; If set, the server will write its process ID to this file
 ; when running in daemon mode (when the -fg flag is not
