--- sapi/apache2handler/config.m4.orig
+++ sapi/apache2handler/config.m4
@@ -122,7 +122,6 @@
       PHP_BUILD_THREAD_SAFE
     fi
   else
-    APACHE_THREADED_MPM=`$APXS_HTTPD -V | grep 'threaded:.*yes'`
     if test -n "$APACHE_THREADED_MPM"; then
       PHP_BUILD_THREAD_SAFE
     fi
--- configure.orig
+++ configure
@@ -6919,7 +6919,6 @@
 
     fi
   else
-    APACHE_THREADED_MPM=`$APXS_HTTPD -V | grep 'threaded:.*yes'`
     if test -n "$APACHE_THREADED_MPM"; then
 
   enable_maintainer_zts=yes
