--- configure.orig	2008-07-03 10:27:06.000000000 +0000
+++ configure	2008-07-03 10:30:56.000000000 +0000
@@ -28399,43 +28399,15 @@
   
   unset THIS_INCLUDE THIS_LIBS THIS_LFLAGS THIS_PREFIX THIS_RESULT
 
-  dbdp="/usr/local/BerkeleyDB.4."
-  for i in $PHP_DB4 ${dbdp}6 ${dbdp}5 ${dbdp}4 ${dbdp}3  ${dbdp}2 ${dbdp}1 ${dbdp}0 /usr/local /usr; do
-    if test -f "$i/db4/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/db4/db.h
-      break
-    elif test -f "$i/include/db4.6/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db4.6/db.h
-      break
-    elif test -f "$i/include/db4.5/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db4.5/db.h
-      break
-    elif test -f "$i/include/db4/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db4/db.h
-      break
-    elif test -f "$i/include/db/db4.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db/db4.h
-      break
-    elif test -f "$i/include/db4.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db4.h
-      break
-    elif test -f "$i/include/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db.h
-      break
-    fi
-  done
+  if test -f "/usr/include/db.h"; then
+    THIS_PREFIX=/usr
+    THIS_INCLUDE=/usr/include/db.h
+  fi
   
   if test -z "$THIS_INCLUDE"; then
     { echo "configure: error: DBA: Could not find necessary header file(s)." 1>&2; exit 1; }
   fi
-  for LIB in db-4.6 db-4.5 db-4.4 db-4.3 db-4.2 db-4.1 db-4.0 db-4 db4 db; do
+  for LIB in db; do
     if test -f $THIS_PREFIX/$PHP_LIBDIR/lib$LIB.a || test -f $THIS_PREFIX/$PHP_LIBDIR/lib$LIB.$SHLIB_SUFFIX_NAME; then
       lib_found="";
       
