--- aaa/ext/dba/config.m4	2022-12-11 16:33:02.721079548 +0100
+++ bbb/ext/dba/config.m4	2022-12-11 16:33:59.821200682 +0100
@@ -308,7 +308,7 @@
         ]])],[
           AC_EGREP_CPP(yes,[
 #include "$THIS_INCLUDE"
-#if DB_VERSION_MAJOR == $1 || ($1 == 4 && DB_VERSION_MAJOR == 5)
+#if DB_VERSION_MAJOR == $1 || ($1 == 4 && DB_VERSION_MAJOR == 5) || ($1 == 4 && DB_VERSION_MAJOR == 6)
             yes
 #endif
           ],[ 
