Index: ext/gd/libgd/gd_compat.c
===================================================================
--- ext/gd/libgd/gd_compat.c	(Revision 286947)
+++ ext/gd/libgd/gd_compat.c	(Revision 286948)
@@ -14,7 +14,7 @@
 	return JPEG_LIB_VERSION;
 }
 
-int gdJpegGetVersionString()
+const char * gdJpegGetVersionString()
 {
 	switch(JPEG_LIB_VERSION) {
 		case 62:
Index: ext/gd/libgd/gd_compat.h
===================================================================
--- ext/gd/libgd/gd_compat.h	(Revision 286947)
+++ ext/gd/libgd/gd_compat.h	(Revision 286948)
@@ -8,7 +8,7 @@
 #endif
 
 const char * gdPngGetVersionString();
-int gdJpegGetVersionString();
+const char * gdJpegGetVersionString();
 int gdJpegGetVersionInt();
 int overflow2(int a, int b);
 
