--- a/adm/cmake/ffmpeg.cmake
+++ b/adm/cmake/ffmpeg.cmake
@@ -93,12 +93,12 @@ if (NOT 3RDPARTY_FFMPEG_INCLUDE_DIR OR NOT EXISTS "${3RDPARTY_FFMPEG_INCLUDE_DIR
   if (3RDPARTY_FFMPEG_DIR AND EXISTS "${3RDPARTY_FFMPEG_DIR}")
     find_path (3RDPARTY_FFMPEG_INCLUDE_DIR NAMES ${HEADER_NAMES}
                                                  PATHS ${3RDPARTY_FFMPEG_DIR}
-                                                 PATH_SUFFIXES include
+                                                 PATH_SUFFIXES include include/ffmpeg4.4
                                                  CMAKE_FIND_ROOT_PATH_BOTH
                                                  NO_DEFAULT_PATH)
   else()
     find_path (3RDPARTY_FFMPEG_INCLUDE_DIR NAMES ${HEADER_NAMES}
-                                                 PATH_SUFFIXES include
+                                                 PATH_SUFFIXES include include/ffmpeg4.4
                                                  CMAKE_FIND_ROOT_PATH_BOTH)
   endif()
 endif()
@@ -116,7 +116,7 @@ foreach (LIBRARY_NAME ${CSF_FFmpeg})
   if (NOT 3RDPARTY_FFMPEG_LIBRARY_${LIBRARY_NAME} OR NOT EXISTS "${3RDPARTY_FFMPEG_LIBRARY_${LIBRARY_NAME}}")
     set (CMAKE_FIND_LIBRARY_SUFFIXES .lib .so .dylib .a)
 
-    set (FFMPEG_PATH_SUFFIXES lib)
+    set (FFMPEG_PATH_SUFFIXES lib lib/ffmpeg4.4)
     if (ANDROID)
       set (FFMPEG_PATH_SUFFIXES ${FFMPEG_PATH_SUFFIXES} libs/${ANDROID_ABI})
     elseif(APPLE)
