diff --git a/src/internal/runtime/gc/scan/scan_amd64.go b/src/internal/runtime/gc/scan/scan_amd64.go
index 0151804fa5..c378c7a7ee 100644
--- a/src/internal/runtime/gc/scan/scan_amd64.go
+++ b/src/internal/runtime/gc/scan/scan_amd64.go
@@ -38,4 +38,5 @@ var avx512ScanPackedReqsMet = cpu.X86.HasAVX512VL &&
 	cpu.X86.HasAVX512BW &&
 	cpu.X86.HasGFNI &&
 	cpu.X86.HasAVX512BITALG &&
-	cpu.X86.HasAVX512VBMI
+	cpu.X86.HasAVX512VBMI &&
+	cpu.X86.HasPOPCNT
