--- src/man_db.conf.in	2020-01-27 13:25:03.000000000 +0100
+++ src/man_db.conf.in.new	2020-02-26 07:40:32.726347889 +0100
@@ -69,7 +69,8 @@
 MANDB_MAP	/usr/local/share/man	/var/cache/man/local
 MANDB_MAP	/usr/X11R6/man		/var/cache/man/X11R6
 MANDB_MAP	/opt/man		/var/cache/man/opt
-MANDB_MAP	/snap/man		/var/cache/man/snap
+MANDB_MAP	/var/lib/snapd/snap/man	/var/cache/man/snap
+
 #
 #---------------------------------------------------------
 # Program definitions.  These are commented out by default as the value
