diff -Nur dhcp-3.0.6.orig/client/scripts/linux dhcp-3.0.6/client/scripts/linux
--- dhcp-3.0.6.orig/client/scripts/linux	2008-03-29 18:44:07.000000000 +0100
+++ dhcp-3.0.6/client/scripts/linux	2008-03-29 18:45:06.000000000 +0100
@@ -36,8 +36,9 @@
 fi;
 
 make_resolv_conf() {
-  if [ "x$new_domain_name" != x ] && [ x"$new_domain_name_servers" != x ]; then
-    echo search $new_domain_name >/etc/resolv.conf
+  if [ -n "$new_domain_name_servers" ]; then
+    : > /etc/resolv.conf
+    [ -n "$new_domain_name" ] && echo search $new_domain_name >/etc/resolv.conf
     chmod 644 /etc/resolv.conf
     for nameserver in $new_domain_name_servers; do
       echo nameserver $nameserver >>/etc/resolv.conf
