From 6abfd7104202121d3e23ddad47e377dfaef22b10 Mon Sep 17 00:00:00 2001
From: BlackEagle <ike.devolder@gmail.com>
Date: Sun, 25 May 2014 12:10:01 +0200
Subject: [PATCH] make sure applications/xbmc.desktop does'nt have executable
 flag

Signed-off-by: BlackEagle <ike.devolder@gmail.com>
---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 754f153..bee1fdd 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -646,7 +646,7 @@ endif
 endif
 	@# Icons and links
 	@install -d $(DESTDIR)$(datarootdir)/applications
-	@install tools/Linux/xbmc.desktop $(DESTDIR)$(datarootdir)/applications/xbmc.desktop
+	@install -m 0644 tools/Linux/xbmc.desktop $(DESTDIR)$(datarootdir)/applications/xbmc.desktop
 	@install -d $(DESTDIR)$(datadir)/icons/hicolor/48x48/apps
 	@install -m 0644 media/icon48x48.png $(DESTDIR)$(datadir)/icons/hicolor/48x48/apps/xbmc.png
 	@install -d $(DESTDIR)$(datadir)/icons/hicolor/256x256/apps
-- 
1.9.1

