diff -rupN a/configure.in b/configure.in
--- a/configure.in	2014-05-03 21:03:25.000000000 +0000
+++ b/configure.in	2014-05-07 09:56:30.039177391 +0000
@@ -600,7 +600,7 @@ PKG_PROG_PKG_CONFIG
 MAKE="${MAKE:-make}"
 OBJDUMP="${OBJDUMP:-objdump}"
 
-use_external_ffmpeg=no
+use_external_ffmpeg=yes
 use_static_ffmpeg=no
 
 # ffmpeg needs the output of uname -s (e.x. linux, darwin) for the target_os
@@ -632,7 +632,7 @@ case $host in
         use_arch="x86"
         use_cpu="i686"
      fi
-     use_static_ffmpeg=yes
+     use_static_ffmpeg=no
      ;;
   x86_64-*-linux-gnu*|x86_64-*-linux-uclibc*)
      ARCH="x86_64-linux"
@@ -640,7 +640,7 @@ case $host in
         use_arch="x86_64"
         use_cpu="x86_64"
      fi
-     use_static_ffmpeg=yes
+     use_static_ffmpeg=no
      ;;
   i386-*-freebsd*)
      ARCH="x86-freebsd"
@@ -696,7 +696,7 @@ case $host in
      use_sdl=no
      use_x11=no
      use_wayland=no
-     use_static_ffmpeg=yes
+     use_static_ffmpeg=no
      ;;
   arm*-*linux-android*)
      target_platform=target_android
diff -rupN a/lib/DllAvCodec.h b/lib/DllAvCodec.h
--- a/lib/DllAvCodec.h	2014-05-03 21:03:25.000000000 +0000
+++ b/lib/DllAvCodec.h	2014-05-07 10:10:06.298165915 +0000
@@ -43,6 +43,7 @@ extern "C" {
 
 #if (defined USE_EXTERNAL_FFMPEG)
   #include <libavcodec/avcodec.h>
+  #include <libavutil/frame.h>
 #else
   #include "libavcodec/avcodec.h"
 #endif
diff -rupN a/lib/DllAvUtil.h b/lib/DllAvUtil.h
--- a/lib/DllAvUtil.h	2014-05-03 21:03:25.000000000 +0000
+++ b/lib/DllAvUtil.h	2014-05-07 10:17:31.415798448 +0000
@@ -31,33 +31,19 @@
 #endif
 
 extern "C" {
-#if (defined USE_EXTERNAL_FFMPEG)
-  #include <libavutil/avutil.h>
-  // for av_get_default_channel_layout
-  #include <libavutil/audioconvert.h>
-  #include <libavutil/crc.h>
-  #include <libavutil/fifo.h>
-  // for LIBAVCODEC_VERSION_INT:
-  #include <libavcodec/avcodec.h>
-  // for enum AVSampleFormat
-  #include <libavutil/samplefmt.h>
-  #include <libavutil/opt.h>
-  #include <libavutil/mem.h>
-  #include <libavutil/mathematics.h>
-  #if (defined USE_LIBAV_HACKS)
-    #include "xbmc-libav-hacks/libav_hacks.h"
-  #endif
-#else
-  #include "libavutil/avutil.h"
-  //for av_get_default_channel_layout
-  #include "libavutil/audioconvert.h"
-  #include "libavutil/crc.h"
-  #include "libavutil/opt.h"
-  #include "libavutil/mem.h"
-  #include "libavutil/fifo.h"
-  // for enum AVSampleFormat
-  #include "libavutil/samplefmt.h"
-#endif
+#include <libavutil/avutil.h>
+#include <libavutil/frame.h>
+// for av_get_default_channel_layout
+#include <libavutil/audioconvert.h>
+#include <libavutil/crc.h>
+#include <libavutil/fifo.h>
+// for LIBAVCODEC_VERSION_INT:
+#include <libavcodec/avcodec.h>
+// for enum AVSampleFormat
+#include <libavutil/samplefmt.h>
+#include <libavutil/opt.h>
+#include <libavutil/mem.h>
+#include <libavutil/mathematics.h>
 }
 
 #if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(52,29,100)
