diff --git a/xbmc/filesystem/NFSDirectory.cpp b/xbmc/filesystem/NFSDirectory.cpp
index 85084a5..8efa7bf 100644
--- a/xbmc/filesystem/NFSDirectory.cpp
+++ b/xbmc/filesystem/NFSDirectory.cpp
@@ -37,6 +37,7 @@
 using namespace XFILE;
 using namespace std;
 #include <limits.h>
+#include <nfsc/libnfs-zdr.h>
 #include <nfsc/libnfs-raw-mount.h>
 #include <nfsc/libnfs-raw-nfs.h>
 
diff --git a/xbmc/filesystem/NFSFile.cpp b/xbmc/filesystem/NFSFile.cpp
index 479f3fe..2aeea32 100644
--- a/xbmc/filesystem/NFSFile.cpp
+++ b/xbmc/filesystem/NFSFile.cpp
@@ -32,6 +32,7 @@
 #include "network/DNSNameCache.h"
 #include "threads/SystemClock.h"
 
+#include <nfsc/libnfs-zdr.h>
 #include <nfsc/libnfs-raw-mount.h>
 
 #ifdef TARGET_WINDOWS