Index: xbmc/Application.cpp
===================================================================
--- xbmc/Application.cpp	(revision 20059)
+++ xbmc/Application.cpp	(working copy)
@@ -873,6 +873,8 @@
 #endif
 
   // set GUI res and force the clear of the screen
+  if (g_guiSettings.GetBool("videoscreen.safefull"))
+    g_guiSettings.m_LookAndFeelResolution = AUTORES;
   g_graphicsContext.SetVideoResolution(g_guiSettings.m_LookAndFeelResolution, TRUE, true);
 
 #ifdef _WIN32PC
@@ -949,6 +951,9 @@
   time(&seconds);
   srand((unsigned int)seconds);
 
+  if (g_guiSettings.GetBool("videoscreen.safefull"))
+    g_graphicsContext.ToggleFullScreenRoot();
+
   return CXBApplicationEx::Create(hWnd);
 }
 
