diff -wbBur XBMC/tools/XBMCTex/XBMCTex.cpp XBMC.my/tools/XBMCTex/XBMCTex.cpp
--- XBMC/tools/XBMCTex/XBMCTex.cpp	2009-05-29 10:26:09.000000000 +0000
+++ XBMC.my/tools/XBMCTex/XBMCTex.cpp	2009-05-29 14:35:02.000000000 +0000
@@ -659,7 +659,7 @@
 			valid = true;
 #ifdef _LINUX
       char *c = NULL;
-      while ((c = strchr(OutputFilename, '\\')) != NULL) *c = '/';
+      while ((c = (char*)strchr(OutputFilename, '\\')) != NULL) *c = '/';
 #endif
 		}
     else if (!stricmp(args[i], "-noprotect") || !stricmp(args[i], "-p"))
diff -wbBur XBMC/xbmc/cores/DllLoader/exports/exports_msvcrt.cpp XBMC.my/xbmc/cores/DllLoader/exports/exports_msvcrt.cpp
--- XBMC/xbmc/cores/DllLoader/exports/exports_msvcrt.cpp	2009-05-29 09:50:32.000000000 +0000
+++ XBMC.my/xbmc/cores/DllLoader/exports/exports_msvcrt.cpp	2009-05-26 11:39:15.000000000 +0000
@@ -372,12 +372,12 @@
   { "sprintf",                    -1, (void*)sprintf,                       NULL },
   { "sqrt",                       -1, (void*)sqrt,                          NULL },
   { "sscanf",                     -1, (void*)sscanf,                        NULL },
-  { "strchr",                     -1, (void*)::strchr,                      NULL },
+  { "strchr",                     -1, (void*)(char* (*)(char*,int))strchr,                        NULL },
   { "strcmp",                     -1, (void*)strcmp,                        NULL },
   { "strcpy",                     -1, (void*)strcpy,                        NULL },
   { "strlen",                     -1, (void*)strlen,                        NULL },
   { "strncpy",                    -1, (void*)strncpy,                       NULL },
-  { "strrchr",                    -1, (void*)::strrchr,                     NULL },
+  { "strrchr",                    -1, (void*)(char* (*)(char*,int))strrchr,                       NULL },
   { "strtod",                     -1, (void*)strtod,                        NULL },
   { "strtok",                     -1, (void*)strtok,                        NULL },
   { "strtol",                     -1, (void*)strtol,                        NULL },
@@ -385,7 +385,7 @@
   { "tan",                        -1, (void*)tan,                           NULL },
   { "tanh",                       -1, (void*)tanh,                          NULL },
   { "time",                       -1, (void*)time,                          NULL },
-  { "toupper",                    -1, (void*)::toupper,                     NULL },
+  { "toupper",                    -1, (void*)toupper,                     NULL },
 #ifndef _LINUX
   { "_memccpy",                   -1, (void*)_memccpy,                      NULL },
 #endif
@@ -423,7 +423,7 @@
   { "getc",                       -1, (void*)dll_getc,                      NULL },
   { "getenv",                     -1, (void*)dll_getenv,                    NULL },
   { "rand",                       -1, (void*)rand,                          NULL },
-  { "remove",                     -1, (void*)::remove,                      NULL },
+  { "remove",                     -1, (void*)remove,                      NULL },
   { "rewind",                     -1, (void*)dll_rewind,                    NULL },
   { "setlocale",                  -1, (void*)setlocale,                     NULL },
   { "signal",                     -1, (void*)dll_signal,                    NULL },
@@ -433,12 +433,12 @@
   { "strerror",                   -1, (void*)dllstrerror,                   NULL },
   { "strncat",                    -1, (void*)strncat,                       NULL },
   { "strncmp",                    -1, (void*)strncmp,                       NULL },
-  { "strpbrk",                    -1, (void*)::strpbrk,                     NULL },
-  { "strstr",                     -1, (void*)::strstr,                      NULL },
-  { "tolower",                    -1, (void*)::tolower,                     NULL },
+  { "strpbrk",                    -1, (void*)(char* (*)(char*,const char*))::strpbrk,                     NULL },
+  { "strstr",                     -1, (void*)(char* (*)(char*,const char*))::strstr,                      NULL },
+  { "tolower",                    -1, (void*)tolower,                     NULL },
   { "acos",                       -1, (void*)acos,                          NULL },
   { "atan",                       -1, (void*)atan,                          NULL },
-  { "memchr",                     -1, (void*)::memchr,                      NULL },
+  { "memchr",                     -1, (void*)(void*(*)(void*,int,size_t))memchr,                      NULL },
   { "fgetc",                      -1, (void*)dll_getc,                      NULL },
 #ifndef _LINUX
   { "_CIpow",                     -1, (void*)_CIpow,                        NULL },
@@ -497,9 +497,9 @@
   { "_fdopen",                    -1, (void*)dll_fdopen,                    NULL },
   { "system",                     -1, (void*)dll_system,                    NULL },
   { "_flsbuf",                    -1, (void*)dll_flsbuf,                    NULL },
-  { "isdigit",                    -1, (void*)::isdigit,                     NULL },
-  { "isalnum",                    -1, (void*)::isalnum,                     NULL },
-  { "isxdigit",                   -1, (void*)::isxdigit,                    NULL },
+  { "isdigit",                    -1, (void*)isdigit,                     NULL },
+  { "isalnum",                    -1, (void*)isalnum,                     NULL },
+  { "isxdigit",                   -1, (void*)isxdigit,                    NULL },
   { "pow",                        -1, (void*)pow,                           NULL },
   { "_onexit",                    -1, (void*)dll_onexit,                    NULL },
   { "modf",                       -1, (void*)modf,                          NULL },
@@ -586,12 +586,12 @@
   { "sprintf",                    -1, (void*)sprintf,                       NULL },
   { "sqrt",                       -1, (void*)sqrt,                          NULL },
   { "sscanf",                     -1, (void*)sscanf,                        NULL },
-  { "strchr",                     -1, (void*)::strchr,                      NULL },
+  { "strchr",                     -1, (void*)(char* (*)(char*,int))strchr,                      NULL },
   { "strcmp",                     -1, (void*)strcmp,                        NULL },
   { "strcpy",                     -1, (void*)strcpy,                        NULL },
   { "strlen",                     -1, (void*)strlen,                        NULL },
   { "strncpy",                    -1, (void*)strncpy,                       NULL },
-  { "strrchr",                    -1, (void*)::strrchr,                     NULL },
+  { "strrchr",                    -1, (void*)(char* (*)(char*,int))strrchr,                     NULL },
   { "strtod",                     -1, (void*)strtod,                        NULL },
   { "strtok",                     -1, (void*)strtok,                        NULL },
   { "strtol",                     -1, (void*)strtol,                        NULL },
@@ -599,7 +599,7 @@
   { "tan",                        -1, (void*)tan,                           NULL },
   { "tanh",                       -1, (void*)tanh,                          NULL },
   { "time",                       -1, (void*)time,                          NULL },
-  { "toupper",                    -1, (void*)::toupper,                       NULL },
+  { "toupper",                    -1, (void*)toupper,                       NULL },
   { "_fstat",                     -1, (void*)dll_fstat,                     NULL },
   { "_mkdir",                     -1, (void*)dll_mkdir,                     NULL },
   { "_pclose",                    -1, (void*)dll_pclose,                    NULL },
@@ -637,7 +637,7 @@
   { "getc",                       -1, (void*)dll_getc,                      NULL },
   { "getenv",                     -1, (void*)dll_getenv,                    NULL },
   { "rand",                       -1, (void*)rand,                          NULL },
-  { "remove",                     -1, (void*)::remove,                      NULL },
+  { "remove",                     -1, (void*)remove,                      NULL },
   { "rewind",                     -1, (void*)dll_rewind,                    NULL },
   { "setlocale",                  -1, (void*)setlocale,                     NULL },
   { "signal",                     -1, (void*)dll_signal,                    NULL },
@@ -647,13 +647,13 @@
   { "strerror",                   -1, (void*)strerror,                      NULL },
   { "strncat",                    -1, (void*)strncat,                       NULL },
   { "strncmp",                    -1, (void*)strncmp,                       NULL },
-  { "strpbrk",                    -1, (void*)::strpbrk,                     NULL },
-  { "strstr",                     -1, (void*)::strstr,                        NULL },
-  { "tolower",                    -1, (void*)::tolower,                     NULL },
+  { "strpbrk",                    -1, (void*)(char* (*)(char*,const char*))strpbrk,                     NULL },
+  { "strstr",                     -1, (void*)(char* (*)(char*,const char*))strstr,                        NULL },
+  { "tolower",                    -1, (void*)tolower,                     NULL },
   { "acos",                       -1, (void*)acos,                          NULL },
   { "atan",                       -1, (void*)atan,                          NULL },
-  { "memchr",                     -1, (void*)::memchr,                      NULL },
-  { "isdigit",                    -1, (void*)::isdigit,                     NULL },
+  { "memchr",                     -1, (void*)(void* (*)(void*,int,size_t))memchr,                      NULL },
+  { "isdigit",                    -1, (void*)isdigit,                     NULL },
   { "_strcmpi",                   -1, (void*)strcmpi,                       NULL },
 #ifndef _LINUX
   { "_CIpow",                     -1, (void*)_CIpow,                        NULL },
@@ -664,7 +664,7 @@
   { "??2@YAPAXI@Z",               -1, (void*)dllmalloc,                     (void*)track_malloc },
   { "??3@YAXPAX@Z",               -1, (void*)dllfree,                       (void*)track_free },
 #ifndef _LINUX
-  { "??_U@YAPAXI@Z",              -1, (void*)(::operator new),              NULL },
+  { "??_U@YAPAXI@Z",              -1, (void*)(operator new),              NULL },
 #endif
   { "_beginthreadex",             -1, (void*)dll_beginthreadex,             NULL },
   { "_fdopen",                    -1, (void*)dll_fdopen,                    NULL },
@@ -702,7 +702,7 @@
 #endif
   { "fgetpos",                    -1, (void*)dll_fgetpos,                   NULL },
   { "??_V@YAXPAX@Z",              -1, (void*)dllfree,                       (void*)track_free},
-  { "isalpha",                    -1, (void*)::isalpha,                       NULL },
+  { "isalpha",                    -1, (void*)isalpha,                       NULL },
 #ifndef _LINUX
   { "_CIacos",                    -1, (void*)_CIacos,                       NULL },
   { "_CIasin",                    -1, (void*)_CIasin,                       NULL },
@@ -718,9 +718,9 @@
 #endif
   { "memcmp",                     -1, (void*)memcmp,                        NULL },
   { "fsetpos",                    -1, (void*)dll_fsetpos,                   NULL },
-  { "isprint",                    -1, (void*)::isprint,                     NULL },
+  { "isprint",                    -1, (void*)isprint,                     NULL },
   { "vsprintf",                   -1, (void*)vsprintf,                      NULL },
-  { "abs",                        -1, (void*)::abs,                         NULL },
+  { "abs",                        -1, (void*)abs,                         NULL },
   { "labs",                       -1, (void*)::labs,                        NULL },
   { "islower",                    -1, (void*)::islower,                     NULL },
   { "isupper",                    -1, (void*)::isupper,                     NULL },
diff -wbBur XBMC/xbmc/cores/paplayer/MACDll/Source/MACLib/APELink.cpp XBMC.my/xbmc/cores/paplayer/MACDll/Source/MACLib/APELink.cpp
--- XBMC/xbmc/cores/paplayer/MACDll/Source/MACLib/APELink.cpp	2009-05-29 09:52:20.000000000 +0000
+++ XBMC.my/xbmc/cores/paplayer/MACDll/Source/MACLib/APELink.cpp	2009-05-26 12:02:22.000000000 +0000
@@ -55,10 +55,10 @@
     if (pData != NULL)
     {
         // parse out the information
-        char * pHeader = strstr(pData, APE_LINK_HEADER);
-        char * pImageFile = strstr(pData, APE_LINK_IMAGE_FILE_TAG);
-        char * pStartBlock = strstr(pData, APE_LINK_START_BLOCK_TAG);
-        char * pFinishBlock = strstr(pData, APE_LINK_FINISH_BLOCK_TAG);
+        char * pHeader = (char*)strstr(pData, APE_LINK_HEADER);
+        char * pImageFile = (char*)strstr(pData, APE_LINK_IMAGE_FILE_TAG);
+        char * pStartBlock = (char*)strstr(pData, APE_LINK_START_BLOCK_TAG);
+        char * pFinishBlock = (char*)strstr(pData, APE_LINK_FINISH_BLOCK_TAG);
 
         if (pHeader && pImageFile && pStartBlock && pFinishBlock)
         {
diff -wbBur XBMC/xbmc/visualizations/XBMCProjectM/libprojectM/BuiltinParams.cpp XBMC.my/xbmc/visualizations/XBMCProjectM/libprojectM/BuiltinParams.cpp
--- XBMC/xbmc/visualizations/XBMCProjectM/libprojectM/BuiltinParams.cpp	2009-05-29 10:20:17.000000000 +0000
+++ XBMC.my/xbmc/visualizations/XBMCProjectM/libprojectM/BuiltinParams.cpp	2009-05-29 14:29:34.000000000 +0000
@@ -1,3 +1,4 @@
+#include <stdio.h>
 
 #include "fatal.h"
 #include "BuiltinParams.hpp"
