--- docs/Makefile.in.orig	2008-11-09 22:45:28.000000000 +0100
+++ docs/Makefile.in	2011-07-23 17:12:26.000000000 +0200
@@ -527,9 +527,9 @@
 @HAVE_TEXI2HTML_TRUE@	  mkdir -lash-manual-html-split; \
 @HAVE_TEXI2HTML_TRUE@	fi
 @HAVE_TEXI2HTML_TRUE@	cd lash-manual-html-one-page && \
-@HAVE_TEXI2HTML_TRUE@	  texi2html --number ../$(top_srcdir)/docs/lash-manual.texi
+@HAVE_TEXI2HTML_TRUE@	  texi2html --number-footnotes --number-sections ../$(top_srcdir)/docs/lash-manual.texi
 @HAVE_TEXI2HTML_TRUE@	cd lash-manual-html-split && \
-@HAVE_TEXI2HTML_TRUE@	  texi2html --number --split=chapter ../$(top_srcdir)/docs/lash-manual.texi
+@HAVE_TEXI2HTML_TRUE@	  texi2html --number-footnotes --number-sections --split=chapter ../$(top_srcdir)/docs/lash-manual.texi
 @HAVE_TEXI2HTML_TRUE@	touch html-manual-stamp
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
