diff --git a/Modules/opengl/opengl.cpp b/Modules/opengl/opengl.cpp
index c87cde8..dcb09ff 100644
--- a/Modules/opengl/opengl.cpp
+++ b/Modules/opengl/opengl.cpp
@@ -818,7 +818,7 @@ static QTreeWidgetItem *get_gl_info_egl(Display *dpy, int scrnum, QTreeWidgetIte
       return result;
    }
 
-   surf = eglCreateWindowSurface(egl_dpy, config, win, NULL);
+   surf = 0;
    if (!surf) {
       qDebug() << "Error: eglCreateWindowSurface failed\n";
       eglDestroyContext(egl_dpy, ctx);
