diff -ur mythplugins-0.21.old/mythmusic/mythmusic/aacdecoder.cpp mythplugins-0.21/mythmusic/mythmusic/aacdecoder.cpp
--- mythplugins-0.21.old/mythmusic/mythmusic/aacdecoder.cpp	2007-08-25 17:49:44.000000000 +0200
+++ mythplugins-0.21/mythmusic/mythmusic/aacdecoder.cpp	2008-03-27 00:16:38.000000000 +0100
@@ -19,6 +19,7 @@
 #include <qobject.h>
 #include <qiodevice.h>
 #include <qfile.h>
+#include <stdlib.h>
 
 #include "aacdecoder.h"
 #include "constants.h"
diff -ur mythplugins-0.21.old/mythmusic/mythmusic/metaiomp4.cpp mythplugins-0.21/mythmusic/mythmusic/metaiomp4.cpp
--- mythplugins-0.21.old/mythmusic/mythmusic/metaiomp4.cpp	2006-08-22 04:06:03.000000000 +0200
+++ mythplugins-0.21/mythmusic/mythmusic/metaiomp4.cpp	2008-03-27 00:17:05.000000000 +0100
@@ -3,6 +3,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <sys/types.h>
+#include <stdlib.h>
 
 using namespace std;
 
