diff -up qca-ossl-2.0.0-beta3/qca-ossl.cpp.no-whirlpool qca-ossl-2.0.0-beta3/qca-ossl.cpp
--- qca-ossl-2.0.0-beta3/qca-ossl.cpp.no-whirlpool	2007-12-11 07:34:57.000000000 +0100
+++ qca-ossl-2.0.0-beta3/qca-ossl.cpp	2009-01-26 14:59:42.000000000 +0100
@@ -6597,7 +6597,7 @@ static QStringList all_hash_types()
 #ifdef SHA512_DIGEST_LENGTH
 	list += "sha512";
 #endif
-#ifdef OBJ_whirlpool
+#ifdef WHIRLPOOL_DIGEST_LENGTH
 	list += "whirlpool";
 #endif
 	return list;
@@ -6810,7 +6810,7 @@ public:
 		else if ( type == "sha512" )
 			return new opensslHashContext( EVP_sha512(), this, type);
 #endif
-#ifdef OBJ_whirlpool
+#ifdef WHIRLPOOL_DIGEST_LENGTH
 		else if ( type == "whirlpool" )
 			return new opensslHashContext( EVP_whirlpool(), this, type);
 #endif
