diff -baur vigra-1.7.1.old/include/vigra/random_forest.hxx vigra-1.7.1/include/vigra/random_forest.hxx
--- vigra-1.7.1.old/include/vigra/random_forest.hxx	2010-12-03 17:40:34.000000000 +0000
+++ vigra-1.7.1/include/vigra/random_forest.hxx	2011-01-28 00:16:32.000000000 +0000
@@ -43,6 +43,7 @@
 #include <set>
 #include <list>
 #include <numeric>
+#include <stddef.h>
 #include "mathutil.hxx"
 #include "array_vector.hxx"
 #include "sized_int.hxx"
diff -baur vigra-1.7.1.old/include/vigra/sifImport.hxx vigra-1.7.1/include/vigra/sifImport.hxx
--- vigra-1.7.1.old/include/vigra/sifImport.hxx	2010-12-03 17:40:34.000000000 +0000
+++ vigra-1.7.1/include/vigra/sifImport.hxx	2011-01-28 00:23:31.000000000 +0000
@@ -57,6 +57,7 @@
 #include <fstream>
 #include <cstring>
 #include <vector> 
+#include <stddef.h>
 #include "vigra/multi_array.hxx"
 
 namespace vigra {
diff -baur vigra-1.7.1.old/include/vigra/multi_iterator.hxx vigra-1.7.1/include/vigra/multi_iterator.hxx
--- vigra-1.7.1.old/include/vigra/multi_iterator.hxx	2010-12-03 17:40:34.000000000 +0000
+++ vigra-1.7.1/include/vigra/multi_iterator.hxx	2011-01-28 00:23:31.000000000 +0000
@@ -41,6 +41,7 @@
 #define VIGRA_MULTI_ITERATOR_HXX
 
 #include <sys/types.h>
+#include <stddef.h>
 #include "tinyvector.hxx"
 #include "iteratortags.hxx"
 
