diff -ruN -x .gitignore -x .cvsignore xf86-input-microtouch-d17c9694e4f83bdae28c9b6154158823ca1d62b0/README xf86-input-microtouch-c77782901cf02272c947f3464a5f5f4d3c534fbf/README
--- xf86-input-microtouch-d17c9694e4f83bdae28c9b6154158823ca1d62b0/README	1970-01-01 00:00:00.000000000 +0000
+++ xf86-input-microtouch-c77782901cf02272c947f3464a5f5f4d3c534fbf/README	2009-01-31 04:27:45.000000000 +0000
@@ -0,0 +1,20 @@
+xf86-input-microtouch - MicroTouch input driver for the Xorg X server
+
+Please submit bugs & patches to the Xorg bugzilla:
+
+        https://bugs.freedesktop.org/enter_bug.cgi?product=xorg
+
+All questions regarding this software should be directed at the
+Xorg mailing list:
+
+        http://lists.freedesktop.org/mailman/listinfo/xorg
+
+The master development code repository can be found at:
+
+        git://anongit.freedesktop.org/git/xorg/driver/xf86-input-microtouch
+
+        http://cgit.freedesktop.org/xorg/driver/xf86-input-microtouch
+
+For more information on the git code manager, see:
+
+        http://wiki.x.org/wiki/GitPage
diff -ruN -x .gitignore -x .cvsignore xf86-input-microtouch-d17c9694e4f83bdae28c9b6154158823ca1d62b0/man/microtouch.man xf86-input-microtouch-c77782901cf02272c947f3464a5f5f4d3c534fbf/man/microtouch.man
--- xf86-input-microtouch-d17c9694e4f83bdae28c9b6154158823ca1d62b0/man/microtouch.man	2008-03-20 20:22:05.000000000 +0000
+++ xf86-input-microtouch-c77782901cf02272c947f3464a5f5f4d3c534fbf/man/microtouch.man	2009-01-31 04:27:45.000000000 +0000
@@ -31,6 +31,6 @@
 .PP
 Config details...
 .SH "SEE ALSO"
-__xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), xorgconfig(__appmansuffix__), Xserver(__appmansuffix__), X(__miscmansuffix__).
+__xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), Xserver(__appmansuffix__), X(__miscmansuffix__).
 .SH AUTHORS
 Authors include...
diff -ruN -x .gitignore -x .cvsignore xf86-input-microtouch-d17c9694e4f83bdae28c9b6154158823ca1d62b0/src/microtouch.c xf86-input-microtouch-c77782901cf02272c947f3464a5f5f4d3c534fbf/src/microtouch.c
--- xf86-input-microtouch-d17c9694e4f83bdae28c9b6154158823ca1d62b0/src/microtouch.c	2008-03-20 20:22:05.000000000 +0000
+++ xf86-input-microtouch-c77782901cf02272c947f3464a5f5f4d3c534fbf/src/microtouch.c	2009-01-31 04:27:45.000000000 +0000
@@ -138,7 +138,9 @@
 	"xf86ErrorFVerb",
 	"xf86FindOptionValue",
 	"xf86FlushInput",
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
 	"xf86GetMotionEvents",
+#endif
 	"xf86GetVerbosity",
 	"xf86LoaderReqSymLists",
 	"xf86MotionHistoryAllocate",
@@ -160,12 +162,6 @@
 #if GET_ABI_MAJOR(ABI_XINPUT_VERSION) == 0
 	"xf86XInputSetSendCoreEvents",
 #endif
-	"xf86memset",
-	"xf86sscanf",
-	"xf86strcmp",
-	"xf86strlen",
-	"xf86strncmp",
-	"xf86strncpy",
         NULL
 };
 
@@ -425,7 +421,10 @@
 	 * Device reports motions on 2 axes in absolute coordinates.
 	 * Axes min and max values are reported in raw coordinates.
 	 */
-	if (InitValuatorClassDeviceStruct (dev, 2, xf86GetMotionEvents,
+	if (InitValuatorClassDeviceStruct (dev, 2,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
+					   xf86GetMotionEvents,
+#endif
 									local->history_size, Absolute) == FALSE)
 	{
 		ErrorF ("Unable to allocate MicroTouch touchscreen ValuatorClassDeviceStruct\n");
