--- cowsay-3.03/install.sh	1999-11-01 21:19:21.000000000 +0100
+++ cowsay-3.03-new/install.sh	2003-08-15 17:14:23.000000000 +0200
@@ -67,7 +67,7 @@
 echo Okay, time to install this puppy.
 
 echo s,%BANGPERL%,!$usethisperl,\; > install.pl
-echo s,%PREFIX%,$PREFIX,\; >> install.pl
+echo s,%PREFIX%,/usr,\; >> install.pl
 set -x
 mkdir -p $PREFIX/bin || (mkdir $PREFIX; mkdir $PREFIX/bin)
 $usethisperl -p install.pl cowsay > $PREFIX/bin/cowsay
