From 96dc0b9e56da64d2c063d1e74c81034b6b404bc2 Mon Sep 17 00:00:00 2001
From: Roman Kyrylych <roman@archlinux.org>
Date: Sat, 15 Mar 2008 21:46:35 +0200
Subject: [PATCH] Do not exit on ferror during psf font loading when built with klibc

Signed-off-by: Roman Kyrylych <roman@archlinux.org>
---
 src/psffontop.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/psffontop.c b/src/psffontop.c
index 1b99e9b..75a6048 100644
--- a/src/psffontop.c
+++ b/src/psffontop.c
@@ -203,9 +203,13 @@ readpsffont(FILE *fontf, char **allbufp, int *allszp,
 			}
 			n += fread(inputbuf+n, 1, inputbuflth-n, fontf);
 			if (ferror(fontf)) {
+				#ifndef __klibc__
 				char *u = _("%s: Error reading input font");
 				fprintf(stderr, u, progname);
 				exit(EX_DATAERR);
+				#else
+				break;
+				#endif
 			}
 			if (feof(fontf))
 				break;
-- 
1.5.3.7

