--- src/corelib/global/qendian.h.orig	2016-05-08 11:03:09.827707031 -0600
+++ src/corelib/global/qendian.h	2016-05-08 11:04:53.968007202 -0600
@@ -136,7 +136,7 @@
         | ((source & 0xff000000) >> 24);
 }
 #endif // GCC & Clang intrinsics
-#if (defined(Q_CC_GNU) && Q_CC_GNU >= 408) || QT_HAS_BUILTIN(__builtin_bswap16)
+#if defined(__ARM_NEON__) && ((defined(Q_CC_GNU) && Q_CC_GNU >= 408) || QT_HAS_BUILTIN(__builtin_bswap16))
 template <> inline quint16 qbswap<quint16>(quint16 source)
 {
     return __builtin_bswap16(source);
