diff -up qtbase-everywhere-src-5.11.1/mkspecs/linux-g++/qplatformdefs.h.glibc qtbase-everywhere-src-5.11.1/mkspecs/linux-g++/qplatformdefs.h
--- qtbase-everywhere-src-5.11.1/mkspecs/linux-g++/qplatformdefs.h.glibc	2018-06-15 09:29:31.000000000 +0200
+++ qtbase-everywhere-src-5.11.1/mkspecs/linux-g++/qplatformdefs.h	2018-07-26 15:40:56.887961897 +0200
@@ -72,7 +72,9 @@
 #include <sys/time.h>
 #include <sys/shm.h>
 #include <sys/socket.h>
+#if 0
 #include <sys/stat.h>
+#endif
 #include <sys/wait.h>
 #include <netinet/in.h>
 
diff -up qtbase-everywhere-src-5.11.1/src/corelib/io/qfilesystemengine_unix.cpp.glibc qtbase-everywhere-src-5.11.1/src/corelib/io/qfilesystemengine_unix.cpp
--- qtbase-everywhere-src-5.11.1/src/corelib/io/qfilesystemengine_unix.cpp.glibc	2018-06-15 09:29:31.000000000 +0200
+++ qtbase-everywhere-src-5.11.1/src/corelib/io/qfilesystemengine_unix.cpp	2018-07-26 16:34:54.346840184 +0200
@@ -50,7 +50,9 @@
 #include <pwd.h>
 #include <stdlib.h> // for realpath()
 #include <sys/types.h>
+#if 0
 #include <sys/stat.h>
+#endif
 #include <unistd.h>
 #include <stdio.h>
 #include <errno.h>
@@ -91,7 +93,9 @@ extern "C" NSString *NSTemporaryDirector
 #  include <sys/syscall.h>
 #  include <sys/sendfile.h>
 #  include <linux/fs.h>
+#if 0
 #  include <linux/stat.h>
+#endif
 
 // in case linux/fs.h is too old and doesn't define it:
 #ifndef FICLONE
@@ -105,13 +109,13 @@ extern "C" NSString *NSTemporaryDirector
 #    undef SYS_renameat2
 #    undef SYS_statx
 #    undef STATX_BASIC_STATS
-#  else
-#    if !QT_CONFIG(renameat2) && defined(SYS_renameat2)
+#  else 
+#    if 0 && !QT_CONFIG(renameat2) && defined(SYS_renameat2)
 static int renameat2(int oldfd, const char *oldpath, int newfd, const char *newpath, unsigned flags)
 { return syscall(SYS_renameat2, oldfd, oldpath, newfd, newpath, flags); }
 #    endif
 
-#    if !QT_CONFIG(statx) && defined(SYS_statx)
+#    if 0 && !QT_CONFIG(statx) && defined(SYS_statx)
 static int statx(int dirfd, const char *pathname, int flag, unsigned mask, struct statx *statxbuf)
 { return syscall(SYS_statx, dirfd, pathname, flag, mask, statxbuf); }
 #    elif !QT_CONFIG(statx) && !defined(SYS_statx)
