diff --git a/mkspecs/common/g++-unix.conf b/mkspecs/common/g++-unix.conf
index a493cd5984..79f2d2fff9 100644
--- a/mkspecs/common/g++-unix.conf
+++ b/mkspecs/common/g++-unix.conf
@@ -10,5 +10,5 @@
 
 include(g++-base.conf)
 
-QMAKE_LFLAGS_RELEASE   += -Wl,-O1
+QMAKE_LFLAGS_RELEASE   += $$(LDFLAGS)
 QMAKE_LFLAGS_NOUNDEF   += -Wl,--no-undefined
diff --git a/mkspecs/common/gcc-base.conf b/mkspecs/common/gcc-base.conf
index 4d82321cba..4a61fc9d25 100644
--- a/mkspecs/common/gcc-base.conf
+++ b/mkspecs/common/gcc-base.conf
@@ -40,8 +40,8 @@ QMAKE_CFLAGS_OPTIMIZE_SIZE = -Os
 QMAKE_CFLAGS_DEPS          += -M
 QMAKE_CFLAGS_WARN_ON       += -Wall -Wextra
 QMAKE_CFLAGS_WARN_OFF      += -w
-QMAKE_CFLAGS_RELEASE       += $$QMAKE_CFLAGS_OPTIMIZE
-QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO += $$QMAKE_CFLAGS_OPTIMIZE -g
+QMAKE_CFLAGS_RELEASE       += $$QMAKE_CFLAGS_OPTIMIZE $$(CFLAGS)
+QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO += $$QMAKE_CFLAGS_OPTIMIZE -g $$(CFLAGS)
 QMAKE_CFLAGS_DEBUG         += -g
 QMAKE_CFLAGS_SHLIB         += $$QMAKE_CFLAGS_PIC
 QMAKE_CFLAGS_STATIC_LIB    += $$QMAKE_CFLAGS_PIC
