--- qtbase-opensource-src-5.6.0-rc/config.tests/unix/alsa/alsatest.cpp.0	2016-02-29 08:15:48.203031809 +0000
+++ qtbase-opensource-src-5.6.0-rc/config.tests/unix/alsa/alsatest.cpp	2016-02-29 08:16:39.712811962 +0000
@@ -32,7 +32,7 @@
 ****************************************************************************/
 
 #include <alsa/asoundlib.h>
-#if(!(SND_LIB_MAJOR == 1 && SND_LIB_MINOR == 0 && SND_LIB_SUBMINOR >= 10))
+#if(!(SND_LIB_MAJOR == 1 && (SND_LIB_MINOR > 0 || SND_LIB_SUBMINOR >= 10)))
 #error "Alsa version found too old, require >= 1.0.10"
 #endif
 
