diff -rupN razorqt-0.5.2/razorqt-lightdm-greeter/CMakeLists.txt razorqt-0.5.2p/razorqt-lightdm-greeter/CMakeLists.txt
--- razorqt-0.5.2/razorqt-lightdm-greeter/CMakeLists.txt	2013-01-12 17:45:52.000000000 +0100
+++ razorqt-0.5.2p/razorqt-lightdm-greeter/CMakeLists.txt	2013-10-11 13:22:28.000000000 +0200
@@ -24,7 +24,11 @@ include ( ${QT_USE_FILE} )
 # check for new version
 include(FindPkgConfig)
 message(STATUS "Checking for lightdm-qt...")
-pkg_check_modules(LIGHTDM_QT liblightdm-qt-2)
+pkg_check_modules(LIGHTDM_QT liblightdm-qt-3)
+if (NOT LIGHTDM_QT_FOUND)
+    message(STATUS "lightdm3 for Qt is not found. Trying to find v2")
+    pkg_check_modules(LIGHTDM_QT liblightdm-qt-2)
+endif()
 # or lightdm v1 if the version 2 is not found
 if (NOT LIGHTDM_QT_FOUND)
     message(STATUS "lightdm2 for Qt is not found. Trying to find v1")
