diff -Naur audacity-src-2.0.6-orig/src/FFmpeg.h audacity-src-2.0.6/src/FFmpeg.h
--- audacity-src-2.0.6-orig/src/FFmpeg.h	2014-09-28 21:54:55.159636208 -0400
+++ audacity-src-2.0.6/src/FFmpeg.h	2014-09-28 21:55:11.246276766 -0400
@@ -688,7 +688,7 @@
    FFMPEG_FUNCTION_WITH_RETURN(
       AVOutputFormat*,
       av_oformat_next,
-      (AVOutputFormat *f),
+      (const AVOutputFormat *f),
       (f)
    );
    FFMPEG_FUNCTION_WITH_RETURN(
@@ -755,7 +755,7 @@
    FFMPEG_FUNCTION_WITH_RETURN(
       int,
       av_fifo_size,
-      (AVFifoBuffer *f),
+      (const AVFifoBuffer *f),
       (f)
    );
    FFMPEG_FUNCTION_WITH_RETURN(
@@ -801,7 +801,7 @@
    FFMPEG_FUNCTION_WITH_RETURN(
       AVDictionaryEntry *,
       av_dict_get,
-      (AVDictionary *m, const char *key, const AVDictionaryEntry *prev, int flags),
+      (const AVDictionary *m, const char *key, const AVDictionaryEntry *prev, int flags),
       (m, key, prev, flags)
    );
    FFMPEG_FUNCTION_WITH_RETURN(
