--- a/H/msgdef.h
+++ b/H/msgdef.h
@@ -8,7 +8,7 @@
  */
 pick( MSG_USAGE,              "usage: UASM [ options ] filelist [@env_var]\n" "Run \"Uasm -?\" or \"Uasm -h\" for more info\n" )
 pick( MSG_ASSEMBLY_RESULTS,   "%s: %lu lines, %u passes, %u ms, %u warnings, %u errors" )
-pick( MSG_UASM,               "UASM v" _UASM_VERSION_ ", " __DATE__ ", Masm-compatible assembler.\n" )
+pick( MSG_UASM,               "UASM v" _UASM_VERSION_ ", @COMMIT_DATE@, Masm-compatible assembler.\n" )
 pick( MSG_UASM2,              "Portions Copyright (c) 1992-2002 Sybase, Inc. All Rights Reserved.\n" "Source code is available under the Sybase Open Watcom Public License.\n" )
 pick( MSG_ERROR_PREFIX,       "Error" )
 pick( MSG_WARNING_PREFIX,     "Warning" )
