--- Builder.py-bak	2007-03-11 02:52:00.000000000 +0200
+++ Builder.py	2007-03-11 02:52:08.000000000 +0200
@@ -872,7 +872,7 @@
         Clears all of the cache entries that use the specified Builder.
         """
         for key, smap in self._memo.items():
-            if builder in smap.builders_used:
+            if builder in smap:
                 del self._memo[key]
 
 SuffixMapCache = _SuffixMapCache()
