--- raggle.old	2006-07-07 00:27:23.000000000 +0200
+++ raggle	2006-07-07 00:28:22.000000000 +0200
@@ -48,6 +48,17 @@
 # As early as possible, ^C and ^\ are common, and dumping a trace is
 # ugly On the other hand, dumping trace is very useful when running
 # tests, therefore disable these unless this file is executed.
+
+if RUBY_VERSION == "1.8.4"
+  class Bignum
+    def to_yaml( opts = {} )
+      YAML::quick_emit( nil, opts ) { |out|
+        out.scalar( nil, to_s, :plain )
+    } 
+    end
+  end
+end
+
 if __FILE__ == $0
   trap('INT') {
     if $config['run_http_server']
