diff -Naur mysql-5.1.32/client/Makefile.in mysql-5.1.32.novo/client/Makefile.in
--- mysql-5.1.32/client/Makefile.in	2009-02-13 23:02:45.000000000 -0200
+++ mysql-5.1.32.novo/client/Makefile.in	2009-03-24 05:32:44.000000000 -0300
@@ -464,15 +464,15 @@
 yassl_libs = @yassl_libs@
 yassl_taocrypt_extra_cxxflags = @yassl_taocrypt_extra_cxxflags@
 zlib_dir = @zlib_dir@
-@THREAD_SAFE_CLIENT_FALSE@LIBMYSQLCLIENT_LA = $(top_builddir)/libmysql/libmysqlclient.la
-@THREAD_SAFE_CLIENT_TRUE@LIBMYSQLCLIENT_LA = $(top_builddir)/libmysql_r/libmysqlclient_r.la
+@THREAD_SAFE_CLIENT_FALSE@LIBMYSQLCLIENT_LA = /usr/lib/mysql/libmysqlclient.so
+@THREAD_SAFE_CLIENT_TRUE@LIBMYSQLCLIENT_LA = /usr/lib/mysql/libmysqlclient_r.so
 INCLUDES = -I$(top_builddir)/include \
 				-I$(top_srcdir)/include \
 				-I$(top_srcdir)/regex \
                                 $(openssl_includes)
 
 LDADD = @CLIENT_EXTRA_LDFLAGS@ $(CLIENT_THREAD_LIBS) \
-				$(top_builddir)/libmysql/libmysqlclient.la
+				/usr/lib/mysql/libmysqlclient.so
 
 noinst_HEADERS = sql_string.h completion_hash.h my_readline.h \
 				client_priv.h
diff -Naur mysql-5.1.32/tests/Makefile.in mysql-5.1.32.novo/tests/Makefile.in
--- mysql-5.1.32/tests/Makefile.in	2009-02-13 23:03:19.000000000 -0200
+++ mysql-5.1.32.novo/tests/Makefile.in	2009-03-24 05:32:35.000000000 -0300
@@ -408,8 +408,8 @@
 yassl_libs = @yassl_libs@
 yassl_taocrypt_extra_cxxflags = @yassl_taocrypt_extra_cxxflags@
 zlib_dir = @zlib_dir@
-@THREAD_SAFE_CLIENT_FALSE@LIBMYSQLCLIENT_LA = $(top_builddir)/libmysql/libmysqlclient.la
-@THREAD_SAFE_CLIENT_TRUE@LIBMYSQLCLIENT_LA = $(top_builddir)/libmysql_r/libmysqlclient_r.la
+@THREAD_SAFE_CLIENT_FALSE@LIBMYSQLCLIENT_LA = /usr/lib/mysql/libmysqlclient.so
+@THREAD_SAFE_CLIENT_TRUE@LIBMYSQLCLIENT_LA = /usr/lib/mysql/libmysqlclient_r.so
 EXTRA_DIST = auto_increment.res auto_increment.tst \
 			function.res function.tst lock_test.pl lock_test.res \
 			export.pl big_record.pl \
