diff -wbBur sdd-1.52/include/schily.h sdd-1.52.my/include/schily.h
--- sdd-1.52/include/schily.h	2004-03-05 00:30:40.000000000 +0000
+++ sdd-1.52.my/include/schily.h	2010-07-08 14:00:07.000000000 +0000
@@ -108,14 +108,14 @@
 		/* 6th arg not const, fexecv forces av[ac] = NULL */
 extern	int	fexecv __PR((const char *, FILE *, FILE *, FILE *, int,
 							char **));
-extern	int	fexecve __PR((const char *, FILE *, FILE *, FILE *,
-					char * const *, char * const *));
+//extern	int	fexecve __PR((const char *, FILE *, FILE *, FILE *,
+//					char * const *, char * const *));
 extern	int	fspawnv __PR((FILE *, FILE *, FILE *, int, char * const *));
 extern	int	fspawnl __PR((FILE *, FILE *, FILE *,
 					const char *, const char *, ...));
 extern	int	fspawnv_nowait __PR((FILE *, FILE *, FILE *,
 					const char *, int, char *const*));
-extern	int	fgetline __PR((FILE *, char *, int));
+extern	int	xfgetline __PR((FILE *, char *, int));
 extern	int	fgetstr __PR((FILE *, char *, int));
 extern	void	file_raise __PR((FILE *, int));
 extern	int	fileclose __PR((FILE *));
@@ -187,7 +187,7 @@
 extern	char	*findbytes __PR((const void *, int, char));
 extern	int	findline __PR((const char *, char, const char *,
 							int, char **, int));
-extern	int	getline __PR((char *, int));
+extern	int	xgetline __PR((char *, int));
 extern	int	getstr __PR((char *, int));
 extern	int	breakline __PR((char *, char, char **, int));
 extern	int	getallargs __PR((int *, char * const**, const char *, ...));
diff -wbBur sdd-1.52/libschily/stdio/fgetline.c sdd-1.52.my/libschily/stdio/fgetline.c
--- sdd-1.52/libschily/stdio/fgetline.c	2004-08-08 11:02:05.000000000 +0000
+++ sdd-1.52.my/libschily/stdio/fgetline.c	2010-07-08 14:02:08.000000000 +0000
@@ -26,7 +26,7 @@
  */
 
 EXPORT int
-fgetline(f, buf, len)
+xfgetline(f, buf, len)
 	register	FILE	*f;
 			char	*buf;
 	register	int	len;
@@ -64,9 +64,9 @@
 }
 
 EXPORT int
-getline(buf, len)
+xgetline(buf, len)
 	char	*buf;
 	int	len;
 {
-	return (fgetline(stdin, buf, len));
+	return (xfgetline(stdin, buf, len));
 }
diff -wbBur sdd-1.52/RULES/mk-gmake.id sdd-1.52.my/RULES/mk-gmake.id
--- sdd-1.52/RULES/mk-gmake.id	2004-08-03 21:33:18.000000000 +0000
+++ sdd-1.52.my/RULES/mk-gmake.id	2010-07-08 14:00:07.000000000 +0000
@@ -37,7 +37,7 @@
 
 ifndef		ARCH_DONE
 
-__gmake_warn:=	$(shell cat $(SRCROOT)/$(RULESDIR)/gmake.wrn 1>&2; sleep 15)
+__gmake_warn:=	$(shell cat $(SRCROOT)/$(RULESDIR)/gmake.wrn 1>&2)
 
 _MACHCMD=	(mach || uname -p || true)	2> /dev/null
 _ARCHCMD=	(arch || /usr/ucb/arch || true)	2> /dev/null
diff -wbBur sdd-1.52/sdd/sdd.c sdd-1.52.my/sdd/sdd.c
--- sdd-1.52/sdd/sdd.c	2004-07-27 10:49:31.000000000 +0000
+++ sdd-1.52.my/sdd/sdd.c	2010-07-08 14:01:20.000000000 +0000
@@ -850,7 +850,7 @@
 						inout, num);
 		(void) fflush(stderr);
 		ap = answer;
-		if (fgetline(tty, ap, 16) == EOF)
+		if (xfgetline(tty, ap, 16) == EOF)
 			return (FALSE);
 		while (*ap == ' ' || *ap == '\t')
 			ap++;
