--- configure.in.orig	2008-06-29 11:55:57.000000000 +0000
+++ configure.in	2008-06-29 12:05:54.000000000 +0000
@@ -79,8 +79,8 @@
 	fi
 	if test "$MOZPLUG_CFLAGS" = ""; then
 		PKG_CHECK_MODULES(MOZPLUG,[libxul],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir libxul`"
-MOZILLA_HOME=`$PKG_CONFIG --variable=libdir libxul`
-GECKO_IDLDIR=`$PKG_CONFIG --variable=idldir libxul`
+MOZILLA_HOME=`$PKG_CONFIG --variable=sdkdir libxul`/bin
+GECKO_IDLDIR=`$PKG_CONFIG --variable=idldir libxul`/stable
 ],[AC_MSG_WARN([libxul not found])])
 	fi
 	if test "$MOZPLUG_CFLAGS" = ""; then
