--- system-tools-backends-2.6.0.edited/system-tools-backends.conf	2009-07-22 03:56:55.000000000 -0400
+++ system-tools-backends-2.6.0/system-tools-backends.conf	2009-07-22 04:01:46.000000000 -0400
@@ -23,11 +23,8 @@
     -->
 
     <!-- configuration modules can't be accessed directly... -->
-    <deny send_interface="org.freedesktop.SystemToolsBackends"/>
-    <deny send_interface="org.freedesktop.SystemToolsBackends.Platform"/>
-
-    <!-- ...so petitions go through the dispatcher instead -->
-    <allow send_destination="org.freedesktop.SystemToolsBackends"/>
+    <deny send_destination="org.freedesktop.SystemToolsBackends"/>
+    <deny send_destination="org.freedesktop.SystemToolsBackends.Platform"/>
   </policy>
 
   <policy user="0">
@@ -49,7 +46,21 @@
 
     <!-- be able to speak to configuration modules,
          so any message to them has to go through the dispatcher -->
-    <allow send_interface="org.freedesktop.SystemToolsBackends"/>
-    <allow send_interface="org.freedesktop.SystemToolsBackends.Platform"/>
+    <allow send_destination="org.freedesktop.SystemToolsBackends"/>
+    <allow send_destination="org.freedesktop.SystemToolsBackends.Platform"/>
+    <allow send_destination="org.freedesktop.SystemToolsBackends.GroupsConfig"/>
+    <allow send_destination="org.freedesktop.SystemToolsBackends.HostsConfig"/>
+    <allow send_destination="org.freedesktop.SystemToolsBackends.IfacesConfig"/>
+    <allow send_destination="org.freedesktop.SystemToolsBackends.NFSConfig"/>
+    <allow send_destination="org.freedesktop.SystemToolsBackends.NTPConfig"/>
+    <allow send_destination="org.freedesktop.SystemToolsBackends.ServicesConfig"/>
+    <allow send_destination="org.freedesktop.SystemToolsBackends.SMBConfig"/>
+    <allow send_destination="org.freedesktop.SystemToolsBackends.TimeConfig"/>
+    <allow send_destination="org.freedesktop.SystemToolsBackends.UserConfig"/>
+    <allow send_destination="org.freedesktop.SystemToolsBackends.UsersConfig"/>
+  </policy>
+  <policy group="stb-admin">
+    <!-- be able to speak to the dispatcher -->
+    <allow send_destination="org.freedesktop.SystemToolsBackends"/>
   </policy>
 </busconfig>
