From 63e8c48149277742bc8a62820177a5343c1ea9de Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sat, 8 Sep 2018 09:24:56 -0600
Subject: [PATCH] fix finding libsensors

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index ba92fb3d85..81668b3ae0 100644
--- a/meson.build
+++ b/meson.build
@@ -1311,7 +1311,7 @@ endif
 
 _sensors = get_option('lmsensors')
 if _sensors != 'false'
-  dep_lmsensors = cc.find_library('libsensors', required : _sensors == 'true')
+  dep_lmsensors = cc.find_library('sensors', required : _sensors == 'true')
   if dep_lmsensors.found()
     pre_args += '-DHAVE_LIBSENSORS=1'
   endif
-- 
2.18.0

