From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Sat, 1 Mar 2025 02:16:28 +0100
Subject: [PATCH] docs: Fix build with Sphinx 8.2

---
 docs/_exts/bootstrap.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/docs/_exts/bootstrap.py b/docs/_exts/bootstrap.py
index 18e46cea1608..6a9cebf59efc 100644
--- a/docs/_exts/bootstrap.py
+++ b/docs/_exts/bootstrap.py
@@ -65,6 +65,7 @@ class BootstrapHTML5TranslatorMixin:
 
         self.body.append(self.starttag(
             node, 'div', CLASS=('alert ' + admonitionclasses[name])))
+        self.context.append("</div>")
         if name:
             self.body.append(
                   self.starttag(node, 'div', '', CLASS='h5'))
