From 161aababc659e23c4a8523366a31f63b3d14e280 Mon Sep 17 00:00:00 2001
From: Ian Romanick <ian.d.romanick@intel.com>
Date: Mon, 08 Oct 2012 22:07:10 +0000
Subject: docs: Add 9.0 release md5sums

Signed-off-by: Ian Romanick <ian.d.romanick@intel.com>
---
diff --git a/docs/relnotes-9.0.html b/docs/relnotes-9.0.html
index d72c5bb..02b7324 100644
--- a/docs/relnotes-9.0.html
+++ b/docs/relnotes-9.0.html
@@ -26,7 +26,9 @@ because GL_ARB_compatibility is not supported.

 <h2>MD5 checksums</h2>
 <pre>
-tbd
+be4cd34c6599a7cb9d254b05c48bdb1f  MesaLib-9.0.tar.gz
+60e557ce407be3732711da484ab3db6c  MesaLib-9.0.tar.bz2
+16b128544cd3f7e237927bb9f8aab7ce  MesaLib-9.0.zip
 </pre>


--
cgit v0.9.0.2-2-gbebe
From 32faf7ab0de8b88bb15a2cb262a73c411dce9d0d Mon Sep 17 00:00:00 2001
From: Brian Paul <brianp@vmware.com>
Date: Fri, 05 Oct 2012 22:47:40 +0000
Subject: mesa: don't call TexImage driver hooks for zero-sized images

This simply avoids some failed assertions but there's no reason to
call the driver hooks for storing a tex image if its size is zero.

Note: This is a candidate for the stable branches.
(cherry picked from commit 91d84096497ff538f55591f7e6bb0b563726db8d)
---
diff --git a/src/mesa/main/teximage.c b/src/mesa/main/teximage.c
index 21646cc..8004876 100644
--- a/src/mesa/main/teximage.c
+++ b/src/mesa/main/teximage.c
@@ -3034,13 +3034,15 @@ teximage(struct gl_context *ctx, GLboolean compressed, GLuint dims,
                                           border, internalFormat, texFormat);

                /* Give the texture to the driver.  <pixels> may be null. */
-               if (compressed) {
-                  ctx->Driver.CompressedTexImage(ctx, dims, texImage,
-                                                 imageSize, pixels);
-               }
-               else {
-                  ctx->Driver.TexImage(ctx, dims, texImage, format,
-                                       type, pixels, unpack);
+               if (width > 0 && height > 0 && depth > 0) {
+                  if (compressed) {
+                     ctx->Driver.CompressedTexImage(ctx, dims, texImage,
+                                                    imageSize, pixels);
+                  }
+                  else {
+                     ctx->Driver.TexImage(ctx, dims, texImage, format,
+                                          type, pixels, unpack);
+                  }
                }

                check_gen_mipmap(ctx, target, texObj, level);
--
cgit v0.9.0.2-2-gbebe
From e75051d1967350ceff0209dde24ae42696b13b5c Mon Sep 17 00:00:00 2001
From: Brian Paul <brianp@vmware.com>
Date: Fri, 05 Oct 2012 22:59:27 +0000
Subject: mesa: fix error check for zero-sized compressed subtexture

For glCompressedTexSubImage, width or height = 0 is legal.
Fixes a failure in piglit's s3tc-errors test.

This is for the 9.0 and 8.0 branches.  Already fixed on master.
---
diff --git a/src/mesa/main/teximage.c b/src/mesa/main/teximage.c
index 8004876..38fa9fa 100644
--- a/src/mesa/main/teximage.c
+++ b/src/mesa/main/teximage.c
@@ -3598,10 +3598,10 @@ compressed_subtexture_error_check(struct gl_context *ctx, GLint dimensions,
    if (!_mesa_is_compressed_format(ctx, format))
       return GL_INVALID_ENUM;

-   if (width < 1 || width > maxTextureSize)
+   if (width < 0 || width > maxTextureSize)
       return GL_INVALID_VALUE;

-   if ((height < 1 || height > maxTextureSize)
+   if ((height < 0 || height > maxTextureSize)
        && dimensions > 1)
       return GL_INVALID_VALUE;

--
cgit v0.9.0.2-2-gbebe
