--- src/gtk/button.cpp.orig	2007-08-30 23:26:57.000000000 +0200
+++ src/gtk/button.cpp	2007-08-30 23:27:52.000000000 +0200
@@ -80,7 +80,7 @@
             right_border += default_border->right;
             top_border += default_border->top;
             bottom_border += default_border->bottom;
-            g_free( default_border );
+            gtk_border_free( default_border );
         }
         win->MoveWindow(
             win->m_x - top_border,
--- src/gtk/window.cpp.orig	2007-08-30 23:52:06.000000000 +0200
+++ src/gtk/window.cpp	2007-08-30 23:52:14.000000000 +0200
@@ -2706,7 +2706,7 @@
                 right_border += default_border->right;
                 top_border += default_border->top;
                 bottom_border += default_border->bottom;
-                g_free( default_border );
+                gtk_border_free( default_border );
             }
         }
 
