diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 1eb691b6..1d8e3466 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -56,7 +56,7 @@ if(OCIO_BUILD_SHARED)
     endif()
 
     set_target_properties(OpenColorIO PROPERTIES
-        OUTPUT_NAME OpenColorIO
+        OUTPUT_NAME OpenColorIO1
         COMPILE_FLAGS   "${EXTERNAL_COMPILE_FLAGS}"
         LINK_FLAGS      "${EXTERNAL_LINK_FLAGS}")
     
@@ -109,7 +109,7 @@ endmacro()
 
 # public interface
 install(FILES ${core_export_headers}
-    DESTINATION ${CMAKE_INSTALL_PREFIX}/include/OpenColorIO/)
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/include/OpenColorIO1/)
 
 # pkg-config
 message(STATUS "Create OpenColorIO.pc from OpenColorIO.pc.in")
