--- a/common.mk	2009-07-19 00:44:37.000000000 -0700
+++ b/common.mk	2009-07-19 00:44:37.000000000 -0700
@@ -74,8 +74,9 @@
 install-apihdr-directory:
 	$(INSTALL_DIRECTORY) $(INSTALL_APIHDR_DIRECTORY)
 
+INSTALL_API_SOCKET_DIRECTORY = $(INSTALL_ROOT)$(API_SOCKET_DIRECTORY)
 install-apisoc-directory:
-	-$(INSTALL_DIRECTORY) -m 1777 $(API_SOCKET_DIRECTORY)
+	-$(INSTALL_DIRECTORY) -m 1777 $(INSTALL_API_SOCKET_DIRECTORY)
 
 clean::
 	-rm -f *.$O *.auto.h *.auto.c core implib.a
