From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Sun, 10 Dec 2023 16:11:13 +0100
Subject: [PATCH] configure.in: Remove assembly files from build

---
 configure.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure.in b/configure.in
index 36e358445827..00936dff2098 100644
--- a/configure.in
+++ b/configure.in
@@ -1584,21 +1584,21 @@ tools are selected during the Xcode/Developer Tools installation.])
         ;;
     i*86)
         AC_DEFINE(i386)
-        PR_MD_ASFILES=os_Linux_x86.s
+        PR_MD_ASFILES=
         ;;
     ia64)
         PR_MD_ASFILES=os_Linux_ia64.s
         ;;
     x86_64)
         if test -n "$USE_64"; then
-            PR_MD_ASFILES=os_Linux_x86_64.s
+            PR_MD_ASFILES=
         elif test -n "$USE_X32"; then
-            PR_MD_ASFILES=os_Linux_x86_64.s
+            PR_MD_ASFILES=
             CC="$CC -mx32"
             CXX="$CXX -mx32"
         else
             AC_DEFINE(i386)
-            PR_MD_ASFILES=os_Linux_x86.s
+            PR_MD_ASFILES=
             CC="$CC -m32"
             CXX="$CXX -m32"
         fi
