diff -r 99f1cf3520d9 -r b1d3888c0ae7 src/os_cpu/linux_zero/vm/thread_linux_zero.hpp
--- a/src/os_cpu/linux_zero/vm/thread_linux_zero.hpp	Tue Feb 16 13:20:38 2016 -0800
+++ b/src/os_cpu/linux_zero/vm/thread_linux_zero.hpp	Wed Feb 17 17:03:31 2016 -0500
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2000, 2010, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2000, 2016, Oracle and/or its affiliates. All rights reserved.
  * Copyright 2007, 2008, 2009, 2010 Red Hat, Inc.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
@@ -110,6 +110,7 @@
                                            void* ucontext,
                                            bool isInJava) {
     ShouldNotCallThis();
+    return false; // silence compile warning
   }
 
   // These routines are only used on cpu architectures that

diff -r 39575526c6d9 -r f5f2a2d13775 src/os/linux/vm/os_linux.inline.hpp
--- a/src/os/linux/vm/os_linux.inline.hpp	Thu Oct 12 01:18:38 2017 +0000
+++ b/src/os/linux/vm/os_linux.inline.hpp	Mon Sep 18 15:06:28 2017 +0200
@@ -98,6 +98,11 @@
 
 inline struct dirent* os::readdir(DIR* dirp, dirent *dbuf)
 {
+// readdir_r has been deprecated since glibc 2.24.
+// See https://sourceware.org/bugzilla/show_bug.cgi?id=19056 for more details.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+
   dirent* p;
   int status;
   assert(dirp != NULL, "just checking");
@@ -111,6 +116,8 @@
     return NULL;
   } else
     return p;
+
+#pragma GCC diagnostic pop
 }
 
 inline int os::closedir(DIR *dirp) {

