diff --git a/config/known-dwarf.awk b/config/known-dwarf.awk
index 0a62d13b2e6e..fb813e539543 100755
--- a/config/known-dwarf.awk
+++ b/config/known-dwarf.awk
@@ -26,7 +26,7 @@ set == "" && $1 ~ /DW_([A-Z_]+)_([^ ]+)/ {
   sub(/_[^[:upper:]_].*$/, "", set);
   if (set ~ /LANG_.+/) set = "LANG";
   if (set ~ /LNAME_.+/) set = "LNAME";
-  if (set ~ /SECT_.+/) set = "SECT"; /* */
+  if (set ~ /SECT_.+/) set = "SECT";
 }
 
 $1 ~ /DW([_A-Z]+)_([^ ]+)/ {
