diff -upr psqlodbc-08.04.0100.orig/odbcapi30.c psqlodbc-08.04.0100/odbcapi30.c
--- psqlodbc-08.04.0100.orig/odbcapi30.c	2008-10-07 00:46:09.000000000 +0300
+++ psqlodbc-08.04.0100/odbcapi30.c	2009-09-19 20:13:57.000000000 +0300
@@ -118,13 +118,7 @@ SQLColAttribute(SQLHSTMT StatementHandle
 			SQLPOINTER CharacterAttribute,
 			SQLSMALLINT BufferLength,
 			SQLSMALLINT *StringLength,
-#if defined(_WIN64)
 			SQLLEN *NumericAttribute
-#elif defined(WITH_UNIXODBC) || defined(WIN32)
-			SQLPOINTER NumericAttribute
-#else
-			SQLLEN *NumericAttribute
-#endif
 			)
 {
 	CSTR	func = "SQLColAttribute";
diff -upr psqlodbc-08.04.0100.orig/odbcapi30w.c psqlodbc-08.04.0100/odbcapi30w.c
--- psqlodbc-08.04.0100.orig/odbcapi30w.c	2009-03-22 16:17:10.000000000 +0200
+++ psqlodbc-08.04.0100/odbcapi30w.c	2009-09-19 20:22:17.000000000 +0300
@@ -262,11 +262,7 @@ SQLRETURN SQL_API SQLColAttributeW(
 	SQLPOINTER	pCharAttr,
 	SQLSMALLINT	cbCharAttrMax,	
 	SQLSMALLINT	*pcbCharAttr,
-#if defined(WITH_UNIXODBC) || (defined(WIN32) && ! defined(_WIN64))
-	SQLPOINTER	pNumAttr
-#else
 	SQLLEN		*pNumAttr
-#endif
 	)
 {
 	CSTR func = "SQLColAttributeW";
diff -upr psqlodbc-08.04.0100.orig/odbcapi.c psqlodbc-08.04.0100/odbcapi.c
--- psqlodbc-08.04.0100.orig/odbcapi.c	2008-11-23 03:00:53.000000000 +0200
+++ psqlodbc-08.04.0100/odbcapi.c	2009-09-19 20:13:56.000000000 +0300
@@ -1111,11 +1111,7 @@ SQLExtendedFetch(
 				 HSTMT hstmt,
 				 SQLUSMALLINT fFetchType,
 				 SQLLEN irow,
-#ifdef WITH_UNIXODBC
-				 SQLROWSETSIZE *pcrow,
-#else
 				 SQLULEN *pcrow,
-#endif /* WITH_UNIXODBC */
 				 SQLUSMALLINT *rgfRowStatus)
 {
 	RETCODE	ret;
