diff -wbBur checkinstall-1.6.1/checkinstall checkinstall-1.6.1.my/checkinstall
--- checkinstall-1.6.1/checkinstall	2006-11-01 10:45:40.000000000 +0300
+++ checkinstall-1.6.1.my/checkinstall	2006-11-02 19:11:31.000000000 +0300
@@ -45,7 +45,7 @@
 
 
 # gettext variables
-export TEXTDOMAINDIR=/usr/local/lib/checkinstall/locale
+export TEXTDOMAINDIR=/usr/share/locale
 export TEXTDOMAIN=checkinstall
 
 # .spec file path
@@ -386,13 +386,13 @@
 ckversion
 echo
 
-if ! [ -f /usr/local/lib/checkinstall/checkinstallrc ]; then
-   echog "The checkinstallrc file was not found at:\n/usr/local/lib/checkinstall/checkinstallrc"
+if ! [ -f /etc/checkinstall/checkinstallrc ]; then
+   echog "The checkinstallrc file was not found at:\n/etc/checkinstall/checkinstallrc"
    echo
    echog "Assuming default values."
 else
 # Get our default settings from the rc file
-source /usr/local/lib/checkinstall/checkinstallrc
+source /etc/checkinstall/checkinstallrc
 fi
 
 
@@ -755,7 +755,7 @@
 
 ################################################################
 # User-configurable variables were moved to the checkinstallrc #
-# file which is probably found at /usr/local/lib/checkinstall  #          
+# file which is probably found at /usr/lib/checkinstall        #
 #                                                              #
 #                  DO NOT modify them here!!                   #
 ################################################################
@@ -764,7 +764,7 @@
 # Debug level
 ! [ "$DEBUG" ] && DEBUG=0
 
-! [ "$INSTALLWATCH_PREFIX" ] && INSTALLWATCH_PREFIX="/usr/local"
+! [ "$INSTALLWATCH_PREFIX" ] && INSTALLWATCH_PREFIX="/usr"
 ! [ "$INSTALLWATCH" ] && INSTALLWATCH=${INSTALLWATCH_PREFIX}/bin/installwatch
 
 # Which makepkg to use
@@ -1654,11 +1654,11 @@
  COMPRESS="gzip -9"
  COMPRESS_EXT=.gz
  
- for d in ./usr/local/man/man* ./usr/local/man/*/man* ./usr/local/info \
- 	./usr/local/share/man/man* ./usr/local/share/man/*/man* \
- 	./usr/local/share/info \
- 	./usr/local/kerberos/man \
- 	./usr/local/share/doc/*/man/man* ./usr/local/lib/*/man/man* \
+ for d in ./usr/man/man* ./usr/man/*/man* ./usr/info \
+ 	./usr/share/man/man* ./usr/share/man/*/man* \
+ 	./usr/share/info \
+ 	./usr/kerberos/man \
+ 	./usr/share/doc/*/man/man* ./usr/lib/*/man/man* \
  	././usr/man/man* ./usr/man/*/man* ./usr/info \
  	./usr/share/man/man* ./usr/share/man/*/man* ./usr/share/info \
  	./usr/kerberos/man ./usr/X11R6/man/man* ./usr/lib/perl5/man/man* \
diff -wbBur checkinstall-1.6.1/checkinstallrc-dist checkinstall-1.6.1.my/checkinstallrc-dist
--- checkinstall-1.6.1/checkinstallrc-dist	2006-11-01 10:45:40.000000000 +0300
+++ checkinstall-1.6.1.my/checkinstallrc-dist	2006-11-02 19:11:31.000000000 +0300
@@ -16,7 +16,7 @@
 DEBUG=0
 
 # Location of the "installwatch" program
-INSTALLWATCH_PREFIX="/usr/local"
+INSTALLWATCH_PREFIX="/usr"
 INSTALLWATCH=${INSTALLWATCH_PREFIX}/bin/installwatch
 
 # Location of the makepkg program. "makepak" is the default, and is 
diff -wbBur checkinstall-1.6.1/installwatch-0.7.0beta5/Makefile checkinstall-1.6.1.my/installwatch-0.7.0beta5/Makefile
--- checkinstall-1.6.1/installwatch-0.7.0beta5/Makefile	2006-11-01 10:45:48.000000000 +0300
+++ checkinstall-1.6.1.my/installwatch-0.7.0beta5/Makefile	2006-11-02 19:11:31.000000000 +0300
@@ -4,14 +4,14 @@
 # Well, the only configurable part is the following variable.
 # Make sure the directory you specify exists.
 
-PREFIX=/usr/local
+PREFIX=/usr
 
 # End of configurable part
 
 VERSION=0.7.0beta5
 
-BINDIR=$(PREFIX)/bin
-LIBDIR=$(PREFIX)/lib
+BINDIR=$(DESTDIR)/$(PREFIX)/bin
+LIBDIR=$(DESTDIR)/$(PREFIX)/lib
 
 all: installwatch.so
 
diff -wbBur checkinstall-1.6.1/Makefile checkinstall-1.6.1.my/Makefile
--- checkinstall-1.6.1/Makefile	2006-11-01 10:45:40.000000000 +0300
+++ checkinstall-1.6.1.my/Makefile	2006-11-02 19:12:03.000000000 +0300
@@ -1,10 +1,10 @@
 # $Id: build-fix.patch,v 1.1 2007/04/06 08:49:39 sergej Exp $
 
 # Where to install.
-PREFIX=/usr/local
-BINDIR=$(PREFIX)/sbin
-LCDIR=$(PREFIX)/lib/checkinstall/locale
-CONFDIR=$(PREFIX)/lib/checkinstall
+PREFIX=/usr
+BINDIR=$(DESTDIR)/$(PREFIX)/sbin
+LCDIR=$(DESTDIR)/$(PREFIX)/share/locale
+CONFDIR=$(DESTDIR)/etc/checkinstall
 
 all:
 	for file in locale/checkinstall-*.po ; do \
@@ -25,7 +25,11 @@
 	export
 	make -C installwatch-0.7.0beta5 install
 	
+	mkdir -p $(DESTDIR)/usr/bin
 	mkdir -p $(BINDIR)
+	mkdir -p $(CONFDIR)
+	mkdir -p $(DESTDIR)/etc/checkinstall
+	
 	install checkinstall makepak $(BINDIR)
 	for file in locale/*.mo ; do \
 		LANG=`echo $$file | sed -e 's|locale/checkinstall-||' \
