diff --git c/bfd/doc/bfd.texinfo i/bfd/doc/bfd.texinfo
index ce1c5af..47a0667 100644
--- c/bfd/doc/bfd.texinfo
+++ i/bfd/doc/bfd.texinfo
@@ -323,7 +323,7 @@ All of BFD lives in one directory.
 @printindex cp
 
 @tex
-% I think something like @colophon should be in texinfo.  In the
+% I think something like @@colophon should be in texinfo.  In the
 % meantime:
 \long\def\colophon{\hbox to0pt{}\vfill
 \centerline{The body of this manual is set in}
@@ -334,7 +334,7 @@ All of BFD lives in one directory.
 \centerline{{\sl\fontname\tensl\/}}
 \centerline{are used for emphasis.}\vfill}
 \page\colophon
-% Blame: doc@cygnus.com, 28mar91.
+% Blame: doc@@cygnus.com, 28mar91.
 @end tex
 
 @bye
