diff -wbBur ipguard-1.02/Makefile ipguard-1.02.my/Makefile
--- ipguard-1.02/Makefile	2008-11-24 15:33:11.000000000 +0300
+++ ipguard-1.02.my/Makefile	2008-12-11 17:23:07.000000000 +0300
@@ -9,16 +9,16 @@
 ETHERS?=/etc/ethers
 
 ## FreeBSD
-LOCALBASE=/usr/local
-LIBNET_CONFIG=libnet11-config
+# LOCALBASE=/usr/local
+# LIBNET_CONFIG=libnet11-config
 
 ## OpenBSD (tested by irix)
 # LOCALBASE=/usr/local
 # LIBNET_CONFIG=libnet-config-1.1
 
 ## Linux
-# LOCALBASE=/usr
-# LIBNET_CONFIG=libnet-config
+LOCALBASE=/usr
+LIBNET_CONFIG=libnet-config
 
 INCLUDES=`${LIBNET_CONFIG} --cflags`
 LIBS=`${LIBNET_CONFIG} --libs` -lpcap -L${LOCALBASE}/lib
