From d48353b54007a39d8ef9cb53c3d18adf8be9b62c Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sat, 14 May 2016 11:08:46 -0600
Subject: [PATCH] remove ineffectual neon check

---
 host/lib/convert/CMakeLists.txt | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/host/lib/convert/CMakeLists.txt b/host/lib/convert/CMakeLists.txt
index 10376ba..4d793c4 100644
--- a/host/lib/convert/CMakeLists.txt
+++ b/host/lib/convert/CMakeLists.txt
@@ -54,22 +54,6 @@ IF(HAVE_EMMINTRIN_H)
 ENDIF(HAVE_EMMINTRIN_H)
 
 ########################################################################
-# Check for NEON SIMD headers
-########################################################################
-IF(CMAKE_COMPILER_IS_GNUCXX)
-    CHECK_INCLUDE_FILE_CXX(arm_neon.h HAVE_ARM_NEON_H)
-ENDIF(CMAKE_COMPILER_IS_GNUCXX)
-
-IF(HAVE_ARM_NEON_H AND (${CMAKE_SIZEOF_VOID_P} EQUAL 4))
-    ENABLE_LANGUAGE(ASM)
-
-    LIBUHD_APPEND_SOURCES(
-        ${CMAKE_CURRENT_SOURCE_DIR}/convert_with_neon.cpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/convert_neon.S
-    )
-ENDIF()
-
-########################################################################
 # Convert types generation
 ########################################################################
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
-- 
2.8.2

