diff -ru encfs-1.5.orig/encfs/NameIO.cpp encfs-1.5/encfs/NameIO.cpp
--- encfs-1.5.orig/encfs/NameIO.cpp	2009-03-03 14:56:44.000000000 +0000
+++ encfs-1.5/encfs/NameIO.cpp	2009-03-03 14:57:02.000000000 +0000
@@ -190,7 +190,7 @@
 	} else
 	{
 	    bool isDotFile = (*path == '.');
-	    char *next = strchr( path, '/' );
+	    const char *next = strchr( path, '/' );
 	    int len = next ? next - path : strlen( path );
 
 	    // at this point we know that len > 0
