diff -ur source.old/texk/am/script_links.am source/texk/am/script_links.am
--- source.old/texk/am/script_links.am	2011-05-02 13:54:33.000000000 +0200
+++ source/texk/am/script_links.am	2011-06-05 11:22:53.118216353 +0200
@@ -51,8 +51,8 @@
 # We support both multiplatform and non-multiplatform builds.
 install-links:
 	case "$(bindir)" in \
-	  */bin) $(MAKE) $(AM_MAKEFLAGS) REL=.. TYPE=$(TYPE) EXT=$(EXT) make-links;; \
-	  */bin/*) $(MAKE) $(AM_MAKEFLAGS) REL=../.. TYPE=$(TYPE) EXT=$(EXT) make-links;; \
+	  */bin) $(MAKE) $(AM_MAKEFLAGS) REL=$(datadir) TYPE=$(TYPE) EXT=$(EXT) make-links;; \
+	  */bin/*) $(MAKE) $(AM_MAKEFLAGS) REL=$(datadir) TYPE=$(TYPE) EXT=$(EXT) make-links;; \
 	  *) echo "strange directory '$(bindir)' for linked $(TYPE) scripts" >&2; \
 	     exit 1;; \
 	esac
diff -ur source.old/texk/tetex/Makefile.am source/texk/tetex/Makefile.am
--- source.old/texk/tetex/Makefile.am	2011-05-31 11:47:04.000000000 +0200
+++ source/texk/tetex/Makefile.am	2011-06-05 10:28:40.287025375 +0200
@@ -35,7 +35,7 @@
 
 include $(srcdir)/../am/script_links.am
 
-texmfdir = ${prefix}/$(scriptsdir)
+texmfdir = $(datadir)/$(scriptsdir)
 dist_texmf_SCRIPTS = $(perl_scripts:=.pl) $(shell_scripts:=.sh)
 
 if !WIN32
diff -ur source.old/texk/texlive/linked_scripts/Makefile.am source/texk/texlive/linked_scripts/Makefile.am
--- source.old/texk/texlive/linked_scripts/Makefile.am	2011-05-31 11:47:04.000000000 +0200
+++ source/texk/texlive/linked_scripts/Makefile.am	2011-06-05 09:41:50.997250443 +0200
@@ -165,8 +165,8 @@
 	done
 else !WIN32
 	case "$(bindir)" in \
-	  */bin) $(MAKE) $(AM_MAKEFLAGS) REL=.. install-links;; \
-	  */bin/*) $(MAKE) $(AM_MAKEFLAGS) REL=../.. install-links;; \
+	  */bin) $(MAKE) $(AM_MAKEFLAGS) REL=$(datadir) install-links;; \
+	  */bin/*) $(MAKE) $(AM_MAKEFLAGS) REL=$(datadir) install-links;; \
 	  *) echo "strange directory '$(bindir)' for linked scripts" >&2; \
 	     exit 1;; \
 	esac
