diff --git a/texk/texlive/linked_scripts/context/stubs/unix/mtxrun b/texk/texlive/linked_scripts/context/stubs/unix/mtxrun
index 6fc17adb0..722e9fc8f 100755
--- a/texk/texlive/linked_scripts/context/stubs/unix/mtxrun
+++ b/texk/texlive/linked_scripts/context/stubs/unix/mtxrun
@@ -25606,7 +25606,7 @@ local function secondaryflags(arguments)
  end
  return concat(flags," ")
 end
-local template=[[--ini %primaryflags% --lua=%luafile% %texfile% %secondaryflags% %redirect%]]
+local template=[[--ini %primaryflags% --socket --shell-escape --lua=%luafile% %texfile% %secondaryflags% %redirect%]]
 local checkers={
  primaryflags="verbose",
  secondaryflags="verbose",
@@ -25741,7 +25741,7 @@ function environment.make_format(formatname)
  report_format()
  lfs.chdir(startupdir)
 end
-local template=[[%primaryflags% --fmt=%fmtfile% --lua=%luafile% %texfile% %secondaryflags%]]
+local template=[[%primaryflags% --socket --shell-escape --fmt=%fmtfile% --lua=%luafile% %texfile% %secondaryflags%]]
 local checkers={
  primaryflags="verbose",
  secondaryflags="verbose",
