--- source/texk/web2c/luatexdir/luafontloader/fontforge/fontforge/noprefs.c.orig	2009-09-30 19:27:00.935039526 +0200
+++ source/texk/web2c/luatexdir/luafontloader/fontforge/fontforge/noprefs.c	2009-09-30 19:28:04.908364646 +0200
@@ -495,6 +495,9 @@
 	{ "Big5HKSCS", e_big5hkscs },
 	{ "Big5-HKSCS", e_big5hkscs },
 	{ "UTF-8", e_utf8 },
+	{ "utf-8", e_utf8 },
+	{ "UTF8", e_utf8 },
+	{ "utf8", e_utf8 },
 	{ "ISO-10646/UTF-8", e_utf8 },
 	{ "ISO_10646/UTF-8", e_utf8 },
 	{ "UCS2", e_unicode },
