--- source/texk/web2c/luatexdir/lua/lepdflib.cc.orig	2011-10-06 06:14:20.000000000 +0000
+++ source/texk/web2c/luatexdir/lua/lepdflib.cc	2012-05-17 11:30:26.327236883 +0000
@@ -1433,7 +1433,7 @@
 
 static int m_Object_getTypeName(lua_State * L)
 {
-    char *s;
+    const char *s;
     udstruct *uin;
     uin = (udstruct *) luaL_checkudata(L, 1, M_Object);
     if (uin->pd != NULL && uin->pd->pc != uin->pc)
--- source/texk/web2c/pdftexdir/pdftosrc.cc.orig	2012-06-24 17:12:34.376329435 +0200
+++ source/texk/web2c/pdftexdir/pdftosrc.cc	2012-06-24 17:12:25.490003152 +0200
@@ -134,7 +134,7 @@
         exit(1);
     }
     if (extract_xref_table) {
-        int size = xref->getSize();
+        int size = xref->getNumObjects();
         int i;
         for (i = 0; i < size; i++) {
             if (xref->getEntry(i)->offset == 0xffffffff)
