diff --git a/texk/ptexenc/ptexenc.c b/texk/ptexenc/ptexenc.c
index 4e81dbee7c..3c98d68125 100644
--- a/texk/ptexenc/ptexenc.c
+++ b/texk/ptexenc/ptexenc.c
@@ -1044,7 +1044,9 @@ long input_line2(FILE *fp, unsigned char *buff, unsigned char *buff2,
             enc = ptenc_guess_enc(fp);
             if (string_to_enc(enc) > 0) {
                 infile_enc[fd] = string_to_enc(enc);
+#ifdef DEBUG
                 fprintf(stderr, "(guessed encoding #%d: %s = %s)", fd, enc, enc_to_string(infile_enc[fd]));
+#endif /* DEBUG */
             } else {
                 infile_enc[fd] = get_file_enc();
             }
