--- source.orig/texk/web2c/synctexdir/am/synctex.am	2016-05-08 19:08:26.231570952 +0200
+++ source/texk/web2c/synctexdir/am/synctex.am	2018-05-27 00:14:26.011125669 +0200
@@ -55,7 +55,8 @@
 if SYNCTEX
 syncinclude_HEADERS = \
 	synctexdir/synctex_parser.h \
-	synctexdir/synctex_parser_utils.h
+	synctexdir/synctex_parser_utils.h \
+	synctexdir/synctex_version.h
 
 pkgconfig_DATA = synctexdir/synctex.pc
 endif SYNCTEX
