diff -ur source/texk/web2c/luatexdir/image/pdftoepdf.w source.new/texk/web2c/luatexdir/image/pdftoepdf.w
--- source/texk/web2c/luatexdir/image/pdftoepdf.w	2018-04-23 08:46:00.031575505 +0200
+++ source.new/texk/web2c/luatexdir/image/pdftoepdf.w	2018-04-23 08:42:06.762986242 +0200
@@ -357,9 +357,9 @@
     pdf->cave = true;
 }
 
-static void copyString(PDF pdf, GooString * string)
+static void copyString(PDF pdf, const GooString * string)
 {
-    char *p;
+    const char *p;
     unsigned char c;
     size_t i, l;
     p = string->getCString();
@@ -389,7 +389,7 @@
     pdf->cave = true;
 }
 
-static void copyName(PDF pdf, char *s)
+static void copyName(PDF pdf, const char *s)
 {
     pdf_out(pdf, '/');
     for (; *s != 0; s++) {
diff -ur source/texk/web2c/luatexdir/lua/lepdflib.cc source.new/texk/web2c/luatexdir/lua/lepdflib.cc
--- source/texk/web2c/luatexdir/lua/lepdflib.cc	2018-04-23 08:46:00.032575486 +0200
+++ source.new/texk/web2c/luatexdir/lua/lepdflib.cc	2018-04-23 08:42:06.762986242 +0200
@@ -516,7 +516,7 @@
 #define m_poppler_get_GOOSTRING(in, function)                  \
 static int m_##in##_##function(lua_State * L)                  \
 {                                                              \
-    GooString *gs;                                             \
+    const GooString *gs;                                             \
     udstruct *uin;                                             \
     uin = (udstruct *) luaL_checkudata(L, 1, M_##in);          \
     if (uin->pd != NULL && uin->pd->pc != uin->pc)             \
@@ -1650,7 +1650,7 @@
 
 static int m_Object_getString(lua_State * L)
 {
-    GooString *gs;
+    const GooString *gs;
     udstruct *uin;
     uin = (udstruct *) luaL_checkudata(L, 1, M_Object);
     if (uin->pd != NULL && uin->pd->pc != uin->pc)
diff -ur source/texk/web2c/pdftexdir/pdftoepdf-newpoppler.cc source.new/texk/web2c/pdftexdir/pdftoepdf-newpoppler.cc
--- source/texk/web2c/pdftexdir/pdftoepdf-newpoppler.cc	2018-04-23 21:27:36.254598208 +0200
+++ source.new/texk/web2c/pdftexdir/pdftoepdf-newpoppler.cc	2018-04-23 21:17:36.475663912 +0200
@@ -275,7 +275,7 @@
 
 static void copyObject(Object *);
 
-static void copyName(char *s)
+static void copyName(const char *s)
 {
     pdf_puts("/");
     for (; *s != 0; s++) {
@@ -418,7 +418,7 @@
         && fontdescRef.isRef()
         && fontdesc.isDict()
         && embeddableFont(&fontdesc)
-        && (fontmap = lookup_fontmap(basefont.getName())) != NULL) {
+        && (fontmap = lookup_fontmap((char *)basefont.getName())) != NULL) {
         // round /StemV value, since the PDF input is a float
         // (see Font Descriptors in PDF reference), but we only store an
         // integer, since we don't want to change the struct.
@@ -427,7 +427,7 @@
         charset = fontdesc.dictLookup("CharSet");
         if (!charset.isNull() &&
             charset.isString() && is_subsetable(fontmap))
-            epdf_mark_glyphs(fd, charset.getString()->getCString());
+            epdf_mark_glyphs(fd, (char *)charset.getString()->getCString());
         else
             embed_whole_font(fd);
         addFontDesc(fontdescRef.getRef(), fd);
@@ -554,8 +554,8 @@
     Object obj1;
     int i, l, c;
     Ref ref;
-    char *p;
-    GString *s;
+    const char *p;
+    const GString *s;
     if (obj->isBool()) {
         pdf_printf("%s", obj->getBool()? "true" : "false");
     } else if (obj->isInt()) {
diff -ur source/texk/web2c/pdftexdir/pdftosrc-newpoppler.cc source.new/texk/web2c/pdftexdir/pdftosrc-newpoppler.cc
--- source/texk/web2c/pdftexdir/pdftosrc-newpoppler.cc	2018-04-23 21:27:36.254598208 +0200
+++ source.new/texk/web2c/pdftexdir/pdftosrc-newpoppler.cc	2018-04-23 08:42:06.762986242 +0200
@@ -68,7 +68,7 @@
     Stream *s;
     Object srcStream, srcName, catalogDict;
     FILE *outfile;
-    char *outname;
+    const char *outname;
     int objnum = 0, objgen = 0;
     bool extract_xref_table = false;
     int c;

