--- a/texk/web2c/pdftexdir/pdftoepdf.cc	2020-03-03 21:11:35.102711802 +0000
+++ b/texk/web2c/pdftexdir/pdftoepdf.cc	2020-03-03 21:13:13.057420111 +0000
@@ -757,7 +757,7 @@
     if (page_name) {
         // get page by name
         GString name(page_name);
-        LinkDest *link = pdf_doc->doc->findDest(&name);
+        LinkDest *link = pdf_doc->doc->findDest(&name).get();
         if (link == 0 || !link->isOk())
             pdftex_fail("PDF inclusion: invalid destination <%s>", page_name);
         Ref ref = link->getPageRef();
