diff -ru texlive-source-orig/texk/web2c/pdftexdir/pdftoepdf.cc texlive-source/texk/web2c/pdftexdir/pdftoepdf.cc
--- texlive-source-orig/texk/web2c/pdftexdir/pdftoepdf.cc	2020-01-07 22:42:03.717667766 +0000
+++ texlive-source/texk/web2c/pdftexdir/pdftoepdf.cc	2020-01-07 22:47:42.818888622 +0000
@@ -723,7 +723,7 @@
 #endif
     // initialize
     if (!isInit) {
-        globalParams = new GlobalParams();
+        globalParams.reset(new GlobalParams());
         globalParams->setErrQuiet(false);
         isInit = true;
     }
@@ -1108,6 +1108,5 @@
             delete_document(p);
         }
         // see above for globalParams
-        delete globalParams;
     }
 }
diff -ru texlive-source-orig/texk/web2c/pdftexdir/pdftosrc.cc texlive-source/texk/web2c/pdftexdir/pdftosrc.cc
--- texlive-source-orig/texk/web2c/pdftexdir/pdftosrc.cc	2020-01-07 22:42:03.714334387 +0000
+++ texlive-source/texk/web2c/pdftexdir/pdftosrc.cc	2020-01-07 22:50:08.070708640 +0000
@@ -79,7 +79,7 @@
         exit(1);
     }
     fileName = new GString(argv[1]);
-    globalParams = new GlobalParams();
+    globalParams.reset(new GlobalParams());
     doc = new PDFDoc(fileName);
     if (!doc->isOk()) {
         fprintf(stderr, "Invalid PDF file\n");
@@ -100,7 +100,7 @@
     if (objnum == 0) {
         srcStream = catalogDict.dictLookup("SourceObject");
         static char const_SourceFile[] = "SourceFile";
-        if (!srcStream.isStream(const_SourceFile)) {
+        if (!srcStream.isDict(const_SourceFile)) {
             fprintf(stderr, "No SourceObject found\n");
             exit(1);
         }
@@ -202,5 +202,4 @@
         fprintf(stderr, "Cross-reference table extracted to %s\n", outname);
     fclose(outfile);
     delete doc;
-    delete globalParams;
 }
