--- documentview.cpp   2012-10-30 14:17:33.000000000 +0000
+++ documentview.cpp   2012-11-22 23:15:58.901398313 +0000
@@ -673,12 +673,12 @@
     {
         PageItem* page = m_pages.at(m_currentPage - 1);
 
-        setScaleFactor(qMin(page->scaleFactor() + 0.1, s_maximumScaleFactor));
+        setScaleFactor(qMin(qreal(page->scaleFactor() + 0.1), s_maximumScaleFactor));
         setScaleMode(ScaleFactor);
     }
     else
     {
-        setScaleFactor(qMin(scaleFactor() + 0.1, s_maximumScaleFactor));
+        setScaleFactor(qMin(qreal(scaleFactor() + 0.1), s_maximumScaleFactor));
     }
 }
 
@@ -689,12 +689,12 @@
     {
         PageItem* page = m_pages.at(m_currentPage - 1);
 
-        setScaleFactor(qMax(page->scaleFactor() - 0.1, s_minimumScaleFactor));
+        setScaleFactor(qMax(qreal(page->scaleFactor() - 0.1), s_minimumScaleFactor));
         setScaleMode(ScaleFactor);
     }
     else
     {
-        setScaleFactor(qMax(scaleFactor() - 0.1, s_minimumScaleFactor));
+        setScaleFactor(qMax(qreal(scaleFactor() - 0.1), s_minimumScaleFactor));
     }
 }
 
@@ -1342,7 +1342,7 @@
 
                 pageHeight = boundingRect.height();
 
-                left = qMin(left, -boundingRect.width() - 1.5 * s_pageSpacing);
+                left = qMin(left, qreal(-boundingRect.width() - 1.5 * s_pageSpacing));
             }
             else
             {
@@ -1350,7 +1350,7 @@
 
                 pageHeight = qMax(pageHeight, boundingRect.height());
 
-                right = qMax(right, boundingRect.width() + 1.5 * s_pageSpacing);
+                right = qMax(right, qreal(boundingRect.width() + 1.5 * s_pageSpacing));
                 height += pageHeight + s_pageSpacing;
             }
         }
@@ -1362,15 +1362,15 @@
 
             pageHeight = boundingRect.height();
 
-            left = qMin(left, -0.5 * boundingRect.width() - s_pageSpacing);
-            right = qMax(right, 0.5 * boundingRect.width() + s_pageSpacing);
+            left = qMin(left, qreal(-0.5 * boundingRect.width() - s_pageSpacing));
+            right = qMax(right, qreal(0.5 * boundingRect.width() + s_pageSpacing));
             height += pageHeight + s_pageSpacing;
         }
     }
 
     if(m_twoPagesMode && m_numberOfPages % 2 != 0)
     {
-        right = qMax(right, 0.5 * s_pageSpacing);
+        right = qMax(right, qreal(0.5 * s_pageSpacing));
         height += pageHeight + s_pageSpacing;
     }
 
@@ -1424,7 +1424,7 @@
                 QRectF boundingRect = page->boundingRect().translated(page->pos());
 
                 top = qMin(top, boundingRect.top() - s_pageSpacing);
-                height = qMax(height, boundingRect.height() + 2.0 * s_pageSpacing);
+                height = qMax(height, qreal(boundingRect.height() + 2.0 * s_pageSpacing));
             }
             else
             {
