diff --git a/src/ldc/intrinsics.di b/src/ldc/intrinsics.di
index f5d2c207..4502ac26 100644
--- a/src/ldc/intrinsics.di
+++ b/src/ldc/intrinsics.di
@@ -26,6 +26,7 @@ else version (LDC_LLVM_800)  enum LLVM_version =  800;
 else version (LDC_LLVM_900)  enum LLVM_version =  900;
 else version (LDC_LLVM_1000) enum LLVM_version = 1000;
 else version (LDC_LLVM_1100) enum LLVM_version = 1100;
+else version (LDC_LLVM_1101) enum LLVM_version = 1101;
 else static assert(false, "LDC LLVM version not supported");
 
 enum LLVM_atleast(int major) = (LLVM_version >= major * 100);
