diff -ubr scummvm-0.13.1.orig/backends/midi/alsa.cpp scummvm-0.13.1/backends/midi/alsa.cpp
--- scummvm-0.13.1.orig/backends/midi/alsa.cpp	2009-04-18 23:32:38.000000000 +0200
+++ scummvm-0.13.1/backends/midi/alsa.cpp	2009-06-11 09:23:01.000000000 +0200
@@ -228,7 +228,7 @@
 }
 
 int MidiDriver_ALSA::parse_addr(const char *arg, int *client, int *port) {
-	char *p;
+	const char *p;
 
 	if (isdigit(*arg)) {
 		if ((p = strpbrk(arg, ADDR_DELIM)) == NULL)
