diff -ur gnokii-0.6.7-old/Docs/sample/gnokiirc gnokii-0.6.7/Docs/sample/gnokiirc
--- gnokii-0.6.7-old/Docs/sample/gnokiirc	2005-05-16 20:55:13.000000000 +0200
+++ gnokii-0.6.7/Docs/sample/gnokiirc	2005-06-03 10:49:42.000000000 +0200
@@ -112,7 +112,7 @@
 # permissions 4750, owned by root, group gnokii.  Ensure you
 # are in the gnokii group and that the group exists...
 [gnokiid]
-bindir = /usr/local/sbin/
+bindir = /usr/sbin/
 
 # Any entries in the following two sections will be set as environment
 # variables when running the scripts.
