--- m4/optimizations.m4.orig	2005-11-27 01:20:08.000000000 +0100
+++ m4/optimizations.m4	2005-11-27 01:22:56.000000000 +0100
@@ -161,10 +161,6 @@
                   ;;
 
               esac
-              if test x"$archopt_val" != x; then
-                  CFLAGS="$sarchopt=$archopt_val $CFLAGS"
-                  DEBUG_CFLAGS="$sarchopt=$archopt_val $DEBUG_CFLAGS"
-              fi
             fi
           else
             dnl we have the Intel compiler
