diff -ruN -x .gitignore -x .cvsignore xf86-input-magellan-90955ce39a01d9dd649b62952ed3fa2f1e63af66/README xf86-input-magellan-fb553dd39e4066198ef93c9227edf496d874f2c6/README
--- xf86-input-magellan-90955ce39a01d9dd649b62952ed3fa2f1e63af66/README	1970-01-01 00:00:00.000000000 +0000
+++ xf86-input-magellan-fb553dd39e4066198ef93c9227edf496d874f2c6/README	2009-01-31 04:24:21.000000000 +0000
@@ -0,0 +1,19 @@
+
+Please submit bugs & patches to the Xorg bugzilla:
+
+        https://bugs.freedesktop.org/enter_bug.cgi?product=xorg
+
+All questions regarding this software should be directed at the
+Xorg mailing list:
+
+        http://lists.freedesktop.org/mailman/listinfo/xorg
+
+The master development code repository can be found at:
+
+        git://anongit.freedesktop.org/git/xorg/driver/xf86-input-magellan
+
+        http://cgit.freedesktop.org/xorg/driver/xf86-input-magellan
+
+For more information on the git code manager, see:
+
+        http://wiki.x.org/wiki/GitPage
diff -ruN -x .gitignore -x .cvsignore xf86-input-magellan-90955ce39a01d9dd649b62952ed3fa2f1e63af66/src/magellan.c xf86-input-magellan-fb553dd39e4066198ef93c9227edf496d874f2c6/src/magellan.c
--- xf86-input-magellan-90955ce39a01d9dd649b62952ed3fa2f1e63af66/src/magellan.c	2008-03-20 20:11:43.000000000 +0000
+++ xf86-input-magellan-fb553dd39e4066198ef93c9227edf496d874f2c6/src/magellan.c	2009-01-31 04:24:21.000000000 +0000
@@ -45,7 +45,9 @@
 
 #include <misc.h>
 #include <xf86.h>
-#define NEED_XF86_TYPES
+#ifndef NEED_XF86_TYPES
+#define NEED_XF86_TYPES 1
+#endif
 #include <xf86_OSproc.h>
 #include <xf86Xinput.h>
 #include <xisb.h>
@@ -120,7 +122,9 @@
         "xf86CollectInputOptions",
         "xf86ErrorFVerb",
         "xf86FindOptionValue",
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
         "xf86GetMotionEvents",
+#endif
         "xf86GetVerbosity",
         "xf86MotionHistoryAllocate",
         "xf86NameCmp",
@@ -364,7 +368,10 @@
       return !Success;
     }
 
-  if (InitValuatorClassDeviceStruct (dev, 6, xf86GetMotionEvents,
+  if (InitValuatorClassDeviceStruct (dev, 6,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
+				     xf86GetMotionEvents,
+#endif
 				     pInfo->history_size, Absolute) == FALSE)
     {
       ErrorF ("Unable to allocate MAGELLAN ValuatorClassDeviceStruct\n");
