--- PyQt-x11-gpl-4.9/configure.py~	2011-12-23 14:53:07.981490550 +0100
+++ PyQt-x11-gpl-4.9/configure.py	2011-12-23 14:52:38.384984221 +0100
@@ -1264,7 +1264,7 @@
     opengl = (mname == "QtOpenGL")

     qt = [mname]
-    if mname in ("QtOpenGL", "QtWebKit"):
+    if mname in ("QtHelp", "QtOpenGL", "QtWebKit"):
         qt.append("QtCore")

     makefile = sipconfig.ProgramMakefile(sipcfg, console=1, qt=qt, warnings=0,
