--- PyQt-x11-gpl-4.9.1/configure.py~	2012-02-13 20:16:11.418786488 +0000
+++ PyQt-x11-gpl-4.9.1/configure.py	2012-02-13 20:16:50.561809527 +0000
@@ -1261,7 +1261,7 @@
     opengl = (mname == "QtOpenGL")
 
     qt = [mname]
-    if mname in ("QtOpenGL", "QtWebKit"):
+    if mname in ("QtHelp", "QtOpenGL", "QtWebKit"):
         qt.append("QtCore")
 
     makefile = sipconfig.ProgramMakefile(sipcfg, console=1, qt=qt, warnings=0,
