From f128aae113cd9d4b98bdfcb19defa433ef4fc244 Mon Sep 17 00:00:00 2001
From: David Runge <dvzrv@archlinux.org>
Date: Sun, 24 May 2026 20:05:19 +0200
Subject: [PATCH 1/4] Use reproducible data for build user and host

Signed-off-by: David Runge <dvzrv@archlinux.org>
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 0a898d1c76..ae92643e8c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4403,11 +4403,11 @@ AC_SUBST(VERSION_REVISION)
 AC_SUBST(VERSION_EXTRA)
 AC_SUBST(COPYRIGHT_YEARS)
 AS_IF([test -z "$VLC_COMPILE_BY"],[
-   VLC_COMPILE_BY=`whoami|sed -e 's/\\\/\\\\\\\/g'`
+   VLC_COMPILE_BY=`echo builduser|sed -e 's/\\\/\\\\\\\/g'`
 ])
 AC_DEFINE_UNQUOTED([VLC_COMPILE_BY], "$VLC_COMPILE_BY", [user who ran configure])
 AS_IF([test -z "$VLC_COMPILE_HOST"],[
-   VLC_COMPILE_HOST=`hostname -f 2>/dev/null || hostname`
+   VLC_COMPILE_HOST=`echo buildhost 2>/dev/null || hostname`
 ])
 AC_DEFINE_UNQUOTED([VLC_COMPILE_HOST], "$VLC_COMPILE_HOST", [host which ran configure])
 AC_DEFINE_UNQUOTED(VLC_COMPILER, "`$CC -v 2>&1 | tail -n 1 | sed -e 's/ *$//'`", [compiler])
