--- a/modules/video_output/caca.c
+++ b/modules/video_output/caca.c
@@ -160,7 +160,11 @@ static int Open(vlc_object_t *object)
     }
 
     const char *driver = NULL;
-#ifdef __APPLE__
+// RPI: If driver is NULL then if we have X but DISPLAY is unset then somehow
+// the GL module becomes unloaded without anything noticing and that then
+// causes a segfault.
+//#ifdef __APPLE__
+#if 1
     // Make sure we don't try to open a window.
     driver = "ncurses";
 #endif
