diff -upr OpenShadingLanguage-Release-1.5.12.orig/src/cmake/externalpackages.cmake OpenShadingLanguage-Release-1.5.12/src/cmake/externalpackages.cmake
--- OpenShadingLanguage-Release-1.5.12.orig/src/cmake/externalpackages.cmake	2014-12-26 22:22:53.000000000 +0200
+++ OpenShadingLanguage-Release-1.5.12/src/cmake/externalpackages.cmake	2015-03-16 03:12:56.052479730 +0200
@@ -211,9 +211,10 @@ if ((LLVM_LIBRARY OR LLVM_STATIC) AND LL
     # if static LLVM libraries were requested, use llvm-config to generate
     # the list of what libraries we need, and substitute that in the right
     # way for LLVM_LIBRARY.
-    execute_process (COMMAND ${LLVM_CONFIG} --libfiles
+    execute_process (COMMAND ${LLVM_CONFIG} --libfiles --system-libs
                      OUTPUT_VARIABLE LLVM_LIBRARY
                      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    string (REPLACE "\n" " " LLVM_LIBRARY ${LLVM_LIBRARY})
     string (REPLACE " " ";" LLVM_LIBRARY ${LLVM_LIBRARY})
   endif ()
   if (VERBOSE)
