diff -up ./src/SerialImp.h.lock ./src/SerialImp.h
--- ./src/SerialImp.h.lock	2010-11-09 15:59:54.891179571 +0100
+++ ./src/SerialImp.h	2010-11-09 16:01:12.257819409 +0100
@@ -158,7 +158,7 @@ struct event_info_struct
 #	endif /* CMSPAR */
 #	
 #	define DEVICEDIR "/dev/"
-#	define LOCKDIR "/var/lock"
+#	define LOCKDIR "/var/lock/lockdev"
 #	define LOCKFILEPREFIX "LCK.."
 #	define FHS
 #endif /* __linux__ */
