--- freeglut-2.4.0.orig/src/freeglut_cursor.c
+++ freeglut-2.4.0/src/freeglut_cursor.c
@@ -130,7 +130,9 @@
                     XCreateFontCursor( fgDisplay.Display, entry->cursorShape );
             }
             cursor = entry->cachedCursor;
-        } else {
+	    if (cursor == None)
+		fgError( "Failed to create cursor" );
+	} else {
             switch( cursorIDToUse )
             {
             case GLUT_CURSOR_NONE:
@@ -147,9 +149,6 @@
             }
         }
 
-        if ( ( cursorIDToUse != GLUT_CURSOR_NONE ) && ( cursor == None ) ) {
-            fgError( "Failed to create cursor" );
-        }
         XDefineCursor( fgDisplay.Display,
                        window->Window.Handle, cursor );
     }
