From cf823a1df66cc258afe1f2c870b4cbd182489f97 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 25 Feb 2018 12:28:01 -0700
Subject: [PATCH] ATF toolchain fixes

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 5b67763e..5571a9c0 100644
--- a/Makefile
+++ b/Makefile
@@ -246,7 +246,7 @@ TF_CFLAGS		+= 	-nostdinc -ffreestanding -Wall			\
 				-std=c99 -c -Os					\
 				$(TF_CFLAGS_$(ARCH))				\
 				${DEFINES} ${INCLUDES}
-TF_CFLAGS		+=	-ffunction-sections -fdata-sections
+TF_CFLAGS		+=	-ffunction-sections -fdata-sections -fno-stack-protector -fno-pie
 
 LDFLAGS			+=	--fatal-warnings -O1
 LDFLAGS			+=	--gc-sections
-- 
2.14.1

