From bacc2b70d8b7258bb4c07c8ac1c490c604d6efb4 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 25 Feb 2018 12:28:30 -0700
Subject: [PATCH] DDR toolchain fixes

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 7c70d75..83bdb5e 100755
--- a/Makefile
+++ b/Makefile
@@ -253,7 +253,7 @@ INCLUDE += $(PLAT_INCLUDES)
 MV_DDR_LIBNAME = mv_ddr_lib.a
 MV_DDR_LIB = $(OBJ_DIR)/$(MV_DDR_LIBNAME)
 
-CFLAGS = -Wall -Werror -Os -ffreestanding -mlittle-endian -g -gdwarf-2 -nostdinc
+CFLAGS = -Wall -Werror -Os -ffreestanding -mlittle-endian -g -gdwarf-2 -nostdinc -fno-stack-protector -fno-pie
 # PLATFORM is set in ble/ble.mk
 CFLAGS += -march=armv8-a -fpie $(INCLUDE)
 CFLAGS += -DMV_DDR_ATF -DCONFIG_APN806
-- 
2.14.1

