diff -ur gnome-alsamixer-0.9.6/src/Makefile.in gnome-alsamixer-0.9.6.new/src/Makefile.in
--- gnome-alsamixer-0.9.6/src/Makefile.in	2003-09-09 12:44:25.000000000 +0100
+++ gnome-alsamixer-0.9.6.new/src/Makefile.in	2004-04-04 18:51:36.000000000 +0100
@@ -86,10 +86,6 @@
 	-DPACKAGE_DATA_DIR=\""$(datadir)/gnome-alsamixer"\"        \
 	-DPIXMAP_ICONDIR=\""$(datadir)/pixmaps/gnome-alsamixer"\"  \
 	-DPACKAGE_LOCALE_DIR=\""$(prefix)/$(DATADIRNAME)/locale"\" \
-	-DG_DISABLE_DEPRECATED					   \
-	-DGDK_DISABLE_DEPRECATED				   \
-	-DGTK_DISABLE_DEPRECATED				   \
-	-DGNOME_DISABLE_DEPRECATED                                 \
 	@PACKAGE_CFLAGS@
 
 
Only in gnome-alsamixer-0.9.6.new/src: Makefile.in~
