Index: apps/dolphin/src/panels/terminal/terminalpanel.cpp
===================================================================
--- apps/dolphin/src/panels/terminal/terminalpanel.cpp	(Revision 1032179)
+++ apps/dolphin/src/panels/terminal/terminalpanel.cpp	(Revision 1032180)
@@ -23,7 +23,6 @@
 #include <kde_terminal_interface_v2.h>
 #include <kparts/part.h>
 #include <kshell.h>
-#include <kio/netaccess.h>
 
 #include <QBoxLayout>
 #include <QShowEvent>
@@ -58,13 +57,12 @@
     }
 
     Panel::setUrl(url);
-    KUrl mostLocalUrl = KIO::NetAccess::mostLocalUrl(url, 0);
     const bool sendInput = (m_terminal != 0)
                            && (m_terminal->foregroundProcessId() == -1)
                            && isVisible()
-                           && mostLocalUrl.isLocalFile();
+                           && url.isLocalFile();
     if (sendInput) {
-        m_terminal->sendInput("cd " + KShell::quoteArg(mostLocalUrl.toLocalFile()) + '\n');
+        m_terminal->sendInput("cd " + KShell::quoteArg(url.toLocalFile()) + '\n');
     }
 
 }
