From aa67bb2825b9c1b8e09c07b69c723c6ed5d3767e Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Mon, 31 Dec 2018 10:03:22 -0700
Subject: [PATCH] fix LF_PINS padding calculation

---
 include/lf.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/lf.h b/include/lf.h
index 973b84cf..9c893c3a 100644
--- a/include/lf.h
+++ b/include/lf.h
@@ -88,7 +88,7 @@ struct LF_PINS {
   std::atomic<uint64> link;
   /* we want sizeof(LF_PINS) to be 64 to avoid false sharing */
 #if 2 * 8 + SIZEOF_CHARP * (LF_PINBOX_PINS + 2) != 64
-  char pad[64 - sizeof(uint32) * 2 - sizeof(void *) * (LF_PINBOX_PINS + 2)];
+  char pad[64 - sizeof(uint64) * 2 - sizeof(void *) * (LF_PINBOX_PINS + 2)];
 #endif
 };
 
-- 
2.19.2

