diff -ru percona-server-8.3.0-1.orig/components/audit_log_filter/event_filter_function/string_find.cc percona-server-8.3.0-1/components/audit_log_filter/event_filter_function/string_find.cc
--- percona-server-8.3.0-1.orig/components/audit_log_filter/event_filter_function/string_find.cc	2024-05-12 10:09:38.860441107 +0200
+++ percona-server-8.3.0-1/components/audit_log_filter/event_filter_function/string_find.cc	2024-05-12 10:10:01.817428593 +0200
@@ -16,6 +16,7 @@
 #include "components/audit_log_filter/event_filter_function/string_find.h"
 
 #include <functional>
+#include <algorithm>
 
 namespace audit_log_filter::event_filter_function {
 
