diff --git a/cmake/zlib.cmake b/cmake/zlib.cmake
index 460d87a..4b98b68 100644
--- a/cmake/zlib.cmake
+++ b/cmake/zlib.cmake
@@ -32,7 +32,7 @@
 SET(MIN_ZLIB_VERSION_REQUIRED "1.2.13")
 
 
-FUNCTION(FIND_ZLIB_VERSION ZLIB_INCLUDE_DIR)
+MACRO(FIND_ZLIB_VERSION ZLIB_INCLUDE_DIR)
   FOREACH(version_part
       ZLIB_VER_MAJOR
       ZLIB_VER_MINOR
@@ -48,9 +48,9 @@ FUNCTION(FIND_ZLIB_VERSION ZLIB_INCLUDE_DIR)
   SET(ZLIB_VERSION "${ZLIB_VERSION}" CACHE INTERNAL "ZLIB major.minor.step")
   MESSAGE(STATUS "ZLIB_VERSION (${WITH_ZLIB}) is ${ZLIB_VERSION}")
   MESSAGE(STATUS "ZLIB_INCLUDE_DIR ${ZLIB_INCLUDE_DIR}")
-ENDFUNCTION(FIND_ZLIB_VERSION)
+ENDMACRO(FIND_ZLIB_VERSION)
 
-FUNCTION(FIND_SYSTEM_ZLIB)
+MACRO(FIND_SYSTEM_ZLIB)
   FIND_PACKAGE(ZLIB)
   IF(ZLIB_FOUND)
     ADD_LIBRARY(zlib_interface INTERFACE)
@@ -61,7 +61,7 @@ FUNCTION(FIND_SYSTEM_ZLIB)
         ${ZLIB_INCLUDE_DIR})
     ENDIF()
   ENDIF()
-ENDFUNCTION(FIND_SYSTEM_ZLIB)
+ENDMACRO(FIND_SYSTEM_ZLIB)
 
 MACRO (RESET_ZLIB_VARIABLES)
   # Reset whatever FIND_PACKAGE may have left behind.
@@ -114,9 +114,9 @@ MACRO (MYSQL_CHECK_ZLIB)
     MYSQL_USE_BUNDLED_ZLIB()
   ELSEIF(WITH_ZLIB STREQUAL "system")
     FIND_SYSTEM_ZLIB()
-    IF(NOT ZLIB_FOUND)
-      MESSAGE(FATAL_ERROR "Cannot find system zlib libraries.")
-    ENDIF()
+    # IF(NOT ZLIB_FOUND)
+    #   MESSAGE(FATAL_ERROR "Cannot find system zlib libraries.")
+    # ENDIF()
   ELSE()
     RESET_ZLIB_VARIABLES()
     MESSAGE(FATAL_ERROR "WITH_ZLIB must be bundled or system")
diff --git a/components/keyrings/common/data/data.h b/components/keyrings/common/data/data.h
index 6340a26..566e036 100644
--- a/components/keyrings/common/data/data.h
+++ b/components/keyrings/common/data/data.h
@@ -23,6 +23,7 @@
 #ifndef DATA_INCLUDED
 #define DATA_INCLUDED
 
+#include <cstdint>
 #include <functional>
 #include <string>
 #include "pfs_string.h"
diff --git a/mysys/buffered_error_log.cc b/mysys/buffered_error_log.cc
index a8abb63..7f8e39a 100644
--- a/mysys/buffered_error_log.cc
+++ b/mysys/buffered_error_log.cc
@@ -1,5 +1,6 @@
 
 #include "buffered_error_log.h"
+#include <string>
 
 Buffered_error_logger buffered_error_log;
 char *buffered_error_log_filename = nullptr;
diff --git a/storage/rocksdb/rocksdb/table/block_based/data_block_hash_index.h b/storage/rocksdb/rocksdb/table/block_based/data_block_hash_index.h
index f356395..c6651ca 100644
--- a/storage/rocksdb/rocksdb/table/block_based/data_block_hash_index.h
+++ b/storage/rocksdb/rocksdb/table/block_based/data_block_hash_index.h
@@ -5,6 +5,8 @@
 
 #pragma once
 
+#include <stdint.h>
+
 #include <string>
 #include <vector>
 
diff --git a/storage/rocksdb/rocksdb/util/string_util.h b/storage/rocksdb/rocksdb/util/string_util.h
index 55d106f..11178fd 100644
--- a/storage/rocksdb/rocksdb/util/string_util.h
+++ b/storage/rocksdb/rocksdb/util/string_util.h
@@ -6,6 +6,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <sstream>
 #include <string>
 #include <unordered_map>
