--- percona-server-8.0.26-16.orig/plugin/auth_ldap/CMakeLists.txt	2021-10-18 15:11:29.000000000 +0300
+++ percona-server-8.0.26-16/plugin/auth_ldap/CMakeLists.txt	2021-11-13 21:44:21.752981358 +0200
@@ -36,7 +36,7 @@ IF(WITH_LDAP)
 
   # libler?
   MYSQL_ADD_PLUGIN(authentication_ldap_simple ${ALP_SOURCES_SIMPLE}
-    LINK_LIBRARIES ldap_r MODULE_ONLY MODULE_OUTPUT_NAME "authentication_ldap_simple")
+    LINK_LIBRARIES ldap MODULE_ONLY MODULE_OUTPUT_NAME "authentication_ldap_simple")
 
   IF(UNIX)
     IF(INSTALL_MYSQLTESTDIR)
