--- lib/roken/Makefile.am.old	2007-11-05 19:42:53.000000000 +0100
+++ lib/roken/Makefile.am	2007-11-05 19:43:12.000000000 +0100
@@ -107,7 +107,7 @@
 	ifaddrs.hin		\
 	vis.hin	
 
-libroken_la_LIBADD = @LTLIBOBJS@
+libroken_la_LIBADD = @LTLIBOBJS@ $(LIB_crypt)
 
 $(LTLIBOBJS) $(libroken_la_OBJECTS): roken.h $(XHEADERS)
 
--- lib/editline/Makefile.am.old	2007-11-05 19:46:08.000000000 +0100
+++ lib/editline/Makefile.am	2007-11-05 20:06:54.000000000 +0100
@@ -36,6 +36,8 @@
 	unix.h \
 	$(EXTRA_SOURCE)
 
+libeditline_la_LIBADD = $(LIB_tgetent)
+
 EXTRA_SOURCE = $(ES) 
 
 libel_compat_la_SOURCES = edit_compat.c edit_compat.h
