diff --git a/cmake/FindJulia.cmake b/cmake/FindJulia.cmake
index 2d4e1f6a..7f3d7a82 100644
--- a/cmake/FindJulia.cmake
+++ b/cmake/FindJulia.cmake
@@ -61,8 +61,14 @@ endif()
 #
 # Julia library location
 #
+if(${JULIA_VERSION_STRING} VERSION_LESS 0.7.0)
+    set(JULIA_LIBDL_COMMAND "abspath(dirname(Libdl.dlpath(\"libjulia\")))")
+else()
+    set(JULIA_LIBDL_COMMAND "using Libdl; abspath(dirname(Libdl.dlpath(\"libjulia\")))")
+endif()
+
 execute_process(
-    COMMAND ${JULIA_EXECUTABLE} -E "abspath(dirname(Libdl.dlpath(\"libjulia\")))"
+    COMMAND ${JULIA_EXECUTABLE} -E ${JULIA_LIBDL_COMMAND}
     OUTPUT_VARIABLE JULIA_LIBRARY_DIR
     RESULT_VARIABLE RESULT
 )
diff --git a/src/backends/julia/juliaserver/juliaserver.cpp b/src/backends/julia/juliaserver/juliaserver.cpp
index 5369fe50..0a1e8889 100644
--- a/src/backends/julia/juliaserver/juliaserver.cpp
+++ b/src/backends/julia/juliaserver/juliaserver.cpp
@@ -43,7 +43,7 @@ JuliaServer::~JuliaServer()
 
 void JuliaServer::login(const QString &path) const
 {
-#if JULIA_VERSION_MINOR > 5
+#if (JULIA_VERSION_MAJOR > 0 || JULIA_VERSION_MINOR > 5)
     Q_UNUSED(path)
     jl_init();
 #else
