diff --git a/src/backends/julia/juliaserver/juliaserver.cpp b/src/backends/julia/juliaserver/juliaserver.cpp
index c68fd46f..e2995cb7 100644
--- a/src/backends/julia/juliaserver/juliaserver.cpp
+++ b/src/backends/julia/juliaserver/juliaserver.cpp
@@ -91,7 +91,9 @@ void JuliaServer::runJuliaCommand(const QString &command)
 
     if (jl_exception_occurred()) { // If exception occurred
         // Show it to user in stderr
-#if QT_VERSION_CHECK(JULIA_VERSION_MAJOR, JULIA_VERSION_MINOR, 0) >= QT_VERSION_CHECK(1, 1, 0)
+#if QT_VERSION_CHECK(JULIA_VERSION_MAJOR, JULIA_VERSION_MINOR, 0) >= QT_VERSION_CHECK(1, 7, 0)
+        jl_value_t *ex = jl_current_task->ptls->previous_exception;
+#elif QT_VERSION_CHECK(JULIA_VERSION_MAJOR, JULIA_VERSION_MINOR, 0) >= QT_VERSION_CHECK(1, 1, 0)
         jl_value_t *ex = jl_get_ptls_states()->previous_exception;
 #else
         jl_value_t *ex = jl_exception_in_transit;
