diff --git a/src/backends/julia/juliaserver/juliaserver.cpp b/src/backends/julia/juliaserver/juliaserver.cpp
index dc8a48e5..91c2c338 100644
--- a/src/backends/julia/juliaserver/juliaserver.cpp
+++ b/src/backends/julia/juliaserver/juliaserver.cpp
@@ -40,7 +40,7 @@ JuliaServer::~JuliaServer()
 void JuliaServer::login(const QString &path) const
 {
     QString dir_path = QFileInfo(path).dir().absolutePath();
-    jl_init(dir_path.toLatin1().constData());
+    jl_init_with_image(dir_path.toLatin1().constData(), NULL);
 }
 
 void JuliaServer::runJuliaCommand(const QString &command)

