diff --git a/src/backends/sage/sagesession.cpp b/src/backends/sage/sagesession.cpp
index c2973a8a..f23914ad 100644
--- a/src/backends/sage/sagesession.cpp
+++ b/src/backends/sage/sagesession.cpp
@@ -118,7 +118,7 @@ void SageSession::login()
     m_process=new KPtyProcess(this);
     updateSageVersion();
     const QString& sageExecFile = SageSettings::self()->path().toLocalFile();
-    if (m_sageVersion >= SageSession::VersionInfo(8, 3))
+    if (m_sageVersion >= SageSession::VersionInfo(8, 4))
         m_process->setProgram(sageExecFile, QStringList() << QLatin1String("--simple-prompt"));
     else
         {
