diff -urN a/scribus/scribus.cpp b/scribus/scribus.cpp
--- a/scribus/scribus.cpp	2012-12-31 02:23:05.000000000 -0700
+++ b/scribus/scribus.cpp	2013-06-06 14:46:51.107008231 -0600
@@ -6688,7 +6688,7 @@
 		if (mode == modeDrawShapes)
 		{
 			doc->SubMode = modeToolBar->SubMode;
-			doc->ShapeValues = modeToolBar->ShapeVals;
+			doc->ShapeValues = (double*)modeToolBar->ShapeVals;
 			doc->ValCount = modeToolBar->ValCount;
 			propertiesPalette->SCustom->setIcon(propertiesPalette->SCustom->getIconPixmap(doc->SubMode));
 		}
