diff -urN a/scribus/scribus.cpp b/scribus/scribus.cpp
--- a/scribus/scribus.cpp	2014-12-21 04:31:54.000000000 -0700
+++ b/scribus/scribus.cpp	2015-01-31 01:03:34.411160428 -0700
@@ -6704,7 +6704,7 @@
 		if (mode == modeDrawShapes)
 		{
 			doc->SubMode = modeToolBar->SubMode;
-			doc->ShapeValues = modeToolBar->ShapeVals;
+			doc->ShapeValues = (double*)modeToolBar->ShapeVals;
 			doc->ValCount = modeToolBar->ValCount;
 			propertiesPalette->SCustom->setIcon(propertiesPalette->SCustom->getIconPixmap(doc->SubMode));
 		}
