diff -Naur gDesklets-0.36.1-old/utils/ErrorFormatter.py gDesklets-0.36.1/utils/ErrorFormatter.py
--- gDesklets-0.36.1-old/utils/ErrorFormatter.py	2008-03-20 09:00:59.000000000 +1000
+++ gDesklets-0.36.1/utils/ErrorFormatter.py	2008-12-07 03:08:41.000000000 +1000
@@ -112,20 +112,20 @@
 # available. Therefore, we have to extend the import handler in order to always
 # give us an absolute path.
 #
-_old_imp = __import__
-def _new_imp(name, globs = {}, locls = {}, fromlist = []):
-
-    module = _old_imp(name, globs, locls, fromlist)
-    # builtin modules have no "__file__" attribute, so we have to check for it
-    if (module):
-        if (hasattr(module, "__file__")):
-            module.__file__ = os.path.abspath(module.__file__)
-        return module
-    else:
-        return ""
-
-import __builtin__
-__builtin__.__import__ = _new_imp
+#_old_imp = __import__
+#def _new_imp(name, globs = {}, locls = {}, fromlist = []):
+#
+#    module = _old_imp(name, globs, locls, fromlist)
+#    # builtin modules have no "__file__" attribute, so we have to check for it
+#    if (module):
+#        if (hasattr(module, "__file__")):
+#            module.__file__ = os.path.abspath(module.__file__)
+#        return module
+#    else:
+#        return ""
+#
+#import __builtin__
+#__builtin__.__import__ = _new_imp
 
 
 
