*** mt-daapd-0.2.4.2/configure.in	2008-04-19 15:31:30.000000000 -0400
--- mt-daapd-0.2.4.2.new/configure.in	2011-05-01 20:56:15.204853627 -0400
***************
*** 20,27 ****
  CPPFLAGS="${CPPFLAGS} -g"
  
  dnl fix freebsd's broken (?) libpthread
! AC_CHECK_LIB(c_r,pthread_creat,LDFLAGS="${LDFLAGS} -lc_r", [
!   AC_CHECK_LIB(pthread,pthread_create,LDFLAGS="${LDFLAGS} -lpthread") ])
  
  AC_ARG_ENABLE(debug,[  --enable-debug          Enable debugging features],
  		       CPPFLAGS="${CPPFLAGS} -Wall")
--- 20,27 ----
  CPPFLAGS="${CPPFLAGS} -g"
  
  dnl fix freebsd's broken (?) libpthread
! AC_CHECK_LIB(c_r,pthread_creat,LIBS="${LIBS} -lc_r", [
!   AC_CHECK_LIB(pthread,pthread_create,LIBS="${LIBS} -lpthread") ])
  
  AC_ARG_ENABLE(debug,[  --enable-debug          Enable debugging features],
  		       CPPFLAGS="${CPPFLAGS} -Wall")
***************
*** 35,47 ****
  		      esac ])
  
  AC_ARG_ENABLE(efence,[  --enable-efence         Enable electric fence],
! 			LDFLAGS="${LDFLAGS} -lefence")
  AC_ARG_ENABLE(nslu2,[  --enable-nslu2          Build for NSLU2/uNSLUng], 
  		       CFLAGS="${CFLAGS} -DNSLU2")
  
  AC_ARG_ENABLE(howl,[  --enable-howl       Use howl 0.9.2 or later],
  		      [ case "${enableval}" in
!                         yes) rend_howl=true; rend_posix=false; rend_avahi=false; LDFLAGS="${LDFLAGS} -lhowl";
  			CPPFLAGS="${CPPFLAGS} -DWITH_HOWL";;
  		        no)  rend_howl=false;;
  		        *)   AC_MSG_ERROR(bad value ${enableval} for --enable-howl);;
--- 35,47 ----
  		      esac ])
  
  AC_ARG_ENABLE(efence,[  --enable-efence         Enable electric fence],
! 			LIBS="${LIBS} -lefence")
  AC_ARG_ENABLE(nslu2,[  --enable-nslu2          Build for NSLU2/uNSLUng], 
  		       CFLAGS="${CFLAGS} -DNSLU2")
  
  AC_ARG_ENABLE(howl,[  --enable-howl       Use howl 0.9.2 or later],
  		      [ case "${enableval}" in
!                         yes) rend_howl=true; rend_posix=false; rend_avahi=false; LIBS="${LIBS} -lhowl";
  			CPPFLAGS="${CPPFLAGS} -DWITH_HOWL";;
  		        no)  rend_howl=false;;
  		        *)   AC_MSG_ERROR(bad value ${enableval} for --enable-howl);;
***************
*** 50,56 ****
  AC_ARG_ENABLE(avahi,[  --enable-avahi          Use avahi 0.6 or later],
  		      [ case "${enableval}" in
                          yes) PKG_CHECK_MODULES(AVAHI, [ avahi-client >= 0.6 ]);
! 			rend_avahi=true; rend_howl=false; rend_posix=false; LDFLAGS="${LDFLAGS} $AVAHI_LIBS";
  			CPPFLAGS="${CPPFLAGS} $AVAHI_CFLAGS -DWITH_AVAHI";;
  		        no)  rend_avahi=false;;
  		        *)   AC_MSG_ERROR(bad value ${enableval} for --enable-avahi);;
--- 50,56 ----
  AC_ARG_ENABLE(avahi,[  --enable-avahi          Use avahi 0.6 or later],
  		      [ case "${enableval}" in
                          yes) PKG_CHECK_MODULES(AVAHI, [ avahi-client >= 0.6 ]);
! 			rend_avahi=true; rend_howl=false; rend_posix=false; LIBS="${LIBS} $AVAHI_LIBS";
  			CPPFLAGS="${CPPFLAGS} $AVAHI_CFLAGS -DWITH_AVAHI";;
  		        no)  rend_avahi=false;;
  		        *)   AC_MSG_ERROR(bad value ${enableval} for --enable-avahi);;
***************
*** 58,65 ****
  
  AC_ARG_ENABLE(oggvorbis,[  --enable-oggvorbis      Enable Ogg/Vorbis support],
                             use_oggvorbis=true;
! #                           LDFLAGS="${LDFLAGS} -logg -lvorbis";
!                            CPPFLAGS="${CPPFLAGS} -DOGGVORBIS")
  
  AM_CONDITIONAL(COND_REND_HOWL, test x$rend_howl = xtrue)
  AM_CONDITIONAL(COND_REND_AVAHI, test x$rend_avahi = xtrue)
--- 58,65 ----
  
  AC_ARG_ENABLE(oggvorbis,[  --enable-oggvorbis      Enable Ogg/Vorbis support],
                             use_oggvorbis=true;
! #                           LIBS="${LIBS} -logg -lvorbis";
!                            CPPFLAGS="${LIBS} -DOGGVORBIS")
  
  AM_CONDITIONAL(COND_REND_HOWL, test x$rend_howl = xtrue)
  AM_CONDITIONAL(COND_REND_AVAHI, test x$rend_avahi = xtrue)
***************
*** 82,88 ****
  	CPPFLAGS="$CPPFLAGS -DNOT_HAVE_SA_LEN -D_XPG4_2"
  	CPPFLAGS="$CPPFLAGS -D__EXTENSIONS__ -DHAVE_BROKEN_RECVIF_NAME"
  	CPPFLAGS="$CPPFLAGS -D_POSIX_PTHREAD_SEMANTICS"
! 	LDFLAGS="$LDFLAGS -lnsl -lsocket";;	
  *freebsd*)
  	CPPFLAGS="$CPPFLAGS -DFREEBSD";;
  *linux*)
--- 82,88 ----
  	CPPFLAGS="$CPPFLAGS -DNOT_HAVE_SA_LEN -D_XPG4_2"
  	CPPFLAGS="$CPPFLAGS -D__EXTENSIONS__ -DHAVE_BROKEN_RECVIF_NAME"
  	CPPFLAGS="$CPPFLAGS -D_POSIX_PTHREAD_SEMANTICS"
! 	LIBS="$LIBS -lnsl -lsocket";;	
  *freebsd*)
  	CPPFLAGS="$CPPFLAGS -DFREEBSD";;
  *linux*)
***************
*** 154,163 ****
  AC_CHECK_LIB(gdbm,gdbm_open,echo "Have gdbm", echo "Must have gdbm";exit)
  
  if test "$STATIC_LIBS" != "no"; then
!   LDFLAGS="${LDFLAGS} ${STATIC_LIBS}/libgdbm.a"
    echo "Adding static libgdbm"
  else
!   LDFLAGS="${LDFLAGS} -lgdbm"
    echo "Adding dynamic libgdbm"
  fi
  
--- 154,163 ----
  AC_CHECK_LIB(gdbm,gdbm_open,echo "Have gdbm", echo "Must have gdbm";exit)
  
  if test "$STATIC_LIBS" != "no"; then
!   LIBS="${LIBS} ${STATIC_LIBS}/libgdbm.a"
    echo "Adding static libgdbm"
  else
!   LIBS="${LIBS} -lgdbm"
    echo "Adding dynamic libgdbm"
  fi
  
***************
*** 170,179 ****
  AC_CHECK_LIB(id3tag,id3_file_open,echo "Have id3tag",echo "Must have libid3tag";exit)
  
  if test "$STATIC_LIBS" != "no"; then
!   LDFLAGS="${LDFLAGS} ${STATIC_LIBS}/libid3tag.a -lz"
    echo "Adding static libid3tag"
  else
!   LDFLAGS="${LDFLAGS} -lid3tag -lz"
    echo "Adding dynamic libid3tag"
  fi
  CFLAGS=$oldcflags
--- 170,179 ----
  AC_CHECK_LIB(id3tag,id3_file_open,echo "Have id3tag",echo "Must have libid3tag";exit)
  
  if test "$STATIC_LIBS" != "no"; then
!   LIBS="${LIBS} ${STATIC_LIBS}/libid3tag.a -lz"
    echo "Adding static libid3tag"
  else
!   LIBS="${LIBS} -lid3tag -lz"
    echo "Adding dynamic libid3tag"
  fi
  CFLAGS=$oldcflags
***************
*** 184,193 ****
    AC_CHECK_LIB(ogg,ogg_sync_init,echo "Have ogg",echo "Must have libogg for Ogg/Vorbis support";exit)
  
    if test x"$STATIC_LIBS" != x"no"; then
!     LDFLAGS="${LDFLAGS} ${STATIC_LIBS}/libogg.a"
      echo "Adding static libogg"
    else
!     LDFLAGS="${LDFLAGS} -logg"
      echo "Adding dynamic libogg"
    fi
  
--- 184,193 ----
    AC_CHECK_LIB(ogg,ogg_sync_init,echo "Have ogg",echo "Must have libogg for Ogg/Vorbis support";exit)
  
    if test x"$STATIC_LIBS" != x"no"; then
!     LIBS="${LIBS} ${STATIC_LIBS}/libogg.a"
      echo "Adding static libogg"
    else
!     LIBS="${LIBS} -logg"
      echo "Adding dynamic libogg"
    fi
  
***************
*** 196,205 ****
    AC_CHECK_LIB(vorbis,vorbis_info_init,echo "Have vorbis",echo "Must have libvorbis for Ogg/Vorbis support";exit)
  
    if test x"$STATIC_LIBS" != x"no"; then
!     LDFLAGS="${LDFLAGS} ${STATIC_LIBS}/libvorbis.a"
      echo "Adding static libvorbis"
    else
!     LDFLAGS="${LDFLAGS} -lvorbis"
      echo "Adding dynamic libvorbis"
    fi
  fi
--- 196,205 ----
    AC_CHECK_LIB(vorbis,vorbis_info_init,echo "Have vorbis",echo "Must have libvorbis for Ogg/Vorbis support";exit)
  
    if test x"$STATIC_LIBS" != x"no"; then
!     LIBS="${LIBS} ${STATIC_LIBS}/libvorbis.a"
      echo "Adding static libvorbis"
    else
!     LIBS="${LIBS} -lvorbis"
      echo "Adding dynamic libvorbis"
    fi
  fi
