diff -urN flashrom-google/internal.c flashrom-google-build/internal.c
--- flashrom-google/internal.c	2013-02-22 21:23:49.091306851 -0500
+++ flashrom-google-build/internal.c	2013-02-22 21:27:34.661306653 -0500
@@ -301,10 +301,10 @@
 	 * fix up any scripts which depend on "-p internal:bus=lpc" for
 	 * flashing EC firmware.
 	 */
-	if ((target_bus == BUS_PROG) || (target_bus == BUS_LPC)) {
+	/*if ((target_bus == BUS_PROG) || (target_bus == BUS_LPC)) {
 		if (!gec_probe_i2c(NULL))
 			return 0;
-	}
+			}*/
 #endif
 
 #if defined(__arm__) && CONFIG_LINUX_SPI == 1
diff -urN flashrom-google/Makefile flashrom-google-build/Makefile
--- flashrom-google/Makefile	2013-02-22 21:23:49.091306851 -0500
+++ flashrom-google-build/Makefile	2013-02-22 21:26:15.056307031 -0500
@@ -362,9 +362,9 @@
 PROGRAMMER_OBJS += gec_lpc.o it87spi.o it85spi.o mec1308.o sb600spi.o wbsio_spi.o mcp6x_spi.o wpce775x.o ene_lpc.o
 PROGRAMMER_OBJS += ichspi.o ich_descriptors.o
 else
-ifeq ($(ARCH),"arm")
-PROGRAMMER_OBJS += gec_i2c.o tegra2_spi.o
-endif
+#ifeq ($(ARCH),"arm")
+#PROGRAMMER_OBJS += gec_i2c.o tegra2_spi.o
+#endif
 NEED_PCI := yes
 endif
 endif
diff -urN flashrom-google/processor_enable.c flashrom-google-build/processor_enable.c
--- flashrom-google/processor_enable.c	2013-02-22 21:23:49.091306851 -0500
+++ flashrom-google-build/processor_enable.c	2013-02-22 21:28:33.701306988 -0500
@@ -159,10 +159,10 @@
 	/* FIXME(yjlou): 0xc09 seems ARMv7, not tegra* specified.
 	 *               Also, the tegra2_spi_init() should move out of
 	 *               the processor enable code logic. */
-	if (is_tegra2()) {
+/*	if (is_tegra2()) {
 		msg_pdbg("Detected NVIDIA Tegra 2.\n");
 		return tegra2_spi_init();
-	}
+		}*/
 #endif
 	/* Not implemented yet. Oh well. */
 	return 1;
