diff -Naur libvpx-1.3.orig/nestegg/halloc/src/align.h libvpx-1.3/nestegg/halloc/src/align.h
--- libvpx-1.3.orig/nestegg/halloc/src/align.h	2015-08-02 13:42:14.783914000 +0200
+++ libvpx-1.3/nestegg/halloc/src/align.h	2015-08-02 13:43:04.100580874 +0200
@@ -30,7 +30,7 @@
 	void (*q)(void);
 };
 
-typedef union max_align max_align_t;
+typedef union max_align h_max_align_t;
 
 #endif
 
diff -Naur libvpx-1.3.orig/nestegg/halloc/src/halloc.c libvpx-1.3/nestegg/halloc/src/halloc.c
--- libvpx-1.3.orig/nestegg/halloc/src/halloc.c	2015-08-02 13:42:14.783914000 +0200
+++ libvpx-1.3/nestegg/halloc/src/halloc.c	2015-08-02 13:43:35.817247310 +0200
@@ -30,7 +30,7 @@
 #endif
 	hlist_item_t  siblings; /* 2 pointers */
 	hlist_head_t  children; /* 1 pointer  */
-	max_align_t   data[1];  /* not allocated, see below */
+	h_max_align_t data[1];  /* not allocated, see below */
 	
 } hblock_t;
 
