diff -urNp wvdial-1.60-orig/wvdialer.cc wvdial-1.60/wvdialer.cc
--- wvdial-1.60-orig/wvdialer.cc	2009-06-11 12:37:13.000000000 +0200
+++ wvdial-1.60/wvdialer.cc	2009-06-11 13:06:06.000000000 +0200
@@ -680,7 +680,7 @@ void WvDialer::load_options()
     	{ NULL,		     NULL, NULL,                   "", 0            }
     };
 
-    char *	d = "Dialer Defaults";
+    const char *	d = "Dialer Defaults";
 
     for( int i=0; opts[i].name != NULL; i++ ) 
     {
Binární soubory wvdial-1.60-orig/.wvdialer.cc.swp a wvdial-1.60/.wvdialer.cc.swp jsou různé
diff -urNp wvdial-1.60-orig/wvdialer.h wvdial-1.60/wvdialer.h
--- wvdial-1.60-orig/wvdialer.h	2009-06-11 12:37:13.000000000 +0200
+++ wvdial-1.60/wvdialer.h	2009-06-11 12:39:30.000000000 +0200
@@ -30,10 +30,10 @@ extern const char wvdial_version_text[];
 struct OptInfo
 /************/
 {
-    char *	name;
+    const char *	name;
     WvString *	str_member;
     int *	int_member;
-    char *	str_default;
+    const char *	str_default;
     int		int_default;
 };
 
diff -urNp wvdial-1.60-orig/wvmodemscan.cc wvdial-1.60/wvmodemscan.cc
--- wvdial-1.60-orig/wvmodemscan.cc	2007-07-19 23:22:26.000000000 +0200
+++ wvdial-1.60/wvmodemscan.cc	2009-06-11 13:16:02.000000000 +0200
@@ -20,7 +20,7 @@ bool     default_asyncmap = false;
 
 // startup at atz atq0 atv1 ate1 ats0 carrier dtr fastdial
 // baudstep reinit done
-static char *commands[WvModemScan::NUM_STAGES] = {
+static const char *commands[WvModemScan::NUM_STAGES] = {
     NULL, "Q0 V1 E1", "Z", "S0=0",
     "&C1", "&D2", "+FCLASS=0", NULL,
     NULL, "", NULL
@@ -496,7 +496,7 @@ static int fileselect(const struct diren
 }
 
 
-static int filesort(const void *_e1, const void *_e2)
+static int filesort(const struct dirent **_e1, const struct dirent **_e2)
 {
     dirent const * const *e1 = (dirent const * const *)_e1;
     dirent const * const *e2 = (dirent const * const *)_e2;
