--- sawfish-1.2-gtk2/man/Makefile.in.old	2002-11-12 22:34:45.000000000 -0500
+++ sawfish-1.2-gtk2/man/Makefile.in	2002-11-12 22:34:40.000000000 -0500
@@ -35,7 +35,7 @@
 TEXI2DVI = texi2dvi
 DVIPS = dvips
 
-all : sawfish.info ../NEWS
+all :
 
 ../NEWS : news.texi
 	$(MAKEINFO) $(MAKEINFOFLAGS) --no-headers $< -o ../NEWS
@@ -54,11 +54,7 @@
 
 man : sawfish.info sawfish.guide sawfish.doc sawfish.dvi sawfish.ps
 
-install : sawfish.info installdirs
-	for f in sawfish.info*; do \
- 	  $(INSTALL_DATA) $$f $(DESTDIR)$(infodir); \
-	done
-	-install-info --info-dir=$(DESTDIR)$(infodir) sawfish.info
+install :
 
 installdirs : mkinstalldirs
 	$(SHELL) $< $(DESTDIR)$(infodir)
