commit e09a9a35a1fdb4e849e63bdaf47d936c272b5e82
Author: Lennart Poettering <lennart@poettering.net>
Date:   2012-03-05 15:16:10 +0100

    sd-login: c++ compatibility

diff --git a/src/systemd/sd-login.h b/src/systemd/sd-login.h
index 2f3c90c..6e99cfc 100644
--- a/src/systemd/sd-login.h
+++ b/src/systemd/sd-login.h
@@ -94,7 +94,7 @@ int sd_session_get_service(const char *session, char **service);
 int sd_session_get_type(const char *session, char **type);
 
 /* Determine the class of this session, i.e. one of "user", "greeter" or "lock-screen". */
-int sd_session_get_class(const char *session, char **class);
+int sd_session_get_class(const char *session, char **clazz);
 
 /* Determine the X11 display of this session. */
 int sd_session_get_display(const char *session, char **display);
