diff -urN a/src/journal/journald.c b/src/journal/journald.c
--- a/src/journal/journald.c	2012-03-15 12:46:57.000000000 -0600
+++ b/src/journal/journald.c	2012-03-17 13:52:36.976437081 -0600
@@ -2144,7 +2144,7 @@
                                 uint8_t buf[CMSG_SPACE(sizeof(struct ucred)) +
                                             CMSG_SPACE(sizeof(struct timeval)) +
                                             CMSG_SPACE(sizeof(int)) +
-                                            CMSG_SPACE(PAGE_SIZE)]; /* selinux label */
+                                            CMSG_SPACE(sysconf(_SC_PAGE_SIZE))]; /* selinux label */
                         } control;
                         ssize_t n;
                         int v;
