From 4144242b59b65d36cf004778c172e351403b1a6a Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Thu, 3 Mar 2016 17:58:16 -0700
Subject: [PATCH] remove default to zero

Signed-off-by: Kevin Mihelich <kevin@archlinuxarm.org>
---
 src/libcec/cmake/CheckPlatformSupport.cmake | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/src/libcec/cmake/CheckPlatformSupport.cmake b/src/libcec/cmake/CheckPlatformSupport.cmake
index b8a16c8..677469e 100644
--- a/src/libcec/cmake/CheckPlatformSupport.cmake
+++ b/src/libcec/cmake/CheckPlatformSupport.cmake
@@ -22,14 +22,6 @@ set(PLATFORM_LIBREQUIRES "")
 include(CheckFunctionExists)
 include(FindPkgConfig)
 
-# defaults
-set(HAVE_RANDR 0)
-set(HAVE_LIBUDEV 0)
-set(HAVE_RPI_API 0)
-set(HAVE_TDA995X_API 0)
-set(HAVE_EXYNOS_API 0)
-set(HAVE_P8_USB_DETECT 0)
-set(HAVE_DRM_EDID_PARSER 0)
 # Pulse-Eight devices are always supported
 set(HAVE_P8_USB 1)
 
-- 
2.7.1

