diff -urN a/src/lib/adapter/RPi/RPiCECAdapterCommunication.cpp b/src/lib/adapter/RPi/RPiCECAdapterCommunication.cpp
--- a/src/lib/adapter/RPi/RPiCECAdapterCommunication.cpp	2013-12-13 08:03:33.000000000 -0700
+++ b/src/lib/adapter/RPi/RPiCECAdapterCommunication.cpp	2014-01-09 13:51:12.250929575 -0700
@@ -299,7 +299,7 @@
   if (bStartListening)
   {
     // enable passive mode
-    vc_cec_set_passive(true);
+    vc_cec_set_passive((bool_t)true);
 
     // register the callbacks
     vc_cec_register_callback(rpi_cec_callback, (void*)this);
@@ -358,7 +358,7 @@
   UnregisterLogicalAddress();
 
   // disable passive mode
-  vc_cec_set_passive(false);
+  vc_cec_set_passive((bool_t)false);
 
   if (!g_bHostInited)
   {
diff -urN a/src/lib/adapter/RPi/RPiCECAdapterMessageQueue.cpp b/src/lib/adapter/RPi/RPiCECAdapterMessageQueue.cpp
--- a/src/lib/adapter/RPi/RPiCECAdapterMessageQueue.cpp	2013-12-13 08:03:33.000000000 -0700
+++ b/src/lib/adapter/RPi/RPiCECAdapterMessageQueue.cpp	2014-01-09 13:51:12.255929509 -0700
@@ -182,7 +182,7 @@
   LIB_CEC->AddLog(CEC_LOG_DEBUG, "sending data: %s", strDump.c_str());
 #endif
 
-   int iReturn = vc_cec_send_message((uint32_t)command.destination, (uint8_t*)&payload, iLength, bIsReply);
+   int iReturn = vc_cec_send_message((uint32_t)command.destination, (uint8_t*)&payload, iLength, (bool_t)bIsReply);
 #endif
 
   if (iReturn != VCHIQ_SUCCESS)
