diff -urN trimslice-kernel-trimslice-kernel-a/arch/arm/mach-tegra/arb_sema.c trimslice-kernel-trimslice-kernel-b/arch/arm/mach-tegra/arb_sema.c
--- trimslice-kernel-trimslice-kernel-a/arch/arm/mach-tegra/arb_sema.c	2012-04-15 09:14:26.000000000 +0100
+++ trimslice-kernel-trimslice-kernel-b/arch/arm/mach-tegra/arb_sema.c	2012-05-11 11:58:43.000000000 +0100
@@ -240,4 +240,4 @@
 }
 subsys_initcall(tegra_arb_init);
 
-MODULE_LICENSE("GPLv2");
+MODULE_LICENSE("GPL v2");
diff -urN trimslice-kernel-trimslice-kernel-a/arch/arm/mach-tegra/headsmp.S trimslice-kernel-trimslice-kernel-b/arch/arm/mach-tegra/headsmp.S
--- trimslice-kernel-trimslice-kernel-a/arch/arm/mach-tegra/headsmp.S	2012-04-15 09:14:26.000000000 +0100
+++ trimslice-kernel-trimslice-kernel-b/arch/arm/mach-tegra/headsmp.S	2012-05-11 11:57:43.000000000 +0100
@@ -298,6 +298,6 @@
 	.rept	TEGRA_RESET_DATA_SIZE
 	.long	0
 	.endr
-	.size	__tegra_cpu_reset_handler_data, .-tegra_cpu_reset_handler_data
+	.size	__tegra_cpu_reset_handler_data, .-__tegra_cpu_reset_handler_data
 	.align L1_CACHE_SHIFT
 ENTRY(__tegra_cpu_reset_handler_end)
diff -urN trimslice-kernel-trimslice-kernel-a/drivers/crypto/tegra-aes.c trimslice-kernel-trimslice-kernel-b/drivers/crypto/tegra-aes.c
--- trimslice-kernel-trimslice-kernel-a/drivers/crypto/tegra-aes.c	2012-04-15 09:14:26.000000000 +0100
+++ trimslice-kernel-trimslice-kernel-b/drivers/crypto/tegra-aes.c	2012-05-11 12:00:29.000000000 +0100
@@ -1447,4 +1447,4 @@
 
 MODULE_DESCRIPTION("Tegra AES hw acceleration support.");
 MODULE_AUTHOR("NVIDIA Corporation");
-MODULE_LICENSE("GPLv2");
+MODULE_LICENSE("GPL v2");
diff -urN trimslice-kernel-trimslice-kernel-a/drivers/misc/tegra-cryptodev.c trimslice-kernel-trimslice-kernel-b/drivers/misc/tegra-cryptodev.c
--- trimslice-kernel-trimslice-kernel-a/drivers/misc/tegra-cryptodev.c	2012-04-15 09:14:26.000000000 +0100
+++ trimslice-kernel-trimslice-kernel-b/drivers/misc/tegra-cryptodev.c	2012-05-11 11:59:38.000000000 +0100
@@ -346,4 +346,4 @@
 
 MODULE_DESCRIPTION("Tegra AES hw device node.");
 MODULE_AUTHOR("NVIDIA Corporation");
-MODULE_LICENSE("GPLv2");
+MODULE_LICENSE("GPL v2");
