diff -ruN weather.orig/translations/weather_pl.ts weather/translations/weather_pl.ts
--- weather.orig/translations/weather_pl.ts	2008-01-22 20:28:27.000000000 +0100
+++ weather/translations/weather_pl.ts	2009-03-22 09:29:01.000000000 +0100
@@ -96,10 +96,6 @@
         <translation>Powiadomienia</translation>
     </message>
     <message>
-        <source>Notifications</source>
-        <translation>Powiadomienia</translation>
-    </message>
-    <message>
         <source>Display</source>
         <translation>Wyświetlanie</translation>
     </message>
