From d820f5c88bea19ba23e4d603d6d9369dd2b525ee Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 9 Jul 2019 07:45:26 -0600
Subject: [PATCH] 32-bit changes

---
 makefile | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/makefile b/makefile
index 06b115c..37e4185 100644
--- a/makefile
+++ b/makefile
@@ -8,10 +8,10 @@
 #try uncommenting next line if cc is the default C compiler
 #CC = gcc
 
-default: lrs redund lrsgmp redundgmp
+#default: lrs redund lrsgmp redundgmp
 
 #choose line below instead if __int128 not supported
-#default: lrs64 redund64 lrsgmp redundgmp
+default: lrs64 redund64 lrsgmp redundgmp
 
 
 #make lrs               lrs,lrsgmp   redund,redundgmp    hybrid and gmp versions 
@@ -195,7 +195,7 @@ SOMINOR ?=.0.0
 SHLIB ?=$(SONAME)$(SOMINOR)
 SHLINK ?=liblrs.so
 
-SHLIBOBJ=lrslong1-shr.o lrslong2-shr.o lrslib1-shr.o lrslib2-shr.o \
+SHLIBOBJ=lrslong1-shr.o lrslib1-shr.o \
 	lrslibgmp-shr.o lrsgmp-shr.o lrsdriver-shr.o
 
 SHLIBBIN=lrs-shared redund-shared lrsnash-shared
@@ -215,11 +215,11 @@ ${SHLINK}: ${SONAME}
 
 all-shared: ${SHLIBBIN}
 
-lrs-shared: ${SHLINK} lrs.o
-	$(CC) lrs.o -o $@ -L . -llrs ${LDFLAGS}
+lrs-shared: ${SHLINK} lrs64.o
+	$(CC) lrs64.o -o $@ -L . -llrs ${LDFLAGS}
 
-redund-shared: ${SHLINK}  redund.o
-	$(CC) redund.o -o $@ -L . -llrs ${LDFLAGS}
+redund-shared: ${SHLINK}  redund64.o
+	$(CC) redund64.o -o $@ -L . -llrs ${LDFLAGS}
 
 lrsnash-shared: ${SHLINK}  lrsnash.c
 	$(CC) -DGMP -DMA lrsnash.c  lrsnashlib.c -I${INCLUDEDIR} -o $@ -L . -llrs -lgmp ${LDFLAGS}
-- 
2.21.0

