--- makefile.orig	2021-06-04 10:48:58.746663307 -0600
+++ makefile	2021-06-04 11:03:59.777040326 -0600
@@ -8,10 +8,10 @@
 #try uncommenting next line if cc is the default C compiler
 #CC = gcc
 
-default: lrs lrsgmp
+#default: lrs lrsgmp
 
 #choose line below instead if __int128 not supported
-#default: lrs64 lrsgmp 
+default: lrs64 lrsgmp 
 
 
 #make lrs               lrs,lrsgmp       hybrid and gmp versions 
@@ -47,12 +47,12 @@
 
 # for 32 bit machines
 
-# BITS=-DB32
-# MPLRSOBJ2=
+BITS=-DB32
+MPLRSOBJ2=
 
 # for 64 bit machines
-BITS=-DB128
-MPLRSOBJ2=lrslib2-mplrs.o lrslong2-mplrs.o
+#BITS=-DB128
+#MPLRSOBJ2=lrslib2-mplrs.o lrslong2-mplrs.o
 
 
 LRSOBJ=lrs.o lrslong1.o lrslong2.o lrslib1.o lrslib2.o lrslibgmp.o lrsgmp.o lrsdriver.o
@@ -171,14 +171,14 @@
 
 single:		lrs.c lrslong.c lrslong.h lrslib.c lrslib.h lrsgmp.c lrsgmp.h lrsdriver.h lrsdriver.c
 		$(CC)  ${CFLAGS}  -DSAFE  -DLRSLONG -o lrs1 lrs.c lrslib.c lrslong.c lrsdriver.c
-		$(CC)  ${CFLAGS} ${BITS} -DSAFE  -DLRSLONG -o lrs2 lrs.c lrslib.c lrslong.c lrsdriver.c
+		#$(CC)  ${CFLAGS} ${BITS} -DSAFE  -DLRSLONG -o lrs2 lrs.c lrslib.c lrslong.c lrsdriver.c
 		ln -s -f lrs1 redund1
 		ln -s -f lrs2 redund2
 
 allmp:		lrs.c lrslib.c lrslib.h lrsmp.c lrsmp.h lrsdriver.h lrsdriver.c
 		$(CC) -Wall -O3  -o lrs lrs.c lrslib.c lrsdriver.c lrsmp.c
 		$(CC) -Wall -O3  -DSAFE -DLRSLONG -o lrs1 lrs.c lrslib.c lrsdriver.c lrslong.c
-		$(CC) -Wall -O3  -DSAFE -DLRSLONG ${BITS} -o lrs2 lrs.c lrslib.c lrsdriver.c lrslong.c
+		#$(CC) -Wall -O3  -DSAFE -DLRSLONG ${BITS} -o lrs2 lrs.c lrslib.c lrsdriver.c lrslong.c
 		$(CC) -O3 -DLRS_QUIET   -o lrsnash lrsnash.c lrsnashlib.c lrslib.c lrsdriver.c lrsmp.c -static
 		$(CC) -O3  -o setnash setupnash.c lrslib.c lrsdriver.c lrsmp.c
 		$(CC) -O3  -o setnash2 setupnash2.c lrslib.c lrsdriver.c lrsmp.c
@@ -194,7 +194,7 @@
 lrsnash:	lrsnash.c nashdemo.c lrsnashlib.c lrslib.c lrsnashlib.h lrslib.h lrsgmp.c lrsgmp.h lrslong.h lrsdriver.h lrsdriver.c
 		$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lrsnashgmp lrsnash.c lrsnashlib.c lrslib.c lrsgmp.c lrsdriver.c  -lgmp -DGMP
 		$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lrsnash1 lrsnash.c lrsnashlib.c lrslib.c lrslong.c lrsdriver.c -DLRSLONG -DSAFE
-		$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lrsnash2 lrsnash.c lrsnashlib.c lrslib.c lrslong.c lrsdriver.c -DLRSLONG -DSAFE ${BITS}
+		#$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lrsnash2 lrsnash.c lrsnashlib.c lrslib.c lrslong.c lrsdriver.c -DLRSLONG -DSAFE ${BITS}
 		$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o nashdemo nashdemo.c lrsnashlib.c lrslib.c lrsgmp.c lrsdriver.c -lgmp -DGMP
 		$(CC) -O3  -I${INCLUDEDIR} -L${LIBDIR} -o 2nash 2nash.c
 		cp lrsnashgmp lrsnash
@@ -202,7 +202,7 @@
 fel:	fel.c lrslib.h lrslib.c lrsgmp.h lrsgmp.c lrslong.c
 	$(CC) -O3 -Wall  -DGMP -I${INCLUDEDIR} fel.c lrslib.c lrsdriver.c lrsgmp.c -L${LIBDIR}  -lgmp -o felgmp
 	$(CC) -O3 -Wall  -I${INCLUDEDIR} fel.c lrslib.c lrsdriver.c lrslong.c -L${LIBDIR}  -DLRSLONG -DSAFE -o fel1
-	$(CC) -O3 -Wall  -I${INCLUDEDIR} fel.c lrslib.c lrsdriver.c lrslong.c -L${LIBDIR}  -DLRSLONG -DSAFE ${BITS} -o fel2
+	#$(CC) -O3 -Wall  -I${INCLUDEDIR} fel.c lrslib.c lrsdriver.c lrslong.c -L${LIBDIR}  -DLRSLONG -DSAFE ${BITS} -o fel2
 
 ######################################################################
 # From here on the author is David Bremner <bremner@unb.ca> to whom you should turn for help             
@@ -213,11 +213,11 @@
 SHLIB ?=$(SONAME)$(SOMINOR)
 SHLINK ?=liblrs.so
 
-SHLIBOBJ2=lrslib2-shr.o lrslong2-shr.o
+#SHLIBOBJ2=lrslib2-shr.o lrslong2-shr.o
 
 # for 32 bit machines
 
-# SHLIBOBJ2=
+SHLIBOBJ2=
 
 SHLIBOBJ=lrslong1-shr.o lrslib1-shr.o  \
 	lrslibgmp-shr.o lrsgmp-shr.o lrsdriver-shr.o \
