diff -urN a/makefile b/makefile
--- a/makefile	2020-06-04 06:47:14.484574508 -0600
+++ b/makefile	2020-06-04 06:52:20.348011716 -0600
@@ -8,10 +8,10 @@
 #try uncommenting next line if cc is the default C compiler
 #CC = gcc
 
-default: lrs
+#default: lrs
 
 #choose line below instead if __int128 not supported
-#default: lrs64 lrsgmp 
+default: lrs64 lrsgmp 
 
 
 #make lrs               lrs,lrsgmp       hybrid and gmp versions 
@@ -185,7 +185,7 @@
 SHLIB ?=$(SONAME)$(SOMINOR)
 SHLINK ?=liblrs.so
 
-SHLIBOBJ=lrslong1-shr.o lrslong2-shr.o lrslib1-shr.o lrslib2-shr.o \
+SHLIBOBJ=lrslong1-shr.o lrslib1-shr.o \
 	lrslibgmp-shr.o lrsgmp-shr.o lrsdriver-shr.o
 
 SHLIBBIN=lrs-shared lrsnash-shared
@@ -205,8 +205,8 @@
 
 all-shared: ${SHLIBBIN}
 
-lrs-shared: ${SHLINK} lrs.o
-	$(CC) lrs.o -o $@ -L . -llrs ${LDFLAGS}
+lrs-shared: ${SHLINK} lrs64.o
+	$(CC) lrs64.o -o $@ -L . -llrs ${LDFLAGS}
 
 
 lrsnash-shared: ${SHLINK}  lrsnash.c
