--- lrslib-071/makefile.orig	2020-06-04 09:58:02.217703517 +0000
+++ lrslib-071/makefile	2020-06-04 09:59:20.505523290 +0000
@@ -239,7 +239,7 @@
 # install targets
 # where to install binaries, libraries, include files
 prefix ?= /usr/local
-INSTALL_INCLUDES=lrslib.h lrsdriver.h lrsgmp.h lrslong.h lrsmp.h
+INSTALL_INCLUDES=lrslib.h lrsdriver.h lrsgmp.h lrslong.h lrsmp.h lrsrestart.h
 
 install: all-shared install-common
 	mkdir -p $(DESTDIR)${prefix}/bin
diff --git a/lrslib-071/lrslib.h.orig b/lrslib-071/lrslib.h
index a2eb4b5..ee48475 100644
--- lrslib-071/lrslib.h.orig
+++ lrslib-071/lrslib.h
@@ -148,6 +148,8 @@
 
 #include ARITH
 
+#include "lrsdriver.h"
+
 #ifndef SIGNALS
 #include <signal.h>
 #include <unistd.h>
--- lrslib-071/lrslib.c.orig    2020-06-04 10:43:56.790948185 +0000
+++ lrslib-071/lrslib.c 2020-06-04 10:44:05.641075898 +0000
@@ -27,7 +27,6 @@
 #include <string.h>
 #include <setjmp.h>
 #include <limits.h>
-#include "lrsrestart.h"
 #include "lrslib.h"
 
 static unsigned long dict_count, dict_limit, cache_tries, cache_misses;
