--- lrslib-070/makefile.orig	2019-04-29 11:53:55.707113154 +0000
+++ lrslib-070/makefile	2019-04-29 11:55:36.485187714 +0000
@@ -35,7 +35,6 @@
 #INCLUDEDIR = /usr/local/include:/usr/include
 #LIBDIR     = /usr/local/lib:/usr/lib
 
-CFLAGS     = -O3 -Wall
 SHLIB_CFLAGS = -fPIC
 mpicxx=mpic++
 
@@ -204,7 +203,7 @@
 # Building (linking) the shared library, and relevant symlinks.
 
 ${SHLIB}: ${SHLIBOBJ}
-	$(CC) -shared -Wl,-soname=$(SONAME) $(SHLIBFLAGS) -o $@ ${SHLIBOBJ} -lgmp
+	$(CC) -shared -Wl,-soname=$(SONAME) $(SHLIBFLAGS) -o $@ ${SHLIBOBJ} -lgmp ${LDFLAGS}
 
 ${SONAME}: ${SHLIB}
 	ln -sf ${SHLIB} ${SONAME}
@@ -217,13 +216,13 @@
 all-shared: ${SHLIBBIN}
 
 lrs-shared: ${SHLINK} lrs.o
-	$(CC) lrs.o -o $@ -L . -llrs
+	$(CC) lrs.o -o $@ -L . -llrs ${LDFLAGS}
 
 redund-shared: ${SHLINK}  redund.o
-	$(CC) redund.o -o $@ -L . -llrs
+	$(CC) redund.o -o $@ -L . -llrs ${LDFLAGS}
 
 lrsnash-shared: ${SHLINK}  lrsnash.c
-	$(CC) -DGMP -DMA lrsnash.c  lrsnashlib.c -I${INCLUDEDIR} -o $@ -L . -llrs -lgmp
+	$(CC) -DGMP -DMA lrsnash.c  lrsnashlib.c -I${INCLUDEDIR} -o $@ -L . -llrs -lgmp ${LDFLAGS}
 
 # build object files for the shared library
 
