--- lrslib-071/makefile.orig	2020-06-04 09:58:02.217703517 +0000
+++ lrslib-071/makefile	2020-06-04 09:59:20.505523290 +0000
@@ -32,8 +32,8 @@
 #LIBDIR     = /usr/lib
 
 #Kyoto machines usage
-INCLUDEDIR = /usr/local/include
-LIBDIR     = /usr/local/lib
+INCLUDEDIR = /usr/include
+LIBDIR     = /usr/lib
 
 CFLAGS     ?= -O3 -Wall
 #CFLAGS     = -g -Wall 
@@ -228,7 +228,7 @@
 # Building (linking) the shared library, and relevant symlinks.
 
 ${SHLIB}: ${SHLIBOBJ}
-	$(CC) -shared -Wl,-soname=$(SONAME) $(SHLIBFLAGS) -o $@ ${SHLIBOBJ} -lgmp
+	$(CC) -shared -Wl,-soname=$(SONAME) $(SHLIBFLAGS) -o $@ ${SHLIBOBJ} -lgmp ${LDFLAGS}
 
 ${SONAME}: ${SHLIB}
 	ln -sf ${SHLIB} ${SONAME}
@@ -241,11 +240,11 @@
 all-shared: ${SHLIBBIN}
 
 lrs-shared: ${SHLINK} lrs-shared.o
-	$(CC) $^ -o $@ -L . -llrs
+	$(CC) $^ -o $@ -L . -llrs ${LDFLAGS}
 
 
 lrsnash-shared: ${SHLINK}  lrsnash.c
-	$(CC) ${CFLAGS} -DGMP -DMA lrsnash.c  lrsnashlib.c -I${INCLUDEDIR} -o $@ -L . -llrs -lgmp
+	$(CC) ${CFLAGS} -DGMP -DMA lrsnash.c  lrsnashlib.c -I${INCLUDEDIR} -o $@ -L . -llrs -lgmp ${LDFLAGS}
 
 # driver object files
 
