diff '--color=auto' -rupN xxHash.orig/Makefile xxHash/Makefile
--- xxHash.orig/Makefile	2021-12-06 11:36:16.742156574 +0100
+++ xxHash/Makefile	2021-12-06 11:37:31.712450289 +0100
@@ -518,9 +518,9 @@ endif
 	$(Q)ln -sf xxhsum $(DESTDIR)$(BINDIR)/xxh128sum
 	@echo Installing man pages
 	$(Q)$(INSTALL_DATA) $(MAN) $(DESTDIR)$(MANDIR)/xxhsum.1
-	$(Q)ln -sf $(MAN) $(DESTDIR)$(MANDIR)/xxh32sum.1
-	$(Q)ln -sf $(MAN) $(DESTDIR)$(MANDIR)/xxh64sum.1
-	$(Q)ln -sf $(MAN) $(DESTDIR)$(MANDIR)/xxh128sum.1
+	$(Q)ln -sf xxhsum.1.gz $(DESTDIR)$(MANDIR)/xxh32sum.1.gz
+	$(Q)ln -sf xxhsum.1.gz $(DESTDIR)$(MANDIR)/xxh64sum.1.gz
+	$(Q)ln -sf xxhsum.1.gz $(DESTDIR)$(MANDIR)/xxh128sum.1.gz
 	@echo xxhash installation completed
 
 .PHONY: uninstall
