--- nx-3.3.0/nxcomp/Loop.cpp.orig	2009-05-17 12:22:19.000000000 +0200
+++ nx-3.3.0/nxcomp/Loop.cpp	2009-05-17 12:22:43.000000000 +0200
@@ -10979,7 +10979,7 @@
                                       packMethod == PACK_LOSSLESS ||
                                           packMethod == PACK_ADAPTIVE)
   {
-    char *dash = rindex(opt, '-');
+    const char *dash = rindex(opt, '-');
 
     if (dash != NULL && strlen(dash) == 2 &&
             *(dash + 1) >= '0' && *(dash + 1) <= '9')

