diff -wbBur tvision/classes/ipstream.cc tvision.my/classes/ipstream.cc
--- tvision/classes/ipstream.cc	2006-11-13 15:50:47.000000000 +0000
+++ tvision.my/classes/ipstream.cc	2009-10-05 09:17:27.000000000 +0000
@@ -19,6 +19,7 @@
  *
  */
 #include <assert.h>
+#include <stdio.h>
 #define Uses_string
 #define Uses_TStreamable
 #define Uses_TStreamableClass
diff -wbBur tvision/config.pl tvision.my/config.pl
--- tvision/config.pl	2007-04-18 15:46:34.000000000 +0000
+++ tvision.my/config.pl	2009-10-05 09:17:33.000000000 +0000
@@ -1199,12 +1199,6 @@
     $rep.="\t".GenInstallFiles('0644',"$makeDir/$ver",'$(libdir)');
     $stripDebug=($OSf eq 'Darwin') ? '-S' : '--strip-debug';
     $rep.="\tstrip $stripDebug \$(libdir)/$ver\n" unless $conf{'debugInfo'} eq 'yes';
-    # FreeBSD: merge data from libdir.
-    # Darwin: doesn't have ldconfig.
-    if ($OSf ne 'Darwin')
-      {
-       $rep.=($OSf eq 'FreeBSD') ? "\t-ldconfig -m \$(libdir)\n" : "\t-ldconfig\n";
-      }
    }
  if ($internac)
    {
diff -wbBur tvision/include/tv/textdev.h tvision.my/include/tv/textdev.h
--- tvision/include/tv/textdev.h	2005-07-08 16:21:55.000000000 +0000
+++ tvision.my/include/tv/textdev.h	2009-10-05 09:17:27.000000000 +0000
@@ -16,6 +16,8 @@
 #if defined( Uses_TTextDevice ) && !defined( __TTextDevice )
 #define __TTextDevice
 
+#include <stdio.h>
+
 class TRect;
 class TScrollBar;
 
diff -wbBur tvision/makes/common.imk tvision.my/makes/common.imk
--- tvision/makes/common.imk	2004-10-01 19:39:19.000000000 +0000
+++ tvision.my/makes/common.imk	2009-10-05 09:17:11.000000000 +0000
@@ -54,7 +54,7 @@
 C_EXTRA_FLAGS=$(SHARED_CODE_OPTION)
 RHIDE_COMPILE_ARCHIVE=$(RHIDE_LD) $(RHIDE_LIBDIRS) $(LDFLAGS) \
 	$(RHIDE_LDFLAGS) $(C_EXTRA_FLAGS) -o $(OUTFILE)  \
-	$(OBJFILES)
+	$(OBJFILES) $(addprefix -l,$(RHIDE_OS_LIBS))
 else
 RHIDE_COMPILE_ARCHIVE=$(RHIDE_AR) $(RHIDE_ARFLAGS) $(OUTFILE)\
 	$(ALL_PREREQ)
