diff -Naur gzip-1.2.4a-orig/Makefile.in gzip-1.2.4a/Makefile.in
--- gzip-1.2.4a-orig/Makefile.in	1993-08-17 10:12:40.000000000 -0700
+++ gzip-1.2.4a/Makefile.in	2005-04-21 11:45:27.000000000 -0700
@@ -348,27 +348,27 @@
 	rm -f _match.s
 
 $(G)zdiff: zdiff.in
-	sed -e "$(SEDCMD)" -e "s|BINDIR|$(bindir)|" $(srcdir)/zdiff.in > $@
+	sed -e "$(SEDCMD)" -e "s|BINDIR|/bin|" $(srcdir)/zdiff.in > $@
 	chmod 755 $@
 
 $(G)zgrep: zgrep.in
-	sed -e "$(SEDCMD)" -e "s|BINDIR|$(bindir)|" $(srcdir)/zgrep.in > $@
+	sed -e "$(SEDCMD)" -e "s|BINDIR|/bin|" $(srcdir)/zgrep.in > $@
 	chmod 755 $@
 
 $(G)zmore: zmore.in
-	sed -e "$(SEDCMD)" -e "s|BINDIR|$(bindir)|" $(srcdir)/zmore.in > $@
+	sed -e "$(SEDCMD)" -e "s|BINDIR|/bin|" $(srcdir)/zmore.in > $@
 	chmod 755 $@
 
 $(G)znew: znew.in
-	sed -e "$(SEDCMD)" -e "s|BINDIR|$(bindir)|" $(srcdir)/znew.in > $@
+	sed -e "$(SEDCMD)" -e "s|BINDIR|/bin|" $(srcdir)/znew.in > $@
 	chmod 755 $@
 
 $(G)zforce: zforce.in
-	sed -e "$(SEDCMD)" -e "s|BINDIR|$(bindir)|" $(srcdir)/zforce.in > $@
+	sed -e "$(SEDCMD)" -e "s|BINDIR|/bin|" $(srcdir)/zforce.in > $@
 	chmod 755 $@
 
 gzexe: gzexe.in
-	sed -e "$(SEDCMD)" -e "s|BINDIR|$(bindir)|" $(srcdir)/gzexe.in > $@
+	sed -e "$(SEDCMD)" -e "s|BINDIR|/bin|" $(srcdir)/gzexe.in > $@
 	chmod 755 $@
 
 gzip.info: gzip.texi
