diff -upr AFL-2.57b.orig/llvm_mode/afl-clang-fast.c AFL-2.57b/llvm_mode/afl-clang-fast.c
--- AFL-2.57b.orig/llvm_mode/afl-clang-fast.c	2023-01-06 15:21:37.896498183 +0200
+++ AFL-2.57b/llvm_mode/afl-clang-fast.c	2023-01-06 15:23:42.810020308 +0200
@@ -113,10 +113,10 @@ static void edit_params(u32 argc, char**
 
   if (!strcmp(name, "afl-clang-fast++")) {
     u8* alt_cxx = getenv("AFL_CXX");
-    cc_params[0] = alt_cxx ? alt_cxx : (u8*)"clang++";
+    cc_params[0] = alt_cxx ? alt_cxx : (u8*)"/usr/lib/llvm14/bin/clang++";
   } else {
     u8* alt_cc = getenv("AFL_CC");
-    cc_params[0] = alt_cc ? alt_cc : (u8*)"clang";
+    cc_params[0] = alt_cc ? alt_cc : (u8*)"/usr/lib/llvm14/bin/clang";
   }
 
   /* There are two ways to compile afl-clang-fast. In the traditional mode, we
