diff -upr AFL-2.57b.orig/llvm_mode/afl-clang-fast.c AFL-2.57b/llvm_mode/afl-clang-fast.c
--- AFL-2.57b.orig/llvm_mode/afl-clang-fast.c	2020-07-04 19:24:18.000000000 +0300
+++ AFL-2.57b/llvm_mode/afl-clang-fast.c	2021-10-25 03:48:15.025835829 +0300
@@ -140,6 +140,7 @@ static void edit_params(u32 argc, char**
 #endif /* ^USE_TRACE_PC */
 
   cc_params[cc_par_cnt++] = "-Qunused-arguments";
+  cc_params[cc_par_cnt++] = "-flegacy-pass-manager";
 
   while (--argc) {
     u8* cur = *(++argv);
