--- a/mimalloc-release.cmake
+++ b/mimalloc-release.cmake
@@ -15,15 +15,5 @@ set_target_properties(mimalloc PROPERTIES
 list(APPEND _cmake_import_check_targets mimalloc )
 list(APPEND _cmake_import_check_files_for_mimalloc "${_IMPORT_PREFIX}/lib/libmimalloc.so.2.2" )
 
-# Import target "mimalloc-static" for configuration "Release"
-set_property(TARGET mimalloc-static APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
-set_target_properties(mimalloc-static PROPERTIES
-  IMPORTED_LINK_INTERFACE_LANGUAGES_RELEASE "C"
-  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/lib/libmimalloc.a"
-  )
-
-list(APPEND _cmake_import_check_targets mimalloc-static )
-list(APPEND _cmake_import_check_files_for_mimalloc-static "${_IMPORT_PREFIX}/lib/libmimalloc.a" )
-
 # Commands beyond this point should not need to know the version.
 set(CMAKE_IMPORT_FILE_VERSION)
--- a/mimalloc.cmake
+++ b/mimalloc.cmake
@@ -19,7 +19,7 @@ set(CMAKE_IMPORT_FILE_VERSION 1)
 set(_cmake_targets_defined "")
 set(_cmake_targets_not_defined "")
 set(_cmake_expected_targets "")
-foreach(_cmake_expected_target IN ITEMS mimalloc mimalloc-static)
+foreach(_cmake_expected_target IN ITEMS mimalloc)
   list(APPEND _cmake_expected_targets "${_cmake_expected_target}")
   if(TARGET "${_cmake_expected_target}")
     list(APPEND _cmake_targets_defined "${_cmake_expected_target}")
@@ -71,14 +71,6 @@ set_target_properties(mimalloc PROPERTIES
   INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
 )
 
-# Create imported target mimalloc-static
-add_library(mimalloc-static STATIC IMPORTED)
-
-set_target_properties(mimalloc-static PROPERTIES
-  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
-  INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY:pthread>;\$<LINK_ONLY:rt>;\$<LINK_ONLY:atomic>"
-)
-
 # Load information for each installed configuration.
 file(GLOB _cmake_config_files "${CMAKE_CURRENT_LIST_DIR}/mimalloc-*.cmake")
 foreach(_cmake_config_file IN LISTS _cmake_config_files)
