*** Makefile.ref.orig	2009-02-16 09:56:29.000000000 -0600
--- Makefile.ref	2009-02-16 09:58:59.000000000 -0600
***************
*** 169,184 ****
--- 169,185 ----
  	jstypes.h	\
  	jsprvtd.h	\
  	jspubtd.h	\
  	jsregexp.h	\
  	jsscan.h	\
  	jsscope.h	\
  	jsscript.h	\
  	jsstr.h		\
+ 	jsutil.h        \
  	jsxdrapi.h	\
  	jsxml.h		\
  	$(NULL)
  
  API_HFILES =		\
  	jsapi.h		\
  	jsdbgapi.h	\
  	$(NULL)
