--- a/boards.cfg	2010-11-07 21:03:05.000000000 -0600
+++ b/boards.cfg	2010-11-07 21:11:49.000000000 -0600
@@ -376,6 +376,7 @@
 mv88f6281gtw_ge	arm	arm926ejs	-		Marvell		kirkwood
 openrd_base	arm	arm926ejs	-		Marvell		kirkwood
 sheevaplug	arm	arm926ejs	-		Marvell		kirkwood
+iconnect	arm	arm926ejs	-		Marvell		kirkwood
 imx27lite	arm	arm926ejs	imx27lite	logicpd		mx27
 magnesium	arm	arm926ejs	imx27lite	logicpd		mx27
 omap5912osk	arm	arm926ejs	-		ti		omap
