diff -rupN qt-creator-opensource-src-3.6.0/src/tools/clangbackend/clangbackend.pro qt-creator-opensource-src-3.6.0-new/src/tools/clangbackend/clangbackend.pro
--- qt-creator-opensource-src-3.6.0/src/tools/clangbackend/clangbackend.pro	2015-12-11 13:45:01.000000000 +0100
+++ qt-creator-opensource-src-3.6.0-new/src/tools/clangbackend/clangbackend.pro	2016-02-08 23:04:03.654903983 +0100
@@ -10,7 +10,7 @@ QT += core network
 QT -= gui
 
 LIBS += $$LLVM_LIBS
-INCLUDEPATH += $$LLVM_INCLUDEPATH
+#INCLUDEPATH += $$LLVM_INCLUDEPATH
 
 SOURCES += clangbackendmain.cpp
 
