diff --git a/src/plugins/clangtools/clangtidyclazyrunner.cpp b/src/plugins/clangtools/clangtidyclazyrunner.cpp
index a580704243..7a8740a5cc 100644
--- a/src/plugins/clangtools/clangtidyclazyrunner.cpp
+++ b/src/plugins/clangtools/clangtidyclazyrunner.cpp
@@ -76,6 +76,7 @@ QStringList ClangTidyClazyRunner::constructCommandLineArguments(const QStringLis
 
     const ClangDiagnosticConfig::TidyMode tidyMode = m_diagnosticConfig.clangTidyMode();
     if (tidyMode != ClangDiagnosticConfig::TidyMode::Disabled) {
+        arguments << XclangArgs({"-load", "libclangTidyPlugin.so"});
         arguments << XclangArgs({"-add-plugin", "clang-tidy"});
         if (tidyMode != ClangDiagnosticConfig::TidyMode::File) {
             const QString tidyChecks = m_diagnosticConfig.clangTidyChecks();
@@ -85,7 +86,7 @@ QStringList ClangTidyClazyRunner::constructCommandLineArguments(const QStringLis
 
     const QString clazyChecks = m_diagnosticConfig.clazyChecks();
     if (!clazyChecks.isEmpty()) {
-        arguments << XclangArgs({"-add-plugin",
+        arguments << XclangArgs({"-load", "ClazyPlugin.so", "-add-plugin",
                                  "clazy",
                                  "-plugin-arg-clazy",
                                  "enable-all-fixits",
