diff --git a/src/plugins/clangcodemodel/clangeditordocumentprocessor.cpp b/src/plugins/clangcodemodel/clangeditordocumentprocessor.cpp
index 25054f14ec..b0e17c3260 100644
--- a/src/plugins/clangcodemodel/clangeditordocumentprocessor.cpp
+++ b/src/plugins/clangcodemodel/clangeditordocumentprocessor.cpp
@@ -524,6 +524,7 @@ private:
         if (tidyMode == Mode::Disabled)
             return;
 
+        addXclangArg("-load", "libclangTidyPlugin.so");
         addXclangArg("-add-plugin", "clang-tidy");
 
         if (tidyMode == Mode::File)
@@ -539,7 +540,8 @@ private:
         if (checks.isEmpty())
             return;

-        addXclangArg("-add-plugin", "clang-lazy");
+        addXclangArg("-load", "ClazyPlugin.so");
+        addXclangArg("-add-plugin", "clazy");
         addXclangArg("-plugin-arg-clang-lazy", "enable-all-fixits");
         addXclangArg("-plugin-arg-clang-lazy", "no-autowrite-fixits");
         addXclangArg("-plugin-arg-clang-lazy", checks);
diff --git a/src/plugins/clangtools/clangtidyclazyrunner.cpp b/src/plugins/clangtools/clangtidyclazyrunner.cpp
index d91724392e..75e1692f49 100644
--- a/src/plugins/clangtools/clangtidyclazyrunner.cpp
+++ b/src/plugins/clangtools/clangtidyclazyrunner.cpp
@@ -83,6 +83,7 @@ QStringList ClangTidyClazyRunner::constructCommandLineArguments(const QStringLis
 
     const ClangDiagnosticConfig::TidyMode tidyMode = m_diagnosticConfig.clangTidyMode();
     if (tidyMode != ClangDiagnosticConfig::TidyMode::Disabled) {
+        addXclangArg(arguments, QString("-load"), QString("libclangTidyPlugin.so"));
         addXclangArg(arguments, QString("-add-plugin"), QString("clang-tidy"));
         if (tidyMode != ClangDiagnosticConfig::TidyMode::File) {
             const QString tidyChecks = m_diagnosticConfig.clangTidyChecks();
@@ -92,7 +93,8 @@ QStringList ClangTidyClazyRunner::constructCommandLineArguments(const QStringLis
 
     const QString clazyChecks = m_diagnosticConfig.clazyChecks();
     if (!clazyChecks.isEmpty()) {
-        addXclangArg(arguments, QString("-add-plugin"), QString("clang-lazy"));
+        addXclangArg(arguments, QString("-load"), QString("ClazyPlugin.so"));
+        addXclangArg(arguments, QString("-add-plugin"), QString("clazy"));
         addXclangArg(arguments, QString("-plugin-arg-clang-lazy"), QString("enable-all-fixits"));
         addXclangArg(arguments, QString("-plugin-arg-clang-lazy"), QString("no-autowrite-fixits"));
         addXclangArg(arguments, QString("-plugin-arg-clang-lazy"), m_diagnosticConfig.clazyChecks());
