diff --git a/src/plugins/clangcodemodel/clangeditordocumentprocessor.cpp b/src/plugins/clangcodemodel/clangeditordocumentprocessor.cpp
index 706c5101a3..dbfaeb8457 100644
--- a/src/plugins/clangcodemodel/clangeditordocumentprocessor.cpp
+++ b/src/plugins/clangcodemodel/clangeditordocumentprocessor.cpp
@@ -527,14 +528,14 @@ private:
             return;
 
         m_options.append(CppTools::XclangArgs({"-add-plugin",
-                                               "clang-lazy",
-                                               "-plugin-arg-clang-lazy",
+                                               "clazy",
+                                               "-plugin-arg-clazy",
                                                "enable-all-fixits",
-                                               "-plugin-arg-clang-lazy",
+                                               "-plugin-arg-clazy",
                                                "no-autowrite-fixits",
-                                               "-plugin-arg-clang-lazy",
+                                               "-plugin-arg-clazy",
                                                checks,
-                                               "-plugin-arg-clang-lazy",
+                                               "-plugin-arg-clazy",
                                                "ignore-included-files"}));
     }
 
diff --git a/src/plugins/clangtools/clangtidyclazyrunner.cpp b/src/plugins/clangtools/clangtidyclazyrunner.cpp
index a580704243..7a8740a5cc 100644
--- a/src/plugins/clangtools/clangtidyclazyrunner.cpp
+++ b/src/plugins/clangtools/clangtidyclazyrunner.cpp
@@ -86,12 +87,12 @@ QStringList ClangTidyClazyRunner::constructCommandLineArguments(const QStringLis
     const QString clazyChecks = m_diagnosticConfig.clazyChecks();
     if (!clazyChecks.isEmpty()) {
         arguments << XclangArgs({"-add-plugin",
-                                 "clang-lazy",
-                                 "-plugin-arg-clang-lazy",
+                                 "clazy",
+                                 "-plugin-arg-clazy",
                                  "enable-all-fixits",
-                                 "-plugin-arg-clang-lazy",
+                                 "-plugin-arg-clazy",
                                  "no-autowrite-fixits",
-                                 "-plugin-arg-clang-lazy",
+                                 "-plugin-arg-clazy",
                                  m_diagnosticConfig.clazyChecks()});
     }
 
