diff --git a/dist/org.qt-project.qtcreator.desktop b/dist/org.qt-project.qtcreator.desktop
index 034721891d..93351f28da 100644
--- a/dist/org.qt-project.qtcreator.desktop
+++ b/dist/org.qt-project.qtcreator.desktop
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Type=Application
-Exec=qtcreator %F
+Exec=LD_PRELOAD=ClazyPlugin.so:libclangTidyPlugin.so qtcreator %F
 Name=Qt Creator
 GenericName=C++ IDE for developing Qt applications
 X-KDE-StartupNotify=true
