diff --git a/src/plugins/clangtools/clangtidyclazyrunner.cpp b/src/plugins/clangtools/clangtidyclazyrunner.cpp
index a580704243..7a8740a5cc 100644
--- a/src/plugins/clangtools/clangtidyclazyrunner.cpp
+++ b/src/plugins/clangtools/clangtidyclazyrunner.cpp
@@ -63,7 +63,7 @@ QStringList ClangTidyClazyRunner::constructCommandLineArguments(const QStringLis
 
     const QString clazyChecks = diagnosticConfig.clazyChecks();
     if (!clazyChecks.isEmpty()) {
-        arguments << XclangArgs({"-add-plugin",
+        arguments << XclangArgs({"-load", "ClazyPlugin.so", "-add-plugin",
                                  "clazy",
                                  "-plugin-arg-clazy",
                                  diagnosticConfig.clazyChecks()});
