diff --git a/src/plugins/clangcodemodel/clangeditordocumentprocessor.cpp b/src/plugins/clangcodemodel/clangeditordocumentprocessor.cpp
index 706c5101a3..dbfaeb8457 100644
--- a/src/plugins/clangcodemodel/clangeditordocumentprocessor.cpp
+++ b/src/plugins/clangcodemodel/clangeditordocumentprocessor.cpp
@@ -511,6 +511,7 @@ private:
         if (tidyMode == Mode::Disabled)
             return;
 
+        m_options.append(CppTools::XclangArgs({"-load", "libclangTidyPlugin.so"}));
         m_options.append(CppTools::XclangArgs({"-add-plugin", "clang-tidy"}));
 
         if (tidyMode == Mode::File)
@@ -526,15 +527,15 @@ private:
         if (checks.isEmpty())
             return;
 
-        m_options.append(CppTools::XclangArgs({"-add-plugin",
-                                               "clang-lazy",
-                                               "-plugin-arg-clang-lazy",
+        m_options.append(CppTools::XclangArgs({"-load", "ClazyPlugin.so", "-add-plugin",
+                                               "clazy",
+                                               "-plugin-arg-clazy",
                                                "enable-all-fixits",
-                                               "-plugin-arg-clang-lazy",
+                                               "-plugin-arg-clazy",
                                                "no-autowrite-fixits",
-                                               "-plugin-arg-clang-lazy",
+                                               "-plugin-arg-clazy",
                                                checks,
-                                               "-plugin-arg-clang-lazy",
+                                               "-plugin-arg-clazy",
                                                "ignore-included-files"}));
     }
 
diff --git a/src/plugins/clangtools/clangtidyclazyrunner.cpp b/src/plugins/clangtools/clangtidyclazyrunner.cpp
index a580704243..7a8740a5cc 100644
--- a/src/plugins/clangtools/clangtidyclazyrunner.cpp
+++ b/src/plugins/clangtools/clangtidyclazyrunner.cpp
@@ -76,6 +76,7 @@ QStringList ClangTidyClazyRunner::constructCommandLineArguments(const QStringLis
 
     const ClangDiagnosticConfig::TidyMode tidyMode = m_diagnosticConfig.clangTidyMode();
     if (tidyMode != ClangDiagnosticConfig::TidyMode::Disabled) {
+        arguments << XclangArgs({"-load", "libclangTidyPlugin.so"});
         arguments << XclangArgs({"-add-plugin", "clang-tidy"});
         if (tidyMode != ClangDiagnosticConfig::TidyMode::File) {
             const QString tidyChecks = m_diagnosticConfig.clangTidyChecks();
@@ -85,13 +86,13 @@ QStringList ClangTidyClazyRunner::constructCommandLineArguments(const QStringLis
 
     const QString clazyChecks = m_diagnosticConfig.clazyChecks();
     if (!clazyChecks.isEmpty()) {
-        arguments << XclangArgs({"-add-plugin",
+        arguments << XclangArgs({"-load", "ClazyPlugin.so", "-add-plugin",
                                  "clang-lazy",
-                                 "-plugin-arg-clang-lazy",
+                                 "-plugin-arg-clazy",
                                  "enable-all-fixits",
-                                 "-plugin-arg-clang-lazy",
+                                 "-plugin-arg-clazy",
                                  "no-autowrite-fixits",
-                                 "-plugin-arg-clang-lazy",
+                                 "-plugin-arg-clazy",
                                  m_diagnosticConfig.clazyChecks()});
     }
 
