diff -Nur xchat-systray-integration-2.4.5/src/menu.c xchat-systray-integration-2.4.5-segfault-fix/src/menu.c
--- xchat-systray-integration-2.4.5/src/menu.c	2004-04-30 14:38:10.000000000 +0200
+++ xchat-systray-integration-2.4.5-segfault-fix/src/menu.c	2004-08-25 00:34:04.321825360 +0200
@@ -10,7 +10,6 @@
 	char command[156]="exec ";
 	strcat(command, bpath);
 	xchat_command( ph, command );
-	g_free(command);
 
 }
 
@@ -20,7 +19,6 @@
 	char command[156]="exec ";
 	strcat(command, epath);
 	xchat_command( ph, command );
-	g_free(command);
 
 }
 
@@ -30,7 +28,6 @@
 	char command[156]="exec ";
 	strcat(command, ppath);
 	xchat_command( ph, command );
-	g_free(command);
 
 }
 
