From 1c95e056224ab6399b84d69fb90572b37a8a53a5 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 18 Aug 2015 17:37:44 -0600
Subject: [PATCH] remove qreal cast for arm

---
 src/gui/widgets/starrating.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui/widgets/starrating.cpp b/src/gui/widgets/starrating.cpp
index 664f12f..d3dbb4c 100644
--- a/src/gui/widgets/starrating.cpp
+++ b/src/gui/widgets/starrating.cpp
@@ -250,7 +250,7 @@ double StarRating::percentForPosition(const QPoint &pos, int numTotalStars, cons
     const int width = starSize * numTotalStars;
     const int x = pos.x() - Private::paintMargin - inside.left();
     const double percent = x * 100.0 / width;
-    return qMax(qreal(0.0), qMin(qreal(100.0), percent));
+    return qMax(0.0, qMin(100.0, percent));
 }
 
 bool StarRatingFieldInput::reset(const Value &value)
-- 
2.5.0

