diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 04222a74..1485acc1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -34,29 +34,6 @@ include(
 ##SET(CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO} -pg")
 ##SET(CMAKE_CXX_FLAGS_DISTRIBUTION "${CMAKE_CXX_FLAGS_DISTRIBUTION} -pg")
 
-# check if QtWebKit is available, which seems to be not available
-# on e.g. RedHat Enterprise Linux 6 or Scientific Linux 6
-if(
-    QT_QTWEBKIT_FOUND
-)
-    message(
-        STATUS
-        "Found QtWebKit, enabling WebKit support"
-    )
-    add_definitions(
-        -DHAVE_QTWEBKIT
-    )
-else(
-    QT_QTWEBKIT_FOUND
-)
-    message(
-        STATUS
-        "QtWebKit not found, disabling WebKit support"
-    )
-endif(
-    QT_QTWEBKIT_FOUND
-)
-
 add_definitions(
     ${QT_DEFINITIONS}
     ${KDE4_DEFINITIONS}
