From 7f41bb239a89d1e66e954577d03b92373ad4da0e Mon Sep 17 00:00:00 2001
From: hannes <hannes@fedoraproject.org>
Date: Sun, 13 Sep 2015 16:40:51 +0200
Subject: [PATCH 3/3] debughelper arm fix

---
 debughelper.cpp | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/debughelper.cpp b/debughelper.cpp
index 4b14ae3..f07c48e 100644
--- a/debughelper.cpp
+++ b/debughelper.cpp
@@ -501,16 +501,12 @@ QString print_backtrace(const QString& message){
 #define FRAME_FROM_UCONTEXT(context) (context)->uc_mcontext.gp_regs[31] //not always used
 #define RETURNTO_FROM_UCONTEXT(context) (context)->uc_mcontext.gp_regs[34]
 #elif defined(CPU_IS_ARM)
-/*
+
 #define PC_FROM_UCONTEXT(context) (context)->uc_mcontext.arm_pc
 #define STACK_FROM_UCONTEXT(context) (context)->uc_mcontext.arm_sp
 #define FRAME_FROM_UCONTEXT(context) (context)->uc_mcontext.arm_fp
 #define RETURNTO_FROM_UCONTEXT(context) (context)->uc_mcontext.arm_lr
-*/
-#define PC_FROM_UCONTEXT(context) (context)->uc_mcontext.__gregs[_REG_R15]
-#define STACK_FROM_UCONTEXT(context) (context)->uc_mcontext.__gregs[_REG_R13]
-#define FRAME_FROM_UCONTEXT(context) (context)->uc_mcontext.__gregs[_REG_R11]
-#define RETURNTO_FROM_UCONTEXT(context) (context)->uc_mcontext.__gregs[_REG_R14]
+
 #elif defined(CPU_IS_IA64)
 #define PC_FROM_UCONTEXT(context) (context)->_u._mc.sc_ip
 #define STACK_FROM_UCONTEXT(context) (context)->_u._mc.sc_gr[12] //is that register 12?
-- 
2.4.3

