From 995e2062399bb1f3d1afcedf5a8128b3c6c39bb5 Mon Sep 17 00:00:00 2001
From: Joe George <joe@zeroc.com>
Date: Fri, 12 Jun 2020 13:58:35 -0400
Subject: [PATCH] Skip Proguard build of IceGridGUI if using Java 14

---
 java/src/IceGridGUI/build.gradle | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/java/src/IceGridGUI/build.gradle b/java/src/IceGridGUI/build.gradle
index 39cecabfe6..ed81a8bcaf 100644
--- a/java/src/IceGridGUI/build.gradle
+++ b/java/src/IceGridGUI/build.gradle
@@ -81,9 +81,14 @@ jar {
 
 project.ext.libJars = []
 
-project.ext.jarBuilder = "plain-jar.gradle"
-if(icegridguiProguard.toBoolean()) {
-    project.ext.jarBuilder = "proguard-jar.gradle"
+project.ext.jarBuilder = "proguard-jar.gradle"
+try {
+    // Proguard does not support Java >= 14
+    if (!icegridguiProguard.toBoolean() || JavaVersion.current() >= JavaVersion.valueOf("VERSION_14")) {
+        project.ext.jarBuilder = "plain-jar.gradle"
+    }
+} catch (IllegalArgumentException) {
+    // Old version of Gralde that that does not know about Java 14
 }
 
 apply from: jarBuilder
