diff --git a/opt/cuda/targets/sbsa-linux/include/crt/math_functions.h b/opt/cuda/targets/sbsa-linux/include/crt/math_functions.h
index a083253..b811f97 100644
--- a/opt/cuda/targets/sbsa-linux/include/crt/math_functions.h
+++ b/opt/cuda/targets/sbsa-linux/include/crt/math_functions.h
@@ -594,7 +594,7 @@ extern __DEVICE_FUNCTIONS_DECL__ __device_builtin__ double         __cdecl sqrt(
  *
  * \note_accuracy_double
  */
-extern __DEVICE_FUNCTIONS_DECL__ __device_builtin__ double                 rsqrt(double x);
+extern __DEVICE_FUNCTIONS_DECL__ __device_builtin__ double                 rsqrt(double x) noexcept(true);

 /**
  * \ingroup CUDA_MATH_SINGLE
@@ -618,7 +618,7 @@ extern __DEVICE_FUNCTIONS_DECL__ __device_builtin__ double                 rsqrt
  *
  * \note_accuracy_single
  */
-extern __DEVICE_FUNCTIONS_DECL__ __device_builtin__ float                  rsqrtf(float x);
+extern __DEVICE_FUNCTIONS_DECL__ __device_builtin__ float                  rsqrtf(float x) noexcept(true);

 #if defined(__QNX__) && !defined(_LIBCPP_VERSION)
 namespace std {
