From 769091d09aa4a83ff91d2db2f7a1311f8b97c737 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Thu, 7 Mar 2019 19:27:59 -0700
Subject: [PATCH 1/2] sd_fusing tweaks

---
 sd_fuse/sd_fusing.sh | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/sd_fuse/sd_fusing.sh b/sd_fuse/sd_fusing.sh
index 451013f119..538fdfa3b5 100755
--- a/sd_fuse/sd_fusing.sh
+++ b/sd_fuse/sd_fusing.sh
@@ -12,12 +12,11 @@ abort() {
 }
 
 [ -z $1 ] && abort "usage: $0 <your/memory/card/device>"
-[ -z ${UBOOT} ] && UBOOT=${PWD}/u-boot.bin
-[ ! -f ${UBOOT} ] && abort "error: ${UBOOT} is not exist"
+[ -z ${UBOOT} ] && UBOOT=/boot/u-boot.bin
+[ ! -f ${UBOOT} ] && abort "Error: ${UBOOT} doesn't exist"
 
-sudo dd if=$UBOOT of=$1 conv=fsync,notrunc bs=512 seek=1
+dd if=$UBOOT of=$1 conv=fsync,notrunc bs=512 seek=1
 
 sync
 
-sudo eject $1
 echo Finished.
-- 
2.20.1

