--- openbox-3.4.7.2/data/autostart.sh.in~	2009-10-03 15:24:20.856675350 +0200
+++ openbox-3.4.7.2/data/autostart.sh.in	2009-10-03 15:28:34.543315964 +0200
@@ -3,13 +3,13 @@
 
 # Set a background color
 BG=""
-if which hsetroot >/dev/null; then
+if which hsetroot >/dev/null 2>&1; then
     BG=hsetroot
 else
-    if which esetroot >/dev/null; then
+    if which esetroot >/dev/null 2>&1; then
 	BG=esetroot
     else
-	if which xsetroot >/dev/null; then
+	if which xsetroot >/dev/null 2>&1; then
 	    BG=xsetroot
 	fi
     fi
@@ -17,28 +17,28 @@
 test -z $BG || $BG -solid "#303030"
 
 # D-bus
-if which dbus-launch >/dev/null && test -z "$DBUS_SESSION_BUS_ADDRESS"; then
+if which dbus-launch >/dev/null 2>&1 && test -z "$DBUS_SESSION_BUS_ADDRESS"; then
        eval `dbus-launch --sh-syntax --exit-with-session`
 fi
 
 # Make GTK apps look and behave how they were set up in the gnome config tools
 if test -x /usr/libexec/gnome-settings-daemon >/dev/null; then
   /usr/libexec/gnome-settings-daemon &
-elif which gnome-settings-daemon >/dev/null; then
+elif which gnome-settings-daemon >/dev/null 2>&1; then
   gnome-settings-daemon &
 # Make GTK apps look and behave how they were set up in the XFCE config tools
-elif which xfce-mcs-manager >/dev/null; then
+elif which xfce-mcs-manager >/dev/null 2>&1; then
   xfce-mcs-manager n &
 fi
 
 # Preload stuff for KDE apps
-if which start_kdeinit >/dev/null; then
+if which start_kdeinit >/dev/null 2>&1; then
   LD_BIND_NOW=true start_kdeinit --new-startup +kcminit_startup &
 fi
 
 # Run XDG autostart things.  By default don't run anything desktop-specific
 # See xdg-autostart --help more info
 DESKTOP_ENV=""
-if which @secretbindir@/xdg-autostart >/dev/null; then
+if which @secretbindir@/xdg-autostart >/dev/null 2>&1; then
   @secretbindir@/xdg-autostart $DESKTOP_ENV
 fi
