From a6b29a74b492f8792c74304081e454ef9f314324 Mon Sep 17 00:00:00 2001
From: graysky <therealgraysky@proton.me>
Date: Tue, 1 Nov 2022 07:44:06 -0400
Subject: [PATCH] store.go: do not call Age()

---
 jump/store.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/jump/store.go b/jump/store.go
index 855e003..8b4f5d0 100644
--- a/jump/store.go
+++ b/jump/store.go
@@ -30,7 +30,7 @@ func (s Store) AddPath(pathToAdd string) error {
 	if err != nil {
 		return err
 	}
-	oldEntries.Age()
+	// oldEntries.Age()
 	newEntries := oldEntries.Update(path, defaultWeight)
 	return s.saveEntries(newEntries)
 }
-- 
2.38.1

