From e742611fb1295dd394a39f9492782dfcbc107142 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Wed, 14 Mar 2018 14:18:25 -0600
Subject: [PATCH] Fix compile on ARM

---
 src/logging.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/logging.c b/src/logging.c
index 1a4f620..0b81b9e 100644
--- a/src/logging.c
+++ b/src/logging.c
@@ -1150,7 +1150,7 @@ _gpgrt_log_printhex (const void *buffer, size_t length,
       va_end (arg_ptr);
     }
   else
-    _gpgrt_logv_printhex (buffer, length, NULL, NULL);
+    _gpgrt_logv_printhex (buffer, length, NULL, arg_ptr);
 }
 
 
-- 
2.16.2

