diff -ur mod_python-3.2.8/configure.in mod_python-3.2.8-new/configure.in
--- mod_python-3.2.8/configure.in	2005-10-28 09:06:22.000000000 -0700
+++ mod_python-3.2.8-new/configure.in	2006-03-06 10:43:28.000000000 -0800
@@ -287,7 +287,7 @@
 # get the mod_python version
 AC_SUBST(MP_VERSION)
 MP_VERSION=`awk '/MPV_STRING/ {print $3}' src/include/mpversion.h`
-MP_VERSION=`echo $MP_VERSION | sed s/\\"//g`                                      
+MP_VERSION=`echo $MP_VERSION | sed s/\"//g`                                      
 
 # get --with-python-src. The python src is required to generate the documentation
 # It is not required to compile or install mod_python itself
