diff -ruN LuaJIT-1.1.4.orig/src/luaconf.h LuaJIT-1.1.3/src/luaconf.h
--- LuaJIT-1.1.4.orig/src/luaconf.h	2007-05-24 13:30:00.000000000 +0200
+++ LuaJIT-1.1.4/src/luaconf.h	2007-11-02 06:28:59.000000000 +0100
@@ -94,7 +94,7 @@
 	".\\?.dll;"  LUA_CDIR"?.dll;" LUA_CDIR"loadall.dll"
 
 #else
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"/usr/"
 #define LUA_LDIR	LUA_ROOT "share/lua/5.1/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/5.1/"
 #define LUA_PATH_DEFAULT  \
