 libgnome-desktop/gnome-desktop-thumbnail-script.c | 1 +
 1 file changed, 1 insertion(+)

diff --git c/libgnome-desktop/gnome-desktop-thumbnail-script.c i/libgnome-desktop/gnome-desktop-thumbnail-script.c
index 99d83ac25d4cb449..7f85da2dec60f454 100644
--- c/libgnome-desktop/gnome-desktop-thumbnail-script.c
+++ i/libgnome-desktop/gnome-desktop-thumbnail-script.c
@@ -505,6 +505,7 @@ add_bwrap (GPtrArray   *array,
 
   add_args (array,
 	    "bwrap",
+	    "--ro-bind", "/var", "/var",
 	    "--ro-bind", "/usr", "/usr",
 	    "--ro-bind", "/lib", "/lib",
 	    "--ro-bind", "/lib64", "/lib64",
