--- soltools/mkdepend/def.h.orig	2009-05-18 15:28:59.000000000 +0200
+++ soltools/mkdepend/def.h	2009-05-18 15:29:36.000000000 +0200
@@ -154,7 +154,7 @@
 
 char			*copy();
 char			*base_name();
-char			*getline();
+char			*ooo_getline();
 char			*isdefined();
 struct filepointer	*getfile();
 struct inclist		*newinclude();
--- soltools/mkdepend/main.c.orig	2009-05-18 15:29:08.000000000 +0200
+++ soltools/mkdepend/main.c	2009-05-18 15:30:23.000000000 +0200
@@ -548,7 +548,7 @@
  * Get the next line.  We only return lines beginning with '#' since that
  * is all this program is ever interested in.
  */
-char *getline(filep)
+char *ooo_getline(filep)
 	register struct filepointer	*filep;
 {
 	register char	*p,	/* walking pointer */
--- soltools/mkdepend/parse.c.orig	2009-05-18 15:29:17.000000000 +0200
+++ soltools/mkdepend/parse.c	2009-05-18 15:30:43.000000000 +0200
@@ -45,7 +45,7 @@
 	register int	type;
 	boolean recfailOK;
 
-	while ((line = getline(filep))) {
+	while ((line = ooo_getline(filep))) {
 		switch(type = deftype(line, filep, file_red, file, TRUE, symbols)) {
 		case IF:
 		doif:
@@ -168,7 +168,7 @@
 	register char	*line;
 	register int	type;
 
-	while ((line = getline(filep))) {
+	while ((line = ooo_getline(filep))) {
 		switch(type = deftype(line, filep, file_red, file, FALSE, symbols)) {
 		case IF:
 		case IFFALSE:

