Index: source/plugin/inc/plugin/unx/plugcon.hxx
===================================================================
RCS file: /cvs/util/extensions/source/plugin/inc/plugin/unx/plugcon.hxx,v
retrieving revision 1.13.64.2
diff -u -r1.13.64.2 plugcon.hxx
--- extensions/source/plugin/inc/plugin/unx/plugcon.hxx	15 Dec 2008 10:29:47 -0000	1.13.64.2
+++ extensions/source/plugin/inc/plugin/unx/plugcon.hxx	23 Dec 2008 09:24:57 -0000
@@ -102,7 +102,7 @@
 extern "C" {
 #include <npsdk/npupp.h>
 }
-#include <npapi.h>
+#include <npsdk/npapi.h>
 
 #if NP_VERSION_MINOR < 17
 // compatibility hack: compile with older NPN api header, but define
