diff -up qtwebkit-opensource-src-5.4.0/Source/JavaScriptCore/runtime/JSObject.cpp.than 
qtwebkit-opensource-src-5.4.0/Source/JavaScriptCore/runtime/JSObject.cpp
--- qtwebkit-opensource-src-5.4.0/Source/JavaScriptCore/runtime/JSObject.cpp.than	2015-03-18 10:24:38.683352327 
-0400
+++ qtwebkit-opensource-src-5.4.0/Source/JavaScriptCore/runtime/JSObject.cpp	2015-03-18 10:25:21.953352327 -0400
@@ -1909,6 +1909,10 @@ void JSObject::putByIndexBeyondVectorLen
     }
 }
 
+template void JSObject::putByIndexBeyondVectorLengthWithoutAttributes<ContiguousShape>(ExecState* exec, unsigned i, JSValue value);
+template void JSObject::putByIndexBeyondVectorLengthWithoutAttributes<DoubleShape>(ExecState* exec, unsigned i, JSValue value);
+template void JSObject::putByIndexBeyondVectorLengthWithoutAttributes<Int32Shape>(ExecState* exec, unsigned i, JSValue value);
+
 void JSObject::putByIndexBeyondVectorLengthWithArrayStorage(ExecState* exec, unsigned i, JSValue value, bool shouldThrow, ArrayStorage* storage)
 {
     VM& vm = exec->vm();

