diff --git a/Makefile.am b/Makefile.am
index 761a772..0a8391b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -108,17 +108,6 @@ pkginclude_HEADERS += extern/HPCombi/include/vect_generic.hpp
 endif
 
 fmtincludedir = $(includedir)/fmt
-fmtinclude_HEADERS =  extern/fmt-5.3.0/include/fmt/chrono.h  
-fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/color.h  
-fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/core.h  
-fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/format-inl.h  
-fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/format.h  
-fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/locale.h  
-fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/ostream.h	
-fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/posix.h  
-fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/printf.h  
-fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/ranges.h  
-fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/time.h
 
 lib_LTLIBRARIES = libsemigroups.la
 
diff --git a/configure.ac b/configure.ac
index c40acdb..832d8a3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -212,11 +212,5 @@ AS_IF([test "x$enable_hpcombi" = xyes],
 AM_CONDITIONAL([HPCOMBI_CONSTEXPR_FUN_ARGS], 
                [test "x$hpcombi_constexpr_fun_args" = xyes])
 
-AC_CHECK_FILE([$srcdir/extern/fmt-5.3.0/README.rst], 
-               [], 
-               []
-               [AC_MSG_ERROR([fmt-5.3.0 is not available, please download fmt-5.3.0 from https://fmt.dev/ and put the unpacked archive in the libsemigroups/extern directory])])
-dnl Output configured files
-
 AC_CONFIG_FILES([Makefile libsemigroups.pc])
 AC_OUTPUT
