--- tachyon/unix/Make-arch.orig	2019-08-24 23:13:05.136655180 +0000
+++ tachyon/unix/Make-arch	2019-08-24 23:13:36.057062493 +0000
@@ -1273,12 +1273,12 @@
 	$(MAKE) all \
 	"ARCH = linux-64-thr" \
 	"CC = cc" \
-	"CFLAGS = -m64 -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux  $(THREADSFLAGS) -D_REENTRANT $(MISCFLAGS)" \
+	"CFLAGS = $(CFLAGS) -DLinux  $(THREADSFLAGS) -D_REENTRANT $(MISCFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = strip" \
 	"RANLIB = ranlib" \
-	"LIBS = -L. -ltachyon $(MISCLIB) -lm -lpthread"
+	"LIBS = -L. -ltachyon $(MISCLIB) -lm -lpthread" "LDFLAGS = $(LDFLAGS)"
 
 # Linux on AMD64/EM64T, using gcc
 linux-64-thr-c11:
