--- Cargo.toml.orig	2023-07-15 15:39:59.618904488 +0000
+++ Cargo.toml	2023-07-15 22:04:36.249321112 +0000
@@ -169,9 +169,9 @@
 
 # NB: the `bench` and `release` profiles must remain EXACTLY the same.
 [profile.release]
-codegen-units = 1
+codegen-units = 32
 incremental = true
-lto = true
+lto = false
 opt-level = 'z' # Optimize for size
 
 # Build release with debug symbols: cargo build --profile=release-with-debug
@@ -181,9 +181,9 @@
 
 # NB: the `bench` and `release` profiles must remain EXACTLY the same.
 [profile.bench]
-codegen-units = 1
+codegen-units = 32
 incremental = true
-lto = true
+lto = false
 opt-level = 'z' # Optimize for size
 
 # Key generation is too slow on `debug`
