diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9a80f95..1bf02ab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -202,7 +202,8 @@ target_link_libraries(openshot-audio PUBLIC ZLIB::ZLIB)
 
 target_link_libraries(openshot-audio PUBLIC
 		${CMAKE_DL_LIBS}
-		${JUCE_PLATFORM_SPECIFIC_LIBRARIES} )
+		${JUCE_PLATFORM_SPECIFIC_LIBRARIES}
+		atomic)
 
 # PROCESS SUB-DIRECTORIES
 add_subdirectory(src)
