--- xfce4-battery-plugin-0.5.0/panel-plugin/libapm.h.orig	2008-03-18 20:10:58.000000000 +0100
+++ xfce4-battery-plugin-0.5.0/panel-plugin/libapm.h	2008-03-18 20:14:45.000000000 +0100
@@ -17,8 +17,8 @@
  * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <linux/apm_bios.h>
 #include <sys/types.h>
+#include <linux/apm_bios.h>
 
 #define APM_PROC   "/proc/apm"
 #define APM_DEVICE "/dev/apm_bios"

