--- Pyro-3.4/setup.cfg	2003-03-07 02:39:00.000000000 +0100
+++ Pyro-3.4/setup.cfg.new	2003-05-22 12:41:04.000000000 +0200
@@ -4,9 +4,9 @@
 ### Enable the following two items for unattended install.
 ### The scripts location is mandatory. Change it so something sensible.
 
-#[install-options]
-#unattended=1
+[install-options]
+unattended=1
 
 [install]
 optimize=1
-#install-scripts=/foo/bar
+install-scripts=/usr/bin
