Index: m4/glib-gettext.m4
===================================================================
RCS file: /cvsroot/gnubg/gnubg/m4/glib-gettext.m4,v
retrieving revision 1.4
diff -u -r1.4 glib-gettext.m4
--- m4/glib-gettext.m4	21 Apr 2016 21:30:20 -0000	1.4
+++ m4/glib-gettext.m4	2 May 2016 18:27:27 -0000
@@ -36,6 +36,8 @@
 dnl try to pull in the installed version of these macros
 dnl when running aclocal in the glib directory.
 dnl
+m4_ifdef([glib_DEFUN],[m4_undefine([glib_DEFUN])])
+m4_ifdef([glib_REQUIRE],[m4_undefine([glib_REQUIRE])])
 m4_copy([AC_DEFUN],[glib_DEFUN])
 m4_copy([AC_REQUIRE],[glib_REQUIRE])
 dnl
