From c821ac6e22b8115ee41f5f76991b3ad03d15b062 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 18 Sep 2018 01:15:09 +0000
Subject: [PATCH] set libm as required

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 4bc51d64..85a75ac9 100644
--- a/meson.build
+++ b/meson.build
@@ -81,7 +81,7 @@ gtk = dependency('gtk+-3.0', version : '>= 3.3.8')
 gdk = dependency('gdk-3.0', version : '>= 2.91.7')
 cairo = dependency('cairo', version : '>= 1.0.0')
 upower = dependency('upower-glib', version : '>= 0.9.1')
-libm = cc.find_library('libm', required: false)
+libm = cc.find_library('m', required: true)
 
 gnome = import('gnome')
 i18n = import('i18n')
-- 
2.18.0

