From 3bbff32456e959c76986921621602360d9d2d67a Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 18 Sep 2018 01:15:09 +0000
Subject: [PATCH] set libm as required

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 04da15cc..b7758201 100644
--- a/meson.build
+++ b/meson.build
@@ -75,7 +75,7 @@ endforeach
 gtk = dependency('gtk4', version: '>= 4.0.0')
 cairo = dependency('cairo', version : '>= 1.0.0')
 upower = dependency('upower-glib', version : '>= 0.99.8')
-libm = cc.find_library('libm', required: false)
+libm = cc.find_library('m', required: true)
 
 gnome = import('gnome')
 i18n = import('i18n')
-- 
2.37.3

