diff -wbBur xkbsel-0.13/configure xkbsel-0.13.my/configure
--- xkbsel-0.13/configure	2000-04-09 13:37:02.000000000 +0400
+++ xkbsel-0.13.my/configure	2006-07-24 17:13:10.000000000 +0400
@@ -6118,7 +6118,7 @@
   { echo "configure: error: "Cannot find library containing dbm_open"" 1>&2; exit 1; }
 fi
 
-x11xkbdir=$ac_x_libraries/X11/xkb
+x11xkbdir=/usr/share/X11/xkb
 test -d $x11xkbdir || { echo "configure: error: "Directory $x11xkbdir does not exist"" 1>&2; exit 1; }
 
 
diff -wbBur xkbsel-0.13/libs/Makefile.in xkbsel-0.13.my/libs/Makefile.in
--- xkbsel-0.13/libs/Makefile.in	2000-04-09 13:51:01.000000000 +0400
+++ xkbsel-0.13.my/libs/Makefile.in	2006-07-24 17:11:52.000000000 +0400
@@ -452,7 +452,7 @@
 
 
 install-data-local:
-	$(mkinstalldirs) ${sysxkmdir}
+	$(mkinstalldirs) $(DESTDIR)${sysxkmdir}
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
