diff -up icewm-1.3.8/src/icesh.cc.narrowingfix icewm-1.3.8/src/icesh.cc
--- icewm-1.3.8/src/icesh.cc.narrowingfix	2017-01-03 15:53:33.537699919 -0500
+++ icewm-1.3.8/src/icesh.cc	2017-01-03 15:53:52.717596858 -0500
@@ -245,7 +245,7 @@ Symbol trayOptionIdentifiers[] = {
 };
 
 SymbolTable layers = {
-    layerIdentifiers, 0, WinLayerCount - 1, WinLayerInvalid
+    layerIdentifiers, 0, WinLayerCount - 1, (long)WinLayerInvalid
 };
 
 SymbolTable states = {
@@ -257,7 +257,7 @@ SymbolTable hints = {
 };
 
 SymbolTable trayOptions = {
-    trayOptionIdentifiers, 0, WinTrayOptionCount - 1, WinTrayInvalid
+    trayOptionIdentifiers, 0, WinTrayOptionCount - 1, (long)WinTrayInvalid
 };
 
 /******************************************************************************/
