Index: icewm-1.3.7/src/wmapp.cc
===================================================================
--- icewm-1.3.7.orig/src/wmapp.cc	2010-10-31 15:09:36.000000000 +0100
+++ icewm-1.3.7/src/wmapp.cc	2012-01-15 17:01:47.284927292 +0100
@@ -187,7 +187,6 @@
         _XA_NET_CLOSE_WINDOW,
         _XA_NET_WM_STRUT,
         _XA_NET_WM_STRUT_PARTIAL,
-        _XA_NET_WORKAREA,
         _XA_NET_WM_STATE,
         _XA_NET_WM_STATE_MAXIMIZED_VERT,
         _XA_NET_WM_STATE_MAXIMIZED_HORZ,
Index: icewm-1.3.7/src/wmmgr.cc
===================================================================
--- icewm-1.3.7.orig/src/wmmgr.cc	2010-10-31 15:09:36.000000000 +0100
+++ icewm-1.3.7/src/wmmgr.cc	2012-01-15 16:59:18.000000000 +0100
@@ -2114,11 +2114,6 @@
         area[ws * 4 + 3] = fWorkArea[ws][0].fMaxY - fWorkArea[ws][0].fMinY;
     }
 
-    XChangeProperty(xapp->display(), handle(),
-                    _XA_NET_WORKAREA,
-                    XA_CARDINAL,
-                    32, PropModeReplace,
-                    (unsigned char *)area, nw * 4);
     delete [] area;
 #endif
 #ifdef GNOME1_HINTS
Index: icewm-1.3.7/src/wmmgr.h
===================================================================
--- icewm-1.3.7.orig/src/wmmgr.h	2010-10-31 15:09:36.000000000 +0100
+++ icewm-1.3.7/src/wmmgr.h	2012-01-15 16:59:18.000000000 +0100
@@ -297,7 +297,6 @@
 extern Atom _XA_NET_CURRENT_DESKTOP;             // OK
 ///extern Atom _XA_NET_DESKTOP_NAMES;            // N/A
 extern Atom _XA_NET_ACTIVE_WINDOW;               // OK
-extern Atom _XA_NET_WORKAREA;                    // OK
 extern Atom _XA_NET_SUPPORTING_WM_CHECK;         // OK
 ///extern Atom _XA_NET_SUPPORTING_WM_CHECK;      // N/A
 extern Atom _XA_NET_CLOSE_WINDOW;                // OK
Index: icewm-1.3.7/src/yxapp.cc
===================================================================
--- icewm-1.3.7.orig/src/yxapp.cc	2010-10-31 15:09:36.000000000 +0100
+++ icewm-1.3.7/src/yxapp.cc	2012-01-15 17:03:59.556423213 +0100
@@ -67,7 +67,7 @@
 Atom _XA_NET_CLIENT_LIST_STACKING;
 Atom _XA_NET_NUMBER_OF_DESKTOPS;
 Atom _XA_NET_CURRENT_DESKTOP;
-Atom _XA_NET_WORKAREA;
+//Atom _XA_NET_WORKAREA;
 Atom _XA_NET_WM_MOVERESIZE;
 
 Atom _XA_NET_WM_STRUT;
@@ -286,7 +286,7 @@
         { &_XA_NET_CLIENT_LIST_STACKING, "_NET_CLIENT_LIST_STACKING" },
         { &_XA_NET_NUMBER_OF_DESKTOPS, "_NET_NUMBER_OF_DESKTOPS" },
         { &_XA_NET_CURRENT_DESKTOP, "_NET_CURRENT_DESKTOP" },
-        { &_XA_NET_WORKAREA, "_NET_WORKAREA" },
+//        { &_XA_NET_WORKAREA, "_NET_WORKAREA" },
         { &_XA_NET_WM_MOVERESIZE, "_NET_WM_MOVERESIZE" },
 
         { &_XA_NET_WM_STRUT, "_NET_WM_STRUT" },
