diff -urNp coreutils-7.2-orig/gnulib-tests/gnulib.mk coreutils-7.2/gnulib-tests/gnulib.mk
--- coreutils-7.2-orig/gnulib-tests/gnulib.mk	2009-03-31 14:28:30.000000000 +0200
+++ coreutils-7.2/gnulib-tests/gnulib.mk	2009-04-01 12:37:00.000000000 +0200
@@ -606,9 +606,9 @@ EXTRA_DIST += test-mbsstr1.c test-mbsstr
 
 ## begin gnulib module memchr-tests
 
-TESTS += test-memchr
-check_PROGRAMS += test-memchr
-EXTRA_DIST += test-memchr.c zerosize-ptr.h
+#TESTS += test-memchr
+#check_PROGRAMS += test-memchr
+#EXTRA_DIST += test-memchr.c zerosize-ptr.h
 
 ## end   gnulib module memchr-tests
 
@@ -910,9 +910,9 @@ EXTRA_DIST += test-strtod.c
 
 ## begin gnulib module strverscmp-tests
 
-TESTS += test-strverscmp
-check_PROGRAMS += test-strverscmp
-EXTRA_DIST += test-strverscmp.c
+#TESTS += test-strverscmp
+#check_PROGRAMS += test-strverscmp
+#EXTRA_DIST += test-strverscmp.c
 
 ## end   gnulib module strverscmp-tests
 
diff -urNp coreutils-6.11-orig/tests/test-lib.sh coreutils-6.11/tests/test-lib.sh
--- coreutils-6.11-orig/tests/test-lib.sh	2008-04-19 23:34:23.000000000 +0200
+++ coreutils-6.11/tests/test-lib.sh	2008-04-24 14:18:59.000000000 +0200
@@ -97,8 +97,8 @@ skip_if_()
 
 require_selinux_()
 {
-  case `ls -Zd .` in
-    '? .'|'unlabeled .')
+  case `ls --scontext -d . | cut -f1 -d" "` in
+    '?'|'unlabeled')
       skip_test_ "this system (or maybe just" \
         "the current file system) lacks SELinux support"
     ;;
diff -urNp coreutils-7.1-orig/src/ls.c coreutils-7.1/src/ls.c
--- coreutils-7.1-orig/src/ls.c	2009-02-25 13:23:59.000000000 +0100
+++ coreutils-7.1/src/ls.c	2009-02-25 13:25:20.000000000 +0100
@@ -38,10 +38,6 @@
 #include <config.h>
 #include <sys/types.h>
 
-#ifdef HAVE_CAP
-# include <sys/capability.h>
-#endif
-
 #if HAVE_TERMIOS_H
 # include <termios.h>
 #endif
@@ -84,6 +80,10 @@
 #include "system.h"
 #include <fnmatch.h>
 
+#ifdef HAVE_CAP
+# include <sys/capability.h>
+#endif
+
 #include "acl.h"
 #include "argmatch.h"
 #include "dev-ino.h"
