From f66618b23bc59f2a6374edd2611e88838dc6e854 Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Thu, 24 Apr 2014 18:21:53 -0500
Subject: [PATCH 26/36] port over to generic bootcommands

---
 include/configs/sunxi-common.h | 24 +++++++++++++++++++-----
 1 file changed, 19 insertions(+), 5 deletions(-)

diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index 56a89b9..65f3244 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -112,8 +112,26 @@
 #define CONFIG_ENV_OFFSET		(544 << 10) /* (8 + 24 + 512) KiB */
 #define CONFIG_ENV_SIZE			(128 << 10)	/* 128 KiB */
 
+#ifndef CONFIG_SPL_BUILD
+#include <config_distro_defaults.h>
+#include <config_distro_bootcmd.h>
+
+#define CONFIG_DEFAULT_FDT_FILE		"sun7i-a20-cubietruck.dtb"
+
 #define CONFIG_EXTRA_ENV_SETTINGS \
-	"bootm_size=0x10000000\0"
+	"kernel_addr_r=0x42000000\0" \
+	"pxe_addr_r=0x4FE00000\0" \
+	"scriptaddr=0x4FF00000\0" \
+	"fdt_addr_r=0x50000000\0" \
+	"ramdisk_addr_r=0x51000000\0" \
+	"bootm_size=0x10000000\0" \
+	"console=ttyS0,115200\0" \
+	"fdt_file=" CONFIG_DEFAULT_FDT_FILE "\0" \
+	BOOTCMDS_COMMON
+
+#define CONFIG_BOOTCOMMAND \
+	"for target in ${boot_targets}; do run bootcmd_${target}; done ;"
+#endif
 
 #define CONFIG_SYS_BOOT_GET_CMDLINE
 
@@ -197,8 +215,4 @@
 #define CONFIG_ENV_IS_NOWHERE
 #endif
 
-#ifndef CONFIG_SPL_BUILD
-#include <config_distro_defaults.h>
-#endif
-
 #endif /* _SUNXI_COMMON_CONFIG_H */
-- 
1.9.0

