diff -ruN stegdetect-0.6.orig/file/Makefile.in stegdetect-0.6/file/Makefile.in
--- stegdetect-0.6.orig/file/Makefile.in	2004-08-29 23:11:06.000000000 +0000
+++ stegdetect-0.6/file/Makefile.in	2007-02-13 21:46:20.000000000 +0000
@@ -423,7 +423,7 @@
 install-exec-am:
 install-exec: install-exec-am
 
-install-data-am: install-man install-dataDATA
+install-data-am: install-dataDATA
 install-data: install-data-am
 
 install-am: all-am
diff -ruN stegdetect-0.6.orig/file/file.c stegdetect-0.6/file/file.c
--- stegdetect-0.6.orig/file/file.c	2004-08-29 23:11:06.000000000 +0000
+++ stegdetect-0.6/file/file.c	2007-02-13 21:23:52.000000000 +0000
@@ -76,7 +76,7 @@
 
 int 			/* Global command-line options 		*/
 	noprint = 1,
-	debug = 0, 	/* debugging 				*/
+	filedebug = 0, 	/* debugging 				*/
 	lflag = 0,	/* follow Symlinks (BSD only) 		*/
 	bflag = 0,	/* brief output format	 		*/
 	zflag = 0,	/* follow (uncompress) compressed files */
diff -ruN stegdetect-0.6.orig/file/file.h stegdetect-0.6/file/file.h
--- stegdetect-0.6.orig/file/file.h	2004-08-29 23:11:06.000000000 +0000
+++ stegdetect-0.6/file/file.h	2007-02-13 21:23:52.000000000 +0000
@@ -163,7 +163,7 @@
 extern struct mlist mlist;	/* list of arrays of magic entries	*/
 
 extern int noprint;
-extern int debug;		/* enable debugging?			*/
+extern int filedebug;		/* enable debugging?			*/
 extern int zflag;		/* process compressed files?		*/
 extern int lflag;		/* follow symbolic links?		*/
 extern int sflag;		/* read/analyze block special files?	*/
diff -ruN stegdetect-0.6.orig/file/softmagic.c stegdetect-0.6/file/softmagic.c
--- stegdetect-0.6.orig/file/softmagic.c	2004-08-29 23:11:06.000000000 +0000
+++ stegdetect-0.6/file/softmagic.c	2007-02-13 21:23:52.000000000 +0000
@@ -589,7 +589,7 @@
 	}
 
 
-	if (debug) {
+	if (filedebug) {
 		mdebug(offset, (char *) p, sizeof(union VALUETYPE));
 		mdump(m);
 	}
@@ -924,7 +924,7 @@
 
 		memcpy(p, s + offset, sizeof(union VALUETYPE));
 
-		if (debug) {
+		if (filedebug) {
 			mdebug(offset, (char *) p, sizeof(union VALUETYPE));
 			mdump(m);
 		}
@@ -1029,21 +1029,21 @@
 
 	switch (m->reln) {
 	case 'x':
-		if (debug)
+		if (filedebug)
 			(void) fprintf(stderr, "%u == *any* = 1\n", v);
 		matched = 1;
 		break;
 
 	case '!':
 		matched = v != l;
-		if (debug)
+		if (filedebug)
 			(void) fprintf(stderr, "%u != %u = %d\n",
 				       v, l, matched);
 		break;
 
 	case '=':
 		matched = v == l;
-		if (debug)
+		if (filedebug)
 			(void) fprintf(stderr, "%u == %u = %d\n",
 				       v, l, matched);
 		break;
@@ -1051,13 +1051,13 @@
 	case '>':
 		if (m->flag & UNSIGNED) {
 			matched = v > l;
-			if (debug)
+			if (filedebug)
 				(void) fprintf(stderr, "%u > %u = %d\n",
 					       v, l, matched);
 		}
 		else {
 			matched = (int32) v > (int32) l;
-			if (debug)
+			if (filedebug)
 				(void) fprintf(stderr, "%d > %d = %d\n",
 					       v, l, matched);
 		}
@@ -1066,13 +1066,13 @@
 	case '<':
 		if (m->flag & UNSIGNED) {
 			matched = v < l;
-			if (debug)
+			if (filedebug)
 				(void) fprintf(stderr, "%u < %u = %d\n",
 					       v, l, matched);
 		}
 		else {
 			matched = (int32) v < (int32) l;
-			if (debug)
+			if (filedebug)
 				(void) fprintf(stderr, "%d < %d = %d\n",
 					       v, l, matched);
 		}
@@ -1080,14 +1080,14 @@
 
 	case '&':
 		matched = (v & l) == l;
-		if (debug)
+		if (filedebug)
 			(void) fprintf(stderr, "((%x & %x) == %x) = %d\n",
 				       v, l, l, matched);
 		break;
 
 	case '^':
 		matched = (v & l) != l;
-		if (debug)
+		if (filedebug)
 			(void) fprintf(stderr, "((%x & %x) != %x) = %d\n",
 				       v, l, l, matched);
 		break;
diff -ruN stegdetect-0.6.orig/jpeg-6b/makefile.cfg stegdetect-0.6/jpeg-6b/makefile.cfg
--- stegdetect-0.6.orig/jpeg-6b/makefile.cfg	2004-08-29 23:11:09.000000000 +0000
+++ stegdetect-0.6/jpeg-6b/makefile.cfg	2007-02-13 22:17:27.000000000 +0000
@@ -194,16 +194,6 @@
 # Installation rules:
 
 install: cjpeg djpeg jpegtran rdjpgcom wrjpgcom @FORCE_INSTALL_LIB@
-	$(INSTALL_PROGRAM) cjpeg $(bindir)/$(binprefix)cjpeg
-	$(INSTALL_PROGRAM) djpeg $(bindir)/$(binprefix)djpeg
-	$(INSTALL_PROGRAM) jpegtran $(bindir)/$(binprefix)jpegtran
-	$(INSTALL_PROGRAM) rdjpgcom $(bindir)/$(binprefix)rdjpgcom
-	$(INSTALL_PROGRAM) wrjpgcom $(bindir)/$(binprefix)wrjpgcom
-	$(INSTALL_DATA) $(srcdir)/cjpeg.1 $(mandir)/$(manprefix)cjpeg.$(manext)
-	$(INSTALL_DATA) $(srcdir)/djpeg.1 $(mandir)/$(manprefix)djpeg.$(manext)
-	$(INSTALL_DATA) $(srcdir)/jpegtran.1 $(mandir)/$(manprefix)jpegtran.$(manext)
-	$(INSTALL_DATA) $(srcdir)/rdjpgcom.1 $(mandir)/$(manprefix)rdjpgcom.$(manext)
-	$(INSTALL_DATA) $(srcdir)/wrjpgcom.1 $(mandir)/$(manprefix)wrjpgcom.$(manext)
 
 install-lib: libjpeg.$(A) install-headers
 	$(INSTALL_LIB) libjpeg.$(A) $(libdir)/$(binprefix)libjpeg.$(A)
