diff -urN a/aclocal.m4 b/aclocal.m4
--- a/aclocal.m4	2014-04-07 12:26:08.000000000 -0600
+++ b/aclocal.m4	2014-05-04 08:43:23.399271800 -0600
@@ -408,7 +408,7 @@
                 )],
                 [changequote(, )dnl
                  ARM_ISA=ARMv7
-                 ARM_ISA_EXT="[VFPv3,NEON]"
+                 ARM_ISA_EXT="[VFPv3D16]"
                  changequote([, ])dnl
                 ])
         ])
