diff -wbBur redis-3.2.1/redis.conf redis-3.2.1.my/redis.conf
--- redis-3.2.1/redis.conf	2016-06-17 16:15:21.000000000 +0300
+++ redis-3.2.1.my/redis.conf	2016-06-22 18:06:23.687046141 +0300
@@ -244,7 +244,7 @@
 # The Append Only File will also be created inside this directory.
 #
 # Note that you must specify a directory here, not a file name.
-dir ./
+dir /var/lib/redis/
 
 ################################# REPLICATION #################################
 
