diff -up slim-1.3.1/switchuser.cpp.orig slim-1.3.1/switchuser.cpp
diff -up slim-1.3.1/switchuser.h.orig slim-1.3.1/switchuser.h
--- slim-1.3.1/switchuser.h.orig	2008-09-26 02:54:15.000000000 +0200
+++ slim-1.3.1/switchuser.h	2009-02-22 10:55:51.000000000 +0100
@@ -13,6 +13,7 @@
 #define _SWITCHUSER_H_
 
 #include <stdlib.h>
+#include <stdio.h> /* remove */
 #include <unistd.h>
 #include <string.h>
 #include <pwd.h>
