--- src/lilypond-2.10.33/flower/file-name.cc~	2008-04-04 22:40:39.000000000 +0200
+++ src/lilypond-2.10.33/flower/file-name.cc	2008-04-04 22:40:54.000000000 +0200
@@ -12,6 +12,7 @@
 #include <cstdio>
 #include <cerrno>
 #include <unistd.h>
+#include <limits.h>
 
 using namespace std;
 
--- src/lilypond-2.10.33/flower/include/rational.hh~	2008-04-04 22:41:47.000000000 +0200
+++ src/lilypond-2.10.33/flower/include/rational.hh	2008-04-04 22:41:55.000000000 +0200
@@ -10,7 +10,7 @@
 #define RATIONAL_HH
 
 #include "std-string.hh"
-
+#include <limits.h>
 #include "std-string.hh"
 
 /**
--- src/lilypond-2.10.33/flower/include/std-vector.hh~	2008-04-04 22:42:32.000000000 +0200
+++ src/lilypond-2.10.33/flower/include/std-vector.hh	2008-04-04 22:42:42.000000000 +0200
@@ -22,6 +22,7 @@
 #include <algorithm>   /* find, reverse, sort */
 #include <functional>  /* unary_function */
 #include <cassert>
+#include <string>
 
 using namespace std;
 
--- src/lilypond-2.10.33/lily/include/lily-guile.hh~	2008-04-04 22:43:15.000000000 +0200
+++ src/lilypond-2.10.33/lily/include/lily-guile.hh	2008-04-04 22:43:23.000000000 +0200
@@ -14,7 +14,7 @@
 #endif
 
 #include <libguile.h>
-
+#include <string.h>
 
 /*
   Hack for various MacOS incarnations.
--- src/lilypond-2.10.33/lily/include/midi-stream.hh~	2008-04-04 22:44:09.000000000 +0200
+++ src/lilypond-2.10.33/lily/include/midi-stream.hh	2008-04-04 22:44:18.000000000 +0200
@@ -10,6 +10,7 @@
 #include <cstdio>
 using namespace std;
 
+#include <string.h>
 #include "std-string.hh"
 
 class Midi_item;
--- src/lilypond-2.10.33/lily/include/engraver.hh~	2008-04-04 22:44:51.000000000 +0200
+++ src/lilypond-2.10.33/lily/include/engraver.hh	2008-04-04 22:44:59.000000000 +0200
@@ -11,6 +11,7 @@
 
 #include "grob-info.hh"
 #include "translator.hh"
+#include <string.h>
 
 /**
    a struct which processes events, and creates the #Grob#s.
