--- a/custom/conf/app.ini.sample
+++ b/custom/conf/app.ini.sample
@@ -7,12 +7,12 @@
 ; App name that shows in every page title
 APP_NAME = Gitea: Git with a cup of tea
 ; Change it if you run locally
-RUN_USER = git
+RUN_USER = gitea
 ; Either "dev", "prod" or "test", default is "dev"
-RUN_MODE = dev
+RUN_MODE = prod

 [repository]
-ROOT =
+ROOT = /var/lib/gitea/repos
 SCRIPT_TYPE = bash
 ; Default ANSI charset
 ANSI_CHARSET =
@@ -378,7 +378,7 @@ SQLITE_TIMEOUT = 500
 ; For iterate buffer, default is 50
 ITERATE_BUFFER_SIZE = 50
 ; Show the database generated SQL
-LOG_SQL = true
+LOG_SQL = false
 ; Maximum number of DB Connect retries
 DB_RETRIES = 10
 ; Backoff time per DB retry (time.Duration)
@@ -751,10 +751,10 @@ FORMAT =
 DEFAULT_UI_LOCATION =

 [log]
-ROOT_PATH =
+ROOT_PATH = /var/log/gitea/
 ; Either "console", "file", "conn", "smtp" or "database", default is "console"
 ; Use comma to separate multiple modes, e.g. "console, file"
-MODE = console
+MODE = console, file
 ; Buffer length of the channel, keep it as it is if you don't know what it is.
 BUFFER_LEN = 10000
 REDIRECT_MACARON_LOG = false
@@ -779,7 +779,7 @@ COLORIZE = false

 ; For "console" mode only
 [log.console]
-LEVEL =
+LEVEL = Info
 STDERR = false

 ; For "file" mode only
