--- a/custom/conf/app.example.ini
+++ b/custom/conf/app.example.ini
@@ -7,6 +7,6 @@
 ; App name that shows in every page title
 APP_NAME = Gitea: Git with a cup of tea
 ; Change it if you run locally
-RUN_USER = git
+RUN_USER = gitea
 ; Application run mode, affects performance and debugging. Either "dev", "prod" or "test", default is "prod"
 RUN_MODE = prod
@@ -18,5 +18,5 @@

 [repository]
-ROOT =
+ROOT = /var/lib/gitea/repos
 SCRIPT_TYPE = bash
 ; DETECTED_CHARSETS_ORDER tie-break order for detected charsets.
@@ -424,7 +424,7 @@ SQLITE_TIMEOUT = 500
 ; For iterate buffer, default is 50
 ITERATE_BUFFER_SIZE = 50
 ; Show the database generated SQL
-LOG_SQL = true
+LOG_SQL = false
 ; Maximum number of DB Connect retries
 DB_RETRIES = 10
 ; Backoff time per DB retry (time.Duration)
@@ -838,10 +838,10 @@ FORMAT =
 DEFAULT_UI_LOCATION =

 [log]
-ROOT_PATH =
+ROOT_PATH = /var/log/gitea/
 ; Either "console", "file", "conn", "smtp" or "database", default is "console"
 ; Use comma to separate multiple modes, e.g. "console, file"
-MODE = console
+MODE = console, file
 ; Buffer length of the channel, keep it as it is if you don't know what it is.
 BUFFER_LEN = 10000
 REDIRECT_MACARON_LOG = false
@@ -866,7 +866,7 @@ COLORIZE = false

 ; For "console" mode only
 [log.console]
-LEVEL =
+LEVEL = Info
 STDERR = false

 ; For "file" mode only
