diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2014-11-12 18:59:40.000000000 -0700
+++ b/CMakeLists.txt	2015-01-24 11:04:26.558025684 -0700
@@ -17,8 +17,7 @@
 
 include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR})
 
-find_package(OpenGL)
-set(HAVE_OPENGL ${OPENGL_FOUND})
+set(HAVE_OPENGL FALSE)
 set_package_properties(OpenGL PROPERTIES TYPE OPTIONAL
                                          PURPOSE "Support for 3D graphs in KAlgebra"
                                          URL "http://opengl.org")
