diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2013-05-02 23:37:45.511551937 -0500
+++ b/CMakeLists.txt	2013-05-21 19:51:54.124577735 -0500
@@ -11,8 +11,11 @@
 include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR} ${KDE4_INCLUDES} ${LIBKDEEDU_INCLUDE_DIR}/qtmml ${ANALITZA_INCLUDE_DIR})
 
 macro_optional_find_package(OpenGL)
-macro_bool_to_01(OPENGL_FOUND HAVE_OPENGL)
-macro_log_feature(OPENGL_FOUND "OpenGL" "Support for 3D graphs in KAlgebra" "http://mesa3d.sourceforge.net" FALSE "" "")
+option(SHOULD_BUILD_OPENGL "Build OpenGL support" ON)
+if(SHOULD_BUILD_OPENGL)
+  macro_bool_to_01(OPENGL_FOUND HAVE_OPENGL)
+endif(SHOULD_BUILD_OPENGL)
+macro_log_feature(HAVE_OPENGL "OpenGL" "Support for 3D graphs in KAlgebra" "http://mesa3d.sourceforge.net" FALSE "" "")
 
 macro_log_feature(QTMML_LIBRARY "QtMathML Widget" "Supports displaying some formulas in a mathematic form" "http://doc.qt.nokia.com/solutions/4/qtmmlwidget/qtmmlwidget.html" FALSE "" "")
 
