From 6601f597373db917feeae4ce49914da724852a8a Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Fri, 2 May 2025 08:09:22 -0600
Subject: [PATCH] disable libyuv neon

---
 src/core/CMakeLists.txt | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index d5e9b715e..bccdd8d74 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -534,13 +534,8 @@ foreach(arch ${archs})
             endif()
             get_gn_arch(cpu ${TEST_architecture_arch})
 
-            if(cpu STREQUAL "arm" AND
-               CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
-               CMAKE_CXX_COMPILER_VERSION VERSION_LESS "11.0.0")
-                list(APPEND gnArgArg libyuv_use_neon=false)
-                #TODO check if libyuv really uses udot in case of arm32
-            endif()
-            if(cpu STREQUAL "arm64" AND NOT QT_FEATURE_webengine_arm64_udot_support)
+            if(cpu STREQUAL "arm" OR cpu STREQUAL "arm64")
+                # udot assembler
                 list(APPEND gnArgArg libyuv_use_neon=false)
             endif()
             if(cpu STREQUAL "arm")
-- 
2.36.1

