From 4e1a63bac5a9df32f3386155414bb97fed33e7e8 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Fri, 2 May 2025 08:09:22 -0600
Subject: [PATCH] disable libyuv neon

---
 src/core/CMakeLists.txt | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index cc992b4d2..d1499a70c 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -534,10 +534,8 @@ foreach(arch ${archs})
             endif()
             get_gn_arch(cpu ${TEST_architecture_arch})
             if(cpu STREQUAL "arm" OR cpu STREQUAL "arm64")
-                if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "11.0.0")
-                    # udot assembler
-                    list(APPEND gnArgArg libyuv_use_neon=false)
-                endif()
+                # udot assembler
+                list(APPEND gnArgArg libyuv_use_neon=false)
             endif()
             if(cpu STREQUAL "arm")
                 check_thumb(armThumb)
-- 
2.36.1

