diff --git a/chromium/media/base/media_switches.cc b/chromium/media/base/media_switches.cc
index 0dfbbd9e38f..fb7bc05cda8 100644
--- a/chromium/media/base/media_switches.cc
+++ b/chromium/media/base/media_switches.cc
@@ -1712,7 +1712,7 @@ BASE_FEATURE(kUseWindowBoundsForPip,
 // Enables FFmpeg allow lists for supported codecs / containers.
 BASE_FEATURE(kFFmpegAllowLists,
              "FFmpegAllowLists",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Enables sending MediaLog to the log stream, which is useful for easier
 // development by ensuring logs can be seen without a remote desktop session.
