diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index e82995f3e..c7c6da168 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -527,6 +527,10 @@ foreach(arch ${archs})
                 ARGS rtc_system_openh264
                 CONDITION QT_FEATURE_webengine_webrtc_system_openh264
             )
+            extend_gn_list(gnArgArg
+                ARGS use_system_openh264
+                CONDITION QT_FEATURE_webengine_webrtc_system_openh264
+            )
             extend_gn_list(gnArgArg
                 ARGS use_vaapi
                 CONDITION QT_FEATURE_webengine_vaapi
