Index: dgen-1.23/fm.h
===================================================================
--- dgen-1.23.orig/fm.h	2007-12-09 20:27:27.000000000 -0200
+++ dgen-1.23/fm.h	2007-12-09 20:27:44.000000000 -0200
@@ -5,6 +5,8 @@
 #ifndef _H_FM_FM_
 #define _H_FM_FM_
 
+#include <stdint.h>
+
 ////// starshine - one chip 4 me!
 #define HAS_YM2612 1
 #define BUILD_YM2612 (HAS_YM2612)       // build YM2612 emulator
@@ -35,12 +37,12 @@
 #endif
 
 #if (FM_OUTPUT_BIT==16)
-typedef unsigned short FMSAMPLE;
-typedef unsigned long FMSAMPLE_MIX;
+typedef uint16_t FMSAMPLE;
+typedef uint32_t FMSAMPLE_MIX;
 #endif
 #if (FM_OUTPUT_BIT==8)
-typedef unsigned char  FMSAMPLE;
-typedef unsigned short FMSAMPLE_MIX;
+typedef uint8_t  FMSAMPLE;
+typedef uint16_t FMSAMPLE_MIX;
 #endif
 
 /* For YM2151/YM2608/YM2612 option */
