--- dgen-sdl-1.23/musa/m68kmake.c	2006-01-08 01:13:20.000000000 +0100
+++ dgen-sdl-1.23/musa/m68kmake.c.new	2006-01-08 01:14:15.000000000 +0100
@@ -1029,9 +1029,9 @@
 	fprintf(filep, "/* ========================= OPCODE TABLE BUILDER ========================= */\n");
 	fprintf(filep, "/* ======================================================================== */\n\n");
 
+	fprintf(filep, "#include <stdlib.h>\n\n");
 	fprintf(filep, "#include \"m68kops.h\"\n");
 	fprintf(filep, "#include \"m68kcpu.h\"\n");
-	fprintf(filep, "#include <stdlib.h>\n\n");
 	fprintf(filep, "#include <string.h>\n\n");
 
 	fprintf(filep, "extern void  (*m68k_instruction_jump_table[])(void); /* opcode handler jump table */\n\n");
--- dgen-sdl-1.23/star/star.c	2007-04-15 00:13:10.000000000 +1000
+++ dgen-sdl-1.23/star/star.c2	2007-04-15 00:15:27.000000000 +1000
@@ -1931,7 +1931,7 @@
 	case aind: case ainc: case adec:
 	case adsp: case axdp:
 		usereg();
-	default:
+	default: ;
 	}
 }
 
