Index: dgen-1.23/main.cpp
===================================================================
--- dgen-1.23.orig/main.cpp	2007-12-09 20:27:28.000000000 -0200
+++ dgen-1.23/main.cpp	2007-12-09 20:27:30.000000000 -0200
@@ -334,6 +334,9 @@
   if(optind >= argc)
     help();
 
+  extern void mz80init(void);
+  mz80init();
+
   // Initialize the platform-dependent stuff.
   if(!pd_graphics_init(dgen_sound, pal_mode))
     {
