From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Thu, 13 Jul 2023 21:16:53 +0200
Subject: [PATCH] bootstrap: Change bash-completion dir

---
 src/bootstrap/dist.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/bootstrap/dist.rs b/src/bootstrap/dist.rs
index 05e106637d74..426b1277fb23 100644
--- a/src/bootstrap/dist.rs
+++ b/src/bootstrap/dist.rs
@@ -1073,7 +1073,7 @@ fn run(self, builder: &Builder<'_>) -> Option<GeneratedTarball> {
         tarball.add_file(etc.join("_cargo"), "share/zsh/site-functions", 0o644);
         tarball.add_renamed_file(
             etc.join("cargo.bashcomp.sh"),
-            "src/etc/bash_completion.d",
+            "share/bash-completion/completions",
             "cargo",
         );
         tarball.add_dir(etc.join("man"), "share/man/man1");
