diff -u -r rustc-1.50.0-src/src/bootstrap/dist.rs rustc-1.50.0-src-nolibexec/src/bootstrap/dist.rs
--- rustc-1.50.0-src/src/bootstrap/dist.rs	2021-02-10 17:36:44.000000000 +0000
+++ rustc-1.50.0-src-nolibexec/src/bootstrap/dist.rs	2021-02-11 18:23:36.312495419 +0000
@@ -980,7 +980,7 @@
         for dirent in fs::read_dir(cargo.parent().unwrap()).expect("read_dir") {
             let dirent = dirent.expect("read dir entry");
             if dirent.file_name().to_str().expect("utf8").starts_with("cargo-credential-") {
-                tarball.add_file(&dirent.path(), "libexec", 0o755);
+                tarball.add_file(&dirent.path(), "lib", 0o755);
             }
         }
 
diff -u -r rustc-1.50.0-src/src/tools/cargo/src/cargo/ops/registry/auth.rs rustc-1.50.0-src-nolibexec/src/tools/cargo/src/cargo/ops/registry/auth.rs
--- rustc-1.50.0-src/src/tools/cargo/src/cargo/ops/registry/auth.rs	2021-02-10 17:37:00.000000000 +0000
+++ rustc-1.50.0-src-nolibexec/src/tools/cargo/src/cargo/ops/registry/auth.rs	2021-02-11 18:24:14.944321464 +0000
@@ -223,7 +223,7 @@
         .parent()
         .and_then(|p| p.parent())
         .ok_or_else(|| format_err!("expected cargo path {}", cargo.display()))?;
-    let exe = root.join("libexec").join(format!(
+    let exe = root.join("lib").join(format!(
         "cargo-credential-{}{}",
         cred_name,
         std::env::consts::EXE_SUFFIX
