From ac560966d09295663fa9516d8d137e0c0fd04c06 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Mon, 12 Jun 2023 07:42:01 +0200
Subject: [PATCH 1/1] support dropins for default configuration

---
 util/grub-mkconfig.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/util/grub-mkconfig.in b/util/grub-mkconfig.in
index 1a945085c..4338dceef 100644
--- a/util/grub-mkconfig.in
+++ b/util/grub-mkconfig.in
@@ -160,6 +160,11 @@ fi
 if test -f ${sysconfdir}/default/grub ; then
   . ${sysconfdir}/default/grub
 fi
+for dropin in ${sysconfdir}/default/grub.d/*.cfg ; do
+  if test -f "${dropin}" ; then
+    . "${dropin}"
+  fi
+done
 
 if [ "x${GRUB_DISABLE_UUID}" = "xtrue" ]; then
   if [ -z "${GRUB_DISABLE_LINUX_UUID}" ]; then
-- 
2.41.0

