From 02dd94a72c71b855827f9ec00d2acf93e0f8c85b Mon Sep 17 00:00:00 2001
From: Menci <huanghaorui301@gmail.com>
Date: Sat, 5 Feb 2022 16:16:50 +0800
Subject: [PATCH] 10_linux: add archlinuxarm's default kernel path

---
 util/grub.d/10_linux.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index ca068038e..7e08a7faf 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -173,7 +173,7 @@ case "x$machine" in
 	done ;;
     *) 
 	list=
-	for i in /boot/vmlinuz-* /boot/vmlinux-* /vmlinuz-* /vmlinux-* /boot/kernel-* ; do
+	for i in /boot/vmlinuz-* /boot/vmlinux-* /vmlinuz-* /vmlinux-* /boot/kernel-* /boot/Image ; do
                   if grub_file_is_not_garbage "$i" ; then list="$list $i" ; fi
 	done ;;
 esac
-- 
2.34.1

