--- a/util/grub.d/10_linux.in   2022-11-13 17:54:27.819444758 +0000
+++ b/util/grub.d/10_linux.in   2022-11-13 17:54:37.351333230 +0000
@@ -212,7 +212,11 @@
   basename=`basename $linux`
   dirname=`dirname $linux`
   rel_dirname=`make_system_path_relative_to_its_root $dirname`
-  version=`echo $basename | sed -e "s,vmlinuz-,,g"`
+  if [ $basename = Image ]; then
+    version=linux
+  else
+    version=`echo $basename | sed -e "s,vmlinuz-,,g"`
+  fi
   alt_version=`echo $version | sed -e "s,\.old$,,g"`
   linux_root_device_thisversion="${LINUX_ROOT_DEVICE}"
 