diff --git a/src/qtractorMeter.cpp b/src/qtractorMeter.cpp
index d1b826c..14c3836 100644
--- a/src/qtractorMeter.cpp
+++ b/src/qtractorMeter.cpp
@@ -29,6 +29,7 @@
 
 #include "qtractorObserverWidget.h"
 
+#include <QAction>
 #include <QHBoxLayout>
 #include <QVBoxLayout>
 #include <QPainter>
diff --git a/src/qtractorTrackButton.cpp b/src/qtractorTrackButton.cpp
index 488eb83..5a314dd 100644
--- a/src/qtractorTrackButton.cpp
+++ b/src/qtractorTrackButton.cpp
@@ -26,6 +26,8 @@
 
 #include "qtractorMidiControlObserver.h"
 
+#include <QAction>
+
 
 
 //----------------------------------------------------------------------------
