diff -ruN a/src/qtractorLv2Plugin.h b/src/qtractorLv2Plugin.h
--- a/src/qtractorLv2Plugin.h	2021-02-12 08:35:41.220805288 +0100
+++ b/src/qtractorLv2Plugin.h	2021-02-12 22:06:40.296752753 +0100
@@ -65,9 +65,7 @@
 #include "lv2_external_ui.h"
 #endif
 #if QT_VERSION >= QT_VERSION_CHECK(5, 1, 0)
-#ifdef CONFIG_LV2_UI_GTK2
 #include <QWindow>
-#endif	// CONFIG_LV2_UI_GTK2
 #endif
 // LV2 UI Request-value support (FAKE).
 #ifdef  CONFIG_LV2_UI_REQ_VALUE_FAKE
