From 303c675477e67e4e9748027e5e37232e5bcc580f Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 18 Sep 2018 01:09:59 +0000
Subject: [PATCH] set libm as required

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 2289eb0..ffcf149 100644
--- a/meson.build
+++ b/meson.build
@@ -94,7 +94,7 @@ libtiff = dependency('libtiff-4')
 colord = dependency('colord', version : '>= 1.3.1')
 libcolord_gtk = dependency('colord-gtk', version : '>= 0.1.20')
 libcanberra = dependency('libcanberra-gtk3', version : '>= 0.10')
-libm = cc.find_library('libm', required: false)
+libm = cc.find_library('m', required: true)
 liblcms = dependency('lcms2', version : '>= 2.2')
 
 libvte = dependency('vte-2.91', version : '>= 0.25.1', required : false)
-- 
2.18.0

