diff -Nru DenyHosts-2.1.old/DenyHosts/constants.py DenyHosts-2.1/DenyHosts/constants.py
--- DenyHosts-2.1.old/DenyHosts/constants.py	2006-02-08 23:59:20.000000000 -0600
+++ DenyHosts-2.1/DenyHosts/constants.py	2006-02-12 22:27:01.000000000 -0600
@@ -34,7 +34,7 @@
 #                           Miscellaneous constants                             #
 #################################################################################
 
-CONFIG_FILE = "denyhosts.cfg"  # default can be overridden on cmd line
+CONFIG_FILE = "/etc/denyhosts/denyhosts.cfg"  # default can be overridden on cmd line
 
 DENY_DELIMITER = "# DenyHosts:"
 ENTRY_DELIMITER = " | "
diff -Nru DenyHosts-2.1.old/setup.py DenyHosts-2.1/setup.py
--- DenyHosts-2.1.old/setup.py	2006-02-09 19:03:39.000000000 -0600
+++ DenyHosts-2.1/setup.py	2006-02-12 22:26:39.000000000 -0600
@@ -23,14 +23,8 @@
       scripts=['denyhosts.py'],
       package_dir={'DenyHosts': 'DenyHosts'},
       packages=["DenyHosts"],
-      data_files=[(libpath, glob("denyhosts.cfg-dist")),
-                  (libpath, glob("setup.py")),
-                  (libpath, glob("daemon-control-dist")),
-                  (libpath, glob("CHANGELOG.txt")),
-                  (libpath, glob("README.txt")),
-                  (scriptspath, glob("scripts/*")),
-                  (pluginspath, glob("plugins/*")),
-                  (libpath, glob("LICENSE.txt"))],
+      data_files=[(scriptspath, glob("scripts/*")),
+                  (pluginspath, glob("plugins/*"))],
       license="GPL v2",
       ##extra_path='denyhosts',
       long_description="""
