--- DenyHosts-2.6/DenyHosts/regex-old.py	2008-01-29 03:07:51.000000000 +0100
+++ DenyHosts-2.6/DenyHosts/regex.py	2008-01-29 03:07:53.000000000 +0100
@@ -17,7 +17,7 @@
 
 FAILED_ENTRY_REGEX4 = re.compile(r"""Authentication failure for (?P<user>.*) .*from (?P<host>.*)""")
 
-FAILED_ENTRY_REGEX5 = re.compile(r"""User (?P<user>.*) .*from (?P<host>.*) not allowed because none of user's groups are listed in AllowGroups""")
+FAILED_ENTRY_REGEX5 = re.compile(r"""User (?P<user>.*) .*from (?P<host>.*) not allowed because none of user's groups are listed in AllowGroups$""")
 
 FAILED_ENTRY_REGEX6 = re.compile(r"""Did not receive identification string .*from (::ffff:)?(?P<host>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})""")
 
--- DenyHosts-2.6/DenyHosts/regex.py	2007-04-23 17:22:21.000000000 +0100
+++ DenyHosts-2.6/DenyHosts/regex.py	2007-04-23 17:24:35.000000000 +0100
@@ -21,7 +21,7 @@
 
 FAILED_ENTRY_REGEX6 = re.compile(r"""Did not receive identification string .*from (::ffff:)?(?P<host>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})""")
 
-FAILED_ENTRY_REGEX7 = re.compile(r"""User (?P<user>.*) not allowed because not listed in AllowUsers""")
+FAILED_ENTRY_REGEX7 = re.compile(r"""User (?P<user>.*) .*from (::ffff:)?(?P<host>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}) not allowed because not listed in AllowUsers""")
 
 
 # these are reserved for future versions
