--- revelation-0.4.11.orig/src/lib/ui.py
+++ revelation-0.4.11/src/lib/ui.py
@@ -1257,8 +1257,9 @@
 	def __init__(self, stock, text = None):
 		gtk.ImageMenuItem.__init__(self, stock)
 
-		self.label = self.get_children()[0]
-		self.image = self.get_children()[1]
+		children=self.get_children()
+		if len(children) > 0 : self.label = children[0]
+		if len(children) > 1 : self.image = children[1]
 
 		if text is not None:
 			self.set_text(text)
