From 403dd9d7b14ba83537a06d15339bd703a64e8987 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Mon, 20 Nov 2017 06:50:15 -0700
Subject: [PATCH] fix boost library path for ARM

---
 configure | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure b/configure
index 3179c1756..e72993d6e 100755
--- a/configure
+++ b/configure
@@ -4707,6 +4707,8 @@ fi
     libsubdirs="lib64 lib lib64" ;; #(
   libsubdirs="lib") :
      ;; #(
+  arm*) :
+    libsubdirs="lib" ;; #(
   *) :
      ;;
 esac
-- 
2.14.1

