From 15c04b09a8c06871635d5fd98c3a37089bbde6d9 Mon Sep 17 00:00:00 2001
From: Jeremie Dimino <jeremie@dimino.org>
Date: Thu, 8 Aug 2019 23:31:27 +0100
Subject: [PATCH] remove nodynlink optimisation

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
---
 src/exe_rules.ml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/exe_rules.ml b/src/exe_rules.ml
index 834afab21..eda12ee4d 100644
--- a/src/exe_rules.ml
+++ b/src/exe_rules.ml
@@ -104,11 +104,11 @@ let executables_rules ~sctx ~dir ~dir_kind ~expander
       else
         Some js_of_ocaml
     in
-    let dynlink =
+    let dynlink = true (*
       Dune_file.Executables.Link_mode.Set.exists exes.modes ~f:(fun mode ->
         match mode.kind with
         | Shared_object -> true
-        | _ -> false)
+                           | _ -> false)*)
     in
     Compilation_context.create ()
       ~super_context:sctx
