--- a/optional/odbc/thread.cpp	2009-07-06 03:56:14.000000000 -0700
+++ b/optional/odbc/thread.cpp	2009-07-06 03:56:14.000000000 -0700
@@ -64,7 +64,7 @@
 	SQLCHAR errmsg[128];
 	SQLSMALLINT mlen, col, cols;
 #if ODBCVER >= 0x0300 && !defined(__ppc__)
-	SQLINTEGER rowcnt;
+	long int rowcnt;
 #else
 	long int rowcnt;
 #endif
--- a/src/script3.h	2007-08-11 04:39:53.000000000 -0700
+++ b/src/script3.h	2009-07-06 04:10:28.000000000 -0700
@@ -40,6 +40,7 @@
  * @short Threaded step execute scripting engine framework.
  **/
 
+#include <cstdlib>
 
 #ifndef	CCXX_SCRIPT3_H_
 #define	CCXX_SCRIPT3_H_
