diff -u -r bcm2835-1.42-orig/src/bcm2835.c bcm2835-1.42/src/bcm2835.c
--- bcm2835-1.42-orig/src/bcm2835.c	2015-03-09 06:00:44.000000000 +0100
+++ bcm2835-1.42/src/bcm2835.c	2015-03-22 19:29:37.531488655 +0100
@@ -122,9 +122,9 @@
        uint32_t ret;
 #ifdef BCM2835_HAVE_DMB
        __asm__(        "\
-  dmb                    \
-  ldr %[ret], [%[paddr]] \
-  dmb								\
+  dmb                   \n\
+  ldr %[ret], [%[paddr]]\n\
+  dmb                   \n\
 " : [ret] "=r" (ret) : [paddr] "r" (paddr) : "memory" );
 #else
        __asm__(            "\
@@ -176,9 +176,9 @@
 	/* Following code provides memory barriers before and after the write */
 #ifdef BCM2835_HAVE_DMB
        __asm__(        "\
-  dmb			   \
-  str %[value], [%[paddr]] \
-  dmb								\
+  dmb                     \n\
+  str %[value], [%[paddr]]\n\
+  dmb                     \n\
 " : : [paddr] "r" (paddr), [value] "r" (value) : "memory" );
 #else
        __asm__(            "\
