diff -urN a/include/internal.h b/include/internal.h
--- a/include/internal.h	2013-10-11 18:51:57.000000000 -0600
+++ b/include/internal.h	2014-08-29 10:44:56.347765456 -0600
@@ -36,6 +36,7 @@
 
 #ifndef POST_PACKED_STRUCTURE
 #ifdef __GNUC__
+#ifndef __arm__
 /* POST_PACKED_STRUCTURE needs to be a macro which
    expands into a compiler directive. The directive must
    tell the compiler to arrange the preceding structure
@@ -44,6 +45,9 @@
    compiler.
 */
 #define POST_PACKED_STRUCTURE __attribute__((__packed__))
+#else
+#define POST_PACKED_STRUCTURE
+#endif
 #else
 /* Add other things here for non-gcc platforms */
 #endif
diff -urN a/include/port.h b/include/port.h
--- a/include/port.h	2013-10-06 09:34:42.000000000 -0600
+++ b/include/port.h	2014-08-29 10:45:25.257670204 -0600
@@ -56,6 +56,7 @@
 
 #ifndef POST_PACKED_STRUCTURE
 #ifdef __GNUC__
+#ifndef __arm__
 /* POST_PACKED_STRUCTURE needs to be a macro which
    expands into a compiler directive. The directive must
    tell the compiler to arrange the preceding structure
@@ -64,6 +65,9 @@
    compiler.
 */
 #define POST_PACKED_STRUCTURE __attribute__((__packed__))
+#else
+#define POST_PACKED_STRUCTURE
+#endif
 #else
 /* Add other things here for non-gcc platforms */
 #endif
diff -urN a/jack/transport.h b/jack/transport.h
--- a/jack/transport.h	2013-10-03 18:46:00.000000000 -0600
+++ b/jack/transport.h	2014-08-29 10:45:09.652721584 -0600
@@ -30,6 +30,7 @@
 
 #ifndef POST_PACKED_STRUCTURE
 #ifdef __GNUC__
+#ifndef __arm__
 /* POST_PACKED_STRUCTURE needs to be a macro which
    expands into a compiler directive. The directive must
    tell the compiler to arrange the preceding structure
@@ -39,6 +40,9 @@
 */
 #define POST_PACKED_STRUCTURE __attribute__((__packed__))
 #else
+#define POST_PACKED_STRUCTURE
+#endif
+#else
 /* Add other things here for non-gcc platforms */
 #endif
 #endif
