From e3d5dd26a3e1057b6927d12bc88427eb13a8f3b1 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 15 Apr 2014 14:27:35 -0500
Subject: [PATCH 5/9] ARM: mxs: boot regression, select CLKSRC_OF

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/mach-mxs/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-mxs/Kconfig b/arch/arm/mach-mxs/Kconfig
index 8479413..b3eb28c 100644
--- a/arch/arm/mach-mxs/Kconfig
+++ b/arch/arm/mach-mxs/Kconfig
@@ -17,6 +17,7 @@ config ARCH_MXS
 	depends on ARCH_MULTI_V5
 	select ARCH_REQUIRE_GPIOLIB
 	select CLKSRC_MMIO
+	select CLKSRC_OF
 	select PINCTRL
 	select SOC_BUS
 	select SOC_IMX23
-- 
1.9.0

