diff -Naur hal-0.5.8.1/tools/linux/hal-luks-remove-linux hal-0.5.8.1-new/tools/linux/hal-luks-remove-linux
--- hal-0.5.8.1/tools/linux/hal-luks-remove-linux	2006-09-19 22:23:25.000000000 +0200
+++ hal-0.5.8.1-new/tools/linux/hal-luks-remove-linux	2007-02-11 13:04:46.000000000 +0100
@@ -6,7 +6,7 @@
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License version 2.
 
-CRYPTSETUP=/sbin/cryptsetup
+CRYPTSETUP=/usr/sbin/cryptsetup
 
 echo LUKS cryptotext device $HAL_PROP_INFO_UDI is going away
 
diff -Naur hal-0.5.8.1/tools/linux/hal-luks-setup-linux hal-0.5.8.1-new/tools/linux/hal-luks-setup-linux
--- hal-0.5.8.1/tools/linux/hal-luks-setup-linux	2006-09-19 22:23:25.000000000 +0200
+++ hal-0.5.8.1-new/tools/linux/hal-luks-setup-linux	2007-02-11 13:04:45.000000000 +0100
@@ -9,7 +9,7 @@
 read password
 PASSWORD=${PASSWORD//[^[:alnum:]_=[:space:]:.+-]/@}
 
-CRYPTSETUP=/sbin/cryptsetup
+CRYPTSETUP=/usr/sbin/cryptsetup
 
 if [ ! -f $CRYPTSETUP ]; then
     echo org.freedesktop.Hal.Device.Volume.Crypto.SetupError >&2
diff -Naur hal-0.5.8.1/tools/linux/hal-luks-teardown-linux hal-0.5.8.1-new/tools/linux/hal-luks-teardown-linux
--- hal-0.5.8.1/tools/linux/hal-luks-teardown-linux	2006-09-19 22:23:25.000000000 +0200
+++ hal-0.5.8.1-new/tools/linux/hal-luks-teardown-linux	2007-02-11 13:04:44.000000000 +0100
@@ -6,7 +6,7 @@
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License version 2.
 
-CRYPTSETUP=/sbin/cryptsetup
+CRYPTSETUP=/usr/sbin/cryptsetup
 
 if [ ! -f $CRYPTSETUP ]; then
 	echo org.freedesktop.Hal.Device.Volume.Crypto.TeardownError >&2
