diff -upr hal-0.5.12/tools/linux/hal-system-killswitch-get-power-linux hal-0.5.12.new/tools/linux/hal-system-killswitch-get-power-linux
--- hal-0.5.12/tools/linux/hal-system-killswitch-get-power-linux	2008-08-22 07:34:47.000000000 +0100
+++ hal-0.5.12.new/tools/linux/hal-system-killswitch-get-power-linux	2009-03-03 10:45:20.000000000 +0000
@@ -8,11 +8,6 @@
 # the Free Software Foundation; either version 2 of the License, or
 # (at your option) any later version.
 
-DELL_WCTL=/usr/bin/dellWirelessCtl
-if [ -x "/usr/sbin/dellWirelessCtl" ]; then
-  DELL_WCTL=/usr/sbin/dellWirelessCtl
-fi
-
 if [ "$HAL_PROP_KILLSWITCH_TYPE" = "bluetooth" ]; then
     if [ "$HAL_PROP_KILLSWITCH_ACCESS_METHOD" = "sonypic" ]; then
 	value="`hal-system-sonypic getbluetooth`"
@@ -22,25 +17,6 @@ if [ "$HAL_PROP_KILLSWITCH_TYPE" = "blue
 	    exit 1
 	fi
 	exit ${value}
-    elif [ "$HAL_PROP_KILLSWITCH_ACCESS_METHOD" = "dell" ]; then
-	if [ -x "$DELL_WCTL" ]; then
-	    # TODO: write our own binary that links with libsmbios?
-	    $DELL_WCTL --st_bt
-	    value=$?
-	    if [ "$value" = "0" ]; then
-	        exit 1
-	    elif [ "$value" = "1" ]; then
-	        exit 0
-	    else
-	        echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-	        echo "dellWirelessCtl returned $value" >&2
-                exit 1
-	    fi
-        else
-	    echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-	    echo "dellWirelessCtl ($DELL_WCTL) not available or executable" >&2
-	    exit 1
-	fi
     elif [ "$HAL_PROP_KILLSWITCH_ACCESS_METHOD" = thinkpad -a -r "$HAL_PROP_LINUX_SYSFS_PATH" ]; then
 	read value < $HAL_PROP_LINUX_SYSFS_PATH 2> /dev/null
 	if [ $? -eq 0 ]; then
@@ -55,56 +31,6 @@ if [ "$HAL_PROP_KILLSWITCH_TYPE" = "blue
 	echo "Access type not supported" >&2
 	exit 1
     fi
-elif [ "$HAL_PROP_KILLSWITCH_TYPE" = "wlan" ]; then
-    if [ "$HAL_PROP_KILLSWITCH_ACCESS_METHOD" = "dell" ]; then
-	if [ -x "$DELL_WCTL" ]; then
- 	    # TODO: write our own binary that links with libsmbios?
-	    $DELL_WCTL --st_wlan
-	    value=$?
-	    if [ "$value" = "0" ]; then
-	        exit 1
-	    elif [ "$value" = "1" ]; then
-	        exit 0
-	    else
-	        echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-	        echo "dellWirelessCtl returned $value" >&2
-                exit 1
-	    fi
-	else 
-	    echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-	    echo "dellWirelessCtl ($DELL_WCTL) not available or executable" >&2
-	    exit 1
-	fi
-    else
-	echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-	echo "Access type not supported" >&2
-	exit 1
-    fi
-elif [ "$HAL_PROP_KILLSWITCH_TYPE" = "wwan" ]; then
-    if [ "$HAL_PROP_KILLSWITCH_ACCESS_METHOD" = "dell" ]; then
-	if [ -x "$DELL_WCTL" ]; then
-   	    # TODO: write our own binary that links with libsmbios?
-	    $DELL_WCTL --st_wwan
-	    value=$?
-	    if [ "$value" = "0" ]; then
-	        exit 1
-	    elif [ "$value" = "1" ]; then
-	        exit 0
-	    else
-	        echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-	        echo "dellWirelessCtl returned $value" >&2
-                exit 1
-	    fi
-	else 
-	    echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-	    echo "dellWirelessCtl ($DELL_WCTL) not available or executable" >&2
-	    exit 1
-	fi
-    else
-	echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-	echo "Access type not supported" >&2
-	exit 1
-    fi
 else
     echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
     echo "Killswitch type not supported" >&2
diff -upr hal-0.5.12/tools/linux/hal-system-killswitch-set-power-linux hal-0.5.12.new/tools/linux/hal-system-killswitch-set-power-linux
--- hal-0.5.12/tools/linux/hal-system-killswitch-set-power-linux	2008-10-24 09:45:54.000000000 +0100
+++ hal-0.5.12.new/tools/linux/hal-system-killswitch-set-power-linux	2009-03-03 10:44:49.000000000 +0000
@@ -8,11 +8,6 @@
 # the Free Software Foundation; either version 2 of the License, or
 # (at your option) any later version.
 
-DELL_WCTL=/usr/bin/dellWirelessCtl
-if [ -x "/usr/sbin/dellWirelessCtl" ]; then
-    DELL_WCTL=/usr/sbin/dellWirelessCtl
-fi
-
 if [ "$HAL_PROP_KILLSWITCH_TYPE" = "bluetooth" ]; then
     if [ "$HAL_PROP_KILLSWITCH_ACCESS_METHOD" = "sonypic" ]; then
 	hal-system-sonypic setbluetooth $value
@@ -23,26 +18,6 @@ if [ "$HAL_PROP_KILLSWITCH_TYPE" = "blue
 	    exit 1
 	fi
 	exit 0
-    elif [ "$HAL_PROP_KILLSWITCH_ACCESS_METHOD" = "dell" ]; then
-	if [ -x "$DELL_WCTL" ]; then
-	    # TODO: write our own binary that links with libsmbios?
-	    if [ "$value" = "true" ]; then
-	        $DELL_WCTL --bt 1
-	        ret=$?
-	    else
-	        $DELL_WCTL --bt 0
-	        ret=$?
-	    fi
-	    if [ "$ret" != "0" ]; then
-	        echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-	        echo "dellWirelessCtl returned $ret" >&2
-	        exit 1
-	    fi
-        else
-            echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-            echo "dellWirelessCtl ($DELL_WCTL) not available or executable" >&2
-            exit 1
-        fi
     elif [ "$HAL_PROP_KILLSWITCH_ACCESS_METHOD" = thinkpad -a -w "$HAL_PROP_LINUX_SYSFS_PATH" ]; then
         if [ "$value" = "true" ]; then 
 	    bit=1; 
@@ -61,60 +36,6 @@ if [ "$HAL_PROP_KILLSWITCH_TYPE" = "blue
 	echo "Access type not supported" >&2
 	exit 1
     fi
-elif [ "$HAL_PROP_KILLSWITCH_TYPE" = "wlan" ]; then
-    if [ "$HAL_PROP_KILLSWITCH_ACCESS_METHOD" = "dell" ]; then
-	if [ -x "$DELL_WCTL" ]; then
-	    # As a side effect we disable the physical kill switch
-	    # TODO: write our own binary that links with libsmbios?
-	    if [ "$value" = "true" ]; then
-	        $DELL_WCTL --sw_wlan 0 --wlan 1
-	        ret=$?
-	    else
-	        $DELL_WCTL --sw_wlan 0 --wlan 0
-	        ret=$?
-	    fi
-	    if [ "$ret" != "0" ]; then
-	        echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-	        echo "dellWirelessCtl returned $ret" >&2
-                exit 1
-	    fi
-        else
-            echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-            echo "dellWirelessCtl ($DELL_WCTL) not available or executable" >&2
-            exit 1
-        fi
-    else
-	echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-	echo "Access type not supported" >&2
-	exit 1
-    fi
-elif [ "$HAL_PROP_KILLSWITCH_TYPE" = "wwan" ]; then
-    if [ "$HAL_PROP_KILLSWITCH_ACCESS_METHOD" = "dell" ]; then 
-	if [ -x "$DELL_WCTL" ]; then
-	    # As a side effect we disable the physical kill switch
-	    # TODO: write our own binary that links with libsmbios?
-	    if [ "$value" = "true" ]; then
-	        $DELL_WCTL --sw_wwan 0 --wwan 1
-	        ret=$?
-	    else
-	        $DELL_WCTL --sw_wwan 0 --wwan 0
-	        ret=$?
-	    fi
-	    if [ "$ret" != "0" ]; then
-	        echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-	        echo "dellWirelessCtl returned $ret" >&2
-                exit 1
-	    fi
-        else
-            echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-            echo "dellWirelessCtl ($DELL_WCTL) not available or executable" >&2
-            exit 1
-        fi
-    else
-	echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
-	echo "Access type not supported" >&2
-	exit 1
-    fi
 else
     echo "org.freedesktop.Hal.Device.KillSwitch.NotSupported" >&2
     echo "Killswitch type not supported" >&2
