--- hal-0.5.9/fdi/policy/10osvendor/20-storage-methods.fdi.orig	2007-05-01 21:39:31.000000000 +0000
+++ hal-0.5.9/fdi/policy/10osvendor/20-storage-methods.fdi	2007-05-01 21:51:14.000000000 +0000
@@ -100,6 +100,11 @@
 	</match>
       </match>
 
+      <!-- Hide partitions marked as Compaq Diagnostics -->
+      <match key="volume.partition.type" string="0x12">
+        <merge key="volume.ignore" type="bool">true</merge>
+      </match>
+
       <!-- EFI firmware partitions -->
       <match key="volume.fstype" string="vfat">
 	<match key="volume.label" string="EFI">
