From 799b445e1cf2ca9810bb7b2a4fca73e897a8da4e Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Mon, 9 Apr 2018 07:56:31 -0600
Subject: [PATCH] Fix ARM build

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index efae71d..8db8c80 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -630,8 +630,8 @@ if (NOT ${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm")
             PURPOSE "Required by the Krita for vectorization")
         macro_bool_to_01(Vc_FOUND HAVE_VC)
     endif()
-    configure_file(config-vc.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-vc.h )
 endif()
+configure_file(config-vc.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-vc.h )
 
 if(HAVE_VC)
     message(STATUS "Vc found!")
-- 
2.16.2

