diff --git a/cmake/modules/FindOCIO.cmake b/cmake/modules/FindOCIO.cmake
index 89b404088c..02a8753221 100644
--- a/cmake/modules/FindOCIO.cmake
+++ b/cmake/modules/FindOCIO.cmake
@@ -34,11 +34,11 @@ find_path(OCIO_INCLUDE_DIR OpenColorIO.h
         /usr/local/include
         /sw/include
         /opt/local/include
-        PATH_SUFFIXES OpenColorIO
+        PATH_SUFFIXES OpenColorIO1
         DOC "The directory where OpenColorIO/OpenColorIO.h resides"
 )
 
-find_library(OCIO_LIBRARIES OpenColorIO
+find_library(OCIO_LIBRARIES OpenColorIO1
         PATHS
         ${OCIO_LIBRARY_PATH}
         ${OCIO_PATH}/lib/
