diff --git a/libs/ui/opengl/KisOpenGLModeProber.cpp b/libs/ui/opengl/KisOpenGLModeProber.cpp
index 6589e67034..01e3040759 100644
--- a/libs/ui/opengl/KisOpenGLModeProber.cpp
+++ b/libs/ui/opengl/KisOpenGLModeProber.cpp
@@ -181,7 +181,10 @@ KisOpenGLModeProber::probeFormat(const KisOpenGL::RendererConfig &rendererConfig
         formatSetter.reset(new SurfaceFormatSetter(format));
 
         QGuiApplication::setDesktopSettingsAware(false);
+        QString platformtheme = qgetenv("QT_QPA_PLATFORMTHEME");
+        qunsetenv("QT_QPA_PLATFORMTHEME");
         application.reset(new QGuiApplication(argc, &argv));
+        qputenv("QT_QPA_PLATFORMTHEME", platformtheme.toLocal8Bit());
         QGuiApplication::setDesktopSettingsAware(true);
     }
 
