diff --git a/cmake/modules/FindPyQt5.py b/cmake/modules/FindPyQt5.py
index 5849f40868..a42ba6c624 100644
--- a/cmake/modules/FindPyQt5.py
+++ b/cmake/modules/FindPyQt5.py
@@ -2,7 +2,7 @@
 # Redistribution and use is allowed according to the terms of the BSD license.
 # For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 
-import sys
+import sys, site
 import os
 
 try:
@@ -41,7 +41,7 @@ except ValueError:
     pass
 
 # FIXME This next line is just a little bit too crude.
-pyqt_sip_dir = os.path.join(sys.prefix, "share", "sip", "PyQt5")
+pyqt_sip_dir = os.path.join(site.getsitepackages()[0], "PyQt5", "bindings")
 print("pyqt_sip_dir:%s" % pyqt_sip_dir)
 
 print("pyqt_sip_flags:%s" % PyQt5.QtCore.PYQT_CONFIGURATION["sip_flags"])
