diff --git a/plugins/extensions/pykrita/sip/CMakeLists.txt b/plugins/extensions/pykrita/sip/CMakeLists.txt
index 1f47f0af1c..d5a24f86ee 100644
--- a/plugins/extensions/pykrita/sip/CMakeLists.txt
+++ b/plugins/extensions/pykrita/sip/CMakeLists.txt
@@ -12,9 +12,13 @@ set(SIP_INCLUDES
     ${PYQT_SIP_DIR_OVERRIDE}
      ./krita)
 
+if(${PYQT5_VERSION} STRGREATER_EQUAL "050b00")
+    set(SIP_NAME -n PyQt5.sip)
+endif()
+
 set(SIP_CONCAT_PARTS 1)
 set(SIP_TAGS ALL WS_X11 ${PYQT5_VERSION_TAG})
-set(SIP_EXTRA_OPTIONS -g -x PyKDE_QVector)
+set(SIP_EXTRA_OPTIONS -g -x PyKDE_QVector ${SIP_NAME})
 
 set(PYTHON_SITE_PACKAGES_INSTALL_DIR ${LIB_INSTALL_DIR}/krita-python-libs)
 file(GLOB PYKRITA_KRITA_sip_files ./krita/*.sip)
