--- dmenu-3.9.orig/config.mk	2008-09-13 20:45:32.000000000 -0300
+++ dmenu-3.9/config.mk	2008-09-13 20:46:42.000000000 -0300
@@ -4,11 +4,11 @@ VERSION = 3.9
 # Customize below to fit your system
 
 # paths
-PREFIX = /usr/local
+PREFIX = /usr
 MANPREFIX = ${PREFIX}/share/man
 
-X11INC = /usr/X11R6/include
-X11LIB = /usr/X11R6/lib
+X11INC = /usr/include/X11
+X11LIB = /usr/lib/X11
 
 # Xinerama, comment if you don't want it
 XINERAMALIBS = -L${X11LIB} -lXinerama
