--- lesstif-0.95.0/include/Motif-2.1/XmI/XpmI.h.BAD	2006-08-30 10:57:12.000000000 -0500
+++ lesstif-0.95.0/include/Motif-2.1/XmI/XpmI.h	2006-08-30 10:58:07.000000000 -0500
@@ -217,8 +217,8 @@
 FUNC(xpmHashSlot, xpmHashAtom *, (xpmHashTable *table, char *s));
 FUNC(xpmHashIntern, int, (xpmHashTable *table, char *tag, void *data));
 
-#define HashAtomData(i) ((void *)i)
-#define HashColorIndex(slot) ((unsigned int)((*slot)->data))
+#define HashAtomData(i) ((void *)((unsigned long)i))
+#define HashColorIndex(slot) ((unsigned long)((*slot)->data))
 #define USE_HASHTABLE (cpp > 2 && ncolors > 4)
 
 /* I/O utility */
