--- lesstif-0.95.0/lib/Xm-2.1/Xpmscan.c.BAD	2006-08-29 14:34:31.000000000 -0500
+++ lesstif-0.95.0/lib/Xm-2.1/Xpmscan.c	2006-08-29 14:36:05.000000000 -0500
@@ -672,8 +672,8 @@
     char *dst;
     unsigned int *iptr;
     char *data;
-    unsigned int x, y, i;
-    int bits, depth, ibu, ibpp, offset;
+    unsigned int x, y;
+    int bits, depth, ibu, ibpp, offset, i;
     unsigned long lbt;
     Pixel pixel, px;
 
@@ -684,6 +684,9 @@
     ibpp = image->bits_per_pixel;
     offset = image->xoffset;
 
+    if (image->bitmap_unit < 0)
+	return (XpmNoMemory);
+
     if ((image->bits_per_pixel | image->depth) == 1) {
 	ibu = image->bitmap_unit;
 	for (y = 0; y < height; y++)
--- lesstif-0.95.0/lib/Xm-2.1/Xpmcreate.c.BAD	2006-08-29 14:36:13.000000000 -0500
+++ lesstif-0.95.0/lib/Xm-2.1/Xpmcreate.c	2006-08-29 14:36:36.000000000 -0500
@@ -1265,10 +1265,10 @@
     register char *src;
     register char *dst;
     register unsigned int *iptr;
-    register unsigned int x, y, i;
+    register unsigned int x, y;
     register char *data;
     Pixel pixel, px;
-    int nbytes, depth, ibu, ibpp;
+    int nbytes, depth, ibu, ibpp, i;
 
     data = image->data;
     iptr = pixelindex;
