--- lesstif-0.95.0/lib/Xm-2.1/Transltns.old	2006-12-31 18:28:29.000000000 +0200
+++ lesstif-0.95.0/lib/Xm-2.1/Transltns.c	2006-12-31 18:29:12.000000000 +0200
@@ -670,8 +670,8 @@
     /* Scroll wheel */
     "\nShift<Btn4Down>:		page-left()"
     "\nShift<Btn5Down>:		page-right()"
-    "\n<Btn5Down>:		scroll-one-line-up()"
-    "\n<Btn4Down>:		scroll-one-line-down()"
+    "\n<Btn4Down>:		scroll-one-line-up()"
+    "\n<Btn5Down>:		scroll-one-line-down()"
 #endif
     ;
 
