--- kdeadmin-4.2.2/ksystemlog/src/modes/system/systemConfiguration.h~	2009-04-22 16:11:44.000000000 +0200
+++ kdeadmin-4.2.2/ksystemlog/src/modes/system/systemConfiguration.h	2009-04-22 16:11:56.000000000 +0200
@@ -39,7 +39,7 @@
 		SystemConfiguration() :
 			GenericLogModeConfiguration(
 					SYSTEM_LOG_MODE_ID, 
-					QStringList() << "/var/log/syslog",
+					QStringList() << "/var/log/messages.log",
 					QList<int>() << INFORMATION_LOG_LEVEL_ID
 			) {
 			
