--- a/acerhk.c
+++ b/acerhk.c
@@ -2668,7 +2668,9 @@ static int acerhk_proc_init(void)
     printk(KERN_INFO"acerhk: could not create /proc/driver/acerhk\n");
   }
   else {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
     proc_acer_dir->owner = THIS_MODULE;
+#endif
     /* now create several files, first general info ... */
     entry = create_proc_read_entry("info",
                                    0444, proc_acer_dir, acerhk_proc_info, NULL);
@@ -2677,7 +2679,9 @@ static int acerhk_proc_init(void)
       remove_proc_entry("driver/acerhk", NULL);
       retval = 0;
     } else {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
       entry->owner = THIS_MODULE;
+#endif
       /* ... last pressed key ... */
       entry = create_proc_read_entry("key",
                                      0444, proc_acer_dir, acerhk_proc_key, NULL);
@@ -2687,7 +2691,9 @@ static int acerhk_proc_init(void)
         remove_proc_entry("driver/acerhk", NULL);
         retval = 0;
       } else {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
         entry->owner = THIS_MODULE;
+#endif
         /* ... and led control file */
         entry = create_proc_entry("led", 0222, proc_acer_dir);
         if (entry == NULL) {
@@ -2699,7 +2705,9 @@ static int acerhk_proc_init(void)
         }
         else {
           entry->write_proc = acerhk_proc_led;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
           entry->owner = THIS_MODULE;
+#endif
           /* ... and wireless led controll file */
           entry = create_proc_entry("wirelessled", 0222, proc_acer_dir);
           if (entry == NULL) {
@@ -2712,7 +2720,9 @@ static int acerhk_proc_init(void)
           }
           else {
             entry->write_proc = acerhk_proc_wirelessled;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
             entry->owner = THIS_MODULE;
+#endif
             /* ... and bluetooth led controll file */
             entry = create_proc_entry("blueled", 0222, proc_acer_dir);
             if (entry == NULL) {
@@ -2725,7 +2735,9 @@ static int acerhk_proc_init(void)
               retval = 0;
             } else {
               entry->write_proc = acerhk_proc_blueled;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
               entry->owner = THIS_MODULE;
+#endif
               retval = 1;
 #ifdef ACERDEBUG
               /* add extra file for debugging purposes */
@@ -2742,7 +2754,9 @@ static int acerhk_proc_init(void)
               }
               else {
                 entry->write_proc = acerhk_proc_debug;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
                 entry->owner = THIS_MODULE;
+#endif
                 retval = 1;
               }
 #endif
