diff -urN a/js/src/config/milestone.pl b/js/src/config/milestone.pl
--- a/js/src/config/milestone.pl	2013-02-11 15:33:22.000000000 -0700
+++ b/js/src/config/milestone.pl	2015-07-11 12:38:09.202030816 -0600
@@ -55,7 +55,7 @@
 #
 my $milestone = Moz::Milestone::getOfficialMilestone($MILESTONE_FILE);
 
-if (defined(@TEMPLATE_FILE)) {
+if ($#TEMPLATE_FILE >= 0) {
   my $TFILE;
 
   foreach $TFILE (@TEMPLATE_FILE) {
