diff -ruN -x .cvsignore -x .gitignore xf86-input-jamstudio-becf2d28fc4946b04cd6486427d885a989710533/README xf86-input-jamstudio-cbc4a590ff9c44ac6ee9b5376bcc6f1ddd8bb9f8/README
--- xf86-input-jamstudio-becf2d28fc4946b04cd6486427d885a989710533/README	1970-01-01 00:00:00.000000000 +0000
+++ xf86-input-jamstudio-cbc4a590ff9c44ac6ee9b5376bcc6f1ddd8bb9f8/README	2009-01-31 04:23:01.000000000 +0000
@@ -0,0 +1,20 @@
+xf86-input-jamstudio - JamStudio pentablet input driver for the Xorg X server
+
+Please submit bugs & patches to the Xorg bugzilla:
+
+        https://bugs.freedesktop.org/enter_bug.cgi?product=xorg
+
+All questions regarding this software should be directed at the
+Xorg mailing list:
+
+        http://lists.freedesktop.org/mailman/listinfo/xorg
+
+The master development code repository can be found at:
+
+        git://anongit.freedesktop.org/git/xorg/driver/xf86-input-jamstudio
+
+        http://cgit.freedesktop.org/xorg/driver/xf86-input-jamstudio
+
+For more information on the git code manager, see:
+
+        http://wiki.x.org/wiki/GitPage
diff -ruN -x .cvsignore -x .gitignore xf86-input-jamstudio-becf2d28fc4946b04cd6486427d885a989710533/man/js_x.man xf86-input-jamstudio-cbc4a590ff9c44ac6ee9b5376bcc6f1ddd8bb9f8/man/js_x.man
--- xf86-input-jamstudio-becf2d28fc4946b04cd6486427d885a989710533/man/js_x.man	2008-03-20 19:55:26.000000000 +0000
+++ xf86-input-jamstudio-cbc4a590ff9c44ac6ee9b5376bcc6f1ddd8bb9f8/man/js_x.man	2009-01-31 04:23:01.000000000 +0000
@@ -62,6 +62,6 @@
 where X equals the value read from the tablet.
 .RE
 .SH "SEE ALSO"
-__xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), xorgconfig(__appmansuffix__), Xserver(__appmansuffix__), X(__miscmansuffix__).
+__xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), Xserver(__appmansuffix__), X(__miscmansuffix__).
 .SH AUTHORS
 Brian Goines <bgoines78@comcast.net>
diff -ruN -x .cvsignore -x .gitignore xf86-input-jamstudio-becf2d28fc4946b04cd6486427d885a989710533/src/js_x.c xf86-input-jamstudio-cbc4a590ff9c44ac6ee9b5376bcc6f1ddd8bb9f8/src/js_x.c
--- xf86-input-jamstudio-becf2d28fc4946b04cd6486427d885a989710533/src/js_x.c	2008-03-20 19:55:26.000000000 +0000
+++ xf86-input-jamstudio-cbc4a590ff9c44ac6ee9b5376bcc6f1ddd8bb9f8/src/js_x.c	2009-01-31 04:23:01.000000000 +0000
@@ -27,10 +27,7 @@
 #endif
 
 #include <sys/types.h>
-#include "xf86Version.h"
-#if XORG_VERSION_CURRENT >= XF86_VERSION_NUMERIC(3,9,0,0,0)
-#define XFREE86_V4 1
-#endif
+#include "xorgVersion.h"
 #include "misc.h"
 #include "xf86.h"
 #include "xf86_OSproc.h"
@@ -38,11 +35,10 @@
 #include "exevents.h"		/* Needed for InitValuator/Proximity stuff */
 #include "mipointer.h"
 
-#ifdef XFree86LOADER
 #include "xf86Module.h"
-#endif
 #include <errno.h>
 #include <string.h>
+#include <unistd.h>
 
 #define JSX_XCOORD	65584
 #define JSX_YCOORD	65585
@@ -51,8 +47,6 @@
 
 #define SYSCALL(call) while(((call) == -1) && (errno == EINTR))
 
-#ifdef XFREE86_V4
-
 struct hiddev_event
 {
    unsigned hid;
@@ -154,8 +148,13 @@
    int width, height;
    int deltaX, deltaY;
 
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) == 0
    width = miPointerCurrentScreen()->width;
    height = miPointerCurrentScreen()->height;
+#else
+   width = miPointerGetScreen(local->dev)->width;
+   height = miPointerGetScreen(local->dev)->height;
+#endif
 /*
 deltaX=(float)width/priv->jsxMaxX; deltaY=(float)height/priv->jsxMaxY;
 */
@@ -192,7 +191,10 @@
 	 return !Success;
       if (InitProximityClassDeviceStruct(pJS_X) == FALSE)
 	 return !Success;
-      if (InitValuatorClassDeviceStruct(pJS_X, nbaxes, xf86GetMotionEvents,
+      if (InitValuatorClassDeviceStruct(pJS_X, nbaxes,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
+					xf86GetMotionEvents,
+#endif
 					local->history_size,
 					Absolute | OutOfProximity) == FALSE)
 	 return !Success;
@@ -330,8 +332,6 @@
 _X_EXPORT InputDriverRec JAMSTUDIO =
       { 1, "js_x", NULL, xf86JS_XInit, xf86JS_XUnInit, NULL, 0 };
 
-#ifdef XFree86LOADER
-
 static void
 xf86JS_XUnplug(pointer p)
 {
@@ -363,5 +363,3 @@
    xf86JS_XPlug,
    xf86JS_XUnplug
 };
-#endif
-#endif
