networking/ifplugd.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/networking/ifplugd.c b/networking/ifplugd.c
index b578f4c..8fad69a 100644
--- a/networking/ifplugd.c
+++ b/networking/ifplugd.c
@@ -34,16 +34,17 @@
 #include "libbb.h"
 
 #include "fix_u32.h"
-#include <linux/if.h>
-#include <linux/mii.h>
-#include <linux/ethtool.h>
 #ifdef HAVE_NET_ETHERNET_H
 # include <net/ethernet.h>
 #endif
+#include <syslog.h>
+
+#include <linux/if.h>
+#include <linux/mii.h>
+#include <linux/ethtool.h>
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
 #include <linux/sockios.h>
-#include <syslog.h>
 
 #define __user
 #include <linux/wireless.h>
 
