From d67dbfa7c195fabce706ca03c439419450f9bd8a Mon Sep 17 00:00:00 2001
From: Derek Buitenhuis <derek.buitenhuis@gmail.com>
Date: Thu, 6 Mar 2014 20:01:17 +0000
Subject: [PATCH] libx265: Support API version 9

Signed-off-by: Derek Buitenhuis <derek.buitenhuis@gmail.com>
---
 configure            |    4 ++--
 libavcodec/libx265.c |   10 +++++-----
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/configure b/configure
index d3f5f33..4d7382a 100755
--- a/configure
+++ b/configure
@@ -3980,8 +3980,8 @@ enabled libx264           && require libx264 x264.h x264_encoder_encode -lx264 &
                              { check_cpp_condition x264.h "X264_BUILD >= 118" ||
                                die "ERROR: libx264 version must be >= 0.118."; }
 enabled libx265           && require_pkg_config x265 x265.h x265_encoder_encode &&
-                             { check_cpp_condition x265.h "X265_BUILD >= 7" ||
-                               die "ERROR: libx265 version must be >= 7."; }
+                             { check_cpp_condition x265.h "X265_BUILD >= 9" ||
+                               die "ERROR: libx265 version must be >= 9."; }
 enabled libxavs           && require libxavs xavs.h xavs_encoder_encode -lxavs
 enabled libxvid           && require libxvid xvid.h xvid_global -lxvidcore
 enabled openssl           && { check_lib openssl/ssl.h SSL_library_init -lssl -lcrypto ||
diff --git a/libavcodec/libx265.c b/libavcodec/libx265.c
index 961fc74..a6e4193 100644
--- a/libavcodec/libx265.c
+++ b/libavcodec/libx265.c
@@ -121,11 +121,11 @@ static av_cold int libx265_encode_init(AVCodecContext *avctx)
     av_reduce(&sar_num, &sar_den,
               avctx->sample_aspect_ratio.num,
               avctx->sample_aspect_ratio.den, 4096);
-    ctx->params->bEnableVuiParametersPresentFlag = 1;
-    ctx->params->bEnableAspectRatioIdc           = 1;
-    ctx->params->aspectRatioIdc                  = 255;
-    ctx->params->sarWidth                        = sar_num;
-    ctx->params->sarHeight                       = sar_den;
+    ctx->params->vui.bEnableVuiParametersPresentFlag = 1;
+    ctx->params->vui.bEnableAspectRatioIdc           = 1;
+    ctx->params->vui.aspectRatioIdc                  = 255;
+    ctx->params->vui.sarWidth                        = sar_num;
+    ctx->params->vui.sarHeight                       = sar_den;
 
     if (x265_max_bit_depth == 8)
         ctx->params->internalBitDepth = 8;
-- 
1.7.10.4

