--- node-v14.21.3/configure	2023-02-16 16:32:25.000000000 +0100
+++ node-v14.21.3/configure.new	2023-06-19 18:34:38.440723045 +0200
@@ -25,7 +25,7 @@
   from distutils.spawn import find_executable as which
 
 print('Node.js configure: Found Python {0}.{1}.{2}...'.format(*sys.version_info))
-acceptable_pythons = ((3,10), (3, 9), (3, 8), (3, 7), (3, 6), (3, 5), (2, 7))
+acceptable_pythons = ((3,11), (3,10), (3, 9), (3, 8), (3, 7), (3, 6), (3, 5), (2, 7))
 if sys.version_info[:2] in acceptable_pythons:
   import configure
 else:
