--- a/project/cmake/scripts/linux/Install.cmake	2017-03-20 17:17:49.000000000 +0100
+++ b/project/cmake/scripts/linux/Install.cmake	2017-05-20 15:42:09.608550173 +0200
@@ -199,7 +199,7 @@
   install(PROGRAMS ${CORE_SOURCE_DIR}/tools/EventClients/lib/python/bt/__init__.py
                    ${CORE_SOURCE_DIR}/tools/EventClients/lib/python/bt/bt.py
                    ${CORE_SOURCE_DIR}/tools/EventClients/lib/python/bt/hid.py
-          DESTINATION lib/python2.7/dist-packages/${APP_NAME_LC}/bt
+          DESTINATION lib/python2.7/site-packages/${APP_NAME_LC}/bt
           COMPONENT kodi-eventclients-common)
 
   # Install kodi-eventclients-common PS3 python files
@@ -208,7 +208,7 @@
                    ${CORE_SOURCE_DIR}/tools/EventClients/lib/python/ps3/sixaxis.py
                    ${CORE_SOURCE_DIR}/tools/EventClients/lib/python/ps3/sixpair.py
                    ${CORE_SOURCE_DIR}/tools/EventClients/lib/python/ps3/sixwatch.py
-          DESTINATION lib/python2.7/dist-packages/${APP_NAME_LC}/ps3
+          DESTINATION lib/python2.7/site-packages/${APP_NAME_LC}/ps3
           COMPONENT kodi-eventclients-common)
 
   # Install kodi-eventclients-common python files
@@ -218,7 +218,7 @@
                    "${CORE_SOURCE_DIR}/tools/EventClients/Clients/PS3 BD Remote/ps3_remote.py"
                    ${CORE_SOURCE_DIR}/tools/EventClients/lib/python/xbmcclient.py
                    ${CORE_SOURCE_DIR}/tools/EventClients/lib/python/zeroconf.py
-          DESTINATION lib/python2.7/dist-packages/${APP_NAME_LC}
+          DESTINATION lib/python2.7/site-packages/${APP_NAME_LC}
           COMPONENT kodi-eventclients-common)
 
   # Install kodi-eventclients-common icons
