--- nedit-5.5/util/misc.c.orig	2007-12-03 21:57:15.000000000 +0000
+++ nedit-5.5/util/misc.c	2007-12-03 21:59:30.000000000 +0000
@@ -475,6 +475,15 @@
     bestClass = 0;
     bestVisual = 0;
     for (i=0; i < nVis; i++) {
+        if (visList[i].depth >= 32 &&          
+           strstr(ServerVendor(display), "X.Org") != 0) {
+               /* Xorg 6.8.* 32-bit visuals (with alpha-channel) cause a lot
+                  of problems, so we have to skip them.
+                  Users can achieve the same effect with older versions of
+                  NEdit by setting the environment variable 
+                  XLIB_SKIP_ARGB_VISUALS. */
+            continue;
+        }
 	if (visList[i].depth > maxDepth) {
 	    maxDepth = visList[i].depth;
 	    bestClass = 0;
