diff --git a/lib/matplotlib/tests/test_polar.py b/lib/matplotlib/tests/test_polar.py
index ff2ad6433e..3457d8dbcf 100644
--- a/lib/matplotlib/tests/test_polar.py
+++ b/lib/matplotlib/tests/test_polar.py
@@ -220,7 +220,7 @@ def test_polar_rlabel_position():
     ax.tick_params(rotation='auto')


-@image_comparison(['polar_title_position.png'], style='mpl20')
+@image_comparison(['polar_title_position.png'], style='mpl20', tol=3.5)
 def test_polar_title_position():
     fig = plt.figure()
     ax = fig.add_subplot(projection='polar')
