diff -ruN a/scripts/murmur.ini b/scripts/murmur.ini
--- a/scripts/murmur.ini	2022-01-18 02:00:00.334013700 +0100
+++ b/scripts/murmur.ini	2022-01-20 22:32:41.267250949 +0100
@@ -13,7 +13,7 @@
 
 ; Path to database. If blank, will search for
 ; murmur.sqlite in default locations or create it if not found.
-database=
+database=/var/db/murmur/murmur.sqlite
 
 ; Murmur defaults to using SQLite with its default rollback journal.
 ; In some situations, using SQLite's write-ahead log (WAL) can be
@@ -99,7 +99,7 @@
 ; logs to the file 'murmur.log'. If you leave this field blank
 ; on Unix-like systems, Murmur will force itself into foreground
 ; mode which logs to the console.
-;logfile=murmur.log
+logfile=
 
 ; If set, Murmur will write its process ID to this file
 ; when running in daemon mode (when the -fg flag is not
