diff -urN a/src/imx_accel.c b/src/imx_accel.c
--- a/src/imx_accel.c	2011-07-26 15:22:47.000000000 -0600
+++ b/src/imx_accel.c	2015-02-15 11:33:42.364386418 -0700
@@ -23,6 +23,7 @@
  */
 
 #include <stdint.h>
+#include <xorg-server.h>
 #include "xf86.h"
 #include "imx_accel.h"
 
diff -urN a/src/imx_display.c b/src/imx_display.c
--- a/src/imx_display.c	2015-02-15 11:29:45.368705377 -0700
+++ b/src/imx_display.c	2015-02-15 11:34:33.749486568 -0700
@@ -32,6 +32,7 @@
 #include <fcntl.h>
 #include <linux/fb.h>
 
+#include <xorg-server.h>
 #include "xf86.h"
 #include "xf86Priv.h"
 #include "xf86Crtc.h"
diff -urN a/src/imx_exa_offscreen.c b/src/imx_exa_offscreen.c
--- a/src/imx_exa_offscreen.c	2015-02-15 11:29:45.378705414 -0700
+++ b/src/imx_exa_offscreen.c	2015-02-15 11:31:19.819025619 -0700
@@ -55,6 +55,7 @@
  */
 
 
+#include <xorg-server.h>
 #include "xf86.h"
 #include "exa.h"
 #include "imx.h"
diff -urN a/src/imx_exa_z160.c b/src/imx_exa_z160.c
--- a/src/imx_exa_z160.c	2015-02-15 11:30:51.368936740 -0700
+++ b/src/imx_exa_z160.c	2015-02-15 11:29:45.403705509 -0700
@@ -22,6 +22,7 @@
  * SOFTWARE.
  */
 
+#include <xorg-server.h>
 #include "xf86.h"
 #include "xf86_OSproc.h"
 #include "exa.h"
diff -urN a/src/imx_ext.c b/src/imx_ext.c
--- a/src/imx_ext.c	2015-02-15 11:29:45.388705452 -0700
+++ b/src/imx_ext.c	2015-02-15 11:36:11.689641208 -0700
@@ -22,6 +22,7 @@
  * SOFTWARE.
  */
 
+#include <xorg-server.h>
 #include <X11/X.h>
 #include <X11/Xproto.h>
 #include <dixstruct.h>
