Description: Fix a type definition conflict for UChar32
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/modules/web/src/main/native/Source/WTF/wtf/unicode/java/UnicodeJava.h
+++ b/modules/web/src/main/native/Source/WTF/wtf/unicode/java/UnicodeJava.h
@@ -26,7 +26,7 @@
 // #endif
 
 #ifndef __UMACHINE_H__ //XXX: recheck
-typedef uint32_t UChar32;
+typedef int32_t UChar32;
 #endif
 
 #define U_MASK(x) ((uint32_t)1<<(x))
