Description: Dynamic linking against ICU
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/modules/web/src/main/native/Source/WebCore/TargetJava.pri
+++ b/modules/web/src/main/native/Source/WebCore/TargetJava.pri
@@ -229,9 +229,9 @@
         # just for build debug: force verboce output from linker 
         QMAKE_LFLAGS +=  -Wl,--verbose
 
-        # statically link with icu libraries in order to avoid version conflict
+        # link with icu libraries
         QMAKE_LFLAGS += `pkg-config --libs-only-L icu-uc`
-        LIBS += -Wl,-Bstatic -licui18n -licuuc -licudata -Wl,-Bdynamic
+        LIBS += -licui18n -licuuc -licudata -ldl
     }
     solaris-cc {
         QMAKE_LFLAGS += -M$$PWD/mapfile-vers
