Index: b/build.gradle
===================================================================
--- a/build.gradle
+++ b/build.gradle
@@ -275,7 +275,7 @@ defineProperty("JAVA", cygpath("$JDK_HOM
 defineProperty("JAVAC", cygpath("$JDK_HOME/bin/javac${IS_WINDOWS ? '.exe' : ''}"))
 defineProperty("JAVAH", cygpath("$JDK_HOME/bin/javah${IS_WINDOWS ? '.exe' : ''}"))
 defineProperty("JAVADOC", cygpath("$JDK_HOME/bin/javadoc${IS_WINDOWS ? '.exe' : ''}"))
-defineProperty("JDK_DOCS", "https://docs.oracle.com/javase/8/docs/api/")
+defineProperty("JDK_DOCS", "file:///usr/share/doc/openjdk-8-doc/api/")
 
 defineProperty("javaRuntimeVersion", System.getProperty("java.runtime.version"))
 defineProperty("javaVersion", javaRuntimeVersion.split("-")[0])
