Description: Use SOURCE_DATE_EPOCH for the BUILD_TIMESTAMP
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/build.gradle
+++ b/build.gradle
@@ -1245,7 +1245,7 @@
     // We need to take the VersionInfo.java file and replace the various
     // properties within it
     def replacements = [
-        "BUILD_TIMESTAMP": new java.util.Date(),
+        "BUILD_TIMESTAMP": new java.util.Date(1000 * Long.parseLong(System.getenv("SOURCE_DATE_EPOCH"))),
         "HUDSON_JOB_NAME": HUDSON_JOB_NAME,
         "HUDSON_BUILD_NUMBER": HUDSON_BUILD_NUMBER,
         "PROMOTED_BUILD_NUMBER": PROMOTED_BUILD_NUMBER,
