Description: Fixes a build error with GCC 6
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
Bug-Debian: https://bugs.debian.org/837021
diff --git a/modules/fxpackager/src/main/native/library/common/PosixPlatform.cpp b/modules/fxpackager/src/main/native/library/common/PosixPlatform.cpp
index 5c8f4b9..7c35213 100644
--- a/modules/fxpackager/src/main/native/library/common/PosixPlatform.cpp
+++ b/modules/fxpackager/src/main/native/library/common/PosixPlatform.cpp
@@ -46,6 +46,7 @@
 #include <iostream>
 #include <dlfcn.h>
 #include <signal.h>
+#include <sys/wait.h>
 
 
 PosixPlatform::PosixPlatform(void) {
@@ -232,7 +233,7 @@ bool PosixProcess::Wait() {
 
     //TODO Use waitpid instead of wait
 #ifdef LINUX
-    wait();
+    wait(&status);
 #endif
 #ifdef MAC
     wpid = wait(&status);
