Description: Fixes an ambiguous call to the pow() function
Origin: backport, https://github.com/WebKit/webkit/commit/2931450725
--- a/modules/web/src/main/native/Source/WebCore/rendering/shapes/BoxShape.cpp
+++ b/modules/web/src/main/native/Source/WebCore/rendering/shapes/BoxShape.cpp
@@ -43,7 +43,7 @@
 
     LayoutUnit ratio = radius / margin;
     if (ratio < 1)
-        return radius + (margin * (1 + pow(ratio - 1, 3)));
+        return radius + (margin * (1 + pow(ratio - 1, 3.0)));
 
     return radius + margin;
 }
