--- a/modules/web/src/main/native/Source/JavaScriptCore/API/JSStringRef.h
+++ b/modules/web/src/main/native/Source/JavaScriptCore/API/JSStringRef.h
@@ -32,6 +32,7 @@
 #include <stdbool.h>
 #endif
 #include <stddef.h> /* for size_t */
+#include <uchar.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -43,7 +44,7 @@
 @typedef JSChar
 @abstract A Unicode character.
 */
-    typedef unsigned short JSChar;
+    typedef char16_t JSChar;
 #else
     typedef wchar_t JSChar;
 #endif
--- a/modules/web/src/main/native/Source/WTF/wtf/unicode/java/UnicodeJava.h
+++ b/modules/web/src/main/native/Source/WTF/wtf/unicode/java/UnicodeJava.h
@@ -18,7 +18,7 @@
 #if PLATFORM(JAVA) && OS(WINDOWS)
 typedef wchar_t UChar;
 #else
-typedef uint16_t UChar;
+typedef char16_t UChar;
 #endif
 
 // #ifdef UChar32
