--- src/effect/Font.cpp.orig	2009-06-21 12:58:06.000000000 +0200
+++ src/effect/Font.cpp		2009-06-21 12:58:06.000000000 +0200
@@ -29 +29 @@
-    int ucsLength = fribidi_utf8_to_unicode(const_cast<char*>(text.c_str()),
+    int ucsLength = fribidi_charset_to_unicode(FRIBIDI_CHAR_SET_UTF8, const_cast<char*>(text.c_str()),
@@ -40 +40 @@
-    int length = fribidi_unicode_to_utf8(visualString, ucsLength, buffer);
+    int length = fribidi_unicode_to_charset(FRIBIDI_CHAR_SET_UTF8, visualString, ucsLength, buffer);
