From a77f11cb0120a0e176fe09c06f37d6d46dfe1caa Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Fri, 19 Feb 2016 21:40:56 -0700
Subject: [PATCH] arch linux arm modifications

---
 include/configs/rpi.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/rpi.h b/include/configs/rpi.h
index e293f2ed31..27c68b151e 100644
--- a/include/configs/rpi.h
+++ b/include/configs/rpi.h
@@ -172,7 +172,7 @@
 	"pxefile_addr_r=0x00100000\0" \
 	"kernel_addr_r=0x01000000\0" \
 	"scriptaddr=0x02000000\0" \
-	"ramdisk_addr_r=0x02100000\0" \
+	"ramdisk_addr_r=0x03000000\0" \
 
 #define BOOT_TARGET_DEVICES(func) \
 	func(MMC, mmc, 0) \
-- 
2.12.0

