From a8956f99a655204a633440f9cac8558bae00df49 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sat, 8 Aug 2020 18:51:00 -0600
Subject: [PATCH] rpi: increase space for kernel

Signed-off-by: Kevin Mihelich <kevin@archlinuxarm.org>
---
 include/configs/rpi.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/configs/rpi.h b/include/configs/rpi.h
index 834f1cd236..7a34049a10 100644
--- a/include/configs/rpi.h
+++ b/include/configs/rpi.h
@@ -163,10 +163,10 @@
 	"fdt_high=" FDT_HIGH "\0" \
 	"initrd_high=" INITRD_HIGH "\0" \
 	"kernel_addr_r=0x00080000\0" \
-	"scriptaddr=0x02400000\0" \
-	"pxefile_addr_r=0x02500000\0" \
-	"fdt_addr_r=0x02600000\0" \
-	"ramdisk_addr_r=0x02700000\0"
+	"scriptaddr=0x02C00000\0" \
+	"pxefile_addr_r=0x02D00000\0" \
+	"fdt_addr_r=0x02E00000\0" \
+	"ramdisk_addr_r=0x02F00000\0"
 
 #if CONFIG_IS_ENABLED(CMD_MMC)
 	#define BOOT_TARGET_MMC(func) \
-- 
2.24.1

