From bd799e156242713c39ad2b68a5e5ef7053f37acb Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Fri, 19 Feb 2016 21:40:56 -0700
Subject: [PATCH] arch linux arm modifications

---
 include/config_distro_defaults.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/config_distro_defaults.h b/include/config_distro_defaults.h
index 0f85cd0..8d6c9e9 100644
--- a/include/config_distro_defaults.h
+++ b/include/config_distro_defaults.h
@@ -7,6 +7,8 @@
 #ifndef _CONFIG_CMD_DISTRO_DEFAULTS_H
 #define _CONFIG_CMD_DISTRO_DEFAULTS_H
 
+#define CONFIG_IDENT_STRING             " Arch Linux ARM"
+
 /*
  * List of all commands and options that when defined enables support for
  * features required by distros to support boards in a standardised and
-- 
2.7.0

