From a8956f99a655204a633440f9cac8558bae00df49 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sat, 8 Aug 2020 18:51:00 -0600
Subject: [PATCH] rpi: increase space for kernel

Signed-off-by: Kevin Mihelich <kevin@archlinuxarm.org>
---
 board/raspberrypi/rpi/rpi.env | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/board/raspberrypi/rpi/rpi.env b/board/raspberrypi/rpi/rpi.env
index 3022828..836c510 100644
--- a/board/raspberrypi/rpi/rpi.env
+++ b/board/raspberrypi/rpi/rpi.env
@@ -69,9 +69,9 @@ fdt_high=ffffffff
 initrd_high=ffffffff
 #endif
 kernel_addr_r=0x00080000
-scriptaddr=0x02400000
-pxefile_addr_r=0x02500000
-fdt_addr_r=0x02600000
-ramdisk_addr_r=0x02700000
+scriptaddr=0x02C00000
+pxefile_addr_r=0x02D00000
+fdt_addr_r=0x02E00000
+ramdisk_addr_r=0x02F00000

 boot_targets=mmc usb pxe dhcp
--
2.43.1

