--- a/Input/mikmod/drv_xmms.c	2003-05-19 23:22:06.000000000 +0200
+++ b/Input/mikmod/drv_xmms.c	2012-11-16 18:52:41.264644767 +0100
@@ -117,6 +117,10 @@
 	return VC_Init();
 }
 
+static void xmms_CommandLine(CHAR * commandLine)
+{
+}
+
 MDRIVER drv_xmms =
 {
 	NULL,
@@ -126,5 +130,6 @@
         "xmms",
         NULL,
 #endif
+	xmms_CommandLine, // Was missing
         xmms_IsThere, 
 	VC_SampleLoad,
