--- xmms/xmms.desktop	2002-02-11 23:15:56.000000000 +0000
+++ xmms.desktop	2006-02-25 14:21:03.000000000 +0000
@@ -1,21 +1,16 @@
 [Desktop Entry]
 Name=XMMS
 Comment=X Multimedia System
-comment[az]=X Multimedya Sistemi
-comment[ca]=Sistema Multimèdia per a X
-comment[cs]=Multimediální pøehrávaè
-comment[es]=Sistema Multimedia para X
-comment[gl]=Sistema Multimedia para X
-comment[hr]=X Multimedijski Sustav
-comment[nn]=X Multimedia-system
-comment[pt_BR]=X Multimedia System
-comment[ro]=X MultiMedia Sistem
-comment[th]=ÃÐººÁÑÅµÔÁÕà´ÕÂº¹ X
-comment[tr]=X Multimedya Sistem
-comment[zh_TW]=X ¦h´CÅé¨t²Î
-Encoding=Legacy-Mixed
+Comment[es]=Sistema Multimedia para X
+Comment[gl]=Sistema Multimedia para X
+Comment[hr]=X Multimedijski Sustav
+Comment[pt_BR]=X Multimedia System
+Comment[ro]=X MultiMedia Sistem
+Comment[tr]=X Multimedya Sistem
+Encoding=UTF-8
 Exec=xmms
-Icon=xmms_mini.xpm
-MimeType=audio/x-scpls;audio/x-mpegurl;audio/mpegurl;audio/mp3;audio/x-mp3;audio/mpeg;audio/x-mpeg;audio/x-wav;application/x-ogg
-Terminal=0
+Icon=xmms
 Type=Application
+Categories=Application;AudioVideo;
+MimeType=audio/x-scpls;audio/x-mpegurl;audio/mpegurl;audio/mp3;audio/x-mp3;audio/mpeg;audio/x-mpeg;audio/x-wav;application/x-ogg
+Terminal=false
