--- xmms-1.2.10/xmms/main.c.fonts	2004-08-10 22:16:06.000000000 +0200
+++ xmms-1.2.10/xmms/main.c	2004-08-12 11:49:17.905598843 +0200
@@ -351,7 +351,8 @@
 	cfg.no_playlist_advance = FALSE;
 	cfg.smooth_title_scroll = TRUE;
 	cfg.random_skin_on_play = FALSE;
-	cfg.mainwin_use_xfont = FALSE;
+	cfg.mainwin_use_xfont = TRUE;
+	cfg.use_fontsets = TRUE;
 	cfg.use_pl_metadata = TRUE;
 
 	cfg.playlist_x = 295;
@@ -499,9 +500,9 @@
 		cfg.mainwin_font = NULL;
 	}
 	if (cfg.playlist_font == NULL)
-		cfg.playlist_font = g_strdup("-adobe-helvetica-bold-r-*-*-10-*");
+		cfg.playlist_font = g_strdup("-*-helvetica-bold-r-*-*-10-*,*");
 	if (cfg.mainwin_font == NULL)
-		cfg.mainwin_font = g_strdup("-adobe-helvetica-medium-r-*-*-8-*");
+		cfg.mainwin_font = g_strdup("-*-helvetica-medium-r-*-*-8-*,*");
 	if (cfg.gentitle_format == NULL)
 		cfg.gentitle_format = g_strdup("%p - %t");
 	if (cfg.outputplugin == NULL)
