--- xmms-1.2.11/Input/mpg123/common.c~	2007-11-18 20:37:41.000000000 +0100
+++ xmms-1.2.11/Input/mpg123/common.c	2007-11-18 20:40:10.000000000 +0100
@@ -274,6 +274,7 @@ static gboolean read_id3v2_tag(unsigned 
 			char *songname = mpg123_title;
 			mpg123_title =
 				mpg123_format_song_title(tag, mpg123_filename);
+			if (filept)
 			mpg123_ip.set_info(mpg123_title, mpg123_length,
 					   mpg123_bitrate * 1000,
 					   mpg123_frequency, mpg123_stereo);
