--- Makefile.in.org	2003-03-02 03:23:00.000000000 +0300
+++ Makefile.in	2006-07-14 08:33:40.000000000 +0400
@@ -172,7 +172,8 @@
   tag_parse_v1.cpp              \
   tag_render.cpp                \
   utils.cpp                     \
-  writers.cpp                   
+  writers.cpp                   \
+  rccpatch.cpp
 
 
 lib_LTLIBRARIES = libid3.la 
@@ -200,7 +201,7 @@
 	io_decorators.lo io_helpers.lo misc_support.lo mp3_parse.lo \
 	readers.lo spec.lo tag.lo tag_file.lo tag_find.lo tag_impl.lo \
 	tag_parse.lo tag_parse_lyrics3.lo tag_parse_musicmatch.lo \
-	tag_parse_v1.lo tag_render.lo utils.lo writers.lo
+	tag_parse_v1.lo tag_render.lo utils.lo writers.lo rccpatch.lo
 am_libid3_la_OBJECTS = $(am__objects_1)
 libid3_la_OBJECTS = $(am_libid3_la_OBJECTS)
 
@@ -233,7 +234,8 @@
 @AMDEP_TRUE@	./$(DEPDIR)/tag_parse_musicmatch.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/tag_parse_v1.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/tag_render.Plo ./$(DEPDIR)/utils.Plo \
-@AMDEP_TRUE@	./$(DEPDIR)/writers.Plo
+@AMDEP_TRUE@	./$(DEPDIR)/writers.Plo \
+@AMDEP_TRUE@	./$(DEPDIR)/rccpatch.Plo
 CXXCOMPILE = $(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
 	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS)
 LTCXXCOMPILE = $(LIBTOOL) --mode=compile $(CXX) $(DEFS) \
@@ -328,6 +330,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/tag_render.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/utils.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/writers.Plo@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/rccpatch.Plo@am__quote@
 
 distclean-depend:
 	-rm -rf ./$(DEPDIR)
