From 9bc4045471dd0f6a23023db6801173b0c624e127 Mon Sep 17 00:00:00 2001
From: graysky <therealgraysky@proton.me>
Date: Sat, 7 May 2022 06:11:27 -0400
Subject: [PATCH 2/2] add dav1d patch to build system

---
 cmake/modules/FindFFMPEG.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/modules/FindFFMPEG.cmake b/cmake/modules/FindFFMPEG.cmake
index a18b21c..fb7326f 100644
--- a/cmake/modules/FindFFMPEG.cmake
+++ b/cmake/modules/FindFFMPEG.cmake
@@ -280,6 +280,7 @@ if(NOT FFMPEG_FOUND)
                                     <SOURCE_DIR> &&
                                     patch -p1 < ${CMAKE_SOURCE_DIR}/tools/depends/target/ffmpeg/0001-rpi-Add-hevc-acceleration.patch &&
                                     patch -p1 < ${CMAKE_SOURCE_DIR}/tools/depends/target/ffmpeg/0002-use-mcpu-avoiding-march-and-mtune.patch &&
+                                    patch -p1 < ${CMAKE_SOURCE_DIR}/tools/depends/target/ffmpeg/0003-add-dav1d-patch-to-build-system.patch &&
                                     echo "########################################## patched ffmpeg ##############################"
                      )
 
-- 
2.36.0

