From fb98e356551cea67f8810a377edf5c1d6205bcb8 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 17 Jan 2017 18:36:03 -0700
Subject: [PATCH] disable LTO

---
 cmake/modules/LXQtCompilerSettings.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/modules/LXQtCompilerSettings.cmake b/cmake/modules/LXQtCompilerSettings.cmake
index 2b1e774..a07b35a 100644
--- a/cmake/modules/LXQtCompilerSettings.cmake
+++ b/cmake/modules/LXQtCompilerSettings.cmake
@@ -150,7 +150,7 @@ endif()
 if (CMAKE_COMPILER_IS_GNUCXX OR LXQT_COMPILER_IS_CLANGCXX)
     # -flto: use link-time optimizations to generate more efficient code
     if (CMAKE_COMPILER_IS_GNUCXX)
-        set(LTO_FLAGS "-flto -fuse-linker-plugin")
+        #set(LTO_FLAGS "-flto -fuse-linker-plugin")
         # When building static libraries with LTO in gcc >= 4.9,
         # "gcc-ar" and "gcc-ranlib" should be used instead of "ar" and "ranlib".
         # references:
-- 
2.11.0

