--- layout/applications.menu	2005-06-28 10:16:39.000000000 +0200
+++ layout/applications.menu	2005-09-12 23:21:20.000000000 +0200
@@ -7,7 +7,6 @@
   <Directory>Applications.directory</Directory>
 
   <!-- Scan legacy dirs first, as later items take priority -->
-  <KDELegacyDirs/>
   <LegacyDir>/etc/X11/applnk</LegacyDir>
   <LegacyDir>/usr/share/gnome/apps</LegacyDir>
 
@@ -18,6 +17,14 @@
   <!-- Read in overrides and child menus from applications-merged/ -->
   <DefaultMergeDirs/>
 
+  <!-- Archlinux submenu -->
+  <Menu>
+    <Name>Archlinux</Name>
+    <Directory>Archlinux.directory</Directory>
+    <Include>
+      <Category>Archlinux</Category>
+    </Include>
+  </Menu> <!-- End Archlinux -->
   <!-- Accessories submenu -->
   <Menu>
     <Name>Accessories</Name>
