From a912cb49eb0b6fa63cfb045e549a51746c704d23 Mon Sep 17 00:00:00 2001
From: Nigel Croxon <nigel.croxon@hpe.com>
Date: Wed, 19 Aug 2015 07:28:45 -0400
Subject: [PATCH 01/10] Add the missing URI device path to the unions.

Signed-off-by: Gary Ching-Pang Lin <chinpang@gmail.com>
SIgned-off-by: Nigel Croxon <nigel.croxon@hpe.com>
---
 inc/efidevp.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/inc/efidevp.h b/inc/efidevp.h
index 38923d9..b20c839 100644
--- a/inc/efidevp.h
+++ b/inc/efidevp.h
@@ -491,6 +491,7 @@ typedef union {
     MAC_ADDR_DEVICE_PATH                 MacAddr;
     IPv4_DEVICE_PATH                     Ipv4;
     IPv6_DEVICE_PATH                     Ipv6;
+    URI_DEVICE_PATH                      Uri;
     INFINIBAND_DEVICE_PATH               InfiniBand;
     UART_DEVICE_PATH                     Uart;
 
@@ -525,6 +526,7 @@ typedef union {
     MAC_ADDR_DEVICE_PATH                 *MacAddr;
     IPv4_DEVICE_PATH                     *Ipv4;
     IPv6_DEVICE_PATH                     *Ipv6;
+    URI_DEVICE_PATH                      *Uri;
     INFINIBAND_DEVICE_PATH               *InfiniBand;
     UART_DEVICE_PATH                     *Uart;
 
-- 
2.5.0

