diff -urN a/gnuefi/elf_aarch64_efi.lds b/gnuefi/elf_aarch64_efi.lds
--- a/gnuefi/elf_aarch64_efi.lds	2015-08-17 14:06:47.000000000 -0600
+++ b/gnuefi/elf_aarch64_efi.lds	2015-12-17 22:20:26.182403513 -0700
@@ -38,6 +38,7 @@
    . = ALIGN(16);
    _bss_end = .;
   }
+  .note.gnu.build-id : { *(.note.gnu.build-id) }
   .rela.dyn : { *(.rela.dyn) }
   .rela.plt : { *(.rela.plt) }
   .rela.got : { *(.rela.got) }
