From b97d0ae34150c75903b645bf49636e67bdd44f46 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Thu, 26 May 2016 06:29:07 -0600
Subject: [PATCH 7/8] add extra errata 843419 build flags

---
 arch/arm64/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index b7da5c34aac1..5e27fa3850c8 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -35,6 +35,7 @@ CHECKFLAGS	+= -D__aarch64__
 
 ifeq ($(CONFIG_ARM64_ERRATUM_843419), y)
 CFLAGS_MODULE	+= -mcmodel=large
+CFLAGS_MODULE	+= $(call cc-option, -mpc-relative-literal-loads)
 endif
 
 # Default value
-- 
2.13.0

