--- makefile.orig	2024-05-31 06:48:10.051261886 -0600
+++ makefile	2024-05-31 06:49:58.003425835 -0600
@@ -24,14 +24,14 @@
 #PLRSFLAGS= 
 
 #compilers supporting _int128 (eg gcc 4.2 and later)
-BITS=-DB128
-MPLRSOBJ2=lrslib2-mplrs.o lrslong2-mplrs.o
-LRSOBJ2=lrslib2.o lrslong2.o
+#BITS=-DB128
+#MPLRSOBJ2=lrslib2-mplrs.o lrslong2-mplrs.o
+#LRSOBJ2=lrslib2.o lrslong2.o
 
 #compilers without _int128 support (equivalent to make lrs64 in v.7.2)
-#BITS=
-#MPLRSOBJ2=
-#LRSOBJ2=
+BITS=
+MPLRSOBJ2=
+LRSOBJ2=
 
 #legacy lrs and mplrs
 #CFLAGS     = -g -Wall -I ${ARITH} 
@@ -231,11 +231,11 @@
 SHLIB ?=$(SONAME)$(SOMINOR)
 SHLINK ?=liblrs.so
 
-SHLIBOBJ2=lrslib2-shr.o lrslong2-shr.o
+#SHLIBOBJ2=lrslib2-shr.o lrslong2-shr.o
 
 # for 32 bit machines
 
-# SHLIBOBJ2=
+SHLIBOBJ2=
 
 SHLIBOBJ=lrslong1-shr.o lrslib1-shr.o  \
 	lrslibgmp-shr.o lrsgmp-shr.o lrsdriver-shr.o \
