From 927178e6fa67ab7f56160f42217dbf5d0b5d243b Mon Sep 17 00:00:00 2001
Date: Sun, 18 Sep 2022 08:12:22 -0400
Subject: [PATCH] add dav1d patch to build system

---
 cmake/modules/FindFFMPEG.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/modules/FindFFMPEG.cmake b/cmake/modules/FindFFMPEG.cmake
index 9ce807c..de4287d 100644
--- a/cmake/modules/FindFFMPEG.cmake
+++ b/cmake/modules/FindFFMPEG.cmake
@@ -279,6 +279,7 @@ if(NOT FFMPEG_FOUND)
                                     ${CMAKE_SOURCE_DIR}/tools/depends/target/ffmpeg/FindGnuTls.cmake
                                     <SOURCE_DIR> &&
                                     patch -p1 < ${CMAKE_SOURCE_DIR}/tools/depends/target/ffmpeg/0001-rpi-Add-hevc-acceleration.patch &&
+                                    patch -p1 < ${CMAKE_SOURCE_DIR}/tools/depends/target/ffmpeg/0002-add-dav1d-patch-to-build-system.patch &&
                                     echo "########################################## patched ffmpeg ##############################"
                      )
 
-- 
2.37.3

