From 22ac2d50eec4070fa58b1c7915a9fbf66070b57b Mon Sep 17 00:00:00 2001
Date: Mon, 9 Jan 2023 11:07:51 -0500
Subject: add dav1d patch to build system

---
 cmake/modules/FindFFMPEG.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/modules/FindFFMPEG.cmake b/cmake/modules/FindFFMPEG.cmake
index 9ce807c..d1c7810 100644
--- a/cmake/modules/FindFFMPEG.cmake
+++ b/cmake/modules/FindFFMPEG.cmake
@@ -279,6 +279,7 @@ if(NOT FFMPEG_FOUND)
                                     ${CMAKE_SOURCE_DIR}/tools/depends/target/ffmpeg/FindGnuTls.cmake
                                     <SOURCE_DIR> &&
                                     patch -p1 < ${CMAKE_SOURCE_DIR}/tools/depends/target/ffmpeg/0001-rpi-Add-hevc-acceleration.patch &&
+				    patch -p1 < ${CMAKE_SOURCE_DIR}/tools/depends/target/ffmpeg/build-fix-for-dav1d-1.0.0.patch &&
                                     echo "########################################## patched ffmpeg ##############################"
                      )
 
-- 
2.39.0

