Index: configure
===================================================================
--- configure	(revision 35421)
+++ configure	(working copy)
@@ -5968,8 +5968,8 @@
         cat > $TMPC << EOF
 #include <stdio.h>
 #include <cdio/version.h>
-#include <cdio/cdda.h>
-#include <cdio/paranoia.h>
+#include <cdio/paranoia/cdda.h>
+#include <cdio/paranoia/paranoia.h>
 int main(void) {
     void *test = cdda_verbose_set;
     printf("%s\n", CDIO_VERSION);
Index: gui/win32/interface.c
===================================================================
--- gui/win32/interface.c	(revision 35421)
+++ gui/win32/interface.c	(working copy)
@@ -24,7 +24,7 @@
 #include "config.h"

 #if defined(CONFIG_LIBCDIO)
-#include <cdio/cdda.h>
+#include <cdio/paranoia/cdda.h>
 #elif defined(CONFIG_CDDA)
 #include <cdda_interface.h>
 #endif
Index: stream/stream_cdda.c
===================================================================
--- stream/stream_cdda.c	(revision 35421)
+++ stream/stream_cdda.c	(working copy)
@@ -21,8 +21,8 @@
 #include <cdda_interface.h>
 #include <cdda_paranoia.h>
 #else
-#include <cdio/cdda.h>
-#include <cdio/paranoia.h>
+#include <cdio/paranoia/cdda.h>
+#include <cdio/paranoia/paranoia.h>
 #endif
 #include <stdio.h>
 #include <stdlib.h>
