--- libmpdemux/demux_gif.c-old  2012-08-20 08:47:01.000000000 -0600
+++ libmpdemux/demux_gif.c	2013-07-05 11:27:55.488387360 -0600
@@ -45,6 +45,16 @@
 
 #define GIF_SIGNATURE (('G' << 16) | ('I' << 8) | 'F')
 
+static void PrintGifError(void)
+{
+	char *Err = GifErrorString();
+
+	if (Err != NULL)
+		fprintf(stderr, "\nGIF-LIB error: %s.\n", Err);
+	else
+		fprintf(stderr, "\nGIF-LIB undefined error %d.\n", GifError());
+}
+
 #ifndef CONFIG_GIF_TVT_HACK
 // not supported by certain versions of the library
 static int my_read_gif(GifFileType *gif, uint8_t *buf, int len)
