--- mplayer.desktop.orig	2012-07-08 22:18:27.000000000 +0300
+++ mplayer.desktop	2012-07-08 22:19:08.000000000 +0300
@@ -23,8 +23,9 @@ Comment[fr]=Lit les films et musiques
 Comment[it]=Riproduce filmati e musica
 Comment[ja]=動画や音声のファイルを再生します
 Icon=mplayer
-TryExec=gmplayer
-Exec=gmplayer %F
+TryExec=mplayer
+Exec=mplayer -really-quiet %F
 Terminal=false
+NoDisplay=true
 Categories=GTK;AudioVideo;Audio;Video;Player;TV;
 MimeType=application/mxf;application/x-netshow-channel;application/ogg;application/ram;application/vnd.rn-realmedia;application/x-shockwave-flash;application/smil;audio/ac3;audio/x-adpcm;audio/x-aiff;audio/AMR;audio/AMR-WB;audio/x-ape;audio/basic;audio/flac;audio/x-flac;audio/x-flac+ogg;audio/x-m4b;audio/x-matroska;audio/mp2;audio/mp4;audio/mpeg;audio/x-mpegurl;audio/x-ms-asx;audio/x-ms-wma;audio/x-musepack;audio/ogg;audio/vnd.rn-realaudio;audio/x-scpls;audio/x-voc;audio/x-vorbis+ogg;audio/x-wav;audio/x-wavpack;video/3gpp;video/3gpp2;video/dv;video/x-flic;video/x-flv;video/x-matroska;video/mp2t;video/mp4;video/mpeg;video/x-ms-asf;video/x-ms-wmv;video/x-msvideo;video/x-nsv;video/ogg;video/x-ogm+ogg;video/quicktime;video/vnd.rn-realvideo;video/x-theora+ogg;video/webm;
