--- gcin-1.4.4/Makefile.orig	2009-04-12 13:37:28.000000000 +0200
+++ gcin-1.4.4/Makefile	2009-04-12 13:37:43.000000000 +0200
@@ -96,7 +96,7 @@
 	if [ $(QT4_IM) = 'Y' ]; then $(MAKE) -C qt4-im; fi
 
 gcin:   $(OBJS) $(IMdkitLIB) $(im-srv)
-	LD_RUN_PATH=.:$(gcinlibdir) \
+	LD_RUN_PATH=$(gcinlibdir) \
 	$(CC) $(EXTRA_LDFLAGS) -o $@ $(OBJS) $(IMdkitLIB) $(im-srv) -lXtst $(LDFLAGS) -L/usr/X11R6/lib
 	rm -f core.*
 	ln -sf $@ $@.test
@@ -108,7 +108,7 @@
 	$(CC) -o $@ $(OBJS_JUYIN_LEARN) $(LDFLAGS)
 	rm -f core.*
 sim2trad:        $(OBJS_sim2trad)
-	LD_RUN_PATH=.:$(gcinlibdir) \
+	LD_RUN_PATH=$(gcinlibdir) \
 	$(CC) -o $@ $(OBJS_sim2trad) $(LDFLAGS)
 	rm -f core.*
 trad2sim:	sim2trad
