diff -Naur kpogre-1.5.4/kpogre/DataTable/kpgdatatable.cpp kpogre-1.5.4-p/kpogre/DataTable/kpgdatatable.cpp
--- kpogre-1.5.4/kpogre/DataTable/kpgdatatable.cpp	2006-12-15 13:47:44.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DataTable/kpgdatatable.cpp	2007-02-02 05:19:15.000000000 +0100
@@ -894,7 +894,7 @@
 	
 	try
 	{
-		pqxx::result pqxxResult = pqxxXaction.Exec((std::string) strSql);
+		pqxx::result pqxxResult = pqxxXaction.exec((std::string) strSql);
 	
 		if(pqxxResult.size() != 1)
 		{
@@ -934,7 +934,7 @@
 	
 	try
 	{
-		pqxx::result pqxxResult = pqxxXaction.Exec((std::string) strSql);
+		pqxx::result pqxxResult = pqxxXaction.exec((std::string) strSql);
 	
 		if(pqxxResult.size() != 1)
 		{
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgaggregatesfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgaggregatesfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgaggregatesfolder.cpp	2006-12-27 21:35:12.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgaggregatesfolder.cpp	2007-02-02 05:19:15.000000000 +0100
@@ -61,7 +61,7 @@
 			
     try
     {
-        m_pqxxResultAggregates = pqxxXaction.Exec(strQuery.latin1());
+        m_pqxxResultAggregates = pqxxXaction.exec(strQuery.latin1());
 
         KPGAggregate *pAggregate = 0;
             
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgcastsfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgcastsfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgcastsfolder.cpp	2006-12-27 21:35:12.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgcastsfolder.cpp	2007-02-02 05:19:15.000000000 +0100
@@ -53,7 +53,7 @@
 		
 	try
     {
-        m_pqxxResultCasts = pqxxXaction.Exec(strQuery.latin1());
+        m_pqxxResultCasts = pqxxXaction.exec(strQuery.latin1());
     
         KPGCast *pCast = 0;
             
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgconnection.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgconnection.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgconnection.cpp	2006-09-12 14:07:52.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgconnection.cpp	2007-02-02 05:18:25.000000000 +0100
@@ -91,11 +91,14 @@
 // Run SQL statement, make conversion regard connection encoding
 pqxx::result KPGConnection::runQuery(const QString& strQuery, eTransType transType)
 {
+ 	QCString cstrDecodedQuery(m_pTextCodec->fromUnicode(strQuery));
+ 	QString strDecodedQuery(cstrDecodedQuery);
+ 	
  	// run query
     if(transType == eTransNormal)
 	{
 		work pqxxXaction(*this, "runQuery");
-		result R(pqxxXaction.exec(m_pTextCodec->fromUnicode(strQuery)));
+		result R(pqxxXaction.exec(strDecodedQuery));
 		pqxxXaction.commit();
   		return R;
 	}
@@ -104,14 +107,14 @@
 		if(transType == eTransNone)
 		{
 			pqxx::nontransaction pqxxXaction(*this, "runQuery");
-			result R(pqxxXaction.exec(m_pTextCodec->fromUnicode(strQuery)));
+			result R(pqxxXaction.exec(strDecodedQuery));
 			return R;
 		}
 	}	
 		
 	//	eTransRobust:
 	pqxx::robusttransaction<> pqxxXaction(*this, "runQuery");
-	result R(pqxxXaction.exec(m_pTextCodec->fromUnicode(strQuery)));
+	result R(pqxxXaction.exec(strDecodedQuery));
 	pqxxXaction.commit();
 	return R;
 }
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgconversionsfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgconversionsfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgconversionsfolder.cpp	2006-12-27 21:35:12.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgconversionsfolder.cpp	2007-02-02 05:19:15.000000000 +0100
@@ -52,7 +52,7 @@
 			
     try
     {
-        m_pqxxResultConversions = pqxxXaction.Exec(strQuery.latin1());
+        m_pqxxResultConversions = pqxxXaction.exec(strQuery.latin1());
     
         KPGConversion *pConversion = 0;
             
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgdatabase.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgdatabase.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgdatabase.cpp	2006-12-28 11:39:51.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgdatabase.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -202,7 +202,7 @@
 	{
 	   work pqxxXaction(*m_pDbConnection, "refreshSchemasList");		
 			
-	   m_pqxxResultSchemas = pqxxXaction.Exec(strQuery.latin1());
+	   m_pqxxResultSchemas = pqxxXaction.exec(strQuery.latin1());
 	   KPGSchema *pSchema = 0;
 			
 	   for (result::size_type i = 0; i != m_pqxxResultSchemas.size(); ++i)
@@ -301,7 +301,7 @@
         
     try
     {
-        pqxx::result pqxxResultDatabases = pqxxXaction.Exec(strQuery.latin1());
+        pqxx::result pqxxResultDatabases = pqxxXaction.exec(strQuery.latin1());
         
         if(pqxxResultDatabases.size() != 1)
         {
@@ -529,7 +529,7 @@
     {
         work pqxxXaction(*m_pDbConnection, "statistics");
             
-        result pqxxResult = pqxxXaction.Exec(strQuery.latin1());
+        result pqxxResult = pqxxXaction.exec(strQuery.latin1());
     
         if(pqxxResult.size() == 1)
         {
@@ -742,7 +742,7 @@
 	try
 	{
 		pqxx::work pqxxXaction(*m_pDbConnection, "displayExtendedStatistics");
-		pqxx::result pqxxResultStatistics = pqxxXaction.Exec(strQuery.latin1());
+		pqxx::result pqxxResultStatistics = pqxxXaction.exec(strQuery.latin1());
 		
 		// fill properties table with result data
 		int nTableCols = fillPropertiesTable(pqxxResultStatistics, pTableStatistics, false);
@@ -806,7 +806,7 @@
 	try
 	{
 		pqxx::work pqxxXaction(*m_pDbConnection, "displayDiskUsage");
-		pqxx::result pqxxResult = pqxxXaction.Exec(strQuery.latin1());
+		pqxx::result pqxxResult = pqxxXaction.exec(strQuery.latin1());
 		
 		// fill properties table with result data
 		int nTableCols = fillPropertiesTable(pqxxResult, pTableDiskUsage, false);
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgdomainsfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgdomainsfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgdomainsfolder.cpp	2006-12-27 21:35:12.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgdomainsfolder.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -52,7 +52,7 @@
 			
     try
     {
-        m_pqxxResultDomains = pqxxXaction.Exec(strQuery.latin1());
+        m_pqxxResultDomains = pqxxXaction.exec(strQuery.latin1());
     
         KPGDomain *pDomain = 0;
             
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgfunction.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgfunction.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgfunction.cpp	2006-12-27 22:11:20.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgfunction.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -148,7 +148,7 @@
         	
         	try
         	{
-        		pqxx::result pqxxResult = pqxxXaction.Exec(strQuery.latin1());
+        		pqxx::result pqxxResult = pqxxXaction.exec(strQuery.latin1());
 	
 				if(pqxxResult.size() != 1)
 				{
@@ -340,7 +340,7 @@
 		
 	try
 	{
-		pqxx::result pqxxResult = pqxxXaction.Exec(strQuery.latin1());
+		pqxx::result pqxxResult = pqxxXaction.exec(strQuery.latin1());
 	
 		if(pqxxResult.size() != 1)
 		{
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgfunctionsfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgfunctionsfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgfunctionsfolder.cpp	2006-12-27 22:11:20.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgfunctionsfolder.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -70,7 +70,7 @@
 			
   try
   {
-    	m_pqxxResultFunctions = pqxxXaction.Exec(strQuery.latin1());
+    	m_pqxxResultFunctions = pqxxXaction.exec(strQuery.latin1());
 
 		KPGFunction *pFunction = 0;
 		
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpggroupsfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpggroupsfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpggroupsfolder.cpp	2006-12-27 22:11:21.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpggroupsfolder.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -46,7 +46,7 @@
                 
     try
     {
-        m_pqxxResultGroups = pqxxXaction.Exec(strQuery.latin1());
+        m_pqxxResultGroups = pqxxXaction.exec(strQuery.latin1());
     
         KPGGroup *pGroup = 0;
             
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpglanguage.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpglanguage.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpglanguage.cpp	2006-12-27 22:11:21.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpglanguage.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -130,7 +130,7 @@
         
     try
     {
-        pqxx::result pqxxResultLanguages = pqxxXaction.Exec(strQuery.latin1());
+        pqxx::result pqxxResultLanguages = pqxxXaction.exec(strQuery.latin1());
         
         if(pqxxResultLanguages.size() != 1)
         {
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpglanguagesfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpglanguagesfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpglanguagesfolder.cpp	2006-12-27 22:11:21.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpglanguagesfolder.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -50,7 +50,7 @@
 			
     try
     {
-        m_pqxxResultLanguages = pqxxXaction.Exec(strQuery.latin1());
+        m_pqxxResultLanguages = pqxxXaction.exec(strQuery.latin1());
     
         KPGLanguage *pLanguage = 0;
             
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgoperatorclassesfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgoperatorclassesfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgoperatorclassesfolder.cpp	2006-12-28 10:33:28.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgoperatorclassesfolder.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -54,7 +54,7 @@
 			
     try
     {
-        m_pqxxResultOperatorClasses = pqxxXaction.Exec(strQuery.latin1());
+        m_pqxxResultOperatorClasses = pqxxXaction.exec(strQuery.latin1());
     
         KPGOperatorClass *pOperatorClass = 0;
             
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgoperatorsfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgoperatorsfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgoperatorsfolder.cpp	2006-12-28 10:33:28.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgoperatorsfolder.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -64,7 +64,7 @@
 			
     try
     {
-        m_pqxxResultOperators = pqxxXaction.Exec(strQuery.latin1());
+        m_pqxxResultOperators = pqxxXaction.exec(strQuery.latin1());
     
         KPGOperator *pOperator = 0;
             
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgrule.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgrule.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgrule.cpp	2006-12-28 11:39:51.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgrule.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -61,7 +61,7 @@
 	
 	try
 	{
-    	pqxx::result pqxxResultRules = pqxxXaction.Exec(strQuery.latin1());
+    	pqxx::result pqxxResultRules = pqxxXaction.exec(strQuery.latin1());
 
 		if(pqxxResultRules.size() != 1)
 		{
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgschema.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgschema.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgschema.cpp	2006-12-28 11:39:51.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgschema.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -171,7 +171,7 @@
     {
         work pqxxXaction(*pDatabase->connection(), "refreshItem");   
         
-        pqxx::result pqxxResultSchemas = pqxxXaction.Exec(strQuery.latin1());
+        pqxx::result pqxxResultSchemas = pqxxXaction.exec(strQuery.latin1());
     
         if(pqxxResultSchemas.size() != 1)
         {
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgsequence.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgsequence.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgsequence.cpp	2006-12-28 11:39:51.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgsequence.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -192,7 +192,7 @@
     {
         KPGConnection *pDbConnection = connection();
         pqxx::work pqxxXaction(*pDbConnection, "refresh sequence statistics");
-        pqxx::result pqxxResultStatistics = pqxxXaction.Exec(strQuery.latin1());
+        pqxx::result pqxxResultStatistics = pqxxXaction.exec(strQuery.latin1());
     
         if(pqxxResultStatistics.size() != 1)
         {
@@ -257,7 +257,7 @@
         
     try
     {
-        pqxx::result pqxxResultSequences = pqxxXaction.Exec(strQuery.latin1());
+        pqxx::result pqxxResultSequences = pqxxXaction.exec(strQuery.latin1());
     
         if(pqxxResultSequences.size() != 1)
         {
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgsequencesfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgsequencesfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgsequencesfolder.cpp	2006-12-28 11:39:51.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgsequencesfolder.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -67,7 +67,7 @@
 			
     try
     {
-        m_pqxxResultSequences = pqxxXaction.Exec(strQuery.latin1());
+        m_pqxxResultSequences = pqxxXaction.exec(strQuery.latin1());
     
         KPGSequence *pSequence = 0;
             
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgserver.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgserver.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgserver.cpp	2006-12-22 00:28:48.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgserver.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -147,7 +147,7 @@
 	pqxx::result pqxxResult; 
     try
     {
-        pqxxResult = pqxxXaction.Exec(strQuery.latin1());
+        pqxxResult = pqxxXaction.exec(strQuery.latin1());
     }
     catch (const std::exception &e)
     {
@@ -214,7 +214,7 @@
 	pqxx::result pqxxResult; 
     try
     {
-        pqxxResult = pqxxXaction.Exec(strQuery.latin1());
+        pqxxResult = pqxxXaction.exec(strQuery.latin1());
     }
     catch (const std::exception &e)
     {
@@ -242,7 +242,7 @@
 	pqxx::result pqxxResult; 
     try
     {
-        pqxxResult = pqxxXaction.Exec(strQuery.latin1());
+        pqxxResult = pqxxXaction.exec(strQuery.latin1());
     }
     catch (const std::exception &e)
     {
@@ -270,7 +270,7 @@
 	pqxx::result pqxxResult; 
     try
     {
-        pqxxResult = pqxxXaction.Exec(strQuery.latin1());
+        pqxxResult = pqxxXaction.exec(strQuery.latin1());
     }
     catch (const std::exception &e)
     {
@@ -336,7 +336,7 @@
         
     try
     {
-        m_pqxxResultDatabases = pqxxXaction.Exec(strQuery.latin1());
+        m_pqxxResultDatabases = pqxxXaction.exec(strQuery.latin1());
     }
     catch (const std::exception &e)
     {
@@ -515,7 +515,7 @@
 	try
 	{
 		pqxx::work pqxxXaction(*m_pDbConnection, "refresh server activity");
-		pqxx::result pqxxResultActivity = pqxxXaction.Exec(strQuery.latin1());
+		pqxx::result pqxxResultActivity = pqxxXaction.exec(strQuery.latin1());
 		
 		// fill properties table with result data
 		int nTableCols = fillPropertiesTable(pqxxResultActivity, pTableActivity);
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgtablecolumn.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgtablecolumn.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgtablecolumn.cpp	2006-12-28 11:39:51.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgtablecolumn.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -75,7 +75,7 @@
 	
 	try
 	{
-    	pqxx::result pqxxResultColumns = pqxxXaction.Exec(strQuery.latin1());
+    	pqxx::result pqxxResultColumns = pqxxXaction.exec(strQuery.latin1());
 
 		if(pqxxResultColumns.size() != 1)
 		{
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgtableconstraint.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgtableconstraint.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgtableconstraint.cpp	2006-12-28 15:03:30.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgtableconstraint.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -58,7 +58,7 @@
 	
 	try
 	{
-    	pqxx::result pqxxResultConstraints = pqxxXaction.Exec(strQuery.latin1());
+    	pqxx::result pqxxResultConstraints = pqxxXaction.exec(strQuery.latin1());
 
 		if(pqxxResultConstraints.size() != 1)
 		{
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgtable.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgtable.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgtable.cpp	2006-12-28 19:51:34.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgtable.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -171,7 +171,7 @@
 		
 	try
 	{
-		pqxx::result pqxxResultTables = pqxxXaction.Exec(strQuery.latin1());
+		pqxx::result pqxxResultTables = pqxxXaction.exec(strQuery.latin1());
 	
 		if(pqxxResultTables.size() != 1)
 		{
@@ -564,7 +564,7 @@
     {
         KPGConnection *pDbConnection = connection();
         pqxx::work pqxxXaction(*pDbConnection, "refresh table statistics");
-        pqxx::result pqxxResultStatistics = pqxxXaction.Exec(strQuery.latin1());
+        pqxx::result pqxxResultStatistics = pqxxXaction.exec(strQuery.latin1());
     
         if(pqxxResultStatistics.size() != 1)
         {
@@ -654,7 +654,7 @@
     {
         KPGConnection *pDbConnection = connection();
         pqxx::work pqxxXaction(*pDbConnection, "refresh table statistics");
-        pqxx::result pqxxResultStatistics = pqxxXaction.Exec(strQuery.latin1());
+        pqxx::result pqxxResultStatistics = pqxxXaction.exec(strQuery.latin1());
     
         if(pqxxResultStatistics.size() != 1) // querying pq_xactloct return no rows... Why ?
         {
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgtableindex.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgtableindex.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgtableindex.cpp	2006-12-22 20:27:08.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgtableindex.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -95,7 +95,7 @@
 	
 	try
 	{
-    	pqxx::result pqxxResultIndexes = pqxxXaction.Exec(strQuery.latin1());
+    	pqxx::result pqxxResultIndexes = pqxxXaction.exec(strQuery.latin1());
 
 		if(pqxxResultIndexes.size() != 1)
 		{
@@ -203,7 +203,7 @@
 	{
 		KPGConnection *pDbConnection = connection();
 		pqxx::work pqxxXaction(*pDbConnection, "refresh table statistics");
-		pqxx::result pqxxResultStatistics = pqxxXaction.Exec(strQuery.latin1());
+		pqxx::result pqxxResultStatistics = pqxxXaction.exec(strQuery.latin1());
 	
 		if(pqxxResultStatistics.size() != 1)
 		{
@@ -264,7 +264,7 @@
 	{
 		KPGConnection *pDbConnection = connection();
 		pqxx::work pqxxXaction(*pDbConnection, "refresh table statistics");
-		pqxx::result pqxxResultStatistics = pqxxXaction.Exec(strQuery.latin1());
+		pqxx::result pqxxResultStatistics = pqxxXaction.exec(strQuery.latin1());
 	
 		if(pqxxResultStatistics.size() != 1)
 		{
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgtablesfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgtablesfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgtablesfolder.cpp	2006-12-28 19:51:34.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgtablesfolder.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -95,7 +95,7 @@
 		
   	try
   	{
-    	m_pqxxResultTables = pqxxXaction.Exec(strQuery.latin1());
+    	m_pqxxResultTables = pqxxXaction.exec(strQuery.latin1());
 
 		KPGTable *pTable = 0;
 		
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgtablespace.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgtablespace.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgtablespace.cpp	2006-12-21 22:36:14.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgtablespace.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -167,7 +167,7 @@
 		
 	try
 	{
-		pqxx::result pqxxResultTablespaces = pqxxXaction.Exec(strQuery.latin1());
+		pqxx::result pqxxResultTablespaces = pqxxXaction.exec(strQuery.latin1());
 	
 		setProperties(pqxxResultTablespaces[0], bVersion81_OrNewer);		
 	}
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgtablespacesfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgtablespacesfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgtablespacesfolder.cpp	2006-12-22 20:27:08.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgtablespacesfolder.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -57,7 +57,7 @@
 			
 	try
 	{
-		m_pqxxResultTablespaces = pqxxXaction.Exec(strQuery.latin1());
+		m_pqxxResultTablespaces = pqxxXaction.exec(strQuery.latin1());
 	
 		KPGTablespace *pTablespace = 0;
 			
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgtrigger.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgtrigger.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgtrigger.cpp	2006-12-28 15:03:30.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgtrigger.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -64,7 +64,7 @@
 	
 	try
 	{
-    	pqxx::result pqxxResultTriggers = pqxxXaction.Exec(strQuery.latin1());
+    	pqxx::result pqxxResultTriggers = pqxxXaction.exec(strQuery.latin1());
 
 		if(pqxxResultTriggers.size() != 1)
 		{
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgtype.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgtype.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgtype.cpp	2006-12-28 15:03:30.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgtype.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -240,7 +240,7 @@
 				
 	try
 	{
-    	pqxx::result pqxxResultTypes = pqxxXaction.Exec(strQuery.latin1());
+    	pqxx::result pqxxResultTypes = pqxxXaction.exec(strQuery.latin1());
 
 		if(pqxxResultTypes.size() != 1)
 		{
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgtypesfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgtypesfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgtypesfolder.cpp	2006-12-28 15:03:30.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgtypesfolder.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -52,7 +52,7 @@
 				
 	try
 	{
-		m_pqxxResultTypes = pqxxXaction.Exec(strQuery.latin1());
+		m_pqxxResultTypes = pqxxXaction.exec(strQuery.latin1());
 	
 		KPGType *pType = 0;
 			
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgusersfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgusersfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgusersfolder.cpp	2006-12-28 15:03:30.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgusersfolder.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -48,7 +48,7 @@
                 
     try
     {
-        m_pqxxResultUsers = pqxxXaction.Exec(strQuery.latin1());
+        m_pqxxResultUsers = pqxxXaction.exec(strQuery.latin1());
     
         KPGUser *pUser = 0;
             
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgview.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgview.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgview.cpp	2006-08-20 14:18:28.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgview.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -127,7 +127,7 @@
 		
 	try
 	{
-		pqxx::result pqxxResultViews = pqxxXaction.Exec(strQuery.latin1());
+		pqxx::result pqxxResultViews = pqxxXaction.exec(strQuery.latin1());
 	
 		if(pqxxResultViews.size() != 1)
 		{
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgviewsfolder.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgviewsfolder.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgviewsfolder.cpp	2006-08-20 14:18:28.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgviewsfolder.cpp	2007-02-02 05:19:16.000000000 +0100
@@ -57,7 +57,7 @@
 		
   try
   {
-    m_pqxxResultViews = pqxxXaction.Exec(strQuery.latin1());
+    m_pqxxResultViews = pqxxXaction.exec(strQuery.latin1());
 
 		KPGView *pgView = 0;
 		
