diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgconnection.cpp kpogre-1.5.4-p/kpogre/DbObjects/kpgconnection.cpp
--- kpogre-1.5.4/kpogre/DbObjects/kpgconnection.cpp	2006-09-12 14:07:52.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgconnection.cpp	2007-02-02 04:16:58.000000000 +0100
@@ -19,8 +19,8 @@
 
 
 // include files for libpgxx 
-#include <pqxx/nontransaction.h> 
-#include <pqxx/robusttransaction.h> 
+#include <pqxx/nontransaction> 
+#include <pqxx/robusttransaction> 
  
 // include files for Qt
 #include <qstring.h>
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgconnection.h kpogre-1.5.4-p/kpogre/DbObjects/kpgconnection.h
--- kpogre-1.5.4/kpogre/DbObjects/kpgconnection.h	2006-09-22 12:10:05.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgconnection.h	2007-02-02 04:16:58.000000000 +0100
@@ -22,7 +22,7 @@
 #define QT_NO_ASCII_CAST
 
 // include files for libpgxx 
-#include <pqxx/connection.h>
+#include <pqxx/connection>
 
 #include "kpgfunction.h"
 
diff -Naur kpogre-1.5.4/kpogre/DbObjects/kpgtreeitem.h kpogre-1.5.4-p/kpogre/DbObjects/kpgtreeitem.h
--- kpogre-1.5.4/kpogre/DbObjects/kpgtreeitem.h	2006-12-28 16:11:29.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/DbObjects/kpgtreeitem.h	2007-02-02 04:16:58.000000000 +0100
@@ -24,8 +24,8 @@
 #include <qlistview.h>
 
 // include files for libpgxx 
-#include <pqxx/transaction.h> 
-#include <pqxx/result.h> 
+#include <pqxx/transaction> 
+#include <pqxx/result> 
 
 class KPGConnection;
 class QTable;
diff -Naur kpogre-1.5.4/kpogre/kpglinklabel.h kpogre-1.5.4-p/kpogre/kpglinklabel.h
--- kpogre-1.5.4/kpogre/kpglinklabel.h	2006-09-15 14:44:09.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/kpglinklabel.h	2007-02-02 04:16:58.000000000 +0100
@@ -13,7 +13,7 @@
 #define KPGLINKLABEL_H
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 #include <qlabel.h>
 //#include <kactivelabel.h>
diff -Naur kpogre-1.5.4/kpogre/kpogreview.cpp kpogre-1.5.4-p/kpogre/kpogreview.cpp
--- kpogre-1.5.4/kpogre/kpogreview.cpp	2006-12-28 20:27:31.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/kpogreview.cpp	2007-02-02 04:16:58.000000000 +0100
@@ -15,7 +15,7 @@
  *                                                                         *
  ***************************************************************************/
 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 // include files for Qt
 #include <qtextcodec.h>
diff -Naur kpogre-1.5.4/kpogre/kpogreview.h kpogre-1.5.4-p/kpogre/kpogreview.h
--- kpogre-1.5.4/kpogre/kpogreview.h	2006-12-14 10:31:58.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/kpogreview.h	2007-02-02 04:16:58.000000000 +0100
@@ -23,7 +23,7 @@
 #endif
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 // include files for Qt
 #include <qwidget.h>
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgaggregateproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgaggregateproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgaggregateproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgaggregateproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 Aggregate properties
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgcastproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgcastproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgcastproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgcastproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 Cast properties
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgcheckconstrproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgcheckconstrproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgcheckconstrproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgcheckconstrproppage.h	2007-02-02 04:16:59.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 Check Constraint properties
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgconversionproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgconversionproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgconversionproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgconversionproppage.h	2007-02-02 04:16:59.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 Conversion properties
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgdatabaseproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgdatabaseproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgdatabaseproppage.h	2006-12-26 23:06:48.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgdatabaseproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -19,7 +19,7 @@
 
 #include "kpgdatabaseproppagebase.h"
 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 // application specific includes
 #include "kpgabstractpropertypage.h"
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgdomainconstrproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgdomainconstrproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgdomainconstrproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgdomainconstrproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 Domain check constraint properties
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgdomainproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgdomainproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgdomainproppage.h	2006-09-18 14:03:22.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgdomainproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 Domain properties
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgforeignkeyconstrproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgforeignkeyconstrproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgforeignkeyconstrproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgforeignkeyconstrproppage.h	2007-02-02 04:16:59.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 Foreign Key Constraint properties
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgfunctionproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgfunctionproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgfunctionproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgfunctionproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -15,7 +15,7 @@
 #include "kpgfunctionproppagebase.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 // application specific includes
 #include "kpgabstractpropertypage.h"
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpggroupproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpggroupproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpggroupproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpggroupproppage.h	2007-02-02 04:16:59.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 User properties
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgindexproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgindexproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgindexproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgindexproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 class KPGSyntaxHighlighter;
 class KPGIndex;
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpglanguageproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpglanguageproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpglanguageproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpglanguageproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -15,7 +15,7 @@
 #include "kpglanguageproppagebase.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 // application specific includes
 #include "kpgabstractpropertypage.h"
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgoperatorclassproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgoperatorclassproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgoperatorclassproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgoperatorclassproppage.h	2007-02-02 04:16:59.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 Operator Class properties
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgoperatorproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgoperatorproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgoperatorproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgoperatorproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 Operator properties
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgprimarykeyconstrproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgprimarykeyconstrproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgprimarykeyconstrproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgprimarykeyconstrproppage.h	2007-02-02 04:16:59.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 Primary Key Constraint properties
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgprimarykeyproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgprimarykeyproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgprimarykeyproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgprimarykeyproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 class KPGSyntaxHighlighter;
 class KPGPrimaryKey;
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgpropertypage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgpropertypage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgpropertypage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgpropertypage.h	2007-02-02 04:16:58.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 
 /**
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgschemaproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgschemaproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgschemaproppage.h	2006-09-15 14:44:09.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgschemaproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -15,7 +15,7 @@
 #include "kpgschemaproppagebase.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 // application specific includes
 #include "kpgabstractpropertypage.h"
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgsequenceproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgsequenceproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgsequenceproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgsequenceproppage.h	2007-02-02 04:16:59.000000000 +0100
@@ -15,7 +15,7 @@
 #include "kpgsequenceproppagebase.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 // application specific includes
 #include "kpgabstractpropertypage.h"
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgserverproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgserverproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgserverproppage.h	2006-12-21 10:47:27.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgserverproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -23,7 +23,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 class KPGServer;
 
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgtablecolumnproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgtablecolumnproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgtablecolumnproppage.h	2006-09-15 13:24:30.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgtablecolumnproppage.h	2007-02-02 04:16:59.000000000 +0100
@@ -23,7 +23,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 Table column property page
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgtableproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgtableproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgtableproppage.h	2006-09-15 14:44:09.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgtableproppage.h	2007-02-02 04:16:59.000000000 +0100
@@ -20,7 +20,7 @@
 #include "kpgtableproppagebase.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 // application specific includes
 #include "kpgabstractpropertypage.h"
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgtablespaceproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgtablespaceproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgtablespaceproppage.h	2006-09-15 13:24:31.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgtablespaceproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -15,7 +15,7 @@
 #include "kpgtablespaceproppagebase.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 // application specific includes
 #include "kpgabstractpropertypage.h"
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgtriggerproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgtriggerproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgtriggerproppage.h	2006-09-15 13:24:31.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgtriggerproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 Table trigger property page
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgtypeproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgtypeproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgtypeproppage.h	2006-09-15 13:24:31.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgtypeproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 Data type properties
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpguniqueconstrproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpguniqueconstrproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpguniqueconstrproppage.h	2006-09-15 13:24:31.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpguniqueconstrproppage.h	2007-02-02 04:16:59.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 Unique Constraint properties
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpguserproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpguserproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpguserproppage.h	2006-09-15 13:24:31.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpguserproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -18,7 +18,7 @@
 #include "kpgabstractpropertypage.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 /**
 User properties
diff -Naur kpogre-1.5.4/kpogre/PropertyPages/kpgviewproppage.h kpogre-1.5.4-p/kpogre/PropertyPages/kpgviewproppage.h
--- kpogre-1.5.4/kpogre/PropertyPages/kpgviewproppage.h	2006-09-15 13:24:31.000000000 +0200
+++ kpogre-1.5.4-p/kpogre/PropertyPages/kpgviewproppage.h	2007-02-02 04:16:58.000000000 +0100
@@ -15,7 +15,7 @@
 #include "kpgviewproppagebase.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
+#include <pqxx/util> 
 
 // application specific includes
 #include "kpgabstractpropertypage.h"
diff -Naur kpogre-1.5.4/kpogre/Wizards/kpgmodifyviewwidget1.h kpogre-1.5.4-p/kpogre/Wizards/kpgmodifyviewwidget1.h
--- kpogre-1.5.4/kpogre/Wizards/kpgmodifyviewwidget1.h	2004-12-19 22:01:11.000000000 +0100
+++ kpogre-1.5.4-p/kpogre/Wizards/kpgmodifyviewwidget1.h	2007-02-02 04:16:59.000000000 +0100
@@ -15,8 +15,8 @@
 #include "kpgmodifyviewwidget1base.h"
 
 // include files for libpgxx 
-#include <pqxx/util.h> 
-#include <pqxx/transaction.h> 
+#include <pqxx/util> 
+#include <pqxx/transaction> 
 
 class KPGDatabase;
 class KPGSyntaxHighlighter;
