diff -ruN rlog-1.4.orig/rlog/common.h rlog-1.4/rlog/common.h
--- rlog-1.4.orig/rlog/common.h	2008-06-15 07:49:52.000000000 +0200
+++ rlog-1.4/rlog/common.h	2008-06-23 19:29:16.000000000 +0200
@@ -52,7 +52,6 @@
 
 # define PRINTF(FMT,X) __attribute__ (( __format__ ( __printf__, FMT, X)))
 # define HAVE_PRINTF_ATTR 1
-# define RLOG_SECTION __attribute__ (( section("RLOG_DATA") ))
 
 #if __GNUC__ >= 3
 # define expect(foo, bar) __builtin_expect((foo),bar)
@@ -77,7 +76,6 @@
 */
 # define PRINTF(FMT,X)
 # define HAVE_PRINTF_ATTR 0
-# define RLOG_SECTION 
 
 /*!
 */
diff -ruN rlog-1.4.orig/rlog/common.h.in rlog-1.4/rlog/common.h.in
--- rlog-1.4.orig/rlog/common.h.in	2008-06-15 05:10:13.000000000 +0200
+++ rlog-1.4/rlog/common.h.in	2008-06-23 19:28:28.000000000 +0200
@@ -52,7 +52,6 @@
 
 # define PRINTF(FMT,X) __attribute__ (( __format__ ( __printf__, FMT, X)))
 # define HAVE_PRINTF_ATTR 1
-# define RLOG_SECTION __attribute__ (( section("RLOG_DATA") ))
 
 #if __GNUC__ >= 3
 # define expect(foo, bar) __builtin_expect((foo),bar)
@@ -77,7 +76,6 @@
 */
 # define PRINTF(FMT,X)
 # define HAVE_PRINTF_ATTR 0
-# define RLOG_SECTION 
 
 /*!
 */
diff -ruN rlog-1.4.orig/rlog/rlog-c99.h rlog-1.4/rlog/rlog-c99.h
--- rlog-1.4.orig/rlog/rlog-c99.h	2008-06-15 05:56:57.000000000 +0200
+++ rlog-1.4/rlog/rlog-c99.h	2008-06-23 19:28:37.000000000 +0200
@@ -22,7 +22,7 @@
   @internal
 */
 #define _rMessageDef(ID, COMPONENT) \
-  static rlog::PublishLoc ID RLOG_SECTION = {& ID ## _enabled, \
+  static rlog::PublishLoc ID = {& ID ## _enabled, \
       &rlog::RLog_Register, 0, STR(COMPONENT), __FILE__, \
       __FUNCTION__, __LINE__, 0};
 
diff -ruN rlog-1.4.orig/rlog/rlog-novariadic.h rlog-1.4/rlog/rlog-novariadic.h
--- rlog-1.4.orig/rlog/rlog-novariadic.h	2008-06-15 05:10:13.000000000 +0200
+++ rlog-1.4/rlog/rlog-novariadic.h	2008-06-23 19:28:45.000000000 +0200
@@ -35,7 +35,7 @@
 
 #define _rMessageDef(ID, COMPONENT) \
   static bool ID ## _enabled = true; \
-  static rlog::PublishLoc ID RLOG_SECTION = {& ID ## _enabled, \
+  static rlog::PublishLoc ID = {& ID ## _enabled, \
       &rlog::RLog_Register, 0, STR(COMPONENT), __FILE__, \
       __FUNCTION__, __LINE__, 0}
 
diff -ruN rlog-1.4.orig/rlog/rlog-prec99.h rlog-1.4/rlog/rlog-prec99.h
--- rlog-1.4.orig/rlog/rlog-prec99.h	2008-06-15 05:59:05.000000000 +0200
+++ rlog-1.4/rlog/rlog-prec99.h	2008-06-23 19:28:06.000000000 +0200
@@ -18,7 +18,7 @@
 
 #define _rMessageDef(ID, COMPONENT) \
   static bool ID ## _enabled = true; \
-  static rlog::PublishLoc ID RLOG_SECTION = {& ID ## _enabled, \
+  static rlog::PublishLoc ID = {& ID ## _enabled, \
       &rlog::RLog_Register, 0, STR(COMPONENT), __FILE__, \
       __FUNCTION__, __LINE__, 0};
 
