--- dbus-0.61/mono/AssemblyInfo.cs.in.old-version	2006-02-24 18:04:34.000000000 -0500
+++ dbus-0.61/mono/AssemblyInfo.cs.in	2006-02-24 18:04:47.000000000 -0500
@@ -1,6 +1,6 @@
 using System.Reflection;
 using System.Runtime.CompilerServices;
  
-[assembly:AssemblyVersion("0.60")]
+[assembly:AssemblyVersion("0.61")]
 [assembly:AssemblyDelaySign(false)]
 [assembly:AssemblyKeyFile("@srcdir@/dbus-sharp.snk")]
