From d85421ec62cdcd0b4b1f162b9908dc3eb0b704b4 Mon Sep 17 00:00:00 2001
From: Pierre Schmitz <pierre@archlinux.de>
Date: Thu, 18 Feb 2010 04:18:10 +0000
Subject: contrib/*_completion: match *.pkg.tar.*

Signed-off-by: Pierre Schmitz <pierre@archlinux.de>
Signed-off-by: Dan McGee <dan@archlinux.org>
---
diff --git a/contrib/bash_completion b/contrib/bash_completion
index 62e5bc9..a231eb6 100644
--- a/contrib/bash_completion
+++ b/contrib/bash_completion
@@ -321,7 +321,7 @@ _pacman ()
     case "${op}" in
       U)
         COMPREPLY=( $( compgen -d -- "$cur" ) \
-                    $( compgen -f -X '!*.pkg.tar.gz' -- "$cur" ) )
+                    $( compgen -f -X '!*.pkg.tar.*' -- "$cur" ) )
         return 0
         ;;
       h|V)
@@ -336,7 +336,7 @@ _pacman ()
                       $( compgen -f -- "$cur" ) )
         elif _instring $mod p; then
           COMPREPLY=( $( compgen -d -- "$cur" ) \
-                      $( compgen -f -X '!*.pkg.tar.gz' -- "$cur" ) )
+                      $( compgen -f -X '!*.pkg.tar.*' -- "$cur" ) )
         elif _instring $mod u; then
           COMPREPLY=''
           return 0
diff --git a/contrib/zsh_completion b/contrib/zsh_completion
index 2f43d9b..01b3c44 100644
--- a/contrib/zsh_completion
+++ b/contrib/zsh_completion
@@ -32,7 +32,7 @@ _pacman_opts_common=(
 _pacman_opts_pkgfile=(
 	'-d[Skip dependency checks]'
 	'-f[Overwrite conflicting files]'
-	'*:package file:_files -g "*.pkg.tar.gz(.)"'
+	'*:package file:_files -g "*.pkg.tar.*(.)"'
 )
 
 # options for passing to _arguments: subactions for --query command
@@ -120,7 +120,7 @@ _pacman_action_query() {
 			_arguments -s : \
 				"$_pacman_opts_common[@]" \
 				"$_pacman_opts_query_modifiers[@]" \
-				'*:package file:_files -g "*.pkg.tar.gz"'
+				'*:package file:_files -g "*.pkg.tar.*"'
 				;;
 		query_group)
 			_arguments -s : \
@@ -295,11 +295,11 @@ _pacman() {
 				"$_pacman_opts_query_modifiers[@]" \
 				'*:package file:_files'
 				;;
-		-Q*p*) # file *.pkg.tar.gz
+		-Q*p*) # file *.pkg.tar.*
 			_arguments -s : \
 				"$_pacman_opts_common[@]" \
 				"$_pacman_opts_query_modifiers[@]" \
-				'*:package file:_files -g "*.pkg.tar.gz"'
+				'*:package file:_files -g "*.pkg.tar.*"'
 				;;
 		-Q*)  _pacman_action_query    ;;
 		-R*)  _pacman_action_remove   ;;
--
cgit v0.8.3.1-3-ga650
