diff -urN subversion-1.6.9/subversion/bindings/swig/python/svn/core.py subversion-1.6.9-fixed/subversion/bindings/swig/python/svn/core.py
--- subversion-1.6.9/subversion/bindings/swig/python/svn/core.py	2009-02-13 11:22:26.000000000 -0500
+++ subversion-1.6.9-fixed/subversion/bindings/swig/python/svn/core.py	2010-02-08 07:46:29.000000000 -0500
@@ -19,6 +19,7 @@
 from libsvn.core import *
 import libsvn.core as _libsvncore
 import atexit as _atexit
+import warnings
 
 class SubversionException(Exception):
   def __init__(self, message=None, apr_err=None, child=None,
@@ -44,7 +45,9 @@
     Exception.__init__(self, *args)
 
     self.apr_err = apr_err
-    self.message = message
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", DeprecationWarning)
+        self.message = message
     self.child = child
     self.file = file
     self.line = line
