--- webboard-0.2/src/webboard	2006-02-09 10:01:34.000000000 -0800
+++ webboard-0.2-new/src/webboard	2006-05-28 01:34:10.000000000 -0700
@@ -27,8 +27,8 @@
     sys.path.insert(0, "")
 
 def main():
-    history = WebBoardHistory()
     config = WebBoardConfig()
+    history = WebBoardHistory(config=config)
     wb = WebBoard(config, history)
     config.add_notifier(wb.on_config_changed)
     del wb
