--- src/lxappearance.c	2011-07-29 16:50:19.000000000 +0200
+++ src/lxappearance1.c	2011-07-30 18:51:40.323368701 +0200
@@ -153,21 +153,33 @@
 
     char* file_path = g_build_filename(g_get_home_dir(), ".gtkrc-2.0", NULL);
     GString* content = g_string_sized_new(512);
+    GString* content3 = g_string_sized_new(512);
     g_string_append(content,
         "# DO NOT EDIT! This file will be overwritten by LXAppearance.\n"
         "# Any customization should be done in ~/.gtkrc-2.0.mine instead.\n\n");
+    g_string_append(content3, "[Settings]\n"
+        "# DO NOT EDIT! This file will be overwritten by LXAppearance.\n");
+    
     if(app.widget_theme)
         g_string_append_printf(content,
             "gtk-theme-name=\"%s\"\n", app.widget_theme);
+        g_string_append_printf(content3,
+            "gtk-theme-name = %s\n", app.widget_theme);
     if(app.icon_theme)
         g_string_append_printf(content,
-            "gtk-icon-theme-name=\"%s\"\n", app.icon_theme);
+            "gtk-icon-theme-name=\"%s\"\n", app.icon_theme);        
+        g_string_append_printf(content3,
+            "gtk-icon-theme-name = %s\n", app.icon_theme);
     if(app.default_font)
         g_string_append_printf(content,
             "gtk-font-name=\"%s\"\n", app.default_font);
+        g_string_append_printf(content3,
+            "gtk-font-name = %s\n", app.default_font);
     if(app.cursor_theme)
         g_string_append_printf(content,
             "gtk-cursor-theme-name=\"%s\"\n", app.cursor_theme);
+        g_string_append_printf(content3,
+            "gtk-cursor-theme-name = %s\n", app.cursor_theme);
     save_cursor_theme_name();
 
     g_string_append_printf(content,
@@ -196,20 +208,46 @@
         app.enable_hinting ? 1 : 0
         );
 
+   g_string_append_printf(content3,
+        "gtk-cursor-theme-size = %d\n"
+        "gtk-toolbar-style = %s\n"
+        "gtk-toolbar-icon-size = %s\n"
+        "gtk-button-images = %d\n"
+        "gtk-menu-images = %d\n"
+        "gtk-enable-event-sounds = %d\n"
+        "gtk-enable-input-feedback-sounds = %d\n"
+        "gtk-xft-antialias = %d\n"
+        "gtk-xft-hinting = %d\n",
+        app.cursor_theme_size,
+        tb_styles[app.toolbar_style],
+        tb_icon_sizes[app.toolbar_icon_size],
+        app.button_images ? 1 : 0,
+        app.menu_images ? 1 : 0,
+        app.enable_event_sound ? 1 : 0,
+        app.enable_input_feedback ? 1 : 0,
+        app.enable_antialising ? 1 : 0,
+        app.enable_hinting ? 1 : 0
+        );
+
     if(app.hinting_style)
         g_string_append_printf(content,
             "gtk-xft-hintstyle=\"%s\"\n", app.hinting_style);
+        g_string_append_printf(content3,
+            "gtk-xft-hintstyle = %s\n", app.hinting_style);
 
     if(app.font_rgba)
         g_string_append_printf(content,
             "gtk-xft-rgba=\"%s\"\n", app.font_rgba);
+        g_string_append_printf(content3,
+            "gtk-xft-rgba = %s\n", app.font_rgba);
 
     if(app.color_scheme)
     {
         char* escaped = g_strescape(app.color_scheme, NULL);
         g_string_append_printf(content,
-            "gtk-color-scheme=\"%s\"\n",
-            escaped);
+            "gtk-color-scheme=\"%s\"\n", escaped);
+        g_string_append_printf(content3,
+            "gtk-color-scheme = %s\n", escaped);
         g_free(escaped);
     }
 
@@ -218,20 +256,20 @@
         g_get_home_dir());
 
     g_file_set_contents(file_path, content->str, content->len, NULL);
+    g_string_free(content, TRUE);
 
     /* Save also for GTK3 */
-    g_string_prepend(content, "[Settings] \n");
-    char* file_path_gtk3 = g_build_filename(g_get_home_dir(), "gtk-3.0", NULL);
-    char* file_path_settings = g_build_filename(g_get_home_dir(), "gtk-3.0", "settings.ini", NULL);
+    char* file_path_gtk3 = g_build_filename(g_get_user_config_dir(), "gtk-3.0", NULL);
+    char* file_path_settings = g_build_filename(g_get_user_config_dir(), "gtk-3.0", "settings.ini", NULL);
 
     if (!g_file_test(file_path_gtk3, G_FILE_TEST_IS_DIR))
     {
         g_mkdir_with_parents(file_path_gtk3, 0755);
     }
 
-    g_file_set_contents(file_path_settings, content->str, content->len, NULL);
+    g_file_set_contents(file_path_settings, content3->str, content3->len, NULL);
 
-    g_string_free(content, TRUE);
+    g_string_free(content3, TRUE);
     g_free(file_path);
 }
