--- src/matos.c	2005-05-30 14:51:12.000000000 -0700
+++ src/matos.c.new	2007-06-11 03:14:11.000000000 -0700
@@ -171,7 +171,10 @@
   gchar Liso[MAXPATHLEN]="";
   const gchar *Lptriso;
 _DEB("DEBUT COPIE !!");
-
+  
+  /*fix failed burns with CD/DVD combo writers */
+  gint Lmedia;
+  
   if (!Lsource && !Lgraveur) {
     g_set_error(Aerror, GRAVEMAN_ERROR, _ERR_INAPPROPRIATE_MODE, 
       _("Cannot duplicate an iso image to an another iso image with graveman, just copy this file !"));
@@ -205,8 +208,18 @@
     } else {
       sc_grave_set_data(Ag, &Liso, "iso"); /* image iso */
 
-      Lstatus = burn_from_image("copy", Ag, Aerror);
-      Lstatus = burn_iso2dvd(Ag, Aerror);
+      /*fix failed burns with CD/DVD combo writers */
+      Lmedia = get_cdinfo(Ag, Lgraveur, NULL); 
+      
+      if ((Lmedia & _MEDIA_CDRW) || (Lmedia & _MEDIA_CDR)) { 
+        Lstatus = burn_from_image("copy", Ag, Aerror); //original line 
+      } 
+      
+      else if (Lmedia & _MEDIA_DVD) { 
+        Lstatus = burn_iso2dvd(Ag, Aerror); //original line 
+      }
+      /*end fix */
+
       sc_grave_del_data(Ag, "iso");
     }
   } else if (!Lgraveur) {
