From bc83c1571b556c8bb906f34d7f3cbd28f95cd275 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 10 Jul 2018 12:50:51 -0600
Subject: [PATCH] remove lto

---
 setup.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/setup.py b/setup.py
index a05c2539..76ecfb1c 100755
--- a/setup.py
+++ b/setup.py
@@ -211,9 +211,6 @@ def init_env(
     cppflags += shlex.split(os.environ.get('CPPFLAGS', ''))
     cflags += shlex.split(os.environ.get('CFLAGS', ''))
     ldflags += shlex.split(os.environ.get('LDFLAGS', ''))
-    if not debug and not sanitize:
-        # See https://github.com/google/sanitizers/issues/647
-        cflags.append('-flto'), ldflags.append('-flto')
 
     if profile:
         cppflags.append('-DWITH_PROFILER')
-- 
2.18.0

