diff --git a/src/gnome-cmd-plain-path.cc b/src/gnome-cmd-plain-path.cc
index 9fb1619..1a3ceb5 100644
--- a/src/gnome-cmd-plain-path.cc
+++ b/src/gnome-cmd-plain-path.cc
@@ -27,7 +27,7 @@
 using namespace std;
 
 
-inline GnomeCmdPath *GnomeCmdPlainPath::get_parent()
+GnomeCmdPath *GnomeCmdPlainPath::get_parent()
 {
     GnomeVFSURI *t = gnome_vfs_uri_new (G_DIR_SEPARATOR_S);
     GnomeVFSURI *u1 = gnome_vfs_uri_append_path (t, path);
@@ -48,7 +48,7 @@ inline GnomeCmdPath *GnomeCmdPlainPath::get_parent()
 }
 
 
-inline GnomeCmdPath *GnomeCmdPlainPath::get_child(const gchar *child)
+GnomeCmdPath *GnomeCmdPlainPath::get_child(const gchar *child)
 {
     GnomeVFSURI *t = gnome_vfs_uri_new (G_DIR_SEPARATOR_S);
     GnomeVFSURI *u1 = gnome_vfs_uri_append_path (t, path);
