diff --git a/src/cmd/go/internal/work/init.go b/src/cmd/go/internal/work/init.go
index 7f894f5..a517887 100644
--- a/src/cmd/go/internal/work/init.go
+++ b/src/cmd/go/internal/work/init.go
@@ -131,7 +131,7 @@ func buildModeInit() {
 		ldBuildmode = "c-shared"
 	case "default":
 		switch cfg.Goos {
-		case "android":
+		case "android", "linux":
 			codegenArg = "-shared"
 			ldBuildmode = "pie"
 		case "darwin":
