diff --git a/jupyterlab/labapp.py b/jupyterlab/labapp.py
index 1eb56b22fe..5a401a0fdf 100644
--- a/jupyterlab/labapp.py
+++ b/jupyterlab/labapp.py
@@ -19,7 +19,7 @@ from jupyterlab_server import (
     WorkspaceImportApp,
     WorkspaceListApp,
 )
-from nbclassic.shim import NBClassicConfigShimMixin
+from notebook_shim.shim import NotebookConfigShimMixin as NBClassicConfigShimMixin
 from traitlets import Bool, Instance, Type, Unicode, default
 
 from ._version import __version__
