diff --git a/jupyterlab/labapp.py b/jupyterlab/labapp.py
index 75af58179c..df2bcdc4cd 100644
--- a/jupyterlab/labapp.py
+++ b/jupyterlab/labapp.py
@@ -14,7 +14,7 @@ from jupyter_server.serverapp import flags
 from jupyter_server.utils import url_path_join as ujoin
 
 from jupyterlab_server import LabServerApp, LicensesApp, WorkspaceExportApp, WorkspaceImportApp, WorkspaceListApp
-from nbclassic.shim import NBClassicConfigShimMixin
+from notebook_shim.shim import NotebookConfigShimMixin as NBClassicConfigShimMixin
 from traitlets import Bool, Instance, Unicode, default
 
 from ._version import __version__
