Upstream commit: https://github.com/PowerDNS/pdns/commit/311f26839819

diff -upr pdns-4.4.1.orig/pdns/Makefile.am pdns-4.4.1/pdns/Makefile.am
--- pdns-4.4.1.orig/pdns/Makefile.am	2021-02-07 00:37:20.000000000 +0200
+++ pdns-4.4.1/pdns/Makefile.am	2021-06-09 00:18:28.264532242 +0300
@@ -6,7 +6,8 @@ AM_CPPFLAGS += \
 	$(LIBEDIT_CFLAGS) \
 	$(LIBCRYPTO_INCLUDES) \
 	$(SYSTEMD_CFLAGS) \
-	$(YAML_CFLAGS)
+	$(YAML_CFLAGS) \
+	-DBOOST_CONTAINER_USE_STD_EXCEPTIONS
 
 AM_CXXFLAGS = \
 	-DSYSCONFDIR=\"$(sysconfdir)\" \
