commit 1bb3ad6dfc885112d05d8d73b066f18a416ca965
Author: Geert Hendrickx <geert@hendrickx.be>
Date:   Thu Aug 27 17:30:03 2020 +0200

    MDEV-23589: Portability: use `uname -n` instead of `hostname`
    
    Closes PR #1663

diff --git a/scripts/CMakeLists.txt b/scripts/CMakeLists.txt
index 33e32ea22f7..eafb051ad60 100644
--- a/scripts/CMakeLists.txt
+++ b/scripts/CMakeLists.txt
@@ -162,7 +162,7 @@ ELSE()
   SET(CHECK_PID "kill -s SIGCONT $PID  > /dev/null 2> /dev/null")
 ENDIF()
 
-SET(HOSTNAME "hostname")
+SET(HOSTNAME "uname -n")
 SET(MYSQLD_USER "mysql")
 ENDIF(UNIX)
 
diff --git a/support-files/CMakeLists.txt b/support-files/CMakeLists.txt
index 80b6ca386df..2a17cf6d0d8 100644
--- a/support-files/CMakeLists.txt
+++ b/support-files/CMakeLists.txt
@@ -30,7 +30,7 @@ ELSE()
   SET(CXXFLAGS ${CMAKE_CXX_FLAGS})
   SET(MYSQLD_USER "mysql")
   SET(ini_file_extension "cnf")
-  SET(HOSTNAME "hostname")
+  SET(HOSTNAME "uname -n")
 ENDIF()
 
 # XXX: shouldn't we just have variables for all this stuff and centralise
