--- libmysqld/CMakeLists.txt.orig	2017-06-02 07:40:08.506002423 -0600
+++ libmysqld/CMakeLists.txt	2017-06-02 07:40:21.575972304 -0600
@@ -137,6 +137,7 @@
   ${LIBWRAP} ${LIBCRYPT} ${LIBDL}
   ${MYSQLD_STATIC_PLUGIN_LIBS}
   sql_embedded
+  atomic
 )
 
 # Some storage engine were compiled for embedded specifically
--- sql/CMakeLists.txt.orig	2017-06-02 17:52:45.041823696 -0600
+++ sql/CMakeLists.txt	2017-06-02 17:52:25.446874245 -0600
@@ -176,7 +176,8 @@
   ${LIBWRAP} ${LIBCRYPT} ${LIBDL} ${CMAKE_THREAD_LIBS_INIT}
   ${WSREP_LIB}
   ${SSL_LIBRARIES}
-  ${LIBSYSTEMD})
+  ${LIBSYSTEMD}
+  atomic)
 
 IF(WIN32)
   SET(MYSQLD_SOURCE main.cc nt_servc.cc message.rc)
