--- libmysqld/CMakeLists.txt.orig	2019-01-04 08:28:48.000000000 -0700
+++ libmysqld/CMakeLists.txt	2019-01-17 06:52:10.668544469 -0700
@@ -150,6 +150,7 @@
   ${LIBWRAP} ${LIBCRYPT} ${LIBDL}
   ${MYSQLD_STATIC_PLUGIN_LIBS}
   sql_embedded
+  atomic
 )
 
 # Some storage engine were compiled for embedded specifically
--- sql/CMakeLists.txt.orig	2019-01-04 08:28:49.000000000 -0700
+++ sql/CMakeLists.txt	2019-01-17 06:52:10.672544438 -0700
@@ -169,7 +169,8 @@
   ${LIBWRAP} ${LIBCRYPT} ${LIBDL} ${CMAKE_THREAD_LIBS_INIT}
   ${WSREP_LIB}
   ${SSL_LIBRARIES}
-  ${LIBSYSTEMD})
+  ${LIBSYSTEMD}
+  atomic)
 
 IF(WIN32)
   SET(MYSQLD_SOURCE main.cc nt_servc.cc message.rc)
--- storage/rocksdb/build_rocksdb.cmake.orig	2019-01-17 07:52:59.896149354 -0700
+++ storage/rocksdb/build_rocksdb.cmake	2019-01-17 07:53:34.996081204 -0700
@@ -423,7 +423,7 @@
 list(APPEND SOURCES ${CMAKE_CURRENT_BINARY_DIR}/build_version.cc)
 
 ADD_CONVENIENCE_LIBRARY(rocksdblib ${SOURCES})
-target_link_libraries(rocksdblib ${THIRDPARTY_LIBS} ${SYSTEM_LIBS})
+target_link_libraries(rocksdblib ${THIRDPARTY_LIBS} ${SYSTEM_LIBS} atomic)
 IF(CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   set_target_properties(rocksdblib PROPERTIES COMPILE_FLAGS "-fPIC -fno-builtin-memcmp -frtti")
 endif()
