diff -urN a/js/src/configure b/js/src/configure
--- a/js/src/configure	2011-03-31 13:08:36.000000000 -0600
+++ b/js/src/configure	2014-02-05 15:19:39.915096224 -0700
@@ -10603,72 +10603,6 @@
 
 esac
 
-echo $ac_n "checking for ARM SIMD support in compiler""... $ac_c" 1>&6
-echo "configure:10608: checking for ARM SIMD support in compiler" >&5
-cat > conftest.$ac_ext <<EOF
-#line 10610 "configure"
-#include "confdefs.h"
-
-int main() {
-asm("uqadd8 r1, r1, r2");
-; return 0; }
-EOF
-if { (eval echo configure:10617: \"$ac_compile\") 1>&5; (eval $ac_compile) 2>&5; }; then
-  rm -rf conftest*
-  result="yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  result="no"
-fi
-rm -f conftest*
-echo "$ac_t"""$result"" 1>&6
-if test "$result" = "yes"; then
-    cat >> confdefs.h <<\EOF
-#define HAVE_ARM_SIMD 1
-EOF
-
-    HAVE_ARM_SIMD=1
-fi
-
-
-echo $ac_n "checking for ARM NEON support in compiler""... $ac_c" 1>&6
-echo "configure:10638: checking for ARM NEON support in compiler" >&5
-_SAVE_CFLAGS="$CFLAGS"
-if test "$GNU_CC"; then
-  # gcc needs -mfpu=neon to recognize NEON instructions
-  CFLAGS="$CFLAGS -mfpu=neon -mfloat-abi=softfp"
-fi
-cat > conftest.$ac_ext <<EOF
-#line 10645 "configure"
-#include "confdefs.h"
-
-int main() {
-asm("vadd.i8 d0, d0, d0");
-; return 0; }
-EOF
-if { (eval echo configure:10652: \"$ac_compile\") 1>&5; (eval $ac_compile) 2>&5; }; then
-  rm -rf conftest*
-  result="yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  result="no"
-fi
-rm -f conftest*
-echo "$ac_t"""$result"" 1>&6
-if test "$result" = "yes"; then
-    cat >> confdefs.h <<\EOF
-#define HAVE_ARM_NEON 1
-EOF
-
-    HAVE_ARM_NEON=1
-fi
-CFLAGS="$_SAVE_CFLAGS"
-
-
 case "$target_os" in
 darwin*)
     USE_PTHREADS=1
