--- modules/ffmpeg_in/ffmpeg_decode.c	2013-03-11 21:43:22.000000000 -0400
+++ modules/ffmpeg_in/ffmpeg_decode.c	2013-07-11 23:38:15.108181474 -0400
@@ -38,7 +38,9 @@
 #undef USE_AVCODEC2
 #endif
 
-#if (LIBAVCODEC_VERSION_MAJOR >= 54) && (LIBAVCODEC_VERSION_MINOR >= 35)
+#if (LIBAVCODEC_VERSION_MAJOR >= 55) 
+#define USE_AVCTX3
+#elif (LIBAVCODEC_VERSION_MAJOR >= 54) && (LIBAVCODEC_VERSION_MINOR >= 35)
 #define USE_AVCTX3
 #endif
 
@@ -625,10 +628,11 @@
 		if (ffd->frame_start>inBufferLength) ffd->frame_start = 0;
 
 redecode:
-		gotpic = AVCODEC_MAX_AUDIO_FRAME_SIZE;
 #ifdef USE_AVCODEC2
+		gotpic = 192000;
 		len = avcodec_decode_audio3(ctx, (short *)ffd->audio_buf, &gotpic, &pkt);
 #else
+		gotpic = AVCODEC_MAX_AUDIO_FRAME_SIZE;
 		len = avcodec_decode_audio2(ctx, (short *)ffd->audio_buf, &gotpic, inBuffer + ffd->frame_start, inBufferLength - ffd->frame_start);
 #endif
 		if (len<0) { ffd->frame_start = 0; return GF_NON_COMPLIANT_BITSTREAM; }
--- modules/ffmpeg_in/ffmpeg_in.h	2013-03-11 21:43:22.000000000 -0400
+++ modules/ffmpeg_in/ffmpeg_in.h	2013-07-11 23:38:15.108181474 -0400
@@ -120,7 +120,7 @@
 
 	/*for audio packed frames*/
 	u32 frame_start;
-	char audio_buf[AVCODEC_MAX_AUDIO_FRAME_SIZE];
+	char audio_buf[192000];
 	Bool check_h264_isma;
 
 	u32 base_ES_ID;
