From 7531c19a2971645e5bae952d2c3e156c93a79e21 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 8 May 2016 14:22:20 -0600
Subject: [PATCH] libudfread: set HAVE_PTHREAD_H

---
 mythtv/external/libudfread/libudfread.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/external/libudfread/libudfread.pro b/mythtv/external/libudfread/libudfread.pro
index 14a9af3..09f2256 100644
--- a/mythtv/external/libudfread/libudfread.pro
+++ b/mythtv/external/libudfread/libudfread.pro
@@ -20,7 +20,7 @@ QMAKE_CLEAN += $(TARGET)
 
 HEADERS += blockinput.h default_blockinput.h ecma167.h udfread.h
 SOURCES += default_blockinput.c ecma167.c udfread.c udfread-version.c
-DEFINES += HAVE_CONFIG_H
+DEFINES += HAVE_CONFIG_H HAVE_PTHREAD_H
 
 inc.path = $${PREFIX}/include/mythtv/
 inc.files = blockinput.h udfread.h
-- 
2.8.3

