diff --git a/mythtv/external/libudfread/libudfread.pro b/mythtv/external/libudfread/libudfread.pro
index 14a9af3..169aabd 100644
--- a/mythtv/external/libudfread/libudfread.pro
+++ b/mythtv/external/libudfread/libudfread.pro
@@ -14,6 +14,8 @@ win32-msvc* {
 
     # needed for nmake
     QMAKE_CFLAGS   += "/FI ../../libs/libmythbase/compat.h"
+} else {
+    DEFINES += HAVE_PTHREAD_H
 }
 
 QMAKE_CLEAN += $(TARGET)
