diff -Nru a/external/FFmpeg/libavdevice/alsa-audio-common.c b/external/FFmpeg/libavdevice/alsa-audio-common.c
--- a/external/FFmpeg/libavdevice/alsa-audio-common.c	2012-10-02 23:30:24.000000000 +1300
+++ b/external/FFmpeg/libavdevice/alsa-audio-common.c	2013-01-09 13:23:49.704360811 +1300
@@ -28,6 +28,7 @@
  * @author Nicolas George ( nicolas george normalesup org )
  */
 
+#include <sys/types.h>
 #include <alsa/asoundlib.h>
 #include "avdevice.h"
 #include "libavutil/avassert.h"
diff -Nru a/external/FFmpeg/libavdevice/alsa-audio-dec.c b/external/FFmpeg/libavdevice/alsa-audio-dec.c
--- a/external/FFmpeg/libavdevice/alsa-audio-dec.c	2012-10-02 23:30:24.000000000 +1300
+++ b/external/FFmpeg/libavdevice/alsa-audio-dec.c	2013-01-09 13:23:47.911005108 +1300
@@ -45,6 +45,7 @@
  * plugin.
  */
 
+#include <sys/types.h>
 #include <alsa/asoundlib.h>
 #include "libavformat/internal.h"
 #include "libavutil/opt.h"
diff -Nru a/external/FFmpeg/libavdevice/alsa-audio-enc.c b/external/FFmpeg/libavdevice/alsa-audio-enc.c
--- a/external/FFmpeg/libavdevice/alsa-audio-enc.c	2012-10-02 23:30:24.000000000 +1300
+++ b/external/FFmpeg/libavdevice/alsa-audio-enc.c	2013-01-09 13:23:46.880992286 +1300
@@ -37,6 +37,7 @@
  * which gives a low latency suitable for real-time playback.
  */
 
+#include <sys/types.h>
 #include <alsa/asoundlib.h>
 
 #include "libavformat/internal.h"
diff -Nru a/external/FFmpeg/libavdevice/alsa-audio.h b/external/FFmpeg/libavdevice/alsa-audio.h
--- a/external/FFmpeg/libavdevice/alsa-audio.h	2012-10-02 23:30:24.000000000 +1300
+++ b/external/FFmpeg/libavdevice/alsa-audio.h	2013-01-09 13:23:51.121045145 +1300
@@ -30,6 +30,7 @@
 #ifndef AVDEVICE_ALSA_AUDIO_H
 #define AVDEVICE_ALSA_AUDIO_H
 
+#include <sys/types.h>
 #include <alsa/asoundlib.h>
 #include "config.h"
 #include "libavutil/log.h"
diff -Nru a/libs/libmyth/audio/audiooutputalsa.h b/libs/libmyth/audio/audiooutputalsa.h
--- a/libs/libmyth/audio/audiooutputalsa.h	2012-10-02 23:30:24.000000000 +1300
+++ b/libs/libmyth/audio/audiooutputalsa.h	2013-01-09 13:22:13.313180438 +1300
@@ -1,6 +1,7 @@
 #ifndef AUDIOOUTPUTALSA
 #define AUDIOOUTPUTALSA
 
+#include <sys/types.h>
 #define ALSA_PCM_NEW_HW_PARAMS_API
 #define ALSA_PCM_NEW_SW_PARAMS_API
 #include <alsa/asoundlib.h>
diff -Nru a/libs/libmythtv/audioinputalsa.h b/libs/libmythtv/audioinputalsa.h
--- a/libs/libmythtv/audioinputalsa.h	2012-10-02 23:30:24.000000000 +1300
+++ b/libs/libmythtv/audioinputalsa.h	2013-01-09 13:22:44.470223553 +1300
@@ -24,6 +24,7 @@
 #include "audioinput.h"
 
 #ifdef USING_ALSA
+#include <sys/types.h>
 #include <alsa/asoundlib.h>
 #else
 typedef int snd_pcm_t;
