diff -Nru a/libs/libmythui/mythrender_opengl.cpp b/libs/libmythui/mythrender_opengl.cpp
--- a/libs/libmythui/mythrender_opengl.cpp	2011-05-16 08:57:52.000000000 +1200
+++ b/libs/libmythui/mythrender_opengl.cpp	2011-10-23 15:05:33.178392264 +1300
@@ -2,6 +2,8 @@
 #include "mythrender_opengl.h"
 #include "mythxdisplay.h"

+#include <GL/glu.h>
+
 #define LOC QString("OpenGL: ")
 #define LOC_ERR QString("OpenGL Error: ")

Binary files a/libs/libmythui/.mythrender_opengl.cpp.swp and b/libs/libmythui/.mythrender_opengl.cpp.swp differ
