--- a/mythtv/libs/libmythbase/loggingserver.cpp	2017-02-13 21:15:14.150339977 +0100
+++ b/mythtv/libs/libmythbase/loggingserver.cpp	2017-02-13 21:15:08.738490693 +0100
@@ -722,7 +722,7 @@
             if (!item)
                 continue;
 
-            if (item->message()[0] != '\0')
+            if (item->message()[0] != QChar('\0'))
             {
                 qLock.unlock();
                 bool logged = m_logger->logqmsg(*query, item);
