*** a/libs/libmythbase/compat.h		2012-03-31 20:52:30.430461503 -0400
--- b/libs/libmythbase/compat.h		2012-03-31 20:59:05.935585911 -0400
***************
*** 44,49 ****
--- 44,51 ----
  # include <sys/resource.h> // for setpriority
  # include <sys/socket.h>
  # include <sys/wait.h>     // For WIFEXITED on Mac OS X
+ # include <stdio.h>        // for snprintf(), used by inline dlerror()
+ # include <unistd.h>       // for usleep()
  #endif
  
  #ifdef USING_MINGW
*** a/libs/libmyth/rssparse.h		2012-03-31 20:52:30.427453729 -0400
--- b/libs/libmyth/rssparse.h		2012-03-31 21:51:11.687438044 -0400
***************
*** 13,18 ****
--- 13,19 ----
  #include <QPair>
  #include <QMap>
  #include <QVariant>
+ #include <sys/types.h>
  
  #include "mythexp.h"
  
*** a/libs/libmythtv/audioinput.h	2012-03-31 22:41:55.626923063 -0400
--- b/libs/libmythtv/audioinput.h	2012-03-31 22:41:22.478248297 -0400
***************
*** 22,27 ****
--- 22,28 ----
  #define _AUDIOINPUT_H_
  
  #include <QString>
+ #include <unistd.h>
  
  class AudioInput
  {
*** a/external/FFmpeg/libavcodec/x86/h264_qpel_mmx.c	2012-03-31 20:52:29.923150437 -0400
--- b/external/FFmpeg/libavcodec/x86/h264_qpel_mmx.c	2012-03-31 21:58:48.422270780 -0400
***************
*** 398,404 ****
              "2:                         \n\t"\
              \
              : "+a"(src), "+c"(dst)\
!             : "S"((x86_reg)srcStride), "D"((x86_reg)dstStride), "g"(h)\
              : "memory"\
          );\
          src += 4-(h+5)*srcStride;\
--- 398,404 ----
              "2:                         \n\t"\
              \
              : "+a"(src), "+c"(dst)\
!             : "S"((x86_reg)srcStride), "D"((x86_reg)dstStride), "rm"(h)\
              : "memory"\
          );\
          src += 4-(h+5)*srcStride;\
***************
*** 446,452 ****
              QPEL_H264HV(%%mm3, %%mm4, %%mm5, %%mm0, %%mm1, %%mm2, 15*48)\
              "2:                     \n\t"\
              : "+a"(src)\
!             : "c"(tmp), "S"((x86_reg)srcStride), "g"(size)\
              : "memory"\
              );\
          tmp += 4;\
--- 446,452 ----
              QPEL_H264HV(%%mm3, %%mm4, %%mm5, %%mm0, %%mm1, %%mm2, 15*48)\
              "2:                     \n\t"\
              : "+a"(src)\
!             : "c"(tmp), "S"((x86_reg)srcStride), "rm"(size)\
              : "memory"\
              );\
          tmp += 4;\
***************
*** 823,829 ****
          "2:                          \n\t"\
          \
          : "+a"(src), "+c"(dst)\
!         : "S"((x86_reg)srcStride), "D"((x86_reg)dstStride), "g"(h)\
          : XMM_CLOBBERS("%xmm0", "%xmm1", "%xmm2", "%xmm3", \
                         "%xmm4", "%xmm5", "%xmm6", "%xmm7",)\
            "memory"\
--- 823,829 ----
          "2:                          \n\t"\
          \
          : "+a"(src), "+c"(dst)\
!         : "S"((x86_reg)srcStride), "D"((x86_reg)dstStride), "rm"(h)\
          : XMM_CLOBBERS("%xmm0", "%xmm1", "%xmm2", "%xmm3", \
                         "%xmm4", "%xmm5", "%xmm6", "%xmm7",)\
            "memory"\
***************
*** 878,884 ****
              QPEL_H264HV_XMM(%%xmm3, %%xmm4, %%xmm5, %%xmm0, %%xmm1, %%xmm2, 15*48)
              "2:                         \n\t"
              : "+a"(src)
!             : "c"(tmp), "S"((x86_reg)srcStride), "g"(size)
              : XMM_CLOBBERS("%xmm0", "%xmm1", "%xmm2", "%xmm3",
                             "%xmm4", "%xmm5", "%xmm6", "%xmm7",)
                "memory"
--- 878,884 ----
              QPEL_H264HV_XMM(%%xmm3, %%xmm4, %%xmm5, %%xmm0, %%xmm1, %%xmm2, 15*48)
              "2:                         \n\t"
              : "+a"(src)
!             : "c"(tmp), "S"((x86_reg)srcStride), "rm"(size)
              : XMM_CLOBBERS("%xmm0", "%xmm1", "%xmm2", "%xmm3",
                             "%xmm4", "%xmm5", "%xmm6", "%xmm7",)
                "memory"
