From 2af1e78b3fd63de03ea4453d69c067a275e70354 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Fri, 26 Jul 2019 12:30:41 -0600
Subject: [PATCH] link with libatomic

---
 modules/core/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/core/CMakeLists.txt b/modules/core/CMakeLists.txt
index 246f23b0b..e4768362a 100644
--- a/modules/core/CMakeLists.txt
+++ b/modules/core/CMakeLists.txt
@@ -31,6 +31,8 @@ if(HAVE_TBB)
   list(APPEND extra_libs tbb)
 endif()
 
+list(APPEND extra_libs atomic)
+
 if(DEFINED WINRT AND NOT DEFINED ENABLE_WINRT_MODE_NATIVE)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /ZW")
 endif()
-- 
2.21.0

