diff -urN a/modules/flann/include/opencv2/flann/dist.h b/modules/flann/include/opencv2/flann/dist.h
--- a/modules/flann/include/opencv2/flann/dist.h	2011-09-12 12:42:27.000000000 -0600
+++ b/modules/flann/include/opencv2/flann/dist.h	2011-11-05 23:29:38.518023435 -0600
@@ -60,7 +60,7 @@
 inline double abs<double>(double x) { return fabs(x); }
 
 template<>
-inline long double abs<long double>(long double x) { return fabsl(x); }
+inline long double abs<long double>(long double x) { return fabs(x); }
 
 
 template<typename T>
