diff -ur opencv-2.4.6.orig/cmake/OpenCVDetectOpenCL.cmake opencv-2.4.6/cmake/OpenCVDetectOpenCL.cmake
--- opencv-2.4.6.orig/cmake/OpenCVDetectOpenCL.cmake	2013-07-11 18:57:27.363199736 +0000
+++ opencv-2.4.6/cmake/OpenCVDetectOpenCL.cmake	2013-07-11 19:04:00.577069207 +0000
@@ -17,8 +17,7 @@
               NAMES OpenCL/cl.h CL/cl.h
               HINTS ${OPENCL_ROOT_DIR}
               PATH_SUFFIXES include include/nvidia-current
-              DOC "OpenCL include directory"
-              NO_DEFAULT_PATH)
+              DOC "OpenCL include directory")
 
     if (X86_64)
       set(OPENCL_POSSIBLE_LIB_SUFFIXES lib/Win64 lib/x86_64 lib/x64)
@@ -30,8 +29,7 @@
               NAMES OpenCL
               HINTS ${OPENCL_ROOT_DIR}
               PATH_SUFFIXES ${OPENCL_POSSIBLE_LIB_SUFFIXES}
-              DOC "OpenCL library"
-              NO_DEFAULT_PATH)
+              DOC "OpenCL library")
 
     mark_as_advanced(OPENCL_INCLUDE_DIR OPENCL_LIBRARY)
     include(FindPackageHandleStandardArgs)
