diff -Nbaur OpenCV-2.4.3.orig/cmake/OpenCVGenPkgconfig.cmake OpenCV-2.4.3/cmake/OpenCVGenPkgconfig.cmake
--- OpenCV-2.4.3.orig/cmake/OpenCVGenPkgconfig.cmake	2012-11-05 18:42:32.460958392 +0000
+++ OpenCV-2.4.3/cmake/OpenCVGenPkgconfig.cmake	2012-11-05 18:52:43.766689315 +0000
@@ -47,7 +47,7 @@
 
   #need better solution....
   if(libpath MATCHES "3rdparty")
-    set(installDir "share/OpenCV/3rdparty/${OPENCV_LIB_INSTALL_PATH}")
+    set(installDir "share/opencv/3rdparty/${OPENCV_LIB_INSTALL_PATH}")
   else()
     set(installDir "${OPENCV_LIB_INSTALL_PATH}")
   endif()
diff -Nbaur OpenCV-2.4.3.orig/CMakeLists.txt OpenCV-2.4.3/CMakeLists.txt
--- OpenCV-2.4.3.orig/CMakeLists.txt	2012-11-05 18:42:32.700956757 +0000
+++ OpenCV-2.4.3/CMakeLists.txt	2012-11-05 18:55:42.358787848 +0000
@@ -221,9 +221,9 @@
 if(ANDROID OR WIN32)
   set(OPENCV_DOC_INSTALL_PATH doc)
 elseif(INSTALL_TO_MANGLED_PATHS)
-  set(OPENCV_DOC_INSTALL_PATH share/OpenCV-${OPENCV_VERSION}/doc)
+  set(OPENCV_DOC_INSTALL_PATH share/doc/opencv-${OPENCV_VERSION})
 else()
-  set(OPENCV_DOC_INSTALL_PATH share/OpenCV/doc)
+  set(OPENCV_DOC_INSTALL_PATH share/doc/opencv)
 endif()
 
 if(ANDROID)
@@ -237,14 +237,14 @@
   set(LIBRARY_OUTPUT_PATH         "${OpenCV_BINARY_DIR}/lib")
   set(3P_LIBRARY_OUTPUT_PATH      "${OpenCV_BINARY_DIR}/3rdparty/lib${LIB_SUFFIX}")
   set(OPENCV_LIB_INSTALL_PATH     lib${LIB_SUFFIX})
-  set(OPENCV_3P_LIB_INSTALL_PATH  share/OpenCV/3rdparty/${OPENCV_LIB_INSTALL_PATH})
+  set(OPENCV_3P_LIB_INSTALL_PATH  share/opencv/3rdparty/${OPENCV_LIB_INSTALL_PATH})
   set(OPENCV_INCLUDE_INSTALL_PATH include)
 
   math(EXPR SIZEOF_VOID_P_BITS "8 * ${CMAKE_SIZEOF_VOID_P}")
   if(LIB_SUFFIX AND NOT SIZEOF_VOID_P_BITS EQUAL LIB_SUFFIX)
     set(OPENCV_CONFIG_INSTALL_PATH lib${LIB_SUFFIX}/cmake/opencv)
   else()
-    set(OPENCV_CONFIG_INSTALL_PATH share/OpenCV)
+    set(OPENCV_CONFIG_INSTALL_PATH share/opencv)
   endif()
 endif()
 
diff -Nbaur OpenCV-2.4.3.orig/data/CMakeLists.txt OpenCV-2.4.3/data/CMakeLists.txt
--- OpenCV-2.4.3.orig/data/CMakeLists.txt	2012-11-05 18:42:32.460958392 +0000
+++ OpenCV-2.4.3/data/CMakeLists.txt	2012-11-05 18:52:43.816688977 +0000
@@ -5,6 +5,6 @@
   install(FILES ${HAAR_CASCADES} DESTINATION sdk/etc/haarcascades COMPONENT main)
   install(FILES ${LBP_CASCADES}  DESTINATION sdk/etc/lbpcascades  COMPONENT main)
 elseif(NOT WIN32)
-  install(FILES ${HAAR_CASCADES} DESTINATION share/OpenCV/haarcascades COMPONENT main)
-  install(FILES ${LBP_CASCADES}  DESTINATION share/OpenCV/lbpcascades  COMPONENT main)
+  install(FILES ${HAAR_CASCADES} DESTINATION share/opencv/haarcascades COMPONENT main)
+  install(FILES ${LBP_CASCADES}  DESTINATION share/opencv/lbpcascades  COMPONENT main)
 endif()
diff -Nbaur OpenCV-2.4.3.orig/samples/c/CMakeLists.txt OpenCV-2.4.3/samples/c/CMakeLists.txt
--- OpenCV-2.4.3.orig/samples/c/CMakeLists.txt	2012-11-05 18:42:32.697623446 +0000
+++ OpenCV-2.4.3/samples/c/CMakeLists.txt	2012-11-05 18:52:43.826688906 +0000
@@ -54,7 +54,7 @@
 if (INSTALL_C_EXAMPLES AND NOT WIN32)
   file(GLOB C_SAMPLES *.c *.cpp *.jpg *.png *.data makefile.* build_all.sh *.dsp *.cmd )
   install(FILES ${C_SAMPLES}
-          DESTINATION share/OpenCV/samples/c
+          DESTINATION share/opencv/samples/c
           PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
 endif ()
 
diff -Nbaur OpenCV-2.4.3.orig/samples/cpp/CMakeLists.txt OpenCV-2.4.3/samples/cpp/CMakeLists.txt
--- OpenCV-2.4.3.orig/samples/cpp/CMakeLists.txt	2012-11-05 18:42:32.677623583 +0000
+++ OpenCV-2.4.3/samples/cpp/CMakeLists.txt	2012-11-05 18:52:43.823355598 +0000
@@ -64,7 +64,7 @@
 if (INSTALL_C_EXAMPLES AND NOT WIN32)
   file(GLOB C_SAMPLES *.c *.cpp *.jpg *.png *.data makefile.* build_all.sh *.dsp *.cmd )
   install(FILES ${C_SAMPLES}
-          DESTINATION share/OpenCV/samples/cpp
+          DESTINATION share/opencv/samples/cpp
           PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
 endif()
 
diff -Nbaur OpenCV-2.4.3.orig/samples/gpu/CMakeLists.txt OpenCV-2.4.3/samples/gpu/CMakeLists.txt
--- OpenCV-2.4.3.orig/samples/gpu/CMakeLists.txt	2012-11-05 18:42:32.670956962 +0000
+++ OpenCV-2.4.3/samples/gpu/CMakeLists.txt	2012-11-05 18:52:43.820022288 +0000
@@ -64,7 +64,7 @@
 if (INSTALL_C_EXAMPLES AND NOT WIN32)
   file(GLOB install_list *.c *.cpp *.jpg *.png *.data makefile.* build_all.sh *.dsp *.cmd )
   install(FILES ${install_list}
-          DESTINATION share/OpenCV/samples/${project}
+          DESTINATION share/opencv/samples/${project}
           PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
 endif()
 
diff -Nbaur OpenCV-2.4.3.orig/samples/gpu/performance/CMakeLists.txt OpenCV-2.4.3/samples/gpu/performance/CMakeLists.txt
--- OpenCV-2.4.3.orig/samples/gpu/performance/CMakeLists.txt	2012-11-05 18:42:32.674290272 +0000
+++ OpenCV-2.4.3/samples/gpu/performance/CMakeLists.txt	2012-11-05 18:52:43.823355598 +0000
@@ -21,6 +21,6 @@
 if(INSTALL_C_EXAMPLES AND NOT WIN32)
   file(GLOB GPU_FILES performance/*.cpp performance/*.h)
   install(FILES ${GPU_FILES}
-          DESTINATION share/OpenCV/samples/gpu/performance
+          DESTINATION share/opencv/samples/gpu/performance
           PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
 endif()
