diff -u -r opencv-4.0.1/cmake/templates/opencv-XXX.pc.in opencv-4.0.1-includedir/cmake/templates/opencv-XXX.pc.in
--- opencv-4.0.1/cmake/templates/opencv-XXX.pc.in	2018-12-22 07:03:30.000000000 +0000
+++ opencv-4.0.1-includedir/cmake/templates/opencv-XXX.pc.in	2019-02-07 15:08:15.016613349 +0000
@@ -11,4 +10,4 @@
 Version: @OPENCV_VERSION_PLAIN@
 Libs: @OPENCV_PC_LIBS@
 Libs.private: @OPENCV_PC_LIBS_PRIVATE@
-Cflags: -I${includedir_old} -I${includedir_new}
+Cflags: -I${includedir_new}
