diff -Naur flex-2.5.31-orig/flex.skl flex-2.5.31/flex.skl
--- flex-2.5.31-orig/flex.skl	2003-03-31 17:51:38.000000000 -0800
+++ flex-2.5.31/flex.skl	2004-04-11 12:19:10.000000000 -0700
@@ -3280,7 +3280,6 @@
 #undef yy_set_bol
 #undef yy_new_buffer
 #undef yy_set_interactive
-#undef yytext_ptr
 #undef YY_DO_BEFORE_ACTION
 
 #ifdef YY_DECL_IS_OURS
diff -Naur flex-2.5.31-orig/gen.c flex-2.5.31/gen.c
--- flex-2.5.31-orig/gen.c	2003-03-30 11:58:44.000000000 -0800
+++ flex-2.5.31/gen.c	2004-04-11 12:19:38.000000000 -0700
@@ -1812,7 +1812,6 @@
 			if (yytext_is_array) {
 				if (!reentrant){
     				indent_puts ("static int yy_more_offset = 0;");
-                }else{
                     indent_puts ("static int yy_prev_more_offset = 0;");
                 }
 			}
