Index: src/btctlimpl.c
===================================================================
--- src/btctlimpl.c	(revision 134)
+++ src/btctlimpl.c	(working copy)
@@ -330,8 +330,8 @@
 	if (dd>=0) {
 		for (i = 0; i < num_rsp; i++) {
 			memset(name, 0, sizeof(name));
-			if (hci_remote_name(dd, &(info+i)->bdaddr,
-						sizeof(name), name, 100000) < 0)
+			if (hci_read_remote_name(dd, &(info+i)->bdaddr,
+						 sizeof(name), name, 100000) < 0)
 				strcpy(name, "n/a");
 
 			baswap(&bdaddr, &(info+i)->bdaddr);
