diff -urN a/linux.bd b/linux.bd
--- a/linux.bd	2010-11-04 01:35:38.000000000 -0600
+++ b/linux.bd	2013-01-15 17:03:51.797178012 -0700
@@ -4,9 +4,9 @@
 	flags = 0x01;
 }
 sources {
-	power_prep="./power_prep/power_prep";
-	sdram_prep="./boot_prep/boot_prep";
-	linux_prep="./linux_prep/output-target/linux_prep";
+	power_prep="imx-bootlets-olinuxino/power_prep";
+	sdram_prep="imx-bootlets-olinuxino/boot_prep";
+	linux_prep="imx-bootlets-olinuxino/linux_prep";
 	zImage = "./zImage";
 }
 
