diff --git a/linux.bd b/linux.bd
index c4956d0..036a609 100644
--- a/linux.bd
+++ b/linux.bd
@@ -4,10 +4,10 @@ options {
 	flags = 0x01;
 }
 sources {
-	power_prep="./power_prep/power_prep";
-	sdram_prep="./boot_prep/boot_prep";
-	linux_prep="./linux_prep/output-target/linux_prep";
-	zImage = "./zImage";
+	power_prep="imx-bootlets-olinuxino/power_prep";
+	sdram_prep="imx-bootlets-olinuxino/boot_prep";
+	linux_prep="imx-bootlets-olinuxino/linux_prep";
+	zImage = "zImage";
 }
 
 section (0) {
