diff --git a/DEPS b/DEPS
index 627823962f8..2a66a2d7e12 100644
--- a/DEPS
+++ b/DEPS
@@ -64,9 +64,8 @@ vars = {
   # Checkout the flute benchmark only when benchmarking.
   "checkout_flute": False,
 
-  # Checkout Android dependencies only on Mac and Linux.
-  "download_android_deps":
-    "host_os == mac or (host_os == linux and host_cpu == x64)",
+  # Do not checkout Android dependencies.
+  "download_android_deps": False,
 
   # Checkout extra javascript engines for testing or benchmarking.
   # d8, the V8 shell, is always checked out.
@@ -81,7 +80,7 @@ vars = {
   "gn_version": "git_revision:e4702d7409069c4f12d45ea7b7f0890717ca3f4b",
 
   "reclient_version": "git_revision:c7349324c93c6e0d85bc1e00b5d7526771006ea0",
-  "download_reclient": True,
+  "download_reclient": False,
 
   # Update from https://chrome-infra-packages.appspot.com/p/fuchsia/sdk/core
   "fuchsia_sdk_version": "version:16.20231105.3.1",
@@ -503,37 +502,7 @@ Var("dart_root") + "/third_party/pkg/tar":
   Var("dart_root") + "/third_party/pkg/yaml":
       Var("dart_git") + "yaml.git" + "@" + Var("yaml_rev"),
 
-  Var("dart_root") + "/buildtools/sysroot/linux": {
-      "packages": [
-          {
-              "package": "fuchsia/third_party/sysroot/linux",
-              "version": "git_revision:fa7a5a9710540f30ff98ae48b62f2cdf72ed2acd",
-          },
-      ],
-      "condition": "host_os == linux",
-      "dep_type": "cipd",
-  },
-  Var("dart_root") + "/buildtools/sysroot/focal": {
-      "packages": [
-          {
-              "package": "fuchsia/third_party/sysroot/focal",
-              "version": "git_revision:fa7a5a9710540f30ff98ae48b62f2cdf72ed2acd",
-          },
-      ],
-      "condition": "host_os == linux",
-      "dep_type": "cipd",
-  },
-
   # Keep consistent with pkg/test_runner/lib/src/options.dart.
-  Var("dart_root") + "/buildtools/linux-x64/clang": {
-      "packages": [
-          {
-              "package": "fuchsia/third_party/clang/linux-amd64",
-              "version": Var("clang_version"),
-          },
-      ],
-      "dep_type": "cipd",
-  },
   Var("dart_root") + "/buildtools/mac-x64/clang": {
       "packages": [
           {
@@ -600,16 +569,6 @@ Var("dart_root") + "/third_party/pkg/tar":
     "dep_type": "cipd",
   },
 
-  Var("dart_root") + "/buildtools": {
-      "packages": [
-          {
-              "package": "gn/gn/${{platform}}",
-              "version": Var("gn_version"),
-          },
-      ],
-      "condition": "host_os != 'win'",
-      "dep_type": "cipd",
-  },
   Var("dart_root") + "/buildtools/win": {
       "packages": [
           {
@@ -621,14 +580,6 @@ Var("dart_root") + "/third_party/pkg/tar":
       "dep_type": "cipd",
   },
 
-  Var("dart_root") + "/buildtools/ninja": {
-      "packages": [{
-          "package": "infra/3pp/tools/ninja/${{platform}}",
-          "version": Var("ninja_tag"),
-      }],
-      "dep_type": "cipd",
-  },
-
   Var("dart_root") + "/third_party/android_tools/ndk": {
       "packages": [
           {
@@ -693,35 +644,6 @@ Var("dart_root") + "/third_party/pkg/tar":
     "dep_type": "cipd",
   },
 
-  # TODO(37531): Remove these cipd packages and build with sdk instead when
-  # benchmark runner gets support for that.
-  Var("dart_root") + "/benchmarks/FfiBoringssl/native/out/": {
-      "packages": [
-          {
-              "package": "dart/benchmarks/ffiboringssl",
-              "version": "commit:a86c69888b9a416f5249aacb4690a765be064969",
-          },
-      ],
-      "dep_type": "cipd",
-  },
-  Var("dart_root") + "/benchmarks/FfiCall/native/out/": {
-      "packages": [
-          {
-              "package": "dart/benchmarks/fficall",
-              "version": "ebF5aRXKDananlaN4Y8b0bbCNHT1MnkGbWqfpCpiND4C",
-          },
-      ],
-          "dep_type": "cipd",
-  },
-  Var("dart_root") + "/benchmarks/NativeCall/native/out/": {
-      "packages": [
-          {
-              "package": "dart/benchmarks/nativecall",
-              "version": "w1JKzCIHSfDNIjqnioMUPq0moCXKwX67aUfhyrvw4E0C",
-          },
-      ],
-          "dep_type": "cipd",
-  },
   Var("dart_root") + "/third_party/browsers/chrome": {
       "packages": [
           {
