--- ARPACK/ARmake.inc~	1996-04-26 21:12:57.000000000 +0200
+++ ARPACK/ARmake.inc	2003-07-19 11:57:10.000000000 +0200
@@ -25,7 +25,7 @@
 # | other than your home directory.      |
 # %--------------------------------------%
 #
-home = $(HOME)/ARPACK
+home = $(PWD)/.
 #
 #  %--------------------------------------%
 #  | The platform identifier to suffix to |
@@ -43,7 +43,7 @@
 UTILdir      = $(home)/UTIL
 SRCdir       = $(home)/SRC
 #
-DIRS        = $(BLASdir) $(LAPACKdir) $(UTILdir) $(SRCdir)
+#DIRS        = $(BLASdir) $(LAPACKdir) $(UTILdir) $(SRCdir)
 #
 # %-------------------------------------------------------------------%
 # | Comment out the previous line and uncomment the following         |
@@ -51,13 +51,13 @@
 # | NOTE: ARPACK assumes the use of LAPACK version 2 codes.           |
 # %-------------------------------------------------------------------%
 #
-#DIRS         = $(UTILdir) $(SRCdir)
+DIRS         = $(UTILdir) $(SRCdir)
 #
 # %---------------------------------------------------%
 # | The name of the libraries to be created/linked to |
 # %---------------------------------------------------%
 #
-ARPACKLIB  = $(home)/libarpack_$(PLAT).a
+ARPACKLIB  = $(home)/libarpack.a
 LAPACKLIB = 
 BLASLIB = 
 #
@@ -112,7 +112,7 @@
 LN      = ln
 LNFLAGS = -s
 
-MAKE    = /bin/make
+MAKE    = make
 
 RM      = rm
 RMFLAGS = -f
--- ARPACK/UTIL/second.f.old 2008-07-08 17:38:51.759444357 +0200
+++ ARPACK/UTIL/second.f 2008-07-08 17:39:37.056301918 +0200
@@ -21,7 +21,6 @@
 *     ..
 *     .. External Functions ..
       REAL               ETIME
-      EXTERNAL           ETIME
 *     ..
 *     .. Executable Statements ..
 *
