diff --git i/vendor/src/github.com/spacemonkeygo/openssl/ctx.go w/vendor/src/github.com/spacemonkeygo/openssl/ctx.go
index 22d6dd1..25b0eec 100644
--- i/vendor/src/github.com/spacemonkeygo/openssl/ctx.go
+++ w/vendor/src/github.com/spacemonkeygo/openssl/ctx.go
@@ -141,7 +141,7 @@ func NewCtxWithVersion(version SSLVersion) (*Ctx, error) {
 	var method *C.SSL_METHOD
 	switch version {
 	case SSLv3:
-		method = C.SSLv3_method()
+		method = C.SSLv23_method()
 	case TLSv1:
 		method = C.TLSv1_method()
 	case TLSv1_1:
