--- libreoffice-7.3.0.3.orig/solenv/gbuild/platform/com_GCC_defs.mk	2022-02-02 23:12:39.000000000 +0000
+++ libreoffice-7.3.0.3/solenv/gbuild/platform/com_GCC_defs.mk	2022-02-02 23:27:44.167129779 +0000
@@ -196,7 +196,7 @@ else
 # use first element of the PARALLELISM, to allow values like "12 -l 14" to specify load limits
 gb_LTOFLAGS := -flto=$(if $(filter-out 0,$(PARALLELISM)),$(firstword $(PARALLELISM)),auto) -fuse-linker-plugin -O2
 # clang does not support -flto=<number>
-gb_CLANG_LTOFLAGS := -flto=thin
+gb_CLANG_LTOFLAGS := -flto
 endif
 endif
 
