--- libreoffice-5.1.3.2/configure.ac		2016-05-12 21:26:13.099948527 +0200
+++ libreoffice-5.1.3.2/configure.ac.new	2016-05-12 21:46:50.613664941 +0200
@@ -11169,6 +11169,9 @@
     if test "x$kde_incdir" = "xno"; then
         AC_MSG_ERROR([KDE4 headers not found.  Please specify the root of your KDE4 installation by exporting KDE4DIR before running "configure".])
     fi
+    # see http://bugs.debian.org/823145. -isystem /usr/include fails with
+    # gcc6
+    if test "$kde_incdir" = "/usr/include"; then kde_incdir=; fi
 
     dnl Check for KDE4 libraries
     AC_MSG_CHECKING([for KDE4 libraries])
