--- libreoffice-6.1.3.2/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx.orig	2018-12-14 20:04:01.633697240 +0000
+++ libreoffice-6.1.3.2/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx	2018-12-14 20:04:18.020410653 +0000
@@ -555,7 +555,7 @@
     LinkAction* pAction = link->getAction();
     if (pAction && pAction->getKind() == actionURI)
     {
-        const char* pURI = static_cast<LinkURI*>(pAction)->getURI()->getCString();
+        const char* pURI = static_cast<LinkURI*>(pAction)->getURI()->c_str();
 
         std::vector<char> aEsc( lcl_escapeLineFeeds(pURI) );
 
@@ -757,7 +757,7 @@
 
             aFont = it->second;
 
-            std::vector<char> aEsc( lcl_escapeLineFeeds(aFont.familyName.getCString()) );
+            std::vector<char> aEsc( lcl_escapeLineFeeds(aFont.familyName.c_str()) );
             printf( " %d %d %d %d %f %d %s",
                     aFont.isEmbedded,
                     aFont.isBold,
