--- libreoffice-25.8.1.1/configure.ac	2025-08-26 00:08:27.000000000 +0200
+++ libreoffice-25.8.1.1/configure.ac.new	2025-09-17 20:34:51.993711385 +0200
@@ -11453,7 +11453,7 @@
 dnl ===================================================================
 dnl Orcus
 dnl ===================================================================
-libo_CHECK_SYSTEM_MODULE([orcus],[ORCUS],[liborcus-0.20 >= 0.20.0])
+libo_CHECK_SYSTEM_MODULE([orcus],[ORCUS],[liborcus-0.21 >= 0.21.0])
 if test "$with_system_orcus" != "yes"; then
     if test "$SYSTEM_BOOST" = "TRUE"; then
         dnl Link with Boost.System
