--- libreoffice-7.2.0.3/external/skia/UnpackedTarball_skia.mk.orig	2021-08-10 19:27:48.000000000 +0200
+++ libreoffice-7.2.0.3/external/skia/UnpackedTarball_skia.mk	2021-08-12 18:09:22.813517161 +0200
@@ -39,6 +39,7 @@
     fast-png-write.patch.1 \
     skia_sk_cpu_sse_level_0_by_default.patch.1 \
     fix-warnings.patch.1 \
+    skia-freetype2.11.patch \
 
 $(eval $(call gb_UnpackedTarball_set_patchlevel,skia,1))
 
