diff -up libreoffice-24.8.2.1/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx.omv~ libreoffice-24.8.2.1/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
--- libreoffice-24.8.2.1/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx.omv~	2024-10-20 19:08:49.935352230 +0200
+++ libreoffice-24.8.2.1/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx	2024-10-20 20:01:30.448965822 +0200
@@ -1306,9 +1306,9 @@ poppler_bool PDFOutDev::tilingPatternFil
         0, nBitmapWidth * nBitmapHeight, Object(objNull)));
     auto aDecode = Object(objNull);
     std::unique_ptr<GfxImageColorMap> pRgbIdentityColorMap(new GfxImageColorMap(8, &aDecode,
-        new GfxDeviceRGBColorSpace()));
+        std::make_unique<GfxDeviceRGBColorSpace>()));
     std::unique_ptr<GfxImageColorMap> pGrayIdentityColorMap(new GfxImageColorMap(8, &aDecode,
-        new GfxDeviceGrayColorSpace()));
+        std::make_unique<GfxDeviceGrayColorSpace>()));
 
     OutputBuffer aBuf; initBuf(aBuf);
     writePng_(aBuf, pRgbStr.get(), nBitmapWidth, nBitmapHeight, pRgbIdentityColorMap.get(),
