diff -wbBur gnubiff-2.2.10/src/support.h gnubiff-2.2.10.my/src/support.h
--- gnubiff-2.2.10/src/support.h	2008-03-22 05:08:51.000000000 +0300
+++ gnubiff-2.2.10.my/src/support.h	2008-04-22 14:21:58.000000000 +0400
@@ -35,6 +35,9 @@
 #endif
 #include "nls.h"
 
+#include <stdlib.h>
+#define EXIT_SUCCESS 0
+#define EXIT_FAILURE 1
 
 #include <glib.h>
 #include <sstream>
@@ -131,6 +134,9 @@
 		if ((pos == end) && (last < num))
 			result << sep << num;
 
+		if ((pos == end) && (last != num))
+		  result << sep << num;
+	
 		// Save new number
 		inf_bound = last = num;
 	}
