--- gpsmap.cc
+++ gpsmap.cc
@@ -874,7 +874,7 @@
 
                 // Break up the path to the gpsxml file and form a path based on that
                 unsigned int lastslash = 0;
-                for (unsigned int x = origxmlfile.find('/'); x != string::npos;
+                for (string::size_type x = origxmlfile.find('/'); x != string::npos;
                      lastslash = x, x = origxmlfile.find('/', lastslash+1)) {
                     // We don't actually need to do anything...
                 }
@@ -882,7 +882,7 @@
                 comp = origxmlfile.substr(0, lastslash);
 
                 lastslash = 0;
-                for (unsigned int x = orignetfile.find('/'); x != string::npos;
+                for (string::size_type x = orignetfile.find('/'); x != string::npos;
                      lastslash = x, x = orignetfile.find('/', lastslash+1)) {
                     // We don't actually need to do anything...
                 }
