--- traceroute-1.4a12/Makefile.in.old	2000-11-23 21:05:29.000000000 +0100
+++ traceroute-1.4a12/Makefile.in	2008-09-27 20:04:30.000000000 +0200
@@ -90,10 +90,10 @@
 	sed -e 's/.*/char version[] = "&";/' $(srcdir)/VERSION > $@
 
 install: force
-	$(INSTALL) -m 4555 -o root -g bin traceroute $(DESTDIR)$(BINDEST)
+	$(INSTALL) -m 4555 -o root -g root traceroute $(DESTDIR)$(BINDEST)
 
 install-man: force
-	$(INSTALL) -m 444 -o bin -g bin traceroute.8 $(DESTDIR)$(MANDEST)/man8
+	$(INSTALL) -m 644 -o root -g root traceroute.8 $(DESTDIR)$(MANDEST)/man8
 
 lint: $(GENSRC) force
 	lint -hbxn $(SRC) | \
