--- openjdk/corba/make/common/shared/Defs-utils.gmk.orig	2013-05-19 19:57:58.530784147 +0000
+++ openjdk/corba/make/common/shared/Defs-utils.gmk	2013-05-19 19:58:44.777147938 +0000
@@ -83,9 +83,9 @@
 CHMOD          = $(UTILS_COMMAND_PATH)chmod
 CMP            = $(UTILS_USR_BIN_PATH)cmp
 COMM           = $(UTILS_USR_BIN_PATH)comm
-COMPRESS       = $(UTILS_USR_BIN_PATH)compress
+COMPRESS       = $(UTILS_COMMAND_PATH)compress
 CP             = $(UTILS_COMMAND_PATH)cp
-CPIO           = $(UTILS_COMMAND_PATH)cpio
+CPIO           = $(UTILS_USR_BIN_PATH)cpio
 CUT            = $(UTILS_USR_BIN_PATH)cut
 DATE           = $(UTILS_COMMAND_PATH)date
 DF             = $(UTILS_COMMAND_PATH)df
@@ -131,7 +131,7 @@
 TAIL           = $(UTILS_USR_BIN_PATH)tail
 TAR            = $(UTILS_COMMAND_PATH)tar
 TEST           = $(UTILS_USR_BIN_PATH)test
-TOUCH          = $(UTILS_COMMAND_PATH)touch
+TOUCH          = $(UTILS_USR_BIN_PATH)touch
 TR             = $(UTILS_USR_BIN_PATH)tr
 TRUE           = $(UTILS_COMMAND_PATH)true
 UNAME          = $(UTILS_COMMAND_PATH)uname
@@ -182,7 +182,7 @@
   # others have it in /usr/bin.
   SORT=$(firstword $(wildcard $(UTILS_COMMAND_PATH)sort) \
                    $(wildcard $(UTILS_USR_BIN_PATH)sort))
-  NAWK           = $(USRBIN_PATH)gawk
+  NAWK           = $(UTILS_COMMAND_PATH)gawk
   # Intrinsic unix command, with backslash-escaped character interpretation
   ECHO           = /bin/echo -e
   # These are really in UTILS_USR_BIN_PATH on Linux (only sccs is not)
