diff -urN a/Makefile.am b/Makefile.am
--- a/Makefile.am	2013-03-11 12:44:50.432980398 -0600
+++ b/Makefile.am	2013-03-21 15:49:27.252081589 -0600
@@ -243,8 +243,7 @@
 ICEDTEA_PATCHES = \
 	patches/8006179-lookup_using_findvirtual.patch \
 	patches/8006882-sun.proxy.patch \
-	patches/pr1303-ifdef_fix.patch \
-	patches/revert-7017193.patch
+	patches/pr1303-ifdef_fix.patch
 
 # Conditional patches
 
diff -urN a/Makefile.in b/Makefile.in
--- a/Makefile.in	2013-03-11 12:45:24.481530253 -0600
+++ b/Makefile.in	2013-03-21 15:49:11.552131813 -0600
@@ -636,7 +636,7 @@
 # Patch list
 ICEDTEA_PATCHES = patches/8006179-lookup_using_findvirtual.patch \
 	patches/8006882-sun.proxy.patch patches/pr1303-ifdef_fix.patch \
-	patches/revert-7017193.patch $(am__append_5) $(am__append_6) \
+	$(am__append_5) $(am__append_6) \
 	$(am__append_7) $(am__append_8) $(am__append_9) \
 	$(am__append_10) $(am__append_11) $(am__append_12) \
 	$(am__append_13) $(am__append_14) $(am__append_15) \
