--- openjdk/jdk/src/share/native/sun/awt/splashscreen/splashscreen_gif.c.orig	2014-05-27 22:17:19.766866065 +0200
+++ openjdk/jdk/src/share/native/sun/awt/splashscreen/splashscreen_gif.c	2014-05-27 22:16:35.452894426 +0200
@@ -310,7 +310,11 @@
     free(pBitmapBits);
     free(pOldBitmapBits);
 
+#if GIFLIB_MAJOR >= 5
+    DGifCloseFile(gif, NULL);
+#else
     DGifCloseFile(gif);
+#endif
 
     return 1;
 }
