--- openjdk/jdk/make/common/shared/Defs-utils.gmk.old	2011-08-06 15:46:01.000000000 +0200
+++ openjdk/jdk/make/common/shared/Defs-utils.gmk	2011-08-06 15:59:26.000000000 +0200
@@ -91,7 +91,7 @@
 CHMOD          = $(UTILS_COMMAND_PATH)chmod
 CMP            = $(UTILS_USR_BIN_PATH)cmp
 COMM           = $(UTILS_USR_BIN_PATH)comm
-COMPRESS       = $(UTILS_USR_BIN_PATH)compress
+COMPRESS       = $(UTILS_COMMAND_PATH)compress
 CP             = $(UTILS_COMMAND_PATH)cp
 CPIO           = $(UTILS_COMMAND_PATH)cpio
 CUT            = $(UTILS_USR_BIN_PATH)cut
@@ -101,13 +101,13 @@
 DIRNAME        = $(UTILS_USR_BIN_PATH)dirname
 DUMP           = $(UTILS_CCS_BIN_PATH)dump
 ECHO           = $(UTILS_COMMAND_PATH)echo
-EGREP          = $(UTILS_COMMAND_PATH)egrep
+EGREP          = $(UTILS_USR_BIN_PATH)egrep
 EXPR           = $(UTILS_USR_BIN_PATH)expr
 FILE           = $(UTILS_USR_BIN_PATH)file
 FIND           = $(UTILS_USR_BIN_PATH)find
 FMT            = $(UTILS_COMMAND_PATH)fmt
 GDB            = $(UTILS_USR_BIN_PATH)gdb
-GREP           = $(UTILS_COMMAND_PATH)grep
+GREP           = $(UTILS_USR_BIN_PATH)grep
 GUNZIP         = $(UTILS_COMMAND_PATH)gunzip
 # GZIP is used for solaris. Linux and windows use tar czf
 GZIP           = $(UTILS_COMMAND_PATH)gzip
@@ -140,7 +140,7 @@
 TAIL           = $(UTILS_USR_BIN_PATH)tail
 TAR            = $(UTILS_COMMAND_PATH)tar
 TEST           = $(UTILS_USR_BIN_PATH)test
-TOUCH          = $(UTILS_COMMAND_PATH)touch
+TOUCH          = $(UTILS_USR_BIN_PATH)touch
 TR             = $(UTILS_USR_BIN_PATH)tr
 TRUE           = $(UTILS_COMMAND_PATH)true
 UNAME          = $(UTILS_COMMAND_PATH)uname
