--- openjdk/hotspot/src/share/vm/runtime/vmStructs.cpp.orig	2015-07-21 19:07:36.946010099 -0600
+++ openjdk/hotspot/src/share/vm/runtime/vmStructs.cpp	2015-07-21 19:07:55.625685939 -0600
@@ -843,10 +843,6 @@
   /***********************************/                                                                                              \
                                                                                                                                      \
      static_field(StubRoutines,                _call_stub_return_address,                     address)                               \
-  AARCH64_ONLY(                                      \
-      static_field(StubRoutines,                _updateBytesCRC32,                             address)                              \
-     static_field(StubRoutines,                _crc_table_adr,                                address)                               \
-  )                                                  \
                                                                                                                                      \
   /***************************************/                                                                                          \
   /* PcDesc and other compiled code info */                                                                                          \
