diff --git a/apps/desktop/desktop_native/index.js b/apps/desktop/desktop_native/index.js
index 1cf7ea943..b12a34bc3 100644
--- a/apps/desktop/desktop_native/index.js
+++ b/apps/desktop/desktop_native/index.js
@@ -151,13 +151,13 @@ switch (platform) {
     switch (arch) {
       case 'x64':
         localFileExisted = existsSync(
-          join(__dirname, 'desktop_native.linux-x64-musl.node')
+          join(__dirname, 'desktop_native.linux-x64-gnu.node')
         )
         try {
           if (localFileExisted) {
-            nativeBinding = require('./desktop_native.linux-x64-musl.node')
+            nativeBinding = require('./desktop_native.linux-x64-gnu.node')
           } else {
-            nativeBinding = require('@bitwarden/desktop-native-linux-x64-musl')
+            nativeBinding = require('@bitwarden/desktop-native-linux-x64-gnu')
           }
         } catch (e) {
           loadError = e
@@ -165,13 +165,13 @@ switch (platform) {
         break
       case 'arm64':
         localFileExisted = existsSync(
-          join(__dirname, 'desktop_native.linux-arm64-musl.node')
+          join(__dirname, 'desktop_native.linux-arm64-gnu.node')
         )
         try {
           if (localFileExisted) {
-            nativeBinding = require('./desktop_native.linux-arm64-musl.node')
+            nativeBinding = require('./desktop_native.linux-arm64-gnu.node')
           } else {
-            nativeBinding = require('@bitwarden/desktop-native-linux-arm64-musl')
+            nativeBinding = require('@bitwarden/desktop-native-linux-arm64-gnu')
           }
         } catch (e) {
           loadError = e
