diff -urN a/tests/backref-multibyte-slow b/tests/backref-multibyte-slow
--- a/tests/backref-multibyte-slow	2010-08-13 23:36:34.000000000 +0000
+++ b/tests/backref-multibyte-slow	2011-01-05 04:32:14.669176000 +0000
@@ -12,7 +12,7 @@
 
 for LOC in en_US.UTF-8; do
   out=out-$LOC
-  LC_ALL=$LOC timeout 5s grep -E '^([a-z]).\1$' in > $out 2>&1
+  LC_ALL=$LOC timeout 120s grep -E '^([a-z]).\1$' in > $out 2>&1
   test $? = 0 || fail=1
   compare $out in || fail=1
 done
