--- src/animation/animation.c.orig	2007-12-01 02:22:12.008115484 +0100
+++ src/animation/animation.c	2007-12-01 02:27:41.541651382 +0100
@@ -1148,7 +1148,7 @@
     { "horizontal_folds_zoom_to_taskbar", "bool", 0, 0, 0 },
     { "magic_lamp_moving_end", "bool", 0, 0, 0 },
     { "magic_lamp_grid_res", "int", "<min>4</min>", 0, 0 },
-    { "magic_lamp_max_waves", "int", "<min>3</min>", 0, 0 },
+    { "magic_lamp_max_waves", "int", "<min>0</min>", 0, 0 },
     { "magic_lamp_amp_min", "float", "<min>200</min>", 0, 0 },
     { "magic_lamp_amp_max", "float", "<min>200</min>", 0, 0 },
     { "magic_lamp_open_start_width", "int", "<min>0</min>", 0, 0 },
--- metadata/animation.xml.in.orig	2007-12-01 02:25:24.011432269 +0100
+++ metadata/animation.xml.in	2007-12-01 02:27:02.724750938 +0100
@@ -1272,7 +1272,7 @@
 	    <_short>Magic Lamp Max Waves</_short>
 	    <_long>The maximum number of waves for Magic Lamp.</_long>
 	    <default>3</default>
-	    <min>3</min>
+	    <min>0</min>
 	    <max>20</max>
 	  </option>
 	  <option name="magic_lamp_amp_min" type="float">
--- metadata/animation.xml.orig	2007-12-01 02:25:14.584765435 +0100
+++ metadata/animation.xml	2007-12-01 02:35:07.068035601 +0100
@@ -3962,7 +3962,7 @@
 	    <long xml:lang="pt">Número máximo de ondas para Lâmpada Mágica.</long>
 	    <long xml:lang="sv">Den Magiska Lampans maximala antal vågor</long>
 	    <default>3</default>
-	    <min>3</min>
+	    <min>0</min>
 	    <max>20</max>
 	  </option>
 	  <option type="float" name="magic_lamp_amp_min">
