--- Makefile.orig	2013-02-22 23:37:25.911542016 -0600
+++ Makefile	2013-02-22 23:37:49.286029980 -0600
@@ -642,7 +642,7 @@
 .PHONY: cgpt
 cgpt: ${CGPT}
 
-${CGPT}: LDFLAGS += -static
+#${CGPT}: LDFLAGS += -static
 ${CGPT}: LDLIBS += -luuid
 
 ${CGPT}: ${CGPT_OBJS} ${LIBS}
@@ -662,7 +662,7 @@
 ${BUILD}/utility/%: INCLUDES += -Ihost/include -Iutility/include
 
 # Utilities for auto-update toolkits must be statically linked.
-${UTIL_BINS_STATIC}: LDFLAGS += -static
+#${UTIL_BINS_STATIC}: LDFLAGS += -static
 
 .PHONY: utils
 utils: ${UTIL_BINS} ${UTIL_SCRIPTS} ${UTIL_SBINS}
