--- nxloadconfig~	2007-10-19 18:32:40.000000000 +0200
+++ nxloadconfig	2007-10-19 18:33:39.000000000 +0200
@@ -306,9 +306,11 @@
 
 if [ "$ENABLE_KDE_CUPS" = "1" ]
 then
-	[ -z "$KDE_PRINTRC" -a -n "$KDEHOME" ] && KDE_PRINTRC="$KDEHOME/share/config/kdeprintrc"
-	[ -z "$KDE_PRINTRC" ] && which kde-config >/dev/null 2>&1 && KDE_PRINTRC=$(kde-config --localprefix 2>/dev/null)"/share/config/kdeprintrc"
-	[ -z "$KDE_PRINTRC" -o ! -r "$KDE_PRINTRC" ] && KDE_PRINTRC="$HOME/.kde/share/config/kdeprintrc"
+	if [ -f /opt/kde/bin/kde-config ]; then
+		[ -z "$KDE_PRINTRC" -a -n "$KDEHOME" ] && KDE_PRINTRC="$KDEHOME/share/config/kdeprintrc"
+		[ -z "$KDE_PRINTRC" ] && KDE_PRINTRC=$(/opt/kde/bin/kde-config --localprefix)"/share/config/kdeprintrc"
+		[ -z "$KDE_PRINTRC" -o ! -r "$KDE_PRINTRC" ] && KDE_PRINTRC="$HOME/.kde/share/config/kdeprintrc"
+	fi
 fi
 
 #########################################################################
