diff --git a/CMakeLists.txt b/CMakeLists.txt
index ef11396..dd10886 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,6 +44,7 @@ set(gRPC_INSTALL_CMAKEDIR "lib/cmake/${PACKAGE_NAME}" CACHE STRING "Installation
 set(gRPC_INSTALL_SHAREDIR "share/grpc" CACHE STRING "Installation directory for root certificates")
 
 # Options
+option(gRPC_BUILD_CLI "Build cli" ON)
 option(gRPC_BUILD_TESTS "Build tests" OFF)
 option(gRPC_BUILD_CODEGEN "Build codegen" ON)
 option(gRPC_BUILD_CSHARP_EXT "Build C# extensions" ON)
@@ -84,7 +85,7 @@ set_property(CACHE gRPC_PROTOBUF_PROVIDER PROPERTY STRINGS "module" "package")
 set(gRPC_PROTOBUF_PACKAGE_TYPE "" CACHE STRING "Algorithm for searching protobuf package")
 set_property(CACHE gRPC_PROTOBUF_PACKAGE_TYPE PROPERTY STRINGS "CONFIG" "MODULE")
 
-if(gRPC_BUILD_TESTS)
+if(gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
   set(gRPC_GFLAGS_PROVIDER "module" CACHE STRING "Provider of gflags library")
   set_property(CACHE gRPC_GFLAGS_PROVIDER PROPERTY STRINGS "module" "package")
 
@@ -4278,7 +4279,7 @@ target_link_libraries(b64_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 
 add_executable(bad_server_response_test
   test/core/end2end/bad_server_response_test.cc
@@ -4401,7 +4402,7 @@ target_link_libraries(bin_decoder_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 
 add_executable(bin_encoder_test
   test/core/transport/chttp2/bin_encoder_test.cc
@@ -5167,7 +5168,7 @@ target_link_libraries(fake_resolver_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 
 add_executable(fake_transport_security_test
   test/core/tsi/fake_transport_security_test.cc
@@ -10762,7 +10763,7 @@ target_link_libraries(global_config_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 
 add_executable(grpc_cli
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/reflection/v1alpha/reflection.pb.cc
