diff --git a/CMakeLists.txt b/CMakeLists.txt
index f7d6191..54ce048 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,6 +44,7 @@ set(gRPC_INSTALL_CMAKEDIR "lib/cmake/${PACKAGE_NAME}" CACHE STRING "Installation
 set(gRPC_INSTALL_SHAREDIR "share/grpc" CACHE STRING "Installation directory for root certificates")
 
 # Options
+option(gRPC_BUILD_CLI "Build cli" ON)
 option(gRPC_BUILD_TESTS "Build tests" OFF)
 option(gRPC_BUILD_CODEGEN "Build codegen" ON)
 option(gRPC_BUILD_CSHARP_EXT "Build C# extensions" ON)
@@ -84,7 +85,7 @@ set_property(CACHE gRPC_PROTOBUF_PROVIDER PROPERTY STRINGS "module" "package")
 set(gRPC_PROTOBUF_PACKAGE_TYPE "" CACHE STRING "Algorithm for searching protobuf package")
 set_property(CACHE gRPC_PROTOBUF_PACKAGE_TYPE PROPERTY STRINGS "CONFIG" "MODULE")
 
-if(gRPC_BUILD_TESTS)
+if(gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
   set(gRPC_GFLAGS_PROVIDER "module" CACHE STRING "Provider of gflags library")
   set_property(CACHE gRPC_GFLAGS_PROVIDER PROPERTY STRINGS "module" "package")
 
@@ -4277,7 +4278,7 @@ if(gRPC_INSTALL)
 endif()
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 
 if(gRPC_BUILD_CODEGEN)
 add_library(grpc++_proto_reflection_desc_db
@@ -4411,7 +4412,7 @@ if(gRPC_INSTALL)
 endif()
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 
 add_library(grpc++_test_config
   test/cpp/util/test_config_cc.cc
@@ -5268,7 +5269,7 @@ target_link_libraries(grpc_benchmark
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 
 if(gRPC_BUILD_CODEGEN)
 add_library(grpc_cli_libs
@@ -5321,6 +5322,7 @@ target_link_libraries(grpc_cli_libs
   grpc++_proto_reflection_desc_db
   grpc++
   grpc
+  gflags
   ${_gRPC_GFLAGS_LIBRARIES}
 )
 
@@ -13003,7 +13005,7 @@ target_link_libraries(grpc_alts_credentials_options_test
 
 
 endif()
-if(gRPC_BUILD_TESTS)
+if(gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 
 add_executable(grpc_cli
   test/cpp/util/grpc_cli.cc
