diff --git a/src/core/lib/promise/detail/basic_seq.h b/src/core/lib/promise/detail/basic_seq.h
index 72b91f9..a7be768 100644
--- a/src/core/lib/promise/detail/basic_seq.h
+++ b/src/core/lib/promise/detail/basic_seq.h
@@ -16,6 +16,7 @@
 #define GRPC_CORE_LIB_PROMISE_DETAIL_BASIC_SEQ_H
 
 #include <grpc/impl/codegen/port_platform.h>
+#include <assert.h>
 
 #include "absl/types/variant.h"
 #include "absl/utility/utility.h"
