diff -ruN a/fs/aufs/debug.c b/fs/aufs/debug.c
--- a/fs/aufs/debug.c	2015-05-01 08:27:35.903754731 +0000
+++ b/fs/aufs/debug.c	2015-05-01 08:30:16.164156908 +0000
@@ -168,7 +168,7 @@
 	struct dentry *d;
 
 	spin_lock(&inode->i_lock);
-	hlist_for_each_entry(d, &inode->i_dentry, d_alias)
+	hlist_for_each_entry(d, &inode->i_dentry, d_u.d_alias)
 		au_dpri_dentry(d);
 	spin_unlock(&inode->i_lock);
 }
diff -ruN a/fs/aufs/export.c b/fs/aufs/export.c
--- a/fs/aufs/export.c	2015-05-01 08:27:35.908755087 +0000
+++ b/fs/aufs/export.c	2015-05-01 08:34:17.023328460 +0000
@@ -243,7 +243,7 @@
 		dentry = d_find_alias(inode);
 	else {
 		spin_lock(&inode->i_lock);
-		hlist_for_each_entry(d, &inode->i_dentry, d_alias) {
+		hlist_for_each_entry(d, &inode->i_dentry, d_u.d_alias) {
 			spin_lock(&d->d_lock);
 			if (!au_test_anon(d)
 			    && d->d_parent->d_inode->i_ino == dir_ino) {
diff -ruN a/fs/aufs/hnotify.c b/fs/aufs/hnotify.c
--- a/fs/aufs/hnotify.c	2015-05-01 08:27:35.914755514 +0000
+++ b/fs/aufs/hnotify.c	2015-05-01 08:32:32.574871642 +0000
@@ -211,7 +211,7 @@
 		AuDebugOn(!name);
 		au_iigen_dec(inode);
 		spin_lock(&inode->i_lock);
-		hlist_for_each_entry(d, &inode->i_dentry, d_alias) {
+		hlist_for_each_entry(d, &inode->i_dentry, d_u.d_alias) {
 			spin_lock(&d->d_lock);
 			dname = &d->d_name;
 			if (dname->len != nlen
@@ -378,7 +378,7 @@
 
 	dentry = NULL;
 	spin_lock(&parent->d_lock);
-	list_for_each_entry(d, &parent->d_subdirs, d_u.d_child) {
+	list_for_each_entry(d, &parent->d_subdirs, d_child) {
 		/* AuDbg("%pd\n", d); */
 		spin_lock_nested(&d->d_lock, DENTRY_D_LOCK_NESTED);
 		dname = &d->d_name;
