diff -urNad ld/Makefile.am ld_/Makefile.am
--- ld/Makefile.am	2010-11-03 16:22:01.000000000 +1300
+++ ld_/Makefile.am	2011-06-14 22:48:48.942861545 +1200
@@ -37,7 +37,7 @@
 # We put the scripts in the directory $(scriptdir)/ldscripts.
 # We can't put the scripts in $(datadir) because the SEARCH_DIR
 # directives need to be different for native and cross linkers.
-scriptdir = $(tooldir)/lib
+scriptdir = $(libdir)
 
 EMUL = @EMUL@
 EMULATION_OFILES = @EMULATION_OFILES@
diff -urNad ld/Makefile.in ld_/Makefile.in
--- ld/Makefile.in	2010-11-05 23:34:22.000000000 +1300
+++ ld_/Makefile.in	2011-06-14 22:48:52.065772461 +1200
@@ -360,7 +360,7 @@
 # We put the scripts in the directory $(scriptdir)/ldscripts.
 # We can't put the scripts in $(datadir) because the SEARCH_DIR
 # directives need to be different for native and cross linkers.
-scriptdir = $(tooldir)/lib
+scriptdir = $(libdir)
 BASEDIR = $(srcdir)/..
 BFDDIR = $(BASEDIR)/bfd
 INCDIR = $(BASEDIR)/include
