diff -urN a/arch/arm/mach-exynos/mach-odroid-xu.c b/arch/arm/mach-exynos/mach-odroid-xu.c
--- a/arch/arm/mach-exynos/mach-odroid-xu.c	2013-12-28 16:16:42.000000000 -0700
+++ b/arch/arm/mach-exynos/mach-odroid-xu.c	2014-01-06 18:11:02.172586515 -0700
@@ -165,7 +165,7 @@
 #ifdef CONFIG_VIDEO_SAMSUNG_MEMSIZE_FIMD
 			CONFIG_VIDEO_SAMSUNG_MEMSIZE_FIMD +
 #endif
-			128) * SZ_1M,
+			1) * SZ_1M,
 			{ .alignment = SZ_1M },
 		},
 #ifdef CONFIG_EXYNOS_CONTENT_PATH_PROTECTION
