diff -urb src.old/lisp/GNUmakefile src/lisp/GNUmakefile
--- src.old/lisp/GNUmakefile	2008-01-18 15:31:50.000000000 +0100
+++ src/lisp/GNUmakefile	2008-09-04 14:37:22.000000000 +0200
@@ -30,10 +30,12 @@
 	$(NM) lisp | grep -v " [F] " >> ,lisp.nm
 	mv ,lisp.nm lisp.nm
 
-lisp: version.c ${OBJS} version
+version.o: version.c version
 	echo '1 + ' `cat version` | bc > ,version
 	mv ,version version
 	$(CC) ${CFLAGS} -DVERSION=`cat version` -c $<
+
+lisp: ${OBJS} version.o
 	$(CC) -g ${OS_LINK_FLAGS} -o ,lisp \
 		${OBJS} version.o \
 		${OS_LIBS} -lm
