diff -ur src.sic/lisp/x86-arch.c src/lisp/x86-arch.c
--- src.sic/lisp/x86-arch.c	2005-10-12 00:22:16.000000000 +0200
+++ src/lisp/x86-arch.c	2005-10-12 00:25:00.000000000 +0200
@@ -55,7 +55,7 @@
       vlen = *(char*) SC_PC(context)++;
       /* Skip lisp error arg data bytes */
       while(vlen-- > 0) 
-	((char*) SC_PC(context))++;
+	SC_PC(context)++;
       break;
 
     case trap_Breakpoint:
@@ -265,7 +265,7 @@
 #if 0
       fprintf(stderr,"*C break\n");
 #endif
-      (char*) SC_PC(context) -= 1;
+      SC_PC(context) -= 1;
 
       handle_breakpoint(signal, code, context);
 #if 0
@@ -274,7 +274,7 @@
       break;
       
     case trap_FunctionEndBreakpoint:
-      (char*) SC_PC(context) -= 1;
+      SC_PC(context) -= 1;
       SC_PC(context) = (int) handle_function_end_breakpoint(signal, code, context);
       break;
       
