diff -wbBur anyfs-tools-0.84.11/anyfs/inode.c anyfs-tools-0.84.11.my/anyfs/inode.c
--- anyfs-tools-0.84.11/anyfs/inode.c	2006-12-16 15:41:19.000000000 +0300
+++ anyfs-tools-0.84.11.my/anyfs/inode.c	2007-02-19 15:01:08.000000000 +0300
@@ -181,7 +181,7 @@
 
 	inode->i_nlink = info->si_inode_table[inode->i_ino].i_links_count;
 
-	inode->i_blksize = info->si_blocksize;
+//	inode->i_blksize = info->si_blocksize;
 	inode->i_blocks = 0;
 	
 	if ( S_ISREG(inode->i_mode) )
diff -wbBur anyfs-tools-0.84.11/anyfs/namei.c anyfs-tools-0.84.11.my/anyfs/namei.c
--- anyfs-tools-0.84.11/anyfs/namei.c	2006-12-16 15:42:03.000000000 +0300
+++ anyfs-tools-0.84.11.my/anyfs/namei.c	2007-02-19 15:05:24.000000000 +0300
@@ -288,7 +288,7 @@
 	inode->i_uid = current->fsuid;
 	inode->i_gid = (dir->i_mode & S_ISGID) ? dir->i_gid : current->fsgid;
 	inode->i_mtime = inode->i_atime = inode->i_ctime = CURRENT_TIME_SEC;
-	inode->i_blocks = inode->i_blksize = 0;
+	inode->i_blocks = 0;
 	inode->i_mode = mode;
 	inode->i_ino = ino;
 	
diff -wbBur anyfs-tools-0.84.11/src/build_xfs/build_xfs.c anyfs-tools-0.84.11.my/src/build_xfs/build_xfs.c
--- anyfs-tools-0.84.11/src/build_xfs/build_xfs.c	2006-12-13 14:59:14.000000000 +0300
+++ anyfs-tools-0.84.11.my/src/build_xfs/build_xfs.c	2007-02-19 15:04:41.000000000 +0300
@@ -27,6 +27,8 @@
 
 #include <math.h>
 
+#define XFS_DFL_RTEXTSIZE       (64 * 1024)
+
 #undef _
 #undef bindtextdomain
 #undef textdomain
