From e0f7380821c868fa0f7cc0621368cfe2ce4d9d4d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sun, 18 Aug 2024 15:50:56 +0000
Subject: [PATCH] Set icon for the window

---
 src/main.rs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/main.rs b/src/main.rs
index cfe067f..b8d0bdc 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -64,6 +64,8 @@ fn main() {
     gio::resources_register(&res);
 
     glib::set_application_name(config::NAME);
+    gtk::Window::set_default_icon_name(config::APP_ID);
+
 
     // Run app itself
     FrgApplication::run();
-- 
GitLab

