index 17f6d3f..cc01dcd 100644 (file)


--- a/tests/mini-loss-time.c
+++ b/tests/mini-loss-time.c
@@ -267,6 +267,7 @@ static void start (int server_packet, int client_packet)
     {
       /* parent */
       server (fd[0], server_packet);
+      kill(child, SIGTERM);
     }
   else 
     {
--- a/src/libopts/m4/libopts.m4
+++ b/src/libopts/m4/libopts.m4
@@ -487,9 +487,9 @@ AC_DEFUN([LIBOPTS_CHECK_COMMON],[
     fi])
 
   AC_ARG_ENABLE([libopts-install],
-    AC_HELP_STRING([--disable-libopts-install],
-       [Do not install libopts with client installation]))
-  AM_CONDITIONAL([INSTALL_LIBOPTS],[test "X${enable_libopts_install}" != Xno])
+    AC_HELP_STRING([--enable-libopts-install],
+       [Install libopts with client installation]))
+  AM_CONDITIONAL([INSTALL_LIBOPTS],[test "X${enable_libopts_install}" = Xyes])
 
   [if test -z "${NEED_LIBOPTS_DIR}" ; then]
      AC_MSG_CHECKING([whether autoopts-config can be found])
