--- client/mount.cifs.c.orig	2010-03-03 13:42:02.143936727 +1000
+++ client/mount.cifs.c	2010-03-04 01:53:22.752879004 +1000
@@ -1563,7 +1563,7 @@
 		}
 	}

-	if (addr->ai_addr->sa_family == AF_INET6 && addr6->sin6_scope_id) {
+	if (addr && addr->ai_addr->sa_family == AF_INET6 && addr6->sin6_scope_id) {
 		strlcat(options, "%", options_size);
 		current_len = strnlen(options, options_size);
 		optionstail = options + current_len; 
