diff -Naur samba-3.0.23-orig/source/Makefile.in samba-3.0.23/source/Makefile.in
--- samba-3.0.23-orig/source/Makefile.in	2006-07-10 09:27:55.000000000 -0700
+++ samba-3.0.23/source/Makefile.in	2006-07-11 14:34:39.000000000 -0700
@@ -138,14 +138,10 @@
 
 ROOT_SBIN_PROGS = @CIFSMOUNT_PROGS@
 
-BIN_PROGS1 = bin/smbclient@EXEEXT@ bin/net@EXEEXT@ bin/smbspool@EXEEXT@ \
-	bin/testparm@EXEEXT@ bin/smbstatus@EXEEXT@ bin/smbget@EXEEXT@
-BIN_PROGS2 = bin/smbcontrol@EXEEXT@ bin/smbtree@EXEEXT@ bin/tdbbackup@EXEEXT@ \
-	bin/nmblookup@EXEEXT@ bin/pdbedit@EXEEXT@ bin/tdbdump@EXEEXT@ \
-	bin/tdbtool@EXEEXT@
-BIN_PROGS3 = bin/smbpasswd@EXEEXT@ bin/rpcclient@EXEEXT@ bin/smbcacls@EXEEXT@ \
-	bin/profiles@EXEEXT@ bin/ntlm_auth@EXEEXT@ bin/sharesec@EXEEXT@ \
-	bin/smbcquotas@EXEEXT@ bin/eventlogadm@EXEEXT@
+BIN_PROGS1 = bin/testparm@EXEEXT@ bin/smbstatus@EXEEXT@
+BIN_PROGS2 = bin/smbcontrol@EXEEXT@ bin/tdbbackup@EXEEXT@ bin/pdbedit@EXEEXT@
+BIN_PROGS3 = bin/smbpasswd@EXEEXT@ bin/profiles@EXEEXT@ bin/ntlm_auth@EXEEXT@ \
+	bin/eventlogadm@EXEEXT@
 BIN_PROGS4 = bin/ldbedit@EXEEXT@ bin/ldbsearch@EXEEXT@ bin/ldbadd@EXEEXT@ \
 	bin/ldbdel@EXEEXT@ bin/ldbmodify@EXEEXT@ bin/ldbrename@EXEEXT@
 
@@ -160,7 +156,7 @@
 
 PAM_MODULES = @PAM_MODULES@
 
-SCRIPTS = $(srcdir)/script/smbtar $(builddir)/script/findsmb
+SCRIPTS = $(builddir)/script/findsmb
 
 VFS_MODULES = @VFS_MODULES@
 PDB_MODULES = @PDB_MODULES@
