--- startkde.cmake.orig	2009-01-15 14:24:44.000000000 +0100
+++ startkde.cmake	2009-01-15 14:33:08.000000000 +0100
@@ -34,22 +34,6 @@
 MALLOC_CHECK_=2 
 export MALLOC_CHECK_
 
-# in case we have been started with full pathname spec without being in PATH
-bindir=`echo "$0" | sed -n 's,^\(/.*\)/[^/][^/]*$,\1,p'`
-if [ -n "$bindir" ]; then
-  qbindir=`$bindir/kde4-config --qt-binaries`
-  if [ -n "$qbindir" ]; then
-    case $PATH in
-      $qbindir|$qbindir:*|*:$qbindir|*:$qbindir:*) ;;
-      *) PATH=$qbindir:$PATH; export PATH;;
-    esac
-  fi
-  case $PATH in
-    $bindir|$bindir:*|*:$bindir|*:$bindir:*) ;;
-    *) PATH=$bindir:$PATH; export PATH;;
-  esac
-fi
-
 # Boot sequence:
 #
 # kdeinit is used to fork off processes which improves memory usage
@@ -206,7 +190,7 @@
 # For anything else (that doesn't set env vars, or that needs a window manager),
 # better use the Autostart folder.
 
-libpath=`kde4-config --path lib | tr : '\n'`
+libpath=`kde4-config --path lib | tr : '\n'`$(echo -e '\n/etc/kde/lib/')
 
 for prefix in `echo "$libpath" | sed -n -e 's,/lib[^/]*/,/env/,p'`; do
   for file in "$prefix"*.sh; do
