--- kdm/kfrontend/genkdmconf.c	2008-02-13 09:40:49.000000000 +0000
+++ kdm/kfrontend/genkdmconf.c	2008-05-16 12:47:36.000000000 +0000
@@ -662,7 +662,7 @@
 "    [ -d /etc/zsh ] && zdir=/etc/zsh || zdir=/etc\n"
 "    zhome=${ZDOTDIR:-$HOME}\n"
 "    # zshenv is always sourced automatically.\n"
-"    [ -f $zdir/zprofile ] && . $zdir/zprofile\n"
+"    [ -f /etc/profile ] && . /etc/profile\n"
 "    [ -f $zhome/.zprofile ] && . $zhome/.zprofile\n"
 "    [ -f $zdir/zlogin ] && . $zdir/zlogin\n"
 "    [ -f $zhome/.zlogin ] && . $zhome/.zlogin\n"
