--- a/powerdevil/daemon/backends/upower/backlighthelper.cpp.orig	2012-01-11 17:29:04.380000890 +0000
+++ b/powerdevil/daemon/backends/upower/backlighthelper.cpp	2012-01-11 17:27:10.590001246 +0000
@@ -37,7 +37,7 @@
     interfaces << "nv_backlight" << "intel_backlight" << "radeon_bl" << "mbp_backlight"
                << "asus_laptop" << "toshiba" << "eeepc" << "thinkpad_screen" << "acpi_video1"
                << "acpi_video0" << "apple_backlight" << "fujitsu-laptop" << "samsung"
-               << "nvidia_backlight" << "dell_backlight" << "sony"
+               << "nvidia_backlight" << "dell_backlight" << "sony" << "pwm-backlight"
                ;
 
     QDir dir;
