--- qgo-1.5.4/src/matrix.cpp.orig	2008-02-15 16:47:04.000000000 -0500
+++ qgo-1.5.4/src/matrix.cpp	2008-02-15 16:47:58.000000000 -0500
@@ -5,7 +5,10 @@
 #include "matrix.h"
 #include <stdlib.h>
 #ifndef NO_DEBUG
-#include <iostream.h>
+#include <iostream>
+
+using namespace std;
+
 #endif
 
 Matrix::Matrix(int s)
--- qgo-1.5.4/src/tree.cpp.orig	2008-02-15 17:01:49.000000000 -0500
+++ qgo-1.5.4/src/tree.cpp	2008-02-15 17:03:06.000000000 -0500
@@ -5,7 +5,10 @@
 #include "tree.h"
 #include "move.h"
 #include "qgo.h"
-#include <iostream.h>
+#include <iostream>
+
+using namespace std;
+
 #include <qptrstack.h>
 
 Tree::Tree(int board_size)
