diff -up qtscriptgenerator-src-0.1.0/generator/build_all.txt.no_phonon qtscriptgenerator-src-0.1.0/generator/build_all.txt
--- qtscriptgenerator-src-0.1.0/generator/build_all.txt.no_phonon	2010-10-26 02:53:53.000000000 -0500
+++ qtscriptgenerator-src-0.1.0/generator/build_all.txt	2010-12-21 12:43:23.717989854 -0600
@@ -6,7 +6,6 @@
   <load-typesystem name="typesystem_svg.xml" generate="yes" />
   <load-typesystem name="typesystem_network.xml" generate="yes" />
   <load-typesystem name="typesystem_xml.xml" generate="yes" />
-  <load-typesystem name="typesystem_phonon.xml" generate="yes" />
   <load-typesystem name="typesystem_webkit.xml" generate="yes" />
   <load-typesystem name="typesystem_xmlpatterns.xml" generate="yes" />
   <load-typesystem name="typesystem_uitools.xml" generate="yes" />
diff -up qtscriptgenerator-src-0.1.0/generator/generate.sh.no_phonon qtscriptgenerator-src-0.1.0/generator/generate.sh
--- qtscriptgenerator-src-0.1.0/generator/generate.sh.no_phonon	2010-10-26 02:53:53.000000000 -0500
+++ qtscriptgenerator-src-0.1.0/generator/generate.sh	2010-12-21 12:43:23.717989854 -0600
@@ -12,8 +12,6 @@ xsltproc --stringparam source $PWD/types
 
 xsltproc --stringparam source $PWD/typesystem_sql-qtscript.xml merge.xsl typesystem_sql-common.xml > typesystem_sql.xml
 
-xsltproc --stringparam source $PWD/typesystem_phonon-qtscript.xml merge.xsl typesystem_phonon-common.xml > typesystem_phonon.xml
-
 xsltproc --stringparam source $PWD/typesystem_webkit-qtscript.xml merge.xsl typesystem_webkit-common.xml > typesystem_webkit.xml
 
 xsltproc --stringparam source $PWD/typesystem_xmlpatterns-qtscript.xml merge.xsl typesystem_xmlpatterns-common.xml > typesystem_xmlpatterns.xml
diff -up qtscriptgenerator-src-0.1.0/qtbindings/qtbindings.pro.no_phonon qtscriptgenerator-src-0.1.0/qtbindings/qtbindings.pro
--- qtscriptgenerator-src-0.1.0/qtbindings/qtbindings.pro.no_phonon	2010-10-26 02:53:53.000000000 -0500
+++ qtscriptgenerator-src-0.1.0/qtbindings/qtbindings.pro	2010-12-21 12:45:01.010979847 -0600
@@ -6,7 +6,6 @@ SUBDIRS = qtscript_core \
 	  qtscript_sql \
 	  qtscript_svg \
 	  qtscript_xml \
-	  qtscript_phonon \
 	  qtscript_webkit \
 	  qtscript_xmlpatterns \
 	  qtscript_uitools \
