diff -up qtscriptgenerator-src-0.1.0/generator/qtscript_masterinclude.h.kde_phonon qtscriptgenerator-src-0.1.0/generator/qtscript_masterinclude.h
--- qtscriptgenerator-src-0.1.0/generator/qtscript_masterinclude.h.kde_phonon	2010-10-26 02:53:53.000000000 -0500
+++ qtscriptgenerator-src-0.1.0/generator/qtscript_masterinclude.h	2010-12-11 17:00:08.456338081 -0600
@@ -49,17 +49,46 @@
 
 #include <QtUiTools/QtUiTools>
 
-#ifndef QT_NO_XMLPATTERNS
-#  include <QtXmlPatterns/QtXmlPatterns>
-#endif
-
-#ifndef QT_NO_WEBKIT
-#  include <QtWebKit/QtWebKit>
-#endif
-
-#ifndef QT_NO_PHONON
-#  include <phonon/phonon>
-#endif
+#include <QtXmlPatterns/QtXmlPatterns>                                                              
+
+#include <QtWebKit/QtWebKit>                                                                        
+
+#include "phonon/abstractaudiooutput.h"
+#include "phonon/abstractmediastream.h"
+#include "phonon/abstractvideooutput.h"
+#include "phonon/addoninterface.h"
+#include "phonon/audiodataoutput.h"
+#include "phonon/audiodataoutputinterface.h"
+#include "phonon/audiooutput.h"
+#include "phonon/audiooutputinterface.h"
+#include "phonon/backendcapabilities.h"
+#include "phonon/backendinterface.h"
+#include "phonon/effect.h"
+#include "phonon/effectinterface.h"
+#include "phonon/effectparameter.h"
+#include "phonon/effectwidget.h"
+#include "phonon/globalconfig.h"
+#include "phonon/mediacontroller.h"
+#include "phonon/medianode.h"
+#include "phonon/mediaobject.h"
+#include "phonon/mediaobjectinterface.h"
+#include "phonon/mediasource.h"
+#include "phonon/objectdescription.h"
+#include "phonon/objectdescriptionmodel.h"
+#include "phonon/path.h"
+#include "phonon/phonon_export.h"
+#include "phonon/phonondefs.h"
+#include "phonon/phononnamespace.h"
+#include "phonon/platformplugin.h"
+#include "phonon/pulsesupport.h"
+#include "phonon/seekslider.h"
+#include "phonon/streaminterface.h"
+#include "phonon/videoplayer.h"
+#include "phonon/videowidget.h"
+#include "phonon/videowidgetinterface.h"
+#include "phonon/volumefadereffect.h"
+#include "phonon/volumefaderinterface.h"
+#include "phonon/volumeslider.h"
 
 #include "../qtbindings/qtscript_core/qtscriptconcurrent.h"
 
