diff --git a/config/ltdl.m4 b/config/ltdl.m4
index ea76f4d..2f1cbfe 100644
--- a/config/ltdl.m4
+++ b/config/ltdl.m4
@@ -162,6 +162,8 @@ m4_defun([_LTDL_INSTALLABLE],
   fi
 fi
 
+enable_ltdl_install=no
+
 # If configure.ac declared an installable ltdl, and the user didn't override
 # with --disable-ltdl-install, we will install the shipped libltdl.
 case $enable_ltdl_install in
diff --git a/ompi/debuggers/Makefile.am b/ompi/debuggers/Makefile.am
index 2adf3fd..5b22a91 100644
--- a/ompi/debuggers/Makefile.am
+++ b/ompi/debuggers/Makefile.am
@@ -46,8 +46,7 @@ headers = \
 # Simple checks to ensure that the DSOs are functional
 
 dlopen_test_SOURCES = dlopen_test.c
-dlopen_test_CPPFLAGS = -I$(top_srcdir)/opal/libltdl
-dlopen_test_LDADD = $(top_builddir)/opal/libltdl/libltdlc.la
+dlopen_test_LDADD = -lltdl
 
 predefined_gap_test_SOURCES = predefined_gap_test.c
 predefined_gap_test_LDFLAGS = $(WRAPPER_EXTRA_LDFLAGS)
diff --git a/ompi/debuggers/dlopen_test.c b/ompi/debuggers/dlopen_test.c
index 029a4ee..023afb8 100644
--- a/ompi/debuggers/dlopen_test.c
+++ b/ompi/debuggers/dlopen_test.c
@@ -13,7 +13,7 @@
 #include <string.h>
 #include <stdlib.h>
 
-#include "opal/libltdl/ltdl.h"
+#include "ltdl.h"
 
 static int do_test(void);
 
diff --git a/test/support/components.c b/test/support/components.c
index 41c4345..6b4b464 100644
--- a/test/support/components.c
+++ b/test/support/components.c
@@ -24,7 +24,7 @@
 
 #include "opal/constants.h"
 #include "opal/mca/mca.h"
-#include "opal/libltdl/ltdl.h"
+#include "ltdl.h"
 
 #include "components.h"
 
diff --git a/test/support/components.h b/test/support/components.h
index 6db1f0d..fc8dfc5 100644
--- a/test/support/components.h
+++ b/test/support/components.h
@@ -20,7 +20,7 @@
 #ifndef OMPI_SUPPORT_COMPONENTS_H
 #define OMPI_SUPPORT_COMPONENTS_H
 
-#include "opal/libltdl/ltdl.h"
+#include "ltdl.h"
 #include "opal/mca/mca.h"
 
 BEGIN_C_DECLS
