diff -urN a/configure b/configure
--- a/configure	2022-05-28 11:52:23.733073347 -0600
+++ b/configure	2022-05-28 11:58:09.110597387 -0600
@@ -197,12 +197,6 @@
 
 parse_opts $@
 
-if [ "${ARCH}" = "aarch64" ] && [ "${SM_VSN}" != "1.8.5" ]
-then
-  echo "ERROR: SpiderMonkey 60 is known broken on ARM 64 (aarch64). Use 1.8.5 instead."
-  exit 1
-fi
-
 echo "==> configuring couchdb in rel/couchdb.config"
 cat > rel/couchdb.config << EOF
 % Licensed under the Apache License, Version 2.0 (the "License"); you may not
diff -urN a/src/jiffy/rebar.config.script b/src/jiffy/rebar.config.script
--- a/src/jiffy/rebar.config.script	2021-10-05 09:07:14.000000000 -0600
+++ b/src/jiffy/rebar.config.script	2022-05-28 11:59:13.368276228 -0600
@@ -28,7 +28,7 @@
         case os:cmd(FLTO_CHECK) of
             [] ->
                 {port_env, PortEnv} = lists:keyfind(port_env, 1, Config1),
-                NewFlag = {".*", "FLTO_FLAG", "-flto"},
+                NewFlag = {".*", "FLTO_FLAG", ""},
                 NewPortEnv = lists:keyreplace("FLTO_FLAG", 2, PortEnv, NewFlag),
                 lists:keyreplace(port_env, 1, Config1, {port_env, NewPortEnv});
             _ ->
