From 6f3e54d404cfe1ba7d1444e6dfcfd77b102d9287 Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Thu, 22 Apr 2021 19:53:15 +0200
Subject: [PATCH 409/576] nptl: Do not build nptl/tst-pthread-gdb-attach as PIE

---
 nptl/Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/nptl/Makefile b/nptl/Makefile
index a3d1ef8d66..294bb2faa4 100644
--- a/nptl/Makefile
+++ b/nptl/Makefile
@@ -377,6 +377,9 @@ endif
 CFLAGS-tst-pthread-gdb-attach-static.c := $(CFLAGS-printers-tests)
 CPPFLAGS-tst-pthread-gdb-attach-static.c := \
   $(CFLAGS-printers-tests) -DDO_ADD_SYMBOL_FILE=0
+# As of version 9.2, GDB cannot attach properly to PIE programs that
+# were launched with an explicit ld.so invocation.
+tst-pthread-gdb-attach-no-pie = yes
 
 ifeq ($(build-shared),yes)
 tests-printers-libs := $(shared-thread-library)
-- 
2.31.1

