diff --git a/sysdeps/x86_64/multiarch/strcmp.S b/sysdeps/x86_64/multiarch/strcmp.S
index f93c83d..d8aa889 100644
--- a/sysdeps/x86_64/multiarch/strcmp.S
+++ b/sysdeps/x86_64/multiarch/strcmp.S
@@ -105,11 +105,6 @@ ENTRY(__strcasecmp)
	jne	1f
	call	__init_cpu_features
 1:
-#  ifdef HAVE_AVX_SUPPORT
-	leaq	__strcasecmp_avx(%rip), %rax
-	testl	$bit_AVX, __cpu_features+CPUID_OFFSET+index_AVX(%rip)
-	jnz	2f
-#  endif
	leaq	__strcasecmp_sse42(%rip), %rax
	testl	$bit_SSE4_2, __cpu_features+CPUID_OFFSET+index_SSE4_2(%rip)
	jnz	2f
@@ -128,11 +123,6 @@ ENTRY(__strncasecmp)
	jne	1f
	call	__init_cpu_features
 1:
-#  ifdef HAVE_AVX_SUPPORT
-	leaq	__strncasecmp_avx(%rip), %rax
-	testl	$bit_AVX, __cpu_features+CPUID_OFFSET+index_AVX(%rip)
-	jnz	2f
-#  endif
	leaq	__strncasecmp_sse42(%rip), %rax
	testl	$bit_SSE4_2, __cpu_features+CPUID_OFFSET+index_SSE4_2(%rip)
	jnz	2f
@@ -152,19 +142,6 @@ weak_alias (__strncasecmp, strncasecmp)
 # include "strcmp-sse42.S"


-# ifdef HAVE_AVX_SUPPORT
-#  if defined USE_AS_STRCASECMP_L || defined USE_AS_STRNCASECMP_L
-#   define LABEL(l) .L##l##_avx
-#   define GLABEL(l) l##_avx
-#   define USE_AVX 1
-#   undef STRCMP_SSE42
-#   define STRCMP_SSE42 STRCMP_AVX
-#   define SECTION avx
-#   include "strcmp-sse42.S"
-#  endif
-# endif
-
-
 # undef ENTRY
 # define ENTRY(name) \
	.type STRCMP_SSE2, @function; \
