From 0c95ab64cb4ec0d22bb222647d9d20c7b4903e38 Mon Sep 17 00:00:00 2001
From: Andreas Schwab <schwab@redhat.com>
Date: Fri, 7 Oct 2011 09:31:27 +0200
Subject: [PATCH] Horrible workaround for horribly broken software

---
 elf/rtld.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/elf/rtld.c b/elf/rtld.c
index 978c609..8422b9f 100644
--- a/elf/rtld.c
+++ b/elf/rtld.c
@@ -1393,7 +1393,9 @@ of this helper program; chances are you did not intend to run this program.\n\
 	      char *copy = malloc (len);
 	      if (copy == NULL)
 		_dl_fatal_printf ("out of memory\n");
-	      l->l_libname->name = l->l_name = memcpy (copy, dsoname, len);
+	      l->l_libname->name = memcpy (copy, dsoname, len);
+	      if (GLRO(dl_debug_mask))
+		l->l_name = copy;
 	    }
 
 	  /* Add the vDSO to the object list.  */
-- 
1.7.3.4

