Description: Fix builds on architectures without _FPU_EXTENDED
Author: Nobuhiro Iwamatsu <iwamatsu@nigauri.org>

--- a/OpenAL32/Include/alu.h
+++ b/OpenAL32/Include/alu.h
@@ -224,6 +224,29 @@
     return a0*mu*mu2 + a1*mu2 + a2*mu + a3;
 }
 
+#if defined(_FPU_GETCW) && defined(_FPU_SETCW)
+ #ifndef _FPU_EXTENDED
+  #define _FPU_EXTENDED	0
+ #endif
+ #ifndef _FPU_DOUBLE
+  #define _FPU_DOUBLE	0
+ #endif
+ #ifndef _FPU_SINGLE
+  #define _FPU_SINGLE	0
+ #endif
+ #ifndef _FPU_RC_NEAREST
+  #define _FPU_RC_NEAREST	0
+ #endif
+ #ifndef _FPU_RC_DOWN
+  #define _FPU_RC_DOWN	0
+ #endif
+ #ifndef _FPU_RC_UP
+  #define _FPU_RC_UP	0
+ #endif
+ #ifndef _FPU_RC_ZERO
+  #define _FPU_RC_ZERO	0
+ #endif
+#endif
 
 static __inline int SetMixerFPUMode(void)
 {
