Index: openal-soft-1.16.0/CMakeLists.txt
===================================================================
--- openal-soft-1.16.0.orig/CMakeLists.txt
+++ openal-soft-1.16.0/CMakeLists.txt
@@ -679,7 +679,7 @@ ENDIF()
 # Check for ARM Neon support
 OPTION(ALSOFT_REQUIRE_NEON "Require ARM Neon support" OFF)
 CHECK_INCLUDE_FILE(arm_neon.h HAVE_ARM_NEON_H)
-IF(HAVE_ARM_NEON_H)
+IF(HAVE_ARM_NEON_H AND CMAKE_SIZEOF_VOID_P EQUAL 4)	# only 32-bit ARM
     OPTION(ALSOFT_CPUEXT_NEON "Enable ARM Neon support" ON)
     IF(ALSOFT_CPUEXT_NEON)
         SET(HAVE_NEON 1)
