diff --git i/examples/alhrtf.c w/examples/alhrtf.c
index 2be28a91..dfa01e5c 100644
--- i/examples/alhrtf.c
+++ w/examples/alhrtf.c
@@ -30,8 +30,6 @@
 #include <string.h>
 
 #include "SDL_sound.h"
-#include "SDL_audio.h"
-#include "SDL_stdinc.h"
 
 #include "AL/al.h"
 #include "AL/alc.h"
diff --git i/examples/allatency.c w/examples/allatency.c
index a61fb820..5705506a 100644
--- i/examples/allatency.c
+++ w/examples/allatency.c
@@ -28,8 +28,6 @@
 #include <assert.h>
 
 #include "SDL_sound.h"
-#include "SDL_audio.h"
-#include "SDL_stdinc.h"
 
 #include "AL/al.h"
 #include "AL/alext.h"
diff --git i/examples/almultireverb.c w/examples/almultireverb.c
index a90b3368..65e444f5 100644
--- i/examples/almultireverb.c
+++ w/examples/almultireverb.c
@@ -35,8 +35,6 @@
 #include <string.h>
 
 #include "SDL_sound.h"
-#include "SDL_audio.h"
-#include "SDL_stdinc.h"
 
 #include "AL/al.h"
 #include "AL/alc.h"
diff --git i/examples/alplay.c w/examples/alplay.c
index 09ad96b4..01cf769d 100644
--- i/examples/alplay.c
+++ w/examples/alplay.c
@@ -28,8 +28,6 @@
 #include <assert.h>
 
 #include "SDL_sound.h"
-#include "SDL_audio.h"
-#include "SDL_stdinc.h"
 
 #include "AL/al.h"
 
diff --git i/examples/alreverb.c w/examples/alreverb.c
index d789dffe..a6dd141d 100644
--- i/examples/alreverb.c
+++ w/examples/alreverb.c
@@ -28,8 +28,6 @@
 #include <assert.h>
 
 #include "SDL_sound.h"
-#include "SDL_audio.h"
-#include "SDL_stdinc.h"
 
 #include "AL/al.h"
 #include "AL/alc.h"
diff --git i/examples/alstream.c w/examples/alstream.c
index 56505ddb..103492c7 100644
--- i/examples/alstream.c
+++ w/examples/alstream.c
@@ -30,8 +30,6 @@
 #include <assert.h>
 
 #include "SDL_sound.h"
-#include "SDL_audio.h"
-#include "SDL_stdinc.h"
 
 #include "AL/al.h"
 
