diff --git a/cloudinit/util.py b/cloudinit/util.py
index f6efb851a..bef9e40f2 100644
--- a/cloudinit/util.py
+++ b/cloudinit/util.py
@@ -614,7 +614,6 @@ def _get_variant(info):
             "almalinux",
             "alpine",
             "aosc",
-            "arch",
             "azurelinux",
             "centos",
             "cloudlinux",
@@ -649,6 +648,11 @@ def _get_variant(info):
             "sles",
         ):
             variant = "suse"
+        elif linux_dist in (
+            "arch",
+            "archarm",
+        ):
+            variant = "arch"
         else:
             variant = "linux"
     elif system in (
