--- qtbase-everywhere-src-5.10.0-beta4/src/corelib/io/qfilesystemengine_unix.cpp.orig	2017-11-14 12:10:28.156094557 +0000
+++ qtbase-everywhere-src-5.10.0-beta4/src/corelib/io/qfilesystemengine_unix.cpp	2017-11-14 12:10:46.156136206 +0000
@@ -95,7 +95,7 @@
 { return syscall(SYS_renameat2, oldfd, oldpath, newfd, newpath, flags); }
 #  endif
 
-#  if !QT_CONFIG(statx) && defined(SYS_statx) && QT_HAS_INCLUDE(<linux/stat.h>)
+#  if 0
 #    include <linux/stat.h>
 static int statx(int dirfd, const char *pathname, int flag, unsigned mask, struct statx *statxbuf)
 { return syscall(SYS_statx, dirfd, pathname, flag, mask, statxbuf); }
