--- gpsbabel-1.2.8-beta20060219/defs.h	2006-01-24 07:30:53.000000000 -0800
+++ gpsbabel-1.2.8-beta20060219-new/defs.h	2006-03-17 21:31:18.000000000 -0800
@@ -407,7 +407,7 @@
  * All shortname functions take a shortname handle as the first arg.
  * This is an opaque pointer.  Callers must not fondle the contents of it.
  */
-typedef struct short_handle * short_handle;
+typedef struct short_handle_ * short_handle;
 #ifndef DEBUG_MEM 
 char *mkshort (short_handle,  const char *);
 void *mkshort_new_handle(void);
