--- chunkedstorage.hh.orig	2009-04-30 01:18:26.000000000 +0200
+++ chunkedstorage.hh		2009-06-21 21:09:28.000000000 +0200
@@ -9,4 +9,5 @@
 
 #include <vector>
+#include <stdint.h>
 
 #ifdef _MSC_VER
--- config.hh.orig	2009-05-18 20:01:50.000000000 +0200
+++ config.hh		2009-06-21 21:09:36.000000000 +0200
@@ -10,4 +10,5 @@
 #include <QDateTime>
 #include <QKeySequence>
+#include <stdint.h>
 #include "ex.hh"
 
