--- goldendict.pro~	2009-06-21 21:49:42.000000000 +0200
+++ goldendict.pro	2009-06-21 21:49:42.000000000 +0200
@@ -36,6 +36,6 @@
     PREFIX = $$(PREFIX)
-    isEmpty( PREFIX ):PREFIX = /usr/local
-    DEFINES += PROGRAM_DATA_DIR=\\\"$$PREFIX/share/apps/goldendict/\\\"
+    isEmpty( PREFIX ):PREFIX = /usr
+    DEFINES += PROGRAM_DATA_DIR=\\\"$$PREFIX/share/goldendict/\\\"
     target.path = $$PREFIX/bin/
-    locale.path = $$PREFIX/share/apps/goldendict/locale/
+    locale.path = $$PREFIX/share/goldendict/locale/
     locale.files = locale/*.qm
