From fea1bacab954d2f112da8a95756f198530e274af Mon Sep 17 00:00:00 2001
From: "Paul G. Banks" <web@paulbanks.org>
Date: Sun, 16 Aug 2020 10:57:36 +0100
Subject: [PATCH] auth: mech-gssapi - GSSAPI can contain NUL.

---
 src/auth/mech-gssapi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/auth/mech-gssapi.c b/src/auth/mech-gssapi.c
index f29e48da88..966273d388 100644
--- a/src/auth/mech-gssapi.c
+++ b/src/auth/mech-gssapi.c
@@ -735,7 +735,7 @@ mech_gssapi_auth_free(struct auth_request *request)
 const struct mech_module mech_gssapi = {
 	"GSSAPI",
 
-	.flags = 0,
+	.flags = MECH_SEC_ALLOW_NULS,
 	.passdb_need = MECH_PASSDB_NEED_NOTHING,
 
 	mech_gssapi_auth_new,
