diff -urN a/include/configs/ox820.h b/include/configs/ox820.h
--- a/include/configs/ox820.h	2014-01-03 05:19:54.000000000 -0700
+++ b/include/configs/ox820.h	2014-01-24 19:47:31.542099246 -0700
@@ -28,7 +28,7 @@
 #define CONFIG_SYS_DCACHE_OFF
 
 /* clock */
-#define CONFIG_PLLA_FREQ_MHZ		800
+#define CONFIG_PLLA_FREQ_MHZ		850
 #define CONFIG_RPSCLK			6250000
 #define CONFIG_SYS_HZ			1000
 #define CONFIG_SYS_CLK_FREQ		CONFIG_RPSCLK
@@ -63,7 +63,7 @@
 #define NAND_ALE_ADDR_PIN		18
 #define MTDPARTS_DEFAULT		"mtdparts=41000000.nand:" \
 						"14m(boot)," \
-                                                "-(data)"
+						"-(data)"
 #define MTDIDS_DEFAULT			"nand0=41000000.nand"
 
 /* net */
@@ -179,6 +179,8 @@
 /* boot */
 #define CONFIG_IDENT_STRING		" for OXNAS"
 #define CONFIG_MACH_TYPE		MACH_TYPE_OXNAS
+#define CONFIG_OF_SEPARATE
+#define CONFIG_DEFAULT_DEVICE_TREE	"ox820-pogoplug-pro"
 #ifndef CONFIG_SPL_BUILD
 /* Enable devicetree support */
 #define CONFIG_OF_LIBFDT
@@ -194,8 +196,9 @@
 /* memtest works on */
 #define CONFIG_SYS_LOAD_ADDR		(CONFIG_SYS_SDRAM_BASE)
 #define CONFIG_SYS_AUTOLOAD		"no"
+#define CONFIG_CMD_BOOTZ
 
-#define CONFIG_BOOTARGS			"console=ttyS0,115200n8 root=/dev/sda2 ubi.mtd=data,512"
+#define CONFIG_BOOTARGS			"console=ttyS0,115200n8 root=/dev/sda1 rw rootwait " MTDPARTS_DEFAULT
 #define CONFIG_BOOTCOMMAND		"run ideboot"
 #define CONFIG_BOOT_RETRY_TIME		-1
 #define CONFIG_RESET_TO_RETRY		60
@@ -205,18 +208,26 @@
 #define CONFIG_IPADDR			192.168.0.100
 
 #define CONFIG_EXTRA_ENV_SETTINGS	\
-	"updboot=" \
-				"bootp;" \
-				"tftp 60000000 u-boot.img;" \
-				"ide write 60000000 400 400\0" \
-	"console=" CONFIG_DEFAULT_CONSOLE \
-	"bootargs=" CONFIG_DEFAULT_CONSOLE \
-	"mtdids=" MTDIDS_DEFAULT "\0" \
-	"mtdparts=" MTDPARTS_DEFAULT "\0" \
+	"console="      CONFIG_DEFAULT_CONSOLE \
+	"bootargs="	CONFIG_BOOTARGS "\0" \
+	"ideboot=" \
+			"load ide 0:1 0x60500000 /boot/zImage;" \
+			"load ide 0:1 0x67e26000 /boot/dtbs/ox820-pogoplug-pro.dtb;" \
+			"bootz 0x60500000 - 0x67e26000" "\0" \
+	"usbboot=" \
+			"usb start;" \
+			"load usb 0:1 0x60500000 /boot/zImage;" \
+			"load usb 0:1 0x67e26000 /boot/dtbs/ox820-pogoplug-pro.dtb;" \
+			"usb stop;" \
+			"setenv bootargs " \
+				"console=ttyS0,115200n8 " \
+				"root=/dev/sdb1 rw rootwait " \
+				MTDPARTS_DEFAULT ";" \
+			"bootz 0x60500000 - 0x67e26000" "\0" \
 	"safeboot=" \
 			"setexpr.b keypressed '*44200005' '&' 1;" \
 			"if test $keypressed = 0;" \
-				"then run netcon;" \
+				"then run serialcon;" \
 			"fi" "\0" \
 	"serialcon=" \
 			"echo switch to serial console;" \
@@ -324,6 +335,7 @@
 #define CONFIG_FAT_WRITE
 #define CONFIG_CMD_EXT2
 #define CONFIG_CMD_EXT4
+#define CONFIG_CMD_FS_GENERIC
 #ifndef CONFIG_SPL_BUILD
 #define CONFIG_CMD_EXT4_WRITE
 #endif
