diff --git i/tests/meson.build w/tests/meson.build
index 0a6a355..0120c29 100644
--- i/tests/meson.build
+++ w/tests/meson.build
@@ -67,7 +67,9 @@ src_build_path = meson.current_build_dir() / '../src'
 
 if build_gir and host_system == 'linux' and not meson.is_cross_build()
   foreach unit: ['introspection.py']
-    install_data(unit, install_dir: installed_test_bindir)
+    if get_option('installed_tests')
+      install_data(unit, install_dir: installed_test_bindir)
+    endif
 
     wrapper = '@0@.test'.format(unit)
     custom_target(wrapper,
