--- a/src/Makefile.am	2011-12-18 00:11:51.000000000 -0600
+++ b/src/Makefile.am	2011-12-17 03:05:58.000000000 -0600
@@ -22,10 +22,10 @@
 MSM_C2D2_SRCS += msm-exa-c2d2.c c2d/c2d.c msm-video-c2d2.c
 endif
 
-NEON_CFLAGS=-march=armv7-a -mfpu=neon -mfloat-abi=softfp -Wa,-mfpu=neon
+NEON_CFLAGS=$(CFLAGS)
 
-NEON_CCASFLAGS=$(NEON_CFLAGS) -mthumb-interwork
-NEON_ASFLAGS=-k -mcpu=cortex-a8 $(NEON_CCASFLAGS)
+NEON_CCASFLAGS=$(NEON_CFLAGS)
+NEON_ASFLAGS=-k $(NEON_CCASFLAGS)
 
 AM_CFLAGS = @XORG_CFLAGS@ @DRI2_CFLAGS@ $(NEON_CFLAGS) -Wall -Werror
 AM_ASFLAGS = $(NEON_ASFLAGS)
