--- a/src/neon/neon-alignedcopy.c	2011-12-18 02:40:06.000000000 
-0600
+++ b/src/neon/neon-alignedcopy.c	2011-12-18 02:40:22.000000000 
-0600
@@ -33,7 +33,6 @@
 
 #include "xf86.h"
 #include "msm.h"
-#include "neon.h"
 
 /* Copy a rectangle that is 1 byte wide, using unrolled loops to make the function as fast as possible. */
 /* This function may use unaligned pointers at no penalty.                                              */
--- a/src/msm-video-mdp.c	2011-12-18 02:37:09.000000000 -0600
+++ b/src/msm-video-mdp.c	2011-12-18 02:37:24.000000000 -0600
@@ -43,8 +43,6 @@
 #include "msm-render.h"
 #include "msm-video.h"
 
-#include "neon/neon.h"
-
 #if ENABLEHWCODEC
 extern struct cmsghdr *cmptr;
 #endif
