--- xfig.3.2.5/Imakefile.fhs	2006-10-11 01:19:22.000000000 +0200
+++ xfig.3.2.5/Imakefile	2007-04-16 14:04:45.000000000 +0200
@@ -169,8 +169,7 @@
 XFIGLIBDIR = $(LIBDIR)/xfig
 
 XCOMM XFIGDOCDIR tells where the html and pdf documentation should go
-XCOMM XFIGDOCDIR = $(DOCDIR)/xfig
-XFIGDOCDIR = /usr/local/xfig/doc
+XFIGDOCDIR = $(DOCDIR)/xfig
 
 XCOMM MANDIR tells where the standard man pages should go (no need to change it
 XCOMM if you want the man pages installed in the standard place on your system
