diff -up xfig.3.2.5/Imakefile.no-Xaw3d xfig.3.2.5/Imakefile
--- xfig.3.2.5/Imakefile.no-Xaw3d	2007-11-16 10:38:18.000000000 +0100
+++ xfig.3.2.5/Imakefile	2007-11-16 10:38:49.000000000 +0100
@@ -50,8 +50,8 @@ XAPPLOADDIR = /usr/share/X11/app-default
 XCOMM Uncomment the following if you have David Hawkey's Xaw3D version 1.5E which has
 XCOMM some new features, including "Tips", which replace xfig's "help balloons"
 XCOMM NOTE: This is the default for many X systems now.

-XCOMM #define XAW3D1_5E
+#define XAW3D1_5E
 
 #ifdef XAW3D1_5E
     DUSEXAW3D = -DXAW3D -DXAW3D1_5E
-    XAWLIB = -lXaw3d15e
+    XAWLIB = -lXaw3d
 #else
     XAW_SRC = w_menuentry.c SmeCascade.c SmeBSB.c SimpleMenu.c
     XAW_OBJ = w_menuentry.o SmeCascade.o SmeBSB.o SimpleMenu.o
