--- xfig.3.2.5/Imakefile.modularX	2007-04-16 14:27:49.000000000 +0200
+++ xfig.3.2.5/Imakefile	2007-04-16 14:29:18.000000000 +0200
@@ -45,7 +45,7 @@
 XCOMM different tree than the "correct" tree that your X system expects. The usual
 XCOMM purpose of DESTDIR is to test an install process by installing in a benign area.
 
-XCOMM XAPPLOADDIR = /home/user/xfig
+XAPPLOADDIR = /usr/share/X11/app-defaults
 
 XCOMM Uncomment the following definition for XAW3D if you want to use
 XCOMM the 3d Athena Widget Set (highly recommended!)
@@ -166,12 +166,12 @@
 XCOMM XFIGLIBDIR = $(LIBDIR)
 
 XCOMM use this if you want the multi-key data base file in the standard X11 tree
-XFIGLIBDIR = $(LIBDIR)/xfig
+XFIGLIBDIR = /usr/share/xfig
 
 XCOMM XFIGDOCDIR tells where the html and pdf documentation should go
 XFIGDOCDIR = $(DOCDIR)/xfig
 
 XCOMM MANDIR tells where the standard man pages should go (no need to change it
 XCOMM if you want the man pages installed in the standard place on your system
-MANDIR = $(MANSOURCEPATH)$(MANSUFFIX)
+MANDIR = /usr/share/man/man1
 XCOMM MANDIR = /usr/local/xfig/man

 XCOMM If your system doesn't have strstr undefine the following definition
