--- a/charconv.cpp	2006-02-14 16:44:58.000000000 -0500
+++ b/charconv.cpp	2009-06-12 17:59:54.332486433 -0400
@@ -5,6 +5,7 @@
 #include <cstdlib>
 #include <clocale>
 #include <climits>
+#include <cstring>
 #if defined(__STDC_ISO_10646__) || defined(__WIN32__)
 #  include <wchar.h>
 #  define fallback(call) (0)
--- a/getid3v2.cpp	2006-02-10 22:23:14.000000000 -0500
+++ b/getid3v2.cpp	2009-06-12 18:06:49.582446332 -0400
@@ -1,5 +1,6 @@
 #include <vector>
 #include <cstdio>
+#include <cstring>
 #include "char_ucs.h"
 #include "id3v2.h"
 #include "getid3v2.h"
--- a/setfname.cpp	2006-03-19 06:13:36.000000000 -0500
+++ b/setfname.cpp	2009-06-12 18:09:35.759097380 -0400
@@ -1,6 +1,7 @@
 #include <cstdio>
 #include <cctype>
 #include <ctime>                                    // borland needs this
+#include <cstring>
 #include <sys/stat.h>
 #include <utime.h>
 #if defined(__WIN32__)
