diff -upr mplayer2-20121128.orig/stream/stream_cdda.c mplayer2-20121128/stream/stream_cdda.c
--- mplayer2-20121128.orig/stream/stream_cdda.c	2013-01-10 19:37:31.000000000 +0200
+++ mplayer2-20121128/stream/stream_cdda.c	2013-01-10 20:23:25.000000000 +0200
@@ -17,8 +17,8 @@
  */
 
 #include "config.h"
-#include <cdio/cdda.h>
-#include <cdio/paranoia.h>
+#include <cdio/paranoia/cdda.h>
+#include <cdio/paranoia/paranoia.h>
 #include <cdio/cdio.h>
 
 #include <stdio.h>
@@ -116,25 +116,25 @@ const m_option_t cdda_opts[] = {
 };
 
 static const char *cdtext_name[] = {
-    [CDTEXT_ARRANGER] = "Arranger",
-    [CDTEXT_COMPOSER] = "Composer",
-    [CDTEXT_MESSAGE]  =  "Message",
-    [CDTEXT_ISRC] =  "ISRC",
-    [CDTEXT_PERFORMER] = "Performer",
-    [CDTEXT_SONGWRITER] =  "Songwriter",
-    [CDTEXT_TITLE] =  "Title",
-    [CDTEXT_UPC_EAN] = "UPC_EAN",
+    [CDTEXT_FIELD_ARRANGER] = "Arranger",
+    [CDTEXT_FIELD_COMPOSER] = "Composer",
+    [CDTEXT_FIELD_MESSAGE]  =  "Message",
+    [CDTEXT_FIELD_ISRC] =  "ISRC",
+    [CDTEXT_FIELD_PERFORMER] = "Performer",
+    [CDTEXT_FIELD_SONGWRITER] =  "Songwriter",
+    [CDTEXT_FIELD_TITLE] =  "Title",
+    [CDTEXT_FIELD_UPC_EAN] = "UPC_EAN",
 };
 
 static bool print_cdtext(stream_t *s, int track)
 {
     cdda_priv* p = (cdda_priv*)s->priv;
-    cdtext_t *text = cdio_get_cdtext(p->cd->p_cdio, track);
+    cdtext_t *text = cdio_get_cdtext(p->cd->p_cdio);
     if (text) {
         mp_msg(MSGT_SEEK, MSGL_INFO, "CD-Text (%s):\n", track ? "track" : "CD");
         for (int i = 0; i < sizeof(cdtext_name) / sizeof(cdtext_name[0]); i++) {
             const char *name = cdtext_name[i];
-            const char *value = cdtext_get_const(i, text);
+            const char *value = cdtext_get_const(text, i, track);
             if (name && value)
                 mp_msg(MSGT_SEEK, MSGL_INFO, "  %s: '%s'\n", name, value);
         }
