diff --git a/tp/Texinfo/ParserNonXS.pm b/tp/Texinfo/ParserNonXS.pm
index 7aafb825e0..9510f12ddf 100644
--- a/tp/Texinfo/ParserNonXS.pm
+++ b/tp/Texinfo/ParserNonXS.pm
@@ -5030,7 +5030,7 @@ sub _handle_menu_entry_separators($$$$$$)
       and ($current->{'parent'}->{'type'} eq 'menu_comment'
            or $current->{'parent'}->{'type'} eq 'menu_entry_description')
       and $asterisk
-      and $current->{'contents'}
+      and ref($current->{'contents'}) eq "ARRAY"
       and $current->{'contents'}->[-1]->{'type'}
       and $current->{'contents'}->[-1]->{'type'} eq 'empty_line'
       and $current->{'contents'}->[-1]->{'text'} eq '') {
