From 766a1b30fe39d397e239e37e77f66fa4cd999722 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 8 May 2018 23:54:19 +0000
Subject: [PATCH] no neon

---
 src/Makefile.am | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 4b0e863f..c1db9e15 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -115,11 +115,6 @@ endif
 
 pkglibexec_PROGRAMS = aircrack-ng--generic
 
-if ARM
-pkglibexec_PROGRAMS += aircrack-ng--neon \
-                       aircrack-ng--asimd
-endif
-
 if PPC
 pkglibexec_PROGRAMS += aircrack-ng--altivec \
                        aircrack-ng--power8
-- 
2.17.0

