--- udev/toudev.awk.orig	2007-03-25 00:16:40.000000000 +0100
+++ udev/toudev.awk	2007-03-25 00:22:12.000000000 +0100
@@ -2,14 +2,14 @@
 BEGIN {
 	FS = ","
 	print "ACTION != \"add\", GOTO=\"lomoco_end\""
-	print "SUBSYSTEM != \"usb\", GOTO=\"lomoco_end\""
-	print "SYSFS{idVendor} != \"046d\", GOTO=\"lomoco_end\""
+	print "SUBSYSTEM != \"usb_device\", GOTO=\"lomoco_end\""
+	print "ATTR{idVendor} != \"046d\", GOTO=\"lomoco_end\""
 	print ""
 }
 
 $1 ~ /0xc[a-f0-9][a-f0-9][a-f0-9]/ {
     print "# " substr($3, index($3, "\"")) ", " $2
-    print "SYSFS{idProduct}==\"" substr($1, index($1, "x")+1) \
+    print "ATTRS{idProduct}==\"" substr($1, index($1, "x")+1) \
 	  "\", RUN=\"lomoco\""
 }
 
