diff -urN a/pjlib/include/pj/config.h b/pjlib/include/pj/config.h
--- a/pjlib/include/pj/config.h	2014-03-14 00:58:10.000000000 -0600
+++ b/pjlib/include/pj/config.h	2014-04-16 15:02:24.469462764 -0600
@@ -244,9 +244,8 @@
 #   define PJ_M_ARMV4		1
 #   define PJ_M_NAME		"armv4"
 #   define PJ_HAS_PENTIUM	0
-#   if !PJ_IS_LITTLE_ENDIAN && !PJ_IS_BIG_ENDIAN
-#   	error Endianness must be declared for this processor
-#   endif
+#   define PJ_IS_LITTLE_ENDIAN	1
+#   define PJ_IS_BIG_ENDIAN	0
 
 #elif defined (PJ_M_POWERPC) || defined(__powerpc) || defined(__powerpc__) || \
 	defined(__POWERPC__) || defined(__ppc__) || defined(_M_PPC) || \
