diff -aur pjproject-2.2.orig/pjmedia/src/pjmedia/ffmpeg_util.h pjproject-2.2/pjmedia/src/pjmedia/ffmpeg_util.h
--- pjproject-2.2.orig/pjmedia/src/pjmedia/ffmpeg_util.h	2011-07-19 05:42:28.000000000 +0200
+++ pjproject-2.2/pjmedia/src/pjmedia/ffmpeg_util.h	2014-03-08 11:30:57.675023916 +0100
@@ -22,6 +22,8 @@
  * that use ffmpeg. This is not a public API.
  */
 
+#define CodecID AVCodecID
+
 #ifndef __PJMEDIA_FFMPEG_UTIL_H__
 #define __PJMEDIA_FFMPEG_UTIL_H__
 
