From 5a422c572991549f0d063af09b6fe0b6b7e958a3 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Fri, 7 Sep 2018 20:03:35 -0600
Subject: [PATCH] arm atomic

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index aa7d8e4f7..f102127fc 100644
--- a/meson.build
+++ b/meson.build
@@ -1555,7 +1555,7 @@ else
     error('GLib must be built with -march=i486 or later.')
   endif
 endif
-glibconfig_conf.set('G_ATOMIC_LOCK_FREE', have_atomic_lock_free)
+#glibconfig_conf.set('G_ATOMIC_LOCK_FREE', have_atomic_lock_free)
 
 # === Threads ===
 
-- 
2.18.0

