diff -rupN linux-2.6.35/arch/arm/include/asm/unistd.h linux-2.6.35.arm-sys_accept4/arch/arm/include/asm/unistd.h
--- linux-2.6.35/arch/arm/include/asm/unistd.h	2010-05-17 19:51:29.000000000 +0200
+++ linux-2.6.35.arm-sys_accept4/arch/arm/include/asm/unistd.h	2010-08-03 14:34:35.000000000 +0200
@@ -392,6 +392,7 @@
 #define __NR_rt_tgsigqueueinfo		(__NR_SYSCALL_BASE+363)
 #define __NR_perf_event_open		(__NR_SYSCALL_BASE+364)
 #define __NR_recvmmsg			(__NR_SYSCALL_BASE+365)
+#define __NR_accept4			(__NR_SYSCALL_BASE+366)

 /*
  * The following SWIs are ARM private.
diff -rupN linux-2.6.35/arch/arm/kernel/calls.S linux-2.6.35.arm-sys_accept4/arch/arm/kernel/calls.S
--- linux-2.6.35/arch/arm/kernel/calls.S	2010-05-17 19:51:29.000000000 +0200
+++ linux-2.6.35.arm-sys_accept4/arch/arm/kernel/calls.S	2010-08-03 14:34:35.000000000 +0200
@@ -375,6 +375,7 @@
 		CALL(sys_rt_tgsigqueueinfo)
 		CALL(sys_perf_event_open)
 /* 365 */	CALL(sys_recvmmsg)
+		CALL(sys_accept4)
 #ifndef syscalls_counted
 .equ syscalls_padding, ((NR_syscalls + 3) & ~3) - NR_syscalls
 #define syscalls_counted
