From 5330ba5af717b983e456be9f576d66b78d9ca214 Mon Sep 17 00:00:00 2001
From: Andrew Resch <andrewresch@gmail.com>
Date: Sat, 2 Jan 2016 14:40:02 -0800
Subject: [PATCH] register_ptr_to_python for boost::shared_ptr<alert>

---
 bindings/python/src/session.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bindings/python/src/session.cpp b/bindings/python/src/session.cpp
index fe53e10..727a54a 100644
--- a/bindings/python/src/session.cpp
+++ b/bindings/python/src/session.cpp
@@ -799,7 +799,7 @@ void bind_session()
         .def("settings", &get_feed_settings)
     ;
 
-    register_ptr_to_python<std::auto_ptr<alert> >();
+    register_ptr_to_python<boost::shared_ptr<alert> >();
 
     def("high_performance_seed", high_performance_seed);
     def("min_memory_usage", min_memory_usage);
