From 75fe906cb0afce76a47338415773a6d4c5f07c5c Mon Sep 17 00:00:00 2001
From: Morten Welinder <terra@gnome.org>
Date: Thu, 30 Apr 2009 23:04:49 +0000
Subject: Work around gtk+ 2.16 ABI break re spinbuttons.

---
diff --git a/src/dialogs/ChangeLog b/src/dialogs/ChangeLog
index 5f63b08..8fcc3f4 100644
--- a/src/dialogs/ChangeLog
+++ b/src/dialogs/ChangeLog
@@ -1,3 +1,7 @@
+2009-04-40  Morten Welinder <terra@gnome.org>
+
+	* *.glade: Work around ABI break for spinbuttons.  #580930.
+
 2009-01-18  Jody Goldberg <jody@gnome.org>
 
 	* Release 1.8.4
diff --git a/src/dialogs/anova-one.glade b/src/dialogs/anova-one.glade
index cf95a5b..a822db5 100644
--- a/src/dialogs/anova-one.glade
+++ b/src/dialogs/anova-one.glade
@@ -318,8 +318,7 @@
 		      <property name="update_policy">GTK_UPDATE_IF_VALID</property>
 		      <property name="snap_to_ticks">False</property>
 		      <property name="wrap">False</property>
-		      <property name="adjustment">0.05 0 1 0.01 10 10</property>
-		    </widget>
+		      <property name="adjustment">0.05 0 1 0.01 10 0</property>		    </widget>
 		    <packing>
 		      <property name="left_attach">1</property>
 		      <property name="right_attach">2</property>
diff --git a/src/dialogs/anova-two.glade b/src/dialogs/anova-two.glade
index 3b101fb..c25e425 100644
--- a/src/dialogs/anova-two.glade
+++ b/src/dialogs/anova-two.glade
@@ -263,8 +263,7 @@
 		      <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 		      <property name="snap_to_ticks">False</property>
 		      <property name="wrap">False</property>
-		      <property name="adjustment">0.05 0 1 0.01 0.05 0.05</property>
-		    </widget>
+		      <property name="adjustment">0.05 0 1 0.01 0.05 0</property>		    </widget>
 		    <packing>
 		      <property name="left_attach">1</property>
 		      <property name="right_attach">2</property>
diff --git a/src/dialogs/cell-format.glade b/src/dialogs/cell-format.glade
index be30f4f..968754d 100644
--- a/src/dialogs/cell-format.glade
+++ b/src/dialogs/cell-format.glade
@@ -356,8 +356,7 @@
 				  <property name="update_policy">GTK_UPDATE_IF_VALID</property>
 				  <property name="snap_to_ticks">False</property>
 				  <property name="wrap">False</property>
-				  <property name="adjustment">0 0 20 1 10 10</property>
-				</widget>
+				  <property name="adjustment">0 0 20 1 10 0</property>				</widget>
 				<packing>
 				  <property name="left_attach">1</property>
 				  <property name="right_attach">2</property>
diff --git a/src/dialogs/col-width.glade b/src/dialogs/col-width.glade
index 907a629..f87b21a 100644
--- a/src/dialogs/col-width.glade
+++ b/src/dialogs/col-width.glade
@@ -148,8 +148,7 @@
 	      <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 	      <property name="snap_to_ticks">False</property>
 	      <property name="wrap">False</property>
-	      <property name="adjustment">5 5 1000 1 5 5</property>
-	    </widget>
+	      <property name="adjustment">5 5 1000 1 5 0</property>	    </widget>
 	    <packing>
 	      <property name="left_attach">1</property>
 	      <property name="right_attach">2</property>
diff --git a/src/dialogs/descriptive-stats.glade b/src/dialogs/descriptive-stats.glade
index 9653553..35bde2c 100644
--- a/src/dialogs/descriptive-stats.glade
+++ b/src/dialogs/descriptive-stats.glade
@@ -474,8 +474,7 @@
 		  <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 		  <property name="snap_to_ticks">False</property>
 		  <property name="wrap">False</property>
-		  <property name="adjustment">0.05 0 1 0.01 10 10</property>
-		</widget>
+		  <property name="adjustment">0.05 0 1 0.01 10 0</property>		</widget>
 		<packing>
 		  <property name="left_attach">2</property>
 		  <property name="right_attach">3</property>
diff --git a/src/dialogs/dialog-stf.glade b/src/dialogs/dialog-stf.glade
index 0673cdd..77c7acd 100644
--- a/src/dialogs/dialog-stf.glade
+++ b/src/dialogs/dialog-stf.glade
@@ -515,8 +515,7 @@
 			  <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 			  <property name="snap_to_ticks">False</property>
 			  <property name="wrap">False</property>
-			  <property name="adjustment">1 1 100 1 10 10</property>
-			</widget>
+			  <property name="adjustment">1 1 100 1 10 0</property>			</widget>
 			<packing>
 			  <property name="padding">0</property>
 			  <property name="expand">False</property>
@@ -557,8 +556,7 @@
 			  <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 			  <property name="snap_to_ticks">False</property>
 			  <property name="wrap">False</property>
-			  <property name="adjustment">1 1 100 1 10 10</property>
-			</widget>
+			  <property name="adjustment">1 1 100 1 10 0</property>			</widget>
 			<packing>
 			  <property name="padding">0</property>
 			  <property name="expand">False</property>
diff --git a/src/dialogs/dialog-zoom.glade b/src/dialogs/dialog-zoom.glade
index 30e65cc..3cd716a 100644
--- a/src/dialogs/dialog-zoom.glade
+++ b/src/dialogs/dialog-zoom.glade
@@ -254,8 +254,7 @@
 			      <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 			      <property name="snap_to_ticks">False</property>
 			      <property name="wrap">False</property>
-			      <property name="adjustment">100 10 500 5 25 25</property>
-			    </widget>
+			      <property name="adjustment">100 10 500 5 25 0</property>			    </widget>
 			    <packing>
 			      <property name="padding">0</property>
 			      <property name="expand">False</property>
diff --git a/src/dialogs/print.glade b/src/dialogs/print.glade
index 196ffd0..f2a1662 100644
--- a/src/dialogs/print.glade
+++ b/src/dialogs/print.glade
@@ -698,8 +698,7 @@
                                       <widget class="GtkSpinButton" id="scale-h-spin">
                                         <property name="visible">True</property>
                                         <property name="can_focus">True</property>
-                                        <property name="adjustment">1 1 100 1 10 10</property>
-                                        <property name="climb_rate">1</property>
+                                        <property name="adjustment">1 1 100 1 10 0</property>                                        <property name="climb_rate">1</property>
                                       </widget>
                                       <packing>
                                         <property name="left_attach">2</property>
@@ -747,8 +746,7 @@
                                       <widget class="GtkSpinButton" id="scale-v-spin">
                                         <property name="visible">True</property>
                                         <property name="can_focus">True</property>
-                                        <property name="adjustment">1 1 100 1 10 10</property>
-                                        <property name="climb_rate">1</property>
+                                        <property name="adjustment">1 1 100 1 10 0</property>                                        <property name="climb_rate">1</property>
                                       </widget>
                                       <packing>
                                         <property name="left_attach">2</property>
@@ -792,8 +790,7 @@
                                       <widget class="GtkSpinButton" id="scale-percent-spin">
                                         <property name="visible">True</property>
                                         <property name="can_focus">True</property>
-                                        <property name="adjustment">10 10 1000 1 10 10</property>
-                                        <property name="climb_rate">1</property>
+                                        <property name="adjustment">10 10 1000 1 10 0</property>                                        <property name="climb_rate">1</property>
                                         <property name="digits">1</property>
                                         <property name="numeric">True</property>
                                       </widget>
diff --git a/src/dialogs/row-height.glade b/src/dialogs/row-height.glade
index 71e62b0..3004d87 100644
--- a/src/dialogs/row-height.glade
+++ b/src/dialogs/row-height.glade
@@ -149,8 +149,7 @@
 	      <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 	      <property name="snap_to_ticks">False</property>
 	      <property name="wrap">False</property>
-	      <property name="adjustment">5 5 1000 1 5 5</property>
-	    </widget>
+	      <property name="adjustment">5 5 1000 1 5 0</property>	    </widget>
 	    <packing>
 	      <property name="left_attach">1</property>
 	      <property name="right_attach">2</property>
diff --git a/src/dialogs/simulation.glade b/src/dialogs/simulation.glade
index d084dfc..356c90b 100644
--- a/src/dialogs/simulation.glade
+++ b/src/dialogs/simulation.glade
@@ -345,8 +345,7 @@
 		      <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 		      <property name="snap_to_ticks">False</property>
 		      <property name="wrap">False</property>
-		      <property name="adjustment">10 1 36000 10 10 10</property>
-		    </widget>
+		      <property name="adjustment">10 1 36000 10 10 0</property>		    </widget>
 		    <packing>
 		      <property name="left_attach">2</property>
 		      <property name="right_attach">3</property>
@@ -366,8 +365,7 @@
 		      <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 		      <property name="snap_to_ticks">False</property>
 		      <property name="wrap">False</property>
-		      <property name="adjustment">1000 1 1e+10 1000 10 10</property>
-		    </widget>
+		      <property name="adjustment">1000 1 1e+10 1000 10 0</property>		    </widget>
 		    <packing>
 		      <property name="left_attach">2</property>
 		      <property name="right_attach">3</property>
@@ -387,8 +385,7 @@
 		      <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 		      <property name="snap_to_ticks">False</property>
 		      <property name="wrap">False</property>
-		      <property name="adjustment">1 1 1e+08 1 10 10</property>
-		    </widget>
+		      <property name="adjustment">1 1 1e+08 1 10 0</property>		    </widget>
 		    <packing>
 		      <property name="left_attach">2</property>
 		      <property name="right_attach">3</property>
@@ -408,8 +405,7 @@
 		      <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 		      <property name="snap_to_ticks">False</property>
 		      <property name="wrap">False</property>
-		      <property name="adjustment">1 1 1e+08 1 10 10</property>
-		    </widget>
+		      <property name="adjustment">1 1 1e+08 1 10 0</property>		    </widget>
 		    <packing>
 		      <property name="left_attach">2</property>
 		      <property name="right_attach">3</property>
diff --git a/src/dialogs/so-scrollbar.glade b/src/dialogs/so-scrollbar.glade
index f4d7de2..8890bbb 100644
--- a/src/dialogs/so-scrollbar.glade
+++ b/src/dialogs/so-scrollbar.glade
@@ -194,8 +194,7 @@
 	      <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 	      <property name="snap_to_ticks">False</property>
 	      <property name="wrap">False</property>
-	      <property name="adjustment">0 0 3001 1 10 1</property>
-	      <accessibility>
+	      <property name="adjustment">0 0 3001 1 10 0</property>	      <accessibility>
 		<atkrelation target="label_min" type="labelled-by"/>
 	      </accessibility>
 	    </widget>
@@ -218,8 +217,7 @@
 	      <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 	      <property name="snap_to_ticks">False</property>
 	      <property name="wrap">False</property>
-	      <property name="adjustment">0 0 3001 1 10 1</property>
-	      <accessibility>
+	      <property name="adjustment">0 0 3001 1 10 0</property>	      <accessibility>
 		<atkrelation target="label_max" type="labelled-by"/>
 	      </accessibility>
 	    </widget>
@@ -242,8 +240,7 @@
 	      <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 	      <property name="snap_to_ticks">False</property>
 	      <property name="wrap">False</property>
-	      <property name="adjustment">0 0 3001 1 10 1</property>
-	      <accessibility>
+	      <property name="adjustment">0 0 3001 1 10 0</property>	      <accessibility>
 		<atkrelation target="label_increment" type="labelled-by"/>
 	      </accessibility>
 	    </widget>
@@ -266,8 +263,7 @@
 	      <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 	      <property name="snap_to_ticks">False</property>
 	      <property name="wrap">False</property>
-	      <property name="adjustment">0 0 3001 1 10 1</property>
-	      <accessibility>
+	      <property name="adjustment">0 0 3001 1 10 0</property>	      <accessibility>
 		<atkrelation target="label_page" type="labelled-by"/>
 	      </accessibility>
 	    </widget>
diff --git a/src/dialogs/solver.glade b/src/dialogs/solver.glade
index 66eab7c..b58fdf3 100644
--- a/src/dialogs/solver.glade
+++ b/src/dialogs/solver.glade
@@ -668,8 +668,7 @@ Bool
 		  <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 		  <property name="snap_to_ticks">False</property>
 		  <property name="wrap">False</property>
-		  <property name="adjustment">100 1 1e+10 10 10 10</property>
-		</widget>
+		  <property name="adjustment">100 1 1e+10 10 10 0</property>		</widget>
 		<packing>
 		  <property name="left_attach">1</property>
 		  <property name="right_attach">2</property>
@@ -689,8 +688,7 @@ Bool
 		  <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 		  <property name="snap_to_ticks">False</property>
 		  <property name="wrap">False</property>
-		  <property name="adjustment">30 1 36000 10 10 10</property>
-		</widget>
+		  <property name="adjustment">30 1 36000 10 10 0</property>		</widget>
 		<packing>
 		  <property name="left_attach">1</property>
 		  <property name="right_attach">2</property>
--
cgit v0.8.2
