--- jbsrc/jb.c.~1~	2008-03-20 16:53:02.000000000 +0100
+++ jbsrc/jb.c	2008-03-26 20:51:45.641363619 +0100
@@ -327,7 +327,6 @@
 jb_package_add_resources (void)
 {
   JBGroup *group;
-  JBRule *rule;
   JBObject *object;
 
   if (jb_variable_get_bool("compile-warnings"))
@@ -362,11 +361,6 @@
   if (jb_variable_get_bool("hotmail"))
     jb_group_add_data_file(group, "hotmail.png", "$pkgdatadir");
 
-  rule = jb_rule_new();
-  jb_rule_set_install_message(rule, "updating the GTK+ icon cache");
-  jb_rule_add_install_command(rule, "-gtk-update-icon-cache -f -t $datadir/icons/hicolor");
-  jb_group_add_resource(group, JB_GROUP_RESOURCE(rule));
-
   jb_group_add(group);
 
   /*** data ******************************************************************/
