From 1ad6e53cf9a4ed81ec421c8d777707a29e58d9c3 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 24 Jun 2014 05:36:29 -0600
Subject: [PATCH] disable xattr mount

---
 src/core/mount-setup.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/core/mount-setup.c b/src/core/mount-setup.c
index ae8447c..42b5677 100644
--- a/src/core/mount-setup.c
+++ b/src/core/mount-setup.c
@@ -93,8 +93,6 @@ static const MountPoint mount_table[] = {
           NULL,       MNT_FATAL|MNT_IN_CONTAINER },
         { "tmpfs",      "/sys/fs/cgroup",            "tmpfs",      "mode=755", MS_NOSUID|MS_NOEXEC|MS_NODEV|MS_STRICTATIME,
           NULL,       MNT_FATAL|MNT_IN_CONTAINER },
-        { "cgroup",     "/sys/fs/cgroup/systemd",    "cgroup",     "none,name=systemd,xattr", MS_NOSUID|MS_NOEXEC|MS_NODEV,
-          NULL,       MNT_FATAL|MNT_IN_CONTAINER },
         { "cgroup",     "/sys/fs/cgroup/systemd",    "cgroup",     "none,name=systemd", MS_NOSUID|MS_NOEXEC|MS_NODEV,
           NULL,       MNT_FATAL|MNT_IN_CONTAINER },
         { "pstore",     "/sys/fs/pstore",            "pstore",     NULL, MS_NOSUID|MS_NOEXEC|MS_NODEV,
-- 
1.9.0

