diff -ru libdrm-2.4.0/configure.ac libdrm-2.4.0-nostubs/configure.ac
--- libdrm-2.4.0/configure.ac	2008-10-09 21:57:09.000000000 +0200
+++ libdrm-2.4.0-nostubs/configure.ac	2008-10-21 10:48:24.000000000 +0200
@@ -32,10 +32,6 @@
 AC_HEADER_STDC
 AC_SYS_LARGEFILE
 
-PKG_CHECK_MODULES(PTHREADSTUBS, pthread-stubs)
-AC_SUBST(PTHREADSTUBS_CFLAGS)
-AC_SUBST(PTHREADSTUBS_LIBS)
-
 pkgconfigdir=${libdir}/pkgconfig
 AC_SUBST(pkgconfigdir)
 AC_ARG_ENABLE(udev,	AS_HELP_STRING([--enable-udev],
diff -ru libdrm-2.4.0/libdrm/intel/Makefile.am libdrm-2.4.0-nostubs/libdrm/intel/Makefile.am
--- libdrm-2.4.0/libdrm/intel/Makefile.am	2008-10-14 01:39:32.000000000 +0200
+++ libdrm-2.4.0-nostubs/libdrm/intel/Makefile.am	2008-10-21 10:49:00.000000000 +0200
@@ -26,13 +26,12 @@
 	$(WARN_CFLAGS) \
 	-I$(top_srcdir)/libdrm \
 	-I$(top_srcdir)/libdrm/intel \
-	$(PTHREADSTUBS_CFLAGS) \
 	-I$(top_srcdir)/shared-core
 
 libdrm_intel_la_LTLIBRARIES = libdrm_intel.la
 libdrm_intel_ladir = $(libdir)
 libdrm_intel_la_LDFLAGS = -version-number 1:0:0 -no-undefined
-libdrm_intel_la_LIBADD = ../libdrm.la @PTHREADSTUBS_LIBS@
+libdrm_intel_la_LIBADD = ../libdrm.la
 
 libdrm_intel_la_SOURCES = \
 	intel_bufmgr.c \
--- libdrm-2.4.5/libdrm/nouveau/Makefile.am.orig	2009-02-28 19:06:24.000000000 +0000
+++ libdrm-2.4.5/libdrm/nouveau/Makefile.am	2009-02-28 19:06:33.000000000 +0000
@@ -2,13 +2,12 @@
 	$(WARN_CFLAGS) \
 	-I$(top_srcdir)/libdrm \
 	-I$(top_srcdir)/libdrm/nouveau \
-	$(PTHREADSTUBS_CFLAGS) \
 	-I$(top_srcdir)/shared-core
 
 libdrm_nouveau_la_LTLIBRARIES = libdrm_nouveau.la
 libdrm_nouveau_ladir = $(libdir)
 libdrm_nouveau_la_LDFLAGS = -version-number 1:0:0 -no-undefined
-libdrm_nouveau_la_LIBADD = ../libdrm.la @PTHREADSTUBS_LIBS@
+libdrm_nouveau_la_LIBADD = ../libdrm.la
 
 libdrm_nouveau_la_SOURCES = \
 			    nouveau_device.c \
