diff -urN a/configure.ac b/configure.ac
--- a/configure.ac	2012-08-24 08:54:42.000000000 -0600
+++ b/configure.ac	2012-08-31 12:05:03.411667257 -0600
@@ -47,10 +47,6 @@
 LT_INIT([disable-static])
 
 
-PKG_CHECK_MODULES(PTHREADSTUBS, pthread-stubs)
-AC_SUBST(PTHREADSTUBS_CFLAGS)
-AC_SUBST(PTHREADSTUBS_LIBS)
-
 pkgconfigdir=${libdir}/pkgconfig
 AC_SUBST(pkgconfigdir)
 AC_ARG_ENABLE([udev],
diff -urN a/exynos/Makefile.am b/exynos/Makefile.am
--- a/exynos/Makefile.am	2012-06-14 15:32:50.000000000 -0600
+++ b/exynos/Makefile.am	2012-08-31 12:05:41.225521950 -0600
@@ -2,13 +2,12 @@
 	$(WARN_CFLAGS) \
 	-I$(top_srcdir) \
 	-I$(top_srcdir)/exynos \
-	$(PTHREADSTUBS_CFLAGS) \
 	-I$(top_srcdir)/include/drm
 
 libdrm_exynos_la_LTLIBRARIES = libdrm_exynos.la
 libdrm_exynos_ladir = $(libdir)
 libdrm_exynos_la_LDFLAGS = -version-number 1:0:0 -no-undefined
-libdrm_exynos_la_LIBADD = ../libdrm.la @PTHREADSTUBS_LIBS@
+libdrm_exynos_la_LIBADD = ../libdrm.la
 
 libdrm_exynos_la_SOURCES = exynos_drm.c
 
diff -urN a/intel/Makefile.am b/intel/Makefile.am
--- a/intel/Makefile.am	2012-07-29 07:39:21.000000000 -0600
+++ b/intel/Makefile.am	2012-08-31 12:05:03.427292817 -0600
@@ -26,7 +26,6 @@
 	$(WARN_CFLAGS) \
 	-I$(top_srcdir) \
 	-I$(top_srcdir)/intel \
-	$(PTHREADSTUBS_CFLAGS) \
 	$(PCIACCESS_CFLAGS) \
 	$(VALGRIND_CFLAGS) \
 	-I$(top_srcdir)/include/drm
@@ -35,7 +34,6 @@
 libdrm_intel_ladir = $(libdir)
 libdrm_intel_la_LDFLAGS = -version-number 1:0:0 -no-undefined
 libdrm_intel_la_LIBADD = ../libdrm.la \
-	@PTHREADSTUBS_LIBS@ \
 	@PCIACCESS_LIBS@ \
 	@CLOCK_LIB@
 
diff -urN a/nouveau/Makefile.am b/nouveau/Makefile.am
--- a/nouveau/Makefile.am	2012-06-14 15:32:50.000000000 -0600
+++ b/nouveau/Makefile.am	2012-08-31 12:05:03.427292817 -0600
@@ -2,14 +2,13 @@
 	$(WARN_CFLAGS) \
 	-I$(top_srcdir) \
 	-I$(top_srcdir)/nouveau \
-	$(PTHREADSTUBS_CFLAGS) \
 	-I$(top_srcdir)/include/drm \
 	-DDEBUG
 
 libdrm_nouveau_la_LTLIBRARIES = libdrm_nouveau.la
 libdrm_nouveau_ladir = $(libdir)
 libdrm_nouveau_la_LDFLAGS = -version-number 2:0:0 -no-undefined
-libdrm_nouveau_la_LIBADD = ../libdrm.la @PTHREADSTUBS_LIBS@
+libdrm_nouveau_la_LIBADD = ../libdrm.la
 
 libdrm_nouveau_la_SOURCES = nouveau.c \
 			    pushbuf.c \
diff -urN a/omap/Makefile.am b/omap/Makefile.am
--- a/omap/Makefile.am	2012-06-14 15:32:50.000000000 -0600
+++ b/omap/Makefile.am	2012-08-31 12:05:59.233979612 -0600
@@ -2,13 +2,12 @@
 	$(WARN_CFLAGS) \
 	-I$(top_srcdir) \
 	-I$(top_srcdir)/omap \
-	$(PTHREADSTUBS_CFLAGS) \
 	-I$(top_srcdir)/include/drm
 
 libdrm_omap_la_LTLIBRARIES = libdrm_omap.la
 libdrm_omap_ladir = $(libdir)
 libdrm_omap_la_LDFLAGS = -version-number 1:0:0 -no-undefined
-libdrm_omap_la_LIBADD = ../libdrm.la @PTHREADSTUBS_LIBS@
+libdrm_omap_la_LIBADD = ../libdrm.la
 
 libdrm_omap_la_SOURCES = omap_drm.c
 
diff -urN a/radeon/Makefile.am b/radeon/Makefile.am
--- a/radeon/Makefile.am	2012-08-24 08:51:13.000000000 -0600
+++ b/radeon/Makefile.am	2012-08-31 12:05:03.435105597 -0600
@@ -26,13 +26,12 @@
 	$(WARN_CFLAGS) \
 	-I$(top_srcdir) \
 	-I$(top_srcdir)/radeon \
-	$(PTHREADSTUBS_CFLAGS) \
 	-I$(top_srcdir)/include/drm
 
 libdrm_radeon_la_LTLIBRARIES = libdrm_radeon.la
 libdrm_radeon_ladir = $(libdir)
 libdrm_radeon_la_LDFLAGS = -version-number 1:0:1 -no-undefined
-libdrm_radeon_la_LIBADD = ../libdrm.la @PTHREADSTUBS_LIBS@
+libdrm_radeon_la_LIBADD = ../libdrm.la
 
 libdrm_radeon_la_SOURCES = \
 	radeon_bo_gem.c \
