diff -ru colortail-0.3.0/CfgFileParser.cc colortail-0.3.0.patched/CfgFileParser.cc
--- colortail-0.3.0/CfgFileParser.cc	2009-08-16 20:21:33.000000000 +0200
+++ colortail-0.3.0.patched/CfgFileParser.cc	2009-08-16 20:19:41.000000000 +0200
@@ -21,6 +21,7 @@
 
 #include <iostream>
 #include <assert.h>
+#include <cstring>
 #include <string>
 #include <malloc.h>
 
diff -ru colortail-0.3.0/Colorizer.cc colortail-0.3.0.patched/Colorizer.cc
--- colortail-0.3.0/Colorizer.cc	2009-08-16 20:21:33.000000000 +0200
+++ colortail-0.3.0.patched/Colorizer.cc	2009-08-16 20:19:55.000000000 +0200
@@ -23,6 +23,7 @@
 
 #include <assert.h>
 #include <iostream>
+#include <cstring>
 #include <string>
 
 using namespace std;
diff -ru colortail-0.3.0/main.cc colortail-0.3.0.patched/main.cc
--- colortail-0.3.0/main.cc	1999-08-03 21:42:59.000000000 +0200
+++ colortail-0.3.0.patched/main.cc	2009-08-16 20:20:34.000000000 +0200
@@ -19,6 +19,7 @@
 
 #include <stdio.h>
 #include <signal.h>
+#include <cstdlib>
 
 #include "main.h"
 #include "ColorTail.h"
