From 784599e99ee97574798512f6daf987cab8fd4cdf Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Wed, 8 Feb 2017 09:12:24 +0100
Subject: [PATCH 1/1] disable unsupported operating system warning

Signed-off-by: Christian Hesse <mail@eworm.de>
---
 backend/wbprivate/workbench/wb_context.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/backend/wbprivate/workbench/wb_context.cpp b/backend/wbprivate/workbench/wb_context.cpp
index 82c0fa6..f06ff99 100644
--- a/backend/wbprivate/workbench/wb_context.cpp
+++ b/backend/wbprivate/workbench/wb_context.cpp
@@ -1009,9 +1009,11 @@ void WBContext::init_finish_(WBOptions *options) {
 
   _frontendCallbacks->show_status_text(_("Ready."));
 
+#if 0
   if (options->open_at_startup_type !=
       "run-script") // <--- so that our runtime tests don't lock up when a modal warning dialog is displayed
     warnIfRunningOnUnsupportedOS();
+#endif
 
   try {
     // execute action requested from command line
-- 
2.11.1

