diff --git a/library/parsers/mysql/MySQLBaseLexer.h b/library/parsers/mysql/MySQLBaseLexer.h
index 1a8b9ac..8bd3d57 100644
--- a/library/parsers/mysql/MySQLBaseLexer.h
+++ b/library/parsers/mysql/MySQLBaseLexer.h
@@ -26,6 +26,7 @@
 #include "Lexer.h"
 #include "MySQLRecognizerCommon.h"
 #include "mysql-recognition-types.h"
+#include <list>
 
 namespace antlr4 {
   class PARSERS_PUBLIC_TYPE Lexer;
