diff --git a/build/cmake/Modules/FindLibSSH.cmake b/build/cmake/Modules/FindLibSSH.cmake
index 3d31ed3..acbe292 100644
--- a/build/cmake/Modules/FindLibSSH.cmake
+++ b/build/cmake/Modules/FindLibSSH.cmake
@@ -51,7 +51,7 @@ else()
   find_library(LibSSH_LIBRARY NAMES ${LibSSH_NAMES})
 endif()
 
-set(LibSSH_HEADER_FILE ${LibSSH_INCLUDE_DIR}/libssh/libssh.h)
+set(LibSSH_HEADER_FILE ${LibSSH_INCLUDE_DIR}/libssh/libssh_version.h)
 file(STRINGS ${LibSSH_HEADER_FILE} LibSSH_VERSION_LINE_MAJOR REGEX "#define LIBSSH_VERSION_MAJOR[ ]+[0-9]+")
 if (LibSSH_VERSION_LINE_MAJOR)
   file(STRINGS ${LibSSH_HEADER_FILE} LibSSH_VERSION_LINE_MINOR REGEX "#define LIBSSH_VERSION_MINOR[ ]+[0-9]+")
