diff --git a/plugins/wb.admin/frontend/wb_admin_export.py b/plugins/wb.admin/frontend/wb_admin_export.py
index 40f5167..ebf6c3d 100644
--- a/plugins/wb.admin/frontend/wb_admin_export.py
+++ b/plugins/wb.admin/frontend/wb_admin_export.py
@@ -1858,8 +1858,6 @@ class WbAdminExportTab(WbAdminSchemaListTab):
                 args.append("--ssl-cipher=%s" % conn["sslCipher"])
 
         # Sets the compatibility parameters if needed
-        if self._compatibility_params:
-          args.append("--set-gtid-purged=OFF")
         if conn.get("OPT_ENABLE_CLEARTEXT_PLUGIN", ""):
             args.append("--enable-cleartext-plugin")
 
diff --git a/plugins/wb.admin/frontend/wb_admin_export_options.py b/plugins/wb.admin/frontend/wb_admin_export_options.py
index e9ec2e7..653f531 100644
--- a/plugins/wb.admin/frontend/wb_admin_export_options.py
+++ b/plugins/wb.admin/frontend/wb_admin_export_options.py
@@ -53,9 +53,9 @@ export_options = {
     "order-by-primary":["Dump each table's rows sorted by its primary key, or by its first unique index.","FALSE"],
     "dump-date":["Include dump date as \"Dump completed on\" comment if --comments is given.","TRUE"],
     "$internal$show-internal-schemas":["Show internal MySQL schemas (mysql, information_schema, performance_schema) in the export schema list.","FALSE"],
-    "tz-utc":["Add SET TIME_ZONE='+00:00' to the dump file.","TRUE"],
+    "tz-utc":["Add SET TIME_ZONE='+00:00' to the dump file.","TRUE"]
 #    "xml":["Produce XML output.","FALSE"]
-    "set-gtid-purged":["Add 'SET @@GLOBAL.GTID_PURGED' to the output.","AUTO","STR",("5.6.9", None)]
+#   "set-gtid-purged":["Add 'SET @@GLOBAL.GTID_PURGED' to the output.","AUTO","STR",("5.6.9", None)]
     }
 
     }
