diff --git a/frontend/linux/workbench/mysql-workbench.in b/frontend/linux/workbench/mysql-workbench.in
index 02a8061..f8176f6 100755
--- a/frontend/linux/workbench/mysql-workbench.in
+++ b/frontend/linux/workbench/mysql-workbench.in
@@ -4,6 +4,12 @@
 # This will cause passwords to be stored only temporarily for the session.
 #WB_NO_GNOME_KEYRING=1
 
+# Looks like WB expects gnome-keyring to be present as soon as XDG_RUNTIME_DIR
+# environment variable is set. Do an extra check here...
+if ! type gnome-keyring-daemon >/dev/null 2>/dev/null; then
+	export WB_NO_GNOME_KEYRING=1
+fi
+
 # force disable the Mac style single menu hack in Ubuntu Unity
 export UBUNTU_MENUPROXY=0
 
