diff --git a/library/forms/swig/mforms.i b/library/forms/swig/mforms.i
index 3f5104f..cdcca30 100644
--- a/library/forms/swig/mforms.i
+++ b/library/forms/swig/mforms.i
@@ -533,7 +533,7 @@ inline boost::function<void (mforms::TextEntryAction)> pycall_void_entryaction_f
 
 #define SWIG_ADD_SIGNAL_BOOL_INT_CALLBACK(method, signal)\
         void add_##method(PyObject *callback) { signal->connect(pycall_bool_int_fun(callback)); }\
-        bool call_##method(int i) { return (*signal)(i); }
+        bool call_##method(int i) { return *( (*signal)(i) ); }
 
 #define SWIG_ADD_SIGNAL_VOID_ENTRYACTION_CALLBACK(method, signal)\
 	void add_##method(PyObject *callback) { signal->connect(pycall_void_entryaction_fun(callback)); }
diff --git a/library/forms/view.cpp b/library/forms/view.cpp
index ec49389..67ff486 100644
--- a/library/forms/view.cpp
+++ b/library/forms/view.cpp
@@ -512,7 +512,7 @@ DropPosition View::get_drop_position()
 bool View::mouse_leave()
 {
   if (_signal_mouse_leave.num_slots() > 0)
-    return _signal_mouse_leave();
+    return *_signal_mouse_leave();
   return false;
 }
