diff --git a/plugins/migration/copytable/copytable.cpp b/plugins/migration/copytable/copytable.cpp
index de52d91..2df53a4 100644
--- a/plugins/migration/copytable/copytable.cpp
+++ b/plugins/migration/copytable/copytable.cpp
@@ -79,7 +79,7 @@ static const char *mysql_field_type_to_name(enum enum_field_types type)
     case MYSQL_TYPE_VAR_STRING: return "MYSQL_TYPE_VAR_STRING";
     case MYSQL_TYPE_STRING: return "MYSQL_TYPE_STRING";
     case MYSQL_TYPE_GEOMETRY: return "MYSQL_TYPE_GEOMETRY";
-    case MYSQL_TYPE_JSON: return "MYSQL_TYPE_JSON";
+//    case MYSQL_TYPE_JSON: return "MYSQL_TYPE_JSON";
     default:
       return "UNKNOWN";
   }
@@ -241,7 +241,7 @@ RowBuffer::RowBuffer(boost::shared_ptr<std::vector<ColumnInfo> > columns,
       case MYSQL_TYPE_STRING:
       case MYSQL_TYPE_VAR_STRING:
       case MYSQL_TYPE_BIT:
-      case MYSQL_TYPE_JSON:
+//      case MYSQL_TYPE_JSON:
         if (!col->is_long_data)
           bind.buffer_length = (unsigned)col->source_length+1;
 
@@ -1551,8 +1551,8 @@ bool MySQLCopyDataSource::fetch_row(RowBuffer &rowbuffer)
             rowbuffer[index].buffer_type == MYSQL_TYPE_LONG_BLOB ||
             rowbuffer[index].buffer_type == MYSQL_TYPE_BLOB ||
             rowbuffer[index].buffer_type == MYSQL_TYPE_STRING ||
-            rowbuffer[index].buffer_type == MYSQL_TYPE_GEOMETRY ||
-            rowbuffer[index].buffer_type == MYSQL_TYPE_JSON)
+            rowbuffer[index].buffer_type == MYSQL_TYPE_GEOMETRY)
+//            rowbuffer[index].buffer_type == MYSQL_TYPE_JSON)
           {
             if (rowbuffer[index].buffer_length)
               free(rowbuffer[index].buffer);
@@ -2357,7 +2357,7 @@ bool MySQLCopyDataTarget::append_bulk_column(size_t col_index)
     case MYSQL_TYPE_STRING:
     case MYSQL_TYPE_ENUM:
     case MYSQL_TYPE_SET:
-    case MYSQL_TYPE_JSON:
+//    case MYSQL_TYPE_JSON:
       _bulk_insert_record.append("'", 1);
       ret_val = _bulk_insert_record.append_escaped((char*)(*_row_buffer)[col_index].buffer, *(*_row_buffer)[col_index].length);
       _bulk_insert_record.append("'", 1);
diff --git a/backend/wbpublic/objimpl/db.query/db_query_Resultset.cpp b/backend/wbpublic/objimpl/db.query/db_query_Resultset.cpp
index f5c2ffb..b42ba6a 100644
--- a/backend/wbpublic/objimpl/db.query/db_query_Resultset.cpp
+++ b/backend/wbpublic/objimpl/db.query/db_query_Resultset.cpp
@@ -300,9 +300,9 @@ public:
         case sql::DataType::SET:
           type= "string";
           break;
-        case sql::DataType::JSON:
+      /*  case sql::DataType::JSON:
           type = "json";
-          break;
+          break; */
         case sql::DataType::SQLNULL:
           type = "null";
           break;          
