--- pandatool/src/xfile/xParser.yxx	2009-02-26 01:34:06.000000000 +0100
+++ pandatool/src/xfile/xParser.yxx	2009-04-08 15:30:53.938782867 +0200
@@ -114,14 +114,14 @@
 template:
          TOKEN_TEMPLATE singleword_name TOKEN_OBRACE class_id
 {
-  $$ = current_node;
+  $<u.node>$ = current_node;
   XFileTemplate *templ = new XFileTemplate(x_file, $2, $4);
   current_node->add_child(templ);
   current_node = templ;
 }
          template_parts TOKEN_CBRACE
 {
-  $$ = current_node;
+  $<u.node>$ = current_node;
   current_node = $<u.node>5;
 }
          ;
@@ -348,7 +348,7 @@
         singleword_name optional_multiword_name TOKEN_OBRACE
 {
   XFileTemplate *xtemplate = x_file->find_template($1);
-  $$ = current_node;
+  $<u.node>$ = current_node;
   
   if (xtemplate == (XFileTemplate *)NULL) {
     yyerror("Unknown template: " + $1);
@@ -367,7 +367,7 @@
     current_template->finalize_parse_data();
   }
 
-  $$ = current_node;
+  $<u.node>$ = current_node;
   current_node = $<u.node>4;
 }
         ;
