diff -Naur libqalculate-0.9.6/libqalculate/Number.cc libqalculate-0.9.6.new/libqalculate/Number.cc
--- libqalculate-0.9.6/libqalculate/Number.cc	2007-05-18 04:03:22.000000000 -0400
+++ libqalculate-0.9.6.new/libqalculate/Number.cc	2008-04-01 10:46:53.000000000 -0400
@@ -15,6 +15,7 @@
 #include "Calculator.h"
 
 #include <sstream>
+#include <climits>
 #include "util.h"
 
 #define REAL_PRECISION_FLOAT_RE(x)		cln::cl_float(cln::realpart(x), cln::float_format(PRECISION + 1))
