diff -urN dnsmasq.conf.example dnsmasq.conf.example.fixed
--- dnsmasq.conf.example	2006-10-18 16:24:50.000000000 -0400
+++ dnsmasq.conf.example.fixed	2006-11-20 22:07:12.000000000 -0500
@@ -58,7 +58,9 @@
 #address=/doubleclick.net/127.0.0.1
 
 # If you want dnsmasq to change uid and gid to something other
-# than the default, edit the following lines.
+# than root, you will need to have CONFIG_SECURITY_CAPABILITIES 
+# enabled in your kernel. The default uid and gid of nobody will 
+# be used if capability is available and this is not set.
 #user=
 #group=
 
