diff -Nura openexr-1.6.1/exrenvmap/main.cpp openexr-1.6.1.new/exrenvmap/main.cpp
--- openexr-1.6.1/exrenvmap/main.cpp	2007-04-25 03:07:51.000000000 +0200
+++ openexr-1.6.1.new/exrenvmap/main.cpp	2008-10-30 17:18:13.000000000 +0100
@@ -45,6 +45,7 @@
 #include <iostream>
 #include <exception>
 #include <stdlib.h>
+#include <string.h>
 
 using namespace Imf;
 using namespace std;
diff -Nura openexr-1.6.1/exrmaketiled/main.cpp openexr-1.6.1.new/exrmaketiled/main.cpp
--- openexr-1.6.1/exrmaketiled/main.cpp	2007-04-25 03:08:45.000000000 +0200
+++ openexr-1.6.1.new/exrmaketiled/main.cpp	2008-10-30 17:18:40.000000000 +0100
@@ -46,6 +46,7 @@
 #include <exception>
 #include <string>
 #include <stdlib.h>
+#include <string.h>
 
 using namespace Imf;
 using namespace std;
