diff -wbBur clearsilver-0.10.5/ruby/ext/hdf/neo_cs.c clearsilver-0.10.5.my/ruby/ext/hdf/neo_cs.c
--- clearsilver-0.10.5/ruby/ext/hdf/neo_cs.c	2007-02-16 00:31:39.000000000 +0000
+++ clearsilver-0.10.5.my/ruby/ext/hdf/neo_cs.c	2009-09-14 09:20:51.000000000 +0000
@@ -19,7 +19,7 @@
 
 VALUE r_neo_error(NEOERR *err);
 
-#define Srb_raise(val) rb_raise(eHdfError, "%s/%d %s",__FILE__,__LINE__,RSTRING(val)->ptr)
+#define Srb_raise(val) rb_raise(eHdfError, "%s/%d %s",__FILE__,__LINE__,RARRAY_PTR(RSTRING(val)))
 
 static void c_free (CSPARSE *csd) {
   if (csd) {
diff -wbBur clearsilver-0.10.5/ruby/ext/hdf/neo_util.c clearsilver-0.10.5.my/ruby/ext/hdf/neo_util.c
--- clearsilver-0.10.5/ruby/ext/hdf/neo_util.c	2007-02-16 00:32:03.000000000 +0000
+++ clearsilver-0.10.5.my/ruby/ext/hdf/neo_util.c	2009-09-14 09:21:41.000000000 +0000
@@ -10,7 +10,7 @@
  */
 
 #include <ruby.h>
-#include <version.h>
+//#include <version.h>
 #include "ClearSilver.h"
 #include "neo_ruby.h"
 
@@ -19,7 +19,7 @@
 VALUE eHdfError;
 static ID id_to_s;
 
-#define Srb_raise(val) rb_raise(eHdfError, "%s/%d %s",__FILE__,__LINE__,RSTRING(val)->ptr)
+#define Srb_raise(val) rb_raise(eHdfError, "%s/%d %s",__FILE__,__LINE__,RARRAY_PTR(RSTRING(val)))
 
 VALUE r_neo_error (NEOERR *err)
 {
