diff -Naur libofx-0.9.0/lib/ofx_container_transaction.cpp libofx-0.9.0.fix/lib/ofx_container_transaction.cpp
--- libofx-0.9.0/lib/ofx_container_transaction.cpp	2007-11-09 18:10:49.000000000 -0500
+++ libofx-0.9.0.fix/lib/ofx_container_transaction.cpp	2008-04-11 21:50:19.237185782 -0400
@@ -21,6 +21,7 @@
 #include <config.h>
 #endif
 
+#include <cstdlib>
 #include <string>
 #include "messages.hh"
 #include "libofx.h"
diff -Naur libofx-0.9.0/lib/ofx_request.cpp libofx-0.9.0.fix/lib/ofx_request.cpp
--- libofx-0.9.0/lib/ofx_request.cpp	2007-10-27 22:37:52.000000000 -0400
+++ libofx-0.9.0.fix/lib/ofx_request.cpp	2008-04-11 21:53:10.511187650 -0400
@@ -21,6 +21,7 @@
 #include <config.h>
 #endif
 
+#include <cstring>
 #include <string>
 #include "messages.hh"
 #include "libofx.h"
diff -Naur libofx-0.9.0/lib/ofx_request_accountinfo.cpp libofx-0.9.0.fix/lib/ofx_request_accountinfo.cpp
--- libofx-0.9.0/lib/ofx_request_accountinfo.cpp	2007-10-27 22:33:43.000000000 -0400
+++ libofx-0.9.0.fix/lib/ofx_request_accountinfo.cpp	2008-04-11 21:55:40.949190524 -0400
@@ -21,6 +21,7 @@
 #include <config.h>
 #endif
 
+#include <cstdlib>
 #include <string>
 #include "libofx.h"
 #include "ofx_request_accountinfo.hh"
diff -Naur libofx-0.9.0/lib/ofx_request_statement.cpp libofx-0.9.0.fix/lib/ofx_request_statement.cpp
--- libofx-0.9.0/lib/ofx_request_statement.cpp	2007-10-28 08:57:18.000000000 -0400
+++ libofx-0.9.0.fix/lib/ofx_request_statement.cpp	2008-04-11 21:57:28.407187014 -0400
@@ -21,6 +21,7 @@
 #include <config.h>
 #endif
 
+#include <cstdlib>
 #include <string>
 #include "libofx.h"
 #include "ofx_utilities.hh"
diff -Naur libofx-0.9.0/lib/ofx_utilities.cpp libofx-0.9.0.fix/lib/ofx_utilities.cpp
--- libofx-0.9.0/lib/ofx_utilities.cpp	2007-10-27 08:12:49.000000000 -0400
+++ libofx-0.9.0.fix/lib/ofx_utilities.cpp	2008-04-11 21:43:01.687185611 -0400
@@ -21,7 +21,8 @@
 
 #include "ParserEventGeneratorKit.h"
 #include "SGMLApplication.h"
-#include <time.h>
+#include <cstdlib>
+#include <ctime>
 #include <string>
 #include <locale.h>
 #include "messages.hh"
diff -Naur libofx-0.9.0/ofxdump/ofxdump.cpp libofx-0.9.0.fix/ofxdump/ofxdump.cpp
--- libofx-0.9.0/ofxdump/ofxdump.cpp	2006-08-02 00:19:22.000000000 -0400
+++ libofx-0.9.0.fix/ofxdump/ofxdump.cpp	2008-04-11 22:05:00.018187231 -0400
@@ -27,8 +27,10 @@
  *   (at your option) any later version.                                   *
  *                                                                         *
  ***************************************************************************/
+#include <cstdlib>
 #include <iostream>
 #include <iomanip>
+#include <cstring>
 #include <string>
 #include "libofx.h"
 #include <stdio.h>		/* for printf() */
