--- zzipfseeko.pc.orig	2007-05-24 15:34:48.000000000 +0300
+++ zzipfseeko.pc	2007-05-24 15:36:57.000000000 +0300
@@ -3,6 +3,7 @@ prefix=/usr
 exec_prefix=${prefix}
 bindir=${exec_prefix}/bin
 libdir=${exec_prefix}/lib
+datarootdir=${prefix}/share
 datadir=${datarootdir}
 sysconfdir=/etc
 includedir=${prefix}/include
--- zziplib.pc.orig	2007-05-24 15:34:48.000000000 +0300
+++ zziplib.pc	2007-05-24 15:37:08.000000000 +0300
@@ -3,6 +3,7 @@ prefix=/usr
 exec_prefix=${prefix}
 bindir=${exec_prefix}/bin
 libdir=${exec_prefix}/lib
+datarootdir=${prefix}/share
 datadir=${datarootdir}
 sysconfdir=/etc
 includedir=${prefix}/include
--- zzipmmapped.pc.orig	2007-05-24 15:34:48.000000000 +0300
+++ zzipmmapped.pc	2007-05-24 15:37:24.000000000 +0300
@@ -3,6 +3,7 @@ prefix=/usr
 exec_prefix=${prefix}
 bindir=${exec_prefix}/bin
 libdir=${exec_prefix}/lib
+datarootdir=${prefix}/share
 datadir=${datarootdir}
 sysconfdir=/etc
 includedir=${prefix}/include
