--- a/arch/arm/mach-kirkwood/Kconfig
+++ b/arch/arm/mach-kirkwood/Kconfig
@@ -27,10 +27,10 @@ config MACH_MV88F6281GTW_GE
 	  Marvell 88F6281 GTW GE Board.
 
 config MACH_SHEEVAPLUG
-	bool "Marvell SheevaPlug Reference Board"
+	bool "Seagate DockStar Board"
 	help
 	  Say 'Y' here if you want your kernel to support the
-	  Marvell SheevaPlug Reference Board.
+	  Seagate DockStar Board.
 
 config MACH_TS219
 	bool "QNAP TS-110, TS-119, TS-210, TS-219 and TS-219P Turbo NAS"

--- a/arch/arm/mach-kirkwood/sheevaplug-setup.c
+++ b/arch/arm/mach-kirkwood/sheevaplug-setup.c
@@ -143,1 +143,1 @@
-MACHINE_START(SHEEVAPLUG, "Marvell SheevaPlug Reference Board")
+MACHINE_START(SHEEVAPLUG, "Seagate DockStar Board")
--
