From 0fc0d0059e0b8c1548da797d4996e20427295401 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 5 Mar 2019 20:05:09 -0700
Subject: [PATCH] Fix for building without valgrind

---
 egg/egg-testing.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/egg/egg-testing.c b/egg/egg-testing.c
index 561e16a..85da119 100644
--- a/egg/egg-testing.c
+++ b/egg/egg-testing.c
@@ -27,7 +27,9 @@
 
 #include <glib-object.h>
 
+#ifdef WITH_VALGRIND
 #include <valgrind/valgrind.h>
+#endif
 
 #include <glib.h>
 #include <glib/gstdio.h>
-- 
2.21.0

