From 253893b4a7eec97e96114bfad6a974d03c0a0bbb Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Thu, 12 Sep 2019 05:44:39 -0600
Subject: [PATCH] disable neon

---
 src/pulsecore/meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/pulsecore/meson.build b/src/pulsecore/meson.build
index 19f6b9e99..09193eb19 100644
--- a/src/pulsecore/meson.build
+++ b/src/pulsecore/meson.build
@@ -174,7 +174,6 @@ simd = import('unstable-simd')
 libpulsecore_simd = simd.check('libpulsecore_simd',
   mmx : ['remap_mmx.c', 'svolume_mmx.c'],
   sse : ['remap_sse.c', 'sconv_sse.c', 'svolume_sse.c'],
-  neon : ['remap_neon.c', 'sconv_neon.c', 'mix_neon.c'],
   c_args : [pa_c_args],
   include_directories : [configinc, topinc],
   implicit_include_directories : false,
-- 
2.23.0

