From bc4cdcd76159b3b2fadd3274322433a9b9e72966 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Thu, 12 Sep 2019 05:44:39 -0600
Subject: [PATCH] disable neon

---
 src/pulsecore/meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/pulsecore/meson.build b/src/pulsecore/meson.build
index 5f78be012..4955e859e 100644
--- a/src/pulsecore/meson.build
+++ b/src/pulsecore/meson.build
@@ -174,7 +174,6 @@ simd = import('unstable-simd')
 libpulsecore_simd = simd.check('libpulsecore_simd',
   mmx : ['remap_mmx.c', 'svolume_mmx.c'],
   sse : ['remap_sse.c', 'sconv_sse.c', 'svolume_sse.c'],
-  neon : ['remap_neon.c', 'sconv_neon.c', 'mix_neon.c'],
   c_args : [pa_c_args],
   include_directories : [configinc, topinc],
   implicit_include_directories : false,
-- 
2.28.0

