diff -u -r foo/usr/bin/padsp bar/usr/bin/padsp
--- foo/usr/bin/padsp	2015-09-09 14:26:20.000000000 +0200
+++ bar/usr/bin/padsp	2015-09-09 14:47:54.828230174 +0200
@@ -74,9 +74,9 @@
 shift $(( $OPTIND - 1 ))
 
 if [ x"$LD_PRELOAD" = x ] ; then
-   LD_PRELOAD="/usr/lib/pulseaudio/libpulsedsp.so"
+   LD_PRELOAD="/usr/\$LIB/pulseaudio/libpulsedsp.so"
 else
-   LD_PRELOAD="$LD_PRELOAD /usr/lib/pulseaudio/libpulsedsp.so"
+   LD_PRELOAD="$LD_PRELOAD /usr/\$LIB/pulseaudio/libpulsedsp.so"
 fi
 
 export LD_PRELOAD
