--- cdrdao-1.2.3.orig/trackdb/TocParser.g	2009/02/21 21:40:56	1.14
+++ cdrdao-1.2.3/trackdb/TocParser.g		2009/05/04 15:49:52	1.15
@@ -511,8 +511,12 @@
        >>
     )
     << if ($st != NULL && $st->length() == 0) {
-          // try to determine length 
-          $st->determineLength();
+         // try to determine length 
+         if ($st->determineLength() != 0) {
+	         log_message(-2, "%s:%d: Cannot determine length of track data specification.",
+		                 filename_, $lineNr);
+	         error_ = 1;
+      	 }
        }
     >> 
     ;
