diff -NrU5 cdrdao-1.2.3rc2.orig/configure.ac cdrdao-1.2.3rc2/configure.ac
--- cdrdao-1.2.3rc2.orig/configure.ac	2009-05-19 16:03:03.616292795 +0200
+++ cdrdao-1.2.3rc2/configure.ac	2009-05-19 16:04:55.826318499 +0200
@@ -32,10 +32,11 @@
 AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_RANLIB
 AC_CHECK_PROG(AR,ar,ar)
 AC_PROG_MAKE_SET
+PKG_PROG_PKG_CONFIG
 
 dnl Checks for header files.
 AC_HEADER_STDC
 AC_HEADER_SYS_WAIT
 AC_CHECK_HEADERS(fcntl.h getopt.h malloc.h unistd.h sys/mman.h sched.h)
