--- cdrdao-1.2.3rc2/trackdb/TempFileManager.cc~	2008-02-17 14:05:05.000000000 +0100
+++ cdrdao-1.2.3rc2/trackdb/TempFileManager.cc	2009-04-07 09:05:41.000000000 +0200
@@ -20,6 +20,7 @@
 #include "TempFileManager.h"
 #include "log.h"
 
+#include <stdio.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
--- cdrdao-1.2.3rc2/dao/CdrDriver.cc~	2009-02-21 11:01:32.000000000 +0100
+++ cdrdao-1.2.3rc2/dao/CdrDriver.cc	2009-04-07 09:07:14.000000000 +0200
@@ -2114,8 +2114,8 @@
 	ret = 1;
     }
     else {
-      if ((caps & CDR_READ_CAP_DATA_RW_RAW|CDR_READ_CAP_DATA_PW_RAW) != 0)
-	ret = 1;
+      if ((caps & (CDR_READ_CAP_DATA_RW_RAW|CDR_READ_CAP_DATA_PW_RAW)) != 0)
+        ret = 1;
     }
     break;
 
