diff -Naur cdrdao-1.2.2/trackdb/TocParser.g cdrdao-1.2.2-cclpatch/trackdb/TocParser.g
--- cdrdao-1.2.2/trackdb/TocParser.g	2008-10-06 08:50:48.000000000 -0700
+++ cdrdao-1.2.2-cclpatch/trackdb/TocParser.g	2008-10-06 08:50:48.000000000 -0700
@@ -774,7 +774,7 @@
       << if (item != NULL) {
            int type = item->packType();
 
-           if (isTrack && ((type >= 0x86 && type <= 0x89) || type == 0x8f)) {
+           if (isTrack && ((type > 0x86 && type <= 0x89) || type == 0x8f)) {
              log_message(-2, "%s:%d: Invalid CD-TEXT item for a track.",
                      filename_, lineNr);
              error_ = 1;
