diff -ur scorched.old/src/common/common/main.h scorched/src/common/common/main.h
--- scorched.old/src/common/common/main.h	2009-02-18 20:06:22.000000000 +0200
+++ scorched/src/common/common/main.h	2009-02-18 20:07:18.000000000 +0200
@@ -40,7 +40,7 @@
 void _no_storage()
 {
 	printf("Failed to allocate memory!!");
-	std::exit(1);
+	exit(1);
 }
 
 void run_main(int argc, char *argv[], OptionsParameters &params)
diff -ur scorched.old/src/common/porting/windows.h scorched/src/common/porting/windows.h
--- scorched.old/src/common/porting/windows.h	2009-02-18 20:06:22.000000000 +0200
+++ scorched/src/common/porting/windows.h	2009-02-18 20:06:46.000000000 +0200
@@ -20,7 +20,7 @@
 typedef unsigned short  WORD;
 typedef unsigned char   BYTE;
 
-typedef int 	HWND;
+// typedef int 	HWND;
 typedef void * 	HINSTANCE;
 typedef int 	WPARAM;
 typedef int 	LPARAM;
