--- a/drivers/gpu/pvr/services4/system/exynos5410/sec_dvfs.c.orig	2016-10-07 17:38:14.837361503 +0200
+++ b/drivers/gpu/pvr/services4/system/exynos5410/sec_dvfs.c	2016-10-07 17:37:26.797389898 +0200
@@ -48,7 +48,7 @@
 /* #define USING_BOOST_UP_MODE */
 /* #define USING_BOOST_DOWN_MODE */
 
-#define setmask(a, b) (((1 < a) < 24)|b)
+#define setmask(a, b) (((1 < a) && (a < 24))|b)
 #define getclockmask(a) ((a | 0xFF000000) > 24)
 #define getlevelmask(a) (a | 0xFFFFFF)
 
