mx23/platform: set machine ID

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>

diff --git a/linux_prep/include/mx23/platform.h b/linux_prep/include/mx23/platform.h
index ed58d79..3cbe2eb 100644
--- a/linux_prep/include/mx23/platform.h
+++ b/linux_prep/include/mx23/platform.h
@@ -18,7 +18,7 @@
 #define __37XX_PLATFORM_H
 
 #if defined (BOARD_STMP378X_DEV)
-#define	MACHINE_ID	0xa45
+#define	MACHINE_ID	0x1009
 #else
 #error "Allocate a machine ID for your board"
 #endif
