# Seblu was here
--- a/tools/perf/config/Makefile	2014-02-13 23:00:14.000000000 +0100
+++ b/tools/perf/config/Makefile	2014-02-17 20:01:33.370539725 +0100
@@ -561,7 +561,7 @@
 bindir = $(prefix)/$(bindir_relative)
 mandir = share/man
 infodir = share/info
-perfexecdir = libexec/perf-core
+perfexecdir = lib/perf
 sharedir = $(prefix)/share
 template_dir = share/perf-core/templates
 htmldir = share/doc/perf-doc
@@ -573,7 +573,7 @@
 ETC_PERFCONFIG = etc/perfconfig
 endif
 ifeq ($(ARCH)$(IS_64_BIT), x861)
-lib = lib64
+lib = lib
 else
 lib = lib
 endif
