diff --git a/tools/power/acpi/tools/acpidump/acpidump.c b/tools/power/acpi/tools/acpidump/acpidump.c
index a84553a..df453cc 100644
--- a/tools/power/acpi/tools/acpidump/acpidump.c
+++ b/tools/power/acpi/tools/acpidump/acpidump.c
@@ -49,8 +49,6 @@ typedef int				s32;
 typedef unsigned char			u8;
 typedef unsigned short			u16;
 typedef unsigned int			u32;
-typedef unsigned long long		u64;
-typedef long long			s64;
 #endif
 
 #include <sys/mman.h>
