# Copyright 2014 Sébastien Luttringer

--- a/tools/usb/usbip/libsrc/usbip_common.h      2014-10-06 02:13:09.295856364 +0200
+++ b/tools/usb/usbip/libsrc/usbip_common.h      2014-10-06 02:13:03.607873728 +0200
@@ -15,7 +15,7 @@
 #include <syslog.h>
 #include <unistd.h>
 #include <linux/usb/ch9.h>
-#include <linux/usbip.h>
+#include "../../../include/uapi/linux/usbip.h"
 
 #ifndef USBIDS_FILE
 #define USBIDS_FILE "/usr/share/hwdata/usb.ids"
