diff -urN a/src/rebar_port_compiler.erl b/src/rebar_port_compiler.erl
--- a/src/rebar_port_compiler.erl	2015-06-19 10:14:28.000000000 -0600
+++ b/src/rebar_port_compiler.erl	2015-09-07 19:16:25.828631245 -0600
@@ -604,11 +604,6 @@
      {"solaris.*-64$", "CXXFLAGS", "-D_REENTRANT -m64 $CXXFLAGS"},
      {"solaris.*-64$", "LDFLAGS", "-m64 $LDFLAGS"},
 
-     %% Linux specific flags for multiarch
-     {"linux.*-64$", "CFLAGS", "-m64 $CFLAGS"},
-     {"linux.*-64$", "CXXFLAGS", "-m64 $CXXFLAGS"},
-     {"linux.*-64$", "LDFLAGS", "$LDFLAGS"},
-
      %% OS X Leopard flags for 64-bit
      {"darwin9.*-64$", "CFLAGS", "-m64 $CFLAGS"},
      {"darwin9.*-64$", "CXXFLAGS", "-m64 $CXXFLAGS"},
