From 0cc831816f5b242e0ad139d363685149b6464091 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Wed, 24 Sep 2014 08:47:43 -0600
Subject: [PATCH 2/2] ffmpeg 2.4 fix

---
 DllAvFilter.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/DllAvFilter.h b/DllAvFilter.h
index 8149274..313e078 100644
--- a/DllAvFilter.h
+++ b/DllAvFilter.h
@@ -124,7 +124,7 @@ public:
   }
   virtual int avfilter_graph_parse(AVFilterGraph *graph, const char *filters, AVFilterInOut **inputs, AVFilterInOut **outputs, void *log_ctx)
   {
-    return ::avfilter_graph_parse(graph, filters, inputs, outputs, log_ctx);
+    return ::avfilter_graph_parse(graph, filters, *inputs, *outputs, log_ctx);
   }
   virtual int avfilter_graph_config(AVFilterGraph *graphctx, void *log_ctx)
   {
-- 
2.1.0

