--- a/arch/arm/boot/dts/armada-370-mirabox.dts	2013-09-16 14:17:51.000000000 -0600
+++ b/arch/arm/boot/dts/armada-370-mirabox.dts	2013-09-21 02:16:27.355026066 -0600
@@ -114,11 +114,12 @@
 				pinctrl-0 = <&sdio_pins3>;
 				pinctrl-names = "default";
 				status = "okay";
+				bus-width = <8>;
 				/*
 				 * No CD or WP GPIOs: SDIO interface used for
 				 * Wifi/Bluetooth chip
 				 */
-				 broken-cd;
+				non-removable;
 			};
 
 			usb@50000 {
--- a/arch/arm/boot/dts/armada-370-xp.dtsi	2013-09-16 14:17:51.000000000 -0600
+++ b/arch/arm/boot/dts/armada-370-xp.dtsi	2013-10-05 18:51:22.198159452 -0600
@@ -209,8 +209,6 @@
 				clocks = <&gateclk 17>;
 				bus-width = <4>;
 				cap-sdio-irq;
-				cap-sd-highspeed;
-				cap-mmc-highspeed;
 				status = "disabled";
 			};
 
