diff -urN a/arch/arm/boot/dts/armada-370-mirabox.dts b/arch/arm/boot/dts/armada-370-mirabox.dts
--- a/arch/arm/boot/dts/armada-370-mirabox.dts	2013-07-12 23:18:35.979098009 -0600
+++ b/arch/arm/boot/dts/armada-370-mirabox.dts	2013-07-13 00:33:39.920059395 -0600
@@ -12,7 +12,7 @@
 /include/ "armada-370.dtsi"
 
 / {
-	model = "Globalscale Mirabox";
+	model = "Marvell SMILE Plug";
 	compatible = "globalscale,mirabox", "marvell,armada370", "marvell,armada-370-xp";
 
 	chosen {
@@ -36,13 +36,8 @@
 			};
 
 			pinctrl {
-				pwr_led_pin: pwr-led-pin {
-					marvell,pins = "mpp63";
-					marvell,function = "gpo";
-				};
-
-				stat_led_pins: stat-led-pins {
-					marvell,pins = "mpp64", "mpp65";
+				smile_led_pins: smile-led-pins {
+					marvell,pins = "mpp63", "mpp64", "mpp47", "mpp59";
 					marvell,function = "gpio";
 				};
 			};
@@ -50,25 +45,31 @@
 			gpio_leds {
 				compatible = "gpio-leds";
 				pinctrl-names = "default";
-				pinctrl-0 = <&pwr_led_pin &stat_led_pins>;
+				pinctrl-0 = <&smile_led_pins>;
 
-				green_pwr_led {
-					label = "mirabox:green:pwr";
-					gpios = <&gpio1 31 1>;
-					linux,default-trigger = "heartbeat";
+				red_eyes_led {
+					label = "smileplug:red:eyes";
+					gpios = <&gpio1 31 0>;
+					default-state = "off";
 				};
 
-				blue_stat_led {
-					label = "mirabox:blue:stat";
-					gpios = <&gpio2 0 1>;
-					linux,default-trigger = "cpu0";
+				green_eyes_led {
+					label = "smileplug:green:eyes";
+					gpios = <&gpio2 0 0>;
+					linux,default-trigger = "default-on";
 				};
 
-				green_stat_led {
-					label = "mirabox:green:stat";
-					gpios = <&gpio2 1 1>;
+				red_smile_led {
+					label = "smileplug:red:smile";
+					gpios = <&gpio1 15 0>;
 					default-state = "off";
 				};
+
+				green_smile_led {
+					label = "smileplug:green:smile";
+					gpios = <&gpio1 27 0>;
+					linux,default-trigger = "default-on";
+				};
 			};
 
 			mdio {
