diff -upr frostwire-4.18.3.noarch.orig/runFrostwire.sh frostwire-4.18.3.noarch/runFrostwire.sh
--- frostwire-4.18.3.noarch.orig/runFrostwire.sh	2009-10-06 18:10:25.000000000 +0300
+++ frostwire-4.18.3.noarch/runFrostwire.sh	2009-10-08 10:02:50.000000000 +0300
@@ -6,9 +6,6 @@
 # this should allow starting limewire from
 # gui-based explorer interfaces
 
-#Attempt to unpack if there's anything to be unpacked.
-python /usr/lib/frostwire/unpack200.py
-
 cd "`dirname "$0"`"
 
 #export AWT_TOOLKIT=MToolkit
