diff -uprb frostwire-4.18.0.noarch.orig/runFrostwire.sh frostwire-4.18.0.noarch/runFrostwire.sh
--- frostwire-4.18.0.noarch.orig/runFrostwire.sh	2009-05-08 04:39:13.000000000 +0300
+++ frostwire-4.18.0.noarch/runFrostwire.sh	2009-07-18 00:43:42.000000000 +0300
@@ -76,7 +76,7 @@ check_version()
       return 1
   fi
 
-  JAVA_HEADER=`${JAVA_PROGRAM_DIR}java -version 2>&1 | head -n 1`
+  JAVA_HEADER=`${JAVA_PROGRAM_DIR}java -version 2>&1 | grep version`
   JAVA_IMPL=`echo ${JAVA_HEADER} | cut -f1 -d' '`
   if [ "$JAVA_IMPL" = "java" ] ; then
     VERSION=`echo ${JAVA_HEADER} | sed "s/java version \"\(.*\)\"/\1/"`
