diff -ur /tmp/dosbox-0.72/src/dos/cdrom_image.cpp dosbox-0.72/src/dos/cdrom_image.cpp
--- /tmp/dosbox-0.72/src/dos/cdrom_image.cpp	2007-08-22 13:54:35.000000000 +0200
+++ dosbox-0.72/src/dos/cdrom_image.cpp	2008-03-06 21:30:00.000000000 +0100
@@ -23,7 +23,7 @@
 #include <cstdio>
 #include <fstream>
 #include <iostream>
-#include <limits.h>
+#include <limits>
 #include <sstream>
 #include <vector>
 #include <sys/stat.h>
diff -ur /tmp/dosbox-0.72/src/hardware/gameblaster.cpp dosbox-0.72/src/hardware/gameblaster.cpp
--- /tmp/dosbox-0.72/src/hardware/gameblaster.cpp	2007-01-10 11:17:37.000000000 +0100
+++ dosbox-0.72/src/hardware/gameblaster.cpp	2008-03-06 21:31:32.000000000 +0100
@@ -24,6 +24,7 @@
 #include "hardware.h"
 #include "setup.h"
 #include "pic.h"
+#include <cstring>
 
 #define LEFT	0x00
 #define RIGHT	0x01
diff -ur /tmp/dosbox-0.72/src/hardware/tandy_sound.cpp dosbox-0.72/src/hardware/tandy_sound.cpp
--- /tmp/dosbox-0.72/src/hardware/tandy_sound.cpp	2007-06-29 20:21:12.000000000 +0200
+++ dosbox-0.72/src/hardware/tandy_sound.cpp	2008-03-06 21:32:02.000000000 +0100
@@ -28,6 +28,7 @@
 #include "setup.h"
 #include "pic.h"
 #include "dma.h"
+#include <cstring>
 
 #define DAC_CLOCK 3570000
 #define MAX_OUTPUT 0x7fff
diff -ur /tmp/dosbox-0.72/src/shell/shell_cmds.cpp dosbox-0.72/src/shell/shell_cmds.cpp
--- /tmp/dosbox-0.72/src/shell/shell_cmds.cpp	2007-08-17 19:58:46.000000000 +0200
+++ dosbox-0.72/src/shell/shell_cmds.cpp	2008-03-06 21:32:47.000000000 +0100
@@ -27,6 +27,7 @@
 #include "regs.h"
 #include "../dos/drives.h"
 #include "support.h"
+#include <cstdlib>
 
 static SHELL_Cmd cmd_list[]={
 {	"CHDIR",	1,			&DOS_Shell::CMD_CHDIR,		"SHELL_CMD_CHDIR_HELP"},
