diff -ru synergy-1.3.1/lib/arch/CArchDaemonUnix.cpp synergy-1.3.1.patched/lib/arch/CArchDaemonUnix.cpp
--- synergy-1.3.1/lib/arch/CArchDaemonUnix.cpp	2005-04-24 05:02:12.000000000 +0200
+++ synergy-1.3.1.patched/lib/arch/CArchDaemonUnix.cpp	2008-10-12 15:57:32.000000000 +0200
@@ -19,6 +19,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <errno.h>
+#include <cstdlib>
 
 //
 // CArchDaemonUnix
diff -ru synergy-1.3.1/lib/arch/CMultibyte.cpp synergy-1.3.1.patched/lib/arch/CMultibyte.cpp
--- synergy-1.3.1/lib/arch/CMultibyte.cpp	2005-11-30 05:33:24.000000000 +0100
+++ synergy-1.3.1.patched/lib/arch/CMultibyte.cpp	2008-10-12 15:58:33.000000000 +0200
@@ -22,6 +22,7 @@
 #if HAVE_LOCALE_H
 #	include <locale.h>
 #endif
+#include <cstdlib>
 #if HAVE_WCHAR_H || defined(_MSC_VER)
 #	include <wchar.h>
 #elif __APPLE__
diff -ru synergy-1.3.1/lib/base/CStringUtil.cpp synergy-1.3.1.patched/lib/base/CStringUtil.cpp
--- synergy-1.3.1/lib/base/CStringUtil.cpp	2005-12-14 18:25:20.000000000 +0100
+++ synergy-1.3.1.patched/lib/base/CStringUtil.cpp	2008-10-12 15:59:04.000000000 +0200
@@ -20,6 +20,7 @@
 #include <cstdio>
 #include <cstdlib>
 #include <algorithm>
+#include <cstring>
 
 //
 // CStringUtil
diff -ru synergy-1.3.1/lib/client/CClient.cpp synergy-1.3.1.patched/lib/client/CClient.cpp
--- synergy-1.3.1/lib/client/CClient.cpp	2006-03-09 05:07:17.000000000 +0100
+++ synergy-1.3.1.patched/lib/client/CClient.cpp	2008-10-12 16:03:32.000000000 +0200
@@ -26,6 +26,8 @@
 #include "CLog.h"
 #include "IEventQueue.h"
 #include "TMethodEventJob.h"
+#include <cstring>
+#include <cstdlib>
 
 //
 // CClient
diff -ru synergy-1.3.1/lib/client/CServerProxy.cpp synergy-1.3.1.patched/lib/client/CServerProxy.cpp
--- synergy-1.3.1/lib/client/CServerProxy.cpp	2006-04-02 03:47:03.000000000 +0200
+++ synergy-1.3.1.patched/lib/client/CServerProxy.cpp	2008-10-12 16:03:59.000000000 +0200
@@ -24,6 +24,7 @@
 #include "TMethodEventJob.h"
 #include "XBase.h"
 #include <memory>
+#include <cstring>
 
 //
 // CServerProxy
diff -ru synergy-1.3.1/lib/net/CTCPSocket.cpp synergy-1.3.1.patched/lib/net/CTCPSocket.cpp
--- synergy-1.3.1/lib/net/CTCPSocket.cpp	2006-04-02 03:47:03.000000000 +0200
+++ synergy-1.3.1.patched/lib/net/CTCPSocket.cpp	2008-10-12 15:59:43.000000000 +0200
@@ -24,6 +24,7 @@
 #include "CArch.h"
 #include "XArch.h"
 #include <string.h>
+#include <cstdlib>
 
 //
 // CTCPSocket
diff -ru synergy-1.3.1/lib/platform/CXWindowsScreen.cpp synergy-1.3.1.patched/lib/platform/CXWindowsScreen.cpp
--- synergy-1.3.1/lib/platform/CXWindowsScreen.cpp	2006-04-02 21:16:39.000000000 +0200
+++ synergy-1.3.1.patched/lib/platform/CXWindowsScreen.cpp	2008-10-12 16:02:25.000000000 +0200
@@ -27,6 +27,7 @@
 #include "IEventQueue.h"
 #include "TMethodEventJob.h"
 #include <cstring>
+#include <cstdlib>
 #if X_DISPLAY_MISSING
 #	error X11 is required to build synergy
 #else
diff -ru synergy-1.3.1/lib/server/CClientProxy1_3.cpp synergy-1.3.1.patched/lib/server/CClientProxy1_3.cpp
--- synergy-1.3.1/lib/server/CClientProxy1_3.cpp	2006-02-12 20:53:43.000000000 +0100
+++ synergy-1.3.1.patched/lib/server/CClientProxy1_3.cpp	2008-10-12 16:04:53.000000000 +0200
@@ -17,6 +17,7 @@
 #include "CLog.h"
 #include "IEventQueue.h"
 #include "TMethodEventJob.h"
+#include <cstring>
 
 //
 // CClientProxy1_3
diff -ru synergy-1.3.1/lib/server/CServer.cpp synergy-1.3.1.patched/lib/server/CServer.cpp
--- synergy-1.3.1/lib/server/CServer.cpp	2006-04-02 03:47:04.000000000 +0200
+++ synergy-1.3.1.patched/lib/server/CServer.cpp	2008-10-12 16:05:27.000000000 +0200
@@ -29,6 +29,7 @@
 #include "TMethodEventJob.h"
 #include "CArch.h"
 #include <string.h>
+#include <cstdlib>
 
 //
 // CServer
diff -ru synergy-1.3.1/lib/synergy/CPacketStreamFilter.cpp synergy-1.3.1.patched/lib/synergy/CPacketStreamFilter.cpp
--- synergy-1.3.1/lib/synergy/CPacketStreamFilter.cpp	2005-04-24 05:02:16.000000000 +0200
+++ synergy-1.3.1.patched/lib/synergy/CPacketStreamFilter.cpp	2008-10-12 16:00:41.000000000 +0200
@@ -16,6 +16,7 @@
 #include "IEventQueue.h"
 #include "CLock.h"
 #include "TMethodEventJob.h"
+#include <cstring>
 
 //
 // CPacketStreamFilter
diff -ru synergy-1.3.1/lib/synergy/IKeyState.cpp synergy-1.3.1.patched/lib/synergy/IKeyState.cpp
--- synergy-1.3.1/lib/synergy/IKeyState.cpp	2006-03-22 06:40:27.000000000 +0100
+++ synergy-1.3.1.patched/lib/synergy/IKeyState.cpp	2008-10-12 16:01:04.000000000 +0200
@@ -14,6 +14,7 @@
 
 #include "IKeyState.h"
 #include <string.h>
+#include <cstdlib>
 
 //
 // IKeyState
diff -ru synergy-1.3.1/lib/synergy/IPrimaryScreen.cpp synergy-1.3.1.patched/lib/synergy/IPrimaryScreen.cpp
--- synergy-1.3.1/lib/synergy/IPrimaryScreen.cpp	2006-03-22 06:40:27.000000000 +0100
+++ synergy-1.3.1.patched/lib/synergy/IPrimaryScreen.cpp	2008-10-12 16:01:20.000000000 +0200
@@ -13,6 +13,7 @@
  */
 
 #include "IPrimaryScreen.h"
+#include <cstdlib>
 
 //
 // IPrimaryScreen
