--- setup.py.bak	2009-08-23 12:13:01.250895899 +0200
+++ setup.py	2009-08-23 12:13:12.060891193 +0200
@@ -4,18 +4,6 @@
 import glob, os.path
 import sys
 
-for arg in sys.argv:
-    if arg == 'install': 
-        print 'Hello.\nYou are trying to do a system-wide install of emesene '\
-              'using this script, which is a very bad thing to do.\n'\
-              'Seriously, you do NOT want to do this, since it can break '\
-              'other python apps, and emesene too!\n'\
-              'Follow my advice: just run the \"emesene\" script that is '\
-              'in this very same directory and you\'re done. emesene is running '\
-              'and your system is safe. It\'s a win-win, don\'t you think?\n'\
-              'Thanks for trying emesene.'
-        quit()
-        
 # From apport's setup.py
 mo_files = []
 for filepath in glob.glob("po/*/LC_MESSAGES/*.mo"):
