--- xserver-xorg-video-imx-viv-3.0.35-4.0.0/DRI_1.10.4/configure.ac	2013-08-01 00:07:52.000000000 +0200
+++ xserver-xorg-video-imx-viv-3.0.35-4.0.0-dirty/DRI_1.10.4/configure.ac	2013-08-01 00:39:31.000000000 +0200
@@ -54,6 +54,7 @@
 
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XORG, [xorg-server >= 1.0.99.901 xproto fontsproto $REQUIRED_MODULES])
+PKG_CHECK_MODULES(LIBDRM, [libdrm >= 2.3.0])
 sdkdir=$(pkg-config --variable=sdkdir xorg-server)
 
 # Checks for libraries.
