--- xserver-xorg-video-imx-viv-3.0.35-4.0.0/EXA/configure.ac	2013-03-19 15:17:14.000000000 +0100
+++ xserver-xorg-video-imx-viv-3.0.35-4.0.0-dirty/EXA/configure.ac	2013-07-30 20:30:29.000000000 +0200
@@ -29,7 +29,6 @@
 AC_CONFIG_SRCDIR([Makefile.am])
 AM_CONFIG_HEADER([config.h])
 AC_CONFIG_AUX_DIR(.)
-AC_CONFIG_MACRO_DIR([m4])
 
 AM_INIT_AUTOMAKE([dist-bzip2])
 
--- xserver-xorg-video-imx-viv-3.0.35-4.0.0/DRI_1.10.4/configure.ac	2013-03-19 15:17:14.000000000 +0100
+++ xserver-xorg-video-imx-viv-3.0.35-4.0.0-dirty/DRI_1.10.4/configure.ac	2013-08-01 00:39:31.000000000 +0200
@@ -29,7 +29,6 @@
 AC_CONFIG_SRCDIR([Makefile.am])
 AM_CONFIG_HEADER([config.h])
 AC_CONFIG_AUX_DIR(.)
-AC_CONFIG_MACRO_DIR([m4])
 
 AM_INIT_AUTOMAKE([dist-bzip2])

