--- partimage-0.6.7/src/shared/net.h~	2008-11-07 16:40:57.000000000 +0100
+++ partimage-0.6.7/src/shared/net.h	2008-11-07 16:41:43.000000000 +0100
@@ -28,7 +28,7 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>
-#include <iostream.h>
+#include <iostream>
 #include <pthread.h>
 
 #include "pathnames.h"
--- partimage-0.6.7/src/server/partimaged.cpp~	2008-11-07 16:41:10.000000000 +0100
+++ partimage-0.6.7/src/server/partimaged.cpp	2008-11-07 16:41:55.000000000 +0100
@@ -22,7 +22,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
 
 #include "net.h"
 #include "netserver.h"
--- partimage-0.6.7/src/server/partimaged-main.cpp~	2008-11-07 16:41:25.000000000 +0100
+++ partimage-0.6.7/src/server/partimaged-main.cpp	2008-11-07 16:42:10.000000000 +0100
@@ -27,7 +27,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <signal.h>
