From bbd274235f80cc8858f27d232f4218de2792250e Mon Sep 17 00:00:00 2001
From: Paulo Cesar Pereira de Andrade <pcpa@mandriva.com>
Date: Thu, 31 Jan 2008 00:58:21 +0000
Subject: Fix typo emoveEnabledDevice -> RemoveEnabledDevice.

Typo detected when searching for missing symbols.

Signed-off-by: Peter Hutterer <peter@cs.unisa.edu.au>
---
diff --git a/src/xf86MagicTouch.c b/src/xf86MagicTouch.c
index 12992e6..4de6559 100644
--- a/src/xf86MagicTouch.c
+++ b/src/xf86MagicTouch.c
@@ -695,7 +695,7 @@ xf86MagicControl(DeviceIntPtr dev,
 			DBG(2, ErrorF("MagicTouch OFF\n") );
 			dev->public.on = FALSE;
 			if (local->fd>=0)
-			   emoveEnabledDevice(local->fd);
+			   RemoveEnabledDevice(local->fd);
 				
 			SYSCALL( close(local->fd) );
 			local->fd = -1;
--
cgit v0.8.2
