From 7f90f6a6266b896da41be0f9b04621556c2e083d Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter@cs.unisa.edu.au>
Date: Mon, 26 May 2008 12:41:25 +0000
Subject: Check for XINPUT ABI 3.

---
diff --git a/src/xf86MagicTouch.c b/src/xf86MagicTouch.c
index 1a784ff..9541318 100644
--- a/src/xf86MagicTouch.c
+++ b/src/xf86MagicTouch.c
@@ -574,7 +574,11 @@ xf86MagicControl(DeviceIntPtr dev,
 			/*
 			 * Il movimento viene eseguito su due assi in coordinate assolute.
 			 */
-			if (InitValuatorClassDeviceStruct(dev, 2, xf86GetMotionEvents, local->history_size, Absolute) == FALSE ) 
+			if (InitValuatorClassDeviceStruct(dev, 2,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
+				    xf86GetMotionEvents,
+#endif
+				    local->history_size, Absolute) == FALSE )
 			{
 				ErrorF("MagicTouch ValuatorClassDeviceStruct: ERRORE\n");
 				return !Success;
--
cgit v0.8.2
