From 0ab3a5722f1fe4d34bcba29bffb0813f0966fc54 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <Alan.Coopersmith@sun.com>
Date: Fri, 03 Feb 2006 23:27:57 +0000
Subject: Driver name is "magictouch", not "void"

---
diff --git a/ChangeLog b/ChangeLog
index e6c8681..40cb99b 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2006-02-03  Alan Coopersmith  <alan.coopersmith@sun.com>
+
+	* man/magictouch.man: 
+	Driver name is "magictouch", not "void"
+
 2005-12-20  Kevin E. Martin  <kem-at-freedesktop-dot-org>
 
 	* configure.ac:
diff --git a/man/magictouch.man b/man/magictouch.man
index 663d115..c9e974d 100644
--- a/man/magictouch.man
+++ b/man/magictouch.man
@@ -1,9 +1,9 @@
-.\" $XFree86$
+.\" $XdotOrg: $
 .\" shorthand for double quote that works everywhere.
 .ds q \N'34'
-.TH VOID __drivermansuffix__ __vendorversion__
+.TH MAGICTOUCH __drivermansuffix__ __vendorversion__
 .SH NAME
-void \- null input driver
+magictouch \- MagicTouch input driver
 .SH SYNOPSIS
 .nf
 .B "Section \*qInputDevice\*q"
--
cgit v0.8.2
