diff -ruN -x .cvsignore -x .gitignore xf86-input-spaceorb-93fd49758565321036891543bbbdea74e759a393/README xf86-input-spaceorb-a2c3a6617773557d9bd5bccbc2b0796781ebec74/README
--- xf86-input-spaceorb-93fd49758565321036891543bbbdea74e759a393/README	1970-01-01 00:00:00.000000000 +0000
+++ xf86-input-spaceorb-a2c3a6617773557d9bd5bccbc2b0796781ebec74/README	2009-01-31 04:29:02.000000000 +0000
@@ -0,0 +1,19 @@
+
+Please submit bugs & patches to the Xorg bugzilla:
+
+        https://bugs.freedesktop.org/enter_bug.cgi?product=xorg
+
+All questions regarding this software should be directed at the
+Xorg mailing list:
+
+        http://lists.freedesktop.org/mailman/listinfo/xorg
+
+The master development code repository can be found at:
+
+        git://anongit.freedesktop.org/git/xorg/driver/xf86-input-spaceorb
+
+        http://cgit.freedesktop.org/xorg/driver/xf86-input-spaceorb
+
+For more information on the git code manager, see:
+
+        http://wiki.x.org/wiki/GitPage
diff -ruN -x .cvsignore -x .gitignore xf86-input-spaceorb-93fd49758565321036891543bbbdea74e759a393/man/Makefile.am xf86-input-spaceorb-a2c3a6617773557d9bd5bccbc2b0796781ebec74/man/Makefile.am
--- xf86-input-spaceorb-93fd49758565321036891543bbbdea74e759a393/man/Makefile.am	2007-09-10 17:12:04.000000000 +0000
+++ xf86-input-spaceorb-a2c3a6617773557d9bd5bccbc2b0796781ebec74/man/Makefile.am	2009-01-31 04:29:02.000000000 +0000
@@ -1,4 +1,3 @@
-# $Id$
 #
 # Copyright 2005 Sun Microsystems, Inc.  All rights reserved.
 # 
diff -ruN -x .cvsignore -x .gitignore xf86-input-spaceorb-93fd49758565321036891543bbbdea74e759a393/src/spaceorb.c xf86-input-spaceorb-a2c3a6617773557d9bd5bccbc2b0796781ebec74/src/spaceorb.c
--- xf86-input-spaceorb-93fd49758565321036891543bbbdea74e759a393/src/spaceorb.c	2007-09-10 17:12:04.000000000 +0000
+++ xf86-input-spaceorb-a2c3a6617773557d9bd5bccbc2b0796781ebec74/src/spaceorb.c	2009-01-31 04:29:02.000000000 +0000
@@ -107,7 +107,9 @@
         "xf86CollectInputOptions",
         "xf86ErrorFVerb",
         "xf86FindOptionValue",
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
         "xf86GetMotionEvents",
+#endif
         "xf86GetVerbosity",
         "xf86MotionHistoryAllocate",
         "xf86NameCmp",
@@ -337,7 +339,10 @@
     return !Success;
   }
 
-  if (InitValuatorClassDeviceStruct (dev, 6, xf86GetMotionEvents,
+  if (InitValuatorClassDeviceStruct (dev, 6,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
+				     xf86GetMotionEvents,
+#endif
 				     pInfo->history_size, Absolute) == FALSE)
     {
       ErrorF ("Unable to allocate SPACEORB ValuatorClassDeviceStruct\n");
