From e31278c05666f8ae0e819bd831c65e8766ba2679 Mon Sep 17 00:00:00 2001
From: Bigard Florian <florian.bigard@gmail.com>
Date: Mon, 9 May 2016 11:55:34 +0200
Subject: [PATCH] Fix projectm compilation with GCC 6 (#5371)

---
 3rdparty/libprojectm/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/3rdparty/libprojectm/CMakeLists.txt b/3rdparty/libprojectm/CMakeLists.txt
index e6287f176..0ef785c49 100644
--- a/3rdparty/libprojectm/CMakeLists.txt
+++ b/3rdparty/libprojectm/CMakeLists.txt
@@ -24,6 +24,8 @@ set(BUILD_PROJECTM_STATIC ON)
 set(DISABLE_NATIVE_PRESETS ON)
 set(DISABLE_MILKDROP_PRESETS OFF)
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++98")
+
 if(DISABLE_NATIVE_PRESETS)
   ADD_DEFINITIONS(-DDISABLE_NATIVE_PRESETS)
 endif(DISABLE_NATIVE_PRESETS)
