diff --git a/src/internet/spotify/spotifyblobdownloader.cpp b/src/internet/spotify/spotifyblobdownloader.cpp
index e34577f5d..045aeeb8a 100644
--- a/src/internet/spotify/spotifyblobdownloader.cpp
+++ b/src/internet/spotify/spotifyblobdownloader.cpp
@@ -189,7 +189,7 @@ bool SpotifyBlobDownloader::CheckSignature(
 
   try {
     CryptoPP::ByteQueue bytes;
-    bytes.Put(reinterpret_cast<const byte*>(public_key_data.constData()),
+    bytes.Put(reinterpret_cast<const CryptoPP::byte*>(public_key_data.constData()),
               public_key_data.size());
     bytes.MessageEnd();
 
@@ -204,9 +204,9 @@ bool SpotifyBlobDownloader::CheckSignature(
       actual_filename.remove(kSignatureSuffix);
 
       const bool result = verifier.VerifyMessage(
-          reinterpret_cast<const byte*>(file_data[actual_filename].constData()),
+          reinterpret_cast<const CryptoPP::byte*>(file_data[actual_filename].constData()),
           file_data[actual_filename].size(),
-          reinterpret_cast<const byte*>(
+          reinterpret_cast<const CryptoPP::byte*>(
               file_data[signature_filename].constData()),
           file_data[signature_filename].size());
       qLog(Debug) << "Verifying" << actual_filename << "against"
