--- Imakefile.config	2003-03-08 15:55:18.000000000 +0100
+++ Imakefile	2005-01-20 14:48:55.630795072 +0100
@@ -5,24 +5,24 @@
 XCOMM Build options. Have you read the README.XAW3D file yet?
 XCOMM
 XCOMM For color pixmaps, define MULTIPLANE_PIXMAPS:
-#undef MULTIPLANE_PIXMAPS
+#define MULTIPLANE_PIXMAPS
 XCOMM For grayed stipple shadows, define GRAY_BLKWHT_STIPPLES:
 #define GRAY_BLKWHT_STIPPLES
 XCOMM For scrollbars with arrows, define ARROW_SCROLLBARS:
-#undef ARROW_SCROLLBARS
+#define ARROW_SCROLLBARS
 
-#define DoNormalLib NormalLibXaw
-#define DoSharedLib SharedLibXaw
-#define DoDebugLib DebugLibXaw 
-#define DoProfileLib ProfileLibXaw 
+#define DoNormalLib NormalLibXaw7
+#define DoSharedLib SharedLibXaw7
+#define DoDebugLib DebugLibXaw7 
+#define DoProfileLib ProfileLibXaw7 
 #define HasSharedData YES
 #define LibName Xaw3d
-#define SoRev SOXAWREV
+#define SoRev SOXAW7REV
 #define IncSubdir X11
 #define IncSubSubdir Xaw3d
 
 XCOMM When building outside an X11 source tree:
-XCOMM EXTRA_INCLUDES = -I.
+EXTRA_INCLUDES = -I.
 
 #ifdef SharedXawReqs
 REQUIREDLIBS = SharedXawReqs
