--- Xaw3d-1.5/xc/lib/Xaw3d/MultiSrc.c.orig	Tue Aug  6 22:51:08 2002
+++ Xaw3d-1.5/xc/lib/Xaw3d/MultiSrc.c	Tue Aug  6 22:51:16 2002
@@ -735,6 +735,7 @@
   
   if ( string_set || (old_src->multi_src.type != src->multi_src.type) ) {
     RemoveOldStringOrFile(old_src, string_set);
+    src->multi_src.allocated_string = old_src->multi_src.allocated_string;
     file = InitStringOrFile(src, string_set);
 
     /* Load pieces does this logic for us, but it shouldn't.  Its messy.*/
