--- Xaw3d-1.5E/xc/lib/Xaw3d/MultiSrc.c	2009-03-15 17:04:35.000000000 +0100
+++ Xaw3d-1.5E.new/xc/lib/Xaw3d/MultiSrc.c	2009-03-15 14:22:05.000000000 +0100
@@ -77,6 +77,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <wchar.h>
 
 /****************************************************************
  *
--- Xaw3d-1.5E/xc/lib/Xaw3d/Panner.c	2000-11-27 14:19:24.000000000 +0100
+++ Xaw3d-1.5E.new/xc/lib/Xaw3d/Panner.c	2009-03-15 13:50:40.000000000 +0100
@@ -36,7 +36,7 @@
 #include <X11/Xmu/Misc.h>		/* for Min */
 #include <X11/Xmu/Drawing.h>
 #include <ctype.h>			/* for isascii() etc. */
-#include <math.h>			/* for atof() */
+#include <stdlib.h>			/* for atof() */
 
 extern Bool XmuDistinguishablePixels(); /* not defined in any Xmu headers */
 
--- Xaw3d-1.5E/xc/lib/Xaw3d/Text.c	2009-03-15 17:04:35.000000000 +0100
+++ Xaw3d-1.5E.new/xc/lib/Xaw3d/Text.c	2009-03-15 13:54:10.000000000 +0100
@@ -75,6 +75,7 @@
 #include <X11/Xfuncs.h>
 #include <ctype.h>		/* for isprint() */
 #include <stdlib.h>
+#include <wchar.h>
 
 #include "XawAlloc.h"
 
--- Xaw3d-1.5E/xc/lib/Xaw3d/TextAction.c	2000-11-27 14:19:36.000000000 +0100
+++ Xaw3d-1.5E.new/xc/lib/Xaw3d/TextAction.c	2009-03-15 14:19:27.000000000 +0100
@@ -45,7 +45,9 @@
 #include "XawI18n.h"
 #endif
 #include <stdio.h>
+#include <stdlib.h>
 #include <ctype.h>
+#include <wchar.h>
 
 #define SrcScan                XawTextSourceScan
 #define FindDist               XawTextSinkFindDistance
--- Xaw3d-1.5E/xc/lib/Xaw3d/TextPop.c	2009-03-15 17:04:35.000000000 +0100
+++ Xaw3d-1.5E.new/xc/lib/Xaw3d/TextPop.c	2009-03-15 14:19:44.000000000 +0100
@@ -65,6 +65,7 @@
 #include <stdio.h>
 #include <X11/Xos.h>		/* for O_RDONLY */
 #include <errno.h>
+#include <wchar.h>
 
 #include "XawAlloc.h"
 
--- Xaw3d-1.5E/xc/lib/Xaw3d/TextSrc.c	2000-11-27 14:19:36.000000000 +0100
+++ Xaw3d-1.5E.new/xc/lib/Xaw3d/TextSrc.c	2009-03-15 14:19:13.000000000 +0100
@@ -49,6 +49,7 @@
 #endif
 #include <stdio.h>
 #include <ctype.h>
+#include <wchar.h>
 
 /****************************************************************
  *
--- Xaw3d-1.5E/xc/lib/Xaw3d/ThreeD.c	2003-02-11 12:41:56.000000000 +0100
+++ Xaw3d-1.5E.new/xc/lib/Xaw3d/ThreeD.c	2009-03-15 14:21:20.000000000 +0100
@@ -39,6 +39,7 @@
 #include <X11/Xaw3d/XawInit.h>
 #include <X11/Xaw3d/ThreeDP.h>
 #include <X11/Xosdefs.h>
+#include <X11/Xmu/CharSet.h>
 
 /* Initialization of defaults */
 
--- Xaw3d-1.5E/xc/lib/Xaw3d/Tip.c	2003-03-09 01:31:49.000000000 +0100
+++ Xaw3d-1.5E.new/xc/lib/Xaw3d/Tip.c	2009-03-15 14:21:36.000000000 +0100
@@ -41,6 +41,7 @@
 #include <X11/Xmu/Converters.h>
 #include <X11/Xaw3d/TipP.h>
 #include <X11/Xaw3d/XawInit.h>
+#include <stdlib.h>
 
 #define	TIP_EVENT_MASK (ButtonPressMask	  |	\
 			ButtonReleaseMask |	\
--- Xaw3d-1.5E/xc/lib/Xaw3d/XawI18n.c	1996-10-15 16:41:25.000000000 +0200
+++ Xaw3d-1.5E.new/xc/lib/Xaw3d/XawI18n.c	2009-03-15 14:23:11.000000000 +0100
@@ -55,6 +55,7 @@
 
 */
 
+#include <stdlib.h>
 #include <X11/IntrinsicP.h>
 #include "XawI18n.h"
 
--- Xaw3d-1.5E/xc/lib/Xaw3d/XawIm.c	1998-04-21 15:42:02.000000000 +0200
+++ Xaw3d-1.5E.new/xc/lib/Xaw3d/XawIm.c	2009-03-15 14:22:41.000000000 +0100
@@ -59,6 +59,7 @@
 #include <X11/Xos.h>
 #include <X11/Xfuncs.h>
 #include <X11/ShellP.h>
+#include <X11/ResourceI.h>
 #include <X11/Xaw3d/TextP.h>
 #include <X11/Xaw3d/MultiSrc.h>
 #include <X11/Xaw3d/MultiSinkP.h>
diff -up Xaw3d-1.5E/xc/lib/Xaw3d/XawI18n.h~ Xaw3d-1.5E/xc/lib/Xaw3d/XawI18n.h
--- Xaw3d-1.5E/xc/lib/Xaw3d/XawI18n.h~	1996-10-15 16:41:26.000000000 +0200
+++ Xaw3d-1.5E/xc/lib/Xaw3d/XawI18n.h	2009-03-15 17:18:04.000000000 +0100
@@ -72,7 +72,4 @@ extern wchar_t _Xaw_atowc (
 
 #ifndef HAS_ISW_FUNCS
 #include <ctype.h>
-#ifndef iswspace
-#define iswspace(c) (isascii(c) && isspace(toascii(c)))
-#endif
 #endif
