diff -up Xaw3d-1.5E/xc/lib/Xaw3d/List.c.prototypes Xaw3d-1.5E/xc/lib/Xaw3d/List.c
--- Xaw3d-1.5E/xc/lib/Xaw3d/List.c.prototypes	2000-11-27 14:19:36.000000000 +0100
+++ Xaw3d-1.5E/xc/lib/Xaw3d/List.c	2008-10-06 10:11:28.000000000 +0200
@@ -38,6 +38,7 @@ in this Software without prior written a
 
 #include "Xaw3dP.h"
 #include <stdio.h>
+#include <stdlib.h>
 #include <ctype.h>
 #include <X11/IntrinsicP.h>
 #include <X11/StringDefs.h>
diff -up Xaw3d-1.5E/xc/lib/Xaw3d/MultiSink.c.prototypes Xaw3d-1.5E/xc/lib/Xaw3d/MultiSink.c
--- Xaw3d-1.5E/xc/lib/Xaw3d/MultiSink.c.prototypes	1996-10-15 16:41:20.000000000 +0200
+++ Xaw3d-1.5E/xc/lib/Xaw3d/MultiSink.c	2008-10-06 10:10:37.000000000 +0200
@@ -81,6 +81,7 @@ SOFTWARE.
 #include <X11/Xaw3d/TextP.h>
 #include "XawI18n.h"
 #include <stdio.h>
+#include <stdlib.h>
 #include <ctype.h>
 
 #ifdef GETLASTPOS
diff -up Xaw3d-1.5E/xc/lib/Xaw3d/Text.c.prototypes Xaw3d-1.5E/xc/lib/Xaw3d/Text.c
--- Xaw3d-1.5E/xc/lib/Xaw3d/Text.c.prototypes	2003-02-10 18:28:13.000000000 +0100
+++ Xaw3d-1.5E/xc/lib/Xaw3d/Text.c	2008-10-06 10:11:11.000000000 +0200
@@ -74,6 +74,7 @@ SOFTWARE.
 #include <X11/Xaw3d/ThreeDP.h>
 #include <X11/Xfuncs.h>
 #include <ctype.h>		/* for isprint() */
+#include <stdlib.h>
 
 #ifndef MAX_LEN_CT
 #define MAX_LEN_CT 6		/* for sequence: ESC $ ( A \xx \xx */
