diff -urN a/inittab b/inittab
--- a/inittab	2012-07-26 23:19:28.000000000 +0000
+++ b/inittab	2012-07-29 00:48:54.000000000 +0000
@@ -24,12 +24,13 @@
 su:S:wait:/sbin/sulogin -p
 
 # -8 options fixes umlauts problem on login
+s0:12345:respawn:/sbin/agetty -8 -s 115200 @TTY@ linux
 c1:2345:respawn:/sbin/agetty -8 -s 38400 tty1 linux
 c2:2345:respawn:/sbin/agetty -8 -s 38400 tty2 linux
-c3:2345:respawn:/sbin/agetty -8 -s 38400 tty3 linux
-c4:2345:respawn:/sbin/agetty -8 -s 38400 tty4 linux
-c5:2345:respawn:/sbin/agetty -8 -s 38400 tty5 linux
-c6:2345:respawn:/sbin/agetty -8 -s 38400 tty6 linux
+#c3:2345:respawn:/sbin/agetty -8 -s 38400 tty3 linux
+#c4:2345:respawn:/sbin/agetty -8 -s 38400 tty4 linux
+#c5:2345:respawn:/sbin/agetty -8 -s 38400 tty5 linux
+#c6:2345:respawn:/sbin/agetty -8 -s 38400 tty6 linux
 
 # Serial Virtual Console for KVM and others VMs
 #s0:2345:respawn:/sbin/agetty -8 -s 9600 ttyS0 linux
diff -urN a/rc.shutdown b/rc.shutdown
--- a/rc.shutdown	2012-07-26 23:19:28.000000000 +0000
+++ b/rc.shutdown	2012-07-29 00:48:54.000000000 +0000
@@ -69,6 +69,10 @@
 
 run_hook shutdown_poweroff
 
+# ALARM: Turn LEDs off
+
+/usr/bin/find /sys/ -path *led* -name *trigger* -exec sh -c '/bin/echo none > {}' \;
+
 if [[ -x /run/initramfs/shutdown ]]; then
 
 	# decide what we want to do
diff -urN a/rc.sysinit b/rc.sysinit
--- a/rc.sysinit	2012-07-26 23:19:28.000000000 +0000
+++ b/rc.sysinit	2012-07-29 00:50:05.000000000 +0000
@@ -99,7 +99,7 @@
 fsck_reboot $fsckret
 
 status "Remounting root and API filesystems" \
-	/usr/lib/systemd/systemd-remount-fs
+	mount -o remount,rw /
 
 # Now mount all the local filesystems
 run_hook sysinit_premount
