diff -uNr HandBrake.orig/contrib/Jamfile HandBrake/contrib/Jamfile
--- HandBrake.orig/contrib/Jamfile	2007-10-24 14:12:36.000000000 +0100
+++ HandBrake/contrib/Jamfile	2007-10-24 14:15:47.000000000 +0100
@@ -98,6 +98,7 @@
     {
         cd `dirname $(>)` && CONTRIB=`pwd` &&
         rm -rf libdvdcss && tar xzf libdvdcss.tar.gz && cd libdvdcss &&
+	sed -i 's|SUBDIRS = src test doc|SUBDIRS = src test|' Makefile.in
         ./configure --prefix=$CONTRIB --cache-file=$CONTRIB/config.cache && make && make install &&
         strip -S $CONTRIB/lib/libdvdcss.a
     }
