--- bind-9.3.1/bin/Makefile.in.orig	2005-05-06 01:52:36.000000000 -0400
+++ bind-9.3.1/bin/Makefile.in	2005-05-06 01:53:24.000000000 -0400
@@ -19,7 +19,7 @@
 VPATH =		@srcdir@
 top_srcdir =	@top_srcdir@
 
-SUBDIRS =	named rndc dig dnssec tests nsupdate check
+SUBDIRS =	dig nsupdate
 TARGETS =
 
 @BIND9_MAKE_RULES@
--- bind-9.3.1/lib/Makefile.in.orig	2005-05-06 02:05:12.000000000 -0400
+++ bind-9.3.1/lib/Makefile.in	2005-05-06 02:05:35.000000000 -0400
@@ -23,7 +23,7 @@
 # Attempt to disable parallel processing.
 .NOTPARALLEL:
 .NO_PARALLEL:
-SUBDIRS =	isc isccc dns isccfg bind9 lwres tests
+SUBDIRS =	isc dns isccfg bind9 lwres
 TARGETS =
 
 @BIND9_MAKE_RULES@
