Description: Adapt testsuite to changed gnupg defaults in Debian
Author: Andreas Metzler <ametzler@debian.org>
Origin: vendor
Bug-Debian: https://bugs.debian.org/1103189
Forwarded: not-needed
Last-Update: 2025-04-20

--- gpgme1.0-1.24.2.orig/lang/qt/tests/t-config.cpp
+++ gpgme1.0-1.24.2/lang/qt/tests/t-config.cpp
@@ -62,7 +62,7 @@ private Q_SLOTS:
         auto entry = conf->entry(QStringLiteral("gpg"), QStringLiteral("compliance"));
         QVERIFY(entry);
         const auto defaultValue = entry->defaultValue().toString();
-        QCOMPARE(defaultValue, QStringLiteral("gnupg"));
+        QCOMPARE(defaultValue, QStringLiteral("openpgp"));
 
         entry->setStringValue("de-vs");
         conf->sync(true);
