diff -ru gmic-4.0.3.orig/cmake/FindOpenCV.cmake gmic-4.0.3/cmake/FindOpenCV.cmake
--- gmic-4.0.3.orig/cmake/FindOpenCV.cmake	2026-07-20 13:53:30.777990860 +0200
+++ gmic-4.0.3/cmake/FindOpenCV.cmake	2026-07-20 13:53:38.030321976 +0200
@@ -1,7 +1,7 @@
 # The CMake modules shipped with OpenCV do not contain targets.
 find_package(PkgConfig)
 
-pkg_check_modules(OPENCV opencv4)
+pkg_check_modules(OPENCV opencv5)
 
 if(NOT OPENCV_FOUND)
   pkg_check_modules(OPENCV opencv)
