diff -urNB webkit-qtwebkit-23/Source/WebCore/WebCore.pri new/Source/WebCore/WebCore.pri
--- webkit-qtwebkit-23/Source/WebCore/WebCore.pri	2013-02-15 16:46:12.000000000 +0100
+++ new/Source/WebCore/WebCore.pri	2013-02-21 15:46:00.298855975 +0100
@@ -197,8 +197,10 @@
 
 contains(DEFINES, WTF_USE_GSTREAMER=1) {
     DEFINES += ENABLE_GLIB_SUPPORT=1
-    PKGCONFIG += glib-2.0 gio-2.0 gstreamer-0.10 gstreamer-app-0.10 gstreamer-base-0.10 gstreamer-interfaces-0.10 gstreamer-pbutils-0.10 gstreamer-plugins-base-0.10 gstreamer-video-0.10
-}
+    DEFINES += GST_API_VERSION=1.0
+    DEFINES += GST_API_VERSION_1
+    PKGCONFIG += glib-2.0 gio-2.0 gstreamer-1.0 gstreamer-app-1.0 gstreamer-base-1.0 gstreamer-pbutils-1.0 gstreamer-plugins-base-1.0 gstreamer-video-1.0 gstreamer-audio-1.0
+ }
 
 contains(DEFINES, ENABLE_VIDEO=1) {
     contains(DEFINES, WTF_USE_QTKIT=1) {
@@ -233,7 +235,7 @@
     contains(DEFINES, WTF_USE_GSTREAMER=1) {
         DEFINES += WTF_USE_WEBAUDIO_GSTREAMER=1
         INCLUDEPATH += $$SOURCE_DIR/platform/audio/gstreamer
-        PKGCONFIG += gstreamer-audio-0.10 gstreamer-fft-0.10
+        PKGCONFIG += gstreamer-audio-1.0 gstreamer-fft-1.0
     }
 }
 
diff -urNB webkit-qtwebkit-23/Source/WebKit/qt/WebCoreSupport/ChromeClientQt.h new/Source/WebKit/qt/WebCoreSupport/ChromeClientQt.h
--- webkit-qtwebkit-23/Source/WebKit/qt/WebCoreSupport/ChromeClientQt.h	2013-02-15 16:46:12.000000000 +0100
+++ new/Source/WebKit/qt/WebCoreSupport/ChromeClientQt.h	2013-02-21 15:46:00.298855975 +0100
@@ -157,7 +157,7 @@
     virtual void needTouchEvents(bool) { }
 #endif
  
-#if ENABLE(VIDEO) && (USE(GSTREAMER) || USE(QT_MULTIMEDIA) || USE(QTKIT))
+#if ENABLE(VIDEO) && ((USE(GSTREAMER) && !defined(GST_API_VERSION_1)) || USE(QT_MULTIMEDIA) || USE(QTKIT))
     virtual bool supportsFullscreenForNode(const Node*);
     virtual void enterFullscreenForNode(Node*);
     virtual void exitFullscreenForNode(Node*);
diff -urNB webkit-qtwebkit-23/Source/WebKit/qt/WebCoreSupport/FullScreenVideoQt.h new/Source/WebKit/qt/WebCoreSupport/FullScreenVideoQt.h
--- webkit-qtwebkit-23/Source/WebKit/qt/WebCoreSupport/FullScreenVideoQt.h	2013-02-15 16:46:12.000000000 +0100
+++ new/Source/WebKit/qt/WebCoreSupport/FullScreenVideoQt.h	2013-02-21 15:52:57.400847675 +0100
@@ -106,10 +106,10 @@
 private:
     ChromeClientQt* m_chromeClient;
     HTMLVideoElement* m_videoElement;
-#if USE(QT_MULTIMEDIA)
+#if USE(QT_MULTIMEDIA) && !defined(GST_API_VERSION_1)
     QWebFullScreenVideoHandler* m_FullScreenVideoHandler;
 #endif
-#if USE(GSTREAMER)
+#if USE(GSTREAMER) && !defined(GST_API_VERSION_1)
     GStreamerFullScreenVideoHandler* m_FullScreenVideoHandlerGStreamer;
 #endif
 #if USE(QTKIT)
diff -urNB webkit-qtwebkit-23/Tools/qmake/mkspecs/features/features.prf new/Tools/qmake/mkspecs/features/features.prf
--- webkit-qtwebkit-23/Tools/qmake/mkspecs/features/features.prf	2013-02-15 16:46:12.000000000 +0100
+++ new/Tools/qmake/mkspecs/features/features.prf	2013-02-21 15:46:00.297855972 +0100
@@ -186,7 +186,7 @@
             DEFINES += WTF_USE_QTKIT=1
         }
     } else: linux-*:!contains(DEFINES, USE_QTMULTIMEDIA=1) {
-        !contains(QT_CONFIG, no-pkg-config):system(pkg-config --exists glib-2.0 gio-2.0 \'gstreamer-0.10 >= 0.10.30\' \'gstreamer-plugins-base-0.10 >= 0.10.30\'): {
+        !contains(QT_CONFIG, no-pkg-config):packagesExist("glib-2.0 gio-2.0 gstreamer-1.0 gstreamer-plugins-base-1.0") {
             DEFINES += ENABLE_VIDEO=1
             DEFINES += WTF_USE_GSTREAMER=1
         } else {
