diff -up libxcb-1.1/configure.ac.pthread-stubs libxcb-1.1/configure.ac
--- libxcb-1.1/configure.ac.pthread-stubs	2007-11-04 18:17:11.000000000 -0500
+++ libxcb-1.1/configure.ac	2007-11-12 10:27:06.000000000 -0500
@@ -31,7 +31,7 @@ AC_SUBST(HTML_CHECK_RESULT)
 
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XCBPROTO, xcb-proto >= 1.1)
-NEEDED="pthread-stubs xau >= 0.99.2"
+NEEDED="xau >= 0.99.2"
 PKG_CHECK_MODULES(NEEDED, $NEEDED)
 
 have_xdmcp="no"
