--- a/src/Makefile.in	2009-07-25 16:36:10.000000000 -0700
+++ b/src/Makefile.in	2009-07-25 16:36:10.000000000 -0700
@@ -208,7 +208,7 @@
  else
     # Libtoolish rules.  These should apply to all POSIX-like systems.
     CVISUAL_LIBS += $(filter-out $(_FILTER_OUT), $(GTK_LIBS) \
-       $(GTHREAD_LIBS) -lboost_python -lboost_thread -lboost_signals)
+       $(GTHREAD_LIBS) -lboost_python-mt -lboost_thread-mt -lboost_signals-mt)
     CXX_RULE = $(LT_CXX_RULE)
     LINK_RULE = $(LT_LINKRULE)
     PLATFORM_TARGET = cvisualmodule.la
