diff --git a/src/crypto.cpp b/src/crypto.cpp
index 1faa7741..7d55954c 100644
--- a/src/crypto.cpp
+++ b/src/crypto.cpp
@@ -12,6 +12,7 @@ extern "C" {
 #include <nettle/aes.h>
 #include <nettle/hmac.h>
 #include <gnutls/crypto.h>
+#include <nettle/version.h>
 
 #include <argon2.h>
 }
@@ -115,19 +116,31 @@ aesEncrypt(const uint8_t* data, size_t data_length, const Blob& key)
         gcm_aes128_set_key(&aes, key.data());
         gcm_aes128_set_iv(&aes, GCM_IV_SIZE, ret.data());
         gcm_aes128_encrypt(&aes, data_length, ret.data() + GCM_IV_SIZE, data);
+#if NETTLE_VERSION_MAJOR > 3
+        gcm_aes128_digest(&aes, ret.data() + GCM_IV_SIZE + data_length);
+#else
         gcm_aes128_digest(&aes, GCM_DIGEST_SIZE, ret.data() + GCM_IV_SIZE + data_length);
+#endif
     } else if (key.size() == AES_LENGTHS[1]) {
         struct gcm_aes192_ctx aes;
         gcm_aes192_set_key(&aes, key.data());
         gcm_aes192_set_iv(&aes, GCM_IV_SIZE, ret.data());
         gcm_aes192_encrypt(&aes, data_length, ret.data() + GCM_IV_SIZE, data);
+#if NETTLE_VERSION_MAJOR > 3
+        gcm_aes192_digest(&aes, ret.data() + GCM_IV_SIZE + data_length);
+#else
         gcm_aes192_digest(&aes, GCM_DIGEST_SIZE, ret.data() + GCM_IV_SIZE + data_length);
+#endif
     } else if (key.size() == AES_LENGTHS[2]) {
         struct gcm_aes256_ctx aes;
         gcm_aes256_set_key(&aes, key.data());
         gcm_aes256_set_iv(&aes, GCM_IV_SIZE, ret.data());
         gcm_aes256_encrypt(&aes, data_length, ret.data() + GCM_IV_SIZE, data);
-        gcm_aes256_digest(&aes, GCM_DIGEST_SIZE, ret.data() + GCM_IV_SIZE + data_length);
+#if NETTLE_VERSION_MAJOR > 3
+        gcm_aes256_digest(&aes, ret.data() + GCM_IV_SIZE + data_length);
+#else
+        gcm_aes256_digest(&aes, GCM_DIGEST_SIZE, GCM_DIGEST_SIZE, ret.data() + GCM_IV_SIZE + data_length);
+#endif
     }
 
     return ret;
@@ -160,19 +173,31 @@ aesDecrypt(const uint8_t* data, size_t data_length, const Blob& key)
         gcm_aes128_set_key(&aes, key.data());
         gcm_aes128_set_iv(&aes, GCM_IV_SIZE, data);
         gcm_aes128_decrypt(&aes, data_sz, ret.data(), data + GCM_IV_SIZE);
+#if NETTLE_VERSION_MAJOR > 3
+        gcm_aes128_digest(&aes, digest.data());
+#else
         gcm_aes128_digest(&aes, GCM_DIGEST_SIZE, digest.data());
+#endif
     } else if (key.size() == AES_LENGTHS[1]) {
         struct gcm_aes192_ctx aes;
         gcm_aes192_set_key(&aes, key.data());
         gcm_aes192_set_iv(&aes, GCM_IV_SIZE, data);
         gcm_aes192_decrypt(&aes, data_sz, ret.data(), data + GCM_IV_SIZE);
+#if NETTLE_VERSION_MAJOR > 3
+        gcm_aes192_digest(&aes, digest.data());
+#else
         gcm_aes192_digest(&aes, GCM_DIGEST_SIZE, digest.data());
+#endif
     } else if (key.size() == AES_LENGTHS[2]) {
         struct gcm_aes256_ctx aes;
         gcm_aes256_set_key(&aes, key.data());
         gcm_aes256_set_iv(&aes, GCM_IV_SIZE, data);
         gcm_aes256_decrypt(&aes, data_sz, ret.data(), data + GCM_IV_SIZE);
+#if NETTLE_VERSION_MAJOR > 3
+        gcm_aes256_digest(&aes, digest.data());
+#else
         gcm_aes256_digest(&aes, GCM_DIGEST_SIZE, digest.data());
+#endif
     }
 
     if (not std::equal(digest.begin(), digest.end(), data + data_length - GCM_DIGEST_SIZE)) {
@@ -2045,7 +2070,11 @@ hkdf_extract(const Blob& salt, const Blob& ikm)
     hmac_sha256_set_key(&ctx, salt.size(), salt.data());
     hmac_sha256_update(&ctx, ikm.size(), ikm.data());
     Blob prk(SHA256_DIGEST_SIZE);
+#if NETTLE_VERSION_MAJOR > 3
+    hmac_sha256_digest(&ctx, prk.data());
+#else
     hmac_sha256_digest(&ctx, SHA256_DIGEST_SIZE, prk.data());
+#endif
     return prk;
 }
 
@@ -2063,7 +2092,11 @@ hkdf_expand(const Blob& prk, const uint8_t* info, size_t info_len, size_t len)
         hmac_sha256_update(&ctx, info_len, info);
         hmac_sha256_update(&ctx, 1, &i);
         T.resize(SHA256_DIGEST_SIZE);
+#if NETTLE_VERSION_MAJOR > 3
+        hmac_sha256_digest(&ctx, T.data());
+#else
         hmac_sha256_digest(&ctx, SHA256_DIGEST_SIZE, T.data());
+#endif
         okm.insert(okm.end(), T.begin(), T.end());
         i++;
     }
@@ -2223,7 +2256,11 @@ webPushEncrypt(const Blob& p256dh,
 
     Blob ciphertext(plaintext.size() + GCM_DIGEST_SIZE);
     gcm_aes128_encrypt(&aes, plaintext.size(), ciphertext.data(), plaintext.data());
+#if NETTLE_VERSION_MAJOR > 3
+    gcm_aes128_digest(&aes, ciphertext.data() + plaintext.size());
+#else
     gcm_aes128_digest(&aes, GCM_DIGEST_SIZE, ciphertext.data() + plaintext.size());
+#endif
 
     // 6. Construct Output
     Blob output;
