diff -Naur rcs-5.7-orig/src/conf.heg rcs-5.7/src/conf.heg
--- rcs-5.7-orig/src/conf.heg	1995-06-02 11:19:00.000000000 -0700
+++ rcs-5.7/src/conf.heg	2004-11-29 12:41:37.000000000 -0800
@@ -210,10 +210,10 @@
 /* Do struct stat s and t describe the same file?  Answer d if unknown.  */
 #define same_file(s,t,d) ((s).st_ino==(t).st_ino && (s).st_dev==(t).st_dev)
 #define has_utimbuf 1 /* Does struct utimbuf work?  */
-#define CO "/usr/local/bin/co" /* name of 'co' program */
+#define CO "/usr/bin/co" /* name of 'co' program */
 #define COMPAT2 0 /* Are version 2 files supported?  */
-#define DIFF "/usr/local/bin/diff" /* name of 'diff' program */
-#define DIFF3 "/usr/local/bin/diff3" /* name of 'diff3' program */
+#define DIFF "/usr/bin/diff" /* name of 'diff' program */
+#define DIFF3 "/usr/bin/diff3" /* name of 'diff3' program */
 #define DIFF3_BIN 1 /* Is diff3 user-visible (not the /usr/lib auxiliary)?  */
 #define DIFFFLAGS "-an" /* Make diff output suitable for RCS.  */
 #define DIFF_L 1 /* Does diff -L work?  */
@@ -221,7 +221,7 @@
 #define DIFF_FAILURE 1 /* DIFF status if differences are found */
 #define DIFF_TROUBLE 2 /* DIFF status if trouble */
 #define ED "/bin/ed" /* name of 'ed' program (used only if !DIFF3_BIN) */
-#define MERGE "/usr/local/bin/merge" /* name of 'merge' program */
+#define MERGE "/usr/bin/merge" /* name of 'merge' program */
 #define TMPDIR "/tmp" /* default directory for temporary files */
 #define SLASH '/' /* principal filename separator */
 #define SLASHes '/' /* `case SLASHes:' labels all filename separators */
diff -Naur rcs-5.7-orig/src/conf.sh rcs-5.7/src/conf.sh
--- rcs-5.7-orig/src/conf.sh	1995-06-15 23:19:24.000000000 -0700
+++ rcs-5.7/src/conf.sh	2004-11-29 12:42:28.000000000 -0800
@@ -32,7 +32,7 @@
 # and can be inspected for clues otherwise.
 
 # The Makefile overrides the following defaults.
-: ${RCSPREFIX=/usr/local/bin/}
+: ${RCSPREFIX=/usr/bin/}
 : ${ALL_CFLAGS=-Dhas_conf_h}
 : ${CC=cc}
 : ${COMPAT2=0}
@@ -42,8 +42,8 @@
 : ${DIFFFLAGS=-an}
 : ${DIFF_L=1}
 : ${DIFF_SUCCESS=0} ${DIFF_FAILURE=1} ${DIFF_TROUBLE=2}
-: ${ED=/bin/ed}
-: ${SENDMAIL='"/usr/lib/sendmail"'}
+: ${ED=/usr/bin/ed}
+: ${SENDMAIL='"/usr/sbin/sendmail"'}
 # : ${LDFLAGS=} ${LIBS=} tickles old shell bug
 
 C="$CC $ALL_CFLAGS"
@@ -73,17 +73,6 @@
 	ech='echo -n' dots='... '
 esac
 
-$ech >&3 "$0: testing permissions $dots"
-rm -f a.d &&
-date >a.d &&
-chmod 0 a.d &&
-{ test -w a.d || cp /dev/null a.d 2>/dev/null; } && {
-	echo >&3 "$n$0: This command should not be run with superuser permissions."
-	exit 1
-}
-echo >&3 OK
-rm -f a.d || exit
-
 $ech >&3 "$0: testing compiler for plausibility $dots"
 echo 'main() { return 0; }' >a.c
 rm -f a.exe a.out || exit
