diff -up koffice-2.3.2/krita/plugins/formats/exr/exr_converter.cc.gcc46 koffice-2.3.2/krita/plugins/formats/exr/exr_converter.cc
--- koffice-2.3.2/krita/plugins/formats/exr/exr_converter.cc.gcc46	2011-02-17 06:13:40.000000000 -0600
+++ koffice-2.3.2/krita/plugins/formats/exr/exr_converter.cc	2011-02-19 21:15:56.597142885 -0600
@@ -80,6 +80,9 @@ ImageType imfTypeToKisType(Imf::PixelTyp
         return IT_FLOAT16;
     case Imf::FLOAT:
         return IT_FLOAT32;
+    default:
+    // shouldn't reach here
+        return IT_UNSUPPORTED;
     }
 }
 
@@ -93,6 +96,9 @@ const KoColorSpace* kisTypeToColorSpace(
     case IT_UNKNOWN:
     case IT_UNSUPPORTED:
         return 0;
+    default:
+    // shouldn't reach here
+        return 0;
     }
 }
 
