--- koffice-2.3.3/plugins/chartshape/kdchart/src/KDChartBackgroundAttributes.cpp	2011-02-17 13:10:38.000000000 +0100
+++ koffice-2.3.3/plugins/chartshape/kdchart/src/KDChartBackgroundAttributes.cpp	2011-09-08 15:03:31.932039128 +0200
@@ -150,7 +150,6 @@ QDebug operator<<(QDebug dbg, const KDCh
 	<< "visible="<<ba.isVisible()
 	<< "brush="<<ba.brush()
 	<< "pixmapmode="<<ba.pixmapMode()
-	<< "pixmap="<<ba.pixmap()
 	<< ")";
     return dbg;
 }
