diff -p -up agave-0.4.7/src/dialogs/gcs-about-window.cc.strfmt agave-0.4.7/src/dialogs/gcs-about-window.cc
--- agave-0.4.7/src/dialogs/gcs-about-window.cc.strfmt	2009-11-25 15:00:03.000000000 +0100
+++ agave-0.4.7/src/dialogs/gcs-about-window.cc	2009-11-25 15:31:43.000000000 +0100
@@ -105,7 +105,7 @@ namespace gcs
                 catch (const Glib::Error& e)
                 {
                     Glib::ustring message = "Couldn't load Agave logo image from theme. " + e.what();
-                    g_warning(message.c_str());
+                    g_warning("%s", message.c_str());
                 }
             }
         }
