diff -Naur agave-0.4.4.orig/src/dialogs/gcs-about-window.cc agave-0.4.4/src/dialogs/gcs-about-window.cc
--- agave-0.4.4.orig/src/dialogs/gcs-about-window.cc	2007-11-02 04:16:16.000000000 +0100
+++ agave-0.4.4/src/dialogs/gcs-about-window.cc	2012-03-20 22:01:55.835021376 +0100
@@ -26,9 +26,6 @@
 #include <cstring>  // for strlen()
 
 #include <config.h>
-#ifdef HAVE_GNOME
-#include <libgnome/gnome-url.h>
-#endif // HAVE_GNOME
 
 #include <gtkmm/icontheme.h>
 
@@ -46,11 +43,8 @@
         {
             set_name(PACKAGE_NAME);
 
-#ifdef HAVE_GNOME
-            set_url_hook(sigc::mem_fun(*this, &AboutWindow::on_link_clicked));
             set_website(PACKAGE_WEBSITE);
             set_website_label(_("Project Website"));
-#endif // HAVE_GNOME
 
             Glib::ustring version(PACKAGE_VERSION);
             set_version(version);
@@ -112,15 +106,6 @@
 
 
         void
-        AboutWindow::on_link_clicked(Gtk::AboutDialog& dialog,
-                const Glib::ustring& link)
-        {
-#ifdef HAVE_GNOME
-            gnome_url_show(link.c_str(), 0);
-#endif // HAVE_GNOME
-        }
-
-        void
         AboutWindow::on_response(int response_id)
         {
             if (response_id == Gtk::RESPONSE_DELETE_EVENT ||
diff -Naur agave-0.4.4.orig/src/gcs-mainwindow-actions.cc agave-0.4.4/src/gcs-mainwindow-actions.cc
--- agave-0.4.4.orig/src/gcs-mainwindow-actions.cc	2007-11-02 04:01:16.000000000 +0100
+++ agave-0.4.4/src/gcs-mainwindow-actions.cc	2012-03-20 22:01:10.637760066 +0100
@@ -25,7 +25,6 @@
 
 #include <vector>
 #include <gtk/gtk.h>
-#include <libgnome/gnome-help.h>
 #include <gtkmm/iconfactory.h>
 #include <gtkmm/stock.h>
 #include <gtkmm/clipboard.h>
@@ -251,9 +250,10 @@
     /* Help Menu Actions */
     void MainWindow::on_action_help_contents(void)
     {
-        gnome_help_display("agave.xml",
-                NULL /* link id */,
-                NULL /* GError */);
+        gtk_show_uri (NULL,
+                "ghelp:agave",
+                gtk_get_current_event_time (),
+                NULL);
     }
 
 
