--- openjdk/corba/make/common/shared/Defs-utils.gmk.old	2008-04-13 13:26:12.000000000 +0300
+++ openjdk/corba/make/common/shared/Defs-utils.gmk	2008-04-14 15:35:13.000000000 +0300
@@ -86,10 +86,10 @@
 CHMOD          = $(UTILS_COMMAND_PATH)chmod
 CMP            = $(UTILS_USR_BIN_PATH)cmp
 COMM           = $(UTILS_USR_BIN_PATH)comm
-COMPRESS       = $(UTILS_USR_BIN_PATH)compress
+COMPRESS       = $(UTILS_COMMAND_PATH)compress
 CP             = $(UTILS_COMMAND_PATH)cp
 CPIO           = $(UTILS_COMMAND_PATH)cpio
-CUT            = $(UTILS_USR_BIN_PATH)cut
+CUT            = $(UTILS_COMMAND_PATH)cut
 DATE           = $(UTILS_COMMAND_PATH)date
 DF             = $(UTILS_COMMAND_PATH)df
 DIFF           = $(UTILS_USR_BIN_PATH)diff
@@ -136,7 +136,7 @@
 TAR            = $(UTILS_COMMAND_PATH)tar
 TEST           = $(UTILS_USR_BIN_PATH)test
 TOUCH          = $(UTILS_COMMAND_PATH)touch
-TR             = $(UTILS_USR_BIN_PATH)tr
+TR             = $(UTILS_COMMAND_PATH)tr
 TRUE           = $(UTILS_COMMAND_PATH)true
 UNAME          = $(UTILS_COMMAND_PATH)uname
 UNIQ           = $(UTILS_USR_BIN_PATH)uniq
@@ -186,7 +186,7 @@
   # others have it in /usr/bin.
   SORT=$(firstword $(wildcard $(UTILS_COMMAND_PATH)sort) \
 		   $(wildcard $(UTILS_USR_BIN_PATH)sort))
-  NAWK           = $(USRBIN_PATH)gawk
+  NAWK           = $(UTILS_COMMAND_PATH)gawk
   # Intrinsic unix command, with backslash-escaped character interpretation
   ECHO           = /bin/echo -e
   # These are really in UTILS_USR_BIN_PATH on Linux (only sccs is not)
