--- Makefile.am	2008-07-21 15:12:07.000000000 +0000
+++ Makefile.am.new	2008-07-22 08:37:49.000000000 +0000
@@ -1176,7 +1176,7 @@
 # gcjwebplugin.so.
 gcjwebplugin.so: gcjwebplugin.cc
 	$(CXX) $(CXXFLAGS) $(MOZILLA_CFLAGS) \
-	  $(MOZILLA_LIBS) $(GLIB_CFLAGS) $(GLIB_LIBS) \
+	  $(MOZILLA_LIBS) -Wl,-R/usr/lib/xulrunner-1.9 $(GLIB_CFLAGS) $(GLIB_LIBS) \
 	  $(GTK_CFLAGS) $(GTK_LIBS) \
 	  -DPACKAGE_VERSION="\"$(PACKAGE_VERSION)\"" \
 	  -fpic -shared -o $@ $<
