--- src/awn-extras-applets-0.2.6/src/weather/weather.py~	2008-05-09 02:20:39.000000000 +0200
+++ src/awn-extras-applets-0.2.6/src/weather/weather.py	2008-05-09 02:21:06.000000000 +0200
@@ -315,7 +315,7 @@
 
 
 	def get_conditions(self):
-		url = 'http://xoap.weather.com/weather/local/' + self.location_code + '?cc=*&prod=xoap&par=1048871467&key=12daac2f3a67cb39'
+		url = 'http://xoap.weather.com/weather/local/' + self.location_code + '?cc=*&link=xoap&par=1048871467&key=12daac2f3a67cb39'
 		if self.units == "Metric":
 			url = url + '&unit=m'
 		try:    
