--- qt-everywhere-opensource-src-4.8.2/tools/assistant/tools/assistant/mainwindow.cpp	2012-06-26 17:37:47.334536023 +0000
+++ qt-everywhere-opensource-src-4.8.2/tools/assistant/tools/assistant/mainwindow.cpp~	2012-06-26 17:37:08.531487286 +0000
@@ -944,8 +944,7 @@
     if (helpEngine.usesAppFont())
         font = helpEngine.appFont();

-    const QWidgetList &widgets = qApp->allWidgets();
-    foreach (QWidget* widget, widgets)
+    foreach (QWidget* widget, QApplication::allWidgets())
         widget->setFont(font);
 }
