diff -ur qt-orig/src/phonon/phonon.pro qt-x11-opensource-src-4.5.2/src/phonon/phonon.pro
--- qt-orig/src/phonon/phonon.pro	2009-06-20 06:57:57.000000000 +0200
+++ qt-x11-opensource-src-4.5.2/src/phonon/phonon.pro	2009-06-25 19:30:00.000000000 +0200
@@ -112,4 +112,9 @@
        DEFINES += QT_NO_DBUS
 }
 
+# install dbus service file
+sources.files = $$PHONON_DIR/org.kde.Phonon.AudioOutput.xml
+sources.path = /usr/share/dbus-1/interfaces
+INSTALLS += sources
+
 contains(QT_CONFIG, reduce_exports): CONFIG += hide_symbols
diff -ur qt-orig/src/plugins/phonon/gstreamer/gstreamer.pro qt-x11-opensource-src-4.5.2/src/plugins/phonon/gstreamer/gstreamer.pro
--- qt-orig/src/plugins/phonon/gstreamer/gstreamer.pro	2009-06-20 06:57:58.000000000 +0200
+++ qt-x11-opensource-src-4.5.2/src/plugins/phonon/gstreamer/gstreamer.pro	2009-06-25 19:31:36.000000000 +0200
@@ -62,6 +62,8 @@
 }
 
 target.path = $$[QT_INSTALL_PLUGINS]/phonon_backend
-INSTALLS += target
+services.files = $$PHONON_GSTREAMER_DIR/gstreamer.desktop
+services.path = /usr/share/kde4/services/phononbackends
+INSTALLS += target services
 
 include(../../qpluginbase.pri)
diff -ur qt-orig/src/3rdparty/phonon/gstreamer/gstreamer.desktop qt-x11-opensource-src-4.5.2/src/3rdparty/phonon/gstreamer/gstreamer.desktop
--- qt-orig/src/3rdparty/phonon/gstreamer/gstreamer.desktop	2009-06-20 06:57:54.000000000 +0200
+++ qt-x11-opensource-src-4.5.2/src/3rdparty/phonon/gstreamer/gstreamer.desktop	2009-06-25 19:30:00.000000000 +0200
@@ -2,7 +2,7 @@
 Type=Service
 X-KDE-ServiceTypes=PhononBackend
 MimeType=application/x-annodex;video/quicktime;video/x-quicktime;audio/x-m4a;application/x-quicktimeplayer;video/mkv;video/msvideo;video/x-msvideo;video/x-flic;audio/x-aiff;audio/aiff;audio/x-pn-aiff;audio/x-realaudio;audio/basic;audio/x-basic;audio/x-pn-au;audio/x-8svx;audio/8svx;audio/x-16sv;audio/168sv;image/x-ilbm;image/ilbm;video/x-anim;video/anim;image/png;image/x-png;video/mng;video/x-mng;audio/x-ogg;audio/x-speex+ogg;application/ogg;application/ogg;audio/vnd.rn-realaudio;audio/x-pn-realaudio-plugin;audio/x-real-audio;application/vnd.rn-realmedia;video/mpeg;video/x-mpeg;audio/x-wav;audio/wav;audio/x-pn-wav;audio/x-pn-windows-acm;audio/mpeg2;audio/x-mpeg2;audio/mpeg3;audio/x-mpeg3;audio/mpeg;audio/x-mpeg;x-mpegurl;audio/x-mpegurl;audio/mp3;audio/mpeg;
-X-KDE-Library=phonon_gstreamer
+X-KDE-Library=libphonon_gstreamer
 X-KDE-PhononBackendInfo-InterfaceVersion=1
 X-KDE-PhononBackendInfo-Version=0.1
 X-KDE-PhononBackendInfo-Website=http://gstreamer.freedesktop.org/
