diff -up qt-everywhere-opensource-src-4.8.0/src/corelib/io/qurl.cpp.toLocalFile qt-everywhere-opensource-src-4.8.0/src/corelib/io/qurl.cpp
--- qt-everywhere-opensource-src-4.8.0/src/corelib/io/qurl.cpp.toLocalFile	2011-10-03 22:44:32.000000000 -0500
+++ qt-everywhere-opensource-src-4.8.0/src/corelib/io/qurl.cpp	2011-10-27 12:58:35.706815049 -0500
@@ -6158,7 +6158,8 @@ QUrl QUrl::fromLocalFile(const QString &
 QString QUrl::toLocalFile() const
 {
     // the call to isLocalFile() also ensures that we're parsed
-    if (!isLocalFile())
+    // Treat URLs with no scheme as local for backward compatibility
+    if (!isLocalFile() && (!d || !d->scheme.isEmpty()))
         return QString();

     QString tmp;
