diff -ru lp_solve_5.5.orig/lp_solve/ccc lp_solve_5.5/lp_solve/ccc
--- lp_solve_5.5.orig/lp_solve/ccc	2009-02-23 16:31:18.000000000 +0000
+++ lp_solve_5.5/lp_solve/ccc	2009-02-23 16:31:29.000000000 +0000
@@ -28,7 +28,7 @@
 fi
 rm /tmp/isnan.c /tmp/isnan >/dev/null 2>&1
 
-opts='-O3'
+opts=$CFLAGS
 
 def=
 if [ "$PLATFORM" = "SCO_UNIX" ]
diff -ru lp_solve_5.5.orig/lpsolve55/ccc lp_solve_5.5/lpsolve55/ccc
--- lp_solve_5.5.orig/lpsolve55/ccc	2009-02-23 16:31:18.000000000 +0000
+++ lp_solve_5.5/lpsolve55/ccc	2009-02-23 16:31:29.000000000 +0000
@@ -35,7 +35,7 @@
      so=y
 fi
 
-opts='-O3'
+opts=$CFLAGS
 
 $c -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd $opts $def $NOISNAN -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
 ar rv bin/$PLATFORM/liblpsolve55.a `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'`

