diff -urN a/libev/ev.c b/libev/ev.c
--- a/libev/ev.c	2014-01-02 23:06:39.749417954 -0700
+++ b/libev/ev.c	2014-01-02 23:08:08.359034557 -0700
@@ -617,9 +617,11 @@
   #if ECB_GCC_VERSION(4,7)
     /* see comment below (stdatomic.h) about the C11 memory model. */
     #define ECB_MEMORY_FENCE         __atomic_thread_fence (__ATOMIC_SEQ_CST)
-  #elif defined __clang && __has_feature (cxx_atomic)
-    /* see comment below (stdatomic.h) about the C11 memory model. */
-    #define ECB_MEMORY_FENCE         __c11_atomic_thread_fence (__ATOMIC_SEQ_CST)
+  #elif defined __clang
+    #if __has_feature (cxx_atomic)
+      /* see comment below (stdatomic.h) about the C11 memory model. */
+      #define ECB_MEMORY_FENCE         __c11_atomic_thread_fence (__ATOMIC_SEQ_CST)
+    #endif
   #elif ECB_GCC_VERSION(4,4) || defined __INTEL_COMPILER || defined __clang__
     #define ECB_MEMORY_FENCE         __sync_synchronize ()
   #elif _MSC_VER >= 1400 /* VC++ 2005 */
