--- icons/cuyo.desktop~	2007-12-20 18:22:32.000000000 +0100
+++ icons/cuyo.desktop	2008-05-15 16:53:35.000000000 +0200
@@ -1,11 +1,13 @@
 [Desktop Entry]
+Type=Application
 Name=Cuyo
+GenericName=Cuyo
 Icon=cuyo
-Type=Application
+Exec=cuyo
+StartupNotify=true
 Terminal=false
-MimeType=application/x-executable
+MimeType=application/x-executable;
 Categories=Game;BlocksGame;
-Encoding=UTF-8
 Comment=A tetris like game with many levels
+Comment[cs]=Hra podobná tetrisu s mnoha úrovněmi
 Comment[de]=Ein Tetris-artiges Spiel mit vielen Leveln
-Exec=cuyo
