--- avogadro-1.1.0/libavogadro/src/pythonengine_p.h~	2013-03-29 12:46:16.306165881 +0000
+++ avogadro-1.1.0/libavogadro/src/pythonengine_p.h	2013-03-29 12:45:43.123089622 +0000
@@ -27,7 +27,9 @@
 
 #include <avogadro/global.h>
 #include <avogadro/engine.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 namespace Avogadro {
 
--- avogadro-1.1.0/libavogadro/src/pythonextension_p.h~	2013-03-29 12:46:32.156043085 +0000
+++ avogadro-1.1.0/libavogadro/src/pythonextension_p.h	2013-03-29 12:47:07.265771087 +0000
@@ -29,7 +29,9 @@
 #include <avogadro/extension.h>
 #include <avogadro/primitive.h>
 #include <avogadro/glwidget.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 #include <QWidget>
 #include <QList>
--- avogadro-1.1.0/libavogadro/src/pythontool_p.h~	2013-03-29 12:47:33.335569185 +0000
+++ avogadro-1.1.0/libavogadro/src/pythontool_p.h	2013-03-29 12:47:50.732101147 +0000
@@ -27,7 +27,9 @@
 
 #include <avogadro/global.h>
 #include <avogadro/tool.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 #include <QObject>
 #include <QAction>
--- avogadro-1.1.0/libavogadro/src/pythoninterpreter.h~	2013-03-29 12:50:46.550738652 +0000
+++ avogadro-1.1.0/libavogadro/src/pythoninterpreter.h	2013-03-29 12:51:00.853961341 +0000
@@ -26,7 +26,9 @@
 #define PYTHONINTERPRETER_H
 
 #include <avogadro/global.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 #include <avogadro/primitive.h>
 #include <QString>
 
--- avogadro-1.1.0/libavogadro/src/pythonscript.h~	2013-03-29 12:52:00.723488935 +0000
+++ avogadro-1.1.0/libavogadro/src/pythonscript.h	2013-03-29 12:52:11.676745364 +0000
@@ -27,7 +27,9 @@
 #define PYTHONSCRIPT_H
 
 #include <avogadro/global.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 #include "pythonerror.h"
 
