 diff -upNr hardinfo-0.4.2.2.orign/util.c hardinfo-0.4.2.2/util.c
--- hardinfo-0.4.2.2.orign/util.c	2007-07-20 16:45:08.000000000 +0200
+++ hardinfo-0.4.2.2/util.c	2007-07-26 10:15:32.000000000 +0200
@@ -996,7 +996,8 @@ gchar *h_strdup_cprintf(const gchar * fo
     if (source) {
 	retn = g_strconcat(source, buffer, NULL);
 	g_free(buffer);
-        g_free(source);
+        if(strlen(source))
+		 g_free(source);
     } else {
 	retn = buffer;
     }
