--- a/src/sasaeval.cpp	2008-08-01 09:39:01.000000000 +0530
+++ b/src/sasaeval.cpp	2008-08-01 09:40:33.000000000 +0530
@@ -21,6 +21,8 @@
 #include "sasaeval.h"
 
 #include "engine.h"
+#include <algorithm>
+using namespace std;
 
 // the surface area code apparently contains some bugs, since it sometimes
 // crashes. another possibility is that the surface area math contains some
--- a/src/sasaeval.h	2008-07-31 00:40:08.000000000 +0530
+++ b/src/sasaeval.h	2008-08-01 09:40:04.000000000 +0530
@@ -24,6 +24,7 @@
 #define SASAEVAL_H
 
 #include "typedef.h"
+#include <stdlib.h>
 
 struct cg_nbt3_nl;	// SASA neighbor list.
 
