From cacc184cb590591fffe5956294380c48f8390110 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sat, 25 Nov 2017 10:06:57 -0700
Subject: [PATCH] disable LTO

---
 cmake/compiler_settings.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/compiler_settings.cmake b/cmake/compiler_settings.cmake
index 6e318d9..87fe59a 100644
--- a/cmake/compiler_settings.cmake
+++ b/cmake/compiler_settings.cmake
@@ -167,5 +167,5 @@ if (CMAKE_COMPILER_IS_GNUCXX OR QTXDG_COMPILER_IS_CLANGCXX)
         # set(LTO_FLAGS "-flto")
     endif()
     # apply these options to "Release" build type only
-    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${LTO_FLAGS}")
+    #set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${LTO_FLAGS}")
 endif()
-- 
2.14.1

