diff --git i/netplan-configure.service w/netplan-configure.service
index 2b68f08..3e9b6a3 100644
--- i/netplan-configure.service
+++ w/netplan-configure.service
@@ -16,8 +16,8 @@ Type=oneshot
 RemainAfterExit=true
 Environment=NETPLAN_PARSER_IGNORE_ERRORS=1
-ExecStart=/usr/libexec/netplan/configure
+ExecStart=/usr/lib/netplan/configure
 # 'udevadm trigger' is needed to rename interfaces at boot time:
 # https://github.com/canonical/netplan/pull/479
 ExecStartPost=udevadm control --reload
 ExecStartPost=udevadm trigger --action=add --subsystem-match=net

diff --git i/netplan_cli/cli/utils.py w/netplan_cli/cli/utils.py
index d0a85df..3afcc07 100644
--- i/netplan_cli/cli/utils.py
+++ w/netplan_cli/cli/utils.py
@@ -43,10 +43,10 @@ def get_generator_path():
-    return os.environ.get('NETPLAN_GENERATE_PATH', '/usr/libexec/netplan/generate')
+    return os.environ.get('NETPLAN_GENERATE_PATH', '/usr/lib/netplan/generate')
 
 
 def get_configure_path():
-    return os.environ.get('NETPLAN_CONFIGURE_PATH', '/usr/libexec/netplan/configure')
+    return os.environ.get('NETPLAN_CONFIGURE_PATH', '/usr/lib/netplan/configure')
 
 
 def is_nm_snap_enabled():
     return subprocess.call(['systemctl', '--quiet', 'is-enabled', NM_SNAP_SERVICE_NAME], stderr=subprocess.DEVNULL) == 0
