diff --git i/netplan_cli/cli/utils.py w/netplan_cli/cli/utils.py
index d0a85df..e091e71 100644
--- i/netplan_cli/cli/utils.py
+++ w/netplan_cli/cli/utils.py
@@ -35,7 +35,7 @@ config_errors = (ConfigurationError, NetplanException, RuntimeError)
 
 
 def get_generator_path():
-    return os.environ.get('NETPLAN_GENERATE_PATH', '/usr/libexec/netplan/generate')
+    return os.environ.get('NETPLAN_GENERATE_PATH', '/usr/lib/netplan/generate')
 
 
 def is_nm_snap_enabled():
