diff --git i/meson.build w/meson.build
index 0da0d7f..28d845b 100644
--- i/meson.build
+++ w/meson.build
@@ -23,10 +23,6 @@ systemd_generator_dir = systemd.get_variable(pkgconfig: 'systemdsystemgeneratord
 bash_completions_dir = completions.get_variable(pkgconfig: 'completionsdir', default_value: '/etc/bash_completion.d')
 
 # Order: Fedora/Mageia/openSUSE || Debian/Ubuntu
-pyflakes = find_program('pyflakes-3', 'pyflakes3', required: get_option('testing'))
-pycodestyle = find_program('pycodestyle-3', 'pycodestyle', 'pep8', required: get_option('testing'))
-pytest = find_program('pytest-3', 'pytest3', required: get_option('testing'))  # also requires the pytest-cov plugin
-pycoverage = find_program('coverage-3', 'python3-coverage', required: get_option('testing'))
 pandoc = find_program('pandoc', required: false)
 find = find_program('find')
 
@@ -76,14 +72,6 @@ if get_option('testing')
     endif
 
     #FIXME: exclude doc/env/
-    test('linting',
-         pyflakes,
-         timeout: 100,
-         args: [meson.current_source_dir()])
-    test('codestyle',
-         pycodestyle,
-         timeout: 100,
-         args: ['--max-line-length=130', '--exclude=doc/env,*meson-private/pycompile.py', meson.current_source_dir()])
     test('documentation',
          find_program('tests/validate_docs.sh'),
          timeout: 100,
@@ -93,11 +81,6 @@ if get_option('testing')
          timeout: 900,
          env: test_env)
     #TODO: split out dbus tests into own test() instance, to run in parallel
-    test('unit-tests',
-         pycoverage,
-         args: ['run', '-a', '-m', 'pytest', '-s', '-v', '--cov-append', meson.current_source_dir()],
-         timeout: 600,
-         env: test_env)
 
     #TODO: the coverage section should probably be cleaned up a bit
     if get_option('b_coverage')
