
Index: devil/build-aux/ltmain.sh
===================================================================
--- devil.orig/build-aux/ltmain.sh	2008-12-30 21:05:37.000000000 +0000
+++ devil/build-aux/ltmain.sh	2008-12-30 21:09:12.000000000 +0000
@@ -4189,6 +4189,11 @@
 	arg=$func_stripname_result
 	;;
 
+    -Wl,--as-needed)
+        deplibs="$deplibs $arg"
+        continue
+    ;;
+
       -Wl,*)
 	func_stripname '-Wl,' '' "$arg"
 	args=$func_stripname_result
@@ -4543,6 +4548,15 @@
 	lib=
 	found=no
 	case $deplib in
+    -Wl,--as-needed)
+        if test "$linkmode,$pass" = "prog,link"; then
+            compile_deplibs="$deplib $compile_deplibs"
+            finalize_deplibs="$deplib $finalize_deplibs"
+        else
+            deplibs="$deplib $deplibs"
+        fi
+        continue
+    ;;
 	-mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe|-threads)
 	  if test "$linkmode,$pass" = "prog,link"; then
 	    compile_deplibs="$deplib $compile_deplibs"

