--- polymake-3.6/bundled/libnormaliz/support/configure.pl.orig	2019-12-11 12:29:31.089671103 +0000
+++ polymake-3.6/bundled/libnormaliz/support/configure.pl	2019-12-11 12:29:52.549969615 +0000
@@ -69,7 +69,7 @@
    }
 
    if ($nmz_path ne "bundled") {
-      my $error=Polymake::Configure::build_test_program(<<'---', LIBS => "-lnormaliz -lgmpxx -lgmp", CXXFLAGS => "$CXXFLAGS", LDFLAGS => "$LDFLAGS");
+      my $error=Polymake::Configure::build_test_program(<<'---', LIBS => "-lnormaliz -lgmpxx -lgmp -leantic", CXXFLAGS => "$CXXFLAGS", LDFLAGS => "$LDFLAGS");
 
 #include <cstddef>
 #include <vector>
