--- polymake-3.1/support/configure.pl.orig	2017-12-03 09:21:44.528067697 +0000
+++ polymake-3.1/support/configure.pl	2017-12-03 09:25:17.258583093 +0000
@@ -450,6 +450,8 @@
 $Libs     =$vars{LIBS}     || "";
 
 $LDsharedFlags=$Config::Config{lddlflags};
+$LDsharedFlags =~ s/,-z,now//g;
 $LDcallableFlags= $options{callable} eq ".none." ? "none" : "$LDsharedFlags $Config::Config{ldflags}";
+$LDcallableFlags =~ s/,-z,now//g;
 
 print "checking C++ library ... ";
