--- libvo/vo_gif89a.c.orig	2014-06-11 10:06:29.243216331 +0000
+++ libvo/vo_gif89a.c	2014-06-11 10:08:37.346078847 +0000
@@ -75,6 +75,9 @@
 #define MakeMapObject GifMakeMapObject
 #define FreeMapObject GifFreeMapObject
 #define QuantizeBuffer GifQuantizeBuffer
+#if defined GIFLIB_MINOR && GIFLIB_MINOR >= 1
+#define EGifCloseFile(a) EGifCloseFile(a, NULL)
+#endif
 #endif
 
 // how many frames per second we are aiming for during output.
--- libmpdemux/demux_gif.c.orig	2014-06-11 10:46:48.120883788 +0000
+++ libmpdemux/demux_gif.c	2014-06-11 10:47:28.290732272 +0000
@@ -50,6 +50,9 @@
 #define DGifOpenFileHandle(a) DGifOpenFileHandle(a, NULL)
 #define GifError() (gif ? gif->Error : 0)
 #define GifErrorString() GifErrorString(gif->Error)
+#if defined GIFLIB_MINOR && GIFLIB_MINOR >= 1
+#define DGifCloseFile(a) DGifCloseFile(a, NULL)
+#endif
 #endif
 
 /* >= 4.2 prior GIFLIB did not have MAJOR/MINOR defines */
