--- mplayer/configure.old	2009-04-16 12:02:10.000000000 +0200
+++ mplayer/configure	2009-05-22 15:23:38.000000000 +0200
@@ -6410,6 +6410,7 @@
 def_liba52='#undef CONFIG_LIBA52'
 def_liba52_internal="#undef CONFIG_LIBA52_INTERNAL"
 if test "$_liba52_internal" = yes ; then
+	test "$cc_vendor" = gnu && test "$cc_version" = 4.4.0 && CFLAGS=$(echo $CFLAGS|sed "s/ *-O4 */ -O2 /")
   _liba52=yes
   def_liba52_internal="#define CONFIG_LIBA52_INTERNAL 1"
   _res_comment="internal"
