diff -ur Pulse-Eight-libcec-178d498/src/lib/adapter/RPi/RPiCECAdapterCommunication.cpp libcec-rpi-2.0.4/src/lib/adapter/RPi/RPiCECAdapterCommunication.cpp
--- Pulse-Eight-libcec-178d498/src/lib/adapter/RPi/RPiCECAdapterCommunication.cpp	2012-11-17 02:33:33.000000000 +0000
+++ libcec-rpi-2.0.4/src/lib/adapter/RPi/RPiCECAdapterCommunication.cpp	2012-11-27 07:36:53.510561048 +0000
@@ -264,7 +264,7 @@
   if (bStartListening)
   {
     // enable passive mode
-    vc_cec_set_passive(true);
+    vc_cec_set_passive((bool_t)true);
 
     // register the callback
     vc_cec_register_callback(((CECSERVICE_CALLBACK_T)rpi_cec_callback), (void*)this);
@@ -320,7 +320,7 @@
   UnregisterLogicalAddress();
 
   // disable passive mode
-  vc_cec_set_passive(false);
+  vc_cec_set_passive((bool_t)false);
 
   if (!g_bHostInited)
   {
diff -ur Pulse-Eight-libcec-178d498/src/lib/adapter/RPi/RPiCECAdapterMessageQueue.cpp libcec-rpi-2.0.4/src/lib/adapter/RPi/RPiCECAdapterMessageQueue.cpp
--- Pulse-Eight-libcec-178d498/src/lib/adapter/RPi/RPiCECAdapterMessageQueue.cpp	2012-11-17 02:33:33.000000000 +0000
+++ libcec-rpi-2.0.4/src/lib/adapter/RPi/RPiCECAdapterMessageQueue.cpp	2012-11-27 07:40:36.500150455 +0000
@@ -182,7 +182,7 @@
   LIB_CEC->AddLog(CEC_LOG_DEBUG, "sending data: %s", strDump.c_str());
 #endif
 
-   int iReturn = vc_cec_send_message((uint32_t)command.destination, (uint8_t*)&payload, iLength, bIsReply);
+   int iReturn = vc_cec_send_message((uint32_t)command.destination, (uint8_t*)&payload, iLength, (bool_t)bIsReply);
 #endif
 
   if (iReturn != VCHIQ_SUCCESS)
