Index: telnet/externs.h
===================================================================
RCS file: /cvsroot/inetutils/inetutils/telnet/externs.h,v
retrieving revision 1.14
diff -p -u -r1.14 externs.h
--- telnet/externs.h	27 Dec 2008 20:36:00 -0000	1.14
+++ telnet/externs.h	3 Jan 2009 21:29:23 -0000
@@ -49,6 +49,8 @@
    to the Free Software Foundation, Inc., 51 Franklin Street,
    Fifth Floor, Boston, MA 02110-1301 USA. */
 
+#include <unistd.h>
+
 #ifndef BSD
 # define BSD 43
 #endif
