From 1d948fd58929eb40ef1c855b6754dbea3f90ab6d Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sat, 23 May 2026 09:55:59 -0600
Subject: [PATCH] add missing include

---
 deps/v8/src/base/memcopy.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/deps/v8/src/base/memcopy.h b/deps/v8/src/base/memcopy.h
index b1d8a36b4cf..82e9a75c24a 100644
--- a/deps/v8/src/base/memcopy.h
+++ b/deps/v8/src/base/memcopy.h
@@ -5,6 +5,7 @@
 #ifndef V8_BASE_MEMCOPY_H_
 #define V8_BASE_MEMCOPY_H_
 
+#include <climits>
 #include <stdlib.h>
 
 #include <atomic>
-- 
2.36.1

