diff -Naur mythtv-0.21-old/libs/libmythtv/mpeg/pespacket.cpp mythtv-0.21/libs/libmythtv/mpeg/pespacket.cpp
--- mythtv-0.21-old/libs/libmythtv/mpeg/pespacket.cpp	2006-09-08 04:55:03.000000000 +1000
+++ mythtv-0.21/libs/libmythtv/mpeg/pespacket.cpp	2009-05-09 19:46:15.000000000 +1000
@@ -6,10 +6,10 @@
 
 extern "C" {
 #include "mythconfig.h"
-#include "../libavcodec/avcodec.h"
-#include "../libavformat/avformat.h"
-#include "../libavutil/crc.h"
-#include "../libavutil/bswap.h"
+#include "libavcodec/avcodec.h"
+#include "libavformat/avformat.h"
+#include "libavutil/crc.h"
+#include "libavutil/bswap.h"
 }
 
 #include <vector>
