diff -Nur udev-128.orig/extras/Makefile.in udev-128/extras/Makefile.in
--- udev-128.orig/extras/Makefile.in	2008-09-11 16:58:01.000000000 +0200
+++ udev-128/extras/Makefile.in	2008-09-18 12:03:09.000000000 +0200
@@ -176,15 +176,11 @@
 udev_prefix = @udev_prefix@
 SUBDIRS = \
 	ata_id \
-	cdrom_id \
 	edd_id \
 	path_id \
 	firmware \
-	collect \
 	floppy \
-	fstab_import \
 	rule_generator \
-	scsi_id \
 	usb_id \
 	volume_id
 
diff -Nur udev-128.orig/extras/volume_id/lib/libvolume_id-private.h udev-128/extras/volume_id/lib/libvolume_id-private.h
--- udev-128.orig/extras/volume_id/lib/libvolume_id-private.h	2008-09-10 02:37:09.000000000 +0200
+++ udev-128/extras/volume_id/lib/libvolume_id-private.h	2008-09-18 12:07:57.000000000 +0200
@@ -35,12 +35,8 @@
 log_null(const char *format, ...) {}
 
 #define err(format, arg...)	volume_id_log_fn(LOG_ERR, __FILE__, __LINE__, format, ##arg)
-#define info(format, arg...)	volume_id_log_fn(LOG_INFO, __FILE__, __LINE__, format, ##arg)
-#ifdef DEBUG
-#define dbg(format, arg...)	volume_id_log_fn(LOG_DEBUG, __FILE__, __LINE__, format, ##arg)
-#else
+#define info(format, arg...)    log_null(format, ##arg)
 #define dbg(format, arg...)	log_null(format, ##arg)
-#endif
 
 #if (__BYTE_ORDER == __LITTLE_ENDIAN)
 #define le16_to_cpu(x) (x)
diff -Nur udev-128.orig/extras/volume_id/lib/md5.c udev-128/extras/volume_id/lib/md5.c
--- udev-128.orig/extras/volume_id/lib/md5.c	2008-09-10 02:18:59.000000000 +0200
+++ udev-128/extras/volume_id/lib/md5.c	2008-09-18 12:03:09.000000000 +0200
@@ -21,7 +21,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include <stdint.h>
+/*#include <stdint.h>*/
 #include <string.h>
 #include <endian.h>
 #include <byteswap.h>
