diff -u -Nr gnome-shell-3.6.3.1/js/ui/components/networkAgent.js gnome-shell-3.6.3.1-nmlibexecdir/js/ui/components/networkAgent.js
--- gnome-shell-3.6.3.1/js/ui/components/networkAgent.js	2013-02-20 20:31:03.000000000 +0100
+++ gnome-shell-3.6.3.1-nmlibexecdir/js/ui/components/networkAgent.js	2013-03-10 17:13:46.120907065 +0100
@@ -686,7 +686,7 @@
                     } catch(e) { } // ignore errors if key does not exist
                     let path = binary;
                     if (!GLib.path_is_absolute(path)) {
-                        path = GLib.build_filenamev([Config.LIBEXECDIR, path]);
+                        path = GLib.build_filenamev(["/usr/lib/networkmanager", path]);
                     }
 
                     if (GLib.file_test(path, GLib.FileTest.IS_EXECUTABLE))
