--- trunk/mcs/class/System.Web.Services/System.Web.Services.Description/BasicProfileChecker.cs	2008/07/01 20:17:46	107006
+++ trunk/mcs/class/System.Web.Services/System.Web.Services.Description/BasicProfileChecker.cs	2008/10/28 09:32:46	117243
@@ -177,7 +177,7 @@
 				foreach (OperationMessage om in op.Messages) {
 					Message msg = ctx.Services.GetMessage (om.Message);
 					foreach (MessagePart part in msg.Parts)
-						parts.Add (part,part);
+						parts [part] = part; // do not use Add() - there could be the same MessagePart instance.
 				}
 			}
 			
