diff -urN a/configure b/configure
--- a/configure	2016-12-09 13:09:49.000000000 -0700
+++ b/configure	2017-03-31 18:28:40.998404499 -0600
@@ -22270,51 +22270,6 @@
 
 
   case $target_cpu in
-    arm*)
-      { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether C compiler accepts -mfpu=neon" >&5
-$as_echo_n "checking whether C compiler accepts -mfpu=neon... " >&6; }
-if ${ax_cv_check_cflags___mfpu_neon+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-
-  ax_check_save_flags=$CFLAGS
-  CFLAGS="$CFLAGS  -mfpu=neon"
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-int
-main ()
-{
-
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_compile "$LINENO"; then :
-  ax_cv_check_cflags___mfpu_neon=yes
-else
-  ax_cv_check_cflags___mfpu_neon=no
-fi
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
-  CFLAGS=$ax_check_save_flags
-fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ax_cv_check_cflags___mfpu_neon" >&5
-$as_echo "$ax_cv_check_cflags___mfpu_neon" >&6; }
-if test x"$ax_cv_check_cflags___mfpu_neon" = xyes; then :
-  ax_cv_support_neon_ext=yes
-else
-  :
-fi
-
-      if test x"$ax_cv_support_neon_ext" = x"yes"; then
-        ARM_NEON_FLAGS="-mfpu=neon -DARM_NEON"
-
-        ARM_FLAGS="$ARM_FLAGS $ARM_NEON_FLAGS"
-
-$as_echo "#define HAVE_NEON /**/" >>confdefs.h
-
-      fi
-    ;;
     aarch64*)
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether C compiler accepts -march=armv8-a" >&5
 $as_echo_n "checking whether C compiler accepts -march=armv8-a... " >&6; }
