diff -urN a/src/rocksdb/Makefile b/src/rocksdb/Makefile
--- a/src/rocksdb/Makefile	2016-06-14 05:44:41.000000000 -0600
+++ b/src/rocksdb/Makefile	2016-09-10 12:40:00.021152948 -0600
@@ -84,9 +84,6 @@
 # compile with -O2 if debug level is not 2
 ifneq ($(DEBUG_LEVEL), 2)
 OPT += -O2 -fno-omit-frame-pointer
-ifneq ($(MACHINE),ppc64) # ppc64 doesn't support -momit-leaf-frame-pointer
-OPT += -momit-leaf-frame-pointer
-endif
 endif
 
 # if we're compiling for release, compile without debug code (-DNDEBUG) and
