diff -uNr a/plugins/php/php_plugin.c b/plugins/php/php_plugin.c
--- a/plugins/php/php_plugin.c	2022-10-06 19:52:07.237882916 +0200
+++ b/plugins/php/php_plugin.c	2022-10-06 19:56:29.232418542 +0200
@@ -607,7 +607,11 @@
 static int php_uwsgi_startup(sapi_module_struct *sapi_module)
 {
 
+#if PHP_VERSION_ID >= 80200
+	if (php_module_startup(&uwsgi_sapi_module, &uwsgi_module_entry)==FAILURE) {
+#else
 	if (php_module_startup(&uwsgi_sapi_module, &uwsgi_module_entry, 1)==FAILURE) {
+#endif
 		return FAILURE;
 	} else {
 		return SUCCESS;
