--- Makefile.in.old	2008-10-19 17:56:51.000000000 +0200
+++ Makefile.in	2008-10-19 17:57:58.000000000 +0200
@@ -68,8 +68,8 @@
 ifdef _MSC_VER
 LUCENE_LIB = $(call EXPAND_LIBNAME_PATH,clucene,$(DEPTH)/flock/clucene)
 else
-INCLUDES += -I$(CLUCENE_DIR)/include -I$(CLUCENE_DIR)/lib
-LUCENE_LIB = -L$(CLUCENE_DIR)/lib -lclucene
+INCLUDES += -I /usr/include -I /usr/lib
+LUCENE_LIB = /usr/lib/libclucene.$(LIB_SUFFIX)
 endif
 
 EXTRA_DSO_LDOPTS = $(XPCOM_GLUE_LDOPTS) $(NSPR_LIBS) $(LUCENE_LIB)
