--- src/command.C.old	2009-05-28 15:21:04.457772957 +0200
+++ src/command.C	2009-05-28 15:18:52.684597530 +0200
@@ -3336,7 +3336,7 @@
           }
         else
           {
-            char *eq = strchr (str, '='); // constness lost, but verified to be ok
+            char *eq = (char *) strchr ( str, '='); // constness lost, but verified to be ok
 
             if (eq)
               {
--- src/misc.C.old	2009-05-28 15:21:16.400270603 +0200
+++ src/misc.C	2009-05-28 15:21:51.697108339 +0200
@@ -170,7 +170,7 @@
 char *
 rxvt_basename (const char *str) NOTHROW
 {
-  char *base = strrchr (str, '/');
+  char *base = (char *) strrchr (str, '/');
 
   return (char *) (base ? base + 1 : str);
 }
