diff -Naur rssh-2.3.2-orig/Makefile.am rssh-2.3.2/Makefile.am
--- rssh-2.3.2-orig/Makefile.am	2006-01-03 09:34:59.000000000 -0800
+++ rssh-2.3.2/Makefile.am	2006-06-29 23:38:56.000000000 -0700
@@ -16,7 +16,7 @@
 	$(CC) -c $(DEFS) $(ourdefs) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $<
 
 install-exec-hook:
-	chmod u+s $(libexecdir)/rssh_chroot_helper
+	chmod u+s $(DESTDIR)$(libexecdir)/rssh_chroot_helper
 
 rpm:	dist
 	rpmbuild -ta --sign $(base).tar.gz
diff -Naur rssh-2.3.2-orig/Makefile.in rssh-2.3.2/Makefile.in
--- rssh-2.3.2-orig/Makefile.in	2006-01-06 18:24:57.000000000 -0800
+++ rssh-2.3.2/Makefile.in	2006-06-29 23:38:30.000000000 -0700
@@ -728,7 +728,7 @@
 	$(CC) -c $(DEFS) $(ourdefs) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $<
 
 install-exec-hook:
-	chmod u+s $(libexecdir)/rssh_chroot_helper
+	chmod u+s $(DESTDIR)$(libexecdir)/rssh_chroot_helper
 
 rpm:	dist
 	rpmbuild -ta --sign $(base).tar.gz
