diff -up 1.0.0.rc15/lib/format/ataraid/isw.c~ 1.0.0.rc15/lib/format/ataraid/isw.c
--- 1.0.0.rc15/lib/format/ataraid/isw.c~	2008-09-17 08:58:52.000000000 -0400
+++ 1.0.0.rc15/lib/format/ataraid/isw.c	2009-02-05 07:04:35.576126020 -0500
@@ -155,7 +155,7 @@ name(struct lib_context *lc, struct raid
 
 			while (i--) {
 				if (disk == isw->disk + i) {
-					id = i % 2;
+					id = i / 2;
 					goto ok;
 				}
 			}
