diff -wbBur avifile-0.7-0.7.45/avifile.pc avifile-0.7-0.7.45.my/avifile.pc
--- avifile-0.7-0.7.45/avifile.pc	2006-03-06 12:12:26.000000000 +0300
+++ avifile-0.7-0.7.45.my/avifile.pc	2009-05-25 13:10:17.347697640 +0400
@@ -1,4 +1,4 @@
-prefix=/var/tmp/avi6
+prefix=/usr
 exec_prefix=${prefix}
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include/avifile-0.7
diff -wbBur avifile-0.7-0.7.45/include/avm_map.h avifile-0.7-0.7.45.my/include/avm_map.h
--- avifile-0.7-0.7.45/include/avm_map.h	2004-02-19 14:39:56.000000000 +0300
+++ avifile-0.7-0.7.45.my/include/avm_map.h	2009-05-25 13:10:21.014364635 +0400
@@ -45,7 +45,7 @@
 
     template <class Key1, class Value1> struct binary_tree_node
     {
-	avm_map::pair<Key1, Value1>* entry;
+	struct pair<Key1, Value1>* entry;
 	binary_tree_node<Key1, Value1>* left;
 	Key1 minval;
 	binary_tree_node<Key1, Value1>* right;
diff -wbBur avifile-0.7-0.7.45/include/version.h avifile-0.7-0.7.45.my/include/version.h
--- avifile-0.7-0.7.45/include/version.h	2006-03-06 12:12:25.000000000 +0300
+++ avifile-0.7-0.7.45.my/include/version.h	2009-05-25 13:10:16.841030758 +0400
@@ -3,5 +3,5 @@
 #define AVIFILE_MINOR_VERSION 7
 #define AVIFILE_PATCHLEVEL 45
 #define AVIFILE_VERSION  ((AVIFILE_MAJOR_VERSION << 16) + (AVIFILE_MINOR_VERSION << 8) + AVIFILE_PATCHLEVEL)
-#define AVIFILE_BUILD "CVS-0.7.45-060306-10:11-4.0.3"
+#define AVIFILE_BUILD "RELEASE-0.7.45-090525-13:10-4.4.0"
 #endif
diff -wbBur avifile-0.7-0.7.45/lib/aviplay/aviplay.cpp avifile-0.7-0.7.45.my/lib/aviplay/aviplay.cpp
--- avifile-0.7-0.7.45/lib/aviplay/aviplay.cpp	2005-11-05 00:56:33.000000000 +0300
+++ avifile-0.7-0.7.45.my/lib/aviplay/aviplay.cpp	2009-05-25 13:13:36.304347364 +0400
@@ -242,7 +242,7 @@
 	//printf("VSTREAMTIME %f\n", m_pVideostream->GetTime());
 	// reading subtitles
 	avm::string fn = m_Filename;
-	char* p = strrchr(fn.c_str(), '.');
+	char* p = (char*)strrchr(fn.c_str(), '.');
 	if (p)
             *p = 0;
 	InitSubtitles(m_Subfilename.size()
diff -wbBur avifile-0.7-0.7.45/lib/codeckeeper.cpp avifile-0.7-0.7.45.my/lib/codeckeeper.cpp
--- avifile-0.7-0.7.45/lib/codeckeeper.cpp	2004-03-10 16:27:38.000000000 +0300
+++ avifile-0.7-0.7.45.my/lib/codeckeeper.cpp	2009-05-25 13:15:11.971004390 +0400
@@ -143,7 +143,7 @@
     {
 	char plgn[100];
 	strcpy(plgn, "avm_codec_plugin_");
-	char* x = strrchr(name, '/');
+	char* x = (char*)strrchr(name, '/');
 	strncat(plgn, x + 1, 50);
 	x = strchr(plgn, '.');
 	*x = 0;
diff -wbBur avifile-0.7-0.7.45/lib/common/Args.cpp avifile-0.7-0.7.45.my/lib/common/Args.cpp
--- avifile-0.7-0.7.45/lib/common/Args.cpp	2003-02-06 23:39:12.000000000 +0300
+++ avifile-0.7-0.7.45.my/lib/common/Args.cpp	2009-05-25 13:14:38.562055811 +0400
@@ -17,7 +17,7 @@
 	{
             char* c = cp;
 	    char* b;
-	    while ((b = strchr(c, ':')))
+	    while ((b = (char*)strchr(c, ':')))
 	    {
 		*b++ = 0;
 		if (strlen(c))
@@ -259,7 +259,7 @@
     }
     for (unsigned i = 0; i < arr.size(); i++)
     {
-	char* par = strchr(arr[i], '=');
+	char* par = (char*)strchr(arr[i], '=');
 	if (par)
 	{
 	    *par = 0;
@@ -335,7 +335,7 @@
 	    {
 		for (unsigned j = 1; j < arr.size(); j++)
 		{
-		    char* p = strchr(arr[j].c_str(), '=');
+		    char* p = (char*)strchr(arr[j].c_str(), '=');
 		    int val = 0;
 		    bool valid = false;
 		    if (p)
diff -wbBur avifile-0.7-0.7.45/player/playercontrol.cpp avifile-0.7-0.7.45.my/player/playercontrol.cpp
--- avifile-0.7-0.7.45/player/playercontrol.cpp	2005-09-27 17:40:09.000000000 +0400
+++ avifile-0.7-0.7.45.my/player/playercontrol.cpp	2009-05-25 13:16:18.274330659 +0400
@@ -764,7 +764,7 @@
 	return;
 
     avm::string path = RS("url", ".");
-    char* cut = strrchr(path.c_str(), '/');
+    char* cut = (char*)strrchr(path.c_str(), '/');
     if (!cut)
 	path = "./";
 #if QT_VERSION <= 220
diff -wbBur avifile-0.7-0.7.45/plugins/libmp3lame_audioenc/mp3encoder.cpp avifile-0.7-0.7.45.my/plugins/libmp3lame_audioenc/mp3encoder.cpp
--- avifile-0.7-0.7.45/plugins/libmp3lame_audioenc/mp3encoder.cpp	2006-03-05 23:45:26.000000000 +0300
+++ avifile-0.7-0.7.45.my/plugins/libmp3lame_audioenc/mp3encoder.cpp	2009-05-25 13:10:21.014364635 +0400
@@ -30,7 +30,7 @@
 	short  nCodecDelay;
     };
 public:
-    MP3Encoder::MP3Encoder(const CodecInfo& info, const WAVEFORMATEX* format)
+    MP3Encoder(const CodecInfo& info, const WAVEFORMATEX* format)
 	:IAudioEncoder(info)
     {
 	in_fmt=*format;
diff -wbBur avifile-0.7-0.7.45/samples/qtvidcap/kv4lsetup.cpp avifile-0.7-0.7.45.my/samples/qtvidcap/kv4lsetup.cpp
--- avifile-0.7-0.7.45/samples/qtvidcap/kv4lsetup.cpp	2005-03-23 18:18:21.000000000 +0300
+++ avifile-0.7-0.7.45.my/samples/qtvidcap/kv4lsetup.cpp	2009-05-25 13:10:21.014364635 +0400
@@ -229,7 +229,7 @@
     if (XF86DGAQueryExtension(dpy, &foo, &bar)) {
 	XF86DGAQueryDirectVideo(dpy, XDefaultScreen(dpy),&flags);
 	if (flags & XF86DGADirectPresent) {
-	    XF86DGAGetVideoLL(dpy,XDefaultScreen(dpy),(int*)&base,&width,&foo,&bar);
+	    XF86DGAGetVideoLL(dpy,XDefaultScreen(dpy),(unsigned int*)&base,&width,&foo,&bar);
 	    set_bpl  = width * set_bpp/8;
 	    set_base = base;
 	    if (verbose == 2) {
diff -wbBur avifile-0.7-0.7.45/samples/qtvidcap/v4lxif.cpp avifile-0.7-0.7.45.my/samples/qtvidcap/v4lxif.cpp
--- avifile-0.7-0.7.45/samples/qtvidcap/v4lxif.cpp	2005-03-23 19:16:09.000000000 +0300
+++ avifile-0.7-0.7.45.my/samples/qtvidcap/v4lxif.cpp	2009-05-25 13:10:21.014364635 +0400
@@ -1115,7 +1115,7 @@
 		     major,minor, XDGA_MAJOR_VERSION,XDGA_MINOR_VERSION);
 		have_dga= false;
 	    } else {
-		XF86DGAGetVideoLL(disp, DefaultScreen(disp), (int*)&base, &width, &bank, &ram);
+		XF86DGAGetVideoLL(disp, DefaultScreen(disp), (unsigned int*)&base, &width, &bank, &ram);
 		if (!base)
 		    avml(AVML_WARN,
 			 "v4l1: can not allocate frame buffer base: %p\n", base);
