diff -urN motion-3.2.12-OLD/motion.h motion-3.2.12-NEW/motion.h
--- motion-3.2.12-OLD/motion.h	2010-05-31 23:48:23.000000000 -0700
+++ motion-3.2.12-NEW/motion.h	2011-08-30 02:11:08.000000000 -0700
@@ -40,7 +40,7 @@

 #define _LINUX_TIME_H 1
 #if !defined(WITHOUT_V4L) && !defined(BSD)
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #endif

 #include <pthread.h>
diff -urN motion-3.2.12-OLD/track.c motion-3.2.12-NEW/track.c
--- motion-3.2.12-OLD/track.c	2010-05-31 23:48:23.000000000 -0700
+++ motion-3.2.12-NEW/track.c	2011-08-30 23:05:09.000000000 -0700
@@ -11,6 +11,7 @@
 #include "motion.h"

 #ifndef WITHOUT_V4L
+#include <linux/videodev2.h>
 #include "pwc-ioctl.h"
 #endif

diff -urN motion-3.2.12-OLD/video.h motion-3.2.12-NEW/video.h
--- motion-3.2.12-OLD/video.h	2010-05-31 23:48:23.000000000 -0700
+++ motion-3.2.12-NEW/video.h	2011-08-30 02:11:12.000000000 -0700
@@ -12,7 +12,7 @@

 #define _LINUX_TIME_H 1
 #ifndef WITHOUT_V4L
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #include <sys/mman.h>
 #include "pwc-ioctl.h"
 #endif
