Index: trunk/src/common/helpers.py
===================================================================
--- trunk/src/common/helpers.py (revision 9282)
+++ trunk/src/common/helpers.py (revision 9283)
@@ -794,5 +794,5 @@
 
 		# our last chance, ask uname and strip it
-		uname_output = get_output_of_command('uname -a | cut -d" " -f1,3')
+		uname_output = get_output_of_command('uname -sr')
 		if uname_output is not None:
 			return uname_output[0] # only first line
