--- Makefile.in.orig	2016-05-15 08:11:58.074191520 +0000
+++ Makefile.in	2016-05-15 08:12:25.599007513 +0000
@@ -11,7 +11,6 @@
 
 DVDPCODECS_DIRS= \
 	lib \
-	lib/libdvd
 
 VideoPlayer_ARCHIVES=xbmc/cores/VideoPlayer/VideoPlayer.a \
                    xbmc/cores/VideoPlayer/DVDCodecs/DVDCodecs.a \
@@ -341,7 +340,6 @@
 	$(MAKE) -C lib/addons/library.xbmc.pvr
 	$(MAKE) -C lib/addons/library.kodi.inputstream
 dvdpcodecs: dllloader
-	$(MAKE) -C lib/libdvd
 
 dvdpextcodecs:
 
