diff -ru podsleuth-0.6.4.old/m4/podsleuth.m4 podsleuth-0.6.4/m4/podsleuth.m4
--- podsleuth-0.6.4.old/m4/podsleuth.m4	2007-12-19 22:37:05.000000000 +0000
+++ podsleuth-0.6.4/m4/podsleuth.m4	2009-04-04 18:35:56.000000000 +0000
@@ -32,8 +32,8 @@
 
 AC_DEFUN([PODSLEUTH_CHECK_SGUTILS],
 [
-	AC_CHECK_LIB(sgutils, sg_ll_inquiry, [], 
-		AC_MSG_ERROR([Error! You need to have libsgutils.])
+	AC_CHECK_LIB(sgutils2, sg_ll_inquiry, [], 
+		AC_MSG_ERROR([Error! You need to have libsgutils2.])
 	)
 ])
 
diff -ru podsleuth-0.6.4.old/src/PodSleuth/PodSleuth/ScsiReader.cs podsleuth-0.6.4/src/PodSleuth/PodSleuth/ScsiReader.cs
--- podsleuth-0.6.4.old/src/PodSleuth/PodSleuth/ScsiReader.cs	2007-09-17 07:51:40.000000000 +0000
+++ podsleuth-0.6.4/src/PodSleuth/PodSleuth/ScsiReader.cs	2009-04-04 18:36:45.000000000 +0000
@@ -50,7 +50,7 @@
             }
         }
         
-        [DllImport("libsgutils.so.1")]
+        [DllImport("libsgutils2.so.2")]
         private static extern int sg_ll_inquiry(int sg_fd, int cmddt, int evpd, int pg_op,
             IntPtr resp, int mx_resp_len, int noisy, int verbose);
             
@@ -74,4 +74,4 @@
             }
         }
     }
-}
\ No newline at end of file
+}
diff -ru podsleuth-0.6.4.old/src/PodSleuth/PodSleuth.dll.config.in podsleuth-0.6.4/src/PodSleuth/PodSleuth.dll.config.in
--- podsleuth-0.6.4.old/src/PodSleuth/PodSleuth.dll.config.in	2007-09-17 07:51:40.000000000 +0000
+++ podsleuth-0.6.4/src/PodSleuth/PodSleuth.dll.config.in	2009-04-04 18:36:29.000000000 +0000
@@ -1,4 +1,4 @@
 <configuration>
-	<dllmap dll="libsgutils.so.1" target="libsgutils.so.1" />
+	<dllmap dll="libsgutils2.so.2" target="libsgutils2.so.2" />
 </configuration>
 
