Index: content/fetchers/curl.c
===================================================================
--- netsurf/content/fetchers/curl.c	(revision 12174)
+++ netsurf/content/fetchers/curl.c	(working copy)
@@ -586,7 +586,7 @@
 		SETOPT(CURLOPT_PROXY, NULL);
 	}
 
-	if (urldb_get_cert_permissions(f->url)) {
+	if (1) { //urldb_get_cert_permissions(f->url)) {
 		/* Disable certificate verification */
 		SETOPT(CURLOPT_SSL_VERIFYPEER, 0L);
 		SETOPT(CURLOPT_SSL_VERIFYHOST, 0L);
