Index: content/content.c
===================================================================
--- netsurf/content/content.c	(revision 12174)
+++ netsurf/content/content.c	(working copy)
@@ -1306,6 +1306,9 @@
  */
 int content_get_width(hlcache_handle *h)
 {
+	if (h == NULL)
+		return 0;
+
 	return content__get_width(hlcache_handle_get_content(h));
 }
 
@@ -1325,6 +1328,9 @@
  */
 int content_get_height(hlcache_handle *h)
 {
+	if (h == NULL)
+		return 0;
+
 	return content__get_height(hlcache_handle_get_content(h));
 }
 
