--- ./src/makemake.in.orig	2013-02-16 15:26:28.000000000 -0700
+++ ./src/makemake.in	2015-02-11 17:20:00.000000000 -0700
@@ -1217,7 +1217,7 @@ if [ $XCC_GCC = true ] ; then
   if [ $CROSS = false ] ; then
     case "$XCC_GCC_VERSION" in
       # gcc 2.7 introduced an annoying warning, but gcc 2.8 has a workaround:
-      2.[8-9]* | 3.* | 4.* | egcs-2.*)
+      2.[8-9]* | 3.* | 4.* | 5.* | egcs-2.*)
         XCFLAGS=$XCFLAGS' -Wno-sign-compare -Wno-format-nonliteral'
         ;;
       * ) ;;
@@ -1226,7 +1226,7 @@ if [ $XCC_GCC = true ] ; then
   if [ $CROSS = false -a $CC_CPLUSPLUS = true ] ; then
     case "$XCC_GCC_VERSION" in
       # g++ 3.4 introduced an annoying warning, but has a workaround:
-      3.[4-9]* | 4.*)
+      3.[4-9]* | 4.* | 5.*)
         XCFLAGS=$XCFLAGS' -Wno-invalid-offsetof'
         ;;
       * ) ;;
@@ -1241,7 +1241,7 @@ if [ $XCC_GCC = true ] ; then
             # gcc-2.7.2 has a strength-reduction bug (fixed in gcc-2.7.2.1,
             # also fixed through the specs file of some Linux distributions).
             XCFLAGS=$XCFLAGS' -O2 -fexpensive-optimizations -fno-strength-reduce' ;;
-          2.3.[2-9]* | 2.[4-9]* | 3.* | 4.* | egcs-2.*)
+          2.3.[2-9]* | 2.[4-9]* | 3.* | 4.* | 5.* | egcs-2.*)
             XCFLAGS=$XCFLAGS' -O2 -fexpensive-optimizations' ;;
           * )
             XCFLAGS=$XCFLAGS' -O' ;;
@@ -1295,7 +1295,7 @@ if [ $XCC_GCC = true ] ; then
     esac
   fi # with_debug
   case "$XCC_GCC_VERSION" in
-    3* | 4*)
+    3* | 4* | 5*)
       # When not optimizing on linux, or with Apple's gcc 4 on MacOS X,
       # or with GCC 4 when "gcc -Os" is used:
       #   C_CODE_ALIGNMENT is wrong. &EVAL-WHEN = 0x806D512
