diff -up clisp-2.47/modules/libsvm/svm.cpp clisp-2.47/modules/libsvm/svm
diff -up clisp-2.47/src/time.d.cpp clisp-2.47/src/time.d
--- clisp-2.47/src/time.d.cpp	2009-03-01 20:14:24.000000000 +0100
+++ clisp-2.47/src/time.d	2009-03-01 20:14:46.000000000 +0100
@@ -302,7 +302,7 @@ global object internal_time_to_I (const 
  #ifdef TIME_WIN32
   return L2_to_I(tp->dwHighDateTime,tp->dwLowDateTime);
  #endif
-#elif TIME_METHOD = 1
+#elif TIME_METHOD == 1
   return UL_to_I(*tp);
 #else
 #error internal_time_to_I: invalid TIME_METHOD
