diff -ru stunnel-4.09/tools/Makefile.in stunnel-4.09.arch/tools/Makefile.in
--- stunnel-4.09/tools/Makefile.in	2005-03-26 15:40:46.000000000 -0500
+++ stunnel-4.09.arch/tools/Makefile.in	2005-04-13 07:48:02.000000000 -0400
@@ -332,8 +332,7 @@
 
 info-am:
 
-install-data-am: install-confDATA install-data-local \
-	install-examplesDATA
+install-data-am: install-confDATA install-examplesDATA
 
 install-exec-am:
 
@@ -366,7 +365,7 @@
 	clean-local distclean distclean-generic distclean-libtool \
 	distdir dvi dvi-am html html-am info info-am install \
 	install-am install-confDATA install-data install-data-am \
-	install-data-local install-examplesDATA install-exec \
+	install-examplesDATA install-exec \
 	install-exec-am install-info install-info-am install-man \
 	install-strip installcheck installcheck-am installdirs \
 	maintainer-clean maintainer-clean-generic mostlyclean \
