--- pingus-0.7.2.orig/src/input/evdev_device.cpp	2007-10-01 14:24:54.000000000 -0300
+++ pingus-0.7.2/src/input/evdev_device.cpp	2009-07-05 23:22:34.000000000 -0300
@@ -23,6 +23,8 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <stdexcept>
+#include <cstdio>
+#include <cstring>
 #include <iostream>
 #include <errno.h>
 
--- pingus-0.7.2.orig/src/tinygettext/po_file_reader.cpp	2007-10-18 14:54:01.000000000 -0300
+++ pingus-0.7.2/src/tinygettext/po_file_reader.cpp	2009-07-05 23:20:25.000000000 -0300
@@ -21,6 +21,7 @@
 #include <vector>
 #include <iconv.h>
 #include <errno.h>
+#include <cstdio>
 #include <sstream>
 #include <stdexcept>
 #include <string.h>
