From 10e4085d20a15d82f2a72c179e240b102041a531 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Tue, 8 Dec 2020 15:16:27 -0700
Subject: [PATCH 2/2] Fix build on ARM

Signed-off-by: Kevin Mihelich <kevin@archlinuxarm.org>
---
 sbc/sbc_primitives.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sbc/sbc_primitives.c b/sbc/sbc_primitives.c
index 97a75be..653ad3f 100644
--- a/sbc/sbc_primitives.c
+++ b/sbc/sbc_primitives.c
@@ -630,7 +630,9 @@ void sbc_init_primitives(struct sbc_encoder_state *state)
 	state->implementation_info = "Generic C";
 
 	/* X86/AMD64 optimizations */
+#if (SBC_BUILD_WITH_MMX_SUPPORT || SBC_BUILD_WITH_SSE_SUPPORT)
 	sbc_init_primitives_x86(state);
+#endif
 
 	/* ARM optimizations */
 #ifdef SBC_BUILD_WITH_ARMV6_SUPPORT
-- 
2.28.0

