diff -urN a/configure b/configure
--- a/configure	2015-07-15 23:13:53.689889673 -0600
+++ b/configure	2015-07-15 23:15:02.609508868 -0600
@@ -789,7 +789,7 @@
 
 if [ $asm = auto -a $ARCH = AARCH64 ] ; then
     # set flags so neon is built by default
-    echo $CFLAGS | grep -Eq '(-mcpu|-march|-mfpu|-arch)' || CFLAGS="$CFLAGS -arch arm64 -mfpu=neon"
+    echo $CFLAGS | grep -Eq '(-mcpu|-march|-mfpu|-arch)' || CFLAGS="$CFLAGS"
 
     if  cc_check '' '' '__asm__("cmeq v0.8h, v0.8h, #0");' ; then define HAVE_NEON
         ASFLAGS="$ASFLAGS -c"
