diff -up gecko-sharp-2.0-0.12/gtkmozembed/gtkgecko-api.raw.xl gecko-sharp-2.0-0.12/gtkmozembed/gtkgecko-api.raw
--- gecko-sharp-2.0-0.12/gtkmozembed/gtkgecko-api.raw.xl	2006-12-22 20:57:47.000000000 +0100
+++ gecko-sharp-2.0-0.12/gtkmozembed/gtkgecko-api.raw	2007-11-27 15:54:41.000000000 +0100
@@ -349,6 +349,12 @@
           <parameter type="const-char*" name="aPath" />
         </parameters>
       </method>
+      <method name="SetPath" cname="gtk_moz_embed_set_path" shared="true">
+        <return-type type="void" />
+        <parameters>
+          <parameter type="const-char*" name="aPath" />
+        </parameters>
+      </method>
       <method name="SetProfilePath" cname="gtk_moz_embed_set_profile_path" shared="true">
         <return-type type="void" />
         <parameters>
diff -up gecko-sharp-2.0-0.12/gtkmozembed/gtkgecko-api.xml.xl gecko-sharp-2.0-0.12/gtkmozembed/gtkgecko-api.xml
--- gecko-sharp-2.0-0.12/gtkmozembed/gtkgecko-api.xml.xl	2006-12-22 20:57:54.000000000 +0100
+++ gecko-sharp-2.0-0.12/gtkmozembed/gtkgecko-api.xml	2007-11-27 15:54:41.000000000 +0100
@@ -349,6 +349,12 @@
           <parameter type="const-char*" name="aPath" />
         </parameters>
       </method>
+      <method name="SetPath" cname="gtk_moz_embed_set_path" shared="true">
+        <return-type type="void" />
+        <parameters>
+          <parameter type="const-char*" name="aPath" />
+        </parameters>
+      </method>
       <method name="SetProfilePath" cname="gtk_moz_embed_set_profile_path" shared="true">
         <return-type type="void" />
         <parameters>
diff -up gecko-sharp-2.0-0.12/gtkmozembed/gecko-sharp.dll.config.in.xl gecko-sharp-2.0-0.12/gtkmozembed/gecko-sharp.dll.config.in
--- gecko-sharp-2.0-0.12/gtkmozembed/gecko-sharp.dll.config.in.xl	2006-12-22 20:57:47.000000000 +0100
+++ gecko-sharp-2.0-0.12/gtkmozembed/gecko-sharp.dll.config.in	2007-11-27 16:05:02.000000000 +0100
@@ -3,5 +3,5 @@
   <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libatk-1.0-0.dll" target="libatk-1.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libgtk-win32-2.0-0.dll" target="libgtk-x11-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="gtkembedmoz.dll" target="libgtkembedmoz@LIB_PREFIX@@LIB_SUFFIX@"/>
+  <dllmap dll="gtkembedmoz.dll" target="libxul@LIB_PREFIX@@LIB_SUFFIX@"/>
 </configuration>
