diff --git a/gcc/d/d-incpath.cc b/gcc/d/d-incpath.cc
index e04ddc6..168a033 100644
--- a/gcc/d/d-incpath.cc
+++ b/gcc/d/d-incpath.cc
@@ -202,7 +202,7 @@ add_import_paths(const char *iprefix, const char *imultilib, bool stdinc)
 	    import_path = xstrdup(p->fname);
 
 	  // Add D-specific suffix.
-	  import_path = concat(import_path, "/d", NULL);
+	  import_path = concat(import_path, "/dlang/gdc", NULL);
 
 	  // Ignore duplicate entries.
 	  bool found = false;
diff --git a/libphobos/configure b/libphobos/configure
index 2ff7114..f7f0660 100755
--- a/libphobos/configure
+++ b/libphobos/configure
@@ -3883,7 +3883,7 @@ fi
 
 
 # Default case for install directory for D sources files.
-gdc_include_dir='${libdir}/gcc/${target_alias}'/${d_gcc_ver}/include/d
+gdc_include_dir=${prefix}/include/dlang/gdc
 
 
 # Sanity check for the cross-compilation case:
diff --git a/libphobos/configure.ac b/libphobos/configure.ac
index 84492dd..947f6a8 100644
--- a/libphobos/configure.ac
+++ b/libphobos/configure.ac
@@ -167,7 +167,7 @@ AC_SUBST(phobos_toolexecdir)
 AC_SUBST(phobos_toolexeclibdir)
 
 # Default case for install directory for D sources files.
-gdc_include_dir='${libdir}/gcc/${target_alias}'/${d_gcc_ver}/include/d
+gdc_include_dir=${prefix}/include/dlang/gdc
 AC_SUBST(gdc_include_dir)
 
 dnl Checks for header files.
