diff --git a/gcc/d/d-incpath.cc b/gcc/d/d-incpath.cc
index e6cc314c..2618f571 100644
--- a/gcc/d/d-incpath.cc
+++ b/gcc/d/d-incpath.cc
@@ -140,7 +140,7 @@ add_import_paths (const char *iprefix, const char *imultilib, bool stdinc)
 	    path = xstrdup (p->fname);
 
 	  /* Add D-specific suffix.  */
-	  path = concat (path, "/d", NULL);
+	  path = concat (path, "/dlang/gdc", NULL);
 
 	  /* Ignore duplicate entries.  */
 	  bool found = false;
diff --git a/libphobos/configure b/libphobos/configure
index 3fdd8dfd..ac6d77a5 100755
--- a/libphobos/configure
+++ b/libphobos/configure
@@ -14547,7 +14547,7 @@ fi
 
 
   # Default case for install directory for D sources files.
-  gdc_include_dir='$(libdir)/gcc/${target_alias}/${gcc_version}/include/d'
+  gdc_include_dir='/usr/include/dlang/gdc/'
 
 
 
diff --git a/libphobos/m4/druntime.m4 b/libphobos/m4/druntime.m4
index 93f01efc..6049eaf9 100644
--- a/libphobos/m4/druntime.m4
+++ b/libphobos/m4/druntime.m4
@@ -97,7 +97,7 @@ AC_DEFUN([DRUNTIME_INSTALL_DIRECTORIES],
   AC_SUBST(toolexeclibdir)
 
   # Default case for install directory for D sources files.
-  gdc_include_dir='$(libdir)/gcc/${target_alias}/${gcc_version}/include/d'
+  gdc_include_dir='/usr/include/dlang/gdc/'
   AC_SUBST(gdc_include_dir)
 ])
 
