diff -ruN a/arch/arm/boot/dts/imx6q-wandboard.dts b/a/arch/arm/boot/dts/imx6q-wandboard.dts
--- a/arch/arm/boot/dts/imx6q-wandboard.dts       2016-04-11 16:29:53.056465812 +0200
+++ b/arch/arm/boot/dts/imx6q-wandboard.dts       2016-04-11 16:30:14.772737499 +0200
@@ -46,3 +46,9 @@
 &sata {
        status = "okay";
 };
+
+&hdmi_cec {
+       pinctrl-names = "default";
+       pinctrl-0 = <&pinctrl_hdmi_cec_2>;
+       status = "okay";
+};
