--- po/Makefile.in.in.org	2007-01-25 15:14:01.000000000 +0300
+++ po/Makefile.in.in	2007-01-25 15:15:11.000000000 +0300
@@ -29,7 +29,7 @@
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS = @MKINSTALLDIRS@
-mkinstalldirs = $(SHELL) `case "$(MKINSTALLDIRS)" in /*) echo "$(MKINSTALLDIRS)" ;; *) echo "$(MKINSTALLDIRS)" ;; esac`
+mkinstalldirs = $(SHELL) $(MKINSTALLDIRS)
 
 CC = @CC@
 GMSGFMT = @GMSGFMT@
