diff -aur hid-tools-0.7.old/doc/source/conf.py hid-tools-0.7/doc/source/conf.py
--- hid-tools-0.7.old/doc/source/conf.py	2024-11-23 11:33:40.921977008 +0100
+++ hid-tools-0.7/doc/source/conf.py	2024-11-23 11:34:04.335873711 +0100
@@ -173,7 +173,7 @@
 ]
 
 # Example configuration for intersphinx: refer to the Python standard library.
-intersphinx_mapping = {"http://docs.python.org/": None}
+intersphinx_mapping = {'python': ('https://docs.python.org/3', None)}
 
 apidoc_module_dir = "../../"
 apidoc_output_dir = "./"
