diff -Naur isdn4k-utils-orig/hisax/Makefile.in isdn4k-utils/hisax/Makefile.in
--- isdn4k-utils-orig/hisax/Makefile.in	2002-07-19 12:03:51.000000000 -0700
+++ isdn4k-utils/hisax/Makefile.in	2004-07-11 10:55:34.000000000 -0700
@@ -30,7 +30,7 @@
 #.SUFFIXES:
 #.SUFFIXES: .c .o
 
-%.8: %.man
+%.8: %.man.in
 	cp $< $@
 
 all: $(PROGRAMS) $(MANPAGES)
diff -Naur isdn4k-utils-orig/isdnlog/Makefile.in isdn4k-utils/isdnlog/Makefile.in
--- isdn4k-utils-orig/isdnlog/Makefile.in	2002-07-19 12:03:55.000000000 -0700
+++ isdn4k-utils/isdnlog/Makefile.in	2004-07-11 10:55:47.000000000 -0700
@@ -1842,9 +1842,6 @@
 		  $(INSTALL_DATA) $$f $(DESTDIR)$(DATADIR); \
 		  echo Installing $(DESTDIR)$(DATADIR)/$$f; \
 		done
-		@(grep isdnlog $(SERVICEFILE) >/dev/null) || \
-		(echo "";echo "";echo "Add a line to the file $(SERVICEFILE)" ;echo "";echo ""; \
-		echo "isdnlog $(SERV_PORT)/tcp        isdnlog" >> $(SERVICEFILE))
 
 install-strip:
 		$(MAKE) INSTALL_BIN='$(INSTALL_BIN) -s' \
diff -Naur isdn4k-utils-orig/isdnlog/isdnlog/isdnlog.h isdn4k-utils/isdnlog/isdnlog/isdnlog.h
--- isdn4k-utils-orig/isdnlog/isdnlog/isdnlog.h	2001-06-21 09:34:56.000000000 -0700
+++ isdn4k-utils/isdnlog/isdnlog/isdnlog.h	2004-07-11 10:55:34.000000000 -0700
@@ -460,7 +460,7 @@
 
 _EXTERN void dotrace(void);
 _EXTERN int  morectrl(int card);
-_EXTERN void moreinfo(void);
+extern void moreinfo(void);
 _EXTERN void morekbd(void);
 _EXTERN void processcint(void);
 _EXTERN void processflow(void);
diff -Naur isdn4k-utils-orig/isdnlog/isdnlog/processor.c isdn4k-utils/isdnlog/isdnlog/processor.c
--- isdn4k-utils-orig/isdnlog/isdnlog/processor.c	2002-04-01 10:06:10.000000000 -0800
+++ isdn4k-utils/isdnlog/isdnlog/processor.c	2004-07-11 10:55:34.000000000 -0700
@@ -5293,7 +5293,7 @@
     tei = BROADCAST; /* Wenn nach einer tei-Zeile keine hex:-Zeile kommt, tei ungueltig machen! */
 
     if ((type == SETUP) && !replay) { /* fetch additional info from "/dev/isdninfo" */
-      static void moreinfo(); /* soviel zu Objektorientiertem Denken ;-) */
+      /*static void moreinfo(); */  /* soviel zu Objektorientiertem Denken ;-) */
       moreinfo();
     } /* if */
 
diff -Naur isdn4k-utils-orig/isdnlog/tools/cdb/cdbmake.c isdn4k-utils/isdnlog/tools/cdb/cdbmake.c
--- isdn4k-utils-orig/isdnlog/tools/cdb/cdbmake.c	2002-02-22 03:10:25.000000000 -0800
+++ isdn4k-utils/isdnlog/tools/cdb/cdbmake.c	2004-07-11 10:55:34.000000000 -0700
@@ -3,8 +3,6 @@
 #include "freecdbmake.h"
 #include <unistd.h>
 
-extern char *malloc();
-
 void diesys(why) char *why; { perror(why); exit(111); }
 
 void writeerror() { diesys("cdbmake: fatal: unable to write"); }
diff -Naur isdn4k-utils-orig/isdnlog/tools/rate.c isdn4k-utils/isdnlog/tools/rate.c
--- isdn4k-utils-orig/isdnlog/tools/rate.c	2002-07-05 16:22:33.000000000 -0700
+++ isdn4k-utils/isdnlog/tools/rate.c	2004-07-11 10:55:34.000000000 -0700
@@ -1773,6 +1773,7 @@
 	  number=realloc(number, numbers*sizeof(int));
 	  number[numbers-1]=i;
 	skip:
+	  ;
 	}
 
 	while (isblank(*s)) s++;
@@ -2344,6 +2345,7 @@
 	}
 	return UNKNOWN;
       done:
+	;
       }
     }
   }
diff -Naur isdn4k-utils-orig/lib/libtools.c isdn4k-utils/lib/libtools.c
--- isdn4k-utils-orig/lib/libtools.c	1999-11-03 08:13:36.000000000 -0800
+++ isdn4k-utils/lib/libtools.c	2004-07-11 10:55:34.000000000 -0700
@@ -711,6 +711,7 @@
 				case 'r': _mode |= R_OK;
 				          break;
 				default :
+					  ;
 			}
 		}
 
