--- ./eicon/eiconctrl.c.gcc4	2001-03-01 15:59:12.000000000 +0100
+++ ./eicon/eiconctrl.c	2005-07-21 19:37:33.000000000 +0200
@@ -1970,7 +1970,7 @@
 				val = strtol(p, &q, 16);
 				p = q;
 				val = strtol(p, &q, 16);
-				(unsigned short) *buffer = (unsigned short) val;
+				*buffer = (unsigned short) val;
 				pos = 2;
 				while ((p != q) && (*q != 0)) {
 					p = q;
--- ./hisax/hisaxctrl.c.gcc4	2000-06-29 19:38:26.000000000 +0200
+++ ./hisax/hisaxctrl.c	2005-07-21 19:21:41.000000000 +0200
@@ -43,7 +43,8 @@
 		fprintf(stderr, "cannot open file %s\n", fname);
 		exit(-1);
 	}
-	p = (int *) buffer = (unsigned char *) malloc(MAX_SIZE+4);
+	buffer = (unsigned char *) malloc(MAX_SIZE+4);
+	p = (int *) buffer;
 	if (!buffer) {
 		fprintf(stderr, "cannot get %d byte memory\n", MAX_SIZE+4);
 		exit(-1);
--- ./ipppd/main.c.gcc4	2002-07-18 02:06:21.000000000 +0200
+++ ./ipppd/main.c	2005-07-21 19:39:38.000000000 +0200
@@ -1041,9 +1041,14 @@
 static char line[256];			/* line to be logged accumulated here */
 static char *linep;
 
+#ifdef __STDC__
+static void pr_log(void *arg, char *fmt, ...);
+#else /* __STDC__ */
+static void pr_log(arg, fmt, va_alist);
+#endif
+
 void log_packet(u_char *p,int len,char *prefix,int linkunit)
 {
-	static void pr_log __P((void *, char *, ...));
 
     int i, n;
     u_short proto;
--- ./isdnlog/tools/isdnrate.c.gcc4	2002-08-15 19:22:43.000000000 +0200
+++ ./isdnlog/tools/isdnrate.c	2005-07-21 19:14:24.000000000 +0200
@@ -309,7 +309,7 @@
 
 static void print_header(void);
 
-static char *myname, *myshortname;
+char *myname, *myshortname;
 static char options[] = "ab:d:f:h:l:op:st:v::x:CD::G:HLNP:O:S:TUVX::Z";
 static char usage[] = "%s: usage: %s [ -%s ] Destination ...\n";
 
@@ -318,7 +318,7 @@
 static int usestat = 0;
 static int duration = LCR_DURATION;
 static time_t start;
-static int day, month, year, hour, min, sec;
+int day, month, year, hour, min, sec;
 static char ignore[MAXPROVIDER];
 static char *fromarea = 0;
 static char wanted_day;
