diff -u -r -N isdn4k-utils-3.25_20130409.orig/ipppd/main.c isdn4k-utils-3.25_20130409/ipppd/main.c
--- isdn4k-utils-3.25_20130409.orig/ipppd/main.c	2013-01-05 14:59:12.000000000 +0100
+++ isdn4k-utils-3.25_20130409/ipppd/main.c	2014-09-21 13:36:09.104294210 +0200
@@ -1205,7 +1205,7 @@
     unsigned long val=0;
     char *str, *f, *buf0;
     unsigned char *p;
-    va_list a;
+    void *a;
     char num[32];
     time_t t;
     static char hexchars[] = "0123456789abcdef";
@@ -1297,7 +1297,7 @@
         break;
     case 'r':
         f = va_arg(args, char *);
-        a = va_arg(args, va_list);
+        a = va_arg(args, void *);
 #ifdef __alpha__       /* always do this? */
 	n = fmtmsg(buf, buflen + 1, f, a);
 #else
