--- configure	2012-01-18 16:43:41.000000000 +0000
+++ configure	2012-01-24 11:43:06.590413001 +0000
@@ -6350,7 +6350,7 @@
 fi
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
-  if test "$result" = "yes"; then
+  if test "$result" = "no"; then
       cat >> confdefs.h <<\EOF
 #define HAVE_ARM_SIMD 1
 EOF
@@ -6381,7 +6381,7 @@
 fi
 rm -f conftest*
   echo "$ac_t"""$result"" 1>&6
-  if test "$result" = "yes"; then
+  if test "$result" = "no"; then
       cat >> confdefs.h <<\EOF
 #define HAVE_ARM_NEON 1
 EOF
@@ -17830,7 +17830,7 @@
     *:arm*)
       if test -n "$GNU_AS" ; then
         VPX_AS=$AS
-                        VPX_ASFLAGS="-march=armv7-a -mfpu=neon"
+                        VPX_ASFLAGS="-march=armv7-a"
         VPX_DASH_C_FLAG="-c"
         VPX_AS_CONVERSION='$(PERL) $(topsrcdir)/media/libvpx/build/make/ads2gas.pl'
         VPX_ASM_SUFFIX="$ASM_SUFFIX"
@@ -24519,8 +24519,6 @@
 s%@SOLARIS_SUNPRO_CC@%$SOLARIS_SUNPRO_CC%g
 s%@SOLARIS_SUNPRO_CXX@%$SOLARIS_SUNPRO_CXX%g
 s%@MOZ_THUMB2@%$MOZ_THUMB2%g
-s%@HAVE_ARM_SIMD@%$HAVE_ARM_SIMD%g
-s%@HAVE_ARM_NEON@%$HAVE_ARM_NEON%g
 s%@HAVE_64BIT_OS@%$HAVE_64BIT_OS%g
 s%@MOZ_OS2_HIGH_MEMORY@%$MOZ_OS2_HIGH_MEMORY%g
 s%@MOZ_VALGRIND@%$MOZ_VALGRIND%g
--- configure.in	2012-01-18 16:24:53.000000000 +0000
+++ configure.in	2012-01-24 11:44:28.520413001 +0000
@@ -1687,7 +1687,7 @@
                  [asm("uqadd8 r1, r1, r2");],
                  result="yes", result="no")
   AC_MSG_RESULT("$result")
-  if test "$result" = "yes"; then
+  if test "$result" = "simd"; then
       AC_DEFINE(HAVE_ARM_SIMD)
       HAVE_ARM_SIMD=1
   fi
@@ -1699,7 +1699,7 @@
                  [asm(".fpu neon\n vadd.i8 d0, d0, d0");],
                  result="yes", result="no")
   AC_MSG_RESULT("$result")
-  if test "$result" = "yes"; then
+  if test "$result" = "no"; then
       AC_DEFINE(HAVE_ARM_NEON)
       HAVE_ARM_NEON=1
   fi
@@ -5722,7 +5722,7 @@
         VPX_AS=$AS
         dnl These flags are a lie; they're just used to enable the requisite
         dnl opcodes; actual arch detection is done at runtime.
-        VPX_ASFLAGS="-march=armv7-a -mfpu=neon"
+        VPX_ASFLAGS="-march=armv7-a"
         VPX_DASH_C_FLAG="-c"
         VPX_AS_CONVERSION='$(PERL) $(topsrcdir)/media/libvpx/build/make/ads2gas.pl'
         VPX_ASM_SUFFIX="$ASM_SUFFIX"
