diff -up mozilla/security/nss/lib/Makefile.nolocalsql mozilla/security/nss/lib/Makefile
--- mozilla/security/nss/lib/Makefile.nolocalsql	2007-11-06 15:12:37.000000000 +0100
+++ mozilla/security/nss/lib/Makefile	2007-11-06 15:13:35.000000000 +0100
@@ -62,11 +62,11 @@ ifeq ($(OS_TARGET), WINCE)
 DIRS := $(filter-out fortcrypt,$(DIRS))
 endif
 
-ifndef MOZILLA_CLIENT
-ifndef NSS_USE_SYSTEM_SQLITE
-DIRS := sqlite $(DIRS)
-endif
-endif
+#ifndef MOZILLA_CLIENT
+#ifndef NSS_USE_SYSTEM_SQLITE
+#DIRS := sqlite $(DIRS)
+#endif
+#endif
 
 #######################################################################
 # (5) Execute "global" rules. (OPTIONAL)                              #
diff -up mozilla/security/nss/lib/softoken/legacydb/manifest.mn.nolocalsql mozilla/security/nss/lib/softoken/legacydb/manifest.mn
--- mozilla/security/nss/lib/softoken/legacydb/manifest.mn.nolocalsql	2007-11-06 15:10:59.000000000 +0100
+++ mozilla/security/nss/lib/softoken/legacydb/manifest.mn	2007-11-06 15:11:07.000000000 +0100
@@ -46,9 +46,9 @@ MAPFILE = $(OBJDIR)/nssdbm.def
 
 DEFINES += -DSHLIB_SUFFIX=\"$(DLL_SUFFIX)\" -DSHLIB_PREFIX=\"$(DLL_PREFIX)\" -DSOFTOKEN_LIB_NAME=\"$(notdir $(SHARED_LIBRARY))\"
 
-ifdef MOZILLA_CLIENT
-INCLUDES += -I$(DIST)/include/sqlite3
-endif
+#ifdef MOZILLA_CLIENT
+#INCLUDES += -I$(DIST)/include/sqlite3
+#endif
 
 CSRCS = \
 	dbmshim.c \
diff -up mozilla/security/nss/lib/softoken/manifest.mn.nolocalsql mozilla/security/nss/lib/softoken/manifest.mn
--- mozilla/security/nss/lib/softoken/manifest.mn.nolocalsql	2007-11-06 15:08:14.000000000 +0100
+++ mozilla/security/nss/lib/softoken/manifest.mn	2007-11-06 15:10:21.000000000 +0100
@@ -47,9 +47,9 @@ MAPFILE = $(OBJDIR)/softokn.def
 
 DEFINES += -DSHLIB_SUFFIX=\"$(DLL_SUFFIX)\" -DSHLIB_PREFIX=\"$(DLL_PREFIX)\" -DSOFTOKEN_LIB_NAME=\"$(notdir $(SHARED_LIBRARY))\" -DSHLIB_VERSION=\"$(LIBRARY_VERSION)\"
 
-ifdef MOZILLA_CLIENT
-INCLUDES += -I$(DIST)/include/sqlite3
-endif
+#ifdef MOZILLA_CLIENT
+#INCLUDES += -I$(DIST)/include/sqlite3
+#endif
 
 EXPORTS = \
 	pkcs11.h \
diff -up mozilla/security/nss/lib/softoken/config.mk.nolocalsql mozilla/security/nss/lib/softoken/config.mk
