--- gtk+-2/pinentry-gtk-2.c.orig	2010-03-03 05:19:55.000000000 -0600
+++ gtk+-2/pinentry-gtk-2.c	2010-07-07 00:11:30.413572124 -0500
@@ -145,7 +145,7 @@
 {
   gdk_keyboard_ungrab (gdk_event_get_time (event));
   /* Unmake window transient for the root window.  */
-  gdk_window_set_transient_for (win->window, NULL);
+  gdk_property_delete (win->window, gdk_atom_intern_static_string ("WM_TRANSIENT_FOR"));
 }
 
 
