diff -ur setserial-2.17/Makefile.in setserial-2.17-new/Makefile.in
--- setserial-2.17/Makefile.in	2000-01-27 16:39:07.000000000 +0100
+++ setserial-2.17-new/Makefile.in	2005-03-09 20:53:31.000000000 +0100
@@ -17,6 +17,11 @@
 INCS = -I.
 TAR = tar
 
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+bindir=@bindir@
+mandir=@mandir@
+
 all: setserial setserial.cat
 
 setserial: setserial.c
@@ -26,9 +31,9 @@
 	nroff -man setserial.8 > setserial.cat
 
 install: setserial setserial.8
-	$(INSTALL_PROGRAM) setserial $(DESTDIR)/bin
-	$(STRIP) $(DESTDIR)/bin/setserial
-	$(INSTALL_DATA) setserial.8 $(DESTDIR)/usr/man/man8
+	$(INSTALL_PROGRAM) setserial $(DESTDIR)$(bindir)
+	$(STRIP) $(DESTDIR)$(bindir)/setserial
+	$(INSTALL_DATA) setserial.8 $(DESTDIR)$(mandir)/man8
 
 clean:
 	$(RM) setserial setserial.o setserial.cat *~
