diff -ur tigervnc-1.4.0.orig/unix/xserver/hw/vnc/Makefile.am tigervnc-1.4.0/unix/xserver/hw/vnc/Makefile.am
--- tigervnc-1.4.0.orig/unix/xserver/hw/vnc/Makefile.am	2014-12-10 11:18:40.000000000 +0100
+++ tigervnc-1.4.0/unix/xserver/hw/vnc/Makefile.am	2014-12-14 23:17:54.569210661 +0100
@@ -7,6 +7,9 @@
 XREGION_LIB=$(LIB_DIR)/Xregion/libXregion.la
 COMMON_LIBS=$(NETWORK_LIB) $(RFB_LIB) $(RDR_LIB) $(XREGION_LIB)
 
+# Hack to get the C headers to work when included from C++ code
+AM_CXXFLAGS = -fpermissive
+
 noinst_LTLIBRARIES = libvnccommon.la
 
 HDRS = RegionHelper.h vncExtInit.h vncHooks.h XserverDesktop.h xorg-version.h \
