diff -up tigervnc-1.3.0/unix/xserver/hw/vnc/xorg-version.h.jx tigervnc-1.3.0/unix/xserver/hw/vnc/xorg-version.h
--- tigervnc-1.3.0/unix/xserver/hw/vnc/xorg-version.h.jx    2013-07-01 08:41:24.000000000 -0400
+++ tigervnc-1.3.0/unix/xserver/hw/vnc/xorg-version.h   2013-11-07 11:39:49.749992669 -0500
@@ -44,8 +44,10 @@
 #define XORG 113
 #elif XORG_VERSION_CURRENT < ((1 * 10000000) + (14 * 100000) + (99 * 1000))
 #define XORG 114
+#elif XORG_VERSION_CURRENT < ((1 * 10000000) + (15 * 100000) + (99 * 1000))
+#define XORG 115
 #else
-#error "X.Org newer than 1.14 is not supported"
+#error "X.Org newer than 1.15 is not supported"
 #endif
 
 #endif
