--- src/Util.cpp.orig   2012-07-19 22:08:18.559690539 +0200
+++ src/Util.cpp        2012-07-19 22:08:42.548436166 +0200
@@ -984,7 +984,7 @@
 
 gzFile utilGzOpen(const char *file, const char *mode)
 {
-  utilGzWriteFunc = (int (*)(void *,void * const, unsigned int))gzwrite;
+  utilGzWriteFunc = (int (ZEXPORT *)(gzFile,const voidp, unsigned int))gzwrite;
   utilGzReadFunc = gzread;
   utilGzCloseFunc = gzclose;
