diff -urN a/xorg-cf-files-1.0.5/xfree86.cf b/xorg-cf-files-1.0.5/xfree86.cf
--- a/xorg-cf-files-1.0.5/xfree86.cf	2013-07-22 23:23:14.000000000 -0600
+++ b/xorg-cf-files-1.0.5/xfree86.cf	2015-07-11 13:51:14.803581068 -0600
@@ -2028,6 +2028,12 @@
 # endif
 #endif
 
+#if HasGcc2 && defined(AArch64Architecture)
+# ifndef DefaultGcc2AArch64Opt
+#  define DefaultGcc2AArch64Opt -O2 GccAliasingArgs
+# endif
+#endif
+
 #if HasGcc2 && defined(AlphaArchitecture)
 # ifndef DefaultGcc2AxpOpt
 #  define DefaultGcc2AxpOpt -O2 GccAliasingArgs
diff -urN a/xorg-cf-files-1.0.5/xorg.cf b/xorg-cf-files-1.0.5/xorg.cf
--- a/xorg-cf-files-1.0.5/xorg.cf	2013-07-22 23:23:14.000000000 -0600
+++ b/xorg-cf-files-1.0.5/xorg.cf	2015-07-11 13:54:31.580466492 -0600
@@ -1561,6 +1561,12 @@
 # endif
 #endif
 
+#if HasGcc2 && defined(AArch64Architecture)
+# ifndef DefaultGcc2AArch64Opt
+#  define DefaultGcc2AArch64Opt -O2 GccAliasingArgs
+# endif
+#endif
+
 #if HasGcc2 && defined(AMD64Architecture)
 # ifndef DefaultGcc2AMD64Opt
 #  define DefaultGcc2AMD64Opt -O2 -fno-strength-reduce GccAliasingArgs
