--- xorg-cf-files-1.0.2/X11.tmpl.xprint	2005-10-01 12:45:48.000000000 -0400
+++ xorg-cf-files-1.0.2/X11.tmpl	2006-04-25 20:24:02.000000000 -0400
@@ -3920,21 +3920,13 @@
 #endif
 
 #ifndef ExtraXawReqs
-# if BuildXaw
-#  define ExtraXawReqs $(XPLIB)
-# else
-#  define ExtraXawReqs /**/
-# endif
+# define ExtraXawReqs /**/
 #endif
 
 EXTRAXAWREQS = ExtraXawReqs
 
 #ifndef ExtraXawClientDepLibs
-# if BuildXaw
-#  define ExtraXawClientDepLibs $(DEPXPLIB)
-# else
-#  define ExtraXawClientDepLibs /**/
-# endif
+# define ExtraXawClientDepLibs /**/
 #endif
 
 EXTRAXAWCLIENTDEPLIBS = ExtraXawClientDepLibs
@@ -3944,11 +3936,7 @@
 #endif
 
 #ifndef ExtraXawClientLibs
-# if BuildXaw
-#  define ExtraXawClientLibs $(XPLIB)
-# else
-#  define ExtraXawClientLibs /**/
-# endif
+# define ExtraXawClientLibs /**/
 #endif
 
 EXTRAXAWCLIENTLIBS = ExtraXawClientLibs
