Description: Add ARM64 support
Author: Colin Watson <cjwatson@ubuntu.com>
Forwarded: https://issues.apache.org/jira/browse/DAEMON-308
Last-Update: 2014-11-06

Index: b/src/native/unix/configure
===================================================================
--- a/src/native/unix/configure
+++ b/src/native/unix/configure
@@ -2697,6 +2697,11 @@
     supported_os="arm"
     HOST_CPU=arm
     ;;
+  aarch64)
+    CFLAGS="$CFLAGS -DCPU=\\\"aarch64\\\""
+    supported_os="aarch64"
+    HOST_CPU=aarch64
+    ;;
   *)
     echo "$as_me:$LINENO: result: failed" >&5
 echo "${ECHO_T}failed" >&6
Index: b/src/native/unix/support/apsupport.m4
===================================================================
--- a/src/native/unix/support/apsupport.m4
+++ b/src/native/unix/support/apsupport.m4
@@ -176,6 +176,11 @@
     supported_os="arm"
     HOST_CPU=arm
     ;;
+  aarch64)
+    CFLAGS="$CFLAGS -DCPU=\\\"aarch64\\\""
+    supported_os="aarch64"
+    HOST_CPU=aarch64
+    ;;
   *)
     AC_MSG_RESULT([failed])
     AC_MSG_ERROR([Unsupported CPU architecture "$host_cpu"]);;
