diff -aur qcad-2.0.5.0-1-community.src.orig/dxflib/src/dl_writer.h qcad-2.0.5.0-1-community.src.new/dxflib/src/dl_writer.h
--- qcad-2.0.5.0-1-community.src.orig/dxflib/src/dl_writer.h	2008-12-11 22:28:20.000000000 +0100
+++ qcad-2.0.5.0-1-community.src.new/dxflib/src/dl_writer.h	2008-12-11 22:29:51.000000000 +0100
@@ -37,6 +37,7 @@
 #endif
 
 #include <iostream>
+#include <cstring>
 
 #include "dl_attributes.h"
 #include "dl_codes.h"
diff -aur qcad-2.0.5.0-1-community.src.orig/qcadactions/src/rs_actionzoompan.cpp qcad-2.0.5.0-1-community.src.new/qcadactions/src/rs_actionzoompan.cpp
--- qcad-2.0.5.0-1-community.src.orig/qcadactions/src/rs_actionzoompan.cpp	2008-12-11 22:28:20.000000000 +0100
+++ qcad-2.0.5.0-1-community.src.new/qcadactions/src/rs_actionzoompan.cpp	2008-12-11 22:29:51.000000000 +0100
@@ -23,7 +23,7 @@
 ** not clear to you.
 **
 **********************************************************************/
-
+#include <cstdlib>
 #include "rs_actionzoompan.h"
 #include "rs_snapper.h"
 #include "rs_point.h"
Nur in qcad-2.0.5.0-1-community.src.new/qcadlib/src/engine: rs_entity.cpp.orig.rej.
Nur in qcad-2.0.5.0-1-community.src.new/qcadlib/src/engine: rs_entity.cpp.rej.
Nur in qcad-2.0.5.0-1-community.src.new/qcadlib/src/engine: rs_layer.cpp.orig.rej.
Nur in qcad-2.0.5.0-1-community.src.new/qcadlib/src/engine: rs_layer.cpp.rej.
diff -aur qcad-2.0.5.0-1-community.src.orig/qcadlib/src/information/rs_information.cpp qcad-2.0.5.0-1-community.src.new/qcadlib/src/information/rs_information.cpp
--- qcad-2.0.5.0-1-community.src.orig/qcadlib/src/information/rs_information.cpp	2008-12-11 22:28:20.000000000 +0100
+++ qcad-2.0.5.0-1-community.src.new/qcadlib/src/information/rs_information.cpp	2008-12-11 22:30:18.000000000 +0100
@@ -25,7 +25,7 @@
 **********************************************************************/
 
 #include "rs_information.h"
-
+#include <cstdlib>
 #include "rs_constructionline.h"
 
 
diff -aur qcad-2.0.5.0-1-community.src.orig/scripts/build_qcad.sh qcad-2.0.5.0-1-community.src.new/scripts/build_qcad.sh
--- qcad-2.0.5.0-1-community.src.orig/scripts/build_qcad.sh	2008-12-11 22:28:20.000000000 +0100
+++ qcad-2.0.5.0-1-community.src.new/scripts/build_qcad.sh	2008-12-11 22:31:37.000000000 +0100
@@ -16,9 +16,6 @@
 echo "  distcc      use distcc for distributed compilation. DISTCC_HOSTS must be set."
 echo
 
-echo "QTDIR is: $QTDIR"
-echo "QMAKESPEC is: $QMAKESPEC"
-
 # detect system:
 if [ "x$OS" == "xWindows_NT" ]
 then
