From 4aa59d4efe5c2001f0de3ac0d28c7aa78696d3b2 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Fri, 6 Jan 2017 20:46:03 -0700
Subject: [PATCH] AArch64 fix

---
 NormalizModule.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/NormalizModule.cpp b/NormalizModule.cpp
index e25de74..ee17989 100644
--- a/NormalizModule.cpp
+++ b/NormalizModule.cpp
@@ -60,7 +60,7 @@ static string cone_name_str_long( cone_name_long );
 
 // Check GCC
 #if __GNUC__
-#if __x86_64__ || __ppc64__
+#if __x86_64__ || __ppc64__ || __aarch64__
 #define ENVIRONMENT64
 #else
 #define ENVIRONMENT32
-- 
2.10.2

