diff -Naur alsa-utils-1.0.9a-orig/alsaconf/alsaconf.in alsa-utils-1.0.9a/alsaconf/alsaconf.in
--- alsa-utils-1.0.9a-orig/alsaconf/alsaconf.in	2005-03-24 00:24:52.000000000 -0800
+++ alsa-utils-1.0.9a/alsaconf/alsaconf.in	2005-07-18 09:25:02.000000000 -0700
@@ -58,6 +58,8 @@
   distribution="redhat"
 elif test -f /etc/fedora-release && grep -q "Fedora" /etc/fedora-release; then
   distribution="fedora"
+elif test -f /etc/arch-release && grep -q "Arch" /etc/arch-release; then
+  distribution="arch"
 elif [ -f /etc/slackware-version -o -f /etc/slamd64-version ]; then
   distribution="slackware"
 else
@@ -382,6 +384,8 @@
     rcalsasound=/etc/rc.d/init.d/alsasound
 elif [ -x /etc/init.d/alsa ]; then
     rcalsasound=/etc/init.d/alsa
+elif [ -x /etc/rc.d/alsa ]; then
+    rcalsasound=/etc/rc.d/alsa
 else
     rcalsasound=rcalsasound
 fi
