From 0d3aeb969592134f7749bd21fd2f641eb13d7090 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Fri, 15 May 2009 12:37:48 +0200
Subject: [PATCH] alsactl: fixed Headphone Playback Volume setting in default rules

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 alsactl/init/default |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/alsactl/init/default b/alsactl/init/default
index c9aa7cc..af0f94f 100644
--- a/alsactl/init/default
+++ b/alsactl/init/default
@@ -46,7 +46,7 @@ CTL{name}="Front Playback Switch",PROGRAM=="__ctl_search", \
   CTL{values}="on"
 
 CTL{reset}="mixer"
-CTL{name}="Headphone Playback Volume",PROGRAM=="__ctl_search",GOTO="headphone0_end"
+CTL{name}="Headphone Playback Volume",PROGRAM!="__ctl_search",GOTO="headphone0_end"
 # if master volume control is present, turn headphone volume to max
 ENV{has_pmaster_vol}=="true",CTL{values}="0dB",RESULT=="0",GOTO="headphone0_end"
 ENV{has_pmaster_vol}=="true",CTL{values)="100%",GOTO="headphone0_end"
@@ -101,7 +101,7 @@ CTL{name}="DAC Playback Switch",PROGRAM=="__ctl_search", \
   CTL{values}="on"
 
 CTL{reset}="mixer"
-CTL{name}="Synth Playback Volume",,PROGRAM=="__ctl_search", \
+CTL{name}="Synth Playback Volume",PROGRAM=="__ctl_search", \
   CTL{values}="$env{pvolume}",RESULT!="0",CTL{values}="$env{ppercent}"
 CTL{name}="Synth Playback Switch",PROGRAM=="__ctl_search", \
   CTL{values}="on"
-- 
1.5.5.1

