Index: trunk/pdns/pdns/aes/aeskey.c
===================================================================
--- a/trunk/pdns/pdns/aes/aeskey.c
+++ b/trunk/pdns/pdns/aes/aeskey.c
@@ -489,4 +489,5 @@
 }
 
+#if 0
 AES_RETURN aes_decrypt_key256(const unsigned char *key, aes_decrypt_ctx cx[1])
 {   uint_32t    ss[9];
@@ -537,4 +538,5 @@
     return EXIT_SUCCESS;
 }
+#endif
 
 #endif
@@ -542,4 +544,5 @@
 #if defined( AES_VAR )
 
+#if 0
 AES_RETURN aes_decrypt_key(const unsigned char *key, int key_len, aes_decrypt_ctx cx[1])
 {
@@ -552,4 +555,5 @@
     }
 }
+#endif
 
 #endif
