diff -up m4-1.4.16/tests/test-readlink.h.orig m4-1.4.16/tests/test-readlink.h
--- m4-1.4.16/tests/test-readlink.h.orig	2011-09-19 12:44:58.745546826 +0200
+++ m4-1.4.16/tests/test-readlink.h	2011-09-19 12:46:00.079548410 +0200
@@ -38,7 +38,7 @@ test_readlink (ssize_t (*func) (char con
   ASSERT (errno == ENOENT);
   errno = 0;
   ASSERT (func ("", buf, sizeof buf) == -1);
-  ASSERT (errno == ENOENT);
+  ASSERT (errno == ENOENT || errno == EINVAL);
   errno = 0;
   ASSERT (func (".", buf, sizeof buf) == -1);
   ASSERT (errno == EINVAL);
