--- valgrind.orig/configure
+++ valgrind/configure
@@ -5279,7 +5279,7 @@
         ARCH_MAX="s390x"
         ;;
 
-     armv7*)
+     arm*)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: ok (${host_cpu})" >&5
 $as_echo "ok (${host_cpu})" >&6; }
 	ARCH_MAX="arm"
Description: Make valgrind believe it's cross compiled so that it builds on armel
Origin: vendor
Author: Pierre Habouzit <madcoder@debian.org>
Reviewed-by: Alessandro Ghedini <ghedo@debian.org>
Last-Update: 2012-05-10

--- valgrind.orig/configure
+++ valgrind/configure
@@ -5133,7 +5133,7 @@
              | head -n 1 \
              | $SED 's/i686-apple-darwin10//' \
              | $SED 's/i686-apple-darwin11//' \
-             | $SED 's/^[^0-9]*\([0-9.]*\).*$/\1/'`
+             | $SED 's/^[^ ]* [^0-9]*\([0-9.]*\).*$/\1/'`
 
 is_clang="notclang"
 if test "x`${CC} --version | head -n 1 | $SED 's/\(clang\) version.*/\1/'`" = "xclang" ; then
--- valgrind.orig/configure.in
+++ valgrind/configure.in
@@ -109,7 +109,7 @@
              | head -n 1 \
              | $SED 's/i686-apple-darwin10//' \
              | $SED 's/i686-apple-darwin11//' \
-             | $SED 's/^[^0-9]*\([0-9.]*\).*$/\1/'`]
+             | $SED 's/^[^ ]* [^0-9]*\([0-9.]*\).*$/\1/'`]
 
 is_clang="notclang"
 if test "x`${CC} --version | head -n 1 | $SED 's/\(clang\) version.*/\1/'`" = "xclang" ; then
