From 88d02c44361bf4d9b354eea7e20559a4a57b866f Mon Sep 17 00:00:00 2001
From: Anatol Pomozov <anatol.pomozov@gmail.com>
Date: Thu, 6 Aug 2015 09:06:25 -0700
Subject: [PATCH 2/2] kernel

---
 configure    | 6 +++---
 configure.ac | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/configure b/configure
index 6ab06b3..a59ef12 100755
--- a/configure
+++ b/configure
@@ -5553,9 +5553,9 @@ $as_echo_n "checking for the kernel version... " >&6; }
         kernel=`uname -r`
 
         case "${kernel}" in
-             2.6.*|3.*)
-        	    { $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.6.x/3.x family (${kernel})" >&5
-$as_echo "2.6.x/3.x family (${kernel})" >&6; }
+             2.6.*|3.*|4.*)
+        	    { $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.6.x/3.x/4.x family (${kernel})" >&5
+$as_echo "2.6.x/3.x/4.x family (${kernel})" >&6; }
 
 $as_echo "#define KERNEL_2_6 1" >>confdefs.h
 
diff --git a/configure.ac b/configure.ac
index 7f389c2..b0bb449 100644
--- a/configure.ac
+++ b/configure.ac
@@ -303,9 +303,9 @@ case "${host_os}" in
         kernel=`uname -r`
 
         case "${kernel}" in
-             2.6.*|3.*) 
-        	    AC_MSG_RESULT([2.6.x/3.x family (${kernel})])
-        	    AC_DEFINE([KERNEL_2_6], 1, [Define to 1 if you're using Linux 2.6.x or Linux 3.x])
+             2.6.*|3.*|4.*) 
+        	    AC_MSG_RESULT([2.6.x/3.x/4.x family (${kernel})])
+        	    AC_DEFINE([KERNEL_2_6], 1, [Define to 1 if you're using Linux 2.6.x or Linux 3.x or Linux 4.x])
         	    ;;
 
              2.4.*) 
-- 
2.5.0

