From ca88e1c63178186a1f6a0d0b1a4ba4ee895af323 Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mark@klomp.org>
Date: Sat, 18 Nov 2023 02:23:10 +0100
Subject: [PATCH] s390x-linux also has memfd_secret, syswrap it

---
 coregrind/m_syswrap/syswrap-s390x-linux.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/coregrind/m_syswrap/syswrap-s390x-linux.c b/coregrind/m_syswrap/syswrap-s390x-linux.c
index 7740bd61f6..3238eef3ff 100644
--- a/coregrind/m_syswrap/syswrap-s390x-linux.c
+++ b/coregrind/m_syswrap/syswrap-s390x-linux.c
@@ -874,6 +874,8 @@ static SyscallTableEntry syscall_table[] = {
    LINX_(__NR_faccessat2,  sys_faccessat2),                           // 439
 
    LINXY(__NR_epoll_pwait2, sys_epoll_pwait2),                        // 441
+
+   LINXY(__NR_memfd_secret, sys_memfd_secret),                        // 447
 };
 
 SyscallTableEntry* ML_(get_linux_syscall_entry) ( UInt sysno )
-- 
2.39.3

