diff -upr gnuradio-3.7.11.orig/gr-blocks/lib/CMakeLists.txt gnuradio-3.7.11/gr-blocks/lib/CMakeLists.txt
--- gnuradio-3.7.11.orig/gr-blocks/lib/CMakeLists.txt	2017-02-28 01:19:00.000000000 +0200
+++ gnuradio-3.7.11/gr-blocks/lib/CMakeLists.txt	2018-06-03 07:12:46.110899996 +0300
@@ -328,6 +328,7 @@ if(ENABLE_TESTING)
     gnuradio-runtime
     gnuradio-blocks
     ${Boost_LIBRARIES}
+    ${CMAKE_THREAD_LIBS_INIT}
     ${CPPUNIT_LIBRARIES}
     ${LOG4CPP_LIBRARIES}
   )
diff -upr gnuradio-3.7.11.orig/gr-uhd/examples/c++/CMakeLists.txt gnuradio-3.7.11/gr-uhd/examples/c++/CMakeLists.txt
--- gnuradio-3.7.11.orig/gr-uhd/examples/c++/CMakeLists.txt	2017-02-28 01:19:00.000000000 +0200
+++ gnuradio-3.7.11/gr-uhd/examples/c++/CMakeLists.txt	2018-06-03 07:12:53.401117802 +0300
@@ -35,7 +35,7 @@ link_directories(${Boost_LIBRARY_DIRS})
 # Build executable
 ########################################################################
 add_executable(tags_demo tags_demo.cc)
-target_link_libraries(tags_demo gnuradio-uhd)
+target_link_libraries(tags_demo gnuradio-uhd ${CMAKE_THREAD_LIBS_INIT})
 
 INSTALL(TARGETS
     tags_demo
