diff -ur a/src/network/md5.cpp b/src/network/md5.cpp
--- a/src/network/md5.cpp	2009-08-24 20:08:14.000000000 +0200
+++ b/src/network/md5.cpp	2009-08-24 20:21:51.000000000 +0200
@@ -75,7 +75,7 @@
  */
 
 #include "md5.h"
-
+#include<cstring>
 #include <string>
 #include <memory>
 
Tylko w b/src/network: md5.cpp~
diff -ur a/src/network/nNetwork.cpp b/src/network/nNetwork.cpp
--- a/src/network/nNetwork.cpp	2009-08-24 20:08:14.000000000 +0200
+++ b/src/network/nNetwork.cpp	2009-08-24 20:20:40.000000000 +0200
@@ -24,7 +24,7 @@
 ***************************************************************************
 
 */
-
+#include<cstring>
 #include "tMemManager.h"
 #include "tInitExit.h"
 #include "nSimulatePing.h"
Tylko w b/src/network: nNetwork.cpp~
diff -ur a/src/network/nNetwork.h b/src/network/nNetwork.h
--- a/src/network/nNetwork.h	2009-08-24 20:08:14.000000000 +0200
+++ b/src/network/nNetwork.h	2009-08-24 20:21:18.000000000 +0200
@@ -27,7 +27,7 @@
 
 #ifndef ArmageTron_NET_H
 #define ArmageTron_NET_H
-
+#include<cstring>
 #include "tError.h"
 #include "tString.h"
 #include "tHeap.h"
Tylko w b/src/network: nNetwork.h~
diff -ur a/src/network/nSocket.h b/src/network/nSocket.h
--- a/src/network/nSocket.h	2009-08-24 20:08:14.000000000 +0200
+++ b/src/network/nSocket.h	2009-08-24 20:22:09.000000000 +0200
@@ -33,7 +33,7 @@
 
 #ifndef NET_SOCKET_H
 #define NET_SOCKET_H
-
+#include<cstring>
 #include <iostream>
 #include "tConsole.h"
 #include "tException.h"
Tylko w b/src/network: nSocket.h~
diff -ur a/src/render/rModel.h b/src/render/rModel.h
--- a/src/render/rModel.h	2009-08-24 20:08:14.000000000 +0200
+++ b/src/render/rModel.h	2009-08-24 20:20:57.000000000 +0200
@@ -27,7 +27,7 @@
 
 #ifndef ArmageTron_MODEL_H
 #define ArmageTron_MODEL_H
-
+#include<cstring>
 #include "defs.h"
 #include "tArray.h"
 #include "tLinkedList.h"
Tylko w b/src/render: rModel.h~
diff -ur a/src/tools/tArray.cpp b/src/tools/tArray.cpp
--- a/src/tools/tArray.cpp	2009-08-24 20:08:14.000000000 +0200
+++ b/src/tools/tArray.cpp	2009-08-24 20:22:25.000000000 +0200
@@ -27,7 +27,7 @@
 */
 
 #define NO_MALLOC_REPLACEMENT
-
+#include<cstring>
 #include "tMemManager.h"
 #include <iostream>
 #include <stdlib.h>
Tylko w b/src/tools: tArray.cpp~
diff -ur a/src/tools/tCommandLine.cpp b/src/tools/tCommandLine.cpp
--- a/src/tools/tCommandLine.cpp	2009-08-24 20:08:14.000000000 +0200
+++ b/src/tools/tCommandLine.cpp	2009-08-24 20:24:19.000000000 +0200
@@ -30,7 +30,7 @@
 #ifndef		TCOMMANDLINE_H_INCLUDED
 #include	"tCommandLine.h"
 #endif
-
+#include<cstring>
 #include    "tLocale.h"
 #include    "tConfiguration.h"
 #include    "tException.h"
Tylko w b/src/tools: tCommandLine.cpp~
diff -ur a/src/tools/tConfiguration.cpp b/src/tools/tConfiguration.cpp
--- a/src/tools/tConfiguration.cpp	2009-08-24 20:08:14.000000000 +0200
+++ b/src/tools/tConfiguration.cpp	2009-08-24 20:22:42.000000000 +0200
@@ -24,7 +24,7 @@
 ***************************************************************************
 
 */
-
+#include<cstring>
 #include "config.h"
 #include <fstream>
 #include <iomanip>
Tylko w b/src/tools: tConfiguration.cpp~
diff -ur a/src/tools/tCrypt.cpp b/src/tools/tCrypt.cpp
--- a/src/tools/tCrypt.cpp	2009-08-24 20:08:14.000000000 +0200
+++ b/src/tools/tCrypt.cpp	2009-08-24 20:22:54.000000000 +0200
@@ -24,7 +24,7 @@
 ***************************************************************************
 
 */
-
+#include<cstring>
 #include "tCrypt.h"
 #include "tMemManager.h"
 
Tylko w b/src/tools: tCrypt.cpp~
diff -ur a/src/tools/tDirectories.cpp b/src/tools/tDirectories.cpp
--- a/src/tools/tDirectories.cpp	2009-08-24 20:08:14.000000000 +0200
+++ b/src/tools/tDirectories.cpp	2009-08-24 20:23:06.000000000 +0200
@@ -27,7 +27,7 @@
 */
 
 #include "config.h"
-
+#include<cstring>
 #include <errno.h>
 #include <sys/types.h>
 #include <stdio.h>
Tylko w b/src/tools: tDirectories.cpp~
diff -ur a/src/tools/tError.cpp b/src/tools/tError.cpp
--- a/src/tools/tError.cpp	2009-08-24 20:08:14.000000000 +0200
+++ b/src/tools/tError.cpp	2009-08-24 20:28:01.000000000 +0200
@@ -24,8 +24,9 @@
 ***************************************************************************
 
 */
-
+#include<cstring>
 #include "config.h"
+#include<cstdlib>
 #include <iostream>
 #if HAVE_UNISTD_H
 #include <unistd.h>
Tylko w b/src/tools: tError.cpp~
diff -ur a/src/tools/tLocale.cpp b/src/tools/tLocale.cpp
--- a/src/tools/tLocale.cpp	2009-08-24 20:08:14.000000000 +0200
+++ b/src/tools/tLocale.cpp	2009-08-24 20:23:49.000000000 +0200
@@ -30,7 +30,7 @@
 #include "tConsole.h"
 #include "tDirectories.h"
 #include "tSafePTR.h"
-
+#include<cstring>
 #include <fstream>
 #include <string>
 #include <map>
Tylko w b/src/tools: tLocale.cpp~
diff -ur a/src/tools/tMemManager.cpp b/src/tools/tMemManager.cpp
--- a/src/tools/tMemManager.cpp	2009-08-24 20:08:14.000000000 +0200
+++ b/src/tools/tMemManager.cpp	2009-08-24 20:24:42.000000000 +0200
@@ -29,7 +29,7 @@
 #define NO_MALLOC_REPLACEMENT
 
 #include "defs.h"
-
+#include<cstring>
 #include <iostream>
 #include <sstream>
 #include <stdio.h>  // need basic C IO since STL IO does memory management
Tylko w b/src/tools: tMemManager.cpp~
diff -ur a/src/tools/tRandom.cpp b/src/tools/tRandom.cpp
--- a/src/tools/tRandom.cpp	2009-08-24 20:08:14.000000000 +0200
+++ b/src/tools/tRandom.cpp	2009-08-24 20:24:31.000000000 +0200
@@ -29,7 +29,7 @@
 #ifndef		TRANDOM_H_INCLUDED
 #include	"tRandom.h"
 #endif
-
+#include<cstdlib>
 #ifdef      HAVE_STDLIB
 #include    <stdlib.h>
 #endif
Tylko w b/src/tools: tRandom.cpp~
diff -ur a/src/tools/tString.cpp b/src/tools/tString.cpp
--- a/src/tools/tString.cpp	2009-08-24 20:08:14.000000000 +0200
+++ b/src/tools/tString.cpp	2009-08-24 20:24:03.000000000 +0200
@@ -24,7 +24,7 @@
 ***************************************************************************
 
 */
-
+#include<cstring>
 #include "tMemManager.h"
 #include "tString.h"
 #include "tLocale.h"
Tylko w b/src/tools: tString.cpp~
diff -ur a/src/tron/gAIBase.h b/src/tron/gAIBase.h
--- a/src/tron/gAIBase.h	2009-08-24 20:08:14.000000000 +0200
+++ b/src/tron/gAIBase.h	2009-08-24 20:21:33.000000000 +0200
@@ -29,7 +29,7 @@
 #define ArmageTron_AIBASE_H
 
 #include "rSDL.h"
-
+#include<memory>
 #include "eTimer.h"
 #include "ePath.h"
 #include "ePlayer.h"
Tylko w b/src/tron: gAIBase.h~
