diff -urN a/include/configs/ti_omap4_common.h b/include/configs/ti_omap4_common.h
--- a/include/configs/ti_omap4_common.h	2014-05-17 13:32:47.384861195 -0600
+++ b/include/configs/ti_omap4_common.h	2014-05-17 13:48:19.361560495 -0600
@@ -85,6 +85,8 @@
 #undef CONFIG_CMD_NET
 #undef CONFIG_CMD_NFS
 
+#define CONFIG_IDENT_STRING " Arch Linux ARM"
+
 /*
  * Environment setup
  */
@@ -95,9 +97,9 @@
 	"fdtaddr=0x815f0000\0" \
 	"fdtfile=undefined\0" \
 	"bootpart=0:1\0" \
-	"bootdir=\0" \
-	"fdtdir=/dtbs\0" \
-	"bootfile=zImage\0" \
+	"rootpart=0:2\0" \
+	"bootdir=/boot\0" \
+	"fdtdir=/boot/dtbs\0" \
 	"usbtty=cdc_acm\0" \
 	"vram=16M\0" \
 	"defaultdisplay=dvi\0" \
@@ -105,24 +107,26 @@
 	"buddy=none\0" \
 	"mmcdev=0\0" \
 	"mmcpart=1\0" \
-	"mmcroot=/dev/mmcblk0p2 ro\0" \
-	"mmcrootfstype=ext4 rootwait fixrtc\0" \
+	"mmcroot=/dev/mmcblk0p2 rw rootwait fixrtc\0" \
 	"mmcargs=setenv bootargs console=${console} " \
 		"buddy=${buddy} "\
 		"vram=${vram} " \
 		"omapfb.mode=${defaultdisplay}:${dvimode} " \
-		"root=${mmcroot} " \
-		"rootfstype=${mmcrootfstype}\0" \
+		"root=${mmcroot}\0" \
 	"loadbootscript=load mmc ${bootpart} ${loadaddr} boot.scr\0" \
 	"bootscript=echo Running bootscript from mmc${mmcdev} ...; " \
 		"source ${loadaddr}\0" \
 	"loadbootenv=load mmc ${bootpart} ${loadaddr} uEnv.txt\0" \
 	"importbootenv=echo Importing environment from mmc${mmcdev} ...; " \
 		"env import -t ${loadaddr} ${filesize}\0" \
-	"loadimage=load mmc ${bootpart} ${loadaddr} ${bootdir}/${bootfile}\0" \
-	"mmcboot=echo Booting from mmc${mmcdev} ...; " \
+	"loadzimage=load mmc ${rootpart} ${loadaddr} ${bootdir}/zImage\0" \
+	"loaduimage=load mmc ${rootpart} ${loadaddr} ${bootdir}/uImage\0" \
+	"mmcbootz=echo Booting from mmc${mmcdev} ...; " \
 		"run mmcargs; " \
 		"bootz ${loadaddr} - ${fdtaddr}\0" \
+	"mmcbootm=echo Booting from mmc${mmcdev} ...; " \
+		"run mmcargs; " \
+		"bootm ${loadaddr}\0" \
 	"mmc_classic_boot=echo Booting from mmc${mmcdev} ...; " \
 		"run mmcargs; " \
 		"bootz ${loadaddr}\0" \
@@ -137,7 +141,7 @@
 			"setenv fdtfile omap4-panda-es.dtb; fi;" \
 		"if test $fdtfile = undefined; then " \
 			"echo WARNING: Could not determine device tree to use; fi; \0" \
-	"loadfdt=load mmc ${bootpart} ${fdtaddr} ${fdtdir}/${fdtfile}\0" \
+	"loadfdt=load mmc ${rootpart} ${fdtaddr} ${fdtdir}/${fdtfile}\0" \
 
 #define CONFIG_BOOTCOMMAND \
 	"run findfdt; " \
@@ -155,9 +159,12 @@
 				"run uenvcmd;" \
 			"fi;" \
 		"fi;" \
-		"if run loadimage; then " \
+		"if run loadzimage; then " \
 			"run loadfdt;" \
-			"run mmcboot; " \
+			"run mmcbootz; " \
+		"fi; " \
+		"if run loaduimage; then " \
+			"run mmcbootm; " \
 		"fi; " \
 	"fi"
 
