diff -urN a/include/configs/omap4_common.h b/include/configs/omap4_common.h
--- a/include/configs/omap4_common.h	2014-01-07 18:45:04.689541145 -0700
+++ b/include/configs/omap4_common.h	2014-01-07 18:58:41.060695680 -0700
@@ -136,6 +136,8 @@
 
 #define CONFIG_ENV_OVERWRITE
 
+#define CONFIG_IDENT_STRING " Arch Linux ARM"
+
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"loadaddr=0x82000000\0" \
 	"console=ttyO2,115200n8\0" \
@@ -144,7 +146,7 @@
 	"fdtfile=undefined\0" \
 	"bootpart=0:2\0" \
 	"bootdir=/boot\0" \
-	"fdtdir=/dtbs\0" \
+	"fdtdir=/boot/dtbs\0" \
 	"bootfile=zImage\0" \
 	"usbtty=cdc_acm\0" \
 	"vram=16M\0" \
@@ -153,14 +155,12 @@
 	"buddy=none\0" \
 	"mmcdev=0\0" \
 	"mmcpart=1\0" \
-	"mmcroot=/dev/mmcblk0p2 ro\0" \
-	"mmcrootfstype=ext4 rootwait fixrtc\0" \
+	"mmcroot=/dev/mmcblk0p2 rw rootwait fixrtc\0" \
 	"mmcargs=setenv bootargs console=${console} " \
 		"buddy=${buddy} "\
 		"vram=${vram} " \
 		"omapfb.mode=${defaultdisplay}:${dvimode} " \
-		"root=${mmcroot} " \
-		"rootfstype=${mmcrootfstype}\0" \
+		"root=${mmcroot}\0" \
 	"loadbootscript=load mmc ${mmcdev}:${mmcpart} ${loadaddr} boot.scr\0" \
 	"bootscript=echo Running bootscript from mmc${mmcdev} ...; " \
 		"source ${loadaddr}\0" \
@@ -168,13 +168,13 @@
 	"importbootenv=echo Importing environment from mmc (uEnv.txt)...; " \
 		"env import -t ${loadaddr} ${filesize}\0" \
 	"loadimage=load mmc ${bootpart} ${loadaddr} ${bootdir}/${bootfile}\0" \
-	"loadzimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} zImage\0" \
+	"loaduimage=load mmc ${bootpart} ${loadaddr} ${bootdir}/uImage\0" \
 	"mmcboot=echo Booting from mmc${mmcdev} ...; " \
 		"run mmcargs; " \
 		"bootz ${loadaddr} - ${fdtaddr}\0" \
 	"mmc_classic_boot=echo Booting from mmc${mmcdev} ...; " \
 		"run mmcargs; " \
-		"bootz ${loadaddr}\0" \
+		"bootm ${loadaddr}\0" \
 	"findfdt="\
 		"if test $board_name = sdp4430; then " \
 			"setenv fdtfile omap4-sdp.dtb; fi; " \
@@ -186,7 +186,7 @@
 			"setenv fdtfile omap4-panda-es.dtb; fi;" \
 		"if test $fdtfile = undefined; then " \
 			"echo WARNING: Could not determine device tree to use; fi; \0" \
-	"loadfdt=load mmc ${mmcdev}:${mmcpart} ${fdtaddr} ${fdtdir}/${fdtfile}\0" \
+	"loadfdt=load mmc ${bootpart} ${fdtaddr} ${fdtdir}/${fdtfile}\0" \
 
 #define CONFIG_BOOTCOMMAND \
 	"run findfdt; " \
@@ -203,8 +203,8 @@
 				"echo Running uenvcmd ...;" \
 				"run uenvcmd;" \
 			"fi;" \
-			"echo Running default loadzimage ...;" \
-			"if run loadzimage; then " \
+			"echo Running default loaduimage ...;" \
+			"if run loaduimage; then " \
 				"run mmc_classic_boot; " \
 			"fi; " \
 		"fi;" \
