--- acconfig.h	2002/12/15 18:05:09	1.15
+++ acconfig.h	2007/10/15 17:27:25	1.16
@@ -12,9 +12,6 @@
 /* Define to `int' if <sys/types.h> doesn't define. */
 #undef ino_t
 
-/* Define to `mode_t' if <sys/types.h> doesn't define. */
-#undef umode_t
-
 /* Define to `int' if <sys/socket.h> doesn't define. */
 #undef socklen_t
 
--- config.h.in	2007/10/13 03:05:27	1.118
+++ config.h.in	2007/10/15 17:27:25	1.119
@@ -37,9 +37,6 @@
 /* Define to `int' if <sys/socket.h> doesn't define. */
 #undef socklen_t
 
-/* Define to `mode_t' if <sys/types.h> doesn't define. */
-#undef umode_t
-
 /* Define if you have AIX send_file() semantics. */
 #undef HAVE_AIX_SENDFILE
 
--- configure.in	2007/10/13 03:05:27	1.261
+++ configure.in	2007/10/15 17:27:25	1.262
@@ -1119,7 +1119,7 @@
 AC_CHECK_SIZEOF(time_t, 0)
 
 dnl Check for generic typedefs
-AC_CHECK_TYPE(umode_t, mode_t)
+AC_CHECK_TYPE(mode_t, mode_t)
 AC_CHECK_TYPE(ino_t, ino_t)
 AC_CHECK_TYPE(intptr_t, AC_DEFINE(HAVE_INTPTR_T),, [
   #if HAVE_INTTYPES_H
