diff -ur mjpegtools-1.8.0-orig/y4mdenoise/Region2D.hh mjpegtools-1.8.0/y4mdenoise/Region2D.hh
--- mjpegtools-1.8.0-orig/y4mdenoise/Region2D.hh	2005-12-19 12:36:54.000000000 +0300
+++ mjpegtools-1.8.0/y4mdenoise/Region2D.hh	2005-12-19 12:44:36.000000000 +0300
@@ -16,7 +16,7 @@
 
 #include "Status_t.h"
 #include <iostream>
-
+#include <cassert>
 
 
 // The 2-dimensional region class.  Parameterized by the numeric type
