--- mjpegtools-1.9.0/mplex/lpcmstrm_in.cpp.orig	2009-05-27 01:31:46.000000000 -0700
+++ mjpegtools-1.9.0/mplex/lpcmstrm_in.cpp	2009-05-27 01:33:09.000000000 -0700
@@ -53,7 +53,7 @@
 
 bool LPCMStream::Probe(IBitStream &bs )
 {
-    char *last_dot = strrchr( bs.StreamName(), '.' );
+    const char *last_dot = strrchr( bs.StreamName(), '.' );
     return 
         last_dot != NULL 
         && strcmp( last_dot+1, "lpcm") == 0;
