From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dylan Baker <dylan@pnwbakers.com>
Date: Thu, 28 Oct 2021 11:46:18 -0700
Subject: [PATCH] modules/gnome: use `install_dir = []` instead of false

---
 mesonbuild/modules/gnome.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/modules/gnome.py b/mesonbuild/modules/gnome.py
index e825981d40f1..94b9793baa06 100644
--- a/mesonbuild/modules/gnome.py
+++ b/mesonbuild/modules/gnome.py
@@ -1458,7 +1458,7 @@ class GnomeModule(ExtensionModule):
             c_kwargs = custom_kwargs.copy()
             # Never install the C file. Complain on bug tracker if you need it.
             c_kwargs['install'] = False
-            c_kwargs['install_dir'] = False
+            c_kwargs['install_dir'] = []
             if h_template is not None:
                 if 'depends' in custom_kwargs:
                     c_kwargs['depends'] += [h_target]
