From c5544aea8d8480262b9792ac85b8e61d1fb08f74 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Corentin=20No=C3=ABl?= <corentin.noel@collabora.com>
Date: Mon, 4 Feb 2019 13:49:01 +0100
Subject: [PATCH] vala: Support position-independent executables

---
 mesonbuild/compilers/vala.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mesonbuild/compilers/vala.py b/mesonbuild/compilers/vala.py
index e64d57f366..5303298a62 100644
--- a/mesonbuild/compilers/vala.py
+++ b/mesonbuild/compilers/vala.py
@@ -49,6 +49,12 @@ def get_compile_only_args(self):
     def get_pic_args(self):
         return []
 
+    def get_pie_args(self):
+        return []
+
+    def get_pie_link_args(self):
+        return []
+
     def get_always_args(self):
         return ['-C']
 
