diff --git i/mesonbuild/modules/gnome.py w/mesonbuild/modules/gnome.py
index e5990f087..835ebfb5a 100644
--- i/mesonbuild/modules/gnome.py
+++ w/mesonbuild/modules/gnome.py
@@ -1477,9 +1477,10 @@ class GnomeModule(ExtensionModule):
         build_by_default = kwargs['build_by_default']
 
         # Annotations are a bit ugly in that they are a list of lists of strings...
-        if kwargs['annotations']:
-            cmd.append('--annotate')
-            cmd.extend(kwargs['annotations'])
+        for i, a in enumerate(kwargs['annotations']):
+            if i % 3 == 0:
+                cmd.append('--annotate')
+            cmd.append(a)
 
         targets = []
         install_header = kwargs['install_header']
