diff --git i/test cases/frameworks/15 llvm/meson.build w/test cases/frameworks/15 llvm/meson.build
index 468094ad4f3a5256..a8cf8f22f4fafb24 100644
--- i/test cases/frameworks/15 llvm/meson.build	
+++ w/test cases/frameworks/15 llvm/meson.build	
@@ -17,5 +17,5 @@ executable('sum', 'sum.c',  dependencies : [
   llvm_dep,
   dependency('zlib'),
   meson.get_compiler('c').find_library('dl', required : false),
-  dependency('tinfo'),
+  dependency('ncursesw'),
   ])
diff --git i/test cases/frameworks/4 qt/meson.build w/test cases/frameworks/4 qt/meson.build
index c6f108bbcf037e6c..ad9d40478d660546 100644
--- i/test cases/frameworks/4 qt/meson.build	
+++ w/test cases/frameworks/4 qt/meson.build	
@@ -71,14 +71,16 @@ foreach qt : ['qt4', 'qt5']
 
     test(qt + 'maninclude', qtmaninclude)
 
+    if qt == 'qt5'
     # building Qt plugins implies to give include path to moc
     plugin_includes = include_directories('pluginInterface', 'plugin')
     pluginpreprocess = qtmodule.preprocess(
       moc_headers : 'plugin/plugin.h',
       include_directories : plugin_includes
     )
-    plugin = library('plugin', 'plugin/plugin.cpp', pluginpreprocess,
+    plugin = library(qt + 'plugin', 'plugin/plugin.cpp', pluginpreprocess,
           include_directories : plugin_includes,
           dependencies : qtcore)
+    endif
   endif
 endforeach
