 mesonbuild/modules/gnome.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git c/mesonbuild/modules/gnome.py i/mesonbuild/modules/gnome.py
index e0c1214d0..5a7160d60 100644
--- c/mesonbuild/modules/gnome.py
+++ i/mesonbuild/modules/gnome.py
@@ -667,6 +667,9 @@ class GnomeModule(ExtensionModule):
                     assert isinstance(girdir, str), 'for mypy'
                     gi_includes.update([girdir])
             if isinstance(dep, InternalDependency):
+                # Ensure BothLibraries are resolved
+                dep = dep.get_as_shared(recursive=True)
+
                 cflags.update(dep.get_compile_args())
                 cflags.update(state.get_include_args(dep.include_directories))
                 for lib in dep.libraries:
