 mesonbuild/cmake/interpreter.py | 1 +
 1 file changed, 1 insertion(+)

diff --git c/mesonbuild/cmake/interpreter.py i/mesonbuild/cmake/interpreter.py
index fafee86ab..12d7049d2 100644
--- c/mesonbuild/cmake/interpreter.py
+++ i/mesonbuild/cmake/interpreter.py
@@ -828,6 +828,7 @@ class CMakeInterpreter:
 
         cmake_args = []
         cmake_args += cmake_get_generator_args(self.env)
+        cmake_args += ['-DCMAKE_LINK_DEPENDS_USE_LINKER=FALSE']
         cmake_args += [f'-DCMAKE_INSTALL_PREFIX={self.install_prefix}']
         cmake_args += extra_cmake_options
         if not any(arg.startswith('-DCMAKE_BUILD_TYPE=') for arg in cmake_args):
