diff --git a/test cases/frameworks/1 boost/meson.build b/test cases/frameworks/1 boost/meson.build
index 8f45dc7f..44bc376d 100644
--- a/test cases/frameworks/1 boost/meson.build	
+++ b/test cases/frameworks/1 boost/meson.build	
@@ -30,7 +30,7 @@ pymod = import('python')
 python2 = pymod.find_installation('python2', required: host_machine.system() == 'linux', disabler: true)
 python3 = pymod.find_installation('python3', required: host_machine.system() == 'linux', disabler: true)
 python2dep = python2.dependency(required: host_machine.system() == 'linux', disabler: true)
-python3dep = python3.dependency(required: host_machine.system() == 'linux', disabler: true)
+python3dep = dependency('python3-embed', required: host_machine.system() == 'linux', disabler: true)
 
 # compile python 2/3 modules only if we found a corresponding python version
 if(python2dep.found() and host_machine.system() == 'linux')
