 unittests/linuxliketests.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git c/unittests/linuxliketests.py i/unittests/linuxliketests.py
index 594e9f0cc..a2c4f8b8b 100644
--- c/unittests/linuxliketests.py
+++ i/unittests/linuxliketests.py
@@ -343,6 +343,7 @@ class LinuxlikeTests(BasePlatformTests):
             raise SkipTest('asan not available on Cygwin')
         if is_openbsd():
             raise SkipTest('-fsanitize=address is not supported on OpenBSD')
+        raise SkipTest('cannot run sanitizers in containers without ptrace')
 
         testdir = os.path.join(self.framework_test_dir, '7 gnome')
         self.init(testdir, extra_args=['-Db_sanitize=address', '-Db_lundef=false'])
@@ -998,6 +999,7 @@ class LinuxlikeTests(BasePlatformTests):
             raise SkipTest('asan not available on Cygwin')
         if is_openbsd():
             raise SkipTest('-fsanitize=address is not supported on OpenBSD')
+        raise SkipTest('cannot run sanitizers in containers without ptrace')
 
         testdir = os.path.join(self.common_test_dir, '13 pch')
         self.init(testdir, extra_args=['-Db_sanitize=address', '-Db_lundef=false'])
