diff --git i/run_unittests.py w/run_unittests.py
index 90d4a628..f83bd61f 100755
--- i/run_unittests.py
+++ w/run_unittests.py
@@ -4368,6 +4368,7 @@ class LinuxlikeTests(BasePlatformTests):
             raise unittest.SkipTest('asan not available on Cygwin')
         if is_openbsd():
             raise unittest.SkipTest('-fsanitize=address is not supported on OpenBSD')
+        raise unittest.SkipTest('cannot run sanitizers in containers without ptrace')
 
         testdir = os.path.join(self.framework_test_dir, '7 gnome')
         self.init(testdir, ['-Db_sanitize=address', '-Db_lundef=false'])
