 run_unittests.py | 1 +
 1 file changed, 1 insertion(+)

diff --git c/run_unittests.py i/run_unittests.py
index 828c80e70..7d4e448f0 100755
--- c/run_unittests.py
+++ i/run_unittests.py
@@ -6699,6 +6699,7 @@ class LinuxlikeTests(BasePlatformTests):
             raise unittest.SkipTest('asan not available on Cygwin')
         if is_openbsd():
             raise unittest.SkipTest('-fsanitize=address is not supported on OpenBSD')
+        raise unittest.SkipTest('cannot run sanitizers in containers without ptrace')
 
         testdir = os.path.join(self.framework_test_dir, '7 gnome')
         self.init(testdir, extra_args=['-Db_sanitize=address', '-Db_lundef=false'])
