diff --git i/test cases/linuxlike/13 cmake dependency/meson.build w/test cases/linuxlike/13 cmake dependency/meson.build
index 193ad184b..e94ba68ac 100644
--- i/test cases/linuxlike/13 cmake dependency/meson.build	
+++ w/test cases/linuxlike/13 cmake dependency/meson.build	
@@ -6,6 +6,8 @@ if not find_program('cmake', required: false).found()
   error('MESON_SKIP_TEST cmake binary not available.')
 endif
 
+error('MESON_SKIP_TEST flaky, see https://github.com/mesonbuild/meson/issues/10104')
+
 # Zlib is probably on all dev machines.
 
 dep = dependency('ZLIB', version : '>=1.2', method : 'cmake')
diff --git i/unittests/linuxliketests.py w/unittests/linuxliketests.py
index 79db0b896..12082ce61 100644
--- i/unittests/linuxliketests.py
+++ w/unittests/linuxliketests.py
@@ -343,6 +343,7 @@ class LinuxlikeTests(BasePlatformTests):
             raise SkipTest('asan not available on Cygwin')
         if is_openbsd():
             raise SkipTest('-fsanitize=address is not supported on OpenBSD')
+        raise SkipTest('cannot run sanitizers in containers without ptrace')
 
         testdir = os.path.join(self.framework_test_dir, '7 gnome')
         self.init(testdir, extra_args=['-Db_sanitize=address', '-Db_lundef=false'])
@@ -998,6 +999,7 @@ class LinuxlikeTests(BasePlatformTests):
             raise SkipTest('asan not available on Cygwin')
         if is_openbsd():
             raise SkipTest('-fsanitize=address is not supported on OpenBSD')
+        raise SkipTest('cannot run sanitizers in containers without ptrace')
 
         testdir = os.path.join(self.common_test_dir, '13 pch')
         self.init(testdir, extra_args=['-Db_sanitize=address', '-Db_lundef=false'])
