diff -ur libgdiplus-1.1.15-old/Makefile.am libgdiplus-1.1.15/Makefile.am
--- libgdiplus-1.1.15-old/Makefile.am	2006-04-14 10:04:55.000000000 -0700
+++ libgdiplus-1.1.15/Makefile.am	2006-04-21 18:06:49.000000000 -0700
@@ -1,4 +1,4 @@
-SUBDIRS = cairo src tests
+SUBDIRS = src tests
 #libpixman
 
 pkgconfigdir = $(libdir)/pkgconfig
diff -ur libgdiplus-1.1.15-old/configure.in libgdiplus-1.1.15/configure.in
--- libgdiplus-1.1.15-old/configure.in	2006-04-14 10:04:55.000000000 -0700
+++ libgdiplus-1.1.15/configure.in	2006-04-21 18:06:49.000000000 -0700
@@ -18,32 +18,32 @@
 	AC_MSG_ERROR("Did not find Glib >= 2.2.3");
 fi
 
-AC_CONFIG_SUBDIRS(cairo)
-CAIRO_LIBS='$(top_builddir)/cairo/src/libcairo.la'
-CAIRO_CFLAGS='-I$(top_builddir)/cairo/pixman/src -I$(top_srcdir)/cairo/pixman/src -I$(top_builddir)/cairo/src -I$(top_srcdir)/cairo/src'
-AC_DEFINE(USE_INCLUDED_CAIRO,1,[Use Cairo bundled in libgdiplus])
+#AC_CONFIG_SUBDIRS(cairo)
+#CAIRO_LIBS='$(top_builddir)/cairo/src/libcairo.la'
+#CAIRO_CFLAGS='-I$(top_builddir)/cairo/pixman/src -I$(top_srcdir)/cairo/pixman/src -I$(top_builddir)/cairo/src -I$(top_srcdir)/cairo/src'
+#AC_DEFINE(USE_INCLUDED_CAIRO,1,[Use Cairo bundled in libgdiplus])
 
 # Activate this when we support linking to an already exiting Cairo installation
-#default_cairo=included
-#AC_ARG_WITH(cairo, [  --with-cairo=included,installed],[cairo_v=$with_cairo],[cairo_v=$default_cairo])
+default_cairo=included
+AC_ARG_WITH(cairo, [  --with-cairo=included,installed],[cairo_v=$with_cairo],[cairo_v=$default_cairo])
 #
-#if test $cairo_v = "included";
-#then
+if test $cairo_v = "included";
+then
 #	#AC_CONFIG_SUBDIRS(libpixman)
-#	AC_CONFIG_SUBDIRS(cairo)
-#	CAIRO_LIBS='$(top_builddir)/cairo/src/libcairo.la'
-#	CAIRO_CFLAGS='-I$(top_builddir)/cairo/pixman/src -I$(top_srcdir)/cairo/pixman/src -I$(top_builddir)/cairo/src -I$(top_srcdir)/cairo/src'
-#	AC_DEFINE(USE_INCLUDED_CAIRO,1,[Use Cairo bundled in libgdiplus])
-#else
-#	if pkg-config --exact-version 0.3.0 cairo; then
-#		echo Cairo installation OK
-#	else
-#		AC_MSG_ERROR("Did not find Cairo == 0.3.0");
-#	fi
-#
-#	CAIRO_LIBS="`pkg-config --libs cairo`"
-#	CAIRO_CFLAGS="`pkg-config --cflags cairo`"
-#fi
+	AC_CONFIG_SUBDIRS(cairo)
+	CAIRO_LIBS='$(top_builddir)/cairo/src/libcairo.la'
+	CAIRO_CFLAGS='-I$(top_builddir)/cairo/pixman/src -I$(top_srcdir)/cairo/pixman/src -I$(top_builddir)/cairo/src -I$(top_srcdir)/cairo/src'
+	AC_DEFINE(USE_INCLUDED_CAIRO,1,[Use Cairo bundled in libgdiplus])
+else
+	if pkg-config --atleast-version 1.0.1 cairo; then
+		echo Cairo installation OK
+	else
+		AC_MSG_ERROR("Did not find Cairo == 1.0.1");
+	fi
+
+	CAIRO_LIBS="`pkg-config --libs cairo`"
+	CAIRO_CFLAGS="`pkg-config --cflags cairo`"
+fi
 
 GDIPLUS_LIBS="$CAIRO_LIBS `pkg-config --libs glib-2.0 ` `pkg-config freetype2 fontconfig --libs`"
 GDIPLUS_CFLAGS="$CAIRO_CFLAGS `pkg-config --cflags glib-2.0 ` `pkg-config freetype2 fontconfig --cflags`"
