From 47ceab32e3af877045a96ef58c1f30ab72bf8260 Mon Sep 17 00:00:00 2001
From: Caitlin Ross <caitlin.ross@kitware.com>
Date: Tue, 6 Dec 2022 10:49:34 -0600
Subject: [PATCH] adios debug mode deprecated

---
 fides/DataSetWriter.cxx                             | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/fides/DataSetWriter.cxx b/fides/DataSetWriter.cxx
index 94c79126..6ca00bdd 100644
--- a/fides/DataSetWriter.cxx
+++ b/fides/DataSetWriter.cxx
@@ -71,9 +71,9 @@ public:
     : DataSets(dataSets)
     , OutputFileName(fname)
 #ifdef FIDES_USE_MPI
-    , Adios(MPI_COMM_WORLD, adios2::DebugON)
+    , Adios(MPI_COMM_WORLD)
 #else
-    , Adios(true)
+    , Adios()
 #endif
     , FieldsToWriteSet(false)
   {
-- 
GitLab

