diff -ru VTK-9.3.1.orig/ThirdParty/ioss/vtkioss/Ioss_IOFactory.C VTK-9.3.1/ThirdParty/ioss/vtkioss/Ioss_IOFactory.C
--- VTK-9.3.1.orig/ThirdParty/ioss/vtkioss/Ioss_IOFactory.C	2024-09-08 10:39:43.785130427 +0200
+++ VTK-9.3.1/ThirdParty/ioss/vtkioss/Ioss_IOFactory.C	2024-09-08 10:42:08.072087946 +0200
@@ -12,6 +12,7 @@
 #include <cstddef> // for nullptr
 #include "vtk_fmt.h"
 #include VTK_FMT(fmt/ostream.h)
+#include VTK_FMT(fmt/ranges.h)
 #include <map>     // for _Rb_tree_iterator, etc
 #include <ostream> // for basic_ostream, etc
 #include <set>
diff -ru VTK-9.3.1.orig/ThirdParty/ioss/vtkioss/Ioss_StructuredBlock.C VTK-9.3.1/ThirdParty/ioss/vtkioss/Ioss_StructuredBlock.C
--- VTK-9.3.1.orig/ThirdParty/ioss/vtkioss/Ioss_StructuredBlock.C	2024-09-08 10:39:43.788463820 +0200
+++ VTK-9.3.1/ThirdParty/ioss/vtkioss/Ioss_StructuredBlock.C	2024-09-08 10:57:28.625305675 +0200
@@ -15,6 +15,7 @@
 #include <Ioss_StructuredBlock.h>
 #include "vtk_fmt.h"
 #include VTK_FMT(fmt/ostream.h)
+#include VTK_FMT(fmt/ranges.h)
 
 #include <cstddef> // for size_t
 #include <numeric>
diff -ru VTK-9.3.1.orig/ThirdParty/ioss/vtkioss/Ioss_Utils.C VTK-9.3.1/ThirdParty/ioss/vtkioss/Ioss_Utils.C
--- VTK-9.3.1.orig/ThirdParty/ioss/vtkioss/Ioss_Utils.C	2024-09-08 10:39:43.791797211 +0200
+++ VTK-9.3.1/ThirdParty/ioss/vtkioss/Ioss_Utils.C	2024-09-08 10:47:06.456187726 +0200
@@ -21,6 +21,7 @@
 #include VTK_FMT(fmt/chrono.h)
 #include VTK_FMT(fmt/format.h)
 #include VTK_FMT(fmt/ostream.h)
+#include VTK_FMT(fmt/ranges.h)
 #include <fstream>
 #include <sstream>
 #include <string>
diff -ru VTK-9.3.1.orig/ThirdParty/ioss/vtkioss/Ioss_ZoneConnectivity.C VTK-9.3.1/ThirdParty/ioss/vtkioss/Ioss_ZoneConnectivity.C
--- VTK-9.3.1.orig/ThirdParty/ioss/vtkioss/Ioss_ZoneConnectivity.C	2024-09-08 10:39:43.791797211 +0200
+++ VTK-9.3.1/ThirdParty/ioss/vtkioss/Ioss_ZoneConnectivity.C	2024-09-08 10:51:18.889659250 +0200
@@ -9,6 +9,7 @@
 #include <cstddef> // for size_t
 #include "vtk_fmt.h"
 #include VTK_FMT(fmt/ostream.h)
+#include VTK_FMT(fmt/ranges.h)
 #include <string> // for string
 #include <vector> // for vector
 
