diff -ru VTK-9.6.1.orig/IO/GDAL/vtkGDALRasterReader.cxx VTK-9.6.1/IO/GDAL/vtkGDALRasterReader.cxx
--- VTK-9.6.1.orig/IO/GDAL/vtkGDALRasterReader.cxx	2026-05-12 23:51:03.286495090 +0200
+++ VTK-9.6.1/IO/GDAL/vtkGDALRasterReader.cxx	2026-05-12 23:52:36.321366950 +0200
@@ -182,7 +182,7 @@
     this->Reader->DriverShortName = GDALGetDriverShortName(driver);
     this->Reader->DriverLongName = GDALGetDriverLongName(driver);
 
-    char** papszMetaData = GDALGetMetadata(this->GDALData, nullptr);
+    CSLConstList papszMetaData = GDALGetMetadata(this->GDALData, nullptr);
     if (CSLCount(papszMetaData) > 0)
     {
       for (int i = 0; papszMetaData[i] != nullptr; ++i)
@@ -882,7 +882,7 @@
 {
   std::vector<std::string> domainMetaData;
 
-  char** papszMetadata = GDALGetMetadata(this->Impl->GDALData, domain.c_str());
+  CSLConstList papszMetadata = GDALGetMetadata(this->Impl->GDALData, domain.c_str());
 
   if (CSLCount(papszMetadata) > 0)
   {
