diff -ru VTK-9.4.1.orig/ThirdParty/exodusII/vtkexodusII/src/ex_put_prop.c VTK-9.4.1/ThirdParty/exodusII/vtkexodusII/src/ex_put_prop.c
--- VTK-9.4.1.orig/ThirdParty/exodusII/vtkexodusII/src/ex_put_prop.c	2025-02-10 13:35:17.671934279 +0100
+++ VTK-9.4.1/ThirdParty/exodusII/vtkexodusII/src/ex_put_prop.c	2025-02-10 13:36:10.298911583 +0100
@@ -243,7 +243,7 @@
     vals[0] = 0; /* fill value */
     /*   create attribute to cause variable to fill with zeros per routine spec
      */
-    if ((status = nc_put_att_longlong(exoid, propid, _FillValue, int_type, 1, vals)) != NC_NOERR) {
+    if ((status = nc_put_att_longlong(exoid, propid, NC_FillValue, int_type, 1, vals)) != NC_NOERR) {
       snprintf(errmsg, MAX_ERR_LENGTH,
                "ERROR: failed to create property name fill attribute in file id %d", exoid);
       ex_err_fn(exoid, __func__, errmsg, status);
diff -ru VTK-9.4.1.orig/ThirdParty/exodusII/vtkexodusII/src/ex_put_prop_names.c VTK-9.4.1/ThirdParty/exodusII/vtkexodusII/src/ex_put_prop_names.c
--- VTK-9.4.1.orig/ThirdParty/exodusII/vtkexodusII/src/ex_put_prop_names.c	2025-02-10 13:35:17.671934279 +0100
+++ VTK-9.4.1/ThirdParty/exodusII/vtkexodusII/src/ex_put_prop_names.c	2025-02-10 13:36:10.298911583 +0100
@@ -172,7 +172,7 @@
 
     /*   create attribute to cause variable to fill with zeros per routine spec
      */
-    if ((status = nc_put_att_longlong(exoid, propid, _FillValue, int_type, 1, vals)) != NC_NOERR) {
+    if ((status = nc_put_att_longlong(exoid, propid, NC_FillValue, int_type, 1, vals)) != NC_NOERR) {
       snprintf(errmsg, MAX_ERR_LENGTH,
                "ERROR: failed to create property name fill attribute in file id %d", exoid);
       ex_err_fn(exoid, __func__, errmsg, status);
