--- a/IO/NanoVDB/vtkNanoVDBWriter.cxx
+++ b/IO/NanoVDB/vtkNanoVDBWriter.cxx
@@ -19,8 +19,8 @@
 #include <string>
 #include <vector>
 
-#include <nanovdb/util/CreateNanoGrid.h>
-#include <nanovdb/util/IO.h>
+#include <nanovdb/tools/CreateNanoGrid.h>
+#include <nanovdb/io/IO.h>
 
 VTK_ABI_NAMESPACE_BEGIN
 namespace
@@ -216,7 +216,7 @@
 struct VectorCopyToNanoVDB
 {
   template <typename TArray>
-  void operator()(TArray* sourceArray, nanovdb::build::Vec3fGrid& destGrid, vtkImageData* imageData,
+  void operator()(TArray* sourceArray, nanovdb::tools::build::Vec3fGrid& destGrid, vtkImageData* imageData,
     int extent[6])
   {
     auto accessor = destGrid.getAccessor();
@@ -241,7 +241,7 @@
 struct ScalarCopyToNanoVDB
 {
   template <typename TArray>
-  void operator()(TArray* sourceArray, nanovdb::build::FloatGrid& destGrid, vtkImageData* imageData,
+  void operator()(TArray* sourceArray, nanovdb::tools::build::FloatGrid& destGrid, vtkImageData* imageData,
     int extent[6])
   {
     auto accessor = destGrid.getAccessor();
@@ -300,7 +300,7 @@
 
       if (numberOfComponents == 3)
       {
-        nanovdb::build::Vec3fGrid vectorGrid(
+        nanovdb::tools::build::Vec3fGrid vectorGrid(
           nanovdb::Vec3f(0.0f, 0.0f, 0.0f), gridName, nanovdb::GridClass::Staggered);
         vectorGrid.mMap = gridMap;
 
@@ -310,11 +310,11 @@
         {
           vectorCopyFunctor(data, vectorGrid, imageData, extent);
         }
-        handles.emplace_back(nanovdb::createNanoGrid(vectorGrid));
+        handles.emplace_back(nanovdb::tools::createNanoGrid(vectorGrid));
       }
       else
       {
-        nanovdb::build::FloatGrid scalarGrid(0.0f, gridName, nanovdb::GridClass::FogVolume);
+        nanovdb::tools::build::FloatGrid scalarGrid(0.0f, gridName, nanovdb::GridClass::FogVolume);
         scalarGrid.mMap = gridMap;
 
         using ScalarCopyDispatch = vtkArrayDispatch::DispatchByValueType<vtkArrayDispatch::Reals>;
@@ -323,7 +323,7 @@
         {
           scalarCopyFunctor(data, scalarGrid, imageData, extent);
         }
-        handles.emplace_back(nanovdb::createNanoGrid(scalarGrid));
+        handles.emplace_back(nanovdb::tools::createNanoGrid(scalarGrid));
       }
     }
   }
