diff -urN a/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.cpp b/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.cpp
--- a/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.cpp	2026-03-26 10:23:37.000000000 -0600
+++ b/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.cpp	2026-04-08 07:17:37.462907226 -0600
@@ -144,7 +144,7 @@
     return XdmfArrayType::Int32();
   }
 
-#if defined(_WIN32) || defined(__APPLE__)
+#if defined(_WIN32) || defined(__APPLE__) || UINTPTR_MAX == UINT32_MAX
   shared_ptr<const XdmfArrayType>
   getArrayType(const long * const) const
   {
@@ -188,7 +188,7 @@
     return XdmfArrayType::UInt32();
   }
 
-#if defined(_WIN32) || defined(__APPLE__)
+#if defined(_WIN32) || defined(__APPLE__) || UINTPTR_MAX == UINT32_MAX
   shared_ptr<const XdmfArrayType>
   getArrayType(const unsigned long * const) const
   {
