Index: paraview/VTK/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.cpp
===================================================================
--- paraview.orig/VTK/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.cpp	2024-10-03 15:22:34.472764387 +0200
+++ paraview/VTK/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.cpp	2024-10-03 15:24:06.658109046 +0200
@@ -180,7 +180,7 @@
     return XdmfArrayType::UInt32();
   }
 
-#if defined(_WIN32) || defined(__APPLE__)
+#if defined(_WIN32) || defined(__APPLE__) || UINTPTR_MAX == UINT32_MAX
   shared_ptr<const XdmfArrayType>
   getArrayType(const unsigned long * const) const
   {
