--- bbdock-0.2.8/src/Icon.cc	2006-09-19 06:46:21.000000000 -0700
+++ bbdock-0.2.8/src/Icon.cc	2008-09-14 14:33:19.000000000 -0700
@@ -28,6 +28,7 @@
  */
 #include <string>
 #include <vector>
+#include <cstdlib>
 
 #include "Icon.hh"

--- bbdock-0.2.8/src/WindowManager.cc	2006-09-19 06:48:23.000000000 -0700
+++ bbdock-0.2.8/src/WindowManager.cc	2008-09-14 14:34:25.000000000 -0700
@@ -32,7 +32,7 @@
 #endif // _GNU_SOURCE
 
 #include <iostream>
-#include <string>
+#include <cstring>
 
 #include "WindowManager.hh"
 
