--- thunar-vfs/thunar-vfs-io-jobs.c.orig	2009-05-19 18:44:00.000000000 +0100
+++ thunar-vfs/thunar-vfs-io-jobs.c	2009-05-19 18:46:04.000000000 +0100
@@ -708,7 +708,7 @@
       _thunar_vfs_job_process_path (job, lp);
 
       /* try to create the target directory */
-      if (!_thunar_vfs_io_ops_mkdir (lp->data, 0777 & ~umask(0), THUNAR_VFS_IO_OPS_NONE, error))
+      if (!_thunar_vfs_io_ops_mkdir (lp->data, 0777, THUNAR_VFS_IO_OPS_NONE, error))
         return FALSE;
     }
 
