diff --git a/Source/WTF/wtf/Platform.h b/Source/WTF/wtf/Platform.h
index 4594ec8..639f28f 100644
--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -342,7 +342,7 @@
 #endif
 #endif
 
-#if CPU(ARM) || CPU(MIPS) || CPU(SH4) || CPU(SPARC) || CPU(MIPS64)
+#if CPU(ARM) || CPU(MIPS) || CPU(SH4) || CPU(SPARC) || CPU(MIPS64) || CPU(AARCH64)
 #define WTF_CPU_NEEDS_ALIGNED_ACCESS 1
 #endif

