--- a/tools/abicommand/xp/AbiCommand.cpp	2007-10-11 13:05:55.000000000 -0700
+++ b/tools/abicommand/xp/AbiCommand.cpp	2009-05-29 12:36:40.000000000 -0700
@@ -551,7 +551,7 @@
 				UT_UTF8String *pCom1 =
 					const_cast < UT_UTF8String * >(static_cast < const UT_UTF8String * >(pToks->getNthItem (1)));
 				printf(" Filename %s \n",pCom1->utf8_str());
-				char *suffix = rindex (pCom1->utf8_str (), '.');
+				const char *suffix = rindex (pCom1->utf8_str (), '.');
 
 				if (suffix != NULL)
 				{
